<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_261) on Mon Nov 02 12:47:50 CST 2020 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Package org.biojava.bio.seq (biojava-legacy 1.9.2 API)</title>
<meta name="date" content="2020-11-02">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Package org.biojava.bio.seq (biojava-legacy 1.9.2 API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/seq/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Uses of Package org.biojava.bio.seq" class="title">Uses of Package<br>org.biojava.bio.seq</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.alignment">org.biojava.bio.alignment</a></td>
<td class="colLast">
<div class="block">
Classes to generate and describe sequence alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.dist">org.biojava.bio.dist</a></td>
<td class="colLast">
<div class="block">
Probability distributions over Alphabets.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.dp">org.biojava.bio.dp</a></td>
<td class="colLast">
<div class="block">HMM and Dynamic Programming Algorithms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.gui.sequence">org.biojava.bio.gui.sequence</a></td>
<td class="colLast">
<div class="block">
Graphical displays of biological sequences and associated annotations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.gui.sequence.tracklayout">org.biojava.bio.gui.sequence.tracklayout</a></td>
<td class="colLast">
<div class="block">Classes for the handling of the layout of a WrappedSequencePanel.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.molbio">org.biojava.bio.molbio</a></td>
<td class="colLast">
<div class="block">The classes and interfaces in this package cover common molecular
biological techniques such as restriction digests and PCR.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.fastq">org.biojava.bio.program.fastq</a></td>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.gff">org.biojava.bio.program.gff</a></td>
<td class="colLast">
<div class="block">GFF manipulation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.gff3">org.biojava.bio.program.gff3</a></td>
<td class="colLast">
<div class="block">
Support for reading and writing GFF3.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.phred">org.biojava.bio.program.phred</a></td>
<td class="colLast">
<div class="block">Parser for Phred output</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.ssbind">org.biojava.bio.program.ssbind</a></td>
<td class="colLast">
<div class="block">
Creation of objects from SAX events using the BioJava BlastLikeDataSetCollection DTD.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.unigene">org.biojava.bio.program.unigene</a></td>
<td class="colLast">
<div class="block">Objects for representing Unigene clusters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.xff">org.biojava.bio.program.xff</a></td>
<td class="colLast">
<div class="block">Event-driven parsing system for the Extensible Feature Format (XFF).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.proteomics">org.biojava.bio.proteomics</a></td>
<td class="colLast">
<div class="block">Utilities to aid in performing various physical analysis of proteins.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.search">org.biojava.bio.search</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for representing sequence similarity search results.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq">org.biojava.bio.seq</a></td>
<td class="colLast">
<div class="block">Classes and interfaces for defining biological sequences and informatics
objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db">org.biojava.bio.seq.db</a></td>
<td class="colLast">
<div class="block">
Collections of biological sequence data.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.biofetch">org.biojava.bio.seq.db.biofetch</a></td>
<td class="colLast">
<div class="block">Client for the OBDA BioFetch protocol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.biosql">org.biojava.bio.seq.db.biosql</a></td>
<td class="colLast">
<div class="block">General purpose Sequence storage in a relational database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.flat">org.biojava.bio.seq.db.flat</a></td>
<td class="colLast">
<div class="block">
Support for OBDA flatfile databases.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.distributed">org.biojava.bio.seq.distributed</a></td>
<td class="colLast">
<div class="block">Sequences and SequenceDBs which are composed from data taken
from a number of data sources.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.filter">org.biojava.bio.seq.filter</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.homol">org.biojava.bio.seq.homol</a></td>
<td class="colLast">
<div class="block">The classes and interfaces for defining sequence similarity and
honology.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.impl">org.biojava.bio.seq.impl</a></td>
<td class="colLast">
<div class="block">Standard in-memory implementations of <code>Sequence</code> and
<code>Feature</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.io">org.biojava.bio.seq.io</a></td>
<td class="colLast">
<div class="block">Classes and interfaces for processing and producing flat-file representations
of sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.io.agave">org.biojava.bio.seq.io.agave</a></td>
<td class="colLast">
<div class="block">Classes for converting between AGAVE XML and BioJava objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.io.filterxml">org.biojava.bio.seq.io.filterxml</a></td>
<td class="colLast">
<div class="block">Tools for reading and writing an XML representation of BioJava's FeatureFilter language.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.io.game">org.biojava.bio.seq.io.game</a></td>
<td class="colLast">
<div class="block">Event-driven parsing system for the Gene Annotation Markup Elements (GAME).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.projection">org.biojava.bio.seq.projection</a></td>
<td class="colLast">
<div class="block">Code for projecting Feature objects and systematically altering their
properties.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.db">org.biojavax.bio.db</a></td>
<td class="colLast">
<div class="block">Interactions between biojavax objects and a DB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.db.biosql">org.biojavax.bio.db.biosql</a></td>
<td class="colLast">
<div class="block">Interface between biojava and biosql databases</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.seq">org.biojavax.bio.seq</a></td>
<td class="colLast">
<div class="block">Rich implementations of Sequences, Locations and Features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.seq.io">org.biojavax.bio.seq.io</a></td>
<td class="colLast">
<div class="block">Classes to support the I/O of RichSequence and 
     Bioentry objects.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.alignment">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/alignment/package-summary.html">org.biojava.bio.alignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/GappedSequence.html#org.biojava.bio.alignment">GappedSequence</a>
<div class="block">Extension of GappedSymbolList which also projects features
 into the gapped coordinate system.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.alignment">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceIterator.html#org.biojava.bio.alignment">SequenceIterator</a>
<div class="block">An iterator over a bag of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dist">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/dist/package-summary.html">org.biojava.bio.dist</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.dist">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dp">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/dp/package-summary.html">org.biojava.bio.dp</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.dp">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceAnnotator.html#org.biojava.bio.dp">SequenceAnnotator</a>
<div class="block">An object which adds some additional information to a Sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.gui.sequence">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/gui/sequence/package-summary.html">org.biojava.bio.gui.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.gui.sequence">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.gui.sequence">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.gui.sequence">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/OptimizableFilter.html#org.biojava.bio.gui.sequence">OptimizableFilter</a>
<div class="block">The interface for filters that can potentialy optimize themselves, and
 compare themselves with other filters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.gui.sequence">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.gui.sequence">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.gui.sequence.tracklayout">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/gui/sequence/tracklayout/package-summary.html">org.biojava.bio.gui.sequence.tracklayout</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.gui.sequence.tracklayout">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.molbio">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/molbio/package-summary.html">org.biojava.bio.molbio</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.molbio">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.molbio">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.molbio">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.molbio">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceAnnotator.html#org.biojava.bio.molbio">SequenceAnnotator</a>
<div class="block">An object which adds some additional information to a Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.html#org.biojava.bio.molbio">StrandedFeature</a>
<div class="block">Adds the concept of 'strand' to features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Template.html#org.biojava.bio.molbio">StrandedFeature.Template</a>
<div class="block">Template class for parameterizing the creation of a new
 <code>StrandedFeature</code>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.fastq">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/program/fastq/package-summary.html">org.biojava.bio.program.fastq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.program.fastq">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.gff">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/program/gff/package-summary.html">org.biojava.bio.program.gff</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.program.gff">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.program.gff">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.program.gff">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceAnnotator.html#org.biojava.bio.program.gff">SequenceAnnotator</a>
<div class="block">An object which adds some additional information to a Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.program.gff">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.gff3">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/program/gff3/package-summary.html">org.biojava.bio.program.gff3</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.program.gff3">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.phred">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/program/phred/package-summary.html">org.biojava.bio.program.phred</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.program.phred">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RealizingFeatureHolder.html#org.biojava.bio.program.phred">RealizingFeatureHolder</a>
<div class="block">Interface for <code>FeatureHolder</code> objects which know how to
 instantiate new child Features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.program.phred">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.ssbind">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.program.ssbind">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.unigene">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/program/unigene/package-summary.html">org.biojava.bio.program.unigene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.program.unigene">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.xff">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/program/xff/package-summary.html">org.biojava.bio.program.xff</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.program.xff">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.program.xff">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.program.xff">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.program.xff">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Template.html#org.biojava.bio.program.xff">StrandedFeature.Template</a>
<div class="block">Template class for parameterizing the creation of a new
 <code>StrandedFeature</code>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.proteomics">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/proteomics/package-summary.html">org.biojava.bio.proteomics</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.proteomics">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.search">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.search">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.search">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/AbstractFeatureHolder.html#org.biojava.bio.seq">AbstractFeatureHolder</a>
<div class="block">An abstract implementation of FeatureHolder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.seq">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.ByLocationComparator.html#org.biojava.bio.seq">Feature.ByLocationComparator</a>
<div class="block"><code>ByLocationComparator</code> compares
 <code>Feature</code>s by the minimum base position of their
 <code>Location</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.seq">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.ByAnnotationType.html#org.biojava.bio.seq">FeatureFilter.ByAnnotationType</a>
<div class="block">A filter that returns all features that have an annotation bundle that is of a given
 annotation type.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureRealizer.html#org.biojava.bio.seq">FeatureRealizer</a>
<div class="block">Interface for translators which map from Feature.Template
 instances to real Feature objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureTypes.Repository.html#org.biojava.bio.seq">FeatureTypes.Repository</a>
<div class="block">A named collection of Types.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureTypes.Type.html#org.biojava.bio.seq">FeatureTypes.Type</a>
<div class="block">A type of feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FilterUtils.FilterTransformer.html#org.biojava.bio.seq">FilterUtils.FilterTransformer</a>
<div class="block">An object able to transform some FeatureFilter instances sytematically into others.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Frame.html#org.biojava.bio.seq">Frame</a>
<div class="block">Title:        Frame.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FramedFeature.ReadingFrame.html#org.biojava.bio.seq">FramedFeature.ReadingFrame</a>
<div class="block">A singleton to hold the frame information</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/GappedSequence.html#org.biojava.bio.seq">GappedSequence</a>
<div class="block">Extension of GappedSymbolList which also projects features
 into the gapped coordinate system.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/OptimizableFilter.html#org.biojava.bio.seq">OptimizableFilter</a>
<div class="block">The interface for filters that can potentialy optimize themselves, and
 compare themselves with other filters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RealizingFeatureHolder.html#org.biojava.bio.seq">RealizingFeatureHolder</a>
<div class="block">Interface for <code>FeatureHolder</code> objects which know how to
 instantiate new child Features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RemoteFeature.html#org.biojava.bio.seq">RemoteFeature</a>
<div class="block">A feature that indicates that there is some remote feature that can't be
 represented entirely on a single Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RemoteFeature.Resolver.html#org.biojava.bio.seq">RemoteFeature.Resolver</a>
<div class="block">The interface for objects that actually can take a RemoteFeature and
 return a Sequence object with the feature resolved into a real feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.html#org.biojava.bio.seq">StrandedFeature</a>
<div class="block">Adds the concept of 'strand' to features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.seq">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Template.html#org.biojava.bio.seq">StrandedFeature.Template</a>
<div class="block">Template class for parameterizing the creation of a new
 <code>StrandedFeature</code>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq.db">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq.db">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.db">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceAnnotator.html#org.biojava.bio.seq.db">SequenceAnnotator</a>
<div class="block">An object which adds some additional information to a Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceIterator.html#org.biojava.bio.seq.db">SequenceIterator</a>
<div class="block">An iterator over a bag of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.biofetch">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/db/biofetch/package-summary.html">org.biojava.bio.seq.db.biofetch</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.db.biofetch">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.biosql">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/db/biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq.db.biosql">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq.db.biosql">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.db.biosql">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceIterator.html#org.biojava.bio.seq.db.biosql">SequenceIterator</a>
<div class="block">An iterator over a bag of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.flat">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/db/flat/package-summary.html">org.biojava.bio.seq.db.flat</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.db.flat">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.distributed">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/distributed/package-summary.html">org.biojava.bio.seq.distributed</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq.distributed">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq.distributed">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.distributed">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.filter">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/filter/package-summary.html">org.biojava.bio.seq.filter</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq.filter">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.And.html#org.biojava.bio.seq.filter">FeatureFilter.And</a>
<div class="block">A filter that returns all features accepted by both child filter.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.ByAncestor.html#org.biojava.bio.seq.filter">FeatureFilter.ByAncestor</a>
<div class="block">Filter by applying a nested <code>FeatureFilter</code> to all
 ancestor features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.ByChild.html#org.biojava.bio.seq.filter">FeatureFilter.ByChild</a>
<div class="block">Filter by applying a nested <code>FeatureFilter</code> to the
 child features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.ByDescendant.html#org.biojava.bio.seq.filter">FeatureFilter.ByDescendant</a>
<div class="block">Filter by applying a nested <code>FeatureFilter</code> to all
 descendant features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.ByParent.html#org.biojava.bio.seq.filter">FeatureFilter.ByParent</a>
<div class="block">Filter by applying a nested <code>FeatureFilter</code> to the
 parent feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.Not.html#org.biojava.bio.seq.filter">FeatureFilter.Not</a>
<div class="block">A filter that returns all features not accepted by a child filter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.OnlyChildren.html#org.biojava.bio.seq.filter">FeatureFilter.OnlyChildren</a>
<div class="block">Accepts features where all immediate children meet the supplied filter.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.OnlyDescendants.html#org.biojava.bio.seq.filter">FeatureFilter.OnlyDescendants</a>
<div class="block">Accepts features where all descendants meet the supplied filter.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.Or.html#org.biojava.bio.seq.filter">FeatureFilter.Or</a>
<div class="block">A filter that returns all features accepted by at least one child filter.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.homol">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/homol/package-summary.html">org.biojava.bio.seq.homol</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.seq.homol">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.seq.homol">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq.homol">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.html#org.biojava.bio.seq.homol">StrandedFeature</a>
<div class="block">Adds the concept of 'strand' to features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Template.html#org.biojava.bio.seq.homol">StrandedFeature.Template</a>
<div class="block">Template class for parameterizing the creation of a new
 <code>StrandedFeature</code>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.impl">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/impl/package-summary.html">org.biojava.bio.seq.impl</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/ComponentFeature.html#org.biojava.bio.seq.impl">ComponentFeature</a>
<div class="block">Feature which represents a component in an assembly (contig).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.seq.impl">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.seq.impl">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq.impl">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq.impl">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureRealizer.html#org.biojava.bio.seq.impl">FeatureRealizer</a>
<div class="block">Interface for translators which map from Feature.Template
 instances to real Feature objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FilterUtils.FilterTransformer.html#org.biojava.bio.seq.impl">FilterUtils.FilterTransformer</a>
<div class="block">An object able to transform some FeatureFilter instances sytematically into others.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Frame.html#org.biojava.bio.seq.impl">Frame</a>
<div class="block">Title:        Frame.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FramedFeature.html#org.biojava.bio.seq.impl">FramedFeature</a>
<div class="block">Title:         FramedFeature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FramedFeature.ReadingFrame.html#org.biojava.bio.seq.impl">FramedFeature.ReadingFrame</a>
<div class="block">A singleton to hold the frame information</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FramedFeature.Template.html#org.biojava.bio.seq.impl">FramedFeature.Template</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/GappedSequence.html#org.biojava.bio.seq.impl">GappedSequence</a>
<div class="block">Extension of GappedSymbolList which also projects features
 into the gapped coordinate system.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RealizingFeatureHolder.html#org.biojava.bio.seq.impl">RealizingFeatureHolder</a>
<div class="block">Interface for <code>FeatureHolder</code> objects which know how to
 instantiate new child Features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RemoteFeature.html#org.biojava.bio.seq.impl">RemoteFeature</a>
<div class="block">A feature that indicates that there is some remote feature that can't be
 represented entirely on a single Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RemoteFeature.Resolver.html#org.biojava.bio.seq.impl">RemoteFeature.Resolver</a>
<div class="block">The interface for objects that actually can take a RemoteFeature and
 return a Sequence object with the feature resolved into a real feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/RemoteFeature.Template.html#org.biojava.bio.seq.impl">RemoteFeature.Template</a>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.impl">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceFactory.html#org.biojava.bio.seq.impl">SequenceFactory</a>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;
<div class="block"><span class="deprecationComment">use org.biojavax.bio.seq.io.RichSequenceBuilder or 
 use org.biojavax.bio.seq.io.SequenceBuilder</span></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SimpleFeatureHolder.html#org.biojava.bio.seq.impl">SimpleFeatureHolder</a>
<div class="block">A no-frills implementation of FeatureHolder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.html#org.biojava.bio.seq.impl">StrandedFeature</a>
<div class="block">Adds the concept of 'strand' to features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.seq.impl">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Template.html#org.biojava.bio.seq.impl">StrandedFeature.Template</a>
<div class="block">Template class for parameterizing the creation of a new
 <code>StrandedFeature</code>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.io">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/ComponentFeature.html#org.biojava.bio.seq.io">ComponentFeature</a>
<div class="block">Feature which represents a component in an assembly (contig).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/ComponentFeature.Template.html#org.biojava.bio.seq.io">ComponentFeature.Template</a>
<div class="block">Template for constructing a new ComponentFeature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.seq.io">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.seq.io">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.io">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceIterator.html#org.biojava.bio.seq.io">SequenceIterator</a>
<div class="block">An iterator over a bag of sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.seq.io">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.io.agave">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/io/agave/package-summary.html">org.biojava.bio.seq.io.agave</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.seq.io.agave">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq.io.agave">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.io.agave">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.seq.io.agave">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.io.filterxml">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/io/filterxml/package-summary.html">org.biojava.bio.seq.io.filterxml</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq.io.filterxml">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.io.game">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/io/game/package-summary.html">org.biojava.bio.seq.io.game</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.seq.io.game">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.seq.io.game">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.projection">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojava/bio/seq/projection/package-summary.html">org.biojava.bio.seq.projection</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/AbstractFeatureHolder.html#org.biojava.bio.seq.projection">AbstractFeatureHolder</a>
<div class="block">An abstract implementation of FeatureHolder.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojava.bio.seq.projection">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojava.bio.seq.projection">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojava.bio.seq.projection">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojava.bio.seq.projection">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FilterUtils.FilterTransformer.html#org.biojava.bio.seq.projection">FilterUtils.FilterTransformer</a>
<div class="block">An object able to transform some FeatureFilter instances sytematically into others.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojava.bio.seq.projection">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojava.bio.seq.projection">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojavax.bio.db">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojavax.bio.db">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojavax.bio.db">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceIterator.html#org.biojavax.bio.db">SequenceIterator</a>
<div class="block">An iterator over a bag of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db.biosql">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojavax.bio.db.biosql">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojavax.bio.db.biosql">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojavax.bio.db.biosql">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.seq">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojavax/bio/seq/package-summary.html">org.biojavax.bio.seq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.html#org.biojavax.bio.seq">Feature</a>
<div class="block">A feature within a sequence, or nested within another feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojavax.bio.seq">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureFilter.html#org.biojavax.bio.seq">FeatureFilter</a>
<div class="block">A filter for accepting or rejecting a feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/FeatureHolder.html#org.biojavax.bio.seq">FeatureHolder</a>
<div class="block">The interface for objects that contain features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojavax.bio.seq">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceIterator.html#org.biojavax.bio.seq">SequenceIterator</a>
<div class="block">An iterator over a bag of sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.html#org.biojavax.bio.seq">StrandedFeature</a>
<div class="block">Adds the concept of 'strand' to features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/StrandedFeature.Strand.html#org.biojavax.bio.seq">StrandedFeature.Strand</a>
<div class="block">Class to represent the 'strandedness' of a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.seq.io">
<!--   -->
</a>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a> used by <a href="../../../../org/biojavax/bio/seq/io/package-summary.html">org.biojavax.bio.seq.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Feature.Template.html#org.biojavax.bio.seq.io">Feature.Template</a>
<div class="block">Template class for a plain feature.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/Sequence.html#org.biojavax.bio.seq.io">Sequence</a>
<div class="block">
 A biological sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../org/biojava/bio/seq/class-use/SequenceIterator.html#org.biojavax.bio.seq.io">SequenceIterator</a>
<div class="block">An iterator over a bag of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" type="text/javascript"></script><script type="text/javascript">_uacct = "UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/seq/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2020 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
