<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence.io, class: FastaGeneWriter">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on 01-21-2010</span>
<span class="source-line-no">021</span><span id="line-21"> */</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">package org.biojava.nbio.core.sequence.io;</span>
<span class="source-line-no">024</span><span id="line-24"></span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.sequence.*;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.io.template.FastaHeaderFormatInterface;</span>
<span class="source-line-no">028</span><span id="line-28">import org.slf4j.Logger;</span>
<span class="source-line-no">029</span><span id="line-29">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">030</span><span id="line-30"></span>
<span class="source-line-no">031</span><span id="line-31">import java.io.OutputStream;</span>
<span class="source-line-no">032</span><span id="line-32">import java.util.Collection;</span>
<span class="source-line-no">033</span><span id="line-33"></span>
<span class="source-line-no">034</span><span id="line-34">/**</span>
<span class="source-line-no">035</span><span id="line-35"> * A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to 3 prime version.</span>
<span class="source-line-no">036</span><span id="line-36"> * It is also an option to write out the gene sequence where the exon regions are upper case</span>
<span class="source-line-no">037</span><span id="line-37"> * 6/22/2010 FastaWriter needs to be sequence aware to handle writing out a GeneSequence which is negative Strand with the proper sequence</span>
<span class="source-line-no">038</span><span id="line-38"> * @author Scooter Willis </span>
<span class="source-line-no">039</span><span id="line-39"> */</span>
<span class="source-line-no">040</span><span id="line-40">public class FastaGeneWriter {</span>
<span class="source-line-no">041</span><span id="line-41"></span>
<span class="source-line-no">042</span><span id="line-42">        private final static Logger logger = LoggerFactory.getLogger(FastaGeneWriter.class);</span>
<span class="source-line-no">043</span><span id="line-43"></span>
<span class="source-line-no">044</span><span id="line-44">        boolean showExonUppercase = false;</span>
<span class="source-line-no">045</span><span id="line-45">        OutputStream os;</span>
<span class="source-line-no">046</span><span id="line-46">        Collection&lt;GeneSequence&gt; sequences;</span>
<span class="source-line-no">047</span><span id="line-47">        FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt; headerFormat;</span>
<span class="source-line-no">048</span><span id="line-48">        private int lineLength = 60;</span>
<span class="source-line-no">049</span><span id="line-49">/**</span>
<span class="source-line-no">050</span><span id="line-50"> *</span>
<span class="source-line-no">051</span><span id="line-51"> * @param os</span>
<span class="source-line-no">052</span><span id="line-52"> * @param sequences</span>
<span class="source-line-no">053</span><span id="line-53"> * @param headerFormat</span>
<span class="source-line-no">054</span><span id="line-54"> * @param showExonUppercase</span>
<span class="source-line-no">055</span><span id="line-55"> */</span>
<span class="source-line-no">056</span><span id="line-56">        public FastaGeneWriter(OutputStream os, Collection&lt;GeneSequence&gt; sequences, FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt; headerFormat, boolean showExonUppercase) {</span>
<span class="source-line-no">057</span><span id="line-57">                this(os, sequences, headerFormat, showExonUppercase, 60);</span>
<span class="source-line-no">058</span><span id="line-58">        }</span>
<span class="source-line-no">059</span><span id="line-59">/**</span>
<span class="source-line-no">060</span><span id="line-60"> *</span>
<span class="source-line-no">061</span><span id="line-61"> * @param os</span>
<span class="source-line-no">062</span><span id="line-62"> * @param sequences</span>
<span class="source-line-no">063</span><span id="line-63"> * @param headerFormat</span>
<span class="source-line-no">064</span><span id="line-64"> * @param showExonUppercase</span>
<span class="source-line-no">065</span><span id="line-65"> * @param lineLength</span>
<span class="source-line-no">066</span><span id="line-66"> */</span>
<span class="source-line-no">067</span><span id="line-67">        public FastaGeneWriter(OutputStream os, Collection&lt;GeneSequence&gt; sequences, FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt; headerFormat, boolean showExonUppercase, int lineLength) {</span>
<span class="source-line-no">068</span><span id="line-68">                this.os = os;</span>
<span class="source-line-no">069</span><span id="line-69">                this.sequences = sequences;</span>
<span class="source-line-no">070</span><span id="line-70">                this.headerFormat = headerFormat;</span>
<span class="source-line-no">071</span><span id="line-71">                this.lineLength = lineLength;</span>
<span class="source-line-no">072</span><span id="line-72">                this.showExonUppercase = showExonUppercase;</span>
<span class="source-line-no">073</span><span id="line-73">        }</span>
<span class="source-line-no">074</span><span id="line-74">/**</span>
<span class="source-line-no">075</span><span id="line-75"> *</span>
<span class="source-line-no">076</span><span id="line-76"> * @throws Exception</span>
<span class="source-line-no">077</span><span id="line-77"> */</span>
<span class="source-line-no">078</span><span id="line-78">        public void process() throws Exception {</span>
<span class="source-line-no">079</span><span id="line-79">                byte[] lineSep = System.getProperty("line.separator").getBytes();</span>
<span class="source-line-no">080</span><span id="line-80"></span>
<span class="source-line-no">081</span><span id="line-81">                for (GeneSequence sequence : sequences) {</span>
<span class="source-line-no">082</span><span id="line-82">                        String header = headerFormat.getHeader(sequence);</span>
<span class="source-line-no">083</span><span id="line-83">                        os.write('&gt;');</span>
<span class="source-line-no">084</span><span id="line-84">                        os.write(header.getBytes());</span>
<span class="source-line-no">085</span><span id="line-85">                        os.write(lineSep);</span>
<span class="source-line-no">086</span><span id="line-86"></span>
<span class="source-line-no">087</span><span id="line-87">                        int compoundCount = 0;</span>
<span class="source-line-no">088</span><span id="line-88">                        String seq = "";</span>
<span class="source-line-no">089</span><span id="line-89">                        //GeneSequence currently has a strand attribute to indicate direction</span>
<span class="source-line-no">090</span><span id="line-90"></span>
<span class="source-line-no">091</span><span id="line-91">                        seq = sequence.getSequence5PrimeTo3Prime().getSequenceAsString();</span>
<span class="source-line-no">092</span><span id="line-92">                        if (showExonUppercase) {</span>
<span class="source-line-no">093</span><span id="line-93">                                StringBuilder sb = new StringBuilder(seq.toLowerCase());</span>
<span class="source-line-no">094</span><span id="line-94">                                int geneBioBegin = sequence.getBioBegin();</span>
<span class="source-line-no">095</span><span id="line-95">                                int geneBioEnd = sequence.getBioEnd();</span>
<span class="source-line-no">096</span><span id="line-96">                                for (ExonSequence exonSequence : sequence.getExonSequences()) {</span>
<span class="source-line-no">097</span><span id="line-97">                                        int featureBioBegin = 0;</span>
<span class="source-line-no">098</span><span id="line-98">                                        int featureBioEnd = 0;</span>
<span class="source-line-no">099</span><span id="line-99">                                        if (sequence.getStrand() != Strand.NEGATIVE) {</span>
<span class="source-line-no">100</span><span id="line-100">                                                featureBioBegin = exonSequence.getBioBegin() - geneBioBegin;</span>
<span class="source-line-no">101</span><span id="line-101">                                                featureBioEnd = exonSequence.getBioEnd() - geneBioBegin;</span>
<span class="source-line-no">102</span><span id="line-102">                                        } else {</span>
<span class="source-line-no">103</span><span id="line-103">                                                featureBioBegin = geneBioEnd - exonSequence.getBioEnd();</span>
<span class="source-line-no">104</span><span id="line-104">                                                featureBioEnd = geneBioEnd - exonSequence.getBioBegin();</span>
<span class="source-line-no">105</span><span id="line-105">                                        }</span>
<span class="source-line-no">106</span><span id="line-106">                                        if (featureBioBegin &lt; 0 || featureBioEnd &lt; 0 || featureBioEnd &gt; sb.length() || featureBioBegin &gt; sb.length()) {</span>
<span class="source-line-no">107</span><span id="line-107">                                                logger.warn("Bad Feature, Accession: {}, Sequence Strand: {}, Gene Begin: {}, Gene End: {}, Exon Begin: {}, Exon End: {}", sequence.getAccession().toString(), sequence.getStrand(), geneBioBegin, geneBioEnd, exonSequence.getBioBegin(), exonSequence.getBioEnd());</span>
<span class="source-line-no">108</span><span id="line-108">                                        } else {</span>
<span class="source-line-no">109</span><span id="line-109">                                                for (int i = featureBioBegin; i &lt;= featureBioEnd; i++) {</span>
<span class="source-line-no">110</span><span id="line-110">                                                        char ch = sb.charAt(i);</span>
<span class="source-line-no">111</span><span id="line-111">                                                        //probably not the fastest but the safest way if language is not standard ASCII</span>
<span class="source-line-no">112</span><span id="line-112">                                                        String temp = String.valueOf(ch);</span>
<span class="source-line-no">113</span><span id="line-113">                                                        ch = temp.toUpperCase().charAt(0);</span>
<span class="source-line-no">114</span><span id="line-114">                                                        sb.setCharAt(i, ch);</span>
<span class="source-line-no">115</span><span id="line-115">                                                }</span>
<span class="source-line-no">116</span><span id="line-116">                                        }</span>
<span class="source-line-no">117</span><span id="line-117">                                }</span>
<span class="source-line-no">118</span><span id="line-118">                                seq = sb.toString();</span>
<span class="source-line-no">119</span><span id="line-119">                        }</span>
<span class="source-line-no">120</span><span id="line-120"></span>
<span class="source-line-no">121</span><span id="line-121">                        for (int i = 0; i &lt; seq.length(); i++) {</span>
<span class="source-line-no">122</span><span id="line-122">                                os.write(seq.charAt(i));</span>
<span class="source-line-no">123</span><span id="line-123">                                compoundCount++;</span>
<span class="source-line-no">124</span><span id="line-124">                                if (compoundCount == lineLength) {</span>
<span class="source-line-no">125</span><span id="line-125">                                        os.write(lineSep);</span>
<span class="source-line-no">126</span><span id="line-126">                                        compoundCount = 0;</span>
<span class="source-line-no">127</span><span id="line-127">                                }</span>
<span class="source-line-no">128</span><span id="line-128"></span>
<span class="source-line-no">129</span><span id="line-129">                        }</span>
<span class="source-line-no">130</span><span id="line-130"></span>
<span class="source-line-no">131</span><span id="line-131"></span>
<span class="source-line-no">132</span><span id="line-132">                        //If we had sequence which was a reciprocal of line length</span>
<span class="source-line-no">133</span><span id="line-133">                        //then don't write the line terminator as this has already written</span>
<span class="source-line-no">134</span><span id="line-134">                        //it</span>
<span class="source-line-no">135</span><span id="line-135">                        if ((sequence.getLength() % getLineLength()) != 0) {</span>
<span class="source-line-no">136</span><span id="line-136">                                os.write(lineSep);</span>
<span class="source-line-no">137</span><span id="line-137">                        }</span>
<span class="source-line-no">138</span><span id="line-138">                }</span>
<span class="source-line-no">139</span><span id="line-139">        }</span>
<span class="source-line-no">140</span><span id="line-140"></span>
<span class="source-line-no">141</span><span id="line-141">        /**</span>
<span class="source-line-no">142</span><span id="line-142">         * @return the lineLength</span>
<span class="source-line-no">143</span><span id="line-143">         */</span>
<span class="source-line-no">144</span><span id="line-144">        public int getLineLength() {</span>
<span class="source-line-no">145</span><span id="line-145">                return lineLength;</span>
<span class="source-line-no">146</span><span id="line-146">        }</span>
<span class="source-line-no">147</span><span id="line-147"></span>
<span class="source-line-no">148</span><span id="line-148">        /**</span>
<span class="source-line-no">149</span><span id="line-149">         * @param lineLength the lineLength to set</span>
<span class="source-line-no">150</span><span id="line-150">         */</span>
<span class="source-line-no">151</span><span id="line-151">        public void setLineLength(int lineLength) {</span>
<span class="source-line-no">152</span><span id="line-152">                this.lineLength = lineLength;</span>
<span class="source-line-no">153</span><span id="line-153">        }</span>
<span class="source-line-no">154</span><span id="line-154"></span>
<span class="source-line-no">155</span><span id="line-155">        </span>
<span class="source-line-no">156</span><span id="line-156">}</span>




























































</pre>
</div>
</main>
</body>
</html>
