<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence.io, class: FastaWriterHelper">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on 01-21-2010</span>
<span class="source-line-no">021</span><span id="line-21"> */</span>
<span class="source-line-no">022</span><span id="line-22">package org.biojava.nbio.core.sequence.io;</span>
<span class="source-line-no">023</span><span id="line-23"></span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.core.sequence.DNASequence;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.sequence.GeneSequence;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.io.template.FastaHeaderFormatInterface;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.template.Compound;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.sequence.template.Sequence;</span>
<span class="source-line-no">032</span><span id="line-32"></span>
<span class="source-line-no">033</span><span id="line-33">import java.io.BufferedOutputStream;</span>
<span class="source-line-no">034</span><span id="line-34">import java.io.File;</span>
<span class="source-line-no">035</span><span id="line-35">import java.io.FileOutputStream;</span>
<span class="source-line-no">036</span><span id="line-36">import java.io.OutputStream;</span>
<span class="source-line-no">037</span><span id="line-37">import java.util.ArrayList;</span>
<span class="source-line-no">038</span><span id="line-38">import java.util.Collection;</span>
<span class="source-line-no">039</span><span id="line-39"></span>
<span class="source-line-no">040</span><span id="line-40">/**</span>
<span class="source-line-no">041</span><span id="line-41"> * The class that should be used to write out fasta file of a sequence collection</span>
<span class="source-line-no">042</span><span id="line-42"> * @author Scooter Willis </span>
<span class="source-line-no">043</span><span id="line-43"> */</span>
<span class="source-line-no">044</span><span id="line-44">public class FastaWriterHelper {</span>
<span class="source-line-no">045</span><span id="line-45"></span>
<span class="source-line-no">046</span><span id="line-46"></span>
<span class="source-line-no">047</span><span id="line-47"></span>
<span class="source-line-no">048</span><span id="line-48">        /**</span>
<span class="source-line-no">049</span><span id="line-49">         * Write collection of protein sequences to a file</span>
<span class="source-line-no">050</span><span id="line-50">         *</span>
<span class="source-line-no">051</span><span id="line-51">         * @param file</span>
<span class="source-line-no">052</span><span id="line-52">         * @param proteinSequences</span>
<span class="source-line-no">053</span><span id="line-53">         * @throws Exception</span>
<span class="source-line-no">054</span><span id="line-54">         */</span>
<span class="source-line-no">055</span><span id="line-55">        public static void writeProteinSequence(File file,</span>
<span class="source-line-no">056</span><span id="line-56">                        Collection&lt;ProteinSequence&gt; proteinSequences) throws Exception {</span>
<span class="source-line-no">057</span><span id="line-57">                FileOutputStream outputStream = new FileOutputStream(file);</span>
<span class="source-line-no">058</span><span id="line-58">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span>
<span class="source-line-no">059</span><span id="line-59">                writeProteinSequence(bo, proteinSequences);</span>
<span class="source-line-no">060</span><span id="line-60">                bo.close();</span>
<span class="source-line-no">061</span><span id="line-61">                outputStream.close();</span>
<span class="source-line-no">062</span><span id="line-62">        }</span>
<span class="source-line-no">063</span><span id="line-63"></span>
<span class="source-line-no">064</span><span id="line-64">        /**</span>
<span class="source-line-no">065</span><span id="line-65">         * Write collection of protein sequences to a stream</span>
<span class="source-line-no">066</span><span id="line-66">         * @param outputStream</span>
<span class="source-line-no">067</span><span id="line-67">         * @param proteinSequences</span>
<span class="source-line-no">068</span><span id="line-68">         * @throws Exception</span>
<span class="source-line-no">069</span><span id="line-69">         */</span>
<span class="source-line-no">070</span><span id="line-70"></span>
<span class="source-line-no">071</span><span id="line-71">        public static void writeProteinSequence(OutputStream outputStream,</span>
<span class="source-line-no">072</span><span id="line-72">                        Collection&lt;ProteinSequence&gt; proteinSequences) throws Exception {</span>
<span class="source-line-no">073</span><span id="line-73"></span>
<span class="source-line-no">074</span><span id="line-74">                FastaWriter&lt;ProteinSequence, AminoAcidCompound&gt; fastaWriter = new FastaWriter&lt;&gt;(</span>
<span class="source-line-no">075</span><span id="line-75">                                outputStream, proteinSequences,</span>
<span class="source-line-no">076</span><span id="line-76">                                new GenericFastaHeaderFormat&lt;ProteinSequence, AminoAcidCompound&gt;());</span>
<span class="source-line-no">077</span><span id="line-77">                fastaWriter.process();</span>
<span class="source-line-no">078</span><span id="line-78"></span>
<span class="source-line-no">079</span><span id="line-79">        }</span>
<span class="source-line-no">080</span><span id="line-80"></span>
<span class="source-line-no">081</span><span id="line-81">                /**</span>
<span class="source-line-no">082</span><span id="line-82">         * Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</span>
<span class="source-line-no">083</span><span id="line-83">         * @param file</span>
<span class="source-line-no">084</span><span id="line-84">         * @param geneSequences</span>
<span class="source-line-no">085</span><span id="line-85">         * @throws Exception</span>
<span class="source-line-no">086</span><span id="line-86">         */</span>
<span class="source-line-no">087</span><span id="line-87"></span>
<span class="source-line-no">088</span><span id="line-88">        public static void writeGeneSequence(File file, Collection&lt;GeneSequence&gt; geneSequences,boolean showExonUppercase) throws Exception {</span>
<span class="source-line-no">089</span><span id="line-89">                FileOutputStream outputStream = new FileOutputStream(file);</span>
<span class="source-line-no">090</span><span id="line-90">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span>
<span class="source-line-no">091</span><span id="line-91">                writeGeneSequence(bo, geneSequences,showExonUppercase);</span>
<span class="source-line-no">092</span><span id="line-92">                bo.close();</span>
<span class="source-line-no">093</span><span id="line-93">                outputStream.close();</span>
<span class="source-line-no">094</span><span id="line-94">        }</span>
<span class="source-line-no">095</span><span id="line-95"></span>
<span class="source-line-no">096</span><span id="line-96">        /**</span>
<span class="source-line-no">097</span><span id="line-97">         * Write a collection of GeneSequences to a file where if the gene is negative strand it will flip and complement the sequence</span>
<span class="source-line-no">098</span><span id="line-98">         * @param outputStream</span>
<span class="source-line-no">099</span><span id="line-99">         * @param geneSequences</span>
<span class="source-line-no">100</span><span id="line-100">         * @throws Exception</span>
<span class="source-line-no">101</span><span id="line-101">         */</span>
<span class="source-line-no">102</span><span id="line-102"></span>
<span class="source-line-no">103</span><span id="line-103">        public static void writeGeneSequence(OutputStream outputStream, Collection&lt;GeneSequence&gt; geneSequences,boolean showExonUppercase) throws Exception {</span>
<span class="source-line-no">104</span><span id="line-104">                FastaGeneWriter fastaWriter = new FastaGeneWriter(</span>
<span class="source-line-no">105</span><span id="line-105">                                outputStream, geneSequences,</span>
<span class="source-line-no">106</span><span id="line-106">                                new GenericFastaHeaderFormat&lt;GeneSequence, NucleotideCompound&gt;(),showExonUppercase);</span>
<span class="source-line-no">107</span><span id="line-107">                fastaWriter.process();</span>
<span class="source-line-no">108</span><span id="line-108"></span>
<span class="source-line-no">109</span><span id="line-109">        }</span>
<span class="source-line-no">110</span><span id="line-110"></span>
<span class="source-line-no">111</span><span id="line-111"></span>
<span class="source-line-no">112</span><span id="line-112">        /**</span>
<span class="source-line-no">113</span><span id="line-113">         * Write a collection of NucleotideSequences to a file</span>
<span class="source-line-no">114</span><span id="line-114">         * @param file</span>
<span class="source-line-no">115</span><span id="line-115">         * @param dnaSequences</span>
<span class="source-line-no">116</span><span id="line-116">         * @throws Exception</span>
<span class="source-line-no">117</span><span id="line-117">         */</span>
<span class="source-line-no">118</span><span id="line-118"></span>
<span class="source-line-no">119</span><span id="line-119">        public static void writeNucleotideSequence(File file, Collection&lt;DNASequence&gt; dnaSequences) throws Exception {</span>
<span class="source-line-no">120</span><span id="line-120">                FileOutputStream outputStream = new FileOutputStream(file);</span>
<span class="source-line-no">121</span><span id="line-121">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span>
<span class="source-line-no">122</span><span id="line-122">                writeNucleotideSequence(bo, dnaSequences);</span>
<span class="source-line-no">123</span><span id="line-123">                bo.close();</span>
<span class="source-line-no">124</span><span id="line-124">                outputStream.close();</span>
<span class="source-line-no">125</span><span id="line-125">        }</span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">        /**</span>
<span class="source-line-no">128</span><span id="line-128">         * Write a collection of NucleotideSequences to a file</span>
<span class="source-line-no">129</span><span id="line-129">         * @param outputStream</span>
<span class="source-line-no">130</span><span id="line-130">         * @param dnaSequences</span>
<span class="source-line-no">131</span><span id="line-131">         * @throws Exception</span>
<span class="source-line-no">132</span><span id="line-132">         */</span>
<span class="source-line-no">133</span><span id="line-133"></span>
<span class="source-line-no">134</span><span id="line-134">        public static void writeNucleotideSequence(OutputStream outputStream, Collection&lt;DNASequence&gt; dnaSequences) throws Exception {</span>
<span class="source-line-no">135</span><span id="line-135">                FastaWriter&lt;DNASequence, NucleotideCompound&gt; fastaWriter = new FastaWriter&lt;&gt;(</span>
<span class="source-line-no">136</span><span id="line-136">                                outputStream, dnaSequences,</span>
<span class="source-line-no">137</span><span id="line-137">                                new GenericFastaHeaderFormat&lt;DNASequence, NucleotideCompound&gt;());</span>
<span class="source-line-no">138</span><span id="line-138">                fastaWriter.process();</span>
<span class="source-line-no">139</span><span id="line-139"></span>
<span class="source-line-no">140</span><span id="line-140">        }</span>
<span class="source-line-no">141</span><span id="line-141"></span>
<span class="source-line-no">142</span><span id="line-142">        /**</span>
<span class="source-line-no">143</span><span id="line-143">         * Write a sequence to a file</span>
<span class="source-line-no">144</span><span id="line-144">         * @param file</span>
<span class="source-line-no">145</span><span id="line-145">         * @param sequence</span>
<span class="source-line-no">146</span><span id="line-146">         * @throws Exception</span>
<span class="source-line-no">147</span><span id="line-147">         */</span>
<span class="source-line-no">148</span><span id="line-148">        public static void writeSequence(File file, Sequence&lt;?&gt; sequence) throws Exception {</span>
<span class="source-line-no">149</span><span id="line-149">                FileOutputStream outputStream = new FileOutputStream(file);</span>
<span class="source-line-no">150</span><span id="line-150">                BufferedOutputStream bo = new BufferedOutputStream(outputStream);</span>
<span class="source-line-no">151</span><span id="line-151">                writeSequences(bo, singleSeqToCollection(sequence));</span>
<span class="source-line-no">152</span><span id="line-152">                bo.close();</span>
<span class="source-line-no">153</span><span id="line-153">                outputStream.close();</span>
<span class="source-line-no">154</span><span id="line-154">        }</span>
<span class="source-line-no">155</span><span id="line-155"></span>
<span class="source-line-no">156</span><span id="line-156">        /**</span>
<span class="source-line-no">157</span><span id="line-157">         * Write a sequence to OutputStream</span>
<span class="source-line-no">158</span><span id="line-158">         * @param outputStream</span>
<span class="source-line-no">159</span><span id="line-159">         * @param sequence</span>
<span class="source-line-no">160</span><span id="line-160">         * @throws Exception</span>
<span class="source-line-no">161</span><span id="line-161">         */</span>
<span class="source-line-no">162</span><span id="line-162">        public static void writeSequence(OutputStream outputStream, Sequence&lt;?&gt; sequence) throws Exception {</span>
<span class="source-line-no">163</span><span id="line-163">                writeSequences(outputStream, singleSeqToCollection(sequence));</span>
<span class="source-line-no">164</span><span id="line-164">        }</span>
<span class="source-line-no">165</span><span id="line-165"></span>
<span class="source-line-no">166</span><span id="line-166">        /**</span>
<span class="source-line-no">167</span><span id="line-167">         *</span>
<span class="source-line-no">168</span><span id="line-168">         * @param sequence</span>
<span class="source-line-no">169</span><span id="line-169">         * @return</span>
<span class="source-line-no">170</span><span id="line-170">         */</span>
<span class="source-line-no">171</span><span id="line-171"></span>
<span class="source-line-no">172</span><span id="line-172">        private static Collection&lt;Sequence&lt;?&gt;&gt; singleSeqToCollection(Sequence&lt;?&gt; sequence) {</span>
<span class="source-line-no">173</span><span id="line-173">                Collection&lt;Sequence&lt;?&gt;&gt; sequences = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">174</span><span id="line-174">                sequences.add(sequence);</span>
<span class="source-line-no">175</span><span id="line-175">                return sequences;</span>
<span class="source-line-no">176</span><span id="line-176">        }</span>
<span class="source-line-no">177</span><span id="line-177"></span>
<span class="source-line-no">178</span><span id="line-178">        /**</span>
<span class="source-line-no">179</span><span id="line-179">         * Method which will write your given Sequences to the specified</span>
<span class="source-line-no">180</span><span id="line-180">         * {@link OutputStream}. This is a very generic method which writes just the</span>
<span class="source-line-no">181</span><span id="line-181">         * AccessionID of the Sequence as the FASTA header.</span>
<span class="source-line-no">182</span><span id="line-182">         *</span>
<span class="source-line-no">183</span><span id="line-183">         * @param outputStream Stream to write to; can be System.out</span>
<span class="source-line-no">184</span><span id="line-184">         * @param sequences The sequences to write out</span>
<span class="source-line-no">185</span><span id="line-185">         * @throws Exception Thrown normally thanks to IO problems</span>
<span class="source-line-no">186</span><span id="line-186">         */</span>
<span class="source-line-no">187</span><span id="line-187">        public static void writeSequences(OutputStream outputStream,</span>
<span class="source-line-no">188</span><span id="line-188">                        Collection&lt;Sequence&lt;?&gt;&gt; sequences) throws Exception {</span>
<span class="source-line-no">189</span><span id="line-189"></span>
<span class="source-line-no">190</span><span id="line-190">                FastaHeaderFormatInterface&lt;Sequence&lt;?&gt;, Compound&gt; fhfi =</span>
<span class="source-line-no">191</span><span id="line-191">                                new FastaHeaderFormatInterface&lt;Sequence&lt;?&gt;, Compound&gt;() {</span>
<span class="source-line-no">192</span><span id="line-192"></span>
<span class="source-line-no">193</span><span id="line-193">                                        @Override</span>
<span class="source-line-no">194</span><span id="line-194">                                        public String getHeader(Sequence&lt;?&gt; sequence) {</span>
<span class="source-line-no">195</span><span id="line-195">                                                return sequence.getAccession().toString();</span>
<span class="source-line-no">196</span><span id="line-196">                                        }</span>
<span class="source-line-no">197</span><span id="line-197"></span>
<span class="source-line-no">198</span><span id="line-198">                                        ;</span>
<span class="source-line-no">199</span><span id="line-199">                                };</span>
<span class="source-line-no">200</span><span id="line-200"></span>
<span class="source-line-no">201</span><span id="line-201">                FastaWriter&lt;Sequence&lt;?&gt;, Compound&gt; fastaWriter =</span>
<span class="source-line-no">202</span><span id="line-202">                                new FastaWriter&lt;&gt;(outputStream,</span>
<span class="source-line-no">203</span><span id="line-203">                                sequences, fhfi);</span>
<span class="source-line-no">204</span><span id="line-204"></span>
<span class="source-line-no">205</span><span id="line-205">                fastaWriter.process();</span>
<span class="source-line-no">206</span><span id="line-206">        }</span>
<span class="source-line-no">207</span><span id="line-207">}</span>




























































</pre>
</div>
</main>
</body>
</html>
