<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence.io, class: GenbankReaderHelper">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on 01-21-2010</span>
<span class="source-line-no">021</span><span id="line-21"> */</span>
<span class="source-line-no">022</span><span id="line-22">package org.biojava.nbio.core.sequence.io;</span>
<span class="source-line-no">023</span><span id="line-23"></span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.core.sequence.DNASequence;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.RNASequence;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.sequence.compound.RNACompoundSet;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.core.sequence.template.AbstractSequence;</span>
<span class="source-line-no">033</span><span id="line-33">import org.slf4j.Logger;</span>
<span class="source-line-no">034</span><span id="line-34">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">035</span><span id="line-35"></span>
<span class="source-line-no">036</span><span id="line-36">import java.io.File;</span>
<span class="source-line-no">037</span><span id="line-37">import java.io.FileInputStream;</span>
<span class="source-line-no">038</span><span id="line-38">import java.io.InputStream;</span>
<span class="source-line-no">039</span><span id="line-39">import java.util.LinkedHashMap;</span>
<span class="source-line-no">040</span><span id="line-40">import java.util.Map;</span>
<span class="source-line-no">041</span><span id="line-41"></span>
<span class="source-line-no">042</span><span id="line-42">/**</span>
<span class="source-line-no">043</span><span id="line-43"> *</span>
<span class="source-line-no">044</span><span id="line-44"> * @author Scooter Willis </span>
<span class="source-line-no">045</span><span id="line-45"> */</span>
<span class="source-line-no">046</span><span id="line-46">public class GenbankReaderHelper {</span>
<span class="source-line-no">047</span><span id="line-47"></span>
<span class="source-line-no">048</span><span id="line-48">        private final static Logger logger = LoggerFactory.getLogger(GenbankReaderHelper.class);</span>
<span class="source-line-no">049</span><span id="line-49"></span>
<span class="source-line-no">050</span><span id="line-50">        /**</span>
<span class="source-line-no">051</span><span id="line-51">         * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects</span>
<span class="source-line-no">052</span><span id="line-52">         * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested</span>
<span class="source-line-no">053</span><span id="line-53">         * in one sequence based on accession id.</span>
<span class="source-line-no">054</span><span id="line-54">         * @param file</span>
<span class="source-line-no">055</span><span id="line-55">         * @param lazySequenceLoad</span>
<span class="source-line-no">056</span><span id="line-56">         * @return</span>
<span class="source-line-no">057</span><span id="line-57">         * @throws Exception</span>
<span class="source-line-no">058</span><span id="line-58">         */</span>
<span class="source-line-no">059</span><span id="line-59">        public static Map&lt;String, DNASequence&gt; readGenbankDNASequence(File file, boolean lazySequenceLoad) throws Exception {</span>
<span class="source-line-no">060</span><span id="line-60">                if (!lazySequenceLoad) {</span>
<span class="source-line-no">061</span><span id="line-61">                        return readGenbankDNASequence(file);</span>
<span class="source-line-no">062</span><span id="line-62">                }</span>
<span class="source-line-no">063</span><span id="line-63"></span>
<span class="source-line-no">064</span><span id="line-64">                GenbankReader&lt;DNASequence, NucleotideCompound&gt; GenbankProxyReader =</span>
<span class="source-line-no">065</span><span id="line-65">                                new GenbankReader&lt;&gt;(</span>
<span class="source-line-no">066</span><span id="line-66">                                                file,</span>
<span class="source-line-no">067</span><span id="line-67">                                                new GenericGenbankHeaderParser&lt;DNASequence, NucleotideCompound&gt;(),</span>
<span class="source-line-no">068</span><span id="line-68">                                                new FileProxyDNASequenceCreator(</span>
<span class="source-line-no">069</span><span id="line-69">                                                                file,</span>
<span class="source-line-no">070</span><span id="line-70">                                                                DNACompoundSet.getDNACompoundSet(),</span>
<span class="source-line-no">071</span><span id="line-71">                                                                new GenbankSequenceParser&lt;AbstractSequence&lt;NucleotideCompound&gt;, NucleotideCompound&gt;()</span>
<span class="source-line-no">072</span><span id="line-72">                                                        )</span>
<span class="source-line-no">073</span><span id="line-73">                                        );</span>
<span class="source-line-no">074</span><span id="line-74">                return GenbankProxyReader.process();</span>
<span class="source-line-no">075</span><span id="line-75"></span>
<span class="source-line-no">076</span><span id="line-76">        }</span>
<span class="source-line-no">077</span><span id="line-77"></span>
<span class="source-line-no">078</span><span id="line-78">        /**</span>
<span class="source-line-no">079</span><span id="line-79">         * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects</span>
<span class="source-line-no">080</span><span id="line-80">         * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested</span>
<span class="source-line-no">081</span><span id="line-81">         * in one sequence based on accession id.</span>
<span class="source-line-no">082</span><span id="line-82">         * @param file</span>
<span class="source-line-no">083</span><span id="line-83">         * @param lazySequenceLoad</span>
<span class="source-line-no">084</span><span id="line-84">         * @return</span>
<span class="source-line-no">085</span><span id="line-85">         * @throws Exception</span>
<span class="source-line-no">086</span><span id="line-86">         */</span>
<span class="source-line-no">087</span><span id="line-87">        public static Map&lt;String, ProteinSequence&gt; readGenbankProteinSequence(File file, boolean lazySequenceLoad) throws Exception {</span>
<span class="source-line-no">088</span><span id="line-88">                if (!lazySequenceLoad) {</span>
<span class="source-line-no">089</span><span id="line-89">                        return readGenbankProteinSequence(file);</span>
<span class="source-line-no">090</span><span id="line-90">                }</span>
<span class="source-line-no">091</span><span id="line-91"></span>
<span class="source-line-no">092</span><span id="line-92">                GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; GenbankProxyReader =</span>
<span class="source-line-no">093</span><span id="line-93">                                new GenbankReader&lt;&gt;(</span>
<span class="source-line-no">094</span><span id="line-94">                                                file,</span>
<span class="source-line-no">095</span><span id="line-95">                                                new GenericGenbankHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</span>
<span class="source-line-no">096</span><span id="line-96">                                                new FileProxyProteinSequenceCreator(</span>
<span class="source-line-no">097</span><span id="line-97">                                                                file,</span>
<span class="source-line-no">098</span><span id="line-98">                                                                AminoAcidCompoundSet.getAminoAcidCompoundSet(),</span>
<span class="source-line-no">099</span><span id="line-99">                                                                new GenbankSequenceParser&lt;AbstractSequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;()</span>
<span class="source-line-no">100</span><span id="line-100">                                                        )</span>
<span class="source-line-no">101</span><span id="line-101">                                        );</span>
<span class="source-line-no">102</span><span id="line-102">                return GenbankProxyReader.process();</span>
<span class="source-line-no">103</span><span id="line-103"></span>
<span class="source-line-no">104</span><span id="line-104">        }</span>
<span class="source-line-no">105</span><span id="line-105"></span>
<span class="source-line-no">106</span><span id="line-106">        /**</span>
<span class="source-line-no">107</span><span id="line-107">         * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects</span>
<span class="source-line-no">108</span><span id="line-108">         * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested</span>
<span class="source-line-no">109</span><span id="line-109">         * in one sequence based on accession id.</span>
<span class="source-line-no">110</span><span id="line-110">         * @param file</span>
<span class="source-line-no">111</span><span id="line-111">         * @param lazySequenceLoad</span>
<span class="source-line-no">112</span><span id="line-112">         * @return</span>
<span class="source-line-no">113</span><span id="line-113">         * @throws Exception</span>
<span class="source-line-no">114</span><span id="line-114">         */</span>
<span class="source-line-no">115</span><span id="line-115">        public static Map&lt;String, RNASequence&gt; readGenbankRNASequence(File file, boolean lazySequenceLoad) throws Exception {</span>
<span class="source-line-no">116</span><span id="line-116">                if (!lazySequenceLoad) {</span>
<span class="source-line-no">117</span><span id="line-117">                        return readGenbankRNASequence(file);</span>
<span class="source-line-no">118</span><span id="line-118">                }</span>
<span class="source-line-no">119</span><span id="line-119"></span>
<span class="source-line-no">120</span><span id="line-120">                GenbankReader&lt;RNASequence, NucleotideCompound&gt; GenbankProxyReader =</span>
<span class="source-line-no">121</span><span id="line-121">                                new GenbankReader&lt;&gt;(</span>
<span class="source-line-no">122</span><span id="line-122">                                                file,</span>
<span class="source-line-no">123</span><span id="line-123">                                                new GenericGenbankHeaderParser&lt;RNASequence, NucleotideCompound&gt;(),</span>
<span class="source-line-no">124</span><span id="line-124">                                                new FileProxyRNASequenceCreator(</span>
<span class="source-line-no">125</span><span id="line-125">                                                                file,</span>
<span class="source-line-no">126</span><span id="line-126">                                                                RNACompoundSet.getRNACompoundSet(),</span>
<span class="source-line-no">127</span><span id="line-127">                                                                new GenbankSequenceParser&lt;AbstractSequence&lt;NucleotideCompound&gt;, NucleotideCompound&gt;()</span>
<span class="source-line-no">128</span><span id="line-128">                                                        )</span>
<span class="source-line-no">129</span><span id="line-129">                                        );</span>
<span class="source-line-no">130</span><span id="line-130">                return GenbankProxyReader.process();</span>
<span class="source-line-no">131</span><span id="line-131"></span>
<span class="source-line-no">132</span><span id="line-132">        }</span>
<span class="source-line-no">133</span><span id="line-133"></span>
<span class="source-line-no">134</span><span id="line-134">        /**</span>
<span class="source-line-no">135</span><span id="line-135">         * Read a Genbank file containing amino acids with setup that would handle most</span>
<span class="source-line-no">136</span><span id="line-136">         * cases.</span>
<span class="source-line-no">137</span><span id="line-137">         *</span>
<span class="source-line-no">138</span><span id="line-138">         * @param file</span>
<span class="source-line-no">139</span><span id="line-139">         * @return</span>
<span class="source-line-no">140</span><span id="line-140">         * @throws Exception</span>
<span class="source-line-no">141</span><span id="line-141">         */</span>
<span class="source-line-no">142</span><span id="line-142">        public static Map&lt;String, ProteinSequence&gt; readGenbankProteinSequence(</span>
<span class="source-line-no">143</span><span id="line-143">                        File file) throws Exception {</span>
<span class="source-line-no">144</span><span id="line-144">                FileInputStream inStream = new FileInputStream(file);</span>
<span class="source-line-no">145</span><span id="line-145">                Map&lt;String, ProteinSequence&gt; proteinSequences = readGenbankProteinSequence(inStream);</span>
<span class="source-line-no">146</span><span id="line-146">                inStream.close();</span>
<span class="source-line-no">147</span><span id="line-147">                return proteinSequences;</span>
<span class="source-line-no">148</span><span id="line-148">        }</span>
<span class="source-line-no">149</span><span id="line-149"></span>
<span class="source-line-no">150</span><span id="line-150">        /**</span>
<span class="source-line-no">151</span><span id="line-151">         * Read a Genbank file containing amino acids with setup that would handle most</span>
<span class="source-line-no">152</span><span id="line-152">         * cases. User is responsible for closing InputStream because you opened it</span>
<span class="source-line-no">153</span><span id="line-153">         *</span>
<span class="source-line-no">154</span><span id="line-154">         * @param inStream</span>
<span class="source-line-no">155</span><span id="line-155">         * @return</span>
<span class="source-line-no">156</span><span id="line-156">         * @throws Exception</span>
<span class="source-line-no">157</span><span id="line-157">         */</span>
<span class="source-line-no">158</span><span id="line-158">        public static Map&lt;String, ProteinSequence&gt; readGenbankProteinSequence(</span>
<span class="source-line-no">159</span><span id="line-159">                        InputStream inStream) throws Exception {</span>
<span class="source-line-no">160</span><span id="line-160">                GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; GenbankReader = new GenbankReader&lt;&gt;(</span>
<span class="source-line-no">161</span><span id="line-161">                                inStream,</span>
<span class="source-line-no">162</span><span id="line-162">                                new GenericGenbankHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</span>
<span class="source-line-no">163</span><span id="line-163">                                new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));</span>
<span class="source-line-no">164</span><span id="line-164">                return GenbankReader.process();</span>
<span class="source-line-no">165</span><span id="line-165">        }</span>
<span class="source-line-no">166</span><span id="line-166"></span>
<span class="source-line-no">167</span><span id="line-167">        /**</span>
<span class="source-line-no">168</span><span id="line-168">         * Read a Genbank DNA sequence</span>
<span class="source-line-no">169</span><span id="line-169">         * @param inStream</span>
<span class="source-line-no">170</span><span id="line-170">         * @return</span>
<span class="source-line-no">171</span><span id="line-171">         * @throws Exception</span>
<span class="source-line-no">172</span><span id="line-172">         */</span>
<span class="source-line-no">173</span><span id="line-173">        public static Map&lt;String, DNASequence&gt; readGenbankDNASequence(</span>
<span class="source-line-no">174</span><span id="line-174">                        InputStream inStream) throws Exception {</span>
<span class="source-line-no">175</span><span id="line-175">                GenbankReader&lt;DNASequence, NucleotideCompound&gt; GenbankReader = new GenbankReader&lt;&gt;(</span>
<span class="source-line-no">176</span><span id="line-176">                                inStream,</span>
<span class="source-line-no">177</span><span id="line-177">                                new GenericGenbankHeaderParser&lt;DNASequence, NucleotideCompound&gt;(),</span>
<span class="source-line-no">178</span><span id="line-178">                                new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));</span>
<span class="source-line-no">179</span><span id="line-179">                return GenbankReader.process();</span>
<span class="source-line-no">180</span><span id="line-180">        }</span>
<span class="source-line-no">181</span><span id="line-181"></span>
<span class="source-line-no">182</span><span id="line-182">        /**</span>
<span class="source-line-no">183</span><span id="line-183">         *</span>
<span class="source-line-no">184</span><span id="line-184">         * @param file</span>
<span class="source-line-no">185</span><span id="line-185">         * @return</span>
<span class="source-line-no">186</span><span id="line-186">         * @throws Exception</span>
<span class="source-line-no">187</span><span id="line-187">         */</span>
<span class="source-line-no">188</span><span id="line-188">        public static Map&lt;String, DNASequence&gt; readGenbankDNASequence(</span>
<span class="source-line-no">189</span><span id="line-189">                        File file) throws Exception {</span>
<span class="source-line-no">190</span><span id="line-190">                FileInputStream inStream = new FileInputStream(file);</span>
<span class="source-line-no">191</span><span id="line-191">                Map&lt;String, DNASequence&gt; dnaSequences = readGenbankDNASequence(inStream);</span>
<span class="source-line-no">192</span><span id="line-192">                inStream.close();</span>
<span class="source-line-no">193</span><span id="line-193">                return dnaSequences;</span>
<span class="source-line-no">194</span><span id="line-194">        }</span>
<span class="source-line-no">195</span><span id="line-195">        /**</span>
<span class="source-line-no">196</span><span id="line-196">         * Read a Genbank RNA sequence</span>
<span class="source-line-no">197</span><span id="line-197">         * @param inStream</span>
<span class="source-line-no">198</span><span id="line-198">         * @return</span>
<span class="source-line-no">199</span><span id="line-199">         * @throws Exception</span>
<span class="source-line-no">200</span><span id="line-200">         */</span>
<span class="source-line-no">201</span><span id="line-201">        public static Map&lt;String, RNASequence&gt; readGenbankRNASequence(</span>
<span class="source-line-no">202</span><span id="line-202">                        InputStream inStream) throws Exception {</span>
<span class="source-line-no">203</span><span id="line-203">                GenbankReader&lt;RNASequence, NucleotideCompound&gt; GenbankReader = new GenbankReader&lt;&gt;(</span>
<span class="source-line-no">204</span><span id="line-204">                                inStream,</span>
<span class="source-line-no">205</span><span id="line-205">                                new GenericGenbankHeaderParser&lt;RNASequence, NucleotideCompound&gt;(),</span>
<span class="source-line-no">206</span><span id="line-206">                                new RNASequenceCreator(RNACompoundSet.getRNACompoundSet()));</span>
<span class="source-line-no">207</span><span id="line-207">                return GenbankReader.process();</span>
<span class="source-line-no">208</span><span id="line-208">        }</span>
<span class="source-line-no">209</span><span id="line-209"></span>
<span class="source-line-no">210</span><span id="line-210">        /**</span>
<span class="source-line-no">211</span><span id="line-211">         *</span>
<span class="source-line-no">212</span><span id="line-212">         * @param file</span>
<span class="source-line-no">213</span><span id="line-213">         * @return</span>
<span class="source-line-no">214</span><span id="line-214">         * @throws Exception</span>
<span class="source-line-no">215</span><span id="line-215">         */</span>
<span class="source-line-no">216</span><span id="line-216">        public static Map&lt;String, RNASequence&gt; readGenbankRNASequence(</span>
<span class="source-line-no">217</span><span id="line-217">                        File file) throws Exception {</span>
<span class="source-line-no">218</span><span id="line-218">                FileInputStream inStream = new FileInputStream(file);</span>
<span class="source-line-no">219</span><span id="line-219">                Map&lt;String, RNASequence&gt; rnaSequences = readGenbankRNASequence(inStream);</span>
<span class="source-line-no">220</span><span id="line-220">                inStream.close();</span>
<span class="source-line-no">221</span><span id="line-221">                return rnaSequences;</span>
<span class="source-line-no">222</span><span id="line-222">        }</span>
<span class="source-line-no">223</span><span id="line-223"></span>
<span class="source-line-no">224</span><span id="line-224">}</span>




























































</pre>
</div>
</main>
</body>
</html>
