<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_131) on Thu Oct 19 12:07:58 PDT 2017 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>TranscriptSequence (biojava 4.2.9 API)</title>
<meta name="date" content="2017-10-19">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="TranscriptSequence (biojava 4.2.9 API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":10,"i1":10,"i2":10,"i3":10,"i4":10,"i5":10,"i6":10,"i7":10,"i8":10,"i9":10,"i10":10,"i11":10,"i12":10};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],8:["t4","Concrete Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TranscriptSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li>Next&nbsp;Class</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/TranscriptSequence.html" target="_top">Frames</a></li>
<li><a href="TranscriptSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.org.biojava.nbio.core.sequence.DNASequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.core.sequence</div>
<h2 title="Class TranscriptSequence" class="title">Class TranscriptSequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template.AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</li>
<li>
<ul class="inheritance">
<li><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence.DNASequence</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.core.sequence.TranscriptSequence</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;, <a href="../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</dd>
</dl>
<hr>
<br>
<pre>public class <a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.40">TranscriptSequence</a>
extends <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></pre>
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence. Need to start with a
 ChromosomeSequence then getting a GeneSequence and then a TranscriptSequence</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Scooter Willis</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested.class.summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested.classes.inherited.from.class.org.biojava.nbio.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava.nbio.core.sequence">DNASequence.DNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested.classes.inherited.from.class.org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#TranscriptSequence-org.biojava.nbio.core.sequence.GeneSequence-int-int-">TranscriptSequence</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&nbsp;parentDNASequence,
                  int&nbsp;begin,
                  int&nbsp;end)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t4" class="tableTab"><span><a href="javascript:show(8);">Concrete Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#addCDS-org.biojava.nbio.core.sequence.AccessionID-int-int-int-">addCDS</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
      int&nbsp;begin,
      int&nbsp;end,
      int&nbsp;phase)</code>
<div class="block">Add a Coding Sequence region with phase to the transcript sequence</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#addStartCodonSequence-org.biojava.nbio.core.sequence.AccessionID-int-int-">addStartCodonSequence</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                     int&nbsp;begin,
                     int&nbsp;end)</code>&nbsp;</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#addStopCodonSequence-org.biojava.nbio.core.sequence.AccessionID-int-int-">addStopCodonSequence</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                    int&nbsp;begin,
                    int&nbsp;end)</code>&nbsp;</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getCDSSequences--">getCDSSequences</a></span>()</code>
<div class="block">Get the CDS sequences that have been added to the TranscriptSequences</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getDNACodingSequence--">getDNACodingSequence</a></span>()</code>
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getLength--">getLength</a></span>()</code>
<div class="block">Returns the length of the Sequence</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getProteinCDSSequences--">getProteinCDSSequences</a></span>()</code>
<div class="block">Return a list of protein sequences based on each CDS sequence
 where the phase shift between two CDS sequences is assigned to the
 CDS sequence that starts the triplet.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getProteinSequence--">getProteinSequence</a></span>()</code>
<div class="block">Get the protein sequence</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getProteinSequence-org.biojava.nbio.core.sequence.transcription.TranscriptionEngine-">getProteinSequence</a></span>(<a href="../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</code>
<div class="block">Get the protein sequence with user defined TranscriptEngine</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getStartCodonSequence--">getStartCodonSequence</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i10" class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getStopCodonSequence--">getStopCodonSequence</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i11" class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getStrand--">getStrand</a></span>()</code>&nbsp;</td>
</tr>
<tr id="i12" class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#removeCDS-java.lang.String-">removeCDS</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Remove a CDS or coding sequence from the transcript sequence</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.org.biojava.nbio.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getComplement--">getComplement</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getDNAType--">getDNAType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getGCCount--">getGCCount</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverse--">getReverse</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverseComplement--">getReverseComplement</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence--">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence-org.biojava.nbio.core.sequence.transcription.Frame-">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence-org.biojava.nbio.core.sequence.transcription.TranscriptionEngine-">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence-org.biojava.nbio.core.sequence.transcription.TranscriptionEngine-org.biojava.nbio.core.sequence.transcription.Frame-">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#main-java.lang.String:A-">main</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#setDNAType-org.biojava.nbio.core.sequence.DNASequence.DNAType-">setDNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature-org.biojava.nbio.core.sequence.features.FeatureInterface-">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature-int-int-org.biojava.nbio.core.sequence.features.FeatureInterface-">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addNote-java.lang.String-">addNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#countCompounds-C...-">countCompounds</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAccession--">getAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAnnotationType--">getAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAsList--">getAsList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioBegin--">getBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioEnd--">getBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundAt-int-">getCompoundAt</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundSet--">getCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDatabaseReferences--">getDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDescription--">getDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatureRetriever--">getFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures--">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures-int-">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures-java.lang.String-int-">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesByType-java.lang.String-">getFeaturesByType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord--">getFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getIndexOf-C-">getIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getInverse--">getInverse</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLastIndexOf-C-">getLastIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getNotesList--">getNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getOriginalHeader--">getOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getParentSequence--">getParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getProxySequenceReader--">getProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString--">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString-java.lang.Integer-java.lang.Integer-org.biojava.nbio.core.sequence.Strand-">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceScore--">getSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSource--">getSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSubSequence-java.lang.Integer-java.lang.Integer-">getSubSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getTaxonomy--">getTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getUserCollection--">getUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#iterator--">iterator</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeFeature-org.biojava.nbio.core.sequence.features.FeatureInterface-">removeFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeNote-java.lang.String-">removeNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAccession-org.biojava.nbio.core.sequence.AccessionID-">setAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAnnotationType-org.biojava.nbio.core.sequence.template.AbstractSequence.AnnotationType-">setAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioBegin-java.lang.Integer-">setBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioEnd-java.lang.Integer-">setBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setCompoundSet-org.biojava.nbio.core.sequence.template.CompoundSet-">setCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDatabaseReferences-org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface-">setDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDescription-java.lang.String-">setDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeatureRetriever-org.biojava.nbio.core.sequence.features.FeatureRetriever-">setFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord-org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface-">setFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setNotesList-java.util.ArrayList-">setNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setOriginalHeader-java.lang.String-">setOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setParentSequence-org.biojava.nbio.core.sequence.template.AbstractSequence-">setParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setProxySequenceReader-org.biojava.nbio.core.sequence.template.SequenceReader-">setProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSequenceScore-java.lang.Double-">setSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSource-java.lang.String-">setSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setTaxonomy-org.biojava.nbio.core.sequence.TaxonomyID-">setTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setUserCollection-java.util.Collection-">setUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#toString--">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone--" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals-java.lang.Object-" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize--" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass--" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode--" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify--" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll--" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait--" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait-long-" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait-long-int-" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Iterable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true#forEach-java.util.function.Consumer-" title="class or interface in java.lang">forEach</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true#spliterator--" title="class or interface in java.lang">spliterator</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor.detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="TranscriptSequence-org.biojava.nbio.core.sequence.GeneSequence-int-int-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>TranscriptSequence</h4>
<pre>public&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.56">TranscriptSequence</a>(<a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&nbsp;parentDNASequence,
                          int&nbsp;begin,
                          int&nbsp;end)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>parentDNASequence</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - inclusive of end</dd>
</dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getLength--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLength</h4>
<pre>public&nbsp;int&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.66">getLength</a>()</pre>
<div class="block"><span class="descfrmTypeLabel">Description copied from interface:&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLength--">Sequence</a></code></span></div>
<div class="block">Returns the length of the Sequence</div>
<dl>
<dt><span class="overrideSpecifyLabel">Specified by:</span></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLength--">getLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt><span class="overrideSpecifyLabel">Overrides:</span></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLength--">getLength</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
</li>
</ul>
<a name="getStrand--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStrand</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.73">getStrand</a>()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the strand</dd>
</dl>
</li>
</ul>
<a name="removeCDS-java.lang.String-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeCDS</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.82">removeCDS</a>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block">Remove a CDS or coding sequence from the transcript sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a name="getCDSSequences--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCDSSequences</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a>&gt;&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.97">getCDSSequences</a>()</pre>
<div class="block">Get the CDS sequences that have been added to the TranscriptSequences</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a name="addCDS-org.biojava.nbio.core.sequence.AccessionID-int-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addCDS</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.109">addCDS</a>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                          int&nbsp;begin,
                          int&nbsp;end,
                          int&nbsp;phase)
                   throws <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></pre>
<div class="block">Add a Coding Sequence region with phase to the transcript sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>accession</code> - </dd>
<dd><code>begin</code> - </dd>
<dd><code>end</code> - </dd>
<dd><code>phase</code> - 0,1,2</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></code></dd>
</dl>
</li>
</ul>
<a name="getProteinCDSSequences--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinCDSSequences</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.139">getProteinCDSSequences</a>()</pre>
<div class="block">Return a list of protein sequences based on each CDS sequence
 where the phase shift between two CDS sequences is assigned to the
 CDS sequence that starts the triplet. This can be used to map
 a CDS/exon region of a protein sequence back to the DNA sequence
 If you have a protein sequence and a predicted gene you can take the
 predict CDS protein sequences and align back to the protein sequence.
 If you have errors in mapping the predicted protein CDS regions to
 an the known protein sequence then you can identify possible errors
 in the prediction</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a name="getDNACodingSequence--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDNACodingSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.202">getDNACodingSequence</a>()</pre>
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a name="getProteinSequence--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.223">getProteinSequence</a>()</pre>
<div class="block">Get the protein sequence</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a name="getProteinSequence-org.biojava.nbio.core.sequence.transcription.TranscriptionEngine-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.232">getProteinSequence</a>(<a href="../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</pre>
<div class="block">Get the protein sequence with user defined TranscriptEngine</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>engine</code> - </dd>
<dt><span class="returnLabel">Returns:</span></dt>
</dl>
</li>
</ul>
<a name="getStartCodonSequence--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStartCodonSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.244">getStartCodonSequence</a>()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the startCodonSequence</dd>
</dl>
</li>
</ul>
<a name="addStartCodonSequence-org.biojava.nbio.core.sequence.AccessionID-int-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addStartCodonSequence</h4>
<pre>public&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.251">addStartCodonSequence</a>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                                  int&nbsp;begin,
                                  int&nbsp;end)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>startCodonSequence</code> - the startCodonSequence to set</dd>
</dl>
</li>
</ul>
<a name="getStopCodonSequence--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStopCodonSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a>&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.259">getStopCodonSequence</a>()</pre>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the stopCodonSequence</dd>
</dl>
</li>
</ul>
<a name="addStopCodonSequence-org.biojava.nbio.core.sequence.AccessionID-int-int-">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>addStopCodonSequence</h4>
<pre>public&nbsp;void&nbsp;<a href="../../../../../src-html/org/biojava/nbio/core/sequence/TranscriptSequence.html#line.266">addStopCodonSequence</a>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                                 int&nbsp;begin,
                                 int&nbsp;end)</pre>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>stopCodonSequence</code> - the stopCodonSequence to set</dd>
</dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/TranscriptSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li>Next&nbsp;Class</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/TranscriptSequence.html" target="_top">Frames</a></li>
<li><a href="TranscriptSequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested.classes.inherited.from.class.org.biojava.nbio.core.sequence.DNASequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor.detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2017 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
