<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.bio.symbol;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.io.BufferedReader;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.io.IOException;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.io.InputStream;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.io.InputStreamReader;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.io.OutputStream;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.io.PrintWriter;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.util.HashMap;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.util.Iterator;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.List;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.Map;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.util.StringTokenizer;<a name="line.34"></a>
<span class="sourceLineNo">035</span><a name="line.35"></a>
<span class="sourceLineNo">036</span>import javax.xml.parsers.DocumentBuilder;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import javax.xml.parsers.DocumentBuilderFactory;<a name="line.37"></a>
<span class="sourceLineNo">038</span><a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.bio.BioException;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.bio.dist.Count;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.bio.dist.Distribution;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.bio.dist.DistributionTools;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.biojava.bio.dist.IndexedCount;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import org.biojava.bio.seq.RNATools;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import org.biojava.bio.seq.io.SymbolTokenization;<a name="line.45"></a>
<span class="sourceLineNo">046</span>import org.biojava.utils.ChangeVetoException;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import org.biojava.utils.ClassTools;<a name="line.47"></a>
<span class="sourceLineNo">048</span>import org.biojava.utils.xml.PrettyXMLWriter;<a name="line.48"></a>
<span class="sourceLineNo">049</span>import org.biojava.utils.xml.XMLWriter;<a name="line.49"></a>
<span class="sourceLineNo">050</span>import org.w3c.dom.Document;<a name="line.50"></a>
<span class="sourceLineNo">051</span>import org.w3c.dom.Element;<a name="line.51"></a>
<span class="sourceLineNo">052</span>import org.w3c.dom.Node;<a name="line.52"></a>
<span class="sourceLineNo">053</span>import org.w3c.dom.NodeList;<a name="line.53"></a>
<span class="sourceLineNo">054</span><a name="line.54"></a>
<span class="sourceLineNo">055</span>/**<a name="line.55"></a>
<span class="sourceLineNo">056</span> * An utility class for codon preferences<a name="line.56"></a>
<span class="sourceLineNo">057</span> *<a name="line.57"></a>
<span class="sourceLineNo">058</span> * @author David Huen<a name="line.58"></a>
<span class="sourceLineNo">059</span> * @author Mark Schreiber<a name="line.59"></a>
<span class="sourceLineNo">060</span> * @since 1.3<a name="line.60"></a>
<span class="sourceLineNo">061</span> */<a name="line.61"></a>
<span class="sourceLineNo">062</span>public class CodonPrefTools<a name="line.62"></a>
<span class="sourceLineNo">063</span>{<a name="line.63"></a>
<span class="sourceLineNo">064</span>    /**<a name="line.64"></a>
<span class="sourceLineNo">065</span>     * constants for model organisms<a name="line.65"></a>
<span class="sourceLineNo">066</span>     */<a name="line.66"></a>
<span class="sourceLineNo">067</span>    static String JUNIT = "jUnit use only!!!!";<a name="line.67"></a>
<span class="sourceLineNo">068</span>    /**<a name="line.68"></a>
<span class="sourceLineNo">069</span>     * Drosophila melanogaster codon preferences<a name="line.69"></a>
<span class="sourceLineNo">070</span>     */<a name="line.70"></a>
<span class="sourceLineNo">071</span>    public static String DROSOPHILA_MELANOGASTER_NUCLEAR = "Drosophila melanogaster";<a name="line.71"></a>
<span class="sourceLineNo">072</span>    /**<a name="line.72"></a>
<span class="sourceLineNo">073</span>     * Homo sapiens codon preferences<a name="line.73"></a>
<span class="sourceLineNo">074</span>     */<a name="line.74"></a>
<span class="sourceLineNo">075</span>    public static String MAN_NUCLEAR = "Homo sapiens";<a name="line.75"></a>
<span class="sourceLineNo">076</span>    /**<a name="line.76"></a>
<span class="sourceLineNo">077</span>     * Mus musculus codon preferences<a name="line.77"></a>
<span class="sourceLineNo">078</span>     */<a name="line.78"></a>
<span class="sourceLineNo">079</span>    public static String MOUSE_NUCLEAR = "Mus musculus";<a name="line.79"></a>
<span class="sourceLineNo">080</span>    /**<a name="line.80"></a>
<span class="sourceLineNo">081</span>     * Rattus norvegicus codon preferences<a name="line.81"></a>
<span class="sourceLineNo">082</span>     */<a name="line.82"></a>
<span class="sourceLineNo">083</span>    public static String RAT_NUCLEAR = "Rattus norvegicus";<a name="line.83"></a>
<span class="sourceLineNo">084</span>    /**<a name="line.84"></a>
<span class="sourceLineNo">085</span>     * Takifugu rubripes codon preferences<a name="line.85"></a>
<span class="sourceLineNo">086</span>     */<a name="line.86"></a>
<span class="sourceLineNo">087</span>    public static String FUGU_NUCLEAR = "Takifugu rubripes";<a name="line.87"></a>
<span class="sourceLineNo">088</span>    /**<a name="line.88"></a>
<span class="sourceLineNo">089</span>     * Caenorhabditis elegans codon preferences<a name="line.89"></a>
<span class="sourceLineNo">090</span>     */<a name="line.90"></a>
<span class="sourceLineNo">091</span>    public static String WORM_NUCLEAR = "Caenorhabditis elegans";<a name="line.91"></a>
<span class="sourceLineNo">092</span>    /**<a name="line.92"></a>
<span class="sourceLineNo">093</span>     * Saccharomyces cerevisiae codon preferences<a name="line.93"></a>
<span class="sourceLineNo">094</span>     */<a name="line.94"></a>
<span class="sourceLineNo">095</span>    public static String CEREVISIAE_NUCLEAR = "Saccharomyces cerevisiae";<a name="line.95"></a>
<span class="sourceLineNo">096</span>    /**<a name="line.96"></a>
<span class="sourceLineNo">097</span>     * Schizosaccharomyces pombe codon preferences<a name="line.97"></a>
<span class="sourceLineNo">098</span>     */<a name="line.98"></a>
<span class="sourceLineNo">099</span>    public static String POMBE_NUCLEAR = "Schizosaccharomyces pombe";<a name="line.99"></a>
<span class="sourceLineNo">100</span>    /**<a name="line.100"></a>
<span class="sourceLineNo">101</span>     * Escherichia coli codon preferences<a name="line.101"></a>
<span class="sourceLineNo">102</span>     */<a name="line.102"></a>
<span class="sourceLineNo">103</span>    public static String ECOLI = "Escherichia coli";<a name="line.103"></a>
<span class="sourceLineNo">104</span><a name="line.104"></a>
<span class="sourceLineNo">105</span>    private static Map prefMap;<a name="line.105"></a>
<span class="sourceLineNo">106</span><a name="line.106"></a>
<span class="sourceLineNo">107</span>    final private static AtomicSymbol [] cutg = new AtomicSymbol[64];<a name="line.107"></a>
<span class="sourceLineNo">108</span><a name="line.108"></a>
<span class="sourceLineNo">109</span>    static {<a name="line.109"></a>
<span class="sourceLineNo">110</span>        prefMap = new HashMap();<a name="line.110"></a>
<span class="sourceLineNo">111</span><a name="line.111"></a>
<span class="sourceLineNo">112</span>        loadCodonPreferences();<a name="line.112"></a>
<span class="sourceLineNo">113</span><a name="line.113"></a>
<span class="sourceLineNo">114</span>        try {<a name="line.114"></a>
<span class="sourceLineNo">115</span>            loadCodonOrder();<a name="line.115"></a>
<span class="sourceLineNo">116</span>        }<a name="line.116"></a>
<span class="sourceLineNo">117</span>        catch (IllegalSymbolException ise) {}<a name="line.117"></a>
<span class="sourceLineNo">118</span>    }<a name="line.118"></a>
<span class="sourceLineNo">119</span><a name="line.119"></a>
<span class="sourceLineNo">120</span>    private static class LoadEverythingSelector implements CodonPrefFilter<a name="line.120"></a>
<span class="sourceLineNo">121</span>    {<a name="line.121"></a>
<span class="sourceLineNo">122</span>        public boolean isRequired(String id) { return true; }<a name="line.122"></a>
<span class="sourceLineNo">123</span>        public void put(CodonPref codonPref)<a name="line.123"></a>
<span class="sourceLineNo">124</span>        {<a name="line.124"></a>
<span class="sourceLineNo">125</span>            prefMap.put(codonPref.getName(), codonPref);<a name="line.125"></a>
<span class="sourceLineNo">126</span>        }<a name="line.126"></a>
<span class="sourceLineNo">127</span>    }<a name="line.127"></a>
<span class="sourceLineNo">128</span><a name="line.128"></a>
<span class="sourceLineNo">129</span>    /**<a name="line.129"></a>
<span class="sourceLineNo">130</span>     * get the specified codon preference.<a name="line.130"></a>
<span class="sourceLineNo">131</span>     */<a name="line.131"></a>
<span class="sourceLineNo">132</span>    public static CodonPref getCodonPreference(String id)<a name="line.132"></a>
<span class="sourceLineNo">133</span>    {<a name="line.133"></a>
<span class="sourceLineNo">134</span>        return (CodonPref) prefMap.get(id);<a name="line.134"></a>
<span class="sourceLineNo">135</span>    }<a name="line.135"></a>
<span class="sourceLineNo">136</span><a name="line.136"></a>
<span class="sourceLineNo">137</span>    private static void loadCodonPreferences()<a name="line.137"></a>
<span class="sourceLineNo">138</span>    {<a name="line.138"></a>
<span class="sourceLineNo">139</span>        try {<a name="line.139"></a>
<span class="sourceLineNo">140</span>            // parse the predefined codon preferences<a name="line.140"></a>
<span class="sourceLineNo">141</span>            InputStream prefStream = ClassTools.getClassLoader(CodonPrefTools.class).getResourceAsStream(<a name="line.141"></a>
<span class="sourceLineNo">142</span>                "org/biojava/bio/symbol/CodonPrefTables.xml"<a name="line.142"></a>
<span class="sourceLineNo">143</span>            );<a name="line.143"></a>
<span class="sourceLineNo">144</span><a name="line.144"></a>
<span class="sourceLineNo">145</span>            CodonPrefFilter select = new LoadEverythingSelector();<a name="line.145"></a>
<span class="sourceLineNo">146</span>            readFromXML(prefStream, select);<a name="line.146"></a>
<span class="sourceLineNo">147</span>        }<a name="line.147"></a>
<span class="sourceLineNo">148</span>        catch (Exception e) { e.printStackTrace(); }<a name="line.148"></a>
<span class="sourceLineNo">149</span>    }<a name="line.149"></a>
<span class="sourceLineNo">150</span><a name="line.150"></a>
<span class="sourceLineNo">151</span>    /**<a name="line.151"></a>
<span class="sourceLineNo">152</span>     * returns an RNA dinucleotide alphabet.<a name="line.152"></a>
<span class="sourceLineNo">153</span>     * Used to represent the non-wobble bases in WobbleDistribution<a name="line.153"></a>
<span class="sourceLineNo">154</span>     */<a name="line.154"></a>
<span class="sourceLineNo">155</span>    public static FiniteAlphabet getDinucleotideAlphabet()<a name="line.155"></a>
<span class="sourceLineNo">156</span>    {<a name="line.156"></a>
<span class="sourceLineNo">157</span>        return (FiniteAlphabet)AlphabetManager.generateCrossProductAlphaFromName("(RNA x RNA)");<a name="line.157"></a>
<span class="sourceLineNo">158</span>    }<a name="line.158"></a>
<span class="sourceLineNo">159</span><a name="line.159"></a>
<span class="sourceLineNo">160</span>    /**<a name="line.160"></a>
<span class="sourceLineNo">161</span>     * write out a specified CodonPref object in XML format.<a name="line.161"></a>
<span class="sourceLineNo">162</span>     */<a name="line.162"></a>
<span class="sourceLineNo">163</span>    public static void writeToXML(CodonPref codonPref, PrintWriter writer)<a name="line.163"></a>
<span class="sourceLineNo">164</span>        throws NullPointerException, IOException, IllegalSymbolException, BioException<a name="line.164"></a>
<span class="sourceLineNo">165</span>    {<a name="line.165"></a>
<span class="sourceLineNo">166</span>        XMLWriter xw = new PrettyXMLWriter(writer);<a name="line.166"></a>
<span class="sourceLineNo">167</span><a name="line.167"></a>
<span class="sourceLineNo">168</span>        dumpToXML(codonPref, xw, true);<a name="line.168"></a>
<span class="sourceLineNo">169</span><a name="line.169"></a>
<span class="sourceLineNo">170</span>        writer.flush();<a name="line.170"></a>
<span class="sourceLineNo">171</span>    }<a name="line.171"></a>
<span class="sourceLineNo">172</span><a name="line.172"></a>
<span class="sourceLineNo">173</span>    /**<a name="line.173"></a>
<span class="sourceLineNo">174</span>     * reads a specified CodonPref from an file.<a name="line.174"></a>
<span class="sourceLineNo">175</span>     * @param name name of organism<a name="line.175"></a>
<span class="sourceLineNo">176</span>     */<a name="line.176"></a>
<span class="sourceLineNo">177</span>    public static CodonPref readFromXML(InputStream prefStream, String name)<a name="line.177"></a>
<span class="sourceLineNo">178</span>        throws BioException<a name="line.178"></a>
<span class="sourceLineNo">179</span>    {<a name="line.179"></a>
<span class="sourceLineNo">180</span>        CodonPrefFilter.ByName filter = new CodonPrefFilter.ByName(name);<a name="line.180"></a>
<span class="sourceLineNo">181</span><a name="line.181"></a>
<span class="sourceLineNo">182</span>        readFromXML(prefStream, filter);<a name="line.182"></a>
<span class="sourceLineNo">183</span><a name="line.183"></a>
<span class="sourceLineNo">184</span>        return filter.getCodonPref();<a name="line.184"></a>
<span class="sourceLineNo">185</span>    }<a name="line.185"></a>
<span class="sourceLineNo">186</span><a name="line.186"></a>
<span class="sourceLineNo">187</span>    public static CodonPref[] readFromXML(InputStream prefStream) throws BioException{<a name="line.187"></a>
<span class="sourceLineNo">188</span>      CodonPrefFilter.AcceptAll filter = new CodonPrefFilter.AcceptAll();<a name="line.188"></a>
<span class="sourceLineNo">189</span>      readFromXML(prefStream, filter);<a name="line.189"></a>
<span class="sourceLineNo">190</span><a name="line.190"></a>
<span class="sourceLineNo">191</span>      List l = filter.getCodonPrefs();<a name="line.191"></a>
<span class="sourceLineNo">192</span>      CodonPref[] cp = new CodonPref[l.size()];<a name="line.192"></a>
<span class="sourceLineNo">193</span>      return (CodonPref[])l.toArray(cp);<a name="line.193"></a>
<span class="sourceLineNo">194</span>    }<a name="line.194"></a>
<span class="sourceLineNo">195</span><a name="line.195"></a>
<span class="sourceLineNo">196</span>    /**<a name="line.196"></a>
<span class="sourceLineNo">197</span>     * read an CodonPref XML stream and handle it with a CodonPrefFilter object.<a name="line.197"></a>
<span class="sourceLineNo">198</span>     */<a name="line.198"></a>
<span class="sourceLineNo">199</span>    public static void readFromXML(InputStream prefStream, CodonPrefFilter filter)<a name="line.199"></a>
<span class="sourceLineNo">200</span>        throws BioException<a name="line.200"></a>
<span class="sourceLineNo">201</span>    {<a name="line.201"></a>
<span class="sourceLineNo">202</span>        try {<a name="line.202"></a>
<span class="sourceLineNo">203</span>            DocumentBuilder parser = DocumentBuilderFactory.newInstance().newDocumentBuilder();<a name="line.203"></a>
<span class="sourceLineNo">204</span>            Document doc = parser.parse(prefStream);<a name="line.204"></a>
<span class="sourceLineNo">205</span><a name="line.205"></a>
<span class="sourceLineNo">206</span>            // get tables for each species<a name="line.206"></a>
<span class="sourceLineNo">207</span>            NodeList children = doc.getDocumentElement().getChildNodes();<a name="line.207"></a>
<span class="sourceLineNo">208</span><a name="line.208"></a>
<span class="sourceLineNo">209</span>            for (int i=0; i&lt;children.getLength(); i++) {<a name="line.209"></a>
<span class="sourceLineNo">210</span>                Node cnode = children.item(i);<a name="line.210"></a>
<span class="sourceLineNo">211</span><a name="line.211"></a>
<span class="sourceLineNo">212</span>                if (!(cnode instanceof Element)) continue;<a name="line.212"></a>
<span class="sourceLineNo">213</span><a name="line.213"></a>
<span class="sourceLineNo">214</span>                Element child = (Element) cnode;<a name="line.214"></a>
<span class="sourceLineNo">215</span><a name="line.215"></a>
<span class="sourceLineNo">216</span>                String name = child.getNodeName();<a name="line.216"></a>
<span class="sourceLineNo">217</span><a name="line.217"></a>
<span class="sourceLineNo">218</span>                // the node must be a CodonPref record<a name="line.218"></a>
<span class="sourceLineNo">219</span>                if (!name.equals("CodonPref")) continue;<a name="line.219"></a>
<span class="sourceLineNo">220</span><a name="line.220"></a>
<span class="sourceLineNo">221</span>                // pick up the id and genetic code<a name="line.221"></a>
<span class="sourceLineNo">222</span>                String codonPrefId = child.getAttribute("id");<a name="line.222"></a>
<span class="sourceLineNo">223</span>                String geneticCodeId = child.getAttribute("geneticCodeId");<a name="line.223"></a>
<span class="sourceLineNo">224</span><a name="line.224"></a>
<span class="sourceLineNo">225</span>                // is this entry one we want?<a name="line.225"></a>
<span class="sourceLineNo">226</span>                if (!filter.isRequired(codonPrefId)) continue;<a name="line.226"></a>
<span class="sourceLineNo">227</span><a name="line.227"></a>
<span class="sourceLineNo">228</span>                // now handle each codon frequency entry<a name="line.228"></a>
<span class="sourceLineNo">229</span>                NodeList freqs = child.getChildNodes();<a name="line.229"></a>
<span class="sourceLineNo">230</span><a name="line.230"></a>
<span class="sourceLineNo">231</span>                // create a Count object for the job<a name="line.231"></a>
<span class="sourceLineNo">232</span>                Count freqCounts = new IndexedCount(RNATools.getCodonAlphabet());<a name="line.232"></a>
<span class="sourceLineNo">233</span><a name="line.233"></a>
<span class="sourceLineNo">234</span>                for (int j=0; j &lt; freqs.getLength(); j++) {<a name="line.234"></a>
<span class="sourceLineNo">235</span>                    // load each entry<a name="line.235"></a>
<span class="sourceLineNo">236</span>                    Node freq = freqs.item(j);<a name="line.236"></a>
<span class="sourceLineNo">237</span><a name="line.237"></a>
<span class="sourceLineNo">238</span>                    if (!(freq instanceof Element)) continue;<a name="line.238"></a>
<span class="sourceLineNo">239</span><a name="line.239"></a>
<span class="sourceLineNo">240</span>                    Element freqElement = (Element) freq;<a name="line.240"></a>
<span class="sourceLineNo">241</span><a name="line.241"></a>
<span class="sourceLineNo">242</span>                    // get attributes<a name="line.242"></a>
<span class="sourceLineNo">243</span>                    String codonString = freqElement.getAttribute("codon");<a name="line.243"></a>
<span class="sourceLineNo">244</span>                    String freqString = freqElement.getAttribute("value");<a name="line.244"></a>
<span class="sourceLineNo">245</span><a name="line.245"></a>
<span class="sourceLineNo">246</span>                    // create codon<a name="line.246"></a>
<span class="sourceLineNo">247</span>                    SymbolList codonSL = RNATools.createRNA(codonString);<a name="line.247"></a>
<span class="sourceLineNo">248</span><a name="line.248"></a>
<span class="sourceLineNo">249</span>                    if (codonSL.length() !=3) throw new BioException("'" + codonString + "' is not a valid codon!");<a name="line.249"></a>
<span class="sourceLineNo">250</span><a name="line.250"></a>
<span class="sourceLineNo">251</span>                    AtomicSymbol codon = (AtomicSymbol) RNATools.getCodonAlphabet().getSymbol(codonSL.toList());<a name="line.251"></a>
<span class="sourceLineNo">252</span><a name="line.252"></a>
<span class="sourceLineNo">253</span>                    // recover frequency value too<a name="line.253"></a>
<span class="sourceLineNo">254</span>                    double freqValue = Double.parseDouble(freqString);<a name="line.254"></a>
<span class="sourceLineNo">255</span>                    freqCounts.increaseCount(codon, freqValue);<a name="line.255"></a>
<span class="sourceLineNo">256</span><a name="line.256"></a>
<span class="sourceLineNo">257</span>                }<a name="line.257"></a>
<span class="sourceLineNo">258</span><a name="line.258"></a>
<span class="sourceLineNo">259</span>                // turn the Counts into a Distribution<a name="line.259"></a>
<span class="sourceLineNo">260</span>                Distribution freqDistribution = DistributionTools.countToDistribution(freqCounts);<a name="line.260"></a>
<span class="sourceLineNo">261</span><a name="line.261"></a>
<span class="sourceLineNo">262</span>                // create a CodonPref object<a name="line.262"></a>
<span class="sourceLineNo">263</span>                CodonPref newCodonPref = new SimpleCodonPref(geneticCodeId, freqDistribution, codonPrefId);<a name="line.263"></a>
<span class="sourceLineNo">264</span><a name="line.264"></a>
<span class="sourceLineNo">265</span>                filter.put(newCodonPref);<a name="line.265"></a>
<span class="sourceLineNo">266</span>            }<a name="line.266"></a>
<span class="sourceLineNo">267</span>        }<a name="line.267"></a>
<span class="sourceLineNo">268</span>        catch (Exception e) {<a name="line.268"></a>
<span class="sourceLineNo">269</span>            throw new BioException(e);<a name="line.269"></a>
<span class="sourceLineNo">270</span>        }<a name="line.270"></a>
<span class="sourceLineNo">271</span>    }<a name="line.271"></a>
<span class="sourceLineNo">272</span><a name="line.272"></a>
<span class="sourceLineNo">273</span>    /**<a name="line.273"></a>
<span class="sourceLineNo">274</span>     * reads in a file in Codon Usage Database format and<a name="line.274"></a>
<span class="sourceLineNo">275</span>     * translate it into our XML format<a name="line.275"></a>
<span class="sourceLineNo">276</span>     * These can be obtained from the<a name="line.276"></a>
<span class="sourceLineNo">277</span>     * &lt;a href="http://www.kazusa.or.jp/codon/"&gt;Codon Usage Database&lt;/a&gt;.<a name="line.277"></a>
<span class="sourceLineNo">278</span>     * &lt;p&gt;<a name="line.278"></a>
<span class="sourceLineNo">279</span>     * Note that the output assumes that the universal genetic code is<a name="line.279"></a>
<span class="sourceLineNo">280</span>     * used as that is not encoded in the CUD files.  Edit the output appropriately<a name="line.280"></a>
<span class="sourceLineNo">281</span>     * to modify the genetic code if necessary.<a name="line.281"></a>
<span class="sourceLineNo">282</span>     */<a name="line.282"></a>
<span class="sourceLineNo">283</span>    public static void translateCUD(InputStream input, OutputStream output)<a name="line.283"></a>
<span class="sourceLineNo">284</span>        throws IOException<a name="line.284"></a>
<span class="sourceLineNo">285</span>    {<a name="line.285"></a>
<span class="sourceLineNo">286</span>        // create a BufferedReader for the job<a name="line.286"></a>
<span class="sourceLineNo">287</span>        BufferedReader rdr = new BufferedReader(new InputStreamReader(input));<a name="line.287"></a>
<span class="sourceLineNo">288</span><a name="line.288"></a>
<span class="sourceLineNo">289</span>        // create a PrintWriter for the job<a name="line.289"></a>
<span class="sourceLineNo">290</span>        PrintWriter pw = new PrintWriter(output);<a name="line.290"></a>
<span class="sourceLineNo">291</span>        CodonPrefFilter.EverythingToXML filter = new CodonPrefFilter.EverythingToXML(pw);<a name="line.291"></a>
<span class="sourceLineNo">292</span><a name="line.292"></a>
<span class="sourceLineNo">293</span>        // now invoke the CUD reader and stream its output to the XML writer<a name="line.293"></a>
<span class="sourceLineNo">294</span>        readFromCUD(rdr, filter);<a name="line.294"></a>
<span class="sourceLineNo">295</span><a name="line.295"></a>
<span class="sourceLineNo">296</span>        filter.close();<a name="line.296"></a>
<span class="sourceLineNo">297</span>    }<a name="line.297"></a>
<span class="sourceLineNo">298</span><a name="line.298"></a>
<span class="sourceLineNo">299</span><a name="line.299"></a>
<span class="sourceLineNo">300</span>    /**<a name="line.300"></a>
<span class="sourceLineNo">301</span>     * converts a String representation of a codon to its Symbol<a name="line.301"></a>
<span class="sourceLineNo">302</span>     */<a name="line.302"></a>
<span class="sourceLineNo">303</span>    private static AtomicSymbol getCodon(String codonString)<a name="line.303"></a>
<span class="sourceLineNo">304</span>        throws IllegalSymbolException<a name="line.304"></a>
<span class="sourceLineNo">305</span>    {<a name="line.305"></a>
<span class="sourceLineNo">306</span>        return (AtomicSymbol) RNATools.getCodonAlphabet().getSymbol(RNATools.createRNA(codonString).toList());<a name="line.306"></a>
<span class="sourceLineNo">307</span>    }<a name="line.307"></a>
<span class="sourceLineNo">308</span><a name="line.308"></a>
<span class="sourceLineNo">309</span>    private static void loadCodonOrder()<a name="line.309"></a>
<span class="sourceLineNo">310</span>        throws IllegalSymbolException<a name="line.310"></a>
<span class="sourceLineNo">311</span>    {<a name="line.311"></a>
<span class="sourceLineNo">312</span>        cutg[0] = getCodon("cga");<a name="line.312"></a>
<span class="sourceLineNo">313</span>        cutg[1] = getCodon("cgc");<a name="line.313"></a>
<span class="sourceLineNo">314</span>        cutg[2] = getCodon("cgg");<a name="line.314"></a>
<span class="sourceLineNo">315</span>        cutg[3] = getCodon("cgu");<a name="line.315"></a>
<span class="sourceLineNo">316</span><a name="line.316"></a>
<span class="sourceLineNo">317</span>        cutg[4] = getCodon("aga");<a name="line.317"></a>
<span class="sourceLineNo">318</span>        cutg[5] = getCodon("agg");<a name="line.318"></a>
<span class="sourceLineNo">319</span><a name="line.319"></a>
<span class="sourceLineNo">320</span>        cutg[6] = getCodon("cua");<a name="line.320"></a>
<span class="sourceLineNo">321</span>        cutg[7] = getCodon("cuc");<a name="line.321"></a>
<span class="sourceLineNo">322</span>        cutg[8] = getCodon("cug");<a name="line.322"></a>
<span class="sourceLineNo">323</span>        cutg[9] = getCodon("cuu");<a name="line.323"></a>
<span class="sourceLineNo">324</span><a name="line.324"></a>
<span class="sourceLineNo">325</span>        cutg[10] = getCodon("uua");<a name="line.325"></a>
<span class="sourceLineNo">326</span>        cutg[11] = getCodon("uug");<a name="line.326"></a>
<span class="sourceLineNo">327</span><a name="line.327"></a>
<span class="sourceLineNo">328</span>        cutg[12] = getCodon("uca");<a name="line.328"></a>
<span class="sourceLineNo">329</span>        cutg[13] = getCodon("ucc");<a name="line.329"></a>
<span class="sourceLineNo">330</span>        cutg[14] = getCodon("ucg");<a name="line.330"></a>
<span class="sourceLineNo">331</span>        cutg[15] = getCodon("ucu");<a name="line.331"></a>
<span class="sourceLineNo">332</span><a name="line.332"></a>
<span class="sourceLineNo">333</span>        cutg[16] = getCodon("agc");<a name="line.333"></a>
<span class="sourceLineNo">334</span>        cutg[17] = getCodon("agu");<a name="line.334"></a>
<span class="sourceLineNo">335</span><a name="line.335"></a>
<span class="sourceLineNo">336</span>        cutg[18] = getCodon("aca");<a name="line.336"></a>
<span class="sourceLineNo">337</span>        cutg[19] = getCodon("acc");<a name="line.337"></a>
<span class="sourceLineNo">338</span>        cutg[20] = getCodon("acg");<a name="line.338"></a>
<span class="sourceLineNo">339</span>        cutg[21] = getCodon("acu");<a name="line.339"></a>
<span class="sourceLineNo">340</span><a name="line.340"></a>
<span class="sourceLineNo">341</span>        cutg[22] = getCodon("cca");<a name="line.341"></a>
<span class="sourceLineNo">342</span>        cutg[23] = getCodon("ccc");<a name="line.342"></a>
<span class="sourceLineNo">343</span>        cutg[24] = getCodon("ccg");<a name="line.343"></a>
<span class="sourceLineNo">344</span>        cutg[25] = getCodon("ccu");<a name="line.344"></a>
<span class="sourceLineNo">345</span><a name="line.345"></a>
<span class="sourceLineNo">346</span>        cutg[26] = getCodon("gca");<a name="line.346"></a>
<span class="sourceLineNo">347</span>        cutg[27] = getCodon("gcc");<a name="line.347"></a>
<span class="sourceLineNo">348</span>        cutg[28] = getCodon("gcg");<a name="line.348"></a>
<span class="sourceLineNo">349</span>        cutg[29] = getCodon("gcu");<a name="line.349"></a>
<span class="sourceLineNo">350</span><a name="line.350"></a>
<span class="sourceLineNo">351</span>        cutg[30] = getCodon("gga");<a name="line.351"></a>
<span class="sourceLineNo">352</span>        cutg[31] = getCodon("ggc");<a name="line.352"></a>
<span class="sourceLineNo">353</span>        cutg[32] = getCodon("ggg");<a name="line.353"></a>
<span class="sourceLineNo">354</span>        cutg[33] = getCodon("ggu");<a name="line.354"></a>
<span class="sourceLineNo">355</span><a name="line.355"></a>
<span class="sourceLineNo">356</span>        cutg[34] = getCodon("gua");<a name="line.356"></a>
<span class="sourceLineNo">357</span>        cutg[35] = getCodon("guc");<a name="line.357"></a>
<span class="sourceLineNo">358</span>        cutg[36] = getCodon("gug");<a name="line.358"></a>
<span class="sourceLineNo">359</span>        cutg[37] = getCodon("guu");<a name="line.359"></a>
<span class="sourceLineNo">360</span><a name="line.360"></a>
<span class="sourceLineNo">361</span>        cutg[38] = getCodon("aaa");<a name="line.361"></a>
<span class="sourceLineNo">362</span>        cutg[39] = getCodon("aag");<a name="line.362"></a>
<span class="sourceLineNo">363</span><a name="line.363"></a>
<span class="sourceLineNo">364</span>        cutg[40] = getCodon("aac");<a name="line.364"></a>
<span class="sourceLineNo">365</span>        cutg[41] = getCodon("aau");<a name="line.365"></a>
<span class="sourceLineNo">366</span><a name="line.366"></a>
<span class="sourceLineNo">367</span>        cutg[42] = getCodon("caa");<a name="line.367"></a>
<span class="sourceLineNo">368</span>        cutg[43] = getCodon("cag");<a name="line.368"></a>
<span class="sourceLineNo">369</span><a name="line.369"></a>
<span class="sourceLineNo">370</span>        cutg[44] = getCodon("cac");<a name="line.370"></a>
<span class="sourceLineNo">371</span>        cutg[45] = getCodon("cau");<a name="line.371"></a>
<span class="sourceLineNo">372</span><a name="line.372"></a>
<span class="sourceLineNo">373</span>        cutg[46] = getCodon("gaa");<a name="line.373"></a>
<span class="sourceLineNo">374</span>        cutg[47] = getCodon("gag");<a name="line.374"></a>
<span class="sourceLineNo">375</span><a name="line.375"></a>
<span class="sourceLineNo">376</span>        cutg[48] = getCodon("gac");<a name="line.376"></a>
<span class="sourceLineNo">377</span>        cutg[49] = getCodon("gau");<a name="line.377"></a>
<span class="sourceLineNo">378</span><a name="line.378"></a>
<span class="sourceLineNo">379</span>        cutg[50] = getCodon("uac");<a name="line.379"></a>
<span class="sourceLineNo">380</span>        cutg[51] = getCodon("uau");<a name="line.380"></a>
<span class="sourceLineNo">381</span><a name="line.381"></a>
<span class="sourceLineNo">382</span>        cutg[52] = getCodon("ugc");<a name="line.382"></a>
<span class="sourceLineNo">383</span>        cutg[53] = getCodon("ugu");<a name="line.383"></a>
<span class="sourceLineNo">384</span><a name="line.384"></a>
<span class="sourceLineNo">385</span>        cutg[54] = getCodon("uuc");<a name="line.385"></a>
<span class="sourceLineNo">386</span>        cutg[55] = getCodon("uuu");<a name="line.386"></a>
<span class="sourceLineNo">387</span><a name="line.387"></a>
<span class="sourceLineNo">388</span>        cutg[56] = getCodon("aua");<a name="line.388"></a>
<span class="sourceLineNo">389</span>        cutg[57] = getCodon("auc");<a name="line.389"></a>
<span class="sourceLineNo">390</span>        cutg[58] = getCodon("auu");<a name="line.390"></a>
<span class="sourceLineNo">391</span><a name="line.391"></a>
<span class="sourceLineNo">392</span>        cutg[59] = getCodon("aug");<a name="line.392"></a>
<span class="sourceLineNo">393</span><a name="line.393"></a>
<span class="sourceLineNo">394</span>        cutg[60] = getCodon("ugg");<a name="line.394"></a>
<span class="sourceLineNo">395</span><a name="line.395"></a>
<span class="sourceLineNo">396</span>        cutg[61] = getCodon("uaa");<a name="line.396"></a>
<span class="sourceLineNo">397</span>        cutg[62] = getCodon("uag");<a name="line.397"></a>
<span class="sourceLineNo">398</span>        cutg[63] = getCodon("uga");<a name="line.398"></a>
<span class="sourceLineNo">399</span>    }<a name="line.399"></a>
<span class="sourceLineNo">400</span><a name="line.400"></a>
<span class="sourceLineNo">401</span>    private static String stringifyCodon(BasisSymbol codon)<a name="line.401"></a>
<span class="sourceLineNo">402</span>        throws IllegalSymbolException, BioException<a name="line.402"></a>
<span class="sourceLineNo">403</span>    {<a name="line.403"></a>
<span class="sourceLineNo">404</span>        // get the component symbols<a name="line.404"></a>
<span class="sourceLineNo">405</span>        List codonList = codon.getSymbols();<a name="line.405"></a>
<span class="sourceLineNo">406</span><a name="line.406"></a>
<span class="sourceLineNo">407</span>        // get a tokenizer<a name="line.407"></a>
<span class="sourceLineNo">408</span>        SymbolTokenization toke = RNATools.getRNA().getTokenization("token");<a name="line.408"></a>
<span class="sourceLineNo">409</span><a name="line.409"></a>
<span class="sourceLineNo">410</span>        String tokenizedCodon = toke.tokenizeSymbol((Symbol) codonList.get(0))<a name="line.410"></a>
<span class="sourceLineNo">411</span>            + toke.tokenizeSymbol((Symbol) codonList.get(1))<a name="line.411"></a>
<span class="sourceLineNo">412</span>            + toke.tokenizeSymbol((Symbol) codonList.get(2));<a name="line.412"></a>
<span class="sourceLineNo">413</span><a name="line.413"></a>
<span class="sourceLineNo">414</span>        return tokenizedCodon;<a name="line.414"></a>
<span class="sourceLineNo">415</span>    }<a name="line.415"></a>
<span class="sourceLineNo">416</span><a name="line.416"></a>
<span class="sourceLineNo">417</span>    /**<a name="line.417"></a>
<span class="sourceLineNo">418</span>     * writes out a CodonPref object in XML form<a name="line.418"></a>
<span class="sourceLineNo">419</span>     */<a name="line.419"></a>
<span class="sourceLineNo">420</span>    static void dumpToXML(CodonPref codonPref, XMLWriter xw, boolean writeWrapper)<a name="line.420"></a>
<span class="sourceLineNo">421</span>        throws NullPointerException, IOException, IllegalSymbolException, BioException<a name="line.421"></a>
<span class="sourceLineNo">422</span>    {<a name="line.422"></a>
<span class="sourceLineNo">423</span>        // validate both objects first<a name="line.423"></a>
<span class="sourceLineNo">424</span>        if ((codonPref == null) || (xw == null))<a name="line.424"></a>
<span class="sourceLineNo">425</span>            throw new NullPointerException();<a name="line.425"></a>
<span class="sourceLineNo">426</span><a name="line.426"></a>
<span class="sourceLineNo">427</span>        // get the CodonPref Distribution<a name="line.427"></a>
<span class="sourceLineNo">428</span>        Distribution codonDist = codonPref.getFrequency();<a name="line.428"></a>
<span class="sourceLineNo">429</span><a name="line.429"></a>
<span class="sourceLineNo">430</span>        // start &lt;CodonPrefs&gt;<a name="line.430"></a>
<span class="sourceLineNo">431</span>        if (writeWrapper) xw.openTag("CodonPrefs");<a name="line.431"></a>
<span class="sourceLineNo">432</span><a name="line.432"></a>
<span class="sourceLineNo">433</span>        xw.openTag("CodonPref");<a name="line.433"></a>
<span class="sourceLineNo">434</span>        xw.attribute("id", codonPref.getName());<a name="line.434"></a>
<span class="sourceLineNo">435</span>        xw.attribute("geneticCodeId", codonPref.getGeneticCodeName());<a name="line.435"></a>
<span class="sourceLineNo">436</span><a name="line.436"></a>
<span class="sourceLineNo">437</span>        // loop over all codons, writing out the stats<a name="line.437"></a>
<span class="sourceLineNo">438</span>        for (Iterator codonI = RNATools.getCodonAlphabet().iterator(); codonI.hasNext(); ) {<a name="line.438"></a>
<span class="sourceLineNo">439</span>            BasisSymbol codon = (BasisSymbol) codonI.next();<a name="line.439"></a>
<span class="sourceLineNo">440</span><a name="line.440"></a>
<span class="sourceLineNo">441</span>            xw.openTag("frequency");<a name="line.441"></a>
<span class="sourceLineNo">442</span><a name="line.442"></a>
<span class="sourceLineNo">443</span>            // convert codon to a three letter string<a name="line.443"></a>
<span class="sourceLineNo">444</span>            xw.attribute("codon", stringifyCodon(codon));<a name="line.444"></a>
<span class="sourceLineNo">445</span>            xw.attribute("value", Double.toString(codonDist.getWeight(codon)));<a name="line.445"></a>
<span class="sourceLineNo">446</span><a name="line.446"></a>
<span class="sourceLineNo">447</span>            xw.closeTag("frequency");<a name="line.447"></a>
<span class="sourceLineNo">448</span>        }<a name="line.448"></a>
<span class="sourceLineNo">449</span><a name="line.449"></a>
<span class="sourceLineNo">450</span>        xw.closeTag("CodonPref");<a name="line.450"></a>
<span class="sourceLineNo">451</span><a name="line.451"></a>
<span class="sourceLineNo">452</span>        if (writeWrapper) xw.closeTag("CodonPrefs");<a name="line.452"></a>
<span class="sourceLineNo">453</span>    }<a name="line.453"></a>
<span class="sourceLineNo">454</span><a name="line.454"></a>
<span class="sourceLineNo">455</span>    /**<a name="line.455"></a>
<span class="sourceLineNo">456</span>     * reads in records in CUD format<a name="line.456"></a>
<span class="sourceLineNo">457</span>     */<a name="line.457"></a>
<span class="sourceLineNo">458</span>    private static void readFromCUD(BufferedReader rdr, CodonPrefFilter filter)<a name="line.458"></a>
<span class="sourceLineNo">459</span>    {<a name="line.459"></a>
<span class="sourceLineNo">460</span>        try {<a name="line.460"></a>
<span class="sourceLineNo">461</span>            String currLine;<a name="line.461"></a>
<span class="sourceLineNo">462</span>            while ((currLine = rdr.readLine()) != null) {<a name="line.462"></a>
<span class="sourceLineNo">463</span><a name="line.463"></a>
<span class="sourceLineNo">464</span>                // process comment line<a name="line.464"></a>
<span class="sourceLineNo">465</span>                StringTokenizer toke = new StringTokenizer(currLine, ":");<a name="line.465"></a>
<span class="sourceLineNo">466</span>                if (toke.hasMoreTokens()) {<a name="line.466"></a>
<span class="sourceLineNo">467</span>                    // get id string<a name="line.467"></a>
<span class="sourceLineNo">468</span>                    String id = (toke.nextToken()).trim();<a name="line.468"></a>
<span class="sourceLineNo">469</span><a name="line.469"></a>
<span class="sourceLineNo">470</span>                    // read the codon count<a name="line.470"></a>
<span class="sourceLineNo">471</span>                    currLine = rdr.readLine();<a name="line.471"></a>
<span class="sourceLineNo">472</span>                    if (currLine == null) break;<a name="line.472"></a>
<span class="sourceLineNo">473</span><a name="line.473"></a>
<span class="sourceLineNo">474</span>                    // do we even want to process this record?<a name="line.474"></a>
<span class="sourceLineNo">475</span>                    if (filter.isRequired(id)) {<a name="line.475"></a>
<span class="sourceLineNo">476</span>                        toke = new StringTokenizer(currLine);<a name="line.476"></a>
<span class="sourceLineNo">477</span><a name="line.477"></a>
<span class="sourceLineNo">478</span>                        int idx = 0;<a name="line.478"></a>
<span class="sourceLineNo">479</span>                        IndexedCount count = new IndexedCount(RNATools.getCodonAlphabet());<a name="line.479"></a>
<span class="sourceLineNo">480</span>                        while (toke.hasMoreTokens()) {<a name="line.480"></a>
<span class="sourceLineNo">481</span>                            // check that I haven't read too many values!<a name="line.481"></a>
<span class="sourceLineNo">482</span>                            if (idx &gt; 63) continue;<a name="line.482"></a>
<span class="sourceLineNo">483</span>                            count.increaseCount(cutg[idx], Double.parseDouble(toke.nextToken()));<a name="line.483"></a>
<span class="sourceLineNo">484</span>                            idx++;<a name="line.484"></a>
<span class="sourceLineNo">485</span>                        }<a name="line.485"></a>
<span class="sourceLineNo">486</span><a name="line.486"></a>
<span class="sourceLineNo">487</span>                        if (idx != 64) continue;<a name="line.487"></a>
<span class="sourceLineNo">488</span><a name="line.488"></a>
<span class="sourceLineNo">489</span>                        // ok, I now have the counts and the name, let's stash it<a name="line.489"></a>
<span class="sourceLineNo">490</span>                        Distribution codonDist = DistributionTools.countToDistribution(count);<a name="line.490"></a>
<span class="sourceLineNo">491</span><a name="line.491"></a>
<span class="sourceLineNo">492</span>                        CodonPref codonPref = new SimpleCodonPref("UNIVERSAL", codonDist, id);<a name="line.492"></a>
<span class="sourceLineNo">493</span>                        filter.put(codonPref);<a name="line.493"></a>
<span class="sourceLineNo">494</span>                    }<a name="line.494"></a>
<span class="sourceLineNo">495</span>                }<a name="line.495"></a>
<span class="sourceLineNo">496</span>            }<a name="line.496"></a>
<span class="sourceLineNo">497</span>        }<a name="line.497"></a>
<span class="sourceLineNo">498</span>        catch (IOException ioe) {}<a name="line.498"></a>
<span class="sourceLineNo">499</span>        catch (IllegalSymbolException ise) {}<a name="line.499"></a>
<span class="sourceLineNo">500</span>        catch (IllegalAlphabetException iae) {}<a name="line.500"></a>
<span class="sourceLineNo">501</span>        catch (ChangeVetoException cve) {}<a name="line.501"></a>
<span class="sourceLineNo">502</span>        catch (BioException be) {}<a name="line.502"></a>
<span class="sourceLineNo">503</span>    }<a name="line.503"></a>
<span class="sourceLineNo">504</span>}<a name="line.504"></a>
<span class="sourceLineNo">505</span><a name="line.505"></a>




























































</pre>
</div>
</body>
</html>
