<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Uses of Class org.biojava.nbio.core.exceptions.CompoundNotFoundException (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="use: package: org.biojava.nbio.core.exceptions, class: CompoundNotFoundException">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class org.biojava.nbio.core.exceptions.CompoundNotFoundException" class="title">Uses of Class<br>org.biojava.nbio.core.exceptions.CompoundNotFoundException</h1>
</div>
<div class="caption"><span>Packages that use <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#demo">demo</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.io.template">org.biojava.nbio.core.sequence.io.template</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.storage">org.biojava.nbio.core.sequence.storage</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.views">org.biojava.nbio.core.sequence.views</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.util">org.biojava.nbio.core.util</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></div>
<div class="col-last odd-row-color">
<div class="block">FASTQ and variants sequence format I/O.</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.genome.util">org.biojava.nbio.genome.util</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></div>
<div class="col-last odd-row-color">
<div class="block">Input and Output of Structures</div>
</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="demo">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../../../../demo/package-summary.html">demo</a></h2>
<div class="caption"><span>Methods in <a href="../../../../../../demo/package-summary.html">demo</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">BioJavaAADemo.</span><code><a href="../../../../../../demo/BioJavaAADemo.html#main(java.lang.String%5B%5D)" class="member-name-link">main</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&nbsp;args)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/package-summary.html">org.biojava.nbio.core.sequence</a></h2>
<div class="caption"><span>Constructors in <a href="../../sequence/package-summary.html">org.biojava.nbio.core.sequence</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/BasicSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">BasicSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/BasicSequence.html" title="type parameter in BasicSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</code></div>
<div class="col-last odd-row-color">
<div class="block">String is king and assume DNA</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/ChromosomeSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../sequence/DNASequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">DNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</code></div>
<div class="col-last odd-row-color">
<div class="block">String is king and create a sequence from DNA with default DNA compound set</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/DNASequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">DNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a sequence from a string with user defined compound set</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">ProteinSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a protein from a string</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/ProteinSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">ProteinSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a protein from a string with a user defined set of amino acids</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../sequence/RNASequence.html#%3Cinit%3E(java.lang.String)" class="member-name-link">RNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create a RNA sequence from a String</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/RNASequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">RNASequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.io">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h2>
<div class="caption"><span>Methods in <a href="../../sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ABITrace.</span><code><a href="../../sequence/io/ABITrace.html#getSequence()" class="member-name-link">getSequence</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the original programmatically determined (unedited) sequence as a <code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code>.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../sequence/io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">DNASequenceCreator.</span><code><a href="../../sequence/io/DNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyDNASequenceCreator.</span><code><a href="../../sequence/io/FileProxyDNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</div>
<div class="col-first even-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../sequence/io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyRNASequenceCreator.</span><code><a href="../../sequence/io/FileProxyRNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</div>
<div class="col-first even-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../sequence/io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RNASequenceCreator.</span><code><a href="../../sequence/io/RNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>int[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ABITrace.</span><code><a href="../../sequence/io/ABITrace.html#getTrace(java.lang.String)" class="member-name-link">getTrace</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;base)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns one of the four traces - all of the y-coordinate values,
 each of which correspond to a single x-coordinate relative to the
 position in the array, so that if element 4 in the array is 972, then
 x is 4 and y is 972 for that point.</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../../sequence/io/GenbankReader.html" title="type parameter in GenbankReader">S</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenbankReader.</span><code><a href="../../sequence/io/GenbankReader.html#process()" class="member-name-link">process</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">The parsing is done in this method.<br>
 This method will return all the available Genbank records
 in the File or InputStream, closes the underlying resource,
 and return the results in <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/LinkedHashMap.html" title="class or interface in java.util" class="external-link"><code>LinkedHashMap</code></a>.<br>
 You don't need to call <a href="../../sequence/io/GenbankReader.html#close()"><code>GenbankReader.close()</code></a> after calling this method.</div>
</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="../../sequence/io/GenbankReader.html" title="type parameter in GenbankReader">S</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenbankReader.</span><code><a href="../../sequence/io/GenbankReader.html#process(int)" class="member-name-link">process</a><wbr>(int&nbsp;max)</code></div>
<div class="col-last even-row-color">
<div class="block">This method tries to parse maximum <code>max</code> records from
 the open File or InputStream, and leaves the underlying resource open.<br>

 Subsequent calls to the same method continue parsing the rest of the file.<br>
 This is particularly useful when dealing with very big data files,
 (e.g.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.io.template">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a></h2>
<div class="caption"><span>Methods in <a href="../../sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../sequence/io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceCreatorInterface.</span><code><a href="../../sequence/io/template/SequenceCreatorInterface.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.loader">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h2>
<div class="caption"><span>Methods in <a href="../../sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceFileProxyLoader.</span><code><a href="../../sequence/loader/SequenceFileProxyLoader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StringProxySequenceReader.</span><code><a href="../../sequence/loader/StringProxySequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StringProxySequenceReader.</span><code><a href="../../sequence/loader/StringProxySequenceReader.html#setContents(java.lang.String,java.util.List)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&nbsp;features)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">UniprotProxySequenceReader.</span><code><a href="../../sequence/loader/UniprotProxySequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last odd-row-color">
<div class="block">Once the sequence is retrieved set the contents and make sure everything this is valid
 Some uniprot records contain white space in the sequence.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/loader/GenbankProxySequenceReader.html#%3Cinit%3E(java.lang.String,java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">GenbankProxySequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;genbankDirectoryCache,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accessionID,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../sequence/loader/SequenceFileProxyLoader.html#%3Cinit%3E(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceParserInterface,long,int,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">SequenceFileProxyLoader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../../sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceParserInterface</a>&nbsp;sequenceParser,
 long&nbsp;sequenceStartIndex,
 int&nbsp;sequenceLength,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/loader/StringProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">StringProxySequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../sequence/loader/UniprotProxySequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">UniprotProxySequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">The UniProt id is used to retrieve the UniProt XML which is then parsed as a DOM object
 so we know everything about the protein.</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/loader/UniprotProxySequenceReader.html#%3Cinit%3E(org.w3c.dom.Document,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">UniprotProxySequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/org/w3c/dom/Document.html" title="class or interface in org.w3c.dom" class="external-link">Document</a>&nbsp;document,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">The xml is passed in as a DOM object so we know everything about the protein.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.storage">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a></h2>
<div class="caption"><span>Methods in <a href="../../sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ArrayListSequenceReader.</span><code><a href="../../sequence/storage/ArrayListSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BitSequenceReader.</span><code><a href="../../sequence/storage/BitSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last odd-row-color">
<div class="block">Class is immutable, so this is unsupported</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">JoiningSequenceReader.</span><code><a href="../../sequence/storage/JoiningSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SingleCompoundSequenceReader.</span><code><a href="../../sequence/storage/SingleCompoundSequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last odd-row-color">
<div class="block">Unsupported</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/storage/ArrayListSequenceReader.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">ArrayListSequenceReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.template">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a></h2>
<div class="caption"><span>Methods in <a href="../../sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>protected void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../../sequence/template/AbstractSequence.html#initSequenceStorage(java.lang.String)" class="member-name-link">initSequenceStorage</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SequenceReader.</span><code><a href="../../sequence/template/SequenceReader.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../sequence/template/AbstractSequence.html#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">AbstractSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="../../sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.views">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a></h2>
<div class="caption"><span>Methods in <a href="../../sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RnaSequenceView.</span><code><a href="../../sequence/views/RnaSequenceView.html#setContents(java.lang.String)" class="member-name-link">setContents</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.util">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../util/package-summary.html">org.biojava.nbio.core.util</a></h2>
<div class="caption"><span>Methods in <a href="../../util/package-summary.html">org.biojava.nbio.core.util</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a><wbr>&lt;?&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceTools.</span><code><a href="../../util/SequenceTools.html#getSequenceFromString(java.lang.String)" class="member-name-link">getSequenceFromString</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence)</code></div>
<div class="col-last even-row-color">
<div class="block">Attempts to parse String as a DNA sequence first.<br/>
 If this fails it tries to  parse as a ProteinSequence.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.genome.io.fastq">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></h2>
<div class="caption"><span>Methods in <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastqTools.</span><code><a href="../../../genome/io/fastq/FastqTools.html#createDNASequence(org.biojava.nbio.genome.io.fastq.Fastq)" class="member-name-link">createDNASequence</a><wbr>(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></div>
<div class="col-last even-row-color">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> from the specified FASTQ formatted sequence.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FastqTools.</span><code><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithErrorProbabilities(org.biojava.nbio.genome.io.fastq.Fastq)" class="member-name-link">createDNASequenceWithErrorProbabilities</a><wbr>(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with error probabilities from the specified
 FASTQ formatted sequence.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastqTools.</span><code><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithQualityScores(org.biojava.nbio.genome.io.fastq.Fastq)" class="member-name-link">createDNASequenceWithQualityScores</a><wbr>(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></div>
<div class="col-last even-row-color">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores from the specified
 FASTQ formatted sequence.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FastqTools.</span><code><a href="../../../genome/io/fastq/FastqTools.html#createDNASequenceWithQualityScoresAndErrorProbabilities(org.biojava.nbio.genome.io.fastq.Fastq)" class="member-name-link">createDNASequenceWithQualityScoresAndErrorProbabilities</a><wbr>(<a href="../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create and return a new <a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores and error probabilities from the
 specified FASTQ formatted sequence.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.genome.util">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a></h2>
<div class="caption"><span>Methods in <a href="../../../genome/util/package-summary.html">org.biojava.nbio.genome.util</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteinMappingTools.</span><code><a href="../../../genome/util/ProteinMappingTools.html#convertDNAtoProteinSequence(java.lang.String)" class="member-name-link">convertDNAtoProteinSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;dnaSequence)</code></div>
<div class="col-last even-row-color">
<div class="block">Converts the DNA sequence to protein sequence.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ProteinMappingTools.</span><code><a href="../../../genome/util/ProteinMappingTools.html#convertDNAtoProteinSequence(org.biojava.nbio.core.sequence.DNASequence)" class="member-name-link">convertDNAtoProteinSequence</a><wbr>(<a href="../../sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;dnaSequence)</code></div>
<div class="col-last odd-row-color">
<div class="block">Converts the DNA sequence to protein sequence.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.multiple.util">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h2>
<div class="caption"><span>Methods in <a href="../../../structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../../../structure/align/multiple/util/MultipleAlignmentTools.html#getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getHSDMTree</a><wbr>(<a href="../../../structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
</div>
<div class="col-first odd-row-color"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../../../structure/align/multiple/util/MultipleAlignmentTools.html#getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getKimuraTree</a><wbr>(<a href="../../../structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a><wbr>&lt;<a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../../../structure/align/multiple/util/MultipleAlignmentTools.html#toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toProteinMSA</a><wbr>(<a href="../../../structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last even-row-color">
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.cluster">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h2>
<div class="caption"><span>Methods in <a href="../../../structure/cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Subunit.</span><code><a href="../../../structure/cluster/Subunit.html#getProteinSequence()" class="member-name-link">getProteinSequence</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get the protein sequence of the Subunit.</div>
</div>
<div class="col-first odd-row-color"><code>boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SubunitCluster.</span><code><a href="../../../structure/cluster/SubunitCluster.html#mergeSequence(org.biojava.nbio.structure.cluster.SubunitCluster,org.biojava.nbio.structure.cluster.SubunitClustererParameters)" class="member-name-link">mergeSequence</a><wbr>(<a href="../../../structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster">SubunitCluster</a>&nbsp;other,
 <a href="../../../structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">
<div class="block">Merges the other SubunitCluster into this one if their representatives
 sequences are similar (according to the criteria in params).</div>
</div>
<div class="col-first even-row-color"><code>boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SubunitCluster.</span><code><a href="../../../structure/cluster/SubunitCluster.html#mergeSequence(org.biojava.nbio.structure.cluster.SubunitCluster,org.biojava.nbio.structure.cluster.SubunitClustererParameters,org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,org.biojava.nbio.alignment.template.GapPenalty,org.biojava.nbio.core.alignment.template.SubstitutionMatrix)" class="member-name-link">mergeSequence</a><wbr>(<a href="../../../structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster">SubunitCluster</a>&nbsp;other,
 <a href="../../../structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a>&nbsp;params,
 <a href="../../../alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;alignerType,
 <a href="../../../alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
 <a href="../../alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;subsMatrix)</code></div>
<div class="col-last even-row-color">
<div class="block">Merges the other SubunitCluster into this one if their representatives
 sequences are similar (according to the criteria in params).</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.io">
<h2>Uses of <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a> in <a href="../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h2>
<div class="caption"><span>Methods in <a href="../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a> that throw <a href="../CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../../structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastaAFPChainConverter.</span><code><a href="../../../structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,int)" class="member-name-link">cpFastaToAfpChain</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;first,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;second,
 <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
 int&nbsp;cpSite)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../../../structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FastaAFPChainConverter.</span><code><a href="../../../structure/io/FastaAFPChainConverter.html#fastaStringToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)" class="member-name-link">fastaStringToAfpChain</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence2,
 <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
 <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../../structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastaAFPChainConverter.</span><code><a href="../../../structure/io/FastaAFPChainConverter.html#fastaToAfpChain(java.lang.String,java.lang.String,org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Structure)" class="member-name-link">fastaToAfpChain</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence2,
 <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
 <a href="../../../structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code></div>
<div class="col-last even-row-color">
<div class="block">TODO Write comment</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
