<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Uses of Class org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="use: package: org.biojava.nbio.genome.parsers.genename, class: GeneChromosomePosition">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition" class="title">Uses of Class<br>org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition</h1>
</div>
<div class="caption"><span>Packages that use <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.genome.parsers.genename">org.biojava.nbio.genome.parsers.genename</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.genome.util">org.biojava.nbio.genome.util</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="org.biojava.nbio.genome.parsers.genename">
<h2>Uses of <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a> in <a href="../package-summary.html">org.biojava.nbio.genome.parsers.genename</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.genome.parsers.genename</a> that return types with arguments of type <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GeneChromosomePositionParser.</span><code><a href="../GeneChromosomePositionParser.html#getChromosomeMappings()" class="member-name-link">getChromosomeMappings</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GeneChromosomePositionParser.</span><code><a href="../GeneChromosomePositionParser.html#getChromosomeMappings(java.io.InputStream)" class="member-name-link">getChromosomeMappings</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;inStream)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.genome.parsers.genename</a> with parameters of type <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GeneChromosomePosition.</span><code><a href="../GeneChromosomePosition.html#compareTo(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">compareTo</a><wbr>(<a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;o)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.genome.util">
<h2>Uses of <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a> in <a href="../../../util/package-summary.html">org.biojava.nbio.genome.util</a></h2>
<div class="caption"><span>Methods in <a href="../../../util/package-summary.html">org.biojava.nbio.genome.util</a> with parameters of type <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">ChromosomeMappingTools.</span><code><a href="../../../util/ChromosomeMappingTools.html#formatExonStructure(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">formatExonStructure</a><wbr>(<a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</code></div>
<div class="col-last even-row-color">
<div class="block">Pretty print the details of a GeneChromosomePosition to a String</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ChromosomeMappingTools.</span><code><a href="../../../util/ChromosomeMappingTools.html#getCDSExonRanges(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getCDSExonRanges</a><wbr>(<a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last odd-row-color">
<div class="block">Extracts the exon boundaries in CDS coordinates.</div>
</div>
<div class="col-first even-row-color"><code>static int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ChromosomeMappingTools.</span><code><a href="../../../util/ChromosomeMappingTools.html#getCDSLength(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getCDSLength</a><wbr>(<a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last even-row-color">
<div class="block">Get the length of the CDS in nucleotides.</div>
</div>
<div class="col-first odd-row-color"><code>static int</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ChromosomeMappingTools.</span><code><a href="../../../util/ChromosomeMappingTools.html#getCDSPosForChromosomeCoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getCDSPosForChromosomeCoordinate</a><wbr>(int&nbsp;coordinate,
 <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</code></div>
<div class="col-last odd-row-color">
<div class="block">I have a genomic coordinate, where is it on the mRNA</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ChromosomeMappingTools.</span><code><a href="../../../util/ChromosomeMappingTools.html#getChromosomalRangesForCDS(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getChromosomalRangesForCDS</a><wbr>(<a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last even-row-color">
<div class="block">Extracts the boundaries of the coding regions in chromosomal coordinates</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ChromosomeMappingTools.</span><code><a href="../../../util/ChromosomeMappingTools.html#getChromosomePosForCDScoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getChromosomePosForCDScoordinate</a><wbr>(int&nbsp;cdsNucleotidePosition,
 <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last odd-row-color">
<div class="block">Maps the position of a CDS nucleotide back to the genome</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">ChromosomeMappingTools.</span><code><a href="../../../util/ChromosomeMappingTools.html#getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getTranscriptDNASequence</a><wbr>(<a href="../../twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
 <a href="../GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;gcp)</code></div>
<div class="col-last even-row-color">
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
