<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:50 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.nbio.structure.Atom (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava.nbio.structure.Atom (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/class-use/Atom.html" target="_top">Frames</a></li>
<li><a href="Atom.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.structure.Atom" class="title">Uses of Interface<br>org.biojava.nbio.structure.Atom</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.protmod.structure">org.biojava.nbio.protmod.structure</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for protein structure (PDB).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></td>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.client">org.biojava.nbio.structure.align.client</a></td>
<td class="colLast">
<div class="block">This package deals with the server communication for auto-downloading pre-calculated alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.fatcat">org.biojava.nbio.structure.align.fatcat</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the FATCAT alignment algorithm, here called jFATCAT.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.fatcat.calc">org.biojava.nbio.structure.align.fatcat.calc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></td>
<td class="colLast">
<div class="block">Utility methods for better interaction with Jmol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.helper">org.biojava.nbio.structure.align.helper</a></td>
<td class="colLast">
<div class="block">Helper classes for structural alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.model">org.biojava.nbio.structure.align.model</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.pairwise">org.biojava.nbio.structure.align.pairwise</a></td>
<td class="colLast">
<div class="block">Classes for the pairwise alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.seq">org.biojava.nbio.structure.align.seq</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.util">org.biojava.nbio.structure.align.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></td>
<td class="colLast">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.asa">org.biojava.nbio.structure.asa</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.contact">org.biojava.nbio.structure.contact</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.domain">org.biojava.nbio.structure.domain</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.domain.pdp">org.biojava.nbio.structure.domain.pdp</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></td>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io.mmcif">org.biojava.nbio.structure.io.mmcif</a></td>
<td class="colLast">
<div class="block">Input and Output of mmcif files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.core">org.biojava.nbio.structure.symmetry.core</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.protmod.structure">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureUtil.</span><code><strong><a href="../../../../../org/biojava/nbio/protmod/structure/StructureUtil.html#findLinkage(org.biojava.nbio.structure.Group,%20org.biojava.nbio.structure.Group,%20java.lang.String,%20java.lang.String,%20double)">findLinkage</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group1,
           <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group2,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;nameOfAtomOnGroup1,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;nameOfAtomOnGroup2,
           double&nbsp;bondLengthTolerance)</code>
<div class="block">Find a linkage between two groups within tolerance of bond length.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureUtil.</span><code><strong><a href="../../../../../org/biojava/nbio/protmod/structure/StructureUtil.html#findNearestAtomLinkage(org.biojava.nbio.structure.Group,%20org.biojava.nbio.structure.Group,%20java.util.List,%20java.util.List,%20boolean,%20double)">findNearestAtomLinkage</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group1,
                      <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group2,
                      <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;potentialNamesOfAtomOnGroup1,
                      <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;potentialNamesOfAtomOnGroup2,
                      boolean&nbsp;ignoreNCLinkage,
                      double&nbsp;bondLengthTolerance)</code>
<div class="block">Find a linkage between two groups within tolerance of bond length,
 from potential atoms.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">StructureUtil.</span><code><strong><a href="../../../../../org/biojava/nbio/protmod/structure/StructureUtil.html#getAtomDistance(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">getAtomDistance</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/protmod/structure/StructureAtom.html" title="class in org.biojava.nbio.protmod.structure">StructureAtom</a></code></td>
<td class="colLast"><span class="strong">StructureUtil.</span><code><strong><a href="../../../../../org/biojava/nbio/protmod/structure/StructureUtil.html#getStructureAtom(org.biojava.nbio.structure.Atom,%20boolean)">getStructureAtom</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
                boolean&nbsp;isParentAminoAcid)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/protmod/structure/StructureAtomLinkage.html" title="class in org.biojava.nbio.protmod.structure">StructureAtomLinkage</a></code></td>
<td class="colLast"><span class="strong">StructureUtil.</span><code><strong><a href="../../../../../org/biojava/nbio/protmod/structure/StructureUtil.html#getStructureAtomLinkage(org.biojava.nbio.structure.Atom,%20boolean,%20org.biojava.nbio.structure.Atom,%20boolean)">getStructureAtomLinkage</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
                       boolean&nbsp;isParentAminoAcid1,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2,
                       boolean&nbsp;isParentAminoAcid2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that implement <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/AtomImpl.html" title="class in org.biojava.nbio.structure">AtomImpl</a></strong></code>
<div class="block">Implementation of an Atom of a PDB file.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> with type parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<td class="colLast"><span class="strong">HetatomImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/HetatomImpl.html#atoms">atoms</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#add(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">add</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">add two atoms ( a + b).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerAtoms(org.biojava.nbio.structure.Atom[])">centerAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code>
<div class="block">Center the atoms at the Centroid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">centerAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code>
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerOfMass(org.biojava.nbio.structure.Atom[])">centerOfMass</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;points)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom[])">cloneAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Provides an equivalent copy of Atoms in a new array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cloneCAArray(org.biojava.nbio.structure.Atom[])">cloneCAArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use the better-named <a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom[])"><code>StructureTools.cloneAtomArray(Atom[])</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#createVirtualCBAtom(org.biojava.nbio.structure.AminoAcid)">createVirtualCBAtom</a></strong>(<a href="../../../../../org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;amino)</code>
<div class="block">creates a virtual C-beta atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#duplicateCA2(org.biojava.nbio.structure.Atom[])">duplicateCA2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Utility method for working with circular permutations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Chain)">getAllAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code>
<div class="block">Returns and array of all atoms of the chain (first model), including
 Hydrogens (if present) and all HETATOMs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Structure)">getAllAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Convert all atoms of the structure (first model) into an Atom array</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Chain,%20boolean)">getAllNonHAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c,
                   boolean&nbsp;hetAtoms)</code>
<div class="block">Returns and array of all non-Hydrogen atoms in the given Chain,
 optionally including HET atoms or not Waters are not included.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Structure,%20boolean)">getAllNonHAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   boolean&nbsp;hetAtoms)</code>
<div class="block">Returns and array of all non-Hydrogen atoms in the given Structure,
 optionally including HET atoms or not.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">HetatomImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/HetatomImpl.html#getAtom(int)">getAtom</a></strong>(int&nbsp;position)</code>
<div class="block">Get at atom by position.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Group.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Group.html#getAtom(int)">getAtom</a></strong>(int&nbsp;position)</code>
<div class="block">Get at atom by position.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">HetatomImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/HetatomImpl.html#getAtom(java.lang.String)">getAtom</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Get an atom given its PDB name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Group.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Group.html#getAtom(java.lang.String)">getAtom</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Get an atom given its PDB name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">BondImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/BondImpl.html#getAtomA()">getAtomA</a></strong>()</code>
<div class="block">Gets atom 'A' of this bond.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Bond.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Bond.html#getAtomA()">getAtomA</a></strong>()</code>
<div class="block">Gets atom 'A' of this bond.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomArray(org.biojava.nbio.structure.Chain,%20java.lang.String[])">getAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;atomNames)</code>
<div class="block">Returns an array of the requested Atoms from the Chain object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomArray(org.biojava.nbio.structure.Structure,%20java.lang.String[])">getAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;atomNames)</code>
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomArrayAllModels(org.biojava.nbio.structure.Structure,%20java.lang.String[])">getAtomArrayAllModels</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;atomNames)</code>
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">BondImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/BondImpl.html#getAtomB()">getAtomB</a></strong>()</code>
<div class="block">Gets atom 'B' of this bond.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Bond.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Bond.html#getAtomB()">getAtomB</a></strong>()</code>
<div class="block">Gets atom 'B' of this bond.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomCAArray(org.biojava.nbio.structure.Chain)">getAtomCAArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code>
<div class="block">Returns an Atom array of the C-alpha atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getAtomCAArray(org.biojava.nbio.structure.Structure)">getAtomCAArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Return an Atom array of the C-alpha atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getBackboneAtomArray(org.biojava.nbio.structure.Structure)">getBackboneAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Return an Atom array of the main chain atoms: CA, C, N, O Any group that
 contains those atoms will be included, be it a standard aminoacid or not</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcidImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcidImpl.html#getC()">getC</a></strong>()</code>
<div class="block">Get C atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcid.html#getC()">getC</a></strong>()</code>
<div class="block">Get C atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcidImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcidImpl.html#getCA()">getCA</a></strong>()</code>
<div class="block">Get CA atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcid.html#getCA()">getCA</a></strong>()</code>
<div class="block">Get CA atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcidImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcidImpl.html#getCB()">getCB</a></strong>()</code>
<div class="block">Get CB atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcid.html#getCB()">getCB</a></strong>()</code>
<div class="block">Get CB atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getCenterVector(org.biojava.nbio.structure.Atom[])">getCenterVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code>
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms
 to the Centroid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getCenterVector(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">getCenterVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code>
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms
 to the Centroid, if the centroid is already known</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getCentroid(org.biojava.nbio.structure.Atom[])">getCentroid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code>
<div class="block">Returns the center  of mass of the set of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcidImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcidImpl.html#getN()">getN</a></strong>()</code>
<div class="block">Get N atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcid.html#getN()">getN</a></strong>()</code>
<div class="block">Get N atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcidImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcidImpl.html#getO()">getO</a></strong>()</code>
<div class="block">Get O atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AminoAcid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AminoAcid.html#getO()">getO</a></strong>()</code>
<div class="block">Get O atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">NucleotideImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/NucleotideImpl.html#getO3Prime()">getO3Prime</a></strong>()</code>
<div class="block">Returns the O3' atom if present, otherwise null</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">NucleotideImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/NucleotideImpl.html#getO5Prime()">getO5Prime</a></strong>()</code>
<div class="block">Returns the O5' atom if present, otherwise null</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">BondImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/BondImpl.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code>
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Bond.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Bond.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code>
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">NucleotideImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/NucleotideImpl.html#getP()">getP</a></strong>()</code>
<div class="block">Returns the P atom if present, otherwise null</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getRepresentativeAtomArray(org.biojava.nbio.structure.Chain)">getRepresentativeAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code>
<div class="block">Gets a representative atom for each group that is part of the chain
 backbone.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getRepresentativeAtomArray(org.biojava.nbio.structure.Structure)">getRepresentativeAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Gets a representative atom for each group that is part of the chain
 backbone.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getTranslation()">getTranslation</a></strong>()</code>
<div class="block">Get the shift vector.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getTranslationVector(javax.vecmath.Matrix4d)">getTranslationVector</a></strong>(javax.vecmath.Matrix4d&nbsp;transform)</code>
<div class="block">Extract the translational vector of a Vecmath transformation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#invert(org.biojava.nbio.structure.Atom)">invert</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AtomIterator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/AtomIterator.html#next()">next</a></strong>()</code>
<div class="block">Return next atom.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#scale(org.biojava.nbio.structure.Atom,%20double)">scale</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     double&nbsp;s)</code>
<div class="block">Multiply elements of a by s</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#scaleAdd(double,%20org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">scaleAdd</a></strong>(double&nbsp;s,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;x,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Perform linear transformation s*X+B, and store the result in b</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#scaleEquals(org.biojava.nbio.structure.Atom,%20double)">scaleEquals</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
           double&nbsp;s)</code>
<div class="block">Multiply elements of a by s (in place)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#subtract(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">subtract</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">subtract two atoms ( a - b).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#unitVector(org.biojava.nbio.structure.Atom)">unitVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">Returns the unit vector of vector a .</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#vectorProduct(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">vectorProduct</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Vector product (cross product).</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that return types with arguments of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<td class="colLast"><span class="strong">HetatomImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/HetatomImpl.html#getAtoms()">getAtoms</a></strong>()</code>
<div class="block">Get list of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<td class="colLast"><span class="strong">Group.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Group.html#getAtoms()">getAtoms</a></strong>()</code>
<div class="block">Get list of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<td class="colLast"><span class="strong">HetatomImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/HetatomImpl.html#iterator()">iterator</a></strong>()</code>
<div class="block">return an AtomIterator.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<td class="colLast"><span class="strong">Group.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Group.html#iterator()">iterator</a></strong>()</code>
<div class="block">get an Atom Iterator.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#add(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">add</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">add two atoms ( a + b).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HetatomImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/HetatomImpl.html#addAtom(org.biojava.nbio.structure.Atom)">addAtom</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code>
<div class="block">Add an atom to this group.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Group.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Group.html#addAtom(org.biojava.nbio.structure.Atom)">addAtom</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code>
<div class="block">Add an atom to this group.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#amount(org.biojava.nbio.structure.Atom)">amount</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">Gets the length of the vector (2-norm)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#angle(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">angle</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Gets the angle between two vectors</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static javax.vecmath.Point3d[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#atomsToPoints(org.biojava.nbio.structure.Atom[])">atomsToPoints</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Convert an array of atoms into an array of vecmath points</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#calcRotationAngleInDegrees(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">calcRotationAngleInDegrees</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centerPt,
                          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;targetPt)</code>
<div class="block">Calculates the angle from centerPt to targetPt in degrees.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerAtoms(org.biojava.nbio.structure.Atom[])">centerAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code>
<div class="block">Center the atoms at the Centroid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">centerAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code>
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">centerAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code>
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerOfMass(org.biojava.nbio.structure.Atom[])">centerOfMass</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;points)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom[])">cloneAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Provides an equivalent copy of Atoms in a new array.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cloneCAArray(org.biojava.nbio.structure.Atom[])">cloneCAArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use the better-named <a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom[])"><code>StructureTools.cloneAtomArray(Atom[])</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#cloneGroups(org.biojava.nbio.structure.Atom[])">cloneGroups</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Clone a set of representative Atoms, but returns the parent groups</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#convertAtomsToSeq(org.biojava.nbio.structure.Atom[])">convertAtomsToSeq</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#duplicateCA2(org.biojava.nbio.structure.Atom[])">duplicateCA2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Utility method for working with circular permutations.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getCenterVector(org.biojava.nbio.structure.Atom[])">getCenterVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code>
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms
 to the Centroid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getCenterVector(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">getCenterVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code>
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms
 to the Centroid, if the centroid is already known</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getCenterVector(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">getCenterVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code>
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms
 to the Centroid, if the centroid is already known</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getCentroid(org.biojava.nbio.structure.Atom[])">getCentroid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code>
<div class="block">Returns the center  of mass of the set of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getDistance(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">getDistance</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">calculate distance between two atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getDistanceFast(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">getDistanceFast</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Will calculate the square of distances between two atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getGroupDistancesWithinShell(org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Atom,%20java.util.Set,%20double,%20boolean,%20boolean)">getGroupDistancesWithinShell</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid,
                            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
                            double&nbsp;radius,
                            boolean&nbsp;includeWater,
                            boolean&nbsp;useAverageDistance)</code>
<div class="block">Finds Groups in <code>structure</code> that contain at least one Atom that is
 within <code>radius</code> Angstroms of <code>centroid</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getGroupsWithinShell(org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Atom,%20java.util.Set,%20double,%20boolean)">getGroupsWithinShell</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
                    <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&lt;<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
                    double&nbsp;distance,
                    boolean&nbsp;includeWater)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">BondImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/BondImpl.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code>
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Bond.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Bond.html#getOther(org.biojava.nbio.structure.Atom)">getOther</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code>
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getRMS(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getRMS</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2)</code>
<div class="block">Calculate the RMS (root mean square) deviation of two sets of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getRMS(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getRMS</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2)</code>
<div class="block">Calculate the RMS (root mean square) deviation of two sets of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getTMScore(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
          int&nbsp;len1,
          int&nbsp;len2)</code>
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getTMScore(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
          int&nbsp;len1,
          int&nbsp;len2)</code>
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getTMScoreAlternate(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getTMScoreAlternate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
                   int&nbsp;len1,
                   int&nbsp;len2)</code>
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getTMScoreAlternate(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getTMScoreAlternate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
                   int&nbsp;len1,
                   int&nbsp;len2)</code>
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static javax.vecmath.Matrix4d</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getTransformation(org.biojava.nbio.structure.jama.Matrix,%20org.biojava.nbio.structure.Atom)">getTransformation</a></strong>(<a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;rot,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;trans)</code>
<div class="block">Convert JAMA rotation and translation to a Vecmath transformation matrix.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getUnalignedGroups(org.biojava.nbio.structure.Atom[])">getUnalignedGroups</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">List of groups from the structure not included in ca (e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#invert(org.biojava.nbio.structure.Atom)">invert</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#rotate(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.jama.Matrix)">rotate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
      <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;matrix)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#rotate(org.biojava.nbio.structure.Atom,%20double[][])">rotate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
      double[][]&nbsp;m)</code>
<div class="block">Rotate a single Atom aroud a rotation matrix.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#rotate(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.jama.Matrix)">rotate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
      <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m)</code>
<div class="block">Rotate an Atom around a Matrix object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#scalarProduct(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">scalarProduct</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Scalar product (dot product).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#scale(org.biojava.nbio.structure.Atom,%20double)">scale</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     double&nbsp;s)</code>
<div class="block">Multiply elements of a by s</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#scaleAdd(double,%20org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">scaleAdd</a></strong>(double&nbsp;s,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;x,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Perform linear transformation s*X+B, and store the result in b</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#scaleEquals(org.biojava.nbio.structure.Atom,%20double)">scaleEquals</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
           double&nbsp;s)</code>
<div class="block">Multiply elements of a by s (in place)</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#shift(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">shift</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Shift an array of atoms at once.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#shift(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">shift</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Shift an array of atoms at once.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#shift(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">shift</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Shift a vector.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#shift(org.biojava.nbio.structure.Group,%20org.biojava.nbio.structure.Atom)">shift</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">Shift a Group with a vector.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#shift(org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Atom)">shift</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">shift a structure with a vector.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#subtract(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">subtract</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">subtract two atoms ( a - b).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#torsionAngle(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">torsionAngle</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;c,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;d)</code>
<div class="block">Calculate the torsion angle, i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#transform(org.biojava.nbio.structure.Atom[],%20javax.vecmath.Matrix4d)">transform</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         javax.vecmath.Matrix4d&nbsp;t)</code>
<div class="block">Transform an array of atoms at once.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#transform(org.biojava.nbio.structure.Atom,%20javax.vecmath.Matrix4d)">transform</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
         javax.vecmath.Matrix4d&nbsp;m)</code>
<div class="block">Transforms an atom object, given a Matrix4d (i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#translate(org.biojava.nbio.structure.Atom,%20javax.vecmath.Vector3d)">translate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
         javax.vecmath.Vector3d&nbsp;v)</code>
<div class="block">Translates an atom object, given a Vector3d (i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#unitVector(org.biojava.nbio.structure.Atom)">unitVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">Returns the unit vector of vector a .</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#vectorProduct(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">vectorProduct</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code>
<div class="block">Vector product (cross product).</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> with type arguments of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">HetatomImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/HetatomImpl.html#setAtoms(java.util.List)">setAtoms</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code>
<div class="block">Set the atoms of this group.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Group.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Group.html#setAtoms(java.util.List)">setAtoms</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code>
<div class="block">Set the atoms of this group.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/AtomPositionMap.html#AtomPositionMap(org.biojava.nbio.structure.Atom[])">AtomPositionMap</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Creates a new AtomPositionMap containing peptide alpha-carbon atoms</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/AtomPositionMap.html#AtomPositionMap(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.AtomPositionMap.GroupMatcher)">AtomPositionMap</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
               <a href="../../../../../org/biojava/nbio/structure/AtomPositionMap.GroupMatcher.html" title="interface in org.biojava.nbio.structure">AtomPositionMap.GroupMatcher</a>&nbsp;matcher)</code>
<div class="block">Creates a new AtomPositionMap containing only atoms matched by <code>matcher</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/BondImpl.html#BondImpl(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom,%20int)">BondImpl</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomA,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomB,
        int&nbsp;bondOrder)</code>
<div class="block">Constructs a new bond from a pair of atoms and the bond order of the bond
 between them.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/BondImpl.html#BondImpl(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom,%20int,%20boolean)">BondImpl</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomA,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomB,
        int&nbsp;bondOrder,
        boolean&nbsp;addSelfToAtoms)</code>
<div class="block">Constructs a new bond from a pair of atoms and the bond order of the bond
 between them.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#SVDSuperimposer(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">SVDSuperimposer</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2)</code>
<div class="block">Create a SVDSuperimposer object and calculate a SVD superimposition of two sets of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#SVDSuperimposer(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">SVDSuperimposer</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2)</code>
<div class="block">Create a SVDSuperimposer object and calculate a SVD superimposition of two sets of atoms.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#getAlignmentAtoms(org.biojava.nbio.structure.Structure)">getAlignmentAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Returns the atoms that are being used for the alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">CallableStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/CallableStructureAlignment.html#getCa1()">getCa1</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>
<div class="block">run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>
<div class="block">run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.StrucAligParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>
<div class="block">calculate the  protein structure superimposition, between two sets of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.StrucAligParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>
<div class="block">calculate the  protein structure superimposition, between two sets of atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CallableStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/CallableStructureAlignment.html#setCa1(org.biojava.nbio.structure.Atom[])">setCa1</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistOptimized(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistOptimized</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">superimposing according to the optimized alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistOptimized(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistOptimized</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">superimposing according to the optimized alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistPDB(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">calculate the total rmsd of the blocks
 output a merged pdb file for both proteins
          protein 1, in chain A
         protein 2 is twisted according to the twists detected, in chain B</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistPDB(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">calculate the total rmsd of the blocks
 output a merged pdb file for both proteins
          protein 1, in chain A
         protein 2 is twisted according to the twists detected, in chain B</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/CallableStructureAlignment.html#CallableStructureAlignment(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20org.biojava.nbio.structure.align.ce.ConfigStrucAligParams)">CallableStructureAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                          <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;algorithmName,
                          <a href="../../../../../org/biojava/nbio/structure/align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce">ConfigStrucAligParams</a>&nbsp;params)</code>
<div class="block">Constructor for all-to-all alignment calculation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/CallableStructureAlignment.html#CallableStructureAlignment(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20org.biojava.nbio.structure.align.ce.ConfigStrucAligParams)">CallableStructureAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                          <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;algorithmName,
                          <a href="../../../../../org/biojava/nbio/structure/align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce">ConfigStrucAligParams</a>&nbsp;params)</code>
<div class="block">Constructor for all-to-all alignment calculation.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.ce">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#getAtomArray(org.biojava.nbio.structure.Atom[],%20java.util.List,%20java.util.List)">getAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatoms,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;nucs)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#getShift()">getShift</a></strong>()</code>
<div class="block">Gets the shift from the last call to
 <a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)"><code>calc_rmsd</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#getShift()">getShift</a></strong>()</code>
<div class="block">Gets the shift from the last call to
 <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)"><code>calc_rmsd</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Aligns ca1 and ca2 using a heuristic to check for CPs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Aligns ca1 and ca2 using a heuristic to check for CPs.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Align ca2 onto ca1.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Align ca2 onto ca1.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20org.biojava.nbio.structure.align.model.AFPChain[])">alignOptimal</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
            <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20org.biojava.nbio.structure.align.model.AFPChain[])">alignOptimal</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
            <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignPermuted(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20int)">alignPermuted</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
             int&nbsp;cp)</code>
<div class="block">Aligns ca1 with ca2 permuted by <i>cp</i> residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignPermuted(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20int)">alignPermuted</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
             int&nbsp;cp)</code>
<div class="block">Aligns ca1 with ca2 permuted by <i>cp</i> residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform)</code>
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform)</code>
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code>
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code>
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)"><code>CECalculator.calc_rmsd(Atom[],Atom[],int,boolean)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)"><code>CECalculator.calc_rmsd(Atom[],Atom[],int,boolean)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">convertAfpChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">convertAfpChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">convertAfpChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">convertAfpChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">filterDuplicateAFPs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code>
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">filterDuplicateAFPs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code>
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECPParameters)">filterDuplicateAFPs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECPParameters)">filterDuplicateAFPs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#getAlignedStructure(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlignedStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#getAlignedStructure(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlignedStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#getAtomArray(org.biojava.nbio.structure.Atom[],%20java.util.List,%20java.util.List)">getAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatoms,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;nucs)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#initSumOfDistances(int,%20int,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">initSumOfDistances</a></strong>(int&nbsp;nse1,
                  int&nbsp;nse2,
                  int&nbsp;winSize,
                  int&nbsp;winSizeComb1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#initSumOfDistances(int,%20int,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">initSumOfDistances</a></strong>(int&nbsp;nse1,
                  int&nbsp;nse2,
                  int&nbsp;winSize,
                  int&nbsp;winSizeComb1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#initSumOfDistances(int,%20int,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">initSumOfDistances</a></strong>(int&nbsp;nse1,
                  int&nbsp;nse2,
                  int&nbsp;winSize,
                  int&nbsp;winSizeComb1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double[][]</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#initSumOfDistances(int,%20int,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">initSumOfDistances</a></strong>(int&nbsp;nse1,
                  int&nbsp;nse2,
                  int&nbsp;winSize,
                  int&nbsp;winSizeComb1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator)">postProcessAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator)">postProcessAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.align.ce.CECPParameters)">postProcessAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.align.ce.CECPParameters)">postProcessAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">prepareGroupsForDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">prepareGroupsForDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#showAlignmentImage(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">showAlignmentImage</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;jmol)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#showAlignmentImage(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">showAlignmentImage</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;jmol)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">traceFragmentMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">traceFragmentMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">traceFragmentMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">traceFragmentMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double[][]</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#updateMatrixWithSequenceConservation(double[][],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CeParameters)">updateMatrixWithSequenceConservation</a></strong>(double[][]&nbsp;max,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code>
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double[][]</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#updateMatrixWithSequenceConservation(double[][],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CeParameters)">updateMatrixWithSequenceConservation</a></strong>(double[][]&nbsp;max,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code>
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double[][]</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#updateMatrixWithSequenceConservation(double[][],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CeParameters)">updateMatrixWithSequenceConservation</a></strong>(double[][]&nbsp;max,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code>
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double[][]</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#updateMatrixWithSequenceConservation(double[][],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CeParameters)">updateMatrixWithSequenceConservation</a></strong>(double[][]&nbsp;max,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code>
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.client">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAFPChainFromServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAFPChainFromServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">getAFPChainFromServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;method,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                     int&nbsp;timeout)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">getAFPChainFromServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;method,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                     int&nbsp;timeout)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#sendAFPChainToServer(java.lang.String,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">sendAFPChainToServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                    <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#sendAFPChainToServer(java.lang.String,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">sendAFPChainToServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                    <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.fatcat">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20boolean)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
     boolean&nbsp;doRigid)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20boolean)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
     boolean&nbsp;doRigid)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignFlexible(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignFlexible</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignFlexible(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignFlexible</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">alignRigid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">alignRigid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignRigid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignRigid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.fatcat.calc">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FatCatAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatAligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean,%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     boolean&nbsp;doRigid,
     <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FatCatAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatAligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean,%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     boolean&nbsp;doRigid,
     <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static double</code></td>
<td class="colLast"><span class="strong">AFPChainer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html#calAfpRmsd(int,%20int[],%20int,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calAfpRmsd</a></strong>(int&nbsp;afpn,
          int[]&nbsp;afpPositions,
          int&nbsp;listStart,
          <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">//return the rmsd of the residues from the segments that form the given AFP list
        //this value can be a measurement (1) for the connectivity of the AFPs</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected static double</code></td>
<td class="colLast"><span class="strong">AFPChainer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html#calAfpRmsd(int,%20int[],%20int,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calAfpRmsd</a></strong>(int&nbsp;afpn,
          int[]&nbsp;afpPositions,
          int&nbsp;listStart,
          <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">//return the rmsd of the residues from the segments that form the given AFP list
        //this value can be a measurement (1) for the connectivity of the AFPs</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html#doChainAfp(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">doChainAfp</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
          <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">// Key function: chain (assembly) the AFPs
        // a AFP (k) is defined as (i, j, k), with i and j are staring points
        // AFP extension (eg.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html#doChainAfp(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">doChainAfp</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
          <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">// Key function: chain (assembly) the AFPs
        // a AFP (k) is defined as (i, j, k), with i and j are staring points
        // AFP extension (eg.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPCalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html#extractAFPChains(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractAFPChains</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
                <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPCalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html#extractAFPChains(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractAFPChains</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
                <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPOptimizer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html#optimizeAln(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">optimizeAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">optimize the alignment by dynamic programming</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPOptimizer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html#optimizeAln(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">optimizeAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">optimize the alignment by dynamic programming</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPPostProcessor.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPPostProcessor.html#postProcess(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">postProcess</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPPostProcessor.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPPostProcessor.html#postProcess(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">postProcess</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPCalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html#sortAfps(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">sortAfps</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPCalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html#sortAfps(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">sortAfps</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/StructureAlignmentOptimizer.html#StructureAlignmentOptimizer(int,%20int,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20int,%20int[][])">StructureAlignmentOptimizer</a></strong>(int&nbsp;b1,
                           int&nbsp;end1,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c1,
                           int&nbsp;b2,
                           int&nbsp;end2,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c2,
                           int&nbsp;iniLen,
                           int[][]&nbsp;iniSet)</code>
<div class="block">optimize the structural alignment by update the equivalent residues
 and then run dynamic programming
 input: len1 the length of structure 1; c1: the structure information of 1
        len2 the length of structure 2; c2: the structure information of 2
        iniLen and iniSet is the length and list of initial equivalent residues</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/StructureAlignmentOptimizer.html#StructureAlignmentOptimizer(int,%20int,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20int,%20int[][])">StructureAlignmentOptimizer</a></strong>(int&nbsp;b1,
                           int&nbsp;end1,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c1,
                           int&nbsp;b2,
                           int&nbsp;end2,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c2,
                           int&nbsp;iniLen,
                           int[][]&nbsp;iniSet)</code>
<div class="block">optimize the structural alignment by update the equivalent residues
 and then run dynamic programming
 input: len1 the length of structure 1; c1: the structure information of 1
        len2 the length of structure 2; c2: the structure information of 2
        iniLen and iniSet is the length and list of initial equivalent residues</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom[],%20java.util.List)">getAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code>
<div class="block">Returns the first atom for each group</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20boolean)">getAtomForAligPos</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 int&nbsp;chainNr,
                 int&nbsp;aligPos,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
                 boolean&nbsp;getPrevious)</code>
<div class="block">return the atom at alignment position aligPos.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">createArtificalStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">createArtificalStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Display an AFPChain alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Display an AFPChain alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Group[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.List,%20java.util.List)">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms2)</code>
<div class="block">Note: ca2, hetatoms2 and nucleotides2 should not be rotated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Group[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.List,%20java.util.List)">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms2)</code>
<div class="block">Note: ca2, hetatoms2 and nucleotides2 should not be rotated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAlignedStructure(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlignedStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Get an artifical Structure containing both chains.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAlignedStructure(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlignedStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Get an artifical Structure containing both chains.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom[],%20java.util.List)">getAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code>
<div class="block">Returns the first atom for each group</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20boolean)">getAtomForAligPos</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 int&nbsp;chainNr,
                 int&nbsp;aligPos,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
                 boolean&nbsp;getPrevious)</code>
<div class="block">return the atom at alignment position aligPos.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getPDBresnum(int,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getPDBresnum</a></strong>(int&nbsp;aligPos,
            <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Return a list of pdb Strings corresponding to the aligned positions of the molecule.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#getPDBresnum(int,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.Atom[])">getPDBresnum</a></strong>(int&nbsp;structNum,
            <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Utility method used in the <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol"><code>MultipleAlignmentJmol</code></a> Frame,
 when the aligned residues of a structure in the alignment have
 to be selected for formatting them (coloring and style).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">prepareGroupsForDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Rotate the Atoms/Groups so they are aligned for the 3D visualisation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">prepareGroupsForDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Rotate the Atoms/Groups so they are aligned for the 3D visualisation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#shiftCA2(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.jama.Matrix,%20org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Group[])">shiftCA2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;shift,
        <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups)</code>
<div class="block">only shift CA positions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#shiftCA2(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.jama.Matrix,%20org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Group[])">shiftCA2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;shift,
        <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups)</code>
<div class="block">only shift CA positions.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#showAlignmentPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showAlignmentPanel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#showAlignmentPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showAlignmentPanel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.aligpanel">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StatusDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html#getCa1()">getCa1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AligPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html#getCa1()">getCa1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StatusDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html#getCa2()">getCa2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AligPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html#getCa2()">getCa2</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StatusDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html#setCa1(org.biojava.nbio.structure.Atom[])">setCa1</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AligPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html#setCa1(org.biojava.nbio.structure.Atom[])">setCa1</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StatusDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html#setCa2(org.biojava.nbio.structure.Atom[])">setCa2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AligPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html#setCa2(org.biojava.nbio.structure.Atom[])">setCa2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html#MultipleAligPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>
<div class="block">Constructor using an afpChain and the atom arrays for pairwise
 alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html#MultipleAligPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>
<div class="block">Constructor using an afpChain and the atom arrays for pairwise
 alignments.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.jmol">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentJmol.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#getJmolScript4Block(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">getJmolScript4Block</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                   int&nbsp;blockNr)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentJmol.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#getJmolScript4Block(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">getJmolScript4Block</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                   int&nbsp;blockNr)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentJmol.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getJmolString</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentJmol.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getJmolString</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">JmolTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/JmolTools.html#getPdbInfo(org.biojava.nbio.structure.Atom)">getPdbInfo</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">get jmol style info:
  jmol style: [MET]508:A.CA/1 #3918
  insertion code: [ASP]1^A:A.CA/1 #2</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">JmolTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/JmolTools.html#getPdbInfo(org.biojava.nbio.structure.Atom,%20boolean)">getPdbInfo</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
          boolean&nbsp;printResName)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractAlignmentJmol.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html#setAtoms(org.biojava.nbio.structure.Atom[])">setAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Create and set a new structure from a given atom array.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#StructureAlignmentJmol(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">StructureAlignmentJmol</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#StructureAlignmentJmol(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">StructureAlignmentJmol</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.helper">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getCenter(org.biojava.nbio.structure.Atom[],%20int,%20int)">getCenter</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         int&nbsp;pos,
         int&nbsp;fragmentLength)</code>
<div class="block">get the centroid for the set of atoms starting fromposition pos, length fragmentLenght</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getFragment(org.biojava.nbio.structure.Atom[],%20int,%20int)">getFragment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
           int&nbsp;pos,
           int&nbsp;fragmentLength)</code>
<div class="block">get a continue subset of Atoms based by the starting position and the length</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getFragmentFromIdxList(org.biojava.nbio.structure.Atom[],%20int[])">getFragmentFromIdxList</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                      int[]&nbsp;idx)</code>
<div class="block">get a subset of Atoms based by their positions</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getFragmentNoClone(org.biojava.nbio.structure.Atom[],%20int,%20int)">getFragmentNoClone</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                  int&nbsp;pos,
                  int&nbsp;fragmentLength)</code>
<div class="block">get a continue subset of Atoms based by the starting position and the length
 does not clone the original atoms.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getCenter(org.biojava.nbio.structure.Atom[],%20int,%20int)">getCenter</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         int&nbsp;pos,
         int&nbsp;fragmentLength)</code>
<div class="block">get the centroid for the set of atoms starting fromposition pos, length fragmentLenght</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double[]</code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getDiagonalAtK(org.biojava.nbio.structure.Atom[],%20int)">getDiagonalAtK</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
              int&nbsp;k)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getDistanceMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getDistanceMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Matrix of all distances between two sets of Atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getDistanceMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getDistanceMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Matrix of all distances between two sets of Atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getFragment(org.biojava.nbio.structure.Atom[],%20int,%20int)">getFragment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
           int&nbsp;pos,
           int&nbsp;fragmentLength)</code>
<div class="block">get a continue subset of Atoms based by the starting position and the length</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getFragmentFromIdxList(org.biojava.nbio.structure.Atom[],%20int[])">getFragmentFromIdxList</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                      int[]&nbsp;idx)</code>
<div class="block">get a subset of Atoms based by their positions</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AlignTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/helper/AlignTools.html#getFragmentNoClone(org.biojava.nbio.structure.Atom[],%20int,%20int)">getFragmentNoClone</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
                  int&nbsp;pos,
                  int&nbsp;fragmentLength)</code>
<div class="block">get a continue subset of Atoms based by the starting position and the length
 does not clone the original atoms.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.model">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/model/package-summary.html">org.biojava.nbio.structure.align.model</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AFPChain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#getBlockShiftVector()">getBlockShiftVector</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AFPChain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setBlockShiftVector(org.biojava.nbio.structure.Atom[])">setBlockShiftVector</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;blockShiftVector)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toAlignedPairs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toAlignedPairs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Prints the alignment in the simplest form: a list of aligned residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toAlignedPairs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toAlignedPairs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Prints the alignment in the simplest form: a list of aligned residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toCE(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toCE</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toCE(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toCE</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#toCE(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toCE</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#toCE(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toCE</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toFatCat(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toFatCat</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toFatCat(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toFatCat</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#toFatcat(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toFatcat</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#toFatcat(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toFatcat</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toFatCatCore(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean,%20boolean,%20boolean,%20boolean)">toFatCatCore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            boolean&nbsp;printLegend,
            boolean&nbsp;longHeader,
            boolean&nbsp;showHTML,
            boolean&nbsp;showAlignmentBlock)</code>
<div class="block">Output in FatCatCore format</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toFatCatCore(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean,%20boolean,%20boolean,%20boolean)">toFatCatCore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            boolean&nbsp;printLegend,
            boolean&nbsp;longHeader,
            boolean&nbsp;showHTML,
            boolean&nbsp;showAlignmentBlock)</code>
<div class="block">Output in FatCatCore format</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toWebSiteDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toWebSiteDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">toWebSiteDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                boolean&nbsp;showAlignmentBlock)</code>
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">toWebSiteDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                boolean&nbsp;showAlignmentBlock)</code>
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#MultipleAlignmentEnsembleImpl(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">MultipleAlignmentEnsembleImpl</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afp,
                             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                             boolean&nbsp;flexible)</code>
<div class="block">Constructor from an AFPChain instance.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#MultipleAlignmentEnsembleImpl(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">MultipleAlignmentEnsembleImpl</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afp,
                             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                             boolean&nbsp;flexible)</code>
<div class="block">Constructor from an AFPChain instance.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20java.util.List,%20int,%20int)">getAtomForSequencePosition</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                          <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;mapSeqToStruct,
                          int&nbsp;str,
                          int&nbsp;sequencePos)</code>
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.pairwise">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#getCenter1()">getCenter1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#getCenter2()">getCenter2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#getShift()">getShift</a></strong>()</code>
<div class="block">returns the shift vector that has to be applied on structure to to shift on structure one</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#getTrans()">getTrans</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#getUnitv()">getUnitv</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#approach_ap3(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.pairwise.FragmentPair[],%20org.biojava.nbio.structure.align.StrucAligParameters)">approach_ap3</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a>[]&nbsp;fraglst,
            <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#approach_ap3(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.pairwise.FragmentPair[],%20org.biojava.nbio.structure.align.StrucAligParameters)">approach_ap3</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a>[]&nbsp;fraglst,
            <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#calcScores(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calcScores</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">calculates  scores for this alignment ( %id )</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#calcScores(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calcScores</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">calculates  scores for this alignment ( %id )</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#calculateSuperpositionByIdx(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calculateSuperpositionByIdx</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#calculateSuperpositionByIdx(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calculateSuperpositionByIdx</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments[],%20org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]&nbsp;fragments,
               <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments[],%20org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]&nbsp;fragments,
               <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments,%20org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;fragments,
               <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments,%20org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;fragments,
               <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#finish(org.biojava.nbio.structure.align.StrucAligParameters,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">finish</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#finish(org.biojava.nbio.structure.align.StrucAligParameters,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">finish</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#getDistanceMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getDistanceMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#getDistanceMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getDistanceMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#getRMS(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments)">getRMS</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
      <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;frag)</code>
<div class="block">get the RMS of the JointFragments pair frag</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#getRMS(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments)">getRMS</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
      <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;frag)</code>
<div class="block">get the RMS of the JointFragments pair frag</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#refine(org.biojava.nbio.structure.align.StrucAligParameters,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Refinement procedure based on superposition and dynamic programming.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#refine(org.biojava.nbio.structure.align.StrucAligParameters,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Refinement procedure based on superposition and dynamic programming.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#setCenter1(org.biojava.nbio.structure.Atom)">setCenter1</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;center1)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#setCenter2(org.biojava.nbio.structure.Atom)">setCenter2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;center2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#setTrans(org.biojava.nbio.structure.Atom)">setTrans</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;trans)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html#setUnitv(org.biojava.nbio.structure.Atom)">setUnitv</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;unitv)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlignmentProgressListener.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlignmentProgressListener.html#startingAlignment(java.lang.String,%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20org.biojava.nbio.structure.Atom[])">startingAlignment</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlignmentProgressListener.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlignmentProgressListener.html#startingAlignment(java.lang.String,%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20org.biojava.nbio.structure.Atom[])">startingAlignment</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.seq">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;parameters)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;parameters)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlignedAtoms1(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getAlignedAtoms1</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlignedAtoms2(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getAlignedAtoms2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getAtoms(java.lang.String)">getAtoms</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Returns the CA atoms for the provided name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getAtoms(org.biojava.nbio.structure.StructureIdentifier)">getAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getOtherTranslation()">getOtherTranslation</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getProjectedPoint(org.biojava.nbio.structure.Atom)">getProjectedPoint</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code>
<div class="block">Projects a given point onto the axis of rotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getRepresentativeAtoms(java.lang.String)">getRepresentativeAtoms</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Returns the representative atoms for the provided name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getRepresentativeAtoms(org.biojava.nbio.structure.StructureIdentifier)">getRepresentativeAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getRotationAxis()">getRotationAxis</a></strong>()</code>
<div class="block">Get a unit vector along the rotation axis</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getRotationPos()">getRotationPos</a></strong>()</code>
<div class="block">Get a position on the rotation axis.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getScrewTranslation()">getScrewTranslation</a></strong>()</code>
<div class="block">Get the component of translation parallel to the axis of rotation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTranslation</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#alignmentToSIF(java.io.Writer,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20java.lang.String)">alignmentToSIF</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/Writer.html?is-external=true" title="class or interface in java.io">Writer</a>&nbsp;out,
              <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;backboneInteraction,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentInteraction)</code>
<div class="block">Creates a simple interaction format (SIF) file for an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#alignmentToSIF(java.io.Writer,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20java.lang.String)">alignmentToSIF</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/Writer.html?is-external=true" title="class or interface in java.io">Writer</a>&nbsp;out,
              <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;backboneInteraction,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentInteraction)</code>
<div class="block">Creates a simple interaction format (SIF) file for an alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#createAFPChain(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.ResidueNumber[],%20org.biojava.nbio.structure.ResidueNumber[])">createAFPChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned1,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned2)</code>
<div class="block">Fundamentally, an alignment is just a list of aligned residues in each
 protein.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#createAFPChain(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.ResidueNumber[],%20org.biojava.nbio.structure.ResidueNumber[])">createAFPChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned1,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned2)</code>
<div class="block">Fundamentally, an alignment is just a list of aligned residues in each
 protein.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">createArtificalStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">createArtificalStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlign</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Extract the alignment output</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlign</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Extract the alignment output</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">getAlign</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        boolean&nbsp;showSeq)</code>
<div class="block">Sets the following properties:
 
 The alignment strings <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnseq1(char[])"><code>alnseq1</code></a>,
  <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnseq2(char[])"><code>alnseq2</code></a>,
  and <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnsymb(char[])"><code>alnsymb</code></a>
 <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnbeg1(int)"><code>alnbeg1</code></a> and 2
 <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnLength(int)"><code>alnLength</code></a> and
  <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setGapLen(int)"><code>gapLen</code></a>
 </div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">getAlign</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        boolean&nbsp;showSeq)</code>
<div class="block">Sets the following properties:
 
 The alignment strings <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnseq1(char[])"><code>alnseq1</code></a>,
  <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnseq2(char[])"><code>alnseq2</code></a>,
  and <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnsymb(char[])"><code>alnsymb</code></a>
 <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnbeg1(int)"><code>alnbeg1</code></a> and 2
 <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnLength(int)"><code>alnLength</code></a> and
  <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setGapLen(int)"><code>gapLen</code></a>
 </div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlignedAtoms1(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getAlignedAtoms1</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlignedAtoms2(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getAlignedAtoms2</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getJmolScript(org.biojava.nbio.structure.Atom[])">getJmolScript</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Returns a Jmol script which will display the axis of rotation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getJmolScript(org.biojava.nbio.structure.Atom[],%20int)">getJmolScript</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
             int&nbsp;axisID)</code>
<div class="block">Returns a Jmol script which will display the axis of rotation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getProjectedDistance(org.biojava.nbio.structure.Atom)">getProjectedDistance</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code>
<div class="block">Get the distance from a point to the axis of rotation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getProjectedPoint(org.biojava.nbio.structure.Atom)">getProjectedPoint</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code>
<div class="block">Projects a given point onto the axis of rotation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getRotMax(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getRotMax</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getRotMax(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getRotMax</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">AFPChainScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">AFPChainScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTranslation</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTranslation</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int[],%20int[][][])">replaceOptAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;blockNum,
             int[]&nbsp;optLens,
             int[][][]&nbsp;optAln)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int[],%20int[][][])">replaceOptAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;blockNum,
             int[]&nbsp;optLens,
             int[][][]&nbsp;optAln)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.Map)">replaceOptAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;alignment)</code>
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.Map)">replaceOptAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;alignment)</code>
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(int[][][],%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">replaceOptAln</a></strong>(int[][][]&nbsp;newAlgn,
             <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(int[][][],%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">replaceOptAln</a></strong>(int[][][]&nbsp;newAlgn,
             <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#rotate(org.biojava.nbio.structure.Atom[],%20double)">rotate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      double&nbsp;theta)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">splitBlocksByTopology</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">splitBlocksByTopology</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#updateSuperposition(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">updateSuperposition</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">After the alignment changes (optAln, optLen, blockNum, at a minimum),
 many other properties which depend on the superposition will be invalid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#updateSuperposition(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">updateSuperposition</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">After the alignment changes (optAln, optLen, blockNum, at a minimum),
 many other properties which depend on the superposition will be invalid.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#RotationAxis(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom,%20double)">RotationAxis</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;axis,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;pos,
            double&nbsp;theta)</code>
<div class="block">Create a rotation axis from a vector, a point, and an angle.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#RotationAxis(org.biojava.nbio.structure.jama.Matrix,%20org.biojava.nbio.structure.Atom)">RotationAxis</a></strong>(<a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;rotation,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;translation)</code>
<div class="block">Determine the location of the rotation axis based on a rotation matrix and a translation vector</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.xml">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">fromXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">fromXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">fromXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">new utility method that checks that the order of the pair in the XML alignment is correct and flips the direction if needed</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">fromXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">new utility method that checks that the order of the pair in the XML alignment is correct and flips the direction if needed</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#printXMLEQRInferPositions(org.biojava.nbio.core.util.PrettyXMLWriter,%20org.biojava.nbio.structure.align.model.AFPChain,%20int,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">printXMLEQRInferPositions</a></strong>(<a href="../../../../../org/biojava/nbio/core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
                         <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                         int&nbsp;bk,
                         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#printXMLEQRInferPositions(org.biojava.nbio.core.util.PrettyXMLWriter,%20org.biojava.nbio.structure.align.model.AFPChain,%20int,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">printXMLEQRInferPositions</a></strong>(<a href="../../../../../org/biojava/nbio/core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
                         <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                         int&nbsp;bk,
                         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#rebuildAFPChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">rebuildAFPChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">replace the PDB res nums with atom positions:</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#rebuildAFPChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">rebuildAFPChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">replace the PDB res nums with atom positions:</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toXML</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Convert an afpChain to a simple XML representation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toXML</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Convert an afpChain to a simple XML representation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,%20java.io.StringWriter,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toXML</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/io/StringWriter.html?is-external=true" title="class or interface in java.io">StringWriter</a>&nbsp;swriter,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Write the XML representation to a StringWriter</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,%20java.io.StringWriter,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toXML</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/io/StringWriter.html?is-external=true" title="class or interface in java.io">StringWriter</a>&nbsp;swriter,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Write the XML representation to a StringWriter</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.asa">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/asa/package-summary.html">org.biojava.nbio.structure.asa</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/asa/package-summary.html">org.biojava.nbio.structure.asa</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">AsaCalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/asa/AsaCalculator.html#getRadius(org.biojava.nbio.structure.Atom)">getRadius</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code>
<div class="block">Gets the van der Waals radius of the given atom following the values defined by
 Chothia (1976) J.Mol.Biol.105,1-14
 NOTE: the vdw values defined by the paper assume no Hydrogens and thus "inflates"
 slightly the heavy atoms to account for Hydrogens.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/asa/package-summary.html">org.biojava.nbio.structure.asa</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/asa/AsaCalculator.html#AsaCalculator(org.biojava.nbio.structure.Atom[],%20double,%20int,%20int)">AsaCalculator</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
             double&nbsp;probe,
             int&nbsp;nSpherePoints,
             int&nbsp;nThreads)</code>
<div class="block">Constructs a new AsaCalculator.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.contact">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureInterface.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/StructureInterface.html#getAtomsForAsa(int)">getAtomsForAsa</a></strong>(int&nbsp;cofactorSizeToUse)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureInterface.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/StructureInterface.html#getFirstAtomsForAsa(int)">getFirstAtomsForAsa</a></strong>(int&nbsp;cofactorSizeToUse)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">StructureInterface.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/StructureInterface.html#getSecondAtomsForAsa(int)">getSecondAtomsForAsa</a></strong>(int&nbsp;cofactorSizeToUse)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a> that return types with arguments of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></td>
<td class="colLast"><span class="strong">AtomContact.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html#getPair()">getPair</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Grid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom[])">addAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Adds a set of atoms, subsequent call to getContacts will produce the interatomic contacts.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Grid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">addAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms)</code>
<div class="block">Adds the i and j atoms and fills the grid.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Grid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">addAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms)</code>
<div class="block">Adds the i and j atoms and fills the grid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Grid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.contact.BoundingBox)">addAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
        <a href="../../../../../org/biojava/nbio/structure/contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;bounds)</code>
<div class="block">Adds a set of atoms, subsequent call to getContacts will produce the interatomic contacts.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Grid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.contact.BoundingBox,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.contact.BoundingBox)">addAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
        <a href="../../../../../org/biojava/nbio/structure/contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;icoordbounds,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
        <a href="../../../../../org/biojava/nbio/structure/contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;jcoordbounds)</code>
<div class="block">Adds the i and j atoms and fills the grid, passing their bounds (array of size 6 with x,y,z minima and x,y,z maxima)
 This way the bounds don't need to be recomputed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">Grid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.contact.BoundingBox,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.contact.BoundingBox)">addAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
        <a href="../../../../../org/biojava/nbio/structure/contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;icoordbounds,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
        <a href="../../../../../org/biojava/nbio/structure/contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;jcoordbounds)</code>
<div class="block">Adds the i and j atoms and fills the grid, passing their bounds (array of size 6 with x,y,z minima and x,y,z maxima)
 This way the bounds don't need to be recomputed.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a></code></td>
<td class="colLast"><span class="strong">AtomContactSet.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/AtomContactSet.html#getContact(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">getContact</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code>
<div class="block">Returns the corresponding AtomContact or null if no contact exists between the 2 given atoms</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a>&gt;</code></td>
<td class="colLast"><span class="strong">GridCell.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/GridCell.html#getContactsToOtherCell(org.biojava.nbio.structure.contact.GridCell,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20double)">getContactsToOtherCell</a></strong>(<a href="../../../../../org/biojava/nbio/structure/contact/GridCell.html" title="class in org.biojava.nbio.structure.contact">GridCell</a>&nbsp;otherCell,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
                      double&nbsp;cutoff)</code>
<div class="block">Calculates all distances of atoms between this cell and the given cell returning those that are
 within the given cutoff as a list of AtomContacts</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a>&gt;</code></td>
<td class="colLast"><span class="strong">GridCell.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/GridCell.html#getContactsToOtherCell(org.biojava.nbio.structure.contact.GridCell,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20double)">getContactsToOtherCell</a></strong>(<a href="../../../../../org/biojava/nbio/structure/contact/GridCell.html" title="class in org.biojava.nbio.structure.contact">GridCell</a>&nbsp;otherCell,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
                      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
                      double&nbsp;cutoff)</code>
<div class="block">Calculates all distances of atoms between this cell and the given cell returning those that are
 within the given cutoff as a list of AtomContacts</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a>&gt;</code></td>
<td class="colLast"><span class="strong">GridCell.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/GridCell.html#getContactsWithinCell(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20double)">getContactsWithinCell</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
                     double&nbsp;cutoff)</code>
<div class="block">Calculates all distances of atoms within this cell returning those that are within the given cutoff
 as a list of AtomContacts</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a>&gt;</code></td>
<td class="colLast"><span class="strong">GridCell.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/GridCell.html#getContactsWithinCell(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20double)">getContactsWithinCell</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
                     double&nbsp;cutoff)</code>
<div class="block">Calculates all distances of atoms within this cell returning those that are within the given cutoff
 as a list of AtomContacts</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">AtomContactSet.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/AtomContactSet.html#hasContact(org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Atom)">hasContact</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a> with type arguments of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AtomContact.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html#setPair(org.biojava.nbio.structure.contact.Pair)">setPair</a></strong>(<a href="../../../../../org/biojava/nbio/structure/contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;pair)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/BoundingBox.html#BoundingBox(org.biojava.nbio.structure.Atom[])">BoundingBox</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Constructs a BoundingBox by calculating maxs and mins of given array of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/StructureInterface.html#StructureInterface(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.contact.AtomContactSet,%20org.biojava.nbio.structure.xtal.CrystalTransform,%20org.biojava.nbio.structure.xtal.CrystalTransform)">StructureInterface</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;firstMolecule,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;secondMolecule,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;firstMoleculeId,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;secondMoleculeId,
                  <a href="../../../../../org/biojava/nbio/structure/contact/AtomContactSet.html" title="class in org.biojava.nbio.structure.contact">AtomContactSet</a>&nbsp;contacts,
                  <a href="../../../../../org/biojava/nbio/structure/xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;firstTransf,
                  <a href="../../../../../org/biojava/nbio/structure/xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;secondTransf)</code>
<div class="block">Constructs a StructureInterface</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/StructureInterface.html#StructureInterface(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.contact.AtomContactSet,%20org.biojava.nbio.structure.xtal.CrystalTransform,%20org.biojava.nbio.structure.xtal.CrystalTransform)">StructureInterface</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;firstMolecule,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;secondMolecule,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;firstMoleculeId,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;secondMoleculeId,
                  <a href="../../../../../org/biojava/nbio/structure/contact/AtomContactSet.html" title="class in org.biojava.nbio.structure.contact">AtomContactSet</a>&nbsp;contacts,
                  <a href="../../../../../org/biojava/nbio/structure/xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;firstTransf,
                  <a href="../../../../../org/biojava/nbio/structure/xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;secondTransf)</code>
<div class="block">Constructs a StructureInterface</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../../../org/biojava/nbio/structure/contact/package-summary.html">org.biojava.nbio.structure.contact</a> with type arguments of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/contact/AtomContact.html#AtomContact(org.biojava.nbio.structure.contact.Pair,%20double)">AtomContact</a></strong>(<a href="../../../../../org/biojava/nbio/structure/contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;pair,
           double&nbsp;distance)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.domain">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/domain/package-summary.html">org.biojava.nbio.structure.domain</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/domain/package-summary.html">org.biojava.nbio.structure.domain</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></td>
<td class="colLast"><span class="strong">LocalProteinDomainParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/LocalProteinDomainParser.html#suggestDomains(org.biojava.nbio.structure.Atom[])">suggestDomains</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Suggest domains for a set of Calpha atoms</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.domain.pdp">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">Cut.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/pdp/Cut.html#cut(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.domain.pdp.Domain,%20org.biojava.nbio.structure.domain.pdp.CutValues,%20int[][],%20org.biojava.nbio.structure.domain.pdp.PDPDistanceMatrix)">cut</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
   <a href="../../../../../org/biojava/nbio/structure/domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&nbsp;dom,
   <a href="../../../../../org/biojava/nbio/structure/domain/pdp/CutValues.html" title="class in org.biojava.nbio.structure.domain.pdp">CutValues</a>&nbsp;val,
   int[][]&nbsp;dist,
   <a href="../../../../../org/biojava/nbio/structure/domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a>&nbsp;pdpMatrix)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a></code></td>
<td class="colLast"><span class="strong">GetDistanceMatrix.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/pdp/GetDistanceMatrix.html#getDistanceMatrix(org.biojava.nbio.structure.Atom[])">getDistanceMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;protein)</code>
<div class="block">A set of Calpha atoms that are representing the protein</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/pdp/CutDomain.html#CutDomain(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.domain.pdp.PDPDistanceMatrix)">CutDomain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
         <a href="../../../../../org/biojava/nbio/structure/domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a>&nbsp;pdpMatrix)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">FileConvert.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom)">toPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code>
<div class="block">Prints the content of an Atom object as a PDB formatted line.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">FileConvert.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,%20java.lang.String)">toPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FileConvert.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,%20java.lang.StringBuffer)">toPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/StringBuffer.html?is-external=true" title="class or interface in java.lang">StringBuffer</a>&nbsp;str)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FileConvert.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,%20java.lang.StringBuffer,%20java.lang.String)">toPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/StringBuffer.html?is-external=true" title="class or interface in java.lang">StringBuffer</a>&nbsp;str,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainID)</code>
<div class="block">Print ATOM record in the following syntax</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io.mmcif">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/io/mmcif/package-summary.html">org.biojava.nbio.structure.io.mmcif</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/mmcif/package-summary.html">org.biojava.nbio.structure.io.mmcif</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model">AtomSite</a></code></td>
<td class="colLast"><span class="strong">MMCIFFileTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/mmcif/MMCIFFileTools.html#convertAtomToAtomSite(org.biojava.nbio.structure.Atom,%20int,%20java.lang.String,%20java.lang.String)">convertAtomToAtomSite</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
                     int&nbsp;model,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;internalChainId)</code>
<div class="block">Converts an Atom object to an <a href="../../../../../org/biojava/nbio/structure/io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model"><code>AtomSite</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model">AtomSite</a></code></td>
<td class="colLast"><span class="strong">MMCIFFileTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/mmcif/MMCIFFileTools.html#convertAtomToAtomSite(org.biojava.nbio.structure.Atom,%20int,%20java.lang.String,%20java.lang.String,%20int)">convertAtomToAtomSite</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
                     int&nbsp;model,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;internalChainId,
                     int&nbsp;atomId)</code>
<div class="block">Converts an Atom object to an <a href="../../../../../org/biojava/nbio/structure/io/mmcif/model/AtomSite.html" title="class in org.biojava.nbio.structure.io.mmcif.model"><code>AtomSite</code></a> object.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.core">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/core/package-summary.html">org.biojava.nbio.structure.symmetry.core</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/core/package-summary.html">org.biojava.nbio.structure.symmetry.core</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">UniqueSequenceList.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/core/UniqueSequenceList.html#getCalphaAtoms()">getCalphaAtoms</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/core/package-summary.html">org.biojava.nbio.structure.symmetry.core</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">UniqueSequenceList.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/core/UniqueSequenceList.html#getSequenceString(org.biojava.nbio.structure.Atom[])">getSequenceString</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caAlphaAtoms)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">SequenceAlignmentCluster.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/core/SequenceAlignmentCluster.html#identityMatch(org.biojava.nbio.structure.Atom[],%20java.lang.String,%20int,%20int,%20java.lang.String)">identityMatch</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;cAlphaAtoms,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
             int&nbsp;modelNumber,
             int&nbsp;structureId,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence)</code>
<div class="block">Match a sequence to this cluster at 100% identity.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">UniqueSequenceList.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/core/UniqueSequenceList.html#isMatch(org.biojava.nbio.structure.Atom[])">isMatch</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caAlphaAtoms)</code>
<div class="block">Return true is the sequence and residues numbers of the passed in array of
 atoms matches those of this unique sequence list</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/symmetry/core/package-summary.html">org.biojava.nbio.structure.symmetry.core</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/core/UniqueSequenceList.html#UniqueSequenceList(org.biojava.nbio.structure.Atom[],%20java.lang.String,%20int,%20int,%20java.lang.String)">UniqueSequenceList</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;cAlphaAtoms,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
                  int&nbsp;modelNumber,
                  int&nbsp;structureId,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqResSequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.internal">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that return <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">CeSymmResult.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html#getAtoms()">getAtoms</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
     <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom[])">analyze</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Analyze the symmetries of the input Atom array using the DEFAULT
 parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyze</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
       <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code>
<div class="block">Analyze the symmetries of the input Atom array using the provided
 parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#analyzeLevel(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyzeLevel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
            <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code>
<div class="block">Analyze a single level of symmetry.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">AngleOrderDetectorPlus.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/AngleOrderDetectorPlus.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">GraphComponentOrderDetector.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/GraphComponentOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">SequenceFunctionOrderDetector.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">OrderDetector.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/OrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymmIterative.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmIterative.html#execute(org.biojava.nbio.structure.Atom[])">execute</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">This method uses iteratively CeSymm to calculate all symmetries in the
 input array of atoms and organize them in a multiple alignment of the
 repeats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">GraphComponentRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">refineSymmetry</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;k)</code>
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">refineSymmetry</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;k)</code>
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeSymmResult.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html#setAtoms(org.biojava.nbio.structure.Atom[])">setAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean[][]</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#blankOutBreakFlag(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20boolean[][],%20int)">blankOutBreakFlag</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 int&nbsp;rows,
                 int&nbsp;cols,
                 <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                 boolean[][]&nbsp;breakFlag,
                 int&nbsp;blankWindowSize)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#blankOutCEOrig(org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.jama.Matrix,%20int)">blankOutCEOrig</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;rows,
              int&nbsp;cols,
              <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
              <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;origM,
              int&nbsp;blankWindowSize)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#blankOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.jama.Matrix,%20int)">blankOutPreviousAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                         int&nbsp;rows,
                         int&nbsp;cols,
                         <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                         <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
                         int&nbsp;blankWindowSize)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#buildSymmetryGraph(java.util.List,%20org.biojava.nbio.structure.Atom[],%20boolean)">buildSymmetryGraph</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&gt;&nbsp;afps,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
                  boolean&nbsp;undirected)</code>
<div class="block">Converts a set of AFP alignments into a Graph of aligned residues, where
 each vertex is a residue and each edge means the connection between the
 two residues in one of the alignments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">fromAFP</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAngle</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Returns the <em>magnitude</em> of the angle between the first and second
 blocks of <code>afpChain</code>, measured in degrees.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAngle</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Returns the <em>magnitude</em> of the angle between the first and second
 blocks of <code>afpChain</code>, measured in degrees.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20double[],%20double[],%20int,%20int)">getDkMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
           int&nbsp;fragmentLength,
           double[]&nbsp;dist1,
           double[]&nbsp;dist2,
           int&nbsp;rows,
           int&nbsp;cols)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20double[],%20double[],%20int,%20int)">getDkMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
           int&nbsp;fragmentLength,
           double[]&nbsp;dist1,
           double[]&nbsp;dist2,
           int&nbsp;rows,
           int&nbsp;cols)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getDkMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
           int&nbsp;k,
           int&nbsp;fragmentLength)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getDkMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
           int&nbsp;k,
           int&nbsp;fragmentLength)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getGroups(org.biojava.nbio.structure.Atom[])">getGroups</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;rAtoms)</code>
<div class="block">Returns the List of Groups of the corresponding representative Atom
 array.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#grayOutCEOrig(org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.jama.Matrix,%20int,%20double[],%20double)">grayOutCEOrig</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;rows,
             int&nbsp;cols,
             <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
             <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;origM,
             int&nbsp;blankWindowSize,
             double[]&nbsp;gradientPolyCoeff,
             double&nbsp;gradientExpCoeff)</code>
<div class="block">Grays out the main diagonal of a duplicated distance matrix.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#grayOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.jama.Matrix,%20int,%20double[],%20double)">grayOutPreviousAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                        int&nbsp;rows,
                        int&nbsp;cols,
                        <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                        <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
                        int&nbsp;blankWindowSize,
                        double[]&nbsp;gradientPolyCoeff,
                        double&nbsp;gradientExpCoeff)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#updateSymmetryTransformation(org.biojava.nbio.structure.symmetry.internal.SymmetryAxes,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.Atom[])">updateSymmetryTransformation</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a>&nbsp;axes,
                            <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Calculates the set of symmetry operation Matrices (transformations) of
 the new alignment, based on the symmetry relations in the SymmetryAxes
 object.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/class-use/Atom.html" target="_top">Frames</a></li>
<li><a href="Atom.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
