<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:50 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Class org.biojava.nbio.structure.StructureException (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.biojava.nbio.structure.StructureException (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/class-use/StructureException.html" target="_top">Frames</a></li>
<li><a href="StructureException.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.structure.StructureException" class="title">Uses of Class<br>org.biojava.nbio.structure.StructureException</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#demo">demo</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for protein structure (PDB).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></td>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.client">org.biojava.nbio.structure.align.client</a></td>
<td class="colLast">
<div class="block">This package deals with the server communication for auto-downloading pre-calculated alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.fatcat">org.biojava.nbio.structure.align.fatcat</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the FATCAT alignment algorithm, here called jFATCAT.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.fatcat.calc">org.biojava.nbio.structure.align.fatcat.calc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple.mc">org.biojava.nbio.structure.align.multiple.mc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.pairwise">org.biojava.nbio.structure.align.pairwise</a></td>
<td class="colLast">
<div class="block">Classes for the pairwise alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.seq">org.biojava.nbio.structure.align.seq</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.util">org.biojava.nbio.structure.align.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></td>
<td class="colLast">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.cath">org.biojava.nbio.structure.cath</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.domain">org.biojava.nbio.structure.domain</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.domain.pdp">org.biojava.nbio.structure.domain.pdp</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.ecod">org.biojava.nbio.structure.ecod</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.gui.util">org.biojava.nbio.structure.gui.util</a></td>
<td class="colLast">
<div class="block">
Some utility classes for the protein structure GUIs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></td>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.quaternary.io">org.biojava.nbio.structure.quaternary.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.scop">org.biojava.nbio.structure.scop</a></td>
<td class="colLast">
<div class="block">Parsers and API for <a href="http://scop.mrc-lmb.cam.ac.uk/scop/">SCOP, Structural Classification of Proteins</a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.secstruc">org.biojava.nbio.structure.secstruc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.gui">org.biojava.nbio.structure.symmetry.gui</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="demo">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../demo/package-summary.html">demo</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../demo/package-summary.html">demo</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoAlignmentFromFasta.</span><code><strong><a href="../../../../../demo/DemoAlignmentFromFasta.html#getAlignmentFromFasta()">getAlignmentFromFasta</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoAlignmentFromFasta.</span><code><strong><a href="../../../../../demo/DemoAlignmentFromFasta.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPFromFasta.</span><code><strong><a href="../../../../../demo/AFPFromFasta.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoContacts.</span><code><strong><a href="../../../../../demo/DemoContacts.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoLoadSecStruc.</span><code><strong><a href="../../../../../demo/DemoLoadSecStruc.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoAsa.</span><code><strong><a href="../../../../../demo/DemoAsa.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoSecStrucCalc.</span><code><strong><a href="../../../../../demo/DemoSecStrucCalc.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoMultipleMC.</span><code><strong><a href="../../../../../demo/DemoMultipleMC.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoCeSymm.</span><code><strong><a href="../../../../../demo/DemoCeSymm.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DemoOrientBioAssembly.</span><code><strong><a href="../../../../../demo/DemoOrientBioAssembly.html#runPDB(java.lang.String,%20int)">runPDB</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbID,
      int&nbsp;bioAssemblyNr)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerAtoms(org.biojava.nbio.structure.Atom[])">centerAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code>
<div class="block">Center the atoms at the Centroid.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#centerAtoms(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom)">centerAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code>
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#createVirtualCBAtom(org.biojava.nbio.structure.AminoAcid)">createVirtualCBAtom</a></strong>(<a href="../../../../../org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;amino)</code>
<div class="block">creates a virtual C-beta atom.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">StructureImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html#findChain(java.lang.String)">findChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>
<div class="block">Request a particular chain from a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">Structure.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Structure.html#findChain(java.lang.String)">findChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>
<div class="block">Request a particular chain from a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">StructureImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html#findChain(java.lang.String,%20int)">findChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
         int&nbsp;modelnr)</code>
<div class="block">Request a particular chain from a particular model</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">Structure.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Structure.html#findChain(java.lang.String,%20int)">findChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
         int&nbsp;modelnr)</code>
<div class="block">Request a particular chain from a particular model</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="strong">StructureImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html#findGroup(java.lang.String,%20java.lang.String)">findGroup</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainName,
         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbResnum)</code>
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="strong">Structure.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Structure.html#findGroup(java.lang.String,%20java.lang.String)">findGroup</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbResnum)</code>
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="strong">StructureImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html#findGroup(java.lang.String,%20java.lang.String,%20int)">findGroup</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbResnum,
         int&nbsp;modelnr)</code>
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="strong">Structure.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Structure.html#findGroup(java.lang.String,%20java.lang.String,%20int)">findGroup</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbResnum,
         int&nbsp;modelnr)</code>
<div class="block">Request a particular group from a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureIO.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly(java.lang.String)">getBiologicalAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Returns the first biologicalAssembly that is available for a protein structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureIO.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly(java.lang.String,%20int)">getBiologicalAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;biolAssemblyNr)</code>
<div class="block">By default the getStructure method loads asym units.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureIO.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly(java.lang.String,%20int,%20org.biojava.nbio.structure.align.util.AtomCache)">getBiologicalAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;biolAssemblyNr,
                     <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureIO.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getBiologicalAssembly(java.lang.String,%20int,%20org.biojava.nbio.structure.align.util.AtomCache,%20org.biojava.nbio.structure.quaternary.io.BioUnitDataProvider)">getBiologicalAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;biolAssemblyNr,
                     <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache,
                     <a href="../../../../../org/biojava/nbio/structure/quaternary/io/BioUnitDataProvider.html" title="interface in org.biojava.nbio.structure.quaternary.io">BioUnitDataProvider</a>&nbsp;provider)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">StructureImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html#getChainByPDB(java.lang.String)">getChainByPDB</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>
<div class="block">Request a chain by its PDB code
 by default takes only the first model</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">Structure.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Structure.html#getChainByPDB(java.lang.String)">getChainByPDB</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>
<div class="block">Request a chain by its PDB code
 by default takes only the first model</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">StructureImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureImpl.html#getChainByPDB(java.lang.String,%20int)">getChainByPDB</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
             int&nbsp;modelnr)</code>
<div class="block">Request a chain by its PDB code
 by default takes only the first model</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></code></td>
<td class="colLast"><span class="strong">Structure.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Structure.html#getChainByPDB(java.lang.String,%20int)">getChainByPDB</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId,
             int&nbsp;modelnr)</code>
<div class="block">Request a chain by its PDB code
 by default takes only the first model</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="strong">ChainImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/ChainImpl.html#getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">getGroupByPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</code>
<div class="block">Get a group by its PDB residue numbering.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="strong">Chain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Chain.html#getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">getGroupByPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</code>
<div class="block">Get a group by its PDB residue numbering.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getGroupByPDBResidueNumber(org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.ResidueNumber)">getGroupByPDBResidueNumber</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;struc,
                          <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbResNum)</code>
<div class="block">Get a group represented by a ResidueNumber.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">ChainImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/ChainImpl.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,%20org.biojava.nbio.structure.ResidueNumber)">getGroupsByPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;start,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;end)</code>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">Chain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Chain.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,%20org.biojava.nbio.structure.ResidueNumber)">getGroupsByPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)</code>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">ChainImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/ChainImpl.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,%20org.biojava.nbio.structure.ResidueNumber,%20boolean)">getGroupsByPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;start,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;end,
              boolean&nbsp;ignoreMissing)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">Chain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Chain.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,%20org.biojava.nbio.structure.ResidueNumber,%20boolean)">getGroupsByPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
              boolean&nbsp;ignoreMissing)</code>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getPhi(org.biojava.nbio.structure.AminoAcid,%20org.biojava.nbio.structure.AminoAcid)">getPhi</a></strong>(<a href="../../../../../org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;a,
      <a href="../../../../../org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;b)</code>
<div class="block">Calculate the phi angle.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#getPsi(org.biojava.nbio.structure.AminoAcid,%20org.biojava.nbio.structure.AminoAcid)">getPsi</a></strong>(<a href="../../../../../org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;a,
      <a href="../../../../../org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;b)</code>
<div class="block">Calculate the psi angle.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getReducedStructure(org.biojava.nbio.structure.Structure,%20int)">getReducedStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   int&nbsp;chainNr)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)"><code>StructureIdentifier.reduce(Structure)</code></a> instead (v. 4.2.0)</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getReducedStructure(org.biojava.nbio.structure.Structure,%20java.lang.String)">getReducedStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
                   <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)"><code>StructureIdentifier.reduce(Structure)</code></a> instead (v. 4.2.0)</i></div>
</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getRMS(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getRMS</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2)</code>
<div class="block">Calculate the RMS (root mean square) deviation of two sets of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getStructure(java.lang.String)">getStructure</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Short version of <a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getStructure(java.lang.String,%20org.biojava.nbio.structure.io.PDBFileParser,%20org.biojava.nbio.structure.align.util.AtomCache)"><code>StructureTools.getStructure(String, PDBFileParser, AtomCache)</code></a>
 which creates new parsers when needed</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureIO.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIO.html#getStructure(java.lang.String)">getStructure</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Loads a structure based on a name.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getStructure(java.lang.String,%20org.biojava.nbio.structure.io.PDBFileParser,%20org.biojava.nbio.structure.align.util.AtomCache)">getStructure</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name,
            <a href="../../../../../org/biojava/nbio/structure/io/PDBFileParser.html" title="class in org.biojava.nbio.structure.io">PDBFileParser</a>&nbsp;parser,
            <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Flexibly get a structure from an input String.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureTools.html#getSubRanges(org.biojava.nbio.structure.Structure,%20java.lang.String)">getSubRanges</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;ranges)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure"><code>StructureIdentifier</code></a> instead (4.2.0)</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getTMScore(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
          int&nbsp;len1,
          int&nbsp;len2)</code>
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SVDSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#getTMScoreAlternate(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int)">getTMScoreAlternate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
                   int&nbsp;len1,
                   int&nbsp;len2)</code>
<div class="block">Calculate the TM-Score for the superposition.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">BioAssemblyIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/BioAssemblyIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">URLIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/URLIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Load the structure from the URL</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PassthroughIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/PassthroughIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Passthrough identifiers don't know how to load a structure</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">SubstructureIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Loads the complete structure based on <a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#getPdbId()"><code>SubstructureIdentifier.getPdbId()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Loads a structure encompassing the structure identified.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">BioAssemblyIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/BioAssemblyIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">URLIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/URLIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PassthroughIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/PassthroughIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">SubstructureIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Takes a complete structure as input and reduces it to residues present in
 the specified ranges</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>
<div class="block">Takes a complete structure as input and reduces it to the substructure
 represented by this StructureIdentifier.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#rotate(org.biojava.nbio.structure.Group,%20double[][])">rotate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group,
      double[][]&nbsp;rotationmatrix)</code>
<div class="block">Rotate a Group.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">Calc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/Calc.html#rotate(org.biojava.nbio.structure.Structure,%20double[][])">rotate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
      double[][]&nbsp;rotationmatrix)</code>
<div class="block">Rotate a structure.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<td class="colLast"><span class="strong">BioAssemblyIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/BioAssemblyIdentifier.html#toCanonical()">toCanonical</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<td class="colLast"><span class="strong">StructureIdentifier.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html#toCanonical()">toCanonical</a></strong>()</code>
<div class="block">Convert to a canonical SubstructureIdentifier.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/SVDSuperimposer.html#SVDSuperimposer(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">SVDSuperimposer</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2)</code>
<div class="block">Create a SVDSuperimposer object and calculate a SVD superimposition of two sets of atoms.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>
<div class="block">run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.StrucAligParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>
<div class="block">calculate the  protein structure superimposition, between two sets of atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleStructureAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/MultipleStructureAligner.html#align(java.util.List)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code>
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleStructureAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/MultipleStructureAligner.html#align(java.util.List,%20java.lang.Object)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>
<div class="block">Run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20java.lang.String)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId2)</code>
<div class="block">Align two chains from the structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20java.lang.String,%20org.biojava.nbio.structure.align.StrucAligParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;chainId2,
     <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>
<div class="block">Aligns two chains from the structures using user provided parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2)</code>
<div class="block">Calculate the alignment between the two full structures with default parameters</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructurePairAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructurePairAligner.html#align(org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.align.StrucAligParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s1,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s2,
     <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>
<div class="block">Calculate the alignment between the two full structures with user provided parameters</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/StructureAlignment.html" title="interface in org.biojava.nbio.structure.align">StructureAlignment</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentFactory.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/StructureAlignmentFactory.html#getAlgorithm(java.lang.String)">getAlgorithm</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistOptimized(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistOptimized</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">superimposing according to the optimized alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistPDB(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistPDB</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">calculate the total rmsd of the blocks
 output a merged pdb file for both proteins
          protein 1, in chain A
         protein 2 is twisted according to the twists detected, in chain B</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.ce">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Aligns ca1 and ca2 using a heuristic to check for CPs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Align ca2 onto ca1.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20org.biojava.nbio.structure.align.model.AFPChain[])">alignOptimal</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
            <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignPermuted(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20int)">alignPermuted</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
             int&nbsp;cp)</code>
<div class="block">Aligns ca1 with ca2 permuted by <i>cp</i> residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform)</code>
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code>
<div class="block">superimpose and get rmsd</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean,%20boolean)">calc_rmsd</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
         int&nbsp;strLen,
         boolean&nbsp;storeTransform,
         boolean&nbsp;show)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Use <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20boolean)"><code>CECalculator.calc_rmsd(Atom[],Atom[],int,boolean)</code></a> instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">filterDuplicateAFPs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code>
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECPParameters)">filterDuplicateAFPs</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
                   <a href="../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator)">postProcessAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.align.ce.CECPParameters)">postProcessAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                    <a href="../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.client">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">FarmJobRunnable.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/FarmJobRunnable.html#alignPair(java.lang.String,%20java.lang.String)">alignPair</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">FarmJobRunnable.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/FarmJobRunnable.html#alignPair(java.lang.String,%20java.lang.String,%20java.lang.String)">alignPair</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;algorithmName)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a></code></td>
<td class="colLast"><span class="strong">StructureName.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#getBaseIdentifier()">getBaseIdentifier</a></strong>()</code>
<div class="block">StructureName wraps another StructureIdentifier.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">PdbPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/PdbPair.html#getPDBCode1()">getPDBCode1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">PdbPair.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/PdbPair.html#getPDBCode2()">getPDBCode2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureName.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#getPdbId()">getPdbId</a></strong>()</code>
<div class="block">Get the PDB ID for this name, if any.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureName.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureName.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></code></td>
<td class="colLast"><span class="strong">StructureName.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html#toCanonical()">toCanonical</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.fatcat">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20boolean)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
     boolean&nbsp;doRigid)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignFlexible(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignFlexible</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">alignRigid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignRigid</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.fatcat.calc">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FatCatAligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatAligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean,%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     boolean&nbsp;doRigid,
     <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPCalculator.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html#extractAFPChains(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractAFPChains</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
                <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPOptimizer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html#optimizeAln(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">optimizeAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">optimize the alignment by dynamic programming</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StructureAlignmentOptimizer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/StructureAlignmentOptimizer.html#runOptimization(int)">runOptimization</a></strong>(int&nbsp;maxi)</code>
<div class="block">run the optimization</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/fatcat/calc/StructureAlignmentOptimizer.html#StructureAlignmentOptimizer(int,%20int,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20int,%20int[][])">StructureAlignmentOptimizer</a></strong>(int&nbsp;b1,
                           int&nbsp;end1,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c1,
                           int&nbsp;b2,
                           int&nbsp;end2,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c2,
                           int&nbsp;iniLen,
                           int[][]&nbsp;iniSet)</code>
<div class="block">optimize the structural alignment by update the equivalent residues
 and then run dynamic programming
 input: len1 the length of structure 1; c1: the structure information of 1
        len2 the length of structure 2; c2: the structure information of 2
        iniLen and iniSet is the length and list of initial equivalent residues</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">createArtificalStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Display an AFPChain alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Group[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.List,%20java.util.List)">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms2)</code>
<div class="block">Note: ca2, hetatoms2 and nucleotides2 should not be rotated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#display(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code>
<div class="block">Display a MultipleAlignment with a JmolPanel.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAlignedStructure(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlignedStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Get an artifical Structure containing both chains.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#getAlignedStructure(java.util.List)">getAlignedStructure</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code>
<div class="block">Get an artifical Structure containing a different model for every
 input structure, so that the alignment result can be viewed in Jmol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom[],%20java.util.List)">getAtomArray</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code>
<div class="block">Returns the first atom for each group</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20boolean)">getAtomForAligPos</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 int&nbsp;chainNr,
                 int&nbsp;aligPos,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
                 boolean&nbsp;getPrevious)</code>
<div class="block">return the atom at alignment position aligPos.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">SelectPDBPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/SelectPDBPanel.html#getStructure1()">getStructure1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">SelectPDBPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/SelectPDBPanel.html#getStructure2()">getStructure2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">prepareGroupsForDisplay</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Rotate the Atoms/Groups so they are aligned for the 3D visualisation</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#showAlignmentPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showAlignmentPanel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentJmolDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html#showMultipleAligmentPanel(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showMultipleAligmentPanel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
                         <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>
<div class="block">Creates a new Frame with the MultipleAlignment Sequence Panel.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.aligpanel">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html#MultipleAligPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>
<div class="block">Constructor using an afpChain and the atom arrays for pairwise
 alignments.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentEnsembleImpl.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#updateAtomArrays()">updateAtomArrays</a></strong>()</code>
<div class="block">Force the atom arrays to regenerate based on
 <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#getStructureIdentifiers()"><code>MultipleAlignmentEnsembleImpl.getStructureIdentifiers()</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple.mc">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/multiple/mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/multiple/mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleMcMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcMain.html#align(java.util.List)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleMcMain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcMain.html#align(java.util.List,%20java.lang.Object)">align</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;parameters)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">MultipleMcOptimizer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/mc/MultipleMcOptimizer.html#optimize()">optimize</a></strong>()</code>
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#calculateScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">calculateScores</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Calculates and puts the RMSD and the average TM-Score of the
 MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getAvgTMScore(java.util.List,%20java.util.List)">getAvgTMScore</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;lengths)</code>
<div class="block">Calculates the average TMScore all the possible pairwise structure
 comparisons of the given a set of superimposed Atoms and the original
 structure lengths.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getAvgTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAvgTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Calculates the average TMScore of all the possible pairwise structure
 comparisons of the given alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getMCScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20double,%20double,%20double)">getMCScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          double&nbsp;gapOpen,
          double&nbsp;gapExtension,
          double&nbsp;dCutoff)</code>
<div class="block">Calculates the MC score, specific for the MultipleAlignment algorithm.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getRefTMScore(java.util.List,%20java.util.List,%20int)">getRefTMScore</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;lengths,
             int&nbsp;reference)</code>
<div class="block">Calculates the average TMScore from all structures to a reference
 structure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html#getRefTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20int)">getRefTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
             int&nbsp;ref)</code>
<div class="block">Calculates the average TMScore from all structures to a reference
 structure, given a set of superimposed atoms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></td>
<td class="colLast"><span class="strong">MultipleAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentDisplay.html#getRotatedAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getRotatedAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code>
<div class="block">New structures are downloaded if they were not cached in the alignment
 and they are entirely transformed here with the superposition information
 in the Multiple Alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CoreSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/CoreSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">MultipleSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>
<div class="block">Superimpose all structures from a <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ReferenceSuperimposer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/ReferenceSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">superimpose</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">MultipleAlignmentWriter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#to3DFormat(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20int,%20int)">to3DFormat</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
          int&nbsp;queryIndex,
          int&nbsp;templateIndex)</code>
<div class="block">Outputs a pairwise alignment in I-TASSER's 3D Format for target-template
 alignment.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.pairwise">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#approach_ap3(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.pairwise.FragmentPair[],%20org.biojava.nbio.structure.align.StrucAligParameters)">approach_ap3</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a>[]&nbsp;fraglst,
            <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#calculateSuperpositionByIdx(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calculateSuperpositionByIdx</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                           <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments[],%20org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]&nbsp;fragments,
               <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments,%20org.biojava.nbio.structure.align.StrucAligParameters)">extendFragments</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
               <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;fragments,
               <a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#finish(org.biojava.nbio.structure.align.StrucAligParameters,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">finish</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">FragmentJoiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html#getRMS(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.helper.JointFragments)">getRMS</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
      <a href="../../../../../org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;frag)</code>
<div class="block">get the RMS of the JointFragments pair frag</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AlternativeAlignment.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html#refine(org.biojava.nbio.structure.align.StrucAligParameters,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Refinement procedure based on superposition and dynamic programming.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.seq">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;parameters)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#alignmentAsMap(org.biojava.nbio.structure.align.model.AFPChain)">alignmentAsMap</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">Creates a Map specifying the alignment as a mapping between residue indices
 of protein 1 and residue indices of protein 2.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#createAFPChain(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.ResidueNumber[],%20org.biojava.nbio.structure.ResidueNumber[])">createAFPChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned1,
              <a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned2)</code>
<div class="block">Fundamentally, an alignment is just a list of aligned residues in each
 protein.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain)">getAngle</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">Calculate the rotation angle for a structure</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getAtoms(java.lang.String)">getAtoms</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Returns the CA atoms for the provided name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getAtoms(org.biojava.nbio.structure.StructureIdentifier)">getAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssembly(java.lang.String)">getBiologicalAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Loads the default biological unit (e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssembly(java.lang.String,%20int)">getBiologicalAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;bioAssemblyId)</code>
<div class="block">Loads the default biological unit (e.g.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssembly(java.lang.String,%20int,%20boolean)">getBiologicalAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
                     int&nbsp;bioAssemblyId,
                     boolean&nbsp;bioAssemblyFallback)</code>
<div class="block">Loads the biological assembly for a given PDB ID and bioAssemblyId.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalUnit(java.lang.String)">getBiologicalUnit</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>Renamed to <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getBiologicalAssembly(java.lang.String)"><code>AtomCache.getBiologicalAssembly(String)</code></a> in 4.2</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getRepresentativeAtoms(java.lang.String)">getRepresentativeAtoms</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Returns the representative atoms for the provided name.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getRepresentativeAtoms(org.biojava.nbio.structure.StructureIdentifier)">getRepresentativeAtoms</a></strong>(<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getRotMax(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getRotMax</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
         <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructure(java.lang.String)">getStructure</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Request a Structure based on a <i>name</i>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructure(org.biojava.nbio.structure.StructureIdentifier)">getStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;strucId)</code>
<div class="block">Get the structure corresponding to the given <a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure"><code>StructureIdentifier</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForCathDomain(org.biojava.nbio.structure.align.client.StructureName)">getStructureForCathDomain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a>&nbsp;structureName)</code>
<div class="block">Returns a <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> corresponding to the CATH identifier supplied in <code>structureName</code>, using the the <a href="../../../../../org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabase</code></a>
 at <a href="../../../../../org/biojava/nbio/structure/cath/CathFactory.html#getCathDatabase()"><code>CathFactory.getCathDatabase()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForCathDomain(org.biojava.nbio.structure.align.client.StructureName,%20org.biojava.nbio.structure.cath.CathDatabase)">getStructureForCathDomain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a>&nbsp;structureName,
                         <a href="../../../../../org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath">CathDatabase</a>&nbsp;cathInstall)</code>
<div class="block">Returns a <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> corresponding to the CATH identifier supplied in <code>structureName</code>, using the specified <a href="../../../../../org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabase</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain(org.biojava.nbio.structure.scop.ScopDomain)">getStructureForDomain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain(org.biojava.nbio.structure.scop.ScopDomain,%20org.biojava.nbio.structure.scop.ScopDatabase)">getStructureForDomain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain,
                     <a href="../../../../../org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain(org.biojava.nbio.structure.scop.ScopDomain,%20org.biojava.nbio.structure.scop.ScopDatabase,%20boolean)">getStructureForDomain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a>&nbsp;domain,
                     <a href="../../../../../org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase,
                     boolean&nbsp;strictLigandHandling)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain(java.lang.String)">getStructureForDomain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;scopId)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForDomain(java.lang.String,%20org.biojava.nbio.structure.scop.ScopDatabase)">getStructureForDomain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;scopId,
                     <a href="../../../../../org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a>&nbsp;scopDatabase)</code>
<div class="block">Returns the representation of a <a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop"><code>ScopDomain</code></a> as a BioJava <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructureForPdbId(java.lang.String)">getStructureForPdbId</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">Loads a structure directly by PDB ID</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#getSymmetryOrder(org.biojava.nbio.structure.align.model.AFPChain,%20int,%20float)">getSymmetryOrder</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                int&nbsp;maxSymmetry,
                float&nbsp;minimumMetricChange)</code>
<div class="block">Guesses the order of symmetry in an alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">AFPChainScorer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTMScore</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTranslation</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#loadStructureFromCifByPdbId(java.lang.String)">loadStructureFromCifByPdbId</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AtomCache.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#loadStructureFromPdbByPdbId(java.lang.String)">loadStructureFromPdbByPdbId</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int[],%20int[][][])">replaceOptAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;blockNum,
             int[]&nbsp;optLens,
             int[][][]&nbsp;optAln)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.Map)">replaceOptAln</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;alignment)</code>
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(int[][][],%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">replaceOptAln</a></strong>(int[][][]&nbsp;newAlgn,
             <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">splitBlocksByTopology</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#updateSuperposition(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">updateSuperposition</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">After the alignment changes (optAln, optLen, blockNum, at a minimum),
 many other properties which depend on the superposition will be invalid.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#RotationAxis(org.biojava.nbio.structure.align.model.AFPChain)">RotationAxis</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">Calculate the rotation axis for the first block of an AFPChain</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.xml">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#flipAlignment(java.lang.String)">flipAlignment</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml)</code>
<div class="block">Takes an XML representation of the alignment and flips the positions of name1 and name2</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainFlipper.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainFlipper.html#flipChain(org.biojava.nbio.structure.align.model.AFPChain)">flipChain</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;o)</code>
<div class="block">Flip the position of name1 and name2 (as well as all underlying data) in an AFPChain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">fromXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">new utility method that checks that the order of the pair in the XML alignment is correct and flips the direction if needed</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.cath">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/cath/package-summary.html">org.biojava.nbio.structure.cath</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/cath/package-summary.html">org.biojava.nbio.structure.cath</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">CathDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/cath/CathDomain.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">CathDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/cath/CathDomain.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.domain">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/domain/package-summary.html">org.biojava.nbio.structure.domain</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/domain/package-summary.html">org.biojava.nbio.structure.domain</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">RemotePDPProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/RemotePDPProvider.html#getDomain(java.lang.String,%20org.biojava.nbio.structure.align.util.AtomCache)">getDomain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdpDomainName,
         <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Get the structure for a particular PDP domain</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDPProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/PDPProvider.html#getDomain(java.lang.String,%20org.biojava.nbio.structure.align.util.AtomCache)">getDomain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdpDomainName,
         <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>
<div class="block">Get the structure for a particular PDP domain</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/SortedSet.html?is-external=true" title="class or interface in java.util">SortedSet</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">RemoteDomainProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/RemoteDomainProvider.html#getDomainNames(java.lang.String)">getDomainNames</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/SortedSet.html?is-external=true" title="class or interface in java.util">SortedSet</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">DomainProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/DomainProvider.html#getDomainNames(java.lang.String)">getDomainNames</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Get a list of constituent domain identifiers</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDPDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/PDPDomain.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">RemoteDomainProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/RemoteDomainProvider.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">RemotePDPProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/RemotePDPProvider.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDPDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/PDPDomain.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></td>
<td class="colLast"><span class="strong">LocalProteinDomainParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/LocalProteinDomainParser.html#suggestDomains(org.biojava.nbio.structure.Atom[])">suggestDomains</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Suggest domains for a set of Calpha atoms</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></td>
<td class="colLast"><span class="strong">LocalProteinDomainParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/LocalProteinDomainParser.html#suggestDomains(org.biojava.nbio.structure.Structure)">suggestDomains</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code>
<div class="block">Suggest domains for a protein structure</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.domain.pdp">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a></code></td>
<td class="colLast"><span class="strong">GetDistanceMatrix.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/domain/pdp/GetDistanceMatrix.html#getDistanceMatrix(org.biojava.nbio.structure.Atom[])">getDistanceMatrix</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;protein)</code>
<div class="block">A set of Calpha atoms that are representing the protein</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.ecod">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/ecod/package-summary.html">org.biojava.nbio.structure.ecod</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/ecod/package-summary.html">org.biojava.nbio.structure.ecod</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">EcodDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/ecod/EcodDomain.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">EcodDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/ecod/EcodDomain.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.gui.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/gui/util/package-summary.html">org.biojava.nbio.structure.gui.util</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDBDirPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBDirPanel.html#getStructure1()">getStructure1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDBServerPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBServerPanel.html#getStructure1()">getStructure1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">ScopSelectPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/ScopSelectPanel.html#getStructure1()">getStructure1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDBUploadPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBUploadPanel.html#getStructure1()">getStructure1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructurePairSelector.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/StructurePairSelector.html#getStructure1()">getStructure1</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDBDirPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBDirPanel.html#getStructure2()">getStructure2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDBServerPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBServerPanel.html#getStructure2()">getStructure2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">ScopSelectPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/ScopSelectPanel.html#getStructure2()">getStructure2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">PDBUploadPanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/PDBUploadPanel.html#getStructure2()">getStructure2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructurePairSelector.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/StructurePairSelector.html#getStructure2()">getStructure2</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&gt;</code></td>
<td class="colLast"><span class="strong">SelectMultiplePanel.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/gui/util/SelectMultiplePanel.html#getStructures()">getStructures</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.io.File,%20org.biojava.nbio.structure.Structure,%20int)">cpFastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
                 <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.structure.Structure,%20int)">cpFastaToAfpChain</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;first,
                 <a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;second,
                 <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20int)">cpFastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;first,
                 <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;second,
                 <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaFileToAfpChain(java.io.File,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaFileToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
                   <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                   <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Reads the file <code>fastaFile</code>, expecting exactly two sequences which give a pairwise alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaStringToAfpChain(java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaStringToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence2,
                     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(java.util.Map,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;sequences,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Uses two sequences each with a corresponding structure to create an AFPChain corresponding to the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence1,
               <a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence2,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.alignment.template.SequencePair,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;alignment,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Provided only for convenience.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence1,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence2,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">TODO Write comment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">SandboxStyleStructureProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/SandboxStyleStructureProvider.html#getStructureById(java.lang.String)">getStructureById</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">StructureProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/StructureProvider.html#getStructureById(java.lang.String)">getStructureById</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId)</code>
<div class="block">get the structure for a PDB ID</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#main(java.lang.String[])">main</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;args)</code>
<div class="block">Prints out the XML representation of an AFPChain from a file containing exactly two FASTA sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">FastaStructureParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#process()">process</a></strong>()</code>
<div class="block">Parses the fasta file and loads it into memory.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.quaternary.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/quaternary/io/package-summary.html">org.biojava.nbio.structure.quaternary.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/quaternary/io/package-summary.html">org.biojava.nbio.structure.quaternary.io</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">MmCifBiolAssemblyProvider.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/quaternary/io/MmCifBiolAssemblyProvider.html#getBiolAssembly(java.lang.String,%20int)">getBiolAssembly</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdbId,
               int&nbsp;biolAssemblyNr)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.scop">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/scop/package-summary.html">org.biojava.nbio.structure.scop</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/scop/package-summary.html">org.biojava.nbio.structure.scop</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">ScopDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html#loadStructure(org.biojava.nbio.structure.align.util.AtomCache)">loadStructure</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">ScopDomain.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/scop/ScopDomain.html#reduce(org.biojava.nbio.structure.Structure)">reduce</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;input)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.secstruc">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/secstruc/package-summary.html">org.biojava.nbio.structure.secstruc</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/secstruc/package-summary.html">org.biojava.nbio.structure.secstruc</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="strong">SecStrucCalc.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucCalc.html#calculate(org.biojava.nbio.structure.Structure,%20boolean)">calculate</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
         boolean&nbsp;assign)</code>
<div class="block">Predicts the secondary structure of this Structure object,
 using a DSSP implementation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="strong">DSSPParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/secstruc/DSSPParser.html#fetch(java.lang.String,%20org.biojava.nbio.structure.Structure,%20boolean)">fetch</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;pdb,
     <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
     boolean&nbsp;assign)</code>
<div class="block">Fetch and parse the DSSP file of the specified pdb code
 from the PDB web server and return the secondary structure
 annotation as a List of <a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="strong">DSSPParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/secstruc/DSSPParser.html#parseFile(java.lang.String,%20org.biojava.nbio.structure.Structure,%20boolean)">parseFile</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dsspPath,
         <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
         boolean&nbsp;assign)</code>
<div class="block">Parse a DSSP output file and return the secondary structure
 annotation as a List of <a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a>&gt;</code></td>
<td class="colLast"><span class="strong">DSSPParser.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/secstruc/DSSPParser.html#parseString(java.lang.String,%20org.biojava.nbio.structure.Structure,%20boolean)">parseString</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dsspOut,
           <a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
           boolean&nbsp;assign)</code>
<div class="block">Parse a DSSP format String and return the secondary structure
 annotation as a List of <a href="../../../../../org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc"><code>SecStrucState</code></a> objects.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/gui/package-summary.html">org.biojava.nbio.structure.symmetry.gui</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/gui/package-summary.html">org.biojava.nbio.structure.symmetry.gui</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">SymmetryDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/gui/SymmetryDisplay.html#display(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">display</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symmResult)</code>
<div class="block">Displays a single structure in a cartoon representation with each
 symmetric repeat colored differently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">SymmetryDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/gui/SymmetryDisplay.html#displayFull(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">displayFull</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code>
<div class="block">Displays a multiple alignment of the whole structure transformations
 colored by blocks, corresponding to the symmetric protodomains.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">SymmetryDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/gui/SymmetryDisplay.html#displayRepeats(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">displayRepeats</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code>
<div class="block">Displays a multiple alignment of the symmetry repeats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">SymmetryDisplay.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/gui/SymmetryDisplay.html#printSymmetryGroup(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">printSymmetryGroup</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code>
<div class="block">Given a symmetry alignment, it draws the symmetry group axes and the
 polyhedron box around the structure.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.internal">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">align</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
     <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom[])">analyze</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Analyze the symmetries of the input Atom array using the DEFAULT
 parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyze</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
       <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code>
<div class="block">Analyze the symmetries of the input Atom array using the provided
 parameters.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymm.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymm.html#analyzeLevel(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.symmetry.internal.CESymmParameters)">analyzeLevel</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
            <a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code>
<div class="block">Analyze a single level of symmetry.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></td>
<td class="colLast"><span class="strong">CeSymmIterative.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmIterative.html#execute(org.biojava.nbio.structure.Atom[])">execute</a></strong>(<a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">This method uses iteratively CeSymm to calculate all symmetries in the
 input array of atoms and organize them in a multiple alignment of the
 repeats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&gt;</code></td>
<td class="colLast"><span class="strong">CeSymmResult.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html#getRepeatsID()">getRepeatsID</a></strong>()</code>
<div class="block">Return the symmetric repeats as structure identifiers, if the result is
 symmetric and it was refined, return null otherwise.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmOptimizer.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SymmOptimizer.html#optimize()">optimize</a></strong>()</code>
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">GraphComponentRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">refineSymmetry</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;k)</code>
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(java.util.Map,%20int)">refineSymmetry</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;alignment,
              int&nbsp;k)</code>
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a> in <a href="../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that throw <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">fromAFP</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
       <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryResults</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getQuaternarySymmetry(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">getQuaternarySymmetry</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;result)</code>
<div class="block">Given a symmetry result, it calculates the overall global symmetry,
 factoring out the alignment and detection steps of
 <a href="../../../../../org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a> algorithm.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#isSignificant(org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20double)">isSignificant</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
             double&nbsp;symmetryThreshold)</code>
<div class="block">Returns true if the symmetry alignment is significant, false otherwise.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#toRepeatsAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)">toRepeatsAlignment</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;result)</code>
<div class="block">Method that converts a symmetry alignment into an alignment of the
 repeats only, as new independent structures.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#updateSymmetryScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">updateSymmetryScores</a></strong>(<a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;symm)</code>
<div class="block">Update the scores (TM-score and RMSD) of a symmetry multiple alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#updateSymmetryTransformation(org.biojava.nbio.structure.symmetry.internal.SymmetryAxes,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment,%20org.biojava.nbio.structure.Atom[])">updateSymmetryTransformation</a></strong>(<a href="../../../../../org/biojava/nbio/structure/symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a>&nbsp;axes,
                            <a href="../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
                            <a href="../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Calculates the set of symmetry operation Matrices (transformations) of
 the new alignment, based on the symmetry relations in the SymmetryAxes
 object.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/class-use/StructureException.html" target="_top">Frames</a></li>
<li><a href="StructureException.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
