<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.genome.homology, class: GFF3FromUniprotBlastHits">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> */</span>
<span class="source-line-no">021</span><span id="line-21">package org.biojava.nbio.genome.homology;</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">import org.biojava.nbio.genome.GeneFeatureHelper;</span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.alignment.Alignments;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.alignment.SimpleGapPenalty;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.alignment.matrices.SimpleSubstitutionMatrix;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.alignment.template.SequencePair;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.*;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.core.sequence.features.DBReferenceInfo;</span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface;</span>
<span class="source-line-no">034</span><span id="line-34">import org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface;</span>
<span class="source-line-no">035</span><span id="line-35">import org.biojava.nbio.core.sequence.loader.UniprotProxySequenceReader;</span>
<span class="source-line-no">036</span><span id="line-36">import org.slf4j.Logger;</span>
<span class="source-line-no">037</span><span id="line-37">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">038</span><span id="line-38"></span>
<span class="source-line-no">039</span><span id="line-39">import java.io.File;</span>
<span class="source-line-no">040</span><span id="line-40">import java.io.FileOutputStream;</span>
<span class="source-line-no">041</span><span id="line-41">import java.io.OutputStream;</span>
<span class="source-line-no">042</span><span id="line-42">import java.util.ArrayList;</span>
<span class="source-line-no">043</span><span id="line-43">import java.util.LinkedHashMap;</span>
<span class="source-line-no">044</span><span id="line-44">import java.util.List;</span>
<span class="source-line-no">045</span><span id="line-45">import java.util.Map;</span>
<span class="source-line-no">046</span><span id="line-46"></span>
<span class="source-line-no">047</span><span id="line-47">/**</span>
<span class="source-line-no">048</span><span id="line-48"> *</span>
<span class="source-line-no">049</span><span id="line-49"> * @author Scooter Willis </span>
<span class="source-line-no">050</span><span id="line-50"> * @author Mark Chapman</span>
<span class="source-line-no">051</span><span id="line-51"> */</span>
<span class="source-line-no">052</span><span id="line-52">public class GFF3FromUniprotBlastHits {</span>
<span class="source-line-no">053</span><span id="line-53"></span>
<span class="source-line-no">054</span><span id="line-54">        private static final Logger logger = LoggerFactory.getLogger(GFF3FromUniprotBlastHits.class);</span>
<span class="source-line-no">055</span><span id="line-55"></span>
<span class="source-line-no">056</span><span id="line-56">        public void process(File xmlBlastHits, double ecutoff, LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceHashMap, OutputStream gff3Output) throws Exception {</span>
<span class="source-line-no">057</span><span id="line-57">                LinkedHashMap&lt;String, ArrayList&lt;String&gt;&gt; hits = BlastHomologyHits.getMatches(xmlBlastHits, ecutoff);</span>
<span class="source-line-no">058</span><span id="line-58">                process(hits, geneSequenceHashMap, gff3Output);</span>
<span class="source-line-no">059</span><span id="line-59">        }</span>
<span class="source-line-no">060</span><span id="line-60"></span>
<span class="source-line-no">061</span><span id="line-61">        public void process(LinkedHashMap&lt;String, ArrayList&lt;String&gt;&gt; hits, LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceHashMap, OutputStream gff3Output) throws Exception {</span>
<span class="source-line-no">062</span><span id="line-62">                int size = hits.size();</span>
<span class="source-line-no">063</span><span id="line-63">                int index = 0;</span>
<span class="source-line-no">064</span><span id="line-64">//              HashMap&lt;String, String&gt; scaffoldsReferencedHashMap = new HashMap&lt;String, String&gt;();</span>
<span class="source-line-no">065</span><span id="line-65">                for (String accessionid : hits.keySet()) {</span>
<span class="source-line-no">066</span><span id="line-66">                        index++;</span>
<span class="source-line-no">067</span><span id="line-67">                        if (index == 12) {</span>
<span class="source-line-no">068</span><span id="line-68">                                index = 12;</span>
<span class="source-line-no">069</span><span id="line-69">                        }</span>
<span class="source-line-no">070</span><span id="line-70">                        logger.error(accessionid + " " + index + "/" + size);</span>
<span class="source-line-no">071</span><span id="line-71">                        try {</span>
<span class="source-line-no">072</span><span id="line-72"></span>
<span class="source-line-no">073</span><span id="line-73">                                String[] data = accessionid.split(" ");</span>
<span class="source-line-no">074</span><span id="line-74">                                String id = data[0];</span>
<span class="source-line-no">075</span><span id="line-75">                                GeneSequence geneSequence = geneSequenceHashMap.get(id);</span>
<span class="source-line-no">076</span><span id="line-76">                                if (geneSequence == null) {</span>
<span class="source-line-no">077</span><span id="line-77">                                        logger.error("Not found " + id);</span>
<span class="source-line-no">078</span><span id="line-78">                                        continue;</span>
<span class="source-line-no">079</span><span id="line-79">                                }</span>
<span class="source-line-no">080</span><span id="line-80">                                ArrayList&lt;String&gt; uniprotProteinHits = hits.get(accessionid);</span>
<span class="source-line-no">081</span><span id="line-81">                                String uniprotBestHit = uniprotProteinHits.get(0);</span>
<span class="source-line-no">082</span><span id="line-82">                                UniprotProxySequenceReader&lt;AminoAcidCompound&gt; uniprotSequence = new UniprotProxySequenceReader&lt;&gt;(uniprotBestHit, AminoAcidCompoundSet.getAminoAcidCompoundSet());</span>
<span class="source-line-no">083</span><span id="line-83"></span>
<span class="source-line-no">084</span><span id="line-84">                                ProteinSequence proteinSequence = new ProteinSequence(uniprotSequence);</span>
<span class="source-line-no">085</span><span id="line-85">                                String hitSequence = proteinSequence.getSequenceAsString();</span>
<span class="source-line-no">086</span><span id="line-86">                                for (TranscriptSequence transcriptSequence : geneSequence.getTranscripts().values()) {</span>
<span class="source-line-no">087</span><span id="line-87"></span>
<span class="source-line-no">088</span><span id="line-88"></span>
<span class="source-line-no">089</span><span id="line-89">                                        String predictedProteinSequence = transcriptSequence.getProteinSequence().getSequenceAsString();</span>
<span class="source-line-no">090</span><span id="line-90">                                        List&lt;ProteinSequence&gt; cdsProteinList = transcriptSequence.getProteinCDSSequences();</span>
<span class="source-line-no">091</span><span id="line-91"></span>
<span class="source-line-no">092</span><span id="line-92">                                        ArrayList&lt;CDSSequence&gt; cdsSequenceList = new ArrayList&lt;&gt;(transcriptSequence.getCDSSequences().values());</span>
<span class="source-line-no">093</span><span id="line-93">                                        String testSequence = "";</span>
<span class="source-line-no">094</span><span id="line-94">                                        for (ProteinSequence cdsProteinSequence : cdsProteinList) {</span>
<span class="source-line-no">095</span><span id="line-95">                                                testSequence = testSequence + cdsProteinSequence.getSequenceAsString();</span>
<span class="source-line-no">096</span><span id="line-96">                                        }</span>
<span class="source-line-no">097</span><span id="line-97">                                        if (!testSequence.equals(predictedProteinSequence) &amp;&amp; (!predictedProteinSequence.equals(testSequence.substring(0, testSequence.length() - 1)))) {</span>
<span class="source-line-no">098</span><span id="line-98">                                                DNASequence codingSequence = transcriptSequence.getDNACodingSequence();</span>
<span class="source-line-no">099</span><span id="line-99">                                                logger.info("Coding Sequence: {}", codingSequence.getSequenceAsString());</span>
<span class="source-line-no">100</span><span id="line-100">                                                logger.info("Sequence agreement error");</span>
<span class="source-line-no">101</span><span id="line-101">                                                logger.info("CDS seq={}", testSequence);</span>
<span class="source-line-no">102</span><span id="line-102">                                                logger.info("PRE seq={}", predictedProteinSequence);</span>
<span class="source-line-no">103</span><span id="line-103">                                                logger.info("UNI seq={}", hitSequence);</span>
<span class="source-line-no">104</span><span id="line-104">                                                //  throw new Exception("Protein Sequence compare error " + id);</span>
<span class="source-line-no">105</span><span id="line-105">                                        }</span>
<span class="source-line-no">106</span><span id="line-106"></span>
<span class="source-line-no">107</span><span id="line-107">                                        SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; alignment = Alignments.getPairwiseAlignment(</span>
<span class="source-line-no">108</span><span id="line-108">                                                        transcriptSequence.getProteinSequence(), proteinSequence,</span>
<span class="source-line-no">109</span><span id="line-109">                                                        PairwiseSequenceAlignerType.LOCAL, new SimpleGapPenalty(),</span>
<span class="source-line-no">110</span><span id="line-110">                                                        SimpleSubstitutionMatrix.getBlosum62()</span>
<span class="source-line-no">111</span><span id="line-111">                                                        );</span>
<span class="source-line-no">112</span><span id="line-112">                                        // System.out.println();</span>
<span class="source-line-no">113</span><span id="line-113">                                        //    System.out.println(alignment.getSummary());</span>
<span class="source-line-no">114</span><span id="line-114">                                        //   System.out.println(new Pair().format(alignment));</span>
<span class="source-line-no">115</span><span id="line-115">                                        int proteinIndex = 0;</span>
<span class="source-line-no">116</span><span id="line-116">                                        int gff3Index = 0;</span>
<span class="source-line-no">117</span><span id="line-117">                                        for (int i = 0; i &lt; cdsProteinList.size(); i++) {</span>
<span class="source-line-no">118</span><span id="line-118">                                                ProteinSequence peptideSequence = cdsProteinList.get(i);</span>
<span class="source-line-no">119</span><span id="line-119">                                                String seq = peptideSequence.getSequenceAsString();</span>
<span class="source-line-no">120</span><span id="line-120">                                                Integer startIndex = null;</span>
<span class="source-line-no">121</span><span id="line-121">                                                int offsetStartIndex = 0;</span>
<span class="source-line-no">122</span><span id="line-122">                                                for (int s = 0; s &lt; seq.length(); s++) {</span>
<span class="source-line-no">123</span><span id="line-123">                                                        startIndex = alignment.getIndexInTargetForQueryAt(proteinIndex + s);</span>
<span class="source-line-no">124</span><span id="line-124">                                                        if (startIndex != null) {</span>
<span class="source-line-no">125</span><span id="line-125">                                                                startIndex = startIndex + 1;</span>
<span class="source-line-no">126</span><span id="line-126">                                                                offsetStartIndex = s;</span>
<span class="source-line-no">127</span><span id="line-127">                                                                break;</span>
<span class="source-line-no">128</span><span id="line-128">                                                        }</span>
<span class="source-line-no">129</span><span id="line-129">                                                }</span>
<span class="source-line-no">130</span><span id="line-130">                                                Integer endIndex = null;</span>
<span class="source-line-no">131</span><span id="line-131"></span>
<span class="source-line-no">132</span><span id="line-132">                                                int offsetEndIndex = 0;</span>
<span class="source-line-no">133</span><span id="line-133">                                                for (int e = 0; e &lt; seq.length(); e++) {</span>
<span class="source-line-no">134</span><span id="line-134">                                                        endIndex = alignment.getIndexInTargetForQueryAt(proteinIndex + seq.length() - 1 - e);</span>
<span class="source-line-no">135</span><span id="line-135">                                                        if (endIndex != null) {</span>
<span class="source-line-no">136</span><span id="line-136">                                                                endIndex = endIndex + 1;</span>
<span class="source-line-no">137</span><span id="line-137">                                                                offsetEndIndex = e;</span>
<span class="source-line-no">138</span><span id="line-138">                                                                break;</span>
<span class="source-line-no">139</span><span id="line-139">                                                        }</span>
<span class="source-line-no">140</span><span id="line-140">                                                }</span>
<span class="source-line-no">141</span><span id="line-141"></span>
<span class="source-line-no">142</span><span id="line-142">                                                proteinIndex = proteinIndex + seq.length();</span>
<span class="source-line-no">143</span><span id="line-143">                                                if (startIndex != null &amp;&amp; endIndex != null &amp;&amp; !startIndex.equals(endIndex)) {</span>
<span class="source-line-no">144</span><span id="line-144">                                                        CDSSequence cdsSequence = cdsSequenceList.get(i);</span>
<span class="source-line-no">145</span><span id="line-145">                                                        String hitLabel = "";</span>
<span class="source-line-no">146</span><span id="line-146">                                                        if (transcriptSequence.getStrand() == Strand.POSITIVE) {</span>
<span class="source-line-no">147</span><span id="line-147">                                                                hitLabel = uniprotBestHit + "_" + startIndex + "_" + endIndex;</span>
<span class="source-line-no">148</span><span id="line-148">                                                        } else {</span>
<span class="source-line-no">149</span><span id="line-149">                                                                hitLabel = uniprotBestHit + "_" + endIndex + "_" + startIndex;</span>
<span class="source-line-no">150</span><span id="line-150">                                                        }</span>
<span class="source-line-no">151</span><span id="line-151">                                                        int dnaBeginIndex = cdsSequence.getBioBegin() + (3 * offsetStartIndex);</span>
<span class="source-line-no">152</span><span id="line-152">                                                        int dnaEndIndex = cdsSequence.getBioEnd() - (3 * offsetEndIndex);</span>
<span class="source-line-no">153</span><span id="line-153">                                                        String scaffold = geneSequence.getParentChromosomeSequence().getAccession().getID();</span>
<span class="source-line-no">154</span><span id="line-154">                                        //        if (scaffoldsReferencedHashMap.containsKey(scaffold) == false) {</span>
<span class="source-line-no">155</span><span id="line-155">                                        //            String gff3line = scaffold + "\t" + geneSequence.getSource() + "\t" + "size" + "\t" + "1" + "\t" + geneSequence.getParentChromosomeSequence().getBioEnd() + "\t.\t.\t.\tName=" + scaffold + "\r\n";</span>
<span class="source-line-no">156</span><span id="line-156">                                        //            gff3Output.write(gff3line.getBytes());</span>
<span class="source-line-no">157</span><span id="line-157">                                        //            scaffoldsReferencedHashMap.put(scaffold, scaffold);</span>
<span class="source-line-no">158</span><span id="line-158">                                        //        }</span>
<span class="source-line-no">159</span><span id="line-159"></span>
<span class="source-line-no">160</span><span id="line-160">                                                        String line = scaffold + "\t" + geneSequence.getSource() + "_" + "UNIPROT\tmatch\t" + dnaBeginIndex + "\t" + dnaEndIndex + "\t.\t" + transcriptSequence.getStrand().getStringRepresentation() + "\t.\t";</span>
<span class="source-line-no">161</span><span id="line-161">                                                        if (gff3Index == 0) {</span>
<span class="source-line-no">162</span><span id="line-162">                                                                FeaturesKeyWordInterface featureKeyWords = proteinSequence.getFeaturesKeyWord();</span>
<span class="source-line-no">163</span><span id="line-163">                                                                String notes = "";</span>
<span class="source-line-no">164</span><span id="line-164">                                                                if (featureKeyWords != null) {</span>
<span class="source-line-no">165</span><span id="line-165">                                                                        List&lt;String&gt; keyWords = featureKeyWords.getKeyWords();</span>
<span class="source-line-no">166</span><span id="line-166">                                                                        if (keyWords.size() &gt; 0) {</span>
<span class="source-line-no">167</span><span id="line-167">                                                                                notes = ";Note=";</span>
<span class="source-line-no">168</span><span id="line-168">                                                                                for (String note : keyWords) {</span>
<span class="source-line-no">169</span><span id="line-169">                                                                                        if ("Complete proteome".equals(note)) {</span>
<span class="source-line-no">170</span><span id="line-170">                                                                                                continue;</span>
<span class="source-line-no">171</span><span id="line-171">                                                                                        }</span>
<span class="source-line-no">172</span><span id="line-172">                                                                                        if ("Direct protein sequencing".equals(note)) {</span>
<span class="source-line-no">173</span><span id="line-173">                                                                                                continue;</span>
<span class="source-line-no">174</span><span id="line-174">                                                                                        }</span>
<span class="source-line-no">175</span><span id="line-175"></span>
<span class="source-line-no">176</span><span id="line-176">                                                                                        notes = notes + " " + note;</span>
<span class="source-line-no">177</span><span id="line-177">                                                                                        geneSequence.addNote(note); // add note/keyword which can be output in fasta header if needed</span>
<span class="source-line-no">178</span><span id="line-178">                                                                                }</span>
<span class="source-line-no">179</span><span id="line-179">                                                                        }</span>
<span class="source-line-no">180</span><span id="line-180"></span>
<span class="source-line-no">181</span><span id="line-181">                                                                }</span>
<span class="source-line-no">182</span><span id="line-182"></span>
<span class="source-line-no">183</span><span id="line-183">                                                                DatabaseReferenceInterface databaseReferences = proteinSequence.getDatabaseReferences();</span>
<span class="source-line-no">184</span><span id="line-184">                                                                if (databaseReferences != null) {</span>
<span class="source-line-no">185</span><span id="line-185">                                                                        Map&lt;String, List&lt;DBReferenceInfo&gt;&gt; databaseReferenceHashMap = databaseReferences.getDatabaseReferences();</span>
<span class="source-line-no">186</span><span id="line-186">                                                                        List&lt;DBReferenceInfo&gt; pfamList = databaseReferenceHashMap.get("Pfam");</span>
<span class="source-line-no">187</span><span id="line-187">                                                                        List&lt;DBReferenceInfo&gt; cazyList = databaseReferenceHashMap.get("CAZy");</span>
<span class="source-line-no">188</span><span id="line-188">                                                                        List&lt;DBReferenceInfo&gt; goList = databaseReferenceHashMap.get("GO");</span>
<span class="source-line-no">189</span><span id="line-189">                                                                        List&lt;DBReferenceInfo&gt; eccList = databaseReferenceHashMap.get("BRENDA");</span>
<span class="source-line-no">190</span><span id="line-190">                                                                        if (pfamList != null &amp;&amp; pfamList.size() &gt; 0) {</span>
<span class="source-line-no">191</span><span id="line-191">                                                                                if (notes.length() == 0) {</span>
<span class="source-line-no">192</span><span id="line-192">                                                                                        notes = ";Note=";</span>
<span class="source-line-no">193</span><span id="line-193">                                                                                }</span>
<span class="source-line-no">194</span><span id="line-194">                                                                                for (DBReferenceInfo note : pfamList) {</span>
<span class="source-line-no">195</span><span id="line-195">                                                                                        notes = notes + " " + note.getId();</span>
<span class="source-line-no">196</span><span id="line-196">                                                                                        geneSequence.addNote(note.getId()); // add note/keyword which can be output in fasta header if needed</span>
<span class="source-line-no">197</span><span id="line-197">                                                                                }</span>
<span class="source-line-no">198</span><span id="line-198">                                                                        }</span>
<span class="source-line-no">199</span><span id="line-199"></span>
<span class="source-line-no">200</span><span id="line-200">                                                                        if (cazyList != null &amp;&amp; cazyList.size() &gt; 0) {</span>
<span class="source-line-no">201</span><span id="line-201">                                                                                if (notes.length() == 0) {</span>
<span class="source-line-no">202</span><span id="line-202">                                                                                        notes = ";Note=";</span>
<span class="source-line-no">203</span><span id="line-203">                                                                                }</span>
<span class="source-line-no">204</span><span id="line-204">                                                                                for (DBReferenceInfo note : cazyList) {</span>
<span class="source-line-no">205</span><span id="line-205">                                                                                        notes = notes + " " + note.getId();</span>
<span class="source-line-no">206</span><span id="line-206">                                                                                        geneSequence.addNote(note.getId()); // add note/keyword which can be output in fasta header if needed</span>
<span class="source-line-no">207</span><span id="line-207">                                                                                        // System.out.println("CAZy=" + note);</span>
<span class="source-line-no">208</span><span id="line-208">                                                                                }</span>
<span class="source-line-no">209</span><span id="line-209">                                                                        }</span>
<span class="source-line-no">210</span><span id="line-210"></span>
<span class="source-line-no">211</span><span id="line-211">                                                                        if (eccList != null &amp;&amp; eccList.size() &gt; 0) {</span>
<span class="source-line-no">212</span><span id="line-212">                                                                                if (notes.length() == 0) {</span>
<span class="source-line-no">213</span><span id="line-213">                                                                                        notes = ";Note=";</span>
<span class="source-line-no">214</span><span id="line-214">                                                                                }</span>
<span class="source-line-no">215</span><span id="line-215">                                                                                for (DBReferenceInfo note : eccList) {</span>
<span class="source-line-no">216</span><span id="line-216">                                                                                        String dbid = note.getId();</span>
<span class="source-line-no">217</span><span id="line-217">                                                                                        dbid = dbid.replace(".", "_"); //replace . with _ to facilitate searching in gbrowse</span>
<span class="source-line-no">218</span><span id="line-218">                                                                                        notes = notes + " " + "EC:" + dbid;</span>
<span class="source-line-no">219</span><span id="line-219">                                                                                        geneSequence.addNote("EC:" + dbid); // add note/keyword which can be output in fasta header if needed</span>
<span class="source-line-no">220</span><span id="line-220"></span>
<span class="source-line-no">221</span><span id="line-221">                                                                                }</span>
<span class="source-line-no">222</span><span id="line-222">                                                                        }</span>
<span class="source-line-no">223</span><span id="line-223"></span>
<span class="source-line-no">224</span><span id="line-224">                                                                        if (goList != null &amp;&amp; goList.size() &gt; 0) {</span>
<span class="source-line-no">225</span><span id="line-225">                                                                                if (notes.length() == 0) {</span>
<span class="source-line-no">226</span><span id="line-226">                                                                                        notes = ";Note=";</span>
<span class="source-line-no">227</span><span id="line-227">                                                                                }</span>
<span class="source-line-no">228</span><span id="line-228">                                                                                for (DBReferenceInfo note : goList) {</span>
<span class="source-line-no">229</span><span id="line-229">                                                                                        notes = notes + " " + note.getId();</span>
<span class="source-line-no">230</span><span id="line-230">                                                                                        geneSequence.addNote(note.getId()); // add note/keyword which can be output in fasta header if needed</span>
<span class="source-line-no">231</span><span id="line-231">                                                                                        Map&lt;String, String&gt; properties = note.getProperties();</span>
<span class="source-line-no">232</span><span id="line-232">                                                                                        for (String propertytype : properties.keySet()) {</span>
<span class="source-line-no">233</span><span id="line-233">                                                                                                if ("evidence".equals(propertytype)) {</span>
<span class="source-line-no">234</span><span id="line-234">                                                                                                        continue;</span>
<span class="source-line-no">235</span><span id="line-235">                                                                                                }</span>
<span class="source-line-no">236</span><span id="line-236">                                                                                                String property = properties.get(propertytype);</span>
<span class="source-line-no">237</span><span id="line-237"></span>
<span class="source-line-no">238</span><span id="line-238">                                                                                                if (property.startsWith("C:")) {</span>
<span class="source-line-no">239</span><span id="line-239">                                                                                                        continue; // skip over the location</span>
<span class="source-line-no">240</span><span id="line-240">                                                                                                }</span>
<span class="source-line-no">241</span><span id="line-241">                                                                                                if (property.endsWith("...")) {</span>
<span class="source-line-no">242</span><span id="line-242">                                                                                                        property = property.substring(0, property.length() - 3);</span>
<span class="source-line-no">243</span><span id="line-243">                                                                                                }</span>
<span class="source-line-no">244</span><span id="line-244">                                                                                                notes = notes + " " + property;</span>
<span class="source-line-no">245</span><span id="line-245">                                                                                                geneSequence.addNote(property);</span>
<span class="source-line-no">246</span><span id="line-246">                                                                                        }</span>
<span class="source-line-no">247</span><span id="line-247">                                                                                }</span>
<span class="source-line-no">248</span><span id="line-248">                                                                        }</span>
<span class="source-line-no">249</span><span id="line-249"></span>
<span class="source-line-no">250</span><span id="line-250">                                                                }</span>
<span class="source-line-no">251</span><span id="line-251"></span>
<span class="source-line-no">252</span><span id="line-252"></span>
<span class="source-line-no">253</span><span id="line-253">                                                                line = line + "Name=" + hitLabel + ";Alias=" + uniprotBestHit + notes + "\n";</span>
<span class="source-line-no">254</span><span id="line-254">                                                        } else {</span>
<span class="source-line-no">255</span><span id="line-255">                                                                line = line + "Name=" + hitLabel + "\n";</span>
<span class="source-line-no">256</span><span id="line-256">                                                        }</span>
<span class="source-line-no">257</span><span id="line-257">                                                        gff3Index++;</span>
<span class="source-line-no">258</span><span id="line-258"></span>
<span class="source-line-no">259</span><span id="line-259">                                                        gff3Output.write(line.getBytes());</span>
<span class="source-line-no">260</span><span id="line-260">                                                }</span>
<span class="source-line-no">261</span><span id="line-261">                                        }</span>
<span class="source-line-no">262</span><span id="line-262">                                }</span>
<span class="source-line-no">263</span><span id="line-263">                        } catch (Exception e) {</span>
<span class="source-line-no">264</span><span id="line-264">                                logger.info("Accession Id: {}", accessionid, e);</span>
<span class="source-line-no">265</span><span id="line-265">                        }</span>
<span class="source-line-no">266</span><span id="line-266">                }</span>
<span class="source-line-no">267</span><span id="line-267"></span>
<span class="source-line-no">268</span><span id="line-268"></span>
<span class="source-line-no">269</span><span id="line-269"></span>
<span class="source-line-no">270</span><span id="line-270">        }</span>
<span class="source-line-no">271</span><span id="line-271"></span>
<span class="source-line-no">272</span><span id="line-272"></span>
<span class="source-line-no">273</span><span id="line-273"></span>
<span class="source-line-no">274</span><span id="line-274"></span>
<span class="source-line-no">275</span><span id="line-275">        public static void main(String[] args) {</span>
<span class="source-line-no">276</span><span id="line-276">                /*</span>
<span class="source-line-no">277</span><span id="line-277">                        try {</span>
<span class="source-line-no">278</span><span id="line-278">                                LinkedHashMap&lt;String, ChromosomeSequence&gt; dnaSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGeneMarkGTF(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/454Scaffolds.fna"), new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/genemark_hmm.gtf"));</span>
<span class="source-line-no">279</span><span id="line-279">                                LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceList = GeneFeatureHelper.getGeneSequences(dnaSequenceList.values());</span>
<span class="source-line-no">280</span><span id="line-280">                                FileOutputStream fo = new FileOutputStream("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/genemark_uniprot_match.gff3");</span>
<span class="source-line-no">281</span><span id="line-281"></span>
<span class="source-line-no">282</span><span id="line-282">                                GFF3FromUniprotBlastHits gff3FromUniprotBlastHits = new GFF3FromUniprotBlastHits();</span>
<span class="source-line-no">283</span><span id="line-283">                                gff3FromUniprotBlastHits.process(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/c1-454Scaffolds-hits-uniprot_fungi.xml"), 1E-10, geneSequenceList, fo);</span>
<span class="source-line-no">284</span><span id="line-284">                                fo.close();</span>
<span class="source-line-no">285</span><span id="line-285"></span>
<span class="source-line-no">286</span><span id="line-286"></span>
<span class="source-line-no">287</span><span id="line-287">                        } catch (Exception e) {</span>
<span class="source-line-no">288</span><span id="line-288">                                logger.error("Exception: ", e);</span>
<span class="source-line-no">289</span><span id="line-289"></span>
<span class="source-line-no">290</span><span id="line-290"></span>
<span class="source-line-no">291</span><span id="line-291">                        }</span>
<span class="source-line-no">292</span><span id="line-292">                */</span>
<span class="source-line-no">293</span><span id="line-293"></span>
<span class="source-line-no">294</span><span id="line-294">                        try {</span>
<span class="source-line-no">295</span><span id="line-295">                                Map&lt;String, ChromosomeSequence&gt; dnaSequenceHashMap = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGlimmerGFF3(new File("/Users/Scooter/scripps/dyadic/analysis/454Scaffolds/454Scaffolds-16.fna"), new File("/Users/Scooter/scripps/dyadic/GlimmerHMM/c1_glimmerhmm-16.gff"));</span>
<span class="source-line-no">296</span><span id="line-296">                                LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceList = GeneFeatureHelper.getGeneSequences(dnaSequenceHashMap.values());</span>
<span class="source-line-no">297</span><span id="line-297">                                FileOutputStream fo = new FileOutputStream("/Users/Scooter/scripps/dyadic/outputGlimmer/genemark_uniprot_match-16.gff3");</span>
<span class="source-line-no">298</span><span id="line-298">                                LinkedHashMap&lt;String, ArrayList&lt;String&gt;&gt; blasthits = BlastHomologyHits.getMatches(new File("/Users/Scooter/scripps/dyadic/blastresults/c1_glimmer_in_uniprot.xml"), 1E-10);</span>
<span class="source-line-no">299</span><span id="line-299">                                logger.error("Number of uniprot hits " + blasthits.size());</span>
<span class="source-line-no">300</span><span id="line-300"></span>
<span class="source-line-no">301</span><span id="line-301">                                GFF3FromUniprotBlastHits gff3FromUniprotBlastHits = new GFF3FromUniprotBlastHits();</span>
<span class="source-line-no">302</span><span id="line-302">                                gff3FromUniprotBlastHits.process(blasthits, geneSequenceList, fo);</span>
<span class="source-line-no">303</span><span id="line-303">                                fo.close();</span>
<span class="source-line-no">304</span><span id="line-304">                        } catch (Exception e) {</span>
<span class="source-line-no">305</span><span id="line-305">                                logger.error("Exception: ", e);</span>
<span class="source-line-no">306</span><span id="line-306">                        }</span>
<span class="source-line-no">307</span><span id="line-307">        }</span>
<span class="source-line-no">308</span><span id="line-308">}</span>




























































</pre>
</div>
</main>
</body>
</html>
