<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.structure, interface: Chain">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on 25.04.2004</span>
<span class="source-line-no">021</span><span id="line-21"> * @author Andreas Prlic</span>
<span class="source-line-no">022</span><span id="line-22"> *</span>
<span class="source-line-no">023</span><span id="line-23"> */</span>
<span class="source-line-no">024</span><span id="line-24">package org.biojava.nbio.structure;</span>
<span class="source-line-no">025</span><span id="line-25"></span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.template.Sequence;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.structure.chem.ChemComp;</span>
<span class="source-line-no">028</span><span id="line-28"></span>
<span class="source-line-no">029</span><span id="line-29">import java.io.Serializable;</span>
<span class="source-line-no">030</span><span id="line-30">import java.util.List;</span>
<span class="source-line-no">031</span><span id="line-31"></span>
<span class="source-line-no">032</span><span id="line-32">/**</span>
<span class="source-line-no">033</span><span id="line-33"> * &lt;p&gt;</span>
<span class="source-line-no">034</span><span id="line-34"> * Defines the interface for a Chain. A Chain corresponds to a Chain in a PDB file.</span>
<span class="source-line-no">035</span><span id="line-35"> * A chain consists of a list of {@link Group} objects. A Group can either be</span>
<span class="source-line-no">036</span><span id="line-36"> * an {@link AminoAcid}, {@link HetatomImpl Hetatom} or {@link NucleotideImpl Nucleotide}.</span>
<span class="source-line-no">037</span><span id="line-37"> * &lt;/p&gt;</span>
<span class="source-line-no">038</span><span id="line-38"> *</span>
<span class="source-line-no">039</span><span id="line-39"> * &lt;p&gt;</span>
<span class="source-line-no">040</span><span id="line-40"> * The BioJava API provides access to both the ATOM and SEQRES records in a PDB file.</span>
<span class="source-line-no">041</span><span id="line-41"> * During parsing of a PDB file it aligns the ATOM and SEQRES groups and joins them.</span>
<span class="source-line-no">042</span><span id="line-42"> * The SEQRES sequence can be accessed via  {@link #getSeqResGroups()} and the</span>
<span class="source-line-no">043</span><span id="line-43"> * ATOM groups via {@link #getAtomGroups()}. Groups that have been observed</span>
<span class="source-line-no">044</span><span id="line-44"> * (i.e. they are in the ATOM records) can be detected by {@link Group}.has3D()</span>
<span class="source-line-no">045</span><span id="line-45"> *  &lt;/p&gt;</span>
<span class="source-line-no">046</span><span id="line-46"> *</span>
<span class="source-line-no">047</span><span id="line-47"> * @author Andreas Prlic</span>
<span class="source-line-no">048</span><span id="line-48"> * @version %I% %G%</span>
<span class="source-line-no">049</span><span id="line-49"> * @since 1.4</span>
<span class="source-line-no">050</span><span id="line-50"> */</span>
<span class="source-line-no">051</span><span id="line-51">public interface Chain extends Serializable {</span>
<span class="source-line-no">052</span><span id="line-52"></span>
<span class="source-line-no">053</span><span id="line-53">        /**</span>
<span class="source-line-no">054</span><span id="line-54">         * Returns an identical copy of this Chain.</span>
<span class="source-line-no">055</span><span id="line-55">         * @return  an identical copy of this Chain</span>
<span class="source-line-no">056</span><span id="line-56">         */</span>
<span class="source-line-no">057</span><span id="line-57">        Object clone();</span>
<span class="source-line-no">058</span><span id="line-58"></span>
<span class="source-line-no">059</span><span id="line-59">        /**</span>
<span class="source-line-no">060</span><span id="line-60">         * Add a group to the list of ATOM record group of this chain.</span>
<span class="source-line-no">061</span><span id="line-61">         * To add SEQRES records a more complex alignment between ATOM and SEQRES residues</span>
<span class="source-line-no">062</span><span id="line-62">         * is required, please see SeqRes2AtomAligner for more details on that.</span>
<span class="source-line-no">063</span><span id="line-63">         * @param group  a Group object</span>
<span class="source-line-no">064</span><span id="line-64">         */</span>
<span class="source-line-no">065</span><span id="line-65">        void addGroup(Group group);</span>
<span class="source-line-no">066</span><span id="line-66"></span>
<span class="source-line-no">067</span><span id="line-67">        /**</span>
<span class="source-line-no">068</span><span id="line-68">         * Get the 'private' asymId (internal chain IDs in mmCif) for this chain.</span>
<span class="source-line-no">069</span><span id="line-69">         *</span>
<span class="source-line-no">070</span><span id="line-70">         * @return the asymId</span>
<span class="source-line-no">071</span><span id="line-71">         * @see #setId(String)</span>
<span class="source-line-no">072</span><span id="line-72">         * @see #getName()</span>
<span class="source-line-no">073</span><span id="line-73">         */</span>
<span class="source-line-no">074</span><span id="line-74">        String getId() ;</span>
<span class="source-line-no">075</span><span id="line-75"></span>
<span class="source-line-no">076</span><span id="line-76"></span>
<span class="source-line-no">077</span><span id="line-77">        /**</span>
<span class="source-line-no">078</span><span id="line-78">         * Set the 'private' asymId (internal chain IDs in mmCif) for this chain.</span>
<span class="source-line-no">079</span><span id="line-79">         *</span>
<span class="source-line-no">080</span><span id="line-80">         * @param asymId the internal chain Id</span>
<span class="source-line-no">081</span><span id="line-81">         */</span>
<span class="source-line-no">082</span><span id="line-82">        void setId(String asymId) ;</span>
<span class="source-line-no">083</span><span id="line-83"></span>
<span class="source-line-no">084</span><span id="line-84"></span>
<span class="source-line-no">085</span><span id="line-85">        /**</span>
<span class="source-line-no">086</span><span id="line-86">         * Set the 'public' authId (chain ID in PDB file)</span>
<span class="source-line-no">087</span><span id="line-87">         *</span>
<span class="source-line-no">088</span><span id="line-88">         * @param authId the 'public' authId (chain ID in PDB file)</span>
<span class="source-line-no">089</span><span id="line-89">         * @see #getId()</span>
<span class="source-line-no">090</span><span id="line-90">         */</span>
<span class="source-line-no">091</span><span id="line-91">        void setName(String authId);</span>
<span class="source-line-no">092</span><span id="line-92"></span>
<span class="source-line-no">093</span><span id="line-93">        /**</span>
<span class="source-line-no">094</span><span id="line-94">         * Get the 'public' authId (chain ID in PDB file)</span>
<span class="source-line-no">095</span><span id="line-95">         *</span>
<span class="source-line-no">096</span><span id="line-96">         * @return the authId for this chain.</span>
<span class="source-line-no">097</span><span id="line-97">         * @see #getId()</span>
<span class="source-line-no">098</span><span id="line-98">         */</span>
<span class="source-line-no">099</span><span id="line-99">        String getName();</span>
<span class="source-line-no">100</span><span id="line-100"></span>
<span class="source-line-no">101</span><span id="line-101"></span>
<span class="source-line-no">102</span><span id="line-102">        /**</span>
<span class="source-line-no">103</span><span id="line-103">         * Return the Group at given position,</span>
<span class="source-line-no">104</span><span id="line-104">         * from within Groups with observed density in the chain, i.e.</span>
<span class="source-line-no">105</span><span id="line-105">         * those with coordinates in ATOM and HETATMS (including waters) records.</span>
<span class="source-line-no">106</span><span id="line-106">         * @param position  an int</span>
<span class="source-line-no">107</span><span id="line-107">         * @return a Group object</span>
<span class="source-line-no">108</span><span id="line-108">         * @see #getAtomLength()</span>
<span class="source-line-no">109</span><span id="line-109">         * @see #getAtomGroups()</span>
<span class="source-line-no">110</span><span id="line-110">         * @see #getSeqResGroup(int)</span>
<span class="source-line-no">111</span><span id="line-111">         */</span>
<span class="source-line-no">112</span><span id="line-112">        Group getAtomGroup (int position);</span>
<span class="source-line-no">113</span><span id="line-113"></span>
<span class="source-line-no">114</span><span id="line-114">        /**</span>
<span class="source-line-no">115</span><span id="line-115">         * Return the Group at given position,</span>
<span class="source-line-no">116</span><span id="line-116">         * from within groups in the SEQRES records of the chain, i.e.</span>
<span class="source-line-no">117</span><span id="line-117">         * the aminoacids/nucleotides in the construct.</span>
<span class="source-line-no">118</span><span id="line-118">         * @param position  an int</span>
<span class="source-line-no">119</span><span id="line-119">         * @return a Group object</span>
<span class="source-line-no">120</span><span id="line-120">         * @see #getSeqResLength()</span>
<span class="source-line-no">121</span><span id="line-121">         * @see #getSeqResGroups()</span>
<span class="source-line-no">122</span><span id="line-122">         * @see #getAtomGroup(int)</span>
<span class="source-line-no">123</span><span id="line-123">         */</span>
<span class="source-line-no">124</span><span id="line-124">        Group getSeqResGroup (int position);</span>
<span class="source-line-no">125</span><span id="line-125"></span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">        /**</span>
<span class="source-line-no">128</span><span id="line-128">         * Return all Groups with observed density in the chain, i.e.</span>
<span class="source-line-no">129</span><span id="line-129">         * those with coordinates in ATOM and HETATMS (including waters) records.</span>
<span class="source-line-no">130</span><span id="line-130">         *</span>
<span class="source-line-no">131</span><span id="line-131">         * @return a List object representing the Groups of this Chain.</span>
<span class="source-line-no">132</span><span id="line-132">         * @see #setAtomGroups(List)</span>
<span class="source-line-no">133</span><span id="line-133">         * @see #getAtomLength()</span>
<span class="source-line-no">134</span><span id="line-134">         * @see #getSeqResGroups()</span>
<span class="source-line-no">135</span><span id="line-135">         */</span>
<span class="source-line-no">136</span><span id="line-136">        List&lt;Group&gt; getAtomGroups();</span>
<span class="source-line-no">137</span><span id="line-137"></span>
<span class="source-line-no">138</span><span id="line-138">        /**</span>
<span class="source-line-no">139</span><span id="line-139">         * Set all Groups with observed density in the chain, i.e.</span>
<span class="source-line-no">140</span><span id="line-140">         * those with coordinates in ATOM and HETATMs (including waters) records.</span>
<span class="source-line-no">141</span><span id="line-141">         * @param groups a List object representing the Groups of this Chain.</span>
<span class="source-line-no">142</span><span id="line-142">         * @see #getAtomGroups()</span>
<span class="source-line-no">143</span><span id="line-143">         */</span>
<span class="source-line-no">144</span><span id="line-144">        void setAtomGroups(List&lt;Group&gt; groups);</span>
<span class="source-line-no">145</span><span id="line-145"></span>
<span class="source-line-no">146</span><span id="line-146">        /**</span>
<span class="source-line-no">147</span><span id="line-147">         * Return a List of all (observed) Groups of a special type, one of: {@link GroupType#AMINOACID},</span>
<span class="source-line-no">148</span><span id="line-148">         * {@link GroupType#HETATM} or {@link GroupType#NUCLEOTIDE}.</span>
<span class="source-line-no">149</span><span id="line-149">         * Note that if a standard aminoacid appears as a HETATM (because it is part of a ligand) then</span>
<span class="source-line-no">150</span><span id="line-150">         * it is still considered as {@link GroupType#AMINOACID} and not as {@link GroupType#HETATM}.</span>
<span class="source-line-no">151</span><span id="line-151">         * @param type  GroupType</span>
<span class="source-line-no">152</span><span id="line-152">         * @return a List object</span>
<span class="source-line-no">153</span><span id="line-153">         * @see #setAtomGroups(List)</span>
<span class="source-line-no">154</span><span id="line-154">         */</span>
<span class="source-line-no">155</span><span id="line-155">        List&lt;Group&gt; getAtomGroups (GroupType type);</span>
<span class="source-line-no">156</span><span id="line-156"></span>
<span class="source-line-no">157</span><span id="line-157"></span>
<span class="source-line-no">158</span><span id="line-158">        /**</span>
<span class="source-line-no">159</span><span id="line-159">         * Get a group by its PDB residue numbering. If the PDB residue number is not known,</span>
<span class="source-line-no">160</span><span id="line-160">         * throws a StructureException.</span>
<span class="source-line-no">161</span><span id="line-161">         *</span>
<span class="source-line-no">162</span><span id="line-162">         * @param resNum the PDB residue number of the group</span>
<span class="source-line-no">163</span><span id="line-163">         * @return the matching group</span>
<span class="source-line-no">164</span><span id="line-164">         * @throws StructureException</span>
<span class="source-line-no">165</span><span id="line-165">         */</span>
<span class="source-line-no">166</span><span id="line-166">        Group getGroupByPDB(ResidueNumber resNum) throws StructureException;</span>
<span class="source-line-no">167</span><span id="line-167"></span>
<span class="source-line-no">168</span><span id="line-168">        /**</span>
<span class="source-line-no">169</span><span id="line-169">         * Get all groups that are located between two PDB residue numbers.</span>
<span class="source-line-no">170</span><span id="line-170">         *</span>
<span class="source-line-no">171</span><span id="line-171">         * @param pdbresnumStart PDB residue number of start. If null, defaults to the chain start.</span>
<span class="source-line-no">172</span><span id="line-172">         * @param pdbresnumEnd PDB residue number of end. If null, defaults to the chain end.</span>
<span class="source-line-no">173</span><span id="line-173">         * @return Groups in between. or throws a StructureException if either start or end can not be found,</span>
<span class="source-line-no">174</span><span id="line-174">         * @throws StructureException</span>
<span class="source-line-no">175</span><span id="line-175">         */</span>
<span class="source-line-no">176</span><span id="line-176">        Group[] getGroupsByPDB(ResidueNumber pdbresnumStart, ResidueNumber pdbresnumEnd) throws StructureException;</span>
<span class="source-line-no">177</span><span id="line-177"></span>
<span class="source-line-no">178</span><span id="line-178"></span>
<span class="source-line-no">179</span><span id="line-179">        /**</span>
<span class="source-line-no">180</span><span id="line-180">         * Get all groups that are located between two PDB residue numbers. In contrast to getGroupsByPDB</span>
<span class="source-line-no">181</span><span id="line-181">         * this method call ignores if the exact outer groups are not found. This is useful e.g. when requesting the range</span>
<span class="source-line-no">182</span><span id="line-182">         * of groups as specified by the DBREF records - these frequently are rather inaccurate.</span>
<span class="source-line-no">183</span><span id="line-183">         *</span>
<span class="source-line-no">184</span><span id="line-184">         *</span>
<span class="source-line-no">185</span><span id="line-185">         * @param pdbresnumStart PDB residue number of start. If null, defaults to the chain start.</span>
<span class="source-line-no">186</span><span id="line-186">         * @param pdbresnumEnd PDB residue number of end. If null, defaults to the chain end.</span>
<span class="source-line-no">187</span><span id="line-187">         * @param ignoreMissing ignore missing groups in this range.</span>
<span class="source-line-no">188</span><span id="line-188">         * @return Groups in between. or throws a StructureException if either start or end can not be found,</span>
<span class="source-line-no">189</span><span id="line-189">         * @throws StructureException</span>
<span class="source-line-no">190</span><span id="line-190">         *</span>
<span class="source-line-no">191</span><span id="line-191">         */</span>
<span class="source-line-no">192</span><span id="line-192">        Group[] getGroupsByPDB(ResidueNumber pdbresnumStart, ResidueNumber pdbresnumEnd,boolean ignoreMissing) throws StructureException;</span>
<span class="source-line-no">193</span><span id="line-193"></span>
<span class="source-line-no">194</span><span id="line-194"></span>
<span class="source-line-no">195</span><span id="line-195">        /**</span>
<span class="source-line-no">196</span><span id="line-196">         * Returns the number of Groups with observed density in the chain, i.e.</span>
<span class="source-line-no">197</span><span id="line-197">         * those with coordinates in ATOM and HETATMs (including waters) records</span>
<span class="source-line-no">198</span><span id="line-198">         *</span>
<span class="source-line-no">199</span><span id="line-199">         * @return the length</span>
<span class="source-line-no">200</span><span id="line-200">         * @see #getAtomGroup(int)</span>
<span class="source-line-no">201</span><span id="line-201">         * @see #getAtomGroups()</span>
<span class="source-line-no">202</span><span id="line-202">         * @see #getSeqResLength()</span>
<span class="source-line-no">203</span><span id="line-203">         */</span>
<span class="source-line-no">204</span><span id="line-204">        int getAtomLength();</span>
<span class="source-line-no">205</span><span id="line-205"></span>
<span class="source-line-no">206</span><span id="line-206">        /**</span>
<span class="source-line-no">207</span><span id="line-207">         * Returns the number of groups in the SEQRES records of the chain, i.e.</span>
<span class="source-line-no">208</span><span id="line-208">         * the number of aminoacids/nucleotides in the construct</span>
<span class="source-line-no">209</span><span id="line-209">         *</span>
<span class="source-line-no">210</span><span id="line-210">         * @return the length</span>
<span class="source-line-no">211</span><span id="line-211">         * @see #getSeqResGroup(int)</span>
<span class="source-line-no">212</span><span id="line-212">         * @see #getSeqResGroups()</span>
<span class="source-line-no">213</span><span id="line-213">         * @see #getAtomLength()</span>
<span class="source-line-no">214</span><span id="line-214">         */</span>
<span class="source-line-no">215</span><span id="line-215">        int getSeqResLength();</span>
<span class="source-line-no">216</span><span id="line-216"></span>
<span class="source-line-no">217</span><span id="line-217">        /**</span>
<span class="source-line-no">218</span><span id="line-218">         * Sets the Entity information</span>
<span class="source-line-no">219</span><span id="line-219">         * @param entityInfo the EntityInfo</span>
<span class="source-line-no">220</span><span id="line-220">         * @see #getEntityInfo()</span>
<span class="source-line-no">221</span><span id="line-221">         */</span>
<span class="source-line-no">222</span><span id="line-222">        void setEntityInfo(EntityInfo entityInfo);</span>
<span class="source-line-no">223</span><span id="line-223"></span>
<span class="source-line-no">224</span><span id="line-224">        /**</span>
<span class="source-line-no">225</span><span id="line-225">         * Returns the EntityInfo for this chain.</span>
<span class="source-line-no">226</span><span id="line-226">         *</span>
<span class="source-line-no">227</span><span id="line-227">         * @return the EntityInfo object</span>
<span class="source-line-no">228</span><span id="line-228">         * @see #setEntityInfo(EntityInfo)</span>
<span class="source-line-no">229</span><span id="line-229">         */</span>
<span class="source-line-no">230</span><span id="line-230">        EntityInfo getEntityInfo();</span>
<span class="source-line-no">231</span><span id="line-231"></span>
<span class="source-line-no">232</span><span id="line-232">        @Override</span>
<span class="source-line-no">233</span><span id="line-233">        String toString();</span>
<span class="source-line-no">234</span><span id="line-234"></span>
<span class="source-line-no">235</span><span id="line-235">        /**</span>
<span class="source-line-no">236</span><span id="line-236">         * Converts the SEQRES groups of a Chain to a Biojava Sequence object.</span>
<span class="source-line-no">237</span><span id="line-237">         *</span>
<span class="source-line-no">238</span><span id="line-238">         * @return the SEQRES groups of the Chain as a Sequence object.</span>
<span class="source-line-no">239</span><span id="line-239">         */</span>
<span class="source-line-no">240</span><span id="line-240">        Sequence&lt;?&gt; getBJSequence()  ;</span>
<span class="source-line-no">241</span><span id="line-241"></span>
<span class="source-line-no">242</span><span id="line-242">        /**</span>
<span class="source-line-no">243</span><span id="line-243">         * Returns the sequence of amino acids as it has been provided in the ATOM records.</span>
<span class="source-line-no">244</span><span id="line-244">         * @return amino acid sequence as string</span>
<span class="source-line-no">245</span><span id="line-245">         * @see #getSeqResSequence()</span>
<span class="source-line-no">246</span><span id="line-246">         */</span>
<span class="source-line-no">247</span><span id="line-247">        String getAtomSequence();</span>
<span class="source-line-no">248</span><span id="line-248"></span>
<span class="source-line-no">249</span><span id="line-249">        /**</span>
<span class="source-line-no">250</span><span id="line-250">         * Returns the PDB SEQRES sequence as a one-letter sequence string.</span>
<span class="source-line-no">251</span><span id="line-251">         * Non-standard residues are represented by an "X".</span>
<span class="source-line-no">252</span><span id="line-252">         * @return one-letter PDB SEQRES sequence as string</span>
<span class="source-line-no">253</span><span id="line-253">         * @see #getAtomSequence()</span>
<span class="source-line-no">254</span><span id="line-254">         */</span>
<span class="source-line-no">255</span><span id="line-255">        String getSeqResSequence();</span>
<span class="source-line-no">256</span><span id="line-256"></span>
<span class="source-line-no">257</span><span id="line-257">        /**</span>
<span class="source-line-no">258</span><span id="line-258">         * Returns a List of all SEQRES groups of a special type, one of: {@link GroupType#AMINOACID},</span>
<span class="source-line-no">259</span><span id="line-259">         * {@link GroupType#HETATM} or {@link GroupType#NUCLEOTIDE}.</span>
<span class="source-line-no">260</span><span id="line-260">         * @param type  a GroupType</span>
<span class="source-line-no">261</span><span id="line-261">         * @return an List object</span>
<span class="source-line-no">262</span><span id="line-262">         * @see #setSeqResGroups(List)</span>
<span class="source-line-no">263</span><span id="line-263">         */</span>
<span class="source-line-no">264</span><span id="line-264">        List&lt;Group&gt; getSeqResGroups (GroupType type);</span>
<span class="source-line-no">265</span><span id="line-265"></span>
<span class="source-line-no">266</span><span id="line-266">        /**</span>
<span class="source-line-no">267</span><span id="line-267">         * Returns a list of all groups in SEQRES records of the chain, i.e.</span>
<span class="source-line-no">268</span><span id="line-268">         * the aminoacids/nucleotides in the construct.</span>
<span class="source-line-no">269</span><span id="line-269">         * @return a List of all Group objects of this chain</span>
<span class="source-line-no">270</span><span id="line-270">         * @see #setSeqResGroups(List)</span>
<span class="source-line-no">271</span><span id="line-271">         * @see #getSeqResLength()</span>
<span class="source-line-no">272</span><span id="line-272">         * @see #getAtomGroups()</span>
<span class="source-line-no">273</span><span id="line-273">         */</span>
<span class="source-line-no">274</span><span id="line-274">        List&lt;Group&gt; getSeqResGroups ();</span>
<span class="source-line-no">275</span><span id="line-275"></span>
<span class="source-line-no">276</span><span id="line-276">        /**</span>
<span class="source-line-no">277</span><span id="line-277">         * Sets the list of SeqResGroups for this chain.</span>
<span class="source-line-no">278</span><span id="line-278">         *</span>
<span class="source-line-no">279</span><span id="line-279">         * @param seqResGroups a List of Group objects that from the SEQRES groups of this chain.</span>
<span class="source-line-no">280</span><span id="line-280">         * @see #getSeqResGroups()</span>
<span class="source-line-no">281</span><span id="line-281">         */</span>
<span class="source-line-no">282</span><span id="line-282">        void setSeqResGroups(List&lt;Group&gt; seqResGroups);</span>
<span class="source-line-no">283</span><span id="line-283"></span>
<span class="source-line-no">284</span><span id="line-284">        /**</span>
<span class="source-line-no">285</span><span id="line-285">         * Sets the back-reference to its parent Structure.</span>
<span class="source-line-no">286</span><span id="line-286">         *</span>
<span class="source-line-no">287</span><span id="line-287">         * @param parent</span>
<span class="source-line-no">288</span><span id="line-288">         */</span>
<span class="source-line-no">289</span><span id="line-289">        void setStructure(Structure parent) ;</span>
<span class="source-line-no">290</span><span id="line-290"></span>
<span class="source-line-no">291</span><span id="line-291">        /**</span>
<span class="source-line-no">292</span><span id="line-292">         * Returns the parent Structure of this chain.</span>
<span class="source-line-no">293</span><span id="line-293">         *</span>
<span class="source-line-no">294</span><span id="line-294">         * @return the parent Structure object</span>
<span class="source-line-no">295</span><span id="line-295">         * @see #setStructure(Structure)</span>
<span class="source-line-no">296</span><span id="line-296">         */</span>
<span class="source-line-no">297</span><span id="line-297">        Structure getStructure() ;</span>
<span class="source-line-no">298</span><span id="line-298"></span>
<span class="source-line-no">299</span><span id="line-299">        /**</span>
<span class="source-line-no">300</span><span id="line-300">         * Convert this Chain to a String in PDB format</span>
<span class="source-line-no">301</span><span id="line-301">         * @return</span>
<span class="source-line-no">302</span><span id="line-302">         */</span>
<span class="source-line-no">303</span><span id="line-303">        String toPDB();</span>
<span class="source-line-no">304</span><span id="line-304"></span>
<span class="source-line-no">305</span><span id="line-305">        /**</span>
<span class="source-line-no">306</span><span id="line-306">         * Convert this Chain to a String in mmCIF format</span>
<span class="source-line-no">307</span><span id="line-307">         * @return</span>
<span class="source-line-no">308</span><span id="line-308">         */</span>
<span class="source-line-no">309</span><span id="line-309">        String toMMCIF();</span>
<span class="source-line-no">310</span><span id="line-310"></span>
<span class="source-line-no">311</span><span id="line-311"></span>
<span class="source-line-no">312</span><span id="line-312">        /**</span>
<span class="source-line-no">313</span><span id="line-313">         * Sets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</span>
<span class="source-line-no">314</span><span id="line-314">         *</span>
<span class="source-line-no">315</span><span id="line-315">         * @param seqMisMatches</span>
<span class="source-line-no">316</span><span id="line-316">         */</span>
<span class="source-line-no">317</span><span id="line-317">        void setSeqMisMatches(List&lt;SeqMisMatch&gt; seqMisMatches);</span>
<span class="source-line-no">318</span><span id="line-318"></span>
<span class="source-line-no">319</span><span id="line-319">        /**</span>
<span class="source-line-no">320</span><span id="line-320">         * Gets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</span>
<span class="source-line-no">321</span><span id="line-321">         *</span>
<span class="source-line-no">322</span><span id="line-322">         * @return a list of sequence mismatches (or null if none found)</span>
<span class="source-line-no">323</span><span id="line-323">         */</span>
<span class="source-line-no">324</span><span id="line-324">        List&lt;SeqMisMatch&gt; getSeqMisMatches();</span>
<span class="source-line-no">325</span><span id="line-325"></span>
<span class="source-line-no">326</span><span id="line-326">        /**</span>
<span class="source-line-no">327</span><span id="line-327">         * Returns the EntityType of this chain. Equivalent to getEntityInfo().getType()</span>
<span class="source-line-no">328</span><span id="line-328">         * @return</span>
<span class="source-line-no">329</span><span id="line-329">         * @see EntityType</span>
<span class="source-line-no">330</span><span id="line-330">         */</span>
<span class="source-line-no">331</span><span id="line-331">        EntityType getEntityType();</span>
<span class="source-line-no">332</span><span id="line-332"></span>
<span class="source-line-no">333</span><span id="line-333">        /**</span>
<span class="source-line-no">334</span><span id="line-334">         * Tests if a chain is consisting of water molecules only</span>
<span class="source-line-no">335</span><span id="line-335">         *</span>
<span class="source-line-no">336</span><span id="line-336">         * @return true if there are only solvent molecules in this chain.</span>
<span class="source-line-no">337</span><span id="line-337">         */</span>
<span class="source-line-no">338</span><span id="line-338">        public boolean isWaterOnly();</span>
<span class="source-line-no">339</span><span id="line-339"></span>
<span class="source-line-no">340</span><span id="line-340">        /**</span>
<span class="source-line-no">341</span><span id="line-341">         * Returns true if the given chain is composed of non-polymeric (including water) groups only.</span>
<span class="source-line-no">342</span><span id="line-342">         *</span>
<span class="source-line-no">343</span><span id="line-343">         * @return true if only non-polymeric groups in this chain.</span>
<span class="source-line-no">344</span><span id="line-344">         */</span>
<span class="source-line-no">345</span><span id="line-345">        public boolean isPureNonPolymer();</span>
<span class="source-line-no">346</span><span id="line-346"></span>
<span class="source-line-no">347</span><span id="line-347">        /**</span>
<span class="source-line-no">348</span><span id="line-348">         * Get the predominant {@link GroupType} for a given Chain, following these</span>
<span class="source-line-no">349</span><span id="line-349">         * rules:</span>
<span class="source-line-no">350</span><span id="line-350">         * &lt;ul&gt;</span>
<span class="source-line-no">351</span><span id="line-351">         * &lt;li&gt;if the ratio of number of residues of a certain</span>
<span class="source-line-no">352</span><span id="line-352">         * {@link GroupType} to total non-water residues is above the threshold</span>
<span class="source-line-no">353</span><span id="line-353">         * {@value org.biojava.nbio.structure.StructureTools#RATIO_RESIDUES_TO_TOTAL}, then that {@link GroupType} is</span>
<span class="source-line-no">354</span><span id="line-354">         * returned&lt;/li&gt; &lt;li&gt;if there is no {@link GroupType} that is above the</span>
<span class="source-line-no">355</span><span id="line-355">         * threshold then the {@link GroupType} with most members is chosen, logging</span>
<span class="source-line-no">356</span><span id="line-356">         * it&lt;/li&gt;</span>
<span class="source-line-no">357</span><span id="line-357">         * &lt;/ul&gt;</span>
<span class="source-line-no">358</span><span id="line-358">         * &lt;p&gt;</span>
<span class="source-line-no">359</span><span id="line-359">         * See also {@link ChemComp#getPolymerType()} and</span>
<span class="source-line-no">360</span><span id="line-360">         * {@link ChemComp#getResidueType()} which follow the PDB chemical component</span>
<span class="source-line-no">361</span><span id="line-361">         * dictionary and provide a much more accurate description of groups and</span>
<span class="source-line-no">362</span><span id="line-362">         * their linking.</span>
<span class="source-line-no">363</span><span id="line-363">         *</span>
<span class="source-line-no">364</span><span id="line-364">         * @return the predominant group type</span>
<span class="source-line-no">365</span><span id="line-365">         */</span>
<span class="source-line-no">366</span><span id="line-366">        GroupType getPredominantGroupType();</span>
<span class="source-line-no">367</span><span id="line-367"></span>
<span class="source-line-no">368</span><span id="line-368">        /**</span>
<span class="source-line-no">369</span><span id="line-369">         * Tell whether given chain is a protein chain</span>
<span class="source-line-no">370</span><span id="line-370">         *</span>
<span class="source-line-no">371</span><span id="line-371">         * @return true if protein, false if nucleotide or ligand</span>
<span class="source-line-no">372</span><span id="line-372">         * @see #getPredominantGroupType()</span>
<span class="source-line-no">373</span><span id="line-373">         */</span>
<span class="source-line-no">374</span><span id="line-374">        boolean isProtein();</span>
<span class="source-line-no">375</span><span id="line-375"></span>
<span class="source-line-no">376</span><span id="line-376">        /**</span>
<span class="source-line-no">377</span><span id="line-377">         * Tell whether given chain is DNA or RNA</span>
<span class="source-line-no">378</span><span id="line-378">         *</span>
<span class="source-line-no">379</span><span id="line-379">         * @return true if nucleic acid, false if protein or ligand</span>
<span class="source-line-no">380</span><span id="line-380">         * @see #getPredominantGroupType()</span>
<span class="source-line-no">381</span><span id="line-381">         */</span>
<span class="source-line-no">382</span><span id="line-382">        boolean isNucleicAcid();</span>
<span class="source-line-no">383</span><span id="line-383">}</span>




























































</pre>
</div>
</main>
</body>
</html>
