<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on 01-21-2010<a name="line.20"></a>
<span class="sourceLineNo">021</span> */<a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.nbio.core.sequence.io;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.core.sequence.DNASequence;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.29"></a>
<span class="sourceLineNo">030</span><a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.io.File;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.io.FileInputStream;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.io.IOException;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.io.InputStream;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import java.util.LinkedHashMap;<a name="line.35"></a>
<span class="sourceLineNo">036</span><a name="line.36"></a>
<span class="sourceLineNo">037</span>/**<a name="line.37"></a>
<span class="sourceLineNo">038</span> *<a name="line.38"></a>
<span class="sourceLineNo">039</span> * @author Scooter Willis &lt;willishf at gmail dot com&gt;<a name="line.39"></a>
<span class="sourceLineNo">040</span> */<a name="line.40"></a>
<span class="sourceLineNo">041</span>public class FastaReaderHelper {<a name="line.41"></a>
<span class="sourceLineNo">042</span><a name="line.42"></a>
<span class="sourceLineNo">043</span>        /**<a name="line.43"></a>
<span class="sourceLineNo">044</span>         * Selecting lazySequenceLoad=true will parse the FASTA file and figure out the accessionid and offsets and return sequence objects<a name="line.44"></a>
<span class="sourceLineNo">045</span>         * that can in the future read the sequence from the disk. This allows the loading of large fasta files where you are only interested<a name="line.45"></a>
<span class="sourceLineNo">046</span>         * in one sequence based on accession id.<a name="line.46"></a>
<span class="sourceLineNo">047</span>         * @param file<a name="line.47"></a>
<span class="sourceLineNo">048</span>         * @param lazySequenceLoad<a name="line.48"></a>
<span class="sourceLineNo">049</span>         * @return<a name="line.49"></a>
<span class="sourceLineNo">050</span>         * @throws IOException<a name="line.50"></a>
<span class="sourceLineNo">051</span>         */<a name="line.51"></a>
<span class="sourceLineNo">052</span>        public static LinkedHashMap&lt;String, DNASequence&gt; readFastaDNASequence(File file, boolean lazySequenceLoad) throws IOException {<a name="line.52"></a>
<span class="sourceLineNo">053</span>                if (!lazySequenceLoad) {<a name="line.53"></a>
<span class="sourceLineNo">054</span>                        return readFastaDNASequence(file);<a name="line.54"></a>
<span class="sourceLineNo">055</span>                }<a name="line.55"></a>
<span class="sourceLineNo">056</span><a name="line.56"></a>
<span class="sourceLineNo">057</span>                FastaReader&lt;DNASequence, NucleotideCompound&gt; fastaProxyReader =<a name="line.57"></a>
<span class="sourceLineNo">058</span>                                new FastaReader&lt;DNASequence, NucleotideCompound&gt;(<a name="line.58"></a>
<span class="sourceLineNo">059</span>                                                file,<a name="line.59"></a>
<span class="sourceLineNo">060</span>                                                new GenericFastaHeaderParser&lt;DNASequence, NucleotideCompound&gt;(),<a name="line.60"></a>
<span class="sourceLineNo">061</span>                                                new FileProxyDNASequenceCreator(<a name="line.61"></a>
<span class="sourceLineNo">062</span>                                                                file,<a name="line.62"></a>
<span class="sourceLineNo">063</span>                                                                DNACompoundSet.getDNACompoundSet(),<a name="line.63"></a>
<span class="sourceLineNo">064</span>                                                                new FastaSequenceParser()<a name="line.64"></a>
<span class="sourceLineNo">065</span>                                                        )<a name="line.65"></a>
<span class="sourceLineNo">066</span>                                        );<a name="line.66"></a>
<span class="sourceLineNo">067</span>                return fastaProxyReader.process();<a name="line.67"></a>
<span class="sourceLineNo">068</span><a name="line.68"></a>
<span class="sourceLineNo">069</span>        }<a name="line.69"></a>
<span class="sourceLineNo">070</span><a name="line.70"></a>
<span class="sourceLineNo">071</span>        /**<a name="line.71"></a>
<span class="sourceLineNo">072</span>         * Read a fasta file containing amino acids with setup that would handle most<a name="line.72"></a>
<span class="sourceLineNo">073</span>         * cases.<a name="line.73"></a>
<span class="sourceLineNo">074</span>         *<a name="line.74"></a>
<span class="sourceLineNo">075</span>         * @param file<a name="line.75"></a>
<span class="sourceLineNo">076</span>         * @return<a name="line.76"></a>
<span class="sourceLineNo">077</span>         * @throws IOException<a name="line.77"></a>
<span class="sourceLineNo">078</span>         */<a name="line.78"></a>
<span class="sourceLineNo">079</span>        public static LinkedHashMap&lt;String, ProteinSequence&gt; readFastaProteinSequence(<a name="line.79"></a>
<span class="sourceLineNo">080</span>                        File file) throws IOException {<a name="line.80"></a>
<span class="sourceLineNo">081</span>                FileInputStream inStream = new FileInputStream(file);<a name="line.81"></a>
<span class="sourceLineNo">082</span>                LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequences = readFastaProteinSequence(inStream);<a name="line.82"></a>
<span class="sourceLineNo">083</span>                inStream.close();<a name="line.83"></a>
<span class="sourceLineNo">084</span>                return proteinSequences;<a name="line.84"></a>
<span class="sourceLineNo">085</span>        }<a name="line.85"></a>
<span class="sourceLineNo">086</span><a name="line.86"></a>
<span class="sourceLineNo">087</span>        /**<a name="line.87"></a>
<span class="sourceLineNo">088</span>         * Read a fasta file containing amino acids with setup that would handle most<a name="line.88"></a>
<span class="sourceLineNo">089</span>         * cases. User is responsible for closing InputStream because you opened it<a name="line.89"></a>
<span class="sourceLineNo">090</span>         *<a name="line.90"></a>
<span class="sourceLineNo">091</span>         * @param inStream<a name="line.91"></a>
<span class="sourceLineNo">092</span>         * @return<a name="line.92"></a>
<span class="sourceLineNo">093</span>         * @throws IOException<a name="line.93"></a>
<span class="sourceLineNo">094</span>         */<a name="line.94"></a>
<span class="sourceLineNo">095</span>        public static LinkedHashMap&lt;String, ProteinSequence&gt; readFastaProteinSequence(<a name="line.95"></a>
<span class="sourceLineNo">096</span>                        InputStream inStream) throws IOException {<a name="line.96"></a>
<span class="sourceLineNo">097</span>                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.97"></a>
<span class="sourceLineNo">098</span>                                inStream,<a name="line.98"></a>
<span class="sourceLineNo">099</span>                                new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),<a name="line.99"></a>
<span class="sourceLineNo">100</span>                                new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));<a name="line.100"></a>
<span class="sourceLineNo">101</span>                return fastaReader.process();<a name="line.101"></a>
<span class="sourceLineNo">102</span>        }<a name="line.102"></a>
<span class="sourceLineNo">103</span><a name="line.103"></a>
<span class="sourceLineNo">104</span>        /**<a name="line.104"></a>
<span class="sourceLineNo">105</span>         * Read a fasta DNA sequence<a name="line.105"></a>
<span class="sourceLineNo">106</span>         * @param inStream<a name="line.106"></a>
<span class="sourceLineNo">107</span>         * @return<a name="line.107"></a>
<span class="sourceLineNo">108</span>         * @throws IOException<a name="line.108"></a>
<span class="sourceLineNo">109</span>         */<a name="line.109"></a>
<span class="sourceLineNo">110</span>        public static LinkedHashMap&lt;String, DNASequence&gt; readFastaDNASequence(<a name="line.110"></a>
<span class="sourceLineNo">111</span>                        InputStream inStream) throws IOException {<a name="line.111"></a>
<span class="sourceLineNo">112</span>                FastaReader&lt;DNASequence, NucleotideCompound&gt; fastaReader = new FastaReader&lt;DNASequence, NucleotideCompound&gt;(<a name="line.112"></a>
<span class="sourceLineNo">113</span>                                inStream,<a name="line.113"></a>
<span class="sourceLineNo">114</span>                                new GenericFastaHeaderParser&lt;DNASequence, NucleotideCompound&gt;(),<a name="line.114"></a>
<span class="sourceLineNo">115</span>                                new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));<a name="line.115"></a>
<span class="sourceLineNo">116</span>                return fastaReader.process();<a name="line.116"></a>
<span class="sourceLineNo">117</span>        }<a name="line.117"></a>
<span class="sourceLineNo">118</span><a name="line.118"></a>
<span class="sourceLineNo">119</span>        /**<a name="line.119"></a>
<span class="sourceLineNo">120</span>         *<a name="line.120"></a>
<span class="sourceLineNo">121</span>         * @param file<a name="line.121"></a>
<span class="sourceLineNo">122</span>         * @return<a name="line.122"></a>
<span class="sourceLineNo">123</span>         * @throws IOException<a name="line.123"></a>
<span class="sourceLineNo">124</span>         */<a name="line.124"></a>
<span class="sourceLineNo">125</span>        public static LinkedHashMap&lt;String, DNASequence&gt; readFastaDNASequence(<a name="line.125"></a>
<span class="sourceLineNo">126</span>                        File file) throws IOException {<a name="line.126"></a>
<span class="sourceLineNo">127</span>                FileInputStream inStream = new FileInputStream(file);<a name="line.127"></a>
<span class="sourceLineNo">128</span>                LinkedHashMap&lt;String, DNASequence&gt; dnaSequences = readFastaDNASequence(inStream);<a name="line.128"></a>
<span class="sourceLineNo">129</span>                inStream.close();<a name="line.129"></a>
<span class="sourceLineNo">130</span>                return dnaSequences;<a name="line.130"></a>
<span class="sourceLineNo">131</span>        }<a name="line.131"></a>
<span class="sourceLineNo">132</span><a name="line.132"></a>
<span class="sourceLineNo">133</span>        public static void main(String args[]) throws Exception {<a name="line.133"></a>
<span class="sourceLineNo">134</span><a name="line.134"></a>
<span class="sourceLineNo">135</span>                LinkedHashMap&lt;String, DNASequence&gt; dnaSequences = FastaReaderHelper.readFastaDNASequence(new File("fasta.fna"));<a name="line.135"></a>
<span class="sourceLineNo">136</span>                for (DNASequence sequence : dnaSequences.values()) {<a name="line.136"></a>
<span class="sourceLineNo">137</span>                        sequence.getRNASequence().getProteinSequence().getSequenceAsString();<a name="line.137"></a>
<span class="sourceLineNo">138</span>                }<a name="line.138"></a>
<span class="sourceLineNo">139</span>        }<a name="line.139"></a>
<span class="sourceLineNo">140</span>}<a name="line.140"></a>




























































</pre>
</div>
</body>
</html>
