<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Uses of Interface org.biojava.nbio.structure.Atom (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="use: package: org.biojava.nbio.structure, interface: Atom">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Interface org.biojava.nbio.structure.Atom" class="title">Uses of Interface<br>org.biojava.nbio.structure.Atom</h1>
</div>
<div class="caption"><span>Packages that use <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.protmod.structure">org.biojava.nbio.protmod.structure</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></div>
<div class="col-last odd-row-color">
<div class="block">Interfaces and classes for protein structure (PDB).</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></div>
<div class="col-last even-row-color">
<div class="block">Classes for the alignment of structures.</div>
</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></div>
<div class="col-last odd-row-color">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.fatcat">org.biojava.nbio.structure.align.fatcat</a></div>
<div class="col-last even-row-color">
<div class="block">Classes related to the implementation of the FATCAT alignment algorithm, here called jFATCAT.</div>
</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.fatcat.calc">org.biojava.nbio.structure.align.fatcat.calc</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></div>
<div class="col-last even-row-color">
<div class="block">Utility methods for better interaction with Jmol.</div>
</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.helper">org.biojava.nbio.structure.align.helper</a></div>
<div class="col-last odd-row-color">
<div class="block">Helper classes for structural alignment.</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.model">org.biojava.nbio.structure.align.model</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.pairwise">org.biojava.nbio.structure.align.pairwise</a></div>
<div class="col-last odd-row-color">
<div class="block">Classes for the pairwise alignment of structures.</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.quaternary">org.biojava.nbio.structure.align.quaternary</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.seq">org.biojava.nbio.structure.align.seq</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.util">org.biojava.nbio.structure.align.util</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></div>
<div class="col-last odd-row-color">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.asa">org.biojava.nbio.structure.asa</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.contact">org.biojava.nbio.structure.contact</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.domain">org.biojava.nbio.structure.domain</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.domain.pdp">org.biojava.nbio.structure.domain.pdp</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></div>
<div class="col-last odd-row-color">
<div class="block">Input and Output of Structures</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.io.cif">org.biojava.nbio.structure.io.cif</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.io.mmtf">org.biojava.nbio.structure.io.mmtf</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="org.biojava.nbio.protmod.structure">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../../protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a></h2>
<div class="caption"><span>Methods in <a href="../../protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureUtil.</span><code><a href="../../protmod/structure/StructureUtil.html#findLinkage(org.biojava.nbio.structure.Group,org.biojava.nbio.structure.Group,java.lang.String,java.lang.String,double)" class="member-name-link">findLinkage</a><wbr>(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group1,
 <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;nameOfAtomOnGroup1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;nameOfAtomOnGroup2,
 double&nbsp;bondLengthTolerance)</code></div>
<div class="col-last even-row-color">
<div class="block">Find a linkage between two groups within tolerance of bond length.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureUtil.</span><code><a href="../../protmod/structure/StructureUtil.html#findNearestAtomLinkage(org.biojava.nbio.structure.Group,org.biojava.nbio.structure.Group,java.util.List,java.util.List,boolean,double)" class="member-name-link">findNearestAtomLinkage</a><wbr>(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group1,
 <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;potentialNamesOfAtomOnGroup1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;&nbsp;potentialNamesOfAtomOnGroup2,
 boolean&nbsp;ignoreNCLinkage,
 double&nbsp;bondLengthTolerance)</code></div>
<div class="col-last odd-row-color">
<div class="block">Find a linkage between two groups within tolerance of bond length,
 from potential atoms.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../protmod/structure/package-summary.html">org.biojava.nbio.protmod.structure</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureUtil.</span><code><a href="../../protmod/structure/StructureUtil.html#getAtomDistance(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">getAtomDistance</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../../protmod/structure/StructureAtom.html" title="class in org.biojava.nbio.protmod.structure">StructureAtom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureUtil.</span><code><a href="../../protmod/structure/StructureUtil.html#getStructureAtom(org.biojava.nbio.structure.Atom,boolean)" class="member-name-link">getStructureAtom</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 boolean&nbsp;isParentAminoAcid)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../../protmod/structure/StructureAtomLinkage.html" title="class in org.biojava.nbio.protmod.structure">StructureAtomLinkage</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureUtil.</span><code><a href="../../protmod/structure/StructureUtil.html#getStructureAtomLinkage(org.biojava.nbio.structure.Atom,boolean,org.biojava.nbio.structure.Atom,boolean)" class="member-name-link">getStructureAtomLinkage</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
 boolean&nbsp;isParentAminoAcid1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2,
 boolean&nbsp;isParentAminoAcid2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../package-summary.html">org.biojava.nbio.structure</a></h2>
<div class="caption"><span>Classes in <a href="../package-summary.html">org.biojava.nbio.structure</a> that implement <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../AtomImpl.html" class="type-name-link" title="class in org.biojava.nbio.structure">AtomImpl</a></code></div>
<div class="col-last even-row-color">
<div class="block">Implementation of an Atom of a PDB file.</div>
</div>
</div>
<div class="caption"><span>Fields in <a href="../package-summary.html">org.biojava.nbio.structure</a> with type parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Field</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>protected <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">HetatomImpl.</span><code><a href="../HetatomImpl.html#atoms" class="member-name-link">atoms</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#add(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">add</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last even-row-color">
<div class="block">add two atoms ( a + b).</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">centerAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Center the atoms at the Centroid.</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)" class="member-name-link">centerAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></div>
<div class="col-last even-row-color">
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#centerOfMass(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">centerOfMass</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;points)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the center of mass of the set of atoms.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">cloneAtomArray</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last even-row-color">
<div class="block">Provides an equivalent copy of Atoms in a new array.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#createVirtualCBAtom(org.biojava.nbio.structure.AminoAcid)" class="member-name-link">createVirtualCBAtom</a><wbr>(<a href="../AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a>&nbsp;amino)</code></div>
<div class="col-last odd-row-color">
<div class="block">creates a virtual C-beta atom. this might be needed when working with GLY

 thanks to Peter Lackner for a python template of this method.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#duplicateCA2(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">duplicateCA2</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Utility method for working with circular permutations.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Chain)" class="member-name-link">getAllAtomArray</a><wbr>(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns and array of all atoms of the chain, including
 Hydrogens (if present) and all HETATOMs.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Structure)" class="member-name-link">getAllAtomArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Convert all atoms of the structure (all models) into an Atom array</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllAtomArray(org.biojava.nbio.structure.Structure,int)" class="member-name-link">getAllAtomArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
 int&nbsp;model)</code></div>
<div class="col-last odd-row-color">
<div class="block">Convert all atoms of the structure (specified model) into an Atom array</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Chain,boolean)" class="member-name-link">getAllNonHAtomArray</a><wbr>(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c,
 boolean&nbsp;hetAtoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns and array of all non-Hydrogen atoms in the given Chain,
 optionally including HET atoms or not Waters are not included.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Structure,boolean)" class="member-name-link">getAllNonHAtomArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
 boolean&nbsp;hetAtoms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns and array of all non-Hydrogen atoms in the given Structure,
 optionally including HET atoms or not.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllNonHAtomArray(org.biojava.nbio.structure.Structure,boolean,int)" class="member-name-link">getAllNonHAtomArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
 boolean&nbsp;hetAtoms,
 int&nbsp;modelNr)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns and array of all non-Hydrogen atoms in the given Structure,
 optionally including HET atoms or not.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Group.</span><code><a href="../Group.html#getAtom(int)" class="member-name-link">getAtom</a><wbr>(int&nbsp;position)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get at atom by position.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Group.</span><code><a href="../Group.html#getAtom(java.lang.String)" class="member-name-link">getAtom</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last even-row-color">
<div class="block">Get an atom given its PDB name.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">HetatomImpl.</span><code><a href="../HetatomImpl.html#getAtom(int)" class="member-name-link">getAtom</a><wbr>(int&nbsp;position)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get at atom by position.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">HetatomImpl.</span><code><a href="../HetatomImpl.html#getAtom(java.lang.String)" class="member-name-link">getAtom</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last even-row-color">
<div class="block">Get an atom given its PDB name.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Bond.</span><code><a href="../Bond.html#getAtomA()" class="member-name-link">getAtomA</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Gets atom 'A' of this bond.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">BondImpl.</span><code><a href="../BondImpl.html#getAtomA()" class="member-name-link">getAtomA</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Gets atom 'A' of this bond.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAtomArray(org.biojava.nbio.structure.Chain,java.lang.String%5B%5D)" class="member-name-link">getAtomArray</a><wbr>(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&nbsp;atomNames)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns an array of the requested Atoms from the Chain object.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAtomArray(org.biojava.nbio.structure.Structure,java.lang.String%5B%5D)" class="member-name-link">getAtomArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&nbsp;atomNames)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAtomArrayAllModels(org.biojava.nbio.structure.Structure,java.lang.String%5B%5D)" class="member-name-link">getAtomArrayAllModels</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]&nbsp;atomNames)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns an array of the requested Atoms from the Structure object.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Bond.</span><code><a href="../Bond.html#getAtomB()" class="member-name-link">getAtomB</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Gets atom 'B' of this bond.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BondImpl.</span><code><a href="../BondImpl.html#getAtomB()" class="member-name-link">getAtomB</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Gets atom 'B' of this bond.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAtomCAArray(org.biojava.nbio.structure.Chain)" class="member-name-link">getAtomCAArray</a><wbr>(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns an Atom array of the C-alpha atoms.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAtomCAArray(org.biojava.nbio.structure.Structure)" class="member-name-link">getAtomCAArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></div>
<div class="col-last odd-row-color">
<div class="block">Return an Atom array of the C-alpha atoms.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getBackboneAtomArray(org.biojava.nbio.structure.Structure)" class="member-name-link">getBackboneAtomArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Return an Atom array of the main chain atoms: CA, C, N, O Any group that
 contains those atoms will be included, be it a standard aminoacid or not</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcid.</span><code><a href="../AminoAcid.html#getC()" class="member-name-link">getC</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get C atom.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidImpl.</span><code><a href="../AminoAcidImpl.html#getC()" class="member-name-link">getC</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get C atom.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcid.</span><code><a href="../AminoAcid.html#getCA()" class="member-name-link">getCA</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get CA atom.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidImpl.</span><code><a href="../AminoAcidImpl.html#getCA()" class="member-name-link">getCA</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get CA atom.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcid.</span><code><a href="../AminoAcid.html#getCB()" class="member-name-link">getCB</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get CB atom.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcidImpl.</span><code><a href="../AminoAcidImpl.html#getCB()" class="member-name-link">getCB</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get CB atom.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getCenterVector</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid.</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)" class="member-name-link">getCenterVector</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid, if the centroid is already known</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getCentroid(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getCentroid</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the centroid of the set of atoms.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcid.</span><code><a href="../AminoAcid.html#getN()" class="member-name-link">getN</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get N atom.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcidImpl.</span><code><a href="../AminoAcidImpl.html#getN()" class="member-name-link">getN</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get N atom.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AminoAcid.</span><code><a href="../AminoAcid.html#getO()" class="member-name-link">getO</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get O atom.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AminoAcidImpl.</span><code><a href="../AminoAcidImpl.html#getO()" class="member-name-link">getO</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get O atom.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">NucleotideImpl.</span><code><a href="../NucleotideImpl.html#getO3Prime()" class="member-name-link">getO3Prime</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the O3' atom if present, otherwise null</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">NucleotideImpl.</span><code><a href="../NucleotideImpl.html#getO5Prime()" class="member-name-link">getO5Prime</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the O5' atom if present, otherwise null</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Bond.</span><code><a href="../Bond.html#getOther(org.biojava.nbio.structure.Atom)" class="member-name-link">getOther</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></div>
<div class="col-last even-row-color">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BondImpl.</span><code><a href="../BondImpl.html#getOther(org.biojava.nbio.structure.Atom)" class="member-name-link">getOther</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></div>
<div class="col-last odd-row-color">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">NucleotideImpl.</span><code><a href="../NucleotideImpl.html#getP()" class="member-name-link">getP</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the P atom if present, otherwise null</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getRepresentativeAtomArray(org.biojava.nbio.structure.Chain)" class="member-name-link">getRepresentativeAtomArray</a><wbr>(<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&nbsp;c)</code></div>
<div class="col-last odd-row-color">
<div class="block">Gets a representative atom for each group that is part of the chain
 backbone.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getRepresentativeAtomArray(org.biojava.nbio.structure.Structure)" class="member-name-link">getRepresentativeAtomArray</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Gets a representative atom for each group that is part of the chain
 backbone.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getTranslationVector(javax.vecmath.Matrix4d)" class="member-name-link">getTranslationVector</a><wbr>(javax.vecmath.Matrix4d&nbsp;transform)</code></div>
<div class="col-last odd-row-color">
<div class="block">Extract the translational vector as an Atom of a transformation matrix.</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#invert(org.biojava.nbio.structure.Atom)" class="member-name-link">invert</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AtomIterator.</span><code><a href="../AtomIterator.html#next()" class="member-name-link">next</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Return next atom.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#scale(org.biojava.nbio.structure.Atom,double)" class="member-name-link">scale</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 double&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Multiply elements of a by s</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#scaleAdd(double,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">scaleAdd</a><wbr>(double&nbsp;s,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;x,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">Perform linear transformation s*X+B, and store the result in b</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#scaleEquals(org.biojava.nbio.structure.Atom,double)" class="member-name-link">scaleEquals</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 double&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Multiply elements of a by s (in place)</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#subtract(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">subtract</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">subtract two atoms ( a - b).</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#unitVector(org.biojava.nbio.structure.Atom)" class="member-name-link">unitVector</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the unit vector of vector a .</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#vectorProduct(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">vectorProduct</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">Vector product (cross product).</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Group.</span><code><a href="../Group.html#getAtoms()" class="member-name-link">getAtoms</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get list of atoms.</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">HetatomImpl.</span><code><a href="../HetatomImpl.html#getAtoms()" class="member-name-link">getAtoms</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get list of atoms.</div>
</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Iterator.html" title="class or interface in java.util" class="external-link">Iterator</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Group.</span><code><a href="../Group.html#iterator()" class="member-name-link">iterator</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get an Atom Iterator.</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Iterator.html" title="class or interface in java.util" class="external-link">Iterator</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">HetatomImpl.</span><code><a href="../HetatomImpl.html#iterator()" class="member-name-link">iterator</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#add(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">add</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last even-row-color">
<div class="block">add two atoms ( a + b).</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Group.</span><code><a href="../Group.html#addAtom(org.biojava.nbio.structure.Atom)" class="member-name-link">addAtom</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code></div>
<div class="col-last odd-row-color">
<div class="block">Add an atom to this group.</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">HetatomImpl.</span><code><a href="../HetatomImpl.html#addAtom(org.biojava.nbio.structure.Atom)" class="member-name-link">addAtom</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code></div>
<div class="col-last even-row-color">
<div class="block">Add an atom to this group.</div>
</div>
<div class="col-first odd-row-color"><code>static final double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#amount(org.biojava.nbio.structure.Atom)" class="member-name-link">amount</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last odd-row-color">
<div class="block">Gets the length of the vector (2-norm)</div>
</div>
<div class="col-first even-row-color"><code>static final double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#angle(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">angle</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last even-row-color">
<div class="block">Gets the angle between two vectors</div>
</div>
<div class="col-first odd-row-color"><code>static javax.vecmath.Point3d[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#atomsToPoints(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">atomsToPoints</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Convert an array of atoms into an array of vecmath points</div>
</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#calcRotationAngleInDegrees(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">calcRotationAngleInDegrees</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centerPt,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;targetPt)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculates the angle from centerPt to targetPt in degrees.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">centerAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></div>
<div class="col-last odd-row-color">
<div class="block">Center the atoms at the Centroid.</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#centerAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)" class="member-name-link">centerAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></div>
<div class="col-last even-row-color">
<div class="block">Center the atoms at the Centroid, if the centroid is already know.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#centerOfMass(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">centerOfMass</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;points)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the center of mass of the set of atoms.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#cloneAtomArray(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">cloneAtomArray</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last even-row-color">
<div class="block">Provides an equivalent copy of Atoms in a new array.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#cloneGroups(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">cloneGroups</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last odd-row-color">
<div class="block">Clone a set of representative Atoms, but returns the parent groups</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#convertAtomsToSeq(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">convertAtomsToSeq</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#duplicateCA2(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">duplicateCA2</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Utility method for working with circular permutations.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html" title="class or interface in java.util" class="external-link">Set</a><wbr>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllGroupsFromSubset(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAllGroupsFromSubset</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Expand a set of atoms into all groups from the same structure.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html" title="class or interface in java.util" class="external-link">Set</a><wbr>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getAllGroupsFromSubset(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.GroupType)" class="member-name-link">getAllGroupsFromSubset</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 <a href="../GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;types)</code></div>
<div class="col-last odd-row-color">
<div class="block">Expand a set of atoms into all groups from the same structure.</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getCenterVector</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getCenterVector(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)" class="member-name-link">getCenterVector</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the Vector that needs to be applied to shift a set of atoms to
 the Centroid, if the centroid is already known</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getCentroid(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getCentroid</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the centroid of the set of atoms.</div>
</div>
<div class="col-first odd-row-color"><code>static final double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getDistance(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">getDistance</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">calculate distance between two atoms.</div>
</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getDistanceFast(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">getDistanceFast</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last even-row-color">
<div class="block">Will calculate the square of distances between two atoms.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a><wbr>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>,<wbr><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Double.html" title="class or interface in java.lang" class="external-link">Double</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getGroupDistancesWithinShell(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Atom,java.util.Set,double,boolean,boolean)" class="member-name-link">getGroupDistancesWithinShell</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;centroid,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html" title="class or interface in java.util" class="external-link">Set</a>&lt;<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
 double&nbsp;radius,
 boolean&nbsp;includeWater,
 boolean&nbsp;useAverageDistance)</code></div>
<div class="col-last odd-row-color">
<div class="block">Finds Groups in <code>structure</code> that contain at least one Atom that is
 within <code>radius</code> Angstroms of <code>centroid</code>.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html" title="class or interface in java.util" class="external-link">Set</a><wbr>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getGroupsWithinShell(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Atom,java.util.Set,double,boolean)" class="member-name-link">getGroupsWithinShell</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Set.html" title="class or interface in java.util" class="external-link">Set</a>&lt;<a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&gt;&nbsp;excludeResidues,
 double&nbsp;distance,
 boolean&nbsp;includeWater)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getLigandsByProximity(java.util.Collection,org.biojava.nbio.structure.Atom%5B%5D,double)" class="member-name-link">getLigandsByProximity</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html" title="class or interface in java.util" class="external-link">Collection</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;target,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;query,
 double&nbsp;cutoff)</code></div>
<div class="col-last odd-row-color">
<div class="block">Finds all ligand groups from the target which fall within the cutoff distance
 of some atom from the query set.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Bond.</span><code><a href="../Bond.html#getOther(org.biojava.nbio.structure.Atom)" class="member-name-link">getOther</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></div>
<div class="col-last even-row-color">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BondImpl.</span><code><a href="../BondImpl.html#getOther(org.biojava.nbio.structure.Atom)" class="member-name-link">getOther</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;exclude)</code></div>
<div class="col-last odd-row-color">
<div class="block">A utility method to get the other atom in a bond, given one of its atoms.</div>
</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getTMScore(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getTMScore</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
 int&nbsp;len1,
 int&nbsp;len2)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate the TM-Score for the superposition.</div>
</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getTMScore(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int,boolean)" class="member-name-link">getTMScore</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atomSet2,
 int&nbsp;len1,
 int&nbsp;len2,
 boolean&nbsp;normalizeMin)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculate the TM-Score for the superposition.</div>
</div>
<div class="col-first even-row-color"><code>static javax.vecmath.Matrix4d</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#getTransformation(org.biojava.nbio.structure.jama.Matrix,org.biojava.nbio.structure.Atom)" class="member-name-link">getTransformation</a><wbr>(<a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;rot,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;trans)</code></div>
<div class="col-last even-row-color">
<div class="block">Convert JAMA rotation and translation to a Vecmath transformation matrix.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#getUnalignedGroups(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getUnalignedGroups</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last odd-row-color">
<div class="block">List of groups from the structure not included in ca (e.g. ligands).</div>
</div>
<div class="col-first even-row-color"><code>boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Atom.</span><code><a href="../Atom.html#hasBond(org.biojava.nbio.structure.Atom)" class="member-name-link">hasBond</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;other)</code></div>
<div class="col-last even-row-color">
<div class="block">Test if another atom has a bond to this atom</div>
</div>
<div class="col-first odd-row-color"><code>boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AtomImpl.</span><code><a href="../AtomImpl.html#hasBond(org.biojava.nbio.structure.Atom)" class="member-name-link">hasBond</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;other)</code></div>
<div class="col-last odd-row-color">
<div class="block">Test if another atom has a bond to this atom</div>
</div>
<div class="col-first even-row-color"><code>static boolean</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#hasDeuteratedEquiv(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Group)" class="member-name-link">hasDeuteratedEquiv</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;currentGroup)</code></div>
<div class="col-last even-row-color">
<div class="block">Check to see if a Hydrogen has a  Deuterated brother in the group.</div>
</div>
<div class="col-first odd-row-color"><code>static boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureTools.</span><code><a href="../StructureTools.html#hasNonDeuteratedEquiv(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Group)" class="member-name-link">hasNonDeuteratedEquiv</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;currentGroup)</code></div>
<div class="col-last odd-row-color">
<div class="block">Check to see if an Deuterated atom has a non deuterated brother in the group.</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#invert(org.biojava.nbio.structure.Atom)" class="member-name-link">invert</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">rmsd</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;x,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;y)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculate the RMSD of two Atom arrays, already superposed.</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#rotate(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.jama.Matrix)" class="member-name-link">rotate</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;matrix)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static final void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#rotate(org.biojava.nbio.structure.Atom,double%5B%5D%5B%5D)" class="member-name-link">rotate</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 double[][]&nbsp;m)</code></div>
<div class="col-last odd-row-color">
<div class="block">Rotate a single Atom aroud a rotation matrix.</div>
</div>
<div class="col-first even-row-color"><code>static final void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#rotate(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.jama.Matrix)" class="member-name-link">rotate</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m)</code></div>
<div class="col-last even-row-color">
<div class="block">Rotate an Atom around a Matrix object.</div>
</div>
<div class="col-first odd-row-color"><code>static final double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#scalarProduct(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">scalarProduct</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">Scalar product (dot product).</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#scale(org.biojava.nbio.structure.Atom,double)" class="member-name-link">scale</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 double&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Multiply elements of a by s</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#scaleAdd(double,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">scaleAdd</a><wbr>(double&nbsp;s,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;x,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">Perform linear transformation s*X+B, and store the result in b</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#scaleEquals(org.biojava.nbio.structure.Atom,double)" class="member-name-link">scaleEquals</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 double&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Multiply elements of a by s (in place)</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#shift(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom)" class="member-name-link">shift</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">Shift an array of atoms at once.</div>
</div>
<div class="col-first even-row-color"><code>static final void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#shift(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">shift</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last even-row-color">
<div class="block">Shift a vector.</div>
</div>
<div class="col-first odd-row-color"><code>static final void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#shift(org.biojava.nbio.structure.Group,org.biojava.nbio.structure.Atom)" class="member-name-link">shift</a><wbr>(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last odd-row-color">
<div class="block">Shift a Group with a vector.</div>
</div>
<div class="col-first even-row-color"><code>static final void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#shift(org.biojava.nbio.structure.Structure,org.biojava.nbio.structure.Atom)" class="member-name-link">shift</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last even-row-color">
<div class="block">shift a structure with a vector.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#subtract(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">subtract</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">subtract two atoms ( a - b).</div>
</div>
<div class="col-first even-row-color"><code>static final double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#torsionAngle(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">torsionAngle</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;c,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;d)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate the torsion angle, i.e. the angle between the normal vectors of
 the two plains a-b-c and b-c-d.</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#transform(org.biojava.nbio.structure.Atom%5B%5D,javax.vecmath.Matrix4d)" class="member-name-link">transform</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 javax.vecmath.Matrix4d&nbsp;t)</code></div>
<div class="col-last odd-row-color">
<div class="block">Transform an array of atoms at once.</div>
</div>
<div class="col-first even-row-color"><code>static final void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#transform(org.biojava.nbio.structure.Atom,javax.vecmath.Matrix4d)" class="member-name-link">transform</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 javax.vecmath.Matrix4d&nbsp;m)</code></div>
<div class="col-last even-row-color">
<div class="block">Transforms an atom object, given a Matrix4d (i.e. the vecmath library
 double-precision 4x4 rotation+translation matrix).</div>
</div>
<div class="col-first odd-row-color"><code>static final void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#translate(org.biojava.nbio.structure.Atom,javax.vecmath.Vector3d)" class="member-name-link">translate</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 javax.vecmath.Vector3d&nbsp;v)</code></div>
<div class="col-last odd-row-color">
<div class="block">Translates an atom object, given a Vector3d (i.e. the vecmath library
 double-precision 3-d vector)</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#unitVector(org.biojava.nbio.structure.Atom)" class="member-name-link">unitVector</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the unit vector of vector a .</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#vectorProduct(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">vectorProduct</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;b)</code></div>
<div class="col-last odd-row-color">
<div class="block">Vector product (cross product).</div>
</div>
</div>
<div class="caption"><span>Method parameters in <a href="../package-summary.html">org.biojava.nbio.structure</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;javax.vecmath.Point3d&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Calc.</span><code><a href="../Calc.html#atomsToPoints(java.util.Collection)" class="member-name-link">atomsToPoints</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Collection.html" title="class or interface in java.util" class="external-link">Collection</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Convert an array of atoms into an array of vecmath points</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Group.</span><code><a href="../Group.html#setAtoms(java.util.List)" class="member-name-link">setAtoms</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Set the atoms of this group.</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">HetatomImpl.</span><code><a href="../HetatomImpl.html#setAtoms(java.util.List)" class="member-name-link">setAtoms</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Set the atoms of this group.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../package-summary.html">org.biojava.nbio.structure</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../AtomPositionMap.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">AtomPositionMap</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Creates a new AtomPositionMap containing peptide alpha-carbon atoms</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../AtomPositionMap.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.AtomPositionMap.GroupMatcher)" class="member-name-link">AtomPositionMap</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 <a href="../AtomPositionMap.GroupMatcher.html" title="interface in org.biojava.nbio.structure">AtomPositionMap.GroupMatcher</a>&nbsp;matcher)</code></div>
<div class="col-last odd-row-color">
<div class="block">Creates a new AtomPositionMap containing only atoms matched by <code>matcher</code>.</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../BondImpl.html#%3Cinit%3E(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,int)" class="member-name-link">BondImpl</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomA,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomB,
 int&nbsp;bondOrder)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructs a new bond from a pair of atoms and the bond order of the bond
 between them.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../BondImpl.html#%3Cinit%3E(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,int,boolean)" class="member-name-link">BondImpl</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomA,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atomB,
 int&nbsp;bondOrder,
 boolean&nbsp;addSelfToAtoms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Constructs a new bond from a pair of atoms and the bond order of the bond
 between them.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a></h2>
<div class="caption"><span>Methods in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructurePairAligner.</span><code><a href="../align/StructurePairAligner.html#getAlignmentAtoms(org.biojava.nbio.structure.Structure)" class="member-name-link">getAlignmentAtoms</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;s)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the atoms that are being used for the alignment.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CallableStructureAlignment.</span><code><a href="../align/CallableStructureAlignment.html#getCa1()" class="member-name-link">getCa1</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>abstract <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractStructureAlignment.</span><code><a href="../align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>abstract <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AbstractStructureAlignment.</span><code><a href="../align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">BioJavaStructureAlignment.</span><code><a href="../align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BioJavaStructureAlignment.</span><code><a href="../align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureAlignment.</span><code><a href="../align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureAlignment.</span><code><a href="../align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">
<div class="block">run an alignment and also send a bean containing the parameters.</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructurePairAligner.</span><code><a href="../align/StructurePairAligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate the protein structure superimposition, between two sets of
 atoms.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CallableStructureAlignment.</span><code><a href="../align/CallableStructureAlignment.html#setCa1(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setCa1</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPTwister.</span><code><a href="../align/AFPTwister.html#twistOptimized(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">twistOptimized</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">superimposing according to the optimized alignment</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPTwister.</span><code><a href="../align/AFPTwister.html#twistPDB(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">twistPDB</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">calculate the total rmsd of the blocks output a merged pdb file for both
 proteins protein 1, in chain A protein 2 is twisted according to the
 twists detected, in chain B</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../align/CallableStructureAlignment.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,org.biojava.nbio.structure.align.ce.ConfigStrucAligParams)" class="member-name-link">CallableStructureAlignment</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;algorithmName,
 <a href="../align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce">ConfigStrucAligParams</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructor for all-to-all alignment calculation.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.ce">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h2>
<div class="caption"><span>Methods in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GuiWrapper.</span><code><a href="../align/ce/GuiWrapper.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List,java.util.List)" class="member-name-link">getAtomArray</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatoms,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;nucs)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#getShift()" class="member-name-link">getShift</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Gets the shift from the last call to
 <a href="../align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean)"><code>calc_rmsd</code></a>.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#getShift()" class="member-name-link">getShift</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Gets the shift from the last call to
 <a href="../align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean,boolean)"><code>calc_rmsd</code></a>.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeCPMain.</span><code><a href="../align/ce/CeCPMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;param)</code></div>
<div class="col-last even-row-color">
<div class="block">Aligns ca1 and ca2 using a heuristic to check for CPs.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeMain.</span><code><a href="../align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeMain.</span><code><a href="../align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;param)</code></div>
<div class="col-last even-row-color">
<div class="block">Align ca2 onto ca1.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">OptimalCECPMain.</span><code><a href="../align/ce/OptimalCECPMain.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;param)</code></div>
<div class="col-last odd-row-color">
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">OptimalCECPMain.</span><code><a href="../align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object,org.biojava.nbio.structure.align.model.AFPChain%5B%5D)" class="member-name-link">alignOptimal</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;param,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code></div>
<div class="col-last even-row-color">
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">OptimalCECPMain.</span><code><a href="../align/ce/OptimalCECPMain.html#alignPermuted(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object,int)" class="member-name-link">alignPermuted</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;param,
 int&nbsp;cp)</code></div>
<div class="col-last odd-row-color">
<div class="block">Aligns ca1 with ca2 permuted by <i>cp</i> residues.</div>
</div>
<div class="col-first even-row-color"><code>double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean)" class="member-name-link">calc_rmsd</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
 int&nbsp;strLen,
 boolean&nbsp;storeTransform)</code></div>
<div class="col-last even-row-color">
<div class="block">superimpose and get rmsd</div>
</div>
<div class="col-first odd-row-color"><code>double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#calc_rmsd(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,boolean,boolean)" class="member-name-link">calc_rmsd</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;pro2,
 int&nbsp;strLen,
 boolean&nbsp;storeTransform,
 boolean&nbsp;show)</code></div>
<div class="col-last odd-row-color">
<div class="block">superimpose and get rmsd</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">convertAfpChain</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">copy data from this class into AFPChain container object.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">convertAfpChain</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">copy data from this class into AFPChain container object.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GuiWrapper.</span><code><a href="../align/ce/GuiWrapper.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">display</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">extractFragments</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">extractFragments</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeCPMain.</span><code><a href="../align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">filterDuplicateAFPs</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code></div>
<div class="col-last odd-row-color">
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeCPMain.</span><code><a href="../align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECPParameters)" class="member-name-link">filterDuplicateAFPs</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
 <a href="../align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GuiWrapper.</span><code><a href="../align/ce/GuiWrapper.html#getAlignedStructure(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlignedStructure</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GuiWrapper.</span><code><a href="../align/ce/GuiWrapper.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List,java.util.List)" class="member-name-link">getAtomArray</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatoms,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;nucs)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>double[][]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#initSumOfDistances(int,int,int,int,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">initSumOfDistances</a><wbr>(int&nbsp;nse1,
 int&nbsp;nse2,
 int&nbsp;winSize,
 int&nbsp;winSizeComb1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>double[][]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#initSumOfDistances(int,int,int,int,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">initSumOfDistances</a><wbr>(int&nbsp;nse1,
 int&nbsp;nse2,
 int&nbsp;winSize,
 int&nbsp;winSizeComb1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">nextStep</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">nextStep</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeCPMain.</span><code><a href="../align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECalculator)" class="member-name-link">postProcessAlignment</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code></div>
<div class="col-last odd-row-color">
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeCPMain.</span><code><a href="../align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.align.ce.CECPParameters)" class="member-name-link">postProcessAlignment</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
 <a href="../align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code></div>
<div class="col-last even-row-color">
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GuiWrapper.</span><code><a href="../align/ce/GuiWrapper.html#showAlignmentImage(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">showAlignmentImage</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;jmol)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">traceFragmentMatrix</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">traceFragmentMatrix</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static double[][]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CECalculator.</span><code><a href="../align/ce/CECalculator.html#updateMatrixWithSequenceConservation(double%5B%5D%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CeParameters)" class="member-name-link">updateMatrixWithSequenceConservation</a><wbr>(double[][]&nbsp;max,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</div>
<div class="col-first odd-row-color"><code>static double[][]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeCalculatorEnhanced.</span><code><a href="../align/ce/CeCalculatorEnhanced.html#updateMatrixWithSequenceConservation(double%5B%5D%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.ce.CeParameters)" class="member-name-link">updateMatrixWithSequenceConservation</a><wbr>(double[][]&nbsp;max,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">
<div class="block">Modifies an alignment matrix by favoring the alignment of similar and identical amino acids and penalizing the alignment of unrelated ones.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.fatcat">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a></h2>
<div class="caption"><span>Methods in <a href="../align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>protected <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FatCat.</span><code><a href="../align/fatcat/FatCat.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,boolean)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
 boolean&nbsp;doRigid)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FatCatFlexible.</span><code><a href="../align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FatCatFlexible.</span><code><a href="../align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;param)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FatCatRigid.</span><code><a href="../align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FatCatRigid.</span><code><a href="../align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;param)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FatCat.</span><code><a href="../align/fatcat/FatCat.html#alignFlexible(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)" class="member-name-link">alignFlexible</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FatCat.</span><code><a href="../align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">alignRigid</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FatCat.</span><code><a href="../align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)" class="member-name-link">alignRigid</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.fatcat.calc">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a></h2>
<div class="caption"><span>Methods in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FatCatAligner.</span><code><a href="../align/fatcat/calc/FatCatAligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean,org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 boolean&nbsp;doRigid,
 <a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>protected static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPChainer.</span><code><a href="../align/fatcat/calc/AFPChainer.html#calAfpRmsd(int,int%5B%5D,int,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">calAfpRmsd</a><wbr>(int&nbsp;afpn,
 int[]&nbsp;afpPositions,
 int&nbsp;listStart,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">//return the rmsd of the residues from the segments that form the given AFP list
        //this value can be a measurement (1) for the connectivity of the AFPs</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChainer.</span><code><a href="../align/fatcat/calc/AFPChainer.html#doChainAfp(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">doChainAfp</a><wbr>(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPCalculator.</span><code><a href="../align/fatcat/calc/AFPCalculator.html#extractAFPChains(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">extractAFPChains</a><wbr>(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPOptimizer.</span><code><a href="../align/fatcat/calc/AFPOptimizer.html#optimizeAln(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">optimizeAln</a><wbr>(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">optimize the alignment by dynamic programming</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPPostProcessor.</span><code><a href="../align/fatcat/calc/AFPPostProcessor.html#postProcess(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">postProcess</a><wbr>(<a href="../align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static final void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPCalculator.</span><code><a href="../align/fatcat/calc/AFPCalculator.html#sortAfps(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">sortAfps</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../align/fatcat/calc/StructureAlignmentOptimizer.html#%3Cinit%3E(int,int,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.Atom%5B%5D,int,int%5B%5D%5B%5D)" class="member-name-link">StructureAlignmentOptimizer</a><wbr>(int&nbsp;b1,
 int&nbsp;end1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c1,
 int&nbsp;b2,
 int&nbsp;end2,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;c2,
 int&nbsp;iniLen,
 int[][]&nbsp;iniSet)</code></div>
<div class="col-last even-row-color">
<div class="block">optimize the structural alignment by update the equivalent residues
 and then run dynamic programming
 input: len1 the length of structure 1; c1: the structure information of 1
        len2 the length of structure 2; c2: the structure information of 2
        iniLen and iniSet is the length and list of initial equivalent residues</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.gui">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h2>
<div class="caption"><span>Methods in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List)" class="member-name-link">getAtomArray</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the first atom for each group</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,int,int,org.biojava.nbio.structure.Atom%5B%5D,boolean)" class="member-name-link">getAtomForAligPos</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 int&nbsp;chainNr,
 int&nbsp;aligPos,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 boolean&nbsp;getPrevious)</code></div>
<div class="col-last odd-row-color">
<div class="block">Return the atom at alignment position aligPos. at the present only works with block 0</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">createArtificalStructure</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Group%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.util.List,java.util.List)" class="member-name-link">display</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Note: ca2, hetatoms2 and nucleotides2 should not be rotated.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureAlignmentDisplay.</span><code><a href="../align/gui/StructureAlignmentDisplay.html#display(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">display</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Display an AFPChain alignment</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#getAtomArray(org.biojava.nbio.structure.Atom%5B%5D,java.util.List)" class="member-name-link">getAtomArray</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the first atom for each group</div>
</div>
<div class="col-first even-row-color"><code>static final <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,int,int,org.biojava.nbio.structure.Atom%5B%5D,boolean)" class="member-name-link">getAtomForAligPos</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 int&nbsp;chainNr,
 int&nbsp;aligPos,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 boolean&nbsp;getPrevious)</code></div>
<div class="col-last even-row-color">
<div class="block">Return the atom at alignment position aligPos. at the present only works with block 0</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#getPDBresnum(int,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getPDBresnum</a><wbr>(int&nbsp;aligPos,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last odd-row-color">
<div class="block">Return a list of pdb Strings corresponding to the aligned positions of the molecule.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentJmolDisplay.</span><code><a href="../align/gui/MultipleAlignmentJmolDisplay.html#getPDBresnum(int,org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getPDBresnum</a><wbr>(int&nbsp;structNum,
 <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last even-row-color">
<div class="block">Utility method used in the <a href="../align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol"><code>MultipleAlignmentJmol</code></a> Frame,
 when the aligned residues of a structure in the alignment have
 to be selected for formatting them (coloring and style).</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">DisplayAFP.</span><code><a href="../align/gui/DisplayAFP.html#showAlignmentPanel(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)" class="member-name-link">showAlignmentPanel</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.gui.aligpanel">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h2>
<div class="caption"><span>Methods in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AligPanel.</span><code><a href="../align/gui/aligpanel/AligPanel.html#getCa1()" class="member-name-link">getCa1</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StatusDisplay.</span><code><a href="../align/gui/aligpanel/StatusDisplay.html#getCa1()" class="member-name-link">getCa1</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AligPanel.</span><code><a href="../align/gui/aligpanel/AligPanel.html#getCa2()" class="member-name-link">getCa2</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StatusDisplay.</span><code><a href="../align/gui/aligpanel/StatusDisplay.html#getCa2()" class="member-name-link">getCa2</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AligPanel.</span><code><a href="../align/gui/aligpanel/AligPanel.html#setCa1(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setCa1</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StatusDisplay.</span><code><a href="../align/gui/aligpanel/StatusDisplay.html#setCa1(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setCa1</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AligPanel.</span><code><a href="../align/gui/aligpanel/AligPanel.html#setCa2(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setCa2</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StatusDisplay.</span><code><a href="../align/gui/aligpanel/StatusDisplay.html#setCa2(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setCa2</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../align/gui/aligpanel/MultipleAligPanel.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)" class="member-name-link">MultipleAligPanel</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructor using an afpChain and the atom arrays for pairwise
 alignments.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.gui.jmol">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h2>
<div class="caption"><span>Methods in <a href="../align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureAlignmentJmol.</span><code><a href="../align/gui/jmol/StructureAlignmentJmol.html#getJmolScript4Block(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">getJmolScript4Block</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;blockNr)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureAlignmentJmol.</span><code><a href="../align/gui/jmol/StructureAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getJmolString</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">JmolTools.</span><code><a href="../align/gui/jmol/JmolTools.html#getPdbInfo(org.biojava.nbio.structure.Atom)" class="member-name-link">getPdbInfo</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last even-row-color">
<div class="block">get jmol style info:
  jmol style: [MET]508:A.CA/1 #3918
  insertion code: [ASP]1^A:A.CA/1 #2</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">JmolTools.</span><code><a href="../align/gui/jmol/JmolTools.html#getPdbInfo(org.biojava.nbio.structure.Atom,boolean)" class="member-name-link">getPdbInfo</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 boolean&nbsp;printResName)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractAlignmentJmol.</span><code><a href="../align/gui/jmol/AbstractAlignmentJmol.html#setAtoms(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Create and set a new structure from a given atom array.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../align/gui/jmol/StructureAlignmentJmol.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">StructureAlignmentJmol</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.helper">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a></h2>
<div class="caption"><span>Methods in <a href="../align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getCenter(org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getCenter</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 int&nbsp;pos,
 int&nbsp;fragmentLength)</code></div>
<div class="col-last even-row-color">
<div class="block">Get the centroid for the set of atoms starting from position pos, length fragmentLenght</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getFragment(org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getFragment</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
 int&nbsp;pos,
 int&nbsp;fragmentLength)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get a continue subset of Atoms based by the starting position and the length</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getFragmentFromIdxList(org.biojava.nbio.structure.Atom%5B%5D,int%5B%5D)" class="member-name-link">getFragmentFromIdxList</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
 int[]&nbsp;idx)</code></div>
<div class="col-last even-row-color">
<div class="block">Get a subset of Atoms based by their positions</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getFragmentNoClone(org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getFragmentNoClone</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
 int&nbsp;pos,
 int&nbsp;fragmentLength)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get a continue subset of Atoms based by the starting position and the length
 does not clone the original atoms.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../align/helper/package-summary.html">org.biojava.nbio.structure.align.helper</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getCenter(org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getCenter</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 int&nbsp;pos,
 int&nbsp;fragmentLength)</code></div>
<div class="col-last even-row-color">
<div class="block">Get the centroid for the set of atoms starting from position pos, length fragmentLenght</div>
</div>
<div class="col-first odd-row-color"><code>static double[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getDiagonalAtK(org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">getDiagonalAtK</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;k)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get distances along diagonal k from coordinate array coords.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getDistanceMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getDistanceMatrix</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Matrix of all distances between two sets of Atoms.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getFragment(org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getFragment</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
 int&nbsp;pos,
 int&nbsp;fragmentLength)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get a continue subset of Atoms based by the starting position and the length</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getFragmentFromIdxList(org.biojava.nbio.structure.Atom%5B%5D,int%5B%5D)" class="member-name-link">getFragmentFromIdxList</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
 int[]&nbsp;idx)</code></div>
<div class="col-last even-row-color">
<div class="block">Get a subset of Atoms based by their positions</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignUtils.</span><code><a href="../align/helper/AlignUtils.html#getFragmentNoClone(org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getFragmentNoClone</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;caall,
 int&nbsp;pos,
 int&nbsp;fragmentLength)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get a continue subset of Atoms based by the starting position and the length
 does not clone the original atoms.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.model">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/model/package-summary.html">org.biojava.nbio.structure.align.model</a></h2>
<div class="caption"><span>Methods in <a href="../align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChain.</span><code><a href="../align/model/AFPChain.html#getBlockShiftVector()" class="member-name-link">getBlockShiftVector</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChain.</span><code><a href="../align/model/AFPChain.html#setBlockShiftVector(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setBlockShiftVector</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;blockShiftVector)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AfpChainWriter.</span><code><a href="../align/model/AfpChainWriter.html#toAlignedPairs(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toAlignedPairs</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Prints the alignment in the simplest form: a list of aligned residues.</div>
</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChain.</span><code><a href="../align/model/AFPChain.html#toCE(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toCE</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AfpChainWriter.</span><code><a href="../align/model/AfpChainWriter.html#toCE(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toCE</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChain.</span><code><a href="../align/model/AFPChain.html#toFatcat(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toFatcat</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AfpChainWriter.</span><code><a href="../align/model/AfpChainWriter.html#toFatCat(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toFatCat</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AfpChainWriter.</span><code><a href="../align/model/AfpChainWriter.html#toFatCatCore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean,boolean,boolean,boolean)" class="member-name-link">toFatCatCore</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 boolean&nbsp;printLegend,
 boolean&nbsp;longHeader,
 boolean&nbsp;showHTML,
 boolean&nbsp;showAlignmentBlock)</code></div>
<div class="col-last even-row-color">
<div class="block">Output in FatCatCore format</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AfpChainWriter.</span><code><a href="../align/model/AfpChainWriter.html#toPrettyAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean,boolean)" class="member-name-link">toPrettyAlignment</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 boolean&nbsp;showHTML,
 boolean&nbsp;showAlignmentBlock)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AfpChainWriter.</span><code><a href="../align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toWebSiteDisplay</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AfpChainWriter.</span><code><a href="../align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)" class="member-name-link">toWebSiteDisplay</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 boolean&nbsp;showAlignmentBlock)</code></div>
<div class="col-last odd-row-color">
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.multiple">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a></h2>
<div class="caption"><span>Constructors in <a href="../align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../align/multiple/MultipleAlignmentEnsembleImpl.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)" class="member-name-link">MultipleAlignmentEnsembleImpl</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afp,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 boolean&nbsp;flexible)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructor from an AFPChain instance.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.multiple.util">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h2>
<div class="caption"><span>Methods in <a href="../align/multiple/util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../align/multiple/util/MultipleAlignmentTools.html#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int,int)" class="member-name-link">getAtomForSequencePosition</a><wbr>(<a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct,
 int&nbsp;str,
 int&nbsp;sequencePos)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.pairwise">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a></h2>
<div class="caption"><span>Methods in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#getCenter1()" class="member-name-link">getCenter1</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#getCenter2()" class="member-name-link">getCenter2</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlternativeAlignment.</span><code><a href="../align/pairwise/AlternativeAlignment.html#getShift()" class="member-name-link">getShift</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">returns the shift vector that has to be applied on structure to to shift on structure one</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#getTrans()" class="member-name-link">getTrans</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#getUnitv()" class="member-name-link">getUnitv</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../align/pairwise/package-summary.html">org.biojava.nbio.structure.align.pairwise</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FragmentJoiner.</span><code><a href="../align/pairwise/FragmentJoiner.html#approach_ap3(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.pairwise.FragmentPair%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)" class="member-name-link">approach_ap3</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a>[]&nbsp;fraglst,
 <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlternativeAlignment.</span><code><a href="../align/pairwise/AlternativeAlignment.html#calcScores(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">calcScores</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">calculates  scores for this alignment ( %id )</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlternativeAlignment.</span><code><a href="../align/pairwise/AlternativeAlignment.html#calculateSuperpositionByIdx(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">calculateSuperpositionByIdx</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FragmentJoiner.</span><code><a href="../align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments%5B%5D,org.biojava.nbio.structure.align.StrucAligParameters)" class="member-name-link">extendFragments</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>[]&nbsp;fragments,
 <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FragmentJoiner.</span><code><a href="../align/pairwise/FragmentJoiner.html#extendFragments(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments,org.biojava.nbio.structure.align.StrucAligParameters)" class="member-name-link">extendFragments</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;fragments,
 <a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlternativeAlignment.</span><code><a href="../align/pairwise/AlternativeAlignment.html#finish(org.biojava.nbio.structure.align.StrucAligParameters,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">finish</a><wbr>(<a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlternativeAlignment.</span><code><a href="../align/pairwise/AlternativeAlignment.html#getDistanceMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getDistanceMatrix</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FragmentJoiner.</span><code><a href="../align/pairwise/FragmentJoiner.html#getRMS(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.align.helper.JointFragments)" class="member-name-link">getRMS</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a>&nbsp;frag)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get the RMS of the JointFragments pair frag</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlternativeAlignment.</span><code><a href="../align/pairwise/AlternativeAlignment.html#refine(org.biojava.nbio.structure.align.StrucAligParameters,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">refine</a><wbr>(<a href="../align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a>&nbsp;params,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Refinement procedure based on superposition and dynamic programming.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#setCenter1(org.biojava.nbio.structure.Atom)" class="member-name-link">setCenter1</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;center1)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#setCenter2(org.biojava.nbio.structure.Atom)" class="member-name-link">setCenter2</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;center2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#setTrans(org.biojava.nbio.structure.Atom)" class="member-name-link">setTrans</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;trans)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FragmentPair.</span><code><a href="../align/pairwise/FragmentPair.html#setUnitv(org.biojava.nbio.structure.Atom)" class="member-name-link">setUnitv</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;unitv)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentProgressListener.</span><code><a href="../align/pairwise/AlignmentProgressListener.html#startingAlignment(java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">startingAlignment</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name2,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.quaternary">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a></h2>
<div class="caption"><span>Methods in <a href="../align/quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">QsAlignResult.</span><code><a href="../align/quaternary/QsAlignResult.html#getAlignedAtomsForSubunits1(int)" class="member-name-link">getAlignedAtomsForSubunits1</a><wbr>(int&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">QsAlignResult.</span><code><a href="../align/quaternary/QsAlignResult.html#getAlignedAtomsForSubunits2(int)" class="member-name-link">getAlignedAtomsForSubunits2</a><wbr>(int&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.seq">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a></h2>
<div class="caption"><span>Methods in <a href="../align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SmithWaterman3Daligner.</span><code><a href="../align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SmithWaterman3Daligner.</span><code><a href="../align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;parameters)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.util">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a></h2>
<div class="caption"><span>Methods in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms1(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlignedAtoms1</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms2(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlignedAtoms2</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AtomCache.</span><code><a href="../align/util/AtomCache.html#getAtoms(java.lang.String)" class="member-name-link">getAtoms</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the CA atoms for the provided name.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AtomCache.</span><code><a href="../align/util/AtomCache.html#getAtoms(org.biojava.nbio.structure.StructureIdentifier)" class="member-name-link">getAtoms</a><wbr>(<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getProjectedPoint(org.biojava.nbio.structure.Atom)" class="member-name-link">getProjectedPoint</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code></div>
<div class="col-last even-row-color">
<div class="block">Projects a given point onto the axis of rotation</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AtomCache.</span><code><a href="../align/util/AtomCache.html#getRepresentativeAtoms(java.lang.String)" class="member-name-link">getRepresentativeAtoms</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the representative atoms for the provided name.</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AtomCache.</span><code><a href="../align/util/AtomCache.html#getRepresentativeAtoms(org.biojava.nbio.structure.StructureIdentifier)" class="member-name-link">getRepresentativeAtoms</a><wbr>(<a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;name)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getRotationAxis()" class="member-name-link">getRotationAxis</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get a unit vector along the rotation axis</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getRotationPos()" class="member-name-link">getRotationPos</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Get a position on the rotation axis.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getScrewTranslation()" class="member-name-link">getScrewTranslation</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get the component of translation parallel to the axis of rotation</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getTranslation</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getAxisEnds(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAxisEnds</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Find a segment of the axis that covers the specified set of atoms.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#alignmentToSIF(java.io.Writer,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,java.lang.String)" class="member-name-link">alignmentToSIF</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Writer.html" title="class or interface in java.io" class="external-link">Writer</a>&nbsp;out,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;backboneInteraction,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;alignmentInteraction)</code></div>
<div class="col-last even-row-color">
<div class="block">Creates a simple interaction format (SIF) file for an alignment.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#createAFPChain(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.ResidueNumber%5B%5D,org.biojava.nbio.structure.ResidueNumber%5B%5D)" class="member-name-link">createAFPChain</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned1,
 <a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Fundamentally, an alignment is just a list of aligned residues in each
 protein.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">createArtificalStructure</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#deleteColumn(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">deleteColumn</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;block,
 int&nbsp;pos)</code></div>
<div class="col-last odd-row-color">
<div class="block">Delete an alignment position from the original alignment object.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#deleteHighestDistanceColumn(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">deleteHighestDistanceColumn</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Find the alignment position with the highest atomic distance between the
 equivalent atomic positions of the arrays and remove it from the
 alignment.</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#fillAlignedAtomArrays(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">fillAlignedAtomArrays</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1aligned,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2aligned)</code></div>
<div class="col-last odd-row-color">
<div class="block">Fill the aligned Atom arrays with the equivalent residues in the afpChain.</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlign</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Extract the alignment output</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)" class="member-name-link">getAlign</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 boolean&nbsp;showSeq)</code></div>
<div class="col-last odd-row-color">
<div class="block">Sets the following properties:
 
 The alignment strings <a href="../align/model/AFPChain.html#setAlnseq1(char%5B%5D)"><code>alnseq1</code></a>,
  <a href="../align/model/AFPChain.html#setAlnseq2(char%5B%5D)"><code>alnseq2</code></a>,
  and <a href="../align/model/AFPChain.html#setAlnsymb(char%5B%5D)"><code>alnsymb</code></a>
 <a href="../align/model/AFPChain.html#setAlnbeg1(int)"><code>alnbeg1</code></a> and 2
 <a href="../align/model/AFPChain.html#setAlnLength(int)"><code>alnLength</code></a> and
  <a href="../align/model/AFPChain.html#setGapLen(int)"><code>gapLen</code></a>
 </div>
</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms1(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlignedAtoms1</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getAlignedAtoms2(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlignedAtoms2</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Chain.html" title="interface in org.biojava.nbio.structure">Chain</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#getAlignedModel(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlignedModel</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last even-row-color">
<div class="block">get an artificial List of chains containing the Atoms and groups.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#getAlignedStructure(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAlignedStructure</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get an artifical Structure containing both chains.</div>
</div>
<div class="col-first even-row-color"><code><a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getAxisEnds(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAxisEnds</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Find a segment of the axis that covers the specified set of atoms.</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getJmolScript(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getJmolScript</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns a Jmol script which will display the axis of rotation.</div>
</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getJmolScript(org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">getJmolScript</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;axisID)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns a Jmol script which will display the axis of rotation.</div>
</div>
<div class="col-first odd-row-color"><code>double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getProjectedDistance(org.biojava.nbio.structure.Atom)" class="member-name-link">getProjectedDistance</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get the distance from a point to the axis of rotation</div>
</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#getProjectedPoint(org.biojava.nbio.structure.Atom)" class="member-name-link">getProjectedPoint</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;point)</code></div>
<div class="col-last even-row-color">
<div class="block">Projects a given point onto the axis of rotation</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getRotMax(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getRotMax</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChainScorer.</span><code><a href="../align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getTMScore</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPChainScorer.</span><code><a href="../align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,boolean)" class="member-name-link">getTMScore</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 boolean&nbsp;normalizeMin)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPAlignmentDisplay.</span><code><a href="../align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getTranslation</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">prepareGroupsForDisplay</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Rotate the Atoms/Groups so they are aligned for the 3D visualisation</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#replaceOptAln(int%5B%5D%5B%5D%5B%5D,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">replaceOptAln</a><wbr>(int[][][]&nbsp;newAlgn,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int%5B%5D,int%5B%5D%5B%5D%5B%5D)" class="member-name-link">replaceOptAln</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;blockNum,
 int[]&nbsp;optLens,
 int[][][]&nbsp;optAln)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.util.Map)" class="member-name-link">replaceOptAln</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>,<wbr><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RotationAxis.</span><code><a href="../align/util/RotationAxis.html#rotate(org.biojava.nbio.structure.Atom%5B%5D,double)" class="member-name-link">rotate</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 double&nbsp;theta)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#shiftCA2(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.jama.Matrix,org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Group%5B%5D)" class="member-name-link">shiftCA2</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;shift,
 <a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups)</code></div>
<div class="col-last even-row-color">
<div class="block">only shift CA positions.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">splitBlocksByTopology</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AlignmentTools.</span><code><a href="../align/util/AlignmentTools.html#updateSuperposition(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">updateSuperposition</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">After the alignment changes (optAln, optLen, blockNum, at a minimum),
 many other properties which depend on the superposition will be invalid.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../align/util/RotationAxis.html#%3Cinit%3E(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom,double)" class="member-name-link">RotationAxis</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;axis,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;pos,
 double&nbsp;theta)</code></div>
<div class="col-last even-row-color">
<div class="block">Create a rotation axis from a vector, a point, and an angle.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../align/util/RotationAxis.html#%3Cinit%3E(org.biojava.nbio.structure.jama.Matrix,org.biojava.nbio.structure.Atom)" class="member-name-link">RotationAxis</a><wbr>(<a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;rotation,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;translation)</code></div>
<div class="col-last odd-row-color">
<div class="block">Determine the location of the rotation axis based on a rotation matrix and a translation vector</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.xml">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h2>
<div class="caption"><span>Methods in <a href="../align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChainXMLParser.</span><code><a href="../align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,java.lang.String,java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">fromXML</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;xml,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name1,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name2,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">new utility method that checks that the order of the pair in the XML alignment is correct and flips the direction if needed</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPChainXMLParser.</span><code><a href="../align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">fromXML</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;xml,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChainXMLConverter.</span><code><a href="../align/xml/AFPChainXMLConverter.html#printXMLEQRInferPositions(org.biojava.nbio.core.util.PrettyXMLWriter,org.biojava.nbio.structure.align.model.AFPChain,int,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">printXMLEQRInferPositions</a><wbr>(<a href="../../core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
 <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 int&nbsp;bk,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPChainXMLParser.</span><code><a href="../align/xml/AFPChainXMLParser.html#rebuildAFPChain(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">rebuildAFPChain</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">replace the PDB res nums with atom positions:</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AFPChainXMLConverter.</span><code><a href="../align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,java.io.StringWriter,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toXML</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/StringWriter.html" title="class or interface in java.io" class="external-link">StringWriter</a>&nbsp;swriter,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last even-row-color">
<div class="block">Write the XML representation to a StringWriter</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AFPChainXMLConverter.</span><code><a href="../align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">toXML</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Convert an afpChain to a simple XML representation</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.asa">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../asa/package-summary.html">org.biojava.nbio.structure.asa</a></h2>
<div class="caption"><span>Methods in <a href="../asa/package-summary.html">org.biojava.nbio.structure.asa</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AsaCalculator.</span><code><a href="../asa/AsaCalculator.html#getRadius(org.biojava.nbio.structure.Atom)" class="member-name-link">getRadius</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom)</code></div>
<div class="col-last even-row-color">
<div class="block">Gets the van der Waals radius of the given atom following the values defined by
 Chothia (1976) J.Mol.Biol.105,1-14
 NOTE: the vdw values defined by the paper assume no Hydrogens and thus "inflates"
 slightly the heavy atoms to account for Hydrogens.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../asa/package-summary.html">org.biojava.nbio.structure.asa</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../asa/AsaCalculator.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,double,int,int)" class="member-name-link">AsaCalculator</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 double&nbsp;probe,
 int&nbsp;nSpherePoints,
 int&nbsp;nThreads)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructs a new AsaCalculator.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.cluster">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h2>
<div class="caption"><span>Methods in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SubunitCluster.</span><code><a href="../cluster/SubunitCluster.html#getAlignedAtomsSubunit(int)" class="member-name-link">getAlignedAtomsSubunit</a><wbr>(int&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Subunit.</span><code><a href="../cluster/Subunit.html#getRepresentativeAtoms()" class="member-name-link">getRepresentativeAtoms</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Get all the representative Atoms of the Subunit.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../cluster/Subunit.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,org.biojava.nbio.structure.StructureIdentifier,org.biojava.nbio.structure.Structure)" class="member-name-link">Subunit</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;reprAtoms,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;name,
 <a href="../StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a>&nbsp;identifier,
 <a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code></div>
<div class="col-last even-row-color">
<div class="block">A Subunit is solely defined by the coordinates of the representative
 Atoms of its residues.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.contact">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a></h2>
<div class="caption"><span>Methods in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>protected <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureInterface.</span><code><a href="../contact/StructureInterface.html#getAtomsForAsa(int)" class="member-name-link">getAtomsForAsa</a><wbr>(int&nbsp;cofactorSizeToUse)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>protected <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StructureInterface.</span><code><a href="../contact/StructureInterface.html#getFirstAtomsForAsa(int)" class="member-name-link">getFirstAtomsForAsa</a><wbr>(int&nbsp;cofactorSizeToUse)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>protected <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StructureInterface.</span><code><a href="../contact/StructureInterface.html#getSecondAtomsForAsa(int)" class="member-name-link">getSecondAtomsForAsa</a><wbr>(int&nbsp;cofactorSizeToUse)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AtomContact.</span><code><a href="../contact/AtomContact.html#getPair()" class="member-name-link">getPair</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Grid.</span><code><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">addAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Adds a set of atoms, subsequent call to <a href="../contact/Grid.html#getIndicesContacts()"><code>Grid.getIndicesContacts()</code></a> or <a href="../contact/Grid.html#getAtomContacts()"><code>Grid.getAtomContacts()</code></a> will produce the interatomic contacts.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Grid.</span><code><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">addAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Adds the i and j atoms and fills the grid.</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Grid.</span><code><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.contact.BoundingBox)" class="member-name-link">addAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 <a href="../contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;bounds)</code></div>
<div class="col-last even-row-color">
<div class="block">Adds a set of atoms, subsequent call to <a href="../contact/Grid.html#getIndicesContacts()"><code>Grid.getIndicesContacts()</code></a> or <a href="../contact/Grid.html#getAtomContacts()"><code>Grid.getAtomContacts()</code></a> will produce the interatomic contacts.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">Grid.</span><code><a href="../contact/Grid.html#addAtoms(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.contact.BoundingBox,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.contact.BoundingBox)" class="member-name-link">addAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;iAtoms,
 <a href="../contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;icoordbounds,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;jAtoms,
 <a href="../contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a>&nbsp;jcoordbounds)</code></div>
<div class="col-last odd-row-color">
<div class="block">Adds the i and j atoms and fills the grid, passing their bounds (array of size 6 with x,y,z minima and x,y,z maxima)
 This way the bounds don't need to be recomputed.</div>
</div>
<div class="col-first even-row-color"><code><a href="../contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AtomContactSet.</span><code><a href="../contact/AtomContactSet.html#getContact(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">getContact</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the corresponding AtomContact or null if no contact exists between the 2 given atoms</div>
</div>
<div class="col-first odd-row-color"><code>boolean</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AtomContactSet.</span><code><a href="../contact/AtomContactSet.html#hasContact(org.biojava.nbio.structure.Atom,org.biojava.nbio.structure.Atom)" class="member-name-link">hasContact</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom2)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Method parameters in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AtomContact.</span><code><a href="../contact/AtomContact.html#setPair(org.biojava.nbio.structure.contact.Pair)" class="member-name-link">setPair</a><wbr>(<a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;pair)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../contact/StructureInterface.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,java.lang.String,java.lang.String,org.biojava.nbio.structure.contact.AtomContactSet,org.biojava.nbio.structure.xtal.CrystalTransform,org.biojava.nbio.structure.xtal.CrystalTransform)" class="member-name-link">StructureInterface</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;firstMolecule,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;secondMolecule,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;firstMoleculeId,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;secondMoleculeId,
 <a href="../contact/AtomContactSet.html" title="class in org.biojava.nbio.structure.contact">AtomContactSet</a>&nbsp;contacts,
 <a href="../xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;firstTransf,
 <a href="../xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a>&nbsp;secondTransf)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructs a StructureInterface</div>
</div>
</div>
<div class="caption"><span>Constructor parameters in <a href="../contact/package-summary.html">org.biojava.nbio.structure.contact</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../contact/AtomContact.html#%3Cinit%3E(org.biojava.nbio.structure.contact.Pair,double)" class="member-name-link">AtomContact</a><wbr>(<a href="../contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;pair,
 double&nbsp;distance)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.domain">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../domain/package-summary.html">org.biojava.nbio.structure.domain</a></h2>
<div class="caption"><span>Methods in <a href="../domain/package-summary.html">org.biojava.nbio.structure.domain</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">LocalProteinDomainParser.</span><code><a href="../domain/LocalProteinDomainParser.html#suggestDomains(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">suggestDomains</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last even-row-color">
<div class="block">Suggest domains for a set of Calpha atoms</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.domain.pdp">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a></h2>
<div class="caption"><span>Methods in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">Cut.</span><code><a href="../domain/pdp/Cut.html#cut(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.domain.pdp.Domain,org.biojava.nbio.structure.domain.pdp.CutValues,int%5B%5D%5B%5D,org.biojava.nbio.structure.domain.pdp.PDPDistanceMatrix)" class="member-name-link">cut</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="../domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a>&nbsp;dom,
 <a href="../domain/pdp/CutValues.html" title="class in org.biojava.nbio.structure.domain.pdp">CutValues</a>&nbsp;val,
 int[][]&nbsp;dist,
 <a href="../domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a>&nbsp;pdpMatrix)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GetDistanceMatrix.</span><code><a href="../domain/pdp/GetDistanceMatrix.html#getDistanceMatrix(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getDistanceMatrix</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;protein)</code></div>
<div class="col-last odd-row-color">
<div class="block">A set of Calpha atoms that are representing the protein</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../domain/pdp/package-summary.html">org.biojava.nbio.structure.domain.pdp</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../domain/pdp/CutDomain.html#%3Cinit%3E(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.domain.pdp.PDPDistanceMatrix)" class="member-name-link">CutDomain</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
 <a href="../domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a>&nbsp;pdpMatrix)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.io">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../io/package-summary.html">org.biojava.nbio.structure.io</a></h2>
<div class="caption"><span>Methods in <a href="../io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileConvert.</span><code><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom)" class="member-name-link">toPDB</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a)</code></div>
<div class="col-last even-row-color">
<div class="block">Prints the content of an Atom object as a PDB formatted line.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileConvert.</span><code><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,java.lang.String)" class="member-name-link">toPDB</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chainId)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileConvert.</span><code><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,java.lang.StringBuffer)" class="member-name-link">toPDB</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/StringBuffer.html" title="class or interface in java.lang" class="external-link">StringBuffer</a>&nbsp;str)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileConvert.</span><code><a href="../io/FileConvert.html#toPDB(org.biojava.nbio.structure.Atom,java.lang.StringBuffer,java.lang.String)" class="member-name-link">toPDB</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;a,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/StringBuffer.html" title="class or interface in java.lang" class="external-link">StringBuffer</a>&nbsp;str,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chainID)</code></div>
<div class="col-last odd-row-color">
<div class="block">Print ATOM record in the following syntax</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.io.cif">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../io/cif/package-summary.html">org.biojava.nbio.structure.io.cif</a></h2>
<div class="caption"><span>Methods in <a href="../io/cif/package-summary.html">org.biojava.nbio.structure.io.cif</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractCifFileSupplier.WrappedAtom.</span><code><a href="../io/cif/AbstractCifFileSupplier.WrappedAtom.html#getAtom()" class="member-name-link">getAtom</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../io/cif/package-summary.html">org.biojava.nbio.structure.io.cif</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../io/cif/AbstractCifFileSupplier.WrappedAtom.html#%3Cinit%3E(int,java.lang.String,java.lang.String,org.biojava.nbio.structure.Atom,int)" class="member-name-link">WrappedAtom</a><wbr>(int&nbsp;model,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chainName,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chainId,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;atom,
 int&nbsp;atomId)</code></div>
<div class="col-last even-row-color">
<div class="block">Construct a new atoms.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.io.mmtf">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a></h2>
<div class="caption"><span>Methods in <a href="../io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a> that return types with arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MmtfSummaryDataBean.</span><code><a href="../io/mmtf/MmtfSummaryDataBean.html#getAllAtoms()" class="member-name-link">getAllAtoms</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MmtfUtils.</span><code><a href="../io/mmtf/MmtfUtils.html#getAtomsForGroup(org.biojava.nbio.structure.Group)" class="member-name-link">getAtomsForGroup</a><wbr>(<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;inputGroup)</code></div>
<div class="col-last odd-row-color">
<div class="block">Get a list of atoms for a group.</div>
</div>
</div>
<div class="caption"><span>Method parameters in <a href="../io/mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a> with type arguments of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MmtfUtils.</span><code><a href="../io/mmtf/MmtfUtils.html#getNumBondsInGroup(java.util.List)" class="member-name-link">getNumBondsInGroup</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;atomsInGroup)</code></div>
<div class="col-last even-row-color">
<div class="block">Find the number of bonds in a group</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MmtfSummaryDataBean.</span><code><a href="../io/mmtf/MmtfSummaryDataBean.html#setAllAtoms(java.util.List)" class="member-name-link">setAllAtoms</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&gt;&nbsp;allAtoms)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.symmetry.internal">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h2>
<div class="caption"><span>Methods in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeSymmResult.</span><code><a href="../symmetry/internal/CeSymmResult.html#getAtoms()" class="member-name-link">getAtoms</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>protected static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeSymm.</span><code><a href="../symmetry/internal/CeSymm.html#align(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)" class="member-name-link">align</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeSymm.</span><code><a href="../symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">analyze</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last odd-row-color">
<div class="block">Analyze the symmetries of the input Atom array using the DEFAULT
 parameters.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeSymm.</span><code><a href="../symmetry/internal/CeSymm.html#analyze(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)" class="member-name-link">analyze</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></div>
<div class="col-last even-row-color">
<div class="block">Analyze the symmetries of the input Atom array using the provided
 parameters.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeSymm.</span><code><a href="../symmetry/internal/CeSymm.html#analyzeLevel(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.symmetry.internal.CESymmParameters)" class="member-name-link">analyzeLevel</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 <a href="../symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">
<div class="block">Analyze a single level of symmetry.</div>
</div>
<div class="col-first even-row-color"><code>int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AngleOrderDetectorPlus.</span><code><a href="../symmetry/internal/AngleOrderDetectorPlus.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">calculateOrder</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>int</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GraphComponentOrderDetector.</span><code><a href="../symmetry/internal/GraphComponentOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">calculateOrder</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">OrderDetector.</span><code><a href="../symmetry/internal/OrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">calculateOrder</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>int</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SequenceFunctionOrderDetector.</span><code><a href="../symmetry/internal/SequenceFunctionOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">calculateOrder</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeSymmIterative.</span><code><a href="../symmetry/internal/CeSymmIterative.html#execute(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">execute</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">This method uses iteratively CeSymm to calculate all symmetries in the
 input array of atoms and organize them in a multiple alignment of the
 repeats.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GraphComponentRefiner.</span><code><a href="../symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">refine</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;order)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceFunctionRefiner.</span><code><a href="../symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">refine</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;order)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryRefiner.</span><code><a href="../symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">refine</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;order)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceFunctionRefiner.</span><code><a href="../symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">refineSymmetry</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;k)</code></div>
<div class="col-last even-row-color">
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CeSymmResult.</span><code><a href="../symmetry/internal/CeSymmResult.html#setAtoms(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">setAtoms</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.symmetry.utils">
<h2>Uses of <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a> in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h2>
<div class="caption"><span>Methods in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that return <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#getRepresentativeAtoms(org.biojava.nbio.structure.Structure)" class="member-name-link">getRepresentativeAtoms</a><wbr>(<a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the representative Atom Array of the first model, if the
 structure is NMR, or the Array for each model, if it is a biological
 assembly with multiple models.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static boolean[][]</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#blankOutBreakFlag(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,boolean%5B%5D%5B%5D,int)" class="member-name-link">blankOutBreakFlag</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;rows,
 int&nbsp;cols,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
 boolean[][]&nbsp;breakFlag,
 int&nbsp;blankWindowSize)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#blankOutCEOrig(org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int)" class="member-name-link">blankOutCEOrig</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;rows,
 int&nbsp;cols,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
 <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;origM,
 int&nbsp;blankWindowSize)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#blankOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int)" class="member-name-link">blankOutPreviousAlignment</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;rows,
 int&nbsp;cols,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
 <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
 int&nbsp;blankWindowSize)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#buildSymmetryGraph(java.util.List,org.biojava.nbio.structure.Atom%5B%5D,boolean)" class="member-name-link">buildSymmetryGraph</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&gt;&nbsp;afps,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 boolean&nbsp;undirected)</code></div>
<div class="col-last odd-row-color">
<div class="block">Converts a set of AFP alignments into a Graph of aligned residues, where
 each vertex is a residue and each edge means the connection between the
 two residues in one of the alignments.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">fromAFP</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getAngle</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the <em>magnitude</em> of the angle between the first and second
 blocks of <code>afpChain</code>, measured in degrees.</div>
</div>
<div class="col-first even-row-color"><code><a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,double%5B%5D,double%5B%5D,int,int)" class="member-name-link">getDkMatrix</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;fragmentLength,
 double[]&nbsp;dist1,
 double[]&nbsp;dist2,
 int&nbsp;rows,
 int&nbsp;cols)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#getDkMatrix(org.biojava.nbio.structure.Atom%5B%5D,org.biojava.nbio.structure.Atom%5B%5D,int,int)" class="member-name-link">getDkMatrix</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;k,
 int&nbsp;fragmentLength)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#getGroups(org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getGroups</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;rAtoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the List of Groups of the corresponding representative Atom
 array.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#grayOutCEOrig(org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int,double%5B%5D,double)" class="member-name-link">grayOutCEOrig</a><wbr>(<a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;rows,
 int&nbsp;cols,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
 <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;origM,
 int&nbsp;blankWindowSize,
 double[]&nbsp;gradientPolyCoeff,
 double&nbsp;gradientExpCoeff)</code></div>
<div class="col-last odd-row-color">
<div class="block">Grays out the main diagonal of a duplicated distance matrix.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../symmetry/utils/SymmetryTools.html#grayOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int,int,org.biojava.nbio.structure.align.ce.CECalculator,org.biojava.nbio.structure.jama.Matrix,int,double%5B%5D,double)" class="member-name-link">grayOutPreviousAlignment</a><wbr>(<a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
 int&nbsp;rows,
 int&nbsp;cols,
 <a href="../align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
 <a href="../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
 int&nbsp;blankWindowSize,
 double[]&nbsp;gradientPolyCoeff,
 double&nbsp;gradientExpCoeff)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
