<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>MultipleAlignmentWriter (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.structure.align.multiple.util, class: MultipleAlignmentWriter">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/MultipleAlignmentWriter.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html#class">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Summary:</p>
<ul>
<li>Nested</li>
<li>Field</li>
<li><a href="#constructor-summary">Constr</a></li>
<li><a href="#method-summary">Method</a></li>
</ul>
</li>
<li>
<p>Detail:</p>
<ul>
<li>Field</li>
<li><a href="#constructor-detail">Constr</a></li>
<li><a href="#method-detail">Method</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></div>
<h1 title="Class MultipleAlignmentWriter" class="title">Class MultipleAlignmentWriter</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">org.biojava.nbio.structure.align.multiple.util.MultipleAlignmentWriter</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-55">MultipleAlignmentWriter</a></span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
<div class="block">This class contains functions for the conversion of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>
 to various String outputs.
 <p>
 Supported formats: FASTA, FatCat, Aligned Residues, Transformation Matrices,
 XML, 3D format.</div>
<dl class="notes">
<dt>Since:</dt>
<dd>4.1.0</dd>
<dt>Author:</dt>
<dd>Aleix Lafita</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">MultipleAlignmentWriter</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab1" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab1', 3)" class="table-tab">Static Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel" aria-labelledby="method-summary-table-tab0">
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#to3DFormat(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int,int)" class="member-name-link">to3DFormat</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 int&nbsp;queryIndex,
 int&nbsp;templateIndex)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Outputs a pairwise alignment in I-TASSER's 3D Format for target-template
 alignment. http://zhanglab.ccmb.med.umich.edu/I-TASSER/option4.html</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#toAlignedResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toAlignedResidues</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Converts the alignment to its simplest form: a list of groups of aligned
 residues.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#toFASTA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toFASTA</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a multiple sequence alignment
 String in FASTA format.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#toFatCat(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toFatCat</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a FatCat String format.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#toTransformMatrices(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toTransformMatrices</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Converts the transformation Matrices of the alignment into a String
 output.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#toXML(org.biojava.nbio.structure.align.multiple.MultipleAlignmentEnsemble)" class="member-name-link">toXML</a><wbr>(<a href="../MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignmentEnsemble</a>&nbsp;ensemble)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Converts all the information of a multiple alignment ensemble into an XML
 String format.</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#clone()" title="class or interface in java.lang" class="external-link">clone</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#finalize()" title="class or interface in java.lang" class="external-link">finalize</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>MultipleAlignmentWriter</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-55">MultipleAlignmentWriter</a></span>()</div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="toFASTA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>toFASTA</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-66">toFASTA</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</span></div>
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a multiple sequence alignment
 String in FASTA format.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>alignment</code> - MultipleAlignment</dd>
<dt>Returns:</dt>
<dd>String multiple sequence alignment in FASTA format</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list-long">
<li><a href="MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)"><code>MultipleAlignmentTools.getSequenceAlignment(MultipleAlignment)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toFatCat(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>toFatCat</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-91">toFatCat</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</span></div>
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a FatCat String format.
 Includes summary information about the alignment in the top and a
 multiple sequence alignment at the bottom.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>alignment</code> - MultipleAlignment</dd>
<dt>Returns:</dt>
<dd>String multiple sequence alignment in FASTA format</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list-long">
<li><a href="MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)"><code>MultipleAlignmentTools.getSequenceAlignment(MultipleAlignment)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toAlignedResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>toAlignedResidues</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-145">toAlignedResidues</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</span></div>
<div class="block">Converts the alignment to its simplest form: a list of groups of aligned
 residues. Format is one line per residue group, tab delimited:
 <ul>
 <li>PDB number (includes insertion code)
 <li>Chain
 <li>Amino Acid (three letter code)</li>
 </ul>
 Example: <code>52      A       ALA     102     A       VAL     154     A       THR</code>
 <p>
 Note that this format loses information about blocks.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>multAln</code> - MultipleAlignment object</dd>
<dt>Returns:</dt>
<dd>a String representation of the aligned residues.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toTransformMatrices(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>toTransformMatrices</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-200">toTransformMatrices</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</span></div>
<div class="block">Converts the transformation Matrices of the alignment into a String
 output.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>alignment</code> - </dd>
<dt>Returns:</dt>
<dd>String transformation Matrices</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toXML(org.biojava.nbio.structure.align.multiple.MultipleAlignmentEnsemble)">
<h3>toXML</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-263">toXML</a></span><wbr><span class="parameters">(<a href="../MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignmentEnsemble</a>&nbsp;ensemble)</span>
                    throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></span></div>
<div class="block">Converts all the information of a multiple alignment ensemble into an XML
 String format. Cached variables, like transformation matrices and scores,
 are also converted.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>ensemble</code> - the MultipleAlignmentEnsemble to convert.</dd>
<dt>Returns:</dt>
<dd>String XML representation of the ensemble</dd>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></code></dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list-long">
<li><a href="../../xml/MultipleAlignmentXMLConverter.html" title="class in org.biojava.nbio.structure.align.xml"><code>Helper methods for XML conversion</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="to3DFormat(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int,int)">
<h3>to3DFormat</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html#line-316">to3DFormat</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 int&nbsp;queryIndex,
 int&nbsp;templateIndex)</span>
                         throws <span class="exceptions"><a href="../../../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span></div>
<div class="block">Outputs a pairwise alignment in I-TASSER's 3D Format for target-template
 alignment. http://zhanglab.ccmb.med.umich.edu/I-TASSER/option4.html

 <p>
 The format is closely related to a standard PDB file, but contains only
 CA atoms and adds two columns for specifying the alignment:

 <pre>
 ATOM   2001  CA  MET     1      41.116 -30.727   6.866  129 THR
 ATOM   2002  CA  ALA     2      39.261 -27.408   6.496  130 ARG
 ATOM   2003  CA  ALA     3      35.665 -27.370   7.726  131 THR
 ATOM   2004  CA  ARG     4      32.662 -25.111   7.172  132 ARG
 ATOM   2005  CA  GLY     5      29.121 -25.194   8.602  133 ARG

 Column 1 -30: Atom and Residue records of query sequence.
 Column 31-54: Coordinates of atoms in query copied from corresponding atoms in template.
 Column 55-59: Corresponding residue number in template based on alignment
 Column 60-64: Corresponding residue name in template
 </pre>

 <p>
 Note that the output is a pairwise alignment. Only the first and second
 rows in the MultipleAlignment will be used, others ignored.

 <p>
 This method supports topology-independent alignments. The output will
 have sequence order matching the query, but include atoms from the
 template.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>alignment</code> - A <em>full</em> multiple alignment between proteins</dd>
<dd><code>queryIndex</code> - index of the query within the multiple alignment</dd>
<dd><code>templateIndex</code> - index of the template within the multiple alignment</dd>
<dt>Returns:</dt>
<dd>The file contents as a string</dd>
<dt>Throws:</dt>
<dd><code><a href="../../../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code> - If an error occurs parsing the alignment's structure names</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
