<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>MultipleAlignmentTools (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.structure.align.multiple.util, class: MultipleAlignmentTools">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/MultipleAlignmentTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html#class">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Summary:</p>
<ul>
<li>Nested</li>
<li>Field</li>
<li><a href="#constructor-summary">Constr</a></li>
<li><a href="#method-summary">Method</a></li>
</ul>
</li>
<li>
<p>Detail:</p>
<ul>
<li>Field</li>
<li><a href="#constructor-detail">Constr</a></li>
<li><a href="#method-detail">Method</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></div>
<h1 title="Class MultipleAlignmentTools" class="title">Class MultipleAlignmentTools</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">org.biojava.nbio.structure.align.multiple.util.MultipleAlignmentTools</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-85">MultipleAlignmentTools</a></span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
<div class="block">Utility functions for working with <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.
 <p>
 Supported functions:
 <ul>
 <li>Obtain the alignment as sequence strings
 <li>Map from sequence alignment position to structure Atom
 <li>Map from sequence alignment position to Block number
 <li>Transform the aligned Atoms of a MultipleAlignment
 <li>Get all the core alignment positions of the alignment
 <li>Calculate the average residue distance of all aligned positions
 <li>Sort Blocks in a MultipleAlignment by a specified row
 <li>Convert a MultipleAlignment to a MultipleSequenceAlignment
 </ul></div>
<dl class="notes">
<dt>Since:</dt>
<dd>4.1.0</dd>
<dt>Author:</dt>
<dd>Spencer Bliven, Aleix Lafita</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">MultipleAlignmentTools</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab1" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab1', 3)" class="table-tab">Static Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel" aria-labelledby="method-summary-table-tab0">
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static final <a href="../../../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getAlignedStructure(java.util.List)" class="member-name-link">getAlignedStructure</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get an artificial Structure containing a different model for every
 input structure, so that the alignment result can be viewed in Jmol.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int,int)" class="member-name-link">getAtomForSequencePosition</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct,
 int&nbsp;str,
 int&nbsp;sequencePos)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getAverageResidueDistances(java.util.List)" class="member-name-link">getAverageResidueDistances</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getAverageResidueDistances(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getAverageResidueDistances</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getBlockForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int)" class="member-name-link">getBlockForSequencePosition</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct,
 int&nbsp;sequencePos)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Returns the block number of a specified position in the sequence
 alignment, given the mapping from structure to function.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getBlockSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;ma)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)" class="member-name-link">getBlockSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCorePositions(org.biojava.nbio.structure.align.multiple.Block)" class="member-name-link">getCorePositions</a><wbr>(<a href="../Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&nbsp;block)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate a List of alignment indicies that correspond to the core of a
 Block, which means that all structures have a residue in that position.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getEquivalentResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment,boolean)" class="member-name-link">getEquivalentResidues</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 boolean&nbsp;coreOnly)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Convert an MSA into a matrix of equivalent residues.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getHSDMTree</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getKimuraTree</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getRMSDMatrix(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getRMSDMatrix</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of
 the matrix contains the RMSD between structures i and j.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)" class="member-name-link">getSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getStructuralTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getStructuralTree</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using RMSD
 distances and the Neighbor Joining algorithm from forester.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#sortBlocks(java.util.List,int)" class="member-name-link">sortBlocks</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&gt;&nbsp;blocks,
 int&nbsp;sortedIndex)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Sort blocks so that the specified row is in sequential order.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#toMultimodelStructure(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)" class="member-name-link">toMultimodelStructure</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../../../core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a><wbr>&lt;<a href="../../../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toProteinMSA</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#transformAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">transformAtoms</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Transforms atoms according to the superposition stored in the alignment.</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#clone()" title="class or interface in java.lang" class="external-link">clone</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#finalize()" title="class or interface in java.lang" class="external-link">finalize</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>MultipleAlignmentTools</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-85">MultipleAlignmentTools</a></span>()</div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">
<h3>getSequenceAlignment</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-108">getSequenceAlignment</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct)</span></div>
<div class="block">Calculate the sequence alignment Strings for the whole alignment. This
 method creates a sequence alignment where aligned residues are in
 uppercase and unaligned residues are in lowercase, thus providing a more
 compact way to represent the alignment.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps are represented by '-'. Separation between different
 Blocks is indicated by a gap in all positions, meaning that there is a
 possible discontinuity.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>alignment</code> - input MultipleAlignment</dd>
<dd><code>mapSeqToStruct</code> - provides a link from the sequence alignment position to the
            structure alignment position. Specially designed for the GUI.
            Has to be initialized previously and will be overwritten.</dd>
<dt>Returns:</dt>
<dd>a string for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>getSequenceAlignment</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-317">getSequenceAlignment</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</span></div>
<div class="block">Calculate the sequence alignment Strings for the whole alignment. This
 method creates a sequence alignment where aligned residues are in
 uppercase and unaligned residues are in lowercase, thus providing a more
 compact way to represent the alignment.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps are represented by '-'. Separation between different
 Blocks is indicated by a gap in all positions, meaning that there is a
 possible discontinuity.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msa</code> - input MultipleAlignment</dd>
<dt>Returns:</dt>
<dd>String for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">
<h3>getBlockSequenceAlignment</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-346">getBlockSequenceAlignment</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct)</span></div>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment. This method creates a sequence alignment where all residues
 are in uppercase and a residue alone with gaps in all the other
 structures represents unaligned residues. Because of this latter
 constraint only the residues within the Blocks are represented, for a
 more compact alignment. For a sequence alignment of the full protein use
 <a href="#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)"><code>getSequenceAlignment(MultipleAlignment)</code></a>.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps between blocks are omitted, while gaps within blocks are
 represented by '-'. Separation between different Blocks is indicated by a
 gap in all positions, meaning that there is something unaligned
 inbetween.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>alignment</code> - input MultipleAlignment</dd>
<dd><code>mapSeqToStruct</code> - provides a link from the sequence alignment position to the
            structure alignment position. Specially designed for the GUI.
            Has to be initialized previously and will be overwritten.</dd>
<dt>Returns:</dt>
<dd>a string for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>getBlockSequenceAlignment</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-474">getBlockSequenceAlignment</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;ma)</span></div>
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment. This method creates a sequence alignment where all residues
 are in uppercase and a residue alone with gaps in all the other
 structures represents unaligned residues. Because of this latter
 constraint only the residues within the Blocks are represented, for a
 more compact alignment. For a sequence alignment of the full protein use
 <a href="#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)"><code>getSequenceAlignment(MultipleAlignment)</code></a>.
 <p>
 Blocks are concatenated in the order returned by
 <a href="../MultipleAlignment.html#getBlocks()"><code>MultipleAlignment.getBlocks()</code></a>, so sequences may not be
 sequential. Gaps between blocks are omitted, while gaps within blocks are
 represented by '-'. Separation between different Blocks is indicated by a
 gap in all positions, meaning that there is something unaligned
 inbetween.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>ma</code> - input MultipleAlignment</dd>
<dt>Returns:</dt>
<dd>String for each row in the alignment, giving the 1-letter code
         for each aligned residue.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int,int)">
<h3>getAtomForSequencePosition</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-494">getAtomForSequencePosition</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct,
 int&nbsp;str,
 int&nbsp;sequencePos)</span></div>
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msa</code> - the MultipleAlignment object from where the sequence alignment
            has been generated</dd>
<dd><code>mapSeqToStruct</code> - the mapping between sequence and structure generated with the
            sequence alignment</dd>
<dd><code>str</code> - the structure index of the alignment (row)</dd>
<dd><code>sequencePos</code> - the sequence alignment position (column)</dd>
<dt>Returns:</dt>
<dd>Atom the atom in that position or null if there is a gap</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getBlockForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int)">
<h3>getBlockForSequencePosition</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-541">getBlockForSequencePosition</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct,
 int&nbsp;sequencePos)</span></div>
<div class="block">Returns the block number of a specified position in the sequence
 alignment, given the mapping from structure to function.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>multAln</code> - the MultipleAlignment object from where the sequence alignment
            has been generated.</dd>
<dd><code>mapSeqToStruct</code> - the mapping between sequence and structure generated with the
            sequence alignment</dd>
<dd><code>sequencePos</code> - the position in the sequence alignment (column)</dd>
<dt>Returns:</dt>
<dd>int the block index, or -1 if the position is not aligned</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAverageResidueDistances(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>getAverageResidueDistances</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-575">getAverageResidueDistances</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</span></div>
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.
 <p>
 Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment
 length.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msa</code> - MultipleAlignment</dd>
<dt>Returns:</dt>
<dd>Matrix containing all average residue distances</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAverageResidueDistances(java.util.List)">
<h3>getAverageResidueDistances</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-593">getAverageResidueDistances</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformed)</span></div>
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.
 <p>
 Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment
 length.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>transformed</code> - List of Atom arrays containing only the aligned atoms of each
            structure, or null if there is a gap.</dd>
<dt>Returns:</dt>
<dd>Matrix containing all average residue distances. Entry -1 means
         there is a gap in the position.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="transformAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>transformAtoms</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-659">transformAtoms</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</span></div>
<div class="block">Transforms atoms according to the superposition stored in the alignment.
 <p>
 For each structure in the alignment, returns an atom for each
 representative atom in the aligned columns, omitting unaligned residues
 (i.e. an array of length <code>alignment.length()</code> ).
 <p>
 All blocks are concatenated together, so Atoms may not appear in the same
 order as in their parent structure. If the alignment blocks contain null
 residues (gaps), then the returned array will also contain null Atoms in
 the same positions.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>alignment</code> - MultipleAlignment</dd>
<dt>Returns:</dt>
<dd>List of Atom arrays of only the aligned atoms of every structure
         (null Atom if a gap position)</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCorePositions(org.biojava.nbio.structure.align.multiple.Block)">
<h3>getCorePositions</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-731">getCorePositions</a></span><wbr><span class="parameters">(<a href="../Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&nbsp;block)</span></div>
<div class="block">Calculate a List of alignment indicies that correspond to the core of a
 Block, which means that all structures have a residue in that position.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>block</code> - alignment Block</dd>
<dt>Returns:</dt>
<dd>List of positions in the core of the alignment</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="sortBlocks(java.util.List,int)">
<h3>sortBlocks</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-758">sortBlocks</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a>&gt;&nbsp;blocks,
 int&nbsp;sortedIndex)</span></div>
<div class="block">Sort blocks so that the specified row is in sequential order. The sort
 happens in place.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>blocks</code> - List of blocks</dd>
<dd><code>sortedIndex</code> - Index of the row to be sorted</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>toProteinMSA</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../../../core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;<a href="../../../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-793">toProteinMSA</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</span>
                                                                                 throws <span class="exceptions"><a href="../../../../core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues. This method is only valid for protein structure alignments.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msta</code> - Multiple Structure Alignment</dd>
<dt>Returns:</dt>
<dd>MultipleSequenceAlignment of protein sequences</dd>
<dt>Throws:</dt>
<dd><code><a href="../../../../core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toMultimodelStructure(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)">
<h3>toMultimodelStructure</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-825">toMultimodelStructure</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms)</span>
                                       throws <span class="exceptions"><a href="../../../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span></div>
<dl class="notes">
<dt>Throws:</dt>
<dd><code><a href="../../../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAlignedStructure(java.util.List)">
<h3>getAlignedStructure</h3>
<div class="member-signature"><span class="modifiers">public static final</span>&nbsp;<span class="return-type"><a href="../../../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-851">getAlignedStructure</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</span>
                                           throws <span class="exceptions"><a href="../../../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span></div>
<div class="block">Get an artificial Structure containing a different model for every
 input structure, so that the alignment result can be viewed in Jmol.
 The Atoms have to be rotated beforehand.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>atomArrays</code> - an array of Atoms for every aligned structure</dd>
<dt>Returns:</dt>
<dd>a structure object containing a set of models,
                        one for each input array of Atoms.</dd>
<dt>Throws:</dt>
<dd><code><a href="../../../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getRMSDMatrix(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>getRMSDMatrix</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-870">getRMSDMatrix</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</span></div>
<div class="block">Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of
 the matrix contains the RMSD between structures i and j.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msa</code> - Multiple Structure Alignment</dd>
<dt>Returns:</dt>
<dd>Matrix of RMSD with size the number of structures squared</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>getKimuraTree</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">org.forester.phylogeny.Phylogeny</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-900">getKimuraTree</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</span>
                                                      throws <span class="exceptions"><a href="../../../../core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a>,
<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></span></div>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msta</code> - MultipleAlignment of protein structures</dd>
<dt>Returns:</dt>
<dd>Phylogeny phylogenetic tree</dd>
<dt>Throws:</dt>
<dd><code><a href="../../../../core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>getHSDMTree</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">org.forester.phylogeny.Phylogeny</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-924">getHSDMTree</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</span>
                                                    throws <span class="exceptions"><a href="../../../../core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a>,
<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></span></div>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msta</code> - MultipleAlignment of protein structures</dd>
<dt>Returns:</dt>
<dd>Phylogeny phylogenetic tree</dd>
<dt>Throws:</dt>
<dd><code><a href="../../../../core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStructuralTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)">
<h3>getStructuralTree</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">org.forester.phylogeny.Phylogeny</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-944">getStructuralTree</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</span></div>
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using RMSD
 distances and the Neighbor Joining algorithm from forester.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msta</code> - MultipleAlignment of protein structures</dd>
<dt>Returns:</dt>
<dd>Phylogeny phylogenetic tree</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getEquivalentResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment,boolean)">
<h3>getEquivalentResidues</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../../../src-html/org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html#line-979">getEquivalentResidues</a></span><wbr><span class="parameters">(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 boolean&nbsp;coreOnly)</span></div>
<div class="block">Convert an MSA into a matrix of equivalent residues.

 This concatenates all blocks, meaning that the indices might not be
 sequential.

 Indices should be consistent with `msa.getAtomArrays()`.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>msa</code> - Multiple alignment</dd>
<dd><code>coreOnly</code> - Include only core (ungapped) columns. Otherwise gaps are
        represented with null.</dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
