<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span>package demo;<a name="line.21"></a>
<span class="sourceLineNo">022</span><a name="line.22"></a>
<span class="sourceLineNo">023</span>import java.io.InputStream;<a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.util.LinkedHashMap;<a name="line.24"></a>
<span class="sourceLineNo">025</span><a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.MultipleSequenceAlignment;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.nbio.core.sequence.io.FastaReader;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.nbio.core.sequence.io.GenericFastaHeaderParser;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.biojava.nbio.core.sequence.io.ProteinSequenceCreator;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.nbio.phylo.DistanceMatrixCalculator;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.nbio.phylo.DistanceTreeEvaluator;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.nbio.phylo.ForesterWrapper;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.nbio.phylo.TreeConstructor;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.nbio.phylo.TreeConstructorType;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.forester.evoinference.matrix.distance.BasicSymmetricalDistanceMatrix;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.forester.evoinference.matrix.distance.DistanceMatrix;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.forester.phylogeny.Phylogeny;<a name="line.42"></a>
<span class="sourceLineNo">043</span><a name="line.43"></a>
<span class="sourceLineNo">044</span>/**<a name="line.44"></a>
<span class="sourceLineNo">045</span> * This demo contains the CookBook example to create a phylogenetic tree from a<a name="line.45"></a>
<span class="sourceLineNo">046</span> * multiple sequence alignment (MSA).<a name="line.46"></a>
<span class="sourceLineNo">047</span> *<a name="line.47"></a>
<span class="sourceLineNo">048</span> * @author Scooter Willis<a name="line.48"></a>
<span class="sourceLineNo">049</span> * @author Aleix Lafita<a name="line.49"></a>
<span class="sourceLineNo">050</span> *<a name="line.50"></a>
<span class="sourceLineNo">051</span> */<a name="line.51"></a>
<span class="sourceLineNo">052</span>public class DemoDistanceTree {<a name="line.52"></a>
<span class="sourceLineNo">053</span><a name="line.53"></a>
<span class="sourceLineNo">054</span>        public static void main(String[] args) throws Exception {<a name="line.54"></a>
<span class="sourceLineNo">055</span><a name="line.55"></a>
<span class="sourceLineNo">056</span>                // 0. This is just to load an example MSA from a FASTA file<a name="line.56"></a>
<span class="sourceLineNo">057</span>                InputStream inStream = TreeConstructor.class<a name="line.57"></a>
<span class="sourceLineNo">058</span>                                .getResourceAsStream("/PF00104_small.fasta");<a name="line.58"></a>
<span class="sourceLineNo">059</span><a name="line.59"></a>
<span class="sourceLineNo">060</span>                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader =<a name="line.60"></a>
<span class="sourceLineNo">061</span>                                new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.61"></a>
<span class="sourceLineNo">062</span>                                inStream,<a name="line.62"></a>
<span class="sourceLineNo">063</span>                                new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),<a name="line.63"></a>
<span class="sourceLineNo">064</span>                                new ProteinSequenceCreator(AminoAcidCompoundSet<a name="line.64"></a>
<span class="sourceLineNo">065</span>                                                .getAminoAcidCompoundSet()));<a name="line.65"></a>
<span class="sourceLineNo">066</span><a name="line.66"></a>
<span class="sourceLineNo">067</span>                LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequences =<a name="line.67"></a>
<span class="sourceLineNo">068</span>                                fastaReader.process();<a name="line.68"></a>
<span class="sourceLineNo">069</span><a name="line.69"></a>
<span class="sourceLineNo">070</span>                inStream.close();<a name="line.70"></a>
<span class="sourceLineNo">071</span><a name="line.71"></a>
<span class="sourceLineNo">072</span>                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa =<a name="line.72"></a>
<span class="sourceLineNo">073</span>                                new MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt;();<a name="line.73"></a>
<span class="sourceLineNo">074</span><a name="line.74"></a>
<span class="sourceLineNo">075</span>                for (ProteinSequence proteinSequence : proteinSequences.values()) {<a name="line.75"></a>
<span class="sourceLineNo">076</span>                        msa.addAlignedSequence(proteinSequence);<a name="line.76"></a>
<span class="sourceLineNo">077</span>                }<a name="line.77"></a>
<span class="sourceLineNo">078</span><a name="line.78"></a>
<span class="sourceLineNo">079</span>                long readT = System.currentTimeMillis();<a name="line.79"></a>
<span class="sourceLineNo">080</span><a name="line.80"></a>
<span class="sourceLineNo">081</span>                // 1. Calculate the evolutionary distance matrix (can take long)<a name="line.81"></a>
<span class="sourceLineNo">082</span>                SubstitutionMatrix&lt;AminoAcidCompound&gt; M = SubstitutionMatrixHelper<a name="line.82"></a>
<span class="sourceLineNo">083</span>                                .getBlosum62();<a name="line.83"></a>
<span class="sourceLineNo">084</span>                DistanceMatrix DM = DistanceMatrixCalculator<a name="line.84"></a>
<span class="sourceLineNo">085</span>                                .dissimilarityScore(msa, M);<a name="line.85"></a>
<span class="sourceLineNo">086</span><a name="line.86"></a>
<span class="sourceLineNo">087</span>                // 2. Construct a distance tree using the NJ algorithm<a name="line.87"></a>
<span class="sourceLineNo">088</span>                Phylogeny phylo = TreeConstructor.distanceTree(<a name="line.88"></a>
<span class="sourceLineNo">089</span>                                (BasicSymmetricalDistanceMatrix) DM, TreeConstructorType.NJ);<a name="line.89"></a>
<span class="sourceLineNo">090</span><a name="line.90"></a>
<span class="sourceLineNo">091</span>                long treeT = System.currentTimeMillis();<a name="line.91"></a>
<span class="sourceLineNo">092</span>                String newick = ForesterWrapper.getNewickString(phylo, true);<a name="line.92"></a>
<span class="sourceLineNo">093</span>                System.out.println(newick);<a name="line.93"></a>
<span class="sourceLineNo">094</span>                System.out.println("Tree Construction: " + (treeT - readT) + " ms.");<a name="line.94"></a>
<span class="sourceLineNo">095</span><a name="line.95"></a>
<span class="sourceLineNo">096</span>                // 3. Evaluate the goodness of fit of the tree<a name="line.96"></a>
<span class="sourceLineNo">097</span>                double cv = DistanceTreeEvaluator.evaluate(phylo, DM);<a name="line.97"></a>
<span class="sourceLineNo">098</span>                System.out.println("CV of the tree: " + (int) (cv * 100) + " %");<a name="line.98"></a>
<span class="sourceLineNo">099</span><a name="line.99"></a>
<span class="sourceLineNo">100</span>        }<a name="line.100"></a>
<span class="sourceLineNo">101</span>}<a name="line.101"></a>




























































</pre>
</div>
</body>
</html>
