<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_65) on Tue Sep 02 20:47:32 CDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>Uses of Package org.biojava.bio.seq.db (biojava-legacy 1.9.1 API)</title>
<meta name="date" content="2014-09-02">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Package org.biojava.bio.seq.db (biojava-legacy 1.9.1 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/bio/seq/db/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Uses of Package org.biojava.bio.seq.db" class="title">Uses of Package<br>org.biojava.bio.seq.db</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.alignment">org.biojava.bio.alignment</a></td>
<td class="colLast">
<div class="block">
Classes to generate and describe sequence alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.dp">org.biojava.bio.dp</a></td>
<td class="colLast">
<div class="block">HMM and Dynamic Programming Algorithms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.gui">org.biojava.bio.gui</a></td>
<td class="colLast">
<div class="block">Graphical interfaces for biojava objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.gff">org.biojava.bio.program.gff</a></td>
<td class="colLast">
<div class="block">GFF manipulation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.phred">org.biojava.bio.program.phred</a></td>
<td class="colLast">
<div class="block">Parser for Phred output</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.ssaha">org.biojava.bio.program.ssaha</a></td>
<td class="colLast">
<div class="block">SSAHA sequence searching API.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.ssbind">org.biojava.bio.program.ssbind</a></td>
<td class="colLast">
<div class="block">
Creation of objects from SAX events using the BioJava BlastLikeDataSetCollection DTD.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.unigene">org.biojava.bio.program.unigene</a></td>
<td class="colLast">
<div class="block">Objects for representing Unigene clusters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.proteomics.aaindex">org.biojava.bio.proteomics.aaindex</a></td>
<td class="colLast">
<div class="block">Classes and interfaces to load Amino Acid Index database files.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.search">org.biojava.bio.search</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for representing sequence similarity search results.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq">org.biojava.bio.seq</a></td>
<td class="colLast">
<div class="block">Classes and interfaces for defining biological sequences and informatics
objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db">org.biojava.bio.seq.db</a></td>
<td class="colLast">
<div class="block">
Collections of biological sequence data.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.biofetch">org.biojava.bio.seq.db.biofetch</a></td>
<td class="colLast">
<div class="block">Client for the OBDA BioFetch protocol.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.biosql">org.biojava.bio.seq.db.biosql</a></td>
<td class="colLast">
<div class="block">General purpose Sequence storage in a relational database.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.flat">org.biojava.bio.seq.db.flat</a></td>
<td class="colLast">
<div class="block">
Support for OBDA flatfile databases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.distributed">org.biojava.bio.seq.distributed</a></td>
<td class="colLast">
<div class="block">Sequences and SequenceDBs which are composed from data taken
from a number of data sources.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.impl">org.biojava.bio.seq.impl</a></td>
<td class="colLast">
<div class="block">Standard in-memory implementations of <code>Sequence</code> and
<code>Feature</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.io">org.biojava.bio.seq.io</a></td>
<td class="colLast">
<div class="block">Classes and interfaces for processing and producing flat-file representations
of sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.directory">org.biojava.directory</a></td>
<td class="colLast">
<div class="block">Open Bio Sequence Database Access (OBDA) registry support.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.db">org.biojavax.bio.db</a></td>
<td class="colLast">
<div class="block">Interactions between biojavax objects and a DB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.db.biosql">org.biojavax.bio.db.biosql</a></td>
<td class="colLast">
<div class="block">Interface between biojava and biosql databases</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.db.ncbi">org.biojavax.bio.db.ncbi</a></td>
<td class="colLast">
<div class="block">Interfaces to NCBI data.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.alignment">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/alignment/package-summary.html">org.biojava.bio.alignment</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.alignment">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dp">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/dp/package-summary.html">org.biojava.bio.dp</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.dp">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.gui">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/gui/package-summary.html">org.biojava.bio.gui</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.gui">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.gff">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/program/gff/package-summary.html">org.biojava.bio.program.gff</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.program.gff">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.program.gff">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.phred">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/program/phred/package-summary.html">org.biojava.bio.program.phred</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.program.phred">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.ssaha">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.program.ssaha">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.ssbind">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.program.ssbind">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.program.ssbind">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBInstallation.html#org.biojava.bio.program.ssbind">SequenceDBInstallation</a>
<div class="block">
 A SequenceDBInstallation has the functionality of a factory for
 SequenceDB objects and additionally manages the SequenceDB objects
 created by itself such that the minimum number of SequenceDB
 objects is created by a particular SequenceDBInstallation
 object.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.unigene">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/program/unigene/package-summary.html">org.biojava.bio.program.unigene</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.program.unigene">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.proteomics.aaindex">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/proteomics/aaindex/package-summary.html">org.biojava.bio.proteomics.aaindex</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.proteomics.aaindex">IllegalIDException</a>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.search">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.search">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/package-summary.html">org.biojava.bio.seq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.seq">IllegalIDException</a>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/AbstractSequenceDB.html#org.biojava.bio.seq.db">AbstractSequenceDB</a>
<div class="block">An abstract implementation of SequenceDB that provides the sequenceIterator
 method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IDMaker.html#org.biojava.bio.seq.db">IDMaker</a>
<div class="block">Interface for objects that define how to make an ID for a sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.seq.db">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/Index.html#org.biojava.bio.seq.db">Index</a>
<div class="block">This defines an index entry for an individual sequence within a set of
 indexed files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IndexStore.html#org.biojava.bio.seq.db">IndexStore</a>
<div class="block">This defines the objects that IndexedSequenceDB uses to store all of the
 database state, such as name, format, sequence builder and the actual file
 offsets.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.seq.db">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBInstallation.html#org.biojava.bio.seq.db">SequenceDBInstallation</a>
<div class="block">
 A SequenceDBInstallation has the functionality of a factory for
 SequenceDB objects and additionally manages the SequenceDB objects
 created by itself such that the minimum number of SequenceDB
 objects is created by a particular SequenceDBInstallation
 object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojava.bio.seq.db">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBWrapper.html#org.biojava.bio.seq.db">SequenceDBWrapper</a>
<div class="block">An abstract implementation of SequenceDB that wraps up another database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/TabIndexStore.html#org.biojava.bio.seq.db">TabIndexStore</a>
<div class="block">
 Implements IndexStore as a serialized file for the java data and a
 tab-delimited file of offsets.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/WebSequenceDB.html#org.biojava.bio.seq.db">WebSequenceDB</a>
<div class="block">Functions for access to a web based database that returns sequences
 in a variety of formats.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.biofetch">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/db/biofetch/package-summary.html">org.biojava.bio.seq.db.biofetch</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.seq.db.biofetch">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojava.bio.seq.db.biofetch">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.biosql">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/db/biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.seq.db.biosql">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojava.bio.seq.db.biosql">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.flat">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/db/flat/package-summary.html">org.biojava.bio.seq.db.flat</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.seq.db.flat">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojava.bio.seq.db.flat">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.distributed">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/distributed/package-summary.html">org.biojava.bio.seq.distributed</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/AbstractSequenceDB.html#org.biojava.bio.seq.distributed">AbstractSequenceDB</a>
<div class="block">An abstract implementation of SequenceDB that provides the sequenceIterator
 method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojava.bio.seq.distributed">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.seq.distributed">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojava.bio.seq.distributed">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.impl">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/impl/package-summary.html">org.biojava.bio.seq.impl</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.seq.impl">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.io">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojava.bio.seq.io">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.directory">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojava/directory/package-summary.html">org.biojava.directory</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojava.directory">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IDMaker.html#org.biojavax.bio.db">IDMaker</a>
<div class="block">Interface for objects that define how to make an ID for a sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojavax.bio.db">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojavax.bio.db">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojavax.bio.db">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db.biosql">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojavax.bio.db.biosql">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojavax.bio.db.biosql">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojavax.bio.db.biosql">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db.ncbi">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> used by <a href="../../../../../org/biojavax/bio/db/ncbi/package-summary.html">org.biojavax.bio.db.ncbi</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/IllegalIDException.html#org.biojavax.bio.db.ncbi">IllegalIDException</a>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDB.html#org.biojavax.bio.db.ncbi">SequenceDB</a>
<div class="block">A database of sequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../org/biojava/bio/seq/db/class-use/SequenceDBLite.html#org.biojavax.bio.db.ncbi">SequenceDBLite</a>
<div class="block">A database of sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/bio/seq/db/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2014 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
