<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:31 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>ChromosomeSequence (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="ChromosomeSequence (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ChromosomeSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/core/sequence/DataSource.html" title="enum in org.biojava.nbio.core.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/ChromosomeSequence.html" target="_top">Frames</a></li>
<li><a href="ChromosomeSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.core.sequence</div>
<h2 title="Class ChromosomeSequence" class="title">Class ChromosomeSequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template.AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</li>
<li>
<ul class="inheritance">
<li><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence.DNASequence</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.core.sequence.ChromosomeSequence</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;, <a href="../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">ChromosomeSequence</span>
extends <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></pre>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
<dl><dt><span class="strong">Author:</span></dt>
  <dd>Scooter Willis</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava.nbio.core.sequence">DNASequence.DNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#ChromosomeSequence()">ChromosomeSequence</a></strong>()</code>
<div class="block">Empty constructor used by tools that need a proper Bean that allows the actual
 sequence data to be set after construction.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#ChromosomeSequence(org.biojava.nbio.core.sequence.template.SequenceReader)">ChromosomeSequence</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</code>
<div class="block">Fairly important constructor given the size of a ChromsomeSequence where the
 ProxySequenceReader could load from disk via RandomAccessFile so that the sequence
 doesn't need to be kept in memory.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#ChromosomeSequence(org.biojava.nbio.core.sequence.template.SequenceReader,%20org.biojava.nbio.core.sequence.template.CompoundSet)">ChromosomeSequence</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
                  <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code>
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#ChromosomeSequence(java.lang.String)">ChromosomeSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString)</code>
<div class="block">String is king and assume DNA</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#ChromosomeSequence(java.lang.String,%20org.biojava.nbio.core.sequence.template.CompoundSet)">ChromosomeSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
                  <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code>
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#addGene(org.biojava.nbio.core.sequence.AccessionID,%20int,%20int,%20org.biojava.nbio.core.sequence.Strand)">addGene</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
       int&nbsp;bioBegin,
       int&nbsp;bioEnd,
       <a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</code>
<div class="block">Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#getChromosomeNumber()">getChromosomeNumber</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#getGene(java.lang.String)">getGene</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Get the gene based on accession.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#getGeneSequences()">getGeneSequences</a></strong>()</code>
<div class="block">Get the list of genes that have been added to the ChromosomeSequence where accession.toString is the key.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#removeGeneSequence(java.lang.String)">removeGeneSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html#setChromosomeNumber(int)">setChromosomeNumber</a></strong>(int&nbsp;chromosomeNumber)</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getComplement()">getComplement</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getDNAType()">getDNAType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getGCCount()">getGCCount</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverse()">getReverse</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverseComplement()">getReverseComplement</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence()">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine)">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine,%20org.biojava.nbio.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#main(java.lang.String[])">main</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#setDNAType(org.biojava.nbio.core.sequence.DNASequence.DNAType)">setDNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(int,%20int,%20org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatureRetriever()">getFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String,%20int)">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLength()">getLength</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,%20java.lang.Integer,%20org.biojava.nbio.core.sequence.Strand)">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSource()">getSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#iterator()">iterator</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAccession(org.biojava.nbio.core.sequence.AccessionID)">setAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava.nbio.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava.nbio.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeatureRetriever(org.biojava.nbio.core.sequence.features.FeatureRetriever)">setFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava.nbio.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava.nbio.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava.nbio.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#toString()">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="ChromosomeSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeSequence</h4>
<pre>public&nbsp;ChromosomeSequence()</pre>
<div class="block">Empty constructor used by tools that need a proper Bean that allows the actual
 sequence data to be set after construction. Not recommended</div>
</li>
</ul>
<a name="ChromosomeSequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeSequence</h4>
<pre>public&nbsp;ChromosomeSequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString)
                   throws <a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">String is king and assume DNA</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqString</code> - </dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd></dl>
</li>
</ul>
<a name="ChromosomeSequence(org.biojava.nbio.core.sequence.template.SequenceReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeSequence</h4>
<pre>public&nbsp;ChromosomeSequence(<a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</pre>
<div class="block">Fairly important constructor given the size of a ChromsomeSequence where the
 ProxySequenceReader could load from disk via RandomAccessFile so that the sequence
 doesn't need to be kept in memory. Could also be a NCBI proxy to load sequence
 data as needed from remote web server.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>proxyLoader</code> - </dd></dl>
</li>
</ul>
<a name="ChromosomeSequence(java.lang.String, org.biojava.nbio.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>ChromosomeSequence</h4>
<pre>public&nbsp;ChromosomeSequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
                  <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)
                   throws <a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqString</code> - </dd><dd><code>compoundSet</code> - </dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd></dl>
</li>
</ul>
<a name="ChromosomeSequence(org.biojava.nbio.core.sequence.template.SequenceReader, org.biojava.nbio.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>ChromosomeSequence</h4>
<pre>public&nbsp;ChromosomeSequence(<a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
                  <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</pre>
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>proxyLoader</code> - </dd><dd><code>compoundSet</code> - </dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getChromosomeNumber()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChromosomeNumber</h4>
<pre>public&nbsp;int&nbsp;getChromosomeNumber()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the chromosomeNumber</dd></dl>
</li>
</ul>
<a name="setChromosomeNumber(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setChromosomeNumber</h4>
<pre>public&nbsp;void&nbsp;setChromosomeNumber(int&nbsp;chromosomeNumber)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>chromosomeNumber</code> - the chromosomeNumber to set</dd></dl>
</li>
</ul>
<a name="getGeneSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGeneSequences</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&gt;&nbsp;getGeneSequences()</pre>
<div class="block">Get the list of genes that have been added to the ChromosomeSequence where accession.toString is the key.
 The list retains the order the genes are added</div>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="removeGeneSequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeGeneSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&nbsp;removeGeneSequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="addGene(org.biojava.nbio.core.sequence.AccessionID, int, int, org.biojava.nbio.core.sequence.Strand)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addGene</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&nbsp;addGene(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                   int&nbsp;bioBegin,
                   int&nbsp;bioEnd,
                   <a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</pre>
<div class="block">Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0. The
 GeneSequence that is returned will have a reference to parent chromosome sequence
 which actually contains the sequence data. Strand is important for positive and negative
 direction where negative strand means we need reverse complement. If negative strand then
 bioBegin will be greater than bioEnd</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd><dd><code>begin</code> - </dd><dd><code>end</code> - </dd><dd><code>strand</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getGene(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getGene</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&nbsp;getGene(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block">Get the gene based on accession. Will return null if not found</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/ChromosomeSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/core/sequence/DataSource.html" title="enum in org.biojava.nbio.core.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/ChromosomeSequence.html" target="_top">Frames</a></li>
<li><a href="ChromosomeSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
