<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:31 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>GeneSequence (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="GeneSequence (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GeneSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/GeneSequence.html" target="_top">Frames</a></li>
<li><a href="GeneSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.core.sequence</div>
<h2 title="Class GeneSequence" class="title">Class GeneSequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template.AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</li>
<li>
<ul class="inheritance">
<li><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence.DNASequence</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.core.sequence.GeneSequence</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;, <a href="../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">GeneSequence</span>
extends <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></pre>
<dl><dt><span class="strong">Author:</span></dt>
  <dd>Scooter Willis</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava.nbio.core.sequence">DNASequence.DNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#GeneSequence(org.biojava.nbio.core.sequence.ChromosomeSequence,%20int,%20int,%20org.biojava.nbio.core.sequence.Strand)">GeneSequence</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&nbsp;parentSequence,
            int&nbsp;begin,
            int&nbsp;end,
            <a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</code>
<div class="block">A class that keeps track of the details of a GeneSequence which is difficult to properly model.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#addExon(org.biojava.nbio.core.sequence.AccessionID,%20int,%20int)">addExon</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
       int&nbsp;begin,
       int&nbsp;end)</code>
<div class="block">Add an ExonSequence mainly used to mark as a feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#addIntron(org.biojava.nbio.core.sequence.AccessionID,%20int,%20int)">addIntron</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
         int&nbsp;begin,
         int&nbsp;end)</code>
<div class="block">Add an Intron Currently used to mark an IntronSequence as a feature</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#addIntronsUsingExons()">addIntronsUsingExons</a></strong>()</code>
<div class="block">Once everything has been added to the gene sequence where you might have added exon sequences only then you
 can infer the intron sequences and add them.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#addTranscript(org.biojava.nbio.core.sequence.AccessionID,%20int,%20int)">addTranscript</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
             int&nbsp;begin,
             int&nbsp;end)</code>
<div class="block">Add a transcription sequence to a gene which describes a ProteinSequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getExonSequences()">getExonSequences</a></strong>()</code>
<div class="block">Get the exons as an ArrayList</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getIntronSequences()">getIntronSequences</a></strong>()</code>
<div class="block">Get the introns as an ArrayList</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getLength()">getLength</a></strong>()</code>
<div class="block">Returns the length of the Sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getParentChromosomeSequence()">getParentChromosomeSequence</a></strong>()</code>
<div class="block">The parent ChromosomeSequence which contains the actual DNA sequence data</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getSequence5PrimeTo3Prime()">getSequence5PrimeTo3Prime</a></strong>()</code>
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction
 Returns the DNASequence representative of the 5' and 3' reading based on strand</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getStrand()">getStrand</a></strong>()</code>
<div class="block">A gene should have Strand</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getTranscript(java.lang.String)">getTranscript</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Get the transcript sequence by accession</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getTranscripts()">getTranscripts</a></strong>()</code>
<div class="block">Get the collection of transcription sequences assigned to this gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#removeExon(java.lang.String)">removeExon</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Remove the exon sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#removeIntron(java.lang.String)">removeIntron</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Remove the intron by accession</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#removeTranscript(java.lang.String)">removeTranscript</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</code>
<div class="block">Remove the transcript sequence from the gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#setStrand(org.biojava.nbio.core.sequence.Strand)">setStrand</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getComplement()">getComplement</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getDNAType()">getDNAType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getGCCount()">getGCCount</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverse()">getReverse</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverseComplement()">getReverseComplement</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence()">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine)">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine,%20org.biojava.nbio.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#main(java.lang.String[])">main</a>, <a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html#setDNAType(org.biojava.nbio.core.sequence.DNASequence.DNAType)">setDNAType</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(int,%20int,%20org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatureRetriever()">getFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String,%20int)">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,%20java.lang.Integer,%20org.biojava.nbio.core.sequence.Strand)">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSource()">getSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#iterator()">iterator</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAccession(org.biojava.nbio.core.sequence.AccessionID)">setAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava.nbio.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava.nbio.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeatureRetriever(org.biojava.nbio.core.sequence.features.FeatureRetriever)">setFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava.nbio.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava.nbio.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava.nbio.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#toString()">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="GeneSequence(org.biojava.nbio.core.sequence.ChromosomeSequence, int, int, org.biojava.nbio.core.sequence.Strand)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>GeneSequence</h4>
<pre>public&nbsp;GeneSequence(<a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&nbsp;parentSequence,
            int&nbsp;begin,
            int&nbsp;end,
            <a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</pre>
<div class="block">A class that keeps track of the details of a GeneSequence which is difficult to properly model. Two important concepts that is difficult
 to make everything flexible but still work. You can have GFF features that only describe Exons or Exons/Introns or CDS regions and one
 or more Transcriptions. You can have exon sequences but that does not imply transcription to the actual protein.

 The GeneSequence will keep track of Exons and Introns but to get a Protein sequence you need to start with a
 TranscriptSequence and then add CDS sequences.

 This is also a key class in the biojava-3-genome module for reading and writing GFF3 files</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>parentDNASequence</code> - </dd><dd><code>begin</code> - </dd><dd><code>end</code> - inclusive of end</dd><dd><code>strand</code> - force a gene to have strand and transcription sequence will inherit</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getParentChromosomeSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getParentChromosomeSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&nbsp;getParentChromosomeSequence()</pre>
<div class="block">The parent ChromosomeSequence which contains the actual DNA sequence data</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>Chromosome sequence</dd></dl>
</li>
</ul>
<a name="getLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLength</h4>
<pre>public&nbsp;int&nbsp;getLength()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLength()">Sequence</a></code></strong></div>
<div class="block">Returns the length of the Sequence</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLength()">getLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLength()">getLength</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
</li>
</ul>
<a name="addIntronsUsingExons()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addIntronsUsingExons</h4>
<pre>public&nbsp;void&nbsp;addIntronsUsingExons()
                          throws <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></pre>
<div class="block">Once everything has been added to the gene sequence where you might have added exon sequences only then you
 can infer the intron sequences and add them. You may also have the case where you only added one or more
 TranscriptSequences and from that you can infer the exon sequences and intron sequences.
 Currently not implement</div>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></code></dd></dl>
</li>
</ul>
<a name="getStrand()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStrand</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;getStrand()</pre>
<div class="block">A gene should have Strand</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the strand</dd></dl>
</li>
</ul>
<a name="setStrand(org.biojava.nbio.core.sequence.Strand)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setStrand</h4>
<pre>public&nbsp;void&nbsp;setStrand(<a href="../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>strand</code> - the strand to set</dd></dl>
</li>
</ul>
<a name="getTranscript(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTranscript</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a>&nbsp;getTranscript(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block">Get the transcript sequence by accession</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>the transcript</dd></dl>
</li>
</ul>
<a name="getTranscripts()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTranscripts</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a>&gt;&nbsp;getTranscripts()</pre>
<div class="block">Get the collection of transcription sequences assigned to this gene</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>transcripts</dd></dl>
</li>
</ul>
<a name="removeTranscript(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeTranscript</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a>&nbsp;removeTranscript(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block">Remove the transcript sequence from the gene</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>transcriptsequence</dd></dl>
</li>
</ul>
<a name="addTranscript(org.biojava.nbio.core.sequence.AccessionID, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addTranscript</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a>&nbsp;addTranscript(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                               int&nbsp;begin,
                               int&nbsp;end)
                                 throws <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></pre>
<div class="block">Add a transcription sequence to a gene which describes a ProteinSequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd><dd><code>begin</code> - </dd><dd><code>end</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>transcript sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></code> - If the accession id is already used</dd></dl>
</li>
</ul>
<a name="removeIntron(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeIntron</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a>&nbsp;removeIntron(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block">Remove the intron by accession</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>intron sequence</dd></dl>
</li>
</ul>
<a name="addIntron(org.biojava.nbio.core.sequence.AccessionID, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addIntron</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a>&nbsp;addIntron(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                       int&nbsp;begin,
                       int&nbsp;end)
                         throws <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></pre>
<div class="block">Add an Intron Currently used to mark an IntronSequence as a feature</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd><dd><code>begin</code> - </dd><dd><code>end</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>intron sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></code></dd></dl>
</li>
</ul>
<a name="removeExon(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeExon</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a>&nbsp;removeExon(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;accession)</pre>
<div class="block">Remove the exon sequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>exon sequence</dd></dl>
</li>
</ul>
<a name="addExon(org.biojava.nbio.core.sequence.AccessionID, int, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addExon</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a>&nbsp;addExon(<a href="../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
                   int&nbsp;begin,
                   int&nbsp;end)
                     throws <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></pre>
<div class="block">Add an ExonSequence mainly used to mark as a feature</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - </dd><dd><code>begin</code> - </dd><dd><code>end</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>exon sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html?is-external=true" title="class or interface in java.lang">Exception</a></code></dd></dl>
</li>
</ul>
<a name="getExonSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getExonSequences</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a>&gt;&nbsp;getExonSequences()</pre>
<div class="block">Get the exons as an ArrayList</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>exons</dd></dl>
</li>
</ul>
<a name="getIntronSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIntronSequences</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a>&gt;&nbsp;getIntronSequences()</pre>
<div class="block">Get the introns as an ArrayList</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>introns</dd></dl>
</li>
</ul>
<a name="getSequence5PrimeTo3Prime()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getSequence5PrimeTo3Prime</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;getSequence5PrimeTo3Prime()</pre>
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction
 Returns the DNASequence representative of the 5' and 3' reading based on strand</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>dna sequence</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/GeneSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/GeneSequence.html" target="_top">Frames</a></li>
<li><a href="GeneSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.DNASequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
