<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:32 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>RNASequence (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="RNASequence (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/RNASequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/core/sequence/SequenceComparator.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/RNASequence.html" target="_top">Frames</a></li>
<li><a href="RNASequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.core.sequence</div>
<h2 title="Class RNASequence" class="title">Class RNASequence</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template.AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.core.sequence.RNASequence</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;, <a href="../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">RNASequence</span>
extends <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</pre>
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
<dl><dt><span class="strong">Author:</span></dt>
  <dd>Scooter Willis <willishf at gmail dot com></dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#RNASequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader)">RNASequence</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</code>
<div class="block">Create a RNA aequence from a proxy reader</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#RNASequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,%20org.biojava.nbio.core.sequence.template.CompoundSet)">RNASequence</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code>
<div class="block">Create a RNA sequence from a proxy reader and user defined RNA compound set</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#RNASequence(java.lang.String)">RNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString)</code>
<div class="block">Create a RNA sequence from a String</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#RNASequence(java.lang.String,%20org.biojava.nbio.core.sequence.template.CompoundSet)">RNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
           <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code>
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getComplement()">getComplement</a></strong>()</code>
<div class="block">Get the complement view of the RNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getGC()">getGC</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getInverse()">getInverse</a></strong>()</code>
<div class="block">Get the inverse view of the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getProteinSequence()">getProteinSequence</a></strong>()</code>
<div class="block">Get the ProteinSequence from the RNA sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getProteinSequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine)">getProteinSequence</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</code>
<div class="block">Get the ProteinSequene from the RNA sequence with user defined
 transcription engine</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getReverseComplement()">getReverseComplement</a></strong>()</code>
<div class="block">Get reverse complement view of the sequence</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(int,%20int,%20org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatureRetriever()">getFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String,%20int)">getFeatures</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLength()">getLength</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,%20java.lang.Integer,%20org.biojava.nbio.core.sequence.Strand)">getSequenceAsString</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSource()">getSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#iterator()">iterator</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAccession(org.biojava.nbio.core.sequence.AccessionID)">setAccession</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava.nbio.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava.nbio.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeatureRetriever(org.biojava.nbio.core.sequence.features.FeatureRetriever)">setFeatureRetriever</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava.nbio.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava.nbio.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava.nbio.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#toString()">toString</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="RNASequence(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>RNASequence</h4>
<pre>public&nbsp;RNASequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString)
            throws <a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Create a RNA sequence from a String</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqString</code> - </dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd></dl>
</li>
</ul>
<a name="RNASequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>RNASequence</h4>
<pre>public&nbsp;RNASequence(<a href="../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</pre>
<div class="block">Create a RNA aequence from a proxy reader</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>proxyLoader</code> - </dd></dl>
</li>
</ul>
<a name="RNASequence(java.lang.String, org.biojava.nbio.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>RNASequence</h4>
<pre>public&nbsp;RNASequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
           <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)
            throws <a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Create a RNA sequence from a string with a user defined RNA compound set</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqString</code> - </dd><dd><code>compoundSet</code> - </dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd></dl>
</li>
</ul>
<a name="RNASequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader, org.biojava.nbio.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>RNASequence</h4>
<pre>public&nbsp;RNASequence(<a href="../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
           <a href="../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</pre>
<div class="block">Create a RNA sequence from a proxy reader and user defined RNA compound set</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>proxyLoader</code> - </dd><dd><code>compoundSet</code> - </dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getReverseComplement()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReverseComplement</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;getReverseComplement()</pre>
<div class="block">Get reverse complement view of the sequence</div>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getInverse()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInverse</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;getInverse()</pre>
<div class="block">Get the inverse view of the sequence. It is the reverse sequence from
 end to begin where use reverse could imply complement. Called getInverse()
 in the hopes of making less confusing.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getInverse()">getInverse</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</a></code>&nbsp;in class&nbsp;<code><a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getComplement()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getComplement</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;getComplement()</pre>
<div class="block">Get the complement view of the RNA sequence</div>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getProteinSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;getProteinSequence()</pre>
<div class="block">Get the ProteinSequence from the RNA sequence</div>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getProteinSequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProteinSequence</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;getProteinSequence(<a href="../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a>&nbsp;engine)</pre>
<div class="block">Get the ProteinSequene from the RNA sequence with user defined
 transcription engine</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>engine</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getGC()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getGC</h4>
<pre>public&nbsp;double&nbsp;getGC()</pre>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/RNASequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/core/sequence/SequenceComparator.html" title="class in org.biojava.nbio.core.sequence"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/core/sequence/RNASequence.html" target="_top">Frames</a></li>
<li><a href="RNASequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_classes_inherited_from_class_org.biojava.nbio.core.sequence.template.AbstractSequence">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
