<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence, class: ChromosomeSequence">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on DATE</span>
<span class="source-line-no">021</span><span id="line-21"> *</span>
<span class="source-line-no">022</span><span id="line-22"> */</span>
<span class="source-line-no">023</span><span id="line-23">package org.biojava.nbio.core.sequence;</span>
<span class="source-line-no">024</span><span id="line-24"></span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.template.CompoundSet;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.template.SequenceReader;</span>
<span class="source-line-no">030</span><span id="line-30"></span>
<span class="source-line-no">031</span><span id="line-31">import java.util.LinkedHashMap;</span>
<span class="source-line-no">032</span><span id="line-32">import java.util.Map;</span>
<span class="source-line-no">033</span><span id="line-33"></span>
<span class="source-line-no">034</span><span id="line-34">/**</span>
<span class="source-line-no">035</span><span id="line-35"> * A ChromosomeSequence is a DNASequence but keeps track of geneSequences</span>
<span class="source-line-no">036</span><span id="line-36"> * @author Scooter Willis</span>
<span class="source-line-no">037</span><span id="line-37"> */</span>
<span class="source-line-no">038</span><span id="line-38">public class ChromosomeSequence extends DNASequence {</span>
<span class="source-line-no">039</span><span id="line-39"></span>
<span class="source-line-no">040</span><span id="line-40">        private int chromosomeNumber;</span>
<span class="source-line-no">041</span><span id="line-41">        private Map&lt;String, GeneSequence&gt; geneSequenceHashMap = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">042</span><span id="line-42"></span>
<span class="source-line-no">043</span><span id="line-43">        /**</span>
<span class="source-line-no">044</span><span id="line-44">         * Empty constructor used by tools that need a proper Bean that allows the actual</span>
<span class="source-line-no">045</span><span id="line-45">         * sequence data to be set after construction. Not recommended</span>
<span class="source-line-no">046</span><span id="line-46">         */</span>
<span class="source-line-no">047</span><span id="line-47">        public ChromosomeSequence() {</span>
<span class="source-line-no">048</span><span id="line-48">//        throw new UnsupportedOperationException("Null constructor not supported");</span>
<span class="source-line-no">049</span><span id="line-49">        }</span>
<span class="source-line-no">050</span><span id="line-50"></span>
<span class="source-line-no">051</span><span id="line-51">        /**</span>
<span class="source-line-no">052</span><span id="line-52">         * String is king and assume DNA</span>
<span class="source-line-no">053</span><span id="line-53">         * @param seqString</span>
<span class="source-line-no">054</span><span id="line-54">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">055</span><span id="line-55">         */</span>
<span class="source-line-no">056</span><span id="line-56">        public ChromosomeSequence(String seqString) throws CompoundNotFoundException {</span>
<span class="source-line-no">057</span><span id="line-57">                super(seqString, DNACompoundSet.getDNACompoundSet());</span>
<span class="source-line-no">058</span><span id="line-58">        }</span>
<span class="source-line-no">059</span><span id="line-59"></span>
<span class="source-line-no">060</span><span id="line-60">        /**</span>
<span class="source-line-no">061</span><span id="line-61">         * Fairly important constructor given the size of a ChromsomeSequence where the</span>
<span class="source-line-no">062</span><span id="line-62">         * ProxySequenceReader could load from disk via RandomAccessFile so that the sequence</span>
<span class="source-line-no">063</span><span id="line-63">         * doesn't need to be kept in memory. Could also be a NCBI proxy to load sequence</span>
<span class="source-line-no">064</span><span id="line-64">         * data as needed from remote web server.</span>
<span class="source-line-no">065</span><span id="line-65">         * @param proxyLoader</span>
<span class="source-line-no">066</span><span id="line-66">         */</span>
<span class="source-line-no">067</span><span id="line-67">        public ChromosomeSequence(SequenceReader&lt;NucleotideCompound&gt; proxyLoader) {</span>
<span class="source-line-no">068</span><span id="line-68">                super(proxyLoader, DNACompoundSet.getDNACompoundSet());</span>
<span class="source-line-no">069</span><span id="line-69">        }</span>
<span class="source-line-no">070</span><span id="line-70"></span>
<span class="source-line-no">071</span><span id="line-71">        /**</span>
<span class="source-line-no">072</span><span id="line-72">         * Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</span>
<span class="source-line-no">073</span><span id="line-73">         * @param seqString</span>
<span class="source-line-no">074</span><span id="line-74">         * @param compoundSet</span>
<span class="source-line-no">075</span><span id="line-75">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">076</span><span id="line-76">         */</span>
<span class="source-line-no">077</span><span id="line-77">        public ChromosomeSequence(String seqString, CompoundSet&lt;NucleotideCompound&gt; compoundSet) throws CompoundNotFoundException {</span>
<span class="source-line-no">078</span><span id="line-78">                super(seqString, compoundSet);</span>
<span class="source-line-no">079</span><span id="line-79">        }</span>
<span class="source-line-no">080</span><span id="line-80"></span>
<span class="source-line-no">081</span><span id="line-81">        /**</span>
<span class="source-line-no">082</span><span id="line-82">         * Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet</span>
<span class="source-line-no">083</span><span id="line-83">         * @param proxyLoader</span>
<span class="source-line-no">084</span><span id="line-84">         * @param compoundSet</span>
<span class="source-line-no">085</span><span id="line-85">         */</span>
<span class="source-line-no">086</span><span id="line-86">        public ChromosomeSequence(SequenceReader&lt;NucleotideCompound&gt; proxyLoader, CompoundSet&lt;NucleotideCompound&gt; compoundSet) {</span>
<span class="source-line-no">087</span><span id="line-87">                super(proxyLoader, compoundSet);</span>
<span class="source-line-no">088</span><span id="line-88">        }</span>
<span class="source-line-no">089</span><span id="line-89"></span>
<span class="source-line-no">090</span><span id="line-90">        /**</span>
<span class="source-line-no">091</span><span id="line-91">         * @return the chromosomeNumber</span>
<span class="source-line-no">092</span><span id="line-92">         */</span>
<span class="source-line-no">093</span><span id="line-93">        public int getChromosomeNumber() {</span>
<span class="source-line-no">094</span><span id="line-94">                return chromosomeNumber;</span>
<span class="source-line-no">095</span><span id="line-95">        }</span>
<span class="source-line-no">096</span><span id="line-96"></span>
<span class="source-line-no">097</span><span id="line-97">        /**</span>
<span class="source-line-no">098</span><span id="line-98">         * @param chromosomeNumber the chromosomeNumber to set</span>
<span class="source-line-no">099</span><span id="line-99">         */</span>
<span class="source-line-no">100</span><span id="line-100">        public void setChromosomeNumber(int chromosomeNumber) {</span>
<span class="source-line-no">101</span><span id="line-101">                this.chromosomeNumber = chromosomeNumber;</span>
<span class="source-line-no">102</span><span id="line-102">        }</span>
<span class="source-line-no">103</span><span id="line-103"></span>
<span class="source-line-no">104</span><span id="line-104">        /**</span>
<span class="source-line-no">105</span><span id="line-105">         * Get the list of genes that have been added to the ChromosomeSequence where accession.toString is the key.</span>
<span class="source-line-no">106</span><span id="line-106">         * The list retains the order the genes are added</span>
<span class="source-line-no">107</span><span id="line-107">         * @return</span>
<span class="source-line-no">108</span><span id="line-108">         */</span>
<span class="source-line-no">109</span><span id="line-109"></span>
<span class="source-line-no">110</span><span id="line-110">        public Map&lt;String, GeneSequence&gt; getGeneSequences() {</span>
<span class="source-line-no">111</span><span id="line-111">                return geneSequenceHashMap;</span>
<span class="source-line-no">112</span><span id="line-112">        }</span>
<span class="source-line-no">113</span><span id="line-113"></span>
<span class="source-line-no">114</span><span id="line-114">        /**</span>
<span class="source-line-no">115</span><span id="line-115">         *</span>
<span class="source-line-no">116</span><span id="line-116">         * @param accession</span>
<span class="source-line-no">117</span><span id="line-117">         * @return</span>
<span class="source-line-no">118</span><span id="line-118">         */</span>
<span class="source-line-no">119</span><span id="line-119">        public GeneSequence removeGeneSequence(String accession) {</span>
<span class="source-line-no">120</span><span id="line-120">                return geneSequenceHashMap.remove(accession);</span>
<span class="source-line-no">121</span><span id="line-121">        }</span>
<span class="source-line-no">122</span><span id="line-122"></span>
<span class="source-line-no">123</span><span id="line-123">        /**</span>
<span class="source-line-no">124</span><span id="line-124">         * Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0. The</span>
<span class="source-line-no">125</span><span id="line-125">         * GeneSequence that is returned will have a reference to parent chromosome sequence</span>
<span class="source-line-no">126</span><span id="line-126">         * which actually contains the sequence data. Strand is important for positive and negative</span>
<span class="source-line-no">127</span><span id="line-127">         * direction where negative strand means we need reverse complement. If negative strand then</span>
<span class="source-line-no">128</span><span id="line-128">         * bioBegin will be greater than bioEnd</span>
<span class="source-line-no">129</span><span id="line-129">         * @param accession</span>
<span class="source-line-no">130</span><span id="line-130">         * @param bioBegin</span>
<span class="source-line-no">131</span><span id="line-131">         * @param bioEnd</span>
<span class="source-line-no">132</span><span id="line-132">         * @param strand</span>
<span class="source-line-no">133</span><span id="line-133">         * @return A GeneSequence</span>
<span class="source-line-no">134</span><span id="line-134">         */</span>
<span class="source-line-no">135</span><span id="line-135">        public GeneSequence addGene(AccessionID accession, int bioBegin, int bioEnd, Strand strand) {</span>
<span class="source-line-no">136</span><span id="line-136">                GeneSequence geneSequence = new GeneSequence(this, bioBegin, bioEnd, strand);</span>
<span class="source-line-no">137</span><span id="line-137">                geneSequence.setAccession(accession);</span>
<span class="source-line-no">138</span><span id="line-138">                geneSequenceHashMap.put(accession.toString(), geneSequence);</span>
<span class="source-line-no">139</span><span id="line-139">                return geneSequence;</span>
<span class="source-line-no">140</span><span id="line-140">        }</span>
<span class="source-line-no">141</span><span id="line-141"></span>
<span class="source-line-no">142</span><span id="line-142">        /**</span>
<span class="source-line-no">143</span><span id="line-143">         * Get the gene based on accession. Will return null if not found</span>
<span class="source-line-no">144</span><span id="line-144">         * @param accession</span>
<span class="source-line-no">145</span><span id="line-145">         * @return</span>
<span class="source-line-no">146</span><span id="line-146">         */</span>
<span class="source-line-no">147</span><span id="line-147">        public GeneSequence getGene(String accession) {</span>
<span class="source-line-no">148</span><span id="line-148">                return geneSequenceHashMap.get(accession);</span>
<span class="source-line-no">149</span><span id="line-149">        }</span>
<span class="source-line-no">150</span><span id="line-150">}</span>




























































</pre>
</div>
</main>
</body>
</html>
