<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence, class: GeneSequence">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on DATE</span>
<span class="source-line-no">021</span><span id="line-21"> *</span>
<span class="source-line-no">022</span><span id="line-22"> */</span>
<span class="source-line-no">023</span><span id="line-23">package org.biojava.nbio.core.sequence;</span>
<span class="source-line-no">024</span><span id="line-24"></span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.template.CompoundSet;</span>
<span class="source-line-no">029</span><span id="line-29">import org.slf4j.Logger;</span>
<span class="source-line-no">030</span><span id="line-30">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">031</span><span id="line-31"></span>
<span class="source-line-no">032</span><span id="line-32">import java.util.ArrayList;</span>
<span class="source-line-no">033</span><span id="line-33">import java.util.Collections;</span>
<span class="source-line-no">034</span><span id="line-34">import java.util.LinkedHashMap;</span>
<span class="source-line-no">035</span><span id="line-35">import java.util.Map;</span>
<span class="source-line-no">036</span><span id="line-36">import java.util.List;</span>
<span class="source-line-no">037</span><span id="line-37"></span>
<span class="source-line-no">038</span><span id="line-38">/**</span>
<span class="source-line-no">039</span><span id="line-39"> *</span>
<span class="source-line-no">040</span><span id="line-40"> * @author Scooter Willis</span>
<span class="source-line-no">041</span><span id="line-41"> */</span>
<span class="source-line-no">042</span><span id="line-42">public class GeneSequence extends DNASequence {</span>
<span class="source-line-no">043</span><span id="line-43"></span>
<span class="source-line-no">044</span><span id="line-44">        private final static Logger logger = LoggerFactory.getLogger(GeneSequence.class);</span>
<span class="source-line-no">045</span><span id="line-45"></span>
<span class="source-line-no">046</span><span id="line-46">        private final Map&lt;String, TranscriptSequence&gt; transcriptSequenceHashMap = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">047</span><span id="line-47">        private final Map&lt;String, IntronSequence&gt; intronSequenceHashMap = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">048</span><span id="line-48">        private final Map&lt;String, ExonSequence&gt; exonSequenceHashMap = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">049</span><span id="line-49">        private final List&lt;IntronSequence&gt; intronSequenceList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">050</span><span id="line-50">        private final List&lt;ExonSequence&gt; exonSequenceList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">051</span><span id="line-51">        boolean intronAdded = false; // need to deal with the problem that typically introns are not added when validating the list and adding in introns as the regions not included in exons</span>
<span class="source-line-no">052</span><span id="line-52">        private Strand strand = Strand.UNDEFINED;</span>
<span class="source-line-no">053</span><span id="line-53">        private ChromosomeSequence chromosomeSequence;</span>
<span class="source-line-no">054</span><span id="line-54"></span>
<span class="source-line-no">055</span><span id="line-55">        /**</span>
<span class="source-line-no">056</span><span id="line-56">         * Use GeneSequence(ChromosomeSequence parentSequence, AccessionID accessionId,  int begin, int end, Strand strand)</span>
<span class="source-line-no">057</span><span id="line-57">         * which mandates an accessionID.</span>
<span class="source-line-no">058</span><span id="line-58">         * @param parentSequence</span>
<span class="source-line-no">059</span><span id="line-59">         * @param begin</span>
<span class="source-line-no">060</span><span id="line-60">         * @param end inclusive of end</span>
<span class="source-line-no">061</span><span id="line-61">         * @param strand force a gene to have strand and transcription sequence will inherit</span>
<span class="source-line-no">062</span><span id="line-62">         * @deprecated</span>
<span class="source-line-no">063</span><span id="line-63">         */</span>
<span class="source-line-no">064</span><span id="line-64">        public GeneSequence(ChromosomeSequence parentSequence,  int begin, int end, Strand strand) {</span>
<span class="source-line-no">065</span><span id="line-65">                setCompoundSet(DNACompoundSet.getDNACompoundSet());</span>
<span class="source-line-no">066</span><span id="line-66">                try {</span>
<span class="source-line-no">067</span><span id="line-67">                        initSequenceStorage(parentSequence.getSequenceAsString());</span>
<span class="source-line-no">068</span><span id="line-68">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">069</span><span id="line-69">                        throw new IllegalArgumentException(e);</span>
<span class="source-line-no">070</span><span id="line-70">                }</span>
<span class="source-line-no">071</span><span id="line-71">                chromosomeSequence = parentSequence;</span>
<span class="source-line-no">072</span><span id="line-72">                setParentSequence(parentSequence);</span>
<span class="source-line-no">073</span><span id="line-73">                setBioBegin(begin);</span>
<span class="source-line-no">074</span><span id="line-74">                setBioEnd(end);</span>
<span class="source-line-no">075</span><span id="line-75">                setStrand(strand);</span>
<span class="source-line-no">076</span><span id="line-76">        }</span>
<span class="source-line-no">077</span><span id="line-77"></span>
<span class="source-line-no">078</span><span id="line-78">        /**</span>
<span class="source-line-no">079</span><span id="line-79">         * A class that keeps track of the details of a GeneSequence which is difficult to properly model. Two important concepts that is difficult</span>
<span class="source-line-no">080</span><span id="line-80">         * to make everything flexible but still work. You can have GFF features that only describe Exons or Exons/Introns or CDS regions and one</span>
<span class="source-line-no">081</span><span id="line-81">         * or more Transcriptions. You can have exon sequences but that does not imply transcription to the actual protein.</span>
<span class="source-line-no">082</span><span id="line-82">         *</span>
<span class="source-line-no">083</span><span id="line-83">         * The GeneSequence will keep track of Exons and Introns but to get a Protein sequence you need to start with a</span>
<span class="source-line-no">084</span><span id="line-84">         * TranscriptSequence and then add CDS sequences.</span>
<span class="source-line-no">085</span><span id="line-85">         *</span>
<span class="source-line-no">086</span><span id="line-86">         * This is also a key class in the biojava-3-genome module for reading and writing GFF3 files</span>
<span class="source-line-no">087</span><span id="line-87">         *</span>
<span class="source-line-no">088</span><span id="line-88">         * @param parentSequence</span>
<span class="source-line-no">089</span><span id="line-89">         * @param accessionId An identifier for the gene.</span>
<span class="source-line-no">090</span><span id="line-90">         * @param begin</span>
<span class="source-line-no">091</span><span id="line-91">         * @param end</span>
<span class="source-line-no">092</span><span id="line-92">         * @param strand force a gene to have strand and transcription sequence will inherit</span>
<span class="source-line-no">093</span><span id="line-93">         */</span>
<span class="source-line-no">094</span><span id="line-94">        public GeneSequence(ChromosomeSequence parentSequence, AccessionID accessionId,  int begin, int end, Strand strand) {</span>
<span class="source-line-no">095</span><span id="line-95">                this(parentSequence,begin,end,strand);</span>
<span class="source-line-no">096</span><span id="line-96">                setAccession(accessionId);</span>
<span class="source-line-no">097</span><span id="line-97">        }</span>
<span class="source-line-no">098</span><span id="line-98"></span>
<span class="source-line-no">099</span><span id="line-99">        /**</span>
<span class="source-line-no">100</span><span id="line-100">         * The parent ChromosomeSequence which contains the actual DNA sequence data</span>
<span class="source-line-no">101</span><span id="line-101">         * @return Chromosome sequence</span>
<span class="source-line-no">102</span><span id="line-102">         */</span>
<span class="source-line-no">103</span><span id="line-103">        public ChromosomeSequence getParentChromosomeSequence() {</span>
<span class="source-line-no">104</span><span id="line-104">                return chromosomeSequence;</span>
<span class="source-line-no">105</span><span id="line-105">        }</span>
<span class="source-line-no">106</span><span id="line-106"></span>
<span class="source-line-no">107</span><span id="line-107">        @Override</span>
<span class="source-line-no">108</span><span id="line-108">        public int getLength() {</span>
<span class="source-line-no">109</span><span id="line-109">                return Math.abs(this.getBioEnd() - this.getBioBegin()) + 1;</span>
<span class="source-line-no">110</span><span id="line-110">        }</span>
<span class="source-line-no">111</span><span id="line-111"></span>
<span class="source-line-no">112</span><span id="line-112">        /**</span>
<span class="source-line-no">113</span><span id="line-113">         * Once everything has been added to the gene sequence where you might have added exon sequences only then you</span>
<span class="source-line-no">114</span><span id="line-114">         * can infer the intron sequences and add them. You may also have the case where you only added one or more</span>
<span class="source-line-no">115</span><span id="line-115">         * TranscriptSequences and from that you can infer the exon sequences and intron sequences.</span>
<span class="source-line-no">116</span><span id="line-116">         * Currently not implement</span>
<span class="source-line-no">117</span><span id="line-117">         */</span>
<span class="source-line-no">118</span><span id="line-118">        public void addIntronsUsingExons() throws Exception {</span>
<span class="source-line-no">119</span><span id="line-119">                if (intronAdded) { //going to assume introns are correct</span>
<span class="source-line-no">120</span><span id="line-120">                        return;</span>
<span class="source-line-no">121</span><span id="line-121">                }</span>
<span class="source-line-no">122</span><span id="line-122">                if (exonSequenceList.size() == 0) {</span>
<span class="source-line-no">123</span><span id="line-123">                        return;</span>
<span class="source-line-no">124</span><span id="line-124">                }</span>
<span class="source-line-no">125</span><span id="line-125">                ExonComparator exonComparator = new ExonComparator();</span>
<span class="source-line-no">126</span><span id="line-126">                //sort based on start position and sense;</span>
<span class="source-line-no">127</span><span id="line-127">                Collections.sort(exonSequenceList, exonComparator);</span>
<span class="source-line-no">128</span><span id="line-128">                int shift = -1;</span>
<span class="source-line-no">129</span><span id="line-129">                if (getStrand() == Strand.NEGATIVE) {</span>
<span class="source-line-no">130</span><span id="line-130">                        shift = 1;</span>
<span class="source-line-no">131</span><span id="line-131">                }</span>
<span class="source-line-no">132</span><span id="line-132">                //ExonSequence firstExonSequence = exonSequenceList.get(0);</span>
<span class="source-line-no">133</span><span id="line-133">                int intronIndex = 1;</span>
<span class="source-line-no">134</span><span id="line-134">//       if (firstExonSequence.getBioBegin().intValue() != getBioBegin().intValue()) {</span>
<span class="source-line-no">135</span><span id="line-135">//           this.addIntron(new AccessionID(this.getAccession().getID() + "-" + "intron" + intronIndex), getBioBegin(), firstExonSequence.getBioBegin() + shift);</span>
<span class="source-line-no">136</span><span id="line-136">//           intronIndex++;</span>
<span class="source-line-no">137</span><span id="line-137">//       }</span>
<span class="source-line-no">138</span><span id="line-138">                for (int i = 0; i &lt; exonSequenceList.size() - 1; i++) {</span>
<span class="source-line-no">139</span><span id="line-139">                        ExonSequence exon1 = exonSequenceList.get(i);</span>
<span class="source-line-no">140</span><span id="line-140">                        ExonSequence exon2 = exonSequenceList.get(i + 1);</span>
<span class="source-line-no">141</span><span id="line-141">                        AccessionID intronId= new AccessionID(this.getAccession().getID() + "-" + "intron" + intronIndex);</span>
<span class="source-line-no">142</span><span id="line-142">                        this.addIntron(intronId, exon1.getBioEnd() - shift, exon2.getBioBegin() + shift);</span>
<span class="source-line-no">143</span><span id="line-143">                        intronIndex++;</span>
<span class="source-line-no">144</span><span id="line-144">                }</span>
<span class="source-line-no">145</span><span id="line-145"></span>
<span class="source-line-no">146</span><span id="line-146">//       ExonSequence lastExonSequence = exonSequenceList.get(exonSequenceList.size() - 1);</span>
<span class="source-line-no">147</span><span id="line-147">//       if (lastExonSequence.getBioEnd().intValue() != getBioEnd().intValue()) {</span>
<span class="source-line-no">148</span><span id="line-148">//           this.addIntron(new AccessionID(this.getAccession().getID() + "-" + "intron" + intronIndex), lastExonSequence.getBioEnd() - shift, getBioEnd());</span>
<span class="source-line-no">149</span><span id="line-149">//           intronIndex++;</span>
<span class="source-line-no">150</span><span id="line-150">//       }</span>
<span class="source-line-no">151</span><span id="line-151"></span>
<span class="source-line-no">152</span><span id="line-152">                //    log.severe("Add in support for building introns based on added exons");</span>
<span class="source-line-no">153</span><span id="line-153"></span>
<span class="source-line-no">154</span><span id="line-154">        }</span>
<span class="source-line-no">155</span><span id="line-155"></span>
<span class="source-line-no">156</span><span id="line-156">        /**</span>
<span class="source-line-no">157</span><span id="line-157">         * A gene should have Strand</span>
<span class="source-line-no">158</span><span id="line-158">         * @return the strand</span>
<span class="source-line-no">159</span><span id="line-159">         */</span>
<span class="source-line-no">160</span><span id="line-160">        public Strand getStrand() {</span>
<span class="source-line-no">161</span><span id="line-161">                return strand;</span>
<span class="source-line-no">162</span><span id="line-162">        }</span>
<span class="source-line-no">163</span><span id="line-163"></span>
<span class="source-line-no">164</span><span id="line-164">        /**</span>
<span class="source-line-no">165</span><span id="line-165">         * @param strand the strand to set</span>
<span class="source-line-no">166</span><span id="line-166">         */</span>
<span class="source-line-no">167</span><span id="line-167">        public void setStrand(Strand strand) {</span>
<span class="source-line-no">168</span><span id="line-168">                this.strand = strand;</span>
<span class="source-line-no">169</span><span id="line-169">        }</span>
<span class="source-line-no">170</span><span id="line-170"></span>
<span class="source-line-no">171</span><span id="line-171">        /**</span>
<span class="source-line-no">172</span><span id="line-172">         * Get the transcript sequence by accession</span>
<span class="source-line-no">173</span><span id="line-173">         * @param accession</span>
<span class="source-line-no">174</span><span id="line-174">         * @return the transcript</span>
<span class="source-line-no">175</span><span id="line-175">         */</span>
<span class="source-line-no">176</span><span id="line-176">        public TranscriptSequence getTranscript(String accession) {</span>
<span class="source-line-no">177</span><span id="line-177">                return transcriptSequenceHashMap.get(accession);</span>
<span class="source-line-no">178</span><span id="line-178">        }</span>
<span class="source-line-no">179</span><span id="line-179"></span>
<span class="source-line-no">180</span><span id="line-180">        /**</span>
<span class="source-line-no">181</span><span id="line-181">         * Get the collection of transcription sequences assigned to this gene</span>
<span class="source-line-no">182</span><span id="line-182">         * @return transcripts</span>
<span class="source-line-no">183</span><span id="line-183">         */</span>
<span class="source-line-no">184</span><span id="line-184">        public Map&lt;String, TranscriptSequence&gt; getTranscripts() {</span>
<span class="source-line-no">185</span><span id="line-185">                return transcriptSequenceHashMap;</span>
<span class="source-line-no">186</span><span id="line-186">        }</span>
<span class="source-line-no">187</span><span id="line-187"></span>
<span class="source-line-no">188</span><span id="line-188">        /**</span>
<span class="source-line-no">189</span><span id="line-189">         * Remove the transcript sequence from the gene</span>
<span class="source-line-no">190</span><span id="line-190">         * @param accession</span>
<span class="source-line-no">191</span><span id="line-191">         * @return transcriptsequence</span>
<span class="source-line-no">192</span><span id="line-192">         */</span>
<span class="source-line-no">193</span><span id="line-193">        public TranscriptSequence removeTranscript(String accession) {</span>
<span class="source-line-no">194</span><span id="line-194">                return transcriptSequenceHashMap.remove(accession);</span>
<span class="source-line-no">195</span><span id="line-195">        }</span>
<span class="source-line-no">196</span><span id="line-196"></span>
<span class="source-line-no">197</span><span id="line-197">        /**</span>
<span class="source-line-no">198</span><span id="line-198">         * Add a transcription sequence to a gene which describes a ProteinSequence</span>
<span class="source-line-no">199</span><span id="line-199">         * @param accession</span>
<span class="source-line-no">200</span><span id="line-200">         * @param begin</span>
<span class="source-line-no">201</span><span id="line-201">         * @param end</span>
<span class="source-line-no">202</span><span id="line-202">         * @return transcript sequence</span>
<span class="source-line-no">203</span><span id="line-203">         * @throws Exception If the accession id is already used</span>
<span class="source-line-no">204</span><span id="line-204">         */</span>
<span class="source-line-no">205</span><span id="line-205">        public TranscriptSequence addTranscript(AccessionID accession, int begin, int end) throws Exception {</span>
<span class="source-line-no">206</span><span id="line-206">                if (transcriptSequenceHashMap.containsKey(accession.getID())) {</span>
<span class="source-line-no">207</span><span id="line-207">                        throw new Exception("Duplicate accesion id " + accession.getID());</span>
<span class="source-line-no">208</span><span id="line-208">                }</span>
<span class="source-line-no">209</span><span id="line-209">                TranscriptSequence transcriptSequence = new TranscriptSequence(this, begin, end);</span>
<span class="source-line-no">210</span><span id="line-210">                transcriptSequence.setAccession(accession);</span>
<span class="source-line-no">211</span><span id="line-211">                transcriptSequenceHashMap.put(accession.getID(), transcriptSequence);</span>
<span class="source-line-no">212</span><span id="line-212">                return transcriptSequence;</span>
<span class="source-line-no">213</span><span id="line-213">        }</span>
<span class="source-line-no">214</span><span id="line-214"></span>
<span class="source-line-no">215</span><span id="line-215">        /**</span>
<span class="source-line-no">216</span><span id="line-216">         * Remove the intron by accession</span>
<span class="source-line-no">217</span><span id="line-217">         * @param accession</span>
<span class="source-line-no">218</span><span id="line-218">         * @return the removed intron sequence, or null if no intron with that accession exists.</span>
<span class="source-line-no">219</span><span id="line-219">         */</span>
<span class="source-line-no">220</span><span id="line-220">        public IntronSequence removeIntron(String accession) {</span>
<span class="source-line-no">221</span><span id="line-221">                for (IntronSequence intronSequence : intronSequenceList) {</span>
<span class="source-line-no">222</span><span id="line-222">                        if (intronSequence.getAccession().getID().equals(accession)) {</span>
<span class="source-line-no">223</span><span id="line-223">                                intronSequenceList.remove(intronSequence);</span>
<span class="source-line-no">224</span><span id="line-224">                                intronSequenceHashMap.remove(accession);</span>
<span class="source-line-no">225</span><span id="line-225">                                return intronSequence;</span>
<span class="source-line-no">226</span><span id="line-226">                        }</span>
<span class="source-line-no">227</span><span id="line-227">                }</span>
<span class="source-line-no">228</span><span id="line-228">                return null;</span>
<span class="source-line-no">229</span><span id="line-229">        }</span>
<span class="source-line-no">230</span><span id="line-230"></span>
<span class="source-line-no">231</span><span id="line-231">        /**</span>
<span class="source-line-no">232</span><span id="line-232">         * Add an Intron Currently used to mark an IntronSequence as a feature</span>
<span class="source-line-no">233</span><span id="line-233">         * @param accession</span>
<span class="source-line-no">234</span><span id="line-234">         * @param begin</span>
<span class="source-line-no">235</span><span id="line-235">         * @param end</span>
<span class="source-line-no">236</span><span id="line-236">         * @return intron sequence</span>
<span class="source-line-no">237</span><span id="line-237">         */</span>
<span class="source-line-no">238</span><span id="line-238">        public IntronSequence addIntron(AccessionID accession, int begin, int end) throws Exception {</span>
<span class="source-line-no">239</span><span id="line-239">                if (intronSequenceHashMap.containsKey(accession.getID())) {</span>
<span class="source-line-no">240</span><span id="line-240">                        throw new Exception("Duplicate accesion id " + accession.getID());</span>
<span class="source-line-no">241</span><span id="line-241">                }</span>
<span class="source-line-no">242</span><span id="line-242">                intronAdded = true;</span>
<span class="source-line-no">243</span><span id="line-243">                IntronSequence intronSequence = new IntronSequence(this, begin, end); // working off the assumption that intron frame is always 0 or doesn't matter and same sense as parent</span>
<span class="source-line-no">244</span><span id="line-244">                intronSequence.setAccession(accession);</span>
<span class="source-line-no">245</span><span id="line-245">                intronSequenceList.add(intronSequence);</span>
<span class="source-line-no">246</span><span id="line-246">                intronSequenceHashMap.put(accession.getID(), intronSequence);</span>
<span class="source-line-no">247</span><span id="line-247">                return intronSequence;</span>
<span class="source-line-no">248</span><span id="line-248">        }</span>
<span class="source-line-no">249</span><span id="line-249"></span>
<span class="source-line-no">250</span><span id="line-250">        /**</span>
<span class="source-line-no">251</span><span id="line-251">         * Remove the exon sequence</span>
<span class="source-line-no">252</span><span id="line-252">         * @param accession</span>
<span class="source-line-no">253</span><span id="line-253">         * @return exon sequence</span>
<span class="source-line-no">254</span><span id="line-254">         */</span>
<span class="source-line-no">255</span><span id="line-255">        public ExonSequence removeExon(String accession) {</span>
<span class="source-line-no">256</span><span id="line-256">                for (ExonSequence exonSequence : exonSequenceList) {</span>
<span class="source-line-no">257</span><span id="line-257">                        if (exonSequence.getAccession().getID().equals(accession)) {</span>
<span class="source-line-no">258</span><span id="line-258">                                exonSequenceList.remove(exonSequence);</span>
<span class="source-line-no">259</span><span id="line-259">                                exonSequenceHashMap.remove(accession);</span>
<span class="source-line-no">260</span><span id="line-260">                                // we now have a new gap which creates an intron</span>
<span class="source-line-no">261</span><span id="line-261">                                intronSequenceList.clear();</span>
<span class="source-line-no">262</span><span id="line-262">                                intronSequenceHashMap.clear();</span>
<span class="source-line-no">263</span><span id="line-263">                                intronAdded = false;</span>
<span class="source-line-no">264</span><span id="line-264">                                try{</span>
<span class="source-line-no">265</span><span id="line-265">                                        addIntronsUsingExons();</span>
<span class="source-line-no">266</span><span id="line-266">                                } catch(Exception e){</span>
<span class="source-line-no">267</span><span id="line-267">                                        logger.error("Remove Exon validate() error " + e.getMessage());</span>
<span class="source-line-no">268</span><span id="line-268">                                }</span>
<span class="source-line-no">269</span><span id="line-269">                                return exonSequence;</span>
<span class="source-line-no">270</span><span id="line-270">                        }</span>
<span class="source-line-no">271</span><span id="line-271">                }</span>
<span class="source-line-no">272</span><span id="line-272">                return null;</span>
<span class="source-line-no">273</span><span id="line-273">        }</span>
<span class="source-line-no">274</span><span id="line-274"></span>
<span class="source-line-no">275</span><span id="line-275">        /**</span>
<span class="source-line-no">276</span><span id="line-276">         * Add an ExonSequence mainly used to mark as a feature</span>
<span class="source-line-no">277</span><span id="line-277">         * @param accession</span>
<span class="source-line-no">278</span><span id="line-278">         * @param begin</span>
<span class="source-line-no">279</span><span id="line-279">         * @param end</span>
<span class="source-line-no">280</span><span id="line-280">         * @return exon sequence</span>
<span class="source-line-no">281</span><span id="line-281">         * @throws IllegalArgumentException if accessionID is already added.</span>
<span class="source-line-no">282</span><span id="line-282">         */</span>
<span class="source-line-no">283</span><span id="line-283">        public ExonSequence addExon(AccessionID accession, int begin, int end) {</span>
<span class="source-line-no">284</span><span id="line-284">                if (exonSequenceHashMap.containsKey(accession.getID())) {</span>
<span class="source-line-no">285</span><span id="line-285">                        throw new IllegalArgumentException("Duplicate accession id: " + accession.getID());</span>
<span class="source-line-no">286</span><span id="line-286">                }</span>
<span class="source-line-no">287</span><span id="line-287"></span>
<span class="source-line-no">288</span><span id="line-288">                ExonSequence exonSequence = new ExonSequence(this, begin, end); //sense should be the same as parent</span>
<span class="source-line-no">289</span><span id="line-289">                exonSequence.setAccession(accession);</span>
<span class="source-line-no">290</span><span id="line-290">                exonSequenceList.add(exonSequence);</span>
<span class="source-line-no">291</span><span id="line-291">                exonSequenceHashMap.put(accession.getID(), exonSequence);</span>
<span class="source-line-no">292</span><span id="line-292">                return exonSequence;</span>
<span class="source-line-no">293</span><span id="line-293">        }</span>
<span class="source-line-no">294</span><span id="line-294"></span>
<span class="source-line-no">295</span><span id="line-295">        /**</span>
<span class="source-line-no">296</span><span id="line-296">         * Get the exons as an ArrayList. Modifying this list will not modify the underlying collection</span>
<span class="source-line-no">297</span><span id="line-297">         * @return exons</span>
<span class="source-line-no">298</span><span id="line-298">         */</span>
<span class="source-line-no">299</span><span id="line-299">        public List&lt;ExonSequence&gt; getExonSequences() {</span>
<span class="source-line-no">300</span><span id="line-300">                return new ArrayList&lt;&gt;(exonSequenceList);</span>
<span class="source-line-no">301</span><span id="line-301">        }</span>
<span class="source-line-no">302</span><span id="line-302"></span>
<span class="source-line-no">303</span><span id="line-303">        /**</span>
<span class="source-line-no">304</span><span id="line-304">         * Get the introns as an ArrayList. Modifying this list will not modify the underlying collection</span>
<span class="source-line-no">305</span><span id="line-305">         * @return introns</span>
<span class="source-line-no">306</span><span id="line-306">         */</span>
<span class="source-line-no">307</span><span id="line-307">        public List&lt;IntronSequence&gt; getIntronSequences() {</span>
<span class="source-line-no">308</span><span id="line-308">                return  new ArrayList&lt;&gt;(intronSequenceList);</span>
<span class="source-line-no">309</span><span id="line-309">        }</span>
<span class="source-line-no">310</span><span id="line-310"></span>
<span class="source-line-no">311</span><span id="line-311">        /**</span>
<span class="source-line-no">312</span><span id="line-312">         * Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction</span>
<span class="source-line-no">313</span><span id="line-313">         * Returns the DNASequence representative of the 5' and 3' reading based on strand</span>
<span class="source-line-no">314</span><span id="line-314">         * @return dna sequence or null if sequence could not be generated.</span>
<span class="source-line-no">315</span><span id="line-315">         */</span>
<span class="source-line-no">316</span><span id="line-316">        public DNASequence getSequence5PrimeTo3Prime() {</span>
<span class="source-line-no">317</span><span id="line-317">                String sequence = getSequenceAsString(this.getBioBegin(), this.getBioEnd(), this.getStrand());</span>
<span class="source-line-no">318</span><span id="line-318">                if (getStrand() == Strand.NEGATIVE) {</span>
<span class="source-line-no">319</span><span id="line-319">                        //need to take complement of sequence because it is negative and we are returning the gene sequence from the opposite strand</span>
<span class="source-line-no">320</span><span id="line-320">                        StringBuilder b = new StringBuilder(getLength());</span>
<span class="source-line-no">321</span><span id="line-321">                        CompoundSet&lt;NucleotideCompound&gt; compoundSet = this.getCompoundSet();</span>
<span class="source-line-no">322</span><span id="line-322">                        for (int i = 0; i &lt; sequence.length(); i++) {</span>
<span class="source-line-no">323</span><span id="line-323">                                String nucleotide = String.valueOf(sequence.charAt(i));</span>
<span class="source-line-no">324</span><span id="line-324">                                NucleotideCompound nucleotideCompound = compoundSet.getCompoundForString(nucleotide);</span>
<span class="source-line-no">325</span><span id="line-325">                                b.append(nucleotideCompound.getComplement().getShortName());</span>
<span class="source-line-no">326</span><span id="line-326">                        }</span>
<span class="source-line-no">327</span><span id="line-327">                        sequence = b.toString();</span>
<span class="source-line-no">328</span><span id="line-328">                }</span>
<span class="source-line-no">329</span><span id="line-329">                DNASequence dnaSequence = null;</span>
<span class="source-line-no">330</span><span id="line-330">                try {</span>
<span class="source-line-no">331</span><span id="line-331">                        dnaSequence = new DNASequence(sequence.toUpperCase());</span>
<span class="source-line-no">332</span><span id="line-332">                    dnaSequence.setAccession(new AccessionID(this.getAccession().getID()));</span>
<span class="source-line-no">333</span><span id="line-333">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">334</span><span id="line-334">                        // this should not happen, the sequence is DNA originally, if it does, there's a bug somewhere</span>
<span class="source-line-no">335</span><span id="line-335">                        logger.error("Could not create new DNA sequence in getSequence5PrimeTo3Prime(). Error: {}",e.getMessage());</span>
<span class="source-line-no">336</span><span id="line-336">                }</span>
<span class="source-line-no">337</span><span id="line-337">                return dnaSequence;</span>
<span class="source-line-no">338</span><span id="line-338">        }</span>
<span class="source-line-no">339</span><span id="line-339">}</span>




























































</pre>
</div>
</main>
</body>
</html>
