<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence, class: ProteinSequence">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on DATE</span>
<span class="source-line-no">021</span><span id="line-21"> *</span>
<span class="source-line-no">022</span><span id="line-22"> */</span>
<span class="source-line-no">023</span><span id="line-23">package org.biojava.nbio.core.sequence;</span>
<span class="source-line-no">024</span><span id="line-24"></span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.compound.*;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.features.FeatureInterface;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.io.DNASequenceCreator;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.io.FastaReader;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.io.PlainFastaHeaderParser;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.sequence.location.InsdcParser;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.core.sequence.location.template.Location;</span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.core.sequence.template.AbstractSequence;</span>
<span class="source-line-no">034</span><span id="line-34">import org.biojava.nbio.core.sequence.template.CompoundSet;</span>
<span class="source-line-no">035</span><span id="line-35">import org.biojava.nbio.core.sequence.template.ProxySequenceReader;</span>
<span class="source-line-no">036</span><span id="line-36">import org.slf4j.Logger;</span>
<span class="source-line-no">037</span><span id="line-37">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">038</span><span id="line-38"></span>
<span class="source-line-no">039</span><span id="line-39">import java.io.IOException;</span>
<span class="source-line-no">040</span><span id="line-40">import java.io.InputStream;</span>
<span class="source-line-no">041</span><span id="line-41">import java.net.URL;</span>
<span class="source-line-no">042</span><span id="line-42">import java.util.LinkedHashMap;</span>
<span class="source-line-no">043</span><span id="line-43">import java.util.List;</span>
<span class="source-line-no">044</span><span id="line-44">import org.biojava.nbio.core.sequence.features.Qualifier;</span>
<span class="source-line-no">045</span><span id="line-45">import java.util.Map;</span>
<span class="source-line-no">046</span><span id="line-46"></span>
<span class="source-line-no">047</span><span id="line-47">/**</span>
<span class="source-line-no">048</span><span id="line-48"> * The representation of a ProteinSequence</span>
<span class="source-line-no">049</span><span id="line-49"> *</span>
<span class="source-line-no">050</span><span id="line-50"> * @author Scooter Willis</span>
<span class="source-line-no">051</span><span id="line-51"> * @author Paolo Pavan</span>
<span class="source-line-no">052</span><span id="line-52"> */</span>
<span class="source-line-no">053</span><span id="line-53">public class ProteinSequence extends AbstractSequence&lt;AminoAcidCompound&gt; {</span>
<span class="source-line-no">054</span><span id="line-54"></span>
<span class="source-line-no">055</span><span id="line-55">        private final static Logger logger = LoggerFactory.getLogger(ProteinSequence.class);</span>
<span class="source-line-no">056</span><span id="line-56"></span>
<span class="source-line-no">057</span><span id="line-57">        /*</span>
<span class="source-line-no">058</span><span id="line-58">         private ArrayList&lt;FeatureInterface&lt;AbstractSequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;&gt; features</span>
<span class="source-line-no">059</span><span id="line-59">         = new ArrayList&lt;FeatureInterface&lt;AbstractSequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;&gt;();</span>
<span class="source-line-no">060</span><span id="line-60">         private LinkedHashMap&lt;String, ArrayList&lt;FeatureInterface&lt;AbstractSequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;&gt;&gt; groupedFeatures</span>
<span class="source-line-no">061</span><span id="line-61">         = new LinkedHashMap&lt;String, ArrayList&lt;FeatureInterface&lt;AbstractSequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;&gt;&gt;();</span>
<span class="source-line-no">062</span><span id="line-62">         */</span>
<span class="source-line-no">063</span><span id="line-63">        /**</span>
<span class="source-line-no">064</span><span id="line-64">         * Create a protein from a string</span>
<span class="source-line-no">065</span><span id="line-65">         *</span>
<span class="source-line-no">066</span><span id="line-66">         * @param seqString</span>
<span class="source-line-no">067</span><span id="line-67">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">068</span><span id="line-68">         */</span>
<span class="source-line-no">069</span><span id="line-69">        public ProteinSequence(String seqString) throws CompoundNotFoundException {</span>
<span class="source-line-no">070</span><span id="line-70">                this(seqString, AminoAcidCompoundSet.getAminoAcidCompoundSet());</span>
<span class="source-line-no">071</span><span id="line-71">        }</span>
<span class="source-line-no">072</span><span id="line-72"></span>
<span class="source-line-no">073</span><span id="line-73">        /**</span>
<span class="source-line-no">074</span><span id="line-74">         * Create a protein from a string with a user defined set of amino acids</span>
<span class="source-line-no">075</span><span id="line-75">         *</span>
<span class="source-line-no">076</span><span id="line-76">         * @param seqString</span>
<span class="source-line-no">077</span><span id="line-77">         * @param compoundSet</span>
<span class="source-line-no">078</span><span id="line-78">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">079</span><span id="line-79">         */</span>
<span class="source-line-no">080</span><span id="line-80">        public ProteinSequence(String seqString, CompoundSet&lt;AminoAcidCompound&gt; compoundSet) throws CompoundNotFoundException {</span>
<span class="source-line-no">081</span><span id="line-81">                super(seqString, compoundSet);</span>
<span class="source-line-no">082</span><span id="line-82">        }</span>
<span class="source-line-no">083</span><span id="line-83"></span>
<span class="source-line-no">084</span><span id="line-84">        /**</span>
<span class="source-line-no">085</span><span id="line-85">         * A protein sequence where the storage of the sequence is somewhere else.</span>
<span class="source-line-no">086</span><span id="line-86">         * Could be loaded from a large Fasta file or via a Uniprot Proxy reader via</span>
<span class="source-line-no">087</span><span id="line-87">         * Uniprot ID</span>
<span class="source-line-no">088</span><span id="line-88">         *</span>
<span class="source-line-no">089</span><span id="line-89">         * @param proxyLoader</span>
<span class="source-line-no">090</span><span id="line-90">         */</span>
<span class="source-line-no">091</span><span id="line-91">        public ProteinSequence(ProxySequenceReader&lt;AminoAcidCompound&gt; proxyLoader) {</span>
<span class="source-line-no">092</span><span id="line-92">                this(proxyLoader, AminoAcidCompoundSet.getAminoAcidCompoundSet());</span>
<span class="source-line-no">093</span><span id="line-93">        }</span>
<span class="source-line-no">094</span><span id="line-94"></span>
<span class="source-line-no">095</span><span id="line-95">        /**</span>
<span class="source-line-no">096</span><span id="line-96">         * A protein sequence where the storage of the sequence is somewhere else</span>
<span class="source-line-no">097</span><span id="line-97">         * with user defined set of amino acids. Could be loaded from a large Fasta</span>
<span class="source-line-no">098</span><span id="line-98">         * file or via a Uniprot Proxy reader via Uniprot ID</span>
<span class="source-line-no">099</span><span id="line-99">         *</span>
<span class="source-line-no">100</span><span id="line-100">         * @param proxyLoader</span>
<span class="source-line-no">101</span><span id="line-101">         * @param compoundSet</span>
<span class="source-line-no">102</span><span id="line-102">         */</span>
<span class="source-line-no">103</span><span id="line-103">        public ProteinSequence(ProxySequenceReader&lt;AminoAcidCompound&gt; proxyLoader, CompoundSet&lt;AminoAcidCompound&gt; compoundSet) {</span>
<span class="source-line-no">104</span><span id="line-104">                super(proxyLoader, compoundSet);</span>
<span class="source-line-no">105</span><span id="line-105"></span>
<span class="source-line-no">106</span><span id="line-106">                // do protein-specific tasks</span>
<span class="source-line-no">107</span><span id="line-107">                // add source if found</span>
<span class="source-line-no">108</span><span id="line-108">                List&lt;FeatureInterface&lt;AbstractSequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;&gt; CDSFeatures = getFeaturesByType("CDS");</span>
<span class="source-line-no">109</span><span id="line-109"></span>
<span class="source-line-no">110</span><span id="line-110">                // cases if a protein has more than 1 parent are not supported yet</span>
<span class="source-line-no">111</span><span id="line-111">                if (CDSFeatures.size() == 1) {</span>
<span class="source-line-no">112</span><span id="line-112">                        Qualifier codedBy = CDSFeatures.get(0).getQualifiers().get("coded_by").get(0);</span>
<span class="source-line-no">113</span><span id="line-113"></span>
<span class="source-line-no">114</span><span id="line-114">                        if (codedBy != null) {</span>
<span class="source-line-no">115</span><span id="line-115">                                String codedBySeq = codedBy.getValue();</span>
<span class="source-line-no">116</span><span id="line-116"></span>
<span class="source-line-no">117</span><span id="line-117">                                InsdcParser parser = new InsdcParser(DataSource.GENBANK);</span>
<span class="source-line-no">118</span><span id="line-118">                                Location location = parser.parse(codedBySeq);</span>
<span class="source-line-no">119</span><span id="line-119"></span>
<span class="source-line-no">120</span><span id="line-120">                                try {</span>
<span class="source-line-no">121</span><span id="line-121">                                        DNASequence dnaSeq = new DNASequence(getSequence(location), DNACompoundSet.getDNACompoundSet());</span>
<span class="source-line-no">122</span><span id="line-122">                                        setParentDNASequence(dnaSeq, location.getStart().getPosition(), location.getEnd().getPosition());</span>
<span class="source-line-no">123</span><span id="line-123">                                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">124</span><span id="line-124">                                        // TODO is there another solution to handle this exception?</span>
<span class="source-line-no">125</span><span id="line-125">                                        logger.error("Could not add 'coded_by' parent DNA location feature, unrecognised compounds found in DNA sequence: {}", e.getMessage());</span>
<span class="source-line-no">126</span><span id="line-126">                                }</span>
<span class="source-line-no">127</span><span id="line-127">                        }</span>
<span class="source-line-no">128</span><span id="line-128">                }</span>
<span class="source-line-no">129</span><span id="line-129"></span>
<span class="source-line-no">130</span><span id="line-130">        }</span>
<span class="source-line-no">131</span><span id="line-131"></span>
<span class="source-line-no">132</span><span id="line-132">        /**</span>
<span class="source-line-no">133</span><span id="line-133">         * A Protein sequence can be stand alone or loaded from a transcript</span>
<span class="source-line-no">134</span><span id="line-134">         * sequence. The design goal is to allow the creation of a Protein sequence</span>
<span class="source-line-no">135</span><span id="line-135">         * from a Uniprot ID or some other Protein ID that based on cross reference</span>
<span class="source-line-no">136</span><span id="line-136">         * you should be able to get the GeneSequence that codes for the protein if</span>
<span class="source-line-no">137</span><span id="line-137">         * the CDS/Gene region is known. From the GeneSequence you should then be</span>
<span class="source-line-no">138</span><span id="line-138">         * able to get the ChromosomeSequence which then allows you explore flaning</span>
<span class="source-line-no">139</span><span id="line-139">         * regions of the gene sequences. The framework is in place to do this but</span>
<span class="source-line-no">140</span><span id="line-140">         * currently hasn't been implement in the reverse direction starting from</span>
<span class="source-line-no">141</span><span id="line-141">         * the Protein sequence.</span>
<span class="source-line-no">142</span><span id="line-142">         *</span>
<span class="source-line-no">143</span><span id="line-143">         * @param parentDNASequence</span>
<span class="source-line-no">144</span><span id="line-144">         * @param begin</span>
<span class="source-line-no">145</span><span id="line-145">         * @param end</span>
<span class="source-line-no">146</span><span id="line-146">         */</span>
<span class="source-line-no">147</span><span id="line-147">        //TODO - Someone needs to check if this is a bug.  Shouldn't a parentDNASequence be something other then AminoAcid?</span>
<span class="source-line-no">148</span><span id="line-148">        //However, due to the derivation of this class, this is the only possible type argument for this parameter...</span>
<span class="source-line-no">149</span><span id="line-149">        public void setParentDNASequence(AbstractSequence&lt;NucleotideCompound&gt; parentDNASequence, Integer begin, Integer end) {</span>
<span class="source-line-no">150</span><span id="line-150">                this.setParentSequence(parentDNASequence);</span>
<span class="source-line-no">151</span><span id="line-151">                setBioBegin(begin);</span>
<span class="source-line-no">152</span><span id="line-152">                setBioEnd(end);</span>
<span class="source-line-no">153</span><span id="line-153">        }</span>
<span class="source-line-no">154</span><span id="line-154"></span>
<span class="source-line-no">155</span><span id="line-155">        private DNASequence getRawParentSequence(String accessId) throws IOException {</span>
<span class="source-line-no">156</span><span id="line-156">                String seqUrlTemplate = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&amp;id=%s&amp;rettype=fasta&amp;retmode=text";</span>
<span class="source-line-no">157</span><span id="line-157">                URL url = new URL(String.format(seqUrlTemplate, accessId));</span>
<span class="source-line-no">158</span><span id="line-158"></span>
<span class="source-line-no">159</span><span id="line-159">                logger.trace("Getting parent DNA sequence from URL: {}", url.toString());</span>
<span class="source-line-no">160</span><span id="line-160"></span>
<span class="source-line-no">161</span><span id="line-161">                InputStream is = url.openConnection().getInputStream();</span>
<span class="source-line-no">162</span><span id="line-162"></span>
<span class="source-line-no">163</span><span id="line-163">                FastaReader&lt;DNASequence, NucleotideCompound&gt; parentReader</span>
<span class="source-line-no">164</span><span id="line-164">                                = new FastaReader&lt;&gt;(is,</span>
<span class="source-line-no">165</span><span id="line-165">                                                new PlainFastaHeaderParser&lt;DNASequence, NucleotideCompound&gt;(),</span>
<span class="source-line-no">166</span><span id="line-166">                                                new DNASequenceCreator(AmbiguityDNACompoundSet.getDNACompoundSet()));</span>
<span class="source-line-no">167</span><span id="line-167">                Map&lt;String, DNASequence&gt; seq = parentReader.process();</span>
<span class="source-line-no">168</span><span id="line-168"></span>
<span class="source-line-no">169</span><span id="line-169">                DNASequence parentSeq = null;</span>
<span class="source-line-no">170</span><span id="line-170">                if (seq.size() == 1) {</span>
<span class="source-line-no">171</span><span id="line-171">                        parentSeq = seq.values().iterator().next();</span>
<span class="source-line-no">172</span><span id="line-172">                }</span>
<span class="source-line-no">173</span><span id="line-173">                is.close();</span>
<span class="source-line-no">174</span><span id="line-174"></span>
<span class="source-line-no">175</span><span id="line-175">                return parentSeq;</span>
<span class="source-line-no">176</span><span id="line-176">        }</span>
<span class="source-line-no">177</span><span id="line-177"></span>
<span class="source-line-no">178</span><span id="line-178">        private String getSequence(Location cdna) {</span>
<span class="source-line-no">179</span><span id="line-179">                DNASequence rawParent;</span>
<span class="source-line-no">180</span><span id="line-180">                if (!cdna.isComplex()) {</span>
<span class="source-line-no">181</span><span id="line-181">                        try {</span>
<span class="source-line-no">182</span><span id="line-182">                                rawParent = getRawParentSequence(cdna.getAccession().getID());</span>
<span class="source-line-no">183</span><span id="line-183">                                return cdna.getSubSequence(rawParent).getSequenceAsString();</span>
<span class="source-line-no">184</span><span id="line-184">                        } catch (IOException e) {</span>
<span class="source-line-no">185</span><span id="line-185">                                // return null</span>
<span class="source-line-no">186</span><span id="line-186">                                logger.error("Caught IOException when getting DNA sequence for id {}. Error: {}", cdna.getAccession().getID(), e.getMessage());</span>
<span class="source-line-no">187</span><span id="line-187">                                return null;</span>
<span class="source-line-no">188</span><span id="line-188">                        }</span>
<span class="source-line-no">189</span><span id="line-189">                } else {</span>
<span class="source-line-no">190</span><span id="line-190">                        // in case of complex</span>
<span class="source-line-no">191</span><span id="line-191">                        StringBuilder sb = new StringBuilder();</span>
<span class="source-line-no">192</span><span id="line-192"></span>
<span class="source-line-no">193</span><span id="line-193">                        for (Location sub : cdna.getSubLocations()) {</span>
<span class="source-line-no">194</span><span id="line-194">                                String sebStr = getSequence(sub);</span>
<span class="source-line-no">195</span><span id="line-195">                                sb.append((sebStr == null ? "" : sebStr));</span>
<span class="source-line-no">196</span><span id="line-196">                        }</span>
<span class="source-line-no">197</span><span id="line-197"></span>
<span class="source-line-no">198</span><span id="line-198">                        return sb.toString();</span>
<span class="source-line-no">199</span><span id="line-199">                }</span>
<span class="source-line-no">200</span><span id="line-200">        }</span>
<span class="source-line-no">201</span><span id="line-201"></span>
<span class="source-line-no">202</span><span id="line-202">}</span>




























































</pre>
</div>
</main>
</body>
</html>
