<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence, class: TranscriptSequence">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on DATE</span>
<span class="source-line-no">021</span><span id="line-21"> *</span>
<span class="source-line-no">022</span><span id="line-22"> */</span>
<span class="source-line-no">023</span><span id="line-23">package org.biojava.nbio.core.sequence;</span>
<span class="source-line-no">024</span><span id="line-24"></span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.transcription.TranscriptionEngine;</span>
<span class="source-line-no">028</span><span id="line-28">import org.slf4j.Logger;</span>
<span class="source-line-no">029</span><span id="line-29">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">030</span><span id="line-30"></span>
<span class="source-line-no">031</span><span id="line-31">import java.util.ArrayList;</span>
<span class="source-line-no">032</span><span id="line-32">import java.util.Collections;</span>
<span class="source-line-no">033</span><span id="line-33">import java.util.LinkedHashMap;</span>
<span class="source-line-no">034</span><span id="line-34">import java.util.List;</span>
<span class="source-line-no">035</span><span id="line-35">import java.util.Map;</span>
<span class="source-line-no">036</span><span id="line-36"></span>
<span class="source-line-no">037</span><span id="line-37">/**</span>
<span class="source-line-no">038</span><span id="line-38"> * This is the sequence if you want to go from a gene sequence to a protein sequence. Need to start with a</span>
<span class="source-line-no">039</span><span id="line-39"> * ChromosomeSequence then getting a GeneSequence and then a TranscriptSequence</span>
<span class="source-line-no">040</span><span id="line-40"> * @author Scooter Willis</span>
<span class="source-line-no">041</span><span id="line-41"> */</span>
<span class="source-line-no">042</span><span id="line-42">public class TranscriptSequence extends DNASequence {</span>
<span class="source-line-no">043</span><span id="line-43"></span>
<span class="source-line-no">044</span><span id="line-44">        private final static Logger logger = LoggerFactory.getLogger(TranscriptSequence.class);</span>
<span class="source-line-no">045</span><span id="line-45"></span>
<span class="source-line-no">046</span><span id="line-46">        private final List&lt;CDSSequence&gt; cdsSequenceList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">047</span><span id="line-47">        private final Map&lt;String, CDSSequence&gt; cdsSequenceHashMap = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">048</span><span id="line-48">        private StartCodonSequence startCodonSequence = null;</span>
<span class="source-line-no">049</span><span id="line-49">        private StopCodonSequence stopCodonSequence = null;</span>
<span class="source-line-no">050</span><span id="line-50">        private GeneSequence parentGeneSequence = null;</span>
<span class="source-line-no">051</span><span id="line-51"></span>
<span class="source-line-no">052</span><span id="line-52">        /**</span>
<span class="source-line-no">053</span><span id="line-53">         * Use {@code}public TranscriptSequence(GeneSequence parentDNASequence, AccessionID accessionID, int begin, int end){@code}</span>
<span class="source-line-no">054</span><span id="line-54">         * that requires an explicit accessionID</span>
<span class="source-line-no">055</span><span id="line-55">         * @deprecated</span>
<span class="source-line-no">056</span><span id="line-56">         */</span>
<span class="source-line-no">057</span><span id="line-57">        public TranscriptSequence(GeneSequence parentDNASequence, int begin, int end) {</span>
<span class="source-line-no">058</span><span id="line-58">                setCompoundSet(DNACompoundSet.getDNACompoundSet());</span>
<span class="source-line-no">059</span><span id="line-59">                try {</span>
<span class="source-line-no">060</span><span id="line-60">                        initSequenceStorage(parentDNASequence.getSequenceAsString());</span>
<span class="source-line-no">061</span><span id="line-61">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">062</span><span id="line-62">                        throw new IllegalArgumentException(e);</span>
<span class="source-line-no">063</span><span id="line-63">                }</span>
<span class="source-line-no">064</span><span id="line-64">                setParentSequence(parentDNASequence);</span>
<span class="source-line-no">065</span><span id="line-65">                this.parentGeneSequence = parentDNASequence;</span>
<span class="source-line-no">066</span><span id="line-66">                setBioBegin(begin);</span>
<span class="source-line-no">067</span><span id="line-67">                setBioEnd(end);</span>
<span class="source-line-no">068</span><span id="line-68">        }</span>
<span class="source-line-no">069</span><span id="line-69"></span>
<span class="source-line-no">070</span><span id="line-70">        /**</span>
<span class="source-line-no">071</span><span id="line-71">         *</span>
<span class="source-line-no">072</span><span id="line-72">         * @param parentDNASequence</span>
<span class="source-line-no">073</span><span id="line-73">         * @param accessionID</span>
<span class="source-line-no">074</span><span id="line-74">         * @param begin</span>
<span class="source-line-no">075</span><span id="line-75">         * @param end inclusive of end</span>
<span class="source-line-no">076</span><span id="line-76">         * @throws  IllegalArgumentException if the parentDNASequence is incompatible with DNACompoundSet</span>
<span class="source-line-no">077</span><span id="line-77">         */</span>
<span class="source-line-no">078</span><span id="line-78">        public TranscriptSequence(GeneSequence parentDNASequence, AccessionID accessionID, int begin, int end) {</span>
<span class="source-line-no">079</span><span id="line-79">                this(parentDNASequence, begin, end);</span>
<span class="source-line-no">080</span><span id="line-80">                setAccession(accessionID);</span>
<span class="source-line-no">081</span><span id="line-81">        }</span>
<span class="source-line-no">082</span><span id="line-82"></span>
<span class="source-line-no">083</span><span id="line-83">                @Override</span>
<span class="source-line-no">084</span><span id="line-84">        public int getLength() {</span>
<span class="source-line-no">085</span><span id="line-85">                return Math.abs(this.getBioEnd() - this.getBioBegin()) + 1;</span>
<span class="source-line-no">086</span><span id="line-86">        }</span>
<span class="source-line-no">087</span><span id="line-87"></span>
<span class="source-line-no">088</span><span id="line-88">        /**</span>
<span class="source-line-no">089</span><span id="line-89">         * @return the strand</span>
<span class="source-line-no">090</span><span id="line-90">         */</span>
<span class="source-line-no">091</span><span id="line-91">        public Strand getStrand() {</span>
<span class="source-line-no">092</span><span id="line-92">                return parentGeneSequence.getStrand();</span>
<span class="source-line-no">093</span><span id="line-93">        }</span>
<span class="source-line-no">094</span><span id="line-94"></span>
<span class="source-line-no">095</span><span id="line-95">        /**</span>
<span class="source-line-no">096</span><span id="line-96">         * Remove a CDS or coding sequence from the transcript sequence</span>
<span class="source-line-no">097</span><span id="line-97">         * @param accession</span>
<span class="source-line-no">098</span><span id="line-98">         * @return</span>
<span class="source-line-no">099</span><span id="line-99">         */</span>
<span class="source-line-no">100</span><span id="line-100">        public CDSSequence removeCDS(String accession) {</span>
<span class="source-line-no">101</span><span id="line-101">                for (CDSSequence cdsSequence : cdsSequenceList) {</span>
<span class="source-line-no">102</span><span id="line-102">                        if (cdsSequence.getAccession().getID().equals(accession)) {</span>
<span class="source-line-no">103</span><span id="line-103">                                cdsSequenceList.remove(cdsSequence);</span>
<span class="source-line-no">104</span><span id="line-104">                                cdsSequenceHashMap.remove(accession);</span>
<span class="source-line-no">105</span><span id="line-105">                                return cdsSequence;</span>
<span class="source-line-no">106</span><span id="line-106">                        }</span>
<span class="source-line-no">107</span><span id="line-107">                }</span>
<span class="source-line-no">108</span><span id="line-108">                return null;</span>
<span class="source-line-no">109</span><span id="line-109">        }</span>
<span class="source-line-no">110</span><span id="line-110"></span>
<span class="source-line-no">111</span><span id="line-111">        /**</span>
<span class="source-line-no">112</span><span id="line-112">         * Get the CDS sequences that have been added to the TranscriptSequences</span>
<span class="source-line-no">113</span><span id="line-113">         * @return</span>
<span class="source-line-no">114</span><span id="line-114">         */</span>
<span class="source-line-no">115</span><span id="line-115">        public Map&lt;String, CDSSequence&gt; getCDSSequences() {</span>
<span class="source-line-no">116</span><span id="line-116">                return cdsSequenceHashMap;</span>
<span class="source-line-no">117</span><span id="line-117">        }</span>
<span class="source-line-no">118</span><span id="line-118"></span>
<span class="source-line-no">119</span><span id="line-119">        /**</span>
<span class="source-line-no">120</span><span id="line-120">         * Add a Coding Sequence region with phase to the transcript sequence</span>
<span class="source-line-no">121</span><span id="line-121">         * @param accession</span>
<span class="source-line-no">122</span><span id="line-122">         * @param begin</span>
<span class="source-line-no">123</span><span id="line-123">         * @param end</span>
<span class="source-line-no">124</span><span id="line-124">         * @param phase 0,1,2</span>
<span class="source-line-no">125</span><span id="line-125">         * @return</span>
<span class="source-line-no">126</span><span id="line-126">         */</span>
<span class="source-line-no">127</span><span id="line-127">        public CDSSequence addCDS(AccessionID accession, int begin, int end, int phase) throws Exception {</span>
<span class="source-line-no">128</span><span id="line-128">                if (cdsSequenceHashMap.containsKey(accession.getID())) {</span>
<span class="source-line-no">129</span><span id="line-129">                        throw new Exception("Duplicate accession id " + accession.getID());</span>
<span class="source-line-no">130</span><span id="line-130">                }</span>
<span class="source-line-no">131</span><span id="line-131">                CDSSequence cdsSequence = new CDSSequence(this, begin, end, phase); //sense should be the same as parent</span>
<span class="source-line-no">132</span><span id="line-132">                cdsSequence.setAccession(accession);</span>
<span class="source-line-no">133</span><span id="line-133">                cdsSequenceList.add(cdsSequence);</span>
<span class="source-line-no">134</span><span id="line-134">                Collections.sort(cdsSequenceList, new CDSComparator());</span>
<span class="source-line-no">135</span><span id="line-135">                cdsSequenceHashMap.put(accession.getID(), cdsSequence);</span>
<span class="source-line-no">136</span><span id="line-136">                return cdsSequence;</span>
<span class="source-line-no">137</span><span id="line-137">        }</span>
<span class="source-line-no">138</span><span id="line-138"></span>
<span class="source-line-no">139</span><span id="line-139">        /**</span>
<span class="source-line-no">140</span><span id="line-140">         * http://www.sequenceontology.org/gff3.shtml</span>
<span class="source-line-no">141</span><span id="line-141">         * http://biowiki.org/~yam/bioe131/GFF.ppt</span>
<span class="source-line-no">142</span><span id="line-142">         * @return</span>
<span class="source-line-no">143</span><span id="line-143">         */</span>
<span class="source-line-no">144</span><span id="line-144">        /**</span>
<span class="source-line-no">145</span><span id="line-145">         * Return a list of protein sequences based on each CDS sequence</span>
<span class="source-line-no">146</span><span id="line-146">         * where the phase shift between two CDS sequences is assigned to the</span>
<span class="source-line-no">147</span><span id="line-147">         * CDS sequence that starts the triplet. This can be used to map</span>
<span class="source-line-no">148</span><span id="line-148">         * a CDS/exon region of a protein sequence back to the DNA sequence</span>
<span class="source-line-no">149</span><span id="line-149">         * If you have a protein sequence and a predicted gene you can take the</span>
<span class="source-line-no">150</span><span id="line-150">         * predict CDS protein sequences and align back to the protein sequence.</span>
<span class="source-line-no">151</span><span id="line-151">         * If you have errors in mapping the predicted protein CDS regions to</span>
<span class="source-line-no">152</span><span id="line-152">         * an the known protein sequence then you can identify possible errors</span>
<span class="source-line-no">153</span><span id="line-153">         * in the prediction</span>
<span class="source-line-no">154</span><span id="line-154">         *</span>
<span class="source-line-no">155</span><span id="line-155">         * @return</span>
<span class="source-line-no">156</span><span id="line-156">         */</span>
<span class="source-line-no">157</span><span id="line-157">        public List&lt;ProteinSequence&gt; getProteinCDSSequences() {</span>
<span class="source-line-no">158</span><span id="line-158">                List&lt;ProteinSequence&gt; proteinSequenceList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">159</span><span id="line-159">                for (int i = 0; i &lt; cdsSequenceList.size(); i++) {</span>
<span class="source-line-no">160</span><span id="line-160">                        CDSSequence cdsSequence = cdsSequenceList.get(i);</span>
<span class="source-line-no">161</span><span id="line-161">                        String codingSequence = cdsSequence.getCodingSequence();</span>
<span class="source-line-no">162</span><span id="line-162">                        //          logger.debug("CDS {} {} = {}", getStrand(), cdsSequence.getPhase(), codingSequence);</span>
<span class="source-line-no">163</span><span id="line-163">                        if (this.getStrand() == Strand.NEGATIVE) {</span>
<span class="source-line-no">164</span><span id="line-164">                                if (cdsSequence.phase == 1) {</span>
<span class="source-line-no">165</span><span id="line-165">                                        codingSequence = codingSequence.substring(1, codingSequence.length());</span>
<span class="source-line-no">166</span><span id="line-166">                                } else if (cdsSequence.phase == 2) {</span>
<span class="source-line-no">167</span><span id="line-167">                                        codingSequence = codingSequence.substring(2, codingSequence.length());</span>
<span class="source-line-no">168</span><span id="line-168">                                }</span>
<span class="source-line-no">169</span><span id="line-169">                                if (i &lt; cdsSequenceList.size() - 1) {</span>
<span class="source-line-no">170</span><span id="line-170">                                        CDSSequence nextCDSSequence = cdsSequenceList.get(i + 1);</span>
<span class="source-line-no">171</span><span id="line-171">                                        if (nextCDSSequence.phase == 1) {</span>
<span class="source-line-no">172</span><span id="line-172">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</span>
<span class="source-line-no">173</span><span id="line-173">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 1);</span>
<span class="source-line-no">174</span><span id="line-174">                                        } else if (nextCDSSequence.phase == 2) {</span>
<span class="source-line-no">175</span><span id="line-175">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</span>
<span class="source-line-no">176</span><span id="line-176">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 2);</span>
<span class="source-line-no">177</span><span id="line-177">                                        }</span>
<span class="source-line-no">178</span><span id="line-178">                                }</span>
<span class="source-line-no">179</span><span id="line-179">                        } else {</span>
<span class="source-line-no">180</span><span id="line-180">                                if (cdsSequence.phase == 1) {</span>
<span class="source-line-no">181</span><span id="line-181">                                        codingSequence = codingSequence.substring(1, codingSequence.length());</span>
<span class="source-line-no">182</span><span id="line-182">                                } else if (cdsSequence.phase == 2) {</span>
<span class="source-line-no">183</span><span id="line-183">                                        codingSequence = codingSequence.substring(2, codingSequence.length());</span>
<span class="source-line-no">184</span><span id="line-184">                                }</span>
<span class="source-line-no">185</span><span id="line-185">                                if (i &lt; cdsSequenceList.size() - 1) {</span>
<span class="source-line-no">186</span><span id="line-186">                                        CDSSequence nextCDSSequence = cdsSequenceList.get(i + 1);</span>
<span class="source-line-no">187</span><span id="line-187">                                        if (nextCDSSequence.phase == 1) {</span>
<span class="source-line-no">188</span><span id="line-188">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</span>
<span class="source-line-no">189</span><span id="line-189">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 1);</span>
<span class="source-line-no">190</span><span id="line-190">                                        } else if (nextCDSSequence.phase == 2) {</span>
<span class="source-line-no">191</span><span id="line-191">                                                String nextCodingSequence = nextCDSSequence.getCodingSequence();</span>
<span class="source-line-no">192</span><span id="line-192">                                                codingSequence = codingSequence + nextCodingSequence.substring(0, 2);</span>
<span class="source-line-no">193</span><span id="line-193">                                        }</span>
<span class="source-line-no">194</span><span id="line-194">                                }</span>
<span class="source-line-no">195</span><span id="line-195">                        }</span>
<span class="source-line-no">196</span><span id="line-196"></span>
<span class="source-line-no">197</span><span id="line-197"></span>
<span class="source-line-no">198</span><span id="line-198">                        //    logger.debug("Coding Sequence: {}", codingSequence);</span>
<span class="source-line-no">199</span><span id="line-199"></span>
<span class="source-line-no">200</span><span id="line-200">                        DNASequence dnaCodingSequence = null;</span>
<span class="source-line-no">201</span><span id="line-201">                        try {</span>
<span class="source-line-no">202</span><span id="line-202">                                dnaCodingSequence = new DNASequence(codingSequence.toUpperCase());</span>
<span class="source-line-no">203</span><span id="line-203">                        } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">204</span><span id="line-204">                                // if I understand this should not happen, please correct if I'm wrong - JD 2014-10-24</span>
<span class="source-line-no">205</span><span id="line-205">                                logger.error("Could not create DNA coding sequence, {}. This is most likely a bug.", e.getMessage());</span>
<span class="source-line-no">206</span><span id="line-206">                        }</span>
<span class="source-line-no">207</span><span id="line-207">                        RNASequence rnaCodingSequence = dnaCodingSequence.getRNASequence(TranscriptionEngine.getDefault());</span>
<span class="source-line-no">208</span><span id="line-208">                        ProteinSequence proteinSequence = rnaCodingSequence.getProteinSequence(TranscriptionEngine.getDefault());</span>
<span class="source-line-no">209</span><span id="line-209">                        proteinSequence.setAccession(new AccessionID(cdsSequence.getAccession().getID()));</span>
<span class="source-line-no">210</span><span id="line-210">                        proteinSequence.setParentDNASequence(cdsSequence, 1, cdsSequence.getLength());</span>
<span class="source-line-no">211</span><span id="line-211">                        proteinSequenceList.add(proteinSequence);</span>
<span class="source-line-no">212</span><span id="line-212">                }</span>
<span class="source-line-no">213</span><span id="line-213">                return proteinSequenceList;</span>
<span class="source-line-no">214</span><span id="line-214">        }</span>
<span class="source-line-no">215</span><span id="line-215"></span>
<span class="source-line-no">216</span><span id="line-216">        /**</span>
<span class="source-line-no">217</span><span id="line-217">         * Get the stitched together CDS sequences then maps to the cDNA</span>
<span class="source-line-no">218</span><span id="line-218">         * @return</span>
<span class="source-line-no">219</span><span id="line-219">         */</span>
<span class="source-line-no">220</span><span id="line-220">        public DNASequence getDNACodingSequence() {</span>
<span class="source-line-no">221</span><span id="line-221">                StringBuilder sb = new StringBuilder();</span>
<span class="source-line-no">222</span><span id="line-222">                for (CDSSequence cdsSequence : cdsSequenceList) {</span>
<span class="source-line-no">223</span><span id="line-223">                        sb.append(cdsSequence.getCodingSequence());</span>
<span class="source-line-no">224</span><span id="line-224">                }</span>
<span class="source-line-no">225</span><span id="line-225"></span>
<span class="source-line-no">226</span><span id="line-226">                DNASequence dnaSequence = null;</span>
<span class="source-line-no">227</span><span id="line-227">                try {</span>
<span class="source-line-no">228</span><span id="line-228">                        dnaSequence = new DNASequence(sb.toString().toUpperCase());</span>
<span class="source-line-no">229</span><span id="line-229">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">230</span><span id="line-230">                        // if I understand this should not happen, please correct if I'm wrong - JD 2014-10-24</span>
<span class="source-line-no">231</span><span id="line-231">                        logger.error("Could not create DNA coding sequence, {}. This is most likely a bug.", e.getMessage());</span>
<span class="source-line-no">232</span><span id="line-232">                }</span>
<span class="source-line-no">233</span><span id="line-233">                dnaSequence.setAccession(new AccessionID(this.getAccession().getID()));</span>
<span class="source-line-no">234</span><span id="line-234">                return dnaSequence;</span>
<span class="source-line-no">235</span><span id="line-235">        }</span>
<span class="source-line-no">236</span><span id="line-236"></span>
<span class="source-line-no">237</span><span id="line-237">        /**</span>
<span class="source-line-no">238</span><span id="line-238">         * Get the protein sequence</span>
<span class="source-line-no">239</span><span id="line-239">         * @return</span>
<span class="source-line-no">240</span><span id="line-240">         */</span>
<span class="source-line-no">241</span><span id="line-241">        public ProteinSequence getProteinSequence() {</span>
<span class="source-line-no">242</span><span id="line-242">                return getProteinSequence(TranscriptionEngine.getDefault());</span>
<span class="source-line-no">243</span><span id="line-243">        }</span>
<span class="source-line-no">244</span><span id="line-244"></span>
<span class="source-line-no">245</span><span id="line-245">        /**</span>
<span class="source-line-no">246</span><span id="line-246">         * Get the protein sequence with user defined TranscriptEngine</span>
<span class="source-line-no">247</span><span id="line-247">         * @param engine</span>
<span class="source-line-no">248</span><span id="line-248">         * @return</span>
<span class="source-line-no">249</span><span id="line-249">         */</span>
<span class="source-line-no">250</span><span id="line-250">        public ProteinSequence getProteinSequence(TranscriptionEngine engine) {</span>
<span class="source-line-no">251</span><span id="line-251">                DNASequence dnaCodingSequence = getDNACodingSequence();</span>
<span class="source-line-no">252</span><span id="line-252">                RNASequence rnaCodingSequence = dnaCodingSequence.getRNASequence(engine);</span>
<span class="source-line-no">253</span><span id="line-253">                ProteinSequence proteinSequence = rnaCodingSequence.getProteinSequence(engine);</span>
<span class="source-line-no">254</span><span id="line-254">                proteinSequence.setAccession(new AccessionID(this.getAccession().getID()));</span>
<span class="source-line-no">255</span><span id="line-255"></span>
<span class="source-line-no">256</span><span id="line-256">                return proteinSequence;</span>
<span class="source-line-no">257</span><span id="line-257">        }</span>
<span class="source-line-no">258</span><span id="line-258"></span>
<span class="source-line-no">259</span><span id="line-259">        /**</span>
<span class="source-line-no">260</span><span id="line-260">         * @return the startCodonSequence</span>
<span class="source-line-no">261</span><span id="line-261">         */</span>
<span class="source-line-no">262</span><span id="line-262">        public StartCodonSequence getStartCodonSequence() {</span>
<span class="source-line-no">263</span><span id="line-263">                return startCodonSequence;</span>
<span class="source-line-no">264</span><span id="line-264">        }</span>
<span class="source-line-no">265</span><span id="line-265"></span>
<span class="source-line-no">266</span><span id="line-266">        /**</span>
<span class="source-line-no">267</span><span id="line-267">         * Sets the start codon sequence at given begin /  end location. Note that calling this method multiple times</span>
<span class="source-line-no">268</span><span id="line-268">         * will replace any existing value.</span>
<span class="source-line-no">269</span><span id="line-269">         * @param accession</span>
<span class="source-line-no">270</span><span id="line-270">         * @param begin</span>
<span class="source-line-no">271</span><span id="line-271">         * @param end</span>
<span class="source-line-no">272</span><span id="line-272">         */</span>
<span class="source-line-no">273</span><span id="line-273">        public void addStartCodonSequence(AccessionID accession, int begin, int end) {</span>
<span class="source-line-no">274</span><span id="line-274">                this.startCodonSequence = new StartCodonSequence(this, begin, end);</span>
<span class="source-line-no">275</span><span id="line-275">                startCodonSequence.setAccession(accession);</span>
<span class="source-line-no">276</span><span id="line-276">        }</span>
<span class="source-line-no">277</span><span id="line-277"></span>
<span class="source-line-no">278</span><span id="line-278">        /**</span>
<span class="source-line-no">279</span><span id="line-279">         * @return the stopCodonSequence</span>
<span class="source-line-no">280</span><span id="line-280">         */</span>
<span class="source-line-no">281</span><span id="line-281">        public StopCodonSequence getStopCodonSequence() {</span>
<span class="source-line-no">282</span><span id="line-282">                return stopCodonSequence;</span>
<span class="source-line-no">283</span><span id="line-283">        }</span>
<span class="source-line-no">284</span><span id="line-284"></span>
<span class="source-line-no">285</span><span id="line-285">        /**</span>
<span class="source-line-no">286</span><span id="line-286">         * Sets the stop codon sequence at given begin /  end location. Note that calling this method multiple times</span>
<span class="source-line-no">287</span><span id="line-287">         * will replace any existing value.</span>
<span class="source-line-no">288</span><span id="line-288">         * @param accession</span>
<span class="source-line-no">289</span><span id="line-289">         * @param begin</span>
<span class="source-line-no">290</span><span id="line-290">         * @param end</span>
<span class="source-line-no">291</span><span id="line-291">         */</span>
<span class="source-line-no">292</span><span id="line-292">        public void addStopCodonSequence(AccessionID accession, int begin, int end) {</span>
<span class="source-line-no">293</span><span id="line-293">                this.stopCodonSequence = new StopCodonSequence(this, begin, end);</span>
<span class="source-line-no">294</span><span id="line-294">                stopCodonSequence.setAccession(accession);</span>
<span class="source-line-no">295</span><span id="line-295">        }</span>
<span class="source-line-no">296</span><span id="line-296">}</span>




























































</pre>
</div>
</main>
</body>
</html>
