<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21"></a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.bio.seq.io;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import java.io.BufferedReader;</a>
<span class="sourceLineNo">025</span><a id="line.25">import java.io.FileReader;</a>
<span class="sourceLineNo">026</span><a id="line.26">import java.io.OutputStream;</a>
<span class="sourceLineNo">027</span><a id="line.27">import java.io.PrintStream;</a>
<span class="sourceLineNo">028</span><a id="line.28">import java.util.Iterator;</a>
<span class="sourceLineNo">029</span><a id="line.29">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">030</span><a id="line.30">import java.util.Vector;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.regex.Matcher;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.regex.Pattern;</a>
<span class="sourceLineNo">033</span><a id="line.33"></a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.bio.BioException;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.bio.alignment.Alignment;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.bio.alignment.SimpleAlignment;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.bio.seq.DNATools;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.biojava.bio.seq.ProteinTools;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.bio.symbol.FiniteAlphabet;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.biojava.bio.symbol.IllegalSymbolException;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.biojava.bio.symbol.Symbol;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.biojava.bio.symbol.SymbolList;</a>
<span class="sourceLineNo">043</span><a id="line.43"></a>
<span class="sourceLineNo">044</span><a id="line.44">/**</a>
<span class="sourceLineNo">045</span><a id="line.45"> * @author raemig</a>
<span class="sourceLineNo">046</span><a id="line.46"> * @author Thomas Down</a>
<span class="sourceLineNo">047</span><a id="line.47"> * @author Keith James</a>
<span class="sourceLineNo">048</span><a id="line.48"> * @author Nimesh Singh</a>
<span class="sourceLineNo">049</span><a id="line.49"> * @author Mark Schreiber</a>
<span class="sourceLineNo">050</span><a id="line.50"> * @author Matthew Pocock</a>
<span class="sourceLineNo">051</span><a id="line.51"> * @author Bradford Powell</a>
<span class="sourceLineNo">052</span><a id="line.52"> */</a>
<span class="sourceLineNo">053</span><a id="line.53"></a>
<span class="sourceLineNo">054</span><a id="line.54">public class MSFAlignmentFormat implements AlignmentFormat {</a>
<span class="sourceLineNo">055</span><a id="line.55">        private static final boolean DEBUGPRINT = false;</a>
<span class="sourceLineNo">056</span><a id="line.56">        private static final int DNA = 1;</a>
<span class="sourceLineNo">057</span><a id="line.57">        private static final int PROTEIN = 2;</a>
<span class="sourceLineNo">058</span><a id="line.58"></a>
<span class="sourceLineNo">059</span><a id="line.59">        public MSFAlignmentFormat() {</a>
<span class="sourceLineNo">060</span><a id="line.60">        }</a>
<span class="sourceLineNo">061</span><a id="line.61"></a>
<span class="sourceLineNo">062</span><a id="line.62">        /**</a>
<span class="sourceLineNo">063</span><a id="line.63">         * used to quick test the code</a>
<span class="sourceLineNo">064</span><a id="line.64">         * </a>
<span class="sourceLineNo">065</span><a id="line.65">         * @param args</a>
<span class="sourceLineNo">066</span><a id="line.66">         */</a>
<span class="sourceLineNo">067</span><a id="line.67">        public static void main(String[] args) {</a>
<span class="sourceLineNo">068</span><a id="line.68">                String filename;</a>
<span class="sourceLineNo">069</span><a id="line.69">                if (args.length &lt; 1) {</a>
<span class="sourceLineNo">070</span><a id="line.70">                        filename = "SimpleMSF.msf"; // change to your favorite</a>
<span class="sourceLineNo">071</span><a id="line.71">                } else {</a>
<span class="sourceLineNo">072</span><a id="line.72">                        filename = args[0];</a>
<span class="sourceLineNo">073</span><a id="line.73">                }</a>
<span class="sourceLineNo">074</span><a id="line.74">                try {</a>
<span class="sourceLineNo">075</span><a id="line.75">                        BufferedReader reader = new BufferedReader(new FileReader(filename));</a>
<span class="sourceLineNo">076</span><a id="line.76">                        MSFAlignmentFormat MSFAlignmentFormat1 = new MSFAlignmentFormat();</a>
<span class="sourceLineNo">077</span><a id="line.77">                        MSFAlignmentFormat1.read(reader);</a>
<span class="sourceLineNo">078</span><a id="line.78">                } catch (Exception E) {</a>
<span class="sourceLineNo">079</span><a id="line.79">                }</a>
<span class="sourceLineNo">080</span><a id="line.80">        }</a>
<span class="sourceLineNo">081</span><a id="line.81"></a>
<span class="sourceLineNo">082</span><a id="line.82">        /**</a>
<span class="sourceLineNo">083</span><a id="line.83">         * Reads an MSF Alignment File</a>
<span class="sourceLineNo">084</span><a id="line.84">         * </a>
<span class="sourceLineNo">085</span><a id="line.85">         * @param reader</a>
<span class="sourceLineNo">086</span><a id="line.86">         *            The file reader</a>
<span class="sourceLineNo">087</span><a id="line.87">         * @return Alignment A SimpleAlignment consisting of the sequences in the</a>
<span class="sourceLineNo">088</span><a id="line.88">         *         file.</a>
<span class="sourceLineNo">089</span><a id="line.89">         */</a>
<span class="sourceLineNo">090</span><a id="line.90">        public Alignment read(BufferedReader reader) {</a>
<span class="sourceLineNo">091</span><a id="line.91">                Vector sequenceNames = new Vector();</a>
<span class="sourceLineNo">092</span><a id="line.92">                String sequenceName = null;</a>
<span class="sourceLineNo">093</span><a id="line.93">                StringBuffer sequenceData[] = null;</a>
<span class="sourceLineNo">094</span><a id="line.94">                int startOfData = 0; // the start of the sequence data in the line</a>
<span class="sourceLineNo">095</span><a id="line.95">                int currSeqCount = 0; // which sequence data you are currently trying to</a>
<span class="sourceLineNo">096</span><a id="line.96">                // get</a>
<span class="sourceLineNo">097</span><a id="line.97">                try {</a>
<span class="sourceLineNo">098</span><a id="line.98">                        Pattern mtc = Pattern</a>
<span class="sourceLineNo">099</span><a id="line.99">                                        .compile("(Name:|NAME:)\\s+(.*?)\\s+(oo|OO|Len:|LEN:)");</a>
<span class="sourceLineNo">100</span><a id="line.100">                        Pattern removewhitespace = Pattern.compile("\\s");</a>
<span class="sourceLineNo">101</span><a id="line.101">                        // REMatch rem = null;</a>
<span class="sourceLineNo">102</span><a id="line.102">                        String line = reader.readLine();</a>
<span class="sourceLineNo">103</span><a id="line.103">                        // parse past header</a>
<span class="sourceLineNo">104</span><a id="line.104">                        while (line.toUpperCase().indexOf("NAME:") == -1) {</a>
<span class="sourceLineNo">105</span><a id="line.105">                                line = reader.readLine();</a>
<span class="sourceLineNo">106</span><a id="line.106">                        }</a>
<span class="sourceLineNo">107</span><a id="line.107">                        // read each name (between Name: and Len:</a>
<span class="sourceLineNo">108</span><a id="line.108">                        while ((line.indexOf("//") == -1) &amp;&amp; ((line.trim()).length() != 0)) {</a>
<span class="sourceLineNo">109</span><a id="line.109">                                Matcher matcher = mtc.matcher(line);</a>
<span class="sourceLineNo">110</span><a id="line.110">                                if (!matcher.find()) {</a>
<span class="sourceLineNo">111</span><a id="line.111">                                        break;</a>
<span class="sourceLineNo">112</span><a id="line.112">                                } // end of sequence names</a>
<span class="sourceLineNo">113</span><a id="line.113">                                // sequenceName = line.substring(rem.getSubStartIndex(1),</a>
<span class="sourceLineNo">114</span><a id="line.114">                                // rem.getSubEndIndex(1));</a>
<span class="sourceLineNo">115</span><a id="line.115">                                if ((line.trim()).length() == 0) {</a>
<span class="sourceLineNo">116</span><a id="line.116">                                        break;</a>
<span class="sourceLineNo">117</span><a id="line.117">                                }</a>
<span class="sourceLineNo">118</span><a id="line.118">                                sequenceName = matcher.group(2).trim();</a>
<span class="sourceLineNo">119</span><a id="line.119">                                sequenceNames.add(sequenceName);</a>
<span class="sourceLineNo">120</span><a id="line.120"></a>
<span class="sourceLineNo">121</span><a id="line.121">                                line = reader.readLine();</a>
<span class="sourceLineNo">122</span><a id="line.122">                        }</a>
<span class="sourceLineNo">123</span><a id="line.123">                        sequenceData = new StringBuffer[sequenceNames.size()];</a>
<span class="sourceLineNo">124</span><a id="line.124">                        for (int it = 0; it &lt; sequenceNames.size(); it++) {</a>
<span class="sourceLineNo">125</span><a id="line.125">                                sequenceData[it] = new StringBuffer();</a>
<span class="sourceLineNo">126</span><a id="line.126">                        }</a>
<span class="sourceLineNo">127</span><a id="line.127">                        // until you get a line that matches the first sequence</a>
<span class="sourceLineNo">128</span><a id="line.128">                        while (line.indexOf((String) sequenceNames.get(0)) == -1) {</a>
<span class="sourceLineNo">129</span><a id="line.129">                                line = reader.readLine();</a>
<span class="sourceLineNo">130</span><a id="line.130">                        }</a>
<span class="sourceLineNo">131</span><a id="line.131">                        // now you on the first line of the sequence data</a>
<span class="sourceLineNo">132</span><a id="line.132">                        while (line != null) {</a>
<span class="sourceLineNo">133</span><a id="line.133">                                for (currSeqCount = 0; currSeqCount &lt; sequenceNames.size(); currSeqCount++) {// you</a>
<span class="sourceLineNo">134</span><a id="line.134">                                        // could</a>
<span class="sourceLineNo">135</span><a id="line.135">                                        // also</a>
<span class="sourceLineNo">136</span><a id="line.136">                                        // check</a>
<span class="sourceLineNo">137</span><a id="line.137">                                        // for</a>
<span class="sourceLineNo">138</span><a id="line.138">                                        // order</a>
<span class="sourceLineNo">139</span><a id="line.139">                                        // of</a>
<span class="sourceLineNo">140</span><a id="line.140">                                        // names</a>
<span class="sourceLineNo">141</span><a id="line.141">                                        if (line.indexOf((String) sequenceNames.get(currSeqCount)) == -1) {</a>
<span class="sourceLineNo">142</span><a id="line.142">                                                break;</a>
<span class="sourceLineNo">143</span><a id="line.143">                                        } // error</a>
<span class="sourceLineNo">144</span><a id="line.144"></a>
<span class="sourceLineNo">145</span><a id="line.145">                                        startOfData = line.indexOf((String) sequenceNames</a>
<span class="sourceLineNo">146</span><a id="line.146">                                                        .get(currSeqCount))</a>
<span class="sourceLineNo">147</span><a id="line.147">                                                        + ((String) sequenceNames.get(currSeqCount))</a>
<span class="sourceLineNo">148</span><a id="line.148">                                                                        .length();</a>
<span class="sourceLineNo">149</span><a id="line.149">                                        line = (line.substring(startOfData));</a>
<span class="sourceLineNo">150</span><a id="line.150">                                        line = removewhitespace.matcher(line).replaceAll("");</a>
<span class="sourceLineNo">151</span><a id="line.151">                                        sequenceData[currSeqCount].append(line); // make into string</a>
<span class="sourceLineNo">152</span><a id="line.152">                                        // buffer</a>
<span class="sourceLineNo">153</span><a id="line.153">                                        line = reader.readLine();</a>
<span class="sourceLineNo">154</span><a id="line.154">                                        if ((currSeqCount &lt; sequenceNames.size() - 1)</a>
<span class="sourceLineNo">155</span><a id="line.155">                                                        &amp;&amp; (line.trim().length() == 0)) {</a>
<span class="sourceLineNo">156</span><a id="line.156">                                                break;</a>
<span class="sourceLineNo">157</span><a id="line.157">                                        } // could be an error</a>
<span class="sourceLineNo">158</span><a id="line.158">                                }</a>
<span class="sourceLineNo">159</span><a id="line.159">                                // until you get a line that matches the first sequence</a>
<span class="sourceLineNo">160</span><a id="line.160">                                while ((line != null)</a>
<span class="sourceLineNo">161</span><a id="line.161">                                                &amp;&amp; (line.indexOf((String) sequenceNames.get(0)) == -1)) // ||</a>
<span class="sourceLineNo">162</span><a id="line.162">                                // (</a>
<span class="sourceLineNo">163</span><a id="line.163">                                // (line.trim())</a>
<span class="sourceLineNo">164</span><a id="line.164">                                // .length()&gt;0</a>
<span class="sourceLineNo">165</span><a id="line.165">                                // )</a>
<span class="sourceLineNo">166</span><a id="line.166">                                // )</a>
<span class="sourceLineNo">167</span><a id="line.167">                                {</a>
<span class="sourceLineNo">168</span><a id="line.168">                                        line = reader.readLine();</a>
<span class="sourceLineNo">169</span><a id="line.169">                                }</a>
<span class="sourceLineNo">170</span><a id="line.170">                        }</a>
<span class="sourceLineNo">171</span><a id="line.171">                        // print them out for testing</a>
<span class="sourceLineNo">172</span><a id="line.172">                        if (DEBUGPRINT) {</a>
<span class="sourceLineNo">173</span><a id="line.173">                                for (currSeqCount = 0; currSeqCount &lt; sequenceNames.size(); currSeqCount++) {</a>
<span class="sourceLineNo">174</span><a id="line.174">                                        System.out.println((String) sequenceNames.get(currSeqCount)</a>
<span class="sourceLineNo">175</span><a id="line.175">                                                        + ":" + sequenceData[currSeqCount]);</a>
<span class="sourceLineNo">176</span><a id="line.176">                                }</a>
<span class="sourceLineNo">177</span><a id="line.177">                        }</a>
<span class="sourceLineNo">178</span><a id="line.178">                        // check DNA, RNA or Prot</a>
<span class="sourceLineNo">179</span><a id="line.179">                        StringBuffer testString = new StringBuffer();</a>
<span class="sourceLineNo">180</span><a id="line.180">                        for (currSeqCount = 0; currSeqCount &lt; sequenceNames.size(); currSeqCount++) {</a>
<span class="sourceLineNo">181</span><a id="line.181">                                testString.append(sequenceData[currSeqCount]);</a>
<span class="sourceLineNo">182</span><a id="line.182">                        }</a>
<span class="sourceLineNo">183</span><a id="line.183">                        String testStringUpper = testString.toString().toUpperCase();</a>
<span class="sourceLineNo">184</span><a id="line.184"></a>
<span class="sourceLineNo">185</span><a id="line.185">                        // now parse through them and create gapped symbol lists</a>
<span class="sourceLineNo">186</span><a id="line.186">                        LinkedHashMap sequenceDataMap = new LinkedHashMap();</a>
<span class="sourceLineNo">187</span><a id="line.187">                        FiniteAlphabet alph = null;</a>
<span class="sourceLineNo">188</span><a id="line.188"></a>
<span class="sourceLineNo">189</span><a id="line.189">                        for (int i = 0; i &lt; testStringUpper.length(); i++) {</a>
<span class="sourceLineNo">190</span><a id="line.190">                                char c = testStringUpper.charAt(i);</a>
<span class="sourceLineNo">191</span><a id="line.191">                                if (c == 'F' || c == 'L' || c == 'I' || c == 'P' || c == 'Q'</a>
<span class="sourceLineNo">192</span><a id="line.192">                                                || c == 'E') {</a>
<span class="sourceLineNo">193</span><a id="line.193">                                        alph = ProteinTools.getTAlphabet();</a>
<span class="sourceLineNo">194</span><a id="line.194">                                        break;</a>
<span class="sourceLineNo">195</span><a id="line.195">                                }</a>
<span class="sourceLineNo">196</span><a id="line.196">                        }</a>
<span class="sourceLineNo">197</span><a id="line.197">                        if (alph == null) {</a>
<span class="sourceLineNo">198</span><a id="line.198">                                alph = DNATools.getDNA();</a>
<span class="sourceLineNo">199</span><a id="line.199">                        }</a>
<span class="sourceLineNo">200</span><a id="line.200">                        for (currSeqCount = 0; currSeqCount &lt; sequenceNames.size(); currSeqCount++) {</a>
<span class="sourceLineNo">201</span><a id="line.201">                                String sd = sequenceData[currSeqCount].toString();</a>
<span class="sourceLineNo">202</span><a id="line.202">                                // change stop codons to specified symbols</a>
<span class="sourceLineNo">203</span><a id="line.203">                                sd = sd.replace('~', '-'); // sometimes this is a term signal</a>
<span class="sourceLineNo">204</span><a id="line.204">                                // not a gap</a>
<span class="sourceLineNo">205</span><a id="line.205">                                sd = sd.replace('.', '-'); // sometimes this is a term signal</a>
<span class="sourceLineNo">206</span><a id="line.206">                                // not a gap</a>
<span class="sourceLineNo">207</span><a id="line.207">                                sequenceDataMap.put((String) sequenceNames.get(currSeqCount),</a>
<span class="sourceLineNo">208</span><a id="line.208">                                                alph == ProteinTools.getTAlphabet() ? ProteinTools</a>
<span class="sourceLineNo">209</span><a id="line.209">                                                                .createGappedProteinSequence(sd,</a>
<span class="sourceLineNo">210</span><a id="line.210">                                                                                (String) sequenceNames</a>
<span class="sourceLineNo">211</span><a id="line.211">                                                                                                .get(currSeqCount)) : DNATools</a>
<span class="sourceLineNo">212</span><a id="line.212">                                                                .createGappedDNASequence(sd,</a>
<span class="sourceLineNo">213</span><a id="line.213">                                                                                (String) sequenceNames</a>
<span class="sourceLineNo">214</span><a id="line.214">                                                                                                .get(currSeqCount)));</a>
<span class="sourceLineNo">215</span><a id="line.215">                        }</a>
<span class="sourceLineNo">216</span><a id="line.216">                        SimpleAlignment sa = new SimpleAlignment(sequenceDataMap);</a>
<span class="sourceLineNo">217</span><a id="line.217">                        return (sa);</a>
<span class="sourceLineNo">218</span><a id="line.218">                } catch (Exception e) {</a>
<span class="sourceLineNo">219</span><a id="line.219">                        e.printStackTrace();</a>
<span class="sourceLineNo">220</span><a id="line.220">                        System.err.println("MSFFormatReader " + e.getMessage());</a>
<span class="sourceLineNo">221</span><a id="line.221">                        // throw (e);</a>
<span class="sourceLineNo">222</span><a id="line.222">                }</a>
<span class="sourceLineNo">223</span><a id="line.223">                return (null);</a>
<span class="sourceLineNo">224</span><a id="line.224">        } // end read it</a>
<span class="sourceLineNo">225</span><a id="line.225"></a>
<span class="sourceLineNo">226</span><a id="line.226">        // This is where I am writing an alignment writer</a>
<span class="sourceLineNo">227</span><a id="line.227">        public void write(OutputStream os, Alignment align, int fileType)</a>
<span class="sourceLineNo">228</span><a id="line.228">                        throws BioException, IllegalSymbolException {</a>
<span class="sourceLineNo">229</span><a id="line.229">                PrintStream out = new PrintStream(os);</a>
<span class="sourceLineNo">230</span><a id="line.230">                Object labels[] = align.getLabels().toArray();</a>
<span class="sourceLineNo">231</span><a id="line.231">                int numSeqs = labels.length;</a>
<span class="sourceLineNo">232</span><a id="line.232">                Iterator&lt;?&gt; seqIts[] = new Iterator&lt;?&gt;[numSeqs];</a>
<span class="sourceLineNo">233</span><a id="line.233">                int maxLabelLength = 0;</a>
<span class="sourceLineNo">234</span><a id="line.234">                for (int i = 0; i &lt; numSeqs; i++) {</a>
<span class="sourceLineNo">235</span><a id="line.235">                        seqIts[i] = align.symbolListForLabel(labels[i].toString())</a>
<span class="sourceLineNo">236</span><a id="line.236">                                        .iterator();</a>
<span class="sourceLineNo">237</span><a id="line.237">                        if (((String) labels[i]).length() &gt; maxLabelLength) {</a>
<span class="sourceLineNo">238</span><a id="line.238">                                maxLabelLength = ((String) labels[i]).length();</a>
<span class="sourceLineNo">239</span><a id="line.239">                        }</a>
<span class="sourceLineNo">240</span><a id="line.240">                }</a>
<span class="sourceLineNo">241</span><a id="line.241">                String nl = System.getProperty("line.separator");</a>
<span class="sourceLineNo">242</span><a id="line.242">                SymbolTokenization toke = null;</a>
<span class="sourceLineNo">243</span><a id="line.243"></a>
<span class="sourceLineNo">244</span><a id="line.244">                // really should determine the filetype based on one of the seqeunces</a>
<span class="sourceLineNo">245</span><a id="line.245">                // alphabet</a>
<span class="sourceLineNo">246</span><a id="line.246"></a>
<span class="sourceLineNo">247</span><a id="line.247">                if (align.symbolListForLabel(labels[0].toString()).getAlphabet() == DNATools</a>
<span class="sourceLineNo">248</span><a id="line.248">                                .getDNA()) {</a>
<span class="sourceLineNo">249</span><a id="line.249">                        fileType = DNA;</a>
<span class="sourceLineNo">250</span><a id="line.250"></a>
<span class="sourceLineNo">251</span><a id="line.251">                } else if (align.symbolListForLabel(labels[0].toString()).getAlphabet() == ProteinTools</a>
<span class="sourceLineNo">252</span><a id="line.252">                                .getAlphabet()</a>
<span class="sourceLineNo">253</span><a id="line.253">                                || align.symbolListForLabel(labels[0].toString()).getAlphabet() == ProteinTools</a>
<span class="sourceLineNo">254</span><a id="line.254">                                                .getTAlphabet()) {</a>
<span class="sourceLineNo">255</span><a id="line.255">                        fileType = PROTEIN;</a>
<span class="sourceLineNo">256</span><a id="line.256">                }</a>
<span class="sourceLineNo">257</span><a id="line.257"></a>
<span class="sourceLineNo">258</span><a id="line.258">                if (fileType == DNA) {</a>
<span class="sourceLineNo">259</span><a id="line.259">                        out.print("PileUp" + nl);</a>
<span class="sourceLineNo">260</span><a id="line.260">                        out.print(nl);</a>
<span class="sourceLineNo">261</span><a id="line.261">                        out.print(" MSF: " + align.length() + "  Type: ");</a>
<span class="sourceLineNo">262</span><a id="line.262">                        out.print("N");</a>
<span class="sourceLineNo">263</span><a id="line.263">                        out.print("   Check: " + 0 + "   .." + nl);</a>
<span class="sourceLineNo">264</span><a id="line.264">                        toke = DNATools.getDNA().getTokenization("token");</a>
<span class="sourceLineNo">265</span><a id="line.265">                } else if (fileType == PROTEIN) {</a>
<span class="sourceLineNo">266</span><a id="line.266">                        out.print("PileUp" + nl);</a>
<span class="sourceLineNo">267</span><a id="line.267">                        out.print(nl);</a>
<span class="sourceLineNo">268</span><a id="line.268">                        out.print(" MSF: " + align.length() + "  Type: ");</a>
<span class="sourceLineNo">269</span><a id="line.269">                        out.print("P");</a>
<span class="sourceLineNo">270</span><a id="line.270">                        out.print("   Check: " + 0 + "   .." + nl);</a>
<span class="sourceLineNo">271</span><a id="line.271">                        toke = ProteinTools.getTAlphabet().getTokenization("token");</a>
<span class="sourceLineNo">272</span><a id="line.272">                } else {</a>
<span class="sourceLineNo">273</span><a id="line.273">                        System.out</a>
<span class="sourceLineNo">274</span><a id="line.274">                                        .println("MSFAlignment.write -- File type not recognized.");</a>
<span class="sourceLineNo">275</span><a id="line.275">                        return;</a>
<span class="sourceLineNo">276</span><a id="line.276">                }</a>
<span class="sourceLineNo">277</span><a id="line.277">                out.print(nl);</a>
<span class="sourceLineNo">278</span><a id="line.278"></a>
<span class="sourceLineNo">279</span><a id="line.279">                for (int i = 0; i &lt; numSeqs; i++) {</a>
<span class="sourceLineNo">280</span><a id="line.280">                        out.print(" Name: " + labels[i]);</a>
<span class="sourceLineNo">281</span><a id="line.281">                        for (int j = 0; j &lt; (maxLabelLength - ((String) labels[i]).length()); j++) {// padding</a>
<span class="sourceLineNo">282</span><a id="line.282">                                out.print(" ");</a>
<span class="sourceLineNo">283</span><a id="line.283">                        }</a>
<span class="sourceLineNo">284</span><a id="line.284">                        out.print("  Len: " + align.length() + "        Check: " + 0</a>
<span class="sourceLineNo">285</span><a id="line.285">                                        + "     Weight: " + 0 + nl); // this really should be seq</a>
<span class="sourceLineNo">286</span><a id="line.286">                        // length?</a>
<span class="sourceLineNo">287</span><a id="line.287">                }</a>
<span class="sourceLineNo">288</span><a id="line.288"></a>
<span class="sourceLineNo">289</span><a id="line.289">                out.println(nl + "//" + nl + nl);</a>
<span class="sourceLineNo">290</span><a id="line.290">                // now should print the numbering line</a>
<span class="sourceLineNo">291</span><a id="line.291"></a>
<span class="sourceLineNo">292</span><a id="line.292">                while (seqIts[0].hasNext()) {</a>
<span class="sourceLineNo">293</span><a id="line.293">                        for (int i = 0; i &lt; numSeqs; i++) {</a>
<span class="sourceLineNo">294</span><a id="line.294">                                while (((String) labels[i]).length() &lt; maxLabelLength + 1) {</a>
<span class="sourceLineNo">295</span><a id="line.295">                                        labels[i] = " " + labels[i];</a>
<span class="sourceLineNo">296</span><a id="line.296">                                }</a>
<span class="sourceLineNo">297</span><a id="line.297">                                out.print(labels[i] + " ");</a>
<span class="sourceLineNo">298</span><a id="line.298">                                theLabel: for (int j = 0; j &lt; 5; j++) {</a>
<span class="sourceLineNo">299</span><a id="line.299">                                        out.print(" ");</a>
<span class="sourceLineNo">300</span><a id="line.300">                                        for (int k = 0; k &lt; 10; k++) {</a>
<span class="sourceLineNo">301</span><a id="line.301">                                                if (seqIts[i].hasNext()) {</a>
<span class="sourceLineNo">302</span><a id="line.302">                                                        out.print(toke.tokenizeSymbol((Symbol) seqIts[i]</a>
<span class="sourceLineNo">303</span><a id="line.303">                                                                        .next()));</a>
<span class="sourceLineNo">304</span><a id="line.304">                                                } else {</a>
<span class="sourceLineNo">305</span><a id="line.305">                                                        break theLabel;</a>
<span class="sourceLineNo">306</span><a id="line.306">                                                }</a>
<span class="sourceLineNo">307</span><a id="line.307">                                        }</a>
<span class="sourceLineNo">308</span><a id="line.308">                                }</a>
<span class="sourceLineNo">309</span><a id="line.309">                                out.print(nl);</a>
<span class="sourceLineNo">310</span><a id="line.310">                        }</a>
<span class="sourceLineNo">311</span><a id="line.311">                        out.print(nl);</a>
<span class="sourceLineNo">312</span><a id="line.312">                }</a>
<span class="sourceLineNo">313</span><a id="line.313"></a>
<span class="sourceLineNo">314</span><a id="line.314">        } // end write</a>
<span class="sourceLineNo">315</span><a id="line.315"></a>
<span class="sourceLineNo">316</span><a id="line.316">        public void writeDna(OutputStream os, Alignment align) throws BioException,</a>
<span class="sourceLineNo">317</span><a id="line.317">                        IllegalSymbolException {</a>
<span class="sourceLineNo">318</span><a id="line.318">                write(os, align, DNA);</a>
<span class="sourceLineNo">319</span><a id="line.319">        }</a>
<span class="sourceLineNo">320</span><a id="line.320"></a>
<span class="sourceLineNo">321</span><a id="line.321">        public void writeProtein(OutputStream os, Alignment align)</a>
<span class="sourceLineNo">322</span><a id="line.322">                        throws BioException, IllegalSymbolException {</a>
<span class="sourceLineNo">323</span><a id="line.323">                write(os, align, PROTEIN);</a>
<span class="sourceLineNo">324</span><a id="line.324">        }</a>
<span class="sourceLineNo">325</span><a id="line.325"></a>
<span class="sourceLineNo">326</span><a id="line.326">} // end class</a>
<span class="sourceLineNo">327</span><a id="line.327"></a>




























































</pre>
</div>
</main>
</body>
</html>
