<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:46 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>FastqTools (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FastqTools (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastqTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqReader.html" title="interface in org.biojava.nbio.sequencing.io.fastq"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqVariant.html" title="enum in org.biojava.nbio.sequencing.io.fastq"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/sequencing/io/fastq/FastqTools.html" target="_top">Frames</a></li>
<li><a href="FastqTools.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.sequencing.io.fastq</div>
<h2 title="Class FastqTools" class="title">Class FastqTools</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.sequencing.io.fastq.FastqTools</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public final class <span class="strong">FastqTools</span>
extends <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Utility methods for FASTQ formatted sequences.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>3.0.3</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#convert(org.biojava.nbio.sequencing.io.fastq.Fastq,%20org.biojava.nbio.sequencing.io.fastq.FastqVariant)">convert</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq,
       <a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqVariant.html" title="enum in org.biojava.nbio.sequencing.io.fastq">FastqVariant</a>&nbsp;variant)</code>
<div class="block">Convert the specified FASTQ formatted sequence to the
 specified FASTQ sequence format variant.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequence(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequence</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> from the specified FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequenceWithErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequenceWithErrorProbabilities</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with error probabilities from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequenceWithQualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequenceWithQualityScores</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequenceWithQualityScoresAndErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequenceWithQualityScoresAndErrorProbabilities</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores and error probabilities from the
 specified FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">createErrorProbabilities</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QuantityFeature</code></a> from the error probabilities of the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createQualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq)">createQualityScores</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a> from the quality scores of the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Number.html?is-external=true" title="class or interface in java.lang">Number</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#errorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">errorProbabilities</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Return the error probabilities from the specified FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#errorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq,%20double[])">errorProbabilities</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq,
                  double[]&nbsp;errorProbabilities)</code>
<div class="block">Copy the error probabilities from the specified FASTQ formatted sequence into the specified double array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Number.html?is-external=true" title="class or interface in java.lang">Number</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#qualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq)">qualityScores</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Return the quality scores from the specified FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#qualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq,%20int[])">qualityScores</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq,
             int[]&nbsp;qualityScores)</code>
<div class="block">Copy the quality scores from the specified FASTQ formatted sequence into the specified int array.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="createDNASequence(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createDNASequence</h4>
<pre>public static&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;createDNASequence(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)
                                     throws <a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> from the specified FASTQ formatted sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> from the specified FASTQ formatted sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code> - if DNA sequence in fastq contains unrecognised compounds</dd></dl>
</li>
</ul>
<a name="createDNASequenceWithQualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createDNASequenceWithQualityScores</h4>
<pre>public static&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;createDNASequenceWithQualityScores(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)
                                                      throws <a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores from the specified
 FASTQ formatted sequence.  The quality scores are stored in a <a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a>
 with a type <code>"qualityScores"</code> the same length as the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores from the specified FASTQ formatted sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code> - if DNA sequence in fastq contains unrecognised compounds</dd></dl>
</li>
</ul>
<a name="createDNASequenceWithErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createDNASequenceWithErrorProbabilities</h4>
<pre>public static&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;createDNASequenceWithErrorProbabilities(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)
                                                           throws <a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with error probabilities from the specified
 FASTQ formatted sequence.  The error probabilities are stored in a <a href="../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QuantityFeature</code></a>
 with a type <code>"errorProbabilities"</code> the same length as the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with error probabilities from the specified FASTQ formatted sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code> - if DNA sequence in fastq contains unrecognised compounds</dd></dl>
</li>
</ul>
<a name="createDNASequenceWithQualityScoresAndErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createDNASequenceWithQualityScoresAndErrorProbabilities</h4>
<pre>public static&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;createDNASequenceWithQualityScoresAndErrorProbabilities(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)
                                                                           throws <a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores and error probabilities from the
 specified FASTQ formatted sequence.  The quality scores are stored in a <a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a>
 with a type <code>"qualityScores"</code> the same length as the sequence and the error
 probabilities are stored in a <a href="../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QuantityFeature</code></a> with a type <code>"errorProbabilities"</code>
 the same length as the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores and error probabilities from the specified
    FASTQ formatted sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code> - if DNA sequence in fastq contains unrecognised compounds</dd></dl>
</li>
</ul>
<a name="createQualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createQualityScores</h4>
<pre>public static&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;createQualityScores(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</pre>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a> from the quality scores of the specified
 FASTQ formatted sequence.  The quality scores feature has a type <code>"qualityScores"</code>
 and will be the same length as the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new <a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a> from the quality scores of the specified FASTQ
    formatted sequence</dd></dl>
</li>
</ul>
<a name="createErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createErrorProbabilities</h4>
<pre>public static&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;createErrorProbabilities(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</pre>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QuantityFeature</code></a> from the error probabilities of the specified
 FASTQ formatted sequence.  The error probabilities feature has a type <code>"errorProbabilities"</code>
 and will be the same length as the sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new <a href="../../../../../../org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a> from the error probabilities of the specified FASTQ
    formatted sequence</dd></dl>
</li>
</ul>
<a name="qualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>qualityScores</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Number.html?is-external=true" title="class or interface in java.lang">Number</a>&gt;&nbsp;qualityScores(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</pre>
<div class="block">Return the quality scores from the specified FASTQ formatted sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>the quality scores from the specified FASTQ formatted sequence</dd></dl>
</li>
</ul>
<a name="qualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq, int[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>qualityScores</h4>
<pre>public static&nbsp;int[]&nbsp;qualityScores(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq,
                  int[]&nbsp;qualityScores)</pre>
<div class="block">Copy the quality scores from the specified FASTQ formatted sequence into the specified int array.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd><dd><code>qualityScores</code> - int array of quality scores, must not be null and must be the same
    length as the FASTQ formatted sequence quality</dd>
<dt><span class="strong">Returns:</span></dt><dd>the specified int array of quality scores</dd></dl>
</li>
</ul>
<a name="errorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>errorProbabilities</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Number.html?is-external=true" title="class or interface in java.lang">Number</a>&gt;&nbsp;errorProbabilities(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</pre>
<div class="block">Return the error probabilities from the specified FASTQ formatted sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>the error probabilities from the specified FASTQ formatted sequence</dd></dl>
</li>
</ul>
<a name="errorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq, double[])">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>errorProbabilities</h4>
<pre>public static&nbsp;double[]&nbsp;errorProbabilities(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq,
                          double[]&nbsp;errorProbabilities)</pre>
<div class="block">Copy the error probabilities from the specified FASTQ formatted sequence into the specified double array.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd><dd><code>errorProbabilities</code> - double array of error probabilities, must not be null and must be the same
    length as the FASTQ formatted sequence quality</dd>
<dt><span class="strong">Returns:</span></dt><dd>the specified double array of error probabilities</dd></dl>
</li>
</ul>
<a name="convert(org.biojava.nbio.sequencing.io.fastq.Fastq, org.biojava.nbio.sequencing.io.fastq.FastqVariant)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>convert</h4>
<pre>public static&nbsp;<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;convert(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq,
            <a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqVariant.html" title="enum in org.biojava.nbio.sequencing.io.fastq">FastqVariant</a>&nbsp;variant)</pre>
<div class="block">Convert the specified FASTQ formatted sequence to the
 specified FASTQ sequence format variant.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastq</code> - FASTQ formatted sequence, must not be null</dd><dd><code>variant</code> - FASTQ sequence format variant, must not be null</dd>
<dt><span class="strong">Returns:</span></dt><dd>the specified FASTQ formatted sequence converted to the
    specified FASTQ sequence format variant</dd><dt><span class="strong">Since:</span></dt>
  <dd>4.2</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastqTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqReader.html" title="interface in org.biojava.nbio.sequencing.io.fastq"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqVariant.html" title="enum in org.biojava.nbio.sequencing.io.fastq"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/sequencing/io/fastq/FastqTools.html" target="_top">Frames</a></li>
<li><a href="FastqTools.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
