<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21"></a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.bio.chromatogram;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.bio.alignment.Alignment;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.bio.symbol.AtomicSymbol;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.bio.symbol.IllegalSymbolException;</a>
<span class="sourceLineNo">027</span><a id="line.27"></a>
<span class="sourceLineNo">028</span><a id="line.28">/** </a>
<span class="sourceLineNo">029</span><a id="line.29"> * Encapsulates the basic information you would want from a chromatogram.</a>
<span class="sourceLineNo">030</span><a id="line.30"> * Read-only.</a>
<span class="sourceLineNo">031</span><a id="line.31"> *</a>
<span class="sourceLineNo">032</span><a id="line.32"> *  @author Rhett Sutphin (&lt;a href="http://genome.uiowa.edu/"&gt;UI CBCB&lt;/a&gt;)</a>
<span class="sourceLineNo">033</span><a id="line.33"> * @author Matthew Pocock</a>
<span class="sourceLineNo">034</span><a id="line.34"> * @since 1.3</a>
<span class="sourceLineNo">035</span><a id="line.35"> */</a>
<span class="sourceLineNo">036</span><a id="line.36">public interface Chromatogram {</a>
<span class="sourceLineNo">037</span><a id="line.37">    /**</a>
<span class="sourceLineNo">038</span><a id="line.38">     * The sequence label for the list of called bases.</a>
<span class="sourceLineNo">039</span><a id="line.39">     */</a>
<span class="sourceLineNo">040</span><a id="line.40">    public static String DNA = "dna";</a>
<span class="sourceLineNo">041</span><a id="line.41"></a>
<span class="sourceLineNo">042</span><a id="line.42">    /**</a>
<span class="sourceLineNo">043</span><a id="line.43">     * The sequence label for the trace offsets of the called bases.</a>
<span class="sourceLineNo">044</span><a id="line.44">     */</a>
<span class="sourceLineNo">045</span><a id="line.45">    public static String OFFSETS = "trace-offsets";</a>
<span class="sourceLineNo">046</span><a id="line.46">    </a>
<span class="sourceLineNo">047</span><a id="line.47">    /** Gets the max intensity from all the traces.  Must be equivalent</a>
<span class="sourceLineNo">048</span><a id="line.48">     *  to the max of calling {@link #getMax(AtomicSymbol)} on each</a>
<span class="sourceLineNo">049</span><a id="line.49">     *  of the four non-ambiguous DNA nucleotides.</a>
<span class="sourceLineNo">050</span><a id="line.50">     *  @return the max intensity </a>
<span class="sourceLineNo">051</span><a id="line.51">     */</a>
<span class="sourceLineNo">052</span><a id="line.52">    public int getMax();</a>
<span class="sourceLineNo">053</span><a id="line.53">    /** Gets the max intensity on the trace for the specified nucleotide.</a>
<span class="sourceLineNo">054</span><a id="line.54">     *  @param nucleotide the trace to examine.  Must be a concrete </a>
<span class="sourceLineNo">055</span><a id="line.55">     *         (non-ambiguous) nucleotide from the DNA alphabet</a>
<span class="sourceLineNo">056</span><a id="line.56">     *  @throws IllegalSymbolException when the nucleotide isn't from the DNA</a>
<span class="sourceLineNo">057</span><a id="line.57">     *          alphabet</a>
<span class="sourceLineNo">058</span><a id="line.58">     *  @return the max intensity</a>
<span class="sourceLineNo">059</span><a id="line.59">     */</a>
<span class="sourceLineNo">060</span><a id="line.60">    public int getMax(AtomicSymbol nucleotide) throws IllegalSymbolException;</a>
<span class="sourceLineNo">061</span><a id="line.61">    </a>
<span class="sourceLineNo">062</span><a id="line.62">    /** Returns the length of the trace of the Chromatogram.</a>
<span class="sourceLineNo">063</span><a id="line.63">     *  @return the number of samples in the trace</a>
<span class="sourceLineNo">064</span><a id="line.64">     *  @see #getTrace(AtomicSymbol)</a>
<span class="sourceLineNo">065</span><a id="line.65">     */</a>
<span class="sourceLineNo">066</span><a id="line.66">    public int getTraceLength();</a>
<span class="sourceLineNo">067</span><a id="line.67">    /** Returns an array containing the intensities of the sampled waveform</a>
<span class="sourceLineNo">068</span><a id="line.68">     *  representing the chromatogram trace for base &lt;code&gt;nucleotide&lt;/code&gt;.  </a>
<span class="sourceLineNo">069</span><a id="line.69">     *  This may be a reference the actual internal representation of the </a>
<span class="sourceLineNo">070</span><a id="line.70">     *  samples, so callers &lt;b&gt;must not modify it&lt;/b&gt;.</a>
<span class="sourceLineNo">071</span><a id="line.71">     *  &lt;p&gt;</a>
<span class="sourceLineNo">072</span><a id="line.72">     *  The resulting array for each nucleotide must be {@link #getTraceLength}</a>
<span class="sourceLineNo">073</span><a id="line.73">     *  &lt;code&gt;int&lt;/code&gt;s long.</a>
<span class="sourceLineNo">074</span><a id="line.74">     *  &lt;/p&gt;</a>
<span class="sourceLineNo">075</span><a id="line.75">     *  @param nucleotide the trace to examine.  Must be the symbol for A, C, G, or T</a>
<span class="sourceLineNo">076</span><a id="line.76">     *         as provided by {@link org.biojava.bio.seq.DNATools}</a>
<span class="sourceLineNo">077</span><a id="line.77">     *  @throws IllegalSymbolException if &lt;code&gt;nucleotide&lt;/code&gt; isn't in the DNA alphabet</a>
<span class="sourceLineNo">078</span><a id="line.78">     *  @return an array of integers representing the values of a particular </a>
<span class="sourceLineNo">079</span><a id="line.79">     *          chromatogram trace.</a>
<span class="sourceLineNo">080</span><a id="line.80">     */</a>
<span class="sourceLineNo">081</span><a id="line.81">    public int[] getTrace(AtomicSymbol nucleotide) throws IllegalSymbolException;</a>
<span class="sourceLineNo">082</span><a id="line.82">    </a>
<span class="sourceLineNo">083</span><a id="line.83">    /** </a>
<span class="sourceLineNo">084</span><a id="line.84">     * Returns the number of bits of the traces which are significant.  For </a>
<span class="sourceLineNo">085</span><a id="line.85">     * instance, if the chromatogram were originally encoded with a single byte </a>
<span class="sourceLineNo">086</span><a id="line.86">     * per trace sample, this method must return 8.</a>
<span class="sourceLineNo">087</span><a id="line.87">     * @return the number of significant bits</a>
<span class="sourceLineNo">088</span><a id="line.88">     */</a>
<span class="sourceLineNo">089</span><a id="line.89">    public int getSignificantBits();</a>
<span class="sourceLineNo">090</span><a id="line.90"></a>
<span class="sourceLineNo">091</span><a id="line.91">    /**</a>
<span class="sourceLineNo">092</span><a id="line.92">     * Returns an alignment that describes the base calls for this chromatogram.</a>
<span class="sourceLineNo">093</span><a id="line.93">     * All of the &lt;code&gt;SymbolList&lt;/code&gt;s in this alignment must be the same</a>
<span class="sourceLineNo">094</span><a id="line.94">     * length and that length must equal {@link #getSequenceLength}.</a>
<span class="sourceLineNo">095</span><a id="line.95">     * &lt;p&gt;</a>
<span class="sourceLineNo">096</span><a id="line.96">     * The alignment must contain, at the least, two sequences:</a>
<span class="sourceLineNo">097</span><a id="line.97">     * &lt;/p&gt;</a>
<span class="sourceLineNo">098</span><a id="line.98">     * &lt;ol&gt;</a>
<span class="sourceLineNo">099</span><a id="line.99">     *   &lt;li&gt;A sequence containing the called bases.  The alphabet of this list</a>
<span class="sourceLineNo">100</span><a id="line.100">     *       must be {@link org.biojava.bio.seq.DNATools#getDNA()}.  </a>
<span class="sourceLineNo">101</span><a id="line.101">     *       The label for this list in the alignment must be </a>
<span class="sourceLineNo">102</span><a id="line.102">     *       &lt;code&gt;Chromatogram.DNA&lt;/code&gt;&lt;/li&gt;</a>
<span class="sourceLineNo">103</span><a id="line.103">     *   &lt;li&gt;A sequence containing the trace offsets at which the called bases </a>
<span class="sourceLineNo">104</span><a id="line.104">     *       were called.  The alphabet of this list must be an </a>
<span class="sourceLineNo">105</span><a id="line.105">     *       {@link org.biojava.bio.symbol.IntegerAlphabet} or a</a>
<span class="sourceLineNo">106</span><a id="line.106">     *       {@link org.biojava.bio.symbol.IntegerAlphabet.SubIntegerAlphabet}.</a>
<span class="sourceLineNo">107</span><a id="line.107">     *       The label for this list in the alignment must be </a>
<span class="sourceLineNo">108</span><a id="line.108">     *       &lt;code&gt;Chromatogram.OFFSETS&lt;/code&gt;.&lt;/li&gt;</a>
<span class="sourceLineNo">109</span><a id="line.109">     * &lt;/ol&gt;</a>
<span class="sourceLineNo">110</span><a id="line.110">     * &lt;p&gt;</a>
<span class="sourceLineNo">111</span><a id="line.111">     * Implementors may provide other sequences as they see fit.</a>
<span class="sourceLineNo">112</span><a id="line.112">     * &lt;/p&gt;</a>
<span class="sourceLineNo">113</span><a id="line.113">     * @return an alignment of at least two sequences, as described above.</a>
<span class="sourceLineNo">114</span><a id="line.114">     */</a>
<span class="sourceLineNo">115</span><a id="line.115">    public Alignment getBaseCalls();</a>
<span class="sourceLineNo">116</span><a id="line.116">    /** </a>
<span class="sourceLineNo">117</span><a id="line.117">     * Returns the number of bases called by whatever base-calling software</a>
<span class="sourceLineNo">118</span><a id="line.118">     * analyzed the chromatogram as loaded.  Must equal </a>
<span class="sourceLineNo">119</span><a id="line.119">     * &lt;code&gt;{@link #getBaseCalls}.length()&lt;/code&gt;.</a>
<span class="sourceLineNo">120</span><a id="line.120">     * @return the number of bases</a>
<span class="sourceLineNo">121</span><a id="line.121">     */</a>
<span class="sourceLineNo">122</span><a id="line.122">    public int getSequenceLength();</a>
<span class="sourceLineNo">123</span><a id="line.123">    </a>
<span class="sourceLineNo">124</span><a id="line.124">    /** </a>
<span class="sourceLineNo">125</span><a id="line.125">     * Returns a new &lt;code&gt;Chromatogram&lt;/code&gt; representing the reverse</a>
<span class="sourceLineNo">126</span><a id="line.126">     * complement of this one.</a>
<span class="sourceLineNo">127</span><a id="line.127">     * &lt;p&gt;</a>
<span class="sourceLineNo">128</span><a id="line.128">     * Implementors should copy the metadata about the chromatogram (i.e., base </a>
<span class="sourceLineNo">129</span><a id="line.129">     * calls) as is appropriate to their formats.</a>
<span class="sourceLineNo">130</span><a id="line.130">     * &lt;/p&gt;</a>
<span class="sourceLineNo">131</span><a id="line.131">     * @return a new chromatogram that is the reverse complement of this one</a>
<span class="sourceLineNo">132</span><a id="line.132">     */</a>
<span class="sourceLineNo">133</span><a id="line.133">    public Chromatogram reverseComplement();</a>
<span class="sourceLineNo">134</span><a id="line.134">}</a>




























































</pre>
</div>
</main>
</body>
</html>
