<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on 01-21-2010<a name="line.20"></a>
<span class="sourceLineNo">021</span> */<a name="line.21"></a>
<span class="sourceLineNo">022</span><a name="line.22"></a>
<span class="sourceLineNo">023</span>package org.biojava.nbio.core.sequence.io;<a name="line.23"></a>
<span class="sourceLineNo">024</span><a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.sequence.*;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.io.template.FastaHeaderFormatInterface;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.slf4j.Logger;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.slf4j.LoggerFactory;<a name="line.29"></a>
<span class="sourceLineNo">030</span><a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.io.OutputStream;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.ArrayList;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.Collection;<a name="line.33"></a>
<span class="sourceLineNo">034</span><a name="line.34"></a>
<span class="sourceLineNo">035</span>/**<a name="line.35"></a>
<span class="sourceLineNo">036</span> * A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to 3 prime version.<a name="line.36"></a>
<span class="sourceLineNo">037</span> * It is also an option to write out the gene sequence where the exon regions are upper case<a name="line.37"></a>
<span class="sourceLineNo">038</span> * 6/22/2010 FastaWriter needs to be sequence aware to handle writing out a GeneSequence which is negative Strand with the proper sequence<a name="line.38"></a>
<span class="sourceLineNo">039</span> * @author Scooter Willis &lt;willishf at gmail dot com&gt;<a name="line.39"></a>
<span class="sourceLineNo">040</span> */<a name="line.40"></a>
<span class="sourceLineNo">041</span>public class FastaGeneWriter {<a name="line.41"></a>
<span class="sourceLineNo">042</span><a name="line.42"></a>
<span class="sourceLineNo">043</span>        private final static Logger logger = LoggerFactory.getLogger(FastaGeneWriter.class);<a name="line.43"></a>
<span class="sourceLineNo">044</span><a name="line.44"></a>
<span class="sourceLineNo">045</span>        boolean showExonUppercase = false;<a name="line.45"></a>
<span class="sourceLineNo">046</span>        OutputStream os;<a name="line.46"></a>
<span class="sourceLineNo">047</span>        Collection&lt;GeneSequence&gt; sequences;<a name="line.47"></a>
<span class="sourceLineNo">048</span>        FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt; headerFormat;<a name="line.48"></a>
<span class="sourceLineNo">049</span>        private int lineLength = 60;<a name="line.49"></a>
<span class="sourceLineNo">050</span>/**<a name="line.50"></a>
<span class="sourceLineNo">051</span> *<a name="line.51"></a>
<span class="sourceLineNo">052</span> * @param os<a name="line.52"></a>
<span class="sourceLineNo">053</span> * @param sequences<a name="line.53"></a>
<span class="sourceLineNo">054</span> * @param headerFormat<a name="line.54"></a>
<span class="sourceLineNo">055</span> * @param showExonUppercase<a name="line.55"></a>
<span class="sourceLineNo">056</span> */<a name="line.56"></a>
<span class="sourceLineNo">057</span>        public FastaGeneWriter(OutputStream os, Collection&lt;GeneSequence&gt; sequences, FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt; headerFormat, boolean showExonUppercase) {<a name="line.57"></a>
<span class="sourceLineNo">058</span>                this(os, sequences, headerFormat, showExonUppercase, 60);<a name="line.58"></a>
<span class="sourceLineNo">059</span>        }<a name="line.59"></a>
<span class="sourceLineNo">060</span>/**<a name="line.60"></a>
<span class="sourceLineNo">061</span> *<a name="line.61"></a>
<span class="sourceLineNo">062</span> * @param os<a name="line.62"></a>
<span class="sourceLineNo">063</span> * @param sequences<a name="line.63"></a>
<span class="sourceLineNo">064</span> * @param headerFormat<a name="line.64"></a>
<span class="sourceLineNo">065</span> * @param showExonUppercase<a name="line.65"></a>
<span class="sourceLineNo">066</span> * @param lineLength<a name="line.66"></a>
<span class="sourceLineNo">067</span> */<a name="line.67"></a>
<span class="sourceLineNo">068</span>        public FastaGeneWriter(OutputStream os, Collection&lt;GeneSequence&gt; sequences, FastaHeaderFormatInterface&lt;GeneSequence, NucleotideCompound&gt; headerFormat, boolean showExonUppercase, int lineLength) {<a name="line.68"></a>
<span class="sourceLineNo">069</span>                this.os = os;<a name="line.69"></a>
<span class="sourceLineNo">070</span>                this.sequences = sequences;<a name="line.70"></a>
<span class="sourceLineNo">071</span>                this.headerFormat = headerFormat;<a name="line.71"></a>
<span class="sourceLineNo">072</span>                this.lineLength = lineLength;<a name="line.72"></a>
<span class="sourceLineNo">073</span>                this.showExonUppercase = showExonUppercase;<a name="line.73"></a>
<span class="sourceLineNo">074</span>        }<a name="line.74"></a>
<span class="sourceLineNo">075</span>/**<a name="line.75"></a>
<span class="sourceLineNo">076</span> *<a name="line.76"></a>
<span class="sourceLineNo">077</span> * @throws Exception<a name="line.77"></a>
<span class="sourceLineNo">078</span> */<a name="line.78"></a>
<span class="sourceLineNo">079</span>        public void process() throws Exception {<a name="line.79"></a>
<span class="sourceLineNo">080</span>                byte[] lineSep = System.getProperty("line.separator").getBytes();<a name="line.80"></a>
<span class="sourceLineNo">081</span><a name="line.81"></a>
<span class="sourceLineNo">082</span>                for (GeneSequence sequence : sequences) {<a name="line.82"></a>
<span class="sourceLineNo">083</span>                        String header = headerFormat.getHeader(sequence);<a name="line.83"></a>
<span class="sourceLineNo">084</span>                        os.write('&gt;');<a name="line.84"></a>
<span class="sourceLineNo">085</span>                        os.write(header.getBytes());<a name="line.85"></a>
<span class="sourceLineNo">086</span>                        os.write(lineSep);<a name="line.86"></a>
<span class="sourceLineNo">087</span><a name="line.87"></a>
<span class="sourceLineNo">088</span>                        int compoundCount = 0;<a name="line.88"></a>
<span class="sourceLineNo">089</span>                        String seq = "";<a name="line.89"></a>
<span class="sourceLineNo">090</span>                        //GeneSequence currently has a strand attribute to indicate direction<a name="line.90"></a>
<span class="sourceLineNo">091</span><a name="line.91"></a>
<span class="sourceLineNo">092</span>                        seq = sequence.getSequence5PrimeTo3Prime().getSequenceAsString();<a name="line.92"></a>
<span class="sourceLineNo">093</span>                        if (showExonUppercase) {<a name="line.93"></a>
<span class="sourceLineNo">094</span>                                StringBuilder sb = new StringBuilder(seq.toLowerCase());<a name="line.94"></a>
<span class="sourceLineNo">095</span>                                int geneBioBegin = sequence.getBioBegin();<a name="line.95"></a>
<span class="sourceLineNo">096</span>                                int geneBioEnd = sequence.getBioEnd();<a name="line.96"></a>
<span class="sourceLineNo">097</span>                                for (ExonSequence exonSequence : sequence.getExonSequences()) {<a name="line.97"></a>
<span class="sourceLineNo">098</span>                                        int featureBioBegin = 0;<a name="line.98"></a>
<span class="sourceLineNo">099</span>                                        int featureBioEnd = 0;<a name="line.99"></a>
<span class="sourceLineNo">100</span>                                        if (sequence.getStrand() != Strand.NEGATIVE) {<a name="line.100"></a>
<span class="sourceLineNo">101</span>                                                featureBioBegin = exonSequence.getBioBegin() - geneBioBegin;<a name="line.101"></a>
<span class="sourceLineNo">102</span>                                                featureBioEnd = exonSequence.getBioEnd() - geneBioBegin;<a name="line.102"></a>
<span class="sourceLineNo">103</span>                                        } else {<a name="line.103"></a>
<span class="sourceLineNo">104</span>                                                featureBioBegin = geneBioEnd - exonSequence.getBioEnd();<a name="line.104"></a>
<span class="sourceLineNo">105</span>                                                featureBioEnd = geneBioEnd - exonSequence.getBioBegin();<a name="line.105"></a>
<span class="sourceLineNo">106</span>                                        }<a name="line.106"></a>
<span class="sourceLineNo">107</span>                                        if (featureBioBegin &lt; 0 || featureBioEnd &lt; 0 || featureBioEnd &gt; sb.length() || featureBioBegin &gt; sb.length()) {<a name="line.107"></a>
<span class="sourceLineNo">108</span>                                                logger.warn("Bad Feature, Accession: {}, Sequence Strand: {}, Gene Begin: {}, Gene End: {}, Exon Begin: {}, Exon End: {}", sequence.getAccession().toString(), sequence.getStrand(), geneBioBegin, geneBioEnd, exonSequence.getBioBegin(), exonSequence.getBioEnd());<a name="line.108"></a>
<span class="sourceLineNo">109</span>                                        } else {<a name="line.109"></a>
<span class="sourceLineNo">110</span>                                                for (int i = featureBioBegin; i &lt;= featureBioEnd; i++) {<a name="line.110"></a>
<span class="sourceLineNo">111</span>                                                        char ch = sb.charAt(i);<a name="line.111"></a>
<span class="sourceLineNo">112</span>                                                        //probably not the fastest but the safest way if language is not standard ASCII<a name="line.112"></a>
<span class="sourceLineNo">113</span>                                                        String temp = String.valueOf(ch);<a name="line.113"></a>
<span class="sourceLineNo">114</span>                                                        ch = temp.toUpperCase().charAt(0);<a name="line.114"></a>
<span class="sourceLineNo">115</span>                                                        sb.setCharAt(i, ch);<a name="line.115"></a>
<span class="sourceLineNo">116</span>                                                }<a name="line.116"></a>
<span class="sourceLineNo">117</span>                                        }<a name="line.117"></a>
<span class="sourceLineNo">118</span>                                }<a name="line.118"></a>
<span class="sourceLineNo">119</span>                                seq = sb.toString();<a name="line.119"></a>
<span class="sourceLineNo">120</span>                        }<a name="line.120"></a>
<span class="sourceLineNo">121</span><a name="line.121"></a>
<span class="sourceLineNo">122</span>                        for (int i = 0; i &lt; seq.length(); i++) {<a name="line.122"></a>
<span class="sourceLineNo">123</span>                                os.write(seq.charAt(i));<a name="line.123"></a>
<span class="sourceLineNo">124</span>                                compoundCount++;<a name="line.124"></a>
<span class="sourceLineNo">125</span>                                if (compoundCount == lineLength) {<a name="line.125"></a>
<span class="sourceLineNo">126</span>                                        os.write(lineSep);<a name="line.126"></a>
<span class="sourceLineNo">127</span>                                        compoundCount = 0;<a name="line.127"></a>
<span class="sourceLineNo">128</span>                                }<a name="line.128"></a>
<span class="sourceLineNo">129</span><a name="line.129"></a>
<span class="sourceLineNo">130</span>                        }<a name="line.130"></a>
<span class="sourceLineNo">131</span><a name="line.131"></a>
<span class="sourceLineNo">132</span><a name="line.132"></a>
<span class="sourceLineNo">133</span>                        //If we had sequence which was a reciprocal of line length<a name="line.133"></a>
<span class="sourceLineNo">134</span>                        //then don't write the line terminator as this has already written<a name="line.134"></a>
<span class="sourceLineNo">135</span>                        //it<a name="line.135"></a>
<span class="sourceLineNo">136</span>                        if ((sequence.getLength() % getLineLength()) != 0) {<a name="line.136"></a>
<span class="sourceLineNo">137</span>                                os.write(lineSep);<a name="line.137"></a>
<span class="sourceLineNo">138</span>                        }<a name="line.138"></a>
<span class="sourceLineNo">139</span>                }<a name="line.139"></a>
<span class="sourceLineNo">140</span>        }<a name="line.140"></a>
<span class="sourceLineNo">141</span><a name="line.141"></a>
<span class="sourceLineNo">142</span>        /**<a name="line.142"></a>
<span class="sourceLineNo">143</span>         * @return the lineLength<a name="line.143"></a>
<span class="sourceLineNo">144</span>         */<a name="line.144"></a>
<span class="sourceLineNo">145</span>        public int getLineLength() {<a name="line.145"></a>
<span class="sourceLineNo">146</span>                return lineLength;<a name="line.146"></a>
<span class="sourceLineNo">147</span>        }<a name="line.147"></a>
<span class="sourceLineNo">148</span><a name="line.148"></a>
<span class="sourceLineNo">149</span>        /**<a name="line.149"></a>
<span class="sourceLineNo">150</span>         * @param lineLength the lineLength to set<a name="line.150"></a>
<span class="sourceLineNo">151</span>         */<a name="line.151"></a>
<span class="sourceLineNo">152</span>        public void setLineLength(int lineLength) {<a name="line.152"></a>
<span class="sourceLineNo">153</span>                this.lineLength = lineLength;<a name="line.153"></a>
<span class="sourceLineNo">154</span>        }<a name="line.154"></a>
<span class="sourceLineNo">155</span><a name="line.155"></a>
<span class="sourceLineNo">156</span>        public static void main(String[] args) {<a name="line.156"></a>
<span class="sourceLineNo">157</span><a name="line.157"></a>
<span class="sourceLineNo">158</span>                try {<a name="line.158"></a>
<span class="sourceLineNo">159</span>                        ArrayList&lt;GeneSequence&gt; sequences = new ArrayList&lt;GeneSequence&gt;();<a name="line.159"></a>
<span class="sourceLineNo">160</span>                        ChromosomeSequence seq1 = new ChromosomeSequence("ATATATATATATATATATATATATATATATATACGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCATATATATATATATATATATATACGCGCGCGCGCGCGCGCATATATATATATATATATATATATATATATATACGCGCGCGCGCGCGCGCGCGCGCGCGCGCGCATATATATATATATATATATATACGCGCGCGCGCGCGCGC");<a name="line.160"></a>
<span class="sourceLineNo">161</span>                        GeneSequence gene1 = seq1.addGene(new AccessionID("gene1"), 1, 20, Strand.POSITIVE);<a name="line.161"></a>
<span class="sourceLineNo">162</span><a name="line.162"></a>
<span class="sourceLineNo">163</span>                        gene1.addExon(new AccessionID("t1_1_10"), 1, 10);<a name="line.163"></a>
<span class="sourceLineNo">164</span>                        gene1.addExon(new AccessionID("t1_12_15"), 12, 15);<a name="line.164"></a>
<span class="sourceLineNo">165</span>                        GeneSequence gene2 = seq1.addGene(new AccessionID("gene2"), 1, 20, Strand.NEGATIVE);<a name="line.165"></a>
<span class="sourceLineNo">166</span><a name="line.166"></a>
<span class="sourceLineNo">167</span>                        gene2.addExon(new AccessionID("t2_1_10"), 1, 10);<a name="line.167"></a>
<span class="sourceLineNo">168</span>                        gene2.addExon(new AccessionID("t2_12_15"), 12, 15);<a name="line.168"></a>
<span class="sourceLineNo">169</span>                        sequences.add(gene1);<a name="line.169"></a>
<span class="sourceLineNo">170</span>                        sequences.add(gene2);<a name="line.170"></a>
<span class="sourceLineNo">171</span><a name="line.171"></a>
<span class="sourceLineNo">172</span><a name="line.172"></a>
<span class="sourceLineNo">173</span>                        FastaGeneWriter fastaWriter = new FastaGeneWriter(System.out, sequences, new GenericFastaHeaderFormat&lt;GeneSequence, NucleotideCompound&gt;(), true);<a name="line.173"></a>
<span class="sourceLineNo">174</span>                        fastaWriter.process();<a name="line.174"></a>
<span class="sourceLineNo">175</span><a name="line.175"></a>
<span class="sourceLineNo">176</span><a name="line.176"></a>
<span class="sourceLineNo">177</span>                } catch (Exception e) {<a name="line.177"></a>
<span class="sourceLineNo">178</span>                        logger.warn("Exception: ", e);<a name="line.178"></a>
<span class="sourceLineNo">179</span>                }<a name="line.179"></a>
<span class="sourceLineNo">180</span>        }<a name="line.180"></a>
<span class="sourceLineNo">181</span>}<a name="line.181"></a>




























































</pre>
</div>
</body>
</html>
