<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on 01-21-2010<a name="line.20"></a>
<span class="sourceLineNo">021</span> */<a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.nbio.core.sequence.io;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.sequence.DNASequence;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.io.template.SequenceParserInterface;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.loader.SequenceFileProxyLoader;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.sequence.template.AbstractSequence;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.sequence.template.CompoundSet;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.nbio.core.sequence.template.ProxySequenceReader;<a name="line.32"></a>
<span class="sourceLineNo">033</span><a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.io.File;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import java.io.IOException;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import java.util.List;<a name="line.36"></a>
<span class="sourceLineNo">037</span><a name="line.37"></a>
<span class="sourceLineNo">038</span>/**<a name="line.38"></a>
<span class="sourceLineNo">039</span> * This class is a good example of using the SequenceCreatorInterface where during parsing of the stream<a name="line.39"></a>
<span class="sourceLineNo">040</span> * the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.<a name="line.40"></a>
<span class="sourceLineNo">041</span> * This way you can load very large fasta files and store accession id and delay loading the sequence to save<a name="line.41"></a>
<span class="sourceLineNo">042</span> * memory. The index is the file stream offset so when a DNASequence has a call to getSequence() the<a name="line.42"></a>
<span class="sourceLineNo">043</span> * SequenceFileProxyLoader will open the file and offset to the index and retrieve the sequence.<a name="line.43"></a>
<span class="sourceLineNo">044</span> *<a name="line.44"></a>
<span class="sourceLineNo">045</span> * Same approach can be used for genome sequence data stored in a local fasta file, in a database or via http<a name="line.45"></a>
<span class="sourceLineNo">046</span> * interface to a remote server<a name="line.46"></a>
<span class="sourceLineNo">047</span> *<a name="line.47"></a>
<span class="sourceLineNo">048</span> * @author Scooter Willis &lt;willishf at gmail dot com&gt;<a name="line.48"></a>
<span class="sourceLineNo">049</span> */<a name="line.49"></a>
<span class="sourceLineNo">050</span>public class FileProxyDNASequenceCreator implements<a name="line.50"></a>
<span class="sourceLineNo">051</span>                SequenceCreatorInterface&lt;NucleotideCompound&gt; {<a name="line.51"></a>
<span class="sourceLineNo">052</span><a name="line.52"></a>
<span class="sourceLineNo">053</span>        CompoundSet&lt;NucleotideCompound&gt; compoundSet = null;<a name="line.53"></a>
<span class="sourceLineNo">054</span>        File file = null;<a name="line.54"></a>
<span class="sourceLineNo">055</span>        SequenceParserInterface sequenceParser;<a name="line.55"></a>
<span class="sourceLineNo">056</span><a name="line.56"></a>
<span class="sourceLineNo">057</span>        /**<a name="line.57"></a>
<span class="sourceLineNo">058</span>         * Need File so that we can store full path name in SequenceFileProxyLoader for Random File access as a quick read<a name="line.58"></a>
<span class="sourceLineNo">059</span>         * @param fastaFile<a name="line.59"></a>
<span class="sourceLineNo">060</span>         * @param compoundSet<a name="line.60"></a>
<span class="sourceLineNo">061</span>         */<a name="line.61"></a>
<span class="sourceLineNo">062</span>        public FileProxyDNASequenceCreator(File file,<a name="line.62"></a>
<span class="sourceLineNo">063</span>                        CompoundSet&lt;NucleotideCompound&gt; compoundSet,<a name="line.63"></a>
<span class="sourceLineNo">064</span>                        SequenceParserInterface sequenceParser) {<a name="line.64"></a>
<span class="sourceLineNo">065</span>                this.compoundSet = compoundSet;<a name="line.65"></a>
<span class="sourceLineNo">066</span>                this.file = file;<a name="line.66"></a>
<span class="sourceLineNo">067</span>                this.sequenceParser = sequenceParser;<a name="line.67"></a>
<span class="sourceLineNo">068</span>        }<a name="line.68"></a>
<span class="sourceLineNo">069</span><a name="line.69"></a>
<span class="sourceLineNo">070</span>        /**<a name="line.70"></a>
<span class="sourceLineNo">071</span>         * Even though we are passing in the sequence we really only care about the length of the sequence and the offset<a name="line.71"></a>
<span class="sourceLineNo">072</span>         * index in the fasta file.<a name="line.72"></a>
<span class="sourceLineNo">073</span>         * @param sequence<a name="line.73"></a>
<span class="sourceLineNo">074</span>         * @param index<a name="line.74"></a>
<span class="sourceLineNo">075</span>         * @return<a name="line.75"></a>
<span class="sourceLineNo">076</span>         * @throws CompoundNotFoundException<a name="line.76"></a>
<span class="sourceLineNo">077</span>         * @throws IOException<a name="line.77"></a>
<span class="sourceLineNo">078</span>         */<a name="line.78"></a>
<span class="sourceLineNo">079</span>        @Override<a name="line.79"></a>
<span class="sourceLineNo">080</span>        public AbstractSequence&lt;NucleotideCompound&gt; getSequence(String sequence, long index ) throws CompoundNotFoundException, IOException {<a name="line.80"></a>
<span class="sourceLineNo">081</span>                SequenceFileProxyLoader&lt;NucleotideCompound&gt; sequenceFileProxyLoader = new SequenceFileProxyLoader&lt;NucleotideCompound&gt;(<a name="line.81"></a>
<span class="sourceLineNo">082</span>                                file,<a name="line.82"></a>
<span class="sourceLineNo">083</span>                                sequenceParser,<a name="line.83"></a>
<span class="sourceLineNo">084</span>                                index,<a name="line.84"></a>
<span class="sourceLineNo">085</span>                                sequence.length(),<a name="line.85"></a>
<span class="sourceLineNo">086</span>                                compoundSet);<a name="line.86"></a>
<span class="sourceLineNo">087</span>                return new DNASequence(sequenceFileProxyLoader, compoundSet);<a name="line.87"></a>
<span class="sourceLineNo">088</span>        }<a name="line.88"></a>
<span class="sourceLineNo">089</span><a name="line.89"></a>
<span class="sourceLineNo">090</span>        /**<a name="line.90"></a>
<span class="sourceLineNo">091</span>         * Should be able to extend the same concept to a remote URL call or database connection. Not supported yet<a name="line.91"></a>
<span class="sourceLineNo">092</span>         * @param proxyLoader<a name="line.92"></a>
<span class="sourceLineNo">093</span>         * @param index<a name="line.93"></a>
<span class="sourceLineNo">094</span>         * @return<a name="line.94"></a>
<span class="sourceLineNo">095</span>         */<a name="line.95"></a>
<span class="sourceLineNo">096</span>        @Override<a name="line.96"></a>
<span class="sourceLineNo">097</span>        public AbstractSequence&lt;NucleotideCompound&gt; getSequence(<a name="line.97"></a>
<span class="sourceLineNo">098</span>                        ProxySequenceReader&lt;NucleotideCompound&gt; proxyLoader, long index) {<a name="line.98"></a>
<span class="sourceLineNo">099</span>                throw new UnsupportedOperationException("Not supported yet.");<a name="line.99"></a>
<span class="sourceLineNo">100</span>        }<a name="line.100"></a>
<span class="sourceLineNo">101</span><a name="line.101"></a>
<span class="sourceLineNo">102</span>        /**<a name="line.102"></a>
<span class="sourceLineNo">103</span>         * Not sure of use case and currently not supported<a name="line.103"></a>
<span class="sourceLineNo">104</span>         * @param list<a name="line.104"></a>
<span class="sourceLineNo">105</span>         * @return<a name="line.105"></a>
<span class="sourceLineNo">106</span>         */<a name="line.106"></a>
<span class="sourceLineNo">107</span>        @Override<a name="line.107"></a>
<span class="sourceLineNo">108</span>        public AbstractSequence&lt;NucleotideCompound&gt; getSequence(<a name="line.108"></a>
<span class="sourceLineNo">109</span>                        List&lt;NucleotideCompound&gt; list) {<a name="line.109"></a>
<span class="sourceLineNo">110</span>                throw new UnsupportedOperationException("Not supported yet.");<a name="line.110"></a>
<span class="sourceLineNo">111</span>        }<a name="line.111"></a>
<span class="sourceLineNo">112</span>}<a name="line.112"></a>




























































</pre>
</div>
</body>
</html>
