<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_261) on Mon Nov 09 18:21:05 CST 2020 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.bio.seq.db.SequenceDB (biojava-legacy 1.9.4 API)</title>
<meta name="date" content="2020-11-09">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava.bio.seq.db.SequenceDB (biojava-legacy 1.9.4 API)";
        }
    }
    catch(err) {
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/bio/seq/db/class-use/SequenceDB.html" target="_top">Frames</a></li>
<li><a href="SequenceDB.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.bio.seq.db.SequenceDB" class="title">Uses of Interface<br>org.biojava.bio.seq.db.SequenceDB</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.alignment">org.biojava.bio.alignment</a></td>
<td class="colLast">
<div class="block">
Classes to generate and describe sequence alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.dp">org.biojava.bio.dp</a></td>
<td class="colLast">
<div class="block">HMM and Dynamic Programming Algorithms.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.gui">org.biojava.bio.gui</a></td>
<td class="colLast">
<div class="block">Graphical interfaces for biojava objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.gff">org.biojava.bio.program.gff</a></td>
<td class="colLast">
<div class="block">GFF manipulation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.phred">org.biojava.bio.program.phred</a></td>
<td class="colLast">
<div class="block">Parser for Phred output</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.ssaha">org.biojava.bio.program.ssaha</a></td>
<td class="colLast">
<div class="block">SSAHA sequence searching API.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.program.ssbind">org.biojava.bio.program.ssbind</a></td>
<td class="colLast">
<div class="block">
Creation of objects from SAX events using the BioJava BlastLikeDataSetCollection DTD.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.program.unigene">org.biojava.bio.program.unigene</a></td>
<td class="colLast">
<div class="block">Objects for representing Unigene clusters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.search">org.biojava.bio.search</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for representing sequence similarity search results.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db">org.biojava.bio.seq.db</a></td>
<td class="colLast">
<div class="block">
Collections of biological sequence data.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.db.biosql">org.biojava.bio.seq.db.biosql</a></td>
<td class="colLast">
<div class="block">General purpose Sequence storage in a relational database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.distributed">org.biojava.bio.seq.distributed</a></td>
<td class="colLast">
<div class="block">Sequences and SequenceDBs which are composed from data taken
from a number of data sources.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.impl">org.biojava.bio.seq.impl</a></td>
<td class="colLast">
<div class="block">Standard in-memory implementations of <code>Sequence</code> and
<code>Feature</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.bio.seq.io">org.biojava.bio.seq.io</a></td>
<td class="colLast">
<div class="block">Classes and interfaces for processing and producing flat-file representations
of sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.db">org.biojavax.bio.db</a></td>
<td class="colLast">
<div class="block">Interactions between biojavax objects and a DB.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojavax.bio.db.biosql">org.biojavax.bio.db.biosql</a></td>
<td class="colLast">
<div class="block">Interface between biojava and biosql databases</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojavax.bio.db.ncbi">org.biojavax.bio.db.ncbi</a></td>
<td class="colLast">
<div class="block">Interfaces to NCBI data.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.bio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/alignment/package-summary.html">org.biojava.bio.alignment</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/alignment/package-summary.html">org.biojava.bio.alignment</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/bio/alignment/AlignmentPair.html" title="class in org.biojava.bio.alignment">AlignmentPair</a>&gt;</code></td>
<td class="colLast"><span class="typeNameLabel">AlignmentAlgorithm.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/alignment/AlignmentAlgorithm.html#alignAll-org.biojava.bio.seq.SequenceIterator-org.biojava.bio.seq.db.SequenceDB-">alignAll</a></span>(<a href="../../../../../../org/biojava/bio/seq/SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;source,
        <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;subjectDB)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.dp">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/dp/package-summary.html">org.biojava.bio.dp</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/dp/package-summary.html">org.biojava.bio.dp</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">TrainingAlgorithm.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/dp/TrainingAlgorithm.html#train-org.biojava.bio.seq.db.SequenceDB-double-org.biojava.bio.dp.StoppingCriteria-">train</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
     double&nbsp;nullWeight,
     <a href="../../../../../../org/biojava/bio/dp/StoppingCriteria.html" title="interface in org.biojava.bio.dp">StoppingCriteria</a>&nbsp;stopper)</code>
<div class="block">Trains the sequences in db untill stopper says to finnish.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">AbstractTrainer.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/dp/AbstractTrainer.html#train-org.biojava.bio.seq.db.SequenceDB-double-org.biojava.bio.dp.StoppingCriteria-">train</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
     double&nbsp;nullModelWeight,
     <a href="../../../../../../org/biojava/bio/dp/StoppingCriteria.html" title="interface in org.biojava.bio.dp">StoppingCriteria</a>&nbsp;stopper)</code>
<div class="block">Trains the sequences in db until stopper says to finnish.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/gui/package-summary.html">org.biojava.bio.gui</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/gui/package-summary.html">org.biojava.bio.gui</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">FeatureTree.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/gui/FeatureTree.html#setSequenceDB-org.biojava.bio.seq.db.SequenceDB-">setSequenceDB</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code>
<div class="block">Use this method to provide the sequences for the tree to work with.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.gff">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/program/gff/package-summary.html">org.biojava.bio.program.gff</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/program/gff/package-summary.html">org.biojava.bio.program.gff</a> that return <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">GFFTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/gff/GFFTools.html#annotateSequences-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.program.gff.GFFEntrySet-">annotateSequences</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqs,
                 <a href="../../../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a>&nbsp;ents)</code>
<div class="block">Annotates all sequences in a sequence DB with features from a GFF entry set.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/program/gff/package-summary.html">org.biojava.bio.program.gff</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">GFFTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/gff/GFFTools.html#annotateSequences-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.program.gff.GFFEntrySet-">annotateSequences</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqs,
                 <a href="../../../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a>&nbsp;ents)</code>
<div class="block">Annotates all sequences in a sequence DB with features from a GFF entry set.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/bio/program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<td class="colLast"><span class="typeNameLabel">GFFTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/gff/GFFTools.html#gffFromSeqDB-org.biojava.bio.seq.db.SequenceDB-">gffFromSeqDB</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code>
<div class="block">Creates a GFFEntrySet containing one entry for each feature on each
 sequence of a SequenceDB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">SequencesAsGFF.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/gff/SequencesAsGFF.html#processDB-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.program.gff.GFFDocumentHandler-">processDB</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
         <a href="../../../../../../org/biojava/bio/program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler)</code>
<div class="block">Process all <span class="type">Sequence</span>s within a
 <span class="type">SequenceDB</span>, informing
 <span class="arg">handler</span> of any suitable features.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.phred">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/program/phred/package-summary.html">org.biojava.bio.program.phred</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/program/phred/package-summary.html">org.biojava.bio.program.phred</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">PhredTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/phred/PhredTools.html#writePhredQuality-java.io.OutputStream-java.io.OutputStream-org.biojava.bio.seq.db.SequenceDB-">writePhredQuality</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;qual,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;seq,
                 <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code>
<div class="block">Writes Phred quality data in a Fasta type format.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.ssaha">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="typeNameLabel">NIODataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssaha/NIODataStoreFactory.html#buildDataStore-java.io.File-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.symbol.Packing-int-int-">buildDataStore</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
              <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="typeNameLabel">DataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssaha/DataStoreFactory.html#buildDataStore-java.io.File-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.symbol.Packing-int-int-">buildDataStore</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
              <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code>
<div class="block">Build a new DataStore.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="typeNameLabel">MappedDataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssaha/MappedDataStoreFactory.html#buildDataStore-java.io.File-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.symbol.Packing-int-int-">buildDataStore</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
              <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<td class="colLast"><span class="typeNameLabel">CompactedDataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssaha/CompactedDataStoreFactory.html#buildDataStore-java.io.File-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.symbol.Packing-int-int-">buildDataStore</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;storeFile,
              <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../../../../org/biojava/bio/symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../org/biojava/bio/program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssaha/SequenceStreamer.SequenceDBStreamer.html#SequenceDBStreamer-org.biojava.bio.seq.db.SequenceDB-">SequenceDBStreamer</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.ssbind">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> declared as <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">ViewSequenceFactory.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssbind/ViewSequenceFactory.html#querySeqHolder">querySeqHolder</a></span></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> that return <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">ViewSequenceFactory.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssbind/ViewSequenceFactory.html#getQuerySeqHolder--">getQuerySeqHolder</a></span>()</code>
<div class="block"><code>getQuerySeqHolder</code> returns the database of query
 sequences used to retrieve sequences for creation of the
 various result objects.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">BlastLikeSearchBuilder.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssbind/BlastLikeSearchBuilder.html#setQuerySeqHolder-org.biojava.bio.seq.db.SequenceDB-">setQuerySeqHolder</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;querySeqHolder)</code>
<div class="block"><code>setQuerySeqHolder</code> sets the query sequence holder
 to a specific database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="typeNameLabel">ViewSequenceFactory.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssbind/ViewSequenceFactory.html#setQuerySeqHolder-org.biojava.bio.seq.db.SequenceDB-">setQuerySeqHolder</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;querySeqHolder)</code>
<div class="block"><code>setQuerySeqHolder</code> sets the query sequence holder
 to a specific database.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../org/biojava/bio/program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/ssbind/BlastLikeSearchBuilder.html#BlastLikeSearchBuilder-java.util.List-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.seq.db.SequenceDBInstallation-">BlastLikeSearchBuilder</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;target,
                      <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;querySeqHolder,
                      <a href="../../../../../../org/biojava/bio/seq/db/SequenceDBInstallation.html" title="interface in org.biojava.bio.seq.db">SequenceDBInstallation</a>&nbsp;subjectDBs)</code>
<div class="block">Creates a new <code>BlastLikeSearchBuilder</code> which will
 instantiate results into the <code>List</code> target.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.program.unigene">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/program/unigene/package-summary.html">org.biojava.bio.program.unigene</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/program/unigene/package-summary.html">org.biojava.bio.program.unigene</a> that return <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">UnigeneCluster.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/program/unigene/UnigeneCluster.html#getAll--">getAll</a></span>()</code>
<div class="block">All sequences that map to this cluster.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.search">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a> that return <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">SequenceDBSearchResult.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/search/SequenceDBSearchResult.html#getSequenceDB--">getSequenceDB</a></span>()</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">SimpleSeqSimilaritySearchResult.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/search/SimpleSeqSimilaritySearchResult.html#getSequenceDB--">getSequenceDB</a></span>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">SeqSimilaritySearchResult.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/search/SeqSimilaritySearchResult.html#getSequenceDB--">getSequenceDB</a></span>()</code>
<div class="block">Returns the sequence database against which the search was
 performed.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/search/SeqSimilaritySearchResult.html" title="interface in org.biojava.bio.search">SeqSimilaritySearchResult</a></code></td>
<td class="colLast"><span class="typeNameLabel">SeqSimilaritySearcher.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/search/SeqSimilaritySearcher.html#search-org.biojava.bio.symbol.SymbolList-org.biojava.bio.seq.db.SequenceDB-java.util.Map-">search</a></span>(<a href="../../../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;querySeq,
      <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;searchParameters)</code>
<div class="block">Using this sequence similarity searcher, search with the given
 sequence against the given sequence database.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../org/biojava/bio/search/package-summary.html">org.biojava.bio.search</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/search/SequenceDBSearchResult.html#SequenceDBSearchResult-org.biojava.bio.seq.Sequence-org.biojava.bio.seq.db.SequenceDB-java.util.Map-java.util.List-org.biojava.bio.Annotation-">SequenceDBSearchResult</a></span>(<a href="../../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;querySequence,
                      <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;sequenceDB,
                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;searchParameters,
                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;hits,
                      <a href="../../../../../../org/biojava/bio/Annotation.html" title="interface in org.biojava.bio">Annotation</a>&nbsp;annotation)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
<div class="block">Creates a new <code>SequenceDBSearchResult</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/search/SimpleSeqSimilaritySearchResult.html#SimpleSeqSimilaritySearchResult-org.biojava.bio.seq.Sequence-org.biojava.bio.seq.db.SequenceDB-java.util.Map-java.util.List-org.biojava.bio.Annotation-">SimpleSeqSimilaritySearchResult</a></span>(<a href="../../../../../../org/biojava/bio/seq/Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;querySequence,
                               <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;sequenceDB,
                               <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;searchParameters,
                               <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;hits,
                               <a href="../../../../../../org/biojava/bio/Annotation.html" title="interface in org.biojava.bio">Annotation</a>&nbsp;annotation)</code>
<div class="block">Creates a new <code>SimpleSeqSimilaritySearchResult</code>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> that implement <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/AbstractSequenceDB.html" title="class in org.biojava.bio.seq.db">AbstractSequenceDB</a></span></code>
<div class="block">An abstract implementation of SequenceDB that provides the sequenceIterator
 method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/AnnotatedSequenceDB.html" title="class in org.biojava.bio.seq.db">AnnotatedSequenceDB</a></span></code>
<div class="block">SequenceDB implementation which lazily applies a SequenceAnnotator
 to sequences retrieved from a SequenceDB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/CachingSequenceDB.html" title="class in org.biojava.bio.seq.db">CachingSequenceDB</a></span></code>
<div class="block">SequenceDB implementation that caches the results of another SequenceDB.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/DummySequenceDB.html" title="class in org.biojava.bio.seq.db">DummySequenceDB</a></span></code>
<div class="block"><code>DummySequenceDB</code> is an implementation which contains
 only a <code>DummySequence</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/HashSequenceDB.html" title="class in org.biojava.bio.seq.db">HashSequenceDB</a></span></code>
<div class="block">An implementation of SequenceDB that uses an underlying HashMap to store the
 sequence objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/IndexedSequenceDB.html" title="class in org.biojava.bio.seq.db">IndexedSequenceDB</a></span></code>
<div class="block">
 This class implements SequenceDB on top of a set of sequence files
 and sequence offsets within these files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/SequenceDBWrapper.html" title="class in org.biojava.bio.seq.db">SequenceDBWrapper</a></span></code>
<div class="block">An abstract implementation of SequenceDB that wraps up another database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/SubSequenceDB.html" title="class in org.biojava.bio.seq.db">SubSequenceDB</a></span></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/ViewingSequenceDB.html" title="class in org.biojava.bio.seq.db">ViewingSequenceDB</a></span></code>
<div class="block">SequenceDB implementation that returns new SequenceView instances
 wrapping the sequences in an underlying database.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> that return <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">SequenceDBWrapper.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/SequenceDBWrapper.html#getParent--">getParent</a></span>()</code>
<div class="block">Return the parent SequenceDB.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">AnnotatedSequenceDB.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/AnnotatedSequenceDB.html#getParent--">getParent</a></span>()</code>
<div class="block">Get the original sequenceDB from this annotated sequenceDB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenbankSequenceDB.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/GenbankSequenceDB.html#getSequences-java.util.Set-">getSequences</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list)</code>
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenbankSequenceDB.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/GenbankSequenceDB.html#getSequences-java.util.Set-org.biojava.bio.seq.db.SequenceDB-">getSequences</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list,
            <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;database)</code>
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">GenbankSequenceDB.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/GenbankSequenceDB.html#getSequences-java.util.Set-org.biojava.bio.seq.db.SequenceDB-">getSequences</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;list,
            <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;database)</code>
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../org/biojava/bio/seq/db/package-summary.html">org.biojava.bio.seq.db</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/AnnotatedSequenceDB.html#AnnotatedSequenceDB-org.biojava.bio.seq.db.SequenceDB-org.biojava.bio.seq.SequenceAnnotator-">AnnotatedSequenceDB</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent,
                   <a href="../../../../../../org/biojava/bio/seq/SequenceAnnotator.html" title="interface in org.biojava.bio.seq">SequenceAnnotator</a>&nbsp;a)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/CachingSequenceDB.html#CachingSequenceDB-org.biojava.bio.seq.db.SequenceDB-">CachingSequenceDB</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent)</code>
<div class="block">Create a new CachingSequenceDB that caches the sequences in parent.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/SequenceDBWrapper.html#SequenceDBWrapper-org.biojava.bio.seq.db.SequenceDB-">SequenceDBWrapper</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/SubSequenceDB.html#SubSequenceDB-org.biojava.bio.seq.db.SequenceDB-java.util.Set-">SubSequenceDB</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util">Set</a>&nbsp;ids)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/ViewingSequenceDB.html#ViewingSequenceDB-org.biojava.bio.seq.db.SequenceDB-">ViewingSequenceDB</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent)</code>
<div class="block">Create a new ViewingSequenceDB that views the sequences in parent.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.db.biosql">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/seq/db/biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/bio/seq/db/biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a> that implement <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/db/biosql/BioSQLSequenceDB.html" title="class in org.biojava.bio.seq.db.biosql">BioSQLSequenceDB</a></span></code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;
<div class="block"><span class="deprecationComment">Use hibernate and org.biojavax.bio.db.*</span></div>
</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.distributed">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/seq/distributed/package-summary.html">org.biojava.bio.seq.distributed</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/bio/seq/distributed/package-summary.html">org.biojava.bio.seq.distributed</a> that implement <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/distributed/DistributedSequenceDB.html" title="class in org.biojava.bio.seq.distributed">DistributedSequenceDB</a></span></code>
<div class="block">Sequence database from the meta-DAS system.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../org/biojava/bio/seq/distributed/package-summary.html">org.biojava.bio.seq.distributed</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/distributed/SequenceDBDataSource.html#SequenceDBDataSource-org.biojava.bio.seq.db.SequenceDB-">SequenceDBDataSource</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.impl">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/seq/impl/package-summary.html">org.biojava.bio.seq.impl</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/seq/impl/package-summary.html">org.biojava.bio.seq.impl</a> that return <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">SimpleRemoteFeature.DBResolver.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/impl/SimpleRemoteFeature.DBResolver.html#getSeqDB--">getSeqDB</a></span>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../org/biojava/bio/seq/impl/package-summary.html">org.biojava.bio.seq.impl</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/impl/SimpleRemoteFeature.DBResolver.html#DBResolver-org.biojava.bio.seq.db.SequenceDB-">DBResolver</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.bio.seq.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a> that return <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<td class="colLast"><span class="typeNameLabel">SeqIOTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/io/SeqIOTools.html#readFasta-java.io.InputStream-org.biojava.bio.symbol.Alphabet-">readFasta</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;seqFile,
         <a href="../../../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
<div class="block">Create a sequence database from a fasta file provided as an
 input stream.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="typeNameLabel">SeqIOTools.</span><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/io/SeqIOTools.html#writeFasta-java.io.OutputStream-org.biojava.bio.seq.db.SequenceDB-">writeFasta</a></span>(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;os,
          <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code>
<div class="block"><span class="deprecatedLabel">Deprecated.</span>&nbsp;</div>
<div class="block">Write a sequenceDB to an output stream in fasta format.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../org/biojava/bio/seq/io/package-summary.html">org.biojava.bio.seq.io</a> with parameters of type <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/bio/seq/io/SequenceDBSequenceBuilder.html#SequenceDBSequenceBuilder-org.biojava.bio.seq.db.SequenceDB-int-">SequenceDBSequenceBuilder</a></span>(<a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
                         int&nbsp;mode)</code>
<div class="block">constructor</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subinterfaces, and an explanation">
<caption><span>Subinterfaces of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></span></code>
<div class="block">A database of RichSequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a> that implement <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojavax/bio/db/AbstractRichSequenceDB.html" title="class in org.biojavax.bio.db">AbstractRichSequenceDB</a></span></code>
<div class="block">An abstract implementation of RichSequenceDB that provides the getRichSequenceIterator
 method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojavax/bio/db/HashRichSequenceDB.html" title="class in org.biojavax.bio.db">HashRichSequenceDB</a></span></code>
<div class="block">An implementation of RichSequenceDB that uses an underlying HashMap to store the
 RichSequence objects.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db.biosql">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a> that implement <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojavax/bio/db/biosql/BioSQLRichSequenceDB.html" title="class in org.biojavax.bio.db.biosql">BioSQLRichSequenceDB</a></span></code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojavax.bio.db.ncbi">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../../org/biojavax/bio/db/ncbi/package-summary.html">org.biojavax.bio.db.ncbi</a></h3>
<table class="useSummary" border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojavax/bio/db/ncbi/package-summary.html">org.biojavax.bio.db.ncbi</a> that implement <a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojavax/bio/db/ncbi/GenbankRichSequenceDB.html" title="class in org.biojavax.bio.db.ncbi">GenbankRichSequenceDB</a></span></code>
<div class="block">This class contains functions accessing DNA sequences in Genbank format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojavax/bio/db/ncbi/GenpeptRichSequenceDB.html" title="class in org.biojavax.bio.db.ncbi">GenpeptRichSequenceDB</a></span></code>
<div class="block">This class contains functions accessing Peptide sequences in Genpept format.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../org/biojava/bio/seq/db/SequenceDB.html" title="interface in org.biojava.bio.seq.db">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" type="text/javascript"></script><script type="text/javascript">_uacct = "UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/bio/seq/db/class-use/SequenceDB.html" target="_top">Frames</a></li>
<li><a href="SequenceDB.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2020 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
