<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_65) on Tue Sep 02 20:47:23 CDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>AbstractChromatogram (biojava-legacy 1.9.1 API)</title>
<meta name="date" content="2014-09-02">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AbstractChromatogram (biojava-legacy 1.9.1 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AbstractChromatogram.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev Class</li>
<li><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/chromatogram/AbstractChromatogram.html" target="_top">Frames</a></li>
<li><a href="AbstractChromatogram.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.bio.chromatogram</div>
<h2 title="Class AbstractChromatogram" class="title">Class AbstractChromatogram</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.bio.chromatogram.AbstractChromatogram</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../org/biojava/bio/program/abi/ABIFChromatogram.html" title="class in org.biojava.bio.program.abi">ABIFChromatogram</a>, <a href="../../../../org/biojava/bio/program/scf/SCF.html" title="class in org.biojava.bio.program.scf">SCF</a>, <a href="../../../../org/biojava/bio/chromatogram/SimpleChromatogram.html" title="class in org.biojava.bio.chromatogram">SimpleChromatogram</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.59">AbstractChromatogram</a>
extends <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></pre>
<div class="block">A basic, abstract implementation of <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram"><code>Chromatogram</code></a>.  Provides
 protected setters so that subclasses may set the value of the various
 properties of a chromatogram.

 Chromatograms should be created using <a href="../../../../org/biojava/bio/chromatogram/ChromatogramFactory.html" title="class in org.biojava.bio.chromatogram"><code>ChromatogramFactory</code></a> or a
 parser for a particular file format.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>1.3</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>Rhett Sutphin (<a href="http://genome.uiowa.edu/">UI CBCB</a>), Matthew Pocock</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_org.biojava.bio.chromatogram.Chromatogram">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;org.biojava.bio.chromatogram.<a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></h3>
<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#DNA">DNA</a>, <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#OFFSETS">OFFSETS</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#AbstractChromatogram()">AbstractChromatogram</a></strong>()</code>
<div class="block">Create a new AbstractChromatogram.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#clearTraces()">clearTraces</a></strong>()</code>
<div class="block">Sets the trace data structures to null.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#createImmutableAlignment(java.util.Map)">createImmutableAlignment</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;labelsToSymLists)</code>
<div class="block">A factory method for creating new immutable alignments, particularly
 for use as base call alignments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#createImmutableSymbolList(org.biojava.bio.symbol.Alphabet,%20java.util.List)">createImmutableSymbolList</a></strong>(<a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha,
                                                  <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;syms)</code>
<div class="block">A factory method for creating new symbol lists with a given alphabet.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#getBaseCalls()">getBaseCalls</a></strong>()</code>
<div class="block">Return the total number of base calls.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#getMax()">getMax</a></strong>()</code>
<div class="block">Gets the max intensity from all the traces.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#getMax(org.biojava.bio.symbol.AtomicSymbol)">getMax</a></strong>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)</code>
<div class="block">Gets the max intensity on the trace for the specified nucleotide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#getSequenceLength()">getSequenceLength</a></strong>()</code>
<div class="block">Return the sequence length.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#getSignificantBits()">getSignificantBits</a></strong>()</code>
<div class="block">Return the number of significant bits.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#getTrace(org.biojava.bio.symbol.AtomicSymbol)">getTrace</a></strong>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)</code>
<div class="block">Returns an array containing the intensities of the sampled waveform
  representing the chromatogram trace for base <code>nucleotide</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#getTraceLength()">getTraceLength</a></strong>()</code>
<div class="block">Returns the length of the trace of the Chromatogram.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static int[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverse(int[])">reverse</a></strong>(int[]&nbsp;src)</code>
<div class="block">Utility method for reversing an int[] array.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplement()">reverseComplement</a></strong>()</code>
<div class="block">Returns a new <code>Chromatogram</code> representing the reverse
 complement of this one.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplementBaseCallList(java.lang.Object)">reverseComplementBaseCallList</a></strong>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;label)</code>
<div class="block">Return a symbol list containing the reverse complement of the base call
 data for the given label.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplementBaseCalls()">reverseComplementBaseCalls</a></strong>()</code>
<div class="block">Returns a new base call alignment that is the reverse complement of
 one in this chromatogram.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract <a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html" title="class in org.biojava.bio.chromatogram">AbstractChromatogram</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplementInstance()">reverseComplementInstance</a></strong>()</code>
<div class="block">Returns a new instance of this AbstractChromatogram subclass for use in
 <a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplement()"><code>reverseComplement()</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#setBaseCallAlignment(org.biojava.bio.alignment.Alignment)">setBaseCallAlignment</a></strong>(<a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align)</code>
<div class="block">Provides the list of base calls.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#setBits(int)">setBits</a></strong>(int&nbsp;bits)</code>
<div class="block">Sets the number of significant bits in the trace samples.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#setTrace(org.biojava.bio.symbol.AtomicSymbol,%20int[],%20int)">setTrace</a></strong>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nuc,
                int[]&nbsp;trace,
                int&nbsp;maxVal)</code>
<div class="block">Provides the trace samples for a particular nucleotide.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="AbstractChromatogram()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AbstractChromatogram</h4>
<pre>public&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.76">AbstractChromatogram</a>()</pre>
<div class="block">Create a new AbstractChromatogram.</div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getTrace(org.biojava.bio.symbol.AtomicSymbol)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrace</h4>
<pre>public&nbsp;int[]&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.82">getTrace</a>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)
               throws <a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTrace(org.biojava.bio.symbol.AtomicSymbol)">Chromatogram</a></code></strong></div>
<div class="block">Returns an array containing the intensities of the sampled waveform
  representing the chromatogram trace for base <code>nucleotide</code>.  
  This may be a reference the actual internal representation of the 
  samples, so callers <b>must not modify it</b>.
  <p>
  The resulting array for each nucleotide must be <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTraceLength()"><code>Chromatogram.getTraceLength()</code></a>
  <code>int</code>s long.
  </p></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTrace(org.biojava.bio.symbol.AtomicSymbol)">getTrace</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>nucleotide</code> - the trace to examine.  Must be the symbol for A, C, G, or T
         as provided by <a href="../../../../org/biojava/bio/seq/DNATools.html" title="class in org.biojava.bio.seq"><code>DNATools</code></a></dd>
<dt><span class="strong">Returns:</span></dt><dd>an array of integers representing the values of a particular 
          chromatogram trace.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - if <code>nucleotide</code> isn't in the DNA alphabet</dd></dl>
</li>
</ul>
<a name="getTraceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTraceLength</h4>
<pre>public&nbsp;int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.87">getTraceLength</a>()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTraceLength()">Chromatogram</a></code></strong></div>
<div class="block">Returns the length of the trace of the Chromatogram.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTraceLength()">getTraceLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the number of samples in the trace</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getTrace(org.biojava.bio.symbol.AtomicSymbol)"><code>Chromatogram.getTrace(AtomicSymbol)</code></a></dd></dl>
</li>
</ul>
<a name="getMax()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMax</h4>
<pre>public&nbsp;int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.91">getMax</a>()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax()">Chromatogram</a></code></strong></div>
<div class="block">Gets the max intensity from all the traces.  Must be equivalent
  to the max of calling <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax(org.biojava.bio.symbol.AtomicSymbol)"><code>Chromatogram.getMax(AtomicSymbol)</code></a> on each
  of the four non-ambiguous DNA nucleotides.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax()">getMax</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the max intensity</dd></dl>
</li>
</ul>
<a name="getMax(org.biojava.bio.symbol.AtomicSymbol)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMax</h4>
<pre>public&nbsp;int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.102">getMax</a>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)
           throws <a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax(org.biojava.bio.symbol.AtomicSymbol)">Chromatogram</a></code></strong></div>
<div class="block">Gets the max intensity on the trace for the specified nucleotide.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getMax(org.biojava.bio.symbol.AtomicSymbol)">getMax</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>nucleotide</code> - the trace to examine.  Must be a concrete 
         (non-ambiguous) nucleotide from the DNA alphabet</dd>
<dt><span class="strong">Returns:</span></dt><dd>the max intensity</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - when the nucleotide isn't from the DNA
          alphabet</dd></dl>
</li>
</ul>
<a name="getBaseCalls()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBaseCalls</h4>
<pre>public&nbsp;<a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.111">getBaseCalls</a>()</pre>
<div class="block">Return the total number of base calls.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getBaseCalls()">getBaseCalls</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the total number of base calls</dd></dl>
</li>
</ul>
<a name="getSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLength</h4>
<pre>public&nbsp;int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.118">getSequenceLength</a>()</pre>
<div class="block">Return the sequence length.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getSequenceLength()">getSequenceLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the sequence length</dd></dl>
</li>
</ul>
<a name="getSignificantBits()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSignificantBits</h4>
<pre>public&nbsp;int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.125">getSignificantBits</a>()</pre>
<div class="block">Return the number of significant bits.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getSignificantBits()">getSignificantBits</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the significant bits</dd></dl>
</li>
</ul>
<a name="setBaseCallAlignment(org.biojava.bio.alignment.Alignment)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBaseCallAlignment</h4>
<pre>protected final&nbsp;void&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.161">setBaseCallAlignment</a>(<a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;align)
                                   throws <a href="../../../../org/biojava/bio/symbol/IllegalAlphabetException.html" title="class in org.biojava.bio.symbol">IllegalAlphabetException</a>,
                                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a>,
                                          <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/NoSuchElementException.html?is-external=true" title="class or interface in java.util">NoSuchElementException</a></pre>
<div class="block">Provides the list of base calls.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>align</code> - the base call alignment</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/NoSuchElementException.html?is-external=true" title="class or interface in java.util">NoSuchElementException</a></code> - when align doesn't contain alignments with
         the required DNA and OFFSETS labels</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - the lists in align aren't all the same length.</dd>
<dd><code><a href="../../../../org/biojava/bio/symbol/IllegalAlphabetException.html" title="class in org.biojava.bio.symbol">IllegalAlphabetException</a></code> - if the required lists don't have the
         correct alphabets.  See the documentation of
         <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getBaseCalls()"><code>Chromatogram.getBaseCalls()</code></a> for details.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#getBaseCalls()"><code>Chromatogram.getBaseCalls()</code></a></dd></dl>
</li>
</ul>
<a name="clearTraces()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>clearTraces</h4>
<pre>protected final&nbsp;void&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.186">clearTraces</a>()</pre>
<div class="block">Sets the trace data structures to null.  If a subclass needs to replace
 the traces with new traces of a different length, this method must be
 called first to avoid provoking an IllegalArgumentException from setTrace
 due to a length mismatch.</div>
</li>
</ul>
<a name="setTrace(org.biojava.bio.symbol.AtomicSymbol, int[], int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTrace</h4>
<pre>protected final&nbsp;void&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.204">setTrace</a>(<a href="../../../../org/biojava/bio/symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nuc,
            int[]&nbsp;trace,
            int&nbsp;maxVal)
                       throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a>,
                              <a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block">Provides the trace samples for a particular nucleotide.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>nuc</code> - A DNA nucleotide</dd><dd><code>trace</code> - the trace samples themselves</dd><dd><code>maxVal</code> - the maximum value in the trace array.  If this value
        is <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Integer.html?is-external=true#MIN_VALUE" title="class or interface in java.lang"><code>Integer.MIN_VALUE</code></a>, this method will do a linear
        search of trace to determine the max.</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - when trace.length is different
         from any of the existing (non-null) traces</dd>
<dd><code><a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - when nuc is not a concrete DNA nucleotide</dd></dl>
</li>
</ul>
<a name="setBits(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBits</h4>
<pre>protected final&nbsp;void&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.236">setBits</a>(int&nbsp;bits)
                      throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></pre>
<div class="block">Sets the number of significant bits in the trace samples.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>bits</code> - a non-negative integer indicating the number of
         significant bits in each trace sample</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - when <code>bits</code> is negative</dd></dl>
</li>
</ul>
<a name="reverseComplementInstance()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplementInstance</h4>
<pre>protected abstract&nbsp;<a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html" title="class in org.biojava.bio.chromatogram">AbstractChromatogram</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.248">reverseComplementInstance</a>()</pre>
<div class="block">Returns a new instance of this AbstractChromatogram subclass for use in
 <a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplement()"><code>reverseComplement()</code></a>.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a reverse-complemented AbstractChromatogram</dd></dl>
</li>
</ul>
<a name="reverseComplement()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplement</h4>
<pre>public&nbsp;<a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.250">reverseComplement</a>()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#reverseComplement()">Chromatogram</a></code></strong></div>
<div class="block">Returns a new <code>Chromatogram</code> representing the reverse
 complement of this one.
 <p>
 Implementors should copy the metadata about the chromatogram (i.e., base 
 calls) as is appropriate to their formats.
 </p></div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#reverseComplement()">reverseComplement</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>a new chromatogram that is the reverse complement of this one</dd></dl>
</li>
</ul>
<a name="reverseComplementBaseCalls()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplementBaseCalls</h4>
<pre>protected&nbsp;<a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.293">reverseComplementBaseCalls</a>()</pre>
<div class="block">Returns a new base call alignment that is the reverse complement of
 one in this chromatogram.  This is achieved by calling
 <a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplementBaseCallList(java.lang.Object)"><code>reverseComplementBaseCallList(java.lang.Object)</code></a> for each label in the current
 base call alignment.  When that method returns null, no list will
 appear in reverse complement base call alignment with the null-provoking
 label.  For this reason, subclasses are encouraged to override
 <a href="../../../../org/biojava/bio/chromatogram/AbstractChromatogram.html#reverseComplementBaseCallList(java.lang.Object)"><code>reverseComplementBaseCallList(java.lang.Object)</code></a> to handle any additional per-base
 metadata that they store.
 <p>
 This implementation should be safely inheritable
 for all chromatogram implementations, unless one just doesn't want
 base calls on its reverse complement output.  If this is the case,
 it should override this method to return null.
 </p></div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a new <a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment"><code>Alignment</code></a> that is the reverse complement of the
          one in the current chromatogram</dd></dl>
</li>
</ul>
<a name="reverseComplementBaseCallList(java.lang.Object)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reverseComplementBaseCallList</h4>
<pre>protected&nbsp;<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.321">reverseComplementBaseCallList</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;label)</pre>
<div class="block">Return a symbol list containing the reverse complement of the base call
 data for the given label.  The returned list will be stored in the
 reverse complement's base call alignment under the same label.
 <p>
 Implementation note: subclasses which do not use an <a href="../../../../org/biojava/bio/symbol/IntegerAlphabet.html" title="class in org.biojava.bio.symbol"><code>IntegerAlphabet</code></a> for
 their offsets lists must override this method, at least for the case where
 <code>label == <a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html#OFFSETS"><code>Chromatogram.OFFSETS</code></a></code>.
 </p></div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>label</code> - the label Object</dd>
<dt><span class="strong">Returns:</span></dt><dd>an appropriately reverse-complemented SymbolList, or null if the
          label is unhandled.</dd></dl>
</li>
</ul>
<a name="createImmutableSymbolList(org.biojava.bio.symbol.Alphabet, java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createImmutableSymbolList</h4>
<pre>protected&nbsp;<a href="../../../../org/biojava/bio/symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.362">createImmutableSymbolList</a>(<a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha,
                                   <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&nbsp;syms)
                                        throws <a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a>,
                                               <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/ClassCastException.html?is-external=true" title="class or interface in java.lang">ClassCastException</a></pre>
<div class="block">A factory method for creating new symbol lists with a given alphabet.
 The default implementation should be fine for nearly all cases, but the
 option is given in case a developer wishes to use a more memory
 efficient implementation.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alpha</code> - the <a href="../../../../org/biojava/bio/symbol/Alphabet.html" title="interface in org.biojava.bio.symbol"><code>Alphabet</code></a> for the new list</dd><dd><code>syms</code> - the symbols to put in the new list</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new immutable SymbolList containing all the given symbols using
          the given Alphabet</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/bio/symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - when alpha and syms are incompatible</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/ClassCastException.html?is-external=true" title="class or interface in java.lang">ClassCastException</a></code> - when any object in syms isn't a Symbol</dd></dl>
</li>
</ul>
<a name="createImmutableAlignment(java.util.Map)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>createImmutableAlignment</h4>
<pre>protected&nbsp;<a href="../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.382">createImmutableAlignment</a>(<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&nbsp;labelsToSymLists)
                                      throws <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a>,
                                             <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/ClassCastException.html?is-external=true" title="class or interface in java.lang">ClassCastException</a></pre>
<div class="block">A factory method for creating new immutable alignments, particularly
 for use as base call alignments.  The default implementation should
 be fine for nearly all cases.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>labelsToSymLists</code> - a <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util"><code>Map</code></a> whose keys are desired labels
        for the alignment and whose values are the SymbolLists.
        All the SymbolLists must be the same length.</dd>
<dt><span class="strong">Returns:</span></dt><dd>a new Alignment</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/IllegalArgumentException.html?is-external=true" title="class or interface in java.lang">IllegalArgumentException</a></code> - if the lists aren't all the same length</dd>
<dd><code><a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/ClassCastException.html?is-external=true" title="class or interface in java.lang">ClassCastException</a></code> - if any of the values in the map aren't
         SymbolLists</dd></dl>
</li>
</ul>
<a name="reverse(int[])">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>reverse</h4>
<pre>protected static final&nbsp;int[]&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/AbstractChromatogram.html#line.396">reverse</a>(int[]&nbsp;src)</pre>
<div class="block">Utility method for reversing an int[] array. Visible for subclass use.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>src</code> - the source array</dd>
<dt><span class="strong">Returns:</span></dt><dd>an array of the same length containing all values in src in
     reverse order</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AbstractChromatogram.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev Class</li>
<li><a href="../../../../org/biojava/bio/chromatogram/Chromatogram.html" title="interface in org.biojava.bio.chromatogram"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/bio/chromatogram/AbstractChromatogram.html" target="_top">Frames</a></li>
<li><a href="AbstractChromatogram.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2014 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
