<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on DATE<a name="line.20"></a>
<span class="sourceLineNo">021</span> *<a name="line.21"></a>
<span class="sourceLineNo">022</span> */<a name="line.22"></a>
<span class="sourceLineNo">023</span>package org.biojava.nbio.core.sequence;<a name="line.23"></a>
<span class="sourceLineNo">024</span><a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.template.CompoundSet;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.slf4j.Logger;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.slf4j.LoggerFactory;<a name="line.30"></a>
<span class="sourceLineNo">031</span><a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.ArrayList;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.Collections;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.util.LinkedHashMap;<a name="line.34"></a>
<span class="sourceLineNo">035</span><a name="line.35"></a>
<span class="sourceLineNo">036</span>/**<a name="line.36"></a>
<span class="sourceLineNo">037</span> *<a name="line.37"></a>
<span class="sourceLineNo">038</span> * @author Scooter Willis<a name="line.38"></a>
<span class="sourceLineNo">039</span> */<a name="line.39"></a>
<span class="sourceLineNo">040</span>public class GeneSequence extends DNASequence {<a name="line.40"></a>
<span class="sourceLineNo">041</span><a name="line.41"></a>
<span class="sourceLineNo">042</span>        private final static Logger logger = LoggerFactory.getLogger(GeneSequence.class);<a name="line.42"></a>
<span class="sourceLineNo">043</span><a name="line.43"></a>
<span class="sourceLineNo">044</span>        private final LinkedHashMap&lt;String, TranscriptSequence&gt; transcriptSequenceHashMap = new LinkedHashMap&lt;String, TranscriptSequence&gt;();<a name="line.44"></a>
<span class="sourceLineNo">045</span>        private final LinkedHashMap&lt;String, IntronSequence&gt; intronSequenceHashMap = new LinkedHashMap&lt;String, IntronSequence&gt;();<a name="line.45"></a>
<span class="sourceLineNo">046</span>        private final LinkedHashMap&lt;String, ExonSequence&gt; exonSequenceHashMap = new LinkedHashMap&lt;String, ExonSequence&gt;();<a name="line.46"></a>
<span class="sourceLineNo">047</span>        private final ArrayList&lt;IntronSequence&gt; intronSequenceList = new ArrayList&lt;IntronSequence&gt;();<a name="line.47"></a>
<span class="sourceLineNo">048</span>        private final ArrayList&lt;ExonSequence&gt; exonSequenceList = new ArrayList&lt;ExonSequence&gt;();<a name="line.48"></a>
<span class="sourceLineNo">049</span>        boolean intronAdded = false; // need to deal with the problem that typically introns are not added when validating the list and adding in introns as the regions not included in exons<a name="line.49"></a>
<span class="sourceLineNo">050</span>        private Strand strand = Strand.UNDEFINED;<a name="line.50"></a>
<span class="sourceLineNo">051</span>        private ChromosomeSequence chromosomeSequence;<a name="line.51"></a>
<span class="sourceLineNo">052</span><a name="line.52"></a>
<span class="sourceLineNo">053</span>        /**<a name="line.53"></a>
<span class="sourceLineNo">054</span>         * A class that keeps track of the details of a GeneSequence which is difficult to properly model. Two important concepts that is difficult<a name="line.54"></a>
<span class="sourceLineNo">055</span>         * to make everything flexible but still work. You can have GFF features that only describe Exons or Exons/Introns or CDS regions and one<a name="line.55"></a>
<span class="sourceLineNo">056</span>         * or more Transcriptions. You can have exon sequences but that does not imply transcription to the actual protein.<a name="line.56"></a>
<span class="sourceLineNo">057</span>         *<a name="line.57"></a>
<span class="sourceLineNo">058</span>         * The GeneSequence will keep track of Exons and Introns but to get a Protein sequence you need to start with a<a name="line.58"></a>
<span class="sourceLineNo">059</span>         * TranscriptSequence and then add CDS sequences.<a name="line.59"></a>
<span class="sourceLineNo">060</span>         *<a name="line.60"></a>
<span class="sourceLineNo">061</span>         * This is also a key class in the biojava-3-genome module for reading and writing GFF3 files<a name="line.61"></a>
<span class="sourceLineNo">062</span>         *<a name="line.62"></a>
<span class="sourceLineNo">063</span>         * @param parentDNASequence<a name="line.63"></a>
<span class="sourceLineNo">064</span>         * @param begin<a name="line.64"></a>
<span class="sourceLineNo">065</span>         * @param end inclusive of end<a name="line.65"></a>
<span class="sourceLineNo">066</span>         * @param strand force a gene to have strand and transcription sequence will inherit<a name="line.66"></a>
<span class="sourceLineNo">067</span>         */<a name="line.67"></a>
<span class="sourceLineNo">068</span>        public GeneSequence(ChromosomeSequence parentSequence, int begin, int end, Strand strand) {<a name="line.68"></a>
<span class="sourceLineNo">069</span>                chromosomeSequence = parentSequence;<a name="line.69"></a>
<span class="sourceLineNo">070</span>                setParentSequence(parentSequence);<a name="line.70"></a>
<span class="sourceLineNo">071</span>                setBioBegin(begin);<a name="line.71"></a>
<span class="sourceLineNo">072</span>                setBioEnd(end);<a name="line.72"></a>
<span class="sourceLineNo">073</span>                setStrand(strand);<a name="line.73"></a>
<span class="sourceLineNo">074</span>                this.setCompoundSet(DNACompoundSet.getDNACompoundSet());<a name="line.74"></a>
<span class="sourceLineNo">075</span>        }<a name="line.75"></a>
<span class="sourceLineNo">076</span><a name="line.76"></a>
<span class="sourceLineNo">077</span>        /**<a name="line.77"></a>
<span class="sourceLineNo">078</span>         * The parent ChromosomeSequence which contains the actual DNA sequence data<a name="line.78"></a>
<span class="sourceLineNo">079</span>         * @return Chromosome sequence<a name="line.79"></a>
<span class="sourceLineNo">080</span>         */<a name="line.80"></a>
<span class="sourceLineNo">081</span>        public ChromosomeSequence getParentChromosomeSequence() {<a name="line.81"></a>
<span class="sourceLineNo">082</span>                return chromosomeSequence;<a name="line.82"></a>
<span class="sourceLineNo">083</span>        }<a name="line.83"></a>
<span class="sourceLineNo">084</span><a name="line.84"></a>
<span class="sourceLineNo">085</span>        @Override<a name="line.85"></a>
<span class="sourceLineNo">086</span>        public int getLength() {<a name="line.86"></a>
<span class="sourceLineNo">087</span>                return Math.abs(this.getBioEnd() - this.getBioBegin()) + 1;<a name="line.87"></a>
<span class="sourceLineNo">088</span>        }<a name="line.88"></a>
<span class="sourceLineNo">089</span><a name="line.89"></a>
<span class="sourceLineNo">090</span>        /**<a name="line.90"></a>
<span class="sourceLineNo">091</span>         * Once everything has been added to the gene sequence where you might have added exon sequences only then you<a name="line.91"></a>
<span class="sourceLineNo">092</span>         * can infer the intron sequences and add them. You may also have the case where you only added one or more<a name="line.92"></a>
<span class="sourceLineNo">093</span>         * TranscriptSequences and from that you can infer the exon sequences and intron sequences.<a name="line.93"></a>
<span class="sourceLineNo">094</span>         * Currently not implement<a name="line.94"></a>
<span class="sourceLineNo">095</span>         */<a name="line.95"></a>
<span class="sourceLineNo">096</span>        public void addIntronsUsingExons() throws Exception {<a name="line.96"></a>
<span class="sourceLineNo">097</span>                if (intronAdded) { //going to assume introns are correct<a name="line.97"></a>
<span class="sourceLineNo">098</span>                        return;<a name="line.98"></a>
<span class="sourceLineNo">099</span>                }<a name="line.99"></a>
<span class="sourceLineNo">100</span>                if (exonSequenceList.size() == 0) {<a name="line.100"></a>
<span class="sourceLineNo">101</span>                        return;<a name="line.101"></a>
<span class="sourceLineNo">102</span>                }<a name="line.102"></a>
<span class="sourceLineNo">103</span>                ExonComparator exonComparator = new ExonComparator();<a name="line.103"></a>
<span class="sourceLineNo">104</span>                //sort based on start position and sense;<a name="line.104"></a>
<span class="sourceLineNo">105</span>                Collections.sort(exonSequenceList, exonComparator);<a name="line.105"></a>
<span class="sourceLineNo">106</span>                int shift = -1;<a name="line.106"></a>
<span class="sourceLineNo">107</span>                if (getStrand() == Strand.NEGATIVE) {<a name="line.107"></a>
<span class="sourceLineNo">108</span>                        shift = 1;<a name="line.108"></a>
<span class="sourceLineNo">109</span>                }<a name="line.109"></a>
<span class="sourceLineNo">110</span>                //ExonSequence firstExonSequence = exonSequenceList.get(0);<a name="line.110"></a>
<span class="sourceLineNo">111</span>                int intronIndex = 1;<a name="line.111"></a>
<span class="sourceLineNo">112</span>//       if (firstExonSequence.getBioBegin().intValue() != getBioBegin().intValue()) {<a name="line.112"></a>
<span class="sourceLineNo">113</span>//           this.addIntron(new AccessionID(this.getAccession().getID() + "-" + "intron" + intronIndex), getBioBegin(), firstExonSequence.getBioBegin() + shift);<a name="line.113"></a>
<span class="sourceLineNo">114</span>//           intronIndex++;<a name="line.114"></a>
<span class="sourceLineNo">115</span>//       }<a name="line.115"></a>
<span class="sourceLineNo">116</span>                for (int i = 0; i &lt; exonSequenceList.size() - 1; i++) {<a name="line.116"></a>
<span class="sourceLineNo">117</span>                        ExonSequence exon1 = exonSequenceList.get(i);<a name="line.117"></a>
<span class="sourceLineNo">118</span>                        ExonSequence exon2 = exonSequenceList.get(i + 1);<a name="line.118"></a>
<span class="sourceLineNo">119</span>                        this.addIntron(new AccessionID(this.getAccession().getID() + "-" + "intron" + intronIndex), exon1.getBioEnd() - shift, exon2.getBioBegin() + shift);<a name="line.119"></a>
<span class="sourceLineNo">120</span>                        intronIndex++;<a name="line.120"></a>
<span class="sourceLineNo">121</span>                }<a name="line.121"></a>
<span class="sourceLineNo">122</span><a name="line.122"></a>
<span class="sourceLineNo">123</span>//       ExonSequence lastExonSequence = exonSequenceList.get(exonSequenceList.size() - 1);<a name="line.123"></a>
<span class="sourceLineNo">124</span>//       if (lastExonSequence.getBioEnd().intValue() != getBioEnd().intValue()) {<a name="line.124"></a>
<span class="sourceLineNo">125</span>//           this.addIntron(new AccessionID(this.getAccession().getID() + "-" + "intron" + intronIndex), lastExonSequence.getBioEnd() - shift, getBioEnd());<a name="line.125"></a>
<span class="sourceLineNo">126</span>//           intronIndex++;<a name="line.126"></a>
<span class="sourceLineNo">127</span>//       }<a name="line.127"></a>
<span class="sourceLineNo">128</span><a name="line.128"></a>
<span class="sourceLineNo">129</span>                //    log.severe("Add in support for building introns based on added exons");<a name="line.129"></a>
<span class="sourceLineNo">130</span><a name="line.130"></a>
<span class="sourceLineNo">131</span>        }<a name="line.131"></a>
<span class="sourceLineNo">132</span><a name="line.132"></a>
<span class="sourceLineNo">133</span>        /**<a name="line.133"></a>
<span class="sourceLineNo">134</span>         * A gene should have Strand<a name="line.134"></a>
<span class="sourceLineNo">135</span>         * @return the strand<a name="line.135"></a>
<span class="sourceLineNo">136</span>         */<a name="line.136"></a>
<span class="sourceLineNo">137</span>        public Strand getStrand() {<a name="line.137"></a>
<span class="sourceLineNo">138</span>                return strand;<a name="line.138"></a>
<span class="sourceLineNo">139</span>        }<a name="line.139"></a>
<span class="sourceLineNo">140</span><a name="line.140"></a>
<span class="sourceLineNo">141</span>        /**<a name="line.141"></a>
<span class="sourceLineNo">142</span>         * @param strand the strand to set<a name="line.142"></a>
<span class="sourceLineNo">143</span>         */<a name="line.143"></a>
<span class="sourceLineNo">144</span>        public void setStrand(Strand strand) {<a name="line.144"></a>
<span class="sourceLineNo">145</span>                this.strand = strand;<a name="line.145"></a>
<span class="sourceLineNo">146</span>        }<a name="line.146"></a>
<span class="sourceLineNo">147</span><a name="line.147"></a>
<span class="sourceLineNo">148</span>        /**<a name="line.148"></a>
<span class="sourceLineNo">149</span>         * Get the transcript sequence by accession<a name="line.149"></a>
<span class="sourceLineNo">150</span>         * @param accession<a name="line.150"></a>
<span class="sourceLineNo">151</span>         * @return the transcript<a name="line.151"></a>
<span class="sourceLineNo">152</span>         */<a name="line.152"></a>
<span class="sourceLineNo">153</span>        public TranscriptSequence getTranscript(String accession) {<a name="line.153"></a>
<span class="sourceLineNo">154</span>                return transcriptSequenceHashMap.get(accession);<a name="line.154"></a>
<span class="sourceLineNo">155</span>        }<a name="line.155"></a>
<span class="sourceLineNo">156</span><a name="line.156"></a>
<span class="sourceLineNo">157</span>        /**<a name="line.157"></a>
<span class="sourceLineNo">158</span>         * Get the collection of transcription sequences assigned to this gene<a name="line.158"></a>
<span class="sourceLineNo">159</span>         * @return transcripts<a name="line.159"></a>
<span class="sourceLineNo">160</span>         */<a name="line.160"></a>
<span class="sourceLineNo">161</span>        public LinkedHashMap&lt;String, TranscriptSequence&gt; getTranscripts() {<a name="line.161"></a>
<span class="sourceLineNo">162</span>                return transcriptSequenceHashMap;<a name="line.162"></a>
<span class="sourceLineNo">163</span>        }<a name="line.163"></a>
<span class="sourceLineNo">164</span><a name="line.164"></a>
<span class="sourceLineNo">165</span>        /**<a name="line.165"></a>
<span class="sourceLineNo">166</span>         * Remove the transcript sequence from the gene<a name="line.166"></a>
<span class="sourceLineNo">167</span>         * @param accession<a name="line.167"></a>
<span class="sourceLineNo">168</span>         * @return transcriptsequence<a name="line.168"></a>
<span class="sourceLineNo">169</span>         */<a name="line.169"></a>
<span class="sourceLineNo">170</span>        public TranscriptSequence removeTranscript(String accession) {<a name="line.170"></a>
<span class="sourceLineNo">171</span><a name="line.171"></a>
<span class="sourceLineNo">172</span><a name="line.172"></a>
<span class="sourceLineNo">173</span>                return transcriptSequenceHashMap.remove(accession);<a name="line.173"></a>
<span class="sourceLineNo">174</span>        }<a name="line.174"></a>
<span class="sourceLineNo">175</span><a name="line.175"></a>
<span class="sourceLineNo">176</span>        /**<a name="line.176"></a>
<span class="sourceLineNo">177</span>         * Add a transcription sequence to a gene which describes a ProteinSequence<a name="line.177"></a>
<span class="sourceLineNo">178</span>         * @param accession<a name="line.178"></a>
<span class="sourceLineNo">179</span>         * @param begin<a name="line.179"></a>
<span class="sourceLineNo">180</span>         * @param end<a name="line.180"></a>
<span class="sourceLineNo">181</span>         * @return transcript sequence<a name="line.181"></a>
<span class="sourceLineNo">182</span>         * @throws Exception If the accession id is already used<a name="line.182"></a>
<span class="sourceLineNo">183</span>         */<a name="line.183"></a>
<span class="sourceLineNo">184</span>        public TranscriptSequence addTranscript(AccessionID accession, int begin, int end) throws Exception {<a name="line.184"></a>
<span class="sourceLineNo">185</span>                if (transcriptSequenceHashMap.containsKey(accession.getID())) {<a name="line.185"></a>
<span class="sourceLineNo">186</span>                        throw new Exception("Duplicate accesion id " + accession.getID());<a name="line.186"></a>
<span class="sourceLineNo">187</span>                }<a name="line.187"></a>
<span class="sourceLineNo">188</span>                TranscriptSequence transcriptSequence = new TranscriptSequence(this, begin, end);<a name="line.188"></a>
<span class="sourceLineNo">189</span>                transcriptSequence.setAccession(accession);<a name="line.189"></a>
<span class="sourceLineNo">190</span>                transcriptSequenceHashMap.put(accession.getID(), transcriptSequence);<a name="line.190"></a>
<span class="sourceLineNo">191</span>                return transcriptSequence;<a name="line.191"></a>
<span class="sourceLineNo">192</span>        }<a name="line.192"></a>
<span class="sourceLineNo">193</span><a name="line.193"></a>
<span class="sourceLineNo">194</span>        /**<a name="line.194"></a>
<span class="sourceLineNo">195</span>         * Remove the intron by accession<a name="line.195"></a>
<span class="sourceLineNo">196</span>         * @param accession<a name="line.196"></a>
<span class="sourceLineNo">197</span>         * @return intron sequence<a name="line.197"></a>
<span class="sourceLineNo">198</span>         */<a name="line.198"></a>
<span class="sourceLineNo">199</span>        public IntronSequence removeIntron(String accession) {<a name="line.199"></a>
<span class="sourceLineNo">200</span>                for (IntronSequence intronSequence : intronSequenceList) {<a name="line.200"></a>
<span class="sourceLineNo">201</span>                        if (intronSequence.getAccession().getID().equals(accession)) {<a name="line.201"></a>
<span class="sourceLineNo">202</span>                                intronSequenceList.remove(intronSequence);<a name="line.202"></a>
<span class="sourceLineNo">203</span>                                intronSequenceHashMap.remove(accession);<a name="line.203"></a>
<span class="sourceLineNo">204</span>                                return intronSequence;<a name="line.204"></a>
<span class="sourceLineNo">205</span>                        }<a name="line.205"></a>
<span class="sourceLineNo">206</span>                }<a name="line.206"></a>
<span class="sourceLineNo">207</span>                return null;<a name="line.207"></a>
<span class="sourceLineNo">208</span>        }<a name="line.208"></a>
<span class="sourceLineNo">209</span><a name="line.209"></a>
<span class="sourceLineNo">210</span>        /**<a name="line.210"></a>
<span class="sourceLineNo">211</span>         * Add an Intron Currently used to mark an IntronSequence as a feature<a name="line.211"></a>
<span class="sourceLineNo">212</span>         * @param accession<a name="line.212"></a>
<span class="sourceLineNo">213</span>         * @param begin<a name="line.213"></a>
<span class="sourceLineNo">214</span>         * @param end<a name="line.214"></a>
<span class="sourceLineNo">215</span>         * @return intron sequence<a name="line.215"></a>
<span class="sourceLineNo">216</span>         */<a name="line.216"></a>
<span class="sourceLineNo">217</span>        public IntronSequence addIntron(AccessionID accession, int begin, int end) throws Exception {<a name="line.217"></a>
<span class="sourceLineNo">218</span>                if (intronSequenceHashMap.containsKey(accession.getID())) {<a name="line.218"></a>
<span class="sourceLineNo">219</span>                        throw new Exception("Duplicate accesion id " + accession.getID());<a name="line.219"></a>
<span class="sourceLineNo">220</span>                }<a name="line.220"></a>
<span class="sourceLineNo">221</span>                intronAdded = true;<a name="line.221"></a>
<span class="sourceLineNo">222</span>                IntronSequence intronSequence = new IntronSequence(this, begin, end); // working off the assumption that intron frame is always 0 or doesn't matter and same sense as parent<a name="line.222"></a>
<span class="sourceLineNo">223</span>                intronSequence.setAccession(accession);<a name="line.223"></a>
<span class="sourceLineNo">224</span>                intronSequenceList.add(intronSequence);<a name="line.224"></a>
<span class="sourceLineNo">225</span>                intronSequenceHashMap.put(accession.getID(), intronSequence);<a name="line.225"></a>
<span class="sourceLineNo">226</span>                return intronSequence;<a name="line.226"></a>
<span class="sourceLineNo">227</span>        }<a name="line.227"></a>
<span class="sourceLineNo">228</span><a name="line.228"></a>
<span class="sourceLineNo">229</span>        /**<a name="line.229"></a>
<span class="sourceLineNo">230</span>         * Remove the exon sequence<a name="line.230"></a>
<span class="sourceLineNo">231</span>         * @param accession<a name="line.231"></a>
<span class="sourceLineNo">232</span>         * @return exon sequence<a name="line.232"></a>
<span class="sourceLineNo">233</span>         */<a name="line.233"></a>
<span class="sourceLineNo">234</span>        public ExonSequence removeExon(String accession) {<a name="line.234"></a>
<span class="sourceLineNo">235</span>                for (ExonSequence exonSequence : exonSequenceList) {<a name="line.235"></a>
<span class="sourceLineNo">236</span>                        if (exonSequence.getAccession().getID().equals(accession)) {<a name="line.236"></a>
<span class="sourceLineNo">237</span>                                exonSequenceList.remove(exonSequence);<a name="line.237"></a>
<span class="sourceLineNo">238</span>                                exonSequenceHashMap.remove(accession);<a name="line.238"></a>
<span class="sourceLineNo">239</span>                                // we now have a new gap which creates an intron<a name="line.239"></a>
<span class="sourceLineNo">240</span>                                intronSequenceList.clear();<a name="line.240"></a>
<span class="sourceLineNo">241</span>                                intronSequenceHashMap.clear();<a name="line.241"></a>
<span class="sourceLineNo">242</span>                                intronAdded = false;<a name="line.242"></a>
<span class="sourceLineNo">243</span>                                try{<a name="line.243"></a>
<span class="sourceLineNo">244</span>                                        addIntronsUsingExons();<a name="line.244"></a>
<span class="sourceLineNo">245</span>                                } catch(Exception e){<a name="line.245"></a>
<span class="sourceLineNo">246</span>                                        logger.error("Remove Exon validate() error " + e.getMessage());<a name="line.246"></a>
<span class="sourceLineNo">247</span>                                }<a name="line.247"></a>
<span class="sourceLineNo">248</span>                                return exonSequence;<a name="line.248"></a>
<span class="sourceLineNo">249</span>                        }<a name="line.249"></a>
<span class="sourceLineNo">250</span>                }<a name="line.250"></a>
<span class="sourceLineNo">251</span>                return null;<a name="line.251"></a>
<span class="sourceLineNo">252</span>        }<a name="line.252"></a>
<span class="sourceLineNo">253</span><a name="line.253"></a>
<span class="sourceLineNo">254</span>        /**<a name="line.254"></a>
<span class="sourceLineNo">255</span>         * Add an ExonSequence mainly used to mark as a feature<a name="line.255"></a>
<span class="sourceLineNo">256</span>         * @param accession<a name="line.256"></a>
<span class="sourceLineNo">257</span>         * @param begin<a name="line.257"></a>
<span class="sourceLineNo">258</span>         * @param end<a name="line.258"></a>
<span class="sourceLineNo">259</span>         * @return exon sequence<a name="line.259"></a>
<span class="sourceLineNo">260</span>         */<a name="line.260"></a>
<span class="sourceLineNo">261</span>        public ExonSequence addExon(AccessionID accession, int begin, int end) throws Exception {<a name="line.261"></a>
<span class="sourceLineNo">262</span>                if (exonSequenceHashMap.containsKey(accession.getID())) {<a name="line.262"></a>
<span class="sourceLineNo">263</span>                        throw new Exception("Duplicate accesion id " + accession.getID());<a name="line.263"></a>
<span class="sourceLineNo">264</span>                }<a name="line.264"></a>
<span class="sourceLineNo">265</span><a name="line.265"></a>
<span class="sourceLineNo">266</span>                ExonSequence exonSequence = new ExonSequence(this, begin, end); //sense should be the same as parent<a name="line.266"></a>
<span class="sourceLineNo">267</span>                exonSequence.setAccession(accession);<a name="line.267"></a>
<span class="sourceLineNo">268</span>                exonSequenceList.add(exonSequence);<a name="line.268"></a>
<span class="sourceLineNo">269</span>                exonSequenceHashMap.put(accession.getID(), exonSequence);<a name="line.269"></a>
<span class="sourceLineNo">270</span>                return exonSequence;<a name="line.270"></a>
<span class="sourceLineNo">271</span>        }<a name="line.271"></a>
<span class="sourceLineNo">272</span><a name="line.272"></a>
<span class="sourceLineNo">273</span>        /**<a name="line.273"></a>
<span class="sourceLineNo">274</span>         * Get the exons as an ArrayList<a name="line.274"></a>
<span class="sourceLineNo">275</span>         * @return exons<a name="line.275"></a>
<span class="sourceLineNo">276</span>         */<a name="line.276"></a>
<span class="sourceLineNo">277</span>        public ArrayList&lt;ExonSequence&gt; getExonSequences() {<a name="line.277"></a>
<span class="sourceLineNo">278</span>                return exonSequenceList;<a name="line.278"></a>
<span class="sourceLineNo">279</span>        }<a name="line.279"></a>
<span class="sourceLineNo">280</span><a name="line.280"></a>
<span class="sourceLineNo">281</span>        /**<a name="line.281"></a>
<span class="sourceLineNo">282</span>         * Get the introns as an ArrayList<a name="line.282"></a>
<span class="sourceLineNo">283</span>         * @return introns<a name="line.283"></a>
<span class="sourceLineNo">284</span>         */<a name="line.284"></a>
<span class="sourceLineNo">285</span>        public ArrayList&lt;IntronSequence&gt; getIntronSequences() {<a name="line.285"></a>
<span class="sourceLineNo">286</span>                return intronSequenceList;<a name="line.286"></a>
<span class="sourceLineNo">287</span>        }<a name="line.287"></a>
<span class="sourceLineNo">288</span><a name="line.288"></a>
<span class="sourceLineNo">289</span>        /**<a name="line.289"></a>
<span class="sourceLineNo">290</span>         * Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction<a name="line.290"></a>
<span class="sourceLineNo">291</span>         * Returns the DNASequence representative of the 5' and 3' reading based on strand<a name="line.291"></a>
<span class="sourceLineNo">292</span>         * @return dna sequence<a name="line.292"></a>
<span class="sourceLineNo">293</span>         */<a name="line.293"></a>
<span class="sourceLineNo">294</span>        public DNASequence getSequence5PrimeTo3Prime() {<a name="line.294"></a>
<span class="sourceLineNo">295</span>                String sequence = getSequenceAsString(this.getBioBegin(), this.getBioEnd(), this.getStrand());<a name="line.295"></a>
<span class="sourceLineNo">296</span>                if (getStrand() == Strand.NEGATIVE) {<a name="line.296"></a>
<span class="sourceLineNo">297</span>                        //need to take complement of sequence because it is negative and we are returning the gene sequence from the opposite strand<a name="line.297"></a>
<span class="sourceLineNo">298</span>                        StringBuilder b = new StringBuilder(getLength());<a name="line.298"></a>
<span class="sourceLineNo">299</span>                        CompoundSet&lt;NucleotideCompound&gt; compoundSet = this.getCompoundSet();<a name="line.299"></a>
<span class="sourceLineNo">300</span>                        for (int i = 0; i &lt; sequence.length(); i++) {<a name="line.300"></a>
<span class="sourceLineNo">301</span>                                String nucleotide = sequence.charAt(i) + "";<a name="line.301"></a>
<span class="sourceLineNo">302</span>                                NucleotideCompound nucleotideCompound = compoundSet.getCompoundForString(nucleotide);<a name="line.302"></a>
<span class="sourceLineNo">303</span>                                b.append(nucleotideCompound.getComplement().getShortName());<a name="line.303"></a>
<span class="sourceLineNo">304</span>                        }<a name="line.304"></a>
<span class="sourceLineNo">305</span>                        sequence = b.toString();<a name="line.305"></a>
<span class="sourceLineNo">306</span>                }<a name="line.306"></a>
<span class="sourceLineNo">307</span>                DNASequence dnaSequence = null;<a name="line.307"></a>
<span class="sourceLineNo">308</span>                try {<a name="line.308"></a>
<span class="sourceLineNo">309</span>                        dnaSequence = new DNASequence(sequence.toUpperCase());<a name="line.309"></a>
<span class="sourceLineNo">310</span>                } catch (CompoundNotFoundException e) {<a name="line.310"></a>
<span class="sourceLineNo">311</span>                        // this should not happen, the sequence is DNA originally, if it does, there's a bug somewhere<a name="line.311"></a>
<span class="sourceLineNo">312</span>                        logger.error("Could not create new DNA sequence in getSequence5PrimeTo3Prime(). Error: {}",e.getMessage());<a name="line.312"></a>
<span class="sourceLineNo">313</span>                }<a name="line.313"></a>
<span class="sourceLineNo">314</span>                dnaSequence.setAccession(new AccessionID(this.getAccession().getID()));<a name="line.314"></a>
<span class="sourceLineNo">315</span>                return dnaSequence;<a name="line.315"></a>
<span class="sourceLineNo">316</span>        }<a name="line.316"></a>
<span class="sourceLineNo">317</span>}<a name="line.317"></a>




























































</pre>
</div>
</body>
</html>
