<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on 01-21-2010<a name="line.20"></a>
<span class="sourceLineNo">021</span> */<a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.nbio.core.sequence.transcription;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.core.sequence.RNASequence;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.template.AbstractCompoundTranslator;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.template.CompoundSet;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.sequence.template.Sequence;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.sequence.template.SequenceView;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.nbio.core.sequence.transcription.Table.Codon;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.nbio.core.sequence.views.WindowedSequence;<a name="line.33"></a>
<span class="sourceLineNo">034</span><a name="line.34"></a>
<span class="sourceLineNo">035</span>import java.util.*;<a name="line.35"></a>
<span class="sourceLineNo">036</span><a name="line.36"></a>
<span class="sourceLineNo">037</span>/**<a name="line.37"></a>
<span class="sourceLineNo">038</span> * Takes a {@link Sequence} of {@link NucleotideCompound} which should represent<a name="line.38"></a>
<span class="sourceLineNo">039</span> * an RNA sequence ({@link RNASequence} is good for this) and returns a list of<a name="line.39"></a>
<span class="sourceLineNo">040</span> * {@link Sequence} which hold {@link AminoAcidCompound}. The translator can<a name="line.40"></a>
<span class="sourceLineNo">041</span> * also trim stop codons as well as changing any valid start codon to an<a name="line.41"></a>
<span class="sourceLineNo">042</span> * initiating met.<a name="line.42"></a>
<span class="sourceLineNo">043</span> *<a name="line.43"></a>
<span class="sourceLineNo">044</span> * @author ayates<a name="line.44"></a>
<span class="sourceLineNo">045</span> */<a name="line.45"></a>
<span class="sourceLineNo">046</span>public class RNAToAminoAcidTranslator extends<a name="line.46"></a>
<span class="sourceLineNo">047</span>                AbstractCompoundTranslator&lt;NucleotideCompound, AminoAcidCompound&gt; {<a name="line.47"></a>
<span class="sourceLineNo">048</span><a name="line.48"></a>
<span class="sourceLineNo">049</span>        private final boolean trimStops;<a name="line.49"></a>
<span class="sourceLineNo">050</span>        private final boolean initMetOnly;<a name="line.50"></a>
<span class="sourceLineNo">051</span>        private final Map&lt;Table.CaseInsensitiveTriplet, Codon&gt; quickLookup;<a name="line.51"></a>
<span class="sourceLineNo">052</span>        private final Map&lt;AminoAcidCompound, List&lt;Codon&gt;&gt; aminoAcidToCodon;<a name="line.52"></a>
<span class="sourceLineNo">053</span>        // Cheeky lookup which uses a hashing value; key is to switch to using this<a name="line.53"></a>
<span class="sourceLineNo">054</span>        // all the time<a name="line.54"></a>
<span class="sourceLineNo">055</span>        private final Codon[] codonArray = new Codon[64000];<a name="line.55"></a>
<span class="sourceLineNo">056</span>        private final AminoAcidCompound unknownAminoAcidCompound;<a name="line.56"></a>
<span class="sourceLineNo">057</span>        private final AminoAcidCompound methionineAminoAcidCompound;<a name="line.57"></a>
<span class="sourceLineNo">058</span>        private final boolean translateNCodons;<a name="line.58"></a>
<span class="sourceLineNo">059</span><a name="line.59"></a>
<span class="sourceLineNo">060</span>        // If true, then translation will stop at the first stop codon encountered<a name="line.60"></a>
<span class="sourceLineNo">061</span>        // in the reading frame (the stop codon will be included as the last residue<a name="line.61"></a>
<span class="sourceLineNo">062</span>        // in the resulting ProteinSequence, unless removed by #trimStops)<a name="line.62"></a>
<span class="sourceLineNo">063</span>        private final boolean stopAtStopCodons;<a name="line.63"></a>
<span class="sourceLineNo">064</span><a name="line.64"></a>
<span class="sourceLineNo">065</span>        // If true, then translation will not start until the first start codon<a name="line.65"></a>
<span class="sourceLineNo">066</span>        // encountered in the reading frame. The start codon will be included as the<a name="line.66"></a>
<span class="sourceLineNo">067</span>        // first residue in the resulting ProteinSequence<a name="line.67"></a>
<span class="sourceLineNo">068</span>        private final boolean waitForStartCodon;<a name="line.68"></a>
<span class="sourceLineNo">069</span><a name="line.69"></a>
<span class="sourceLineNo">070</span><a name="line.70"></a>
<span class="sourceLineNo">071</span>        public RNAToAminoAcidTranslator(<a name="line.71"></a>
<span class="sourceLineNo">072</span>                        SequenceCreatorInterface&lt;AminoAcidCompound&gt; creator,<a name="line.72"></a>
<span class="sourceLineNo">073</span>                        CompoundSet&lt;NucleotideCompound&gt; nucleotides,<a name="line.73"></a>
<span class="sourceLineNo">074</span>                        CompoundSet&lt;Codon&gt; codons,<a name="line.74"></a>
<span class="sourceLineNo">075</span>                        CompoundSet&lt;AminoAcidCompound&gt; aminoAcids, Table table,<a name="line.75"></a>
<span class="sourceLineNo">076</span>                        boolean trimStops, boolean initMetOnly, boolean translateNCodons,<a name="line.76"></a>
<span class="sourceLineNo">077</span>                        boolean stopAtStopCodons, boolean waitForStartCodon) {<a name="line.77"></a>
<span class="sourceLineNo">078</span><a name="line.78"></a>
<span class="sourceLineNo">079</span>                super(creator, nucleotides, aminoAcids);<a name="line.79"></a>
<span class="sourceLineNo">080</span>                this.trimStops = trimStops;<a name="line.80"></a>
<span class="sourceLineNo">081</span>                this.initMetOnly = initMetOnly;<a name="line.81"></a>
<span class="sourceLineNo">082</span>                this.translateNCodons = translateNCodons;<a name="line.82"></a>
<span class="sourceLineNo">083</span><a name="line.83"></a>
<span class="sourceLineNo">084</span>                quickLookup = new HashMap&lt;Table.CaseInsensitiveTriplet, Codon&gt;(codons<a name="line.84"></a>
<span class="sourceLineNo">085</span>                                .getAllCompounds().size());<a name="line.85"></a>
<span class="sourceLineNo">086</span>                aminoAcidToCodon = new HashMap&lt;AminoAcidCompound, List&lt;Codon&gt;&gt;();<a name="line.86"></a>
<span class="sourceLineNo">087</span><a name="line.87"></a>
<span class="sourceLineNo">088</span>                List&lt;Codon&gt; codonList = table.getCodons(nucleotides, aminoAcids);<a name="line.88"></a>
<span class="sourceLineNo">089</span>                for (Codon codon : codonList) {<a name="line.89"></a>
<span class="sourceLineNo">090</span>                        quickLookup.put(codon.getTriplet(), codon);<a name="line.90"></a>
<span class="sourceLineNo">091</span>                        codonArray[codon.getTriplet().intValue()] = codon;<a name="line.91"></a>
<span class="sourceLineNo">092</span><a name="line.92"></a>
<span class="sourceLineNo">093</span>                        List&lt;Codon&gt; codonL = aminoAcidToCodon.get(codon.getAminoAcid());<a name="line.93"></a>
<span class="sourceLineNo">094</span>                        if (codonL == null) {<a name="line.94"></a>
<span class="sourceLineNo">095</span>                                codonL = new ArrayList&lt;Codon&gt;();<a name="line.95"></a>
<span class="sourceLineNo">096</span>                                aminoAcidToCodon.put(codon.getAminoAcid(), codonL);<a name="line.96"></a>
<span class="sourceLineNo">097</span>                        }<a name="line.97"></a>
<span class="sourceLineNo">098</span>                        codonL.add(codon);<a name="line.98"></a>
<span class="sourceLineNo">099</span><a name="line.99"></a>
<span class="sourceLineNo">100</span>                }<a name="line.100"></a>
<span class="sourceLineNo">101</span>                unknownAminoAcidCompound = aminoAcids.getCompoundForString("X");<a name="line.101"></a>
<span class="sourceLineNo">102</span>                methionineAminoAcidCompound = aminoAcids.getCompoundForString("M");<a name="line.102"></a>
<span class="sourceLineNo">103</span>                this.stopAtStopCodons = stopAtStopCodons;<a name="line.103"></a>
<span class="sourceLineNo">104</span>                this.waitForStartCodon = waitForStartCodon;<a name="line.104"></a>
<span class="sourceLineNo">105</span>        }<a name="line.105"></a>
<span class="sourceLineNo">106</span><a name="line.106"></a>
<span class="sourceLineNo">107</span>        /**<a name="line.107"></a>
<span class="sourceLineNo">108</span>         * Performs the core conversion of RNA to Peptide. It does this by walking a<a name="line.108"></a>
<span class="sourceLineNo">109</span>         * windowed version of the given sequence. Any trailing DNA base pairs are<a name="line.109"></a>
<span class="sourceLineNo">110</span>         * ignored according to the specification of {@link WindowedSequence}.<a name="line.110"></a>
<span class="sourceLineNo">111</span>         */<a name="line.111"></a>
<span class="sourceLineNo">112</span><a name="line.112"></a>
<span class="sourceLineNo">113</span>        @Override<a name="line.113"></a>
<span class="sourceLineNo">114</span>        public List&lt;Sequence&lt;AminoAcidCompound&gt;&gt; createSequences(<a name="line.114"></a>
<span class="sourceLineNo">115</span>                        Sequence&lt;NucleotideCompound&gt; originalSequence) {<a name="line.115"></a>
<span class="sourceLineNo">116</span><a name="line.116"></a>
<span class="sourceLineNo">117</span>                List&lt;List&lt;AminoAcidCompound&gt;&gt; workingList = new ArrayList&lt;List&lt;AminoAcidCompound&gt;&gt;();<a name="line.117"></a>
<span class="sourceLineNo">118</span><a name="line.118"></a>
<span class="sourceLineNo">119</span>                Iterable&lt;SequenceView&lt;NucleotideCompound&gt;&gt; iter = new WindowedSequence&lt;NucleotideCompound&gt;(<a name="line.119"></a>
<span class="sourceLineNo">120</span>                                originalSequence, 3);<a name="line.120"></a>
<span class="sourceLineNo">121</span><a name="line.121"></a>
<span class="sourceLineNo">122</span>                boolean first = true;<a name="line.122"></a>
<span class="sourceLineNo">123</span><a name="line.123"></a>
<span class="sourceLineNo">124</span>                // If not waiting for a start codon, start translating immediately<a name="line.124"></a>
<span class="sourceLineNo">125</span>                boolean doTranslate = !waitForStartCodon;<a name="line.125"></a>
<span class="sourceLineNo">126</span><a name="line.126"></a>
<span class="sourceLineNo">127</span>                for (SequenceView&lt;NucleotideCompound&gt; element : iter) {<a name="line.127"></a>
<span class="sourceLineNo">128</span>                        AminoAcidCompound aminoAcid = null;<a name="line.128"></a>
<span class="sourceLineNo">129</span><a name="line.129"></a>
<span class="sourceLineNo">130</span>                        int i = 1;<a name="line.130"></a>
<span class="sourceLineNo">131</span>                        Table.CaseInsensitiveTriplet triplet = new Table.CaseInsensitiveTriplet(<a name="line.131"></a>
<span class="sourceLineNo">132</span>                                        element.getCompoundAt(i++), element.getCompoundAt(i++),<a name="line.132"></a>
<span class="sourceLineNo">133</span>                                        element.getCompoundAt(i++));<a name="line.133"></a>
<span class="sourceLineNo">134</span><a name="line.134"></a>
<span class="sourceLineNo">135</span>                        Codon target = null;<a name="line.135"></a>
<span class="sourceLineNo">136</span><a name="line.136"></a>
<span class="sourceLineNo">137</span>                        target = quickLookup.get(triplet);<a name="line.137"></a>
<span class="sourceLineNo">138</span><a name="line.138"></a>
<span class="sourceLineNo">139</span>                        // Check for a start<a name="line.139"></a>
<span class="sourceLineNo">140</span>                        if (!doTranslate &amp;&amp; target.isStart()) {<a name="line.140"></a>
<span class="sourceLineNo">141</span>                                doTranslate = true;<a name="line.141"></a>
<span class="sourceLineNo">142</span>                        }<a name="line.142"></a>
<span class="sourceLineNo">143</span><a name="line.143"></a>
<span class="sourceLineNo">144</span>                        if (doTranslate) {<a name="line.144"></a>
<span class="sourceLineNo">145</span>                                if (target != null)<a name="line.145"></a>
<span class="sourceLineNo">146</span>                                        aminoAcid = target.getAminoAcid();<a name="line.146"></a>
<span class="sourceLineNo">147</span>                                if (aminoAcid == null &amp;&amp; translateNCodons()) {<a name="line.147"></a>
<span class="sourceLineNo">148</span>                                        aminoAcid = unknownAminoAcidCompound;<a name="line.148"></a>
<span class="sourceLineNo">149</span>                                } else {<a name="line.149"></a>
<span class="sourceLineNo">150</span>                                        if (first &amp;&amp; initMetOnly &amp;&amp; target.isStart()) {<a name="line.150"></a>
<span class="sourceLineNo">151</span>                                                aminoAcid = methionineAminoAcidCompound;<a name="line.151"></a>
<span class="sourceLineNo">152</span>                                        }<a name="line.152"></a>
<span class="sourceLineNo">153</span>                                }<a name="line.153"></a>
<span class="sourceLineNo">154</span><a name="line.154"></a>
<span class="sourceLineNo">155</span>                                addCompoundsToList(Arrays.asList(aminoAcid), workingList);<a name="line.155"></a>
<span class="sourceLineNo">156</span>                        }<a name="line.156"></a>
<span class="sourceLineNo">157</span><a name="line.157"></a>
<span class="sourceLineNo">158</span>                        if (doTranslate &amp;&amp; stopAtStopCodons &amp;&amp; target.isStop()) {<a name="line.158"></a>
<span class="sourceLineNo">159</span>                                // Check if we need to stop, but dont stop until started!<a name="line.159"></a>
<span class="sourceLineNo">160</span>                                break;<a name="line.160"></a>
<span class="sourceLineNo">161</span>                        }<a name="line.161"></a>
<span class="sourceLineNo">162</span><a name="line.162"></a>
<span class="sourceLineNo">163</span>                        first = false;<a name="line.163"></a>
<span class="sourceLineNo">164</span>                }<a name="line.164"></a>
<span class="sourceLineNo">165</span>                postProcessCompoundLists(workingList);<a name="line.165"></a>
<span class="sourceLineNo">166</span><a name="line.166"></a>
<span class="sourceLineNo">167</span>                return workingListToSequences(workingList);<a name="line.167"></a>
<span class="sourceLineNo">168</span>        }<a name="line.168"></a>
<span class="sourceLineNo">169</span><a name="line.169"></a>
<span class="sourceLineNo">170</span>        /**<a name="line.170"></a>
<span class="sourceLineNo">171</span>         * Performs the trimming of stop codons and the conversion of a valid start<a name="line.171"></a>
<span class="sourceLineNo">172</span>         * amino acid to M<a name="line.172"></a>
<span class="sourceLineNo">173</span>         */<a name="line.173"></a>
<span class="sourceLineNo">174</span>        @Override<a name="line.174"></a>
<span class="sourceLineNo">175</span>        protected void postProcessCompoundLists(<a name="line.175"></a>
<span class="sourceLineNo">176</span>                        List&lt;List&lt;AminoAcidCompound&gt;&gt; compoundLists) {<a name="line.176"></a>
<span class="sourceLineNo">177</span>                for (List&lt;AminoAcidCompound&gt; compounds : compoundLists) {<a name="line.177"></a>
<span class="sourceLineNo">178</span>                        if (trimStops) {<a name="line.178"></a>
<span class="sourceLineNo">179</span>                                trimStop(compounds);<a name="line.179"></a>
<span class="sourceLineNo">180</span>                        }<a name="line.180"></a>
<span class="sourceLineNo">181</span>                }<a name="line.181"></a>
<span class="sourceLineNo">182</span>        }<a name="line.182"></a>
<span class="sourceLineNo">183</span><a name="line.183"></a>
<span class="sourceLineNo">184</span>        /**<a name="line.184"></a>
<span class="sourceLineNo">185</span>         * Imperfect code. Checks the last amino acid to see if a codon could have<a name="line.185"></a>
<span class="sourceLineNo">186</span>         * translated a stop for it. Left in for the moment<a name="line.186"></a>
<span class="sourceLineNo">187</span>         */<a name="line.187"></a>
<span class="sourceLineNo">188</span>        protected void trimStop(List&lt;AminoAcidCompound&gt; sequence) {<a name="line.188"></a>
<span class="sourceLineNo">189</span>                AminoAcidCompound stop = sequence.get(sequence.size() - 1);<a name="line.189"></a>
<span class="sourceLineNo">190</span>                boolean isStop = false;<a name="line.190"></a>
<span class="sourceLineNo">191</span>                if (aminoAcidToCodon.containsKey(stop)) {<a name="line.191"></a>
<span class="sourceLineNo">192</span>                        for (Codon c : aminoAcidToCodon.get(stop)) {<a name="line.192"></a>
<span class="sourceLineNo">193</span>                                if (c.isStop()) {<a name="line.193"></a>
<span class="sourceLineNo">194</span>                                        isStop = true;<a name="line.194"></a>
<span class="sourceLineNo">195</span>                                        break;<a name="line.195"></a>
<span class="sourceLineNo">196</span>                                }<a name="line.196"></a>
<span class="sourceLineNo">197</span>                        }<a name="line.197"></a>
<span class="sourceLineNo">198</span>                }<a name="line.198"></a>
<span class="sourceLineNo">199</span><a name="line.199"></a>
<span class="sourceLineNo">200</span>                if (isStop) {<a name="line.200"></a>
<span class="sourceLineNo">201</span>                        sequence.remove(sequence.size() - 1);<a name="line.201"></a>
<span class="sourceLineNo">202</span>                }<a name="line.202"></a>
<span class="sourceLineNo">203</span>        }<a name="line.203"></a>
<span class="sourceLineNo">204</span><a name="line.204"></a>
<span class="sourceLineNo">205</span>        /**<a name="line.205"></a>
<span class="sourceLineNo">206</span>         * Indicates if we want to force exact translation of compounds or not i.e.<a name="line.206"></a>
<span class="sourceLineNo">207</span>         * those with internal N RNA bases. This will cause a translation to an X<a name="line.207"></a>
<span class="sourceLineNo">208</span>         * amino acid<a name="line.208"></a>
<span class="sourceLineNo">209</span>         */<a name="line.209"></a>
<span class="sourceLineNo">210</span>        public boolean translateNCodons() {<a name="line.210"></a>
<span class="sourceLineNo">211</span>                return translateNCodons;<a name="line.211"></a>
<span class="sourceLineNo">212</span>        }<a name="line.212"></a>
<span class="sourceLineNo">213</span>}<a name="line.213"></a>




























































</pre>
</div>
</body>
</html>
