<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Uses of Class org.biojava.nbio.core.sequence.template.AbstractSequence (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="use: package: org.biojava.nbio.core.sequence.template, class: AbstractSequence">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Class org.biojava.nbio.core.sequence.template.AbstractSequence" class="title">Uses of Class<br>org.biojava.nbio.core.sequence.template.AbstractSequence</h1>
</div>
<div class="caption"><span>Packages that use <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.alignment.io">org.biojava.nbio.alignment.io</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.features">org.biojava.nbio.core.sequence.features</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.io.template">org.biojava.nbio.core.sequence.io.template</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.location">org.biojava.nbio.core.sequence.location</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></div>
<div class="col-last even-row-color">
<div class="block">FASTQ and variants sequence format I/O.</div>
</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="org.biojava.nbio.alignment.io">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../alignment/io/package-summary.html">org.biojava.nbio.alignment.io</a></h2>
<div class="caption"><span>Methods in <a href="../../../../alignment/io/package-summary.html">org.biojava.nbio.alignment.io</a> that return types with arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;? extends <a href="../AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a>&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StockholmStructure.</span><code><a href="../../../../alignment/io/StockholmStructure.html#getBioSequences()" class="member-name-link">getBioSequences</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">used to retrieve sequences from the structure</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;? extends <a href="../AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">StockholmStructure.</span><code><a href="../../../../alignment/io/StockholmStructure.html#getBioSequences(boolean)" class="member-name-link">getBioSequences</a><wbr>(boolean&nbsp;ignoreCase)</code></div>
<div class="col-last odd-row-color">
<div class="block">Because some database files have incorrectly small letters (e.g.</div>
</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;? extends <a href="../AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a>&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">StockholmStructure.</span><code><a href="../../../../alignment/io/StockholmStructure.html#getBioSequences(boolean,java.lang.String)" class="member-name-link">getBioSequences</a><wbr>(boolean&nbsp;ignoreCase,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;forcedSequenceType)</code></div>
<div class="col-last even-row-color">
<div class="block">This function tolerates mixed case letters, and allows for forcing the output biosequence type (PFAM/RFAM).</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a></h2>
<div class="caption"><span>Subclasses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../BasicSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>&lt;C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../CDSSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></code></div>
<div class="col-last odd-row-color">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../ChromosomeSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></code></div>
<div class="col-last even-row-color">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../DNASequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-last odd-row-color">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../ExonSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></code></div>
<div class="col-last even-row-color">
<div class="block">A gene contains a collection of Exon sequences</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../GeneSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../IntronSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../ProteinSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></code></div>
<div class="col-last odd-row-color">
<div class="block">The representation of a ProteinSequence</div>
</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../RNASequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">RNASequence</a></code></div>
<div class="col-last even-row-color">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../StartCodonSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></code></div>
<div class="col-last odd-row-color">
<div class="block">Used to map the start codon feature on a gene</div>
</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../StopCodonSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></code></div>
<div class="col-last even-row-color">
<div class="block">Used to map the stop codon sequence on a gene</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../TranscriptSequence.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></code></div>
<div class="col-last odd-row-color">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../package-summary.html">org.biojava.nbio.core.sequence</a> with parameters of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceComparator.</span><code><a href="../../SequenceComparator.html#compare(org.biojava.nbio.core.sequence.template.AbstractSequence,org.biojava.nbio.core.sequence.template.AbstractSequence)" class="member-name-link">compare</a><wbr>(<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o1,
 <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;o2)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ProteinSequence.</span><code><a href="../../ProteinSequence.html#setParentDNASequence(org.biojava.nbio.core.sequence.template.AbstractSequence,java.lang.Integer,java.lang.Integer)" class="member-name-link">setParentDNASequence</a><wbr>(<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;parentDNASequence,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;begin,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&nbsp;end)</code></div>
<div class="col-last odd-row-color">
<div class="block">A Protein sequence can be stand alone or loaded from a transcript
 sequence.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.features">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../features/package-summary.html">org.biojava.nbio.core.sequence.features</a></h2>
<div class="caption"><span>Classes in <a href="../../features/package-summary.html">org.biojava.nbio.core.sequence.features</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../features/AbstractFeature.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../features/FeatureDbReferenceInfo.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.features">FeatureDbReferenceInfo</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">
<div class="block">It is <a href="../../features/DBReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features"><code>DBReferenceInfo</code></a> which implements <a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features"><code>FeatureInterface</code></a>.</div>
</div>
<div class="col-first even-row-color"><code>interface&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../features/FeatureInterface.html" class="type-name-link" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">Interface class to handle describing arbitrary features.</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../features/QualityFeature.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../features/QuantityFeature.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../features/TextFeature.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.features">TextFeature</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">
<div class="block">A implmentation of AbstractFeature</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../features/package-summary.html">org.biojava.nbio.core.sequence.features</a> that return types with arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../features/FeatureRetriever.html" title="type parameter in FeatureRetriever">C</a>&gt;,<wbr><a href="../../features/FeatureRetriever.html" title="type parameter in FeatureRetriever">C</a>&gt;&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FeatureRetriever.</span><code><a href="../../features/FeatureRetriever.html#getFeatures()" class="member-name-link">getFeatures</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.io">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h2>
<div class="caption"><span>Classes in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/GenbankReader.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenbankReader</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">Use <a href="../../io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> as an example of how to use this class where <a href="../../io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> should be the
 primary class used to read Genbank files</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/GenbankSequenceParser.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenbankSequenceParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/GenericFastaHeaderFormat.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/GenericFastaHeaderParser.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">
<div class="block">The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources

 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface</div>
</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/GenericGenbankHeaderFormat.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/GenericGenbankHeaderParser.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../io/GenericInsdcHeaderFormat.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenericInsdcHeaderFormat</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../io/PlainFastaHeaderParser.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">PlainFastaHeaderParser</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">
<div class="block">The plain fasta header takes everything in the header as a single entity.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that return <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ABITrace.</span><code><a href="../../io/ABITrace.html#getSequence()" class="member-name-link">getSequence</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the original programmatically determined (unedited) sequence as a <code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code>.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">
<div class="block">Assumes all compounds were uppercase</div>
</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CasePreservingProteinSequenceCreator.</span><code><a href="../../io/CasePreservingProteinSequenceCreator.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">DNASequenceCreator.</span><code><a href="../../io/DNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">DNASequenceCreator.</span><code><a href="../../io/DNASequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">DNASequenceCreator.</span><code><a href="../../io/DNASequenceCreator.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyDNASequenceCreator.</span><code><a href="../../io/FileProxyDNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileProxyDNASequenceCreator.</span><code><a href="../../io/FileProxyDNASequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">
<div class="block">Not sure of use case and currently not supported</div>
</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyDNASequenceCreator.</span><code><a href="../../io/FileProxyDNASequenceCreator.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last odd-row-color">
<div class="block">Not sure of use case and currently not supported</div>
</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileProxyProteinSequenceCreator.</span><code><a href="../../io/FileProxyProteinSequenceCreator.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyRNASequenceCreator.</span><code><a href="../../io/FileProxyRNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Even though we are passing in the sequence we really only care about the length of the sequence and the offset
 index in the fasta file.</div>
</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FileProxyRNASequenceCreator.</span><code><a href="../../io/FileProxyRNASequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">
<div class="block">Not sure of use case and currently not supported</div>
</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FileProxyRNASequenceCreator.</span><code><a href="../../io/FileProxyRNASequenceCreator.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Should be able to extend the same concept to a remote URL call or database connection.</div>
</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">ProteinSequenceCreator.</span><code><a href="../../io/ProteinSequenceCreator.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RNASequenceCreator.</span><code><a href="../../io/RNASequenceCreator.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">RNASequenceCreator.</span><code><a href="../../io/RNASequenceCreator.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;list)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">RNASequenceCreator.</span><code><a href="../../io/RNASequenceCreator.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that return types with arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a>&gt;,<wbr><a href="../../io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a>&gt;&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenbankSequenceParser.</span><code><a href="../../io/GenbankSequenceParser.html#getFeatures()" class="member-name-link">getFeatures</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a>&gt;,<wbr><a href="../../io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenbankSequenceParser.</span><code><a href="../../io/GenbankSequenceParser.html#getFeatures(java.lang.String)" class="member-name-link">getFeatures</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;keyword)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenbankSequenceParser.</span><code><a href="../../io/GenbankSequenceParser.html#parseFeatures(org.biojava.nbio.core.sequence.template.AbstractSequence)" class="member-name-link">parseFeatures</a><wbr>(<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../io/GenbankSequenceParser.html" title="type parameter in GenbankSequenceParser">C</a>&gt;&nbsp;sequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Method parameters in <a href="../../io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>protected <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenericInsdcHeaderFormat.</span><code><a href="../../io/GenericInsdcHeaderFormat.html#_write_feature(org.biojava.nbio.core.sequence.features.FeatureInterface,int)" class="member-name-link">_write_feature</a><wbr>(<a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../io/GenericInsdcHeaderFormat.html" title="type parameter in GenericInsdcHeaderFormat">C</a>&gt;,<wbr><a href="../../io/GenericInsdcHeaderFormat.html" title="type parameter in GenericInsdcHeaderFormat">C</a>&gt;&nbsp;feature,
 int&nbsp;record_length)</code></div>
<div class="col-last even-row-color">
<div class="block">Write a single SeqFeature object to features table.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.io.template">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a></h2>
<div class="caption"><span>Methods in <a href="../../io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a> that return <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceCreatorInterface.</span><code><a href="../../io/template/SequenceCreatorInterface.html#getSequence(java.lang.String,long)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;sequence,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SequenceCreatorInterface.</span><code><a href="../../io/template/SequenceCreatorInterface.html#getSequence(java.util.List)" class="member-name-link">getSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;&nbsp;list)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SequenceCreatorInterface.</span><code><a href="../../io/template/SequenceCreatorInterface.html#getSequence(org.biojava.nbio.core.sequence.template.ProxySequenceReader,long)" class="member-name-link">getSequence</a><wbr>(<a href="../ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;<a href="../../io/template/SequenceCreatorInterface.html" title="type parameter in SequenceCreatorInterface">C</a>&gt;&nbsp;proxyLoader,
 long&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.loader">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h2>
<div class="caption"><span>Methods in <a href="../../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that return types with arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;,<wbr><a href="../../loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">GenbankProxySequenceReader.</span><code><a href="../../loader/GenbankProxySequenceReader.html#getFeatures()" class="member-name-link">getFeatures</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../../io/GenericGenbankHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a><wbr>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;,<wbr><a href="../../loader/GenbankProxySequenceReader.html" title="type parameter in GenbankProxySequenceReader">C</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">GenbankProxySequenceReader.</span><code><a href="../../loader/GenbankProxySequenceReader.html#getHeaderParser()" class="member-name-link">getHeaderParser</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.location">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../location/package-summary.html">org.biojava.nbio.core.sequence.location</a></h2>
<div class="caption"><span>Classes in <a href="../../location/package-summary.html">org.biojava.nbio.core.sequence.location</a> with type parameters of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../location/SequenceLocation.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.location">SequenceLocation</a>&lt;S extends <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.template">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a></h2>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> that return <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a><wbr>&lt;?&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#getParentSequence()" class="member-name-link">getParentSequence</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> that return types with arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#getFeatures()" class="member-name-link">getFeatures</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#getFeatures(int)" class="member-name-link">getFeatures</a><wbr>(int&nbsp;bioSequencePosition)</code></div>
<div class="col-last odd-row-color">
<div class="block">Return features at a sequence position</div>
</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#getFeatures(java.lang.String,int)" class="member-name-link">getFeatures</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;featureType,
 int&nbsp;bioSequencePosition)</code></div>
<div class="col-last even-row-color">
<div class="block">Return features at a sequence position by type</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#getFeaturesByType(java.lang.String)" class="member-name-link">getFeaturesByType</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;type)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with parameters of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#setParentSequence(org.biojava.nbio.core.sequence.template.AbstractSequence)" class="member-name-link">setParentSequence</a><wbr>(<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;parentSequence)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Method parameters in <a href="../package-summary.html">org.biojava.nbio.core.sequence.template</a> with type arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#addFeature(int,int,org.biojava.nbio.core.sequence.features.FeatureInterface)" class="member-name-link">addFeature</a><wbr>(int&nbsp;bioStart,
 int&nbsp;bioEnd,
 <a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></div>
<div class="col-last even-row-color">
<div class="block">Method to help set the proper details for a feature as it relates to a sequence
 where the feature needs to have a location on the sequence</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#addFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)" class="member-name-link">addFeature</a><wbr>(<a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></div>
<div class="col-last odd-row-color">
<div class="block">Add a feature to this sequence.</div>
</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">AbstractSequence.</span><code><a href="../AbstractSequence.html#removeFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)" class="member-name-link">removeFeature</a><wbr>(<a href="../../features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<wbr><a href="../AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code></div>
<div class="col-last even-row-color">
<div class="block">Remove a feature from the sequence</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.genome.io.fastq">
<h2>Uses of <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a> in <a href="../../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></h2>
<div class="caption"><span>Methods in <a href="../../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a> that return types with arguments of type <a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature</a><wbr>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<wbr><a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastqTools.</span><code><a href="../../../../genome/io/fastq/FastqTools.html#createErrorProbabilities(org.biojava.nbio.genome.io.fastq.Fastq)" class="member-name-link">createErrorProbabilities</a><wbr>(<a href="../../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></div>
<div class="col-last even-row-color">
<div class="block">Create and return a new <a href="../../features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QuantityFeature</code></a> from the error probabilities of the specified
 FASTQ formatted sequence.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature</a><wbr>&lt;<a href="../AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;,<wbr><a href="../../compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">FastqTools.</span><code><a href="../../../../genome/io/fastq/FastqTools.html#createQualityScores(org.biojava.nbio.genome.io.fastq.Fastq)" class="member-name-link">createQualityScores</a><wbr>(<a href="../../../../genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a>&nbsp;fastq)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create and return a new <a href="../../features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features"><code>QualityFeature</code></a> from the quality scores of the specified
 FASTQ formatted sequence.</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
