<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.aaproperties, class: CommandPrompt, enum: PropertyName">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> */</span>
<span class="source-line-no">021</span><span id="line-21">package org.biojava.nbio.aaproperties;</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">import org.biojava.nbio.aaproperties.xml.AminoAcidCompositionTable;</span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.aaproperties.xml.CaseFreeAminoAcidCompoundSet;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.io.*;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.template.CompoundSet;</span>
<span class="source-line-no">031</span><span id="line-31"></span>
<span class="source-line-no">032</span><span id="line-32">import java.io.File;</span>
<span class="source-line-no">033</span><span id="line-33">import java.io.FileInputStream;</span>
<span class="source-line-no">034</span><span id="line-34">import java.io.IOException;</span>
<span class="source-line-no">035</span><span id="line-35">import java.io.PrintStream;</span>
<span class="source-line-no">036</span><span id="line-36">import java.util.ArrayList;</span>
<span class="source-line-no">037</span><span id="line-37">import java.util.LinkedHashMap;</span>
<span class="source-line-no">038</span><span id="line-38">import java.util.List;</span>
<span class="source-line-no">039</span><span id="line-39">import java.util.Map;</span>
<span class="source-line-no">040</span><span id="line-40">import java.util.Map.Entry;</span>
<span class="source-line-no">041</span><span id="line-41"></span>
<span class="source-line-no">042</span><span id="line-42"></span>
<span class="source-line-no">043</span><span id="line-43">public class CommandPrompt {</span>
<span class="source-line-no">044</span><span id="line-44"></span>
<span class="source-line-no">045</span><span id="line-45">        /**</span>
<span class="source-line-no">046</span><span id="line-46">         * The main method</span>
<span class="source-line-no">047</span><span id="line-47">         * @param args</span>
<span class="source-line-no">048</span><span id="line-48">         *      See showHelp for a list of available arguments</span>
<span class="source-line-no">049</span><span id="line-49">         * @throws Exception</span>
<span class="source-line-no">050</span><span id="line-50">         *  To handle exception thrown by reading of XML files</span>
<span class="source-line-no">051</span><span id="line-51">         */</span>
<span class="source-line-no">052</span><span id="line-52">        public static void main(String[] args) throws Exception{</span>
<span class="source-line-no">053</span><span id="line-53">                run(args);</span>
<span class="source-line-no">054</span><span id="line-54">        }</span>
<span class="source-line-no">055</span><span id="line-55"></span>
<span class="source-line-no">056</span><span id="line-56">        private static AminoAcidCompositionTable checkForValidityAndObtainAATable(String inputLocation, int propertyListSize, String aminoAcidCompositionLocation,</span>
<span class="source-line-no">057</span><span id="line-57">                        String elementMassLocation) throws Exception{</span>
<span class="source-line-no">058</span><span id="line-58">                if(inputLocation == null) {</span>
<span class="source-line-no">059</span><span id="line-59">                        showHelp();</span>
<span class="source-line-no">060</span><span id="line-60">                        throw new Error("Please do provide location of input file.");</span>
<span class="source-line-no">061</span><span id="line-61">                }</span>
<span class="source-line-no">062</span><span id="line-62">                if(propertyListSize == 0){</span>
<span class="source-line-no">063</span><span id="line-63">                        showHelp();</span>
<span class="source-line-no">064</span><span id="line-64">                        throw new Error("Please at least specify a property to compute.");</span>
<span class="source-line-no">065</span><span id="line-65">                }</span>
<span class="source-line-no">066</span><span id="line-66">                AminoAcidCompositionTable aaTable = null;</span>
<span class="source-line-no">067</span><span id="line-67">                if(aminoAcidCompositionLocation != null &amp;&amp; elementMassLocation == null){</span>
<span class="source-line-no">068</span><span id="line-68">                        aaTable = PeptideProperties.obtainAminoAcidCompositionTable(new File(aminoAcidCompositionLocation));</span>
<span class="source-line-no">069</span><span id="line-69">                }else if(aminoAcidCompositionLocation != null &amp;&amp; elementMassLocation != null){</span>
<span class="source-line-no">070</span><span id="line-70">                        aaTable = PeptideProperties.obtainAminoAcidCompositionTable(new File(aminoAcidCompositionLocation, elementMassLocation));</span>
<span class="source-line-no">071</span><span id="line-71">                }else if(aminoAcidCompositionLocation == null &amp;&amp; elementMassLocation != null){</span>
<span class="source-line-no">072</span><span id="line-72">                        throw new Error("You have define the location of Element Mass XML file. Please also define the location of Amino Acid Composition XML file");</span>
<span class="source-line-no">073</span><span id="line-73">                }</span>
<span class="source-line-no">074</span><span id="line-74">                return aaTable;</span>
<span class="source-line-no">075</span><span id="line-75">        }</span>
<span class="source-line-no">076</span><span id="line-76"></span>
<span class="source-line-no">077</span><span id="line-77">        private static void readInputAndGenerateOutput(String outputLocation, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList,</span>
<span class="source-line-no">078</span><span id="line-78">                        String delimiter, String inputLocation, AminoAcidCompositionTable aaTable, int decimalPlace) throws Exception{</span>
<span class="source-line-no">079</span><span id="line-79">                PrintStream output;</span>
<span class="source-line-no">080</span><span id="line-80">                if(outputLocation != null)</span>
<span class="source-line-no">081</span><span id="line-81">                        output = new PrintStream(new File(outputLocation));</span>
<span class="source-line-no">082</span><span id="line-82">                else</span>
<span class="source-line-no">083</span><span id="line-83">                        output = System.out;</span>
<span class="source-line-no">084</span><span id="line-84">                printHeader(output, propertyList, specificList, delimiter);</span>
<span class="source-line-no">085</span><span id="line-85">                Map&lt;String, ProteinSequence&gt; a = readInputFile(inputLocation, aaTable);</span>
<span class="source-line-no">086</span><span id="line-86">                //Need for the last sequence</span>
<span class="source-line-no">087</span><span id="line-87">                for(Entry&lt;String, ProteinSequence&gt; entry:a.entrySet()){</span>
<span class="source-line-no">088</span><span id="line-88">                        compute(output, entry.getValue().getOriginalHeader(), entry.getValue().getSequenceAsString().trim(), delimiter, aaTable, propertyList, specificList,</span>
<span class="source-line-no">089</span><span id="line-89">                                        decimalPlace);</span>
<span class="source-line-no">090</span><span id="line-90">                }</span>
<span class="source-line-no">091</span><span id="line-91">                output.close();</span>
<span class="source-line-no">092</span><span id="line-92">        }</span>
<span class="source-line-no">093</span><span id="line-93"></span>
<span class="source-line-no">094</span><span id="line-94">        public static void run(String[] args) throws Exception{</span>
<span class="source-line-no">095</span><span id="line-95">                /*</span>
<span class="source-line-no">096</span><span id="line-96">                 * Parse input arguments</span>
<span class="source-line-no">097</span><span id="line-97">                 */</span>
<span class="source-line-no">098</span><span id="line-98">                List&lt;Character&gt; propertyList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">099</span><span id="line-99">                List&lt;Character&gt; specificList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">100</span><span id="line-100">                String inputLocation = null;</span>
<span class="source-line-no">101</span><span id="line-101">                String outputLocation = null;</span>
<span class="source-line-no">102</span><span id="line-102">                String aminoAcidCompositionLocation = null;</span>
<span class="source-line-no">103</span><span id="line-103">                String elementMassLocation = null;</span>
<span class="source-line-no">104</span><span id="line-104">                String delimiter = ",";</span>
<span class="source-line-no">105</span><span id="line-105">                int decimalPlace = 4;</span>
<span class="source-line-no">106</span><span id="line-106"></span>
<span class="source-line-no">107</span><span id="line-107">                for(int i = 0; i &lt; args.length; i++){</span>
<span class="source-line-no">108</span><span id="line-108">                        if(args[i].charAt(0) != '-' || args[i].length() != 2){</span>
<span class="source-line-no">109</span><span id="line-109">                                showHelp();</span>
<span class="source-line-no">110</span><span id="line-110">                                throw new Error("Unknown option: " + args[i]);</span>
<span class="source-line-no">111</span><span id="line-111">                        }else{</span>
<span class="source-line-no">112</span><span id="line-112">                                switch(args[i].charAt(1)){</span>
<span class="source-line-no">113</span><span id="line-113">                                //Required</span>
<span class="source-line-no">114</span><span id="line-114">                                case 'i': inputLocation = args[++i]; break;</span>
<span class="source-line-no">115</span><span id="line-115">                                //Optional</span>
<span class="source-line-no">116</span><span id="line-116">                                case 'o': outputLocation = args[++i]; break;</span>
<span class="source-line-no">117</span><span id="line-117">                                case 'f':</span>
<span class="source-line-no">118</span><span id="line-118">                                        i++;</span>
<span class="source-line-no">119</span><span id="line-119">                                        if("csv".equalsIgnoreCase(args[i])) delimiter = ",";</span>
<span class="source-line-no">120</span><span id="line-120">                                        else if("tsv".equalsIgnoreCase(args[i])) delimiter = "\t";</span>
<span class="source-line-no">121</span><span id="line-121">                                        else throw new Error("Invalid value for -f: " + args[i] + ". Please choose either csv or tsv only.");</span>
<span class="source-line-no">122</span><span id="line-122">                                        break;</span>
<span class="source-line-no">123</span><span id="line-123">                                case 'x': aminoAcidCompositionLocation = args[++i]; break;</span>
<span class="source-line-no">124</span><span id="line-124">                                case 'y': elementMassLocation = args[++i]; break;</span>
<span class="source-line-no">125</span><span id="line-125">                                case 'd': decimalPlace = Integer.parseInt(args[++i]); break;</span>
<span class="source-line-no">126</span><span id="line-126">                                //Properties</span>
<span class="source-line-no">127</span><span id="line-127">                                case 'a':</span>
<span class="source-line-no">128</span><span id="line-128">                                        propertyList.add('1');</span>
<span class="source-line-no">129</span><span id="line-129">                                        propertyList.add('2');</span>
<span class="source-line-no">130</span><span id="line-130">                                        propertyList.add('3');</span>
<span class="source-line-no">131</span><span id="line-131">                                        propertyList.add('4');</span>
<span class="source-line-no">132</span><span id="line-132">                                        propertyList.add('5');</span>
<span class="source-line-no">133</span><span id="line-133">                                        propertyList.add('6');</span>
<span class="source-line-no">134</span><span id="line-134">                                        propertyList.add('7');</span>
<span class="source-line-no">135</span><span id="line-135">                                        propertyList.add('8');</span>
<span class="source-line-no">136</span><span id="line-136">                                        propertyList.add('9');</span>
<span class="source-line-no">137</span><span id="line-137">                                        break;</span>
<span class="source-line-no">138</span><span id="line-138">                                case '1': propertyList.add('1'); break;</span>
<span class="source-line-no">139</span><span id="line-139">                                case '2': propertyList.add('2'); break;</span>
<span class="source-line-no">140</span><span id="line-140">                                case '3': propertyList.add('3'); break;</span>
<span class="source-line-no">141</span><span id="line-141">                                case '4': propertyList.add('4'); break;</span>
<span class="source-line-no">142</span><span id="line-142">                                case '5': propertyList.add('5'); break;</span>
<span class="source-line-no">143</span><span id="line-143">                                case '6': propertyList.add('6'); break;</span>
<span class="source-line-no">144</span><span id="line-144">                                case '7': propertyList.add('7'); break;</span>
<span class="source-line-no">145</span><span id="line-145">                                case '8': propertyList.add('8'); break;</span>
<span class="source-line-no">146</span><span id="line-146">                                case '9': propertyList.add('9'); break;</span>
<span class="source-line-no">147</span><span id="line-147">                                case '0':</span>
<span class="source-line-no">148</span><span id="line-148">                                        propertyList.add('0');</span>
<span class="source-line-no">149</span><span id="line-149">                                        i++;</span>
<span class="source-line-no">150</span><span id="line-150">                                        if(args[i].length() != 1) throw new Error("Invalid value: " + args[i] + ". Amino Acid Symbol should be of single character");</span>
<span class="source-line-no">151</span><span id="line-151">                                        specificList.add(args[i].toUpperCase().charAt(0));</span>
<span class="source-line-no">152</span><span id="line-152">                                        break;</span>
<span class="source-line-no">153</span><span id="line-153">                                default:</span>
<span class="source-line-no">154</span><span id="line-154">                                        showHelp();</span>
<span class="source-line-no">155</span><span id="line-155">                                        throw new Error("Unknown option: " + args[i]);</span>
<span class="source-line-no">156</span><span id="line-156">                                }</span>
<span class="source-line-no">157</span><span id="line-157">                        }</span>
<span class="source-line-no">158</span><span id="line-158">                }</span>
<span class="source-line-no">159</span><span id="line-159"></span>
<span class="source-line-no">160</span><span id="line-160">                /*</span>
<span class="source-line-no">161</span><span id="line-161">                 * Check for validity of input arguments</span>
<span class="source-line-no">162</span><span id="line-162">                 */</span>
<span class="source-line-no">163</span><span id="line-163">                AminoAcidCompositionTable aaTable = checkForValidityAndObtainAATable(inputLocation, propertyList.size(), aminoAcidCompositionLocation,</span>
<span class="source-line-no">164</span><span id="line-164">                                elementMassLocation);</span>
<span class="source-line-no">165</span><span id="line-165"></span>
<span class="source-line-no">166</span><span id="line-166">                /*</span>
<span class="source-line-no">167</span><span id="line-167">                 * Read input file and generate output</span>
<span class="source-line-no">168</span><span id="line-168">                 */</span>
<span class="source-line-no">169</span><span id="line-169">                readInputAndGenerateOutput(outputLocation, propertyList, specificList, delimiter, inputLocation, aaTable, decimalPlace);</span>
<span class="source-line-no">170</span><span id="line-170">        }</span>
<span class="source-line-no">171</span><span id="line-171"></span>
<span class="source-line-no">172</span><span id="line-172">        private static Map&lt;String, ProteinSequence&gt; readInputFile(String inputLocation, AminoAcidCompositionTable aaTable) throws Exception{</span>
<span class="source-line-no">173</span><span id="line-173">                FileInputStream inStream = new FileInputStream(inputLocation);</span>
<span class="source-line-no">174</span><span id="line-174">                CompoundSet&lt;AminoAcidCompound&gt;  set;</span>
<span class="source-line-no">175</span><span id="line-175">                if(aaTable == null){</span>
<span class="source-line-no">176</span><span id="line-176">                        set = CaseFreeAminoAcidCompoundSet.getAminoAcidCompoundSet();</span>
<span class="source-line-no">177</span><span id="line-177">                }else{</span>
<span class="source-line-no">178</span><span id="line-178">                        set = aaTable.getAminoAcidCompoundSet();</span>
<span class="source-line-no">179</span><span id="line-179">                }</span>
<span class="source-line-no">180</span><span id="line-180">                Map&lt;String, ProteinSequence&gt; ret;</span>
<span class="source-line-no">181</span><span id="line-181">                if ( inputLocation.toLowerCase().contains(".gb")) {</span>
<span class="source-line-no">182</span><span id="line-182">                        GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; genbankReader = new GenbankReader&lt;&gt;(</span>
<span class="source-line-no">183</span><span id="line-183">                                        inStream, new GenericGenbankHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</span>
<span class="source-line-no">184</span><span id="line-184">                                        new ProteinSequenceCreator(set));</span>
<span class="source-line-no">185</span><span id="line-185">                        ret = genbankReader.process();</span>
<span class="source-line-no">186</span><span id="line-186"></span>
<span class="source-line-no">187</span><span id="line-187"></span>
<span class="source-line-no">188</span><span id="line-188">                } else {</span>
<span class="source-line-no">189</span><span id="line-189">                        FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;&gt;(</span>
<span class="source-line-no">190</span><span id="line-190">                                        inStream, new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</span>
<span class="source-line-no">191</span><span id="line-191">                                        new ProteinSequenceCreator(set));</span>
<span class="source-line-no">192</span><span id="line-192">                        ret = fastaReader.process();</span>
<span class="source-line-no">193</span><span id="line-193"></span>
<span class="source-line-no">194</span><span id="line-194">                }</span>
<span class="source-line-no">195</span><span id="line-195">                return ret;</span>
<span class="source-line-no">196</span><span id="line-196">        }</span>
<span class="source-line-no">197</span><span id="line-197"></span>
<span class="source-line-no">198</span><span id="line-198">        public enum PropertyName{MolecularWeight, Absorbance_True, Absorbance_False, ExtinctionCoefficient_True, ExtinctionCoefficient_False,</span>
<span class="source-line-no">199</span><span id="line-199">                InstabilityIndex, ApliphaticIndex, AverageHydropathyValue, IsoelectricPoint, NetCharge_pH_7, A, R,</span>
<span class="source-line-no">200</span><span id="line-200">                N, D, C, E, Q, G, H, I, L,</span>
<span class="source-line-no">201</span><span id="line-201">                K, M, F, P, S, T, W, Y, V};</span>
<span class="source-line-no">202</span><span id="line-202"></span>
<span class="source-line-no">203</span><span id="line-203">        private static void printHeader(PrintStream output, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList, String delimiter) throws IOException{</span>
<span class="source-line-no">204</span><span id="line-204">                int specificCount = 0;</span>
<span class="source-line-no">205</span><span id="line-205">                /*</span>
<span class="source-line-no">206</span><span id="line-206">                 * 1 Molecular weight</span>
<span class="source-line-no">207</span><span id="line-207">                 * 2 Absorbance (assumed Cys reduced and assume Cys to form cystines)</span>
<span class="source-line-no">208</span><span id="line-208">                 * 3 Extinction coefficient (assumed Cys reduced and assume Cys to form cystines)</span>
<span class="source-line-no">209</span><span id="line-209">                 * 4 Instability index</span>
<span class="source-line-no">210</span><span id="line-210">                 * 5 Apliphatic index</span>
<span class="source-line-no">211</span><span id="line-211">                 * 6 Average hydropathy value</span>
<span class="source-line-no">212</span><span id="line-212">                 * 7 Isoelectric point</span>
<span class="source-line-no">213</span><span id="line-213">                 * 8 Net charge at pH 7</span>
<span class="source-line-no">214</span><span id="line-214">                 * 9 Composition of the 20 standard amino acid</span>
<span class="source-line-no">215</span><span id="line-215">                 * 0 Composition of the specific amino acid</span>
<span class="source-line-no">216</span><span id="line-216">                 */</span>
<span class="source-line-no">217</span><span id="line-217">                List&lt;String&gt; sList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">218</span><span id="line-218">                sList.add("SequenceName");</span>
<span class="source-line-no">219</span><span id="line-219">                for(Character c:propertyList){</span>
<span class="source-line-no">220</span><span id="line-220">                        switch(c){</span>
<span class="source-line-no">221</span><span id="line-221">                        case '1': sList.add(PropertyName.MolecularWeight.toString()); break;</span>
<span class="source-line-no">222</span><span id="line-222">                        case '2': sList.add(PropertyName.Absorbance_True.toString()); sList.add(PropertyName.Absorbance_False.toString()); break;</span>
<span class="source-line-no">223</span><span id="line-223">                        case '3': sList.add(PropertyName.ExtinctionCoefficient_True.toString()); sList.add(PropertyName.ExtinctionCoefficient_False.toString()); break;</span>
<span class="source-line-no">224</span><span id="line-224">                        case '4': sList.add(PropertyName.InstabilityIndex.toString()); break;</span>
<span class="source-line-no">225</span><span id="line-225">                        case '5': sList.add(PropertyName.ApliphaticIndex.toString()); break;</span>
<span class="source-line-no">226</span><span id="line-226">                        case '6': sList.add(PropertyName.AverageHydropathyValue.toString()); break;</span>
<span class="source-line-no">227</span><span id="line-227">                        case '7': sList.add(PropertyName.IsoelectricPoint.toString()); break;</span>
<span class="source-line-no">228</span><span id="line-228">                        case '8': sList.add(PropertyName.NetCharge_pH_7.toString()); break;</span>
<span class="source-line-no">229</span><span id="line-229">                        case '9':</span>
<span class="source-line-no">230</span><span id="line-230">                                sList.add(PropertyName.A.toString()); sList.add(PropertyName.R.toString());</span>
<span class="source-line-no">231</span><span id="line-231">                                sList.add(PropertyName.N.toString()); sList.add(PropertyName.D.toString());</span>
<span class="source-line-no">232</span><span id="line-232">                                sList.add(PropertyName.C.toString()); sList.add(PropertyName.E.toString());</span>
<span class="source-line-no">233</span><span id="line-233">                                sList.add(PropertyName.Q.toString()); sList.add(PropertyName.G.toString());</span>
<span class="source-line-no">234</span><span id="line-234">                                sList.add(PropertyName.H.toString()); sList.add(PropertyName.I.toString());</span>
<span class="source-line-no">235</span><span id="line-235">                                sList.add(PropertyName.L.toString()); sList.add(PropertyName.K.toString());</span>
<span class="source-line-no">236</span><span id="line-236">                                sList.add(PropertyName.M.toString()); sList.add(PropertyName.F.toString());</span>
<span class="source-line-no">237</span><span id="line-237">                                sList.add(PropertyName.P.toString()); sList.add(PropertyName.S.toString());</span>
<span class="source-line-no">238</span><span id="line-238">                                sList.add(PropertyName.T.toString()); sList.add(PropertyName.W.toString());</span>
<span class="source-line-no">239</span><span id="line-239">                                sList.add(PropertyName.Y.toString()); sList.add(PropertyName.V.toString());</span>
<span class="source-line-no">240</span><span id="line-240">                                break;</span>
<span class="source-line-no">241</span><span id="line-241">                        case '0': sList.add("" + specificList.get(specificCount++)); break;</span>
<span class="source-line-no">242</span><span id="line-242">                        }</span>
<span class="source-line-no">243</span><span id="line-243">                }</span>
<span class="source-line-no">244</span><span id="line-244">                for(int i = 0; i &lt; sList.size(); i++){</span>
<span class="source-line-no">245</span><span id="line-245">                        if(i != 0) output.print(delimiter);</span>
<span class="source-line-no">246</span><span id="line-246">                        output.print(sList.get(i));</span>
<span class="source-line-no">247</span><span id="line-247">                }</span>
<span class="source-line-no">248</span><span id="line-248">                output.println();</span>
<span class="source-line-no">249</span><span id="line-249">                output.flush();</span>
<span class="source-line-no">250</span><span id="line-250">        }</span>
<span class="source-line-no">251</span><span id="line-251"></span>
<span class="source-line-no">252</span><span id="line-252">        private static void compute(PrintStream output, String header, String sequence, String delimiter,</span>
<span class="source-line-no">253</span><span id="line-253">                        AminoAcidCompositionTable aaTable, List&lt;Character&gt; propertyList, List&lt;Character&gt; specificList, int decimalPlace) throws CompoundNotFoundException{</span>
<span class="source-line-no">254</span><span id="line-254">                /*</span>
<span class="source-line-no">255</span><span id="line-255">                 * 1 Molecular weight</span>
<span class="source-line-no">256</span><span id="line-256">                 * 2 Absorbance (assumed Cys reduced and assume Cys to form cystines)</span>
<span class="source-line-no">257</span><span id="line-257">                 * 3 Extinction coefficient</span>
<span class="source-line-no">258</span><span id="line-258">                 * 4 Instability index</span>
<span class="source-line-no">259</span><span id="line-259">                 * 5 Apliphatic index</span>
<span class="source-line-no">260</span><span id="line-260">                 * 6 Average hydropathy value</span>
<span class="source-line-no">261</span><span id="line-261">                 * 7 Isoelectric point</span>
<span class="source-line-no">262</span><span id="line-262">                 * 8 Net charge at pH 7</span>
<span class="source-line-no">263</span><span id="line-263">                 * 9 Composition of the 20 standard amino acid</span>
<span class="source-line-no">264</span><span id="line-264">                 * 0 Composition of the specific amino acid</span>
<span class="source-line-no">265</span><span id="line-265">                 */</span>
<span class="source-line-no">266</span><span id="line-266">                ProteinSequence pSequence;</span>
<span class="source-line-no">267</span><span id="line-267">                CompoundSet&lt;AminoAcidCompound&gt; aaSet;</span>
<span class="source-line-no">268</span><span id="line-268">                if(aaTable != null){</span>
<span class="source-line-no">269</span><span id="line-269">                        sequence = Utils.checkSequence(sequence, aaTable.getSymbolSet());</span>
<span class="source-line-no">270</span><span id="line-270">                        pSequence = new ProteinSequence(sequence, aaTable.getAminoAcidCompoundSet());</span>
<span class="source-line-no">271</span><span id="line-271">                        aaSet = aaTable.getAminoAcidCompoundSet();</span>
<span class="source-line-no">272</span><span id="line-272">                }else{</span>
<span class="source-line-no">273</span><span id="line-273">                        sequence = Utils.checkSequence(sequence);</span>
<span class="source-line-no">274</span><span id="line-274">                        pSequence = new ProteinSequence(sequence);</span>
<span class="source-line-no">275</span><span id="line-275">                        aaSet = AminoAcidCompoundSet.getAminoAcidCompoundSet();</span>
<span class="source-line-no">276</span><span id="line-276">                }</span>
<span class="source-line-no">277</span><span id="line-277">                IPeptideProperties pp = new PeptidePropertiesImpl();</span>
<span class="source-line-no">278</span><span id="line-278"></span>
<span class="source-line-no">279</span><span id="line-279">                int specificCount = 0;</span>
<span class="source-line-no">280</span><span id="line-280">                List&lt;Double&gt; dList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">281</span><span id="line-281">                for(Character c:propertyList){</span>
<span class="source-line-no">282</span><span id="line-282">                        switch(c){</span>
<span class="source-line-no">283</span><span id="line-283">                        case '1':</span>
<span class="source-line-no">284</span><span id="line-284">                                if(aaTable == null)</span>
<span class="source-line-no">285</span><span id="line-285">                                        dList.add(pp.getMolecularWeight(pSequence));</span>
<span class="source-line-no">286</span><span id="line-286">                                else</span>
<span class="source-line-no">287</span><span id="line-287">                                        dList.add(pp.getMolecularWeight(pSequence));</span>
<span class="source-line-no">288</span><span id="line-288">                                break;</span>
<span class="source-line-no">289</span><span id="line-289">                        case '2':</span>
<span class="source-line-no">290</span><span id="line-290">                                dList.add(pp.getAbsorbance(pSequence, true));</span>
<span class="source-line-no">291</span><span id="line-291">                                dList.add(pp.getAbsorbance(pSequence, false));</span>
<span class="source-line-no">292</span><span id="line-292">                                break;</span>
<span class="source-line-no">293</span><span id="line-293">                        case '3':</span>
<span class="source-line-no">294</span><span id="line-294">                                dList.add(pp.getExtinctionCoefficient(pSequence, true));</span>
<span class="source-line-no">295</span><span id="line-295">                                dList.add(pp.getExtinctionCoefficient(pSequence, false));</span>
<span class="source-line-no">296</span><span id="line-296">                                break;</span>
<span class="source-line-no">297</span><span id="line-297">                        case '4': dList.add(pp.getInstabilityIndex(pSequence)); break;</span>
<span class="source-line-no">298</span><span id="line-298">                        case '5': dList.add(pp.getApliphaticIndex(pSequence)); break;</span>
<span class="source-line-no">299</span><span id="line-299">                        case '6': dList.add(pp.getAvgHydropathy(pSequence)); break;</span>
<span class="source-line-no">300</span><span id="line-300">                        case '7': dList.add(pp.getIsoelectricPoint(pSequence)); break;</span>
<span class="source-line-no">301</span><span id="line-301">                        case '8': dList.add(pp.getNetCharge(pSequence)); break;</span>
<span class="source-line-no">302</span><span id="line-302">                        case '9':</span>
<span class="source-line-no">303</span><span id="line-303">                                Map&lt;AminoAcidCompound, Double&gt; aaCompound2Double = pp.getAAComposition(pSequence);</span>
<span class="source-line-no">304</span><span id="line-304">                                //(A, R, N, D, C, E, Q, G, H, I, L, K, M, F, P, S, T, W, Y, V)</span>
<span class="source-line-no">305</span><span id="line-305">                                dList.add(aaCompound2Double.get(Constraints.A));</span>
<span class="source-line-no">306</span><span id="line-306">                                dList.add(aaCompound2Double.get(Constraints.R));</span>
<span class="source-line-no">307</span><span id="line-307">                                dList.add(aaCompound2Double.get(Constraints.N));</span>
<span class="source-line-no">308</span><span id="line-308">                                dList.add(aaCompound2Double.get(Constraints.D));</span>
<span class="source-line-no">309</span><span id="line-309">                                dList.add(aaCompound2Double.get(Constraints.C));</span>
<span class="source-line-no">310</span><span id="line-310">                                dList.add(aaCompound2Double.get(Constraints.E));</span>
<span class="source-line-no">311</span><span id="line-311">                                dList.add(aaCompound2Double.get(Constraints.Q));</span>
<span class="source-line-no">312</span><span id="line-312">                                dList.add(aaCompound2Double.get(Constraints.G));</span>
<span class="source-line-no">313</span><span id="line-313">                                dList.add(aaCompound2Double.get(Constraints.H));</span>
<span class="source-line-no">314</span><span id="line-314">                                dList.add(aaCompound2Double.get(Constraints.I));</span>
<span class="source-line-no">315</span><span id="line-315">                                dList.add(aaCompound2Double.get(Constraints.L));</span>
<span class="source-line-no">316</span><span id="line-316">                                dList.add(aaCompound2Double.get(Constraints.K));</span>
<span class="source-line-no">317</span><span id="line-317">                                dList.add(aaCompound2Double.get(Constraints.M));</span>
<span class="source-line-no">318</span><span id="line-318">                                dList.add(aaCompound2Double.get(Constraints.F));</span>
<span class="source-line-no">319</span><span id="line-319">                                dList.add(aaCompound2Double.get(Constraints.P));</span>
<span class="source-line-no">320</span><span id="line-320">                                dList.add(aaCompound2Double.get(Constraints.S));</span>
<span class="source-line-no">321</span><span id="line-321">                                dList.add(aaCompound2Double.get(Constraints.T));</span>
<span class="source-line-no">322</span><span id="line-322">                                dList.add(aaCompound2Double.get(Constraints.W));</span>
<span class="source-line-no">323</span><span id="line-323">                                dList.add(aaCompound2Double.get(Constraints.Y));</span>
<span class="source-line-no">324</span><span id="line-324">                                dList.add(aaCompound2Double.get(Constraints.V));</span>
<span class="source-line-no">325</span><span id="line-325">                                break;</span>
<span class="source-line-no">326</span><span id="line-326">                        case '0': dList.add(pp.getEnrichment(pSequence, aaSet.getCompoundForString("" + specificList.get(specificCount++)))); break;</span>
<span class="source-line-no">327</span><span id="line-327">                        }</span>
<span class="source-line-no">328</span><span id="line-328">                }</span>
<span class="source-line-no">329</span><span id="line-329">                output.print(header.replace(delimiter, "_"));</span>
<span class="source-line-no">330</span><span id="line-330">                dList.stream().forEach(item -&gt; output.print(delimiter + Utils.roundToDecimals(item, decimalPlace)));</span>
<span class="source-line-no">331</span><span id="line-331">                output.println();</span>
<span class="source-line-no">332</span><span id="line-332">                output.flush();</span>
<span class="source-line-no">333</span><span id="line-333">        }</span>
<span class="source-line-no">334</span><span id="line-334"></span>
<span class="source-line-no">335</span><span id="line-335">        private static void showHelp(){</span>
<span class="source-line-no">336</span><span id="line-336">                System.err.println("NAME");</span>
<span class="source-line-no">337</span><span id="line-337">                System.err.println("\tAn executable to generate physico-chemical properties of protein sequences.");</span>
<span class="source-line-no">338</span><span id="line-338">                System.err.println();</span>
<span class="source-line-no">339</span><span id="line-339"></span>
<span class="source-line-no">340</span><span id="line-340">                System.err.println("EXAMPLES");</span>
<span class="source-line-no">341</span><span id="line-341">                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -a");</span>
<span class="source-line-no">342</span><span id="line-342">                System.err.println("\t\tGenerates all possible properties.");</span>
<span class="source-line-no">343</span><span id="line-343">                System.err.println();</span>
<span class="source-line-no">344</span><span id="line-344">                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -1 -3 -7");</span>
<span class="source-line-no">345</span><span id="line-345">                System.err.println("\t\tGenerates only molecular weight, extinction coefficient and isoelectric point.");</span>
<span class="source-line-no">346</span><span id="line-346">                System.err.println();</span>
<span class="source-line-no">347</span><span id="line-347">                System.err.println("\tjava -jar AAProperties.jar -i test.fasta -0 A -0 N -1");</span>
<span class="source-line-no">348</span><span id="line-348">                System.err.println("\t\tGenerates composition of two specific amino acid symbol and molecular weight.");</span>
<span class="source-line-no">349</span><span id="line-349">                System.err.println();</span>
<span class="source-line-no">350</span><span id="line-350"></span>
<span class="source-line-no">351</span><span id="line-351">                System.err.println("OPTIONS");</span>
<span class="source-line-no">352</span><span id="line-352">                System.err.println("\tRequired");</span>
<span class="source-line-no">353</span><span id="line-353">                System.err.println("\t\t-i location of input FASTA file");</span>
<span class="source-line-no">354</span><span id="line-354">                System.err.println();</span>
<span class="source-line-no">355</span><span id="line-355"></span>
<span class="source-line-no">356</span><span id="line-356">                System.err.println("\tOptional");</span>
<span class="source-line-no">357</span><span id="line-357">                System.err.println("\t\t-o location of output file [standard output (default)]");</span>
<span class="source-line-no">358</span><span id="line-358">                System.err.println("\t\t-f output format [csv (default) or tsv]");</span>
<span class="source-line-no">359</span><span id="line-359">                System.err.println("\t\t-x location of Amino Acid Composition XML file for defining amino acid composition");</span>
<span class="source-line-no">360</span><span id="line-360">                System.err.println("\t\t-y location of Element Mass XML file for defining mass of elements");</span>
<span class="source-line-no">361</span><span id="line-361">                System.err.println("\t\t-d number of decimals (int) [4 (default)]");</span>
<span class="source-line-no">362</span><span id="line-362">                System.err.println();</span>
<span class="source-line-no">363</span><span id="line-363"></span>
<span class="source-line-no">364</span><span id="line-364">                System.err.println("\tProvide at least one of them");</span>
<span class="source-line-no">365</span><span id="line-365">                System.err.println("\t\t-a compute properties of option 1-9");</span>
<span class="source-line-no">366</span><span id="line-366">                System.err.println("\t\t-1 compute molecular weight");</span>
<span class="source-line-no">367</span><span id="line-367">                System.err.println("\t\t-2 compute absorbance");</span>
<span class="source-line-no">368</span><span id="line-368">                System.err.println("\t\t-3 compute extinction coefficient");</span>
<span class="source-line-no">369</span><span id="line-369">                System.err.println("\t\t-4 compute instability index");</span>
<span class="source-line-no">370</span><span id="line-370">                System.err.println("\t\t-5 compute apliphatic index");</span>
<span class="source-line-no">371</span><span id="line-371">                System.err.println("\t\t-6 compute average hydropathy value");</span>
<span class="source-line-no">372</span><span id="line-372">                System.err.println("\t\t-7 compute isoelectric point");</span>
<span class="source-line-no">373</span><span id="line-373">                System.err.println("\t\t-8 compute net charge at pH 7");</span>
<span class="source-line-no">374</span><span id="line-374">                System.err.println("\t\t-9 compute composition of 20 standard amino acid (A, R, N, D, C, E, Q, G, H, I, L, K, M, F, P, S, T, W, Y, V)");</span>
<span class="source-line-no">375</span><span id="line-375">                System.err.println("\t\t-0 compute composition of specific amino acid symbol");</span>
<span class="source-line-no">376</span><span id="line-376">                System.err.println();</span>
<span class="source-line-no">377</span><span id="line-377">        }</span>
<span class="source-line-no">378</span><span id="line-378">}</span>




























































</pre>
</div>
</main>
</body>
</html>
