<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:50 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../../index.html?org/biojava/nbio/core/sequence/io/template/class-use/SequenceHeaderParserInterface.html" target="_top">Frames</a></li>
<li><a href="SequenceHeaderParserInterface.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface" class="title">Uses of Interface<br>org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></td>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a> in <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that implement <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources

 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a>&lt;S extends <a href="../../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/PlainFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">PlainFastaHeaderParser</a>&lt;S extends <a href="../../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,C extends <a href="../../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">The plain fasta header takes everything in the header as a single entity.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html#FastaReader(java.io.File,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)">FastaReader</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
           <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">S</a>,<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
           <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code>
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you
 need to use this constructor because we need details about
 the location of the file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html#FastaReader(java.io.InputStream,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)">FastaReader</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;is,
           <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">S</a>,<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
           <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code>
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because we need details about
 local file offsets for quick reads.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html#GenbankReader(java.io.File,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)">GenbankReader</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
             <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">S</a>,<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;headerParser,
             <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;sequenceCreator)</code>
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you
 need to use this constructor because we need details about
 the location of the file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html#GenbankReader(java.io.InputStream,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)">GenbankReader</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;is,
             <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">S</a>,<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;headerParser,
             <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/io/GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;sequenceCreator)</code>
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because we need details about
 local file offsets for quick reads.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a> in <a href="../../../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser(java.io.File,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,%20org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                    <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser(java.io.InputStream,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,%20org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;is,
                    <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../../index.html?org/biojava/nbio/core/sequence/io/template/class-use/SequenceHeaderParserInterface.html" target="_top">Frames</a></li>
<li><a href="SequenceHeaderParserInterface.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
