<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.genome, class: GeneFeatureHelper">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> */</span>
<span class="source-line-no">021</span><span id="line-21">package org.biojava.nbio.genome;</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">import org.biojava.nbio.genome.parsers.gff.*;</span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.core.sequence.*;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.sequence.io.FastaReaderHelper;</span>
<span class="source-line-no">026</span><span id="line-26">import org.slf4j.Logger;</span>
<span class="source-line-no">027</span><span id="line-27">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">028</span><span id="line-28"></span>
<span class="source-line-no">029</span><span id="line-29">import java.io.File;</span>
<span class="source-line-no">030</span><span id="line-30">import java.io.FileWriter;</span>
<span class="source-line-no">031</span><span id="line-31">import java.util.ArrayList;</span>
<span class="source-line-no">032</span><span id="line-32">import java.util.Collection;</span>
<span class="source-line-no">033</span><span id="line-33">import java.util.Collections;</span>
<span class="source-line-no">034</span><span id="line-34">import java.util.LinkedHashMap;</span>
<span class="source-line-no">035</span><span id="line-35">import java.util.Map;</span>
<span class="source-line-no">036</span><span id="line-36"></span>
<span class="source-line-no">037</span><span id="line-37">/**</span>
<span class="source-line-no">038</span><span id="line-38"> *</span>
<span class="source-line-no">039</span><span id="line-39"> * @author Scooter Willis </span>
<span class="source-line-no">040</span><span id="line-40"> */</span>
<span class="source-line-no">041</span><span id="line-41">public class GeneFeatureHelper {</span>
<span class="source-line-no">042</span><span id="line-42"></span>
<span class="source-line-no">043</span><span id="line-43">        private static final Logger logger = LoggerFactory.getLogger(GeneFeatureHelper.class);</span>
<span class="source-line-no">044</span><span id="line-44"></span>
<span class="source-line-no">045</span><span id="line-45">        static public LinkedHashMap&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromUpperCaseExonFastaFile(File fastaSequenceFile, File uppercaseFastaFile, boolean throwExceptionGeneNotFound) throws Exception {</span>
<span class="source-line-no">046</span><span id="line-46">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">047</span><span id="line-47">                Map&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</span>
<span class="source-line-no">048</span><span id="line-48">                for (String accession : dnaSequenceList.keySet()) {</span>
<span class="source-line-no">049</span><span id="line-49">                        DNASequence contigSequence = dnaSequenceList.get(accession);</span>
<span class="source-line-no">050</span><span id="line-50">                        ChromosomeSequence chromsomeSequence = new ChromosomeSequence(contigSequence.getSequenceAsString());</span>
<span class="source-line-no">051</span><span id="line-51">                        chromsomeSequence.setAccession(contigSequence.getAccession());</span>
<span class="source-line-no">052</span><span id="line-52">                        chromosomeSequenceList.put(accession, chromsomeSequence);</span>
<span class="source-line-no">053</span><span id="line-53">                }</span>
<span class="source-line-no">054</span><span id="line-54"></span>
<span class="source-line-no">055</span><span id="line-55"></span>
<span class="source-line-no">056</span><span id="line-56">                Map&lt;String, DNASequence&gt; geneSequenceList = FastaReaderHelper.readFastaDNASequence(uppercaseFastaFile);</span>
<span class="source-line-no">057</span><span id="line-57">                for (DNASequence dnaSequence : geneSequenceList.values()) {</span>
<span class="source-line-no">058</span><span id="line-58">                        String geneSequence = dnaSequence.getSequenceAsString();</span>
<span class="source-line-no">059</span><span id="line-59">                        String lcGeneSequence = geneSequence.toLowerCase();</span>
<span class="source-line-no">060</span><span id="line-60">                        String reverseGeneSequence = dnaSequence.getReverse().getSequenceAsString();</span>
<span class="source-line-no">061</span><span id="line-61">                        String lcReverseGeneSequence = reverseGeneSequence.toLowerCase();</span>
<span class="source-line-no">062</span><span id="line-62">                        Integer bioStart = null;</span>
<span class="source-line-no">063</span><span id="line-63">                        Integer bioEnd = null;</span>
<span class="source-line-no">064</span><span id="line-64">                        Strand strand = Strand.POSITIVE;</span>
<span class="source-line-no">065</span><span id="line-65">                        boolean geneFound = false;</span>
<span class="source-line-no">066</span><span id="line-66">                        String accession = "";</span>
<span class="source-line-no">067</span><span id="line-67">                        DNASequence contigDNASequence = null;</span>
<span class="source-line-no">068</span><span id="line-68">                        for (String id : dnaSequenceList.keySet()) {</span>
<span class="source-line-no">069</span><span id="line-69">                                accession = id;</span>
<span class="source-line-no">070</span><span id="line-70">                                contigDNASequence = dnaSequenceList.get(id);</span>
<span class="source-line-no">071</span><span id="line-71">                                String contigSequence = contigDNASequence.getSequenceAsString().toLowerCase();</span>
<span class="source-line-no">072</span><span id="line-72">                                bioStart = contigSequence.indexOf(lcGeneSequence);</span>
<span class="source-line-no">073</span><span id="line-73">                                if (bioStart != -1) {</span>
<span class="source-line-no">074</span><span id="line-74">                                        bioStart = bioStart + 1;</span>
<span class="source-line-no">075</span><span id="line-75">                                        bioEnd = bioStart + geneSequence.length() - 1;</span>
<span class="source-line-no">076</span><span id="line-76">                                        geneFound = true;</span>
<span class="source-line-no">077</span><span id="line-77">                                        break;</span>
<span class="source-line-no">078</span><span id="line-78">                                } else {</span>
<span class="source-line-no">079</span><span id="line-79">                                        bioStart = contigSequence.indexOf(lcReverseGeneSequence);</span>
<span class="source-line-no">080</span><span id="line-80">                                        if (bioStart != -1) {</span>
<span class="source-line-no">081</span><span id="line-81">                                                bioStart = bioStart + 1;</span>
<span class="source-line-no">082</span><span id="line-82">                                                bioEnd = bioStart - geneSequence.length() - 1;</span>
<span class="source-line-no">083</span><span id="line-83">                                                strand = Strand.NEGATIVE;</span>
<span class="source-line-no">084</span><span id="line-84">                                                geneFound = true;</span>
<span class="source-line-no">085</span><span id="line-85">                                                break;</span>
<span class="source-line-no">086</span><span id="line-86">                                        }</span>
<span class="source-line-no">087</span><span id="line-87">                                }</span>
<span class="source-line-no">088</span><span id="line-88">                        }</span>
<span class="source-line-no">089</span><span id="line-89"></span>
<span class="source-line-no">090</span><span id="line-90">                        if (geneFound) {</span>
<span class="source-line-no">091</span><span id="line-91">                                logger.info("Gene {} found at {} {} {} {}",</span>
<span class="source-line-no">092</span><span id="line-92">                                                dnaSequence.getAccession().toString(), contigDNASequence.getAccession().toString(), bioStart, bioEnd, strand);</span>
<span class="source-line-no">093</span><span id="line-93">                                ChromosomeSequence chromosomeSequence = chromosomeSequenceList.get(accession);</span>
<span class="source-line-no">094</span><span id="line-94"></span>
<span class="source-line-no">095</span><span id="line-95">                                ArrayList&lt;Integer&gt; exonBoundries = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">096</span><span id="line-96"></span>
<span class="source-line-no">097</span><span id="line-97">                                //look for transitions from lowercase to upper case</span>
<span class="source-line-no">098</span><span id="line-98">                                for (int i = 0; i &lt; geneSequence.length(); i++) {</span>
<span class="source-line-no">099</span><span id="line-99">                                        if (i == 0 &amp;&amp; Character.isUpperCase(geneSequence.charAt(i))) {</span>
<span class="source-line-no">100</span><span id="line-100">                                                exonBoundries.add(i);</span>
<span class="source-line-no">101</span><span id="line-101">                                        } else if (i == geneSequence.length() - 1) {</span>
<span class="source-line-no">102</span><span id="line-102">                                                exonBoundries.add(i);</span>
<span class="source-line-no">103</span><span id="line-103">                                        } else if (Character.isUpperCase(geneSequence.charAt(i)) &amp;&amp; Character.isLowerCase(geneSequence.charAt(i - 1))) {</span>
<span class="source-line-no">104</span><span id="line-104">                                                exonBoundries.add(i);</span>
<span class="source-line-no">105</span><span id="line-105">                                        } else if (Character.isUpperCase(geneSequence.charAt(i)) &amp;&amp; Character.isLowerCase(geneSequence.charAt(i + 1))) {</span>
<span class="source-line-no">106</span><span id="line-106">                                                exonBoundries.add(i);</span>
<span class="source-line-no">107</span><span id="line-107">                                        }</span>
<span class="source-line-no">108</span><span id="line-108">                                }</span>
<span class="source-line-no">109</span><span id="line-109">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">110</span><span id="line-110">                                        Collections.reverse(exonBoundries);</span>
<span class="source-line-no">111</span><span id="line-111">                                }</span>
<span class="source-line-no">112</span><span id="line-112"></span>
<span class="source-line-no">113</span><span id="line-113"></span>
<span class="source-line-no">114</span><span id="line-114">                                String geneaccession = dnaSequence.getAccession().getID();</span>
<span class="source-line-no">115</span><span id="line-115">                                String note = geneaccession;</span>
<span class="source-line-no">116</span><span id="line-116">                                String[] values = geneaccession.split(" ");</span>
<span class="source-line-no">117</span><span id="line-117">                                geneaccession = values[0];</span>
<span class="source-line-no">118</span><span id="line-118"></span>
<span class="source-line-no">119</span><span id="line-119"></span>
<span class="source-line-no">120</span><span id="line-120"></span>
<span class="source-line-no">121</span><span id="line-121">                                GeneSequence geneSeq = chromosomeSequence.addGene(new AccessionID(geneaccession), bioStart, bioEnd, strand);</span>
<span class="source-line-no">122</span><span id="line-122">                                geneSeq.addNote(note);</span>
<span class="source-line-no">123</span><span id="line-123">                                geneSeq.setSource(uppercaseFastaFile.getName());</span>
<span class="source-line-no">124</span><span id="line-124">                                //String transcriptName = geneaccession + "-transcript";</span>
<span class="source-line-no">125</span><span id="line-125">                                //TranscriptSequence transcriptSequence = geneSeq.addTranscript(new AccessionID(transcriptName), bioStart, bioEnd);</span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">                                int runningFrameLength = 0;</span>
<span class="source-line-no">128</span><span id="line-128">                                for (int i = 0; i &lt; exonBoundries.size() - 1; i = i + 2) {</span>
<span class="source-line-no">129</span><span id="line-129">                                        int cdsBioStart = exonBoundries.get(i) + bioStart;</span>
<span class="source-line-no">130</span><span id="line-130">                                        int cdsBioEnd = exonBoundries.get(i + 1) + bioStart;</span>
<span class="source-line-no">131</span><span id="line-131">                                        runningFrameLength = runningFrameLength + Math.abs(cdsBioEnd - cdsBioStart) + 1;</span>
<span class="source-line-no">132</span><span id="line-132">                                        //String cdsName = transcriptName + "-cds-" + cdsBioStart + "-" + cdsBioEnd;</span>
<span class="source-line-no">133</span><span id="line-133"></span>
<span class="source-line-no">134</span><span id="line-134">                                        //AccessionID cdsAccessionID = new AccessionID(cdsName);</span>
<span class="source-line-no">135</span><span id="line-135">                                        //ExonSequence exonSequence = geneSeq.addExon(cdsAccessionID, cdsBioStart, cdsBioEnd);</span>
<span class="source-line-no">136</span><span id="line-136">                                        int remainder = runningFrameLength % 3;</span>
<span class="source-line-no">137</span><span id="line-137">                                        //int frame = 0;</span>
<span class="source-line-no">138</span><span id="line-138">                                        if (remainder == 1) {</span>
<span class="source-line-no">139</span><span id="line-139">                                                //frame = 2; // borrow 2 from next CDS region</span>
<span class="source-line-no">140</span><span id="line-140">                                        } else if (remainder == 2) {</span>
<span class="source-line-no">141</span><span id="line-141">                                                //frame = 1;</span>
<span class="source-line-no">142</span><span id="line-142">                                        }</span>
<span class="source-line-no">143</span><span id="line-143">                                        //CDSSequence cdsSequence = transcriptSequence.addCDS(cdsAccessionID, cdsBioStart, cdsBioEnd, frame);</span>
<span class="source-line-no">144</span><span id="line-144">                                }</span>
<span class="source-line-no">145</span><span id="line-145"></span>
<span class="source-line-no">146</span><span id="line-146"></span>
<span class="source-line-no">147</span><span id="line-147">                        } else {</span>
<span class="source-line-no">148</span><span id="line-148">                                if (throwExceptionGeneNotFound) {</span>
<span class="source-line-no">149</span><span id="line-149">                                        throw new Exception(dnaSequence.getAccession().toString() + " not found");</span>
<span class="source-line-no">150</span><span id="line-150">                                }</span>
<span class="source-line-no">151</span><span id="line-151">                                logger.info("Gene not found {}", dnaSequence.getAccession().toString());</span>
<span class="source-line-no">152</span><span id="line-152">                        }</span>
<span class="source-line-no">153</span><span id="line-153"></span>
<span class="source-line-no">154</span><span id="line-154">                }</span>
<span class="source-line-no">155</span><span id="line-155">                return chromosomeSequenceList;</span>
<span class="source-line-no">156</span><span id="line-156">        }</span>
<span class="source-line-no">157</span><span id="line-157"></span>
<span class="source-line-no">158</span><span id="line-158">        /**</span>
<span class="source-line-no">159</span><span id="line-159">         * Output a gff3 feature file that will give the length of each scaffold/chromosome in the fasta file.</span>
<span class="source-line-no">160</span><span id="line-160">         * Used for gbrowse so it knows length.</span>
<span class="source-line-no">161</span><span id="line-161">         * @param fastaSequenceFile</span>
<span class="source-line-no">162</span><span id="line-162">         * @param gffFile</span>
<span class="source-line-no">163</span><span id="line-163">         * @throws Exception</span>
<span class="source-line-no">164</span><span id="line-164">         */</span>
<span class="source-line-no">165</span><span id="line-165">        static public void outputFastaSequenceLengthGFF3(File fastaSequenceFile, File gffFile) throws Exception {</span>
<span class="source-line-no">166</span><span id="line-166">                Map&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</span>
<span class="source-line-no">167</span><span id="line-167">                String fileName = fastaSequenceFile.getName();</span>
<span class="source-line-no">168</span><span id="line-168">                FileWriter fw = new FileWriter(gffFile);</span>
<span class="source-line-no">169</span><span id="line-169">                String newLine = System.getProperty("line.separator");</span>
<span class="source-line-no">170</span><span id="line-170">                fw.write("##gff-version 3" + newLine);</span>
<span class="source-line-no">171</span><span id="line-171">                for (DNASequence dnaSequence : dnaSequenceList.values()) {</span>
<span class="source-line-no">172</span><span id="line-172">                        String gff3line = dnaSequence.getAccession().getID() + "\t" + fileName + "\t" + "contig" + "\t" + "1" + "\t" + dnaSequence.getBioEnd() + "\t.\t.\t.\tName=" + dnaSequence.getAccession().getID() + newLine;</span>
<span class="source-line-no">173</span><span id="line-173">                        fw.write(gff3line);</span>
<span class="source-line-no">174</span><span id="line-174">                }</span>
<span class="source-line-no">175</span><span id="line-175">                fw.close();</span>
<span class="source-line-no">176</span><span id="line-176">        }</span>
<span class="source-line-no">177</span><span id="line-177"></span>
<span class="source-line-no">178</span><span id="line-178">        /**</span>
<span class="source-line-no">179</span><span id="line-179">         * Loads Fasta file and GFF2 feature file generated from the geneid prediction algorithm</span>
<span class="source-line-no">180</span><span id="line-180">         *</span>
<span class="source-line-no">181</span><span id="line-181">         * @param fastaSequenceFile</span>
<span class="source-line-no">182</span><span id="line-182">         * @param gffFile</span>
<span class="source-line-no">183</span><span id="line-183">         * @return</span>
<span class="source-line-no">184</span><span id="line-184">         * @throws Exception</span>
<span class="source-line-no">185</span><span id="line-185">         */</span>
<span class="source-line-no">186</span><span id="line-186">        static public Map&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGeneIDGFF2(File fastaSequenceFile, File gffFile) throws Exception {</span>
<span class="source-line-no">187</span><span id="line-187">                Map&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</span>
<span class="source-line-no">188</span><span id="line-188">                Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</span>
<span class="source-line-no">189</span><span id="line-189">                FeatureList listGenes = GeneIDGFF2Reader.read(gffFile.getAbsolutePath());</span>
<span class="source-line-no">190</span><span id="line-190">                addGeneIDGFF2GeneFeatures(chromosomeSequenceList, listGenes);</span>
<span class="source-line-no">191</span><span id="line-191">                return chromosomeSequenceList;</span>
<span class="source-line-no">192</span><span id="line-192">        }</span>
<span class="source-line-no">193</span><span id="line-193"></span>
<span class="source-line-no">194</span><span id="line-194">        /**</span>
<span class="source-line-no">195</span><span id="line-195">         * Load GFF2 feature file generated from the geneid prediction algorithm and map features onto the chromosome sequences</span>
<span class="source-line-no">196</span><span id="line-196">         *</span>
<span class="source-line-no">197</span><span id="line-197">         * @param chromosomeSequenceList</span>
<span class="source-line-no">198</span><span id="line-198">         * @param listGenes</span>
<span class="source-line-no">199</span><span id="line-199">         * @throws Exception</span>
<span class="source-line-no">200</span><span id="line-200">         */</span>
<span class="source-line-no">201</span><span id="line-201">        static public void addGeneIDGFF2GeneFeatures(Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</span>
<span class="source-line-no">202</span><span id="line-202">                Collection&lt;String&gt; geneIds = listGenes.attributeValues("gene_id");</span>
<span class="source-line-no">203</span><span id="line-203">                for (String geneid : geneIds) {</span>
<span class="source-line-no">204</span><span id="line-204">                        FeatureList gene = listGenes.selectByAttribute("gene_id", geneid);</span>
<span class="source-line-no">205</span><span id="line-205">                        FeatureI geneFeature = gene.get(0);</span>
<span class="source-line-no">206</span><span id="line-206">                        ChromosomeSequence seq = chromosomeSequenceList.get(geneFeature.seqname());</span>
<span class="source-line-no">207</span><span id="line-207">                        geneid = geneid.replaceAll("_", ".G");</span>
<span class="source-line-no">208</span><span id="line-208">                        AccessionID geneAccessionID = new AccessionID(geneid);</span>
<span class="source-line-no">209</span><span id="line-209">                        GeneSequence geneSequence = null;</span>
<span class="source-line-no">210</span><span id="line-210">                        Collection&lt;String&gt; transcriptids = gene.attributeValues("gene_id");</span>
<span class="source-line-no">211</span><span id="line-211">                        for (String transcriptid : transcriptids) {</span>
<span class="source-line-no">212</span><span id="line-212">                                // get all the individual features (exons, CDS regions, etc.) of this gene</span>
<span class="source-line-no">213</span><span id="line-213">                                FeatureList transcriptFeature = listGenes.selectByAttribute("gene_id", transcriptid);</span>
<span class="source-line-no">214</span><span id="line-214">                                transcriptid = transcriptid.replaceAll("_", ".G");</span>
<span class="source-line-no">215</span><span id="line-215"></span>
<span class="source-line-no">216</span><span id="line-216"></span>
<span class="source-line-no">217</span><span id="line-217"></span>
<span class="source-line-no">218</span><span id="line-218"></span>
<span class="source-line-no">219</span><span id="line-219">                                //      String seqName = feature.seqname();</span>
<span class="source-line-no">220</span><span id="line-220">                                //FeatureI startCodon = null;</span>
<span class="source-line-no">221</span><span id="line-221">                                //FeatureI stopCodon = null;</span>
<span class="source-line-no">222</span><span id="line-222">                                Integer startCodonBegin = null;</span>
<span class="source-line-no">223</span><span id="line-223">                                Integer stopCodonEnd = null;</span>
<span class="source-line-no">224</span><span id="line-224">                                //String startCodonName = "";</span>
<span class="source-line-no">225</span><span id="line-225">                                //String stopCodonName = "";</span>
<span class="source-line-no">226</span><span id="line-226"></span>
<span class="source-line-no">227</span><span id="line-227"></span>
<span class="source-line-no">228</span><span id="line-228">                                // now select only the coding regions of this gene</span>
<span class="source-line-no">229</span><span id="line-229">                                FeatureList firstFeatures = transcriptFeature.selectByType("First");</span>
<span class="source-line-no">230</span><span id="line-230">                                FeatureList terminalFeatures = transcriptFeature.selectByType("Terminal");</span>
<span class="source-line-no">231</span><span id="line-231">                                FeatureList internalFeatures = transcriptFeature.selectByType("Internal");</span>
<span class="source-line-no">232</span><span id="line-232">                                FeatureList singleFeatures = transcriptFeature.selectByType("Single");</span>
<span class="source-line-no">233</span><span id="line-233">                                FeatureList cdsFeatures = new FeatureList();</span>
<span class="source-line-no">234</span><span id="line-234">                                cdsFeatures.add(firstFeatures);</span>
<span class="source-line-no">235</span><span id="line-235">                                cdsFeatures.add(terminalFeatures);</span>
<span class="source-line-no">236</span><span id="line-236">                                cdsFeatures.add(internalFeatures);</span>
<span class="source-line-no">237</span><span id="line-237">                                cdsFeatures.add(singleFeatures);</span>
<span class="source-line-no">238</span><span id="line-238">                                // sort them</span>
<span class="source-line-no">239</span><span id="line-239">                                cdsFeatures = cdsFeatures.sortByStart();</span>
<span class="source-line-no">240</span><span id="line-240">                                Strand strand = Strand.POSITIVE;</span>
<span class="source-line-no">241</span><span id="line-241">                                FeatureI feature = cdsFeatures.get(0);</span>
<span class="source-line-no">242</span><span id="line-242">                                if (feature.location().isNegative()) {</span>
<span class="source-line-no">243</span><span id="line-243">                                        strand = Strand.NEGATIVE;</span>
<span class="source-line-no">244</span><span id="line-244">                                }</span>
<span class="source-line-no">245</span><span id="line-245">                                if (startCodonBegin == null) {</span>
<span class="source-line-no">246</span><span id="line-246">                                        FeatureI firstFeature = cdsFeatures.get(0);</span>
<span class="source-line-no">247</span><span id="line-247">                                        if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">248</span><span id="line-248">                                                startCodonBegin = firstFeature.location().bioEnd();</span>
<span class="source-line-no">249</span><span id="line-249">                                        } else {</span>
<span class="source-line-no">250</span><span id="line-250">                                                startCodonBegin = firstFeature.location().bioStart();</span>
<span class="source-line-no">251</span><span id="line-251">                                        }</span>
<span class="source-line-no">252</span><span id="line-252">                                }</span>
<span class="source-line-no">253</span><span id="line-253"></span>
<span class="source-line-no">254</span><span id="line-254">                                if (stopCodonEnd == null) {</span>
<span class="source-line-no">255</span><span id="line-255"></span>
<span class="source-line-no">256</span><span id="line-256">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</span>
<span class="source-line-no">257</span><span id="line-257">                                        if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">258</span><span id="line-258">                                                stopCodonEnd = lastFeature.location().bioStart();</span>
<span class="source-line-no">259</span><span id="line-259">                                        } else {</span>
<span class="source-line-no">260</span><span id="line-260">                                                stopCodonEnd = lastFeature.location().bioEnd();</span>
<span class="source-line-no">261</span><span id="line-261">                                        }</span>
<span class="source-line-no">262</span><span id="line-262">                                }</span>
<span class="source-line-no">263</span><span id="line-263">                                //for gtf ordering can be strand based so first is last and last is first</span>
<span class="source-line-no">264</span><span id="line-264">                                if (startCodonBegin &gt; stopCodonEnd) {</span>
<span class="source-line-no">265</span><span id="line-265">                                        int temp = startCodonBegin;</span>
<span class="source-line-no">266</span><span id="line-266">                                        startCodonBegin = stopCodonEnd;</span>
<span class="source-line-no">267</span><span id="line-267">                                        stopCodonEnd = temp;</span>
<span class="source-line-no">268</span><span id="line-268">                                }</span>
<span class="source-line-no">269</span><span id="line-269"></span>
<span class="source-line-no">270</span><span id="line-270">                                AccessionID transcriptAccessionID = new AccessionID(transcriptid);</span>
<span class="source-line-no">271</span><span id="line-271">                                if (geneSequence == null) {</span>
<span class="source-line-no">272</span><span id="line-272">                                        geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</span>
<span class="source-line-no">273</span><span id="line-273">                                        geneSequence.setSource(((Feature) feature).source());</span>
<span class="source-line-no">274</span><span id="line-274">                                } else {</span>
<span class="source-line-no">275</span><span id="line-275">                                        //if multiple transcripts for one gene make sure the gene is defined as the min and max start/end</span>
<span class="source-line-no">276</span><span id="line-276"></span>
<span class="source-line-no">277</span><span id="line-277">                                        if (startCodonBegin &lt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">278</span><span id="line-278">                                                geneSequence.setBioBegin(startCodonBegin);</span>
<span class="source-line-no">279</span><span id="line-279">                                        }</span>
<span class="source-line-no">280</span><span id="line-280">                                        if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">281</span><span id="line-281">                                                geneSequence.setBioEnd(stopCodonEnd);</span>
<span class="source-line-no">282</span><span id="line-282">                                        }</span>
<span class="source-line-no">283</span><span id="line-283"></span>
<span class="source-line-no">284</span><span id="line-284">                                }</span>
<span class="source-line-no">285</span><span id="line-285">                                TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</span>
<span class="source-line-no">286</span><span id="line-286">                                /*</span>
<span class="source-line-no">287</span><span id="line-287">                                if (startCodon != null) {</span>
<span class="source-line-no">288</span><span id="line-288">                                        if (startCodonName == null || startCodonName.length() == 0) {</span>
<span class="source-line-no">289</span><span id="line-289">                                                startCodonName = transcriptid + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</span>
<span class="source-line-no">290</span><span id="line-290">                                        }</span>
<span class="source-line-no">291</span><span id="line-291">                                        transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</span>
<span class="source-line-no">292</span><span id="line-292">                                }</span>
<span class="source-line-no">293</span><span id="line-293">                                if (stopCodon != null) {</span>
<span class="source-line-no">294</span><span id="line-294">                                        if (stopCodonName == null || stopCodonName.length() == 0) {</span>
<span class="source-line-no">295</span><span id="line-295">                                                stopCodonName = transcriptid + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</span>
<span class="source-line-no">296</span><span id="line-296">                                        }</span>
<span class="source-line-no">297</span><span id="line-297">                                        transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</span>
<span class="source-line-no">298</span><span id="line-298">                                }</span>
<span class="source-line-no">299</span><span id="line-299">                                */</span>
<span class="source-line-no">300</span><span id="line-300"></span>
<span class="source-line-no">301</span><span id="line-301">                                for (FeatureI cdsFeature : cdsFeatures) {</span>
<span class="source-line-no">302</span><span id="line-302">                                        Feature cds = (Feature) cdsFeature;</span>
<span class="source-line-no">303</span><span id="line-303">                                        String cdsName = cds.getAttribute("transcript_name");</span>
<span class="source-line-no">304</span><span id="line-304">                                        if (cdsName == null || cdsName.length() == 0) {</span>
<span class="source-line-no">305</span><span id="line-305">                                                cdsName = transcriptid + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</span>
<span class="source-line-no">306</span><span id="line-306">                                        }</span>
<span class="source-line-no">307</span><span id="line-307">                                        AccessionID cdsAccessionID = new AccessionID(cdsName);</span>
<span class="source-line-no">308</span><span id="line-308">                                        //ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</span>
<span class="source-line-no">309</span><span id="line-309">                                        CDSSequence cdsSequence = transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), cds.frame());</span>
<span class="source-line-no">310</span><span id="line-310">                                        cdsSequence.setSequenceScore(cds.score());</span>
<span class="source-line-no">311</span><span id="line-311">                                }</span>
<span class="source-line-no">312</span><span id="line-312">                        }</span>
<span class="source-line-no">313</span><span id="line-313">                }</span>
<span class="source-line-no">314</span><span id="line-314"></span>
<span class="source-line-no">315</span><span id="line-315">        }</span>
<span class="source-line-no">316</span><span id="line-316"></span>
<span class="source-line-no">317</span><span id="line-317">        static public Map&lt;String, ChromosomeSequence&gt; getChromosomeSequenceFromDNASequence(Map&lt;String, DNASequence&gt; dnaSequenceList) {</span>
<span class="source-line-no">318</span><span id="line-318">                LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">319</span><span id="line-319">                for (String key : dnaSequenceList.keySet()) {</span>
<span class="source-line-no">320</span><span id="line-320">                        DNASequence dnaSequence = dnaSequenceList.get(key);</span>
<span class="source-line-no">321</span><span id="line-321">                        ChromosomeSequence chromosomeSequence = new ChromosomeSequence(dnaSequence.getProxySequenceReader()); //we want the underlying sequence but don't need storage</span>
<span class="source-line-no">322</span><span id="line-322">                        chromosomeSequence.setAccession(dnaSequence.getAccession());</span>
<span class="source-line-no">323</span><span id="line-323">                        chromosomeSequenceList.put(key, chromosomeSequence);</span>
<span class="source-line-no">324</span><span id="line-324">                }</span>
<span class="source-line-no">325</span><span id="line-325">                return chromosomeSequenceList;</span>
<span class="source-line-no">326</span><span id="line-326">        }</span>
<span class="source-line-no">327</span><span id="line-327"></span>
<span class="source-line-no">328</span><span id="line-328">        /**</span>
<span class="source-line-no">329</span><span id="line-329">         * Lots of variations in the ontology or descriptors that can be used in GFF3 which requires writing a custom parser to handle a GFF3 generated or used</span>
<span class="source-line-no">330</span><span id="line-330">         * by a specific application. Probably could be abstracted out but for now easier to handle with custom code to deal with gff3 elements that are not</span>
<span class="source-line-no">331</span><span id="line-331">         * included but can be extracted from other data elements.</span>
<span class="source-line-no">332</span><span id="line-332">         * @param fastaSequenceFile</span>
<span class="source-line-no">333</span><span id="line-333">         * @param gffFile</span>
<span class="source-line-no">334</span><span id="line-334">         * @param lazyloadsequences If set to true then the fasta file will be parsed for accession id but sequences will be read from disk when needed to save memory</span>
<span class="source-line-no">335</span><span id="line-335">         * @return</span>
<span class="source-line-no">336</span><span id="line-336">         * @throws Exception</span>
<span class="source-line-no">337</span><span id="line-337">         */</span>
<span class="source-line-no">338</span><span id="line-338">        static public Map&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGmodGFF3(File fastaSequenceFile, File gffFile,boolean lazyloadsequences) throws Exception {</span>
<span class="source-line-no">339</span><span id="line-339">                Map&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile,lazyloadsequences);</span>
<span class="source-line-no">340</span><span id="line-340">                Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</span>
<span class="source-line-no">341</span><span id="line-341">                FeatureList listGenes = GFF3Reader.read(gffFile.getAbsolutePath());</span>
<span class="source-line-no">342</span><span id="line-342">                addGmodGFF3GeneFeatures(chromosomeSequenceList, listGenes);</span>
<span class="source-line-no">343</span><span id="line-343">                return chromosomeSequenceList;</span>
<span class="source-line-no">344</span><span id="line-344">        }</span>
<span class="source-line-no">345</span><span id="line-345"></span>
<span class="source-line-no">346</span><span id="line-346">        /**</span>
<span class="source-line-no">347</span><span id="line-347">         * Load GFF3 file using mRNA as the gene feature as not all GFF3 files are complete</span>
<span class="source-line-no">348</span><span id="line-348">         * @param chromosomeSequenceList</span>
<span class="source-line-no">349</span><span id="line-349">         * @param listGenes</span>
<span class="source-line-no">350</span><span id="line-350">         * @throws Exception</span>
<span class="source-line-no">351</span><span id="line-351">         */</span>
<span class="source-line-no">352</span><span id="line-352">        static public void addGmodGFF3GeneFeatures(Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</span>
<span class="source-line-no">353</span><span id="line-353"></span>
<span class="source-line-no">354</span><span id="line-354"></span>
<span class="source-line-no">355</span><span id="line-355">                // key off mRNA as being a known feature that may or may not have a parent gene</span>
<span class="source-line-no">356</span><span id="line-356"></span>
<span class="source-line-no">357</span><span id="line-357"></span>
<span class="source-line-no">358</span><span id="line-358">                FeatureList mRNAFeatures = listGenes.selectByType("mRNA");</span>
<span class="source-line-no">359</span><span id="line-359">                LinkedHashMap&lt;String,FeatureList&gt; featureIDHashMap = FeatureHelper.buildFeatureAtrributeIndex("ID", listGenes);</span>
<span class="source-line-no">360</span><span id="line-360">                LinkedHashMap&lt;String,FeatureList&gt; featureParentHashMap = FeatureHelper.buildFeatureAtrributeIndex("Parent", listGenes);</span>
<span class="source-line-no">361</span><span id="line-361"></span>
<span class="source-line-no">362</span><span id="line-362">                for (FeatureI f : mRNAFeatures) {</span>
<span class="source-line-no">363</span><span id="line-363">                        String geneID;</span>
<span class="source-line-no">364</span><span id="line-364">                        String geneNote = null;</span>
<span class="source-line-no">365</span><span id="line-365">                        String geneSource = null;</span>
<span class="source-line-no">366</span><span id="line-366">                        String sequenceName = null;</span>
<span class="source-line-no">367</span><span id="line-367">                        ChromosomeSequence seq = null;</span>
<span class="source-line-no">368</span><span id="line-368">                        GeneSequence geneSequence = null;</span>
<span class="source-line-no">369</span><span id="line-369"></span>
<span class="source-line-no">370</span><span id="line-370">                        Feature mRNAFeature = (Feature) f;</span>
<span class="source-line-no">371</span><span id="line-371">                        String mRNAID = mRNAFeature.getAttribute("ID");</span>
<span class="source-line-no">372</span><span id="line-372">                        String mRNAsource = mRNAFeature.source();</span>
<span class="source-line-no">373</span><span id="line-373">                        String mRNANote = mRNAFeature.getAttribute("Note");</span>
<span class="source-line-no">374</span><span id="line-374">                        String mRNAParent = mRNAFeature.getAttribute("Parent");</span>
<span class="source-line-no">375</span><span id="line-375">                        if (mRNAParent != null &amp;&amp; mRNAParent.length() &gt; 0) {</span>
<span class="source-line-no">376</span><span id="line-376">                           // FeatureList geneFeatureList = listGenes.selectByAttribute("ID", mRNAParent);</span>
<span class="source-line-no">377</span><span id="line-377">                                FeatureList geneFeatureList = featureIDHashMap.get(mRNAParent);</span>
<span class="source-line-no">378</span><span id="line-378">                                Feature geneFeature = (Feature) geneFeatureList.get(0);</span>
<span class="source-line-no">379</span><span id="line-379">                                geneID = geneFeature.getAttribute("ID");</span>
<span class="source-line-no">380</span><span id="line-380">                                geneNote = geneFeature.getAttribute("Note");</span>
<span class="source-line-no">381</span><span id="line-381">                                geneSource = geneFeature.source();</span>
<span class="source-line-no">382</span><span id="line-382">                                sequenceName = geneFeature.seqname();</span>
<span class="source-line-no">383</span><span id="line-383"></span>
<span class="source-line-no">384</span><span id="line-384">                                //</span>
<span class="source-line-no">385</span><span id="line-385">                        } else {</span>
<span class="source-line-no">386</span><span id="line-386">                                //deal with cases where no parent gene is given</span>
<span class="source-line-no">387</span><span id="line-387">                                geneID = mRNAID;</span>
<span class="source-line-no">388</span><span id="line-388">                                geneSource = mRNAsource;</span>
<span class="source-line-no">389</span><span id="line-389">                                sequenceName = mRNAFeature.seqname();</span>
<span class="source-line-no">390</span><span id="line-390">                        }</span>
<span class="source-line-no">391</span><span id="line-391"></span>
<span class="source-line-no">392</span><span id="line-392">                        seq = chromosomeSequenceList.get(sequenceName);</span>
<span class="source-line-no">393</span><span id="line-393"></span>
<span class="source-line-no">394</span><span id="line-394">                        AccessionID geneAccessionID = new AccessionID(geneID);</span>
<span class="source-line-no">395</span><span id="line-395"></span>
<span class="source-line-no">396</span><span id="line-396">                  //  FeatureList mRNAChildren = listGenes.selectByAttribute("Parent", mRNAID);</span>
<span class="source-line-no">397</span><span id="line-397">                        FeatureList mRNAChildren = featureParentHashMap.get(mRNAID);</span>
<span class="source-line-no">398</span><span id="line-398">                        FeatureList cdsFeatures = mRNAChildren.selectByType("CDS");</span>
<span class="source-line-no">399</span><span id="line-399">                        FeatureI feature = cdsFeatures.get(0);</span>
<span class="source-line-no">400</span><span id="line-400">                        Strand strand = Strand.POSITIVE;</span>
<span class="source-line-no">401</span><span id="line-401"></span>
<span class="source-line-no">402</span><span id="line-402">                        if (feature.location().isNegative()) {</span>
<span class="source-line-no">403</span><span id="line-403">                                strand = Strand.NEGATIVE;</span>
<span class="source-line-no">404</span><span id="line-404">                        }</span>
<span class="source-line-no">405</span><span id="line-405">                        cdsFeatures = cdsFeatures.sortByStart();</span>
<span class="source-line-no">406</span><span id="line-406"></span>
<span class="source-line-no">407</span><span id="line-407"></span>
<span class="source-line-no">408</span><span id="line-408"></span>
<span class="source-line-no">409</span><span id="line-409"></span>
<span class="source-line-no">410</span><span id="line-410"></span>
<span class="source-line-no">411</span><span id="line-411"></span>
<span class="source-line-no">412</span><span id="line-412"></span>
<span class="source-line-no">413</span><span id="line-413">                        //String seqName = feature.seqname();</span>
<span class="source-line-no">414</span><span id="line-414">                        FeatureI startCodon = null;</span>
<span class="source-line-no">415</span><span id="line-415">                        FeatureI stopCodon = null;</span>
<span class="source-line-no">416</span><span id="line-416">                        Integer startCodonBegin = null;</span>
<span class="source-line-no">417</span><span id="line-417">                        Integer stopCodonEnd = null;</span>
<span class="source-line-no">418</span><span id="line-418">                        String startCodonName = "";</span>
<span class="source-line-no">419</span><span id="line-419">                        String stopCodonName = "";</span>
<span class="source-line-no">420</span><span id="line-420">                        FeatureList startCodonList = mRNAChildren.selectByType("five_prime_UTR");</span>
<span class="source-line-no">421</span><span id="line-421">                        if (startCodonList != null &amp;&amp; startCodonList.size() &gt; 0) {</span>
<span class="source-line-no">422</span><span id="line-422">                                startCodon = startCodonList.get(0);</span>
<span class="source-line-no">423</span><span id="line-423">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">424</span><span id="line-424">                                        startCodonBegin = startCodon.location().bioEnd();</span>
<span class="source-line-no">425</span><span id="line-425">                                } else {</span>
<span class="source-line-no">426</span><span id="line-426">                                        startCodonBegin = startCodon.location().bioStart();</span>
<span class="source-line-no">427</span><span id="line-427">                                }</span>
<span class="source-line-no">428</span><span id="line-428">                                startCodonName = startCodon.getAttribute("ID");</span>
<span class="source-line-no">429</span><span id="line-429">                        }</span>
<span class="source-line-no">430</span><span id="line-430"></span>
<span class="source-line-no">431</span><span id="line-431">                        FeatureList stopCodonList = mRNAChildren.selectByType("three_prime_UTR");</span>
<span class="source-line-no">432</span><span id="line-432"></span>
<span class="source-line-no">433</span><span id="line-433">                        if (stopCodonList != null &amp;&amp; stopCodonList.size() &gt; 0) {</span>
<span class="source-line-no">434</span><span id="line-434">                                stopCodon = stopCodonList.get(0);</span>
<span class="source-line-no">435</span><span id="line-435">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">436</span><span id="line-436">                                        stopCodonEnd = stopCodon.location().bioStart();</span>
<span class="source-line-no">437</span><span id="line-437">                                } else {</span>
<span class="source-line-no">438</span><span id="line-438">                                        stopCodonEnd = stopCodon.location().bioEnd();</span>
<span class="source-line-no">439</span><span id="line-439">                                }</span>
<span class="source-line-no">440</span><span id="line-440">                                stopCodonName = stopCodon.getAttribute("ID");</span>
<span class="source-line-no">441</span><span id="line-441"></span>
<span class="source-line-no">442</span><span id="line-442">                        }</span>
<span class="source-line-no">443</span><span id="line-443"></span>
<span class="source-line-no">444</span><span id="line-444"></span>
<span class="source-line-no">445</span><span id="line-445"></span>
<span class="source-line-no">446</span><span id="line-446"></span>
<span class="source-line-no">447</span><span id="line-447">                        if (startCodonBegin == null) {</span>
<span class="source-line-no">448</span><span id="line-448">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">449</span><span id="line-449">                                        FeatureI firstFeature = cdsFeatures.get(0);</span>
<span class="source-line-no">450</span><span id="line-450"></span>
<span class="source-line-no">451</span><span id="line-451">                                        startCodonBegin = firstFeature.location().bioEnd();</span>
<span class="source-line-no">452</span><span id="line-452">                                } else {</span>
<span class="source-line-no">453</span><span id="line-453">                                        FeatureI firstFeature = cdsFeatures.get(0);</span>
<span class="source-line-no">454</span><span id="line-454"></span>
<span class="source-line-no">455</span><span id="line-455">                                        startCodonBegin = firstFeature.location().bioStart();</span>
<span class="source-line-no">456</span><span id="line-456">                                }</span>
<span class="source-line-no">457</span><span id="line-457">                        }</span>
<span class="source-line-no">458</span><span id="line-458"></span>
<span class="source-line-no">459</span><span id="line-459">                        if (stopCodonEnd == null) {</span>
<span class="source-line-no">460</span><span id="line-460">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">461</span><span id="line-461">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</span>
<span class="source-line-no">462</span><span id="line-462">                                        stopCodonEnd = lastFeature.location().bioStart();</span>
<span class="source-line-no">463</span><span id="line-463">                                } else {</span>
<span class="source-line-no">464</span><span id="line-464">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</span>
<span class="source-line-no">465</span><span id="line-465">                                        stopCodonEnd = lastFeature.location().bioEnd();</span>
<span class="source-line-no">466</span><span id="line-466">                                }</span>
<span class="source-line-no">467</span><span id="line-467">                        }</span>
<span class="source-line-no">468</span><span id="line-468">                        //for gtf ordering can be strand based so first is last and last is first</span>
<span class="source-line-no">469</span><span id="line-469">                        if (startCodonBegin &gt; stopCodonEnd) {</span>
<span class="source-line-no">470</span><span id="line-470">                                int temp = startCodonBegin;</span>
<span class="source-line-no">471</span><span id="line-471">                                startCodonBegin = stopCodonEnd;</span>
<span class="source-line-no">472</span><span id="line-472">                                stopCodonEnd = temp;</span>
<span class="source-line-no">473</span><span id="line-473">                        }</span>
<span class="source-line-no">474</span><span id="line-474"></span>
<span class="source-line-no">475</span><span id="line-475"></span>
<span class="source-line-no">476</span><span id="line-476"></span>
<span class="source-line-no">477</span><span id="line-477">                        AccessionID transcriptAccessionID = new AccessionID(mRNAID);</span>
<span class="source-line-no">478</span><span id="line-478">                        geneSequence = seq.getGene(geneID);</span>
<span class="source-line-no">479</span><span id="line-479">                        if (geneSequence == null) {</span>
<span class="source-line-no">480</span><span id="line-480">                                geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</span>
<span class="source-line-no">481</span><span id="line-481">                                geneSequence.setSource(geneSource);</span>
<span class="source-line-no">482</span><span id="line-482">                                if (geneNote != null &amp;&amp; geneNote.length() &gt; 0) {</span>
<span class="source-line-no">483</span><span id="line-483">                                        geneSequence.addNote(geneNote);</span>
<span class="source-line-no">484</span><span id="line-484">                                }</span>
<span class="source-line-no">485</span><span id="line-485">                        } else {</span>
<span class="source-line-no">486</span><span id="line-486"></span>
<span class="source-line-no">487</span><span id="line-487">                                if (startCodonBegin &lt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">488</span><span id="line-488">                                        geneSequence.setBioBegin(startCodonBegin);</span>
<span class="source-line-no">489</span><span id="line-489">                                }</span>
<span class="source-line-no">490</span><span id="line-490">                                if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">491</span><span id="line-491">                                        geneSequence.setBioEnd(stopCodonEnd);</span>
<span class="source-line-no">492</span><span id="line-492">                                }</span>
<span class="source-line-no">493</span><span id="line-493"></span>
<span class="source-line-no">494</span><span id="line-494">                        }</span>
<span class="source-line-no">495</span><span id="line-495">                        TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</span>
<span class="source-line-no">496</span><span id="line-496">                        transcriptSequence.setSource(mRNAsource);</span>
<span class="source-line-no">497</span><span id="line-497">                        if (mRNANote != null &amp;&amp; mRNANote.length() &gt; 0) {</span>
<span class="source-line-no">498</span><span id="line-498">                                transcriptSequence.addNote(mRNANote);</span>
<span class="source-line-no">499</span><span id="line-499"></span>
<span class="source-line-no">500</span><span id="line-500">                        }</span>
<span class="source-line-no">501</span><span id="line-501">                        if (startCodon != null) {</span>
<span class="source-line-no">502</span><span id="line-502">                                if (startCodonName == null || startCodonName.length() == 0) {</span>
<span class="source-line-no">503</span><span id="line-503">                                        startCodonName = mRNAID + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</span>
<span class="source-line-no">504</span><span id="line-504">                                }</span>
<span class="source-line-no">505</span><span id="line-505">                                transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</span>
<span class="source-line-no">506</span><span id="line-506">                        }</span>
<span class="source-line-no">507</span><span id="line-507">                        if (stopCodon != null) {</span>
<span class="source-line-no">508</span><span id="line-508">                                if (stopCodonName == null || stopCodonName.length() == 0) {</span>
<span class="source-line-no">509</span><span id="line-509">                                        stopCodonName = mRNAID + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</span>
<span class="source-line-no">510</span><span id="line-510">                                }</span>
<span class="source-line-no">511</span><span id="line-511">                                transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</span>
<span class="source-line-no">512</span><span id="line-512">                        }</span>
<span class="source-line-no">513</span><span id="line-513"></span>
<span class="source-line-no">514</span><span id="line-514">                        for (FeatureI cdsFeature : cdsFeatures) {</span>
<span class="source-line-no">515</span><span id="line-515">                                Feature cds = (Feature) cdsFeature;</span>
<span class="source-line-no">516</span><span id="line-516">                                String cdsNote = cdsFeature.getAttribute("Note");</span>
<span class="source-line-no">517</span><span id="line-517">                                String cdsSource = cds.source();</span>
<span class="source-line-no">518</span><span id="line-518">                                String cdsName = cds.getAttribute("ID");</span>
<span class="source-line-no">519</span><span id="line-519">                                if (cdsName == null || cdsName.length() == 0) {</span>
<span class="source-line-no">520</span><span id="line-520">                                        cdsName = mRNAID + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</span>
<span class="source-line-no">521</span><span id="line-521">                                }</span>
<span class="source-line-no">522</span><span id="line-522">                                AccessionID cdsAccessionID = new AccessionID(cdsName);</span>
<span class="source-line-no">523</span><span id="line-523">                                ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</span>
<span class="source-line-no">524</span><span id="line-524">                                exonSequence.setSource(cdsSource);</span>
<span class="source-line-no">525</span><span id="line-525">                                if (cdsNote != null &amp;&amp; cdsNote.length() &gt; 0) {</span>
<span class="source-line-no">526</span><span id="line-526">                                        exonSequence.addNote(cdsNote);</span>
<span class="source-line-no">527</span><span id="line-527">                                }</span>
<span class="source-line-no">528</span><span id="line-528">                                transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), cds.frame());</span>
<span class="source-line-no">529</span><span id="line-529">                        }</span>
<span class="source-line-no">530</span><span id="line-530">                        geneSequence.addIntronsUsingExons();</span>
<span class="source-line-no">531</span><span id="line-531"></span>
<span class="source-line-no">532</span><span id="line-532">                }</span>
<span class="source-line-no">533</span><span id="line-533"></span>
<span class="source-line-no">534</span><span id="line-534">        }</span>
<span class="source-line-no">535</span><span id="line-535"></span>
<span class="source-line-no">536</span><span id="line-536">        static public Map&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGlimmerGFF3(File fastaSequenceFile, File gffFile) throws Exception {</span>
<span class="source-line-no">537</span><span id="line-537">                Map&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</span>
<span class="source-line-no">538</span><span id="line-538">                Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</span>
<span class="source-line-no">539</span><span id="line-539">                FeatureList listGenes = GFF3Reader.read(gffFile.getAbsolutePath());</span>
<span class="source-line-no">540</span><span id="line-540">                addGlimmerGFF3GeneFeatures(chromosomeSequenceList, listGenes);</span>
<span class="source-line-no">541</span><span id="line-541">                return chromosomeSequenceList;</span>
<span class="source-line-no">542</span><span id="line-542">        }</span>
<span class="source-line-no">543</span><span id="line-543"></span>
<span class="source-line-no">544</span><span id="line-544">        static public void addGlimmerGFF3GeneFeatures(Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</span>
<span class="source-line-no">545</span><span id="line-545">                FeatureList mRNAFeatures = listGenes.selectByType("mRNA");</span>
<span class="source-line-no">546</span><span id="line-546">                for (FeatureI f : mRNAFeatures) {</span>
<span class="source-line-no">547</span><span id="line-547">                        Feature mRNAFeature = (Feature) f;</span>
<span class="source-line-no">548</span><span id="line-548">                        String geneid = mRNAFeature.getAttribute("ID");</span>
<span class="source-line-no">549</span><span id="line-549">                        String source = mRNAFeature.source();</span>
<span class="source-line-no">550</span><span id="line-550"></span>
<span class="source-line-no">551</span><span id="line-551">                        FeatureList gene = listGenes.selectByAttribute("Parent", geneid);</span>
<span class="source-line-no">552</span><span id="line-552">                        FeatureI geneFeature = gene.get(0);</span>
<span class="source-line-no">553</span><span id="line-553">                        ChromosomeSequence seq = chromosomeSequenceList.get(geneFeature.seqname());</span>
<span class="source-line-no">554</span><span id="line-554">                        AccessionID geneAccessionID = new AccessionID(geneid);</span>
<span class="source-line-no">555</span><span id="line-555">                        GeneSequence geneSequence = null;</span>
<span class="source-line-no">556</span><span id="line-556"></span>
<span class="source-line-no">557</span><span id="line-557">                        FeatureList cdsFeatures = gene.selectByType("CDS");</span>
<span class="source-line-no">558</span><span id="line-558">                        FeatureI feature = cdsFeatures.get(0);</span>
<span class="source-line-no">559</span><span id="line-559">                        Strand strand = Strand.POSITIVE;</span>
<span class="source-line-no">560</span><span id="line-560"></span>
<span class="source-line-no">561</span><span id="line-561">                        if (feature.location().isNegative()) {</span>
<span class="source-line-no">562</span><span id="line-562">                                strand = Strand.NEGATIVE;</span>
<span class="source-line-no">563</span><span id="line-563">                        }</span>
<span class="source-line-no">564</span><span id="line-564">                        cdsFeatures = cdsFeatures.sortByStart();</span>
<span class="source-line-no">565</span><span id="line-565"></span>
<span class="source-line-no">566</span><span id="line-566"></span>
<span class="source-line-no">567</span><span id="line-567"></span>
<span class="source-line-no">568</span><span id="line-568"></span>
<span class="source-line-no">569</span><span id="line-569"></span>
<span class="source-line-no">570</span><span id="line-570"></span>
<span class="source-line-no">571</span><span id="line-571"></span>
<span class="source-line-no">572</span><span id="line-572">                        //String seqName = feature.seqname();</span>
<span class="source-line-no">573</span><span id="line-573">                        FeatureI startCodon = null;</span>
<span class="source-line-no">574</span><span id="line-574">                        FeatureI stopCodon = null;</span>
<span class="source-line-no">575</span><span id="line-575">                        Integer startCodonBegin = null;</span>
<span class="source-line-no">576</span><span id="line-576">                        Integer stopCodonEnd = null;</span>
<span class="source-line-no">577</span><span id="line-577">                        String startCodonName = "";</span>
<span class="source-line-no">578</span><span id="line-578">                        String stopCodonName = "";</span>
<span class="source-line-no">579</span><span id="line-579">                        FeatureList startCodonList = gene.selectByAttribute("Note", "initial-exon");</span>
<span class="source-line-no">580</span><span id="line-580">                        if (startCodonList != null &amp;&amp; startCodonList.size() &gt; 0) {</span>
<span class="source-line-no">581</span><span id="line-581">                                startCodon = startCodonList.get(0);</span>
<span class="source-line-no">582</span><span id="line-582">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">583</span><span id="line-583">                                        startCodonBegin = startCodon.location().bioEnd();</span>
<span class="source-line-no">584</span><span id="line-584">                                } else {</span>
<span class="source-line-no">585</span><span id="line-585">                                        startCodonBegin = startCodon.location().bioStart();</span>
<span class="source-line-no">586</span><span id="line-586">                                }</span>
<span class="source-line-no">587</span><span id="line-587">                                startCodonName = startCodon.getAttribute("ID");</span>
<span class="source-line-no">588</span><span id="line-588">                        }</span>
<span class="source-line-no">589</span><span id="line-589"></span>
<span class="source-line-no">590</span><span id="line-590">                        FeatureList stopCodonList = gene.selectByAttribute("Note", "final-exon");</span>
<span class="source-line-no">591</span><span id="line-591"></span>
<span class="source-line-no">592</span><span id="line-592">                        if (stopCodonList != null &amp;&amp; stopCodonList.size() &gt; 0) {</span>
<span class="source-line-no">593</span><span id="line-593">                                stopCodon = stopCodonList.get(0);</span>
<span class="source-line-no">594</span><span id="line-594">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">595</span><span id="line-595">                                        stopCodonEnd = stopCodon.location().bioStart();</span>
<span class="source-line-no">596</span><span id="line-596">                                } else {</span>
<span class="source-line-no">597</span><span id="line-597">                                        stopCodonEnd = stopCodon.location().bioEnd();</span>
<span class="source-line-no">598</span><span id="line-598">                                }</span>
<span class="source-line-no">599</span><span id="line-599">                                stopCodonName = stopCodon.getAttribute("ID");</span>
<span class="source-line-no">600</span><span id="line-600"></span>
<span class="source-line-no">601</span><span id="line-601">                        }</span>
<span class="source-line-no">602</span><span id="line-602"></span>
<span class="source-line-no">603</span><span id="line-603"></span>
<span class="source-line-no">604</span><span id="line-604"></span>
<span class="source-line-no">605</span><span id="line-605"></span>
<span class="source-line-no">606</span><span id="line-606">                        if (startCodonBegin == null) {</span>
<span class="source-line-no">607</span><span id="line-607">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">608</span><span id="line-608">                                        FeatureI firstFeature = cdsFeatures.get(0);</span>
<span class="source-line-no">609</span><span id="line-609"></span>
<span class="source-line-no">610</span><span id="line-610">                                        startCodonBegin = firstFeature.location().bioEnd();</span>
<span class="source-line-no">611</span><span id="line-611">                                } else {</span>
<span class="source-line-no">612</span><span id="line-612">                                        FeatureI firstFeature = cdsFeatures.get(0);</span>
<span class="source-line-no">613</span><span id="line-613"></span>
<span class="source-line-no">614</span><span id="line-614">                                        startCodonBegin = firstFeature.location().bioStart();</span>
<span class="source-line-no">615</span><span id="line-615">                                }</span>
<span class="source-line-no">616</span><span id="line-616">                        }</span>
<span class="source-line-no">617</span><span id="line-617"></span>
<span class="source-line-no">618</span><span id="line-618">                        if (stopCodonEnd == null) {</span>
<span class="source-line-no">619</span><span id="line-619">                                if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">620</span><span id="line-620">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</span>
<span class="source-line-no">621</span><span id="line-621">                                        stopCodonEnd = lastFeature.location().bioStart();</span>
<span class="source-line-no">622</span><span id="line-622">                                } else {</span>
<span class="source-line-no">623</span><span id="line-623">                                        FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</span>
<span class="source-line-no">624</span><span id="line-624">                                        stopCodonEnd = lastFeature.location().bioEnd();</span>
<span class="source-line-no">625</span><span id="line-625">                                }</span>
<span class="source-line-no">626</span><span id="line-626">                        }</span>
<span class="source-line-no">627</span><span id="line-627">                        //for gtf ordering can be strand based so first is last and last is first</span>
<span class="source-line-no">628</span><span id="line-628">                        if (startCodonBegin &gt; stopCodonEnd) {</span>
<span class="source-line-no">629</span><span id="line-629">                                int temp = startCodonBegin;</span>
<span class="source-line-no">630</span><span id="line-630">                                startCodonBegin = stopCodonEnd;</span>
<span class="source-line-no">631</span><span id="line-631">                                stopCodonEnd = temp;</span>
<span class="source-line-no">632</span><span id="line-632">                        }</span>
<span class="source-line-no">633</span><span id="line-633"></span>
<span class="source-line-no">634</span><span id="line-634"></span>
<span class="source-line-no">635</span><span id="line-635"></span>
<span class="source-line-no">636</span><span id="line-636">                        AccessionID transcriptAccessionID = new AccessionID(geneid);</span>
<span class="source-line-no">637</span><span id="line-637">                        if (geneSequence == null) {</span>
<span class="source-line-no">638</span><span id="line-638">                                geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</span>
<span class="source-line-no">639</span><span id="line-639">                                geneSequence.setSource(source);</span>
<span class="source-line-no">640</span><span id="line-640">                        }</span>
<span class="source-line-no">641</span><span id="line-641">                        /*</span>
<span class="source-line-no">642</span><span id="line-642">                        else {</span>
<span class="source-line-no">643</span><span id="line-643"></span>
<span class="source-line-no">644</span><span id="line-644">                                if (startCodonBegin &lt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">645</span><span id="line-645">                                        geneSequence.setBioBegin(startCodonBegin);</span>
<span class="source-line-no">646</span><span id="line-646">                                }</span>
<span class="source-line-no">647</span><span id="line-647">                                if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">648</span><span id="line-648">                                        geneSequence.setBioEnd(stopCodonEnd);</span>
<span class="source-line-no">649</span><span id="line-649">                                }</span>
<span class="source-line-no">650</span><span id="line-650">                        }</span>
<span class="source-line-no">651</span><span id="line-651">                        */</span>
<span class="source-line-no">652</span><span id="line-652">                        TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</span>
<span class="source-line-no">653</span><span id="line-653">                        if (startCodon != null) {</span>
<span class="source-line-no">654</span><span id="line-654">                                if (startCodonName == null || startCodonName.length() == 0) {</span>
<span class="source-line-no">655</span><span id="line-655">                                        startCodonName = geneid + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</span>
<span class="source-line-no">656</span><span id="line-656">                                }</span>
<span class="source-line-no">657</span><span id="line-657">                                transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</span>
<span class="source-line-no">658</span><span id="line-658">                        }</span>
<span class="source-line-no">659</span><span id="line-659">                        if (stopCodon != null) {</span>
<span class="source-line-no">660</span><span id="line-660">                                if (stopCodonName == null || stopCodonName.length() == 0) {</span>
<span class="source-line-no">661</span><span id="line-661">                                        stopCodonName = geneid + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</span>
<span class="source-line-no">662</span><span id="line-662">                                }</span>
<span class="source-line-no">663</span><span id="line-663">                                transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</span>
<span class="source-line-no">664</span><span id="line-664">                        }</span>
<span class="source-line-no">665</span><span id="line-665"></span>
<span class="source-line-no">666</span><span id="line-666">                        for (FeatureI cdsFeature : cdsFeatures) {</span>
<span class="source-line-no">667</span><span id="line-667">                                Feature cds = (Feature) cdsFeature;</span>
<span class="source-line-no">668</span><span id="line-668">                                String cdsName = cds.getAttribute("ID");</span>
<span class="source-line-no">669</span><span id="line-669">                                if (cdsName == null || cdsName.length() == 0) {</span>
<span class="source-line-no">670</span><span id="line-670">                                        cdsName = geneid + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</span>
<span class="source-line-no">671</span><span id="line-671">                                }</span>
<span class="source-line-no">672</span><span id="line-672">                                AccessionID cdsAccessionID = new AccessionID(cdsName);</span>
<span class="source-line-no">673</span><span id="line-673">                                //ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</span>
<span class="source-line-no">674</span><span id="line-674">                                transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), cds.frame());</span>
<span class="source-line-no">675</span><span id="line-675">                        }</span>
<span class="source-line-no">676</span><span id="line-676"></span>
<span class="source-line-no">677</span><span id="line-677">                }</span>
<span class="source-line-no">678</span><span id="line-678"></span>
<span class="source-line-no">679</span><span id="line-679">        }</span>
<span class="source-line-no">680</span><span id="line-680"></span>
<span class="source-line-no">681</span><span id="line-681">        static public Map&lt;String, ChromosomeSequence&gt; loadFastaAddGeneFeaturesFromGeneMarkGTF(File fastaSequenceFile, File gffFile) throws Exception {</span>
<span class="source-line-no">682</span><span id="line-682">                Map&lt;String, DNASequence&gt; dnaSequenceList = FastaReaderHelper.readFastaDNASequence(fastaSequenceFile);</span>
<span class="source-line-no">683</span><span id="line-683">                Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.getChromosomeSequenceFromDNASequence(dnaSequenceList);</span>
<span class="source-line-no">684</span><span id="line-684">                FeatureList listGenes = GeneMarkGTFReader.read(gffFile.getAbsolutePath());</span>
<span class="source-line-no">685</span><span id="line-685">                addGeneMarkGTFGeneFeatures(chromosomeSequenceList, listGenes);</span>
<span class="source-line-no">686</span><span id="line-686">                return chromosomeSequenceList;</span>
<span class="source-line-no">687</span><span id="line-687">        }</span>
<span class="source-line-no">688</span><span id="line-688"></span>
<span class="source-line-no">689</span><span id="line-689">        static public void addGeneMarkGTFGeneFeatures(Map&lt;String, ChromosomeSequence&gt; chromosomeSequenceList, FeatureList listGenes) throws Exception {</span>
<span class="source-line-no">690</span><span id="line-690">                Collection&lt;String&gt; geneIds = listGenes.attributeValues("gene_id");</span>
<span class="source-line-no">691</span><span id="line-691">                for (String geneid : geneIds) {</span>
<span class="source-line-no">692</span><span id="line-692">                        //       if(geneid.equals("45_g")){</span>
<span class="source-line-no">693</span><span id="line-693">                        //           int dummy =1;</span>
<span class="source-line-no">694</span><span id="line-694">                        //       }</span>
<span class="source-line-no">695</span><span id="line-695">                        FeatureList gene = listGenes.selectByAttribute("gene_id", geneid);</span>
<span class="source-line-no">696</span><span id="line-696">                        FeatureI geneFeature = gene.get(0);</span>
<span class="source-line-no">697</span><span id="line-697">                        ChromosomeSequence seq = chromosomeSequenceList.get(geneFeature.seqname());</span>
<span class="source-line-no">698</span><span id="line-698">                        AccessionID geneAccessionID = new AccessionID(geneid);</span>
<span class="source-line-no">699</span><span id="line-699">                        GeneSequence geneSequence = null;</span>
<span class="source-line-no">700</span><span id="line-700">                        Collection&lt;String&gt; transcriptids = gene.attributeValues("transcript_id");</span>
<span class="source-line-no">701</span><span id="line-701">                        for (String transcriptid : transcriptids) {</span>
<span class="source-line-no">702</span><span id="line-702">                                // get all the individual features (exons, CDS regions, etc.) of this gene</span>
<span class="source-line-no">703</span><span id="line-703"></span>
<span class="source-line-no">704</span><span id="line-704"></span>
<span class="source-line-no">705</span><span id="line-705">                                FeatureList transcriptFeature = listGenes.selectByAttribute("transcript_id", transcriptid);</span>
<span class="source-line-no">706</span><span id="line-706">                                // now select only the coding regions of this gene</span>
<span class="source-line-no">707</span><span id="line-707">                                FeatureList cdsFeatures = transcriptFeature.selectByType("CDS");</span>
<span class="source-line-no">708</span><span id="line-708">                                // sort them</span>
<span class="source-line-no">709</span><span id="line-709">                                cdsFeatures = cdsFeatures.sortByStart();</span>
<span class="source-line-no">710</span><span id="line-710"></span>
<span class="source-line-no">711</span><span id="line-711">                                FeatureI feature = cdsFeatures.get(0);</span>
<span class="source-line-no">712</span><span id="line-712">                                Strand strand = Strand.POSITIVE;</span>
<span class="source-line-no">713</span><span id="line-713"></span>
<span class="source-line-no">714</span><span id="line-714">                                if (feature.location().isNegative()) {</span>
<span class="source-line-no">715</span><span id="line-715">                                        strand = Strand.NEGATIVE;</span>
<span class="source-line-no">716</span><span id="line-716">                                }</span>
<span class="source-line-no">717</span><span id="line-717"></span>
<span class="source-line-no">718</span><span id="line-718">                                //String seqName = feature.seqname();</span>
<span class="source-line-no">719</span><span id="line-719">                                FeatureI startCodon = null;</span>
<span class="source-line-no">720</span><span id="line-720">                                FeatureI stopCodon = null;</span>
<span class="source-line-no">721</span><span id="line-721">                                Integer startCodonBegin = null;</span>
<span class="source-line-no">722</span><span id="line-722">                                Integer stopCodonEnd = null;</span>
<span class="source-line-no">723</span><span id="line-723">                                String startCodonName = "";</span>
<span class="source-line-no">724</span><span id="line-724">                                String stopCodonName = "";</span>
<span class="source-line-no">725</span><span id="line-725">                                FeatureList startCodonList = transcriptFeature.selectByType("start_codon");</span>
<span class="source-line-no">726</span><span id="line-726">                                if (startCodonList != null &amp;&amp; startCodonList.size() &gt; 0) {</span>
<span class="source-line-no">727</span><span id="line-727">                                        startCodon = startCodonList.get(0);</span>
<span class="source-line-no">728</span><span id="line-728">                                        if (strand == Strand.POSITIVE) {</span>
<span class="source-line-no">729</span><span id="line-729">                                                startCodonBegin = startCodon.location().bioStart();</span>
<span class="source-line-no">730</span><span id="line-730">                                        } else {</span>
<span class="source-line-no">731</span><span id="line-731">                                                startCodonBegin = startCodon.location().bioEnd();</span>
<span class="source-line-no">732</span><span id="line-732">                                        }</span>
<span class="source-line-no">733</span><span id="line-733">                                        startCodonName = startCodon.getAttribute("transcript_name");</span>
<span class="source-line-no">734</span><span id="line-734">                                }</span>
<span class="source-line-no">735</span><span id="line-735"></span>
<span class="source-line-no">736</span><span id="line-736">                                FeatureList stopCodonList = transcriptFeature.selectByType("stop_codon");</span>
<span class="source-line-no">737</span><span id="line-737"></span>
<span class="source-line-no">738</span><span id="line-738">                                if (stopCodonList != null &amp;&amp; stopCodonList.size() &gt; 0) {</span>
<span class="source-line-no">739</span><span id="line-739">                                        stopCodon = stopCodonList.get(0);</span>
<span class="source-line-no">740</span><span id="line-740">                                        if (strand == Strand.POSITIVE) {</span>
<span class="source-line-no">741</span><span id="line-741">                                                stopCodonEnd = stopCodon.location().bioEnd();</span>
<span class="source-line-no">742</span><span id="line-742">                                        } else {</span>
<span class="source-line-no">743</span><span id="line-743">                                                stopCodonEnd = stopCodon.location().bioStart();</span>
<span class="source-line-no">744</span><span id="line-744">                                        }</span>
<span class="source-line-no">745</span><span id="line-745"></span>
<span class="source-line-no">746</span><span id="line-746">                                        stopCodonName = stopCodon.getAttribute("transcript_name");</span>
<span class="source-line-no">747</span><span id="line-747"></span>
<span class="source-line-no">748</span><span id="line-748">                                }</span>
<span class="source-line-no">749</span><span id="line-749"></span>
<span class="source-line-no">750</span><span id="line-750"></span>
<span class="source-line-no">751</span><span id="line-751"></span>
<span class="source-line-no">752</span><span id="line-752"></span>
<span class="source-line-no">753</span><span id="line-753">                                if (startCodonBegin == null) {</span>
<span class="source-line-no">754</span><span id="line-754">                                        if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">755</span><span id="line-755">                                                FeatureI firstFeature = cdsFeatures.get(0);</span>
<span class="source-line-no">756</span><span id="line-756"></span>
<span class="source-line-no">757</span><span id="line-757">                                                startCodonBegin = firstFeature.location().bioEnd();</span>
<span class="source-line-no">758</span><span id="line-758">                                        } else {</span>
<span class="source-line-no">759</span><span id="line-759">                                                FeatureI firstFeature = cdsFeatures.get(0);</span>
<span class="source-line-no">760</span><span id="line-760"></span>
<span class="source-line-no">761</span><span id="line-761">                                                startCodonBegin = firstFeature.location().bioStart();</span>
<span class="source-line-no">762</span><span id="line-762">                                        }</span>
<span class="source-line-no">763</span><span id="line-763">                                }</span>
<span class="source-line-no">764</span><span id="line-764"></span>
<span class="source-line-no">765</span><span id="line-765">                                if (stopCodonEnd == null) {</span>
<span class="source-line-no">766</span><span id="line-766">                                        if (strand == Strand.NEGATIVE) {</span>
<span class="source-line-no">767</span><span id="line-767">                                                FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</span>
<span class="source-line-no">768</span><span id="line-768">                                                stopCodonEnd = lastFeature.location().bioStart();</span>
<span class="source-line-no">769</span><span id="line-769">                                        } else {</span>
<span class="source-line-no">770</span><span id="line-770">                                                FeatureI lastFeature = cdsFeatures.get(cdsFeatures.size() - 1);</span>
<span class="source-line-no">771</span><span id="line-771">                                                stopCodonEnd = lastFeature.location().bioEnd();</span>
<span class="source-line-no">772</span><span id="line-772">                                        }</span>
<span class="source-line-no">773</span><span id="line-773">                                }</span>
<span class="source-line-no">774</span><span id="line-774">                                //for gtf ordering can be strand based so first is last and last is first</span>
<span class="source-line-no">775</span><span id="line-775">                                if (startCodonBegin &gt; stopCodonEnd) {</span>
<span class="source-line-no">776</span><span id="line-776">                                        int temp = startCodonBegin;</span>
<span class="source-line-no">777</span><span id="line-777">                                        startCodonBegin = stopCodonEnd;</span>
<span class="source-line-no">778</span><span id="line-778">                                        stopCodonEnd = temp;</span>
<span class="source-line-no">779</span><span id="line-779">                                }</span>
<span class="source-line-no">780</span><span id="line-780"></span>
<span class="source-line-no">781</span><span id="line-781">                                AccessionID transcriptAccessionID = new AccessionID(transcriptid);</span>
<span class="source-line-no">782</span><span id="line-782">                                if (geneSequence == null) {</span>
<span class="source-line-no">783</span><span id="line-783">                                        geneSequence = seq.addGene(geneAccessionID, startCodonBegin, stopCodonEnd, strand);</span>
<span class="source-line-no">784</span><span id="line-784">                                        geneSequence.setSource(((Feature) feature).source());</span>
<span class="source-line-no">785</span><span id="line-785">                                } else {</span>
<span class="source-line-no">786</span><span id="line-786">                                        //if multiple transcripts for one gene make sure the gene is defined as the min and max start/end</span>
<span class="source-line-no">787</span><span id="line-787"></span>
<span class="source-line-no">788</span><span id="line-788">                                        if (startCodonBegin &lt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">789</span><span id="line-789">                                                geneSequence.setBioBegin(startCodonBegin);</span>
<span class="source-line-no">790</span><span id="line-790">                                        }</span>
<span class="source-line-no">791</span><span id="line-791">                                        if (stopCodonEnd &gt; geneSequence.getBioBegin()) {</span>
<span class="source-line-no">792</span><span id="line-792">                                                geneSequence.setBioEnd(stopCodonEnd);</span>
<span class="source-line-no">793</span><span id="line-793">                                        }</span>
<span class="source-line-no">794</span><span id="line-794"></span>
<span class="source-line-no">795</span><span id="line-795">                                }</span>
<span class="source-line-no">796</span><span id="line-796">                                TranscriptSequence transcriptSequence = geneSequence.addTranscript(transcriptAccessionID, startCodonBegin, stopCodonEnd);</span>
<span class="source-line-no">797</span><span id="line-797">                                if (startCodon != null) {</span>
<span class="source-line-no">798</span><span id="line-798">                                        if (startCodonName == null || startCodonName.length() == 0) {</span>
<span class="source-line-no">799</span><span id="line-799">                                                startCodonName = transcriptid + "-start_codon-" + startCodon.location().bioStart() + "-" + startCodon.location().bioEnd();</span>
<span class="source-line-no">800</span><span id="line-800">                                        }</span>
<span class="source-line-no">801</span><span id="line-801">                                        transcriptSequence.addStartCodonSequence(new AccessionID(startCodonName), startCodon.location().bioStart(), startCodon.location().bioEnd());</span>
<span class="source-line-no">802</span><span id="line-802">                                }</span>
<span class="source-line-no">803</span><span id="line-803">                                if (stopCodon != null) {</span>
<span class="source-line-no">804</span><span id="line-804">                                        if (stopCodonName == null || stopCodonName.length() == 0) {</span>
<span class="source-line-no">805</span><span id="line-805">                                                stopCodonName = transcriptid + "-stop_codon-" + stopCodon.location().bioStart() + "-" + stopCodon.location().bioEnd();</span>
<span class="source-line-no">806</span><span id="line-806">                                        }</span>
<span class="source-line-no">807</span><span id="line-807">                                        transcriptSequence.addStopCodonSequence(new AccessionID(stopCodonName), stopCodon.location().bioStart(), stopCodon.location().bioEnd());</span>
<span class="source-line-no">808</span><span id="line-808">                                }</span>
<span class="source-line-no">809</span><span id="line-809"></span>
<span class="source-line-no">810</span><span id="line-810">                                for (FeatureI cdsFeature : cdsFeatures) {</span>
<span class="source-line-no">811</span><span id="line-811">                                        Feature cds = (Feature) cdsFeature;</span>
<span class="source-line-no">812</span><span id="line-812">                                        // for genemark it appears frame of 2 =1 and frame of 1 = 2</span>
<span class="source-line-no">813</span><span id="line-813">                                        // doesn't matter when you string cds regions together as one block</span>
<span class="source-line-no">814</span><span id="line-814">                                        // but does make a difference when you try to make a protein sequence for each CDS region where</span>
<span class="source-line-no">815</span><span id="line-815">                                        // you give up or borrow based on the frame value</span>
<span class="source-line-no">816</span><span id="line-816">                                        // compared with gff like files and docs for geneid and glimmer where geneid and glimmer both do it the same</span>
<span class="source-line-no">817</span><span id="line-817">                                        // way that appears to match the gff3 docs.</span>
<span class="source-line-no">818</span><span id="line-818">                                        int frame = cds.frame();</span>
<span class="source-line-no">819</span><span id="line-819">                                        if (frame == 1) {</span>
<span class="source-line-no">820</span><span id="line-820">                                                frame = 2;</span>
<span class="source-line-no">821</span><span id="line-821">                                        } else if (frame == 2) {</span>
<span class="source-line-no">822</span><span id="line-822">                                                frame = 1;</span>
<span class="source-line-no">823</span><span id="line-823">                                        } else {</span>
<span class="source-line-no">824</span><span id="line-824">                                                frame = 0;</span>
<span class="source-line-no">825</span><span id="line-825">                                        }</span>
<span class="source-line-no">826</span><span id="line-826">                                        String cdsName = cds.getAttribute("transcript_name");</span>
<span class="source-line-no">827</span><span id="line-827">                                        if (cdsName == null || cdsName.length() == 0) {</span>
<span class="source-line-no">828</span><span id="line-828">                                                cdsName = transcriptid + "-cds-" + cds.location().bioStart() + "-" + cds.location().bioEnd();</span>
<span class="source-line-no">829</span><span id="line-829">                                        }</span>
<span class="source-line-no">830</span><span id="line-830">                                        AccessionID cdsAccessionID = new AccessionID(cdsName);</span>
<span class="source-line-no">831</span><span id="line-831">                                        //ExonSequence exonSequence = geneSequence.addExon(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd());</span>
<span class="source-line-no">832</span><span id="line-832">                                        transcriptSequence.addCDS(cdsAccessionID, cdsFeature.location().bioStart(), cdsFeature.location().bioEnd(), frame);</span>
<span class="source-line-no">833</span><span id="line-833">                                }</span>
<span class="source-line-no">834</span><span id="line-834">                        }</span>
<span class="source-line-no">835</span><span id="line-835">                }</span>
<span class="source-line-no">836</span><span id="line-836"></span>
<span class="source-line-no">837</span><span id="line-837">        }</span>
<span class="source-line-no">838</span><span id="line-838"></span>
<span class="source-line-no">839</span><span id="line-839">        static public LinkedHashMap&lt;String, ProteinSequence&gt; getProteinSequences(Collection&lt;ChromosomeSequence&gt; chromosomeSequences) throws Exception {</span>
<span class="source-line-no">840</span><span id="line-840">                LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequenceHashMap = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">841</span><span id="line-841">                for (ChromosomeSequence dnaSequence : chromosomeSequences) {</span>
<span class="source-line-no">842</span><span id="line-842">                        for (GeneSequence geneSequence : dnaSequence.getGeneSequences().values()) {</span>
<span class="source-line-no">843</span><span id="line-843">                                for (TranscriptSequence transcriptSequence : geneSequence.getTranscripts().values()) {</span>
<span class="source-line-no">844</span><span id="line-844">                                        //TODO remove?</span>
<span class="source-line-no">845</span><span id="line-845">//                    DNASequence dnaCodingSequence = transcriptSequence.getDNACodingSequence();</span>
<span class="source-line-no">846</span><span id="line-846">//                    logger.info("CDS={}", dnaCodingSequence.getSequenceAsString());</span>
<span class="source-line-no">847</span><span id="line-847"></span>
<span class="source-line-no">848</span><span id="line-848">                                        try {</span>
<span class="source-line-no">849</span><span id="line-849">                                                ProteinSequence proteinSequence = transcriptSequence.getProteinSequence();</span>
<span class="source-line-no">850</span><span id="line-850"></span>
<span class="source-line-no">851</span><span id="line-851">//                        logger.info("{} {}", proteinSequence.getAccession().getID(), proteinSequence);</span>
<span class="source-line-no">852</span><span id="line-852">                                                if (proteinSequenceHashMap.containsKey(proteinSequence.getAccession().getID())) {</span>
<span class="source-line-no">853</span><span id="line-853">                                                        throw new Exception("Duplicate protein sequence id=" + proteinSequence.getAccession().getID() + " found at Gene id=" + geneSequence.getAccession().getID());</span>
<span class="source-line-no">854</span><span id="line-854">                                                } else {</span>
<span class="source-line-no">855</span><span id="line-855">                                                        proteinSequenceHashMap.put(proteinSequence.getAccession().getID(), proteinSequence);</span>
<span class="source-line-no">856</span><span id="line-856">                                                }</span>
<span class="source-line-no">857</span><span id="line-857">                                        } catch (Exception e) {</span>
<span class="source-line-no">858</span><span id="line-858">                                                logger.error("Exception: ", e);</span>
<span class="source-line-no">859</span><span id="line-859">                                        }</span>
<span class="source-line-no">860</span><span id="line-860"></span>
<span class="source-line-no">861</span><span id="line-861">                                }</span>
<span class="source-line-no">862</span><span id="line-862"></span>
<span class="source-line-no">863</span><span id="line-863">                        }</span>
<span class="source-line-no">864</span><span id="line-864">                }</span>
<span class="source-line-no">865</span><span id="line-865">                return proteinSequenceHashMap;</span>
<span class="source-line-no">866</span><span id="line-866">        }</span>
<span class="source-line-no">867</span><span id="line-867"></span>
<span class="source-line-no">868</span><span id="line-868">        static public LinkedHashMap&lt;String, GeneSequence&gt; getGeneSequences(Collection&lt;ChromosomeSequence&gt; chromosomeSequences) throws Exception {</span>
<span class="source-line-no">869</span><span id="line-869">                LinkedHashMap&lt;String, GeneSequence&gt; geneSequenceHashMap = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">870</span><span id="line-870">                for (ChromosomeSequence chromosomeSequence : chromosomeSequences) {</span>
<span class="source-line-no">871</span><span id="line-871">                        for (GeneSequence geneSequence : chromosomeSequence.getGeneSequences().values()) {</span>
<span class="source-line-no">872</span><span id="line-872">                                geneSequenceHashMap.put(geneSequence.getAccession().getID(), geneSequence);</span>
<span class="source-line-no">873</span><span id="line-873">                        }</span>
<span class="source-line-no">874</span><span id="line-874">                }</span>
<span class="source-line-no">875</span><span id="line-875"></span>
<span class="source-line-no">876</span><span id="line-876">                return geneSequenceHashMap;</span>
<span class="source-line-no">877</span><span id="line-877">        }</span>
<span class="source-line-no">878</span><span id="line-878"></span>
<span class="source-line-no">879</span><span id="line-879">        public static void main(String[] args) throws Exception {</span>
<span class="source-line-no">880</span><span id="line-880">/*        if (false) {</span>
<span class="source-line-no">881</span><span id="line-881">                        LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGeneMarkGTF(new File("Scaffolds.fna"), new File("genemark_hmm.gtf"));</span>
<span class="source-line-no">882</span><span id="line-882">                        LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequenceList = GeneFeatureHelper.getProteinSequences(chromosomeSequenceList.values());</span>
<span class="source-line-no">883</span><span id="line-883">                }</span>
<span class="source-line-no">884</span><span id="line-884"></span>
<span class="source-line-no">885</span><span id="line-885">                if (false) {</span>
<span class="source-line-no">886</span><span id="line-886">                        LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceList = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGlimmerGFF3(new File("Scaffolds.fna"), new File("glimmerhmm.gff"));</span>
<span class="source-line-no">887</span><span id="line-887">                        LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequenceList = GeneFeatureHelper.getProteinSequences(chromosomeSequenceList.values());</span>
<span class="source-line-no">888</span><span id="line-888">                        //  for (ProteinSequence proteinSequence : proteinSequenceList.values()) {</span>
<span class="source-line-no">889</span><span id="line-889">                        //      logger.info(proteinSequence.getAccession().getID() + " " + proteinSequence);</span>
<span class="source-line-no">890</span><span id="line-890">                        //  }</span>
<span class="source-line-no">891</span><span id="line-891">                        FastaWriterHelper.writeProteinSequence(new File("predicted_glimmer.faa"), proteinSequenceList.values());</span>
<span class="source-line-no">892</span><span id="line-892"></span>
<span class="source-line-no">893</span><span id="line-893">                }</span>
<span class="source-line-no">894</span><span id="line-894">                if (false) {</span>
<span class="source-line-no">895</span><span id="line-895">                        GeneFeatureHelper.outputFastaSequenceLengthGFF3(new File("Scaffolds.fna"), new File("scaffolds.gff3"));</span>
<span class="source-line-no">896</span><span id="line-896">                }</span>
<span class="source-line-no">897</span><span id="line-897"></span>
<span class="source-line-no">898</span><span id="line-898"> */</span>
<span class="source-line-no">899</span><span id="line-899"></span>
<span class="source-line-no">900</span><span id="line-900">                try {</span>
<span class="source-line-no">901</span><span id="line-901"></span>
<span class="source-line-no">902</span><span id="line-902">                        if (true) {</span>
<span class="source-line-no">903</span><span id="line-903">                                //File fastaSequenceFile = new File("Scaffolds.fna");</span>
<span class="source-line-no">904</span><span id="line-904">                                //File gff3File = new File("geneid-v6.gff3");</span>
<span class="source-line-no">905</span><span id="line-905">                                //LinkedHashMap&lt;String, ChromosomeSequence&gt; chromosomeSequenceHashMap = GeneFeatureHelper.loadFastaAddGeneFeaturesFromGmodGFF3(fastaSequenceFile, gff3File,true);</span>
<span class="source-line-no">906</span><span id="line-906">                        }</span>
<span class="source-line-no">907</span><span id="line-907"></span>
<span class="source-line-no">908</span><span id="line-908">                } catch (Exception e) {</span>
<span class="source-line-no">909</span><span id="line-909">                        logger.error("Exception: ", e);</span>
<span class="source-line-no">910</span><span id="line-910">                }</span>
<span class="source-line-no">911</span><span id="line-911"></span>
<span class="source-line-no">912</span><span id="line-912">        }</span>
<span class="source-line-no">913</span><span id="line-913">}</span>




























































</pre>
</div>
</main>
</body>
</html>
