<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21"></a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.bio.program.ssbind;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import java.util.ArrayList;</a>
<span class="sourceLineNo">025</span><a id="line.25">import java.util.HashMap;</a>
<span class="sourceLineNo">026</span><a id="line.26">import java.util.List;</a>
<span class="sourceLineNo">027</span><a id="line.27">import java.util.Map;</a>
<span class="sourceLineNo">028</span><a id="line.28"></a>
<span class="sourceLineNo">029</span><a id="line.29">import org.biojava.bio.BioException;</a>
<span class="sourceLineNo">030</span><a id="line.30">import org.biojava.bio.alignment.Alignment;</a>
<span class="sourceLineNo">031</span><a id="line.31">import org.biojava.bio.alignment.SimpleAlignment;</a>
<span class="sourceLineNo">032</span><a id="line.32">import org.biojava.bio.search.SearchContentHandler;</a>
<span class="sourceLineNo">033</span><a id="line.33">import org.biojava.bio.seq.Sequence;</a>
<span class="sourceLineNo">034</span><a id="line.34">import org.biojava.bio.seq.StrandedFeature;</a>
<span class="sourceLineNo">035</span><a id="line.35">import org.biojava.bio.seq.StrandedFeature.Strand;</a>
<span class="sourceLineNo">036</span><a id="line.36">import org.biojava.bio.seq.homol.Homology;</a>
<span class="sourceLineNo">037</span><a id="line.37">import org.biojava.bio.seq.homol.HomologyFeature;</a>
<span class="sourceLineNo">038</span><a id="line.38">import org.biojava.bio.seq.homol.SimpleHomology;</a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.bio.seq.io.SymbolTokenization;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.biojava.bio.symbol.FiniteAlphabet;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.biojava.bio.symbol.RangeLocation;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.biojava.bio.symbol.SimpleSymbolList;</a>
<span class="sourceLineNo">043</span><a id="line.43">import org.biojava.utils.ChangeVetoException;</a>
<span class="sourceLineNo">044</span><a id="line.44"></a>
<span class="sourceLineNo">045</span><a id="line.45">/**</a>
<span class="sourceLineNo">046</span><a id="line.46"> * &lt;p&gt;&lt;code&gt;BlastLikeHomologyBuilder&lt;/code&gt; populates a</a>
<span class="sourceLineNo">047</span><a id="line.47"> * &lt;code&gt;List&lt;/code&gt; with &lt;code&gt;Homology&lt;/code&gt; instances created from</a>
<span class="sourceLineNo">048</span><a id="line.48"> * SAX events supplied via a &lt;code&gt;SeqSimilarityAdapter&lt;/code&gt;. The</a>
<span class="sourceLineNo">049</span><a id="line.49"> * SAX events should describe elements conforming to the BioJava</a>
<span class="sourceLineNo">050</span><a id="line.50"> * BlastLikeDataSetCollection DTD. Suitable sources are</a>
<span class="sourceLineNo">051</span><a id="line.51"> * &lt;code&gt;BlastLikeSAXParser&lt;/code&gt; or</a>
<span class="sourceLineNo">052</span><a id="line.52"> * &lt;code&gt;FastaSAXParser&lt;/code&gt;. Annotated &lt;code&gt;ViewSequence&lt;/code&gt;s</a>
<span class="sourceLineNo">053</span><a id="line.53"> * wrapping both query and subject sequences are created and populated</a>
<span class="sourceLineNo">054</span><a id="line.54"> * with &lt;code&gt;HomologyFeature&lt;/code&gt;s. See the documentation of</a>
<span class="sourceLineNo">055</span><a id="line.55"> * &lt;code&gt;Homology&lt;/code&gt; and &lt;code&gt;HomologyFeature&lt;/code&gt;.&lt;/p&gt;</a>
<span class="sourceLineNo">056</span><a id="line.56"> *</a>
<span class="sourceLineNo">057</span><a id="line.57"> * &lt;p&gt;As &lt;code&gt;SimpleHomologyFeature&lt;/code&gt;s are created on views of</a>
<span class="sourceLineNo">058</span><a id="line.58"> * the query and subject sequences, both query and subject should be</a>
<span class="sourceLineNo">059</span><a id="line.59"> * nucleotide sequences (&lt;code&gt;SimpleHomologyFeature&lt;/code&gt; extends</a>
<span class="sourceLineNo">060</span><a id="line.60"> * &lt;code&gt;StrandedFeature&lt;/code&gt;.). This limits the searches currently</a>
<span class="sourceLineNo">061</span><a id="line.61"> * handled to BLASTN, TBLASTX and Fasta DNA.&lt;/p&gt;</a>
<span class="sourceLineNo">062</span><a id="line.62"> *</a>
<span class="sourceLineNo">063</span><a id="line.63"> * @author Keith James</a>
<span class="sourceLineNo">064</span><a id="line.64"> * @author Greg Cox</a>
<span class="sourceLineNo">065</span><a id="line.65"> * @since 1.2</a>
<span class="sourceLineNo">066</span><a id="line.66"> */</a>
<span class="sourceLineNo">067</span><a id="line.67">public class BlastLikeHomologyBuilder extends ViewSequenceFactory</a>
<span class="sourceLineNo">068</span><a id="line.68">    implements SearchContentHandler</a>
<span class="sourceLineNo">069</span><a id="line.69">{</a>
<span class="sourceLineNo">070</span><a id="line.70">    /**</a>
<span class="sourceLineNo">071</span><a id="line.71">     * &lt;code&gt;HOMOLOGY_FEATURE_TYPE&lt;/code&gt; is the type String used by</a>
<span class="sourceLineNo">072</span><a id="line.72">     * &lt;code&gt;BlastLikeHomologyBuilder&lt;/code&gt; when creating</a>
<span class="sourceLineNo">073</span><a id="line.73">     * &lt;code&gt;HomologyFeature&lt;/code&gt;s. This is the String which is</a>
<span class="sourceLineNo">074</span><a id="line.74">     * returned when an &lt;code&gt;HomologyFeature&lt;/code&gt;'s</a>
<span class="sourceLineNo">075</span><a id="line.75">     * &lt;code&gt;getType()&lt;/code&gt; method is called.</a>
<span class="sourceLineNo">076</span><a id="line.76">     */</a>
<span class="sourceLineNo">077</span><a id="line.77">    public static final String HOMOLOGY_FEATURE_TYPE = "homology";</a>
<span class="sourceLineNo">078</span><a id="line.78"></a>
<span class="sourceLineNo">079</span><a id="line.79">    // Identifiers for query and database</a>
<span class="sourceLineNo">080</span><a id="line.80">    private String queryID;</a>
<span class="sourceLineNo">081</span><a id="line.81"></a>
<span class="sourceLineNo">082</span><a id="line.82">    // Data holders for search result properties</a>
<span class="sourceLineNo">083</span><a id="line.83">    private Map resultData;</a>
<span class="sourceLineNo">084</span><a id="line.84">    private Map hitData;</a>
<span class="sourceLineNo">085</span><a id="line.85">    private Map subHitData;</a>
<span class="sourceLineNo">086</span><a id="line.86"></a>
<span class="sourceLineNo">087</span><a id="line.87">    private SymbolTokenization tokenParser;</a>
<span class="sourceLineNo">088</span><a id="line.88">    private StringBuffer       tokenBuffer;</a>
<span class="sourceLineNo">089</span><a id="line.89"></a>
<span class="sourceLineNo">090</span><a id="line.90">    // List for holding homologies from current search. There may be</a>
<span class="sourceLineNo">091</span><a id="line.91">    // more than one search result in a stream</a>
<span class="sourceLineNo">092</span><a id="line.92">    private List homologies;</a>
<span class="sourceLineNo">093</span><a id="line.93">    // Flag indicating whether there are more results in the stream</a>
<span class="sourceLineNo">094</span><a id="line.94">    private boolean moreSearchesAvailable = false;</a>
<span class="sourceLineNo">095</span><a id="line.95">    // List to accept homologies from all results in the stream</a>
<span class="sourceLineNo">096</span><a id="line.96">    private List target;</a>
<span class="sourceLineNo">097</span><a id="line.97"></a>
<span class="sourceLineNo">098</span><a id="line.98">    /**</a>
<span class="sourceLineNo">099</span><a id="line.99">     * Creates a new &lt;code&gt;BlastLikeHomologyBuilder&lt;/code&gt; which will</a>
<span class="sourceLineNo">100</span><a id="line.100">     * instantiate &lt;code&gt;Homology&lt;/code&gt; objects into the</a>
<span class="sourceLineNo">101</span><a id="line.101">     * &lt;code&gt;List&lt;/code&gt; target.</a>
<span class="sourceLineNo">102</span><a id="line.102">     *</a>
<span class="sourceLineNo">103</span><a id="line.103">     * @param target a &lt;code&gt;List&lt;/code&gt;.</a>
<span class="sourceLineNo">104</span><a id="line.104">     */</a>
<span class="sourceLineNo">105</span><a id="line.105">    public BlastLikeHomologyBuilder(List target)</a>
<span class="sourceLineNo">106</span><a id="line.106">    {</a>
<span class="sourceLineNo">107</span><a id="line.107">        this.target = target;</a>
<span class="sourceLineNo">108</span><a id="line.108"></a>
<span class="sourceLineNo">109</span><a id="line.109">        resultData       = new HashMap();</a>
<span class="sourceLineNo">110</span><a id="line.110">        hitData          = new HashMap();</a>
<span class="sourceLineNo">111</span><a id="line.111">        subHitData       = new HashMap();</a>
<span class="sourceLineNo">112</span><a id="line.112">        queryViewCache   = new HashMap();</a>
<span class="sourceLineNo">113</span><a id="line.113">        subjectViewCache = new HashMap();</a>
<span class="sourceLineNo">114</span><a id="line.114">        tokenBuffer      = new StringBuffer(1024);</a>
<span class="sourceLineNo">115</span><a id="line.115">    }</a>
<span class="sourceLineNo">116</span><a id="line.116"></a>
<span class="sourceLineNo">117</span><a id="line.117">    public void setQueryID(String queryID)</a>
<span class="sourceLineNo">118</span><a id="line.118">    {</a>
<span class="sourceLineNo">119</span><a id="line.119">        this.queryID = queryID;</a>
<span class="sourceLineNo">120</span><a id="line.120">    }</a>
<span class="sourceLineNo">121</span><a id="line.121"></a>
<span class="sourceLineNo">122</span><a id="line.122">    public boolean getMoreSearches()</a>
<span class="sourceLineNo">123</span><a id="line.123">    {</a>
<span class="sourceLineNo">124</span><a id="line.124">        return moreSearchesAvailable;</a>
<span class="sourceLineNo">125</span><a id="line.125">    }</a>
<span class="sourceLineNo">126</span><a id="line.126"></a>
<span class="sourceLineNo">127</span><a id="line.127">    public void setMoreSearches(boolean value)</a>
<span class="sourceLineNo">128</span><a id="line.128">    {</a>
<span class="sourceLineNo">129</span><a id="line.129">        moreSearchesAvailable = value;</a>
<span class="sourceLineNo">130</span><a id="line.130">    }</a>
<span class="sourceLineNo">131</span><a id="line.131"></a>
<span class="sourceLineNo">132</span><a id="line.132">    public void startSearch()</a>
<span class="sourceLineNo">133</span><a id="line.133">    {</a>
<span class="sourceLineNo">134</span><a id="line.134">        subjectViewCache.clear();</a>
<span class="sourceLineNo">135</span><a id="line.135">        homologies = new ArrayList();</a>
<span class="sourceLineNo">136</span><a id="line.136">    }</a>
<span class="sourceLineNo">137</span><a id="line.137"></a>
<span class="sourceLineNo">138</span><a id="line.138">    public void endSearch()</a>
<span class="sourceLineNo">139</span><a id="line.139">    {</a>
<span class="sourceLineNo">140</span><a id="line.140">        target.addAll(homologies);</a>
<span class="sourceLineNo">141</span><a id="line.141">    }</a>
<span class="sourceLineNo">142</span><a id="line.142"></a>
<span class="sourceLineNo">143</span><a id="line.143">    public void startHeader()</a>
<span class="sourceLineNo">144</span><a id="line.144">    {</a>
<span class="sourceLineNo">145</span><a id="line.145">        resultData.clear();</a>
<span class="sourceLineNo">146</span><a id="line.146">    }</a>
<span class="sourceLineNo">147</span><a id="line.147"></a>
<span class="sourceLineNo">148</span><a id="line.148">    public void endHeader() { }</a>
<span class="sourceLineNo">149</span><a id="line.149"></a>
<span class="sourceLineNo">150</span><a id="line.150">    public void startHit()</a>
<span class="sourceLineNo">151</span><a id="line.151">    {</a>
<span class="sourceLineNo">152</span><a id="line.152">        hitData.clear();</a>
<span class="sourceLineNo">153</span><a id="line.153">        subHitData.clear();</a>
<span class="sourceLineNo">154</span><a id="line.154">    }</a>
<span class="sourceLineNo">155</span><a id="line.155"></a>
<span class="sourceLineNo">156</span><a id="line.156">    public void endHit() { }</a>
<span class="sourceLineNo">157</span><a id="line.157"></a>
<span class="sourceLineNo">158</span><a id="line.158">    public void startSubHit() { }</a>
<span class="sourceLineNo">159</span><a id="line.159"></a>
<span class="sourceLineNo">160</span><a id="line.160">    public void endSubHit()</a>
<span class="sourceLineNo">161</span><a id="line.161">    {</a>
<span class="sourceLineNo">162</span><a id="line.162">        try</a>
<span class="sourceLineNo">163</span><a id="line.163">        {</a>
<span class="sourceLineNo">164</span><a id="line.164">            homologies.add(makeHomology());</a>
<span class="sourceLineNo">165</span><a id="line.165">        }</a>
<span class="sourceLineNo">166</span><a id="line.166">        catch (BioException be)</a>
<span class="sourceLineNo">167</span><a id="line.167">        {</a>
<span class="sourceLineNo">168</span><a id="line.168">            System.err.println("Failed to build Homology:");</a>
<span class="sourceLineNo">169</span><a id="line.169">            be.printStackTrace();</a>
<span class="sourceLineNo">170</span><a id="line.170">        }</a>
<span class="sourceLineNo">171</span><a id="line.171">    }</a>
<span class="sourceLineNo">172</span><a id="line.172"></a>
<span class="sourceLineNo">173</span><a id="line.173">    public void addSearchProperty(Object key, Object value)</a>
<span class="sourceLineNo">174</span><a id="line.174">    {</a>
<span class="sourceLineNo">175</span><a id="line.175">        resultData.put(key, value);</a>
<span class="sourceLineNo">176</span><a id="line.176">    }</a>
<span class="sourceLineNo">177</span><a id="line.177"></a>
<span class="sourceLineNo">178</span><a id="line.178">    public void addHitProperty(Object key, Object value)</a>
<span class="sourceLineNo">179</span><a id="line.179">    {</a>
<span class="sourceLineNo">180</span><a id="line.180">        hitData.put(key, value);</a>
<span class="sourceLineNo">181</span><a id="line.181">    }</a>
<span class="sourceLineNo">182</span><a id="line.182"></a>
<span class="sourceLineNo">183</span><a id="line.183">    public void addSubHitProperty(Object key, Object value)</a>
<span class="sourceLineNo">184</span><a id="line.184">    {</a>
<span class="sourceLineNo">185</span><a id="line.185">        subHitData.put(key, value);</a>
<span class="sourceLineNo">186</span><a id="line.186">    }</a>
<span class="sourceLineNo">187</span><a id="line.187"></a>
<span class="sourceLineNo">188</span><a id="line.188">    /**</a>
<span class="sourceLineNo">189</span><a id="line.189">     * &lt;code&gt;makeHomology&lt;/code&gt; creates a new</a>
<span class="sourceLineNo">190</span><a id="line.190">     * &lt;code&gt;SimpleHomology&lt;/code&gt; describing the similarity between</a>
<span class="sourceLineNo">191</span><a id="line.191">     * the query and subject sequences. The</a>
<span class="sourceLineNo">192</span><a id="line.192">     * &lt;code&gt;HomologyFeatures&lt;/code&gt; created are added to</a>
<span class="sourceLineNo">193</span><a id="line.193">     * &lt;code&gt;ViewSequence&lt;/code&gt;s wrapping the query and subject</a>
<span class="sourceLineNo">194</span><a id="line.194">     * sequences.</a>
<span class="sourceLineNo">195</span><a id="line.195">     *</a>
<span class="sourceLineNo">196</span><a id="line.196">     * @return an &lt;code&gt;Homology&lt;/code&gt;.</a>
<span class="sourceLineNo">197</span><a id="line.197">     *</a>
<span class="sourceLineNo">198</span><a id="line.198">     * @exception BioException if an error occurs.</a>
<span class="sourceLineNo">199</span><a id="line.199">     */</a>
<span class="sourceLineNo">200</span><a id="line.200">    private Homology makeHomology() throws BioException</a>
<span class="sourceLineNo">201</span><a id="line.201">    {</a>
<span class="sourceLineNo">202</span><a id="line.202">        subHitData.putAll(resultData);</a>
<span class="sourceLineNo">203</span><a id="line.203">        subHitData.putAll(hitData);</a>
<span class="sourceLineNo">204</span><a id="line.204"></a>
<span class="sourceLineNo">205</span><a id="line.205">        if (tokenParser == null)</a>
<span class="sourceLineNo">206</span><a id="line.206">        {</a>
<span class="sourceLineNo">207</span><a id="line.207">            String identifier;</a>
<span class="sourceLineNo">208</span><a id="line.208">            // Try explicit sequence type first</a>
<span class="sourceLineNo">209</span><a id="line.209">            if (subHitData.containsKey("hitSequenceType"))</a>
<span class="sourceLineNo">210</span><a id="line.210">                identifier = (String) subHitData.get("hitSequenceType");</a>
<span class="sourceLineNo">211</span><a id="line.211">            // Otherwise try to resolve from the program name (only</a>
<span class="sourceLineNo">212</span><a id="line.212">            // works for Blast)</a>
<span class="sourceLineNo">213</span><a id="line.213">            else if (subHitData.containsKey("program"))</a>
<span class="sourceLineNo">214</span><a id="line.214">                identifier = (String) subHitData.get("program");</a>
<span class="sourceLineNo">215</span><a id="line.215">            else</a>
<span class="sourceLineNo">216</span><a id="line.216">                throw new BioException("Failed to determine sequence type");</a>
<span class="sourceLineNo">217</span><a id="line.217"></a>
<span class="sourceLineNo">218</span><a id="line.218">            FiniteAlphabet alpha = AlphabetResolver.resolveAlphabet(identifier);</a>
<span class="sourceLineNo">219</span><a id="line.219">            tokenParser = alpha.getTokenization("token");</a>
<span class="sourceLineNo">220</span><a id="line.220">        }</a>
<span class="sourceLineNo">221</span><a id="line.221"></a>
<span class="sourceLineNo">222</span><a id="line.222">        Strand qStrand = StrandedFeature.POSITIVE;</a>
<span class="sourceLineNo">223</span><a id="line.223">        Strand sStrand = StrandedFeature.POSITIVE;</a>
<span class="sourceLineNo">224</span><a id="line.224"></a>
<span class="sourceLineNo">225</span><a id="line.225">        // In cases where an explicit strand is given (FASTA DNA, BLASTN)</a>
<span class="sourceLineNo">226</span><a id="line.226">        if (subHitData.containsKey("queryStrand") &amp;&amp;</a>
<span class="sourceLineNo">227</span><a id="line.227">            subHitData.get("queryStrand").equals("minus"))</a>
<span class="sourceLineNo">228</span><a id="line.228">            qStrand = StrandedFeature.NEGATIVE;</a>
<span class="sourceLineNo">229</span><a id="line.229"></a>
<span class="sourceLineNo">230</span><a id="line.230">        if (subHitData.containsKey("subjectStrand") &amp;&amp;</a>
<span class="sourceLineNo">231</span><a id="line.231">            subHitData.get("subjectStrand").equals("minus"))</a>
<span class="sourceLineNo">232</span><a id="line.232">            sStrand = StrandedFeature.NEGATIVE;</a>
<span class="sourceLineNo">233</span><a id="line.233"></a>
<span class="sourceLineNo">234</span><a id="line.234">        // In cases where a frame is given as this contains strand</a>
<span class="sourceLineNo">235</span><a id="line.235">        // information (TBLASTN for hit, TBLASTX for both query and</a>
<span class="sourceLineNo">236</span><a id="line.236">        // hit)</a>
<span class="sourceLineNo">237</span><a id="line.237">        if (subHitData.containsKey("queryFrame") &amp;&amp;</a>
<span class="sourceLineNo">238</span><a id="line.238">            ((String) subHitData.get("queryFrame")).startsWith("minus"))</a>
<span class="sourceLineNo">239</span><a id="line.239">            qStrand = StrandedFeature.NEGATIVE;</a>
<span class="sourceLineNo">240</span><a id="line.240"></a>
<span class="sourceLineNo">241</span><a id="line.241">        if (subHitData.containsKey("subjectFrame") &amp;&amp;</a>
<span class="sourceLineNo">242</span><a id="line.242">            ((String) subHitData.get("subjectFrame")).startsWith("minus"))</a>
<span class="sourceLineNo">243</span><a id="line.243">            sStrand = StrandedFeature.NEGATIVE;</a>
<span class="sourceLineNo">244</span><a id="line.244"></a>
<span class="sourceLineNo">245</span><a id="line.245">        int qStart = Integer.parseInt((String) subHitData.get("querySequenceStart"));</a>
<span class="sourceLineNo">246</span><a id="line.246">        int   qEnd = Integer.parseInt((String) subHitData.get("querySequenceEnd"));</a>
<span class="sourceLineNo">247</span><a id="line.247">        int sStart = Integer.parseInt((String) subHitData.get("subjectSequenceStart"));</a>
<span class="sourceLineNo">248</span><a id="line.248">        int   sEnd = Integer.parseInt((String) subHitData.get("subjectSequenceEnd"));</a>
<span class="sourceLineNo">249</span><a id="line.249"></a>
<span class="sourceLineNo">250</span><a id="line.250">        // The start/end coordinates from BioJava XML don't follow the</a>
<span class="sourceLineNo">251</span><a id="line.251">        // BioJava paradigm of start &lt; end, with orientation given by</a>
<span class="sourceLineNo">252</span><a id="line.252">        // the strand property. Rather, they present start/end as</a>
<span class="sourceLineNo">253</span><a id="line.253">        // displayed in BLAST output, with the coordinates being</a>
<span class="sourceLineNo">254</span><a id="line.254">        // inverted on the reverse strand. We account for this here.</a>
<span class="sourceLineNo">255</span><a id="line.255">        if (qStrand == StrandedFeature.NEGATIVE)</a>
<span class="sourceLineNo">256</span><a id="line.256">        {</a>
<span class="sourceLineNo">257</span><a id="line.257">            int swap = qStart;</a>
<span class="sourceLineNo">258</span><a id="line.258">            qStart = qEnd;</a>
<span class="sourceLineNo">259</span><a id="line.259">            qEnd   = swap;</a>
<span class="sourceLineNo">260</span><a id="line.260">        }</a>
<span class="sourceLineNo">261</span><a id="line.261"></a>
<span class="sourceLineNo">262</span><a id="line.262">        if (sStrand == StrandedFeature.NEGATIVE)</a>
<span class="sourceLineNo">263</span><a id="line.263">        {</a>
<span class="sourceLineNo">264</span><a id="line.264">            int swap = sStart;</a>
<span class="sourceLineNo">265</span><a id="line.265">            sStart = sEnd;</a>
<span class="sourceLineNo">266</span><a id="line.266">            sEnd   = swap;</a>
<span class="sourceLineNo">267</span><a id="line.267">        }</a>
<span class="sourceLineNo">268</span><a id="line.268"></a>
<span class="sourceLineNo">269</span><a id="line.269">        String subjectID = (String) hitData.get("subjectId");</a>
<span class="sourceLineNo">270</span><a id="line.270"></a>
<span class="sourceLineNo">271</span><a id="line.271">        Sequence   queryView = makeQueryViewSequence(queryID);</a>
<span class="sourceLineNo">272</span><a id="line.272">        Sequence subjectView = makeSubjectViewSequence(subjectID);</a>
<span class="sourceLineNo">273</span><a id="line.273"></a>
<span class="sourceLineNo">274</span><a id="line.274">        SimpleHomology homology = new SimpleHomology();</a>
<span class="sourceLineNo">275</span><a id="line.275"></a>
<span class="sourceLineNo">276</span><a id="line.276">        // Map of HomologyFeatures to Alignment sequences</a>
<span class="sourceLineNo">277</span><a id="line.277">        Map labelMap = new HashMap();</a>
<span class="sourceLineNo">278</span><a id="line.278"></a>
<span class="sourceLineNo">279</span><a id="line.279">        try</a>
<span class="sourceLineNo">280</span><a id="line.280">        {</a>
<span class="sourceLineNo">281</span><a id="line.281">            String source = "unknown";</a>
<span class="sourceLineNo">282</span><a id="line.282">            if (subHitData.containsKey("program"))</a>
<span class="sourceLineNo">283</span><a id="line.283">                source = (String) subHitData.get("program");</a>
<span class="sourceLineNo">284</span><a id="line.284"></a>
<span class="sourceLineNo">285</span><a id="line.285">            tokenBuffer.setLength(0);</a>
<span class="sourceLineNo">286</span><a id="line.286">            tokenBuffer.append((String) subHitData.get("querySequence"));</a>
<span class="sourceLineNo">287</span><a id="line.287"></a>
<span class="sourceLineNo">288</span><a id="line.288">            // Query sequence feature</a>
<span class="sourceLineNo">289</span><a id="line.289">            HomologyFeature.Template qt = new HomologyFeature.Template();</a>
<span class="sourceLineNo">290</span><a id="line.290">            qt.type       = HOMOLOGY_FEATURE_TYPE;</a>
<span class="sourceLineNo">291</span><a id="line.291">            qt.source     = source;</a>
<span class="sourceLineNo">292</span><a id="line.292">            qt.location   = new RangeLocation(qStart, qEnd);</a>
<span class="sourceLineNo">293</span><a id="line.293">            qt.strand     = qStrand;</a>
<span class="sourceLineNo">294</span><a id="line.294">            qt.annotation = AnnotationFactory.makeAnnotation(subHitData);</a>
<span class="sourceLineNo">295</span><a id="line.295">            qt.homology   = homology;</a>
<span class="sourceLineNo">296</span><a id="line.296"></a>
<span class="sourceLineNo">297</span><a id="line.297">            // Map the new feature to the alignment SymbolList</a>
<span class="sourceLineNo">298</span><a id="line.298">            labelMap.put(queryView.createFeature(qt),</a>
<span class="sourceLineNo">299</span><a id="line.299">                         new SimpleSymbolList(tokenParser, tokenBuffer.substring(0)));</a>
<span class="sourceLineNo">300</span><a id="line.300"></a>
<span class="sourceLineNo">301</span><a id="line.301">            tokenBuffer.setLength(0);</a>
<span class="sourceLineNo">302</span><a id="line.302">            tokenBuffer.append((String) subHitData.get("subjectSequence"));</a>
<span class="sourceLineNo">303</span><a id="line.303"></a>
<span class="sourceLineNo">304</span><a id="line.304">            // Subject sequence feature</a>
<span class="sourceLineNo">305</span><a id="line.305">            HomologyFeature.Template st = new HomologyFeature.Template();</a>
<span class="sourceLineNo">306</span><a id="line.306">            st.type       = HOMOLOGY_FEATURE_TYPE;</a>
<span class="sourceLineNo">307</span><a id="line.307">            st.source     = source;</a>
<span class="sourceLineNo">308</span><a id="line.308">            st.location   = new RangeLocation(sStart, sEnd);</a>
<span class="sourceLineNo">309</span><a id="line.309">            st.strand     = sStrand;</a>
<span class="sourceLineNo">310</span><a id="line.310">            st.annotation = AnnotationFactory.makeAnnotation(subHitData);</a>
<span class="sourceLineNo">311</span><a id="line.311">            st.homology   = homology;</a>
<span class="sourceLineNo">312</span><a id="line.312"></a>
<span class="sourceLineNo">313</span><a id="line.313">            // Map the new feature to the alignment SymbolList</a>
<span class="sourceLineNo">314</span><a id="line.314">            labelMap.put(subjectView.createFeature(st),</a>
<span class="sourceLineNo">315</span><a id="line.315">                         new SimpleSymbolList(tokenParser, tokenBuffer.substring(0)));</a>
<span class="sourceLineNo">316</span><a id="line.316"></a>
<span class="sourceLineNo">317</span><a id="line.317">            Alignment a = new SimpleAlignment(labelMap);</a>
<span class="sourceLineNo">318</span><a id="line.318">            homology.setAlignment(a);</a>
<span class="sourceLineNo">319</span><a id="line.319"></a>
<span class="sourceLineNo">320</span><a id="line.320">            return homology;</a>
<span class="sourceLineNo">321</span><a id="line.321">        }</a>
<span class="sourceLineNo">322</span><a id="line.322">        catch (ChangeVetoException cve)</a>
<span class="sourceLineNo">323</span><a id="line.323">        {</a>
<span class="sourceLineNo">324</span><a id="line.324">            throw new BioException( "Failed to create HomologyFeature",cve);</a>
<span class="sourceLineNo">325</span><a id="line.325">        }</a>
<span class="sourceLineNo">326</span><a id="line.326">    }</a>
<span class="sourceLineNo">327</span><a id="line.327">}</a>




























































</pre>
</div>
</main>
</body>
</html>
