<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_65) on Tue Sep 02 20:47:17 CDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>SimilarityPairFeature (biojava-legacy 1.9.1 API)</title>
<meta name="date" content="2014-09-02">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SimilarityPairFeature (biojava-legacy 1.9.1 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SimilarityPairFeature.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/bio/seq/homol/HomologyFeature.Template.html" title="class in org.biojava.bio.seq.homol"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.EmptyPairwiseAlignment.html" title="class in org.biojava.bio.seq.homol"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/bio/seq/homol/SimilarityPairFeature.html" target="_top">Frames</a></li>
<li><a href="SimilarityPairFeature.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.bio.seq.homol</div>
<h2 title="Interface SimilarityPairFeature" class="title">Interface SimilarityPairFeature</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Superinterfaces:</dt>
<dd><a href="../../../../../org/biojava/bio/Annotatable.html" title="interface in org.biojava.bio">Annotatable</a>, <a href="../../../../../org/biojava/utils/Changeable.html" title="interface in org.biojava.utils">Changeable</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a>, <a href="../../../../../org/biojava/bio/seq/StrandedFeature.html" title="interface in org.biojava.bio.seq">StrandedFeature</a></dd>
</dl>
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><a href="../../../../../org/biojava/bio/seq/impl/SimpleSimilarityPairFeature.html" title="class in org.biojava.bio.seq.impl">SimpleSimilarityPairFeature</a></dd>
</dl>
<hr>
<br>
<pre>public interface <a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.68">SimilarityPairFeature</a>
extends <a href="../../../../../org/biojava/bio/seq/StrandedFeature.html" title="interface in org.biojava.bio.seq">StrandedFeature</a></pre>
<div class="block"><p><code>SimilarityPairFeature</code> describes a pairwise
 similarity between two nucleotide sequences (as it extends
 <code>StrandedFeature</code>). It is analagous to, and based on,
 the BioPerl Bio::SeqFeature::SimilarityPair.</p>

 <p>It is different from <code>HomologyFeature</code> in that it
 expresses a relationship between only two sequence regions (rather
 than >= 2), with one clearly defined as the query sequence and the
 other as the subject (database hit). These are identified by
 constant labels in the
 <code>Alignment</code>. <code>HomologyFeature</code> identifies the
 related sequence regions by means of an <code>Homology</code>
 instance which contains an <code>Alignment</code> which uses the
 <code>HomologyFeature</code>s themselves as labels.</p>

 <p>In cases where there is no alignment available, for example when
 MSPCrunch output or GFF have been used, the
 <code>EmptyPairwiseAlignment</code> in the EMPTY_PAIRWISE field may
 be used. This may also be useful if an implementation elides the
 alignment data for some reason.</p></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>1.2</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>Keith James</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.EmptyPairwiseAlignment.html" title="class in org.biojava.bio.seq.homol">SimilarityPairFeature.EmptyPairwiseAlignment</a></strong></code>
<div class="block"><code>EmptyPairwiseAlignment</code> empty pairwise alignment
 which has labels to empty symbol lists.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.Template.html" title="class in org.biojava.bio.seq.homol">SimilarityPairFeature.Template</a></strong></code>
<div class="block"><code>Template</code> for construction of
 <code>SimilarityPairFeature</code>s.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.bio.seq.StrandedFeature">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/StrandedFeature.html" title="interface in org.biojava.bio.seq">StrandedFeature</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/StrandedFeature.Strand.html" title="class in org.biojava.bio.seq">StrandedFeature.Strand</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.bio.seq.Feature">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/Feature.ByLocationComparator.html" title="class in org.biojava.bio.seq">Feature.ByLocationComparator</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.bio.seq.FeatureHolder">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/FeatureHolder.EmptyFeatureHolder.html" title="class in org.biojava.bio.seq">FeatureHolder.EmptyFeatureHolder</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="nested_classes_inherited_from_class_org.biojava.bio.Annotatable">
<!--   -->
</a>
<h3>Nested classes/interfaces inherited from interface&nbsp;org.biojava.bio.<a href="../../../../../org/biojava/bio/Annotatable.html" title="interface in org.biojava.bio">Annotatable</a></h3>
<code><a href="../../../../../org/biojava/bio/Annotatable.AnnotationForwarder.html" title="class in org.biojava.bio">Annotatable.AnnotationForwarder</a></code></li>
</ul>
</li>
</ul>
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Field Summary table, listing fields, and an explanation">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#EMPTY_PAIRWISE">EMPTY_PAIRWISE</a></strong></code>
<div class="block">Constant <code>EMPTY_PAIRWISE</code> is an empty alignment for
 situations where there is no available alignment data or the
 implementation does not want to create one.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#QUERY_LABEL">QUERY_LABEL</a></strong></code>
<div class="block">Constant <code>QUERY_LABEL</code> is the alignment label used
 for all query sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../org/biojava/utils/ChangeType.html" title="class in org.biojava.utils">ChangeType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#SIBLING">SIBLING</a></strong></code>
<div class="block">The sibling of this feature has altered.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#SUBJECT_LABEL">SUBJECT_LABEL</a></strong></code>
<div class="block">Constant <code>SUBJECT_LABEL</code> is the alignment label used
 for all subject sequences.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_org.biojava.bio.seq.StrandedFeature">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/StrandedFeature.html" title="interface in org.biojava.bio.seq">StrandedFeature</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/StrandedFeature.html#NEGATIVE">NEGATIVE</a>, <a href="../../../../../org/biojava/bio/seq/StrandedFeature.html#POSITIVE">POSITIVE</a>, <a href="../../../../../org/biojava/bio/seq/StrandedFeature.html#STRAND">STRAND</a>, <a href="../../../../../org/biojava/bio/seq/StrandedFeature.html#UNKNOWN">UNKNOWN</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_org.biojava.bio.seq.Feature">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/Feature.html#byLocationOrder">byLocationOrder</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#LOCATION">LOCATION</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#PROPERTY_DATA_KEY">PROPERTY_DATA_KEY</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#SOURCE">SOURCE</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#SOURCETERM">SOURCETERM</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#TYPE">TYPE</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#TYPETERM">TYPETERM</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_org.biojava.bio.seq.FeatureHolder">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#EMPTY_FEATURE_HOLDER">EMPTY_FEATURE_HOLDER</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#FEATURES">FEATURES</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#SCHEMA">SCHEMA</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_org.biojava.bio.Annotatable">
<!--   -->
</a>
<h3>Fields inherited from interface&nbsp;org.biojava.bio.<a href="../../../../../org/biojava/bio/Annotatable.html" title="interface in org.biojava.bio">Annotatable</a></h3>
<code><a href="../../../../../org/biojava/bio/Annotatable.html#ANNOTATION">ANNOTATION</a></code></li>
</ul>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#getAlignment()">getAlignment</a></strong>()</code>
<div class="block"><code>getAlignment</code> returns the <code>Alignment</code> of
 two similar features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#getScore()">getScore</a></strong>()</code>
<div class="block"><code>getScore</code> returns the alignment score.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html" title="interface in org.biojava.bio.seq.homol">SimilarityPairFeature</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#getSibling()">getSibling</a></strong>()</code>
<div class="block"><code>getSibling</code> returns the sibling
 <code>Feature</code>, query for subject and vice versa.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html#setSibling(org.biojava.bio.seq.homol.SimilarityPairFeature)">setSibling</a></strong>(<a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html" title="interface in org.biojava.bio.seq.homol">SimilarityPairFeature</a>&nbsp;sibling)</code>
<div class="block"><code>setSibling</code> sets the sibling feature of the
 pair.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.bio.seq.StrandedFeature">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/StrandedFeature.html" title="interface in org.biojava.bio.seq">StrandedFeature</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/StrandedFeature.html#getStrand()">getStrand</a>, <a href="../../../../../org/biojava/bio/seq/StrandedFeature.html#getSymbols()">getSymbols</a>, <a href="../../../../../org/biojava/bio/seq/StrandedFeature.html#setStrand(org.biojava.bio.seq.StrandedFeature.Strand)">setStrand</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.bio.seq.Feature">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/Feature.html" title="interface in org.biojava.bio.seq">Feature</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/Feature.html#features()">features</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#getLocation()">getLocation</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#getParent()">getParent</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#getSequence()">getSequence</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#getSource()">getSource</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#getSourceTerm()">getSourceTerm</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#getType()">getType</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#getTypeTerm()">getTypeTerm</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#makeTemplate()">makeTemplate</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#setLocation(org.biojava.bio.symbol.Location)">setLocation</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#setSource(java.lang.String)">setSource</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#setSourceTerm(org.biojava.ontology.Term)">setSourceTerm</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#setType(java.lang.String)">setType</a>, <a href="../../../../../org/biojava/bio/seq/Feature.html#setTypeTerm(org.biojava.ontology.Term)">setTypeTerm</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.bio.seq.FeatureHolder">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;org.biojava.bio.seq.<a href="../../../../../org/biojava/bio/seq/FeatureHolder.html" title="interface in org.biojava.bio.seq">FeatureHolder</a></h3>
<code><a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#containsFeature(org.biojava.bio.seq.Feature)">containsFeature</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#countFeatures()">countFeatures</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#createFeature(org.biojava.bio.seq.Feature.Template)">createFeature</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#filter(org.biojava.bio.seq.FeatureFilter)">filter</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#filter(org.biojava.bio.seq.FeatureFilter,%20boolean)">filter</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#getSchema()">getSchema</a>, <a href="../../../../../org/biojava/bio/seq/FeatureHolder.html#removeFeature(org.biojava.bio.seq.Feature)">removeFeature</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.bio.Annotatable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;org.biojava.bio.<a href="../../../../../org/biojava/bio/Annotatable.html" title="interface in org.biojava.bio">Annotatable</a></h3>
<code><a href="../../../../../org/biojava/bio/Annotatable.html#getAnnotation()">getAnnotation</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_org.biojava.utils.Changeable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;org.biojava.utils.<a href="../../../../../org/biojava/utils/Changeable.html" title="interface in org.biojava.utils">Changeable</a></h3>
<code><a href="../../../../../org/biojava/utils/Changeable.html#addChangeListener(org.biojava.utils.ChangeListener)">addChangeListener</a>, <a href="../../../../../org/biojava/utils/Changeable.html#addChangeListener(org.biojava.utils.ChangeListener,%20org.biojava.utils.ChangeType)">addChangeListener</a>, <a href="../../../../../org/biojava/utils/Changeable.html#isUnchanging(org.biojava.utils.ChangeType)">isUnchanging</a>, <a href="../../../../../org/biojava/utils/Changeable.html#removeChangeListener(org.biojava.utils.ChangeListener)">removeChangeListener</a>, <a href="../../../../../org/biojava/utils/Changeable.html#removeChangeListener(org.biojava.utils.ChangeListener,%20org.biojava.utils.ChangeType)">removeChangeListener</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a name="SIBLING">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SIBLING</h4>
<pre>static final&nbsp;<a href="../../../../../org/biojava/utils/ChangeType.html" title="class in org.biojava.utils">ChangeType</a> <a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.73">SIBLING</a></pre>
<div class="block">The sibling of this feature has altered.</div>
</li>
</ul>
<a name="QUERY_LABEL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>QUERY_LABEL</h4>
<pre>static final&nbsp;<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> <a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.80">QUERY_LABEL</a></pre>
<div class="block">Constant <code>QUERY_LABEL</code> is the alignment label used
 for all query sequences.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#org.biojava.bio.seq.homol.SimilarityPairFeature.QUERY_LABEL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="SUBJECT_LABEL">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>SUBJECT_LABEL</h4>
<pre>static final&nbsp;<a href="http://docs.oracle.com/javase/1.5.0/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a> <a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.86">SUBJECT_LABEL</a></pre>
<div class="block">Constant <code>SUBJECT_LABEL</code> is the alignment label used
 for all subject sequences.</div>
<dl><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../constant-values.html#org.biojava.bio.seq.homol.SimilarityPairFeature.SUBJECT_LABEL">Constant Field Values</a></dd></dl>
</li>
</ul>
<a name="EMPTY_PAIRWISE">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>EMPTY_PAIRWISE</h4>
<pre>static final&nbsp;<a href="../../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a> <a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.93">EMPTY_PAIRWISE</a></pre>
<div class="block">Constant <code>EMPTY_PAIRWISE</code> is an empty alignment for
 situations where there is no available alignment data or the
 implementation does not want to create one.</div>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getSibling()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSibling</h4>
<pre><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html" title="interface in org.biojava.bio.seq.homol">SimilarityPairFeature</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.101">getSibling</a>()</pre>
<div class="block"><code>getSibling</code> returns the sibling
 <code>Feature</code>, query for subject and vice versa.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a <code>Feature</code>.</dd></dl>
</li>
</ul>
<a name="setSibling(org.biojava.bio.seq.homol.SimilarityPairFeature)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSibling</h4>
<pre>void&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.109">setSibling</a>(<a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.html" title="interface in org.biojava.bio.seq.homol">SimilarityPairFeature</a>&nbsp;sibling)
                throws <a href="../../../../../org/biojava/utils/ChangeVetoException.html" title="class in org.biojava.utils">ChangeVetoException</a></pre>
<div class="block"><code>setSibling</code> sets the sibling feature of the
 pair. This is used to set the reciprocal
 <code>SimilarityPairFeature</code> as both cannot be set using
 the <code>Template</code>.</div>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../org/biojava/utils/ChangeVetoException.html" title="class in org.biojava.utils">ChangeVetoException</a></code></dd></dl>
</li>
</ul>
<a name="getAlignment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAlignment</h4>
<pre><a href="../../../../../org/biojava/bio/alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.118">getAlignment</a>()</pre>
<div class="block"><code>getAlignment</code> returns the <code>Alignment</code> of
 two similar features.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>an <code>Alignment</code> value.</dd></dl>
</li>
</ul>
<a name="getScore()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getScore</h4>
<pre>double&nbsp;<a href="../../../../../src-html/org/biojava/bio/seq/homol/SimilarityPairFeature.html#line.125">getScore</a>()</pre>
<div class="block"><code>getScore</code> returns the alignment score.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a <code>double</code>.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SimilarityPairFeature.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/bio/seq/homol/HomologyFeature.Template.html" title="class in org.biojava.bio.seq.homol"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/bio/seq/homol/SimilarityPairFeature.EmptyPairwiseAlignment.html" title="class in org.biojava.bio.seq.homol"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/bio/seq/homol/SimilarityPairFeature.html" target="_top">Frames</a></li>
<li><a href="SimilarityPairFeature.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li><a href="#field_summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field_detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2014 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
