<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.ronn, class: Jronn, class: Range">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*        BioJava development code</span>
<span class="source-line-no">002</span><span id="line-2"> *</span>
<span class="source-line-no">003</span><span id="line-3"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">004</span><span id="line-4"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">005</span><span id="line-5"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">006</span><span id="line-6"> * see:</span>
<span class="source-line-no">007</span><span id="line-7"> *</span>
<span class="source-line-no">008</span><span id="line-8"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">009</span><span id="line-9"> *</span>
<span class="source-line-no">010</span><span id="line-10"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">011</span><span id="line-11"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">012</span><span id="line-12"> *</span>
<span class="source-line-no">013</span><span id="line-13"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">014</span><span id="line-14"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">015</span><span id="line-15"> * at:</span>
<span class="source-line-no">016</span><span id="line-16"> *</span>
<span class="source-line-no">017</span><span id="line-17"> *      http://www.biojava.org/</span>
<span class="source-line-no">018</span><span id="line-18"> *</span>
<span class="source-line-no">019</span><span id="line-19"> */</span>
<span class="source-line-no">020</span><span id="line-20">package org.biojava.nbio.ronn;</span>
<span class="source-line-no">021</span><span id="line-21"></span>
<span class="source-line-no">022</span><span id="line-22">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">023</span><span id="line-23">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.data.sequence.FastaSequence;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.data.sequence.SequenceUtil;</span>
<span class="source-line-no">026</span><span id="line-26"></span>
<span class="source-line-no">027</span><span id="line-27">import java.io.FileInputStream;</span>
<span class="source-line-no">028</span><span id="line-28">import java.io.FileNotFoundException;</span>
<span class="source-line-no">029</span><span id="line-29">import java.io.IOException;</span>
<span class="source-line-no">030</span><span id="line-30">import java.io.Serializable;</span>
<span class="source-line-no">031</span><span id="line-31">import java.util.ArrayList;</span>
<span class="source-line-no">032</span><span id="line-32">import java.util.List;</span>
<span class="source-line-no">033</span><span id="line-33">import java.util.Map;</span>
<span class="source-line-no">034</span><span id="line-34">import java.util.TreeMap;</span>
<span class="source-line-no">035</span><span id="line-35">import java.util.stream.Collectors;</span>
<span class="source-line-no">036</span><span id="line-36"></span>
<span class="source-line-no">037</span><span id="line-37"></span>
<span class="source-line-no">038</span><span id="line-38">/**</span>
<span class="source-line-no">039</span><span id="line-39"> * This class gives public API to RONN functions.</span>
<span class="source-line-no">040</span><span id="line-40"> * It is build on top of the command line client. Due to this fact a few things</span>
<span class="source-line-no">041</span><span id="line-41"> * could be improved and extended pending the refactoring of the command line client.</span>
<span class="source-line-no">042</span><span id="line-42"> *</span>
<span class="source-line-no">043</span><span id="line-43"> * The input sequence limitations - the input sequence must not contain any ambiguous characters,</span>
<span class="source-line-no">044</span><span id="line-44"> * and have a minimum length of 19 amino acids.</span>
<span class="source-line-no">045</span><span id="line-45"> *</span>
<span class="source-line-no">046</span><span id="line-46"> * @author Peter Troshin</span>
<span class="source-line-no">047</span><span id="line-47"> * @version 1.0</span>
<span class="source-line-no">048</span><span id="line-48"> * @since 3.0.2</span>
<span class="source-line-no">049</span><span id="line-49"> *</span>
<span class="source-line-no">050</span><span id="line-50"> */</span>
<span class="source-line-no">051</span><span id="line-51">public class Jronn implements Serializable {</span>
<span class="source-line-no">052</span><span id="line-52"></span>
<span class="source-line-no">053</span><span id="line-53">        /**</span>
<span class="source-line-no">054</span><span id="line-54">         *</span>
<span class="source-line-no">055</span><span id="line-55">         */</span>
<span class="source-line-no">056</span><span id="line-56">        private static final long serialVersionUID = 8104272449130849946L;</span>
<span class="source-line-no">057</span><span id="line-57">        // Load models</span>
<span class="source-line-no">058</span><span id="line-58">        private static final ModelLoader loader = new ModelLoader();</span>
<span class="source-line-no">059</span><span id="line-59">        static {</span>
<span class="source-line-no">060</span><span id="line-60">                try {</span>
<span class="source-line-no">061</span><span id="line-61">                        loader.loadModels();</span>
<span class="source-line-no">062</span><span id="line-62">                } catch (NumberFormatException e) {</span>
<span class="source-line-no">063</span><span id="line-63">                        throw new RuntimeException("Fails to load models!" + e.getMessage(), e);</span>
<span class="source-line-no">064</span><span id="line-64">                } catch (IOException e) {</span>
<span class="source-line-no">065</span><span id="line-65">                        throw new RuntimeException("Fails to load models!" + e.getMessage(), e);</span>
<span class="source-line-no">066</span><span id="line-66">                }</span>
<span class="source-line-no">067</span><span id="line-67">        }</span>
<span class="source-line-no">068</span><span id="line-68"></span>
<span class="source-line-no">069</span><span id="line-69"></span>
<span class="source-line-no">070</span><span id="line-70">        /**</span>
<span class="source-line-no">071</span><span id="line-71">         * Holder for the ranges, contain pointers to starting and ending position</span>
<span class="source-line-no">072</span><span id="line-72">         * on the sequence which comprises a disordered region. Immutable.</span>
<span class="source-line-no">073</span><span id="line-73">         * @author pvtroshin</span>
<span class="source-line-no">074</span><span id="line-74">         */</span>
<span class="source-line-no">075</span><span id="line-75">        public static class Range {</span>
<span class="source-line-no">076</span><span id="line-76">                /**</span>
<span class="source-line-no">077</span><span id="line-77">                 * Range starting position counts from 1 (the first position on the sequence is 1)</span>
<span class="source-line-no">078</span><span id="line-78">                 */</span>
<span class="source-line-no">079</span><span id="line-79">                public final int from;</span>
<span class="source-line-no">080</span><span id="line-80">                /**</span>
<span class="source-line-no">081</span><span id="line-81">                 * The range ending position includes the last residue.</span>
<span class="source-line-no">082</span><span id="line-82">                 */</span>
<span class="source-line-no">083</span><span id="line-83">                public final int to;</span>
<span class="source-line-no">084</span><span id="line-84"></span>
<span class="source-line-no">085</span><span id="line-85">                public final float score;</span>
<span class="source-line-no">086</span><span id="line-86">                public Range(int from, int to, float score) {</span>
<span class="source-line-no">087</span><span id="line-87">                        assert from&gt;=0;</span>
<span class="source-line-no">088</span><span id="line-88">                        assert from&lt;to;</span>
<span class="source-line-no">089</span><span id="line-89">                        this.from = from;</span>
<span class="source-line-no">090</span><span id="line-90">                        this.to = to;</span>
<span class="source-line-no">091</span><span id="line-91">                        this.score = score;</span>
<span class="source-line-no">092</span><span id="line-92">                }</span>
<span class="source-line-no">093</span><span id="line-93"></span>
<span class="source-line-no">094</span><span id="line-94">                @Override</span>
<span class="source-line-no">095</span><span id="line-95">                public String toString() {</span>
<span class="source-line-no">096</span><span id="line-96">                        return "Range" + " From:" + from + "\t" + "to: " + to + "\n";</span>
<span class="source-line-no">097</span><span id="line-97">                }</span>
<span class="source-line-no">098</span><span id="line-98"></span>
<span class="source-line-no">099</span><span id="line-99">                @Override</span>
<span class="source-line-no">100</span><span id="line-100">                public int hashCode() {</span>
<span class="source-line-no">101</span><span id="line-101">                        final int prime = 31;</span>
<span class="source-line-no">102</span><span id="line-102">                        int result = 1;</span>
<span class="source-line-no">103</span><span id="line-103">                        result = prime * result + from;</span>
<span class="source-line-no">104</span><span id="line-104">                        result = prime * result + to;</span>
<span class="source-line-no">105</span><span id="line-105">                        return result;</span>
<span class="source-line-no">106</span><span id="line-106">                }</span>
<span class="source-line-no">107</span><span id="line-107"></span>
<span class="source-line-no">108</span><span id="line-108">                @Override</span>
<span class="source-line-no">109</span><span id="line-109">                public boolean equals(Object obj) {</span>
<span class="source-line-no">110</span><span id="line-110">                        if (this == obj)</span>
<span class="source-line-no">111</span><span id="line-111">                                return true;</span>
<span class="source-line-no">112</span><span id="line-112">                        if (obj == null)</span>
<span class="source-line-no">113</span><span id="line-113">                                return false;</span>
<span class="source-line-no">114</span><span id="line-114">                        if (getClass() != obj.getClass())</span>
<span class="source-line-no">115</span><span id="line-115">                                return false;</span>
<span class="source-line-no">116</span><span id="line-116">                        Range other = (Range) obj;</span>
<span class="source-line-no">117</span><span id="line-117">                        if (from != other.from)</span>
<span class="source-line-no">118</span><span id="line-118">                                return false;</span>
<span class="source-line-no">119</span><span id="line-119">                        if (to != other.to)</span>
<span class="source-line-no">120</span><span id="line-120">                                return false;</span>
<span class="source-line-no">121</span><span id="line-121">                        return true;</span>
<span class="source-line-no">122</span><span id="line-122">                }</span>
<span class="source-line-no">123</span><span id="line-123"></span>
<span class="source-line-no">124</span><span id="line-124"></span>
<span class="source-line-no">125</span><span id="line-125">        }</span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">        /**</span>
<span class="source-line-no">128</span><span id="line-128">         * Calculates the probability value for each residue in the protein sequence,</span>
<span class="source-line-no">129</span><span id="line-129">         * telling the probability that the residue belongs to disordered region.</span>
<span class="source-line-no">130</span><span id="line-130">         * In general, values greater than 0.5 considered to be in the disordered regions.</span>
<span class="source-line-no">131</span><span id="line-131">         *</span>
<span class="source-line-no">132</span><span id="line-132">         * @param sequence an instance of FastaSequence object, holding the name and the sequence.</span>
<span class="source-line-no">133</span><span id="line-133">         * @return the probability scores for each residue in the sequence</span>
<span class="source-line-no">134</span><span id="line-134">         */</span>
<span class="source-line-no">135</span><span id="line-135">        public static float[] getDisorderScores(FastaSequence sequence) {</span>
<span class="source-line-no">136</span><span id="line-136">                    return predictSerial(sequence);</span>
<span class="source-line-no">137</span><span id="line-137">        }</span>
<span class="source-line-no">138</span><span id="line-138"></span>
<span class="source-line-no">139</span><span id="line-139">        /**</span>
<span class="source-line-no">140</span><span id="line-140">         * Calculates the probability value for each residue in the protein sequence,</span>
<span class="source-line-no">141</span><span id="line-141">         * telling the probability that the residue belongs to disordered region.</span>
<span class="source-line-no">142</span><span id="line-142">         * In general, values greater than 0.5 considered to be in the disordered regions.</span>
<span class="source-line-no">143</span><span id="line-143">         *</span>
<span class="source-line-no">144</span><span id="line-144">         * @param sequence an instance of FastaSequence object, holding the name and the sequence.</span>
<span class="source-line-no">145</span><span id="line-145">         * @return the probability scores for each residue in the sequence</span>
<span class="source-line-no">146</span><span id="line-146">         */</span>
<span class="source-line-no">147</span><span id="line-147">        public static float[] getDisorderScores(ProteinSequence sequence) {</span>
<span class="source-line-no">148</span><span id="line-148"></span>
<span class="source-line-no">149</span><span id="line-149">                FastaSequence seq = convertProteinSequencetoFasta(sequence);</span>
<span class="source-line-no">150</span><span id="line-150"></span>
<span class="source-line-no">151</span><span id="line-151">                return predictSerial(seq);</span>
<span class="source-line-no">152</span><span id="line-152">        }</span>
<span class="source-line-no">153</span><span id="line-153"></span>
<span class="source-line-no">154</span><span id="line-154">        /** Utility method to convert a BioJava ProteinSequence object to the FastaSequence</span>
<span class="source-line-no">155</span><span id="line-155">         *  object used internally in JRonn.</span>
<span class="source-line-no">156</span><span id="line-156">         *</span>
<span class="source-line-no">157</span><span id="line-157">         * @param sequence</span>
<span class="source-line-no">158</span><span id="line-158">         * @return</span>
<span class="source-line-no">159</span><span id="line-159">         */</span>
<span class="source-line-no">160</span><span id="line-160">        public static FastaSequence convertProteinSequencetoFasta(ProteinSequence sequence){</span>
<span class="source-line-no">161</span><span id="line-161">                StringBuffer buf = new StringBuffer();</span>
<span class="source-line-no">162</span><span id="line-162">                for (AminoAcidCompound compound : sequence) {</span>
<span class="source-line-no">163</span><span id="line-163"></span>
<span class="source-line-no">164</span><span id="line-164">                        String c = compound.getShortName();</span>
<span class="source-line-no">165</span><span id="line-165"></span>
<span class="source-line-no">166</span><span id="line-166">                        if (! SequenceUtil.NON_AA.matcher(c).find()) {</span>
<span class="source-line-no">167</span><span id="line-167">                                buf.append(c);</span>
<span class="source-line-no">168</span><span id="line-168">                        } else {</span>
<span class="source-line-no">169</span><span id="line-169">                                buf.append("X");</span>
<span class="source-line-no">170</span><span id="line-170">                        }</span>
<span class="source-line-no">171</span><span id="line-171">                }</span>
<span class="source-line-no">172</span><span id="line-172"></span>
<span class="source-line-no">173</span><span id="line-173">                return new FastaSequence(sequence.getAccession().getID(),buf.toString());</span>
<span class="source-line-no">174</span><span id="line-174">        }</span>
<span class="source-line-no">175</span><span id="line-175"></span>
<span class="source-line-no">176</span><span id="line-176">        private static float[] predictSerial(FastaSequence fsequence) {</span>
<span class="source-line-no">177</span><span id="line-177">                ORonn.validateSequenceForRonn(fsequence);</span>
<span class="source-line-no">178</span><span id="line-178">                ORonn ronn;</span>
<span class="source-line-no">179</span><span id="line-179">                float[] disorder = null;</span>
<span class="source-line-no">180</span><span id="line-180">                try {</span>
<span class="source-line-no">181</span><span id="line-181">                        ronn = new ORonn(fsequence, loader);</span>
<span class="source-line-no">182</span><span id="line-182">                        disorder = ronn.call().getMeanScores();</span>
<span class="source-line-no">183</span><span id="line-183">                } catch (NumberFormatException | IOException e) {</span>
<span class="source-line-no">184</span><span id="line-184">                        throw new RuntimeException("Jronn fails to load models " + e.getLocalizedMessage(), e);</span>
<span class="source-line-no">185</span><span id="line-185">                } </span>
<span class="source-line-no">186</span><span id="line-186">                return disorder;</span>
<span class="source-line-no">187</span><span id="line-187">        }</span>
<span class="source-line-no">188</span><span id="line-188"></span>
<span class="source-line-no">189</span><span id="line-189">        /**</span>
<span class="source-line-no">190</span><span id="line-190">         * Calculates the disordered regions of the sequence. More formally, the regions for which the</span>
<span class="source-line-no">191</span><span id="line-191">         * probability of disorder is greater then 0.50.</span>
<span class="source-line-no">192</span><span id="line-192">         *</span>
<span class="source-line-no">193</span><span id="line-193">         *</span>
<span class="source-line-no">194</span><span id="line-194">         * @param sequence an instance of FastaSequence object, holding the name and the sequence.</span>
<span class="source-line-no">195</span><span id="line-195">         * @return the array of ranges if there are any residues predicted to have the</span>
<span class="source-line-no">196</span><span id="line-196">         * probability of disorder greater then 0.5, null otherwise.</span>
<span class="source-line-no">197</span><span id="line-197">         *</span>
<span class="source-line-no">198</span><span id="line-198">         */</span>
<span class="source-line-no">199</span><span id="line-199">        public static Range[] getDisorder(FastaSequence sequence) {</span>
<span class="source-line-no">200</span><span id="line-200">                float[] scores = getDisorderScores(sequence);</span>
<span class="source-line-no">201</span><span id="line-201">                return scoresToRanges(scores, RonnConstraint.DEFAULT_RANGE_PROBABILITY_THRESHOLD);</span>
<span class="source-line-no">202</span><span id="line-202">        }</span>
<span class="source-line-no">203</span><span id="line-203"></span>
<span class="source-line-no">204</span><span id="line-204">        /**</span>
<span class="source-line-no">205</span><span id="line-205">         * Convert raw scores to ranges. Gives ranges for given probability of disorder value</span>
<span class="source-line-no">206</span><span id="line-206">         * @param scores the raw probability of disorder scores for each residue in the sequence.</span>
<span class="source-line-no">207</span><span id="line-207">         * @param probability the cut off threshold. Include all residues with the probability of disorder greater then this value</span>
<span class="source-line-no">208</span><span id="line-208">         * @return the array of ranges if there are any residues predicted to have the</span>
<span class="source-line-no">209</span><span id="line-209">         * probability of disorder greater then {@code probability}, null otherwise.</span>
<span class="source-line-no">210</span><span id="line-210">         */</span>
<span class="source-line-no">211</span><span id="line-211">        public static Range[] scoresToRanges(float[] scores, float probability)  {</span>
<span class="source-line-no">212</span><span id="line-212">                assert scores!=null &amp;&amp; scores.length&gt;0;</span>
<span class="source-line-no">213</span><span id="line-213">                assert probability&gt;0 &amp;&amp; probability&lt;1;</span>
<span class="source-line-no">214</span><span id="line-214"></span>
<span class="source-line-no">215</span><span id="line-215">                int count=0;</span>
<span class="source-line-no">216</span><span id="line-216">                int regionLen=0;</span>
<span class="source-line-no">217</span><span id="line-217">                List&lt;Range&gt; ranges = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">218</span><span id="line-218">                for(float score: scores) {</span>
<span class="source-line-no">219</span><span id="line-219">                        count++;</span>
<span class="source-line-no">220</span><span id="line-220">                        // Round to 2 decimal points before comparison</span>
<span class="source-line-no">221</span><span id="line-221">                        score = (float) (Math.round(score*100.0)/100.0);</span>
<span class="source-line-no">222</span><span id="line-222">                        if(score&gt;probability) {</span>
<span class="source-line-no">223</span><span id="line-223">                                regionLen++;</span>
<span class="source-line-no">224</span><span id="line-224">                        } else {</span>
<span class="source-line-no">225</span><span id="line-225">                                if(regionLen&gt;0) {</span>
<span class="source-line-no">226</span><span id="line-226">                                        ranges.add(new Range(count-regionLen, count-1,score));</span>
<span class="source-line-no">227</span><span id="line-227">                                }</span>
<span class="source-line-no">228</span><span id="line-228">                                regionLen=0;</span>
<span class="source-line-no">229</span><span id="line-229">                        }</span>
<span class="source-line-no">230</span><span id="line-230">                }</span>
<span class="source-line-no">231</span><span id="line-231">                // In case of the range to boundary runs to the very end of the sequence</span>
<span class="source-line-no">232</span><span id="line-232">                if(regionLen&gt;1) {</span>
<span class="source-line-no">233</span><span id="line-233">                        ranges.add(new Range(count-regionLen+1, count,scores[scores.length-1]));</span>
<span class="source-line-no">234</span><span id="line-234">                }</span>
<span class="source-line-no">235</span><span id="line-235">                return ranges.toArray(new Range[ranges.size()]);</span>
<span class="source-line-no">236</span><span id="line-236"></span>
<span class="source-line-no">237</span><span id="line-237">        }</span>
<span class="source-line-no">238</span><span id="line-238"></span>
<span class="source-line-no">239</span><span id="line-239">        /**</span>
<span class="source-line-no">240</span><span id="line-240">         * Calculates the probability of disorder scores for each residue in the sequence for</span>
<span class="source-line-no">241</span><span id="line-241">         * many sequences in the input.</span>
<span class="source-line-no">242</span><span id="line-242">         *</span>
<span class="source-line-no">243</span><span id="line-243">         * @param sequences the list of the FastaSequence objects</span>
<span class="source-line-no">244</span><span id="line-244">         * @return the Map with key-&gt;FastaSequence, value-&gt;probability of disorder for each residue</span>
<span class="source-line-no">245</span><span id="line-245">         * @see #getDisorder(FastaSequence)</span>
<span class="source-line-no">246</span><span id="line-246">         */</span>
<span class="source-line-no">247</span><span id="line-247">        public static Map&lt;FastaSequence,float[]&gt; getDisorderScores(List&lt;FastaSequence&gt; sequences) {</span>
<span class="source-line-no">248</span><span id="line-248">                Map&lt;FastaSequence,float[]&gt; results = new TreeMap&lt;&gt;();</span>
<span class="source-line-no">249</span><span id="line-249">                results = sequences.stream().collect(Collectors.toMap(fastaSequence -&gt;  fastaSequence, fastaSequence -&gt; predictSerial(fastaSequence)));</span>
<span class="source-line-no">250</span><span id="line-250">                return results;</span>
<span class="source-line-no">251</span><span id="line-251">        }</span>
<span class="source-line-no">252</span><span id="line-252"></span>
<span class="source-line-no">253</span><span id="line-253">        /**</span>
<span class="source-line-no">254</span><span id="line-254">         * Calculates the disordered regions of the sequence for many sequences in the input.</span>
<span class="source-line-no">255</span><span id="line-255">         *</span>
<span class="source-line-no">256</span><span id="line-256">         * @param sequences sequences the list of the FastaSequence objects</span>
<span class="source-line-no">257</span><span id="line-257">         * @return</span>
<span class="source-line-no">258</span><span id="line-258">         * @see #getDisorder(FastaSequence)</span>
<span class="source-line-no">259</span><span id="line-259">         */</span>
<span class="source-line-no">260</span><span id="line-260">        public static Map&lt;FastaSequence,Range[]&gt; getDisorder(List&lt;FastaSequence&gt; sequences) {</span>
<span class="source-line-no">261</span><span id="line-261">                Map&lt;FastaSequence,Range[]&gt; disorderRanges = new TreeMap&lt;&gt;();</span>
<span class="source-line-no">262</span><span id="line-262">                disorderRanges = sequences.stream().collect(Collectors.toMap(fastaSequence -&gt; fastaSequence, fastaSequence -&gt; getDisorder(fastaSequence) ));</span>
<span class="source-line-no">263</span><span id="line-263">                return disorderRanges;</span>
<span class="source-line-no">264</span><span id="line-264">        }</span>
<span class="source-line-no">265</span><span id="line-265"></span>
<span class="source-line-no">266</span><span id="line-266">        /**</span>
<span class="source-line-no">267</span><span id="line-267">         * Calculates the disordered regions of the protein sequence.</span>
<span class="source-line-no">268</span><span id="line-268">         * @param fastaFile input file name containing the sequence in FASTA</span>
<span class="source-line-no">269</span><span id="line-269">         * @return the Map with key-&gt;FastaSequence, value-&gt;the list of disordered regions for each sequence</span>
<span class="source-line-no">270</span><span id="line-270">         * @throws FileNotFoundException if the input file cannot be found</span>
<span class="source-line-no">271</span><span id="line-271">         * @throws IOException of the system cannot access or read from the input file</span>
<span class="source-line-no">272</span><span id="line-272">         * @see #getDisorder(FastaSequence)</span>
<span class="source-line-no">273</span><span id="line-273">         */</span>
<span class="source-line-no">274</span><span id="line-274">        public static Map&lt;FastaSequence,Range[]&gt; getDisorder(String fastaFile) throws IOException {</span>
<span class="source-line-no">275</span><span id="line-275">                final List&lt;FastaSequence&gt; sequences = SequenceUtil.readFasta(new FileInputStream(fastaFile));</span>
<span class="source-line-no">276</span><span id="line-276">                return getDisorder(sequences);</span>
<span class="source-line-no">277</span><span id="line-277">        }</span>
<span class="source-line-no">278</span><span id="line-278"></span>
<span class="source-line-no">279</span><span id="line-279">        /**</span>
<span class="source-line-no">280</span><span id="line-280">         * TODO</span>
<span class="source-line-no">281</span><span id="line-281">         *</span>
<span class="source-line-no">282</span><span id="line-282">         * High performance method for calculating disorder. Use multiple threads to achieve the speedup.</span>
<span class="source-line-no">283</span><span id="line-283">         *</span>
<span class="source-line-no">284</span><span id="line-284">         * @param fastaFile  fully qualified path to the input FASTA file</span>
<span class="source-line-no">285</span><span id="line-285">         * @param outputFile file name of the file for the results</span>
<span class="source-line-no">286</span><span id="line-286">         * @param threadNumber the number of threads to use, default</span>
<span class="source-line-no">287</span><span id="line-287">         * @param controls the format of the result file</span>
<span class="source-line-no">288</span><span id="line-288">         * @throws FileNotFoundException if input file in not found</span>
<span class="source-line-no">289</span><span id="line-289">         * @throws IOException if the input or the output files cannot be accessed</span>
<span class="source-line-no">290</span><span id="line-290">         * @see ORonn.ResultLayout</span>
<span class="source-line-no">291</span><span id="line-291"></span>
<span class="source-line-no">292</span><span id="line-292">        public static void calculateDisorder(String fastaFile, String outputFile, int threadNumber, ResultLayout layout) throws FileNotFoundException, IOException {</span>
<span class="source-line-no">293</span><span id="line-293">                final List&lt;FastaSequence&gt; sequences = SequenceUtil.readFasta(new FileInputStream(fastaFile));</span>
<span class="source-line-no">294</span><span id="line-294">                InputParameters in = new InputParameters();</span>
<span class="source-line-no">295</span><span id="line-295">                in.setFilePrm(fastaFile, InputParameters.inputKey);</span>
<span class="source-line-no">296</span><span id="line-296">                in.setFilePrm(outputFile, InputParameters.outputKey);</span>
<span class="source-line-no">297</span><span id="line-297">                //in.setThreadNum(Integer.toString(threadNumber));</span>
<span class="source-line-no">298</span><span id="line-298">                ORonn.predictParallel(sequences, in, loader);</span>
<span class="source-line-no">299</span><span id="line-299">        }</span>
<span class="source-line-no">300</span><span id="line-300">        */</span>
<span class="source-line-no">301</span><span id="line-301">}</span>




























































</pre>
</div>
</main>
</body>
</html>
