<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>ChromosomeSequence (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.core.sequence, class: ChromosomeSequence">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/ChromosomeSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html#class">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Summary:</p>
<ul>
<li><a href="#nested-class-summary">Nested</a></li>
<li>Field</li>
<li><a href="#constructor-summary">Constr</a></li>
<li><a href="#method-summary">Method</a></li>
</ul>
</li>
<li>
<p>Detail:</p>
<ul>
<li>Field</li>
<li><a href="#constructor-detail">Constr</a></li>
<li><a href="#method-detail">Method</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li><a href="#nested-class-summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.core.sequence</a></div>
<h1 title="Class ChromosomeSequence" class="title">Class ChromosomeSequence</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance"><a href="template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template.AbstractSequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;
<div class="inheritance"><a href="DNASequence.html" title="class in org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence.DNASequence</a>
<div class="inheritance">org.biojava.nbio.core.sequence.ChromosomeSequence</div>
</div>
</div>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>All Implemented Interfaces:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code>, <code><a href="template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a></code>, <code><a href="template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-38">ChromosomeSequence</a></span>
<span class="extends-implements">extends <a href="DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span></div>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
<dl class="notes">
<dt>Author:</dt>
<dd>Scooter Willis</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<li>
<section class="nested-class-summary" id="nested-class-summary">
<h2>Nested Class Summary</h2>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-org.biojava.nbio.core.sequence.DNASequence">Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h2>
<code><a href="DNASequence.DNAType.html" title="enum in org.biojava.nbio.core.sequence">DNASequence.DNAType</a></code></div>
<div class="inherited-list">
<h2 id="nested-classes-inherited-from-class-org.biojava.nbio.core.sequence.template.AbstractSequence">Nested classes/interfaces inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h2>
<code><a href="template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></code></div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">ChromosomeSequence</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Empty constructor used by tools that need a proper Bean that allows the actual
 sequence data to be set after construction.</div>
</div>
<div class="col-constructor-name odd-row-color"><code><a href="#%3Cinit%3E(java.lang.String)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</code></div>
<div class="col-last odd-row-color">
<div class="block">String is king and assume DNA</div>
</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</div>
</div>
<div class="col-constructor-name odd-row-color"><code><a href="#%3Cinit%3E(org.biojava.nbio.core.sequence.template.SequenceReader)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</code></div>
<div class="col-last odd-row-color">
<div class="block">Fairly important constructor given the size of a ChromsomeSequence where the
 ProxySequenceReader could load from disk via RandomAccessFile so that the sequence
 doesn't need to be kept in memory.</div>
</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(org.biojava.nbio.core.sequence.template.SequenceReader,org.biojava.nbio.core.sequence.template.CompoundSet)" class="member-name-link">ChromosomeSequence</a><wbr>(<a href="template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</code></div>
<div class="col-last even-row-color">
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet</div>
</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel" aria-labelledby="method-summary-table-tab0">
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#addGene(org.biojava.nbio.core.sequence.AccessionID,int,int,org.biojava.nbio.core.sequence.Strand)" class="member-name-link">addGene</a><wbr>(<a href="AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;bioBegin,
 int&nbsp;bioEnd,
 <a href="Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getChromosomeNumber()" class="member-name-link">getChromosomeNumber</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getGene(java.lang.String)" class="member-name-link">getGene</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the gene based on accession.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getGeneSequences()" class="member-name-link">getGeneSequences</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Get the list of genes that have been added to the ChromosomeSequence where accession.toString is the key.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#removeGeneSequence(java.lang.String)" class="member-name-link">removeGeneSequence</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#setChromosomeNumber(int)" class="member-name-link">setChromosomeNumber</a><wbr>(int&nbsp;chromosomeNumber)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">&nbsp;</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava.nbio.core.sequence.DNASequence">Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.<a href="DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></h3>
<code><a href="DNASequence.html#getComplement()">getComplement</a>, <a href="DNASequence.html#getDNAType()">getDNAType</a>, <a href="DNASequence.html#getGCCount()">getGCCount</a>, <a href="DNASequence.html#getReverse()">getReverse</a>, <a href="DNASequence.html#getReverseComplement()">getReverseComplement</a>, <a href="DNASequence.html#getRNASequence()">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine)">getRNASequence</a>, <a href="DNASequence.html#getRNASequence(org.biojava.nbio.core.sequence.transcription.TranscriptionEngine,org.biojava.nbio.core.sequence.transcription.Frame)">getRNASequence</a>, <a href="DNASequence.html#setDNAType(org.biojava.nbio.core.sequence.DNASequence.DNAType)">setDNAType</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava.nbio.core.sequence.template.AbstractSequence">Methods inherited from class&nbsp;org.biojava.nbio.core.sequence.template.<a href="template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a></h3>
<code><a href="template/AbstractSequence.html#addFeature(int,int,org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a>, <a href="template/AbstractSequence.html#addNote(java.lang.String)">addNote</a>, <a href="template/AbstractSequence.html#countCompounds(C...)">countCompounds</a>, <a href="template/AbstractSequence.html#equals(java.lang.Object)">equals</a>, <a href="template/AbstractSequence.html#getAccession()">getAccession</a>, <a href="template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a>, <a href="template/AbstractSequence.html#getAsList()">getAsList</a>, <a href="template/AbstractSequence.html#getBioBegin()">getBioBegin</a>, <a href="template/AbstractSequence.html#getBioEnd()">getBioEnd</a>, <a href="template/AbstractSequence.html#getComments()">getComments</a>, <a href="template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a>, <a href="template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a>, <a href="template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a>, <a href="template/AbstractSequence.html#getDescription()">getDescription</a>, <a href="template/AbstractSequence.html#getFeatureRetriever()">getFeatureRetriever</a>, <a href="template/AbstractSequence.html#getFeatures()">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeatures(java.lang.String,int)">getFeatures</a>, <a href="template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a>, <a href="template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a>, <a href="template/AbstractSequence.html#getInverse()">getInverse</a>, <a href="template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a>, <a href="template/AbstractSequence.html#getLength()">getLength</a>, <a href="template/AbstractSequence.html#getNotesList()">getNotesList</a>, <a href="template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a>, <a href="template/AbstractSequence.html#getParentSequence()">getParentSequence</a>, <a href="template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a>, <a href="template/AbstractSequence.html#getReferences()">getReferences</a>, <a href="template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,java.lang.Integer,org.biojava.nbio.core.sequence.Strand)">getSequenceAsString</a>, <a href="template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a>, <a href="template/AbstractSequence.html#getSource()">getSource</a>, <a href="template/AbstractSequence.html#getSubSequence(java.lang.Integer,java.lang.Integer)">getSubSequence</a>, <a href="template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a>, <a href="template/AbstractSequence.html#getUserCollection()">getUserCollection</a>, <a href="template/AbstractSequence.html#hashCode()">hashCode</a>, <a href="template/AbstractSequence.html#initSequenceStorage(java.lang.String)">initSequenceStorage</a>, <a href="template/AbstractSequence.html#iterator()">iterator</a>, <a href="template/AbstractSequence.html#removeFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">removeFeature</a>, <a href="template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a>, <a href="template/AbstractSequence.html#setAccession(org.biojava.nbio.core.sequence.AccessionID)">setAccession</a>, <a href="template/AbstractSequence.html#setAnnotationType(org.biojava.nbio.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a>, <a href="template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a>, <a href="template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a>, <a href="template/AbstractSequence.html#setComments(java.util.List)">setComments</a>, <a href="template/AbstractSequence.html#setCompoundSet(org.biojava.nbio.core.sequence.template.CompoundSet)">setCompoundSet</a>, <a href="template/AbstractSequence.html#setDatabaseReferences(org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a>, <a href="template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a>, <a href="template/AbstractSequence.html#setFeatureRetriever(org.biojava.nbio.core.sequence.features.FeatureRetriever)">setFeatureRetriever</a>, <a href="template/AbstractSequence.html#setFeaturesKeyWord(org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a>, <a href="template/AbstractSequence.html#setNotesList(java.util.List)">setNotesList</a>, <a href="template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a>, <a href="template/AbstractSequence.html#setParentSequence(org.biojava.nbio.core.sequence.template.AbstractSequence)">setParentSequence</a>, <a href="template/AbstractSequence.html#setProxySequenceReader(org.biojava.nbio.core.sequence.template.SequenceReader)">setProxySequenceReader</a>, <a href="template/AbstractSequence.html#setReferences(java.util.List)">setReferences</a>, <a href="template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a>, <a href="template/AbstractSequence.html#setSource(java.lang.String)">setSource</a>, <a href="template/AbstractSequence.html#setTaxonomy(org.biojava.nbio.core.sequence.TaxonomyID)">setTaxonomy</a>, <a href="template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a>, <a href="template/AbstractSequence.html#toString()">toString</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#clone()" title="class or interface in java.lang" class="external-link">clone</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#finalize()" title="class or interface in java.lang" class="external-link">finalize</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Iterable">Methods inherited from interface&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html" title="class or interface in java.lang" class="external-link">Iterable</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#forEach(java.util.function.Consumer)" title="class or interface in java.lang" class="external-link">forEach</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Iterable.html#spliterator()" title="class or interface in java.lang" class="external-link">spliterator</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>ChromosomeSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-47">ChromosomeSequence</a></span>()</div>
<div class="block">Empty constructor used by tools that need a proper Bean that allows the actual
 sequence data to be set after construction. Not recommended</div>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(java.lang.String)">
<h3>ChromosomeSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-56">ChromosomeSequence</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString)</span>
                   throws <span class="exceptions"><a href="../exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="block">String is king and assume DNA</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seqString</code> - </dd>
<dt>Throws:</dt>
<dd><code><a href="../exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(org.biojava.nbio.core.sequence.template.SequenceReader)">
<h3>ChromosomeSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-67">ChromosomeSequence</a></span><wbr><span class="parameters">(<a href="template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader)</span></div>
<div class="block">Fairly important constructor given the size of a ChromsomeSequence where the
 ProxySequenceReader could load from disk via RandomAccessFile so that the sequence
 doesn't need to be kept in memory. Could also be a NCBI proxy to load sequence
 data as needed from remote web server.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>proxyLoader</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(java.lang.String,org.biojava.nbio.core.sequence.template.CompoundSet)">
<h3>ChromosomeSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-77">ChromosomeSequence</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;seqString,
 <a href="template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</span>
                   throws <span class="exceptions"><a href="../exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></span></div>
<div class="block">Allows the creation of a ChromosomeSequence using String for the sequence with a custom CompoundSet</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seqString</code> - </dd>
<dd><code>compoundSet</code> - </dd>
<dt>Throws:</dt>
<dd><code><a href="../exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(org.biojava.nbio.core.sequence.template.SequenceReader,org.biojava.nbio.core.sequence.template.CompoundSet)">
<h3>ChromosomeSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-86">ChromosomeSequence</a></span><wbr><span class="parameters">(<a href="template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;proxyLoader,
 <a href="template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;compoundSet)</span></div>
<div class="block">Allows the creation of a ChromosomeSequence using a ProxyResequenceReader for the sequence with a custom CompoundSet</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>proxyLoader</code> - </dd>
<dd><code>compoundSet</code> - </dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getChromosomeNumber()">
<h3>getChromosomeNumber</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-93">getChromosomeNumber</a></span>()</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the chromosomeNumber</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setChromosomeNumber(int)">
<h3>setChromosomeNumber</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-100">setChromosomeNumber</a></span><wbr><span class="parameters">(int&nbsp;chromosomeNumber)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromosomeNumber</code> - the chromosomeNumber to set</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGeneSequences()">
<h3>getGeneSequences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html" title="class or interface in java.util" class="external-link">Map</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>,<wbr><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-110">getGeneSequences</a></span>()</div>
<div class="block">Get the list of genes that have been added to the ChromosomeSequence where accession.toString is the key.
 The list retains the order the genes are added</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="removeGeneSequence(java.lang.String)">
<h3>removeGeneSequence</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-119">removeGeneSequence</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</span></div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="addGene(org.biojava.nbio.core.sequence.AccessionID,int,int,org.biojava.nbio.core.sequence.Strand)">
<h3>addGene</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-135">addGene</a></span><wbr><span class="parameters">(<a href="AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession,
 int&nbsp;bioBegin,
 int&nbsp;bioEnd,
 <a href="Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</span></div>
<div class="block">Add a gene to the chromosome sequence using bioIndexing starts at 1 instead of 0. The
 GeneSequence that is returned will have a reference to parent chromosome sequence
 which actually contains the sequence data. Strand is important for positive and negative
 direction where negative strand means we need reverse complement. If negative strand then
 bioBegin will be greater than bioEnd</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dd><code>bioBegin</code> - </dd>
<dd><code>bioEnd</code> - </dd>
<dd><code>strand</code> - </dd>
<dt>Returns:</dt>
<dd>A GeneSequence</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGene(java.lang.String)">
<h3>getGene</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/core/sequence/ChromosomeSequence.html#line-147">getGene</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;accession)</span></div>
<div class="block">Get the gene based on accession. Will return null if not found</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>accession</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
