<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.bio.chromatogram;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.util.ArrayList;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.util.HashMap;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.util.Iterator;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.util.List;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.util.Map;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.util.NoSuchElementException;<a name="line.29"></a>
<span class="sourceLineNo">030</span><a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.bio.BioError;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.bio.alignment.Alignment;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.bio.alignment.SimpleAlignment;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.biojava.bio.seq.DNATools;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.bio.symbol.Alphabet;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.bio.symbol.AtomicSymbol;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.bio.symbol.IllegalAlphabetException;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.bio.symbol.IllegalSymbolException;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.bio.symbol.IntegerAlphabet;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.bio.symbol.SimpleSymbolList;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.bio.symbol.SymbolList;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.bio.symbol.SymbolListViews;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.biojava.utils.ChangeListener;<a name="line.43"></a>
<span class="sourceLineNo">044</span><a name="line.44"></a>
<span class="sourceLineNo">045</span>/**<a name="line.45"></a>
<span class="sourceLineNo">046</span> * A basic, abstract implementation of {@link Chromatogram}.  Provides<a name="line.46"></a>
<span class="sourceLineNo">047</span> * protected setters so that subclasses may set the value of the various<a name="line.47"></a>
<span class="sourceLineNo">048</span> * properties of a chromatogram.<a name="line.48"></a>
<span class="sourceLineNo">049</span> *<a name="line.49"></a>
<span class="sourceLineNo">050</span> * Chromatograms should be created using {@link ChromatogramFactory} or a<a name="line.50"></a>
<span class="sourceLineNo">051</span> * parser for a particular file format.<a name="line.51"></a>
<span class="sourceLineNo">052</span> * <a name="line.52"></a>
<span class="sourceLineNo">053</span> * @author Rhett Sutphin (&lt;a href="http://genome.uiowa.edu/"&gt;UI CBCB&lt;/a&gt;)<a name="line.53"></a>
<span class="sourceLineNo">054</span> * @author Matthew Pocock<a name="line.54"></a>
<span class="sourceLineNo">055</span> * @since 1.3<a name="line.55"></a>
<span class="sourceLineNo">056</span> *<a name="line.56"></a>
<span class="sourceLineNo">057</span> * <a name="line.57"></a>
<span class="sourceLineNo">058</span> */<a name="line.58"></a>
<span class="sourceLineNo">059</span>public abstract class AbstractChromatogram implements Chromatogram {<a name="line.59"></a>
<span class="sourceLineNo">060</span>    private static int A = 0; // Array index for 'A'-related material<a name="line.60"></a>
<span class="sourceLineNo">061</span>    private static int C = 1; // Array index for 'C'-related material<a name="line.61"></a>
<span class="sourceLineNo">062</span>    private static int G = 2; // Array index for 'G'-related material<a name="line.62"></a>
<span class="sourceLineNo">063</span>    private static int T = 3; // Array index for 'T'-related material<a name="line.63"></a>
<span class="sourceLineNo">064</span><a name="line.64"></a>
<span class="sourceLineNo">065</span>    /** 2D array containing all the samples for all 4 traces. */<a name="line.65"></a>
<span class="sourceLineNo">066</span>    private int[][] traceSample;<a name="line.66"></a>
<span class="sourceLineNo">067</span>    /** Array containing the single highest value for each trace. */<a name="line.67"></a>
<span class="sourceLineNo">068</span>    private int[] maxTraceValue;<a name="line.68"></a>
<span class="sourceLineNo">069</span>    /** The immutable Alignment that will be returned by getBaseCalls(). */<a name="line.69"></a>
<span class="sourceLineNo">070</span>    private Alignment baseCalls;<a name="line.70"></a>
<span class="sourceLineNo">071</span>    private int significantBits;<a name="line.71"></a>
<span class="sourceLineNo">072</span><a name="line.72"></a>
<span class="sourceLineNo">073</span>  /**<a name="line.73"></a>
<span class="sourceLineNo">074</span>   * Create a new AbstractChromatogram.<a name="line.74"></a>
<span class="sourceLineNo">075</span>   */<a name="line.75"></a>
<span class="sourceLineNo">076</span>    public AbstractChromatogram() {<a name="line.76"></a>
<span class="sourceLineNo">077</span>        maxTraceValue = new int[4];<a name="line.77"></a>
<span class="sourceLineNo">078</span>        traceSample = new int[4][];<a name="line.78"></a>
<span class="sourceLineNo">079</span>        significantBits = 0;<a name="line.79"></a>
<span class="sourceLineNo">080</span>    }<a name="line.80"></a>
<span class="sourceLineNo">081</span><a name="line.81"></a>
<span class="sourceLineNo">082</span>    public int[] getTrace(AtomicSymbol nucleotide)<a name="line.82"></a>
<span class="sourceLineNo">083</span>    throws IllegalSymbolException {<a name="line.83"></a>
<span class="sourceLineNo">084</span>        return traceSample[nucToIndex(nucleotide)];<a name="line.84"></a>
<span class="sourceLineNo">085</span>    }<a name="line.85"></a>
<span class="sourceLineNo">086</span><a name="line.86"></a>
<span class="sourceLineNo">087</span>    public int getTraceLength() {<a name="line.87"></a>
<span class="sourceLineNo">088</span>        return traceSample[0].length;<a name="line.88"></a>
<span class="sourceLineNo">089</span>    }<a name="line.89"></a>
<span class="sourceLineNo">090</span><a name="line.90"></a>
<span class="sourceLineNo">091</span>    public int getMax() {<a name="line.91"></a>
<span class="sourceLineNo">092</span>        try {<a name="line.92"></a>
<span class="sourceLineNo">093</span>            return Math.max(<a name="line.93"></a>
<span class="sourceLineNo">094</span>                    Math.max(getMax(DNATools.a()), getMax(DNATools.c())),<a name="line.94"></a>
<span class="sourceLineNo">095</span>                    Math.max(getMax(DNATools.g()), getMax(DNATools.t()))<a name="line.95"></a>
<span class="sourceLineNo">096</span>                   );<a name="line.96"></a>
<span class="sourceLineNo">097</span>        } catch (IllegalSymbolException ise) {<a name="line.97"></a>
<span class="sourceLineNo">098</span>            throw new BioError("Can't happen", ise);<a name="line.98"></a>
<span class="sourceLineNo">099</span>        }<a name="line.99"></a>
<span class="sourceLineNo">100</span>    }<a name="line.100"></a>
<span class="sourceLineNo">101</span><a name="line.101"></a>
<span class="sourceLineNo">102</span>    public int getMax(AtomicSymbol nucleotide) throws IllegalSymbolException {<a name="line.102"></a>
<span class="sourceLineNo">103</span>        return maxTraceValue[nucToIndex(nucleotide)];<a name="line.103"></a>
<span class="sourceLineNo">104</span>    }<a name="line.104"></a>
<span class="sourceLineNo">105</span><a name="line.105"></a>
<span class="sourceLineNo">106</span>  /**<a name="line.106"></a>
<span class="sourceLineNo">107</span>   * Return the total number of base calls.<a name="line.107"></a>
<span class="sourceLineNo">108</span>   *<a name="line.108"></a>
<span class="sourceLineNo">109</span>   * @return the total number of base calls<a name="line.109"></a>
<span class="sourceLineNo">110</span>   */<a name="line.110"></a>
<span class="sourceLineNo">111</span>    public Alignment getBaseCalls()  { return baseCalls; }<a name="line.111"></a>
<span class="sourceLineNo">112</span><a name="line.112"></a>
<span class="sourceLineNo">113</span>  /**<a name="line.113"></a>
<span class="sourceLineNo">114</span>   * Return the sequence length.<a name="line.114"></a>
<span class="sourceLineNo">115</span>   *<a name="line.115"></a>
<span class="sourceLineNo">116</span>   * @return the sequence length<a name="line.116"></a>
<span class="sourceLineNo">117</span>   */<a name="line.117"></a>
<span class="sourceLineNo">118</span>    public int getSequenceLength()   { return baseCalls.length(); }<a name="line.118"></a>
<span class="sourceLineNo">119</span><a name="line.119"></a>
<span class="sourceLineNo">120</span>  /**<a name="line.120"></a>
<span class="sourceLineNo">121</span>   * Return the number of significant bits.<a name="line.121"></a>
<span class="sourceLineNo">122</span>   *<a name="line.122"></a>
<span class="sourceLineNo">123</span>   * @return the significant bits<a name="line.123"></a>
<span class="sourceLineNo">124</span>   */<a name="line.124"></a>
<span class="sourceLineNo">125</span>    public int getSignificantBits()  { return significantBits; }<a name="line.125"></a>
<span class="sourceLineNo">126</span><a name="line.126"></a>
<span class="sourceLineNo">127</span>    /**<a name="line.127"></a>
<span class="sourceLineNo">128</span>     * Convert a DNA symbol to an internal array index.<a name="line.128"></a>
<span class="sourceLineNo">129</span>     *<a name="line.129"></a>
<span class="sourceLineNo">130</span>     * @param nuc  the nucleotide to convert<a name="line.130"></a>
<span class="sourceLineNo">131</span>     * @return an integer giving it's internal index value<a name="line.131"></a>
<span class="sourceLineNo">132</span>     * @throws IllegalSymbolException if the symbol is not recognised<a name="line.132"></a>
<span class="sourceLineNo">133</span>     */<a name="line.133"></a>
<span class="sourceLineNo">134</span>    private final int nucToIndex(AtomicSymbol nuc) throws IllegalSymbolException {<a name="line.134"></a>
<span class="sourceLineNo">135</span>        if      (nuc == DNATools.a()) return A;<a name="line.135"></a>
<span class="sourceLineNo">136</span>        else if (nuc == DNATools.c()) return C;<a name="line.136"></a>
<span class="sourceLineNo">137</span>        else if (nuc == DNATools.g()) return G;<a name="line.137"></a>
<span class="sourceLineNo">138</span>        else if (nuc == DNATools.t()) return T;<a name="line.138"></a>
<span class="sourceLineNo">139</span>        else<a name="line.139"></a>
<span class="sourceLineNo">140</span>            throw new IllegalSymbolException("The symbol " +<a name="line.140"></a>
<span class="sourceLineNo">141</span>                nuc.getName() + " (" +<a name="line.141"></a>
<span class="sourceLineNo">142</span>                nuc.getClass().getName() +<a name="line.142"></a>
<span class="sourceLineNo">143</span>                ") is not in the DNA alphabet");<a name="line.143"></a>
<span class="sourceLineNo">144</span>    }<a name="line.144"></a>
<span class="sourceLineNo">145</span><a name="line.145"></a>
<span class="sourceLineNo">146</span>    /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * *<a name="line.146"></a>
<span class="sourceLineNo">147</span>     * Protected mutators for subclasses to initialize fields<a name="line.147"></a>
<span class="sourceLineNo">148</span>     * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */<a name="line.148"></a>
<span class="sourceLineNo">149</span><a name="line.149"></a>
<span class="sourceLineNo">150</span>    /**<a name="line.150"></a>
<span class="sourceLineNo">151</span>     * Provides the list of base calls.<a name="line.151"></a>
<span class="sourceLineNo">152</span>     * @param align the base call alignment<a name="line.152"></a>
<span class="sourceLineNo">153</span>     * @throws NoSuchElementException when align doesn't contain alignments with<a name="line.153"></a>
<span class="sourceLineNo">154</span>     *         the required DNA and OFFSETS labels<a name="line.154"></a>
<span class="sourceLineNo">155</span>     * @throws IllegalArgumentException the lists in align aren't all the same length.<a name="line.155"></a>
<span class="sourceLineNo">156</span>     * @throws IllegalAlphabetException if the required lists don't have the<a name="line.156"></a>
<span class="sourceLineNo">157</span>     *         correct alphabets.  See the documentation of<a name="line.157"></a>
<span class="sourceLineNo">158</span>     *         {@link Chromatogram#getBaseCalls() } for details.<a name="line.158"></a>
<span class="sourceLineNo">159</span>     * @see Chromatogram#getBaseCalls<a name="line.159"></a>
<span class="sourceLineNo">160</span>     */<a name="line.160"></a>
<span class="sourceLineNo">161</span>    protected final void setBaseCallAlignment(Alignment align)<a name="line.161"></a>
<span class="sourceLineNo">162</span>    throws IllegalAlphabetException, IllegalArgumentException,<a name="line.162"></a>
<span class="sourceLineNo">163</span>           NoSuchElementException {<a name="line.163"></a>
<span class="sourceLineNo">164</span>        SymbolList dna, offsets;<a name="line.164"></a>
<span class="sourceLineNo">165</span>        try {<a name="line.165"></a>
<span class="sourceLineNo">166</span>            dna = align.symbolListForLabel(Chromatogram.DNA);<a name="line.166"></a>
<span class="sourceLineNo">167</span>            offsets = align.symbolListForLabel(Chromatogram.OFFSETS);<a name="line.167"></a>
<span class="sourceLineNo">168</span>        } catch (NoSuchElementException nsee) {<a name="line.168"></a>
<span class="sourceLineNo">169</span>            throw nsee;<a name="line.169"></a>
<span class="sourceLineNo">170</span>        }<a name="line.170"></a>
<span class="sourceLineNo">171</span>        if (dna.getAlphabet() != DNATools.getDNA())<a name="line.171"></a>
<span class="sourceLineNo">172</span>            throw new IllegalAlphabetException("DNA list has inappropriate alphabet");<a name="line.172"></a>
<span class="sourceLineNo">173</span>        if (! (offsets.getAlphabet() instanceof IntegerAlphabet ||<a name="line.173"></a>
<span class="sourceLineNo">174</span>               offsets.getAlphabet() instanceof IntegerAlphabet.SubIntegerAlphabet))<a name="line.174"></a>
<span class="sourceLineNo">175</span>            throw new IllegalAlphabetException("Offsets list has inappropriate alphabet");<a name="line.175"></a>
<span class="sourceLineNo">176</span><a name="line.176"></a>
<span class="sourceLineNo">177</span>        baseCalls = align;<a name="line.177"></a>
<span class="sourceLineNo">178</span>    }<a name="line.178"></a>
<span class="sourceLineNo">179</span><a name="line.179"></a>
<span class="sourceLineNo">180</span>    /**<a name="line.180"></a>
<span class="sourceLineNo">181</span>     * Sets the trace data structures to null.  If a subclass needs to replace<a name="line.181"></a>
<span class="sourceLineNo">182</span>     * the traces with new traces of a different length, this method must be<a name="line.182"></a>
<span class="sourceLineNo">183</span>     * called first to avoid provoking an IllegalArgumentException from setTrace<a name="line.183"></a>
<span class="sourceLineNo">184</span>     * due to a length mismatch.<a name="line.184"></a>
<span class="sourceLineNo">185</span>     */<a name="line.185"></a>
<span class="sourceLineNo">186</span>    protected final void clearTraces() {<a name="line.186"></a>
<span class="sourceLineNo">187</span>        maxTraceValue[A] = maxTraceValue[C] = maxTraceValue[G]<a name="line.187"></a>
<span class="sourceLineNo">188</span>                         = maxTraceValue[T] = Integer.MIN_VALUE;<a name="line.188"></a>
<span class="sourceLineNo">189</span>        traceSample[A] = traceSample[C] = traceSample[G]<a name="line.189"></a>
<span class="sourceLineNo">190</span>                       = traceSample[T] = null;<a name="line.190"></a>
<span class="sourceLineNo">191</span>    }<a name="line.191"></a>
<span class="sourceLineNo">192</span><a name="line.192"></a>
<span class="sourceLineNo">193</span>    /**<a name="line.193"></a>
<span class="sourceLineNo">194</span>     * Provides the trace samples for a particular nucleotide.<a name="line.194"></a>
<span class="sourceLineNo">195</span>     * @param nuc A DNA nucleotide<a name="line.195"></a>
<span class="sourceLineNo">196</span>     * @param trace the trace samples themselves<a name="line.196"></a>
<span class="sourceLineNo">197</span>     * @param maxVal the maximum value in the trace array.  If this value<a name="line.197"></a>
<span class="sourceLineNo">198</span>     *        is {@link Integer#MIN_VALUE}, this method will do a linear<a name="line.198"></a>
<span class="sourceLineNo">199</span>     *        search of trace to determine the max.<a name="line.199"></a>
<span class="sourceLineNo">200</span>     * @throws IllegalArgumentException when trace.length is different<a name="line.200"></a>
<span class="sourceLineNo">201</span>     *         from any of the existing (non-null) traces<a name="line.201"></a>
<span class="sourceLineNo">202</span>     * @throws IllegalSymbolException when nuc is not a concrete DNA nucleotide<a name="line.202"></a>
<span class="sourceLineNo">203</span>     */<a name="line.203"></a>
<span class="sourceLineNo">204</span>    protected final void setTrace(AtomicSymbol nuc, int[] trace, int maxVal)<a name="line.204"></a>
<span class="sourceLineNo">205</span>    throws IllegalArgumentException, IllegalSymbolException {<a name="line.205"></a>
<span class="sourceLineNo">206</span>        int idx = nucToIndex(nuc);<a name="line.206"></a>
<span class="sourceLineNo">207</span>        if (trace == null) {<a name="line.207"></a>
<span class="sourceLineNo">208</span>            traceSample[idx] = null;<a name="line.208"></a>
<span class="sourceLineNo">209</span>            maxTraceValue[idx] = Integer.MIN_VALUE;<a name="line.209"></a>
<span class="sourceLineNo">210</span>        }<a name="line.210"></a>
<span class="sourceLineNo">211</span>        else {<a name="line.211"></a>
<span class="sourceLineNo">212</span>            // check for length mismatches<a name="line.212"></a>
<span class="sourceLineNo">213</span>            for (int i = 0 ; i &lt; 4 ; i++) {<a name="line.213"></a>
<span class="sourceLineNo">214</span>                if (traceSample[i] != null &amp;&amp; traceSample[i].length != trace.length) {<a name="line.214"></a>
<span class="sourceLineNo">215</span>                    throw new IllegalArgumentException(<a name="line.215"></a>
<span class="sourceLineNo">216</span>                        "All traces must be the same length.  " + trace.length  +<a name="line.216"></a>
<span class="sourceLineNo">217</span>                        " != " + traceSample[i].length);<a name="line.217"></a>
<span class="sourceLineNo">218</span>                }<a name="line.218"></a>
<span class="sourceLineNo">219</span>            }<a name="line.219"></a>
<span class="sourceLineNo">220</span>            // special case to do linear search for max<a name="line.220"></a>
<span class="sourceLineNo">221</span>            if (maxVal == Integer.MIN_VALUE) {<a name="line.221"></a>
<span class="sourceLineNo">222</span>                maxVal = trace[0];<a name="line.222"></a>
<span class="sourceLineNo">223</span>                for (int i = 1 ; i &lt; trace.length ; i++)<a name="line.223"></a>
<span class="sourceLineNo">224</span>                    maxVal = Math.max(maxVal, trace[i]);<a name="line.224"></a>
<span class="sourceLineNo">225</span>            }<a name="line.225"></a>
<span class="sourceLineNo">226</span>            traceSample[idx] = trace;<a name="line.226"></a>
<span class="sourceLineNo">227</span>            maxTraceValue[idx] = maxVal;<a name="line.227"></a>
<span class="sourceLineNo">228</span>        }<a name="line.228"></a>
<span class="sourceLineNo">229</span>    }<a name="line.229"></a>
<span class="sourceLineNo">230</span><a name="line.230"></a>
<span class="sourceLineNo">231</span>    /** Sets the number of significant bits in the trace samples.<a name="line.231"></a>
<span class="sourceLineNo">232</span>     *  @param bits a non-negative integer indicating the number of<a name="line.232"></a>
<span class="sourceLineNo">233</span>     *         significant bits in each trace sample<a name="line.233"></a>
<span class="sourceLineNo">234</span>     *  @throws IllegalArgumentException when &lt;code&gt;bits&lt;/code&gt; is negative<a name="line.234"></a>
<span class="sourceLineNo">235</span>     */<a name="line.235"></a>
<span class="sourceLineNo">236</span>    protected final void setBits(int bits)<a name="line.236"></a>
<span class="sourceLineNo">237</span>    throws IllegalArgumentException {<a name="line.237"></a>
<span class="sourceLineNo">238</span>        if (bits &lt; 0)<a name="line.238"></a>
<span class="sourceLineNo">239</span>            throw new IllegalArgumentException("Invalid number of significant bits");<a name="line.239"></a>
<span class="sourceLineNo">240</span>        this.significantBits = bits;<a name="line.240"></a>
<span class="sourceLineNo">241</span>    }<a name="line.241"></a>
<span class="sourceLineNo">242</span><a name="line.242"></a>
<span class="sourceLineNo">243</span>    /** Returns a new instance of this AbstractChromatogram subclass for use in<a name="line.243"></a>
<span class="sourceLineNo">244</span>     * {@link #reverseComplement}.<a name="line.244"></a>
<span class="sourceLineNo">245</span>     *<a name="line.245"></a>
<span class="sourceLineNo">246</span>     * @return a reverse-complemented AbstractChromatogram<a name="line.246"></a>
<span class="sourceLineNo">247</span>     */<a name="line.247"></a>
<span class="sourceLineNo">248</span>    protected abstract AbstractChromatogram reverseComplementInstance();<a name="line.248"></a>
<span class="sourceLineNo">249</span><a name="line.249"></a>
<span class="sourceLineNo">250</span>    public Chromatogram reverseComplement() {<a name="line.250"></a>
<span class="sourceLineNo">251</span>        AbstractChromatogram rev = this.reverseComplementInstance();<a name="line.251"></a>
<span class="sourceLineNo">252</span>        try {<a name="line.252"></a>
<span class="sourceLineNo">253</span>            rev.setTrace(DNATools.a(), reverse(traceSample[T]), maxTraceValue[T]);<a name="line.253"></a>
<span class="sourceLineNo">254</span>            rev.setTrace(DNATools.c(), reverse(traceSample[G]), maxTraceValue[G]);<a name="line.254"></a>
<span class="sourceLineNo">255</span>            rev.setTrace(DNATools.g(), reverse(traceSample[C]), maxTraceValue[C]);<a name="line.255"></a>
<span class="sourceLineNo">256</span>            rev.setTrace(DNATools.t(), reverse(traceSample[A]), maxTraceValue[A]);<a name="line.256"></a>
<span class="sourceLineNo">257</span>        } catch (IllegalSymbolException ise) {<a name="line.257"></a>
<span class="sourceLineNo">258</span>            throw new BioError( "Can't happen -- all symbols are explicit and legal", ise);<a name="line.258"></a>
<span class="sourceLineNo">259</span>        }<a name="line.259"></a>
<span class="sourceLineNo">260</span>        Alignment revBC = reverseComplementBaseCalls();<a name="line.260"></a>
<span class="sourceLineNo">261</span>        try {<a name="line.261"></a>
<span class="sourceLineNo">262</span>            rev.setBaseCallAlignment(revBC);<a name="line.262"></a>
<span class="sourceLineNo">263</span>        } catch (IllegalAlphabetException iae) {<a name="line.263"></a>
<span class="sourceLineNo">264</span>            throw new BioError("Can't happen unless reverseComplementBaseCalls or reverseComplementBaseCallList have been overridden out-of-spec", iae);<a name="line.264"></a>
<span class="sourceLineNo">265</span>        } catch (NoSuchElementException nsee) {<a name="line.265"></a>
<span class="sourceLineNo">266</span>            throw new BioError("Can't happen unless reverseComplementBaseCalls or reverseComplementBaseCallList have been overridden out-of-spec", nsee);<a name="line.266"></a>
<span class="sourceLineNo">267</span>        } catch (IllegalArgumentException iae) {<a name="line.267"></a>
<span class="sourceLineNo">268</span>            throw new BioError("Can't happen unless reverseComplementBaseCalls or reverseComplementBaseCallList have been overridden out-of-spec", iae);<a name="line.268"></a>
<span class="sourceLineNo">269</span>        }<a name="line.269"></a>
<span class="sourceLineNo">270</span>        rev.setBits(this.significantBits);<a name="line.270"></a>
<span class="sourceLineNo">271</span>        return rev;<a name="line.271"></a>
<span class="sourceLineNo">272</span>    }<a name="line.272"></a>
<span class="sourceLineNo">273</span><a name="line.273"></a>
<span class="sourceLineNo">274</span>    /**<a name="line.274"></a>
<span class="sourceLineNo">275</span>     * Returns a new base call alignment that is the reverse complement of<a name="line.275"></a>
<span class="sourceLineNo">276</span>     * one in this chromatogram.  This is achieved by calling<a name="line.276"></a>
<span class="sourceLineNo">277</span>     * {@link #reverseComplementBaseCallList} for each label in the current<a name="line.277"></a>
<span class="sourceLineNo">278</span>     * base call alignment.  When that method returns null, no list will<a name="line.278"></a>
<span class="sourceLineNo">279</span>     * appear in reverse complement base call alignment with the null-provoking<a name="line.279"></a>
<span class="sourceLineNo">280</span>     * label.  For this reason, subclasses are encouraged to override<a name="line.280"></a>
<span class="sourceLineNo">281</span>     * {@link #reverseComplementBaseCallList} to handle any additional per-base<a name="line.281"></a>
<span class="sourceLineNo">282</span>     * metadata that they store.<a name="line.282"></a>
<span class="sourceLineNo">283</span>     * &lt;p&gt;<a name="line.283"></a>
<span class="sourceLineNo">284</span>     * This implementation should be safely inheritable<a name="line.284"></a>
<span class="sourceLineNo">285</span>     * for all chromatogram implementations, unless one just doesn't want<a name="line.285"></a>
<span class="sourceLineNo">286</span>     * base calls on its reverse complement output.  If this is the case,<a name="line.286"></a>
<span class="sourceLineNo">287</span>     * it should override this method to return null.<a name="line.287"></a>
<span class="sourceLineNo">288</span>     * &lt;/p&gt;<a name="line.288"></a>
<span class="sourceLineNo">289</span>     *<a name="line.289"></a>
<span class="sourceLineNo">290</span>     * @return a new {@link Alignment} that is the reverse complement of the<a name="line.290"></a>
<span class="sourceLineNo">291</span>     *          one in the current chromatogram<a name="line.291"></a>
<span class="sourceLineNo">292</span>     */<a name="line.292"></a>
<span class="sourceLineNo">293</span>    protected Alignment reverseComplementBaseCalls() {<a name="line.293"></a>
<span class="sourceLineNo">294</span>        Alignment current = this.getBaseCalls();<a name="line.294"></a>
<span class="sourceLineNo">295</span>        Map revBCMap = new HashMap();<a name="line.295"></a>
<span class="sourceLineNo">296</span>        Object curLabel;<a name="line.296"></a>
<span class="sourceLineNo">297</span>        SymbolList revList;<a name="line.297"></a>
<span class="sourceLineNo">298</span>        for (Iterator it = current.getLabels().iterator() ; it.hasNext() ;) {<a name="line.298"></a>
<span class="sourceLineNo">299</span>            curLabel = it.next();<a name="line.299"></a>
<span class="sourceLineNo">300</span>            revList = reverseComplementBaseCallList(curLabel);<a name="line.300"></a>
<span class="sourceLineNo">301</span>            if (revList != null)<a name="line.301"></a>
<span class="sourceLineNo">302</span>                revBCMap.put(curLabel, revList);<a name="line.302"></a>
<span class="sourceLineNo">303</span>        }<a name="line.303"></a>
<span class="sourceLineNo">304</span>        return createImmutableAlignment(revBCMap);<a name="line.304"></a>
<span class="sourceLineNo">305</span>    }<a name="line.305"></a>
<span class="sourceLineNo">306</span><a name="line.306"></a>
<span class="sourceLineNo">307</span>    /**<a name="line.307"></a>
<span class="sourceLineNo">308</span>     * Return a symbol list containing the reverse complement of the base call<a name="line.308"></a>
<span class="sourceLineNo">309</span>     * data for the given label.  The returned list will be stored in the<a name="line.309"></a>
<span class="sourceLineNo">310</span>     * reverse complement's base call alignment under the same label.<a name="line.310"></a>
<span class="sourceLineNo">311</span>     * &lt;p&gt;<a name="line.311"></a>
<span class="sourceLineNo">312</span>     * Implementation note: subclasses which do not use an {@link IntegerAlphabet} for<a name="line.312"></a>
<span class="sourceLineNo">313</span>     * their offsets lists must override this method, at least for the case where<a name="line.313"></a>
<span class="sourceLineNo">314</span>     * &lt;code&gt;label == {@link #OFFSETS}&lt;/code&gt;.<a name="line.314"></a>
<span class="sourceLineNo">315</span>     * &lt;/p&gt;<a name="line.315"></a>
<span class="sourceLineNo">316</span>     *<a name="line.316"></a>
<span class="sourceLineNo">317</span>     * @param label the label Object<a name="line.317"></a>
<span class="sourceLineNo">318</span>     * @return an appropriately reverse-complemented SymbolList, or null if the<a name="line.318"></a>
<span class="sourceLineNo">319</span>     *          label is unhandled.<a name="line.319"></a>
<span class="sourceLineNo">320</span>     */<a name="line.320"></a>
<span class="sourceLineNo">321</span>    protected SymbolList reverseComplementBaseCallList(Object label) {<a name="line.321"></a>
<span class="sourceLineNo">322</span>        if (label == DNA) {<a name="line.322"></a>
<span class="sourceLineNo">323</span>            try {<a name="line.323"></a>
<span class="sourceLineNo">324</span>                return SymbolListViews.reverse(DNATools.complement(this.getBaseCalls().symbolListForLabel(DNA)));<a name="line.324"></a>
<span class="sourceLineNo">325</span>            } catch (IllegalAlphabetException iae) {<a name="line.325"></a>
<span class="sourceLineNo">326</span>                throw new BioError("Can't happen unless the DNA list has been set improperly", iae);<a name="line.326"></a>
<span class="sourceLineNo">327</span>            }<a name="line.327"></a>
<span class="sourceLineNo">328</span>        }<a name="line.328"></a>
<span class="sourceLineNo">329</span>        else if (label == OFFSETS) {<a name="line.329"></a>
<span class="sourceLineNo">330</span>            SymbolList curOffsets = this.getBaseCalls().symbolListForLabel(OFFSETS);<a name="line.330"></a>
<span class="sourceLineNo">331</span>            List revOffsets = new ArrayList(curOffsets.length());<a name="line.331"></a>
<span class="sourceLineNo">332</span>            IntegerAlphabet alpha = (IntegerAlphabet) curOffsets.getAlphabet();<a name="line.332"></a>
<span class="sourceLineNo">333</span>            IntegerAlphabet.IntegerSymbol sym;<a name="line.333"></a>
<span class="sourceLineNo">334</span>            for (int i = curOffsets.length() ; i &gt; 0 ; i--) {<a name="line.334"></a>
<span class="sourceLineNo">335</span>                sym = (IntegerAlphabet.IntegerSymbol) curOffsets.symbolAt(i);<a name="line.335"></a>
<span class="sourceLineNo">336</span>                revOffsets.add(alpha.getSymbol(this.getTraceLength() - sym.intValue()));<a name="line.336"></a>
<span class="sourceLineNo">337</span>            }<a name="line.337"></a>
<span class="sourceLineNo">338</span>            try {<a name="line.338"></a>
<span class="sourceLineNo">339</span>                return createImmutableSymbolList(alpha, revOffsets);<a name="line.339"></a>
<span class="sourceLineNo">340</span>            } catch (IllegalSymbolException ise) {<a name="line.340"></a>
<span class="sourceLineNo">341</span>                throw new BioError("Can't happen -- revOffsets was just created with only symbols from alpha");<a name="line.341"></a>
<span class="sourceLineNo">342</span>            }<a name="line.342"></a>
<span class="sourceLineNo">343</span>        }<a name="line.343"></a>
<span class="sourceLineNo">344</span>        else {<a name="line.344"></a>
<span class="sourceLineNo">345</span>            return null;<a name="line.345"></a>
<span class="sourceLineNo">346</span>        }<a name="line.346"></a>
<span class="sourceLineNo">347</span>    }<a name="line.347"></a>
<span class="sourceLineNo">348</span><a name="line.348"></a>
<span class="sourceLineNo">349</span>    /**<a name="line.349"></a>
<span class="sourceLineNo">350</span>     * A factory method for creating new symbol lists with a given alphabet.<a name="line.350"></a>
<span class="sourceLineNo">351</span>     * The default implementation should be fine for nearly all cases, but the<a name="line.351"></a>
<span class="sourceLineNo">352</span>     * option is given in case a developer wishes to use a more memory<a name="line.352"></a>
<span class="sourceLineNo">353</span>     * efficient implementation.<a name="line.353"></a>
<span class="sourceLineNo">354</span>     *<a name="line.354"></a>
<span class="sourceLineNo">355</span>     * @param alpha the {@link Alphabet} for the new list<a name="line.355"></a>
<span class="sourceLineNo">356</span>     * @param syms the symbols to put in the new list<a name="line.356"></a>
<span class="sourceLineNo">357</span>     * @throws IllegalSymbolException when alpha and syms are incompatible<a name="line.357"></a>
<span class="sourceLineNo">358</span>     * @throws ClassCastException when any object in syms isn't a Symbol<a name="line.358"></a>
<span class="sourceLineNo">359</span>     * @return a new immutable SymbolList containing all the given symbols using<a name="line.359"></a>
<span class="sourceLineNo">360</span>     *          the given Alphabet<a name="line.360"></a>
<span class="sourceLineNo">361</span>     */<a name="line.361"></a>
<span class="sourceLineNo">362</span>    protected SymbolList createImmutableSymbolList(Alphabet alpha, List syms)<a name="line.362"></a>
<span class="sourceLineNo">363</span>    throws IllegalSymbolException, ClassCastException {<a name="line.363"></a>
<span class="sourceLineNo">364</span>        SymbolList symlist = new SimpleSymbolList(alpha, syms);<a name="line.364"></a>
<span class="sourceLineNo">365</span>        symlist.addChangeListener(ChangeListener.ALWAYS_VETO);<a name="line.365"></a>
<span class="sourceLineNo">366</span>        return symlist;<a name="line.366"></a>
<span class="sourceLineNo">367</span>    }<a name="line.367"></a>
<span class="sourceLineNo">368</span><a name="line.368"></a>
<span class="sourceLineNo">369</span>    /**<a name="line.369"></a>
<span class="sourceLineNo">370</span>     * A factory method for creating new immutable alignments, particularly<a name="line.370"></a>
<span class="sourceLineNo">371</span>     * for use as base call alignments.  The default implementation should<a name="line.371"></a>
<span class="sourceLineNo">372</span>     * be fine for nearly all cases.<a name="line.372"></a>
<span class="sourceLineNo">373</span>     *<a name="line.373"></a>
<span class="sourceLineNo">374</span>     * @param labelsToSymLists a {@link Map} whose keys are desired labels<a name="line.374"></a>
<span class="sourceLineNo">375</span>     *        for the alignment and whose values are the SymbolLists.<a name="line.375"></a>
<span class="sourceLineNo">376</span>     *        All the SymbolLists must be the same length.<a name="line.376"></a>
<span class="sourceLineNo">377</span>     * @return a new Alignment<a name="line.377"></a>
<span class="sourceLineNo">378</span>     * @throws IllegalArgumentException if the lists aren't all the same length<a name="line.378"></a>
<span class="sourceLineNo">379</span>     * @throws ClassCastException if any of the values in the map aren't<a name="line.379"></a>
<span class="sourceLineNo">380</span>     *         SymbolLists<a name="line.380"></a>
<span class="sourceLineNo">381</span>     */<a name="line.381"></a>
<span class="sourceLineNo">382</span>    protected Alignment createImmutableAlignment(Map labelsToSymLists)<a name="line.382"></a>
<span class="sourceLineNo">383</span>    throws IllegalArgumentException, ClassCastException {<a name="line.383"></a>
<span class="sourceLineNo">384</span>        Alignment sa = new SimpleAlignment(labelsToSymLists);<a name="line.384"></a>
<span class="sourceLineNo">385</span>        sa.addChangeListener(ChangeListener.ALWAYS_VETO);<a name="line.385"></a>
<span class="sourceLineNo">386</span>        return sa;<a name="line.386"></a>
<span class="sourceLineNo">387</span>    }<a name="line.387"></a>
<span class="sourceLineNo">388</span><a name="line.388"></a>
<span class="sourceLineNo">389</span>    /**<a name="line.389"></a>
<span class="sourceLineNo">390</span>     *  Utility method for reversing an int[] array. Visible for subclass use.<a name="line.390"></a>
<span class="sourceLineNo">391</span>     *<a name="line.391"></a>
<span class="sourceLineNo">392</span>     * @param src  the source array<a name="line.392"></a>
<span class="sourceLineNo">393</span>     * @return an array of the same length containing all values in src in<a name="line.393"></a>
<span class="sourceLineNo">394</span>     *     reverse order<a name="line.394"></a>
<span class="sourceLineNo">395</span>     */<a name="line.395"></a>
<span class="sourceLineNo">396</span>    protected final static int[] reverse(int[] src) {<a name="line.396"></a>
<span class="sourceLineNo">397</span>        int[] dst = new int[src.length];<a name="line.397"></a>
<span class="sourceLineNo">398</span>        for (int i = 0 ; i &lt; src.length ; i++)<a name="line.398"></a>
<span class="sourceLineNo">399</span>            dst[src.length - i - 1] = src[i];<a name="line.399"></a>
<span class="sourceLineNo">400</span>        return dst;<a name="line.400"></a>
<span class="sourceLineNo">401</span>    }<a name="line.401"></a>
<span class="sourceLineNo">402</span>}<a name="line.402"></a>




























































</pre>
</div>
</body>
</html>
