<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.bio.chromatogram;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.bio.alignment.Alignment;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.bio.symbol.AtomicSymbol;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.bio.symbol.IllegalSymbolException;<a name="line.26"></a>
<span class="sourceLineNo">027</span><a name="line.27"></a>
<span class="sourceLineNo">028</span>/** <a name="line.28"></a>
<span class="sourceLineNo">029</span> * Encapsulates the basic information you would want from a chromatogram.<a name="line.29"></a>
<span class="sourceLineNo">030</span> * Read-only.<a name="line.30"></a>
<span class="sourceLineNo">031</span> *<a name="line.31"></a>
<span class="sourceLineNo">032</span> *  @author Rhett Sutphin (&lt;a href="http://genome.uiowa.edu/"&gt;UI CBCB&lt;/a&gt;)<a name="line.32"></a>
<span class="sourceLineNo">033</span> * @author Matthew Pocock<a name="line.33"></a>
<span class="sourceLineNo">034</span> * @since 1.3<a name="line.34"></a>
<span class="sourceLineNo">035</span> */<a name="line.35"></a>
<span class="sourceLineNo">036</span>public interface Chromatogram {<a name="line.36"></a>
<span class="sourceLineNo">037</span>    /**<a name="line.37"></a>
<span class="sourceLineNo">038</span>     * The sequence label for the list of called bases.<a name="line.38"></a>
<span class="sourceLineNo">039</span>     */<a name="line.39"></a>
<span class="sourceLineNo">040</span>    public static String DNA = "dna";<a name="line.40"></a>
<span class="sourceLineNo">041</span><a name="line.41"></a>
<span class="sourceLineNo">042</span>    /**<a name="line.42"></a>
<span class="sourceLineNo">043</span>     * The sequence label for the trace offsets of the called bases.<a name="line.43"></a>
<span class="sourceLineNo">044</span>     */<a name="line.44"></a>
<span class="sourceLineNo">045</span>    public static String OFFSETS = "trace-offsets";<a name="line.45"></a>
<span class="sourceLineNo">046</span>    <a name="line.46"></a>
<span class="sourceLineNo">047</span>    /** Gets the max intensity from all the traces.  Must be equivalent<a name="line.47"></a>
<span class="sourceLineNo">048</span>     *  to the max of calling {@link #getMax(AtomicSymbol)} on each<a name="line.48"></a>
<span class="sourceLineNo">049</span>     *  of the four non-ambiguous DNA nucleotides.<a name="line.49"></a>
<span class="sourceLineNo">050</span>     *  @return the max intensity <a name="line.50"></a>
<span class="sourceLineNo">051</span>     */<a name="line.51"></a>
<span class="sourceLineNo">052</span>    public int getMax();<a name="line.52"></a>
<span class="sourceLineNo">053</span>    /** Gets the max intensity on the trace for the specified nucleotide.<a name="line.53"></a>
<span class="sourceLineNo">054</span>     *  @param nucleotide the trace to examine.  Must be a concrete <a name="line.54"></a>
<span class="sourceLineNo">055</span>     *         (non-ambiguous) nucleotide from the DNA alphabet<a name="line.55"></a>
<span class="sourceLineNo">056</span>     *  @throws IllegalSymbolException when the nucleotide isn't from the DNA<a name="line.56"></a>
<span class="sourceLineNo">057</span>     *          alphabet<a name="line.57"></a>
<span class="sourceLineNo">058</span>     *  @return the max intensity<a name="line.58"></a>
<span class="sourceLineNo">059</span>     */<a name="line.59"></a>
<span class="sourceLineNo">060</span>    public int getMax(AtomicSymbol nucleotide) throws IllegalSymbolException;<a name="line.60"></a>
<span class="sourceLineNo">061</span>    <a name="line.61"></a>
<span class="sourceLineNo">062</span>    /** Returns the length of the trace of the Chromatogram.<a name="line.62"></a>
<span class="sourceLineNo">063</span>     *  @return the number of samples in the trace<a name="line.63"></a>
<span class="sourceLineNo">064</span>     *  @see #getTrace(AtomicSymbol)<a name="line.64"></a>
<span class="sourceLineNo">065</span>     */<a name="line.65"></a>
<span class="sourceLineNo">066</span>    public int getTraceLength();<a name="line.66"></a>
<span class="sourceLineNo">067</span>    /** Returns an array containing the intensities of the sampled waveform<a name="line.67"></a>
<span class="sourceLineNo">068</span>     *  representing the chromatogram trace for base &lt;code&gt;nucleotide&lt;/code&gt;.  <a name="line.68"></a>
<span class="sourceLineNo">069</span>     *  This may be a reference the actual internal representation of the <a name="line.69"></a>
<span class="sourceLineNo">070</span>     *  samples, so callers &lt;b&gt;must not modify it&lt;/b&gt;.<a name="line.70"></a>
<span class="sourceLineNo">071</span>     *  &lt;p&gt;<a name="line.71"></a>
<span class="sourceLineNo">072</span>     *  The resulting array for each nucleotide must be {@link #getTraceLength}<a name="line.72"></a>
<span class="sourceLineNo">073</span>     *  &lt;code&gt;int&lt;/code&gt;s long.<a name="line.73"></a>
<span class="sourceLineNo">074</span>     *  &lt;/p&gt;<a name="line.74"></a>
<span class="sourceLineNo">075</span>     *  @param nucleotide the trace to examine.  Must be the symbol for A, C, G, or T<a name="line.75"></a>
<span class="sourceLineNo">076</span>     *         as provided by {@link org.biojava.bio.seq.DNATools}<a name="line.76"></a>
<span class="sourceLineNo">077</span>     *  @throws IllegalSymbolException if &lt;code&gt;nucleotide&lt;/code&gt; isn't in the DNA alphabet<a name="line.77"></a>
<span class="sourceLineNo">078</span>     *  @return an array of integers representing the values of a particular <a name="line.78"></a>
<span class="sourceLineNo">079</span>     *          chromatogram trace.<a name="line.79"></a>
<span class="sourceLineNo">080</span>     */<a name="line.80"></a>
<span class="sourceLineNo">081</span>    public int[] getTrace(AtomicSymbol nucleotide) throws IllegalSymbolException;<a name="line.81"></a>
<span class="sourceLineNo">082</span>    <a name="line.82"></a>
<span class="sourceLineNo">083</span>    /** <a name="line.83"></a>
<span class="sourceLineNo">084</span>     * Returns the number of bits of the traces which are significant.  For <a name="line.84"></a>
<span class="sourceLineNo">085</span>     * instance, if the chromatogram were originally encoded with a single byte <a name="line.85"></a>
<span class="sourceLineNo">086</span>     * per trace sample, this method must return 8.<a name="line.86"></a>
<span class="sourceLineNo">087</span>     * @return the number of significant bits<a name="line.87"></a>
<span class="sourceLineNo">088</span>     */<a name="line.88"></a>
<span class="sourceLineNo">089</span>    public int getSignificantBits();<a name="line.89"></a>
<span class="sourceLineNo">090</span><a name="line.90"></a>
<span class="sourceLineNo">091</span>    /**<a name="line.91"></a>
<span class="sourceLineNo">092</span>     * Returns an alignment that describes the base calls for this chromatogram.<a name="line.92"></a>
<span class="sourceLineNo">093</span>     * All of the &lt;code&gt;SymbolList&lt;/code&gt;s in this alignment must be the same<a name="line.93"></a>
<span class="sourceLineNo">094</span>     * length and that length must equal {@link #getSequenceLength}.<a name="line.94"></a>
<span class="sourceLineNo">095</span>     * &lt;p&gt;<a name="line.95"></a>
<span class="sourceLineNo">096</span>     * The alignment must contain, at the least, two sequences:<a name="line.96"></a>
<span class="sourceLineNo">097</span>     * &lt;/p&gt;<a name="line.97"></a>
<span class="sourceLineNo">098</span>     * &lt;ol&gt;<a name="line.98"></a>
<span class="sourceLineNo">099</span>     *   &lt;li&gt;A sequence containing the called bases.  The alphabet of this list<a name="line.99"></a>
<span class="sourceLineNo">100</span>     *       must be {@link org.biojava.bio.seq.DNATools#getDNA()}.  <a name="line.100"></a>
<span class="sourceLineNo">101</span>     *       The label for this list in the alignment must be <a name="line.101"></a>
<span class="sourceLineNo">102</span>     *       &lt;code&gt;Chromatogram.DNA&lt;/code&gt;&lt;/li&gt;<a name="line.102"></a>
<span class="sourceLineNo">103</span>     *   &lt;li&gt;A sequence containing the trace offsets at which the called bases <a name="line.103"></a>
<span class="sourceLineNo">104</span>     *       were called.  The alphabet of this list must be an <a name="line.104"></a>
<span class="sourceLineNo">105</span>     *       {@link org.biojava.bio.symbol.IntegerAlphabet} or a<a name="line.105"></a>
<span class="sourceLineNo">106</span>     *       {@link org.biojava.bio.symbol.IntegerAlphabet.SubIntegerAlphabet}.<a name="line.106"></a>
<span class="sourceLineNo">107</span>     *       The label for this list in the alignment must be <a name="line.107"></a>
<span class="sourceLineNo">108</span>     *       &lt;code&gt;Chromatogram.OFFSETS&lt;/code&gt;.&lt;/li&gt;<a name="line.108"></a>
<span class="sourceLineNo">109</span>     * &lt;/ol&gt;<a name="line.109"></a>
<span class="sourceLineNo">110</span>     * &lt;p&gt;<a name="line.110"></a>
<span class="sourceLineNo">111</span>     * Implementors may provide other sequences as they see fit.<a name="line.111"></a>
<span class="sourceLineNo">112</span>     * &lt;/p&gt;<a name="line.112"></a>
<span class="sourceLineNo">113</span>     * @return an alignment of at least two sequences, as described above.<a name="line.113"></a>
<span class="sourceLineNo">114</span>     */<a name="line.114"></a>
<span class="sourceLineNo">115</span>    public Alignment getBaseCalls();<a name="line.115"></a>
<span class="sourceLineNo">116</span>    /** <a name="line.116"></a>
<span class="sourceLineNo">117</span>     * Returns the number of bases called by whatever base-calling software<a name="line.117"></a>
<span class="sourceLineNo">118</span>     * analyzed the chromatogram as loaded.  Must equal <a name="line.118"></a>
<span class="sourceLineNo">119</span>     * &lt;code&gt;{@link #getBaseCalls}.length()&lt;/code&gt;.<a name="line.119"></a>
<span class="sourceLineNo">120</span>     * @return the number of bases<a name="line.120"></a>
<span class="sourceLineNo">121</span>     */<a name="line.121"></a>
<span class="sourceLineNo">122</span>    public int getSequenceLength();<a name="line.122"></a>
<span class="sourceLineNo">123</span>    <a name="line.123"></a>
<span class="sourceLineNo">124</span>    /** <a name="line.124"></a>
<span class="sourceLineNo">125</span>     * Returns a new &lt;code&gt;Chromatogram&lt;/code&gt; representing the reverse<a name="line.125"></a>
<span class="sourceLineNo">126</span>     * complement of this one.<a name="line.126"></a>
<span class="sourceLineNo">127</span>     * &lt;p&gt;<a name="line.127"></a>
<span class="sourceLineNo">128</span>     * Implementors should copy the metadata about the chromatogram (i.e., base <a name="line.128"></a>
<span class="sourceLineNo">129</span>     * calls) as is appropriate to their formats.<a name="line.129"></a>
<span class="sourceLineNo">130</span>     * &lt;/p&gt;<a name="line.130"></a>
<span class="sourceLineNo">131</span>     * @return a new chromatogram that is the reverse complement of this one<a name="line.131"></a>
<span class="sourceLineNo">132</span>     */<a name="line.132"></a>
<span class="sourceLineNo">133</span>    public Chromatogram reverseComplement();<a name="line.133"></a>
<span class="sourceLineNo">134</span>}<a name="line.134"></a>




























































</pre>
</div>
</body>
</html>
