<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.bio.chromatogram;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.util.Map;<a name="line.24"></a>
<span class="sourceLineNo">025</span><a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.bio.symbol.AtomicSymbol;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.bio.symbol.IllegalAlphabetException;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.bio.symbol.IllegalSymbolException;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.bio.symbol.SymbolList;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.utils.SmallMap;<a name="line.30"></a>
<span class="sourceLineNo">031</span><a name="line.31"></a>
<span class="sourceLineNo">032</span>/**<a name="line.32"></a>
<span class="sourceLineNo">033</span> * A basic chromatogram implementation which provides public mutators <a name="line.33"></a>
<span class="sourceLineNo">034</span> * for setting the various attributes of the chromatogram.  <a name="line.34"></a>
<span class="sourceLineNo">035</span> * &lt;p&gt;<a name="line.35"></a>
<span class="sourceLineNo">036</span> * In general, &lt;b&gt;new chromatogram implementations&lt;/b&gt; should be derived from<a name="line.36"></a>
<span class="sourceLineNo">037</span> * {@link AbstractChromatogram}, &lt;b&gt;not this class&lt;/b&gt;, as it is generally <a name="line.37"></a>
<span class="sourceLineNo">038</span> * undesirable to allow the internal structures of a {@link Chromatogram} to be<a name="line.38"></a>
<span class="sourceLineNo">039</span> * manipulated externally. This class could still be useful, however, for <a name="line.39"></a>
<span class="sourceLineNo">040</span> * programatically generated "chromatograms".<a name="line.40"></a>
<span class="sourceLineNo">041</span> * &lt;/p&gt;<a name="line.41"></a>
<span class="sourceLineNo">042</span> *<a name="line.42"></a>
<span class="sourceLineNo">043</span> * @author Rhett Sutphin (&lt;a href="http://genome.uiowa.edu/"&gt;UI CBCB&lt;/a&gt;)<a name="line.43"></a>
<span class="sourceLineNo">044</span> * @author Matthew Pocock<a name="line.44"></a>
<span class="sourceLineNo">045</span> * @since 1.3<a name="line.45"></a>
<span class="sourceLineNo">046</span> */<a name="line.46"></a>
<span class="sourceLineNo">047</span>public class SimpleChromatogram extends AbstractChromatogram {<a name="line.47"></a>
<span class="sourceLineNo">048</span>    <a name="line.48"></a>
<span class="sourceLineNo">049</span>    /** Creates a new instance of SimpleChromatogram. */<a name="line.49"></a>
<span class="sourceLineNo">050</span>    public SimpleChromatogram() {<a name="line.50"></a>
<span class="sourceLineNo">051</span>        super();<a name="line.51"></a>
<span class="sourceLineNo">052</span>    }<a name="line.52"></a>
<span class="sourceLineNo">053</span>    <a name="line.53"></a>
<span class="sourceLineNo">054</span>    /**<a name="line.54"></a>
<span class="sourceLineNo">055</span>     * Set the DNA and OFFSETS symbol lists for the basecall alignment.  Beware:<a name="line.55"></a>
<span class="sourceLineNo">056</span>     * this method does no consistency checks to be sure that all the offsets<a name="line.56"></a>
<span class="sourceLineNo">057</span>     * are valid indices into the trace arrays.<a name="line.57"></a>
<span class="sourceLineNo">058</span>     *<a name="line.58"></a>
<span class="sourceLineNo">059</span>     * @param dna a symbol list in the DNA alphabet that contains the base calls<a name="line.59"></a>
<span class="sourceLineNo">060</span>     *        for this chromatogram<a name="line.60"></a>
<span class="sourceLineNo">061</span>     * @param offsets a symbol list in an integer or sub-integer alphabet that<a name="line.61"></a>
<span class="sourceLineNo">062</span>     *        contains the locations in the chromatogram for the bases called <a name="line.62"></a>
<span class="sourceLineNo">063</span>     *        in &lt;code&gt;dna&lt;/code&gt;<a name="line.63"></a>
<span class="sourceLineNo">064</span>     * @throws IllegalAlphabetException when the alphabets aren't as specified<a name="line.64"></a>
<span class="sourceLineNo">065</span>     * @throws IllegalArgumentException when the lists aren't the same length<a name="line.65"></a>
<span class="sourceLineNo">066</span>     */<a name="line.66"></a>
<span class="sourceLineNo">067</span>    public void setSymbolLists(SymbolList dna, SymbolList offsets)<a name="line.67"></a>
<span class="sourceLineNo">068</span>    throws IllegalAlphabetException, IllegalArgumentException {<a name="line.68"></a>
<span class="sourceLineNo">069</span>        if (dna.length() != offsets.length())<a name="line.69"></a>
<span class="sourceLineNo">070</span>            throw new IllegalArgumentException("The SymbolLists must be the same length");<a name="line.70"></a>
<span class="sourceLineNo">071</span>        Map map = new SmallMap(2);<a name="line.71"></a>
<span class="sourceLineNo">072</span>        map.put(Chromatogram.DNA, dna);<a name="line.72"></a>
<span class="sourceLineNo">073</span>        map.put(Chromatogram.OFFSETS, offsets);<a name="line.73"></a>
<span class="sourceLineNo">074</span>        super.setBaseCallAlignment(super.createImmutableAlignment(map));<a name="line.74"></a>
<span class="sourceLineNo">075</span>    }<a name="line.75"></a>
<span class="sourceLineNo">076</span>    <a name="line.76"></a>
<span class="sourceLineNo">077</span>    /**<a name="line.77"></a>
<span class="sourceLineNo">078</span>     * Sets the trace array for one of the DNA nucleotides.  The provided<a name="line.78"></a>
<span class="sourceLineNo">079</span>     * array &lt;i&gt;will not be copied&lt;/i&gt;, so any modifications to it will be<a name="line.79"></a>
<span class="sourceLineNo">080</span>     * reflected in calls to {@link #getTrace}.<a name="line.80"></a>
<span class="sourceLineNo">081</span>     * &lt;p&gt;<a name="line.81"></a>
<span class="sourceLineNo">082</span>     * If you need to set a new set of traces whose length is different<a name="line.82"></a>
<span class="sourceLineNo">083</span>     * from the old set, you must call {@link #clearTraceValues} first,<a name="line.83"></a>
<span class="sourceLineNo">084</span>     * or you will provoke an &lt;code&gt;IllegalArgumentException&lt;/code&gt;.<a name="line.84"></a>
<span class="sourceLineNo">085</span>     * &lt;/p&gt;<a name="line.85"></a>
<span class="sourceLineNo">086</span>     *<a name="line.86"></a>
<span class="sourceLineNo">087</span>     * @param nuc the nucleotide for which to set the trace<a name="line.87"></a>
<span class="sourceLineNo">088</span>     * @param trace the sampled intensities along the trace<a name="line.88"></a>
<span class="sourceLineNo">089</span>     * @param maxVal the maximum value on the trace, or {@link java.lang.Integer#MIN_VALUE}<a name="line.89"></a>
<span class="sourceLineNo">090</span>     *        to force this method to calculate it<a name="line.90"></a>
<span class="sourceLineNo">091</span>     * @throws IllegalArgumentException when trace.length is different<a name="line.91"></a>
<span class="sourceLineNo">092</span>     *         from any of the existing (non-null) traces<a name="line.92"></a>
<span class="sourceLineNo">093</span>     * @throws IllegalSymbolException when nuc is not a concrete DNA nucleotide<a name="line.93"></a>
<span class="sourceLineNo">094</span>     */<a name="line.94"></a>
<span class="sourceLineNo">095</span>    public void setTraceValues(AtomicSymbol nuc, int[] trace, int maxVal)<a name="line.95"></a>
<span class="sourceLineNo">096</span>    throws IllegalArgumentException, IllegalSymbolException {<a name="line.96"></a>
<span class="sourceLineNo">097</span>        super.setTrace(nuc, trace, maxVal);<a name="line.97"></a>
<span class="sourceLineNo">098</span>    }<a name="line.98"></a>
<span class="sourceLineNo">099</span>    <a name="line.99"></a>
<span class="sourceLineNo">100</span>    /**<a name="line.100"></a>
<span class="sourceLineNo">101</span>     * Sets all the traces to null.<a name="line.101"></a>
<span class="sourceLineNo">102</span>     */<a name="line.102"></a>
<span class="sourceLineNo">103</span>    public void clearTraceValues() {<a name="line.103"></a>
<span class="sourceLineNo">104</span>        super.clearTraces();<a name="line.104"></a>
<span class="sourceLineNo">105</span>    }<a name="line.105"></a>
<span class="sourceLineNo">106</span>    <a name="line.106"></a>
<span class="sourceLineNo">107</span>    /**<a name="line.107"></a>
<span class="sourceLineNo">108</span>     * Sets the number of significant bits in the data.<a name="line.108"></a>
<span class="sourceLineNo">109</span>     * @param bits how many bits of the trace samples are significant<a name="line.109"></a>
<span class="sourceLineNo">110</span>     * @see Chromatogram#getSignificantBits<a name="line.110"></a>
<span class="sourceLineNo">111</span>     */<a name="line.111"></a>
<span class="sourceLineNo">112</span>    public void setSignificantBits(int bits) {<a name="line.112"></a>
<span class="sourceLineNo">113</span>        super.setBits(bits);<a name="line.113"></a>
<span class="sourceLineNo">114</span>    }<a name="line.114"></a>
<span class="sourceLineNo">115</span>    <a name="line.115"></a>
<span class="sourceLineNo">116</span>    protected AbstractChromatogram reverseComplementInstance() {<a name="line.116"></a>
<span class="sourceLineNo">117</span>        return new SimpleChromatogram();<a name="line.117"></a>
<span class="sourceLineNo">118</span>    }<a name="line.118"></a>
<span class="sourceLineNo">119</span>}<a name="line.119"></a>




























































</pre>
</div>
</body>
</html>
