<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*        BioJava development code<a name="line.1"></a>
<span class="sourceLineNo">002</span> *<a name="line.2"></a>
<span class="sourceLineNo">003</span> * This code may be freely distributed and modified under the<a name="line.3"></a>
<span class="sourceLineNo">004</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.4"></a>
<span class="sourceLineNo">005</span> * be distributed with the code.  If you do not have a copy,<a name="line.5"></a>
<span class="sourceLineNo">006</span> * see:<a name="line.6"></a>
<span class="sourceLineNo">007</span> *<a name="line.7"></a>
<span class="sourceLineNo">008</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.8"></a>
<span class="sourceLineNo">009</span> *<a name="line.9"></a>
<span class="sourceLineNo">010</span> * Copyright for this code is held jointly by the individual<a name="line.10"></a>
<span class="sourceLineNo">011</span> * authors.  These should be listed in @author doc comments.<a name="line.11"></a>
<span class="sourceLineNo">012</span> *<a name="line.12"></a>
<span class="sourceLineNo">013</span> * For more information on the BioJava project and its aims,<a name="line.13"></a>
<span class="sourceLineNo">014</span> * or to join the biojava-l mailing list, visit the home page<a name="line.14"></a>
<span class="sourceLineNo">015</span> * at:<a name="line.15"></a>
<span class="sourceLineNo">016</span> *<a name="line.16"></a>
<span class="sourceLineNo">017</span> *      http://www.biojava.org/<a name="line.17"></a>
<span class="sourceLineNo">018</span> *<a name="line.18"></a>
<span class="sourceLineNo">019</span> */<a name="line.19"></a>
<span class="sourceLineNo">020</span>package org.biojava.nbio.ronn;<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.22"></a>
<span class="sourceLineNo">023</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.data.sequence.FastaSequence;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.data.sequence.SequenceUtil;<a name="line.25"></a>
<span class="sourceLineNo">026</span><a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.io.FileInputStream;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.io.FileNotFoundException;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.io.IOException;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.util.ArrayList;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.util.List;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.Map;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.TreeMap;<a name="line.33"></a>
<span class="sourceLineNo">034</span><a name="line.34"></a>
<span class="sourceLineNo">035</span><a name="line.35"></a>
<span class="sourceLineNo">036</span>/**<a name="line.36"></a>
<span class="sourceLineNo">037</span> * This class gives public API to RONN functions.<a name="line.37"></a>
<span class="sourceLineNo">038</span> * It is build on top of the command line client. Due to this fact a few things<a name="line.38"></a>
<span class="sourceLineNo">039</span> * could be improved and extended pending the refactoring of the command line client.<a name="line.39"></a>
<span class="sourceLineNo">040</span> *<a name="line.40"></a>
<span class="sourceLineNo">041</span> * The input sequence limitations - the input sequence must not contain any ambiguous characters,<a name="line.41"></a>
<span class="sourceLineNo">042</span> * and have a minimum length of 19 amino acids.<a name="line.42"></a>
<span class="sourceLineNo">043</span> *<a name="line.43"></a>
<span class="sourceLineNo">044</span> * @author Peter Troshin<a name="line.44"></a>
<span class="sourceLineNo">045</span> * @version 1.0<a name="line.45"></a>
<span class="sourceLineNo">046</span> * @since 3.0.2<a name="line.46"></a>
<span class="sourceLineNo">047</span> *<a name="line.47"></a>
<span class="sourceLineNo">048</span> */<a name="line.48"></a>
<span class="sourceLineNo">049</span>public class Jronn {<a name="line.49"></a>
<span class="sourceLineNo">050</span><a name="line.50"></a>
<span class="sourceLineNo">051</span>        // Load models<a name="line.51"></a>
<span class="sourceLineNo">052</span>        private static final ModelLoader loader = new ModelLoader();<a name="line.52"></a>
<span class="sourceLineNo">053</span>        static {<a name="line.53"></a>
<span class="sourceLineNo">054</span>                try {<a name="line.54"></a>
<span class="sourceLineNo">055</span>                        loader.loadModels();<a name="line.55"></a>
<span class="sourceLineNo">056</span>                } catch (NumberFormatException e) {<a name="line.56"></a>
<span class="sourceLineNo">057</span>                        throw new RuntimeException("Fails to load models!" + e.getMessage(), e);<a name="line.57"></a>
<span class="sourceLineNo">058</span>                } catch (IOException e) {<a name="line.58"></a>
<span class="sourceLineNo">059</span>                        throw new RuntimeException("Fails to load models!" + e.getMessage(), e);<a name="line.59"></a>
<span class="sourceLineNo">060</span>                }<a name="line.60"></a>
<span class="sourceLineNo">061</span>        }<a name="line.61"></a>
<span class="sourceLineNo">062</span><a name="line.62"></a>
<span class="sourceLineNo">063</span><a name="line.63"></a>
<span class="sourceLineNo">064</span>        /**<a name="line.64"></a>
<span class="sourceLineNo">065</span>         * Holder for the ranges, contain pointers to starting and ending position<a name="line.65"></a>
<span class="sourceLineNo">066</span>         * on the sequence which comprises a disordered region. Immutable.<a name="line.66"></a>
<span class="sourceLineNo">067</span>         * @author pvtroshin<a name="line.67"></a>
<span class="sourceLineNo">068</span>         */<a name="line.68"></a>
<span class="sourceLineNo">069</span>        public static class Range {<a name="line.69"></a>
<span class="sourceLineNo">070</span>                /**<a name="line.70"></a>
<span class="sourceLineNo">071</span>                 * Range starting position counts from 1 (the first position on the sequence is 1)<a name="line.71"></a>
<span class="sourceLineNo">072</span>                 */<a name="line.72"></a>
<span class="sourceLineNo">073</span>                public final int from;<a name="line.73"></a>
<span class="sourceLineNo">074</span>                /**<a name="line.74"></a>
<span class="sourceLineNo">075</span>                 * The range ending position includes the last residue.<a name="line.75"></a>
<span class="sourceLineNo">076</span>                 */<a name="line.76"></a>
<span class="sourceLineNo">077</span>                public final int to;<a name="line.77"></a>
<span class="sourceLineNo">078</span><a name="line.78"></a>
<span class="sourceLineNo">079</span>                public final float score;<a name="line.79"></a>
<span class="sourceLineNo">080</span>                public Range(int from, int to, float score) {<a name="line.80"></a>
<span class="sourceLineNo">081</span>                        assert from&gt;=0;<a name="line.81"></a>
<span class="sourceLineNo">082</span>                        assert from&lt;to;<a name="line.82"></a>
<span class="sourceLineNo">083</span>                        this.from = from;<a name="line.83"></a>
<span class="sourceLineNo">084</span>                        this.to = to;<a name="line.84"></a>
<span class="sourceLineNo">085</span>                        this.score = score;<a name="line.85"></a>
<span class="sourceLineNo">086</span>                }<a name="line.86"></a>
<span class="sourceLineNo">087</span><a name="line.87"></a>
<span class="sourceLineNo">088</span>                @Override<a name="line.88"></a>
<span class="sourceLineNo">089</span>                public String toString() {<a name="line.89"></a>
<span class="sourceLineNo">090</span>                        return "Range" + " From:" + from + "\t" + "to: " + to + "\n";<a name="line.90"></a>
<span class="sourceLineNo">091</span>                }<a name="line.91"></a>
<span class="sourceLineNo">092</span><a name="line.92"></a>
<span class="sourceLineNo">093</span>                @Override<a name="line.93"></a>
<span class="sourceLineNo">094</span>                public int hashCode() {<a name="line.94"></a>
<span class="sourceLineNo">095</span>                        final int prime = 31;<a name="line.95"></a>
<span class="sourceLineNo">096</span>                        int result = 1;<a name="line.96"></a>
<span class="sourceLineNo">097</span>                        result = prime * result + from;<a name="line.97"></a>
<span class="sourceLineNo">098</span>                        result = prime * result + to;<a name="line.98"></a>
<span class="sourceLineNo">099</span>                        return result;<a name="line.99"></a>
<span class="sourceLineNo">100</span>                }<a name="line.100"></a>
<span class="sourceLineNo">101</span><a name="line.101"></a>
<span class="sourceLineNo">102</span>                @Override<a name="line.102"></a>
<span class="sourceLineNo">103</span>                public boolean equals(Object obj) {<a name="line.103"></a>
<span class="sourceLineNo">104</span>                        if (this == obj)<a name="line.104"></a>
<span class="sourceLineNo">105</span>                                return true;<a name="line.105"></a>
<span class="sourceLineNo">106</span>                        if (obj == null)<a name="line.106"></a>
<span class="sourceLineNo">107</span>                                return false;<a name="line.107"></a>
<span class="sourceLineNo">108</span>                        if (getClass() != obj.getClass())<a name="line.108"></a>
<span class="sourceLineNo">109</span>                                return false;<a name="line.109"></a>
<span class="sourceLineNo">110</span>                        Range other = (Range) obj;<a name="line.110"></a>
<span class="sourceLineNo">111</span>                        if (from != other.from)<a name="line.111"></a>
<span class="sourceLineNo">112</span>                                return false;<a name="line.112"></a>
<span class="sourceLineNo">113</span>                        if (to != other.to)<a name="line.113"></a>
<span class="sourceLineNo">114</span>                                return false;<a name="line.114"></a>
<span class="sourceLineNo">115</span>                        return true;<a name="line.115"></a>
<span class="sourceLineNo">116</span>                }<a name="line.116"></a>
<span class="sourceLineNo">117</span><a name="line.117"></a>
<span class="sourceLineNo">118</span><a name="line.118"></a>
<span class="sourceLineNo">119</span>        }<a name="line.119"></a>
<span class="sourceLineNo">120</span><a name="line.120"></a>
<span class="sourceLineNo">121</span>        /**<a name="line.121"></a>
<span class="sourceLineNo">122</span>         * Calculates the probability value for each residue in the protein sequence,<a name="line.122"></a>
<span class="sourceLineNo">123</span>         * telling the probability that the residue belongs to disordered region.<a name="line.123"></a>
<span class="sourceLineNo">124</span>         * In general, values greater than 0.5 considered to be in the disordered regions.<a name="line.124"></a>
<span class="sourceLineNo">125</span>         *<a name="line.125"></a>
<span class="sourceLineNo">126</span>         * @param sequence an instance of FastaSequence object, holding the name and the sequence.<a name="line.126"></a>
<span class="sourceLineNo">127</span>         * @return the probability scores for each residue in the sequence<a name="line.127"></a>
<span class="sourceLineNo">128</span>         */<a name="line.128"></a>
<span class="sourceLineNo">129</span>        public static float[] getDisorderScores(FastaSequence sequence) {<a name="line.129"></a>
<span class="sourceLineNo">130</span>                    return predictSerial(sequence);<a name="line.130"></a>
<span class="sourceLineNo">131</span>        }<a name="line.131"></a>
<span class="sourceLineNo">132</span><a name="line.132"></a>
<span class="sourceLineNo">133</span>        /**<a name="line.133"></a>
<span class="sourceLineNo">134</span>         * Calculates the probability value for each residue in the protein sequence,<a name="line.134"></a>
<span class="sourceLineNo">135</span>         * telling the probability that the residue belongs to disordered region.<a name="line.135"></a>
<span class="sourceLineNo">136</span>         * In general, values greater than 0.5 considered to be in the disordered regions.<a name="line.136"></a>
<span class="sourceLineNo">137</span>         *<a name="line.137"></a>
<span class="sourceLineNo">138</span>         * @param sequence an instance of FastaSequence object, holding the name and the sequence.<a name="line.138"></a>
<span class="sourceLineNo">139</span>         * @return the probability scores for each residue in the sequence<a name="line.139"></a>
<span class="sourceLineNo">140</span>         */<a name="line.140"></a>
<span class="sourceLineNo">141</span>        public static float[] getDisorderScores(ProteinSequence sequence) {<a name="line.141"></a>
<span class="sourceLineNo">142</span><a name="line.142"></a>
<span class="sourceLineNo">143</span>                FastaSequence seq = convertProteinSequencetoFasta(sequence);<a name="line.143"></a>
<span class="sourceLineNo">144</span><a name="line.144"></a>
<span class="sourceLineNo">145</span>                return predictSerial(seq);<a name="line.145"></a>
<span class="sourceLineNo">146</span>        }<a name="line.146"></a>
<span class="sourceLineNo">147</span><a name="line.147"></a>
<span class="sourceLineNo">148</span>        /** Utility method to convert a BioJava ProteinSequence object to the FastaSequence<a name="line.148"></a>
<span class="sourceLineNo">149</span>         *  object used internally in JRonn.<a name="line.149"></a>
<span class="sourceLineNo">150</span>         *<a name="line.150"></a>
<span class="sourceLineNo">151</span>         * @param sequence<a name="line.151"></a>
<span class="sourceLineNo">152</span>         * @return<a name="line.152"></a>
<span class="sourceLineNo">153</span>         */<a name="line.153"></a>
<span class="sourceLineNo">154</span>        public static FastaSequence convertProteinSequencetoFasta(ProteinSequence sequence){<a name="line.154"></a>
<span class="sourceLineNo">155</span>                StringBuffer buf = new StringBuffer();<a name="line.155"></a>
<span class="sourceLineNo">156</span>                for (AminoAcidCompound compound : sequence) {<a name="line.156"></a>
<span class="sourceLineNo">157</span><a name="line.157"></a>
<span class="sourceLineNo">158</span>                        String c = compound.getShortName();<a name="line.158"></a>
<span class="sourceLineNo">159</span><a name="line.159"></a>
<span class="sourceLineNo">160</span>                        if (! SequenceUtil.NON_AA.matcher(c).find()) {<a name="line.160"></a>
<span class="sourceLineNo">161</span>                                buf.append(c);<a name="line.161"></a>
<span class="sourceLineNo">162</span>                        } else {<a name="line.162"></a>
<span class="sourceLineNo">163</span>                                buf.append("X");<a name="line.163"></a>
<span class="sourceLineNo">164</span>                        }<a name="line.164"></a>
<span class="sourceLineNo">165</span>                }<a name="line.165"></a>
<span class="sourceLineNo">166</span><a name="line.166"></a>
<span class="sourceLineNo">167</span>                return new FastaSequence(sequence.getAccession().getID(),buf.toString());<a name="line.167"></a>
<span class="sourceLineNo">168</span>        }<a name="line.168"></a>
<span class="sourceLineNo">169</span><a name="line.169"></a>
<span class="sourceLineNo">170</span>        private static float[] predictSerial(FastaSequence fsequence) {<a name="line.170"></a>
<span class="sourceLineNo">171</span>                ORonn.validateSequenceForRonn(fsequence);<a name="line.171"></a>
<span class="sourceLineNo">172</span>                ORonn ronn;<a name="line.172"></a>
<span class="sourceLineNo">173</span>                float[] disorder = null;<a name="line.173"></a>
<span class="sourceLineNo">174</span>                try {<a name="line.174"></a>
<span class="sourceLineNo">175</span>                        ronn = new ORonn(fsequence, loader);<a name="line.175"></a>
<span class="sourceLineNo">176</span>                        disorder = ronn.call().getMeanScores();<a name="line.176"></a>
<span class="sourceLineNo">177</span>                } catch (NumberFormatException e) {<a name="line.177"></a>
<span class="sourceLineNo">178</span>                        throw new RuntimeException("Jronn fails to load models " + e.getLocalizedMessage(), e);<a name="line.178"></a>
<span class="sourceLineNo">179</span>                } catch (IOException e) {<a name="line.179"></a>
<span class="sourceLineNo">180</span>                        throw new RuntimeException("Jronn fails to load models " + e.getLocalizedMessage(), e);<a name="line.180"></a>
<span class="sourceLineNo">181</span>                }<a name="line.181"></a>
<span class="sourceLineNo">182</span>                return disorder;<a name="line.182"></a>
<span class="sourceLineNo">183</span>        }<a name="line.183"></a>
<span class="sourceLineNo">184</span><a name="line.184"></a>
<span class="sourceLineNo">185</span>        /**<a name="line.185"></a>
<span class="sourceLineNo">186</span>         * Calculates the disordered regions of the sequence. More formally, the regions for which the<a name="line.186"></a>
<span class="sourceLineNo">187</span>         * probability of disorder is greater then 0.50.<a name="line.187"></a>
<span class="sourceLineNo">188</span>         *<a name="line.188"></a>
<span class="sourceLineNo">189</span>         *<a name="line.189"></a>
<span class="sourceLineNo">190</span>         * @param sequence an instance of FastaSequence object, holding the name and the sequence.<a name="line.190"></a>
<span class="sourceLineNo">191</span>         * @return the array of ranges if there are any residues predicted to have the<a name="line.191"></a>
<span class="sourceLineNo">192</span>         * probability of disorder greater then 0.5, null otherwise.<a name="line.192"></a>
<span class="sourceLineNo">193</span>         *<a name="line.193"></a>
<span class="sourceLineNo">194</span>         */<a name="line.194"></a>
<span class="sourceLineNo">195</span>        public static Range[] getDisorder(FastaSequence sequence) {<a name="line.195"></a>
<span class="sourceLineNo">196</span>                float[] scores = getDisorderScores(sequence);<a name="line.196"></a>
<span class="sourceLineNo">197</span>                return scoresToRanges(scores, RonnConstraint.DEFAULT_RANGE_PROBABILITY_THRESHOLD);<a name="line.197"></a>
<span class="sourceLineNo">198</span>        }<a name="line.198"></a>
<span class="sourceLineNo">199</span><a name="line.199"></a>
<span class="sourceLineNo">200</span>        /**<a name="line.200"></a>
<span class="sourceLineNo">201</span>         * Convert raw scores to ranges. Gives ranges for given probability of disorder value<a name="line.201"></a>
<span class="sourceLineNo">202</span>         * @param scores the raw probability of disorder scores for each residue in the sequence.<a name="line.202"></a>
<span class="sourceLineNo">203</span>         * @param probability the cut off threshold. Include all residues with the probability of disorder greater then this value<a name="line.203"></a>
<span class="sourceLineNo">204</span>         * @return the array of ranges if there are any residues predicted to have the<a name="line.204"></a>
<span class="sourceLineNo">205</span>         * probability of disorder greater then {@code probability}, null otherwise.<a name="line.205"></a>
<span class="sourceLineNo">206</span>         */<a name="line.206"></a>
<span class="sourceLineNo">207</span>        public static Range[] scoresToRanges(float[] scores, float probability)  {<a name="line.207"></a>
<span class="sourceLineNo">208</span>                assert scores!=null &amp;&amp; scores.length&gt;0;<a name="line.208"></a>
<span class="sourceLineNo">209</span>                assert probability&gt;0 &amp;&amp; probability&lt;1;<a name="line.209"></a>
<span class="sourceLineNo">210</span><a name="line.210"></a>
<span class="sourceLineNo">211</span>                int count=0;<a name="line.211"></a>
<span class="sourceLineNo">212</span>                int regionLen=0;<a name="line.212"></a>
<span class="sourceLineNo">213</span>                List&lt;Range&gt; ranges = new ArrayList&lt;Range&gt;();<a name="line.213"></a>
<span class="sourceLineNo">214</span>                for(float score: scores) {<a name="line.214"></a>
<span class="sourceLineNo">215</span>                        count++;<a name="line.215"></a>
<span class="sourceLineNo">216</span>                        // Round to 2 decimal points before comparison<a name="line.216"></a>
<span class="sourceLineNo">217</span>                        score = (float) (Math.round(score*100.0)/100.0);<a name="line.217"></a>
<span class="sourceLineNo">218</span>                        if(score&gt;probability) {<a name="line.218"></a>
<span class="sourceLineNo">219</span>                                regionLen++;<a name="line.219"></a>
<span class="sourceLineNo">220</span>                        } else {<a name="line.220"></a>
<span class="sourceLineNo">221</span>                                if(regionLen&gt;0) {<a name="line.221"></a>
<span class="sourceLineNo">222</span>                                        ranges.add(new Range(count-regionLen, count-1,score));<a name="line.222"></a>
<span class="sourceLineNo">223</span>                                }<a name="line.223"></a>
<span class="sourceLineNo">224</span>                                regionLen=0;<a name="line.224"></a>
<span class="sourceLineNo">225</span>                        }<a name="line.225"></a>
<span class="sourceLineNo">226</span>                }<a name="line.226"></a>
<span class="sourceLineNo">227</span>                // In case of the range to boundary runs to the very end of the sequence<a name="line.227"></a>
<span class="sourceLineNo">228</span>                if(regionLen&gt;1) {<a name="line.228"></a>
<span class="sourceLineNo">229</span>                        ranges.add(new Range(count-regionLen+1, count,scores[scores.length-1]));<a name="line.229"></a>
<span class="sourceLineNo">230</span>                }<a name="line.230"></a>
<span class="sourceLineNo">231</span>                return ranges.toArray(new Range[ranges.size()]);<a name="line.231"></a>
<span class="sourceLineNo">232</span><a name="line.232"></a>
<span class="sourceLineNo">233</span>        }<a name="line.233"></a>
<span class="sourceLineNo">234</span><a name="line.234"></a>
<span class="sourceLineNo">235</span>        /**<a name="line.235"></a>
<span class="sourceLineNo">236</span>         * Calculates the probability of disorder scores for each residue in the sequence for<a name="line.236"></a>
<span class="sourceLineNo">237</span>         * many sequences in the input.<a name="line.237"></a>
<span class="sourceLineNo">238</span>         *<a name="line.238"></a>
<span class="sourceLineNo">239</span>         * @param sequences the list of the FastaSequence objects<a name="line.239"></a>
<span class="sourceLineNo">240</span>         * @return the Map with key-&gt;FastaSequence, value-&gt;probability of disorder for each residue<a name="line.240"></a>
<span class="sourceLineNo">241</span>         * @see #getDisorder(FastaSequence)<a name="line.241"></a>
<span class="sourceLineNo">242</span>         */<a name="line.242"></a>
<span class="sourceLineNo">243</span>        public static Map&lt;FastaSequence,float[]&gt; getDisorderScores(List&lt;FastaSequence&gt; sequences) {<a name="line.243"></a>
<span class="sourceLineNo">244</span>                Map&lt;FastaSequence,float[]&gt; results = new TreeMap&lt;FastaSequence, float[]&gt;();<a name="line.244"></a>
<span class="sourceLineNo">245</span>                for(FastaSequence fsequence : sequences) {<a name="line.245"></a>
<span class="sourceLineNo">246</span>                        results.put(fsequence, predictSerial(fsequence));<a name="line.246"></a>
<span class="sourceLineNo">247</span>                }<a name="line.247"></a>
<span class="sourceLineNo">248</span>                return results;<a name="line.248"></a>
<span class="sourceLineNo">249</span>        }<a name="line.249"></a>
<span class="sourceLineNo">250</span><a name="line.250"></a>
<span class="sourceLineNo">251</span>        /**<a name="line.251"></a>
<span class="sourceLineNo">252</span>         * Calculates the disordered regions of the sequence for many sequences in the input.<a name="line.252"></a>
<span class="sourceLineNo">253</span>         *<a name="line.253"></a>
<span class="sourceLineNo">254</span>         * @param sequences sequences the list of the FastaSequence objects<a name="line.254"></a>
<span class="sourceLineNo">255</span>         * @return<a name="line.255"></a>
<span class="sourceLineNo">256</span>         * @see #getDisorder(FastaSequence)<a name="line.256"></a>
<span class="sourceLineNo">257</span>         */<a name="line.257"></a>
<span class="sourceLineNo">258</span>        public static Map&lt;FastaSequence,Range[]&gt; getDisorder(List&lt;FastaSequence&gt; sequences) {<a name="line.258"></a>
<span class="sourceLineNo">259</span>                Map&lt;FastaSequence,Range[]&gt; disorderRanges = new TreeMap&lt;FastaSequence,Range[]&gt;();<a name="line.259"></a>
<span class="sourceLineNo">260</span>                for(FastaSequence fs: sequences) {<a name="line.260"></a>
<span class="sourceLineNo">261</span>                        disorderRanges.put(fs, getDisorder(fs));<a name="line.261"></a>
<span class="sourceLineNo">262</span>                }<a name="line.262"></a>
<span class="sourceLineNo">263</span>                return disorderRanges;<a name="line.263"></a>
<span class="sourceLineNo">264</span>        }<a name="line.264"></a>
<span class="sourceLineNo">265</span><a name="line.265"></a>
<span class="sourceLineNo">266</span>        /**<a name="line.266"></a>
<span class="sourceLineNo">267</span>         * Calculates the disordered regions of the protein sequence.<a name="line.267"></a>
<span class="sourceLineNo">268</span>         * @param fastaFile input file name containing the sequence in FASTA<a name="line.268"></a>
<span class="sourceLineNo">269</span>         * @return the Map with key-&gt;FastaSequence, value-&gt;the list of disordered regions for each sequence<a name="line.269"></a>
<span class="sourceLineNo">270</span>         * @throws FileNotFoundException if the input file cannot be found<a name="line.270"></a>
<span class="sourceLineNo">271</span>         * @throws IOException of the system cannot access or read from the input file<a name="line.271"></a>
<span class="sourceLineNo">272</span>         * @see #getDisorder(FastaSequence)<a name="line.272"></a>
<span class="sourceLineNo">273</span>         * @see #Jronn.Range<a name="line.273"></a>
<span class="sourceLineNo">274</span>         */<a name="line.274"></a>
<span class="sourceLineNo">275</span>        public static Map&lt;FastaSequence,Range[]&gt; getDisorder(String fastaFile) throws FileNotFoundException, IOException {<a name="line.275"></a>
<span class="sourceLineNo">276</span>                final List&lt;FastaSequence&gt; sequences = SequenceUtil.readFasta(new FileInputStream(fastaFile));<a name="line.276"></a>
<span class="sourceLineNo">277</span>                return getDisorder(sequences);<a name="line.277"></a>
<span class="sourceLineNo">278</span>        }<a name="line.278"></a>
<span class="sourceLineNo">279</span><a name="line.279"></a>
<span class="sourceLineNo">280</span>        /**<a name="line.280"></a>
<span class="sourceLineNo">281</span>         * TODO<a name="line.281"></a>
<span class="sourceLineNo">282</span>         *<a name="line.282"></a>
<span class="sourceLineNo">283</span>         * High performance method for calculating disorder. Use multiple threads to achieve the speedup.<a name="line.283"></a>
<span class="sourceLineNo">284</span>         *<a name="line.284"></a>
<span class="sourceLineNo">285</span>         * @param fastaFile  fully qualified path to the input FASTA file<a name="line.285"></a>
<span class="sourceLineNo">286</span>         * @param outputFile file name of the file for the results<a name="line.286"></a>
<span class="sourceLineNo">287</span>         * @param threadNumber the number of threads to use, default<a name="line.287"></a>
<span class="sourceLineNo">288</span>         * @param controls the format of the result file<a name="line.288"></a>
<span class="sourceLineNo">289</span>         * @throws FileNotFoundException if input file in not found<a name="line.289"></a>
<span class="sourceLineNo">290</span>         * @throws IOException if the input or the output files cannot be accessed<a name="line.290"></a>
<span class="sourceLineNo">291</span>         * @see ORonn.ResultLayout<a name="line.291"></a>
<span class="sourceLineNo">292</span><a name="line.292"></a>
<span class="sourceLineNo">293</span>        public static void calculateDisorder(String fastaFile, String outputFile, int threadNumber, ResultLayout layout) throws FileNotFoundException, IOException {<a name="line.293"></a>
<span class="sourceLineNo">294</span>                final List&lt;FastaSequence&gt; sequences = SequenceUtil.readFasta(new FileInputStream(fastaFile));<a name="line.294"></a>
<span class="sourceLineNo">295</span>                InputParameters in = new InputParameters();<a name="line.295"></a>
<span class="sourceLineNo">296</span>                in.setFilePrm(fastaFile, InputParameters.inputKey);<a name="line.296"></a>
<span class="sourceLineNo">297</span>                in.setFilePrm(outputFile, InputParameters.outputKey);<a name="line.297"></a>
<span class="sourceLineNo">298</span>                //in.setThreadNum(Integer.toString(threadNumber));<a name="line.298"></a>
<span class="sourceLineNo">299</span>                ORonn.predictParallel(sequences, in, loader);<a name="line.299"></a>
<span class="sourceLineNo">300</span>        }<a name="line.300"></a>
<span class="sourceLineNo">301</span>        */<a name="line.301"></a>
<span class="sourceLineNo">302</span>}<a name="line.302"></a>




























































</pre>
</div>
</body>
</html>
