<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:36 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Chain (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Chain (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chain.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../org/biojava/nbio/structure/Calc.html" title="class in org.biojava.nbio.structure"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../org/biojava/nbio/structure/ChainImpl.html" title="class in org.biojava.nbio.structure"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/nbio/structure/Chain.html" target="_top">Frames</a></li>
<li><a href="Chain.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.structure</div>
<h2 title="Interface Chain" class="title">Interface Chain</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><a href="../../../../org/biojava/nbio/structure/ChainImpl.html" title="class in org.biojava.nbio.structure">ChainImpl</a></dd>
</dl>
<hr>
<br>
<pre>public interface <span class="strong">Chain</span></pre>
<div class="block"><p>
 Defines the interface for a Chain. A Chain corresponds to a Chain in a PDB file.
 A chain consists out of a list of <a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a> objects. A Group can either be
 an <a href="../../../../org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure"><code>AminoAcid</code></a>, <a href="../../../../org/biojava/nbio/structure/HetatomImpl.html" title="class in org.biojava.nbio.structure"><code>Hetatom</code></a> or <a href="../../../../org/biojava/nbio/structure/NucleotideImpl.html" title="class in org.biojava.nbio.structure"><code>Nucleotide</code></a>.
 </p>

 <p>
 The BioJava API provides access to both the ATOM and SEQRES records in a PDB file.
 During parsing of a PDB file it aligns the ATOM and SEQRES groups and joins them.
 The SEQRES sequence can be accessed via  <a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroups()"><code>getSeqResGroups()</code></a> and the
 ATOM groups via <a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroups()"><code>getAtomGroups()</code></a>. Groups that have been observed
 (i.e. they are in the ATOM records) can be detected by <a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a>.has3D()
  </p></div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>1.4</dd>
<dt><span class="strong">Version:</span></dt>
  <dd>%I% %G%</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>Andreas Prlic</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#addGroup(org.biojava.nbio.structure.Group)">addGroup</a></strong>(<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group)</code>
<div class="block">add a group to the list of ATOM record group of this chain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#clone()">clone</a></strong>()</code>
<div class="block">returns an identical copy of this Chain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroup(int)">getAtomGroup</a></strong>(int&nbsp;position)</code>
<div class="block">Return the Group at given position,
 from within Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroups()">getAtomGroups</a></strong>()</code>
<div class="block">Return all Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroups(org.biojava.nbio.structure.GroupType)">getAtomGroups</a></strong>(<a href="../../../../org/biojava/nbio/structure/GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</code>
<div class="block">Return a List of all (observed) Groups of a special type, one of: <a href="../../../../org/biojava/nbio/structure/GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="../../../../org/biojava/nbio/structure/GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="../../../../org/biojava/nbio/structure/GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomLength()">getAtomLength</a></strong>()</code>
<div class="block">Returns the number of Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomLigands()">getAtomLigands</a></strong>()</code>
<div class="block">Gets all groups that are not polymer groups and that are not solvent groups.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomSequence()">getAtomSequence</a></strong>()</code>
<div class="block">Returns the sequence of amino acids as it has been provided in the ATOM records.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getBJSequence()">getBJSequence</a></strong>()</code>
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getChainID()">getChainID</a></strong>()</code>
<div class="block">Gets the name of this chain (Chain id in PDB file ).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Compound.html" title="class in org.biojava.nbio.structure">Compound</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getCompound()">getCompound</a></strong>()</code>
<div class="block">Returns the Compound for this chain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">getGroupByPDB</a></strong>(<a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</code>
<div class="block">Get a group by its PDB residue numbering.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,%20org.biojava.nbio.structure.ResidueNumber)">getGroupsByPDB</a></strong>(<a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)</code>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,%20org.biojava.nbio.structure.ResidueNumber,%20boolean)">getGroupsByPDB</a></strong>(<a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
              <a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
              boolean&nbsp;ignoreMissing)</code>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Long.html?is-external=true" title="class or interface in java.lang">Long</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getId()">getId</a></strong>()</code>
<div class="block">Get the ID used by Hibernate.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getInternalChainID()">getInternalChainID</a></strong>()</code>
<div class="block">If available, returns the internal chain ID that is used in mmCIF files (asym_id), otherwise null</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getParent()">getParent</a></strong>()</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use getStructure(Structure) instead.</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqMisMatches()">getSeqMisMatches</a></strong>()</code>
<div class="block">Get annotated sequence mismatches for this chain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroup(int)">getSeqResGroup</a></strong>(int&nbsp;position)</code>
<div class="block">Return the Group at given position,
 from within groups in the SEQRES records of the chain, i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroups()">getSeqResGroups</a></strong>()</code>
<div class="block">Returns a list of all groups in SEQRES records of the chain, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroups(org.biojava.nbio.structure.GroupType)">getSeqResGroups</a></strong>(<a href="../../../../org/biojava/nbio/structure/GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</code>
<div class="block">Returns a List of all SEQRES groups of a special type, one of: <a href="../../../../org/biojava/nbio/structure/GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="../../../../org/biojava/nbio/structure/GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="../../../../org/biojava/nbio/structure/GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResLength()">getSeqResLength</a></strong>()</code>
<div class="block">Returns the number of groups in the SEQRES records of the chain, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResSequence()">getSeqResSequence</a></strong>()</code>
<div class="block">Returns the PDB SEQRES sequence as a one-letter sequence string.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getStructure()">getStructure</a></strong>()</code>
<div class="block">Returns the parent Structure of this chain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#getSwissprotId()">getSwissprotId</a></strong>()</code>
<div class="block">Gets the Swissprot id of this chain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setAtomGroups(java.util.List)">setAtomGroups</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groups)</code>
<div class="block">Set all Groups with observed density in the chain, i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setChainID(java.lang.String)">setChainID</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Sets the name of this chain (Chain id in PDB file ).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setCompound(org.biojava.nbio.structure.Compound)">setCompound</a></strong>(<a href="../../../../org/biojava/nbio/structure/Compound.html" title="class in org.biojava.nbio.structure">Compound</a>&nbsp;compound)</code>
<div class="block">Sets the Compound</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setId(java.lang.Long)">setId</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Long.html?is-external=true" title="class or interface in java.lang">Long</a>&nbsp;id)</code>
<div class="block">Set the ID used by Hibernate.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setInternalChainID(java.lang.String)">setInternalChainID</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;internalChainID)</code>
<div class="block">Sets the internal chain ID that is used in mmCif files</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setParent(org.biojava.nbio.structure.Structure)">setParent</a></strong>(<a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use setStructure instead</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setSeqMisMatches(java.util.List)">setSeqMisMatches</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;seqMisMatches)</code>
<div class="block">Set annotated sequence mismatches for this chain.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setSeqResGroups(java.util.List)">setSeqResGroups</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;seqResGroups)</code>
<div class="block">Sets the list of SeqResGroups for this chain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setStructure(org.biojava.nbio.structure.Structure)">setStructure</a></strong>(<a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code>
<div class="block">Sets the back-reference to its parent Structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#setSwissprotId(java.lang.String)">setSwissprotId</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sp_id)</code>
<div class="block">Sets the Swissprot id of this chain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#toMMCIF()">toMMCIF</a></strong>()</code>
<div class="block">Convert this Chain to a String in mmCIF format</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#toPDB()">toPDB</a></strong>()</code>
<div class="block">Convert this Chain to a String in PDB format</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/structure/Chain.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="clone()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>clone</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;clone()</pre>
<div class="block">returns an identical copy of this Chain.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>an identical copy of this Chain</dd></dl>
</li>
</ul>
<a name="addGroup(org.biojava.nbio.structure.Group)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addGroup</h4>
<pre>void&nbsp;addGroup(<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group)</pre>
<div class="block">add a group to the list of ATOM record group of this chain.
 To add SEQRES records a more complex alignment between ATOM and SEQRES residues
 is required, please see SeqRes2AtomAligner for more details on that.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>group</code> - a Group object</dd></dl>
</li>
</ul>
<a name="getId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getId</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Long.html?is-external=true" title="class or interface in java.lang">Long</a>&nbsp;getId()</pre>
<div class="block">Get the ID used by Hibernate.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the ID used by Hibernate</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setId(java.lang.Long)"><code>setId(Long)</code></a></dd></dl>
</li>
</ul>
<a name="setId(java.lang.Long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setId</h4>
<pre>void&nbsp;setId(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Long.html?is-external=true" title="class or interface in java.lang">Long</a>&nbsp;id)</pre>
<div class="block">Set the ID used by Hibernate.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>id</code> - assigned by Hibernate</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getId()"><code>getId()</code></a></dd></dl>
</li>
</ul>
<a name="getAtomGroup(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomGroup</h4>
<pre><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;getAtomGroup(int&nbsp;position)</pre>
<div class="block">Return the Group at given position,
 from within Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMS (including waters) records.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>position</code> - an int</dd>
<dt><span class="strong">Returns:</span></dt><dd>a Group object</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomLength()"><code>getAtomLength()</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroups()"><code>getAtomGroups()</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroup(int)"><code>getSeqResGroup(int)</code></a></dd></dl>
</li>
</ul>
<a name="getSeqResGroup(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResGroup</h4>
<pre><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;getSeqResGroup(int&nbsp;position)</pre>
<div class="block">Return the Group at given position,
 from within groups in the SEQRES records of the chain, i.e.
 the aminoacids/nucleotides in the construct.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>position</code> - an int</dd>
<dt><span class="strong">Returns:</span></dt><dd>a Group object</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResLength()"><code>getSeqResLength()</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroups()"><code>getSeqResGroups()</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroup(int)"><code>getAtomGroup(int)</code></a></dd></dl>
</li>
</ul>
<a name="getAtomGroups()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomGroups</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;getAtomGroups()</pre>
<div class="block">Return all Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMS (including waters) records.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a List object representing the Groups of this Chain.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setAtomGroups(java.util.List)"><code>setAtomGroups(List)</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomLength()"><code>getAtomLength()</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroups()"><code>getSeqResGroups()</code></a></dd></dl>
</li>
</ul>
<a name="setAtomGroups(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAtomGroups</h4>
<pre>void&nbsp;setAtomGroups(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groups)</pre>
<div class="block">Set all Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMs (including waters) records.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>groups</code> - a List object representing the Groups of this Chain.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroups()"><code>getAtomGroups()</code></a></dd></dl>
</li>
</ul>
<a name="getAtomGroups(org.biojava.nbio.structure.GroupType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomGroups</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;getAtomGroups(<a href="../../../../org/biojava/nbio/structure/GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</pre>
<div class="block">Return a List of all (observed) Groups of a special type, one of: <a href="../../../../org/biojava/nbio/structure/GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="../../../../org/biojava/nbio/structure/GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="../../../../org/biojava/nbio/structure/GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.
 Note that if a standard aminoacid appears as a HETATM (because it is part of a ligand) then
 it is still considered as <a href="../../../../org/biojava/nbio/structure/GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a> and not as <a href="../../../../org/biojava/nbio/structure/GroupType.html#HETATM"><code>GroupType.HETATM</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>type</code> - GroupType</dd>
<dt><span class="strong">Returns:</span></dt><dd>a List object</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setAtomGroups(java.util.List)"><code>setAtomGroups(List)</code></a></dd></dl>
</li>
</ul>
<a name="getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGroupByPDB</h4>
<pre><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;getGroupByPDB(<a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)
                    throws <a href="../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Get a group by its PDB residue numbering. If the PDB residue number is not known,
 throws a StructureException.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>resNum</code> - the PDB residue number of the group</dd>
<dt><span class="strong">Returns:</span></dt><dd>the matching group</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd></dl>
</li>
</ul>
<a name="getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber, org.biojava.nbio.structure.ResidueNumber)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGroupsByPDB</h4>
<pre><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;getGroupsByPDB(<a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
                     <a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)
                       throws <a href="../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>pdbresnumStart</code> - PDB residue number of start</dd><dd><code>pdbresnumEnd</code> - PDB residue number of end</dd>
<dt><span class="strong">Returns:</span></dt><dd>Groups in between. or throws a StructureException if either start or end can not be found,</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd></dl>
</li>
</ul>
<a name="getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber, org.biojava.nbio.structure.ResidueNumber, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getGroupsByPDB</h4>
<pre><a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;getGroupsByPDB(<a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
                     <a href="../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
                     boolean&nbsp;ignoreMissing)
                       throws <a href="../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Get all groups that are located between two PDB residue numbers. In contrast to getGroupsByPDB
 this method call ignores if the exact outer groups are not found. This is useful e.g. when requesting the range
 of groups as specified by the DBREF records - these frequently are rather inaccurate.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>pdbresnumStart</code> - PDB residue number of start</dd><dd><code>pdbresnumEnd</code> - PDB residue number of end</dd><dd><code>ignoreMissing</code> - ignore missing groups in this range.</dd>
<dt><span class="strong">Returns:</span></dt><dd>Groups in between. or throws a StructureException if either start or end can not be found,</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd></dl>
</li>
</ul>
<a name="getAtomLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomLength</h4>
<pre>int&nbsp;getAtomLength()</pre>
<div class="block">Returns the number of Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMs (including waters) records</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the length</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroup(int)"><code>getAtomGroup(int)</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroups()"><code>getAtomGroups()</code></a>, 
<code>#getSeqResLength())</code></dd></dl>
</li>
</ul>
<a name="getSeqResLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResLength</h4>
<pre>int&nbsp;getSeqResLength()</pre>
<div class="block">Returns the number of groups in the SEQRES records of the chain, i.e.
 the number of aminoacids/nucleotides in the construct</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the length</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroup(int)"><code>getSeqResGroup(int)</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroups()"><code>getSeqResGroups()</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomLength()"><code>getAtomLength()</code></a></dd></dl>
</li>
</ul>
<a name="setCompound(org.biojava.nbio.structure.Compound)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCompound</h4>
<pre>void&nbsp;setCompound(<a href="../../../../org/biojava/nbio/structure/Compound.html" title="class in org.biojava.nbio.structure">Compound</a>&nbsp;compound)</pre>
<div class="block">Sets the Compound</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>compound</code> - the Compound</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getCompound()"><code>getCompound()</code></a></dd></dl>
</li>
</ul>
<a name="getCompound()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompound</h4>
<pre><a href="../../../../org/biojava/nbio/structure/Compound.html" title="class in org.biojava.nbio.structure">Compound</a>&nbsp;getCompound()</pre>
<div class="block">Returns the Compound for this chain.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the Compound object</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setCompound(org.biojava.nbio.structure.Compound)"><code>setCompound(Compound)</code></a></dd></dl>
</li>
</ul>
<a name="setChainID(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setChainID</h4>
<pre>void&nbsp;setChainID(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</pre>
<div class="block">Sets the name of this chain (Chain id in PDB file ).</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>name</code> - a String specifying the name value</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getChainID()"><code>getChainID()</code></a></dd></dl>
</li>
</ul>
<a name="getChainID()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getChainID</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getChainID()</pre>
<div class="block">Gets the name of this chain (Chain id in PDB file ).</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a String representing the name value</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setChainID(java.lang.String)"><code>setChainID(String)</code></a></dd></dl>
</li>
</ul>
<a name="getInternalChainID()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getInternalChainID</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getInternalChainID()</pre>
<div class="block">If available, returns the internal chain ID that is used in mmCIF files (asym_id), otherwise null</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>String or null</dd><dt><span class="strong">Since:</span></dt>
  <dd>3.0.5</dd></dl>
</li>
</ul>
<a name="setInternalChainID(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setInternalChainID</h4>
<pre>void&nbsp;setInternalChainID(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;internalChainID)</pre>
<div class="block">Sets the internal chain ID that is used in mmCif files</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>internalChainID</code> - </dd><dt><span class="strong">Since:</span></dt>
  <dd>3.0.5</dd></dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="getBJSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBJSequence</h4>
<pre><a href="../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;getBJSequence()</pre>
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the SEQRES groups of the Chain as a Sequence object.</dd></dl>
</li>
</ul>
<a name="getAtomSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomSequence</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getAtomSequence()</pre>
<div class="block">Returns the sequence of amino acids as it has been provided in the ATOM records.
 Non-standard residues will be present in the string only if the property
  has been set.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>amino acid sequence as string</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResSequence()"><code>getSeqResSequence()</code></a></dd></dl>
</li>
</ul>
<a name="getSeqResSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResSequence</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSeqResSequence()</pre>
<div class="block">Returns the PDB SEQRES sequence as a one-letter sequence string.
 Non-standard residues are represented by an "X".</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>one-letter PDB SEQRES sequence as string</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomSequence()"><code>getAtomSequence()</code></a></dd></dl>
</li>
</ul>
<a name="setSwissprotId(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSwissprotId</h4>
<pre>void&nbsp;setSwissprotId(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sp_id)</pre>
<div class="block">Sets the Swissprot id of this chain.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sp_id</code> - a String specifying the swissprot id value</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getSwissprotId()"><code>getSwissprotId()</code></a></dd></dl>
</li>
</ul>
<a name="getSwissprotId()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSwissprotId</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSwissprotId()</pre>
<div class="block">Gets the Swissprot id of this chain.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a String representing the swissprot id value</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setSwissprotId(java.lang.String)"><code>setSwissprotId(String sp_id)</code></a></dd></dl>
</li>
</ul>
<a name="getSeqResGroups(org.biojava.nbio.structure.GroupType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResGroups</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;getSeqResGroups(<a href="../../../../org/biojava/nbio/structure/GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</pre>
<div class="block">Returns a List of all SEQRES groups of a special type, one of: <a href="../../../../org/biojava/nbio/structure/GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="../../../../org/biojava/nbio/structure/GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="../../../../org/biojava/nbio/structure/GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>type</code> - a GroupType</dd>
<dt><span class="strong">Returns:</span></dt><dd>an List object</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setSeqResGroups(java.util.List)"><code>setSeqResGroups(List)</code></a></dd></dl>
</li>
</ul>
<a name="getSeqResGroups()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSeqResGroups</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;getSeqResGroups()</pre>
<div class="block">Returns a list of all groups in SEQRES records of the chain, i.e.
 the aminoacids/nucleotides in the construct.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>a List of all Group objects of this chain</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setSeqResGroups(java.util.List)"><code>setSeqResGroups(List)</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResLength()"><code>getSeqResLength()</code></a>, 
<a href="../../../../org/biojava/nbio/structure/Chain.html#getAtomGroups()"><code>getAtomGroups()</code></a></dd></dl>
</li>
</ul>
<a name="setSeqResGroups(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSeqResGroups</h4>
<pre>void&nbsp;setSeqResGroups(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;seqResGroups)</pre>
<div class="block">Sets the list of SeqResGroups for this chain.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqResGroups</code> - a List of Group objects that from the SEQRES groups of this chain.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getSeqResGroups()"><code>getSeqResGroups()</code></a></dd></dl>
</li>
</ul>
<a name="setParent(org.biojava.nbio.structure.Structure)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setParent</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
void&nbsp;setParent(<a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use setStructure instead</i></div>
<div class="block">Sets the back-reference to its parent Structure.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>parent</code> - the parent Structure object for this Chain</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#getStructure()"><code>getStructure()</code></a></dd></dl>
</li>
</ul>
<a name="setStructure(org.biojava.nbio.structure.Structure)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setStructure</h4>
<pre>void&nbsp;setStructure(<a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</pre>
<div class="block">Sets the back-reference to its parent Structure.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>parent</code> - </dd></dl>
</li>
</ul>
<a name="getParent()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getParent</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Deprecated.html?is-external=true" title="class or interface in java.lang">@Deprecated</a>
<a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;getParent()</pre>
<div class="block"><span class="strong">Deprecated.</span>&nbsp;<i>use getStructure(Structure) instead.</i></div>
<div class="block">Returns the parent Structure of this chain.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the parent Structure object</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setStructure(org.biojava.nbio.structure.Structure)"><code>setStructure(Structure)</code></a></dd></dl>
</li>
</ul>
<a name="getStructure()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStructure</h4>
<pre><a href="../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;getStructure()</pre>
<div class="block">Returns the parent Structure of this chain.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the parent Structure object</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/structure/Chain.html#setStructure(org.biojava.nbio.structure.Structure)"><code>setStructure(Structure)</code></a></dd></dl>
</li>
</ul>
<a name="getAtomLigands()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAtomLigands</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;getAtomLigands()</pre>
<div class="block">Gets all groups that are not polymer groups and that are not solvent groups.
 Will automatically fetch Chemical Component files from the PDB web site, even if
 <code>FileParsingParameters#setLoadChemCompInfo(boolean)</code> has not been set to true.
 Otherwise the Ligands could not correctly be identified.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>list of Groups that are ligands</dd></dl>
</li>
</ul>
<a name="toPDB()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toPDB</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toPDB()</pre>
<div class="block">Convert this Chain to a String in PDB format</div>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="toMMCIF()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toMMCIF</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toMMCIF()</pre>
<div class="block">Convert this Chain to a String in mmCIF format</div>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="setSeqMisMatches(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSeqMisMatches</h4>
<pre>void&nbsp;setSeqMisMatches(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;seqMisMatches)</pre>
<div class="block">Set annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqMisMatches</code> - </dd></dl>
</li>
</ul>
<a name="getSeqMisMatches()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getSeqMisMatches</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/structure/SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;getSeqMisMatches()</pre>
<div class="block">Get annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chain.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../org/biojava/nbio/structure/Calc.html" title="class in org.biojava.nbio.structure"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../org/biojava/nbio/structure/ChainImpl.html" title="class in org.biojava.nbio.structure"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/nbio/structure/Chain.html" target="_top">Frames</a></li>
<li><a href="Chain.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
