<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.core.sequence.transcription, class: RNAToAminoAcidTranslator">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on 01-21-2010</span>
<span class="source-line-no">021</span><span id="line-21"> */</span>
<span class="source-line-no">022</span><span id="line-22">package org.biojava.nbio.core.sequence.transcription;</span>
<span class="source-line-no">023</span><span id="line-23"></span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.core.sequence.RNASequence;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.template.AbstractCompoundTranslator;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.template.CompoundSet;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.template.Sequence;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.sequence.template.SequenceView;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.core.sequence.transcription.Table.Codon;</span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.core.sequence.views.WindowedSequence;</span>
<span class="source-line-no">034</span><span id="line-34"></span>
<span class="source-line-no">035</span><span id="line-35">import java.util.*;</span>
<span class="source-line-no">036</span><span id="line-36"></span>
<span class="source-line-no">037</span><span id="line-37">/**</span>
<span class="source-line-no">038</span><span id="line-38"> * Takes a {@link Sequence} of {@link NucleotideCompound} which should represent</span>
<span class="source-line-no">039</span><span id="line-39"> * an RNA sequence ({@link RNASequence} is good for this) and returns a list of</span>
<span class="source-line-no">040</span><span id="line-40"> * {@link Sequence} which hold {@link AminoAcidCompound}. The translator can</span>
<span class="source-line-no">041</span><span id="line-41"> * also trim stop codons as well as changing any valid start codon to an</span>
<span class="source-line-no">042</span><span id="line-42"> * initiating met.</span>
<span class="source-line-no">043</span><span id="line-43"> *</span>
<span class="source-line-no">044</span><span id="line-44"> * @author ayates</span>
<span class="source-line-no">045</span><span id="line-45"> */</span>
<span class="source-line-no">046</span><span id="line-46">public class RNAToAminoAcidTranslator extends</span>
<span class="source-line-no">047</span><span id="line-47">                AbstractCompoundTranslator&lt;NucleotideCompound, AminoAcidCompound&gt; {</span>
<span class="source-line-no">048</span><span id="line-48"></span>
<span class="source-line-no">049</span><span id="line-49">        private final boolean trimStops;</span>
<span class="source-line-no">050</span><span id="line-50">        private final boolean initMetOnly;</span>
<span class="source-line-no">051</span><span id="line-51">        private final Map&lt;Table.CaseInsensitiveTriplet, Codon&gt; quickLookup;</span>
<span class="source-line-no">052</span><span id="line-52">        private final Map&lt;AminoAcidCompound, List&lt;Codon&gt;&gt; aminoAcidToCodon;</span>
<span class="source-line-no">053</span><span id="line-53">        // Cheeky lookup which uses a hashing value; key is to switch to using this</span>
<span class="source-line-no">054</span><span id="line-54">        // all the time</span>
<span class="source-line-no">055</span><span id="line-55">        private final Codon[] codonArray = new Codon[64000];</span>
<span class="source-line-no">056</span><span id="line-56">        private final AminoAcidCompound unknownAminoAcidCompound;</span>
<span class="source-line-no">057</span><span id="line-57">        private final AminoAcidCompound methionineAminoAcidCompound;</span>
<span class="source-line-no">058</span><span id="line-58">        private final boolean translateNCodons;</span>
<span class="source-line-no">059</span><span id="line-59"></span>
<span class="source-line-no">060</span><span id="line-60">        // If true, then translation will stop at the first stop codon encountered</span>
<span class="source-line-no">061</span><span id="line-61">        // in the reading frame (the stop codon will be included as the last residue</span>
<span class="source-line-no">062</span><span id="line-62">        // in the resulting ProteinSequence, unless removed by #trimStops)</span>
<span class="source-line-no">063</span><span id="line-63">        private final boolean stopAtStopCodons;</span>
<span class="source-line-no">064</span><span id="line-64"></span>
<span class="source-line-no">065</span><span id="line-65">        // If true, then translation will not start until the first start codon</span>
<span class="source-line-no">066</span><span id="line-66">        // encountered in the reading frame. The start codon will be included as the</span>
<span class="source-line-no">067</span><span id="line-67">        // first residue in the resulting ProteinSequence</span>
<span class="source-line-no">068</span><span id="line-68">        private final boolean waitForStartCodon;</span>
<span class="source-line-no">069</span><span id="line-69"></span>
<span class="source-line-no">070</span><span id="line-70"></span>
<span class="source-line-no">071</span><span id="line-71">        public RNAToAminoAcidTranslator(</span>
<span class="source-line-no">072</span><span id="line-72">                        SequenceCreatorInterface&lt;AminoAcidCompound&gt; creator,</span>
<span class="source-line-no">073</span><span id="line-73">                        CompoundSet&lt;NucleotideCompound&gt; nucleotides,</span>
<span class="source-line-no">074</span><span id="line-74">                        CompoundSet&lt;Codon&gt; codons,</span>
<span class="source-line-no">075</span><span id="line-75">                        CompoundSet&lt;AminoAcidCompound&gt; aminoAcids, Table table,</span>
<span class="source-line-no">076</span><span id="line-76">                        boolean trimStops, boolean initMetOnly, boolean translateNCodons,</span>
<span class="source-line-no">077</span><span id="line-77">                        boolean stopAtStopCodons, boolean waitForStartCodon) {</span>
<span class="source-line-no">078</span><span id="line-78"></span>
<span class="source-line-no">079</span><span id="line-79">                super(creator, nucleotides, aminoAcids);</span>
<span class="source-line-no">080</span><span id="line-80">                this.trimStops = trimStops;</span>
<span class="source-line-no">081</span><span id="line-81">                this.initMetOnly = initMetOnly;</span>
<span class="source-line-no">082</span><span id="line-82">                this.translateNCodons = translateNCodons;</span>
<span class="source-line-no">083</span><span id="line-83"></span>
<span class="source-line-no">084</span><span id="line-84">                quickLookup = new HashMap&lt;&gt;(codons</span>
<span class="source-line-no">085</span><span id="line-85">                                .getAllCompounds().size());</span>
<span class="source-line-no">086</span><span id="line-86">                aminoAcidToCodon = new HashMap&lt;&gt;();</span>
<span class="source-line-no">087</span><span id="line-87"></span>
<span class="source-line-no">088</span><span id="line-88">                List&lt;Codon&gt; codonList = table.getCodons(nucleotides, aminoAcids);</span>
<span class="source-line-no">089</span><span id="line-89">                for (Codon codon : codonList) {</span>
<span class="source-line-no">090</span><span id="line-90">                        quickLookup.put(codon.getTriplet(), codon);</span>
<span class="source-line-no">091</span><span id="line-91">                        codonArray[codon.getTriplet().intValue()] = codon;</span>
<span class="source-line-no">092</span><span id="line-92"></span>
<span class="source-line-no">093</span><span id="line-93">                        List&lt;Codon&gt; codonL = aminoAcidToCodon.get(codon.getAminoAcid());</span>
<span class="source-line-no">094</span><span id="line-94">                        if (codonL == null) {</span>
<span class="source-line-no">095</span><span id="line-95">                                codonL = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">096</span><span id="line-96">                                aminoAcidToCodon.put(codon.getAminoAcid(), codonL);</span>
<span class="source-line-no">097</span><span id="line-97">                        }</span>
<span class="source-line-no">098</span><span id="line-98">                        codonL.add(codon);</span>
<span class="source-line-no">099</span><span id="line-99"></span>
<span class="source-line-no">100</span><span id="line-100">                }</span>
<span class="source-line-no">101</span><span id="line-101">                unknownAminoAcidCompound = aminoAcids.getCompoundForString("X");</span>
<span class="source-line-no">102</span><span id="line-102">                methionineAminoAcidCompound = aminoAcids.getCompoundForString("M");</span>
<span class="source-line-no">103</span><span id="line-103">                this.stopAtStopCodons = stopAtStopCodons;</span>
<span class="source-line-no">104</span><span id="line-104">                this.waitForStartCodon = waitForStartCodon;</span>
<span class="source-line-no">105</span><span id="line-105">        }</span>
<span class="source-line-no">106</span><span id="line-106"></span>
<span class="source-line-no">107</span><span id="line-107">        /**</span>
<span class="source-line-no">108</span><span id="line-108">         * Performs the core conversion of RNA to Peptide. It does this by walking a</span>
<span class="source-line-no">109</span><span id="line-109">         * windowed version of the given sequence. Any trailing DNA base pairs are</span>
<span class="source-line-no">110</span><span id="line-110">         * ignored according to the specification of {@link WindowedSequence}.</span>
<span class="source-line-no">111</span><span id="line-111">         */</span>
<span class="source-line-no">112</span><span id="line-112"></span>
<span class="source-line-no">113</span><span id="line-113">        @Override</span>
<span class="source-line-no">114</span><span id="line-114">        public List&lt;Sequence&lt;AminoAcidCompound&gt;&gt; createSequences(</span>
<span class="source-line-no">115</span><span id="line-115">                        Sequence&lt;NucleotideCompound&gt; originalSequence) {</span>
<span class="source-line-no">116</span><span id="line-116"></span>
<span class="source-line-no">117</span><span id="line-117">                List&lt;List&lt;AminoAcidCompound&gt;&gt; workingList = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">118</span><span id="line-118"></span>
<span class="source-line-no">119</span><span id="line-119">                Iterable&lt;SequenceView&lt;NucleotideCompound&gt;&gt; iter = new WindowedSequence&lt;NucleotideCompound&gt;(</span>
<span class="source-line-no">120</span><span id="line-120">                                originalSequence, 3);</span>
<span class="source-line-no">121</span><span id="line-121"></span>
<span class="source-line-no">122</span><span id="line-122">                boolean first = true;</span>
<span class="source-line-no">123</span><span id="line-123"></span>
<span class="source-line-no">124</span><span id="line-124">                // If not waiting for a start codon, start translating immediately</span>
<span class="source-line-no">125</span><span id="line-125">                boolean doTranslate = !waitForStartCodon;</span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">                for (SequenceView&lt;NucleotideCompound&gt; element : iter) {</span>
<span class="source-line-no">128</span><span id="line-128">                        AminoAcidCompound aminoAcid = null;</span>
<span class="source-line-no">129</span><span id="line-129"></span>
<span class="source-line-no">130</span><span id="line-130">                        int i = 1;</span>
<span class="source-line-no">131</span><span id="line-131">                        Table.CaseInsensitiveTriplet triplet = new Table.CaseInsensitiveTriplet(</span>
<span class="source-line-no">132</span><span id="line-132">                                        element.getCompoundAt(i++), element.getCompoundAt(i++),</span>
<span class="source-line-no">133</span><span id="line-133">                                        element.getCompoundAt(i++));</span>
<span class="source-line-no">134</span><span id="line-134"></span>
<span class="source-line-no">135</span><span id="line-135">                        Codon target = null;</span>
<span class="source-line-no">136</span><span id="line-136"></span>
<span class="source-line-no">137</span><span id="line-137">                        target = quickLookup.get(triplet);</span>
<span class="source-line-no">138</span><span id="line-138"></span>
<span class="source-line-no">139</span><span id="line-139">                        // Check for a start</span>
<span class="source-line-no">140</span><span id="line-140">                        if (!doTranslate &amp;&amp; target.isStart()) {</span>
<span class="source-line-no">141</span><span id="line-141">                                doTranslate = true;</span>
<span class="source-line-no">142</span><span id="line-142">                        }</span>
<span class="source-line-no">143</span><span id="line-143"></span>
<span class="source-line-no">144</span><span id="line-144">                        if (doTranslate) {</span>
<span class="source-line-no">145</span><span id="line-145">                                if (target != null)</span>
<span class="source-line-no">146</span><span id="line-146">                                        aminoAcid = target.getAminoAcid();</span>
<span class="source-line-no">147</span><span id="line-147">                                if (aminoAcid == null &amp;&amp; translateNCodons()) {</span>
<span class="source-line-no">148</span><span id="line-148">                                        aminoAcid = unknownAminoAcidCompound;</span>
<span class="source-line-no">149</span><span id="line-149">                                } else {</span>
<span class="source-line-no">150</span><span id="line-150">                                        if (first &amp;&amp; initMetOnly &amp;&amp; target.isStart()) {</span>
<span class="source-line-no">151</span><span id="line-151">                                                aminoAcid = methionineAminoAcidCompound;</span>
<span class="source-line-no">152</span><span id="line-152">                                        }</span>
<span class="source-line-no">153</span><span id="line-153">                                }</span>
<span class="source-line-no">154</span><span id="line-154"></span>
<span class="source-line-no">155</span><span id="line-155">                                addCompoundsToList(Arrays.asList(aminoAcid), workingList);</span>
<span class="source-line-no">156</span><span id="line-156">                        }</span>
<span class="source-line-no">157</span><span id="line-157"></span>
<span class="source-line-no">158</span><span id="line-158">                        if (doTranslate &amp;&amp; stopAtStopCodons &amp;&amp; target.isStop()) {</span>
<span class="source-line-no">159</span><span id="line-159">                                // Check if we need to stop, but dont stop until started!</span>
<span class="source-line-no">160</span><span id="line-160">                                break;</span>
<span class="source-line-no">161</span><span id="line-161">                        }</span>
<span class="source-line-no">162</span><span id="line-162"></span>
<span class="source-line-no">163</span><span id="line-163">                        first = false;</span>
<span class="source-line-no">164</span><span id="line-164">                }</span>
<span class="source-line-no">165</span><span id="line-165">                postProcessCompoundLists(workingList);</span>
<span class="source-line-no">166</span><span id="line-166"></span>
<span class="source-line-no">167</span><span id="line-167">                return workingListToSequences(workingList);</span>
<span class="source-line-no">168</span><span id="line-168">        }</span>
<span class="source-line-no">169</span><span id="line-169"></span>
<span class="source-line-no">170</span><span id="line-170">        /**</span>
<span class="source-line-no">171</span><span id="line-171">         * Performs the trimming of stop codons and the conversion of a valid start</span>
<span class="source-line-no">172</span><span id="line-172">         * amino acid to M</span>
<span class="source-line-no">173</span><span id="line-173">         */</span>
<span class="source-line-no">174</span><span id="line-174">        @Override</span>
<span class="source-line-no">175</span><span id="line-175">        protected void postProcessCompoundLists(</span>
<span class="source-line-no">176</span><span id="line-176">                        List&lt;List&lt;AminoAcidCompound&gt;&gt; compoundLists) {</span>
<span class="source-line-no">177</span><span id="line-177">                for (List&lt;AminoAcidCompound&gt; compounds : compoundLists) {</span>
<span class="source-line-no">178</span><span id="line-178">                        if (trimStops) {</span>
<span class="source-line-no">179</span><span id="line-179">                                trimStop(compounds);</span>
<span class="source-line-no">180</span><span id="line-180">                        }</span>
<span class="source-line-no">181</span><span id="line-181">                }</span>
<span class="source-line-no">182</span><span id="line-182">        }</span>
<span class="source-line-no">183</span><span id="line-183"></span>
<span class="source-line-no">184</span><span id="line-184">        /**</span>
<span class="source-line-no">185</span><span id="line-185">         * Imperfect code. Checks the last amino acid to see if a codon could have</span>
<span class="source-line-no">186</span><span id="line-186">         * translated a stop for it. Left in for the moment</span>
<span class="source-line-no">187</span><span id="line-187">         */</span>
<span class="source-line-no">188</span><span id="line-188">        protected void trimStop(List&lt;AminoAcidCompound&gt; sequence) {</span>
<span class="source-line-no">189</span><span id="line-189">                AminoAcidCompound stop = sequence.get(sequence.size() - 1);</span>
<span class="source-line-no">190</span><span id="line-190">                boolean isStop = false;</span>
<span class="source-line-no">191</span><span id="line-191">                if (aminoAcidToCodon.containsKey(stop)) {</span>
<span class="source-line-no">192</span><span id="line-192">                        for (Codon c : aminoAcidToCodon.get(stop)) {</span>
<span class="source-line-no">193</span><span id="line-193">                                if (c.isStop()) {</span>
<span class="source-line-no">194</span><span id="line-194">                                        isStop = true;</span>
<span class="source-line-no">195</span><span id="line-195">                                        break;</span>
<span class="source-line-no">196</span><span id="line-196">                                }</span>
<span class="source-line-no">197</span><span id="line-197">                        }</span>
<span class="source-line-no">198</span><span id="line-198">                }</span>
<span class="source-line-no">199</span><span id="line-199"></span>
<span class="source-line-no">200</span><span id="line-200">                if (isStop) {</span>
<span class="source-line-no">201</span><span id="line-201">                        sequence.remove(sequence.size() - 1);</span>
<span class="source-line-no">202</span><span id="line-202">                }</span>
<span class="source-line-no">203</span><span id="line-203">        }</span>
<span class="source-line-no">204</span><span id="line-204"></span>
<span class="source-line-no">205</span><span id="line-205">        /**</span>
<span class="source-line-no">206</span><span id="line-206">         * Indicates if we want to force exact translation of compounds or not i.e.</span>
<span class="source-line-no">207</span><span id="line-207">         * those with internal N RNA bases. This will cause a translation to an X</span>
<span class="source-line-no">208</span><span id="line-208">         * amino acid</span>
<span class="source-line-no">209</span><span id="line-209">         */</span>
<span class="source-line-no">210</span><span id="line-210">        public boolean translateNCodons() {</span>
<span class="source-line-no">211</span><span id="line-211">                return translateNCodons;</span>
<span class="source-line-no">212</span><span id="line-212">        }</span>
<span class="source-line-no">213</span><span id="line-213">}</span>




























































</pre>
</div>
</main>
</body>
</html>
