<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Uses of Package org.biojava.nbio.core.sequence.features (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="use: package: org.biojava.nbio.core.sequence.features">
<meta name="generator" content="javadoc/PackageUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="package-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Package org.biojava.nbio.core.sequence.features" class="title">Uses of Package<br>org.biojava.nbio.core.sequence.features</h1>
</div>
<div class="caption"><span>Packages that use <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.features">org.biojava.nbio.core.sequence.features</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.genome.io.fastq">org.biojava.nbio.genome.io.fastq</a></div>
<div class="col-last even-row-color">
<div class="block">FASTQ and variants sequence format I/O.</div>
</div>
</div>
<section class="package-uses">
<ul class="block-list">
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.features">
<div class="caption"><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.features">AbstractFeature</a></div>
<div class="col-last even-row-color">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.features">DBReferenceInfo</a></div>
<div class="col-last odd-row-color">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</div>
<div class="col-first even-row-color"><a href="class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.features">FeatureInterface</a></div>
<div class="col-last even-row-color">
<div class="block">Interface class to handle describing arbitrary features.</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/Qualifier.html#org.biojava.nbio.core.sequence.features">Qualifier</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.io">
<div class="caption"><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../io/package-summary.html">org.biojava.nbio.core.sequence.io</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.io">AbstractFeature</a></div>
<div class="col-last even-row-color">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.io">DBReferenceInfo</a></div>
<div class="col-last odd-row-color">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</div>
<div class="col-first even-row-color"><a href="class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.io">FeatureInterface</a></div>
<div class="col-last even-row-color">
<div class="block">Interface class to handle describing arbitrary features.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.loader">
<div class="caption"><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.loader">AbstractFeature</a></div>
<div class="col-last even-row-color">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/DatabaseReferenceInterface.html#org.biojava.nbio.core.sequence.loader">DatabaseReferenceInterface</a></div>
<div class="col-last odd-row-color">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</div>
<div class="col-first even-row-color"><a href="class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.loader">DBReferenceInfo</a></div>
<div class="col-last even-row-color">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/FeatureRetriever.html#org.biojava.nbio.core.sequence.loader">FeatureRetriever</a></div>
<div class="col-last odd-row-color">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</div>
<div class="col-first even-row-color"><a href="class-use/FeaturesKeyWordInterface.html#org.biojava.nbio.core.sequence.loader">FeaturesKeyWordInterface</a></div>
<div class="col-last even-row-color">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.template">
<div class="caption"><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../template/package-summary.html">org.biojava.nbio.core.sequence.template</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/DatabaseReferenceInterface.html#org.biojava.nbio.core.sequence.template">DatabaseReferenceInterface</a></div>
<div class="col-last even-row-color">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.template">FeatureInterface</a></div>
<div class="col-last odd-row-color">
<div class="block">Interface class to handle describing arbitrary features.</div>
</div>
<div class="col-first even-row-color"><a href="class-use/FeatureRetriever.html#org.biojava.nbio.core.sequence.template">FeatureRetriever</a></div>
<div class="col-last even-row-color">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/FeaturesKeyWordInterface.html#org.biojava.nbio.core.sequence.template">FeaturesKeyWordInterface</a></div>
<div class="col-last odd-row-color">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.genome.io.fastq">
<div class="caption"><span>Classes in <a href="package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../../../genome/io/fastq/package-summary.html">org.biojava.nbio.genome.io.fastq</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="class-use/QualityFeature.html#org.biojava.nbio.genome.io.fastq">QualityFeature</a></div>
<div class="col-last even-row-color">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</div>
<div class="col-first odd-row-color"><a href="class-use/QuantityFeature.html#org.biojava.nbio.genome.io.fastq">QuantityFeature</a></div>
<div class="col-last odd-row-color">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
