<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.10) on Fri Mar 12 13:02:30 CST 2021 -->
<title>Uses of Interface org.biojava.bio.seq.db.SequenceDB (biojava-legacy 1.9.5 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2021-03-12">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Uses of Interface org.biojava.bio.seq.db.SequenceDB (biojava-legacy 1.9.5 API)";
        }
    }
    catch(err) {
    }
//-->
var pathtoroot = "../../../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<main role="main">
<div class="header">
<h2 title="Uses of Interface org.biojava.bio.seq.db.SequenceDB" class="title">Uses of Interface<br>org.biojava.bio.seq.db.SequenceDB</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table class="useSummary">
<caption><span>Packages that use <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.alignment">org.biojava.bio.alignment</a></th>
<td class="colLast">
<div class="block">
Classes to generate and describe sequence alignments.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.dp">org.biojava.bio.dp</a></th>
<td class="colLast">
<div class="block">HMM and Dynamic Programming Algorithms.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.gui">org.biojava.bio.gui</a></th>
<td class="colLast">
<div class="block">Graphical interfaces for biojava objects.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.program.gff">org.biojava.bio.program.gff</a></th>
<td class="colLast">
<div class="block">GFF manipulation.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.program.phred">org.biojava.bio.program.phred</a></th>
<td class="colLast">
<div class="block">Parser for Phred output</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.program.ssaha">org.biojava.bio.program.ssaha</a></th>
<td class="colLast">
<div class="block">SSAHA sequence searching API.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.program.ssbind">org.biojava.bio.program.ssbind</a></th>
<td class="colLast">
<div class="block">
Creation of objects from SAX events using the BioJava BlastLikeDataSetCollection DTD.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.program.unigene">org.biojava.bio.program.unigene</a></th>
<td class="colLast">
<div class="block">Objects for representing Unigene clusters.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.search">org.biojava.bio.search</a></th>
<td class="colLast">
<div class="block">
Interfaces and classes for representing sequence similarity search results.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.seq.db">org.biojava.bio.seq.db</a></th>
<td class="colLast">
<div class="block">
Collections of biological sequence data.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.seq.db.biosql">org.biojava.bio.seq.db.biosql</a></th>
<td class="colLast">
<div class="block">General purpose Sequence storage in a relational database.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.seq.distributed">org.biojava.bio.seq.distributed</a></th>
<td class="colLast">
<div class="block">Sequences and SequenceDBs which are composed from data taken
from a number of data sources.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.seq.impl">org.biojava.bio.seq.impl</a></th>
<td class="colLast">
<div class="block">Standard in-memory implementations of <code>Sequence</code> and
<code>Feature</code>.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojava.bio.seq.io">org.biojava.bio.seq.io</a></th>
<td class="colLast">
<div class="block">Classes and interfaces for processing and producing flat-file representations
of sequences.</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojavax.bio.db">org.biojavax.bio.db</a></th>
<td class="colLast">
<div class="block">Interactions between biojavax objects and a DB.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colFirst" scope="row"><a href="#org.biojavax.bio.db.biosql">org.biojavax.bio.db.biosql</a></th>
<td class="colLast">
<div class="block">Interface between biojava and biosql databases</div>
</td>
</tr>
<tr class="altColor">
<th class="colFirst" scope="row"><a href="#org.biojavax.bio.db.ncbi">org.biojavax.bio.db.ncbi</a></th>
<td class="colLast">
<div class="block">Interfaces to NCBI data.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList">
<section role="region"><a id="org.biojava.bio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../alignment/package-summary.html">org.biojava.bio.alignment</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../alignment/package-summary.html">org.biojava.bio.alignment</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&lt;<a href="../../../alignment/AlignmentPair.html" title="class in org.biojava.bio.alignment">AlignmentPair</a>&gt;</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AlignmentAlgorithm.</span><code><span class="memberNameLink"><a href="../../../alignment/AlignmentAlgorithm.html#alignAll(org.biojava.bio.seq.SequenceIterator,org.biojava.bio.seq.db.SequenceDB)">alignAll</a></span>&#8203;(<a href="../../SequenceIterator.html" title="interface in org.biojava.bio.seq">SequenceIterator</a>&nbsp;source,
        <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;subjectDB)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.dp">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../dp/package-summary.html">org.biojava.bio.dp</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../dp/package-summary.html">org.biojava.bio.dp</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AbstractTrainer.</span><code><span class="memberNameLink"><a href="../../../dp/AbstractTrainer.html#train(org.biojava.bio.seq.db.SequenceDB,double,org.biojava.bio.dp.StoppingCriteria)">train</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
     double&nbsp;nullModelWeight,
     <a href="../../../dp/StoppingCriteria.html" title="interface in org.biojava.bio.dp">StoppingCriteria</a>&nbsp;stopper)</code></th>
<td class="colLast">
<div class="block">Trains the sequences in db until stopper says to finnish.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">TrainingAlgorithm.</span><code><span class="memberNameLink"><a href="../../../dp/TrainingAlgorithm.html#train(org.biojava.bio.seq.db.SequenceDB,double,org.biojava.bio.dp.StoppingCriteria)">train</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
     double&nbsp;nullWeight,
     <a href="../../../dp/StoppingCriteria.html" title="interface in org.biojava.bio.dp">StoppingCriteria</a>&nbsp;stopper)</code></th>
<td class="colLast">
<div class="block">Trains the sequences in db untill stopper says to finnish.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.gui">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../gui/package-summary.html">org.biojava.bio.gui</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../gui/package-summary.html">org.biojava.bio.gui</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">FeatureTree.</span><code><span class="memberNameLink"><a href="../../../gui/FeatureTree.html#setSequenceDB(org.biojava.bio.seq.db.SequenceDB)">setSequenceDB</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code></th>
<td class="colLast">
<div class="block">Use this method to provide the sequences for the tree to work with.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.program.gff">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../program/gff/package-summary.html">org.biojava.bio.program.gff</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../program/gff/package-summary.html">org.biojava.bio.program.gff</a> that return <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GFFTools.</span><code><span class="memberNameLink"><a href="../../../program/gff/GFFTools.html#annotateSequences(org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.program.gff.GFFEntrySet)">annotateSequences</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqs,
                 <a href="../../../program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a>&nbsp;ents)</code></th>
<td class="colLast">
<div class="block">Annotates all sequences in a sequence DB with features from a GFF entry set.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../program/gff/package-summary.html">org.biojava.bio.program.gff</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GFFTools.</span><code><span class="memberNameLink"><a href="../../../program/gff/GFFTools.html#annotateSequences(org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.program.gff.GFFEntrySet)">annotateSequences</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqs,
                 <a href="../../../program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a>&nbsp;ents)</code></th>
<td class="colLast">
<div class="block">Annotates all sequences in a sequence DB with features from a GFF entry set.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../program/gff/GFFEntrySet.html" title="class in org.biojava.bio.program.gff">GFFEntrySet</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GFFTools.</span><code><span class="memberNameLink"><a href="../../../program/gff/GFFTools.html#gffFromSeqDB(org.biojava.bio.seq.db.SequenceDB)">gffFromSeqDB</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code></th>
<td class="colLast">
<div class="block">Creates a GFFEntrySet containing one entry for each feature on each
 sequence of a SequenceDB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequencesAsGFF.</span><code><span class="memberNameLink"><a href="../../../program/gff/SequencesAsGFF.html#processDB(org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.program.gff.GFFDocumentHandler)">processDB</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
         <a href="../../../program/gff/GFFDocumentHandler.html" title="interface in org.biojava.bio.program.gff">GFFDocumentHandler</a>&nbsp;handler)</code></th>
<td class="colLast">
<div class="block">Process all <span class="type">Sequence</span>s within a
 <span class="type">SequenceDB</span>, informing
 <span class="arg">handler</span> of any suitable features.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.program.phred">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../program/phred/package-summary.html">org.biojava.bio.program.phred</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../program/phred/package-summary.html">org.biojava.bio.program.phred</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">PhredTools.</span><code><span class="memberNameLink"><a href="../../../program/phred/PhredTools.html#writePhredQuality(java.io.OutputStream,java.io.OutputStream,org.biojava.bio.seq.db.SequenceDB)">writePhredQuality</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;qual,
                 <a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;seq,
                 <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code></th>
<td class="colLast">
<div class="block">Writes Phred quality data in a Fasta type format.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.program.ssaha">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">CompactedDataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../program/ssaha/CompactedDataStoreFactory.html#buildDataStore(java.io.File,org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.symbol.Packing,int,int)">buildDataStore</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;storeFile,
              <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">DataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../program/ssaha/DataStoreFactory.html#buildDataStore(java.io.File,org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.symbol.Packing,int,int)">buildDataStore</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;storeFile,
              <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code></th>
<td class="colLast">
<div class="block">Build a new DataStore.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">MappedDataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../program/ssaha/MappedDataStoreFactory.html#buildDataStore(java.io.File,org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.symbol.Packing,int,int)">buildDataStore</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;storeFile,
              <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../program/ssaha/DataStore.html" title="interface in org.biojava.bio.program.ssaha">DataStore</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">NIODataStoreFactory.</span><code><span class="memberNameLink"><a href="../../../program/ssaha/NIODataStoreFactory.html#buildDataStore(java.io.File,org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.symbol.Packing,int,int)">buildDataStore</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io" class="externalLink">File</a>&nbsp;storeFile,
              <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB,
              <a href="../../../symbol/Packing.html" title="interface in org.biojava.bio.symbol">Packing</a>&nbsp;packing,
              int&nbsp;wordLength,
              int&nbsp;threshold)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../program/ssaha/package-summary.html">org.biojava.bio.program.ssaha</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../program/ssaha/SequenceStreamer.SequenceDBStreamer.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB)">SequenceDBStreamer</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.program.ssbind">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a></h3>
<table class="useSummary">
<caption><span>Fields in <a href="../../../program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> declared as <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ViewSequenceFactory.</span><code><span class="memberNameLink"><a href="../../../program/ssbind/ViewSequenceFactory.html#querySeqHolder">querySeqHolder</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> that return <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ViewSequenceFactory.</span><code><span class="memberNameLink"><a href="../../../program/ssbind/ViewSequenceFactory.html#getQuerySeqHolder()">getQuerySeqHolder</a></span>()</code></th>
<td class="colLast">
<div class="block"><code>getQuerySeqHolder</code> returns the database of query
 sequences used to retrieve sequences for creation of the
 various result objects.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">BlastLikeSearchBuilder.</span><code><span class="memberNameLink"><a href="../../../program/ssbind/BlastLikeSearchBuilder.html#setQuerySeqHolder(org.biojava.bio.seq.db.SequenceDB)">setQuerySeqHolder</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;querySeqHolder)</code></th>
<td class="colLast">
<div class="block"><code>setQuerySeqHolder</code> sets the query sequence holder
 to a specific database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">ViewSequenceFactory.</span><code><span class="memberNameLink"><a href="../../../program/ssbind/ViewSequenceFactory.html#setQuerySeqHolder(org.biojava.bio.seq.db.SequenceDB)">setQuerySeqHolder</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;querySeqHolder)</code></th>
<td class="colLast">
<div class="block"><code>setQuerySeqHolder</code> sets the query sequence holder
 to a specific database.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../program/ssbind/package-summary.html">org.biojava.bio.program.ssbind</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../program/ssbind/BlastLikeSearchBuilder.html#%3Cinit%3E(java.util.List,org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.seq.db.SequenceDBInstallation)">BlastLikeSearchBuilder</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&nbsp;target,
                      <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;querySeqHolder,
                      <a href="../SequenceDBInstallation.html" title="interface in org.biojava.bio.seq.db">SequenceDBInstallation</a>&nbsp;subjectDBs)</code></th>
<td class="colLast">
<div class="block">Creates a new <code>BlastLikeSearchBuilder</code> which will
 instantiate results into the <code>List</code> target.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.program.unigene">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../program/unigene/package-summary.html">org.biojava.bio.program.unigene</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../program/unigene/package-summary.html">org.biojava.bio.program.unigene</a> that return <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">UnigeneCluster.</span><code><span class="memberNameLink"><a href="../../../program/unigene/UnigeneCluster.html#getAll()">getAll</a></span>()</code></th>
<td class="colLast">
<div class="block">All sequences that map to this cluster.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.search">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../search/package-summary.html">org.biojava.bio.search</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../../search/package-summary.html">org.biojava.bio.search</a> that return <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SeqSimilaritySearchResult.</span><code><span class="memberNameLink"><a href="../../../search/SeqSimilaritySearchResult.html#getSequenceDB()">getSequenceDB</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the sequence database against which the search was
 performed.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceDBSearchResult.</span><code><span class="memberNameLink"><a href="../../../search/SequenceDBSearchResult.html#getSequenceDB()">getSequenceDB</a></span>()</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SimpleSeqSimilaritySearchResult.</span><code><span class="memberNameLink"><a href="../../../search/SimpleSeqSimilaritySearchResult.html#getSequenceDB()">getSequenceDB</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../../search/package-summary.html">org.biojava.bio.search</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../search/SeqSimilaritySearchResult.html" title="interface in org.biojava.bio.search">SeqSimilaritySearchResult</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SeqSimilaritySearcher.</span><code><span class="memberNameLink"><a href="../../../search/SeqSimilaritySearcher.html#search(org.biojava.bio.symbol.SymbolList,org.biojava.bio.seq.db.SequenceDB,java.util.Map)">search</a></span>&#8203;(<a href="../../../symbol/SymbolList.html" title="interface in org.biojava.bio.symbol">SymbolList</a>&nbsp;querySeq,
      <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&nbsp;searchParameters)</code></th>
<td class="colLast">
<div class="block">Using this sequence similarity searcher, search with the given
 sequence against the given sequence database.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../../search/package-summary.html">org.biojava.bio.search</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../search/SequenceDBSearchResult.html#%3Cinit%3E(org.biojava.bio.seq.Sequence,org.biojava.bio.seq.db.SequenceDB,java.util.Map,java.util.List,org.biojava.bio.Annotation)">SequenceDBSearchResult</a></span>&#8203;(<a href="../../Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;querySequence,
                      <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;sequenceDB,
                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&nbsp;searchParameters,
                      <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&nbsp;hits,
                      <a href="../../../Annotation.html" title="interface in org.biojava.bio">Annotation</a>&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Creates a new <code>SequenceDBSearchResult</code>.</div>
</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../search/SimpleSeqSimilaritySearchResult.html#%3Cinit%3E(org.biojava.bio.seq.Sequence,org.biojava.bio.seq.db.SequenceDB,java.util.Map,java.util.List,org.biojava.bio.Annotation)">SimpleSeqSimilaritySearchResult</a></span>&#8203;(<a href="../../Sequence.html" title="interface in org.biojava.bio.seq">Sequence</a>&nbsp;querySequence,
                               <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;sequenceDB,
                               <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util" class="externalLink">Map</a>&nbsp;searchParameters,
                               <a href="https://docs.oracle.com/javase/8/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util" class="externalLink">List</a>&nbsp;hits,
                               <a href="../../../Annotation.html" title="interface in org.biojava.bio">Annotation</a>&nbsp;annotation)</code></th>
<td class="colLast">
<div class="block">Creates a new <code>SimpleSeqSimilaritySearchResult</code>.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.seq.db">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../package-summary.html">org.biojava.bio.seq.db</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../package-summary.html">org.biojava.bio.seq.db</a> that implement <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AbstractSequenceDB.html" title="class in org.biojava.bio.seq.db">AbstractSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">An abstract implementation of SequenceDB that provides the sequenceIterator
 method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AnnotatedSequenceDB.html" title="class in org.biojava.bio.seq.db">AnnotatedSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">SequenceDB implementation which lazily applies a SequenceAnnotator
 to sequences retrieved from a SequenceDB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CachingSequenceDB.html" title="class in org.biojava.bio.seq.db">CachingSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">SequenceDB implementation that caches the results of another SequenceDB.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../DummySequenceDB.html" title="class in org.biojava.bio.seq.db">DummySequenceDB</a></span></code></th>
<td class="colLast">
<div class="block"><code>DummySequenceDB</code> is an implementation which contains
 only a <code>DummySequence</code>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../HashSequenceDB.html" title="class in org.biojava.bio.seq.db">HashSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">An implementation of SequenceDB that uses an underlying HashMap to store the
 sequence objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../IndexedSequenceDB.html" title="class in org.biojava.bio.seq.db">IndexedSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">
 This class implements SequenceDB on top of a set of sequence files
 and sequence offsets within these files.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceDBWrapper.html" title="class in org.biojava.bio.seq.db">SequenceDBWrapper</a></span></code></th>
<td class="colLast">
<div class="block">An abstract implementation of SequenceDB that wraps up another database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SubSequenceDB.html" title="class in org.biojava.bio.seq.db">SubSequenceDB</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ViewingSequenceDB.html" title="class in org.biojava.bio.seq.db">ViewingSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">SequenceDB implementation that returns new SequenceView instances
 wrapping the sequences in an underlying database.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.bio.seq.db</a> that return <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">AnnotatedSequenceDB.</span><code><span class="memberNameLink"><a href="../AnnotatedSequenceDB.html#getParent()">getParent</a></span>()</code></th>
<td class="colLast">
<div class="block">Get the original sequenceDB from this annotated sequenceDB.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SequenceDBWrapper.</span><code><span class="memberNameLink"><a href="../SequenceDBWrapper.html#getParent()">getParent</a></span>()</code></th>
<td class="colLast">
<div class="block">Return the parent SequenceDB.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankSequenceDB.</span><code><span class="memberNameLink"><a href="../GenbankSequenceDB.html#getSequences(java.util.Set)">getSequences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&nbsp;list)</code></th>
<td class="colLast">
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankSequenceDB.</span><code><span class="memberNameLink"><a href="../GenbankSequenceDB.html#getSequences(java.util.Set,org.biojava.bio.seq.db.SequenceDB)">getSequences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&nbsp;list,
            <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;database)</code></th>
<td class="colLast">
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../package-summary.html">org.biojava.bio.seq.db</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">GenbankSequenceDB.</span><code><span class="memberNameLink"><a href="../GenbankSequenceDB.html#getSequences(java.util.Set,org.biojava.bio.seq.db.SequenceDB)">getSequences</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&nbsp;list,
            <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;database)</code></th>
<td class="colLast">
<div class="block">Retrieve sequences from a Genbank</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../package-summary.html">org.biojava.bio.seq.db</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../AnnotatedSequenceDB.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB,org.biojava.bio.seq.SequenceAnnotator)">AnnotatedSequenceDB</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent,
                   <a href="../../SequenceAnnotator.html" title="interface in org.biojava.bio.seq">SequenceAnnotator</a>&nbsp;a)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../CachingSequenceDB.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB)">CachingSequenceDB</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent)</code></th>
<td class="colLast">
<div class="block">Create a new CachingSequenceDB that caches the sequences in parent.</div>
</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SequenceDBWrapper.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB)">SequenceDBWrapper</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../SubSequenceDB.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB,java.util.Set)">SubSequenceDB</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent,
             <a href="https://docs.oracle.com/javase/8/docs/api/java/util/Set.html?is-external=true" title="class or interface in java.util" class="externalLink">Set</a>&nbsp;ids)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../ViewingSequenceDB.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB)">ViewingSequenceDB</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;parent)</code></th>
<td class="colLast">
<div class="block">Create a new ViewingSequenceDB that views the sequences in parent.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.seq.db.biosql">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../biosql/package-summary.html">org.biojava.bio.seq.db.biosql</a> that implement <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../biosql/BioSQLSequenceDB.html" title="class in org.biojava.bio.seq.db.biosql">BioSQLSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span>
<div class="deprecationComment">Use hibernate and org.biojavax.bio.db</div>
</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.seq.distributed">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../distributed/package-summary.html">org.biojava.bio.seq.distributed</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../distributed/package-summary.html">org.biojava.bio.seq.distributed</a> that implement <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../distributed/DistributedSequenceDB.html" title="class in org.biojava.bio.seq.distributed">DistributedSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">Sequence database from the meta-DAS system.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../distributed/package-summary.html">org.biojava.bio.seq.distributed</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../distributed/SequenceDBDataSource.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB)">SequenceDBDataSource</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.seq.impl">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../impl/package-summary.html">org.biojava.bio.seq.impl</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../impl/package-summary.html">org.biojava.bio.seq.impl</a> that return <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SimpleRemoteFeature.DBResolver.</span><code><span class="memberNameLink"><a href="../../impl/SimpleRemoteFeature.DBResolver.html#getSeqDB()">getSeqDB</a></span>()</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../impl/package-summary.html">org.biojava.bio.seq.impl</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../impl/SimpleRemoteFeature.DBResolver.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB)">DBResolver</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;seqDB)</code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojava.bio.seq.io">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../io/package-summary.html">org.biojava.bio.seq.io</a></h3>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava.bio.seq.io</a> that return <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SeqIOTools.</span><code><span class="memberNameLink"><a href="../../io/SeqIOTools.html#readFasta(java.io.InputStream,org.biojava.bio.symbol.Alphabet)">readFasta</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">InputStream</a>&nbsp;seqFile,
         <a href="../../../symbol/Alphabet.html" title="interface in org.biojava.bio.symbol">Alphabet</a>&nbsp;alpha)</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Create a sequence database from a fasta file provided as an
 input stream.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Methods in <a href="../../io/package-summary.html">org.biojava.bio.seq.io</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<th class="colSecond" scope="row"><span class="typeNameLabel">SeqIOTools.</span><code><span class="memberNameLink"><a href="../../io/SeqIOTools.html#writeFasta(java.io.OutputStream,org.biojava.bio.seq.db.SequenceDB)">writeFasta</a></span>&#8203;(<a href="https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io" class="externalLink">OutputStream</a>&nbsp;os,
          <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db)</code></th>
<td class="colLast">
<div class="block"><span class="deprecatedLabel">Deprecated.</span></div>
<div class="block">Write a sequenceDB to an output stream in fasta format.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Constructors in <a href="../../io/package-summary.html">org.biojava.bio.seq.io</a> with parameters of type <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Constructor</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../io/SequenceDBSequenceBuilder.html#%3Cinit%3E(org.biojava.bio.seq.db.SequenceDB,int)">SequenceDBSequenceBuilder</a></span>&#8203;(<a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a>&nbsp;db,
                         int&nbsp;mode)</code></th>
<td class="colLast">
<div class="block">constructor</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojavax.bio.db">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a></h3>
<table class="useSummary">
<caption><span>Subinterfaces of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Interface</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../biojavax/bio/db/RichSequenceDB.html" title="interface in org.biojavax.bio.db">RichSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">A database of RichSequences with accessible keys and iterators over all
 sequences.</div>
</td>
</tr>
</tbody>
</table>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../../biojavax/bio/db/package-summary.html">org.biojavax.bio.db</a> that implement <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../biojavax/bio/db/AbstractRichSequenceDB.html" title="class in org.biojavax.bio.db">AbstractRichSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">An abstract implementation of RichSequenceDB that provides the getRichSequenceIterator
 method.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../biojavax/bio/db/HashRichSequenceDB.html" title="class in org.biojavax.bio.db">HashRichSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">An implementation of RichSequenceDB that uses an underlying HashMap to store the
 RichSequence objects.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojavax.bio.db.biosql">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../../biojavax/bio/db/biosql/package-summary.html">org.biojavax.bio.db.biosql</a> that implement <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../biojavax/bio/db/biosql/BioSQLRichSequenceDB.html" title="class in org.biojavax.bio.db.biosql">BioSQLRichSequenceDB</a></span></code></th>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</section>
</li>
<li class="blockList">
<section role="region"><a id="org.biojavax.bio.db.ncbi">
<!--   -->
</a>
<h3>Uses of <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a> in <a href="../../../../../biojavax/bio/db/ncbi/package-summary.html">org.biojavax.bio.db.ncbi</a></h3>
<table class="useSummary">
<caption><span>Classes in <a href="../../../../../biojavax/bio/db/ncbi/package-summary.html">org.biojavax.bio.db.ncbi</a> that implement <a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">SequenceDB</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Class</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../biojavax/bio/db/ncbi/GenbankRichSequenceDB.html" title="class in org.biojavax.bio.db.ncbi">GenbankRichSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">This class contains functions accessing DNA sequences in Genbank format.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="../../../../../biojavax/bio/db/ncbi/GenpeptRichSequenceDB.html" title="class in org.biojavax.bio.db.ncbi">GenpeptRichSequenceDB</a></span></code></th>
<td class="colLast">
<div class="block">This class contains functions accessing Peptide sequences in Genpept format.</div>
</td>
</tr>
</tbody>
</table>
</section>
</li>
</ul>
</li>
</ul>
</div>
</main>
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../SequenceDB.html" title="interface in org.biojava.bio.seq.db">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" type="text/javascript"></script><script type="text/javascript">_uacct = "UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2021 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
