<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.bio.program.ssbind;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.util.ArrayList;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.util.HashMap;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.util.List;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.util.Map;<a name="line.27"></a>
<span class="sourceLineNo">028</span><a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.bio.BioException;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.bio.alignment.Alignment;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.bio.alignment.SimpleAlignment;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.bio.search.SearchContentHandler;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.bio.seq.Sequence;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.biojava.bio.seq.StrandedFeature;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.bio.seq.StrandedFeature.Strand;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.bio.seq.homol.Homology;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.bio.seq.homol.HomologyFeature;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.bio.seq.homol.SimpleHomology;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.bio.seq.io.SymbolTokenization;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.bio.symbol.FiniteAlphabet;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.bio.symbol.RangeLocation;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.bio.symbol.SimpleSymbolList;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.biojava.utils.ChangeVetoException;<a name="line.43"></a>
<span class="sourceLineNo">044</span><a name="line.44"></a>
<span class="sourceLineNo">045</span>/**<a name="line.45"></a>
<span class="sourceLineNo">046</span> * &lt;p&gt;&lt;code&gt;BlastLikeHomologyBuilder&lt;/code&gt; populates a<a name="line.46"></a>
<span class="sourceLineNo">047</span> * &lt;code&gt;List&lt;/code&gt; with &lt;code&gt;Homology&lt;/code&gt; instances created from<a name="line.47"></a>
<span class="sourceLineNo">048</span> * SAX events supplied via a &lt;code&gt;SeqSimilarityAdapter&lt;/code&gt;. The<a name="line.48"></a>
<span class="sourceLineNo">049</span> * SAX events should describe elements conforming to the BioJava<a name="line.49"></a>
<span class="sourceLineNo">050</span> * BlastLikeDataSetCollection DTD. Suitable sources are<a name="line.50"></a>
<span class="sourceLineNo">051</span> * &lt;code&gt;BlastLikeSAXParser&lt;/code&gt; or<a name="line.51"></a>
<span class="sourceLineNo">052</span> * &lt;code&gt;FastaSAXParser&lt;/code&gt;. Annotated &lt;code&gt;ViewSequence&lt;/code&gt;s<a name="line.52"></a>
<span class="sourceLineNo">053</span> * wrapping both query and subject sequences are created and populated<a name="line.53"></a>
<span class="sourceLineNo">054</span> * with &lt;code&gt;HomologyFeature&lt;/code&gt;s. See the documentation of<a name="line.54"></a>
<span class="sourceLineNo">055</span> * &lt;code&gt;Homology&lt;/code&gt; and &lt;code&gt;HomologyFeature&lt;/code&gt;.&lt;/p&gt;<a name="line.55"></a>
<span class="sourceLineNo">056</span> *<a name="line.56"></a>
<span class="sourceLineNo">057</span> * &lt;p&gt;As &lt;code&gt;SimpleHomologyFeature&lt;/code&gt;s are created on views of<a name="line.57"></a>
<span class="sourceLineNo">058</span> * the query and subject sequences, both query and subject should be<a name="line.58"></a>
<span class="sourceLineNo">059</span> * nucleotide sequences (&lt;code&gt;SimpleHomologyFeature&lt;/code&gt; extends<a name="line.59"></a>
<span class="sourceLineNo">060</span> * &lt;code&gt;StrandedFeature&lt;/code&gt;.). This limits the searches currently<a name="line.60"></a>
<span class="sourceLineNo">061</span> * handled to BLASTN, TBLASTX and Fasta DNA.&lt;/p&gt;<a name="line.61"></a>
<span class="sourceLineNo">062</span> *<a name="line.62"></a>
<span class="sourceLineNo">063</span> * @author Keith James<a name="line.63"></a>
<span class="sourceLineNo">064</span> * @author Greg Cox<a name="line.64"></a>
<span class="sourceLineNo">065</span> * @since 1.2<a name="line.65"></a>
<span class="sourceLineNo">066</span> */<a name="line.66"></a>
<span class="sourceLineNo">067</span>public class BlastLikeHomologyBuilder extends ViewSequenceFactory<a name="line.67"></a>
<span class="sourceLineNo">068</span>    implements SearchContentHandler<a name="line.68"></a>
<span class="sourceLineNo">069</span>{<a name="line.69"></a>
<span class="sourceLineNo">070</span>    /**<a name="line.70"></a>
<span class="sourceLineNo">071</span>     * &lt;code&gt;HOMOLOGY_FEATURE_TYPE&lt;/code&gt; is the type String used by<a name="line.71"></a>
<span class="sourceLineNo">072</span>     * &lt;code&gt;BlastLikeHomologyBuilder&lt;/code&gt; when creating<a name="line.72"></a>
<span class="sourceLineNo">073</span>     * &lt;code&gt;HomologyFeature&lt;/code&gt;s. This is the String which is<a name="line.73"></a>
<span class="sourceLineNo">074</span>     * returned when an &lt;code&gt;HomologyFeature&lt;/code&gt;'s<a name="line.74"></a>
<span class="sourceLineNo">075</span>     * &lt;code&gt;getType()&lt;/code&gt; method is called.<a name="line.75"></a>
<span class="sourceLineNo">076</span>     */<a name="line.76"></a>
<span class="sourceLineNo">077</span>    public static final String HOMOLOGY_FEATURE_TYPE = "homology";<a name="line.77"></a>
<span class="sourceLineNo">078</span><a name="line.78"></a>
<span class="sourceLineNo">079</span>    // Identifiers for query and database<a name="line.79"></a>
<span class="sourceLineNo">080</span>    private String queryID;<a name="line.80"></a>
<span class="sourceLineNo">081</span><a name="line.81"></a>
<span class="sourceLineNo">082</span>    // Data holders for search result properties<a name="line.82"></a>
<span class="sourceLineNo">083</span>    private Map resultData;<a name="line.83"></a>
<span class="sourceLineNo">084</span>    private Map hitData;<a name="line.84"></a>
<span class="sourceLineNo">085</span>    private Map subHitData;<a name="line.85"></a>
<span class="sourceLineNo">086</span><a name="line.86"></a>
<span class="sourceLineNo">087</span>    private SymbolTokenization tokenParser;<a name="line.87"></a>
<span class="sourceLineNo">088</span>    private StringBuffer       tokenBuffer;<a name="line.88"></a>
<span class="sourceLineNo">089</span><a name="line.89"></a>
<span class="sourceLineNo">090</span>    // List for holding homologies from current search. There may be<a name="line.90"></a>
<span class="sourceLineNo">091</span>    // more than one search result in a stream<a name="line.91"></a>
<span class="sourceLineNo">092</span>    private List homologies;<a name="line.92"></a>
<span class="sourceLineNo">093</span>    // Flag indicating whether there are more results in the stream<a name="line.93"></a>
<span class="sourceLineNo">094</span>    private boolean moreSearchesAvailable = false;<a name="line.94"></a>
<span class="sourceLineNo">095</span>    // List to accept homologies from all results in the stream<a name="line.95"></a>
<span class="sourceLineNo">096</span>    private List target;<a name="line.96"></a>
<span class="sourceLineNo">097</span><a name="line.97"></a>
<span class="sourceLineNo">098</span>    /**<a name="line.98"></a>
<span class="sourceLineNo">099</span>     * Creates a new &lt;code&gt;BlastLikeHomologyBuilder&lt;/code&gt; which will<a name="line.99"></a>
<span class="sourceLineNo">100</span>     * instantiate &lt;code&gt;Homology&lt;/code&gt; objects into the<a name="line.100"></a>
<span class="sourceLineNo">101</span>     * &lt;code&gt;List&lt;/code&gt; target.<a name="line.101"></a>
<span class="sourceLineNo">102</span>     *<a name="line.102"></a>
<span class="sourceLineNo">103</span>     * @param target a &lt;code&gt;List&lt;/code&gt;.<a name="line.103"></a>
<span class="sourceLineNo">104</span>     */<a name="line.104"></a>
<span class="sourceLineNo">105</span>    public BlastLikeHomologyBuilder(List target)<a name="line.105"></a>
<span class="sourceLineNo">106</span>    {<a name="line.106"></a>
<span class="sourceLineNo">107</span>        this.target = target;<a name="line.107"></a>
<span class="sourceLineNo">108</span><a name="line.108"></a>
<span class="sourceLineNo">109</span>        resultData       = new HashMap();<a name="line.109"></a>
<span class="sourceLineNo">110</span>        hitData          = new HashMap();<a name="line.110"></a>
<span class="sourceLineNo">111</span>        subHitData       = new HashMap();<a name="line.111"></a>
<span class="sourceLineNo">112</span>        queryViewCache   = new HashMap();<a name="line.112"></a>
<span class="sourceLineNo">113</span>        subjectViewCache = new HashMap();<a name="line.113"></a>
<span class="sourceLineNo">114</span>        tokenBuffer      = new StringBuffer(1024);<a name="line.114"></a>
<span class="sourceLineNo">115</span>    }<a name="line.115"></a>
<span class="sourceLineNo">116</span><a name="line.116"></a>
<span class="sourceLineNo">117</span>    public void setQueryID(String queryID)<a name="line.117"></a>
<span class="sourceLineNo">118</span>    {<a name="line.118"></a>
<span class="sourceLineNo">119</span>        this.queryID = queryID;<a name="line.119"></a>
<span class="sourceLineNo">120</span>    }<a name="line.120"></a>
<span class="sourceLineNo">121</span><a name="line.121"></a>
<span class="sourceLineNo">122</span>    public boolean getMoreSearches()<a name="line.122"></a>
<span class="sourceLineNo">123</span>    {<a name="line.123"></a>
<span class="sourceLineNo">124</span>        return moreSearchesAvailable;<a name="line.124"></a>
<span class="sourceLineNo">125</span>    }<a name="line.125"></a>
<span class="sourceLineNo">126</span><a name="line.126"></a>
<span class="sourceLineNo">127</span>    public void setMoreSearches(boolean value)<a name="line.127"></a>
<span class="sourceLineNo">128</span>    {<a name="line.128"></a>
<span class="sourceLineNo">129</span>        moreSearchesAvailable = value;<a name="line.129"></a>
<span class="sourceLineNo">130</span>    }<a name="line.130"></a>
<span class="sourceLineNo">131</span><a name="line.131"></a>
<span class="sourceLineNo">132</span>    public void startSearch()<a name="line.132"></a>
<span class="sourceLineNo">133</span>    {<a name="line.133"></a>
<span class="sourceLineNo">134</span>        subjectViewCache.clear();<a name="line.134"></a>
<span class="sourceLineNo">135</span>        homologies = new ArrayList();<a name="line.135"></a>
<span class="sourceLineNo">136</span>    }<a name="line.136"></a>
<span class="sourceLineNo">137</span><a name="line.137"></a>
<span class="sourceLineNo">138</span>    public void endSearch()<a name="line.138"></a>
<span class="sourceLineNo">139</span>    {<a name="line.139"></a>
<span class="sourceLineNo">140</span>        target.addAll(homologies);<a name="line.140"></a>
<span class="sourceLineNo">141</span>    }<a name="line.141"></a>
<span class="sourceLineNo">142</span><a name="line.142"></a>
<span class="sourceLineNo">143</span>    public void startHeader()<a name="line.143"></a>
<span class="sourceLineNo">144</span>    {<a name="line.144"></a>
<span class="sourceLineNo">145</span>        resultData.clear();<a name="line.145"></a>
<span class="sourceLineNo">146</span>    }<a name="line.146"></a>
<span class="sourceLineNo">147</span><a name="line.147"></a>
<span class="sourceLineNo">148</span>    public void endHeader() { }<a name="line.148"></a>
<span class="sourceLineNo">149</span><a name="line.149"></a>
<span class="sourceLineNo">150</span>    public void startHit()<a name="line.150"></a>
<span class="sourceLineNo">151</span>    {<a name="line.151"></a>
<span class="sourceLineNo">152</span>        hitData.clear();<a name="line.152"></a>
<span class="sourceLineNo">153</span>        subHitData.clear();<a name="line.153"></a>
<span class="sourceLineNo">154</span>    }<a name="line.154"></a>
<span class="sourceLineNo">155</span><a name="line.155"></a>
<span class="sourceLineNo">156</span>    public void endHit() { }<a name="line.156"></a>
<span class="sourceLineNo">157</span><a name="line.157"></a>
<span class="sourceLineNo">158</span>    public void startSubHit() { }<a name="line.158"></a>
<span class="sourceLineNo">159</span><a name="line.159"></a>
<span class="sourceLineNo">160</span>    public void endSubHit()<a name="line.160"></a>
<span class="sourceLineNo">161</span>    {<a name="line.161"></a>
<span class="sourceLineNo">162</span>        try<a name="line.162"></a>
<span class="sourceLineNo">163</span>        {<a name="line.163"></a>
<span class="sourceLineNo">164</span>            homologies.add(makeHomology());<a name="line.164"></a>
<span class="sourceLineNo">165</span>        }<a name="line.165"></a>
<span class="sourceLineNo">166</span>        catch (BioException be)<a name="line.166"></a>
<span class="sourceLineNo">167</span>        {<a name="line.167"></a>
<span class="sourceLineNo">168</span>            System.err.println("Failed to build Homology:");<a name="line.168"></a>
<span class="sourceLineNo">169</span>            be.printStackTrace();<a name="line.169"></a>
<span class="sourceLineNo">170</span>        }<a name="line.170"></a>
<span class="sourceLineNo">171</span>    }<a name="line.171"></a>
<span class="sourceLineNo">172</span><a name="line.172"></a>
<span class="sourceLineNo">173</span>    public void addSearchProperty(Object key, Object value)<a name="line.173"></a>
<span class="sourceLineNo">174</span>    {<a name="line.174"></a>
<span class="sourceLineNo">175</span>        resultData.put(key, value);<a name="line.175"></a>
<span class="sourceLineNo">176</span>    }<a name="line.176"></a>
<span class="sourceLineNo">177</span><a name="line.177"></a>
<span class="sourceLineNo">178</span>    public void addHitProperty(Object key, Object value)<a name="line.178"></a>
<span class="sourceLineNo">179</span>    {<a name="line.179"></a>
<span class="sourceLineNo">180</span>        hitData.put(key, value);<a name="line.180"></a>
<span class="sourceLineNo">181</span>    }<a name="line.181"></a>
<span class="sourceLineNo">182</span><a name="line.182"></a>
<span class="sourceLineNo">183</span>    public void addSubHitProperty(Object key, Object value)<a name="line.183"></a>
<span class="sourceLineNo">184</span>    {<a name="line.184"></a>
<span class="sourceLineNo">185</span>        subHitData.put(key, value);<a name="line.185"></a>
<span class="sourceLineNo">186</span>    }<a name="line.186"></a>
<span class="sourceLineNo">187</span><a name="line.187"></a>
<span class="sourceLineNo">188</span>    /**<a name="line.188"></a>
<span class="sourceLineNo">189</span>     * &lt;code&gt;makeHomology&lt;/code&gt; creates a new<a name="line.189"></a>
<span class="sourceLineNo">190</span>     * &lt;code&gt;SimpleHomology&lt;/code&gt; describing the similarity between<a name="line.190"></a>
<span class="sourceLineNo">191</span>     * the query and subject sequences. The<a name="line.191"></a>
<span class="sourceLineNo">192</span>     * &lt;code&gt;HomologyFeatures&lt;/code&gt; created are added to<a name="line.192"></a>
<span class="sourceLineNo">193</span>     * &lt;code&gt;ViewSequence&lt;/code&gt;s wrapping the query and subject<a name="line.193"></a>
<span class="sourceLineNo">194</span>     * sequences.<a name="line.194"></a>
<span class="sourceLineNo">195</span>     *<a name="line.195"></a>
<span class="sourceLineNo">196</span>     * @return an &lt;code&gt;Homology&lt;/code&gt;.<a name="line.196"></a>
<span class="sourceLineNo">197</span>     *<a name="line.197"></a>
<span class="sourceLineNo">198</span>     * @exception BioException if an error occurs.<a name="line.198"></a>
<span class="sourceLineNo">199</span>     */<a name="line.199"></a>
<span class="sourceLineNo">200</span>    private Homology makeHomology() throws BioException<a name="line.200"></a>
<span class="sourceLineNo">201</span>    {<a name="line.201"></a>
<span class="sourceLineNo">202</span>        subHitData.putAll(resultData);<a name="line.202"></a>
<span class="sourceLineNo">203</span>        subHitData.putAll(hitData);<a name="line.203"></a>
<span class="sourceLineNo">204</span><a name="line.204"></a>
<span class="sourceLineNo">205</span>        if (tokenParser == null)<a name="line.205"></a>
<span class="sourceLineNo">206</span>        {<a name="line.206"></a>
<span class="sourceLineNo">207</span>            String identifier;<a name="line.207"></a>
<span class="sourceLineNo">208</span>            // Try explicit sequence type first<a name="line.208"></a>
<span class="sourceLineNo">209</span>            if (subHitData.containsKey("hitSequenceType"))<a name="line.209"></a>
<span class="sourceLineNo">210</span>                identifier = (String) subHitData.get("hitSequenceType");<a name="line.210"></a>
<span class="sourceLineNo">211</span>            // Otherwise try to resolve from the program name (only<a name="line.211"></a>
<span class="sourceLineNo">212</span>            // works for Blast)<a name="line.212"></a>
<span class="sourceLineNo">213</span>            else if (subHitData.containsKey("program"))<a name="line.213"></a>
<span class="sourceLineNo">214</span>                identifier = (String) subHitData.get("program");<a name="line.214"></a>
<span class="sourceLineNo">215</span>            else<a name="line.215"></a>
<span class="sourceLineNo">216</span>                throw new BioException("Failed to determine sequence type");<a name="line.216"></a>
<span class="sourceLineNo">217</span><a name="line.217"></a>
<span class="sourceLineNo">218</span>            FiniteAlphabet alpha = AlphabetResolver.resolveAlphabet(identifier);<a name="line.218"></a>
<span class="sourceLineNo">219</span>            tokenParser = alpha.getTokenization("token");<a name="line.219"></a>
<span class="sourceLineNo">220</span>        }<a name="line.220"></a>
<span class="sourceLineNo">221</span><a name="line.221"></a>
<span class="sourceLineNo">222</span>        Strand qStrand = StrandedFeature.POSITIVE;<a name="line.222"></a>
<span class="sourceLineNo">223</span>        Strand sStrand = StrandedFeature.POSITIVE;<a name="line.223"></a>
<span class="sourceLineNo">224</span><a name="line.224"></a>
<span class="sourceLineNo">225</span>        // In cases where an explicit strand is given (FASTA DNA, BLASTN)<a name="line.225"></a>
<span class="sourceLineNo">226</span>        if (subHitData.containsKey("queryStrand") &amp;&amp;<a name="line.226"></a>
<span class="sourceLineNo">227</span>            subHitData.get("queryStrand").equals("minus"))<a name="line.227"></a>
<span class="sourceLineNo">228</span>            qStrand = StrandedFeature.NEGATIVE;<a name="line.228"></a>
<span class="sourceLineNo">229</span><a name="line.229"></a>
<span class="sourceLineNo">230</span>        if (subHitData.containsKey("subjectStrand") &amp;&amp;<a name="line.230"></a>
<span class="sourceLineNo">231</span>            subHitData.get("subjectStrand").equals("minus"))<a name="line.231"></a>
<span class="sourceLineNo">232</span>            sStrand = StrandedFeature.NEGATIVE;<a name="line.232"></a>
<span class="sourceLineNo">233</span><a name="line.233"></a>
<span class="sourceLineNo">234</span>        // In cases where a frame is given as this contains strand<a name="line.234"></a>
<span class="sourceLineNo">235</span>        // information (TBLASTN for hit, TBLASTX for both query and<a name="line.235"></a>
<span class="sourceLineNo">236</span>        // hit)<a name="line.236"></a>
<span class="sourceLineNo">237</span>        if (subHitData.containsKey("queryFrame") &amp;&amp;<a name="line.237"></a>
<span class="sourceLineNo">238</span>            ((String) subHitData.get("queryFrame")).startsWith("minus"))<a name="line.238"></a>
<span class="sourceLineNo">239</span>            qStrand = StrandedFeature.NEGATIVE;<a name="line.239"></a>
<span class="sourceLineNo">240</span><a name="line.240"></a>
<span class="sourceLineNo">241</span>        if (subHitData.containsKey("subjectFrame") &amp;&amp;<a name="line.241"></a>
<span class="sourceLineNo">242</span>            ((String) subHitData.get("subjectFrame")).startsWith("minus"))<a name="line.242"></a>
<span class="sourceLineNo">243</span>            sStrand = StrandedFeature.NEGATIVE;<a name="line.243"></a>
<span class="sourceLineNo">244</span><a name="line.244"></a>
<span class="sourceLineNo">245</span>        int qStart = Integer.parseInt((String) subHitData.get("querySequenceStart"));<a name="line.245"></a>
<span class="sourceLineNo">246</span>        int   qEnd = Integer.parseInt((String) subHitData.get("querySequenceEnd"));<a name="line.246"></a>
<span class="sourceLineNo">247</span>        int sStart = Integer.parseInt((String) subHitData.get("subjectSequenceStart"));<a name="line.247"></a>
<span class="sourceLineNo">248</span>        int   sEnd = Integer.parseInt((String) subHitData.get("subjectSequenceEnd"));<a name="line.248"></a>
<span class="sourceLineNo">249</span><a name="line.249"></a>
<span class="sourceLineNo">250</span>        // The start/end coordinates from BioJava XML don't follow the<a name="line.250"></a>
<span class="sourceLineNo">251</span>        // BioJava paradigm of start &lt; end, with orientation given by<a name="line.251"></a>
<span class="sourceLineNo">252</span>        // the strand property. Rather, they present start/end as<a name="line.252"></a>
<span class="sourceLineNo">253</span>        // displayed in BLAST output, with the coordinates being<a name="line.253"></a>
<span class="sourceLineNo">254</span>        // inverted on the reverse strand. We account for this here.<a name="line.254"></a>
<span class="sourceLineNo">255</span>        if (qStrand == StrandedFeature.NEGATIVE)<a name="line.255"></a>
<span class="sourceLineNo">256</span>        {<a name="line.256"></a>
<span class="sourceLineNo">257</span>            int swap = qStart;<a name="line.257"></a>
<span class="sourceLineNo">258</span>            qStart = qEnd;<a name="line.258"></a>
<span class="sourceLineNo">259</span>            qEnd   = swap;<a name="line.259"></a>
<span class="sourceLineNo">260</span>        }<a name="line.260"></a>
<span class="sourceLineNo">261</span><a name="line.261"></a>
<span class="sourceLineNo">262</span>        if (sStrand == StrandedFeature.NEGATIVE)<a name="line.262"></a>
<span class="sourceLineNo">263</span>        {<a name="line.263"></a>
<span class="sourceLineNo">264</span>            int swap = sStart;<a name="line.264"></a>
<span class="sourceLineNo">265</span>            sStart = sEnd;<a name="line.265"></a>
<span class="sourceLineNo">266</span>            sEnd   = swap;<a name="line.266"></a>
<span class="sourceLineNo">267</span>        }<a name="line.267"></a>
<span class="sourceLineNo">268</span><a name="line.268"></a>
<span class="sourceLineNo">269</span>        String subjectID = (String) hitData.get("subjectId");<a name="line.269"></a>
<span class="sourceLineNo">270</span><a name="line.270"></a>
<span class="sourceLineNo">271</span>        Sequence   queryView = makeQueryViewSequence(queryID);<a name="line.271"></a>
<span class="sourceLineNo">272</span>        Sequence subjectView = makeSubjectViewSequence(subjectID);<a name="line.272"></a>
<span class="sourceLineNo">273</span><a name="line.273"></a>
<span class="sourceLineNo">274</span>        SimpleHomology homology = new SimpleHomology();<a name="line.274"></a>
<span class="sourceLineNo">275</span><a name="line.275"></a>
<span class="sourceLineNo">276</span>        // Map of HomologyFeatures to Alignment sequences<a name="line.276"></a>
<span class="sourceLineNo">277</span>        Map labelMap = new HashMap();<a name="line.277"></a>
<span class="sourceLineNo">278</span><a name="line.278"></a>
<span class="sourceLineNo">279</span>        try<a name="line.279"></a>
<span class="sourceLineNo">280</span>        {<a name="line.280"></a>
<span class="sourceLineNo">281</span>            String source = "unknown";<a name="line.281"></a>
<span class="sourceLineNo">282</span>            if (subHitData.containsKey("program"))<a name="line.282"></a>
<span class="sourceLineNo">283</span>                source = (String) subHitData.get("program");<a name="line.283"></a>
<span class="sourceLineNo">284</span><a name="line.284"></a>
<span class="sourceLineNo">285</span>            tokenBuffer.setLength(0);<a name="line.285"></a>
<span class="sourceLineNo">286</span>            tokenBuffer.append((String) subHitData.get("querySequence"));<a name="line.286"></a>
<span class="sourceLineNo">287</span><a name="line.287"></a>
<span class="sourceLineNo">288</span>            // Query sequence feature<a name="line.288"></a>
<span class="sourceLineNo">289</span>            HomologyFeature.Template qt = new HomologyFeature.Template();<a name="line.289"></a>
<span class="sourceLineNo">290</span>            qt.type       = HOMOLOGY_FEATURE_TYPE;<a name="line.290"></a>
<span class="sourceLineNo">291</span>            qt.source     = source;<a name="line.291"></a>
<span class="sourceLineNo">292</span>            qt.location   = new RangeLocation(qStart, qEnd);<a name="line.292"></a>
<span class="sourceLineNo">293</span>            qt.strand     = qStrand;<a name="line.293"></a>
<span class="sourceLineNo">294</span>            qt.annotation = AnnotationFactory.makeAnnotation(subHitData);<a name="line.294"></a>
<span class="sourceLineNo">295</span>            qt.homology   = homology;<a name="line.295"></a>
<span class="sourceLineNo">296</span><a name="line.296"></a>
<span class="sourceLineNo">297</span>            // Map the new feature to the alignment SymbolList<a name="line.297"></a>
<span class="sourceLineNo">298</span>            labelMap.put(queryView.createFeature(qt),<a name="line.298"></a>
<span class="sourceLineNo">299</span>                         new SimpleSymbolList(tokenParser, tokenBuffer.substring(0)));<a name="line.299"></a>
<span class="sourceLineNo">300</span><a name="line.300"></a>
<span class="sourceLineNo">301</span>            tokenBuffer.setLength(0);<a name="line.301"></a>
<span class="sourceLineNo">302</span>            tokenBuffer.append((String) subHitData.get("subjectSequence"));<a name="line.302"></a>
<span class="sourceLineNo">303</span><a name="line.303"></a>
<span class="sourceLineNo">304</span>            // Subject sequence feature<a name="line.304"></a>
<span class="sourceLineNo">305</span>            HomologyFeature.Template st = new HomologyFeature.Template();<a name="line.305"></a>
<span class="sourceLineNo">306</span>            st.type       = HOMOLOGY_FEATURE_TYPE;<a name="line.306"></a>
<span class="sourceLineNo">307</span>            st.source     = source;<a name="line.307"></a>
<span class="sourceLineNo">308</span>            st.location   = new RangeLocation(sStart, sEnd);<a name="line.308"></a>
<span class="sourceLineNo">309</span>            st.strand     = sStrand;<a name="line.309"></a>
<span class="sourceLineNo">310</span>            st.annotation = AnnotationFactory.makeAnnotation(subHitData);<a name="line.310"></a>
<span class="sourceLineNo">311</span>            st.homology   = homology;<a name="line.311"></a>
<span class="sourceLineNo">312</span><a name="line.312"></a>
<span class="sourceLineNo">313</span>            // Map the new feature to the alignment SymbolList<a name="line.313"></a>
<span class="sourceLineNo">314</span>            labelMap.put(subjectView.createFeature(st),<a name="line.314"></a>
<span class="sourceLineNo">315</span>                         new SimpleSymbolList(tokenParser, tokenBuffer.substring(0)));<a name="line.315"></a>
<span class="sourceLineNo">316</span><a name="line.316"></a>
<span class="sourceLineNo">317</span>            Alignment a = new SimpleAlignment(labelMap);<a name="line.317"></a>
<span class="sourceLineNo">318</span>            homology.setAlignment(a);<a name="line.318"></a>
<span class="sourceLineNo">319</span><a name="line.319"></a>
<span class="sourceLineNo">320</span>            return homology;<a name="line.320"></a>
<span class="sourceLineNo">321</span>        }<a name="line.321"></a>
<span class="sourceLineNo">322</span>        catch (ChangeVetoException cve)<a name="line.322"></a>
<span class="sourceLineNo">323</span>        {<a name="line.323"></a>
<span class="sourceLineNo">324</span>            throw new BioException( "Failed to create HomologyFeature",cve);<a name="line.324"></a>
<span class="sourceLineNo">325</span>        }<a name="line.325"></a>
<span class="sourceLineNo">326</span>    }<a name="line.326"></a>
<span class="sourceLineNo">327</span>}<a name="line.327"></a>




























































</pre>
</div>
</body>
</html>
