<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:50 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.nbio.core.sequence.template.Sequence (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava.nbio.core.sequence.template.Sequence (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/sequence/template/class-use/Sequence.html" target="_top">Frames</a></li>
<li><a href="Sequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.core.sequence.template.Sequence" class="title">Uses of Interface<br>org.biojava.nbio.core.sequence.template.Sequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.aaproperties.xml">org.biojava.nbio.aaproperties.xml</a></td>
<td class="colLast">
<div class="block">Set of classes that handles the reading and writing of xml files.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.alignment">org.biojava.nbio.alignment</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.alignment.routines">org.biojava.nbio.alignment.routines</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.alignment.template">org.biojava.nbio.alignment.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.alignment">org.biojava.nbio.core.alignment</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.alignment.template">org.biojava.nbio.core.alignment.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.search.io">org.biojava.nbio.core.search.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.search.io.blast">org.biojava.nbio.core.search.io.blast</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.compound">org.biojava.nbio.core.sequence.compound</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.edits">org.biojava.nbio.core.sequence.edits</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io.template">org.biojava.nbio.core.sequence.io.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io.util">org.biojava.nbio.core.sequence.io.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.location.template">org.biojava.nbio.core.sequence.location.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.storage">org.biojava.nbio.core.sequence.storage</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.transcription">org.biojava.nbio.core.sequence.transcription</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.views">org.biojava.nbio.core.sequence.views</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.util">org.biojava.nbio.core.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.phylo">org.biojava.nbio.phylo</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure">org.biojava.nbio.structure</a></td>
<td class="colLast">
<div class="block">
Interfaces and classes for protein structure (PDB).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></td>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.ws.alignment">org.biojava.nbio.ws.alignment</a></td>
<td class="colLast">
<div class="block">This package contains the interfaces that need to be implemented by discrete alignment services.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.ws.alignment.qblast">org.biojava.nbio.ws.alignment.qblast</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.aaproperties.xml">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/aaproperties/xml/package-summary.html">org.biojava.nbio.aaproperties.xml</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/aaproperties/xml/package-summary.html">org.biojava.nbio.aaproperties.xml</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">CaseFreeAminoAcidCompoundSet.</span><code><strong><a href="../../../../../../../org/biojava/nbio/aaproperties/xml/CaseFreeAminoAcidCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">ModifiedAminoAcidCompoundSet.</span><code><strong><a href="../../../../../../../org/biojava/nbio/aaproperties/xml/ModifiedAminoAcidCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/alignment/package-summary.html">org.biojava.nbio.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/alignment/package-summary.html">org.biojava.nbio.alignment</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/FractionalIdentityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityInProfileScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/FractionalIdentityScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/FractionalSimilarityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityInProfileScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/FractionalSimilarityScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a data structure for a guide tree used during progressive multiple sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/NeedlemanWunsch.html" title="class in org.biojava.nbio.alignment">NeedlemanWunsch</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Needleman and Wunsch defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="class in org.biojava.nbio.alignment">SimpleProfileProfileAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a simple (naive) <a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SmithWaterman.html" title="class in org.biojava.nbio.alignment">SmithWaterman</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Smith and Waterman defined an algorithm for pairwise local sequence alignments (best match of sections from each
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/StandardRescoreRefiner.html" title="class in org.biojava.nbio.alignment">StandardRescoreRefiner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="class in org.biojava.nbio.alignment">SubstitutionMatrixScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Scores using a substitution matrix.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/alignment/package-summary.html">org.biojava.nbio.alignment</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getAllPairsAlignments(java.util.List,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsAlignments</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences,
                     <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                     <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                     <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which computes a sequence alignment for all <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S,C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getAllPairsScorers(java.util.List,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScorers</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences,
                  <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                  <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which sets up a sequence pair scorer for all <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>double[]</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getAllPairsScores(java.util.List,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScores</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences,
                 <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                 <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                 <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which computes a sequence pair score for all <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S,C&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getMultipleSequenceAlignment(java.util.List,%20java.lang.Object...)">getMultipleSequenceAlignment</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences,
                            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>...&nbsp;settings)</code>
<div class="block">Factory method which computes a multiple sequence alignment for the given <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a> of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;S,C&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getPairwiseAligner(S,%20S,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAligner</a></strong>(S&nbsp;query,
                  S&nbsp;target,
                  <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                  <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which constructs a pairwise sequence aligner.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,C&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getPairwiseAlignment(S,%20S,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAlignment</a></strong>(S&nbsp;query,
                    S&nbsp;target,
                    <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                    <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                    <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which computes a sequence alignment for the given <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pair.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S,C&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getProgressiveAlignment(org.biojava.nbio.alignment.GuideTree,%20org.biojava.nbio.alignment.Alignments.ProfileProfileAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getProgressiveAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;S,C&gt;&nbsp;tree,
                       <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.ProfileProfileAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.ProfileProfileAlignerType</a>&nbsp;type,
                       <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                       <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method to run the profile-profile alignments of a progressive multiple sequence alignment concurrently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>double[]</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#runPairwiseScorers(java.util.List)">runPairwiseScorers</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S,C&gt;&gt;&nbsp;scorers)</code>
<div class="block">Factory method to run a list of scorers concurrently.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.alignment.routines">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.routines">AnchoredPairwiseSequenceAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">This algorithm uses a divide-and-conquer approach to find optimal pairwise global sequence alignments (from the
 first until the last <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>) with the restriction that any alignment produced
 will connect the query sequence to the target sequence at the <em>anchors</em>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="class in org.biojava.nbio.alignment.routines">GuanUberbacher</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Guan and Uberbacher defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractMatrixAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements common code for an <a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractPairwiseSequenceAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements common code for an <a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractProfileProfileAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements common code for an <a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template">Aligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an algorithm which computes an alignment <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a> from a list of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/CallablePairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a concurrency wrapper for a <a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceAligner</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/CallablePairwiseSequenceScorer.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a concurrency wrapper for a <a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceScorer</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/CallableProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">CallableProfileProfileAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a concurrency wrapper for a <a href="../../../../../../../org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>ProfileProfileAligner</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/GuideTreeNode.html" title="interface in org.biojava.nbio.alignment.template">GuideTreeNode</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a data structure for the node in a guide tree used during progressive multiple sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/HierarchicalClusterer.html" title="interface in org.biojava.nbio.alignment.template">HierarchicalClusterer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a clustering algorithm that converts a distance matrix into a tree.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/MatrixAligner.html" title="interface in org.biojava.nbio.alignment.template">MatrixAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an <a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/PairInProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">PairInProfileScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an algorithm which computes a score for a sequence alignment pair picked from an alignment <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an <a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an algorithm which computes a score for a pair of sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/PartitionRefiner.html" title="interface in org.biojava.nbio.alignment.template">PartitionRefiner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by splitting a current alignment and realigning.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileAligner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an <a href="../../../../../../../org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <code>Profile</code>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/ProfileProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileScorer</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an algorithm which computes a score for a pairing of alignment profiles.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/RescoreRefiner.html" title="interface in org.biojava.nbio.alignment.template">RescoreRefiner</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines an algorithm which computes a new alignment <code>Profile</code> by rescoring all pairs in an alignment and
 realigning.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a data structure for a <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleProfile.html" title="class in org.biojava.nbio.core.alignment">SimpleProfile</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a data structure for the results of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleProfilePair.html" title="class in org.biojava.nbio.core.alignment">SimpleProfilePair</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a data structure for the results of the alignment of a pair of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleSequencePair.html" title="class in org.biojava.nbio.core.alignment">SimpleSequencePair</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a data structure for a <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleProfile.html" title="type parameter in SimpleProfile">S</a>,<a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleProfile.html" title="type parameter in SimpleProfile">C</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">SimpleProfile.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleProfile.html#getAlignedSequences(S...)">getAlignedSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleProfile.html" title="type parameter in SimpleProfile">S</a>...&nbsp;sequences)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a data structure for a <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a mutable (editable) data structure for an <a href="../../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/MutableProfile.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfile</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a mutable (editable) data structure for a <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/MutableProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfilePair</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a mutable (editable) data structure for a <a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template"><code>ProfilePair</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/MutableSequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableSequencePair</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a mutable (editable) data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a data structure for the results of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a data structure for the results of the alignment of a pair of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfileView.html" title="interface in org.biojava.nbio.core.alignment.template">ProfileView</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a data structure for a view of sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a data structure for the results of pairwise sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subinterfaces, and an explanation">
<caption><span>Subinterfaces of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a data structure for a <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a mutable (editable) data structure for an <a href="../../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="type parameter in Profile">S</a>,<a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="type parameter in Profile">C</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">Profile.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html#getAlignedSequences(S...)">getAlignedSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="type parameter in Profile">S</a>...&nbsp;sequences)</code>
<div class="block">Returns a <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a> containing some of the individual <a href="../../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>s of this alignment.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.search.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/search/io/package-summary.html">org.biojava.nbio.core.search.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/search/io/package-summary.html">org.biojava.nbio.core.search.io</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/Hsp.html" title="class in org.biojava.nbio.core.search.io">Hsp</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">This class models a search Hsp.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/search/io/package-summary.html">org.biojava.nbio.core.search.io</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></code></td>
<td class="colLast"><span class="strong">Hit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/Hit.html#getHitSequence()">getHitSequence</a></strong>()</code>
<div class="block">returns the reference to the original and whole sequence hit in the database.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></code></td>
<td class="colLast"><span class="strong">Result.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/Result.html#getQuerySequence()">getQuerySequence</a></strong>()</code>
<div class="block">returns the reference to the original and whole sequence used to query the database.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/core/search/io/package-summary.html">org.biojava.nbio.core.search.io</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ResultFactory.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/ResultFactory.html#setDatabaseReferences(java.util.List)">setDatabaseReferences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code>
<div class="block">Specify the collection of sequences objects used as database in the Search run.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">ResultFactory.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/ResultFactory.html#setQueryReferences(java.util.List)">setQueryReferences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code>
<div class="block">Specify the collection of sequences objects used as queries in the Search run.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/core/search/io/package-summary.html">org.biojava.nbio.core.search.io</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/Hit.html#Hit(int,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20int,%20java.util.List,%20org.biojava.nbio.core.sequence.template.Sequence)">Hit</a></strong>(int&nbsp;hitNum,
   <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;hitId,
   <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;hitDef,
   <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;hitAccession,
   int&nbsp;hitLen,
   <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/search/io/Hsp.html" title="class in org.biojava.nbio.core.search.io">Hsp</a>&gt;&nbsp;hsps,
   <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;hitSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/Result.html#Result(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.util.HashMap,%20int,%20java.lang.String,%20java.lang.String,%20int,%20java.util.List,%20org.biojava.nbio.core.sequence.template.Sequence)">Result</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;program,
      <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;version,
      <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;reference,
      <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbFile,
      <a href="http://docs.oracle.com/javase/7/docs/api/java/util/HashMap.html?is-external=true" title="class or interface in java.util">HashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;programSpecificParameters,
      int&nbsp;iterationNumber,
      <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;queryID,
      <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;queryDef,
      int&nbsp;queryLength,
      <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/search/io/Hit.html" title="class in org.biojava.nbio.core.search.io">Hit</a>&gt;&nbsp;hits,
      <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;querySequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.search.io.blast">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastHitBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastHitBuilder</a></code></td>
<td class="colLast"><span class="strong">BlastHitBuilder.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastHitBuilder.html#setHitSequence(org.biojava.nbio.core.sequence.template.Sequence)">setHitSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;s)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastResultBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastResultBuilder</a></code></td>
<td class="colLast"><span class="strong">BlastResultBuilder.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastResultBuilder.html#setQuerySequence(org.biojava.nbio.core.sequence.template.Sequence)">setQuerySequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;s)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/core/search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BlastXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastXMLParser.html#setDatabaseReferences(java.util.List)">setDatabaseReferences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BlastTabularParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastTabularParser.html#setDatabaseReferences(java.util.List)">setDatabaseReferences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code>
<div class="block">Intended for use with run module.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BlastXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastXMLParser.html#setQueryReferences(java.util.List)">setQueryReferences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BlastTabularParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastTabularParser.html#setQueryReferences(java.util.List)">setQueryReferences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&gt;&nbsp;sequences)</code>
<div class="block">Intended for use with run module.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/core/search/io/blast/package-summary.html">org.biojava.nbio.core.search.io.blast</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/search/io/blast/BlastResult.html#BlastResult(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20java.util.HashMap,%20int,%20java.lang.String,%20java.lang.String,%20int,%20java.util.List,%20org.biojava.nbio.core.sequence.template.Sequence)">BlastResult</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;program,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;version,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;reference,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;dbFile,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/util/HashMap.html?is-external=true" title="class or interface in java.util">HashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;programSpecificParameters,
           int&nbsp;iterationNumber,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;queryID,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;queryDef,
           int&nbsp;queryLength,
           <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/search/io/Hit.html" title="class in org.biojava.nbio.core.search.io">Hit</a>&gt;&nbsp;hits,
           <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&nbsp;querySequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></strong></code>
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></strong></code>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></strong></code>
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></strong></code>
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></strong></code>
<div class="block">The representation of a ProteinSequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></strong></code>
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></strong></code>
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></strong></code>
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></strong></code>
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.compound">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/compound/package-summary.html">org.biojava.nbio.core.sequence.compound</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/compound/package-summary.html">org.biojava.nbio.core.sequence.compound</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">AminoAcidCompoundSet.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.edits">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getEmptySequence(org.biojava.nbio.core.sequence.template.Sequence)">getEmptySequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Returns an empty sequence with the given compound set of the editing
 sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Should return the 5-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Delete.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Insert.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Substitute.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava.nbio.core.sequence.template.Sequence)">getTargetSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Returns the Sequence which is our edit.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Should return the 3-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Delete.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Insert.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Substitute.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.html" title="type parameter in Edit">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#edit(org.biojava.nbio.core.sequence.template.Sequence)">edit</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getEmptySequence(org.biojava.nbio.core.sequence.template.Sequence)">getEmptySequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Returns an empty sequence with the given compound set of the editing
 sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">Edit.Substitute.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html#getEnd(org.biojava.nbio.core.sequence.template.Sequence)">getEnd</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;sequence)</code>
<div class="block">Must use this rather than the no-args getEnd as this can return
 -1 and the length of a sub is dependent on the length of the
 Sequence; we cannot assume 1:1 mapping between characters in a
 String and the number of compounds we will have to insert.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Should return the 5-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Delete.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Insert.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Substitute.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html#getFivePrime(org.biojava.nbio.core.sequence.template.Sequence)">getFivePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getTargetSequence(org.biojava.nbio.core.sequence.template.Sequence)">getTargetSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Returns the Sequence which is our edit.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected abstract <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;editingSequence)</code>
<div class="block">Should return the 3-prime end of the given Sequence according to
 the edit.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Delete.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="type parameter in Edit.Delete">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Insert.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Edit.Substitute.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html#getThreePrime(org.biojava.nbio.core.sequence.template.Sequence)">getThreePrime</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;editingSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">Edit.AbstractEdit.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html#setSequence(org.biojava.nbio.core.sequence.template.Sequence)">setSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="type parameter in Edit.AbstractEdit">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/core/sequence/edits/package-summary.html">org.biojava.nbio.core.sequence.edits</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html#Edit.Insert(org.biojava.nbio.core.sequence.template.Sequence,%20int)">Edit.Insert</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;sequence,
           int&nbsp;position)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html#Edit.Insert(org.biojava.nbio.core.sequence.template.Sequence,%20int,%20int)">Edit.Insert</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="type parameter in Edit.Insert">C</a>&gt;&nbsp;sequence,
           int&nbsp;start,
           int&nbsp;stop)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html#Edit.Substitute(org.biojava.nbio.core.sequence.template.Sequence,%20int)">Edit.Substitute</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="type parameter in Edit.Substitute">C</a>&gt;&nbsp;sequence,
               int&nbsp;position)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the
 primary class used to read Fasta files</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FastaWriter.html" title="class in org.biojava.nbio.core.sequence.io">FastaWriter</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriter.html" title="class in org.biojava.nbio.core.sequence.io">GenbankWriter</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FastaWriterHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FastaWriterHelper.html#writeSequence(java.io.File,%20org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
             <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code>
<div class="block">Write a sequence to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GenbankWriterHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeSequence(java.io.File,%20org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
             <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code>
<div class="block">Write a sequence to a file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FastaWriterHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FastaWriterHelper.html#writeSequence(java.io.OutputStream,%20org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
             <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code>
<div class="block">Write a sequence to OutputStream</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GenbankWriterHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeSequence(java.io.OutputStream,%20org.biojava.nbio.core.sequence.template.Sequence)">writeSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
             <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&nbsp;sequence)</code>
<div class="block">Write a sequence to OutputStream</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FastaWriterHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/FastaWriterHelper.html#writeSequences(java.io.OutputStream,%20java.util.Collection)">writeSequences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&gt;&nbsp;sequences)</code>
<div class="block">Method which will write your given Sequences to the specified
 <a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io"><code>OutputStream</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GenbankWriterHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeSequences(java.io.OutputStream,%20java.util.Collection)">writeSequences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;&gt;&nbsp;sequences)</code>
<div class="block">Method which will write your given Sequences to the specified
 <a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io"><code>OutputStream</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/package-summary.html">org.biojava.nbio.core.sequence.io.template</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">FastaHeaderFormatInterface</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/GenbankHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">GenbankHeaderFormatInterface</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/util/package-summary.html">org.biojava.nbio.core.sequence.io.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/io/util/package-summary.html">org.biojava.nbio.core.sequence.io.util</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>int</code></td>
<td class="colLast"><span class="strong">IOUtils.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/util/IOUtils.html#getGCGChecksum(java.util.List)">getGCGChecksum</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences)</code>
<div class="block">Calculates GCG checksum for entire list of sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>int</code></td>
<td class="colLast"><span class="strong">IOUtils.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/util/IOUtils.html#getGCGChecksum(S)">getGCGChecksum</a></strong>(S&nbsp;sequence)</code>
<div class="block">Calculates GCG checksum for a given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">IOUtils.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/util/IOUtils.html#getGCGHeader(java.util.List)">getGCGHeader</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences)</code>
<div class="block">Assembles a GCG file header</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">IOUtils.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/io/util/IOUtils.html#getIDFormat(java.util.List)">getIDFormat</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences)</code>
<div class="block">Creates format String for accession IDs</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">ArrayListProxySequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">GenbankProxySequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava.nbio.core.sequence.loader">SequenceFileProxyLoader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">StringProxySequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">UniprotProxySequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the sequence data and other data elements
 associated with the ProteinSequence by Uniprot.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.location.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">Location.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/Location.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Will return a SequenceReader object which offers a view of all resolved
 locations i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">AbstractLocation.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">Location.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/Location.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Will return a SequenceReader object which represents the outer bounds
 of this Location</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">AbstractLocation.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/AbstractLocation.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">If circular this will return the sequence represented by the sub
 locations joined.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">AbstractLocation.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/AbstractLocation.html#reverseSequence(org.biojava.nbio.core.sequence.template.Sequence)">reverseSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Reverses and (if possible) complements the Sequence so as to represent
 the reverse strand (if one exists).</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/package-summary.html">org.biojava.nbio.core.sequence.location.template</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>boolean</code></td>
<td class="colLast"><span class="strong">AbstractLocation.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/AbstractLocation.html#canComplement(org.biojava.nbio.core.sequence.template.Sequence)">canComplement</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Uses the Sequence's CompoundSet to decide if a compound can
 be assgined to ComplementCompound meaning it can complement</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">Location.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/Location.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Will return a SequenceReader object which offers a view of all resolved
 locations i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">AbstractLocation.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/AbstractLocation.html#getRelevantSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getRelevantSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">Location.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/Location.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Will return a SequenceReader object which represents the outer bounds
 of this Location</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">AbstractLocation.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/AbstractLocation.html#getSubSequence(org.biojava.nbio.core.sequence.template.Sequence)">getSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">If circular this will return the sequence represented by the sub
 locations joined.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">AbstractLocation.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/location/template/AbstractLocation.html#reverseSequence(org.biojava.nbio.core.sequence.template.Sequence)">reverseSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Reverses and (if possible) complements the Sequence so as to represent
 the reverse strand (if one exists).</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.storage">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">ArrayListSequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">An implementation of the popular bit encodings.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Four bit encoding of the bit formats.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">JoiningSequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">This reader actually proxies onto multiple types of sequence in order
 to allow a number of sequence objects to act as if they are one sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">SingleCompoundSequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">TwoBitSequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</strong></code>
<div class="block">Implementation of the 2bit encoding.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">BitSequenceReader.BitArrayWorker.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#populate(org.biojava.nbio.core.sequence.template.Sequence)">populate</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;sequence)</code>
<div class="block">Loops through the Compounds in a Sequence and passes them onto
 <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#setCompoundAt(C,%20int)"><code>BitSequenceReader.BitArrayWorker.setCompoundAt(Compound, int)</code></a></div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html#BitSequenceReader.BitArrayWorker(org.biojava.nbio.core.sequence.template.Sequence)">BitSequenceReader.BitArrayWorker</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="type parameter in BitSequenceReader.BitArrayWorker">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html#FourBitSequenceReader.FourBitArrayWorker(org.biojava.nbio.core.sequence.template.Sequence)">FourBitSequenceReader.FourBitArrayWorker</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="type parameter in FourBitSequenceReader.FourBitArrayWorker">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.html#FourBitSequenceReader(org.biojava.nbio.core.sequence.template.Sequence)">FourBitSequenceReader</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.html" title="type parameter in FourBitSequenceReader">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava.nbio.core.sequence.template.CompoundSet,%20org.biojava.nbio.core.sequence.template.Sequence...)">JoiningSequenceReader</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
                     <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;...&nbsp;sequences)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava.nbio.core.sequence.template.Sequence...)">JoiningSequenceReader</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;...&nbsp;sequences)</code>
<div class="block">Allows creation of the store from Vargs Sequence<C> objects.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html#TwoBitSequenceReader.TwoBitArrayWorker(org.biojava.nbio.core.sequence.template.Sequence)">TwoBitSequenceReader.TwoBitArrayWorker</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="type parameter in TwoBitSequenceReader.TwoBitArrayWorker">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.html#TwoBitSequenceReader(org.biojava.nbio.core.sequence.template.Sequence)">TwoBitSequenceReader</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.html" title="type parameter in TwoBitSequenceReader">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructor parameters in <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(org.biojava.nbio.core.sequence.template.CompoundSet,%20java.util.List)">JoiningSequenceReader</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&nbsp;compoundSet,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&gt;&nbsp;sequences)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html#JoiningSequenceReader(java.util.List)">JoiningSequenceReader</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;&gt;&nbsp;sequences)</code>
<div class="block">Allows creation of the store from List<Sequence<C>>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/LightweightProfile.html" title="interface in org.biojava.nbio.core.sequence.template">LightweightProfile</a>&lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subinterfaces, and an explanation">
<caption><span>Subinterfaces of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Interface and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>interface&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template">SequenceProxyView</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;</code></td>
<td class="colLast"><span class="strong">CompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;</code></td>
<td class="colLast"><span class="strong">AbstractCompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceProxyView.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html#getViewedSequence()">getViewedSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="type parameter in SequenceView">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceView.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html#getViewedSequence()">getViewedSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#shuffle(org.biojava.nbio.core.sequence.template.Sequence)">shuffle</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Implements sequence shuffling by first materializing the given
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> into a <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>, applying
 <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html?is-external=true#shuffle(java.util.List)" title="class or interface in java.util"><code>Collections.shuffle(List)</code></a> and then returning the shuffled
 elements in a new instance of <code>SequenceBackingStore</code> which behaves
 as a <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">CompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">AbstractCompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">AbstractCompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html#workingListToSequences(java.util.List)">workingListToSequences</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;&nbsp;workingList)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#checksum(org.biojava.nbio.core.sequence.template.Sequence)">checksum</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Performs a simple CRC64 checksum on any given sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#countAT(org.biojava.nbio.core.sequence.template.Sequence)">countAT</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sequence)</code>
<div class="block">Returns the count of AT in the given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>int</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#countCompounds(org.biojava.nbio.core.sequence.template.Sequence,%20C...)">countCompounds</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
              C...&nbsp;compounds)</code>
<div class="block">For the given vargs of compounds this method counts the number of
 times those compounds appear in the given sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#countGC(org.biojava.nbio.core.sequence.template.Sequence)">countGC</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sequence)</code>
<div class="block">Returns the count of GC in the given sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#createIterator(org.biojava.nbio.core.sequence.template.Sequence)">createIterator</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Creates a simple sequence iterator which moves through a sequence going
 from 1 to the length of the Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;</code></td>
<td class="colLast"><span class="strong">CompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;</code></td>
<td class="colLast"><span class="strong">AbstractCompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">T</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">CompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="type parameter in CompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">T</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">AbstractCompoundTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="type parameter in AbstractCompoundTranslator">F</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#createSubSequence(org.biojava.nbio.core.sequence.template.Sequence,%20int,%20int)">createSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                 int&nbsp;start,
                 int&nbsp;end)</code>
<div class="block">Creates a simple sub sequence view delimited by the given start and end.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;C,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#getComposition(org.biojava.nbio.core.sequence.template.Sequence)">getComposition</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Does a linear scan over the given Sequence and records the number of
 times each base appears.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;C,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#getDistribution(org.biojava.nbio.core.sequence.template.Sequence)">getDistribution</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Analogous to <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#getComposition(org.biojava.nbio.core.sequence.template.Sequence)"><code>SequenceMixin.getComposition(Sequence)</code></a> but returns the
 distribution of that <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> over the given sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>int</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#indexOf(org.biojava.nbio.core.sequence.template.Sequence,%20C)">indexOf</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
       C&nbsp;compound)</code>
<div class="block">Performs a linear search of the given Sequence for the given compound.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#inverse(org.biojava.nbio.core.sequence.template.Sequence)">inverse</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">A method which attempts to do the right thing when is comes to a
 reverse/reverse complement</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">AbstractCompoundSet.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractCompoundSet.html" title="type parameter in AbstractCompoundSet">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><span class="strong">CompoundSet.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html#isValidSequence(org.biojava.nbio.core.sequence.template.Sequence)">isValidSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="type parameter in CompoundSet">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>int</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#lastIndexOf(org.biojava.nbio.core.sequence.template.Sequence,%20C)">lastIndexOf</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
           C&nbsp;compound)</code>
<div class="block">Performs a reversed linear search of the given Sequence by wrapping
 it in a <a href="../../../../../../../org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views"><code>ReversedSequenceView</code></a> and passing it into
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#indexOf(org.biojava.nbio.core.sequence.template.Sequence,%20C)"><code>SequenceMixin.indexOf(Sequence, Compound)</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#nonOverlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,%20int)">nonOverlappingKmers</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                   int&nbsp;kmer)</code>
<div class="block">Produces kmers of the specified size e.g.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#overlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,%20int)">overlappingKmers</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                int&nbsp;kmer)</code>
<div class="block">Used to generate overlapping k-mers such i.e.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>boolean</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.template.Sequence)">sequenceEquality</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code>
<div class="block">A case-sensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>boolean</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#sequenceEquality(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.template.Sequence)">sequenceEquality</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code>
<div class="block">A case-sensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>boolean</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                          <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code>
<div class="block">A case-insensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>boolean</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#sequenceEqualityIgnoreCase(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.template.Sequence)">sequenceEqualityIgnoreCase</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;source,
                          <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;target)</code>
<div class="block">A case-insensitive manner of comparing two sequence objects together.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#shuffle(org.biojava.nbio.core.sequence.template.Sequence)">shuffle</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Implements sequence shuffling by first materializing the given
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> into a <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>, applying
 <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html?is-external=true#shuffle(java.util.List)" title="class or interface in java.util"><code>Collections.shuffle(List)</code></a> and then returning the shuffled
 elements in a new instance of <code>SequenceBackingStore</code> which behaves
 as a <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#toList(org.biojava.nbio.core.sequence.template.Sequence)">toList</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">For the given <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> this will return a <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a> filled with
 the Compounds of that <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#toString(org.biojava.nbio.core.sequence.template.Sequence)">toString</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Shortcut to <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)"><code>SequenceMixin.toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)</code></a>
 which calls toString() on the resulting object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html?is-external=true" title="class or interface in java.lang">StringBuilder</a></code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#toStringBuilder(org.biojava.nbio.core.sequence.template.Sequence)">toStringBuilder</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">For the given Sequence this will return a <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html?is-external=true" title="class or interface in java.lang"><code>StringBuilder</code></a> object
 filled with the results of <code>Compound#toString()</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>void</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#write(java.lang.Appendable,%20org.biojava.nbio.core.sequence.template.Sequence)">write</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Appendable.html?is-external=true" title="class or interface in java.lang">Appendable</a>&nbsp;appendable,
     <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">Used as a way of sending a Sequence to a writer without the cost of
 converting to a full length String and then writing the data out</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.SequenceIterator.html#SequenceMixin.SequenceIterator(org.biojava.nbio.core.sequence.template.Sequence)">SequenceMixin.SequenceIterator</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.SequenceIterator.html" title="type parameter in SequenceMixin.SequenceIterator">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html#SequenceProxyView(org.biojava.nbio.core.sequence.template.Sequence)">SequenceProxyView</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html#SequenceProxyView(org.biojava.nbio.core.sequence.template.Sequence,%20java.lang.Integer,%20java.lang.Integer)">SequenceProxyView</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;&nbsp;sequence,
                 <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioStart,
                 <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>
<div class="block">Main constructor for working with SequenceProxyViews</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.transcription">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DNAToRNATranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DNAToRNATranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.transcription.Frame)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence,
              <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>&nbsp;frame)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">TranscriptionEngine.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html#translate(org.biojava.nbio.core.sequence.template.Sequence)">translate</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code>
<div class="block">Quick method to let you go from a CDS to a Peptide quickly.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">Frame.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html#wrap(org.biojava.nbio.core.sequence.template.Sequence)">wrap</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;incoming)</code>
<div class="block">Optionally wraps a Sequence in a reverse complementing view (if the
 frame is on the reverse strand) and creates a sub sequence view if
 it is required.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">DNAToRNATranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code>
<div class="block">Overloaded local version which delegates to an optional translator
 when told to (specified during construction).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">RNAToAminoAcidTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code>
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">TranscriptionEngine.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.transcription.Frame...)">multipleFrameTranslation</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
                        <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>...&nbsp;frames)</code>
<div class="block">A way of translating DNA in a number of frames</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/package-summary.html">org.biojava.nbio.core.sequence.transcription</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DNAToRNATranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DNAToRNATranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html#createSequence(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.transcription.Frame)">createSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence,
              <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>&nbsp;frame)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">DNAToRNATranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code>
<div class="block">Overloaded local version which delegates to an optional translator
 when told to (specified during construction).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">RNAToAminoAcidTranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/RNAToAminoAcidTranslator.html#createSequences(org.biojava.nbio.core.sequence.template.Sequence)">createSequences</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;originalSequence)</code>
<div class="block">Performs the core conversion of RNA to Peptide.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">TranscriptionEngine.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html#multipleFrameTranslation(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.transcription.Frame...)">multipleFrameTranslation</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna,
                        <a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a>...&nbsp;frames)</code>
<div class="block">A way of translating DNA in a number of frames</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">TranscriptionEngine.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html#translate(org.biojava.nbio.core.sequence.template.Sequence)">translate</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code>
<div class="block">Quick method to let you go from a CDS to a Peptide quickly.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">Frame.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/Frame.html#wrap(org.biojava.nbio.core.sequence.template.Sequence)">wrap</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;incoming)</code>
<div class="block">Optionally wraps a Sequence in a reverse complementing view (if the
 frame is on the reverse strand) and creates a sub sequence view if
 it is required.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></code></td>
<td class="colLast"><span class="strong">DNAToRNATranslator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html#wrapToRna(org.biojava.nbio.core.sequence.template.Sequence)">wrapToRna</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;dna)</code>
<div class="block">Takes in the given DNA Sequence and returns an instance of RNASequence
 which is using <a href="../../../../../../../org/biojava/nbio/core/sequence/views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views"><code>RnaSequenceView</code></a> as a
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template"><code>ProxySequenceReader</code></a>.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.views">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ComplementSequenceView</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/ComplementCompound.html" title="interface in org.biojava.nbio.core.sequence.template">ComplementCompound</a>&gt;</strong></code>
<div class="block">For a given sequence this class will create a view over the top of it
 and for every request the code will return the complement of the underlying
 base e.g.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ReversedSequenceView</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">For a given sequence this class will return the base at the reversed
 position i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">RnaSequenceView</a></strong></code>
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds
 until asked.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">WindowedSequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html#getBackingSequence()">getBackingSequence</a></strong>()</code>
<div class="block">Access the sequence which backs this window</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/ComplementSequenceView.html#ComplementSequenceView(org.biojava.nbio.core.sequence.template.Sequence)">ComplementSequenceView</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/views/ComplementSequenceView.html" title="type parameter in ComplementSequenceView">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/ReversedSequenceView.html#ReversedSequenceView(org.biojava.nbio.core.sequence.template.Sequence)">ReversedSequenceView</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="type parameter in ReversedSequenceView">C</a>&gt;&nbsp;sequence)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/RnaSequenceView.html#RnaSequenceView(org.biojava.nbio.core.sequence.template.Sequence)">RnaSequenceView</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sourceDna)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/RnaSequenceView.html#RnaSequenceView(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.core.sequence.template.CompoundSet)">RnaSequenceView</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;sourceDna,
               <a href="../../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;&nbsp;rnaCompounds)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html#WindowedSequence(org.biojava.nbio.core.sequence.template.Sequence,%20int)">WindowedSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a>&gt;&nbsp;sequence,
                int&nbsp;windowSize)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/core/util/package-summary.html">org.biojava.nbio.core.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/util/package-summary.html">org.biojava.nbio.core.util</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<td class="colLast"><span class="strong">SequenceTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/util/SequenceTools.html#getSequenceFromString(java.lang.String)">getSequenceFromString</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.phylo">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/phylo/package-summary.html">org.biojava.nbio.phylo</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/phylo/package-summary.html">org.biojava.nbio.phylo</a> with type parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.msa.Msa</code></td>
<td class="colLast"><span class="strong">ForesterWrapper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/ForesterWrapper.html#convert(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">convert</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa)</code>
<div class="block">Convert a BioJava <a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence"><code>MultipleSequenceAlignment</code></a> to a forester
 <code>Msa</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#dissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">dissimilarityScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa,
                  <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code>
<div class="block">The dissimilarity score is the additive inverse of the similarity score
 (sum of scores) between two aligned sequences using a substitution model
 (Substitution Matrix).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#fractionalDissimilarity(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">fractionalDissimilarity</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa)</code>
<div class="block">The fractional dissimilarity (D) is defined as the percentage of sites
 that differ between two aligned sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#fractionalDissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">fractionalDissimilarityScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa,
                            <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code>
<div class="block">The fractional dissimilarity score (Ds) is a relative measure of the
 dissimilarity between two aligned sequences.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#jointSeqStrucDistance(double[][])">jointSeqStrucDistance</a></strong>(double[][]&nbsp;rmsdMat)</code>
<div class="block">The joint sequence-structure distance (d<sub>SS</sub>) is a combination
 of the sequence-based and the structure-based distances.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#kimuraDistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">kimuraDistance</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa)</code>
<div class="block">The Kimura evolutionary distance (d) is a correction of the fractional
 dissimilarity (D) specially needed for large evolutionary distances.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#pamMLdistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">pamMLdistance</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa)</code>
<div class="block">The PAM (Point Accepted Mutations) distance is a measure of evolutionary
 distance in protein sequences.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#percentageIdentity(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">percentageIdentity</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa)</code>
<div class="block">BioJava implementation for percentage of identity (PID).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#poissonDistance(org.biojava.nbio.core.sequence.MultipleSequenceAlignment)">poissonDistance</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa)</code>
<div class="block">The Poisson (correction) evolutionary distance (d) is a function of the
 fractional dissimilarity (D), given by:</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#structuralDistance(double[][],%20double,%20double,%20double)">structuralDistance</a></strong>(double[][]&nbsp;rmsdMat,
                  double&nbsp;alpha,
                  double&nbsp;rmsdMax,
                  double&nbsp;rmsd0)</code>
<div class="block">The structural distance (d<sub>S</sub>) uses the structural similarity
 (or dissimilarity) from a the structural alignment of two protein
 strutures.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/package-summary.html">org.biojava.nbio.structure</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<td class="colLast"><span class="strong">ChainImpl.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/ChainImpl.html#getBJSequence()">getBJSequence</a></strong>()</code>
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</code></td>
<td class="colLast"><span class="strong">Chain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/Chain.html#getBJSequence()">getBJSequence</a></strong>()</code>
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.alignment.template.SequencePair,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;alignment,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Provided only for convenience.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.ws.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/ws/alignment/package-summary.html">org.biojava.nbio.ws.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/ws/alignment/package-summary.html">org.biojava.nbio.ws.alignment</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">RemotePairwiseAlignmentService.</span><code><strong><a href="../../../../../../../org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentService.html#sendAlignmentRequest(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;seq,
                    <a href="../../../../../../../org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code>
<div class="block">Doing the actual analysis on the instantiated service using specified parameters and the RichSequence object</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.ws.alignment.qblast">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a> in <a href="../../../../../../../org/biojava/nbio/ws/alignment/qblast/package-summary.html">org.biojava.nbio.ws.alignment.qblast</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/ws/alignment/qblast/package-summary.html">org.biojava.nbio.ws.alignment.qblast</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">NCBIQBlastService.</span><code><strong><a href="../../../../../../../org/biojava/nbio/ws/alignment/qblast/NCBIQBlastService.html#sendAlignmentRequest(org.biojava.nbio.core.sequence.template.Sequence,%20org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)">sendAlignmentRequest</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;seq,
                    <a href="../../../../../../../org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a>&nbsp;rpa)</code>
<div class="block">Converts given sequence to String and calls
 <a href="../../../../../../../org/biojava/nbio/ws/alignment/qblast/NCBIQBlastService.html#sendAlignmentRequest(java.lang.String,%20org.biojava.nbio.ws.alignment.RemotePairwiseAlignmentProperties)"><code>NCBIQBlastService.sendAlignmentRequest(String, RemotePairwiseAlignmentProperties)</code></a></div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/sequence/template/class-use/Sequence.html" target="_top">Frames</a></li>
<li><a href="Sequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
