<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:50 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.nbio.core.sequence.template.SequenceView (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava.nbio.core.sequence.template.SequenceView (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/sequence/template/class-use/SequenceView.html" target="_top">Frames</a></li>
<li><a href="SequenceView.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.core.sequence.template.SequenceView" class="title">Uses of Interface<br>org.biojava.nbio.core.sequence.template.SequenceView</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.alignment">org.biojava.nbio.core.alignment</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.storage">org.biojava.nbio.core.sequence.storage</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.views">org.biojava.nbio.core.sequence.views</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.core.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a> in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="type parameter in SimpleAlignedSequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SimpleAlignedSequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleAlignedSequence.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="type parameter in SimpleAlignedSequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SimpleAlignedSequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleAlignedSequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;start,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">RNASequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getComplement()">getComplement</a></strong>()</code>
<div class="block">Get the complement view of the RNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DNASequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getComplement()">getComplement</a></strong>()</code>
<div class="block">Returns a Sequence which will complement every base</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">RNASequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getInverse()">getInverse</a></strong>()</code>
<div class="block">Get the inverse view of the sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DNASequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverse()">getReverse</a></strong>()</code>
<div class="block">Returns a Sequence which runs in the current reverse order</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">RNASequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/RNASequence.html#getReverseComplement()">getReverseComplement</a></strong>()</code>
<div class="block">Get reverse complement view of the sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DNASequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/DNASequence.html#getReverseComplement()">getReverseComplement</a></strong>()</code>
<div class="block">Delegates to <a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getInverse()"><code>AbstractSequence.getInverse()</code></a> for the reverse complement</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceFileProxyLoader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">StringProxySequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">UniprotProxySequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html" title="type parameter in SequenceFileProxyLoader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceFileProxyLoader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioBegin,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html" title="type parameter in StringProxySequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">StringProxySequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioBegin,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html" title="type parameter in UniprotProxySequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">UniprotProxySequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioBegin,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.storage">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/storage/package-summary.html">org.biojava.nbio.core.sequence.storage</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">JoiningSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">BitSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">ArrayListSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SingleCompoundSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="type parameter in JoiningSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">JoiningSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;start,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">BitSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;start,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html" title="type parameter in ArrayListSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">ArrayListSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioBegin,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html" title="type parameter in SingleCompoundSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SingleCompoundSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;start,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>
<div class="block">Creates a <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template"><code>SequenceProxyView</code></a> for the given coordinates</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="type parameter in BitSequenceReader">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">BitSequenceReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html#getSubSequence(int,%20int)">getSubSequence</a></strong>(int&nbsp;start,
              int&nbsp;end)</code>
<div class="block">Returns a sub sequence view</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template">SequenceProxyView</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#createSubSequence(org.biojava.nbio.core.sequence.template.Sequence,%20int,%20int)">createSubSequence</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                 int&nbsp;start,
                 int&nbsp;end)</code>
<div class="block">Creates a simple sub sequence view delimited by the given start and end.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceProxyView.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">AbstractSequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Sequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getInverse()">getInverse</a></strong>()</code>
<div class="block">Does the <em>right thing</em> to get the inverse of the current
 Sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="type parameter in SequenceProxyView">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SequenceProxyView.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioStart,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">AbstractSequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioStart,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">Sequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;start,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>
<div class="block">Returns a portion of the sequence from the different positions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#inverse(org.biojava.nbio.core.sequence.template.Sequence)">inverse</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence)</code>
<div class="block">A method which attempts to do the right thing when is comes to a
 reverse/reverse complement</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#nonOverlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,%20int)">nonOverlappingKmers</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                   int&nbsp;kmer)</code>
<div class="block">Produces kmers of the specified size e.g.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">SequenceMixin.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html#overlappingKmers(org.biojava.nbio.core.sequence.template.Sequence,%20int)">overlappingKmers</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;&nbsp;sequence,
                int&nbsp;kmer)</code>
<div class="block">Used to generate overlapping k-mers such i.e.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.views">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a> in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that implement <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ComplementSequenceView</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/ComplementCompound.html" title="interface in org.biojava.nbio.core.sequence.template">ComplementCompound</a>&gt;</strong></code>
<div class="block">For a given sequence this class will create a view over the top of it
 and for every request the code will return the complement of the underlying
 base e.g.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ReversedSequenceView</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">For a given sequence this class will return the base at the reversed
 position i.e.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">RnaSequenceView</a></strong></code>
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds
 until asked.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that return <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">WindowedSequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html#get(int)">get</a></strong>(int&nbsp;index)</code>
<div class="block">Returns the window specified at the given index in offsets i.e.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/sequence/views/package-summary.html">org.biojava.nbio.core.sequence.views</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html" title="type parameter in WindowedSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">WindowedSequence.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/sequence/views/WindowedSequence.html#iterator()">iterator</a></strong>()</code>
<div class="block">Returns an iterator which will return the windows in a sequence in
 sequential order.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/sequence/template/class-use/SequenceView.html" target="_top">Frames</a></li>
<li><a href="SequenceView.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
