<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.structure.align.multiple.util, class: MultipleAlignmentTools">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> */</span>
<span class="source-line-no">021</span><span id="line-21">package org.biojava.nbio.structure.align.multiple.util;</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">import java.io.IOException;</span>
<span class="source-line-no">024</span><span id="line-24">import java.util.ArrayList;</span>
<span class="source-line-no">025</span><span id="line-25">import java.util.Arrays;</span>
<span class="source-line-no">026</span><span id="line-26">import java.util.Collections;</span>
<span class="source-line-no">027</span><span id="line-27">import java.util.Comparator;</span>
<span class="source-line-no">028</span><span id="line-28">import java.util.HashMap;</span>
<span class="source-line-no">029</span><span id="line-29">import java.util.List;</span>
<span class="source-line-no">030</span><span id="line-30">import java.util.Map;</span>
<span class="source-line-no">031</span><span id="line-31">import java.util.SortedSet;</span>
<span class="source-line-no">032</span><span id="line-32">import java.util.TreeSet;</span>
<span class="source-line-no">033</span><span id="line-33"></span>
<span class="source-line-no">034</span><span id="line-34">import javax.vecmath.Matrix4d;</span>
<span class="source-line-no">035</span><span id="line-35"></span>
<span class="source-line-no">036</span><span id="line-36">import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;</span>
<span class="source-line-no">037</span><span id="line-37">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">038</span><span id="line-38">import org.biojava.nbio.core.sequence.AccessionID;</span>
<span class="source-line-no">039</span><span id="line-39">import org.biojava.nbio.core.sequence.MultipleSequenceAlignment;</span>
<span class="source-line-no">040</span><span id="line-40">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">041</span><span id="line-41">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">042</span><span id="line-42">import org.biojava.nbio.phylo.DistanceMatrixCalculator;</span>
<span class="source-line-no">043</span><span id="line-43">import org.biojava.nbio.phylo.TreeConstructor;</span>
<span class="source-line-no">044</span><span id="line-44">import org.biojava.nbio.phylo.TreeConstructorType;</span>
<span class="source-line-no">045</span><span id="line-45">import org.biojava.nbio.structure.AminoAcid;</span>
<span class="source-line-no">046</span><span id="line-46">import org.biojava.nbio.structure.Atom;</span>
<span class="source-line-no">047</span><span id="line-47">import org.biojava.nbio.structure.Calc;</span>
<span class="source-line-no">048</span><span id="line-48">import org.biojava.nbio.structure.Chain;</span>
<span class="source-line-no">049</span><span id="line-49">import org.biojava.nbio.structure.Group;</span>
<span class="source-line-no">050</span><span id="line-50">import org.biojava.nbio.structure.PDBHeader;</span>
<span class="source-line-no">051</span><span id="line-51">import org.biojava.nbio.structure.Structure;</span>
<span class="source-line-no">052</span><span id="line-52">import org.biojava.nbio.structure.StructureException;</span>
<span class="source-line-no">053</span><span id="line-53">import org.biojava.nbio.structure.StructureIdentifier;</span>
<span class="source-line-no">054</span><span id="line-54">import org.biojava.nbio.structure.StructureImpl;</span>
<span class="source-line-no">055</span><span id="line-55">import org.biojava.nbio.structure.StructureTools;</span>
<span class="source-line-no">056</span><span id="line-56">import org.biojava.nbio.structure.align.multiple.Block;</span>
<span class="source-line-no">057</span><span id="line-57">import org.biojava.nbio.structure.align.multiple.BlockSet;</span>
<span class="source-line-no">058</span><span id="line-58">import org.biojava.nbio.structure.align.multiple.MultipleAlignment;</span>
<span class="source-line-no">059</span><span id="line-59">import org.biojava.nbio.structure.align.util.AlignmentTools;</span>
<span class="source-line-no">060</span><span id="line-60">import org.biojava.nbio.structure.cluster.SubunitCluster;</span>
<span class="source-line-no">061</span><span id="line-61">import org.biojava.nbio.structure.jama.Matrix;</span>
<span class="source-line-no">062</span><span id="line-62">import org.forester.evoinference.matrix.distance.BasicSymmetricalDistanceMatrix;</span>
<span class="source-line-no">063</span><span id="line-63">import org.forester.phylogeny.Phylogeny;</span>
<span class="source-line-no">064</span><span id="line-64"></span>
<span class="source-line-no">065</span><span id="line-65">/**</span>
<span class="source-line-no">066</span><span id="line-66"> * Utility functions for working with {@link MultipleAlignment}.</span>
<span class="source-line-no">067</span><span id="line-67"> * &lt;p&gt;</span>
<span class="source-line-no">068</span><span id="line-68"> * Supported functions:</span>
<span class="source-line-no">069</span><span id="line-69"> * &lt;ul&gt;</span>
<span class="source-line-no">070</span><span id="line-70"> * &lt;li&gt;Obtain the alignment as sequence strings</span>
<span class="source-line-no">071</span><span id="line-71"> * &lt;li&gt;Map from sequence alignment position to structure Atom</span>
<span class="source-line-no">072</span><span id="line-72"> * &lt;li&gt;Map from sequence alignment position to Block number</span>
<span class="source-line-no">073</span><span id="line-73"> * &lt;li&gt;Transform the aligned Atoms of a MultipleAlignment</span>
<span class="source-line-no">074</span><span id="line-74"> * &lt;li&gt;Get all the core alignment positions of the alignment</span>
<span class="source-line-no">075</span><span id="line-75"> * &lt;li&gt;Calculate the average residue distance of all aligned positions</span>
<span class="source-line-no">076</span><span id="line-76"> * &lt;li&gt;Sort Blocks in a MultipleAlignment by a specified row</span>
<span class="source-line-no">077</span><span id="line-77"> * &lt;li&gt;Convert a MultipleAlignment to a MultipleSequenceAlignment</span>
<span class="source-line-no">078</span><span id="line-78"> * &lt;/ul&gt;</span>
<span class="source-line-no">079</span><span id="line-79"> *</span>
<span class="source-line-no">080</span><span id="line-80"> * @author Spencer Bliven</span>
<span class="source-line-no">081</span><span id="line-81"> * @author Aleix Lafita</span>
<span class="source-line-no">082</span><span id="line-82"> * @since 4.1.0</span>
<span class="source-line-no">083</span><span id="line-83"> *</span>
<span class="source-line-no">084</span><span id="line-84"> */</span>
<span class="source-line-no">085</span><span id="line-85">public class MultipleAlignmentTools {</span>
<span class="source-line-no">086</span><span id="line-86"></span>
<span class="source-line-no">087</span><span id="line-87">        /**</span>
<span class="source-line-no">088</span><span id="line-88">         * Calculate the sequence alignment Strings for the whole alignment. This</span>
<span class="source-line-no">089</span><span id="line-89">         * method creates a sequence alignment where aligned residues are in</span>
<span class="source-line-no">090</span><span id="line-90">         * uppercase and unaligned residues are in lowercase, thus providing a more</span>
<span class="source-line-no">091</span><span id="line-91">         * compact way to represent the alignment.</span>
<span class="source-line-no">092</span><span id="line-92">         * &lt;p&gt;</span>
<span class="source-line-no">093</span><span id="line-93">         * Blocks are concatenated in the order returned by</span>
<span class="source-line-no">094</span><span id="line-94">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</span>
<span class="source-line-no">095</span><span id="line-95">         * sequential. Gaps are represented by '-'. Separation between different</span>
<span class="source-line-no">096</span><span id="line-96">         * Blocks is indicated by a gap in all positions, meaning that there is a</span>
<span class="source-line-no">097</span><span id="line-97">         * possible discontinuity.</span>
<span class="source-line-no">098</span><span id="line-98">         *</span>
<span class="source-line-no">099</span><span id="line-99">         * @param alignment</span>
<span class="source-line-no">100</span><span id="line-100">         *            input MultipleAlignment</span>
<span class="source-line-no">101</span><span id="line-101">         * @param mapSeqToStruct</span>
<span class="source-line-no">102</span><span id="line-102">         *            provides a link from the sequence alignment position to the</span>
<span class="source-line-no">103</span><span id="line-103">         *            structure alignment position. Specially designed for the GUI.</span>
<span class="source-line-no">104</span><span id="line-104">         *            Has to be initialized previously and will be overwritten.</span>
<span class="source-line-no">105</span><span id="line-105">         * @return a string for each row in the alignment, giving the 1-letter code</span>
<span class="source-line-no">106</span><span id="line-106">         *         for each aligned residue.</span>
<span class="source-line-no">107</span><span id="line-107">         */</span>
<span class="source-line-no">108</span><span id="line-108">        public static List&lt;String&gt; getSequenceAlignment(</span>
<span class="source-line-no">109</span><span id="line-109">                        MultipleAlignment alignment, final List&lt;Integer&gt; mapSeqToStruct) {</span>
<span class="source-line-no">110</span><span id="line-110"></span>
<span class="source-line-no">111</span><span id="line-111">                // Initialize sequence variables</span>
<span class="source-line-no">112</span><span id="line-112">                List&lt;String&gt; alnSequences = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">113</span><span id="line-113">                for (int str = 0; str &lt; alignment.size(); str++)</span>
<span class="source-line-no">114</span><span id="line-114">                        alnSequences.add("");</span>
<span class="source-line-no">115</span><span id="line-115">                mapSeqToStruct.clear();</span>
<span class="source-line-no">116</span><span id="line-116">                List&lt;Atom[]&gt; atoms = alignment.getAtomArrays();</span>
<span class="source-line-no">117</span><span id="line-117">                int globalPos = -1;</span>
<span class="source-line-no">118</span><span id="line-118"></span>
<span class="source-line-no">119</span><span id="line-119">                // Initialize helper variables in constucting the sequence alignment</span>
<span class="source-line-no">120</span><span id="line-120">                List&lt;SortedSet&lt;Integer&gt;&gt; freePool = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">121</span><span id="line-121">                List&lt;SortedSet&lt;Integer&gt;&gt; blockStarts = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">122</span><span id="line-122">                List&lt;List&lt;Integer&gt;&gt; aligned = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">123</span><span id="line-123"></span>
<span class="source-line-no">124</span><span id="line-124">                // Generate freePool residues from the ones not aligned</span>
<span class="source-line-no">125</span><span id="line-125">                for (int i = 0; i &lt; alignment.size(); i++) {</span>
<span class="source-line-no">126</span><span id="line-126">                        List&lt;Integer&gt; residues = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">127</span><span id="line-127">                        freePool.add(new TreeSet&lt;Integer&gt;());</span>
<span class="source-line-no">128</span><span id="line-128">                        blockStarts.add(new TreeSet&lt;Integer&gt;());</span>
<span class="source-line-no">129</span><span id="line-129">                        for (BlockSet bs : alignment.getBlockSets()) {</span>
<span class="source-line-no">130</span><span id="line-130">                                for (Block b : bs.getBlocks()) {</span>
<span class="source-line-no">131</span><span id="line-131">                                        boolean first = true;</span>
<span class="source-line-no">132</span><span id="line-132">                                        for (int l = 0; l &lt; b.length(); l++) {</span>
<span class="source-line-no">133</span><span id="line-133">                                                Integer residue = b.getAlignRes().get(i).get(l);</span>
<span class="source-line-no">134</span><span id="line-134">                                                if (residue != null) {</span>
<span class="source-line-no">135</span><span id="line-135">                                                        if (first)</span>
<span class="source-line-no">136</span><span id="line-136">                                                                blockStarts.get(i).add(residue);</span>
<span class="source-line-no">137</span><span id="line-137">                                                        residues.add(residue);</span>
<span class="source-line-no">138</span><span id="line-138">                                                        first = false;</span>
<span class="source-line-no">139</span><span id="line-139">                                                }</span>
<span class="source-line-no">140</span><span id="line-140">                                        }</span>
<span class="source-line-no">141</span><span id="line-141">                                }</span>
<span class="source-line-no">142</span><span id="line-142">                        }</span>
<span class="source-line-no">143</span><span id="line-143">                        aligned.add(residues);</span>
<span class="source-line-no">144</span><span id="line-144">                }</span>
<span class="source-line-no">145</span><span id="line-145">                // Add any residue not aligned to the free pool for every structure</span>
<span class="source-line-no">146</span><span id="line-146">                for (int i = 0; i &lt; alignment.size(); i++) {</span>
<span class="source-line-no">147</span><span id="line-147">                        for (int k = 0; k &lt; atoms.get(i).length; k++) {</span>
<span class="source-line-no">148</span><span id="line-148">                                if (!aligned.get(i).contains(k))</span>
<span class="source-line-no">149</span><span id="line-149">                                        freePool.get(i).add(k);</span>
<span class="source-line-no">150</span><span id="line-150">                        }</span>
<span class="source-line-no">151</span><span id="line-151">                }</span>
<span class="source-line-no">152</span><span id="line-152"></span>
<span class="source-line-no">153</span><span id="line-153">                for (int b = 0; b &lt; alignment.getBlocks().size(); b++) {</span>
<span class="source-line-no">154</span><span id="line-154">                        if (b != 0) {</span>
<span class="source-line-no">155</span><span id="line-155">                                // Add a gap to all structures to separate visually the Blocks</span>
<span class="source-line-no">156</span><span id="line-156">                                for (int str = 0; str &lt; alignment.size(); str++)</span>
<span class="source-line-no">157</span><span id="line-157">                                        alnSequences.set(str, alnSequences.get(str).concat("-"));</span>
<span class="source-line-no">158</span><span id="line-158">                                mapSeqToStruct.add(-1); // means unaligned position</span>
<span class="source-line-no">159</span><span id="line-159">                        }</span>
<span class="source-line-no">160</span><span id="line-160">                        // Store the previous position added to the sequence alignment</span>
<span class="source-line-no">161</span><span id="line-161">                        int[] previousPos = new int[alignment.size()];</span>
<span class="source-line-no">162</span><span id="line-162">                        Arrays.fill(previousPos, -1);</span>
<span class="source-line-no">163</span><span id="line-163">                        // Store provisional characters</span>
<span class="source-line-no">164</span><span id="line-164">                        char[] provisionalChar = new char[alignment.size()];</span>
<span class="source-line-no">165</span><span id="line-165">                        Arrays.fill(provisionalChar, '-');</span>
<span class="source-line-no">166</span><span id="line-166"></span>
<span class="source-line-no">167</span><span id="line-167">                        for (int pos = 0; pos &lt; alignment.getBlocks().get(b).length(); pos++) {</span>
<span class="source-line-no">168</span><span id="line-168">                                globalPos++;</span>
<span class="source-line-no">169</span><span id="line-169">                                boolean gaps = true; // true if consecutive with the previous</span>
<span class="source-line-no">170</span><span id="line-170">                                while (gaps) {</span>
<span class="source-line-no">171</span><span id="line-171">                                        gaps = false;</span>
<span class="source-line-no">172</span><span id="line-172">                                        // Loop through all the structures</span>
<span class="source-line-no">173</span><span id="line-173">                                        for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">174</span><span id="line-174">                                                // If it is the first position or before it was null</span>
<span class="source-line-no">175</span><span id="line-175">                                                if (previousPos[str] == -1) {</span>
<span class="source-line-no">176</span><span id="line-176">                                                        Integer residue = alignment.getBlocks().get(b)</span>
<span class="source-line-no">177</span><span id="line-177">                                                                        .getAlignRes().get(str).get(pos);</span>
<span class="source-line-no">178</span><span id="line-178">                                                        if (residue == null)</span>
<span class="source-line-no">179</span><span id="line-179">                                                                provisionalChar[str] = '-';</span>
<span class="source-line-no">180</span><span id="line-180">                                                        else {</span>
<span class="source-line-no">181</span><span id="line-181">                                                                Atom a = atoms.get(str)[residue];</span>
<span class="source-line-no">182</span><span id="line-182">                                                                String group = a.getGroup().getPDBName();</span>
<span class="source-line-no">183</span><span id="line-183">                                                                provisionalChar[str] = StructureTools</span>
<span class="source-line-no">184</span><span id="line-184">                                                                                .get1LetterCode(group);</span>
<span class="source-line-no">185</span><span id="line-185">                                                        }</span>
<span class="source-line-no">186</span><span id="line-186">                                                } else {</span>
<span class="source-line-no">187</span><span id="line-187">                                                        Integer residue = alignment.getBlocks().get(b)</span>
<span class="source-line-no">188</span><span id="line-188">                                                                        .getAlignRes().get(str).get(pos);</span>
<span class="source-line-no">189</span><span id="line-189">                                                        int nextPos = previousPos[str] + 1;</span>
<span class="source-line-no">190</span><span id="line-190">                                                        if (residue == null) {</span>
<span class="source-line-no">191</span><span id="line-191">                                                                if (freePool.get(str).contains(nextPos)) {</span>
<span class="source-line-no">192</span><span id="line-192">                                                                        Atom a = atoms.get(str)[nextPos];</span>
<span class="source-line-no">193</span><span id="line-193">                                                                        String g = a.getGroup().getPDBName();</span>
<span class="source-line-no">194</span><span id="line-194">                                                                        char aa = StructureTools.get1LetterCode(g);</span>
<span class="source-line-no">195</span><span id="line-195">                                                                        provisionalChar[str] = Character</span>
<span class="source-line-no">196</span><span id="line-196">                                                                                        .toLowerCase(aa);</span>
<span class="source-line-no">197</span><span id="line-197">                                                                } else</span>
<span class="source-line-no">198</span><span id="line-198">                                                                        provisionalChar[str] = '-';</span>
<span class="source-line-no">199</span><span id="line-199">                                                        } else if (nextPos == residue) {</span>
<span class="source-line-no">200</span><span id="line-200">                                                                Atom a = atoms.get(str)[nextPos];</span>
<span class="source-line-no">201</span><span id="line-201">                                                                String group = a.getGroup().getPDBName();</span>
<span class="source-line-no">202</span><span id="line-202">                                                                provisionalChar[str] = StructureTools</span>
<span class="source-line-no">203</span><span id="line-203">                                                                                .get1LetterCode(group);</span>
<span class="source-line-no">204</span><span id="line-204">                                                        } else {</span>
<span class="source-line-no">205</span><span id="line-205">                                                                // This means non-consecutive</span>
<span class="source-line-no">206</span><span id="line-206">                                                                provisionalChar[str] = ' ';</span>
<span class="source-line-no">207</span><span id="line-207">                                                                gaps = true;</span>
<span class="source-line-no">208</span><span id="line-208">                                                        }</span>
<span class="source-line-no">209</span><span id="line-209">                                                }</span>
<span class="source-line-no">210</span><span id="line-210">                                        }// End all structure analysis</span>
<span class="source-line-no">211</span><span id="line-211"></span>
<span class="source-line-no">212</span><span id="line-212">                                        if (gaps) {</span>
<span class="source-line-no">213</span><span id="line-213">                                                for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">214</span><span id="line-214">                                                        if (provisionalChar[str] == ' ') {</span>
<span class="source-line-no">215</span><span id="line-215">                                                                // It means this residue was non-consecutive</span>
<span class="source-line-no">216</span><span id="line-216">                                                                Atom a = atoms.get(str)[previousPos[str] + 1];</span>
<span class="source-line-no">217</span><span id="line-217">                                                                String group = a.getGroup().getPDBName();</span>
<span class="source-line-no">218</span><span id="line-218">                                                                char aa = StructureTools.get1LetterCode(group);</span>
<span class="source-line-no">219</span><span id="line-219">                                                                alnSequences</span>
<span class="source-line-no">220</span><span id="line-220">                                                                                .set(str,</span>
<span class="source-line-no">221</span><span id="line-221">                                                                                                alnSequences</span>
<span class="source-line-no">222</span><span id="line-222">                                                                                                                .get(str)</span>
<span class="source-line-no">223</span><span id="line-223">                                                                                                                .concat(String.valueOf(Character</span>
<span class="source-line-no">224</span><span id="line-224">                                                                                                                                                .toLowerCase(aa))) );</span>
<span class="source-line-no">225</span><span id="line-225">                                                                previousPos[str]++;</span>
<span class="source-line-no">226</span><span id="line-226">                                                        } else {</span>
<span class="source-line-no">227</span><span id="line-227">                                                                // Insert a gap otherwise</span>
<span class="source-line-no">228</span><span id="line-228">                                                                alnSequences.set(str, alnSequences.get(str)</span>
<span class="source-line-no">229</span><span id="line-229">                                                                                .concat("-"));</span>
<span class="source-line-no">230</span><span id="line-230">                                                        }</span>
<span class="source-line-no">231</span><span id="line-231">                                                }</span>
<span class="source-line-no">232</span><span id="line-232">                                                mapSeqToStruct.add(-1); // unaligned position</span>
<span class="source-line-no">233</span><span id="line-233">                                        } else {</span>
<span class="source-line-no">234</span><span id="line-234">                                                // Add provisional and update the indices</span>
<span class="source-line-no">235</span><span id="line-235">                                                for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">236</span><span id="line-236">                                                        alnSequences.set(</span>
<span class="source-line-no">237</span><span id="line-237">                                                                        str,</span>
<span class="source-line-no">238</span><span id="line-238">                                                                        alnSequences.get(str).concat(</span>
<span class="source-line-no">239</span><span id="line-239">                                                                                        String.valueOf(provisionalChar[str])));</span>
<span class="source-line-no">240</span><span id="line-240"></span>
<span class="source-line-no">241</span><span id="line-241">                                                        if (provisionalChar[str] != '-') {</span>
<span class="source-line-no">242</span><span id="line-242">                                                                if (alignment.getBlocks().get(b).getAlignRes()</span>
<span class="source-line-no">243</span><span id="line-243">                                                                                .get(str).get(pos) == null) {</span>
<span class="source-line-no">244</span><span id="line-244">                                                                        previousPos[str]++;</span>
<span class="source-line-no">245</span><span id="line-245">                                                                } else {</span>
<span class="source-line-no">246</span><span id="line-246">                                                                        previousPos[str] = alignment.getBlocks()</span>
<span class="source-line-no">247</span><span id="line-247">                                                                                        .get(b).getAlignRes().get(str)</span>
<span class="source-line-no">248</span><span id="line-248">                                                                                        .get(pos);</span>
<span class="source-line-no">249</span><span id="line-249">                                                                }</span>
<span class="source-line-no">250</span><span id="line-250">                                                        }</span>
<span class="source-line-no">251</span><span id="line-251">                                                }</span>
<span class="source-line-no">252</span><span id="line-252">                                                mapSeqToStruct.add(globalPos); // alignment index</span>
<span class="source-line-no">253</span><span id="line-253">                                        }</span>
<span class="source-line-no">254</span><span id="line-254">                                }</span>
<span class="source-line-no">255</span><span id="line-255">                        } // All positions in the Block considered so far</span>
<span class="source-line-no">256</span><span id="line-256"></span>
<span class="source-line-no">257</span><span id="line-257">                        // Calculate the index of the next Block for every structure</span>
<span class="source-line-no">258</span><span id="line-258">                        int[] blockEnds = new int[alignment.size()];</span>
<span class="source-line-no">259</span><span id="line-259">                        for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">260</span><span id="line-260">                                for (int res : blockStarts.get(str)) {</span>
<span class="source-line-no">261</span><span id="line-261">                                        if (previousPos[str] &gt; res)</span>
<span class="source-line-no">262</span><span id="line-262">                                                blockEnds[str] = res;</span>
<span class="source-line-no">263</span><span id="line-263">                                        else {</span>
<span class="source-line-no">264</span><span id="line-264">                                                blockEnds[str] = res;</span>
<span class="source-line-no">265</span><span id="line-265">                                                break;</span>
<span class="source-line-no">266</span><span id="line-266">                                        }</span>
<span class="source-line-no">267</span><span id="line-267">                                }</span>
<span class="source-line-no">268</span><span id="line-268">                        }</span>
<span class="source-line-no">269</span><span id="line-269"></span>
<span class="source-line-no">270</span><span id="line-270">                        // Add the unaligned residues in between Blocks (lowercase)</span>
<span class="source-line-no">271</span><span id="line-271">                        boolean allGaps = false; // true means no more residues to add</span>
<span class="source-line-no">272</span><span id="line-272">                        while (!allGaps) {</span>
<span class="source-line-no">273</span><span id="line-273">                                allGaps = true;</span>
<span class="source-line-no">274</span><span id="line-274">                                for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">275</span><span id="line-275">                                        if (previousPos[str] + 1 &lt; blockEnds[str]) {</span>
<span class="source-line-no">276</span><span id="line-276">                                                Atom a = atoms.get(str)[previousPos[str] + 1];</span>
<span class="source-line-no">277</span><span id="line-277">                                                String group = a.getGroup().getPDBName();</span>
<span class="source-line-no">278</span><span id="line-278">                                                char letter = StructureTools.get1LetterCode(group);</span>
<span class="source-line-no">279</span><span id="line-279"></span>
<span class="source-line-no">280</span><span id="line-280">                                                provisionalChar[str] = Character.toLowerCase(letter);</span>
<span class="source-line-no">281</span><span id="line-281">                                                previousPos[str]++;</span>
<span class="source-line-no">282</span><span id="line-282">                                                allGaps = false;</span>
<span class="source-line-no">283</span><span id="line-283">                                        } else</span>
<span class="source-line-no">284</span><span id="line-284">                                                provisionalChar[str] = '-';</span>
<span class="source-line-no">285</span><span id="line-285">                                }</span>
<span class="source-line-no">286</span><span id="line-286">                                if (!allGaps) {</span>
<span class="source-line-no">287</span><span id="line-287">                                        for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">288</span><span id="line-288">                                                alnSequences.set(</span>
<span class="source-line-no">289</span><span id="line-289">                                                                str,</span>
<span class="source-line-no">290</span><span id="line-290">                                                                alnSequences.get(str).concat(</span>
<span class="source-line-no">291</span><span id="line-291">                                                                                String.valueOf(provisionalChar[str])) );</span>
<span class="source-line-no">292</span><span id="line-292">                                        }</span>
<span class="source-line-no">293</span><span id="line-293">                                        mapSeqToStruct.add(-1); // unaligned position</span>
<span class="source-line-no">294</span><span id="line-294">                                }</span>
<span class="source-line-no">295</span><span id="line-295">                        }</span>
<span class="source-line-no">296</span><span id="line-296">                }</span>
<span class="source-line-no">297</span><span id="line-297">                return alnSequences;</span>
<span class="source-line-no">298</span><span id="line-298">        }</span>
<span class="source-line-no">299</span><span id="line-299"></span>
<span class="source-line-no">300</span><span id="line-300">        /**</span>
<span class="source-line-no">301</span><span id="line-301">         * Calculate the sequence alignment Strings for the whole alignment. This</span>
<span class="source-line-no">302</span><span id="line-302">         * method creates a sequence alignment where aligned residues are in</span>
<span class="source-line-no">303</span><span id="line-303">         * uppercase and unaligned residues are in lowercase, thus providing a more</span>
<span class="source-line-no">304</span><span id="line-304">         * compact way to represent the alignment.</span>
<span class="source-line-no">305</span><span id="line-305">         * &lt;p&gt;</span>
<span class="source-line-no">306</span><span id="line-306">         * Blocks are concatenated in the order returned by</span>
<span class="source-line-no">307</span><span id="line-307">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</span>
<span class="source-line-no">308</span><span id="line-308">         * sequential. Gaps are represented by '-'. Separation between different</span>
<span class="source-line-no">309</span><span id="line-309">         * Blocks is indicated by a gap in all positions, meaning that there is a</span>
<span class="source-line-no">310</span><span id="line-310">         * possible discontinuity.</span>
<span class="source-line-no">311</span><span id="line-311">         *</span>
<span class="source-line-no">312</span><span id="line-312">         * @param msa</span>
<span class="source-line-no">313</span><span id="line-313">         *            input MultipleAlignment</span>
<span class="source-line-no">314</span><span id="line-314">         * @return String for each row in the alignment, giving the 1-letter code</span>
<span class="source-line-no">315</span><span id="line-315">         *         for each aligned residue.</span>
<span class="source-line-no">316</span><span id="line-316">         */</span>
<span class="source-line-no">317</span><span id="line-317">        public static List&lt;String&gt; getSequenceAlignment(MultipleAlignment msa) {</span>
<span class="source-line-no">318</span><span id="line-318">                return getSequenceAlignment(msa, new ArrayList&lt;Integer&gt;());</span>
<span class="source-line-no">319</span><span id="line-319">        }</span>
<span class="source-line-no">320</span><span id="line-320"></span>
<span class="source-line-no">321</span><span id="line-321">        /**</span>
<span class="source-line-no">322</span><span id="line-322">         * Calculate the sequence alignment Strings for the alignment Blocks in an</span>
<span class="source-line-no">323</span><span id="line-323">         * alignment. This method creates a sequence alignment where all residues</span>
<span class="source-line-no">324</span><span id="line-324">         * are in uppercase and a residue alone with gaps in all the other</span>
<span class="source-line-no">325</span><span id="line-325">         * structures represents unaligned residues. Because of this latter</span>
<span class="source-line-no">326</span><span id="line-326">         * constraint only the residues within the Blocks are represented, for a</span>
<span class="source-line-no">327</span><span id="line-327">         * more compact alignment. For a sequence alignment of the full protein use</span>
<span class="source-line-no">328</span><span id="line-328">         * {@link #getSequenceAlignment(MultipleAlignment)}.</span>
<span class="source-line-no">329</span><span id="line-329">         * &lt;p&gt;</span>
<span class="source-line-no">330</span><span id="line-330">         * Blocks are concatenated in the order returned by</span>
<span class="source-line-no">331</span><span id="line-331">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</span>
<span class="source-line-no">332</span><span id="line-332">         * sequential. Gaps between blocks are omitted, while gaps within blocks are</span>
<span class="source-line-no">333</span><span id="line-333">         * represented by '-'. Separation between different Blocks is indicated by a</span>
<span class="source-line-no">334</span><span id="line-334">         * gap in all positions, meaning that there is something unaligned</span>
<span class="source-line-no">335</span><span id="line-335">         * inbetween.</span>
<span class="source-line-no">336</span><span id="line-336">         *</span>
<span class="source-line-no">337</span><span id="line-337">         * @param alignment</span>
<span class="source-line-no">338</span><span id="line-338">         *            input MultipleAlignment</span>
<span class="source-line-no">339</span><span id="line-339">         * @param mapSeqToStruct</span>
<span class="source-line-no">340</span><span id="line-340">         *            provides a link from the sequence alignment position to the</span>
<span class="source-line-no">341</span><span id="line-341">         *            structure alignment position. Specially designed for the GUI.</span>
<span class="source-line-no">342</span><span id="line-342">         *            Has to be initialized previously and will be overwritten.</span>
<span class="source-line-no">343</span><span id="line-343">         * @return a string for each row in the alignment, giving the 1-letter code</span>
<span class="source-line-no">344</span><span id="line-344">         *         for each aligned residue.</span>
<span class="source-line-no">345</span><span id="line-345">         */</span>
<span class="source-line-no">346</span><span id="line-346">        public static List&lt;String&gt; getBlockSequenceAlignment(</span>
<span class="source-line-no">347</span><span id="line-347">                        MultipleAlignment alignment, List&lt;Integer&gt; mapSeqToStruct) {</span>
<span class="source-line-no">348</span><span id="line-348"></span>
<span class="source-line-no">349</span><span id="line-349">                // Initialize sequence variables</span>
<span class="source-line-no">350</span><span id="line-350">                List&lt;String&gt; alnSequences = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">351</span><span id="line-351">                for (int str = 0; str &lt; alignment.size(); str++)</span>
<span class="source-line-no">352</span><span id="line-352">                        alnSequences.add("");</span>
<span class="source-line-no">353</span><span id="line-353">                mapSeqToStruct.clear();</span>
<span class="source-line-no">354</span><span id="line-354">                List&lt;Atom[]&gt; atoms = alignment.getAtomArrays();</span>
<span class="source-line-no">355</span><span id="line-355">                int globalPos = -1;</span>
<span class="source-line-no">356</span><span id="line-356"></span>
<span class="source-line-no">357</span><span id="line-357">                // Loop through all the alignment Blocks in the order given</span>
<span class="source-line-no">358</span><span id="line-358">                for (int b = 0; b &lt; alignment.getBlocks().size(); b++) {</span>
<span class="source-line-no">359</span><span id="line-359">                        if (b != 0) {</span>
<span class="source-line-no">360</span><span id="line-360">                                // Add a gap to all structures to separate Blocks</span>
<span class="source-line-no">361</span><span id="line-361">                                for (int str = 0; str &lt; alignment.size(); str++)</span>
<span class="source-line-no">362</span><span id="line-362">                                        alnSequences.set(str, alnSequences.get(str).concat("-"));</span>
<span class="source-line-no">363</span><span id="line-363">                                mapSeqToStruct.add(-1); // means unaligned position</span>
<span class="source-line-no">364</span><span id="line-364">                        }</span>
<span class="source-line-no">365</span><span id="line-365"></span>
<span class="source-line-no">366</span><span id="line-366">                        // Store the previous position added to the sequence alignment</span>
<span class="source-line-no">367</span><span id="line-367">                        int[] previousPos = new int[alignment.size()];</span>
<span class="source-line-no">368</span><span id="line-368">                        Arrays.fill(previousPos, -1);</span>
<span class="source-line-no">369</span><span id="line-369">                        // Store provisional characters</span>
<span class="source-line-no">370</span><span id="line-370">                        char[] provisionalChar = new char[alignment.size()];</span>
<span class="source-line-no">371</span><span id="line-371">                        Arrays.fill(provisionalChar, '-');</span>
<span class="source-line-no">372</span><span id="line-372"></span>
<span class="source-line-no">373</span><span id="line-373">                        for (int pos = 0; pos &lt; alignment.getBlocks().get(b).length(); pos++) {</span>
<span class="source-line-no">374</span><span id="line-374">                                globalPos++;</span>
<span class="source-line-no">375</span><span id="line-375">                                boolean gaps = true;</span>
<span class="source-line-no">376</span><span id="line-376">                                while (gaps) {</span>
<span class="source-line-no">377</span><span id="line-377">                                        gaps = false;</span>
<span class="source-line-no">378</span><span id="line-378">                                        // Loop through all the structures</span>
<span class="source-line-no">379</span><span id="line-379">                                        for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">380</span><span id="line-380">                                                // If it is the first position or before it was null</span>
<span class="source-line-no">381</span><span id="line-381">                                                if (previousPos[str] == -1) {</span>
<span class="source-line-no">382</span><span id="line-382">                                                        Integer residue = alignment.getBlocks().get(b)</span>
<span class="source-line-no">383</span><span id="line-383">                                                                        .getAlignRes().get(str).get(pos);</span>
<span class="source-line-no">384</span><span id="line-384">                                                        if (residue == null)</span>
<span class="source-line-no">385</span><span id="line-385">                                                                provisionalChar[str] = '-';</span>
<span class="source-line-no">386</span><span id="line-386">                                                        else {</span>
<span class="source-line-no">387</span><span id="line-387">                                                                Atom a = atoms.get(str)[residue];</span>
<span class="source-line-no">388</span><span id="line-388">                                                                String g = a.getGroup().getPDBName();</span>
<span class="source-line-no">389</span><span id="line-389">                                                                char aa = StructureTools.get1LetterCode(g);</span>
<span class="source-line-no">390</span><span id="line-390">                                                                provisionalChar[str] = aa;</span>
<span class="source-line-no">391</span><span id="line-391">                                                        }</span>
<span class="source-line-no">392</span><span id="line-392">                                                } else {</span>
<span class="source-line-no">393</span><span id="line-393">                                                        Integer residue = alignment.getBlocks().get(b)</span>
<span class="source-line-no">394</span><span id="line-394">                                                                        .getAlignRes().get(str).get(pos);</span>
<span class="source-line-no">395</span><span id="line-395">                                                        if (residue == null)</span>
<span class="source-line-no">396</span><span id="line-396">                                                                provisionalChar[str] = '-';</span>
<span class="source-line-no">397</span><span id="line-397">                                                        else if (previousPos[str] + 1 == residue) {</span>
<span class="source-line-no">398</span><span id="line-398">                                                                Atom a = atoms.get(str)[residue];</span>
<span class="source-line-no">399</span><span id="line-399">                                                                String g = a.getGroup().getPDBName();</span>
<span class="source-line-no">400</span><span id="line-400">                                                                char aa = StructureTools.get1LetterCode(g);</span>
<span class="source-line-no">401</span><span id="line-401">                                                                provisionalChar[str] = aa;</span>
<span class="source-line-no">402</span><span id="line-402">                                                        } else {</span>
<span class="source-line-no">403</span><span id="line-403">                                                                provisionalChar[str] = ' ';</span>
<span class="source-line-no">404</span><span id="line-404">                                                                gaps = true;</span>
<span class="source-line-no">405</span><span id="line-405">                                                        }</span>
<span class="source-line-no">406</span><span id="line-406">                                                }</span>
<span class="source-line-no">407</span><span id="line-407">                                        }// End all structures analysis</span>
<span class="source-line-no">408</span><span id="line-408"></span>
<span class="source-line-no">409</span><span id="line-409">                                        if (gaps) {</span>
<span class="source-line-no">410</span><span id="line-410">                                                for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">411</span><span id="line-411">                                                        if (provisionalChar[str] == ' ') {</span>
<span class="source-line-no">412</span><span id="line-412">                                                                // It means this residue was non-consecutive</span>
<span class="source-line-no">413</span><span id="line-413">                                                                for (int s2 = 0; s2 &lt; alignment.size(); s2++) {</span>
<span class="source-line-no">414</span><span id="line-414">                                                                        if (str == s2) {</span>
<span class="source-line-no">415</span><span id="line-415">                                                                                int next = previousPos[str] + 1;</span>
<span class="source-line-no">416</span><span id="line-416">                                                                                Atom a = atoms.get(s2)[next];</span>
<span class="source-line-no">417</span><span id="line-417">                                                                                String g = a.getGroup().getPDBName();</span>
<span class="source-line-no">418</span><span id="line-418">                                                                                char aa = StructureTools</span>
<span class="source-line-no">419</span><span id="line-419">                                                                                                .get1LetterCode(g);</span>
<span class="source-line-no">420</span><span id="line-420">                                                                                alnSequences.set(</span>
<span class="source-line-no">421</span><span id="line-421">                                                                                                s2,</span>
<span class="source-line-no">422</span><span id="line-422">                                                                                                alnSequences.get(s2).concat(</span>
<span class="source-line-no">423</span><span id="line-423">                                                                                                                String.valueOf(aa)) );</span>
<span class="source-line-no">424</span><span id="line-424">                                                                        } else {</span>
<span class="source-line-no">425</span><span id="line-425">                                                                                alnSequences.set(s2,</span>
<span class="source-line-no">426</span><span id="line-426">                                                                                                alnSequences.get(s2)</span>
<span class="source-line-no">427</span><span id="line-427">                                                                                                                .concat("-"));</span>
<span class="source-line-no">428</span><span id="line-428">                                                                        }</span>
<span class="source-line-no">429</span><span id="line-429">                                                                }</span>
<span class="source-line-no">430</span><span id="line-430">                                                                mapSeqToStruct.add(-1); // unaligned</span>
<span class="source-line-no">431</span><span id="line-431">                                                                previousPos[str] += 1;</span>
<span class="source-line-no">432</span><span id="line-432">                                                        }</span>
<span class="source-line-no">433</span><span id="line-433">                                                }</span>
<span class="source-line-no">434</span><span id="line-434">                                        } else { // Append the provisional and update the indices</span>
<span class="source-line-no">435</span><span id="line-435">                                                for (int str = 0; str &lt; alignment.size(); str++) {</span>
<span class="source-line-no">436</span><span id="line-436">                                                        alnSequences.set(</span>
<span class="source-line-no">437</span><span id="line-437">                                                                        str,</span>
<span class="source-line-no">438</span><span id="line-438">                                                                        alnSequences.get(str).concat(</span>
<span class="source-line-no">439</span><span id="line-439">                                                                                        String.valueOf(provisionalChar[str])) );</span>
<span class="source-line-no">440</span><span id="line-440">                                                        if (provisionalChar[str] != '-') {</span>
<span class="source-line-no">441</span><span id="line-441">                                                                previousPos[str] = alignment.getBlocks().get(b)</span>
<span class="source-line-no">442</span><span id="line-442">                                                                                .getAlignRes().get(str).get(pos);</span>
<span class="source-line-no">443</span><span id="line-443">                                                        }</span>
<span class="source-line-no">444</span><span id="line-444">                                                }</span>
<span class="source-line-no">445</span><span id="line-445">                                                mapSeqToStruct.add(globalPos);</span>
<span class="source-line-no">446</span><span id="line-446">                                        }</span>
<span class="source-line-no">447</span><span id="line-447">                                }</span>
<span class="source-line-no">448</span><span id="line-448">                        }</span>
<span class="source-line-no">449</span><span id="line-449">                }</span>
<span class="source-line-no">450</span><span id="line-450">                return alnSequences;</span>
<span class="source-line-no">451</span><span id="line-451">        }</span>
<span class="source-line-no">452</span><span id="line-452"></span>
<span class="source-line-no">453</span><span id="line-453">        /**</span>
<span class="source-line-no">454</span><span id="line-454">         * Calculate the sequence alignment Strings for the alignment Blocks in an</span>
<span class="source-line-no">455</span><span id="line-455">         * alignment. This method creates a sequence alignment where all residues</span>
<span class="source-line-no">456</span><span id="line-456">         * are in uppercase and a residue alone with gaps in all the other</span>
<span class="source-line-no">457</span><span id="line-457">         * structures represents unaligned residues. Because of this latter</span>
<span class="source-line-no">458</span><span id="line-458">         * constraint only the residues within the Blocks are represented, for a</span>
<span class="source-line-no">459</span><span id="line-459">         * more compact alignment. For a sequence alignment of the full protein use</span>
<span class="source-line-no">460</span><span id="line-460">         * {@link #getSequenceAlignment(MultipleAlignment)}.</span>
<span class="source-line-no">461</span><span id="line-461">         * &lt;p&gt;</span>
<span class="source-line-no">462</span><span id="line-462">         * Blocks are concatenated in the order returned by</span>
<span class="source-line-no">463</span><span id="line-463">         * {@link MultipleAlignment#getBlocks()}, so sequences may not be</span>
<span class="source-line-no">464</span><span id="line-464">         * sequential. Gaps between blocks are omitted, while gaps within blocks are</span>
<span class="source-line-no">465</span><span id="line-465">         * represented by '-'. Separation between different Blocks is indicated by a</span>
<span class="source-line-no">466</span><span id="line-466">         * gap in all positions, meaning that there is something unaligned</span>
<span class="source-line-no">467</span><span id="line-467">         * inbetween.</span>
<span class="source-line-no">468</span><span id="line-468">         *</span>
<span class="source-line-no">469</span><span id="line-469">         * @param ma</span>
<span class="source-line-no">470</span><span id="line-470">         *            input MultipleAlignment</span>
<span class="source-line-no">471</span><span id="line-471">         * @return String for each row in the alignment, giving the 1-letter code</span>
<span class="source-line-no">472</span><span id="line-472">         *         for each aligned residue.</span>
<span class="source-line-no">473</span><span id="line-473">         */</span>
<span class="source-line-no">474</span><span id="line-474">        public static List&lt;String&gt; getBlockSequenceAlignment(MultipleAlignment ma) {</span>
<span class="source-line-no">475</span><span id="line-475">                return getBlockSequenceAlignment(ma, new ArrayList&lt;Integer&gt;());</span>
<span class="source-line-no">476</span><span id="line-476">        }</span>
<span class="source-line-no">477</span><span id="line-477"></span>
<span class="source-line-no">478</span><span id="line-478">        /**</span>
<span class="source-line-no">479</span><span id="line-479">         * Returns the Atom of the specified structure that is aligned in the</span>
<span class="source-line-no">480</span><span id="line-480">         * sequence alignment position specified.</span>
<span class="source-line-no">481</span><span id="line-481">         *</span>
<span class="source-line-no">482</span><span id="line-482">         * @param msa</span>
<span class="source-line-no">483</span><span id="line-483">         *            the MultipleAlignment object from where the sequence alignment</span>
<span class="source-line-no">484</span><span id="line-484">         *            has been generated</span>
<span class="source-line-no">485</span><span id="line-485">         * @param mapSeqToStruct</span>
<span class="source-line-no">486</span><span id="line-486">         *            the mapping between sequence and structure generated with the</span>
<span class="source-line-no">487</span><span id="line-487">         *            sequence alignment</span>
<span class="source-line-no">488</span><span id="line-488">         * @param str</span>
<span class="source-line-no">489</span><span id="line-489">         *            the structure index of the alignment (row)</span>
<span class="source-line-no">490</span><span id="line-490">         * @param sequencePos</span>
<span class="source-line-no">491</span><span id="line-491">         *            the sequence alignment position (column)</span>
<span class="source-line-no">492</span><span id="line-492">         * @return Atom the atom in that position or null if there is a gap</span>
<span class="source-line-no">493</span><span id="line-493">         */</span>
<span class="source-line-no">494</span><span id="line-494">        public static Atom getAtomForSequencePosition(MultipleAlignment msa,</span>
<span class="source-line-no">495</span><span id="line-495">                        List&lt;Integer&gt; mapSeqToStruct, int str, int sequencePos) {</span>
<span class="source-line-no">496</span><span id="line-496"></span>
<span class="source-line-no">497</span><span id="line-497">                int seqPos = mapSeqToStruct.get(sequencePos);</span>
<span class="source-line-no">498</span><span id="line-498"></span>
<span class="source-line-no">499</span><span id="line-499">                // Check if the position selected is an aligned position</span>
<span class="source-line-no">500</span><span id="line-500">                if (seqPos == -1)</span>
<span class="source-line-no">501</span><span id="line-501">                        return null;</span>
<span class="source-line-no">502</span><span id="line-502">                else {</span>
<span class="source-line-no">503</span><span id="line-503">                        Atom a = null;</span>
<span class="source-line-no">504</span><span id="line-504">                        // Calculate the corresponding structure position</span>
<span class="source-line-no">505</span><span id="line-505">                        int sum = 0;</span>
<span class="source-line-no">506</span><span id="line-506">                        for (Block b : msa.getBlocks()) {</span>
<span class="source-line-no">507</span><span id="line-507">                                if (sum + b.length() &lt;= seqPos) {</span>
<span class="source-line-no">508</span><span id="line-508">                                        sum += b.length();</span>
<span class="source-line-no">509</span><span id="line-509">                                        continue;</span>
<span class="source-line-no">510</span><span id="line-510">                                } else {</span>
<span class="source-line-no">511</span><span id="line-511">                                        for (Integer p : b.getAlignRes().get(str)) {</span>
<span class="source-line-no">512</span><span id="line-512">                                                if (sum == seqPos) {</span>
<span class="source-line-no">513</span><span id="line-513">                                                        if (p != null) {</span>
<span class="source-line-no">514</span><span id="line-514">                                                                a = msa.getAtomArrays().get(str)[p];</span>
<span class="source-line-no">515</span><span id="line-515">                                                        }</span>
<span class="source-line-no">516</span><span id="line-516">                                                        break;</span>
<span class="source-line-no">517</span><span id="line-517">                                                }</span>
<span class="source-line-no">518</span><span id="line-518">                                                sum++;</span>
<span class="source-line-no">519</span><span id="line-519">                                        }</span>
<span class="source-line-no">520</span><span id="line-520">                                        break;</span>
<span class="source-line-no">521</span><span id="line-521">                                }</span>
<span class="source-line-no">522</span><span id="line-522">                        }</span>
<span class="source-line-no">523</span><span id="line-523">                        return a;</span>
<span class="source-line-no">524</span><span id="line-524">                }</span>
<span class="source-line-no">525</span><span id="line-525">        }</span>
<span class="source-line-no">526</span><span id="line-526"></span>
<span class="source-line-no">527</span><span id="line-527">        /**</span>
<span class="source-line-no">528</span><span id="line-528">         * Returns the block number of a specified position in the sequence</span>
<span class="source-line-no">529</span><span id="line-529">         * alignment, given the mapping from structure to function.</span>
<span class="source-line-no">530</span><span id="line-530">         *</span>
<span class="source-line-no">531</span><span id="line-531">         * @param multAln</span>
<span class="source-line-no">532</span><span id="line-532">         *            the MultipleAlignment object from where the sequence alignment</span>
<span class="source-line-no">533</span><span id="line-533">         *            has been generated.</span>
<span class="source-line-no">534</span><span id="line-534">         * @param mapSeqToStruct</span>
<span class="source-line-no">535</span><span id="line-535">         *            the mapping between sequence and structure generated with the</span>
<span class="source-line-no">536</span><span id="line-536">         *            sequence alignment</span>
<span class="source-line-no">537</span><span id="line-537">         * @param sequencePos</span>
<span class="source-line-no">538</span><span id="line-538">         *            the position in the sequence alignment (column)</span>
<span class="source-line-no">539</span><span id="line-539">         * @return int the block index, or -1 if the position is not aligned</span>
<span class="source-line-no">540</span><span id="line-540">         */</span>
<span class="source-line-no">541</span><span id="line-541">        public static int getBlockForSequencePosition(MultipleAlignment multAln,</span>
<span class="source-line-no">542</span><span id="line-542">                        List&lt;Integer&gt; mapSeqToStruct, int sequencePos) {</span>
<span class="source-line-no">543</span><span id="line-543"></span>
<span class="source-line-no">544</span><span id="line-544">                int seqPos = mapSeqToStruct.get(sequencePos);</span>
<span class="source-line-no">545</span><span id="line-545">                // Check if the position selected is an aligned position</span>
<span class="source-line-no">546</span><span id="line-546">                if (seqPos == -1)</span>
<span class="source-line-no">547</span><span id="line-547">                        return -1;</span>
<span class="source-line-no">548</span><span id="line-548">                else {</span>
<span class="source-line-no">549</span><span id="line-549">                        // Calculate the corresponding block (by iterating all Blocks)</span>
<span class="source-line-no">550</span><span id="line-550">                        int sum = 0;</span>
<span class="source-line-no">551</span><span id="line-551">                        int block = 0;</span>
<span class="source-line-no">552</span><span id="line-552">                        for (Block b : multAln.getBlocks()) {</span>
<span class="source-line-no">553</span><span id="line-553">                                if (sum + b.length() &lt;= seqPos) {</span>
<span class="source-line-no">554</span><span id="line-554">                                        sum += b.length();</span>
<span class="source-line-no">555</span><span id="line-555">                                        block++;</span>
<span class="source-line-no">556</span><span id="line-556">                                        continue;</span>
<span class="source-line-no">557</span><span id="line-557">                                } else</span>
<span class="source-line-no">558</span><span id="line-558">                                        break;</span>
<span class="source-line-no">559</span><span id="line-559">                        }</span>
<span class="source-line-no">560</span><span id="line-560">                        return block;</span>
<span class="source-line-no">561</span><span id="line-561">                }</span>
<span class="source-line-no">562</span><span id="line-562">        }</span>
<span class="source-line-no">563</span><span id="line-563"></span>
<span class="source-line-no">564</span><span id="line-564">        /**</span>
<span class="source-line-no">565</span><span id="line-565">         * The average residue distance Matrix contains the average distance from</span>
<span class="source-line-no">566</span><span id="line-566">         * each residue to all other residues aligned with it.</span>
<span class="source-line-no">567</span><span id="line-567">         * &lt;p&gt;</span>
<span class="source-line-no">568</span><span id="line-568">         * Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment</span>
<span class="source-line-no">569</span><span id="line-569">         * length.</span>
<span class="source-line-no">570</span><span id="line-570">         *</span>
<span class="source-line-no">571</span><span id="line-571">         * @param msa</span>
<span class="source-line-no">572</span><span id="line-572">         *            MultipleAlignment</span>
<span class="source-line-no">573</span><span id="line-573">         * @return Matrix containing all average residue distances</span>
<span class="source-line-no">574</span><span id="line-574">         */</span>
<span class="source-line-no">575</span><span id="line-575">        public static Matrix getAverageResidueDistances(MultipleAlignment msa) {</span>
<span class="source-line-no">576</span><span id="line-576">                List&lt;Atom[]&gt; transformed = transformAtoms(msa);</span>
<span class="source-line-no">577</span><span id="line-577">                return getAverageResidueDistances(transformed);</span>
<span class="source-line-no">578</span><span id="line-578">        }</span>
<span class="source-line-no">579</span><span id="line-579"></span>
<span class="source-line-no">580</span><span id="line-580">        /**</span>
<span class="source-line-no">581</span><span id="line-581">         * The average residue distance Matrix contains the average distance from</span>
<span class="source-line-no">582</span><span id="line-582">         * each residue to all other residues aligned with it.</span>
<span class="source-line-no">583</span><span id="line-583">         * &lt;p&gt;</span>
<span class="source-line-no">584</span><span id="line-584">         * Complexity: T(n,l) = O(l*n^2), if n=number of structures and l=alignment</span>
<span class="source-line-no">585</span><span id="line-585">         * length.</span>
<span class="source-line-no">586</span><span id="line-586">         *</span>
<span class="source-line-no">587</span><span id="line-587">         * @param transformed</span>
<span class="source-line-no">588</span><span id="line-588">         *            List of Atom arrays containing only the aligned atoms of each</span>
<span class="source-line-no">589</span><span id="line-589">         *            structure, or null if there is a gap.</span>
<span class="source-line-no">590</span><span id="line-590">         * @return Matrix containing all average residue distances. Entry -1 means</span>
<span class="source-line-no">591</span><span id="line-591">         *         there is a gap in the position.</span>
<span class="source-line-no">592</span><span id="line-592">         */</span>
<span class="source-line-no">593</span><span id="line-593">        public static Matrix getAverageResidueDistances(List&lt;Atom[]&gt; transformed) {</span>
<span class="source-line-no">594</span><span id="line-594"></span>
<span class="source-line-no">595</span><span id="line-595">                int size = transformed.size();</span>
<span class="source-line-no">596</span><span id="line-596">                int length = transformed.get(0).length;</span>
<span class="source-line-no">597</span><span id="line-597">                Matrix resDist = new Matrix(size, length, -1);</span>
<span class="source-line-no">598</span><span id="line-598"></span>
<span class="source-line-no">599</span><span id="line-599">                // Calculate the average residue distances</span>
<span class="source-line-no">600</span><span id="line-600">                for (int r1 = 0; r1 &lt; size; r1++) {</span>
<span class="source-line-no">601</span><span id="line-601">                        for (int c = 0; c &lt; transformed.get(r1).length; c++) {</span>
<span class="source-line-no">602</span><span id="line-602">                                Atom refAtom = transformed.get(r1)[c];</span>
<span class="source-line-no">603</span><span id="line-603">                                if (refAtom == null)</span>
<span class="source-line-no">604</span><span id="line-604">                                        continue;</span>
<span class="source-line-no">605</span><span id="line-605"></span>
<span class="source-line-no">606</span><span id="line-606">                                for (int r2 = r1 + 1; r2 &lt; size; r2++) {</span>
<span class="source-line-no">607</span><span id="line-607">                                        Atom atom = transformed.get(r2)[c];</span>
<span class="source-line-no">608</span><span id="line-608">                                        if (atom != null) {</span>
<span class="source-line-no">609</span><span id="line-609">                                                double distance = Calc.getDistance(refAtom, atom);</span>
<span class="source-line-no">610</span><span id="line-610"></span>
<span class="source-line-no">611</span><span id="line-611">                                                if (resDist.get(r1, c) == -1) {</span>
<span class="source-line-no">612</span><span id="line-612">                                                        resDist.set(r1, c, 1 + distance);</span>
<span class="source-line-no">613</span><span id="line-613">                                                } else {</span>
<span class="source-line-no">614</span><span id="line-614">                                                        resDist.set(r1, c, resDist.get(r1, c) + distance);</span>
<span class="source-line-no">615</span><span id="line-615">                                                }</span>
<span class="source-line-no">616</span><span id="line-616"></span>
<span class="source-line-no">617</span><span id="line-617">                                                if (resDist.get(r2, c) == -1) {</span>
<span class="source-line-no">618</span><span id="line-618">                                                        resDist.set(r2, c, 1 + distance);</span>
<span class="source-line-no">619</span><span id="line-619">                                                } else {</span>
<span class="source-line-no">620</span><span id="line-620">                                                        resDist.set(r2, c, resDist.get(r2, c) + distance);</span>
<span class="source-line-no">621</span><span id="line-621">                                                }</span>
<span class="source-line-no">622</span><span id="line-622">                                        }</span>
<span class="source-line-no">623</span><span id="line-623">                                }</span>
<span class="source-line-no">624</span><span id="line-624">                        }</span>
<span class="source-line-no">625</span><span id="line-625">                }</span>
<span class="source-line-no">626</span><span id="line-626"></span>
<span class="source-line-no">627</span><span id="line-627">                for (int c = 0; c &lt; length; c++) {</span>
<span class="source-line-no">628</span><span id="line-628">                        int nonNullRes = 0;</span>
<span class="source-line-no">629</span><span id="line-629">                        for (int r = 0; r &lt; size; r++) {</span>
<span class="source-line-no">630</span><span id="line-630">                                if (resDist.get(r, c) != -1)</span>
<span class="source-line-no">631</span><span id="line-631">                                        nonNullRes++;</span>
<span class="source-line-no">632</span><span id="line-632">                        }</span>
<span class="source-line-no">633</span><span id="line-633">                        for (int r = 0; r &lt; size; r++) {</span>
<span class="source-line-no">634</span><span id="line-634">                                if (resDist.get(r, c) != -1) {</span>
<span class="source-line-no">635</span><span id="line-635">                                        resDist.set(r, c, resDist.get(r, c) / nonNullRes);</span>
<span class="source-line-no">636</span><span id="line-636">                                }</span>
<span class="source-line-no">637</span><span id="line-637">                        }</span>
<span class="source-line-no">638</span><span id="line-638">                }</span>
<span class="source-line-no">639</span><span id="line-639">                return resDist;</span>
<span class="source-line-no">640</span><span id="line-640">        }</span>
<span class="source-line-no">641</span><span id="line-641"></span>
<span class="source-line-no">642</span><span id="line-642">        /**</span>
<span class="source-line-no">643</span><span id="line-643">         * Transforms atoms according to the superposition stored in the alignment.</span>
<span class="source-line-no">644</span><span id="line-644">         * &lt;p&gt;</span>
<span class="source-line-no">645</span><span id="line-645">         * For each structure in the alignment, returns an atom for each</span>
<span class="source-line-no">646</span><span id="line-646">         * representative atom in the aligned columns, omitting unaligned residues</span>
<span class="source-line-no">647</span><span id="line-647">         * (i.e. an array of length &lt;code&gt;alignment.length()&lt;/code&gt; ).</span>
<span class="source-line-no">648</span><span id="line-648">         * &lt;p&gt;</span>
<span class="source-line-no">649</span><span id="line-649">         * All blocks are concatenated together, so Atoms may not appear in the same</span>
<span class="source-line-no">650</span><span id="line-650">         * order as in their parent structure. If the alignment blocks contain null</span>
<span class="source-line-no">651</span><span id="line-651">         * residues (gaps), then the returned array will also contain null Atoms in</span>
<span class="source-line-no">652</span><span id="line-652">         * the same positions.</span>
<span class="source-line-no">653</span><span id="line-653">         *</span>
<span class="source-line-no">654</span><span id="line-654">         * @param alignment</span>
<span class="source-line-no">655</span><span id="line-655">         *            MultipleAlignment</span>
<span class="source-line-no">656</span><span id="line-656">         * @return List of Atom arrays of only the aligned atoms of every structure</span>
<span class="source-line-no">657</span><span id="line-657">         *         (null Atom if a gap position)</span>
<span class="source-line-no">658</span><span id="line-658">         */</span>
<span class="source-line-no">659</span><span id="line-659">        public static List&lt;Atom[]&gt; transformAtoms(MultipleAlignment alignment) {</span>
<span class="source-line-no">660</span><span id="line-660">                if (alignment.getEnsemble() == null) {</span>
<span class="source-line-no">661</span><span id="line-661">                        throw new NullPointerException("No ensemble set for this alignment");</span>
<span class="source-line-no">662</span><span id="line-662">                }</span>
<span class="source-line-no">663</span><span id="line-663"></span>
<span class="source-line-no">664</span><span id="line-664">                List&lt;Atom[]&gt; atomArrays = alignment.getAtomArrays();</span>
<span class="source-line-no">665</span><span id="line-665">                List&lt;Atom[]&gt; transformed = new ArrayList&lt;&gt;(atomArrays.size());</span>
<span class="source-line-no">666</span><span id="line-666"></span>
<span class="source-line-no">667</span><span id="line-667">                // Loop through structures</span>
<span class="source-line-no">668</span><span id="line-668">                for (int i = 0; i &lt; atomArrays.size(); i++) {</span>
<span class="source-line-no">669</span><span id="line-669"></span>
<span class="source-line-no">670</span><span id="line-670">                        Matrix4d transform = null;</span>
<span class="source-line-no">671</span><span id="line-671">                        Atom[] curr = atomArrays.get(i); // all CA atoms from structure</span>
<span class="source-line-no">672</span><span id="line-672"></span>
<span class="source-line-no">673</span><span id="line-673">                        // Concatenated list of all blocks for this structure</span>
<span class="source-line-no">674</span><span id="line-674">                        Atom[] transformedAtoms = new Atom[alignment.length()];</span>
<span class="source-line-no">675</span><span id="line-675">                        int transformedAtomsLength = 0;</span>
<span class="source-line-no">676</span><span id="line-676"></span>
<span class="source-line-no">677</span><span id="line-677">                        // Each blockset gets transformed independently</span>
<span class="source-line-no">678</span><span id="line-678">                        for (BlockSet bs : alignment.getBlockSets()) {</span>
<span class="source-line-no">679</span><span id="line-679"></span>
<span class="source-line-no">680</span><span id="line-680">                                Atom[] blocksetAtoms = new Atom[bs.length()];</span>
<span class="source-line-no">681</span><span id="line-681">                                int blockPos = 0;</span>
<span class="source-line-no">682</span><span id="line-682"></span>
<span class="source-line-no">683</span><span id="line-683">                                for (Block blk : bs.getBlocks()) {</span>
<span class="source-line-no">684</span><span id="line-684">                                        if (blk.size() != atomArrays.size()) {</span>
<span class="source-line-no">685</span><span id="line-685">                                                throw new IllegalStateException(String.format(</span>
<span class="source-line-no">686</span><span id="line-686">                                                                "Mismatched block size. Expected %d "</span>
<span class="source-line-no">687</span><span id="line-687">                                                                                + "structures, found %d.",</span>
<span class="source-line-no">688</span><span id="line-688">                                                                atomArrays.size(), blk.size()));</span>
<span class="source-line-no">689</span><span id="line-689">                                        }</span>
<span class="source-line-no">690</span><span id="line-690">                                        // Extract aligned atoms</span>
<span class="source-line-no">691</span><span id="line-691">                                        for (int j = 0; j &lt; blk.length(); j++) {</span>
<span class="source-line-no">692</span><span id="line-692">                                                Integer alignedPos = blk.getAlignRes().get(i).get(j);</span>
<span class="source-line-no">693</span><span id="line-693">                                                if (alignedPos != null) {</span>
<span class="source-line-no">694</span><span id="line-694">                                                        blocksetAtoms[blockPos] = (Atom) curr[alignedPos]</span>
<span class="source-line-no">695</span><span id="line-695">                                                                        .clone();</span>
<span class="source-line-no">696</span><span id="line-696">                                                }</span>
<span class="source-line-no">697</span><span id="line-697">                                                blockPos++;</span>
<span class="source-line-no">698</span><span id="line-698">                                        }</span>
<span class="source-line-no">699</span><span id="line-699">                                }</span>
<span class="source-line-no">700</span><span id="line-700"></span>
<span class="source-line-no">701</span><span id="line-701">                                // Transform according to the blockset or alignment matrix</span>
<span class="source-line-no">702</span><span id="line-702">                                Matrix4d blockTrans = null;</span>
<span class="source-line-no">703</span><span id="line-703">                                if (bs.getTransformations() != null)</span>
<span class="source-line-no">704</span><span id="line-704">                                        blockTrans = bs.getTransformations().get(i);</span>
<span class="source-line-no">705</span><span id="line-705">                                if (blockTrans == null) {</span>
<span class="source-line-no">706</span><span id="line-706">                                        blockTrans = transform;</span>
<span class="source-line-no">707</span><span id="line-707">                                }</span>
<span class="source-line-no">708</span><span id="line-708"></span>
<span class="source-line-no">709</span><span id="line-709">                                for (Atom a : blocksetAtoms) {</span>
<span class="source-line-no">710</span><span id="line-710">                                        if (a != null)</span>
<span class="source-line-no">711</span><span id="line-711">                                                Calc.transform(a, blockTrans);</span>
<span class="source-line-no">712</span><span id="line-712">                                        transformedAtoms[transformedAtomsLength] = a;</span>
<span class="source-line-no">713</span><span id="line-713">                                        transformedAtomsLength++;</span>
<span class="source-line-no">714</span><span id="line-714">                                }</span>
<span class="source-line-no">715</span><span id="line-715">                        }</span>
<span class="source-line-no">716</span><span id="line-716">                        assert (transformedAtomsLength == alignment.length());</span>
<span class="source-line-no">717</span><span id="line-717"></span>
<span class="source-line-no">718</span><span id="line-718">                        transformed.add(transformedAtoms);</span>
<span class="source-line-no">719</span><span id="line-719">                }</span>
<span class="source-line-no">720</span><span id="line-720">                return transformed;</span>
<span class="source-line-no">721</span><span id="line-721">        }</span>
<span class="source-line-no">722</span><span id="line-722"></span>
<span class="source-line-no">723</span><span id="line-723">        /**</span>
<span class="source-line-no">724</span><span id="line-724">         * Calculate a List of alignment indicies that correspond to the core of a</span>
<span class="source-line-no">725</span><span id="line-725">         * Block, which means that all structures have a residue in that position.</span>
<span class="source-line-no">726</span><span id="line-726">         *</span>
<span class="source-line-no">727</span><span id="line-727">         * @param block</span>
<span class="source-line-no">728</span><span id="line-728">         *            alignment Block</span>
<span class="source-line-no">729</span><span id="line-729">         * @return List of positions in the core of the alignment</span>
<span class="source-line-no">730</span><span id="line-730">         */</span>
<span class="source-line-no">731</span><span id="line-731">        public static List&lt;Integer&gt; getCorePositions(Block block) {</span>
<span class="source-line-no">732</span><span id="line-732"></span>
<span class="source-line-no">733</span><span id="line-733">                List&lt;Integer&gt; corePositions = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">734</span><span id="line-734"></span>
<span class="source-line-no">735</span><span id="line-735">                for (int col = 0; col &lt; block.length(); col++) {</span>
<span class="source-line-no">736</span><span id="line-736">                        boolean core = true;</span>
<span class="source-line-no">737</span><span id="line-737">                        for (int str = 0; str &lt; block.size(); str++) {</span>
<span class="source-line-no">738</span><span id="line-738">                                if (block.getAlignRes().get(str).get(col) == null) {</span>
<span class="source-line-no">739</span><span id="line-739">                                        core = false;</span>
<span class="source-line-no">740</span><span id="line-740">                                        break;</span>
<span class="source-line-no">741</span><span id="line-741">                                }</span>
<span class="source-line-no">742</span><span id="line-742">                        }</span>
<span class="source-line-no">743</span><span id="line-743">                        if (core)</span>
<span class="source-line-no">744</span><span id="line-744">                                corePositions.add(col);</span>
<span class="source-line-no">745</span><span id="line-745">                }</span>
<span class="source-line-no">746</span><span id="line-746">                return corePositions;</span>
<span class="source-line-no">747</span><span id="line-747">        }</span>
<span class="source-line-no">748</span><span id="line-748"></span>
<span class="source-line-no">749</span><span id="line-749">        /**</span>
<span class="source-line-no">750</span><span id="line-750">         * Sort blocks so that the specified row is in sequential order. The sort</span>
<span class="source-line-no">751</span><span id="line-751">         * happens in place.</span>
<span class="source-line-no">752</span><span id="line-752">         *</span>
<span class="source-line-no">753</span><span id="line-753">         * @param blocks</span>
<span class="source-line-no">754</span><span id="line-754">         *            List of blocks</span>
<span class="source-line-no">755</span><span id="line-755">         * @param sortedIndex</span>
<span class="source-line-no">756</span><span id="line-756">         *            Index of the row to be sorted</span>
<span class="source-line-no">757</span><span id="line-757">         */</span>
<span class="source-line-no">758</span><span id="line-758">        public static void sortBlocks(List&lt;Block&gt; blocks, final int sortedIndex) {</span>
<span class="source-line-no">759</span><span id="line-759">                Collections.sort(blocks, new Comparator&lt;Block&gt;() {</span>
<span class="source-line-no">760</span><span id="line-760">                        @Override</span>
<span class="source-line-no">761</span><span id="line-761">                        public int compare(Block o1, Block o2) {</span>
<span class="source-line-no">762</span><span id="line-762">                                // Compare the first non-null residue of each block</span>
<span class="source-line-no">763</span><span id="line-763">                                List&lt;Integer&gt; alignres1 = o1.getAlignRes().get(sortedIndex);</span>
<span class="source-line-no">764</span><span id="line-764">                                List&lt;Integer&gt; alignres2 = o2.getAlignRes().get(sortedIndex);</span>
<span class="source-line-no">765</span><span id="line-765">                                Integer res1 = null;</span>
<span class="source-line-no">766</span><span id="line-766">                                Integer res2 = null;</span>
<span class="source-line-no">767</span><span id="line-767">                                for (Integer r : alignres1) {</span>
<span class="source-line-no">768</span><span id="line-768">                                        if (r != null) {</span>
<span class="source-line-no">769</span><span id="line-769">                                                res1 = r;</span>
<span class="source-line-no">770</span><span id="line-770">                                                break;</span>
<span class="source-line-no">771</span><span id="line-771">                                        }</span>
<span class="source-line-no">772</span><span id="line-772">                                }</span>
<span class="source-line-no">773</span><span id="line-773">                                for (Integer r : alignres2) {</span>
<span class="source-line-no">774</span><span id="line-774">                                        if (r != null) {</span>
<span class="source-line-no">775</span><span id="line-775">                                                res2 = r;</span>
<span class="source-line-no">776</span><span id="line-776">                                                break;</span>
<span class="source-line-no">777</span><span id="line-777">                                        }</span>
<span class="source-line-no">778</span><span id="line-778">                                }</span>
<span class="source-line-no">779</span><span id="line-779">                                return res1.compareTo(res2);</span>
<span class="source-line-no">780</span><span id="line-780">                        }</span>
<span class="source-line-no">781</span><span id="line-781">                });</span>
<span class="source-line-no">782</span><span id="line-782">        }</span>
<span class="source-line-no">783</span><span id="line-783"></span>
<span class="source-line-no">784</span><span id="line-784">        /**</span>
<span class="source-line-no">785</span><span id="line-785">         * Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid</span>
<span class="source-line-no">786</span><span id="line-786">         * residues. This method is only valid for protein structure alignments.</span>
<span class="source-line-no">787</span><span id="line-787">         *</span>
<span class="source-line-no">788</span><span id="line-788">         * @param msta</span>
<span class="source-line-no">789</span><span id="line-789">         *            Multiple Structure Alignment</span>
<span class="source-line-no">790</span><span id="line-790">         * @return MultipleSequenceAlignment of protein sequences</span>
<span class="source-line-no">791</span><span id="line-791">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">792</span><span id="line-792">         */</span>
<span class="source-line-no">793</span><span id="line-793">        public static MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; toProteinMSA(</span>
<span class="source-line-no">794</span><span id="line-794">                        MultipleAlignment msta) throws CompoundNotFoundException {</span>
<span class="source-line-no">795</span><span id="line-795"></span>
<span class="source-line-no">796</span><span id="line-796">                // Check that the alignment is of protein structures</span>
<span class="source-line-no">797</span><span id="line-797">                Group g = msta.getAtomArrays().get(0)[0].getGroup();</span>
<span class="source-line-no">798</span><span id="line-798">                if (!(g instanceof AminoAcid)) {</span>
<span class="source-line-no">799</span><span id="line-799">                        throw new IllegalArgumentException(</span>
<span class="source-line-no">800</span><span id="line-800">                                        "Cannot convert to multiple sequence alignment: "</span>
<span class="source-line-no">801</span><span id="line-801">                                                        + "the structures aligned are not proteins");</span>
<span class="source-line-no">802</span><span id="line-802">                }</span>
<span class="source-line-no">803</span><span id="line-803"></span>
<span class="source-line-no">804</span><span id="line-804">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = new MultipleSequenceAlignment&lt;&gt;();</span>
<span class="source-line-no">805</span><span id="line-805"></span>
<span class="source-line-no">806</span><span id="line-806">                Map&lt;String, Integer&gt; uniqueID = new HashMap&lt;&gt;();</span>
<span class="source-line-no">807</span><span id="line-807">                List&lt;String&gt; seqs = getSequenceAlignment(msta);</span>
<span class="source-line-no">808</span><span id="line-808">                for (int i = 0; i &lt; msta.size(); i++) {</span>
<span class="source-line-no">809</span><span id="line-809">                        // Make sure the identifiers are unique (required by AccessionID)</span>
<span class="source-line-no">810</span><span id="line-810">                        String id = msta.getStructureIdentifier(i).toString();</span>
<span class="source-line-no">811</span><span id="line-811">                        if (uniqueID.containsKey(id)) {</span>
<span class="source-line-no">812</span><span id="line-812">                                uniqueID.put(id, uniqueID.get(id) + 1);</span>
<span class="source-line-no">813</span><span id="line-813">                                id += "_" + uniqueID.get(id);</span>
<span class="source-line-no">814</span><span id="line-814">                        } else</span>
<span class="source-line-no">815</span><span id="line-815">                                uniqueID.put(id, 1);</span>
<span class="source-line-no">816</span><span id="line-816"></span>
<span class="source-line-no">817</span><span id="line-817">                        AccessionID acc = new AccessionID(id);</span>
<span class="source-line-no">818</span><span id="line-818">                        ProteinSequence pseq = new ProteinSequence(seqs.get(i));</span>
<span class="source-line-no">819</span><span id="line-819">                        pseq.setAccession(acc);</span>
<span class="source-line-no">820</span><span id="line-820">                        msa.addAlignedSequence(pseq);</span>
<span class="source-line-no">821</span><span id="line-821">                }</span>
<span class="source-line-no">822</span><span id="line-822">                return msa;</span>
<span class="source-line-no">823</span><span id="line-823">        }</span>
<span class="source-line-no">824</span><span id="line-824"></span>
<span class="source-line-no">825</span><span id="line-825">        public static Structure toMultimodelStructure(MultipleAlignment multAln, List&lt;Atom[]&gt; transformedAtoms) throws StructureException {</span>
<span class="source-line-no">826</span><span id="line-826">                PDBHeader header = new PDBHeader();</span>
<span class="source-line-no">827</span><span id="line-827">                String title = multAln.getEnsemble().getAlgorithmName() + " V."</span>
<span class="source-line-no">828</span><span id="line-828">                                + multAln.getEnsemble().getVersion() + " : ";</span>
<span class="source-line-no">829</span><span id="line-829"></span>
<span class="source-line-no">830</span><span id="line-830">                for (StructureIdentifier name : multAln.getEnsemble()</span>
<span class="source-line-no">831</span><span id="line-831">                                .getStructureIdentifiers()) {</span>
<span class="source-line-no">832</span><span id="line-832">                        title += name.getIdentifier() + " ";</span>
<span class="source-line-no">833</span><span id="line-833">                }</span>
<span class="source-line-no">834</span><span id="line-834">                Structure artificial = getAlignedStructure(transformedAtoms);</span>
<span class="source-line-no">835</span><span id="line-835"></span>
<span class="source-line-no">836</span><span id="line-836">                artificial.setPDBHeader(header);</span>
<span class="source-line-no">837</span><span id="line-837">                header.setTitle(title);</span>
<span class="source-line-no">838</span><span id="line-838">                return artificial;</span>
<span class="source-line-no">839</span><span id="line-839">        }</span>
<span class="source-line-no">840</span><span id="line-840"></span>
<span class="source-line-no">841</span><span id="line-841">        /**</span>
<span class="source-line-no">842</span><span id="line-842">         * Get an artificial Structure containing a different model for every</span>
<span class="source-line-no">843</span><span id="line-843">         * input structure, so that the alignment result can be viewed in Jmol.</span>
<span class="source-line-no">844</span><span id="line-844">         * The Atoms have to be rotated beforehand.</span>
<span class="source-line-no">845</span><span id="line-845">         *</span>
<span class="source-line-no">846</span><span id="line-846">         * @param atomArrays an array of Atoms for every aligned structure</span>
<span class="source-line-no">847</span><span id="line-847">         * @return a structure object containing a set of models,</span>
<span class="source-line-no">848</span><span id="line-848">         *                      one for each input array of Atoms.</span>
<span class="source-line-no">849</span><span id="line-849">         * @throws StructureException</span>
<span class="source-line-no">850</span><span id="line-850">         */</span>
<span class="source-line-no">851</span><span id="line-851">        public static final Structure getAlignedStructure(List&lt;Atom[]&gt; atomArrays)</span>
<span class="source-line-no">852</span><span id="line-852">                        throws StructureException {</span>
<span class="source-line-no">853</span><span id="line-853"></span>
<span class="source-line-no">854</span><span id="line-854">                Structure s = new StructureImpl();</span>
<span class="source-line-no">855</span><span id="line-855">                for (int i=0; i&lt;atomArrays.size(); i++){</span>
<span class="source-line-no">856</span><span id="line-856">                        List&lt;Chain&gt; model = AlignmentTools.getAlignedModel(atomArrays.get(i));</span>
<span class="source-line-no">857</span><span id="line-857">                        s.addModel(model);</span>
<span class="source-line-no">858</span><span id="line-858">                }</span>
<span class="source-line-no">859</span><span id="line-859">                return s;</span>
<span class="source-line-no">860</span><span id="line-860">        }</span>
<span class="source-line-no">861</span><span id="line-861"></span>
<span class="source-line-no">862</span><span id="line-862">        /**</span>
<span class="source-line-no">863</span><span id="line-863">         * Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of</span>
<span class="source-line-no">864</span><span id="line-864">         * the matrix contains the RMSD between structures i and j.</span>
<span class="source-line-no">865</span><span id="line-865">         *</span>
<span class="source-line-no">866</span><span id="line-866">         * @param msa</span>
<span class="source-line-no">867</span><span id="line-867">         *            Multiple Structure Alignment</span>
<span class="source-line-no">868</span><span id="line-868">         * @return Matrix of RMSD with size the number of structures squared</span>
<span class="source-line-no">869</span><span id="line-869">         */</span>
<span class="source-line-no">870</span><span id="line-870">        public static Matrix getRMSDMatrix(MultipleAlignment msa) {</span>
<span class="source-line-no">871</span><span id="line-871"></span>
<span class="source-line-no">872</span><span id="line-872">                Matrix rmsdMat = new Matrix(msa.size(), msa.size());</span>
<span class="source-line-no">873</span><span id="line-873">                List&lt;Atom[]&gt; superposed = transformAtoms(msa);</span>
<span class="source-line-no">874</span><span id="line-874"></span>
<span class="source-line-no">875</span><span id="line-875">                for (int i = 0; i &lt; msa.size(); i++) {</span>
<span class="source-line-no">876</span><span id="line-876">                        for (int j = i; j &lt; msa.size(); j++) {</span>
<span class="source-line-no">877</span><span id="line-877">                                if (i == j)</span>
<span class="source-line-no">878</span><span id="line-878">                                        rmsdMat.set(i, j, 0.0);</span>
<span class="source-line-no">879</span><span id="line-879">                                List&lt;Atom[]&gt; compared = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">880</span><span id="line-880">                                compared.add(superposed.get(i));</span>
<span class="source-line-no">881</span><span id="line-881">                                compared.add(superposed.get(j));</span>
<span class="source-line-no">882</span><span id="line-882">                                double rmsd = MultipleAlignmentScorer.getRMSD(compared);</span>
<span class="source-line-no">883</span><span id="line-883">                                rmsdMat.set(i, j, rmsd);</span>
<span class="source-line-no">884</span><span id="line-884">                                rmsdMat.set(j, i, rmsd);</span>
<span class="source-line-no">885</span><span id="line-885">                        }</span>
<span class="source-line-no">886</span><span id="line-886">                }</span>
<span class="source-line-no">887</span><span id="line-887">                return rmsdMat;</span>
<span class="source-line-no">888</span><span id="line-888">        }</span>
<span class="source-line-no">889</span><span id="line-889"></span>
<span class="source-line-no">890</span><span id="line-890">        /**</span>
<span class="source-line-no">891</span><span id="line-891">         * Calculate a phylogenetic tree of the MultipleAlignment using Kimura</span>
<span class="source-line-no">892</span><span id="line-892">         * distances and the Neighbor Joining algorithm from forester.</span>
<span class="source-line-no">893</span><span id="line-893">         *</span>
<span class="source-line-no">894</span><span id="line-894">         * @param msta</span>
<span class="source-line-no">895</span><span id="line-895">         *            MultipleAlignment of protein structures</span>
<span class="source-line-no">896</span><span id="line-896">         * @return Phylogeny phylogenetic tree</span>
<span class="source-line-no">897</span><span id="line-897">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">898</span><span id="line-898">         * @throws IOException</span>
<span class="source-line-no">899</span><span id="line-899">         */</span>
<span class="source-line-no">900</span><span id="line-900">        public static Phylogeny getKimuraTree(MultipleAlignment msta)</span>
<span class="source-line-no">901</span><span id="line-901">                        throws CompoundNotFoundException, IOException {</span>
<span class="source-line-no">902</span><span id="line-902">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = MultipleAlignmentTools</span>
<span class="source-line-no">903</span><span id="line-903">                                .toProteinMSA(msta);</span>
<span class="source-line-no">904</span><span id="line-904">                BasicSymmetricalDistanceMatrix distmat = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator</span>
<span class="source-line-no">905</span><span id="line-905">                                .kimuraDistance(msa);</span>
<span class="source-line-no">906</span><span id="line-906">                Phylogeny tree = TreeConstructor.distanceTree(distmat,</span>
<span class="source-line-no">907</span><span id="line-907">                                TreeConstructorType.NJ);</span>
<span class="source-line-no">908</span><span id="line-908">                tree.setName("Kimura Tree");</span>
<span class="source-line-no">909</span><span id="line-909">                return tree;</span>
<span class="source-line-no">910</span><span id="line-910">        }</span>
<span class="source-line-no">911</span><span id="line-911"></span>
<span class="source-line-no">912</span><span id="line-912">        /**</span>
<span class="source-line-no">913</span><span id="line-913">         * Calculate a phylogenetic tree of the MultipleAlignment using</span>
<span class="source-line-no">914</span><span id="line-914">         * dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for</span>
<span class="source-line-no">915</span><span id="line-915">         * distantly related proteins, structure-derived) and the Neighbor Joining</span>
<span class="source-line-no">916</span><span id="line-916">         * algorithm from forester.</span>
<span class="source-line-no">917</span><span id="line-917">         *</span>
<span class="source-line-no">918</span><span id="line-918">         * @param msta</span>
<span class="source-line-no">919</span><span id="line-919">         *            MultipleAlignment of protein structures</span>
<span class="source-line-no">920</span><span id="line-920">         * @return Phylogeny phylogenetic tree</span>
<span class="source-line-no">921</span><span id="line-921">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">922</span><span id="line-922">         * @throws IOException</span>
<span class="source-line-no">923</span><span id="line-923">         */</span>
<span class="source-line-no">924</span><span id="line-924">        public static Phylogeny getHSDMTree(MultipleAlignment msta)</span>
<span class="source-line-no">925</span><span id="line-925">                        throws CompoundNotFoundException, IOException {</span>
<span class="source-line-no">926</span><span id="line-926">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa = MultipleAlignmentTools</span>
<span class="source-line-no">927</span><span id="line-927">                                .toProteinMSA(msta);</span>
<span class="source-line-no">928</span><span id="line-928">                BasicSymmetricalDistanceMatrix distmat = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator</span>
<span class="source-line-no">929</span><span id="line-929">                                .dissimilarityScore(msa, SubstitutionMatrixHelper.getAminoAcidSubstitutionMatrix("PRLA000102"));</span>
<span class="source-line-no">930</span><span id="line-930">                Phylogeny tree = TreeConstructor.distanceTree(distmat,</span>
<span class="source-line-no">931</span><span id="line-931">                                TreeConstructorType.NJ);</span>
<span class="source-line-no">932</span><span id="line-932">                tree.setName("HSDM Tree");</span>
<span class="source-line-no">933</span><span id="line-933">                return tree;</span>
<span class="source-line-no">934</span><span id="line-934">        }</span>
<span class="source-line-no">935</span><span id="line-935"></span>
<span class="source-line-no">936</span><span id="line-936">        /**</span>
<span class="source-line-no">937</span><span id="line-937">         * Calculate a phylogenetic tree of the MultipleAlignment using RMSD</span>
<span class="source-line-no">938</span><span id="line-938">         * distances and the Neighbor Joining algorithm from forester.</span>
<span class="source-line-no">939</span><span id="line-939">         *</span>
<span class="source-line-no">940</span><span id="line-940">         * @param msta</span>
<span class="source-line-no">941</span><span id="line-941">         *            MultipleAlignment of protein structures</span>
<span class="source-line-no">942</span><span id="line-942">         * @return Phylogeny phylogenetic tree</span>
<span class="source-line-no">943</span><span id="line-943">         */</span>
<span class="source-line-no">944</span><span id="line-944">        public static Phylogeny getStructuralTree(MultipleAlignment msta) {</span>
<span class="source-line-no">945</span><span id="line-945">                double[][] rmsdMat = MultipleAlignmentTools.getRMSDMatrix(msta)</span>
<span class="source-line-no">946</span><span id="line-946">                                .getArray();</span>
<span class="source-line-no">947</span><span id="line-947">                BasicSymmetricalDistanceMatrix rmsdDist = (BasicSymmetricalDistanceMatrix) DistanceMatrixCalculator</span>
<span class="source-line-no">948</span><span id="line-948">                                .structuralDistance(rmsdMat, 1, 5, 0.4);</span>
<span class="source-line-no">949</span><span id="line-949">                // Set the identifiers of the matrix</span>
<span class="source-line-no">950</span><span id="line-950">                Map&lt;String, Integer&gt; alreadySeen = new HashMap&lt;&gt;();</span>
<span class="source-line-no">951</span><span id="line-951">                for (int i = 0; i &lt; msta.size(); i++) {</span>
<span class="source-line-no">952</span><span id="line-952">                        // Make sure the identifiers are unique</span>
<span class="source-line-no">953</span><span id="line-953">                        String id = msta.getStructureIdentifier(i).toString();</span>
<span class="source-line-no">954</span><span id="line-954">                        if (alreadySeen.containsKey(id)) {</span>
<span class="source-line-no">955</span><span id="line-955">                                alreadySeen.put(id, alreadySeen.get(id) + 1);</span>
<span class="source-line-no">956</span><span id="line-956">                                id += "_" + alreadySeen.get(id);</span>
<span class="source-line-no">957</span><span id="line-957">                        } else</span>
<span class="source-line-no">958</span><span id="line-958">                                alreadySeen.put(id, 1);</span>
<span class="source-line-no">959</span><span id="line-959">                        rmsdDist.setIdentifier(i, id);</span>
<span class="source-line-no">960</span><span id="line-960">                }</span>
<span class="source-line-no">961</span><span id="line-961">                Phylogeny tree = TreeConstructor.distanceTree(rmsdDist,</span>
<span class="source-line-no">962</span><span id="line-962">                                TreeConstructorType.NJ);</span>
<span class="source-line-no">963</span><span id="line-963">                tree.setName("Structural Tree");</span>
<span class="source-line-no">964</span><span id="line-964">                return tree;</span>
<span class="source-line-no">965</span><span id="line-965">        }</span>
<span class="source-line-no">966</span><span id="line-966"></span>
<span class="source-line-no">967</span><span id="line-967">        /**</span>
<span class="source-line-no">968</span><span id="line-968">         * Convert an MSA into a matrix of equivalent residues.</span>
<span class="source-line-no">969</span><span id="line-969">         *</span>
<span class="source-line-no">970</span><span id="line-970">         * This concatenates all blocks, meaning that the indices might not be</span>
<span class="source-line-no">971</span><span id="line-971">         * sequential.</span>
<span class="source-line-no">972</span><span id="line-972">         *</span>
<span class="source-line-no">973</span><span id="line-973">         * Indices should be consistent with `msa.getAtomArrays()`.</span>
<span class="source-line-no">974</span><span id="line-974">         * @param msa Multiple alignment</span>
<span class="source-line-no">975</span><span id="line-975">         * @param coreOnly Include only core (ungapped) columns. Otherwise gaps are</span>
<span class="source-line-no">976</span><span id="line-976">         *        represented with null.</span>
<span class="source-line-no">977</span><span id="line-977">         * @return</span>
<span class="source-line-no">978</span><span id="line-978">         */</span>
<span class="source-line-no">979</span><span id="line-979">        public static List&lt;List&lt;Integer&gt;&gt; getEquivalentResidues(MultipleAlignment msa, boolean coreOnly) {</span>
<span class="source-line-no">980</span><span id="line-980">                List&lt;List&lt;Integer&gt;&gt; eqr = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">981</span><span id="line-981">                for (int str = 0; str &lt; msa.size(); str++) {</span>
<span class="source-line-no">982</span><span id="line-982">                        eqr.add(new ArrayList&lt;&gt;());</span>
<span class="source-line-no">983</span><span id="line-983">                }</span>
<span class="source-line-no">984</span><span id="line-984"></span>
<span class="source-line-no">985</span><span id="line-985">                for(Block block: msa.getBlocks()) {</span>
<span class="source-line-no">986</span><span id="line-986">                        List&lt;List&lt;Integer&gt;&gt; aln = block.getAlignRes();</span>
<span class="source-line-no">987</span><span id="line-987">                        for (int col = 0; col &lt; block.length(); col++) {</span>
<span class="source-line-no">988</span><span id="line-988">                                // skip non-core columns</span>
<span class="source-line-no">989</span><span id="line-989">                                if(coreOnly) {</span>
<span class="source-line-no">990</span><span id="line-990">                                        boolean core = true;</span>
<span class="source-line-no">991</span><span id="line-991">                                        for (int str = 0; str &lt; block.size(); str++) {</span>
<span class="source-line-no">992</span><span id="line-992">                                                if (aln.get(str).get(col) == null) {</span>
<span class="source-line-no">993</span><span id="line-993">                                                        core = false;</span>
<span class="source-line-no">994</span><span id="line-994">                                                        break;</span>
<span class="source-line-no">995</span><span id="line-995">                                                }</span>
<span class="source-line-no">996</span><span id="line-996">                                        }</span>
<span class="source-line-no">997</span><span id="line-997">                                        if(!core) {</span>
<span class="source-line-no">998</span><span id="line-998">                                                continue;</span>
<span class="source-line-no">999</span><span id="line-999">                                        }</span>
<span class="source-line-no">1000</span><span id="line-1000">                                }</span>
<span class="source-line-no">1001</span><span id="line-1001">                                // add column to eqr</span>
<span class="source-line-no">1002</span><span id="line-1002">                                for (int str = 0; str &lt; block.size(); str++) {</span>
<span class="source-line-no">1003</span><span id="line-1003">                                        eqr.get(str).add(aln.get(str).get(col));</span>
<span class="source-line-no">1004</span><span id="line-1004">                                }</span>
<span class="source-line-no">1005</span><span id="line-1005">                        }</span>
<span class="source-line-no">1006</span><span id="line-1006">                }</span>
<span class="source-line-no">1007</span><span id="line-1007">                return eqr;</span>
<span class="source-line-no">1008</span><span id="line-1008">        }</span>
<span class="source-line-no">1009</span><span id="line-1009">}</span>




























































</pre>
</div>
</main>
</body>
</html>
