<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.data.sequence, class: SequenceUtil">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> * @(#)SequenceUtil.java 1.0 September 2009</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * Copyright (c) 2009 Peter Troshin</span>
<span class="source-line-no">005</span><span id="line-5"> *</span>
<span class="source-line-no">006</span><span id="line-6"> *        BioJava development code</span>
<span class="source-line-no">007</span><span id="line-7"> *</span>
<span class="source-line-no">008</span><span id="line-8"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">009</span><span id="line-9"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">010</span><span id="line-10"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">011</span><span id="line-11"> * see:</span>
<span class="source-line-no">012</span><span id="line-12"> *</span>
<span class="source-line-no">013</span><span id="line-13"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">014</span><span id="line-14"> *</span>
<span class="source-line-no">015</span><span id="line-15"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">016</span><span id="line-16"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">019</span><span id="line-19"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">020</span><span id="line-20"> * at:</span>
<span class="source-line-no">021</span><span id="line-21"> *</span>
<span class="source-line-no">022</span><span id="line-22"> *      http://www.biojava.org/</span>
<span class="source-line-no">023</span><span id="line-23"> *</span>
<span class="source-line-no">024</span><span id="line-24"> */</span>
<span class="source-line-no">025</span><span id="line-25"></span>
<span class="source-line-no">026</span><span id="line-26">package org.biojava.nbio.data.sequence;</span>
<span class="source-line-no">027</span><span id="line-27"></span>
<span class="source-line-no">028</span><span id="line-28">import org.slf4j.Logger;</span>
<span class="source-line-no">029</span><span id="line-29">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">030</span><span id="line-30"></span>
<span class="source-line-no">031</span><span id="line-31">import java.io.*;</span>
<span class="source-line-no">032</span><span id="line-32">import java.util.ArrayList;</span>
<span class="source-line-no">033</span><span id="line-33">import java.util.List;</span>
<span class="source-line-no">034</span><span id="line-34">import java.util.regex.Matcher;</span>
<span class="source-line-no">035</span><span id="line-35">import java.util.regex.Pattern;</span>
<span class="source-line-no">036</span><span id="line-36"></span>
<span class="source-line-no">037</span><span id="line-37">/**</span>
<span class="source-line-no">038</span><span id="line-38"> * Utility class for operations on sequences</span>
<span class="source-line-no">039</span><span id="line-39"> *</span>
<span class="source-line-no">040</span><span id="line-40"> * @author Peter Troshin</span>
<span class="source-line-no">041</span><span id="line-41"> * @version 1.0</span>
<span class="source-line-no">042</span><span id="line-42"> * @since 3.0.2</span>
<span class="source-line-no">043</span><span id="line-43"> */</span>
<span class="source-line-no">044</span><span id="line-44">public final class SequenceUtil {</span>
<span class="source-line-no">045</span><span id="line-45"></span>
<span class="source-line-no">046</span><span id="line-46">        private static final Logger logger = LoggerFactory.getLogger(SequenceUtil.class);</span>
<span class="source-line-no">047</span><span id="line-47"></span>
<span class="source-line-no">048</span><span id="line-48">        /**</span>
<span class="source-line-no">049</span><span id="line-49">         * A whitespace character: [\t\n\x0B\f\r]</span>
<span class="source-line-no">050</span><span id="line-50">         */</span>
<span class="source-line-no">051</span><span id="line-51">        public static final Pattern WHITE_SPACE = Pattern.compile("\\s");</span>
<span class="source-line-no">052</span><span id="line-52"></span>
<span class="source-line-no">053</span><span id="line-53">        /**</span>
<span class="source-line-no">054</span><span id="line-54">         * A digit</span>
<span class="source-line-no">055</span><span id="line-55">         */</span>
<span class="source-line-no">056</span><span id="line-56">        public static final Pattern DIGIT = Pattern.compile("\\d");</span>
<span class="source-line-no">057</span><span id="line-57"></span>
<span class="source-line-no">058</span><span id="line-58">        /**</span>
<span class="source-line-no">059</span><span id="line-59">         * Non word</span>
<span class="source-line-no">060</span><span id="line-60">         */</span>
<span class="source-line-no">061</span><span id="line-61">        public static final Pattern NONWORD = Pattern.compile("\\W");</span>
<span class="source-line-no">062</span><span id="line-62"></span>
<span class="source-line-no">063</span><span id="line-63">        /**</span>
<span class="source-line-no">064</span><span id="line-64">         * Valid Amino acids</span>
<span class="source-line-no">065</span><span id="line-65">         */</span>
<span class="source-line-no">066</span><span id="line-66">        public static final Pattern AA = Pattern.compile("[ARNDCQEGHILKMFPSTWYVUO]+",</span>
<span class="source-line-no">067</span><span id="line-67">                Pattern.CASE_INSENSITIVE);</span>
<span class="source-line-no">068</span><span id="line-68"></span>
<span class="source-line-no">069</span><span id="line-69">        /**</span>
<span class="source-line-no">070</span><span id="line-70">         * inversion of AA pattern</span>
<span class="source-line-no">071</span><span id="line-71">         */</span>
<span class="source-line-no">072</span><span id="line-72">        public static final Pattern NON_AA = Pattern.compile(</span>
<span class="source-line-no">073</span><span id="line-73">                "[^ARNDCQEGHILKMFPSTWYVXUO]+", Pattern.CASE_INSENSITIVE);</span>
<span class="source-line-no">074</span><span id="line-74"></span>
<span class="source-line-no">075</span><span id="line-75">        /**</span>
<span class="source-line-no">076</span><span id="line-76">         * Same as AA pattern but with one additional letters - X</span>
<span class="source-line-no">077</span><span id="line-77">         */</span>
<span class="source-line-no">078</span><span id="line-78">        public static final Pattern AMBIGUOUS_AA = Pattern.compile(</span>
<span class="source-line-no">079</span><span id="line-79">                "[ARNDCQEGHILKMFPSTWYVXUO]+", Pattern.CASE_INSENSITIVE);</span>
<span class="source-line-no">080</span><span id="line-80"></span>
<span class="source-line-no">081</span><span id="line-81">        /**</span>
<span class="source-line-no">082</span><span id="line-82">         * Nucleotides a, t, g, c, u</span>
<span class="source-line-no">083</span><span id="line-83">         */</span>
<span class="source-line-no">084</span><span id="line-84">        public static final Pattern NUCLEOTIDE = Pattern.compile("[AGTCU]+",</span>
<span class="source-line-no">085</span><span id="line-85">                Pattern.CASE_INSENSITIVE);</span>
<span class="source-line-no">086</span><span id="line-86"></span>
<span class="source-line-no">087</span><span id="line-87">        /**</span>
<span class="source-line-no">088</span><span id="line-88">         * Ambiguous nucleotide</span>
<span class="source-line-no">089</span><span id="line-89">         */</span>
<span class="source-line-no">090</span><span id="line-90">        public static final Pattern AMBIGUOUS_NUCLEOTIDE = Pattern.compile(</span>
<span class="source-line-no">091</span><span id="line-91">                "[AGTCRYMKSWHBVDNU]+", Pattern.CASE_INSENSITIVE); // see IUPAC</span>
<span class="source-line-no">092</span><span id="line-92">        /**</span>
<span class="source-line-no">093</span><span id="line-93">         * Non nucleotide</span>
<span class="source-line-no">094</span><span id="line-94">         */</span>
<span class="source-line-no">095</span><span id="line-95">        public static final Pattern NON_NUCLEOTIDE = Pattern.compile("[^AGTCU]+",</span>
<span class="source-line-no">096</span><span id="line-96">                Pattern.CASE_INSENSITIVE);</span>
<span class="source-line-no">097</span><span id="line-97"></span>
<span class="source-line-no">098</span><span id="line-98">        private SequenceUtil() {</span>
<span class="source-line-no">099</span><span id="line-99">        } // utility class, no instantiation</span>
<span class="source-line-no">100</span><span id="line-100"></span>
<span class="source-line-no">101</span><span id="line-101">        /*</span>
<span class="source-line-no">102</span><span id="line-102">         * public static void write_PirSeq(OutputStream os, FastaSequence seq)</span>
<span class="source-line-no">103</span><span id="line-103">         * throws IOException { BufferedWriter pir_out = new BufferedWriter(new</span>
<span class="source-line-no">104</span><span id="line-104">         * OutputStreamWriter(os)); pir_out.write("&gt;P1;" + seq.getId() +</span>
<span class="source-line-no">105</span><span id="line-105">         * SysPrefs.newlinechar); pir_out.write(seq.getSequence() +</span>
<span class="source-line-no">106</span><span id="line-106">         * SysPrefs.newlinechar); pir_out.close(); }</span>
<span class="source-line-no">107</span><span id="line-107">         *</span>
<span class="source-line-no">108</span><span id="line-108">         * public static void write_FastaSeq(OutputStream os, FastaSequence seq)</span>
<span class="source-line-no">109</span><span id="line-109">         * throws IOException { BufferedWriter fasta_out = new BufferedWriter( new</span>
<span class="source-line-no">110</span><span id="line-110">         * OutputStreamWriter(os)); fasta_out.write("&gt;" + seq.getId() +</span>
<span class="source-line-no">111</span><span id="line-111">         * SysPrefs.newlinechar); fasta_out.write(seq.getSequence() +</span>
<span class="source-line-no">112</span><span id="line-112">         * SysPrefs.newlinechar); fasta_out.close(); }</span>
<span class="source-line-no">113</span><span id="line-113">         */</span>
<span class="source-line-no">114</span><span id="line-114"></span>
<span class="source-line-no">115</span><span id="line-115">        /**</span>
<span class="source-line-no">116</span><span id="line-116">         * @return true is the sequence contains only letters a,c, t, g, u</span>
<span class="source-line-no">117</span><span id="line-117">         */</span>
<span class="source-line-no">118</span><span id="line-118">        public static boolean isNucleotideSequence(final FastaSequence s) {</span>
<span class="source-line-no">119</span><span id="line-119">        return SequenceUtil.isNonAmbNucleotideSequence(s.getSequence());</span>
<span class="source-line-no">120</span><span id="line-120">        }</span>
<span class="source-line-no">121</span><span id="line-121"></span>
<span class="source-line-no">122</span><span id="line-122">        /**</span>
<span class="source-line-no">123</span><span id="line-123">         * Ambiguous DNA chars : AGTCRYMKSWHBVDN // differs from protein in only one</span>
<span class="source-line-no">124</span><span id="line-124">         * (!) - B char</span>
<span class="source-line-no">125</span><span id="line-125">         */</span>
<span class="source-line-no">126</span><span id="line-126">        public static boolean isNonAmbNucleotideSequence(String sequence) {</span>
<span class="source-line-no">127</span><span id="line-127">        sequence = SequenceUtil.cleanSequence(sequence);</span>
<span class="source-line-no">128</span><span id="line-128">        if (SequenceUtil.DIGIT.matcher(sequence).find()) {</span>
<span class="source-line-no">129</span><span id="line-129">                return false;</span>
<span class="source-line-no">130</span><span id="line-130">        }</span>
<span class="source-line-no">131</span><span id="line-131">        if (SequenceUtil.NON_NUCLEOTIDE.matcher(sequence).find()) {</span>
<span class="source-line-no">132</span><span id="line-132">                return false;</span>
<span class="source-line-no">133</span><span id="line-133">                /*</span>
<span class="source-line-no">134</span><span id="line-134">                 * System.out.format("I found the text starting at " +</span>
<span class="source-line-no">135</span><span id="line-135">                 * "index %d and ending at index %d.%n", nonDNAmatcher .start(),</span>
<span class="source-line-no">136</span><span id="line-136">                 * nonDNAmatcher.end());</span>
<span class="source-line-no">137</span><span id="line-137">                 */</span>
<span class="source-line-no">138</span><span id="line-138">        }</span>
<span class="source-line-no">139</span><span id="line-139">        final Matcher DNAmatcher = SequenceUtil.NUCLEOTIDE.matcher(sequence);</span>
<span class="source-line-no">140</span><span id="line-140">        return DNAmatcher.find();</span>
<span class="source-line-no">141</span><span id="line-141">        }</span>
<span class="source-line-no">142</span><span id="line-142"></span>
<span class="source-line-no">143</span><span id="line-143">        /**</span>
<span class="source-line-no">144</span><span id="line-144">         * Removes all whitespace chars in the sequence string</span>
<span class="source-line-no">145</span><span id="line-145">         *</span>
<span class="source-line-no">146</span><span id="line-146">         * @param sequence</span>
<span class="source-line-no">147</span><span id="line-147">         * @return cleaned up sequence</span>
<span class="source-line-no">148</span><span id="line-148">         */</span>
<span class="source-line-no">149</span><span id="line-149">        public static String cleanSequence(String sequence) {</span>
<span class="source-line-no">150</span><span id="line-150">        assert sequence != null;</span>
<span class="source-line-no">151</span><span id="line-151">        final Matcher m = SequenceUtil.WHITE_SPACE.matcher(sequence);</span>
<span class="source-line-no">152</span><span id="line-152">        sequence = m.replaceAll("").toUpperCase();</span>
<span class="source-line-no">153</span><span id="line-153">        return sequence;</span>
<span class="source-line-no">154</span><span id="line-154">        }</span>
<span class="source-line-no">155</span><span id="line-155"></span>
<span class="source-line-no">156</span><span id="line-156">        /**</span>
<span class="source-line-no">157</span><span id="line-157">         * Removes all special characters and digits as well as whitespace chars</span>
<span class="source-line-no">158</span><span id="line-158">         * from the sequence</span>
<span class="source-line-no">159</span><span id="line-159">         *</span>
<span class="source-line-no">160</span><span id="line-160">         * @param sequence</span>
<span class="source-line-no">161</span><span id="line-161">         * @return cleaned up sequence</span>
<span class="source-line-no">162</span><span id="line-162">         */</span>
<span class="source-line-no">163</span><span id="line-163">        public static String deepCleanSequence(String sequence) {</span>
<span class="source-line-no">164</span><span id="line-164">        sequence = SequenceUtil.cleanSequence(sequence);</span>
<span class="source-line-no">165</span><span id="line-165">        sequence = SequenceUtil.DIGIT.matcher(sequence).replaceAll("");</span>
<span class="source-line-no">166</span><span id="line-166">        sequence = SequenceUtil.NONWORD.matcher(sequence).replaceAll("");</span>
<span class="source-line-no">167</span><span id="line-167">        final Pattern othernonSeqChars = Pattern.compile("[_-]+");</span>
<span class="source-line-no">168</span><span id="line-168">        sequence = othernonSeqChars.matcher(sequence).replaceAll("");</span>
<span class="source-line-no">169</span><span id="line-169">        return sequence;</span>
<span class="source-line-no">170</span><span id="line-170">        }</span>
<span class="source-line-no">171</span><span id="line-171"></span>
<span class="source-line-no">172</span><span id="line-172">        /**</span>
<span class="source-line-no">173</span><span id="line-173">         *</span>
<span class="source-line-no">174</span><span id="line-174">         * @param sequence</span>
<span class="source-line-no">175</span><span id="line-175">         * @return true is the sequence is a protein sequence, false overwise</span>
<span class="source-line-no">176</span><span id="line-176">         */</span>
<span class="source-line-no">177</span><span id="line-177">        public static boolean isProteinSequence(String sequence) {</span>
<span class="source-line-no">178</span><span id="line-178">        sequence = SequenceUtil.cleanSequence(sequence);</span>
<span class="source-line-no">179</span><span id="line-179">        if (SequenceUtil.isNonAmbNucleotideSequence(sequence)) {</span>
<span class="source-line-no">180</span><span id="line-180">                return false;</span>
<span class="source-line-no">181</span><span id="line-181">        }</span>
<span class="source-line-no">182</span><span id="line-182">        if (SequenceUtil.DIGIT.matcher(sequence).find()) {</span>
<span class="source-line-no">183</span><span id="line-183">                return false;</span>
<span class="source-line-no">184</span><span id="line-184">        }</span>
<span class="source-line-no">185</span><span id="line-185">        if (SequenceUtil.NON_AA.matcher(sequence).find()) {</span>
<span class="source-line-no">186</span><span id="line-186">                logger.info("Found non aa: {}", sequence);</span>
<span class="source-line-no">187</span><span id="line-187">                return false;</span>
<span class="source-line-no">188</span><span id="line-188">        }</span>
<span class="source-line-no">189</span><span id="line-189">        final Matcher protmatcher = SequenceUtil.AA.matcher(sequence);</span>
<span class="source-line-no">190</span><span id="line-190">        return protmatcher.find();</span>
<span class="source-line-no">191</span><span id="line-191">        }</span>
<span class="source-line-no">192</span><span id="line-192"></span>
<span class="source-line-no">193</span><span id="line-193">        /**</span>
<span class="source-line-no">194</span><span id="line-194">         * Check whether the sequence confirms to amboguous protein sequence</span>
<span class="source-line-no">195</span><span id="line-195">         *</span>
<span class="source-line-no">196</span><span id="line-196">         * @param sequence</span>
<span class="source-line-no">197</span><span id="line-197">         * @return return true only if the sequence if ambiguous protein sequence</span>
<span class="source-line-no">198</span><span id="line-198">         *         Return false otherwise. e.g. if the sequence is non-ambiguous</span>
<span class="source-line-no">199</span><span id="line-199">         *         protein or DNA</span>
<span class="source-line-no">200</span><span id="line-200">         */</span>
<span class="source-line-no">201</span><span id="line-201">        public static boolean isAmbiguosProtein(String sequence) {</span>
<span class="source-line-no">202</span><span id="line-202">        sequence = SequenceUtil.cleanSequence(sequence);</span>
<span class="source-line-no">203</span><span id="line-203">        if (SequenceUtil.isNonAmbNucleotideSequence(sequence)) {</span>
<span class="source-line-no">204</span><span id="line-204">                return false;</span>
<span class="source-line-no">205</span><span id="line-205">        }</span>
<span class="source-line-no">206</span><span id="line-206">        if (SequenceUtil.DIGIT.matcher(sequence).find()) {</span>
<span class="source-line-no">207</span><span id="line-207">                return false;</span>
<span class="source-line-no">208</span><span id="line-208">        }</span>
<span class="source-line-no">209</span><span id="line-209">        if (SequenceUtil.NON_AA.matcher(sequence).find()) {</span>
<span class="source-line-no">210</span><span id="line-210">                return false;</span>
<span class="source-line-no">211</span><span id="line-211">        }</span>
<span class="source-line-no">212</span><span id="line-212">        if (SequenceUtil.AA.matcher(sequence).find()) {</span>
<span class="source-line-no">213</span><span id="line-213">                return false;</span>
<span class="source-line-no">214</span><span id="line-214">        }</span>
<span class="source-line-no">215</span><span id="line-215">        final Matcher amb_prot = SequenceUtil.AMBIGUOUS_AA.matcher(sequence);</span>
<span class="source-line-no">216</span><span id="line-216">        return amb_prot.find();</span>
<span class="source-line-no">217</span><span id="line-217">        }</span>
<span class="source-line-no">218</span><span id="line-218"></span>
<span class="source-line-no">219</span><span id="line-219">        /**</span>
<span class="source-line-no">220</span><span id="line-220">         * Writes list of FastaSequeces into the outstream formatting the sequence</span>
<span class="source-line-no">221</span><span id="line-221">         * so that it contains width chars on each line</span>
<span class="source-line-no">222</span><span id="line-222">         *</span>
<span class="source-line-no">223</span><span id="line-223">         * @param outstream</span>
<span class="source-line-no">224</span><span id="line-224">         * @param sequences</span>
<span class="source-line-no">225</span><span id="line-225">         * @param width</span>
<span class="source-line-no">226</span><span id="line-226">         *            - the maximum number of characters to write in one line</span>
<span class="source-line-no">227</span><span id="line-227">         * @throws IOException</span>
<span class="source-line-no">228</span><span id="line-228">         */</span>
<span class="source-line-no">229</span><span id="line-229">        public static void writeFasta(final OutputStream outstream,</span>
<span class="source-line-no">230</span><span id="line-230">                final List&lt;FastaSequence&gt; sequences, final int width)</span>
<span class="source-line-no">231</span><span id="line-231">                throws IOException {</span>
<span class="source-line-no">232</span><span id="line-232">        final OutputStreamWriter writer = new OutputStreamWriter(outstream);</span>
<span class="source-line-no">233</span><span id="line-233">        final BufferedWriter fastawriter = new BufferedWriter(writer);</span>
<span class="source-line-no">234</span><span id="line-234">        for (final FastaSequence fs : sequences) {</span>
<span class="source-line-no">235</span><span id="line-235">                fastawriter.write(fs.getFormatedSequence(width));</span>
<span class="source-line-no">236</span><span id="line-236">        }</span>
<span class="source-line-no">237</span><span id="line-237">        outstream.flush();</span>
<span class="source-line-no">238</span><span id="line-238">        fastawriter.close();</span>
<span class="source-line-no">239</span><span id="line-239">        writer.close();</span>
<span class="source-line-no">240</span><span id="line-240">        }</span>
<span class="source-line-no">241</span><span id="line-241"></span>
<span class="source-line-no">242</span><span id="line-242">        /**</span>
<span class="source-line-no">243</span><span id="line-243">         * Reads fasta sequences from inStream into the list of FastaSequence</span>
<span class="source-line-no">244</span><span id="line-244">         * objects</span>
<span class="source-line-no">245</span><span id="line-245">         *</span>
<span class="source-line-no">246</span><span id="line-246">         * @param inStream</span>
<span class="source-line-no">247</span><span id="line-247">         *            from</span>
<span class="source-line-no">248</span><span id="line-248">         * @return list of FastaSequence objects</span>
<span class="source-line-no">249</span><span id="line-249">         * @throws IOException</span>
<span class="source-line-no">250</span><span id="line-250">         */</span>
<span class="source-line-no">251</span><span id="line-251">        public static List&lt;FastaSequence&gt; readFasta(final InputStream inStream)</span>
<span class="source-line-no">252</span><span id="line-252">                throws IOException {</span>
<span class="source-line-no">253</span><span id="line-253">        final List&lt;FastaSequence&gt; seqs = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">254</span><span id="line-254"></span>
<span class="source-line-no">255</span><span id="line-255">        final BufferedReader infasta = new BufferedReader(</span>
<span class="source-line-no">256</span><span id="line-256">                new InputStreamReader(inStream, "UTF8"), 16000);</span>
<span class="source-line-no">257</span><span id="line-257">        final Pattern pattern = Pattern.compile("//s+");</span>
<span class="source-line-no">258</span><span id="line-258"></span>
<span class="source-line-no">259</span><span id="line-259">        String line;</span>
<span class="source-line-no">260</span><span id="line-260">        String sname = "", seqstr = null;</span>
<span class="source-line-no">261</span><span id="line-261">        do {</span>
<span class="source-line-no">262</span><span id="line-262">                line = infasta.readLine();</span>
<span class="source-line-no">263</span><span id="line-263">                if ((line == null) || line.startsWith("&gt;")) {</span>
<span class="source-line-no">264</span><span id="line-264">                if (seqstr != null) {</span>
<span class="source-line-no">265</span><span id="line-265">                        seqs.add(new FastaSequence(sname.substring(1), seqstr));</span>
<span class="source-line-no">266</span><span id="line-266">                }</span>
<span class="source-line-no">267</span><span id="line-267">                sname = line; // remove &gt;</span>
<span class="source-line-no">268</span><span id="line-268">                seqstr = "";</span>
<span class="source-line-no">269</span><span id="line-269">                } else {</span>
<span class="source-line-no">270</span><span id="line-270">                final String subseq = pattern.matcher(line).replaceAll("");</span>
<span class="source-line-no">271</span><span id="line-271">                seqstr += subseq;</span>
<span class="source-line-no">272</span><span id="line-272">                }</span>
<span class="source-line-no">273</span><span id="line-273">        } while (line != null);</span>
<span class="source-line-no">274</span><span id="line-274"></span>
<span class="source-line-no">275</span><span id="line-275">        infasta.close();</span>
<span class="source-line-no">276</span><span id="line-276">        return seqs;</span>
<span class="source-line-no">277</span><span id="line-277">        }</span>
<span class="source-line-no">278</span><span id="line-278"></span>
<span class="source-line-no">279</span><span id="line-279">        /**</span>
<span class="source-line-no">280</span><span id="line-280">         * Writes FastaSequence in the file, each sequence will take one line only</span>
<span class="source-line-no">281</span><span id="line-281">         *</span>
<span class="source-line-no">282</span><span id="line-282">         * @param os</span>
<span class="source-line-no">283</span><span id="line-283">         * @param sequences</span>
<span class="source-line-no">284</span><span id="line-284">         * @throws IOException</span>
<span class="source-line-no">285</span><span id="line-285">         */</span>
<span class="source-line-no">286</span><span id="line-286">        public static void writeFasta(final OutputStream os,</span>
<span class="source-line-no">287</span><span id="line-287">                final List&lt;FastaSequence&gt; sequences) throws IOException {</span>
<span class="source-line-no">288</span><span id="line-288">        final OutputStreamWriter outWriter = new OutputStreamWriter(os);</span>
<span class="source-line-no">289</span><span id="line-289">        final BufferedWriter fasta_out = new BufferedWriter(outWriter);</span>
<span class="source-line-no">290</span><span id="line-290">        for (final FastaSequence fs : sequences) {</span>
<span class="source-line-no">291</span><span id="line-291">                fasta_out.write(fs.getOnelineFasta());</span>
<span class="source-line-no">292</span><span id="line-292">        }</span>
<span class="source-line-no">293</span><span id="line-293">        fasta_out.close();</span>
<span class="source-line-no">294</span><span id="line-294">        outWriter.close();</span>
<span class="source-line-no">295</span><span id="line-295">        }</span>
<span class="source-line-no">296</span><span id="line-296"></span>
<span class="source-line-no">297</span><span id="line-297">}</span>




























































</pre>
</div>
</main>
</body>
</html>
