<!DOCTYPE HTML>
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (11.0.10) on Fri Mar 12 13:02:30 CST 2021 -->
<title>Chromatogram (biojava-legacy 1.9.5 API)</title>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2021-03-12">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../jquery/jquery-ui.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip/dist/jszip.min.js"></script>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils.min.js"></script>
<!--[if IE]>
<script type="text/javascript" src="../../../../jquery/jszip-utils/dist/jszip-utils-ie.min.js"></script>
<![endif]-->
<script type="text/javascript" src="../../../../jquery/jquery-3.5.1.js"></script>
<script type="text/javascript" src="../../../../jquery/jquery-ui.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Chromatogram (biojava-legacy 1.9.5 API)";
        }
    }
    catch(err) {
    }
//-->
var data = {"i0":6,"i1":6,"i2":6,"i3":6,"i4":6,"i5":6,"i6":6,"i7":6};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],4:["t3","Abstract Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
var pathtoroot = "../../../../";
var useModuleDirectories = true;
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<header role="banner">
<nav role="navigation">
<div class="fixedNav">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a id="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromatogram.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<ul class="navListSearch">
<li><label for="search">SEARCH:</label>
<input type="text" id="search" value="search" disabled="disabled">
<input type="reset" id="reset" value="reset" disabled="disabled">
</li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
</div>
<div class="navPadding">&nbsp;</div>
<script type="text/javascript"><!--
$('.navPadding').css('padding-top', $('.fixedNav').css("height"));
//-->
</script>
</nav>
</header>
<!-- ======== START OF CLASS DATA ======== -->
<main role="main">
<div class="header">
<div class="subTitle"><span class="packageLabelInType">Package</span>&nbsp;<a href="package-summary.html">org.biojava.bio.chromatogram</a></div>
<h2 title="Interface Chromatogram" class="title">Interface Chromatogram</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><code><a href="../program/abi/ABIFChromatogram.html" title="class in org.biojava.bio.program.abi">ABIFChromatogram</a></code>, <code><a href="AbstractChromatogram.html" title="class in org.biojava.bio.chromatogram">AbstractChromatogram</a></code>, <code><a href="../program/scf/SCF.html" title="class in org.biojava.bio.program.scf">SCF</a></code>, <code><a href="SimpleChromatogram.html" title="class in org.biojava.bio.chromatogram">SimpleChromatogram</a></code></dd>
</dl>
<hr>
<pre>public interface <a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.36">Chromatogram</a></pre>
<div class="block">Encapsulates the basic information you would want from a chromatogram.
 Read-only.</div>
<dl>
<dt><span class="simpleTagLabel">Since:</span></dt>
<dd>1.3</dd>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Rhett Sutphin (<a href="http://genome.uiowa.edu/">UI CBCB</a>), Matthew Pocock</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<table class="memberSummary">
<caption><span>Fields</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Field</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#DNA">DNA</a></span></code></th>
<td class="colLast">
<div class="block">The sequence label for the list of called bases.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#OFFSETS">OFFSETS</a></span></code></th>
<td class="colLast">
<div class="block">The sequence label for the trace offsets of the called bases.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
<!-- ========== METHOD SUMMARY =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Abstract Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colSecond" scope="col">Method</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code><a href="../alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getBaseCalls()">getBaseCalls</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns an alignment that describes the base calls for this chromatogram.</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMax()">getMax</a></span>()</code></th>
<td class="colLast">
<div class="block">Gets the max intensity from all the traces.</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getMax(org.biojava.bio.symbol.AtomicSymbol)">getMax</a></span>&#8203;(<a href="../symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)</code></th>
<td class="colLast">
<div class="block">Gets the max intensity on the trace for the specified nucleotide.</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSequenceLength()">getSequenceLength</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the number of bases called by whatever base-calling software
 analyzed the chromatogram as loaded.</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getSignificantBits()">getSignificantBits</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the number of bits of the traces which are significant.</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code>int[]</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTrace(org.biojava.bio.symbol.AtomicSymbol)">getTrace</a></span>&#8203;(<a href="../symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)</code></th>
<td class="colLast">
<div class="block">Returns an array containing the intensities of the sampled waveform
  representing the chromatogram trace for base <code>nucleotide</code>.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>int</code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#getTraceLength()">getTraceLength</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns the length of the trace of the Chromatogram.</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code><a href="Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a></code></td>
<th class="colSecond" scope="row"><code><span class="memberNameLink"><a href="#reverseComplement()">reverseComplement</a></span>()</code></th>
<td class="colLast">
<div class="block">Returns a new <code>Chromatogram</code> representing the reverse
 complement of this one.</div>
</td>
</tr>
</table>
</li>
</ul>
</section>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ FIELD DETAIL =========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="field.detail">
<!--   -->
</a>
<h3>Field Detail</h3>
<a id="DNA">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>DNA</h4>
<pre>static final&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a> <a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.40">DNA</a></pre>
<div class="block">The sequence label for the list of called bases.</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../../constant-values.html#org.biojava.bio.chromatogram.Chromatogram.DNA">Constant Field Values</a></dd>
</dl>
</li>
</ul>
<a id="OFFSETS">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>OFFSETS</h4>
<pre>static final&nbsp;<a href="https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang" class="externalLink">String</a> <a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.45">OFFSETS</a></pre>
<div class="block">The sequence label for the trace offsets of the called bases.</div>
<dl>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="../../../../constant-values.html#org.biojava.bio.chromatogram.Chromatogram.OFFSETS">Constant Field Values</a></dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
<!-- ============ METHOD DETAIL ========== -->
<section role="region">
<ul class="blockList">
<li class="blockList"><a id="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a id="getMax()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMax</h4>
<pre class="methodSignature">int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.52">getMax</a>()</pre>
<div class="block">Gets the max intensity from all the traces.  Must be equivalent
  to the max of calling <a href="#getMax(org.biojava.bio.symbol.AtomicSymbol)"><code>getMax(AtomicSymbol)</code></a> on each
  of the four non-ambiguous DNA nucleotides.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the max intensity</dd>
</dl>
</li>
</ul>
<a id="getMax(org.biojava.bio.symbol.AtomicSymbol)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getMax</h4>
<pre class="methodSignature">int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.60">getMax</a>&#8203;(<a href="../symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)
    throws <a href="../symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block">Gets the max intensity on the trace for the specified nucleotide.</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nucleotide</code> - the trace to examine.  Must be a concrete 
         (non-ambiguous) nucleotide from the DNA alphabet</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the max intensity</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - when the nucleotide isn't from the DNA
          alphabet</dd>
</dl>
</li>
</ul>
<a id="getTraceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTraceLength</h4>
<pre class="methodSignature">int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.66">getTraceLength</a>()</pre>
<div class="block">Returns the length of the trace of the Chromatogram.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the number of samples in the trace</dd>
<dt><span class="seeLabel">See Also:</span></dt>
<dd><a href="#getTrace(org.biojava.bio.symbol.AtomicSymbol)"><code>getTrace(AtomicSymbol)</code></a></dd>
</dl>
</li>
</ul>
<a id="getTrace(org.biojava.bio.symbol.AtomicSymbol)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTrace</h4>
<pre class="methodSignature">int[]&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.81">getTrace</a>&#8203;(<a href="../symbol/AtomicSymbol.html" title="interface in org.biojava.bio.symbol">AtomicSymbol</a>&nbsp;nucleotide)
        throws <a href="../symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></pre>
<div class="block">Returns an array containing the intensities of the sampled waveform
  representing the chromatogram trace for base <code>nucleotide</code>.  
  This may be a reference the actual internal representation of the 
  samples, so callers <b>must not modify it</b>.
  <p>
  The resulting array for each nucleotide must be <a href="#getTraceLength()"><code>getTraceLength()</code></a>
  <code>int</code>s long.
  </p></div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>nucleotide</code> - the trace to examine.  Must be the symbol for A, C, G, or T
         as provided by <a href="../seq/DNATools.html" title="class in org.biojava.bio.seq"><code>DNATools</code></a></dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an array of integers representing the values of a particular 
          chromatogram trace.</dd>
<dt><span class="throwsLabel">Throws:</span></dt>
<dd><code><a href="../symbol/IllegalSymbolException.html" title="class in org.biojava.bio.symbol">IllegalSymbolException</a></code> - if <code>nucleotide</code> isn't in the DNA alphabet</dd>
</dl>
</li>
</ul>
<a id="getSignificantBits()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSignificantBits</h4>
<pre class="methodSignature">int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.89">getSignificantBits</a>()</pre>
<div class="block">Returns the number of bits of the traces which are significant.  For 
 instance, if the chromatogram were originally encoded with a single byte 
 per trace sample, this method must return 8.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the number of significant bits</dd>
</dl>
</li>
</ul>
<a id="getBaseCalls()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBaseCalls</h4>
<pre class="methodSignature"><a href="../alignment/Alignment.html" title="interface in org.biojava.bio.alignment">Alignment</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.115">getBaseCalls</a>()</pre>
<div class="block">Returns an alignment that describes the base calls for this chromatogram.
 All of the <code>SymbolList</code>s in this alignment must be the same
 length and that length must equal <a href="#getSequenceLength()"><code>getSequenceLength()</code></a>.
 <p>
 The alignment must contain, at the least, two sequences:
 </p>
 <ol>
   <li>A sequence containing the called bases.  The alphabet of this list
       must be <a href="../seq/DNATools.html#getDNA()"><code>DNATools.getDNA()</code></a>.  
       The label for this list in the alignment must be 
       <code>Chromatogram.DNA</code></li>
   <li>A sequence containing the trace offsets at which the called bases 
       were called.  The alphabet of this list must be an 
       <a href="../symbol/IntegerAlphabet.html" title="class in org.biojava.bio.symbol"><code>IntegerAlphabet</code></a> or a
       <a href="../symbol/IntegerAlphabet.SubIntegerAlphabet.html" title="class in org.biojava.bio.symbol"><code>IntegerAlphabet.SubIntegerAlphabet</code></a>.
       The label for this list in the alignment must be 
       <code>Chromatogram.OFFSETS</code>.</li>
 </ol>
 <p>
 Implementors may provide other sequences as they see fit.
 </p></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>an alignment of at least two sequences, as described above.</dd>
</dl>
</li>
</ul>
<a id="getSequenceLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceLength</h4>
<pre class="methodSignature">int&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.122">getSequenceLength</a>()</pre>
<div class="block">Returns the number of bases called by whatever base-calling software
 analyzed the chromatogram as loaded.  Must equal 
 <code><a href="#getBaseCalls()"><code>getBaseCalls()</code></a>.length()</code>.</div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>the number of bases</dd>
</dl>
</li>
</ul>
<a id="reverseComplement()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>reverseComplement</h4>
<pre class="methodSignature"><a href="Chromatogram.html" title="interface in org.biojava.bio.chromatogram">Chromatogram</a>&nbsp;<a href="../../../../src-html/org/biojava/bio/chromatogram/Chromatogram.html#line.133">reverseComplement</a>()</pre>
<div class="block">Returns a new <code>Chromatogram</code> representing the reverse
 complement of this one.
 <p>
 Implementors should copy the metadata about the chromatogram (i.e., base 
 calls) as is appropriate to their formats.
 </p></div>
<dl>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>a new chromatogram that is the reverse complement of this one</dd>
</dl>
</li>
</ul>
</li>
</ul>
</section>
</li>
</ul>
</div>
</div>
</main>
<!-- ========= END OF CLASS DATA ========= -->
<footer role="contentinfo">
<nav role="navigation">
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a id="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a id="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Chromatogram.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><script src="http://www.google-analytics.com/urchin.js" type="text/javascript"></script><script type="text/javascript">_uacct = "UA-1326640-1";urchinTracker();</script></div>
</div>
<div class="subNav">
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field.summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li><a href="#field.detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a id="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
</nav>
<p class="legalCopy"><small>Copyright &#169; 2021 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</body>
</html>
