<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21"></a>
<span class="sourceLineNo">022</span><a id="line.22">package org.biojava.bio.symbol;</a>
<span class="sourceLineNo">023</span><a id="line.23"></a>
<span class="sourceLineNo">024</span><a id="line.24">import java.io.BufferedReader;</a>
<span class="sourceLineNo">025</span><a id="line.25">import java.io.IOException;</a>
<span class="sourceLineNo">026</span><a id="line.26">import java.io.InputStream;</a>
<span class="sourceLineNo">027</span><a id="line.27">import java.io.InputStreamReader;</a>
<span class="sourceLineNo">028</span><a id="line.28">import java.io.OutputStream;</a>
<span class="sourceLineNo">029</span><a id="line.29">import java.io.PrintWriter;</a>
<span class="sourceLineNo">030</span><a id="line.30">import java.util.HashMap;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.Iterator;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.List;</a>
<span class="sourceLineNo">033</span><a id="line.33">import java.util.Map;</a>
<span class="sourceLineNo">034</span><a id="line.34">import java.util.StringTokenizer;</a>
<span class="sourceLineNo">035</span><a id="line.35"></a>
<span class="sourceLineNo">036</span><a id="line.36">import javax.xml.parsers.DocumentBuilder;</a>
<span class="sourceLineNo">037</span><a id="line.37">import javax.xml.parsers.DocumentBuilderFactory;</a>
<span class="sourceLineNo">038</span><a id="line.38"></a>
<span class="sourceLineNo">039</span><a id="line.39">import org.biojava.bio.BioException;</a>
<span class="sourceLineNo">040</span><a id="line.40">import org.biojava.bio.dist.Count;</a>
<span class="sourceLineNo">041</span><a id="line.41">import org.biojava.bio.dist.Distribution;</a>
<span class="sourceLineNo">042</span><a id="line.42">import org.biojava.bio.dist.DistributionTools;</a>
<span class="sourceLineNo">043</span><a id="line.43">import org.biojava.bio.dist.IndexedCount;</a>
<span class="sourceLineNo">044</span><a id="line.44">import org.biojava.bio.seq.RNATools;</a>
<span class="sourceLineNo">045</span><a id="line.45">import org.biojava.bio.seq.io.SymbolTokenization;</a>
<span class="sourceLineNo">046</span><a id="line.46">import org.biojava.utils.ChangeVetoException;</a>
<span class="sourceLineNo">047</span><a id="line.47">import org.biojava.utils.ClassTools;</a>
<span class="sourceLineNo">048</span><a id="line.48">import org.biojava.utils.xml.PrettyXMLWriter;</a>
<span class="sourceLineNo">049</span><a id="line.49">import org.biojava.utils.xml.XMLWriter;</a>
<span class="sourceLineNo">050</span><a id="line.50">import org.w3c.dom.Document;</a>
<span class="sourceLineNo">051</span><a id="line.51">import org.w3c.dom.Element;</a>
<span class="sourceLineNo">052</span><a id="line.52">import org.w3c.dom.Node;</a>
<span class="sourceLineNo">053</span><a id="line.53">import org.w3c.dom.NodeList;</a>
<span class="sourceLineNo">054</span><a id="line.54"></a>
<span class="sourceLineNo">055</span><a id="line.55">/**</a>
<span class="sourceLineNo">056</span><a id="line.56"> * An utility class for codon preferences</a>
<span class="sourceLineNo">057</span><a id="line.57"> *</a>
<span class="sourceLineNo">058</span><a id="line.58"> * @author David Huen</a>
<span class="sourceLineNo">059</span><a id="line.59"> * @author Mark Schreiber</a>
<span class="sourceLineNo">060</span><a id="line.60"> * @since 1.3</a>
<span class="sourceLineNo">061</span><a id="line.61"> */</a>
<span class="sourceLineNo">062</span><a id="line.62">public class CodonPrefTools</a>
<span class="sourceLineNo">063</span><a id="line.63">{</a>
<span class="sourceLineNo">064</span><a id="line.64">    /**</a>
<span class="sourceLineNo">065</span><a id="line.65">     * constants for model organisms</a>
<span class="sourceLineNo">066</span><a id="line.66">     */</a>
<span class="sourceLineNo">067</span><a id="line.67">    static String JUNIT = "jUnit use only!!!!";</a>
<span class="sourceLineNo">068</span><a id="line.68">    /**</a>
<span class="sourceLineNo">069</span><a id="line.69">     * Drosophila melanogaster codon preferences</a>
<span class="sourceLineNo">070</span><a id="line.70">     */</a>
<span class="sourceLineNo">071</span><a id="line.71">    public static String DROSOPHILA_MELANOGASTER_NUCLEAR = "Drosophila melanogaster";</a>
<span class="sourceLineNo">072</span><a id="line.72">    /**</a>
<span class="sourceLineNo">073</span><a id="line.73">     * Homo sapiens codon preferences</a>
<span class="sourceLineNo">074</span><a id="line.74">     */</a>
<span class="sourceLineNo">075</span><a id="line.75">    public static String MAN_NUCLEAR = "Homo sapiens";</a>
<span class="sourceLineNo">076</span><a id="line.76">    /**</a>
<span class="sourceLineNo">077</span><a id="line.77">     * Mus musculus codon preferences</a>
<span class="sourceLineNo">078</span><a id="line.78">     */</a>
<span class="sourceLineNo">079</span><a id="line.79">    public static String MOUSE_NUCLEAR = "Mus musculus";</a>
<span class="sourceLineNo">080</span><a id="line.80">    /**</a>
<span class="sourceLineNo">081</span><a id="line.81">     * Rattus norvegicus codon preferences</a>
<span class="sourceLineNo">082</span><a id="line.82">     */</a>
<span class="sourceLineNo">083</span><a id="line.83">    public static String RAT_NUCLEAR = "Rattus norvegicus";</a>
<span class="sourceLineNo">084</span><a id="line.84">    /**</a>
<span class="sourceLineNo">085</span><a id="line.85">     * Takifugu rubripes codon preferences</a>
<span class="sourceLineNo">086</span><a id="line.86">     */</a>
<span class="sourceLineNo">087</span><a id="line.87">    public static String FUGU_NUCLEAR = "Takifugu rubripes";</a>
<span class="sourceLineNo">088</span><a id="line.88">    /**</a>
<span class="sourceLineNo">089</span><a id="line.89">     * Caenorhabditis elegans codon preferences</a>
<span class="sourceLineNo">090</span><a id="line.90">     */</a>
<span class="sourceLineNo">091</span><a id="line.91">    public static String WORM_NUCLEAR = "Caenorhabditis elegans";</a>
<span class="sourceLineNo">092</span><a id="line.92">    /**</a>
<span class="sourceLineNo">093</span><a id="line.93">     * Saccharomyces cerevisiae codon preferences</a>
<span class="sourceLineNo">094</span><a id="line.94">     */</a>
<span class="sourceLineNo">095</span><a id="line.95">    public static String CEREVISIAE_NUCLEAR = "Saccharomyces cerevisiae";</a>
<span class="sourceLineNo">096</span><a id="line.96">    /**</a>
<span class="sourceLineNo">097</span><a id="line.97">     * Schizosaccharomyces pombe codon preferences</a>
<span class="sourceLineNo">098</span><a id="line.98">     */</a>
<span class="sourceLineNo">099</span><a id="line.99">    public static String POMBE_NUCLEAR = "Schizosaccharomyces pombe";</a>
<span class="sourceLineNo">100</span><a id="line.100">    /**</a>
<span class="sourceLineNo">101</span><a id="line.101">     * Escherichia coli codon preferences</a>
<span class="sourceLineNo">102</span><a id="line.102">     */</a>
<span class="sourceLineNo">103</span><a id="line.103">    public static String ECOLI = "Escherichia coli";</a>
<span class="sourceLineNo">104</span><a id="line.104"></a>
<span class="sourceLineNo">105</span><a id="line.105">    private static Map prefMap;</a>
<span class="sourceLineNo">106</span><a id="line.106"></a>
<span class="sourceLineNo">107</span><a id="line.107">    final private static AtomicSymbol [] cutg = new AtomicSymbol[64];</a>
<span class="sourceLineNo">108</span><a id="line.108"></a>
<span class="sourceLineNo">109</span><a id="line.109">    static {</a>
<span class="sourceLineNo">110</span><a id="line.110">        prefMap = new HashMap();</a>
<span class="sourceLineNo">111</span><a id="line.111"></a>
<span class="sourceLineNo">112</span><a id="line.112">        loadCodonPreferences();</a>
<span class="sourceLineNo">113</span><a id="line.113"></a>
<span class="sourceLineNo">114</span><a id="line.114">        try {</a>
<span class="sourceLineNo">115</span><a id="line.115">            loadCodonOrder();</a>
<span class="sourceLineNo">116</span><a id="line.116">        }</a>
<span class="sourceLineNo">117</span><a id="line.117">        catch (IllegalSymbolException ise) {}</a>
<span class="sourceLineNo">118</span><a id="line.118">    }</a>
<span class="sourceLineNo">119</span><a id="line.119"></a>
<span class="sourceLineNo">120</span><a id="line.120">    private static class LoadEverythingSelector implements CodonPrefFilter</a>
<span class="sourceLineNo">121</span><a id="line.121">    {</a>
<span class="sourceLineNo">122</span><a id="line.122">        public boolean isRequired(String id) { return true; }</a>
<span class="sourceLineNo">123</span><a id="line.123">        public void put(CodonPref codonPref)</a>
<span class="sourceLineNo">124</span><a id="line.124">        {</a>
<span class="sourceLineNo">125</span><a id="line.125">            prefMap.put(codonPref.getName(), codonPref);</a>
<span class="sourceLineNo">126</span><a id="line.126">        }</a>
<span class="sourceLineNo">127</span><a id="line.127">    }</a>
<span class="sourceLineNo">128</span><a id="line.128"></a>
<span class="sourceLineNo">129</span><a id="line.129">    /**</a>
<span class="sourceLineNo">130</span><a id="line.130">     * get the specified codon preference.</a>
<span class="sourceLineNo">131</span><a id="line.131">     */</a>
<span class="sourceLineNo">132</span><a id="line.132">    public static CodonPref getCodonPreference(String id)</a>
<span class="sourceLineNo">133</span><a id="line.133">    {</a>
<span class="sourceLineNo">134</span><a id="line.134">        return (CodonPref) prefMap.get(id);</a>
<span class="sourceLineNo">135</span><a id="line.135">    }</a>
<span class="sourceLineNo">136</span><a id="line.136"></a>
<span class="sourceLineNo">137</span><a id="line.137">    private static void loadCodonPreferences()</a>
<span class="sourceLineNo">138</span><a id="line.138">    {</a>
<span class="sourceLineNo">139</span><a id="line.139">        try {</a>
<span class="sourceLineNo">140</span><a id="line.140">            // parse the predefined codon preferences</a>
<span class="sourceLineNo">141</span><a id="line.141">            InputStream prefStream = ClassTools.getClassLoader(CodonPrefTools.class).getResourceAsStream(</a>
<span class="sourceLineNo">142</span><a id="line.142">                "org/biojava/bio/symbol/CodonPrefTables.xml"</a>
<span class="sourceLineNo">143</span><a id="line.143">            );</a>
<span class="sourceLineNo">144</span><a id="line.144"></a>
<span class="sourceLineNo">145</span><a id="line.145">            CodonPrefFilter select = new LoadEverythingSelector();</a>
<span class="sourceLineNo">146</span><a id="line.146">            readFromXML(prefStream, select);</a>
<span class="sourceLineNo">147</span><a id="line.147">        }</a>
<span class="sourceLineNo">148</span><a id="line.148">        catch (Exception e) { e.printStackTrace(); }</a>
<span class="sourceLineNo">149</span><a id="line.149">    }</a>
<span class="sourceLineNo">150</span><a id="line.150"></a>
<span class="sourceLineNo">151</span><a id="line.151">    /**</a>
<span class="sourceLineNo">152</span><a id="line.152">     * returns an RNA dinucleotide alphabet.</a>
<span class="sourceLineNo">153</span><a id="line.153">     * Used to represent the non-wobble bases in WobbleDistribution</a>
<span class="sourceLineNo">154</span><a id="line.154">     */</a>
<span class="sourceLineNo">155</span><a id="line.155">    public static FiniteAlphabet getDinucleotideAlphabet()</a>
<span class="sourceLineNo">156</span><a id="line.156">    {</a>
<span class="sourceLineNo">157</span><a id="line.157">        return (FiniteAlphabet)AlphabetManager.generateCrossProductAlphaFromName("(RNA x RNA)");</a>
<span class="sourceLineNo">158</span><a id="line.158">    }</a>
<span class="sourceLineNo">159</span><a id="line.159"></a>
<span class="sourceLineNo">160</span><a id="line.160">    /**</a>
<span class="sourceLineNo">161</span><a id="line.161">     * write out a specified CodonPref object in XML format.</a>
<span class="sourceLineNo">162</span><a id="line.162">     */</a>
<span class="sourceLineNo">163</span><a id="line.163">    public static void writeToXML(CodonPref codonPref, PrintWriter writer)</a>
<span class="sourceLineNo">164</span><a id="line.164">        throws NullPointerException, IOException, IllegalSymbolException, BioException</a>
<span class="sourceLineNo">165</span><a id="line.165">    {</a>
<span class="sourceLineNo">166</span><a id="line.166">        XMLWriter xw = new PrettyXMLWriter(writer);</a>
<span class="sourceLineNo">167</span><a id="line.167"></a>
<span class="sourceLineNo">168</span><a id="line.168">        dumpToXML(codonPref, xw, true);</a>
<span class="sourceLineNo">169</span><a id="line.169"></a>
<span class="sourceLineNo">170</span><a id="line.170">        writer.flush();</a>
<span class="sourceLineNo">171</span><a id="line.171">    }</a>
<span class="sourceLineNo">172</span><a id="line.172"></a>
<span class="sourceLineNo">173</span><a id="line.173">    /**</a>
<span class="sourceLineNo">174</span><a id="line.174">     * reads a specified CodonPref from an file.</a>
<span class="sourceLineNo">175</span><a id="line.175">     * @param name name of organism</a>
<span class="sourceLineNo">176</span><a id="line.176">     */</a>
<span class="sourceLineNo">177</span><a id="line.177">    public static CodonPref readFromXML(InputStream prefStream, String name)</a>
<span class="sourceLineNo">178</span><a id="line.178">        throws BioException</a>
<span class="sourceLineNo">179</span><a id="line.179">    {</a>
<span class="sourceLineNo">180</span><a id="line.180">        CodonPrefFilter.ByName filter = new CodonPrefFilter.ByName(name);</a>
<span class="sourceLineNo">181</span><a id="line.181"></a>
<span class="sourceLineNo">182</span><a id="line.182">        readFromXML(prefStream, filter);</a>
<span class="sourceLineNo">183</span><a id="line.183"></a>
<span class="sourceLineNo">184</span><a id="line.184">        return filter.getCodonPref();</a>
<span class="sourceLineNo">185</span><a id="line.185">    }</a>
<span class="sourceLineNo">186</span><a id="line.186"></a>
<span class="sourceLineNo">187</span><a id="line.187">    public static CodonPref[] readFromXML(InputStream prefStream) throws BioException{</a>
<span class="sourceLineNo">188</span><a id="line.188">      CodonPrefFilter.AcceptAll filter = new CodonPrefFilter.AcceptAll();</a>
<span class="sourceLineNo">189</span><a id="line.189">      readFromXML(prefStream, filter);</a>
<span class="sourceLineNo">190</span><a id="line.190"></a>
<span class="sourceLineNo">191</span><a id="line.191">      List l = filter.getCodonPrefs();</a>
<span class="sourceLineNo">192</span><a id="line.192">      CodonPref[] cp = new CodonPref[l.size()];</a>
<span class="sourceLineNo">193</span><a id="line.193">      return (CodonPref[])l.toArray(cp);</a>
<span class="sourceLineNo">194</span><a id="line.194">    }</a>
<span class="sourceLineNo">195</span><a id="line.195"></a>
<span class="sourceLineNo">196</span><a id="line.196">    /**</a>
<span class="sourceLineNo">197</span><a id="line.197">     * read an CodonPref XML stream and handle it with a CodonPrefFilter object.</a>
<span class="sourceLineNo">198</span><a id="line.198">     */</a>
<span class="sourceLineNo">199</span><a id="line.199">    public static void readFromXML(InputStream prefStream, CodonPrefFilter filter)</a>
<span class="sourceLineNo">200</span><a id="line.200">        throws BioException</a>
<span class="sourceLineNo">201</span><a id="line.201">    {</a>
<span class="sourceLineNo">202</span><a id="line.202">        try {</a>
<span class="sourceLineNo">203</span><a id="line.203">            DocumentBuilder parser = DocumentBuilderFactory.newInstance().newDocumentBuilder();</a>
<span class="sourceLineNo">204</span><a id="line.204">            Document doc = parser.parse(prefStream);</a>
<span class="sourceLineNo">205</span><a id="line.205"></a>
<span class="sourceLineNo">206</span><a id="line.206">            // get tables for each species</a>
<span class="sourceLineNo">207</span><a id="line.207">            NodeList children = doc.getDocumentElement().getChildNodes();</a>
<span class="sourceLineNo">208</span><a id="line.208"></a>
<span class="sourceLineNo">209</span><a id="line.209">            for (int i=0; i&lt;children.getLength(); i++) {</a>
<span class="sourceLineNo">210</span><a id="line.210">                Node cnode = children.item(i);</a>
<span class="sourceLineNo">211</span><a id="line.211"></a>
<span class="sourceLineNo">212</span><a id="line.212">                if (!(cnode instanceof Element)) continue;</a>
<span class="sourceLineNo">213</span><a id="line.213"></a>
<span class="sourceLineNo">214</span><a id="line.214">                Element child = (Element) cnode;</a>
<span class="sourceLineNo">215</span><a id="line.215"></a>
<span class="sourceLineNo">216</span><a id="line.216">                String name = child.getNodeName();</a>
<span class="sourceLineNo">217</span><a id="line.217"></a>
<span class="sourceLineNo">218</span><a id="line.218">                // the node must be a CodonPref record</a>
<span class="sourceLineNo">219</span><a id="line.219">                if (!name.equals("CodonPref")) continue;</a>
<span class="sourceLineNo">220</span><a id="line.220"></a>
<span class="sourceLineNo">221</span><a id="line.221">                // pick up the id and genetic code</a>
<span class="sourceLineNo">222</span><a id="line.222">                String codonPrefId = child.getAttribute("id");</a>
<span class="sourceLineNo">223</span><a id="line.223">                String geneticCodeId = child.getAttribute("geneticCodeId");</a>
<span class="sourceLineNo">224</span><a id="line.224"></a>
<span class="sourceLineNo">225</span><a id="line.225">                // is this entry one we want?</a>
<span class="sourceLineNo">226</span><a id="line.226">                if (!filter.isRequired(codonPrefId)) continue;</a>
<span class="sourceLineNo">227</span><a id="line.227"></a>
<span class="sourceLineNo">228</span><a id="line.228">                // now handle each codon frequency entry</a>
<span class="sourceLineNo">229</span><a id="line.229">                NodeList freqs = child.getChildNodes();</a>
<span class="sourceLineNo">230</span><a id="line.230"></a>
<span class="sourceLineNo">231</span><a id="line.231">                // create a Count object for the job</a>
<span class="sourceLineNo">232</span><a id="line.232">                Count freqCounts = new IndexedCount(RNATools.getCodonAlphabet());</a>
<span class="sourceLineNo">233</span><a id="line.233"></a>
<span class="sourceLineNo">234</span><a id="line.234">                for (int j=0; j &lt; freqs.getLength(); j++) {</a>
<span class="sourceLineNo">235</span><a id="line.235">                    // load each entry</a>
<span class="sourceLineNo">236</span><a id="line.236">                    Node freq = freqs.item(j);</a>
<span class="sourceLineNo">237</span><a id="line.237"></a>
<span class="sourceLineNo">238</span><a id="line.238">                    if (!(freq instanceof Element)) continue;</a>
<span class="sourceLineNo">239</span><a id="line.239"></a>
<span class="sourceLineNo">240</span><a id="line.240">                    Element freqElement = (Element) freq;</a>
<span class="sourceLineNo">241</span><a id="line.241"></a>
<span class="sourceLineNo">242</span><a id="line.242">                    // get attributes</a>
<span class="sourceLineNo">243</span><a id="line.243">                    String codonString = freqElement.getAttribute("codon");</a>
<span class="sourceLineNo">244</span><a id="line.244">                    String freqString = freqElement.getAttribute("value");</a>
<span class="sourceLineNo">245</span><a id="line.245"></a>
<span class="sourceLineNo">246</span><a id="line.246">                    // create codon</a>
<span class="sourceLineNo">247</span><a id="line.247">                    SymbolList codonSL = RNATools.createRNA(codonString);</a>
<span class="sourceLineNo">248</span><a id="line.248"></a>
<span class="sourceLineNo">249</span><a id="line.249">                    if (codonSL.length() !=3) throw new BioException("'" + codonString + "' is not a valid codon!");</a>
<span class="sourceLineNo">250</span><a id="line.250"></a>
<span class="sourceLineNo">251</span><a id="line.251">                    AtomicSymbol codon = (AtomicSymbol) RNATools.getCodonAlphabet().getSymbol(codonSL.toList());</a>
<span class="sourceLineNo">252</span><a id="line.252"></a>
<span class="sourceLineNo">253</span><a id="line.253">                    // recover frequency value too</a>
<span class="sourceLineNo">254</span><a id="line.254">                    double freqValue = Double.parseDouble(freqString);</a>
<span class="sourceLineNo">255</span><a id="line.255">                    freqCounts.increaseCount(codon, freqValue);</a>
<span class="sourceLineNo">256</span><a id="line.256"></a>
<span class="sourceLineNo">257</span><a id="line.257">                }</a>
<span class="sourceLineNo">258</span><a id="line.258"></a>
<span class="sourceLineNo">259</span><a id="line.259">                // turn the Counts into a Distribution</a>
<span class="sourceLineNo">260</span><a id="line.260">                Distribution freqDistribution = DistributionTools.countToDistribution(freqCounts);</a>
<span class="sourceLineNo">261</span><a id="line.261"></a>
<span class="sourceLineNo">262</span><a id="line.262">                // create a CodonPref object</a>
<span class="sourceLineNo">263</span><a id="line.263">                CodonPref newCodonPref = new SimpleCodonPref(geneticCodeId, freqDistribution, codonPrefId);</a>
<span class="sourceLineNo">264</span><a id="line.264"></a>
<span class="sourceLineNo">265</span><a id="line.265">                filter.put(newCodonPref);</a>
<span class="sourceLineNo">266</span><a id="line.266">            }</a>
<span class="sourceLineNo">267</span><a id="line.267">        }</a>
<span class="sourceLineNo">268</span><a id="line.268">        catch (Exception e) {</a>
<span class="sourceLineNo">269</span><a id="line.269">            throw new BioException(e);</a>
<span class="sourceLineNo">270</span><a id="line.270">        }</a>
<span class="sourceLineNo">271</span><a id="line.271">    }</a>
<span class="sourceLineNo">272</span><a id="line.272"></a>
<span class="sourceLineNo">273</span><a id="line.273">    /**</a>
<span class="sourceLineNo">274</span><a id="line.274">     * reads in a file in Codon Usage Database format and</a>
<span class="sourceLineNo">275</span><a id="line.275">     * translate it into our XML format</a>
<span class="sourceLineNo">276</span><a id="line.276">     * These can be obtained from the</a>
<span class="sourceLineNo">277</span><a id="line.277">     * &lt;a href="http://www.kazusa.or.jp/codon/"&gt;Codon Usage Database&lt;/a&gt;.</a>
<span class="sourceLineNo">278</span><a id="line.278">     * &lt;p&gt;</a>
<span class="sourceLineNo">279</span><a id="line.279">     * Note that the output assumes that the universal genetic code is</a>
<span class="sourceLineNo">280</span><a id="line.280">     * used as that is not encoded in the CUD files.  Edit the output appropriately</a>
<span class="sourceLineNo">281</span><a id="line.281">     * to modify the genetic code if necessary.</a>
<span class="sourceLineNo">282</span><a id="line.282">     */</a>
<span class="sourceLineNo">283</span><a id="line.283">    public static void translateCUD(InputStream input, OutputStream output)</a>
<span class="sourceLineNo">284</span><a id="line.284">        throws IOException</a>
<span class="sourceLineNo">285</span><a id="line.285">    {</a>
<span class="sourceLineNo">286</span><a id="line.286">        // create a BufferedReader for the job</a>
<span class="sourceLineNo">287</span><a id="line.287">        BufferedReader rdr = new BufferedReader(new InputStreamReader(input));</a>
<span class="sourceLineNo">288</span><a id="line.288"></a>
<span class="sourceLineNo">289</span><a id="line.289">        // create a PrintWriter for the job</a>
<span class="sourceLineNo">290</span><a id="line.290">        PrintWriter pw = new PrintWriter(output);</a>
<span class="sourceLineNo">291</span><a id="line.291">        CodonPrefFilter.EverythingToXML filter = new CodonPrefFilter.EverythingToXML(pw);</a>
<span class="sourceLineNo">292</span><a id="line.292"></a>
<span class="sourceLineNo">293</span><a id="line.293">        // now invoke the CUD reader and stream its output to the XML writer</a>
<span class="sourceLineNo">294</span><a id="line.294">        readFromCUD(rdr, filter);</a>
<span class="sourceLineNo">295</span><a id="line.295"></a>
<span class="sourceLineNo">296</span><a id="line.296">        filter.close();</a>
<span class="sourceLineNo">297</span><a id="line.297">    }</a>
<span class="sourceLineNo">298</span><a id="line.298"></a>
<span class="sourceLineNo">299</span><a id="line.299"></a>
<span class="sourceLineNo">300</span><a id="line.300">    /**</a>
<span class="sourceLineNo">301</span><a id="line.301">     * converts a String representation of a codon to its Symbol</a>
<span class="sourceLineNo">302</span><a id="line.302">     */</a>
<span class="sourceLineNo">303</span><a id="line.303">    private static AtomicSymbol getCodon(String codonString)</a>
<span class="sourceLineNo">304</span><a id="line.304">        throws IllegalSymbolException</a>
<span class="sourceLineNo">305</span><a id="line.305">    {</a>
<span class="sourceLineNo">306</span><a id="line.306">        return (AtomicSymbol) RNATools.getCodonAlphabet().getSymbol(RNATools.createRNA(codonString).toList());</a>
<span class="sourceLineNo">307</span><a id="line.307">    }</a>
<span class="sourceLineNo">308</span><a id="line.308"></a>
<span class="sourceLineNo">309</span><a id="line.309">    private static void loadCodonOrder()</a>
<span class="sourceLineNo">310</span><a id="line.310">        throws IllegalSymbolException</a>
<span class="sourceLineNo">311</span><a id="line.311">    {</a>
<span class="sourceLineNo">312</span><a id="line.312">        cutg[0] = getCodon("cga");</a>
<span class="sourceLineNo">313</span><a id="line.313">        cutg[1] = getCodon("cgc");</a>
<span class="sourceLineNo">314</span><a id="line.314">        cutg[2] = getCodon("cgg");</a>
<span class="sourceLineNo">315</span><a id="line.315">        cutg[3] = getCodon("cgu");</a>
<span class="sourceLineNo">316</span><a id="line.316"></a>
<span class="sourceLineNo">317</span><a id="line.317">        cutg[4] = getCodon("aga");</a>
<span class="sourceLineNo">318</span><a id="line.318">        cutg[5] = getCodon("agg");</a>
<span class="sourceLineNo">319</span><a id="line.319"></a>
<span class="sourceLineNo">320</span><a id="line.320">        cutg[6] = getCodon("cua");</a>
<span class="sourceLineNo">321</span><a id="line.321">        cutg[7] = getCodon("cuc");</a>
<span class="sourceLineNo">322</span><a id="line.322">        cutg[8] = getCodon("cug");</a>
<span class="sourceLineNo">323</span><a id="line.323">        cutg[9] = getCodon("cuu");</a>
<span class="sourceLineNo">324</span><a id="line.324"></a>
<span class="sourceLineNo">325</span><a id="line.325">        cutg[10] = getCodon("uua");</a>
<span class="sourceLineNo">326</span><a id="line.326">        cutg[11] = getCodon("uug");</a>
<span class="sourceLineNo">327</span><a id="line.327"></a>
<span class="sourceLineNo">328</span><a id="line.328">        cutg[12] = getCodon("uca");</a>
<span class="sourceLineNo">329</span><a id="line.329">        cutg[13] = getCodon("ucc");</a>
<span class="sourceLineNo">330</span><a id="line.330">        cutg[14] = getCodon("ucg");</a>
<span class="sourceLineNo">331</span><a id="line.331">        cutg[15] = getCodon("ucu");</a>
<span class="sourceLineNo">332</span><a id="line.332"></a>
<span class="sourceLineNo">333</span><a id="line.333">        cutg[16] = getCodon("agc");</a>
<span class="sourceLineNo">334</span><a id="line.334">        cutg[17] = getCodon("agu");</a>
<span class="sourceLineNo">335</span><a id="line.335"></a>
<span class="sourceLineNo">336</span><a id="line.336">        cutg[18] = getCodon("aca");</a>
<span class="sourceLineNo">337</span><a id="line.337">        cutg[19] = getCodon("acc");</a>
<span class="sourceLineNo">338</span><a id="line.338">        cutg[20] = getCodon("acg");</a>
<span class="sourceLineNo">339</span><a id="line.339">        cutg[21] = getCodon("acu");</a>
<span class="sourceLineNo">340</span><a id="line.340"></a>
<span class="sourceLineNo">341</span><a id="line.341">        cutg[22] = getCodon("cca");</a>
<span class="sourceLineNo">342</span><a id="line.342">        cutg[23] = getCodon("ccc");</a>
<span class="sourceLineNo">343</span><a id="line.343">        cutg[24] = getCodon("ccg");</a>
<span class="sourceLineNo">344</span><a id="line.344">        cutg[25] = getCodon("ccu");</a>
<span class="sourceLineNo">345</span><a id="line.345"></a>
<span class="sourceLineNo">346</span><a id="line.346">        cutg[26] = getCodon("gca");</a>
<span class="sourceLineNo">347</span><a id="line.347">        cutg[27] = getCodon("gcc");</a>
<span class="sourceLineNo">348</span><a id="line.348">        cutg[28] = getCodon("gcg");</a>
<span class="sourceLineNo">349</span><a id="line.349">        cutg[29] = getCodon("gcu");</a>
<span class="sourceLineNo">350</span><a id="line.350"></a>
<span class="sourceLineNo">351</span><a id="line.351">        cutg[30] = getCodon("gga");</a>
<span class="sourceLineNo">352</span><a id="line.352">        cutg[31] = getCodon("ggc");</a>
<span class="sourceLineNo">353</span><a id="line.353">        cutg[32] = getCodon("ggg");</a>
<span class="sourceLineNo">354</span><a id="line.354">        cutg[33] = getCodon("ggu");</a>
<span class="sourceLineNo">355</span><a id="line.355"></a>
<span class="sourceLineNo">356</span><a id="line.356">        cutg[34] = getCodon("gua");</a>
<span class="sourceLineNo">357</span><a id="line.357">        cutg[35] = getCodon("guc");</a>
<span class="sourceLineNo">358</span><a id="line.358">        cutg[36] = getCodon("gug");</a>
<span class="sourceLineNo">359</span><a id="line.359">        cutg[37] = getCodon("guu");</a>
<span class="sourceLineNo">360</span><a id="line.360"></a>
<span class="sourceLineNo">361</span><a id="line.361">        cutg[38] = getCodon("aaa");</a>
<span class="sourceLineNo">362</span><a id="line.362">        cutg[39] = getCodon("aag");</a>
<span class="sourceLineNo">363</span><a id="line.363"></a>
<span class="sourceLineNo">364</span><a id="line.364">        cutg[40] = getCodon("aac");</a>
<span class="sourceLineNo">365</span><a id="line.365">        cutg[41] = getCodon("aau");</a>
<span class="sourceLineNo">366</span><a id="line.366"></a>
<span class="sourceLineNo">367</span><a id="line.367">        cutg[42] = getCodon("caa");</a>
<span class="sourceLineNo">368</span><a id="line.368">        cutg[43] = getCodon("cag");</a>
<span class="sourceLineNo">369</span><a id="line.369"></a>
<span class="sourceLineNo">370</span><a id="line.370">        cutg[44] = getCodon("cac");</a>
<span class="sourceLineNo">371</span><a id="line.371">        cutg[45] = getCodon("cau");</a>
<span class="sourceLineNo">372</span><a id="line.372"></a>
<span class="sourceLineNo">373</span><a id="line.373">        cutg[46] = getCodon("gaa");</a>
<span class="sourceLineNo">374</span><a id="line.374">        cutg[47] = getCodon("gag");</a>
<span class="sourceLineNo">375</span><a id="line.375"></a>
<span class="sourceLineNo">376</span><a id="line.376">        cutg[48] = getCodon("gac");</a>
<span class="sourceLineNo">377</span><a id="line.377">        cutg[49] = getCodon("gau");</a>
<span class="sourceLineNo">378</span><a id="line.378"></a>
<span class="sourceLineNo">379</span><a id="line.379">        cutg[50] = getCodon("uac");</a>
<span class="sourceLineNo">380</span><a id="line.380">        cutg[51] = getCodon("uau");</a>
<span class="sourceLineNo">381</span><a id="line.381"></a>
<span class="sourceLineNo">382</span><a id="line.382">        cutg[52] = getCodon("ugc");</a>
<span class="sourceLineNo">383</span><a id="line.383">        cutg[53] = getCodon("ugu");</a>
<span class="sourceLineNo">384</span><a id="line.384"></a>
<span class="sourceLineNo">385</span><a id="line.385">        cutg[54] = getCodon("uuc");</a>
<span class="sourceLineNo">386</span><a id="line.386">        cutg[55] = getCodon("uuu");</a>
<span class="sourceLineNo">387</span><a id="line.387"></a>
<span class="sourceLineNo">388</span><a id="line.388">        cutg[56] = getCodon("aua");</a>
<span class="sourceLineNo">389</span><a id="line.389">        cutg[57] = getCodon("auc");</a>
<span class="sourceLineNo">390</span><a id="line.390">        cutg[58] = getCodon("auu");</a>
<span class="sourceLineNo">391</span><a id="line.391"></a>
<span class="sourceLineNo">392</span><a id="line.392">        cutg[59] = getCodon("aug");</a>
<span class="sourceLineNo">393</span><a id="line.393"></a>
<span class="sourceLineNo">394</span><a id="line.394">        cutg[60] = getCodon("ugg");</a>
<span class="sourceLineNo">395</span><a id="line.395"></a>
<span class="sourceLineNo">396</span><a id="line.396">        cutg[61] = getCodon("uaa");</a>
<span class="sourceLineNo">397</span><a id="line.397">        cutg[62] = getCodon("uag");</a>
<span class="sourceLineNo">398</span><a id="line.398">        cutg[63] = getCodon("uga");</a>
<span class="sourceLineNo">399</span><a id="line.399">    }</a>
<span class="sourceLineNo">400</span><a id="line.400"></a>
<span class="sourceLineNo">401</span><a id="line.401">    private static String stringifyCodon(BasisSymbol codon)</a>
<span class="sourceLineNo">402</span><a id="line.402">        throws IllegalSymbolException, BioException</a>
<span class="sourceLineNo">403</span><a id="line.403">    {</a>
<span class="sourceLineNo">404</span><a id="line.404">        // get the component symbols</a>
<span class="sourceLineNo">405</span><a id="line.405">        List codonList = codon.getSymbols();</a>
<span class="sourceLineNo">406</span><a id="line.406"></a>
<span class="sourceLineNo">407</span><a id="line.407">        // get a tokenizer</a>
<span class="sourceLineNo">408</span><a id="line.408">        SymbolTokenization toke = RNATools.getRNA().getTokenization("token");</a>
<span class="sourceLineNo">409</span><a id="line.409"></a>
<span class="sourceLineNo">410</span><a id="line.410">        String tokenizedCodon = toke.tokenizeSymbol((Symbol) codonList.get(0))</a>
<span class="sourceLineNo">411</span><a id="line.411">            + toke.tokenizeSymbol((Symbol) codonList.get(1))</a>
<span class="sourceLineNo">412</span><a id="line.412">            + toke.tokenizeSymbol((Symbol) codonList.get(2));</a>
<span class="sourceLineNo">413</span><a id="line.413"></a>
<span class="sourceLineNo">414</span><a id="line.414">        return tokenizedCodon;</a>
<span class="sourceLineNo">415</span><a id="line.415">    }</a>
<span class="sourceLineNo">416</span><a id="line.416"></a>
<span class="sourceLineNo">417</span><a id="line.417">    /**</a>
<span class="sourceLineNo">418</span><a id="line.418">     * writes out a CodonPref object in XML form</a>
<span class="sourceLineNo">419</span><a id="line.419">     */</a>
<span class="sourceLineNo">420</span><a id="line.420">    static void dumpToXML(CodonPref codonPref, XMLWriter xw, boolean writeWrapper)</a>
<span class="sourceLineNo">421</span><a id="line.421">        throws NullPointerException, IOException, IllegalSymbolException, BioException</a>
<span class="sourceLineNo">422</span><a id="line.422">    {</a>
<span class="sourceLineNo">423</span><a id="line.423">        // validate both objects first</a>
<span class="sourceLineNo">424</span><a id="line.424">        if ((codonPref == null) || (xw == null))</a>
<span class="sourceLineNo">425</span><a id="line.425">            throw new NullPointerException();</a>
<span class="sourceLineNo">426</span><a id="line.426"></a>
<span class="sourceLineNo">427</span><a id="line.427">        // get the CodonPref Distribution</a>
<span class="sourceLineNo">428</span><a id="line.428">        Distribution codonDist = codonPref.getFrequency();</a>
<span class="sourceLineNo">429</span><a id="line.429"></a>
<span class="sourceLineNo">430</span><a id="line.430">        // start &lt;CodonPrefs&gt;</a>
<span class="sourceLineNo">431</span><a id="line.431">        if (writeWrapper) xw.openTag("CodonPrefs");</a>
<span class="sourceLineNo">432</span><a id="line.432"></a>
<span class="sourceLineNo">433</span><a id="line.433">        xw.openTag("CodonPref");</a>
<span class="sourceLineNo">434</span><a id="line.434">        xw.attribute("id", codonPref.getName());</a>
<span class="sourceLineNo">435</span><a id="line.435">        xw.attribute("geneticCodeId", codonPref.getGeneticCodeName());</a>
<span class="sourceLineNo">436</span><a id="line.436"></a>
<span class="sourceLineNo">437</span><a id="line.437">        // loop over all codons, writing out the stats</a>
<span class="sourceLineNo">438</span><a id="line.438">        for (Iterator codonI = RNATools.getCodonAlphabet().iterator(); codonI.hasNext(); ) {</a>
<span class="sourceLineNo">439</span><a id="line.439">            BasisSymbol codon = (BasisSymbol) codonI.next();</a>
<span class="sourceLineNo">440</span><a id="line.440"></a>
<span class="sourceLineNo">441</span><a id="line.441">            xw.openTag("frequency");</a>
<span class="sourceLineNo">442</span><a id="line.442"></a>
<span class="sourceLineNo">443</span><a id="line.443">            // convert codon to a three letter string</a>
<span class="sourceLineNo">444</span><a id="line.444">            xw.attribute("codon", stringifyCodon(codon));</a>
<span class="sourceLineNo">445</span><a id="line.445">            xw.attribute("value", Double.toString(codonDist.getWeight(codon)));</a>
<span class="sourceLineNo">446</span><a id="line.446"></a>
<span class="sourceLineNo">447</span><a id="line.447">            xw.closeTag("frequency");</a>
<span class="sourceLineNo">448</span><a id="line.448">        }</a>
<span class="sourceLineNo">449</span><a id="line.449"></a>
<span class="sourceLineNo">450</span><a id="line.450">        xw.closeTag("CodonPref");</a>
<span class="sourceLineNo">451</span><a id="line.451"></a>
<span class="sourceLineNo">452</span><a id="line.452">        if (writeWrapper) xw.closeTag("CodonPrefs");</a>
<span class="sourceLineNo">453</span><a id="line.453">    }</a>
<span class="sourceLineNo">454</span><a id="line.454"></a>
<span class="sourceLineNo">455</span><a id="line.455">    /**</a>
<span class="sourceLineNo">456</span><a id="line.456">     * reads in records in CUD format</a>
<span class="sourceLineNo">457</span><a id="line.457">     */</a>
<span class="sourceLineNo">458</span><a id="line.458">    private static void readFromCUD(BufferedReader rdr, CodonPrefFilter filter)</a>
<span class="sourceLineNo">459</span><a id="line.459">    {</a>
<span class="sourceLineNo">460</span><a id="line.460">        try {</a>
<span class="sourceLineNo">461</span><a id="line.461">            String currLine;</a>
<span class="sourceLineNo">462</span><a id="line.462">            while ((currLine = rdr.readLine()) != null) {</a>
<span class="sourceLineNo">463</span><a id="line.463"></a>
<span class="sourceLineNo">464</span><a id="line.464">                // process comment line</a>
<span class="sourceLineNo">465</span><a id="line.465">                StringTokenizer toke = new StringTokenizer(currLine, ":");</a>
<span class="sourceLineNo">466</span><a id="line.466">                if (toke.hasMoreTokens()) {</a>
<span class="sourceLineNo">467</span><a id="line.467">                    // get id string</a>
<span class="sourceLineNo">468</span><a id="line.468">                    String id = (toke.nextToken()).trim();</a>
<span class="sourceLineNo">469</span><a id="line.469"></a>
<span class="sourceLineNo">470</span><a id="line.470">                    // read the codon count</a>
<span class="sourceLineNo">471</span><a id="line.471">                    currLine = rdr.readLine();</a>
<span class="sourceLineNo">472</span><a id="line.472">                    if (currLine == null) break;</a>
<span class="sourceLineNo">473</span><a id="line.473"></a>
<span class="sourceLineNo">474</span><a id="line.474">                    // do we even want to process this record?</a>
<span class="sourceLineNo">475</span><a id="line.475">                    if (filter.isRequired(id)) {</a>
<span class="sourceLineNo">476</span><a id="line.476">                        toke = new StringTokenizer(currLine);</a>
<span class="sourceLineNo">477</span><a id="line.477"></a>
<span class="sourceLineNo">478</span><a id="line.478">                        int idx = 0;</a>
<span class="sourceLineNo">479</span><a id="line.479">                        IndexedCount count = new IndexedCount(RNATools.getCodonAlphabet());</a>
<span class="sourceLineNo">480</span><a id="line.480">                        while (toke.hasMoreTokens()) {</a>
<span class="sourceLineNo">481</span><a id="line.481">                            // check that I haven't read too many values!</a>
<span class="sourceLineNo">482</span><a id="line.482">                            if (idx &gt; 63) continue;</a>
<span class="sourceLineNo">483</span><a id="line.483">                            count.increaseCount(cutg[idx], Double.parseDouble(toke.nextToken()));</a>
<span class="sourceLineNo">484</span><a id="line.484">                            idx++;</a>
<span class="sourceLineNo">485</span><a id="line.485">                        }</a>
<span class="sourceLineNo">486</span><a id="line.486"></a>
<span class="sourceLineNo">487</span><a id="line.487">                        if (idx != 64) continue;</a>
<span class="sourceLineNo">488</span><a id="line.488"></a>
<span class="sourceLineNo">489</span><a id="line.489">                        // ok, I now have the counts and the name, let's stash it</a>
<span class="sourceLineNo">490</span><a id="line.490">                        Distribution codonDist = DistributionTools.countToDistribution(count);</a>
<span class="sourceLineNo">491</span><a id="line.491"></a>
<span class="sourceLineNo">492</span><a id="line.492">                        CodonPref codonPref = new SimpleCodonPref("UNIVERSAL", codonDist, id);</a>
<span class="sourceLineNo">493</span><a id="line.493">                        filter.put(codonPref);</a>
<span class="sourceLineNo">494</span><a id="line.494">                    }</a>
<span class="sourceLineNo">495</span><a id="line.495">                }</a>
<span class="sourceLineNo">496</span><a id="line.496">            }</a>
<span class="sourceLineNo">497</span><a id="line.497">        }</a>
<span class="sourceLineNo">498</span><a id="line.498">        catch (IOException ioe) {}</a>
<span class="sourceLineNo">499</span><a id="line.499">        catch (IllegalSymbolException ise) {}</a>
<span class="sourceLineNo">500</span><a id="line.500">        catch (IllegalAlphabetException iae) {}</a>
<span class="sourceLineNo">501</span><a id="line.501">        catch (ChangeVetoException cve) {}</a>
<span class="sourceLineNo">502</span><a id="line.502">        catch (BioException be) {}</a>
<span class="sourceLineNo">503</span><a id="line.503">    }</a>
<span class="sourceLineNo">504</span><a id="line.504">}</a>
<span class="sourceLineNo">505</span><a id="line.505"></a>




























































</pre>
</div>
</main>
</body>
</html>
