<!DOCTYPE HTML>
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<main role="main">
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span><a id="line.1">/*</a>
<span class="sourceLineNo">002</span><a id="line.2"> *                    BioJava development code</a>
<span class="sourceLineNo">003</span><a id="line.3"> *</a>
<span class="sourceLineNo">004</span><a id="line.4"> * This code may be freely distributed and modified under the</a>
<span class="sourceLineNo">005</span><a id="line.5"> * terms of the GNU Lesser General Public Licence.  This should</a>
<span class="sourceLineNo">006</span><a id="line.6"> * be distributed with the code.  If you do not have a copy,</a>
<span class="sourceLineNo">007</span><a id="line.7"> * see:</a>
<span class="sourceLineNo">008</span><a id="line.8"> *</a>
<span class="sourceLineNo">009</span><a id="line.9"> *      http://www.gnu.org/copyleft/lesser.html</a>
<span class="sourceLineNo">010</span><a id="line.10"> *</a>
<span class="sourceLineNo">011</span><a id="line.11"> * Copyright for this code is held jointly by the individual</a>
<span class="sourceLineNo">012</span><a id="line.12"> * authors.  These should be listed in @author doc comments.</a>
<span class="sourceLineNo">013</span><a id="line.13"> *</a>
<span class="sourceLineNo">014</span><a id="line.14"> * For more information on the BioJava project and its aims,</a>
<span class="sourceLineNo">015</span><a id="line.15"> * or to join the biojava-l mailing list, visit the home page</a>
<span class="sourceLineNo">016</span><a id="line.16"> * at:</a>
<span class="sourceLineNo">017</span><a id="line.17"> *</a>
<span class="sourceLineNo">018</span><a id="line.18"> *      http://www.biojava.org/</a>
<span class="sourceLineNo">019</span><a id="line.19"> *</a>
<span class="sourceLineNo">020</span><a id="line.20"> */</a>
<span class="sourceLineNo">021</span><a id="line.21">package org.biojava.nbio.genome.query;</a>
<span class="sourceLineNo">022</span><a id="line.22"></a>
<span class="sourceLineNo">023</span><a id="line.23">import org.biojava.nbio.genome.parsers.gff.Feature;</a>
<span class="sourceLineNo">024</span><a id="line.24">import org.biojava.nbio.genome.parsers.gff.FeatureI;</a>
<span class="sourceLineNo">025</span><a id="line.25">import org.biojava.nbio.genome.parsers.gff.FeatureList;</a>
<span class="sourceLineNo">026</span><a id="line.26">import org.biojava.nbio.genome.parsers.gff.GeneMarkGTFReader;</a>
<span class="sourceLineNo">027</span><a id="line.27">import org.slf4j.Logger;</a>
<span class="sourceLineNo">028</span><a id="line.28">import org.slf4j.LoggerFactory;</a>
<span class="sourceLineNo">029</span><a id="line.29"></a>
<span class="sourceLineNo">030</span><a id="line.30">import java.io.File;</a>
<span class="sourceLineNo">031</span><a id="line.31">import java.util.ArrayList;</a>
<span class="sourceLineNo">032</span><a id="line.32">import java.util.LinkedHashMap;</a>
<span class="sourceLineNo">033</span><a id="line.33"></a>
<span class="sourceLineNo">034</span><a id="line.34">/**</a>
<span class="sourceLineNo">035</span><a id="line.35"> *</a>
<span class="sourceLineNo">036</span><a id="line.36"> * @author Scooter Willis &lt;willishf at gmail dot com&gt;</a>
<span class="sourceLineNo">037</span><a id="line.37"> */</a>
<span class="sourceLineNo">038</span><a id="line.38">public class OutputHitsGFF {</a>
<span class="sourceLineNo">039</span><a id="line.39"></a>
<span class="sourceLineNo">040</span><a id="line.40">        private static final Logger logger = LoggerFactory.getLogger(OutputHitsGFF.class);</a>
<span class="sourceLineNo">041</span><a id="line.41"></a>
<span class="sourceLineNo">042</span><a id="line.42">        public void process(File blastXMLFile, File gffFile, File gffOutputFile, double maxEScore, double percentageAligned, boolean includeFrameShift, boolean includeNegativeStrand) throws Exception {</a>
<span class="sourceLineNo">043</span><a id="line.43">                BlastXMLQuery blastXMLQuery = new BlastXMLQuery(blastXMLFile.getAbsolutePath());</a>
<span class="sourceLineNo">044</span><a id="line.44">                LinkedHashMap&lt;String, ArrayList&lt;String&gt;&gt; hits = blastXMLQuery.getHitsQueryDef(maxEScore);</a>
<span class="sourceLineNo">045</span><a id="line.45">                FeatureList listGenes = GeneMarkGTFReader.read(gffFile.getAbsolutePath());</a>
<span class="sourceLineNo">046</span><a id="line.46">                FeatureList hitGenes = new FeatureList();</a>
<span class="sourceLineNo">047</span><a id="line.47">                for (String id : hits.keySet()) {</a>
<span class="sourceLineNo">048</span><a id="line.48">                        String[] values = id.split(" ");</a>
<span class="sourceLineNo">049</span><a id="line.49">                        String gene_id = values[0];</a>
<span class="sourceLineNo">050</span><a id="line.50">                        FeatureList gene = listGenes.selectByAttribute("gene_id", gene_id);</a>
<span class="sourceLineNo">051</span><a id="line.51">                        for (FeatureI geneFeature : gene) {</a>
<span class="sourceLineNo">052</span><a id="line.52"></a>
<span class="sourceLineNo">053</span><a id="line.53">                                if (!includeNegativeStrand &amp;&amp; geneFeature.location().isNegative()) {</a>
<span class="sourceLineNo">054</span><a id="line.54">                                        continue;</a>
<span class="sourceLineNo">055</span><a id="line.55">                                }</a>
<span class="sourceLineNo">056</span><a id="line.56">                                if (!includeFrameShift) {</a>
<span class="sourceLineNo">057</span><a id="line.57">                                        boolean frameShift = false;</a>
<span class="sourceLineNo">058</span><a id="line.58">                                        FeatureList cdsList = gene.selectByType("CDS");</a>
<span class="sourceLineNo">059</span><a id="line.59">                                        for(FeatureI cdsFeature : cdsList){</a>
<span class="sourceLineNo">060</span><a id="line.60">                                                int frame = ((Feature)cdsFeature).frame();</a>
<span class="sourceLineNo">061</span><a id="line.61">                                                if(frame != 0){</a>
<span class="sourceLineNo">062</span><a id="line.62">                                                        frameShift = true;</a>
<span class="sourceLineNo">063</span><a id="line.63">                                                        break;</a>
<span class="sourceLineNo">064</span><a id="line.64">                                                }</a>
<span class="sourceLineNo">065</span><a id="line.65">                                        }</a>
<span class="sourceLineNo">066</span><a id="line.66">                                        if(frameShift)</a>
<span class="sourceLineNo">067</span><a id="line.67">                                                continue;</a>
<span class="sourceLineNo">068</span><a id="line.68">                                }</a>
<span class="sourceLineNo">069</span><a id="line.69">                                hitGenes.add(geneFeature);</a>
<span class="sourceLineNo">070</span><a id="line.70">                        }</a>
<span class="sourceLineNo">071</span><a id="line.71">                }</a>
<span class="sourceLineNo">072</span><a id="line.72"></a>
<span class="sourceLineNo">073</span><a id="line.73">        //    GeneMarkGTFReader.write(hitGenes, gffOutputFile.getAbsolutePath());</a>
<span class="sourceLineNo">074</span><a id="line.74">        }</a>
<span class="sourceLineNo">075</span><a id="line.75"></a>
<span class="sourceLineNo">076</span><a id="line.76"></a>
<span class="sourceLineNo">077</span><a id="line.77">                public static void main(String[] args) {</a>
<span class="sourceLineNo">078</span><a id="line.78">                try {</a>
<span class="sourceLineNo">079</span><a id="line.79">                        OutputHitsGFF outputHitsGFF = new OutputHitsGFF();</a>
<span class="sourceLineNo">080</span><a id="line.80">                        outputHitsGFF.process(new File("hits-uniprot_fungi.xml"),</a>
<span class="sourceLineNo">081</span><a id="line.81">                                        new File("genemark_hmm.gtf"),</a>
<span class="sourceLineNo">082</span><a id="line.82">                                        new File("genemark_hits_hmm.gtf"), 0, 100, true, true);</a>
<span class="sourceLineNo">083</span><a id="line.83"></a>
<span class="sourceLineNo">084</span><a id="line.84"></a>
<span class="sourceLineNo">085</span><a id="line.85">                } catch (Exception e) {</a>
<span class="sourceLineNo">086</span><a id="line.86">                        logger.error("Execution: ", e);</a>
<span class="sourceLineNo">087</span><a id="line.87">                }</a>
<span class="sourceLineNo">088</span><a id="line.88">        }</a>
<span class="sourceLineNo">089</span><a id="line.89">}</a>




























































</pre>
</div>
</main>
</body>
</html>
