<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:51 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Class org.biojava.nbio.genome.parsers.gff.FeatureList (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.biojava.nbio.genome.parsers.gff.FeatureList (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/genome/parsers/gff/class-use/FeatureList.html" target="_top">Frames</a></li>
<li><a href="FeatureList.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.genome.parsers.gff.FeatureList" class="title">Uses of Class<br>org.biojava.nbio.genome.parsers.gff.FeatureList</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome">org.biojava.nbio.genome</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome.parsers.gff">org.biojava.nbio.genome.parsers.gff</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.genome">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a> in <a href="../../../../../../../org/biojava/nbio/genome/package-summary.html">org.biojava.nbio.genome</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/genome/package-summary.html">org.biojava.nbio.genome</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GeneFeatureHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/GeneFeatureHelper.html#addGeneIDGFF2GeneFeatures(java.util.LinkedHashMap,%20org.biojava.nbio.genome.parsers.gff.FeatureList)">addGeneIDGFF2GeneFeatures</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                         <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code>
<div class="block">Load GFF2 feature file generated from the geneid prediction algorithm and map features onto the chromosome sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GeneFeatureHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/GeneFeatureHelper.html#addGeneMarkGTFGeneFeatures(java.util.LinkedHashMap,%20org.biojava.nbio.genome.parsers.gff.FeatureList)">addGeneMarkGTFGeneFeatures</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                          <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GeneFeatureHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/GeneFeatureHelper.html#addGlimmerGFF3GeneFeatures(java.util.LinkedHashMap,%20org.biojava.nbio.genome.parsers.gff.FeatureList)">addGlimmerGFF3GeneFeatures</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                          <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GeneFeatureHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/GeneFeatureHelper.html#addGmodGFF3GeneFeatures(java.util.LinkedHashMap,%20org.biojava.nbio.genome.parsers.gff.FeatureList)">addGmodGFF3GeneFeatures</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&gt;&nbsp;chromosomeSequenceList,
                       <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&nbsp;listGenes)</code>
<div class="block">Load GFF3 file using mRNA as the gene feature as not all GFF3 files are complete</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome.parsers.gff">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a> in <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a> that return <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#omitOverlapping(java.lang.String,%20org.biojava.nbio.genome.parsers.gff.Location,%20boolean)">omitOverlapping</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqname,
               <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/Location.html" title="class in org.biojava.nbio.genome.parsers.gff">Location</a>&nbsp;location,
               boolean&nbsp;useBothStrands)</code>
<div class="block">Create a list of all features that do not overlap the specified location on the specified sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">GFF3Reader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/GFF3Reader.html#read(java.lang.String)">read</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;filename)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">GeneIDGFF2Reader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/GeneIDGFF2Reader.html#read(java.lang.String)">read</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;filename)</code>
<div class="block">Read a file into a FeatureList.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">GeneMarkGTFReader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/GeneMarkGTFReader.html#read(java.lang.String)">read</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;filename)</code>
<div class="block">Read a file into a FeatureList.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">GFF3Reader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/GFF3Reader.html#read(java.lang.String,%20java.util.List)">read</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;filename,
    <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;indexes)</code>
<div class="block">Read a file into a FeatureList.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#selectByAttribute(java.lang.String)">selectByAttribute</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;key)</code>
<div class="block">Create a list of all features that include the specified attribute key.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#selectByAttribute(java.lang.String,%20java.lang.String)">selectByAttribute</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;key,
                 <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;value)</code>
<div class="block">Create a list of all features that include the specified attribute key/value pair.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#selectByGroup(java.lang.String)">selectByGroup</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;groupid)</code>
<div class="block">Create a list of all features that have the specified group id, as defined by
 the group() method of the features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#selectByType(java.lang.String)">selectByType</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>
<div class="block">Create a list of all features that are of the specified type, as defined by
 the type() method of the features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#selectByUserData(java.lang.String)">selectByUserData</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;key)</code>
<div class="block">Create a list of all features that include the specified key in their userMap().</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#selectByUserData(java.lang.String,%20java.lang.Object)">selectByUserData</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;key,
                <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;value)</code>
<div class="block">Create a list of all features that include the specified key/value pair in their userMap().</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#selectOverlapping(java.lang.String,%20org.biojava.nbio.genome.parsers.gff.Location,%20boolean)">selectOverlapping</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqname,
                 <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/Location.html" title="class in org.biojava.nbio.genome.parsers.gff">Location</a>&nbsp;location,
                 boolean&nbsp;useBothStrands)</code>
<div class="block">Create a list of all features that overlap the specified location on the specified
 sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#sortByStart()">sortByStart</a></strong>()</code>
<div class="block">Create a new list that is ordered by the starting index of the features' locations.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&gt;</code></td>
<td class="colLast"><span class="strong">FeatureHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureHelper.html#buildFeatureAtrributeIndex(java.lang.String,%20org.biojava.nbio.genome.parsers.gff.FeatureList)">buildFeatureAtrributeIndex</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;attribute,
                          <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&nbsp;list)</code>
<div class="block">Build a list of individual features to allow easy indexing and to avoid iterating through large genome gff3 files
 The index for the returned HashMap is the value of the attribute used to build the index</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&gt;</code></td>
<td class="colLast"><span class="strong">FeatureHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureHelper.html#buildFeatureAtrributeIndex(java.lang.String,%20org.biojava.nbio.genome.parsers.gff.FeatureList)">buildFeatureAtrributeIndex</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;attribute,
                          <a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&nbsp;list)</code>
<div class="block">Build a list of individual features to allow easy indexing and to avoid iterating through large genome gff3 files
 The index for the returned HashMap is the value of the attribute used to build the index</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GeneIDGFF2Reader.</span><code><strong><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/GeneIDGFF2Reader.html#write(org.biojava.nbio.genome.parsers.gff.FeatureList,%20java.lang.String)">write</a></strong>(<a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a>&nbsp;features,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;filename)</code>
<div class="block">Write features in FeatureList to file.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/genome/parsers/gff/class-use/FeatureList.html" target="_top">Frames</a></li>
<li><a href="FeatureList.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
