<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:46 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Jronn (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Jronn (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Jronn.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev Class</li>
<li><a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/nbio/ronn/Jronn.html" target="_top">Frames</a></li>
<li><a href="Jronn.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.ronn</div>
<h2 title="Class Jronn" class="title">Class Jronn</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.ronn.Jronn</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">Jronn</span>
extends <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">This class gives public API to RONN functions.
 It is build on top of the command line client. Due to this fact a few things
 could be improved and extended pending the refactoring of the command line client.

 The input sequence limitations - the input sequence must not contain any ambiguous characters,
 and have a minimum length of 19 amino acids.</div>
<dl><dt><span class="strong">Since:</span></dt>
  <dd>3.0.2</dd>
<dt><span class="strong">Version:</span></dt>
  <dd>1.0</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>Peter Troshin</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a></strong></code>
<div class="block">Holder for the ranges, contain pointers to starting and ending position
 on the sequence which comprises a disordered region.</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#Jronn()">Jronn</a></strong>()</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a></code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#convertProteinSequencetoFasta(org.biojava.nbio.core.sequence.ProteinSequence)">convertProteinSequencetoFasta</a></strong>(<a href="../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Utility method to convert a BioJava ProteinSequence object to the FastaSequence
  object used internally in JRonn.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)">getDisorder</a></strong>(<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</code>
<div class="block">Calculates the disordered regions of the sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,<a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorder(java.util.List)">getDisorder</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</code>
<div class="block">Calculates the disordered regions of the sequence for many sequences in the input.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,<a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorder(java.lang.String)">getDisorder</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fastaFile)</code>
<div class="block">Calculates the disordered regions of the protein sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static float[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorderScores(org.biojava.nbio.data.sequence.FastaSequence)">getDisorderScores</a></strong>(<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</code>
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,float[]&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorderScores(java.util.List)">getDisorderScores</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</code>
<div class="block">Calculates the probability of disorder scores for each residue in the sequence for
 many sequences in the input.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static float[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorderScores(org.biojava.nbio.core.sequence.ProteinSequence)">getDisorderScores</a></strong>(<a href="../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</code>
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../org/biojava/nbio/ronn/Jronn.html#scoresToRanges(float[],%20float)">scoresToRanges</a></strong>(float[]&nbsp;scores,
              float&nbsp;probability)</code>
<div class="block">Convert raw scores to ranges.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="Jronn()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>Jronn</h4>
<pre>public&nbsp;Jronn()</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getDisorderScores(org.biojava.nbio.data.sequence.FastaSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorderScores</h4>
<pre>public static&nbsp;float[]&nbsp;getDisorderScores(<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</pre>
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.
 In general, values greater than 0.5 considered to be in the disordered regions.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - an instance of FastaSequence object, holding the name and the sequence.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the probability scores for each residue in the sequence</dd></dl>
</li>
</ul>
<a name="getDisorderScores(org.biojava.nbio.core.sequence.ProteinSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorderScores</h4>
<pre>public static&nbsp;float[]&nbsp;getDisorderScores(<a href="../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</pre>
<div class="block">Calculates the probability value for each residue in the protein sequence,
 telling the probability that the residue belongs to disordered region.
 In general, values greater than 0.5 considered to be in the disordered regions.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - an instance of FastaSequence object, holding the name and the sequence.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the probability scores for each residue in the sequence</dd></dl>
</li>
</ul>
<a name="convertProteinSequencetoFasta(org.biojava.nbio.core.sequence.ProteinSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>convertProteinSequencetoFasta</h4>
<pre>public static&nbsp;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;convertProteinSequencetoFasta(<a href="../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence)</pre>
<div class="block">Utility method to convert a BioJava ProteinSequence object to the FastaSequence
  object used internally in JRonn.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getDisorder(org.biojava.nbio.data.sequence.FastaSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorder</h4>
<pre>public static&nbsp;<a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&nbsp;getDisorder(<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&nbsp;sequence)</pre>
<div class="block">Calculates the disordered regions of the sequence. More formally, the regions for which the
 probability of disorder is greater then 0.50.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequence</code> - an instance of FastaSequence object, holding the name and the sequence.</dd>
<dt><span class="strong">Returns:</span></dt><dd>the array of ranges if there are any residues predicted to have the
 probability of disorder greater then 0.5, null otherwise.</dd></dl>
</li>
</ul>
<a name="scoresToRanges(float[], float)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>scoresToRanges</h4>
<pre>public static&nbsp;<a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&nbsp;scoresToRanges(float[]&nbsp;scores,
                           float&nbsp;probability)</pre>
<div class="block">Convert raw scores to ranges. Gives ranges for given probability of disorder value</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>scores</code> - the raw probability of disorder scores for each residue in the sequence.</dd><dd><code>probability</code> - the cut off threshold. Include all residues with the probability of disorder greater then this value</dd>
<dt><span class="strong">Returns:</span></dt><dd>the array of ranges if there are any residues predicted to have the
 probability of disorder greater then <code>probability</code>, null otherwise.</dd></dl>
</li>
</ul>
<a name="getDisorderScores(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorderScores</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,float[]&gt;&nbsp;getDisorderScores(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</pre>
<div class="block">Calculates the probability of disorder scores for each residue in the sequence for
 many sequences in the input.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - the list of the FastaSequence objects</dd>
<dt><span class="strong">Returns:</span></dt><dd>the Map with key->FastaSequence, value->probability of disorder for each residue</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)"><code>getDisorder(FastaSequence)</code></a></dd></dl>
</li>
</ul>
<a name="getDisorder(java.util.List)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDisorder</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,<a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;&nbsp;getDisorder(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>&gt;&nbsp;sequences)</pre>
<div class="block">Calculates the disordered regions of the sequence for many sequences in the input.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequences</code> - sequences the list of the FastaSequence objects</dd>
<dt><span class="strong">Returns:</span></dt><dd></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)"><code>getDisorder(FastaSequence)</code></a></dd></dl>
</li>
</ul>
<a name="getDisorder(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getDisorder</h4>
<pre>public static&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="../../../../org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a>,<a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a>[]&gt;&nbsp;getDisorder(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;fastaFile)
                                                    throws <a href="http://docs.oracle.com/javase/7/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io">FileNotFoundException</a>,
                                                           <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Calculates the disordered regions of the protein sequence.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fastaFile</code> - input file name containing the sequence in FASTA</dd>
<dt><span class="strong">Returns:</span></dt><dd>the Map with key->FastaSequence, value->the list of disordered regions for each sequence</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io">FileNotFoundException</a></code> - if the input file cannot be found</dd>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - of the system cannot access or read from the input file</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../org/biojava/nbio/ronn/Jronn.html#getDisorder(org.biojava.nbio.data.sequence.FastaSequence)"><code>getDisorder(FastaSequence)</code></a>, 
<code>#Jronn.Range</code></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Jronn.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev Class</li>
<li><a href="../../../../org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?org/biojava/nbio/ronn/Jronn.html" target="_top">Frames</a></li>
<li><a href="Jronn.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
