<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: demo, class: DemoDistanceTree">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> */</span>
<span class="source-line-no">021</span><span id="line-21">package demo;</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">import java.io.InputStream;</span>
<span class="source-line-no">024</span><span id="line-24">import java.util.LinkedHashMap;</span>
<span class="source-line-no">025</span><span id="line-25">import java.util.Map;</span>
<span class="source-line-no">026</span><span id="line-26"></span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.MultipleSequenceAlignment;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.core.sequence.io.FastaReader;</span>
<span class="source-line-no">034</span><span id="line-34">import org.biojava.nbio.core.sequence.io.GenericFastaHeaderParser;</span>
<span class="source-line-no">035</span><span id="line-35">import org.biojava.nbio.core.sequence.io.ProteinSequenceCreator;</span>
<span class="source-line-no">036</span><span id="line-36">import org.biojava.nbio.phylo.DistanceMatrixCalculator;</span>
<span class="source-line-no">037</span><span id="line-37">import org.biojava.nbio.phylo.DistanceTreeEvaluator;</span>
<span class="source-line-no">038</span><span id="line-38">import org.biojava.nbio.phylo.ForesterWrapper;</span>
<span class="source-line-no">039</span><span id="line-39">import org.biojava.nbio.phylo.TreeConstructor;</span>
<span class="source-line-no">040</span><span id="line-40">import org.biojava.nbio.phylo.TreeConstructorType;</span>
<span class="source-line-no">041</span><span id="line-41">import org.forester.evoinference.matrix.distance.BasicSymmetricalDistanceMatrix;</span>
<span class="source-line-no">042</span><span id="line-42">import org.forester.evoinference.matrix.distance.DistanceMatrix;</span>
<span class="source-line-no">043</span><span id="line-43">import org.forester.phylogeny.Phylogeny;</span>
<span class="source-line-no">044</span><span id="line-44"></span>
<span class="source-line-no">045</span><span id="line-45">/**</span>
<span class="source-line-no">046</span><span id="line-46"> * This demo contains the CookBook example to create a phylogenetic tree from a</span>
<span class="source-line-no">047</span><span id="line-47"> * multiple sequence alignment (MSA).</span>
<span class="source-line-no">048</span><span id="line-48"> *</span>
<span class="source-line-no">049</span><span id="line-49"> * @author Scooter Willis</span>
<span class="source-line-no">050</span><span id="line-50"> * @author Aleix Lafita</span>
<span class="source-line-no">051</span><span id="line-51"> *</span>
<span class="source-line-no">052</span><span id="line-52"> */</span>
<span class="source-line-no">053</span><span id="line-53">public class DemoDistanceTree {</span>
<span class="source-line-no">054</span><span id="line-54"></span>
<span class="source-line-no">055</span><span id="line-55">        public static void main(String[] args) throws Exception {</span>
<span class="source-line-no">056</span><span id="line-56"></span>
<span class="source-line-no">057</span><span id="line-57">                // 0. This is just to load an example MSA from a FASTA file</span>
<span class="source-line-no">058</span><span id="line-58">                InputStream inStream = TreeConstructor.class</span>
<span class="source-line-no">059</span><span id="line-59">                                .getResourceAsStream("/PF00104_small.fasta");</span>
<span class="source-line-no">060</span><span id="line-60"></span>
<span class="source-line-no">061</span><span id="line-61">                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader =</span>
<span class="source-line-no">062</span><span id="line-62">                                new FastaReader&lt;&gt;(</span>
<span class="source-line-no">063</span><span id="line-63">                                inStream,</span>
<span class="source-line-no">064</span><span id="line-64">                                new GenericFastaHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),</span>
<span class="source-line-no">065</span><span id="line-65">                                new ProteinSequenceCreator(AminoAcidCompoundSet</span>
<span class="source-line-no">066</span><span id="line-66">                                                .getAminoAcidCompoundSet()));</span>
<span class="source-line-no">067</span><span id="line-67"></span>
<span class="source-line-no">068</span><span id="line-68">                Map&lt;String, ProteinSequence&gt; proteinSequences =</span>
<span class="source-line-no">069</span><span id="line-69">                                fastaReader.process();</span>
<span class="source-line-no">070</span><span id="line-70"></span>
<span class="source-line-no">071</span><span id="line-71">                inStream.close();</span>
<span class="source-line-no">072</span><span id="line-72"></span>
<span class="source-line-no">073</span><span id="line-73">                MultipleSequenceAlignment&lt;ProteinSequence, AminoAcidCompound&gt; msa =</span>
<span class="source-line-no">074</span><span id="line-74">                                new MultipleSequenceAlignment&lt;&gt;();</span>
<span class="source-line-no">075</span><span id="line-75"></span>
<span class="source-line-no">076</span><span id="line-76">                for (ProteinSequence proteinSequence : proteinSequences.values()) {</span>
<span class="source-line-no">077</span><span id="line-77">                        msa.addAlignedSequence(proteinSequence);</span>
<span class="source-line-no">078</span><span id="line-78">                }</span>
<span class="source-line-no">079</span><span id="line-79"></span>
<span class="source-line-no">080</span><span id="line-80">                long readT = System.currentTimeMillis();</span>
<span class="source-line-no">081</span><span id="line-81"></span>
<span class="source-line-no">082</span><span id="line-82">                // 1. Calculate the evolutionary distance matrix (can take long)</span>
<span class="source-line-no">083</span><span id="line-83">                SubstitutionMatrix&lt;AminoAcidCompound&gt; M = SubstitutionMatrixHelper</span>
<span class="source-line-no">084</span><span id="line-84">                                .getBlosum62();</span>
<span class="source-line-no">085</span><span id="line-85">                DistanceMatrix DM = DistanceMatrixCalculator</span>
<span class="source-line-no">086</span><span id="line-86">                                .dissimilarityScore(msa, M);</span>
<span class="source-line-no">087</span><span id="line-87"></span>
<span class="source-line-no">088</span><span id="line-88">                // 2. Construct a distance tree using the NJ algorithm</span>
<span class="source-line-no">089</span><span id="line-89">                Phylogeny phylo = TreeConstructor.distanceTree(</span>
<span class="source-line-no">090</span><span id="line-90">                                (BasicSymmetricalDistanceMatrix) DM, TreeConstructorType.NJ);</span>
<span class="source-line-no">091</span><span id="line-91"></span>
<span class="source-line-no">092</span><span id="line-92">                long treeT = System.currentTimeMillis();</span>
<span class="source-line-no">093</span><span id="line-93">                String newick = ForesterWrapper.getNewickString(phylo, true);</span>
<span class="source-line-no">094</span><span id="line-94">                System.out.println(newick);</span>
<span class="source-line-no">095</span><span id="line-95">                System.out.println("Tree Construction: " + (treeT - readT) + " ms.");</span>
<span class="source-line-no">096</span><span id="line-96"></span>
<span class="source-line-no">097</span><span id="line-97">                // 3. Evaluate the goodness of fit of the tree</span>
<span class="source-line-no">098</span><span id="line-98">                double cv = DistanceTreeEvaluator.evaluate(phylo, DM);</span>
<span class="source-line-no">099</span><span id="line-99">                System.out.println("CV of the tree: " + (int) (cv * 100) + " %");</span>
<span class="source-line-no">100</span><span id="line-100"></span>
<span class="source-line-no">101</span><span id="line-101">        }</span>
<span class="source-line-no">102</span><span id="line-102">}</span>




























































</pre>
</div>
</main>
</body>
</html>
