<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Uses of Interface org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="use: package: org.biojava.nbio.core.sequence.io.template, interface: SequenceHeaderParserInterface">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../../../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Interface org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface" class="title">Uses of Interface<br>org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface</h1>
</div>
<div class="caption"><span>Packages that use <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></div>
<div class="col-last odd-row-color">
<div class="block">Input and Output of Structures</div>
</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="org.biojava.nbio.core.sequence.io">
<h2>Uses of <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a> in <a href="../../package-summary.html">org.biojava.nbio.core.sequence.io</a></h2>
<div class="caption"><span>Classes in <a href="../../package-summary.html">org.biojava.nbio.core.sequence.io</a> that implement <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../GenericFastaHeaderParser.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderParser</a>&lt;S extends <a href="../../../template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../../../template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources

 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface</div>
</div>
<div class="col-first odd-row-color"><code>class&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../GenericGenbankHeaderParser.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser</a>&lt;S extends <a href="../../../template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../../../template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../PlainFastaHeaderParser.html" class="type-name-link" title="class in org.biojava.nbio.core.sequence.io">PlainFastaHeaderParser</a>&lt;S extends <a href="../../../template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;C&gt;,<wbr>C extends <a href="../../../template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></div>
<div class="col-last even-row-color">
<div class="block">The plain fasta header takes everything in the header as a single entity.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../package-summary.html">org.biojava.nbio.core.sequence.io</a> that return <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>protected <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a><wbr>&lt;<a href="../../../ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastaStreamer.</span><code><a href="../../FastaStreamer.html#getHeaderParser()" class="member-name-link">getHeaderParser</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../../FastaStreamer.html" title="class in org.biojava.nbio.core.sequence.io">FastaStreamer</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">FastaStreamer.</span><code><a href="../../FastaStreamer.html#withHeaderParser(org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface)" class="member-name-link">withHeaderParser</a><wbr>(<a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../package-summary.html">org.biojava.nbio.core.sequence.io</a> with parameters of type <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../FastaReader.html#%3Cinit%3E(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">S</a>,<wbr><a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last even-row-color">
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you
 need to use this constructor because we need details about
 the location of the file.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../FastaReader.html#%3Cinit%3E(java.io.InputStream,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">FastaReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;is,
 <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">S</a>,<wbr><a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../FastaReader.html" title="type parameter in FastaReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last odd-row-color">
<div class="block">If you are going to use FileProxyProteinSequenceCreator then do not use this constructor because we need details about
 local file offsets for quick reads.</div>
</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../GenbankReader.html#%3Cinit%3E(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">GenbankReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../GenbankReader.html" title="type parameter in GenbankReader">S</a>,<wbr><a href="../../GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last even-row-color">
<div class="block">If you are going to use the FileProxyProteinSequenceCreator then you
 need to use this constructor because we need details about
 the location of the file.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../GenbankReader.html#%3Cinit%3E(java.io.InputStream,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface)" class="member-name-link">GenbankReader</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;is,
 <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../GenbankReader.html" title="type parameter in GenbankReader">S</a>,<wbr><a href="../../GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../GenbankReader.html" title="type parameter in GenbankReader">C</a>&gt;&nbsp;sequenceCreator)</code></div>
<div class="col-last odd-row-color">
<div class="block">If you are going to use <a href="../../FileProxyProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io"><code>FileProxyProteinSequenceCreator</code></a> then do not use this constructor because we need details about
 local file offsets for quick reads.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.io">
<h2>Uses of <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a> in <a href="../../../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h2>
<div class="caption"><span>Constructors in <a href="../../../../../structure/io/package-summary.html">org.biojava.nbio.structure.io</a> with parameters of type <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../../../../structure/io/FastaStructureParser.html#%3Cinit%3E(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)" class="member-name-link">FastaStructureParser</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
 <a href="../../../../../structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../../../../structure/io/FastaStructureParser.html#%3Cinit%3E(java.io.InputStream,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)" class="member-name-link">FastaStructureParser</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;is,
 <a href="../SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
 <a href="../SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
 <a href="../../../../../structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
