<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>FastaStructureParser (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.structure.io, class: FastaStructureParser">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/FastaStructureParser.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html#class">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Summary:</p>
<ul>
<li>Nested</li>
<li>Field</li>
<li><a href="#constructor-summary">Constr</a></li>
<li><a href="#method-summary">Method</a></li>
</ul>
</li>
<li>
<p>Detail:</p>
<ul>
<li>Field</li>
<li><a href="#constructor-detail">Constr</a></li>
<li><a href="#method-detail">Method</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.structure.io</a></div>
<h1 title="Class FastaStructureParser" class="title">Class FastaStructureParser</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">org.biojava.nbio.structure.io.FastaStructureParser</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-58">FastaStructureParser</a></span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
<div class="block">Reads a protein sequence from a fasta file and attempts to match it to a
 3D structure. Any gaps ('-') in the fasta file are preserved as null atoms in
 the output, allowing structural alignments to be read from fasta files.

 <p>Structures are loaded from an AtomCache. For this to work, the accession
 for each protein should be parsed from the fasta header line into a form
 understood by <a href="../align/util/AtomCache.html#getStructure(java.lang.String)"><code>AtomCache.getStructure(String)</code></a>.

 <p>Lowercase letters are sometimes used to specify unaligned residues.
 This information can be preserved by using a CasePreservingSequenceCreator,
 which allows the case of residues to be accessed through the
 <a href="../../core/sequence/template/AbstractSequence.html#getUserCollection()"><code>AbstractSequence.getUserCollection()</code></a> method.</div>
<dl class="notes">
<dt>Author:</dt>
<dd>Spencer Bliven</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)" class="member-name-link">FastaStructureParser</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
 <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
 <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-constructor-name odd-row-color"><code><a href="#%3Cinit%3E(java.io.InputStream,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)" class="member-name-link">FastaStructureParser</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;is,
 <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
 <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
 <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E(org.biojava.nbio.core.sequence.io.FastaReader,org.biojava.nbio.structure.align.util.AtomCache)" class="member-name-link">FastaStructureParser</a><wbr>(<a href="../../core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;reader,
 <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel" aria-labelledby="method-summary-table-tab0">
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getAccessions()" class="member-name-link">getAccessions</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Gets the protein accessions mapped from the Fasta file.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[][]</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getResidues()" class="member-name-link">getResidues</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">For each residue in the fasta file, return the ResidueNumber in the
 corresponding structure.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getSequences()" class="member-name-link">getSequences</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Gets the protein sequences read from the Fasta file.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>[]</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#getStructures()" class="member-name-link">getStructures</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Gets the protein structures mapped from the Fasta file.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4"><code><a href="#process()" class="member-name-link">process</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab4">
<div class="block">Parses the fasta file and loads it into memory.</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#clone()" title="class or interface in java.lang" class="external-link">clone</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#finalize()" title="class or interface in java.lang" class="external-link">finalize</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;(java.io.InputStream,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)">
<h3>FastaStructureParser</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-70">FastaStructureParser</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html" title="class or interface in java.io" class="external-link">InputStream</a>&nbsp;is,
 <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
 <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
 <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</span></div>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(java.io.File,org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,org.biojava.nbio.structure.align.util.AtomCache)">
<h3>FastaStructureParser</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-79">FastaStructureParser</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html" title="class or interface in java.io" class="external-link">File</a>&nbsp;file,
 <a href="../../core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
 <a href="../../core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
 <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</span>
                     throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/FileNotFoundException.html" title="class or interface in java.io" class="external-link">FileNotFoundException</a></span></div>
<dl class="notes">
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/FileNotFoundException.html" title="class or interface in java.io" class="external-link">FileNotFoundException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="&lt;init&gt;(org.biojava.nbio.core.sequence.io.FastaReader,org.biojava.nbio.structure.align.util.AtomCache)">
<h3>FastaStructureParser</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-88">FastaStructureParser</a></span><wbr><span class="parameters">(<a href="../../core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;reader,
 <a href="../align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</span></div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="process()">
<h3>process</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-111">process</a></span>()
             throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a>,
<a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span></div>
<div class="block">Parses the fasta file and loads it into memory.

 Information can be subsequently accessed through
 <a href="#getSequences()"><code>getSequences()</code></a>,
 <a href="#getStructures()"><code>getStructures()</code></a>,
 <a href="#getResidues()"><code>getResidues()</code></a>, and
 <a href="#getAccessions()"><code>getAccessions()</code></a>.</div>
<dl class="notes">
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/IOException.html" title="class or interface in java.io" class="external-link">IOException</a></code></dd>
<dd><code><a href="../StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSequences()">
<h3>getSequences</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-141">getSequences</a></span>()</div>
<div class="block">Gets the protein sequences read from the Fasta file.
 Returns null if <a href="#process()"><code>process()</code></a> has not been called.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>An array ProteinSequences from
  parsing the fasta file, or null if process() hasn't been called.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStructures()">
<h3>getStructures</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>[]</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-151">getStructures</a></span>()</div>
<div class="block">Gets the protein structures mapped from the Fasta file.
 Returns null if <a href="#process()"><code>process()</code></a> has not been called.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>An array of Structures for each protein
  in the fasta file, or null if process() hasn't been called.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getResidues()">
<h3>getResidues</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="../ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[][]</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-164">getResidues</a></span>()</div>
<div class="block">For each residue in the fasta file, return the ResidueNumber in the
 corresponding structure. If the residue cannot be found in the structure,
 that entry will be null. This can happen if that residue was not included
 in the PDB file (eg disordered residues), if the fasta sequence does not
 match the PDB sequence, or if errors occur during the matching process.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>A 2D array of ResidueNumbers, or null if process() hasn't been called.</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list-long">
<li><a href="StructureSequenceMatcher.html#matchSequenceToStructure(org.biojava.nbio.core.sequence.ProteinSequence,org.biojava.nbio.structure.Structure)"><code>StructureSequenceMatcher.matchSequenceToStructure(ProteinSequence, Structure)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAccessions()">
<h3>getAccessions</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>[]</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/structure/io/FastaStructureParser.html#line-174">getAccessions</a></span>()</div>
<div class="block">Gets the protein accessions mapped from the Fasta file.
 Returns null if <a href="#process()"><code>process()</code></a> has not been called.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>An array of Structures for each protein
  in the fasta file, or null if process() hasn't been called.</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
