<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>org.biojava.nbio.structure.io (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.structure.io">
<meta name="generator" content="javadoc/PackageWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="package-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li class="nav-bar-cell1-rev">Package</li>
<li>Class</li>
<li><a href="package-use.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html#package">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Package:</p>
<ul>
<li><a href="#package-description">Description</a></li>
<li><a href="#related-package-summary">Related Packages</a></li>
<li><a href="#class-summary">Classes and Interfaces</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Package:&nbsp;</li>
<li><a href="#package-description">Description</a>&nbsp;|&nbsp;</li>
<li><a href="#related-package-summary">Related Packages</a>&nbsp;|&nbsp;</li>
<li><a href="#class-summary">Classes and Interfaces</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Package org.biojava.nbio.structure.io" class="title">Package org.biojava.nbio.structure.io</h1>
</div>
<hr>
<div class="package-signature">package <span class="element-name">org.biojava.nbio.structure.io</span></div>
<section class="package-description" id="package-description">
<div class="block">Input and Output of Structures</div>
</section>
<section class="summary">
<ul class="summary-list">
<li>
<div id="related-package-summary">
<div class="caption"><span>Related Packages</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="../package-summary.html">org.biojava.nbio.structure</a></div>
<div class="col-last even-row-color">
<div class="block">Interfaces and classes for protein structure (PDB).</div>
</div>
<div class="col-first odd-row-color"><a href="cif/package-summary.html">org.biojava.nbio.structure.io.cif</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="mmtf/package-summary.html">org.biojava.nbio.structure.io.mmtf</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="sifts/package-summary.html">org.biojava.nbio.structure.io.sifts</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="util/package-summary.html">org.biojava.nbio.structure.io.util</a></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</div>
</li>
<li>
<div id="class-summary">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="class-summary-tab0" role="tab" aria-selected="true" aria-controls="class-summary.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('class-summary', 'class-summary', 2)" class="active-table-tab">All Classes and Interfaces</button><button id="class-summary-tab1" role="tab" aria-selected="false" aria-controls="class-summary.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('class-summary', 'class-summary-tab1', 2)" class="table-tab">Interfaces</button><button id="class-summary-tab2" role="tab" aria-selected="false" aria-controls="class-summary.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('class-summary', 'class-summary-tab2', 2)" class="table-tab">Classes</button><button id="class-summary-tab3" role="tab" aria-selected="false" aria-controls="class-summary.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('class-summary', 'class-summary-tab3', 2)" class="table-tab">Enums</button><button id="class-summary-tab5" role="tab" aria-selected="false" aria-controls="class-summary.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('class-summary', 'class-summary-tab5', 2)" class="table-tab">Exception Classes</button></div>
<div id="class-summary.tabpanel" role="tabpanel" aria-labelledby="class-summary-tab0">
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="BcifFileReader.html" title="class in org.biojava.nbio.structure.io">BcifFileReader</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Parse binary Cif files and provide capabilities to store them locally.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="BondMaker.html" title="class in org.biojava.nbio.structure.io">BondMaker</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Adds polymer bonds for peptides and nucleotides based on distance cutoffs and
 intra-group (residue) bonds based on data from the Chemical Component Dictionary
 to the Structure object.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="CAConverter.html" title="class in org.biojava.nbio.structure.io">CAConverter</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Converts full atom representations to Calpha only ones.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="ChargeAdder.html" title="class in org.biojava.nbio.structure.io">ChargeAdder</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">A class to add appropriate charge information to a structure.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="CifFileReader.html" title="class in org.biojava.nbio.structure.io">CifFileReader</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Parse text Cif files and provide capabilities to store them locally.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="EntityFinder.html" title="class in org.biojava.nbio.structure.io">EntityFinder</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Heuristical finding of Entities (called Compounds in legacy PDB format)
 in a given Structure.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io">FastaAFPChainConverter</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">A collection of static utilities to convert between <a href="../align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model"><code>AFPChains</code></a> and FastaSequences.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="FastaStructureParser.html" title="class in org.biojava.nbio.structure.io">FastaStructureParser</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Reads a protein sequence from a fasta file and attempts to match it to a
 3D structure.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="FileConvert.html" title="class in org.biojava.nbio.structure.io">FileConvert</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">Methods to convert a structure object into different file formats.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="FileParsingParameters.html" title="class in org.biojava.nbio.structure.io">FileParsingParameters</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">A class that configures parameters that can be sent to the PDB file parsers

 
  <a href="FileParsingParameters.html#setParseCAOnly(boolean)"><code>FileParsingParameters.setParseCAOnly(boolean)</code></a> - parse only the Atom records for C-alpha atoms
  <a href="FileParsingParameters.html#setParseSecStruc(boolean)"><code>FileParsingParameters.setParseSecStruc(boolean)</code></a> - a flag if the secondary structure information from the PDB file (author's assignment) should be parsed.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="GroupToSDF.html" title="class in org.biojava.nbio.structure.io">GroupToSDF</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">&nbsp;</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="LocalPDBDirectory.html" title="class in org.biojava.nbio.structure.io">LocalPDBDirectory</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Superclass for classes which download and interact with the PDB's FTP server,
 specifically <a href="PDBFileReader.html" title="class in org.biojava.nbio.structure.io"><code>PDBFileReader</code></a> and <a href="CifFileReader.html" title="class in org.biojava.nbio.structure.io"><code>CifFileReader</code></a>.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab3"><a href="LocalPDBDirectory.FetchBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.FetchBehavior</a></div>
<div class="col-last even-row-color class-summary class-summary-tab3">
<div class="block">Controls when the class should fetch files from the ftp server</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab3"><a href="LocalPDBDirectory.ObsoleteBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.ObsoleteBehavior</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab3">
<div class="block">Behaviors for when an obsolete structure is requested.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="MMTFFileReader.html" title="class in org.biojava.nbio.structure.io">MMTFFileReader</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">A class to read MMTF files and cache them locally.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="PDBBioAssemblyParser.html" title="class in org.biojava.nbio.structure.io">PDBBioAssemblyParser</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Parses REMARK 350 records in a PDB file and creates transformations to
 construct the quaternary structure of a protein from an asymmetric unit</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="PDBFileParser.html" title="class in org.biojava.nbio.structure.io">PDBFileParser</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">This class implements the actual PDB file parsing.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="PDBFileReader.html" title="class in org.biojava.nbio.structure.io">PDBFileReader</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">The wrapper class for parsing a PDB file.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab5"><a href="PDBParseException.html" title="class in org.biojava.nbio.structure.io">PDBParseException</a></div>
<div class="col-last even-row-color class-summary class-summary-tab5">
<div class="block">An exception during the parsing of a PDB file.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab2"><a href="SeqRes2AtomAligner.html" title="class in org.biojava.nbio.structure.io">SeqRes2AtomAligner</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab2">
<div class="block">Aligns the SEQRES residues to the ATOM residues.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="SSBondImpl.html" title="class in org.biojava.nbio.structure.io">SSBondImpl</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">A simple bean to store disulfide bridge information, the SSBOND records in the PDB files.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab3"><a href="StructureFiletype.html" title="enum in org.biojava.nbio.structure.io">StructureFiletype</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab3">
<div class="block">An enum of supported file formats.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab1"><a href="StructureIOFile.html" title="interface in org.biojava.nbio.structure.io">StructureIOFile</a></div>
<div class="col-last even-row-color class-summary class-summary-tab1">
<div class="block">StructureIOFile extends StructureProvider with methods specific to
  parsing files from the filesystem.</div>
</div>
<div class="col-first odd-row-color class-summary class-summary-tab1"><a href="StructureProvider.html" title="interface in org.biojava.nbio.structure.io">StructureProvider</a></div>
<div class="col-last odd-row-color class-summary class-summary-tab1">
<div class="block">A class that can provide a protein structure object from somewhere.</div>
</div>
<div class="col-first even-row-color class-summary class-summary-tab2"><a href="StructureSequenceMatcher.html" title="class in org.biojava.nbio.structure.io">StructureSequenceMatcher</a></div>
<div class="col-last even-row-color class-summary class-summary-tab2">
<div class="block">A utility class with methods for matching ProteinSequences with
 Structures.</div>
</div>
</div>
</div>
</div>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
