<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojavax.bio.seq.io;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.io.BufferedInputStream;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.io.BufferedReader;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.io.File;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.io.FileReader;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.io.IOException;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.io.InputStreamReader;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.io.PrintStream;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.io.PrintWriter;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.ArrayList;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.Iterator;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import java.util.List;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import java.util.Map;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import java.util.Set;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import java.util.TreeMap;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import java.util.TreeSet;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import java.util.regex.Matcher;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import java.util.regex.Pattern;<a name="line.40"></a>
<span class="sourceLineNo">041</span><a name="line.41"></a>
<span class="sourceLineNo">042</span>import javax.xml.parsers.ParserConfigurationException;<a name="line.42"></a>
<span class="sourceLineNo">043</span><a name="line.43"></a>
<span class="sourceLineNo">044</span>import org.biojava.bio.proteomics.MassCalc;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import org.biojava.bio.seq.Sequence;<a name="line.45"></a>
<span class="sourceLineNo">046</span>import org.biojava.bio.seq.io.ParseException;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import org.biojava.bio.seq.io.SeqIOListener;<a name="line.47"></a>
<span class="sourceLineNo">048</span>import org.biojava.bio.seq.io.SymbolTokenization;<a name="line.48"></a>
<span class="sourceLineNo">049</span>import org.biojava.bio.symbol.IllegalSymbolException;<a name="line.49"></a>
<span class="sourceLineNo">050</span>import org.biojava.bio.symbol.Location;<a name="line.50"></a>
<span class="sourceLineNo">051</span>import org.biojava.bio.symbol.SimpleSymbolList;<a name="line.51"></a>
<span class="sourceLineNo">052</span>import org.biojava.bio.symbol.Symbol;<a name="line.52"></a>
<span class="sourceLineNo">053</span>import org.biojava.bio.symbol.SymbolList;<a name="line.53"></a>
<span class="sourceLineNo">054</span>import org.biojava.utils.ChangeVetoException;<a name="line.54"></a>
<span class="sourceLineNo">055</span>import org.biojava.utils.xml.PrettyXMLWriter;<a name="line.55"></a>
<span class="sourceLineNo">056</span>import org.biojava.utils.xml.XMLWriter;<a name="line.56"></a>
<span class="sourceLineNo">057</span>import org.biojavax.Comment;<a name="line.57"></a>
<span class="sourceLineNo">058</span>import org.biojavax.CrossRef;<a name="line.58"></a>
<span class="sourceLineNo">059</span>import org.biojavax.DocRef;<a name="line.59"></a>
<span class="sourceLineNo">060</span>import org.biojavax.DocRefAuthor;<a name="line.60"></a>
<span class="sourceLineNo">061</span>import org.biojavax.Namespace;<a name="line.61"></a>
<span class="sourceLineNo">062</span>import org.biojavax.Note;<a name="line.62"></a>
<span class="sourceLineNo">063</span>import org.biojavax.RankedCrossRef;<a name="line.63"></a>
<span class="sourceLineNo">064</span>import org.biojavax.RankedDocRef;<a name="line.64"></a>
<span class="sourceLineNo">065</span>import org.biojavax.RichAnnotation;<a name="line.65"></a>
<span class="sourceLineNo">066</span>import org.biojavax.RichObjectFactory;<a name="line.66"></a>
<span class="sourceLineNo">067</span>import org.biojavax.SimpleCrossRef;<a name="line.67"></a>
<span class="sourceLineNo">068</span>import org.biojavax.SimpleDocRef;<a name="line.68"></a>
<span class="sourceLineNo">069</span>import org.biojavax.SimpleDocRefAuthor;<a name="line.69"></a>
<span class="sourceLineNo">070</span>import org.biojavax.SimpleNamespace;<a name="line.70"></a>
<span class="sourceLineNo">071</span>import org.biojavax.SimpleNote;<a name="line.71"></a>
<span class="sourceLineNo">072</span>import org.biojavax.SimpleRankedCrossRef;<a name="line.72"></a>
<span class="sourceLineNo">073</span>import org.biojavax.SimpleRankedDocRef;<a name="line.73"></a>
<span class="sourceLineNo">074</span>import org.biojavax.SimpleRichAnnotation;<a name="line.74"></a>
<span class="sourceLineNo">075</span>import org.biojavax.bio.seq.Position;<a name="line.75"></a>
<span class="sourceLineNo">076</span>import org.biojavax.bio.seq.RichFeature;<a name="line.76"></a>
<span class="sourceLineNo">077</span>import org.biojavax.bio.seq.RichLocation;<a name="line.77"></a>
<span class="sourceLineNo">078</span>import org.biojavax.bio.seq.RichSequence;<a name="line.78"></a>
<span class="sourceLineNo">079</span>import org.biojavax.bio.seq.io.UniProtCommentParser.Event;<a name="line.79"></a>
<span class="sourceLineNo">080</span>import org.biojavax.bio.seq.io.UniProtCommentParser.Interaction;<a name="line.80"></a>
<span class="sourceLineNo">081</span>import org.biojavax.bio.seq.io.UniProtCommentParser.Isoform;<a name="line.81"></a>
<span class="sourceLineNo">082</span>import org.biojavax.bio.taxa.NCBITaxon;<a name="line.82"></a>
<span class="sourceLineNo">083</span>import org.biojavax.bio.taxa.SimpleNCBITaxon;<a name="line.83"></a>
<span class="sourceLineNo">084</span>import org.biojavax.ontology.ComparableOntology;<a name="line.84"></a>
<span class="sourceLineNo">085</span>import org.biojavax.ontology.ComparableTerm;<a name="line.85"></a>
<span class="sourceLineNo">086</span>import org.biojavax.ontology.SimpleComparableOntology;<a name="line.86"></a>
<span class="sourceLineNo">087</span>import org.biojavax.utils.CRC64Checksum;<a name="line.87"></a>
<span class="sourceLineNo">088</span>import org.biojavax.utils.StringTools;<a name="line.88"></a>
<span class="sourceLineNo">089</span>import org.biojavax.utils.XMLTools;<a name="line.89"></a>
<span class="sourceLineNo">090</span>import org.xml.sax.Attributes;<a name="line.90"></a>
<span class="sourceLineNo">091</span>import org.xml.sax.SAXException;<a name="line.91"></a>
<span class="sourceLineNo">092</span>import org.xml.sax.helpers.DefaultHandler;<a name="line.92"></a>
<span class="sourceLineNo">093</span><a name="line.93"></a>
<span class="sourceLineNo">094</span>/**<a name="line.94"></a>
<span class="sourceLineNo">095</span> * Format reader for UniProtXML files. This version of UniProtXML format will generate<a name="line.95"></a>
<span class="sourceLineNo">096</span> * and write RichSequence objects. Loosely Based on code from the old, deprecated,<a name="line.96"></a>
<span class="sourceLineNo">097</span> * org.biojava.bio.seq.io.GenbankXmlFormat object.<a name="line.97"></a>
<span class="sourceLineNo">098</span> *<a name="line.98"></a>
<span class="sourceLineNo">099</span> * Understands http://www.ebi.uniprot.org/support/docs/uniprot.xsd<a name="line.99"></a>
<span class="sourceLineNo">100</span> *<a name="line.100"></a>
<span class="sourceLineNo">101</span> * @author Alan Li (code based on his work)<a name="line.101"></a>
<span class="sourceLineNo">102</span> * @author Richard Holland<a name="line.102"></a>
<span class="sourceLineNo">103</span> * @since 1.5<a name="line.103"></a>
<span class="sourceLineNo">104</span> */<a name="line.104"></a>
<span class="sourceLineNo">105</span>public class UniProtXMLFormat extends RichSequenceFormat.BasicFormat {<a name="line.105"></a>
<span class="sourceLineNo">106</span>                        <a name="line.106"></a>
<span class="sourceLineNo">107</span>    // Register this format with the format auto-guesser.<a name="line.107"></a>
<span class="sourceLineNo">108</span>    static {<a name="line.108"></a>
<span class="sourceLineNo">109</span>        RichSequence.IOTools.registerFormat(UniProtXMLFormat.class);<a name="line.109"></a>
<span class="sourceLineNo">110</span>    }<a name="line.110"></a>
<span class="sourceLineNo">111</span>    <a name="line.111"></a>
<span class="sourceLineNo">112</span>    /**<a name="line.112"></a>
<span class="sourceLineNo">113</span>     * The name of this format<a name="line.113"></a>
<span class="sourceLineNo">114</span>     */<a name="line.114"></a>
<span class="sourceLineNo">115</span>    public static final String UNIPROTXML_FORMAT = "UniProtXML";<a name="line.115"></a>
<span class="sourceLineNo">116</span>    <a name="line.116"></a>
<span class="sourceLineNo">117</span>    protected static final String ENTRY_GROUP_TAG = "uniprot";<a name="line.117"></a>
<span class="sourceLineNo">118</span>    protected static final String ENTRY_TAG = "entry";<a name="line.118"></a>
<span class="sourceLineNo">119</span>    protected static final String ENTRY_VERSION_ATTR = "version";<a name="line.119"></a>
<span class="sourceLineNo">120</span>    protected static final String ENTRY_NAMESPACE_ATTR = "dataset";<a name="line.120"></a>
<span class="sourceLineNo">121</span>    protected static final String ENTRY_CREATED_ATTR = "created";<a name="line.121"></a>
<span class="sourceLineNo">122</span>    protected static final String ENTRY_UPDATED_ATTR = "modified";<a name="line.122"></a>
<span class="sourceLineNo">123</span>    protected static final String COPYRIGHT_TAG = "copyright";<a name="line.123"></a>
<span class="sourceLineNo">124</span>    <a name="line.124"></a>
<span class="sourceLineNo">125</span>    protected static final String ACCESSION_TAG = "accession";<a name="line.125"></a>
<span class="sourceLineNo">126</span>    protected static final String NAME_TAG = "name";<a name="line.126"></a>
<span class="sourceLineNo">127</span>    protected static final String TEXT_TAG = "text";<a name="line.127"></a>
<span class="sourceLineNo">128</span>    <a name="line.128"></a>
<span class="sourceLineNo">129</span>    protected static final String REF_ATTR = "ref";<a name="line.129"></a>
<span class="sourceLineNo">130</span>    protected static final String TYPE_ATTR = "type";<a name="line.130"></a>
<span class="sourceLineNo">131</span>    protected static final String KEY_ATTR = "key";<a name="line.131"></a>
<span class="sourceLineNo">132</span>    protected static final String ID_ATTR = "id";<a name="line.132"></a>
<span class="sourceLineNo">133</span>    protected static final String EVIDENCE_ATTR = "evidence";<a name="line.133"></a>
<span class="sourceLineNo">134</span>    protected static final String VALUE_ATTR = "value";<a name="line.134"></a>
<span class="sourceLineNo">135</span>    protected static final String STATUS_ATTR = "value";<a name="line.135"></a>
<span class="sourceLineNo">136</span>    protected static final String NAME_ATTR = "name";<a name="line.136"></a>
<span class="sourceLineNo">137</span>    <a name="line.137"></a>
<span class="sourceLineNo">138</span>    protected static final String PROTEIN_TAG = "protein";<a name="line.138"></a>
<span class="sourceLineNo">139</span>    protected static final String PROTEIN_TYPE_ATTR = "type";<a name="line.139"></a>
<span class="sourceLineNo">140</span>    <a name="line.140"></a>
<span class="sourceLineNo">141</span>    protected static final String DOMAIN_TAG = "domain";<a name="line.141"></a>
<span class="sourceLineNo">142</span>    protected static final String COMPONENT_TAG = "component";<a name="line.142"></a>
<span class="sourceLineNo">143</span>    protected static final String GENE_TAG = "gene";<a name="line.143"></a>
<span class="sourceLineNo">144</span>    protected static final String ORGANISM_TAG = "organism";<a name="line.144"></a>
<span class="sourceLineNo">145</span>    protected static final String DBXREF_TAG = "dbReference";<a name="line.145"></a>
<span class="sourceLineNo">146</span>    protected static final String PROPERTY_TAG = "property";<a name="line.146"></a>
<span class="sourceLineNo">147</span>    protected static final String LINEAGE_TAG = "lineage";<a name="line.147"></a>
<span class="sourceLineNo">148</span>    protected static final String TAXON_TAG = "taxon";<a name="line.148"></a>
<span class="sourceLineNo">149</span>    protected static final String GENELOCATION_TAG = "geneLocation";<a name="line.149"></a>
<span class="sourceLineNo">150</span>    protected static final String GENELOCATION_NAME_TAG = "name";<a name="line.150"></a>
<span class="sourceLineNo">151</span>    <a name="line.151"></a>
<span class="sourceLineNo">152</span>    protected static final String REFERENCE_TAG = "reference";<a name="line.152"></a>
<span class="sourceLineNo">153</span>    protected static final String CITATION_TAG = "citation";<a name="line.153"></a>
<span class="sourceLineNo">154</span>    protected static final String TITLE_TAG = "title";<a name="line.154"></a>
<span class="sourceLineNo">155</span>    protected static final String EDITOR_LIST_TAG = "editorList";<a name="line.155"></a>
<span class="sourceLineNo">156</span>    protected static final String AUTHOR_LIST_TAG = "authorList";<a name="line.156"></a>
<span class="sourceLineNo">157</span>    protected static final String PERSON_TAG = "person";<a name="line.157"></a>
<span class="sourceLineNo">158</span>    protected static final String CONSORTIUM_TAG = "consortium";<a name="line.158"></a>
<span class="sourceLineNo">159</span>    protected static final String LOCATOR_TAG = "locator";<a name="line.159"></a>
<span class="sourceLineNo">160</span>    protected static final String RP_LINE_TAG = "scope";<a name="line.160"></a>
<span class="sourceLineNo">161</span>    protected static final String RC_LINE_TAG = "source";<a name="line.161"></a>
<span class="sourceLineNo">162</span>    protected static final String RC_SPECIES_TAG = "species";<a name="line.162"></a>
<span class="sourceLineNo">163</span>    protected static final String RC_TISSUE_TAG = "tissue";<a name="line.163"></a>
<span class="sourceLineNo">164</span>    protected static final String RC_TRANSP_TAG = "transposon";<a name="line.164"></a>
<span class="sourceLineNo">165</span>    protected static final String RC_STRAIN_TAG = "strain";<a name="line.165"></a>
<span class="sourceLineNo">166</span>    protected static final String RC_PLASMID_TAG = "plasmid";<a name="line.166"></a>
<span class="sourceLineNo">167</span>    <a name="line.167"></a>
<span class="sourceLineNo">168</span>    protected static final String COMMENT_TAG = "comment";<a name="line.168"></a>
<span class="sourceLineNo">169</span>    protected static final String COMMENT_MASS_ATTR = "mass";<a name="line.169"></a>
<span class="sourceLineNo">170</span>    protected static final String COMMENT_ERROR_ATTR = "error";<a name="line.170"></a>
<span class="sourceLineNo">171</span>    protected static final String COMMENT_METHOD_ATTR = "method";<a name="line.171"></a>
<span class="sourceLineNo">172</span>    protected static final String COMMENT_LOCTYPE_ATTR = "locationType";<a name="line.172"></a>
<span class="sourceLineNo">173</span>    <a name="line.173"></a>
<span class="sourceLineNo">174</span>    protected static final String COMMENT_ABSORPTION_TAG = "absorption";<a name="line.174"></a>
<span class="sourceLineNo">175</span>    protected static final String COMMENT_ABS_MAX_TAG = "max";<a name="line.175"></a>
<span class="sourceLineNo">176</span>    protected static final String COMMENT_KINETICS_TAG = "kinetics";<a name="line.176"></a>
<span class="sourceLineNo">177</span>    protected static final String COMMENT_KIN_KM_TAG = "KM";<a name="line.177"></a>
<span class="sourceLineNo">178</span>    protected static final String COMMENT_KIN_VMAX_TAG = "VMax";<a name="line.178"></a>
<span class="sourceLineNo">179</span>    protected static final String COMMENT_PH_TAG = "phDependence";<a name="line.179"></a>
<span class="sourceLineNo">180</span>    protected static final String COMMENT_REDOX_TAG = "redoxPotential";<a name="line.180"></a>
<span class="sourceLineNo">181</span>    protected static final String COMMENT_TEMPERATURE_TAG = "temperatureDependence";<a name="line.181"></a>
<span class="sourceLineNo">182</span>    protected static final String COMMENT_LINK_TAG = "link";<a name="line.182"></a>
<span class="sourceLineNo">183</span>    protected static final String COMMENT_LINK_URI_ATTR = "uri";<a name="line.183"></a>
<span class="sourceLineNo">184</span>    protected static final String COMMENT_EVENT_TAG = "event";<a name="line.184"></a>
<span class="sourceLineNo">185</span>    protected static final String COMMENT_ISOFORM_TAG = "isoform";<a name="line.185"></a>
<span class="sourceLineNo">186</span>    protected static final String COMMENT_INTERACTANT_TAG = "interactant";<a name="line.186"></a>
<span class="sourceLineNo">187</span>    protected static final String COMMENT_INTERACT_INTACT_ATTR = "intactId";<a name="line.187"></a>
<span class="sourceLineNo">188</span>    protected static final String COMMENT_INTERACT_LABEL_TAG = "label";<a name="line.188"></a>
<span class="sourceLineNo">189</span>    protected static final String COMMENT_ORGANISMS_TAG = "organismsDiffer";<a name="line.189"></a>
<span class="sourceLineNo">190</span>    protected static final String COMMENT_EXPERIMENTS_TAG = "experiments";<a name="line.190"></a>
<span class="sourceLineNo">191</span>    <a name="line.191"></a>
<span class="sourceLineNo">192</span>    protected static final String NOTE_TAG = "note";<a name="line.192"></a>
<span class="sourceLineNo">193</span>    protected static final String KEYWORD_TAG = "keyword";<a name="line.193"></a>
<span class="sourceLineNo">194</span>    protected static final String PROTEIN_EXISTS_TAG = "proteinExistence";<a name="line.194"></a>
<span class="sourceLineNo">195</span>    protected static final String ID_TAG = "id";<a name="line.195"></a>
<span class="sourceLineNo">196</span>    <a name="line.196"></a>
<span class="sourceLineNo">197</span>    protected static final String FEATURE_TAG = "feature";<a name="line.197"></a>
<span class="sourceLineNo">198</span>    protected static final String FEATURE_DESC_ATTR = "description";<a name="line.198"></a>
<span class="sourceLineNo">199</span>    protected static final String FEATURE_ORIGINAL_TAG = "original";<a name="line.199"></a>
<span class="sourceLineNo">200</span>    protected static final String FEATURE_VARIATION_TAG = "variation";<a name="line.200"></a>
<span class="sourceLineNo">201</span>    <a name="line.201"></a>
<span class="sourceLineNo">202</span>    protected static final String EVIDENCE_TAG = "evidence";<a name="line.202"></a>
<span class="sourceLineNo">203</span>    protected static final String EVIDENCE_CATEGORY_ATTR = "category";<a name="line.203"></a>
<span class="sourceLineNo">204</span>    protected static final String EVIDENCE_ATTRIBUTE_ATTR = "attribute";<a name="line.204"></a>
<span class="sourceLineNo">205</span>    protected static final String EVIDENCE_DATE_ATTR = "date";<a name="line.205"></a>
<span class="sourceLineNo">206</span>    <a name="line.206"></a>
<span class="sourceLineNo">207</span>    protected static final String LOCATION_TAG = "location";<a name="line.207"></a>
<span class="sourceLineNo">208</span>    protected static final String LOCATION_SEQ_ATTR = "sequence";<a name="line.208"></a>
<span class="sourceLineNo">209</span>    protected static final String LOCATION_BEGIN_TAG = "begin";<a name="line.209"></a>
<span class="sourceLineNo">210</span>    protected static final String LOCATION_END_TAG = "end";<a name="line.210"></a>
<span class="sourceLineNo">211</span>    protected static final String LOCATION_POSITION_ATTR = "position";<a name="line.211"></a>
<span class="sourceLineNo">212</span>    protected static final String LOCATION_POSITION_TAG = "position";<a name="line.212"></a>
<span class="sourceLineNo">213</span>    <a name="line.213"></a>
<span class="sourceLineNo">214</span>    protected static final String SEQUENCE_TAG = "sequence";<a name="line.214"></a>
<span class="sourceLineNo">215</span>    protected static final String SEQUENCE_VERSION_ATTR = "version";<a name="line.215"></a>
<span class="sourceLineNo">216</span>    protected static final String SEQUENCE_LENGTH_ATTR = "length";<a name="line.216"></a>
<span class="sourceLineNo">217</span>    protected static final String SEQUENCE_MASS_ATTR = "mass";<a name="line.217"></a>
<span class="sourceLineNo">218</span>    protected static final String SEQUENCE_CHECKSUM_ATTR = "checksum";<a name="line.218"></a>
<span class="sourceLineNo">219</span>    protected static final String SEQUENCE_MODIFIED_ATTR = "modified";<a name="line.219"></a>
<span class="sourceLineNo">220</span>    <a name="line.220"></a>
<span class="sourceLineNo">221</span>    // RP line parser<a name="line.221"></a>
<span class="sourceLineNo">222</span>    protected static final Pattern rppat = Pattern.compile("SEQUENCE OF (\\d+)-(\\d+)");<a name="line.222"></a>
<span class="sourceLineNo">223</span>    <a name="line.223"></a>
<span class="sourceLineNo">224</span>    protected static final Pattern xmlSchema = Pattern.compile(".*http://www\\.uniprot\\.org/support/docs/uniprot\\.xsd.*");<a name="line.224"></a>
<span class="sourceLineNo">225</span>    <a name="line.225"></a>
<span class="sourceLineNo">226</span>    /**<a name="line.226"></a>
<span class="sourceLineNo">227</span>     * Implements some UniProtXML-specific terms.<a name="line.227"></a>
<span class="sourceLineNo">228</span>     */<a name="line.228"></a>
<span class="sourceLineNo">229</span>    public static class Terms extends RichSequence.Terms {        <a name="line.229"></a>
<span class="sourceLineNo">230</span>        public static final String CONTAINS_PREFIX = "Contains:";<a name="line.230"></a>
<span class="sourceLineNo">231</span>        public static final String INCLUDES_PREFIX = "Includes:";<a name="line.231"></a>
<span class="sourceLineNo">232</span>        <a name="line.232"></a>
<span class="sourceLineNo">233</span>        public static final String GENENAME_KEY = "primary";<a name="line.233"></a>
<span class="sourceLineNo">234</span>        public static final String GENESYNONYM_KEY = "synonym";<a name="line.234"></a>
<span class="sourceLineNo">235</span>        public static final String ORDLOCNAME_KEY = "ordered locus";<a name="line.235"></a>
<span class="sourceLineNo">236</span>        public static final String ORFNAME_KEY = "ORF";<a name="line.236"></a>
<span class="sourceLineNo">237</span>        <a name="line.237"></a>
<span class="sourceLineNo">238</span>        public static final String NCBI_TAXON_KEY = "NCBI Taxonomy";<a name="line.238"></a>
<span class="sourceLineNo">239</span>        public static final String COMMON_NAME_KEY = "common";<a name="line.239"></a>
<span class="sourceLineNo">240</span>        public static final String FULL_NAME_KEY = "full";<a name="line.240"></a>
<span class="sourceLineNo">241</span>        public static final String SCIENTIFIC_NAME_KEY = "scientific";<a name="line.241"></a>
<span class="sourceLineNo">242</span>        public static final String SYNONYM_NAME_KEY = "synonym";<a name="line.242"></a>
<span class="sourceLineNo">243</span>        public static final String ABBREV_NAME_KEY = "abbreviation";<a name="line.243"></a>
<span class="sourceLineNo">244</span>        <a name="line.244"></a>
<span class="sourceLineNo">245</span>        public static final String LOC_FUZZY_START_KEY = "less than";<a name="line.245"></a>
<span class="sourceLineNo">246</span>        public static final String LOC_FUZZY_END_KEY = "greater than";<a name="line.246"></a>
<span class="sourceLineNo">247</span>        <a name="line.247"></a>
<span class="sourceLineNo">248</span>        // Ontology for uniprot keywords (because they have identifiers, aaargh...)<a name="line.248"></a>
<span class="sourceLineNo">249</span>        private static ComparableOntology uniprotKWOnto = null;<a name="line.249"></a>
<span class="sourceLineNo">250</span>        <a name="line.250"></a>
<span class="sourceLineNo">251</span>        /**<a name="line.251"></a>
<span class="sourceLineNo">252</span>         * Getter for the protein exists term<a name="line.252"></a>
<span class="sourceLineNo">253</span>         * @return The protein exists Term<a name="line.253"></a>
<span class="sourceLineNo">254</span>         */<a name="line.254"></a>
<span class="sourceLineNo">255</span>        public static ComparableTerm getProteinExistsTerm() {<a name="line.255"></a>
<span class="sourceLineNo">256</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("UniProt protein exists");<a name="line.256"></a>
<span class="sourceLineNo">257</span>        }<a name="line.257"></a>
<span class="sourceLineNo">258</span>        <a name="line.258"></a>
<span class="sourceLineNo">259</span>        /**<a name="line.259"></a>
<span class="sourceLineNo">260</span>         * Getter for the private uniprot ontology.<a name="line.260"></a>
<span class="sourceLineNo">261</span>         * @return the ontology.<a name="line.261"></a>
<span class="sourceLineNo">262</span>         */<a name="line.262"></a>
<span class="sourceLineNo">263</span>        public static ComparableOntology getUniprotKWOnto() {<a name="line.263"></a>
<span class="sourceLineNo">264</span>            return (ComparableOntology)RichObjectFactory.getObject(SimpleComparableOntology.class, new Object[]{"uniprot_kw"});<a name="line.264"></a>
<span class="sourceLineNo">265</span>        }<a name="line.265"></a>
<span class="sourceLineNo">266</span>        <a name="line.266"></a>
<span class="sourceLineNo">267</span>        /**<a name="line.267"></a>
<span class="sourceLineNo">268</span>         * Getter for the UniProtXML term<a name="line.268"></a>
<span class="sourceLineNo">269</span>         * @return The UniProtXML Term<a name="line.269"></a>
<span class="sourceLineNo">270</span>         */<a name="line.270"></a>
<span class="sourceLineNo">271</span>        public static ComparableTerm getUniProtXMLTerm() {<a name="line.271"></a>
<span class="sourceLineNo">272</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("UniProtXML");<a name="line.272"></a>
<span class="sourceLineNo">273</span>        }<a name="line.273"></a>
<span class="sourceLineNo">274</span>        <a name="line.274"></a>
<span class="sourceLineNo">275</span>        /**<a name="line.275"></a>
<span class="sourceLineNo">276</span>         * Getter for the protein type term<a name="line.276"></a>
<span class="sourceLineNo">277</span>         * @return The protein type Term<a name="line.277"></a>
<span class="sourceLineNo">278</span>         */<a name="line.278"></a>
<span class="sourceLineNo">279</span>        public static ComparableTerm getProteinTypeTerm() {<a name="line.279"></a>
<span class="sourceLineNo">280</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("protein_type");<a name="line.280"></a>
<span class="sourceLineNo">281</span>        }<a name="line.281"></a>
<span class="sourceLineNo">282</span>        <a name="line.282"></a>
<span class="sourceLineNo">283</span>        /**<a name="line.283"></a>
<span class="sourceLineNo">284</span>         * Getter for the evidence category term<a name="line.284"></a>
<span class="sourceLineNo">285</span>         * @return The evidence category Term<a name="line.285"></a>
<span class="sourceLineNo">286</span>         */<a name="line.286"></a>
<span class="sourceLineNo">287</span>        public static ComparableTerm getEvidenceCategoryTerm() {<a name="line.287"></a>
<span class="sourceLineNo">288</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("evidence_category");<a name="line.288"></a>
<span class="sourceLineNo">289</span>        }<a name="line.289"></a>
<span class="sourceLineNo">290</span>        <a name="line.290"></a>
<span class="sourceLineNo">291</span>        /**<a name="line.291"></a>
<span class="sourceLineNo">292</span>         * Getter for the evidence type term<a name="line.292"></a>
<span class="sourceLineNo">293</span>         * @return The evidence type Term<a name="line.293"></a>
<span class="sourceLineNo">294</span>         */<a name="line.294"></a>
<span class="sourceLineNo">295</span>        public static ComparableTerm getEvidenceTypeTerm() {<a name="line.295"></a>
<span class="sourceLineNo">296</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("evidence_type");<a name="line.296"></a>
<span class="sourceLineNo">297</span>        }<a name="line.297"></a>
<span class="sourceLineNo">298</span>        <a name="line.298"></a>
<span class="sourceLineNo">299</span>        /**<a name="line.299"></a>
<span class="sourceLineNo">300</span>         * Getter for the evidence date term<a name="line.300"></a>
<span class="sourceLineNo">301</span>         * @return The evidence date Term<a name="line.301"></a>
<span class="sourceLineNo">302</span>         */<a name="line.302"></a>
<span class="sourceLineNo">303</span>        public static ComparableTerm getEvidenceDateTerm() {<a name="line.303"></a>
<span class="sourceLineNo">304</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("evidence_date");<a name="line.304"></a>
<span class="sourceLineNo">305</span>        }<a name="line.305"></a>
<span class="sourceLineNo">306</span>        <a name="line.306"></a>
<span class="sourceLineNo">307</span>        /**<a name="line.307"></a>
<span class="sourceLineNo">308</span>         * Getter for the evidence attr term<a name="line.308"></a>
<span class="sourceLineNo">309</span>         * @return The evidence attr Term<a name="line.309"></a>
<span class="sourceLineNo">310</span>         */<a name="line.310"></a>
<span class="sourceLineNo">311</span>        public static ComparableTerm getEvidenceAttrTerm() {<a name="line.311"></a>
<span class="sourceLineNo">312</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("evidence_attr");<a name="line.312"></a>
<span class="sourceLineNo">313</span>        }<a name="line.313"></a>
<span class="sourceLineNo">314</span>        <a name="line.314"></a>
<span class="sourceLineNo">315</span>        /**<a name="line.315"></a>
<span class="sourceLineNo">316</span>         * Getter for the feature ref term<a name="line.316"></a>
<span class="sourceLineNo">317</span>         * @return The feature ref Term<a name="line.317"></a>
<span class="sourceLineNo">318</span>         */<a name="line.318"></a>
<span class="sourceLineNo">319</span>        public static ComparableTerm getFeatureRefTerm() {<a name="line.319"></a>
<span class="sourceLineNo">320</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("feature_ref");<a name="line.320"></a>
<span class="sourceLineNo">321</span>        }<a name="line.321"></a>
<span class="sourceLineNo">322</span>        <a name="line.322"></a>
<span class="sourceLineNo">323</span>        /**<a name="line.323"></a>
<span class="sourceLineNo">324</span>         * Getter for the feature status term<a name="line.324"></a>
<span class="sourceLineNo">325</span>         * @return The feature status Term<a name="line.325"></a>
<span class="sourceLineNo">326</span>         */<a name="line.326"></a>
<span class="sourceLineNo">327</span>        public static ComparableTerm getFeatureStatusTerm() {<a name="line.327"></a>
<span class="sourceLineNo">328</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("feature_status");<a name="line.328"></a>
<span class="sourceLineNo">329</span>        }<a name="line.329"></a>
<span class="sourceLineNo">330</span>        <a name="line.330"></a>
<span class="sourceLineNo">331</span>        /**<a name="line.331"></a>
<span class="sourceLineNo">332</span>         * Getter for the feature original term<a name="line.332"></a>
<span class="sourceLineNo">333</span>         * @return The feature original Term<a name="line.333"></a>
<span class="sourceLineNo">334</span>         */<a name="line.334"></a>
<span class="sourceLineNo">335</span>        public static ComparableTerm getFeatureOriginalTerm() {<a name="line.335"></a>
<span class="sourceLineNo">336</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("feature_original");<a name="line.336"></a>
<span class="sourceLineNo">337</span>        }<a name="line.337"></a>
<span class="sourceLineNo">338</span>        <a name="line.338"></a>
<span class="sourceLineNo">339</span>        /**<a name="line.339"></a>
<span class="sourceLineNo">340</span>         * Getter for the feature variation term<a name="line.340"></a>
<span class="sourceLineNo">341</span>         * @return The feature variation Term<a name="line.341"></a>
<span class="sourceLineNo">342</span>         */<a name="line.342"></a>
<span class="sourceLineNo">343</span>        public static ComparableTerm getFeatureVariationTerm() {<a name="line.343"></a>
<span class="sourceLineNo">344</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("feature_variation");<a name="line.344"></a>
<span class="sourceLineNo">345</span>        }<a name="line.345"></a>
<span class="sourceLineNo">346</span>        <a name="line.346"></a>
<span class="sourceLineNo">347</span>        /**<a name="line.347"></a>
<span class="sourceLineNo">348</span>         * Getter for the location seq term<a name="line.348"></a>
<span class="sourceLineNo">349</span>         * @return The location seq Term<a name="line.349"></a>
<span class="sourceLineNo">350</span>         */<a name="line.350"></a>
<span class="sourceLineNo">351</span>        public static ComparableTerm getLocationSequenceTerm() {<a name="line.351"></a>
<span class="sourceLineNo">352</span>            return RichObjectFactory.getDefaultOntology().getOrCreateTerm("locseq");<a name="line.352"></a>
<span class="sourceLineNo">353</span>        }<a name="line.353"></a>
<span class="sourceLineNo">354</span>    }    <a name="line.354"></a>
<span class="sourceLineNo">355</span>    <a name="line.355"></a>
<span class="sourceLineNo">356</span>    /**<a name="line.356"></a>
<span class="sourceLineNo">357</span>     * {@inheritDoc}<a name="line.357"></a>
<span class="sourceLineNo">358</span>     * A file is in UniProtXML format if the second XML line contains the phrase "http://www.uniprot.org/support/docs/uniprot.xsd".<a name="line.358"></a>
<span class="sourceLineNo">359</span>     */<a name="line.359"></a>
<span class="sourceLineNo">360</span>    @Override<a name="line.360"></a>
<span class="sourceLineNo">361</span>    public boolean canRead(File file) throws IOException {<a name="line.361"></a>
<span class="sourceLineNo">362</span>        BufferedReader br = new BufferedReader(new FileReader(file));<a name="line.362"></a>
<span class="sourceLineNo">363</span>        br.readLine(); // skip first line<a name="line.363"></a>
<span class="sourceLineNo">364</span>        String secondLine = br.readLine();<a name="line.364"></a>
<span class="sourceLineNo">365</span>        boolean readable = secondLine!=null &amp;&amp; xmlSchema.matcher(secondLine).matches(); // check on second line<a name="line.365"></a>
<span class="sourceLineNo">366</span>        br.close();<a name="line.366"></a>
<span class="sourceLineNo">367</span>        return readable;<a name="line.367"></a>
<span class="sourceLineNo">368</span>    }<a name="line.368"></a>
<span class="sourceLineNo">369</span>    <a name="line.369"></a>
<span class="sourceLineNo">370</span>    /**<a name="line.370"></a>
<span class="sourceLineNo">371</span>     * {@inheritDoc}<a name="line.371"></a>
<span class="sourceLineNo">372</span>     * Always returns a protein tokenizer.<a name="line.372"></a>
<span class="sourceLineNo">373</span>     */<a name="line.373"></a>
<span class="sourceLineNo">374</span>    @Override<a name="line.374"></a>
<span class="sourceLineNo">375</span>    public SymbolTokenization guessSymbolTokenization(File file) throws IOException {<a name="line.375"></a>
<span class="sourceLineNo">376</span>        return RichSequence.IOTools.getProteinParser();<a name="line.376"></a>
<span class="sourceLineNo">377</span>    }<a name="line.377"></a>
<span class="sourceLineNo">378</span>    <a name="line.378"></a>
<span class="sourceLineNo">379</span>    /**<a name="line.379"></a>
<span class="sourceLineNo">380</span>     * {@inheritDoc}<a name="line.380"></a>
<span class="sourceLineNo">381</span>     * A stream is in UniProtXML format if the second XML line contains the phrase "http://www.uniprot.org/support/docs/uniprot.xsd".<a name="line.381"></a>
<span class="sourceLineNo">382</span>     */<a name="line.382"></a>
<span class="sourceLineNo">383</span>    public boolean canRead(BufferedInputStream stream) throws IOException {<a name="line.383"></a>
<span class="sourceLineNo">384</span>        stream.mark(2000); // some streams may not support this<a name="line.384"></a>
<span class="sourceLineNo">385</span>        BufferedReader br = new BufferedReader(new InputStreamReader(stream));<a name="line.385"></a>
<span class="sourceLineNo">386</span>        br.readLine(); // skip first line<a name="line.386"></a>
<span class="sourceLineNo">387</span>        String secondLine = br.readLine();<a name="line.387"></a>
<span class="sourceLineNo">388</span>        boolean readable = secondLine!=null &amp;&amp; xmlSchema.matcher(secondLine).matches(); // check on second line<a name="line.388"></a>
<span class="sourceLineNo">389</span>        // don't close the reader as it'll close the stream too.<a name="line.389"></a>
<span class="sourceLineNo">390</span>        // br.close();<a name="line.390"></a>
<span class="sourceLineNo">391</span>        stream.reset();<a name="line.391"></a>
<span class="sourceLineNo">392</span>        return readable;<a name="line.392"></a>
<span class="sourceLineNo">393</span>    }<a name="line.393"></a>
<span class="sourceLineNo">394</span>    <a name="line.394"></a>
<span class="sourceLineNo">395</span>    /**<a name="line.395"></a>
<span class="sourceLineNo">396</span>     * {@inheritDoc}<a name="line.396"></a>
<span class="sourceLineNo">397</span>     * Always returns a protein tokenizer.<a name="line.397"></a>
<span class="sourceLineNo">398</span>     */<a name="line.398"></a>
<span class="sourceLineNo">399</span>    public SymbolTokenization guessSymbolTokenization(BufferedInputStream stream) throws IOException {<a name="line.399"></a>
<span class="sourceLineNo">400</span>        return RichSequence.IOTools.getProteinParser();<a name="line.400"></a>
<span class="sourceLineNo">401</span>    }<a name="line.401"></a>
<span class="sourceLineNo">402</span>    <a name="line.402"></a>
<span class="sourceLineNo">403</span>    /**<a name="line.403"></a>
<span class="sourceLineNo">404</span>     * {@inheritDoc}<a name="line.404"></a>
<span class="sourceLineNo">405</span>     */<a name="line.405"></a>
<span class="sourceLineNo">406</span>    public boolean readSequence(BufferedReader reader,<a name="line.406"></a>
<span class="sourceLineNo">407</span>            SymbolTokenization symParser,<a name="line.407"></a>
<span class="sourceLineNo">408</span>            SeqIOListener listener)<a name="line.408"></a>
<span class="sourceLineNo">409</span>            throws IllegalSymbolException, IOException, ParseException {<a name="line.409"></a>
<span class="sourceLineNo">410</span>        if (!(listener instanceof RichSeqIOListener)) throw new IllegalArgumentException("Only accepting RichSeqIOListeners today");<a name="line.410"></a>
<span class="sourceLineNo">411</span>        return this.readRichSequence(reader,symParser,(RichSeqIOListener)listener,null);<a name="line.411"></a>
<span class="sourceLineNo">412</span>    }<a name="line.412"></a>
<span class="sourceLineNo">413</span>    <a name="line.413"></a>
<span class="sourceLineNo">414</span>    /**<a name="line.414"></a>
<span class="sourceLineNo">415</span>     * {@inheritDoc}<a name="line.415"></a>
<span class="sourceLineNo">416</span>     * If namespace is null, then the namespace of the sequence in the fasta is used.<a name="line.416"></a>
<span class="sourceLineNo">417</span>     * If the namespace is null and so is the namespace of the sequence in the fasta,<a name="line.417"></a>
<span class="sourceLineNo">418</span>     * then the default namespace is used.<a name="line.418"></a>
<span class="sourceLineNo">419</span>     */<a name="line.419"></a>
<span class="sourceLineNo">420</span>    public boolean readRichSequence(BufferedReader reader,<a name="line.420"></a>
<span class="sourceLineNo">421</span>            SymbolTokenization symParser,<a name="line.421"></a>
<span class="sourceLineNo">422</span>            RichSeqIOListener rlistener,<a name="line.422"></a>
<span class="sourceLineNo">423</span>            Namespace ns)<a name="line.423"></a>
<span class="sourceLineNo">424</span>            throws IllegalSymbolException, IOException, ParseException {<a name="line.424"></a>
<span class="sourceLineNo">425</span>        <a name="line.425"></a>
<span class="sourceLineNo">426</span>        Pattern copyright = Pattern.compile(".*&lt;"+COPYRIGHT_TAG+".*");<a name="line.426"></a>
<span class="sourceLineNo">427</span>        <a name="line.427"></a>
<span class="sourceLineNo">428</span>        try {<a name="line.428"></a>
<span class="sourceLineNo">429</span>            rlistener.startSequence();                   <a name="line.429"></a>
<span class="sourceLineNo">430</span>            DefaultHandler m_handler = new UniProtXMLHandler(this,symParser,rlistener,ns);<a name="line.430"></a>
<span class="sourceLineNo">431</span>            boolean hasMore=XMLTools.readXMLChunk(reader, m_handler, ENTRY_TAG);<a name="line.431"></a>
<span class="sourceLineNo">432</span>            // deal with copyright chunk<a name="line.432"></a>
<span class="sourceLineNo">433</span>            reader.mark(10000);<a name="line.433"></a>
<span class="sourceLineNo">434</span>            String line = reader.readLine();<a name="line.434"></a>
<span class="sourceLineNo">435</span>            reader.reset();<a name="line.435"></a>
<span class="sourceLineNo">436</span>            if (copyright.matcher(line).matches()) XMLTools.readXMLChunk(reader, m_handler, COPYRIGHT_TAG);<a name="line.436"></a>
<span class="sourceLineNo">437</span>            // all done!<a name="line.437"></a>
<span class="sourceLineNo">438</span>            rlistener.endSequence();<a name="line.438"></a>
<span class="sourceLineNo">439</span>            return hasMore;<a name="line.439"></a>
<span class="sourceLineNo">440</span>        } catch (ParserConfigurationException e) {<a name="line.440"></a>
<span class="sourceLineNo">441</span>            throw new ParseException(e);<a name="line.441"></a>
<span class="sourceLineNo">442</span>        } catch (SAXException e) {<a name="line.442"></a>
<span class="sourceLineNo">443</span>            throw new ParseException(e);<a name="line.443"></a>
<span class="sourceLineNo">444</span>        }<a name="line.444"></a>
<span class="sourceLineNo">445</span>    }<a name="line.445"></a>
<span class="sourceLineNo">446</span>    <a name="line.446"></a>
<span class="sourceLineNo">447</span>    private PrintWriter pw;<a name="line.447"></a>
<span class="sourceLineNo">448</span>    private XMLWriter xml;<a name="line.448"></a>
<span class="sourceLineNo">449</span>    <a name="line.449"></a>
<span class="sourceLineNo">450</span>    /**<a name="line.450"></a>
<span class="sourceLineNo">451</span>     * {@inheritDoc}<a name="line.451"></a>
<span class="sourceLineNo">452</span>     */<a name="line.452"></a>
<span class="sourceLineNo">453</span>    public void beginWriting() throws IOException {<a name="line.453"></a>
<span class="sourceLineNo">454</span>        // make an XML writer<a name="line.454"></a>
<span class="sourceLineNo">455</span>        pw = new PrintWriter(this.getPrintStream());<a name="line.455"></a>
<span class="sourceLineNo">456</span>        xml = new PrettyXMLWriter(pw);<a name="line.456"></a>
<span class="sourceLineNo">457</span>        xml.printRaw("&lt;?xml version=\"1.0\" encoding=\"UTF-8\" ?&gt;");<a name="line.457"></a>
<span class="sourceLineNo">458</span>        xml.openTag(ENTRY_GROUP_TAG);<a name="line.458"></a>
<span class="sourceLineNo">459</span>        xml.attribute("xmlns","http://uniprot.org/uniprot");<a name="line.459"></a>
<span class="sourceLineNo">460</span>        xml.attribute("xmlns:xsi","http://www.w3.org/2001/XMLSchema-instance");<a name="line.460"></a>
<span class="sourceLineNo">461</span>        xml.attribute("xsi:schemaLocation","http://uniprot.org/uniprot http://www.uniprot.org/support/docs/uniprot.xsd");<a name="line.461"></a>
<span class="sourceLineNo">462</span>    }<a name="line.462"></a>
<span class="sourceLineNo">463</span>    <a name="line.463"></a>
<span class="sourceLineNo">464</span>    /**<a name="line.464"></a>
<span class="sourceLineNo">465</span>     * {@inheritDoc}<a name="line.465"></a>
<span class="sourceLineNo">466</span>     */<a name="line.466"></a>
<span class="sourceLineNo">467</span>    public void finishWriting() throws IOException {<a name="line.467"></a>
<span class="sourceLineNo">468</span>        xml.closeTag(ENTRY_GROUP_TAG);<a name="line.468"></a>
<span class="sourceLineNo">469</span>        pw.flush();<a name="line.469"></a>
<span class="sourceLineNo">470</span>    }<a name="line.470"></a>
<span class="sourceLineNo">471</span>    <a name="line.471"></a>
<span class="sourceLineNo">472</span>    /**<a name="line.472"></a>
<span class="sourceLineNo">473</span>     * {@inheritDoc}<a name="line.473"></a>
<span class="sourceLineNo">474</span>     */<a name="line.474"></a>
<span class="sourceLineNo">475</span>    public void writeSequence(Sequence seq, PrintStream os) throws IOException {<a name="line.475"></a>
<span class="sourceLineNo">476</span>        if (this.getPrintStream()==null) this.setPrintStream(this.getPrintStream());<a name="line.476"></a>
<span class="sourceLineNo">477</span>        this.writeSequence(seq, RichObjectFactory.getDefaultNamespace());<a name="line.477"></a>
<span class="sourceLineNo">478</span>    }<a name="line.478"></a>
<span class="sourceLineNo">479</span>    <a name="line.479"></a>
<span class="sourceLineNo">480</span>    /**<a name="line.480"></a>
<span class="sourceLineNo">481</span>     * {@inheritDoc}<a name="line.481"></a>
<span class="sourceLineNo">482</span>     */<a name="line.482"></a>
<span class="sourceLineNo">483</span>    public void writeSequence(Sequence seq, String format, PrintStream os) throws IOException {<a name="line.483"></a>
<span class="sourceLineNo">484</span>        if (this.getPrintStream()==null) this.setPrintStream(this.getPrintStream());<a name="line.484"></a>
<span class="sourceLineNo">485</span>        if (!format.equals(this.getDefaultFormat())) throw new IllegalArgumentException("Unknown format: "+format);<a name="line.485"></a>
<span class="sourceLineNo">486</span>        this.writeSequence(seq, RichObjectFactory.getDefaultNamespace());<a name="line.486"></a>
<span class="sourceLineNo">487</span>    }<a name="line.487"></a>
<span class="sourceLineNo">488</span>    <a name="line.488"></a>
<span class="sourceLineNo">489</span>    /**<a name="line.489"></a>
<span class="sourceLineNo">490</span>     * {@inheritDoc}<a name="line.490"></a>
<span class="sourceLineNo">491</span>     * If namespace is null, then the sequence's own namespace is used.<a name="line.491"></a>
<span class="sourceLineNo">492</span>     */<a name="line.492"></a>
<span class="sourceLineNo">493</span>    public void writeSequence(Sequence seq, Namespace ns) throws IOException {<a name="line.493"></a>
<span class="sourceLineNo">494</span>        RichSequence rs;<a name="line.494"></a>
<span class="sourceLineNo">495</span>        try {<a name="line.495"></a>
<span class="sourceLineNo">496</span>            if (seq instanceof RichSequence) rs = (RichSequence)seq;<a name="line.496"></a>
<span class="sourceLineNo">497</span>            else rs = RichSequence.Tools.enrich(seq);<a name="line.497"></a>
<span class="sourceLineNo">498</span>        } catch (ChangeVetoException e) {<a name="line.498"></a>
<span class="sourceLineNo">499</span>            IOException e2 = new IOException("Unable to enrich sequence");<a name="line.499"></a>
<span class="sourceLineNo">500</span>            e2.initCause(e);<a name="line.500"></a>
<span class="sourceLineNo">501</span>            throw e2;<a name="line.501"></a>
<span class="sourceLineNo">502</span>        }<a name="line.502"></a>
<span class="sourceLineNo">503</span>        <a name="line.503"></a>
<span class="sourceLineNo">504</span>        int key = 1;<a name="line.504"></a>
<span class="sourceLineNo">505</span>        <a name="line.505"></a>
<span class="sourceLineNo">506</span>        Set&lt;Note&gt; notes = rs.getNoteSet();<a name="line.506"></a>
<span class="sourceLineNo">507</span>        List accessions = new ArrayList();<a name="line.507"></a>
<span class="sourceLineNo">508</span>        List kws = new ArrayList();<a name="line.508"></a>
<span class="sourceLineNo">509</span>        String cdat = null;<a name="line.509"></a>
<span class="sourceLineNo">510</span>        String udat = null;<a name="line.510"></a>
<span class="sourceLineNo">511</span>        String arel = null;<a name="line.511"></a>
<span class="sourceLineNo">512</span>        String adat = null;<a name="line.512"></a>
<span class="sourceLineNo">513</span>        String copyright = null;<a name="line.513"></a>
<span class="sourceLineNo">514</span>        String proteinType = null;<a name="line.514"></a>
<span class="sourceLineNo">515</span>        String proteinExists = null;<a name="line.515"></a>
<span class="sourceLineNo">516</span>        Map genenames = new TreeMap();<a name="line.516"></a>
<span class="sourceLineNo">517</span>        Map genesynonyms = new TreeMap();<a name="line.517"></a>
<span class="sourceLineNo">518</span>        Map orfnames = new TreeMap();<a name="line.518"></a>
<span class="sourceLineNo">519</span>        Map ordlocnames = new TreeMap();<a name="line.519"></a>
<span class="sourceLineNo">520</span>        Set evidenceIDs = new TreeSet();<a name="line.520"></a>
<span class="sourceLineNo">521</span>        Set organelles = new TreeSet();<a name="line.521"></a>
<span class="sourceLineNo">522</span>        Map evcats = new TreeMap();<a name="line.522"></a>
<span class="sourceLineNo">523</span>        Map evtypes = new TreeMap();<a name="line.523"></a>
<span class="sourceLineNo">524</span>        Map evdates = new TreeMap();<a name="line.524"></a>
<span class="sourceLineNo">525</span>        Map evattrs = new TreeMap();<a name="line.525"></a>
<span class="sourceLineNo">526</span>        Map speciesRecs = new TreeMap();<a name="line.526"></a>
<span class="sourceLineNo">527</span>        Map strainRecs = new TreeMap();<a name="line.527"></a>
<span class="sourceLineNo">528</span>        Map tissueRecs = new TreeMap();<a name="line.528"></a>
<span class="sourceLineNo">529</span>        Map transpRecs = new TreeMap();<a name="line.529"></a>
<span class="sourceLineNo">530</span>        Map plasmidRecs = new TreeMap();<a name="line.530"></a>
<span class="sourceLineNo">531</span>        for (Iterator&lt;Note&gt; i = notes.iterator(); i.hasNext();) {<a name="line.531"></a>
<span class="sourceLineNo">532</span>            Note n = i.next();<a name="line.532"></a>
<span class="sourceLineNo">533</span>            if (n.getTerm().equals(Terms.getDateCreatedTerm())) cdat=n.getValue();<a name="line.533"></a>
<span class="sourceLineNo">534</span>            else if (n.getTerm().equals(Terms.getDateUpdatedTerm())) udat=n.getValue();<a name="line.534"></a>
<span class="sourceLineNo">535</span>            else if (n.getTerm().equals(Terms.getRelAnnotatedTerm())) arel=n.getValue();<a name="line.535"></a>
<span class="sourceLineNo">536</span>            else if (n.getTerm().equals(Terms.getDateAnnotatedTerm())) adat=n.getValue();<a name="line.536"></a>
<span class="sourceLineNo">537</span>            else if (n.getTerm().equals(Terms.getAdditionalAccessionTerm())) accessions.add(n.getValue());<a name="line.537"></a>
<span class="sourceLineNo">538</span>            else if (n.getTerm().equals(Terms.getOrganelleTerm())) organelles.add(n.getValue());<a name="line.538"></a>
<span class="sourceLineNo">539</span>            else if (n.getTerm().equals(Terms.getKeywordTerm())) {<a name="line.539"></a>
<span class="sourceLineNo">540</span>                ComparableTerm t = Terms.getUniprotKWOnto().getOrCreateTerm(n.getValue());<a name="line.540"></a>
<span class="sourceLineNo">541</span>                try {<a name="line.541"></a>
<span class="sourceLineNo">542</span>                    if (t.getIdentifier()==null || t.getIdentifier().length()==0) t.setIdentifier("UNKNOWN");<a name="line.542"></a>
<span class="sourceLineNo">543</span>                } catch (ChangeVetoException ce) {<a name="line.543"></a>
<span class="sourceLineNo">544</span>                    IOException e = new IOException("Failed to assign keyword identifier");<a name="line.544"></a>
<span class="sourceLineNo">545</span>                    e.initCause(ce);<a name="line.545"></a>
<span class="sourceLineNo">546</span>                    throw e;<a name="line.546"></a>
<span class="sourceLineNo">547</span>                }<a name="line.547"></a>
<span class="sourceLineNo">548</span>                kws.add(t);<a name="line.548"></a>
<span class="sourceLineNo">549</span>            } else if (n.getTerm().equals(Terms.getCopyrightTerm())) copyright=n.getValue();<a name="line.549"></a>
<span class="sourceLineNo">550</span>            else if (n.getTerm().equals(Terms.getProteinTypeTerm())) proteinType=n.getValue();<a name="line.550"></a>
<span class="sourceLineNo">551</span>            else if (n.getTerm().equals(Terms.getProteinExistsTerm())) proteinExists=n.getValue();<a name="line.551"></a>
<span class="sourceLineNo">552</span>            // use the nasty hack to split the reference rank away from the actual value in this field<a name="line.552"></a>
<span class="sourceLineNo">553</span>            else if (n.getTerm().equals(Terms.getGeneNameTerm()))  {<a name="line.553"></a>
<span class="sourceLineNo">554</span>                String ref = n.getValue();<a name="line.554"></a>
<span class="sourceLineNo">555</span>                int colon = ref.indexOf(':');<a name="line.555"></a>
<span class="sourceLineNo">556</span>                Integer refID = new Integer(0);<a name="line.556"></a>
<span class="sourceLineNo">557</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.557"></a>
<span class="sourceLineNo">558</span>                genenames.put(refID, ref.substring(colon+1)); // map of id -&gt; string as only one name per gene<a name="line.558"></a>
<span class="sourceLineNo">559</span>            } else if (n.getTerm().equals(Terms.getGeneSynonymTerm())) {<a name="line.559"></a>
<span class="sourceLineNo">560</span>                String ref = n.getValue();<a name="line.560"></a>
<span class="sourceLineNo">561</span>                int colon = ref.indexOf(':');<a name="line.561"></a>
<span class="sourceLineNo">562</span>                Integer refID = new Integer(0);<a name="line.562"></a>
<span class="sourceLineNo">563</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.563"></a>
<span class="sourceLineNo">564</span>                if (genesynonyms.get(refID)==null) genesynonyms.put(refID, new ArrayList());<a name="line.564"></a>
<span class="sourceLineNo">565</span>                ((List)genesynonyms.get(refID)).add(ref.substring(colon+1));<a name="line.565"></a>
<span class="sourceLineNo">566</span>            } else if (n.getTerm().equals(Terms.getOrderedLocusNameTerm())) {<a name="line.566"></a>
<span class="sourceLineNo">567</span>                String ref = n.getValue();<a name="line.567"></a>
<span class="sourceLineNo">568</span>                int colon = ref.indexOf(':');<a name="line.568"></a>
<span class="sourceLineNo">569</span>                Integer refID = new Integer(0);<a name="line.569"></a>
<span class="sourceLineNo">570</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.570"></a>
<span class="sourceLineNo">571</span>                if (ordlocnames.get(refID)==null) ordlocnames.put(refID, new ArrayList());<a name="line.571"></a>
<span class="sourceLineNo">572</span>                ((List)ordlocnames.get(refID)).add(ref.substring(colon+1));<a name="line.572"></a>
<span class="sourceLineNo">573</span>            } else if (n.getTerm().equals(Terms.getORFNameTerm())) {<a name="line.573"></a>
<span class="sourceLineNo">574</span>                String ref = n.getValue();<a name="line.574"></a>
<span class="sourceLineNo">575</span>                int colon = ref.indexOf(':');<a name="line.575"></a>
<span class="sourceLineNo">576</span>                Integer refID = new Integer(0);<a name="line.576"></a>
<span class="sourceLineNo">577</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.577"></a>
<span class="sourceLineNo">578</span>                if (orfnames.get(refID)==null) orfnames.put(refID, new ArrayList());<a name="line.578"></a>
<span class="sourceLineNo">579</span>                ((List)orfnames.get(refID)).add(ref.substring(colon+1));<a name="line.579"></a>
<span class="sourceLineNo">580</span>            }<a name="line.580"></a>
<span class="sourceLineNo">581</span>            // use the nasty hack to split the reference rank away from the actual value in this field<a name="line.581"></a>
<span class="sourceLineNo">582</span>            else if (n.getTerm().equals(Terms.getEvidenceCategoryTerm()))  {<a name="line.582"></a>
<span class="sourceLineNo">583</span>                String ref = n.getValue();<a name="line.583"></a>
<span class="sourceLineNo">584</span>                int colon = ref.indexOf(':');<a name="line.584"></a>
<span class="sourceLineNo">585</span>                Integer refID = new Integer(0);<a name="line.585"></a>
<span class="sourceLineNo">586</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.586"></a>
<span class="sourceLineNo">587</span>                evcats.put(refID, ref.substring(colon+1)); // map of id -&gt; string as only one name per gene<a name="line.587"></a>
<span class="sourceLineNo">588</span>                evidenceIDs.add(refID);<a name="line.588"></a>
<span class="sourceLineNo">589</span>            } else if (n.getTerm().equals(Terms.getEvidenceTypeTerm()))  {<a name="line.589"></a>
<span class="sourceLineNo">590</span>                String ref = n.getValue();<a name="line.590"></a>
<span class="sourceLineNo">591</span>                int colon = ref.indexOf(':');<a name="line.591"></a>
<span class="sourceLineNo">592</span>                Integer refID = new Integer(0);<a name="line.592"></a>
<span class="sourceLineNo">593</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.593"></a>
<span class="sourceLineNo">594</span>                evtypes.put(refID, ref.substring(colon+1)); // map of id -&gt; string as only one name per gene<a name="line.594"></a>
<span class="sourceLineNo">595</span>                evidenceIDs.add(refID);<a name="line.595"></a>
<span class="sourceLineNo">596</span>            } else if (n.getTerm().equals(Terms.getEvidenceDateTerm()))  {<a name="line.596"></a>
<span class="sourceLineNo">597</span>                String ref = n.getValue();<a name="line.597"></a>
<span class="sourceLineNo">598</span>                int colon = ref.indexOf(':');<a name="line.598"></a>
<span class="sourceLineNo">599</span>                Integer refID = new Integer(0);<a name="line.599"></a>
<span class="sourceLineNo">600</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.600"></a>
<span class="sourceLineNo">601</span>                evdates.put(refID, ref.substring(colon+1)); // map of id -&gt; string as only one name per gene<a name="line.601"></a>
<span class="sourceLineNo">602</span>                evidenceIDs.add(refID);<a name="line.602"></a>
<span class="sourceLineNo">603</span>            } else if (n.getTerm().equals(Terms.getEvidenceAttrTerm()))  {<a name="line.603"></a>
<span class="sourceLineNo">604</span>                String ref = n.getValue();<a name="line.604"></a>
<span class="sourceLineNo">605</span>                int colon = ref.indexOf(':');<a name="line.605"></a>
<span class="sourceLineNo">606</span>                Integer refID = new Integer(0);<a name="line.606"></a>
<span class="sourceLineNo">607</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.607"></a>
<span class="sourceLineNo">608</span>                evattrs.put(refID, ref.substring(colon+1)); // map of id -&gt; string as only one name per gene<a name="line.608"></a>
<span class="sourceLineNo">609</span>                evidenceIDs.add(refID);<a name="line.609"></a>
<span class="sourceLineNo">610</span>            }<a name="line.610"></a>
<span class="sourceLineNo">611</span>            // use the nasty hack to split the reference rank away from the actual value in this field<a name="line.611"></a>
<span class="sourceLineNo">612</span>            // we'll end up with a bunch in key 0 for those which did not come from us. We ignore these for now.<a name="line.612"></a>
<span class="sourceLineNo">613</span>            else if (n.getTerm().equals(Terms.getSpeciesTerm())) {<a name="line.613"></a>
<span class="sourceLineNo">614</span>                String ref = n.getValue();<a name="line.614"></a>
<span class="sourceLineNo">615</span>                int colon = ref.indexOf(':');<a name="line.615"></a>
<span class="sourceLineNo">616</span>                Integer refID = new Integer(0);<a name="line.616"></a>
<span class="sourceLineNo">617</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.617"></a>
<span class="sourceLineNo">618</span>                if (speciesRecs.get(refID)==null) speciesRecs.put(refID, new ArrayList());<a name="line.618"></a>
<span class="sourceLineNo">619</span>                ((List)speciesRecs.get(refID)).add(ref.substring(colon+1));<a name="line.619"></a>
<span class="sourceLineNo">620</span>            } else if (n.getTerm().equals(Terms.getStrainTerm()))  {<a name="line.620"></a>
<span class="sourceLineNo">621</span>                String ref = n.getValue();<a name="line.621"></a>
<span class="sourceLineNo">622</span>                int colon = ref.indexOf(':');<a name="line.622"></a>
<span class="sourceLineNo">623</span>                Integer refID = new Integer(0);<a name="line.623"></a>
<span class="sourceLineNo">624</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.624"></a>
<span class="sourceLineNo">625</span>                if (strainRecs.get(refID)==null) strainRecs.put(refID, new ArrayList());<a name="line.625"></a>
<span class="sourceLineNo">626</span>                ((List)strainRecs.get(refID)).add(ref.substring(colon+1));<a name="line.626"></a>
<span class="sourceLineNo">627</span>            } else if (n.getTerm().equals(Terms.getTissueTerm()))  {<a name="line.627"></a>
<span class="sourceLineNo">628</span>                String ref = n.getValue();<a name="line.628"></a>
<span class="sourceLineNo">629</span>                int colon = ref.indexOf(':');<a name="line.629"></a>
<span class="sourceLineNo">630</span>                Integer refID = new Integer(0);<a name="line.630"></a>
<span class="sourceLineNo">631</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.631"></a>
<span class="sourceLineNo">632</span>                if (tissueRecs.get(refID)==null) tissueRecs.put(refID, new ArrayList());<a name="line.632"></a>
<span class="sourceLineNo">633</span>                ((List)tissueRecs.get(refID)).add(ref.substring(colon+1));<a name="line.633"></a>
<span class="sourceLineNo">634</span>            } else if (n.getTerm().equals(Terms.getTransposonTerm()))  {<a name="line.634"></a>
<span class="sourceLineNo">635</span>                String ref = n.getValue();<a name="line.635"></a>
<span class="sourceLineNo">636</span>                int colon = ref.indexOf(':');<a name="line.636"></a>
<span class="sourceLineNo">637</span>                Integer refID = new Integer(0);<a name="line.637"></a>
<span class="sourceLineNo">638</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.638"></a>
<span class="sourceLineNo">639</span>                if (transpRecs.get(refID)==null) transpRecs.put(refID, new ArrayList());<a name="line.639"></a>
<span class="sourceLineNo">640</span>                ((List)transpRecs.get(refID)).add(ref.substring(colon+1));<a name="line.640"></a>
<span class="sourceLineNo">641</span>            } else if (n.getTerm().equals(Terms.getPlasmidTerm()))  {<a name="line.641"></a>
<span class="sourceLineNo">642</span>                String ref = n.getValue();<a name="line.642"></a>
<span class="sourceLineNo">643</span>                int colon = ref.indexOf(':');<a name="line.643"></a>
<span class="sourceLineNo">644</span>                Integer refID = new Integer(0);<a name="line.644"></a>
<span class="sourceLineNo">645</span>                if (colon&gt;=1) refID = new Integer(ref.substring(0,colon));<a name="line.645"></a>
<span class="sourceLineNo">646</span>                if (plasmidRecs.get(refID)==null) plasmidRecs.put(refID, new ArrayList());<a name="line.646"></a>
<span class="sourceLineNo">647</span>                ((List)plasmidRecs.get(refID)).add(ref.substring(colon+1));<a name="line.647"></a>
<span class="sourceLineNo">648</span>            }<a name="line.648"></a>
<span class="sourceLineNo">649</span>        }<a name="line.649"></a>
<span class="sourceLineNo">650</span>        <a name="line.650"></a>
<span class="sourceLineNo">651</span>        xml.openTag(ENTRY_TAG);<a name="line.651"></a>
<span class="sourceLineNo">652</span>        xml.attribute(ENTRY_VERSION_ATTR,""+(arel==null?""+rs.getVersion():arel));<a name="line.652"></a>
<span class="sourceLineNo">653</span>        xml.attribute(ENTRY_NAMESPACE_ATTR,(ns==null?rs.getNamespace().getName():ns.getName()));<a name="line.653"></a>
<span class="sourceLineNo">654</span>        xml.attribute(ENTRY_CREATED_ATTR,cdat);<a name="line.654"></a>
<span class="sourceLineNo">655</span>        xml.attribute(ENTRY_UPDATED_ATTR,(adat==null?cdat:adat)); // annotation update<a name="line.655"></a>
<span class="sourceLineNo">656</span>        <a name="line.656"></a>
<span class="sourceLineNo">657</span>        xml.openTag(ACCESSION_TAG);<a name="line.657"></a>
<span class="sourceLineNo">658</span>        xml.print(rs.getAccession());<a name="line.658"></a>
<span class="sourceLineNo">659</span>        xml.closeTag(ACCESSION_TAG);<a name="line.659"></a>
<span class="sourceLineNo">660</span>        <a name="line.660"></a>
<span class="sourceLineNo">661</span>        xml.openTag(NAME_TAG);<a name="line.661"></a>
<span class="sourceLineNo">662</span>        xml.print(rs.getName());<a name="line.662"></a>
<span class="sourceLineNo">663</span>        xml.closeTag(NAME_TAG);<a name="line.663"></a>
<span class="sourceLineNo">664</span>        <a name="line.664"></a>
<span class="sourceLineNo">665</span>        xml.openTag(PROTEIN_TAG);<a name="line.665"></a>
<span class="sourceLineNo">666</span>        if (proteinType!=null) xml.attribute(TYPE_ATTR,proteinType);<a name="line.666"></a>
<span class="sourceLineNo">667</span>        String desc = rs.getDescription().trim(); // this is only going to make sense if it was a UniProt seq to start with<a name="line.667"></a>
<span class="sourceLineNo">668</span>        if (desc.endsWith(".")) desc = desc.substring(0, desc.length()-1); // chomp trailing dot<a name="line.668"></a>
<span class="sourceLineNo">669</span>        String[] parts = desc.split("\\[");<a name="line.669"></a>
<span class="sourceLineNo">670</span>        for (int j = 0 ; j &lt; parts.length; j++) {<a name="line.670"></a>
<span class="sourceLineNo">671</span>            if (parts[j].startsWith(Terms.CONTAINS_PREFIX)) {<a name="line.671"></a>
<span class="sourceLineNo">672</span>                // contains section<a name="line.672"></a>
<span class="sourceLineNo">673</span>                String chunk = parts[j].substring(Terms.CONTAINS_PREFIX.length()+1).trim();<a name="line.673"></a>
<span class="sourceLineNo">674</span>                if (chunk.endsWith("]")) chunk = chunk.substring(0, chunk.length()-1); // chomp trailing ]<a name="line.674"></a>
<span class="sourceLineNo">675</span>                String[] moreparts = chunk.split(";");<a name="line.675"></a>
<span class="sourceLineNo">676</span>                for (int k = 0; k &lt; moreparts.length; k++) {<a name="line.676"></a>
<span class="sourceLineNo">677</span>                    xml.openTag(DOMAIN_TAG);<a name="line.677"></a>
<span class="sourceLineNo">678</span>                    String[] names = moreparts[k].split("\\(");<a name="line.678"></a>
<span class="sourceLineNo">679</span>                    for (int l = 0; l &lt; names.length; l++) {<a name="line.679"></a>
<span class="sourceLineNo">680</span>                        String name = names[l].trim();<a name="line.680"></a>
<span class="sourceLineNo">681</span>                        if (name.endsWith(")")) name = name.substring(0,name.length()-1); // chomp trailing )<a name="line.681"></a>
<span class="sourceLineNo">682</span>                        xml.openTag(NAME_TAG);<a name="line.682"></a>
<span class="sourceLineNo">683</span>                        xml.print(name);<a name="line.683"></a>
<span class="sourceLineNo">684</span>                        xml.closeTag(NAME_TAG);<a name="line.684"></a>
<span class="sourceLineNo">685</span>                    }<a name="line.685"></a>
<span class="sourceLineNo">686</span>                    xml.closeTag(DOMAIN_TAG);<a name="line.686"></a>
<span class="sourceLineNo">687</span>                }<a name="line.687"></a>
<span class="sourceLineNo">688</span>            } else if (parts[j].startsWith(Terms.INCLUDES_PREFIX)) {<a name="line.688"></a>
<span class="sourceLineNo">689</span>                // includes section<a name="line.689"></a>
<span class="sourceLineNo">690</span>                String chunk = parts[j].substring(Terms.INCLUDES_PREFIX.length()+1).trim();<a name="line.690"></a>
<span class="sourceLineNo">691</span>                if (chunk.endsWith("]")) chunk = chunk.substring(0, chunk.length()-1); // chomp trailing ]<a name="line.691"></a>
<span class="sourceLineNo">692</span>                String[] moreparts = chunk.split(";");<a name="line.692"></a>
<span class="sourceLineNo">693</span>                for (int k = 0; k &lt; moreparts.length; k++) {<a name="line.693"></a>
<span class="sourceLineNo">694</span>                    xml.openTag(COMPONENT_TAG);<a name="line.694"></a>
<span class="sourceLineNo">695</span>                    String[] names = moreparts[k].split("\\(");<a name="line.695"></a>
<span class="sourceLineNo">696</span>                    for (int l = 0; l &lt; names.length; l++) {<a name="line.696"></a>
<span class="sourceLineNo">697</span>                        String name = names[l].trim();<a name="line.697"></a>
<span class="sourceLineNo">698</span>                        if (name.endsWith(")")) name = name.substring(0,name.length()-1); // chomp trailing )<a name="line.698"></a>
<span class="sourceLineNo">699</span>                        xml.openTag(NAME_TAG);<a name="line.699"></a>
<span class="sourceLineNo">700</span>                        xml.print(name);<a name="line.700"></a>
<span class="sourceLineNo">701</span>                        xml.closeTag(NAME_TAG);<a name="line.701"></a>
<span class="sourceLineNo">702</span>                    }<a name="line.702"></a>
<span class="sourceLineNo">703</span>                    xml.closeTag(COMPONENT_TAG);<a name="line.703"></a>
<span class="sourceLineNo">704</span>                }<a name="line.704"></a>
<span class="sourceLineNo">705</span>            } else {<a name="line.705"></a>
<span class="sourceLineNo">706</span>                // plain names<a name="line.706"></a>
<span class="sourceLineNo">707</span>                String[] names = parts[j].split("\\(");<a name="line.707"></a>
<span class="sourceLineNo">708</span>                for (int l = 0; l &lt; names.length; l++) {<a name="line.708"></a>
<span class="sourceLineNo">709</span>                    String name = names[l].trim();<a name="line.709"></a>
<span class="sourceLineNo">710</span>                    if (name.endsWith(")")) name = name.substring(0,name.length()-1); // chomp trailing )<a name="line.710"></a>
<span class="sourceLineNo">711</span>                    xml.openTag(NAME_TAG);<a name="line.711"></a>
<span class="sourceLineNo">712</span>                    xml.print(name);<a name="line.712"></a>
<span class="sourceLineNo">713</span>                    xml.closeTag(NAME_TAG);<a name="line.713"></a>
<span class="sourceLineNo">714</span>                }<a name="line.714"></a>
<span class="sourceLineNo">715</span>            }<a name="line.715"></a>
<span class="sourceLineNo">716</span>        }<a name="line.716"></a>
<span class="sourceLineNo">717</span>        xml.closeTag(PROTEIN_TAG);<a name="line.717"></a>
<span class="sourceLineNo">718</span>        <a name="line.718"></a>
<span class="sourceLineNo">719</span>        // gene line<a name="line.719"></a>
<span class="sourceLineNo">720</span>        for (Iterator i = genenames.keySet().iterator(); i.hasNext(); ) {<a name="line.720"></a>
<span class="sourceLineNo">721</span>            Integer geneid = (Integer)i.next();<a name="line.721"></a>
<span class="sourceLineNo">722</span>            String genename = (String)genenames.get(geneid);<a name="line.722"></a>
<span class="sourceLineNo">723</span>            List synonyms = (List)genesynonyms.get(geneid);<a name="line.723"></a>
<span class="sourceLineNo">724</span>            List orfs = (List)orfnames.get(geneid);<a name="line.724"></a>
<span class="sourceLineNo">725</span>            List ordlocs = (List)ordlocnames.get(geneid);<a name="line.725"></a>
<span class="sourceLineNo">726</span>            <a name="line.726"></a>
<span class="sourceLineNo">727</span>            xml.openTag(GENE_TAG);<a name="line.727"></a>
<span class="sourceLineNo">728</span>            <a name="line.728"></a>
<span class="sourceLineNo">729</span>            xml.openTag(NAME_TAG);<a name="line.729"></a>
<span class="sourceLineNo">730</span>            xml.attribute(TYPE_ATTR,Terms.GENENAME_KEY);<a name="line.730"></a>
<span class="sourceLineNo">731</span>            xml.print(genename);<a name="line.731"></a>
<span class="sourceLineNo">732</span>            xml.closeTag(NAME_TAG);<a name="line.732"></a>
<span class="sourceLineNo">733</span>            <a name="line.733"></a>
<span class="sourceLineNo">734</span>            if (synonyms!=null) {<a name="line.734"></a>
<span class="sourceLineNo">735</span>                for (Iterator j = synonyms.iterator(); j.hasNext(); ) {<a name="line.735"></a>
<span class="sourceLineNo">736</span>                    xml.openTag(NAME_TAG);<a name="line.736"></a>
<span class="sourceLineNo">737</span>                    xml.attribute(TYPE_ATTR,Terms.GENESYNONYM_KEY);<a name="line.737"></a>
<span class="sourceLineNo">738</span>                    xml.print((String)j.next());<a name="line.738"></a>
<span class="sourceLineNo">739</span>                    xml.closeTag(NAME_TAG);<a name="line.739"></a>
<span class="sourceLineNo">740</span>                }<a name="line.740"></a>
<span class="sourceLineNo">741</span>            }<a name="line.741"></a>
<span class="sourceLineNo">742</span>            if (ordlocs!=null) {<a name="line.742"></a>
<span class="sourceLineNo">743</span>                for (Iterator j = synonyms.iterator(); j.hasNext(); ) {<a name="line.743"></a>
<span class="sourceLineNo">744</span>                    xml.openTag(NAME_TAG);<a name="line.744"></a>
<span class="sourceLineNo">745</span>                    xml.attribute(TYPE_ATTR,Terms.ORDLOCNAME_KEY);<a name="line.745"></a>
<span class="sourceLineNo">746</span>                    xml.print((String)j.next());<a name="line.746"></a>
<span class="sourceLineNo">747</span>                    xml.closeTag(NAME_TAG);<a name="line.747"></a>
<span class="sourceLineNo">748</span>                }<a name="line.748"></a>
<span class="sourceLineNo">749</span>            }<a name="line.749"></a>
<span class="sourceLineNo">750</span>            if (orfs!=null) {<a name="line.750"></a>
<span class="sourceLineNo">751</span>                for (Iterator j = synonyms.iterator(); j.hasNext(); ) {<a name="line.751"></a>
<span class="sourceLineNo">752</span>                    xml.openTag(NAME_TAG);<a name="line.752"></a>
<span class="sourceLineNo">753</span>                    xml.attribute(TYPE_ATTR,Terms.ORFNAME_KEY);<a name="line.753"></a>
<span class="sourceLineNo">754</span>                    xml.print((String)j.next());<a name="line.754"></a>
<span class="sourceLineNo">755</span>                    xml.closeTag(NAME_TAG);<a name="line.755"></a>
<span class="sourceLineNo">756</span>                }<a name="line.756"></a>
<span class="sourceLineNo">757</span>            }<a name="line.757"></a>
<span class="sourceLineNo">758</span>            <a name="line.758"></a>
<span class="sourceLineNo">759</span>            xml.closeTag(GENE_TAG);<a name="line.759"></a>
<span class="sourceLineNo">760</span>        }<a name="line.760"></a>
<span class="sourceLineNo">761</span>        <a name="line.761"></a>
<span class="sourceLineNo">762</span>        // source line (from taxon)<a name="line.762"></a>
<span class="sourceLineNo">763</span>        //   organism line<a name="line.763"></a>
<span class="sourceLineNo">764</span>        NCBITaxon tax = rs.getTaxon();<a name="line.764"></a>
<span class="sourceLineNo">765</span>        if (tax!=null) {<a name="line.765"></a>
<span class="sourceLineNo">766</span>            xml.openTag(ORGANISM_TAG);<a name="line.766"></a>
<span class="sourceLineNo">767</span>            xml.attribute(KEY_ATTR,""+(key++));<a name="line.767"></a>
<span class="sourceLineNo">768</span>            <a name="line.768"></a>
<span class="sourceLineNo">769</span>            for (Iterator i = tax.getNameClasses().iterator(); i.hasNext(); ) {<a name="line.769"></a>
<span class="sourceLineNo">770</span>                String nameclass = (String)i.next();<a name="line.770"></a>
<span class="sourceLineNo">771</span>                String ournameclass = Terms.COMMON_NAME_KEY;<a name="line.771"></a>
<span class="sourceLineNo">772</span>                if (nameclass.equalsIgnoreCase(Terms.FULL_NAME_KEY)) ournameclass = NCBITaxon.EQUIVALENT;<a name="line.772"></a>
<span class="sourceLineNo">773</span>                else if (nameclass.equalsIgnoreCase(Terms.SCIENTIFIC_NAME_KEY)) ournameclass = NCBITaxon.SCIENTIFIC;<a name="line.773"></a>
<span class="sourceLineNo">774</span>                else if (nameclass.equalsIgnoreCase(Terms.SYNONYM_NAME_KEY)) ournameclass = NCBITaxon.SYNONYM;<a name="line.774"></a>
<span class="sourceLineNo">775</span>                else if (nameclass.equalsIgnoreCase(Terms.ABBREV_NAME_KEY)) ournameclass = NCBITaxon.ACRONYM;<a name="line.775"></a>
<span class="sourceLineNo">776</span>                for (Iterator j = tax.getNames(nameclass).iterator(); j.hasNext(); ) {<a name="line.776"></a>
<span class="sourceLineNo">777</span>                    xml.openTag(NAME_TAG);<a name="line.777"></a>
<span class="sourceLineNo">778</span>                    xml.attribute(TYPE_ATTR,ournameclass);<a name="line.778"></a>
<span class="sourceLineNo">779</span>                    xml.print((String)j.next());<a name="line.779"></a>
<span class="sourceLineNo">780</span>                    xml.closeTag(NAME_TAG);<a name="line.780"></a>
<span class="sourceLineNo">781</span>                }<a name="line.781"></a>
<span class="sourceLineNo">782</span>            }<a name="line.782"></a>
<span class="sourceLineNo">783</span>            <a name="line.783"></a>
<span class="sourceLineNo">784</span>            xml.openTag(DBXREF_TAG);<a name="line.784"></a>
<span class="sourceLineNo">785</span>            xml.attribute(KEY_ATTR,""+(key++));<a name="line.785"></a>
<span class="sourceLineNo">786</span>            xml.attribute(TYPE_ATTR,Terms.NCBI_TAXON_KEY);<a name="line.786"></a>
<span class="sourceLineNo">787</span>            xml.attribute(ID_ATTR,""+tax.getNCBITaxID());<a name="line.787"></a>
<span class="sourceLineNo">788</span>            xml.closeTag(DBXREF_TAG);<a name="line.788"></a>
<span class="sourceLineNo">789</span>            <a name="line.789"></a>
<span class="sourceLineNo">790</span>            String h = tax.getNameHierarchy();<a name="line.790"></a>
<span class="sourceLineNo">791</span>            h = h.substring(0, h.length()-1); // chomp dot<a name="line.791"></a>
<span class="sourceLineNo">792</span>            String[] hierarch = h.split(";");<a name="line.792"></a>
<span class="sourceLineNo">793</span>            xml.openTag(LINEAGE_TAG);<a name="line.793"></a>
<span class="sourceLineNo">794</span>            for (int j = 0; j &lt; hierarch.length; j++) {<a name="line.794"></a>
<span class="sourceLineNo">795</span>                xml.openTag(TAXON_TAG);<a name="line.795"></a>
<span class="sourceLineNo">796</span>                xml.print(hierarch[j].trim());<a name="line.796"></a>
<span class="sourceLineNo">797</span>                xml.closeTag(TAXON_TAG);<a name="line.797"></a>
<span class="sourceLineNo">798</span>            }<a name="line.798"></a>
<span class="sourceLineNo">799</span>            xml.closeTag(LINEAGE_TAG);<a name="line.799"></a>
<span class="sourceLineNo">800</span>            <a name="line.800"></a>
<span class="sourceLineNo">801</span>            xml.closeTag(ORGANISM_TAG);<a name="line.801"></a>
<span class="sourceLineNo">802</span>        }<a name="line.802"></a>
<span class="sourceLineNo">803</span>        <a name="line.803"></a>
<span class="sourceLineNo">804</span>        // gene location line (organelle)<a name="line.804"></a>
<span class="sourceLineNo">805</span>        for (Iterator i = organelles.iterator(); i.hasNext(); ) {<a name="line.805"></a>
<span class="sourceLineNo">806</span>            String org = (String)i.next();<a name="line.806"></a>
<span class="sourceLineNo">807</span>            xml.openTag(GENELOCATION_TAG);<a name="line.807"></a>
<span class="sourceLineNo">808</span>            if (org.startsWith("Plasmid")) {<a name="line.808"></a>
<span class="sourceLineNo">809</span>                xml.attribute(TYPE_ATTR,"plasmid");<a name="line.809"></a>
<span class="sourceLineNo">810</span>                String[] subparts = org.split(",");<a name="line.810"></a>
<span class="sourceLineNo">811</span>                for (int j = 0; j &lt; parts.length; j++) {<a name="line.811"></a>
<span class="sourceLineNo">812</span>                    org = subparts[j].trim();<a name="line.812"></a>
<span class="sourceLineNo">813</span>                    if (org.startsWith("and")) org = org.substring(3).trim();<a name="line.813"></a>
<span class="sourceLineNo">814</span>                    org = org.substring("Plasmid".length()).trim();<a name="line.814"></a>
<span class="sourceLineNo">815</span>                    xml.openTag(GENELOCATION_NAME_TAG);<a name="line.815"></a>
<span class="sourceLineNo">816</span>                    xml.attribute(STATUS_ATTR,"known");<a name="line.816"></a>
<span class="sourceLineNo">817</span>                    xml.print(org);<a name="line.817"></a>
<span class="sourceLineNo">818</span>                    xml.closeTag(GENELOCATION_NAME_TAG);<a name="line.818"></a>
<span class="sourceLineNo">819</span>                }<a name="line.819"></a>
<span class="sourceLineNo">820</span>            } else {<a name="line.820"></a>
<span class="sourceLineNo">821</span>                xml.attribute(TYPE_ATTR,org.toLowerCase()); // uniprotxml must have lower case<a name="line.821"></a>
<span class="sourceLineNo">822</span>            }<a name="line.822"></a>
<span class="sourceLineNo">823</span>            xml.closeTag(GENELOCATION_TAG);<a name="line.823"></a>
<span class="sourceLineNo">824</span>        }<a name="line.824"></a>
<span class="sourceLineNo">825</span>        <a name="line.825"></a>
<span class="sourceLineNo">826</span>        // docrefs<a name="line.826"></a>
<span class="sourceLineNo">827</span>        for (Iterator&lt;RankedDocRef&gt; i = rs.getRankedDocRefs().iterator(); i.hasNext(); ) {<a name="line.827"></a>
<span class="sourceLineNo">828</span>            RankedDocRef rdr = i.next();<a name="line.828"></a>
<span class="sourceLineNo">829</span>            DocRef dr = rdr.getDocumentReference();<a name="line.829"></a>
<span class="sourceLineNo">830</span>            <a name="line.830"></a>
<span class="sourceLineNo">831</span>            xml.openTag(REFERENCE_TAG);<a name="line.831"></a>
<span class="sourceLineNo">832</span>            xml.attribute(KEY_ATTR,""+(key++));<a name="line.832"></a>
<span class="sourceLineNo">833</span>            <a name="line.833"></a>
<span class="sourceLineNo">834</span>            xml.openTag(CITATION_TAG);<a name="line.834"></a>
<span class="sourceLineNo">835</span>            xml.attribute(TYPE_ATTR,"journal article"); // faking it i know<a name="line.835"></a>
<span class="sourceLineNo">836</span>            <a name="line.836"></a>
<span class="sourceLineNo">837</span>            if (dr.getTitle()!=null) {<a name="line.837"></a>
<span class="sourceLineNo">838</span>                xml.openTag(TITLE_TAG);<a name="line.838"></a>
<span class="sourceLineNo">839</span>                xml.print(dr.getTitle());<a name="line.839"></a>
<span class="sourceLineNo">840</span>                xml.closeTag(TITLE_TAG);<a name="line.840"></a>
<span class="sourceLineNo">841</span>            }<a name="line.841"></a>
<span class="sourceLineNo">842</span>            <a name="line.842"></a>
<span class="sourceLineNo">843</span>            List&lt;DocRefAuthor&gt; auths = new ArrayList(dr.getAuthorList());<a name="line.843"></a>
<span class="sourceLineNo">844</span>            List&lt;DocRefAuthor&gt; editors = new ArrayList&lt;DocRefAuthor&gt;(auths);<a name="line.844"></a>
<span class="sourceLineNo">845</span>            for (final Iterator&lt;DocRefAuthor&gt; j = editors.iterator(); j.hasNext(); ) {<a name="line.845"></a>
<span class="sourceLineNo">846</span>                DocRefAuthor a = j.next();<a name="line.846"></a>
<span class="sourceLineNo">847</span>                if (!a.isEditor())<a name="line.847"></a>
<span class="sourceLineNo">848</span>                    j.remove();<a name="line.848"></a>
<span class="sourceLineNo">849</span>                else<a name="line.849"></a>
<span class="sourceLineNo">850</span>                    auths.remove(a);<a name="line.850"></a>
<span class="sourceLineNo">851</span>            }<a name="line.851"></a>
<span class="sourceLineNo">852</span>            if (!editors.isEmpty()) {<a name="line.852"></a>
<span class="sourceLineNo">853</span>                xml.openTag(EDITOR_LIST_TAG);<a name="line.853"></a>
<span class="sourceLineNo">854</span>                for (Iterator&lt;DocRefAuthor&gt; j = editors.iterator(); j.hasNext(); ) {<a name="line.854"></a>
<span class="sourceLineNo">855</span>                    DocRefAuthor a = j.next();<a name="line.855"></a>
<span class="sourceLineNo">856</span>                    if (a.isEditor()) {<a name="line.856"></a>
<span class="sourceLineNo">857</span>                        if (a.isConsortium()) {<a name="line.857"></a>
<span class="sourceLineNo">858</span>                            xml.openTag(CONSORTIUM_TAG);<a name="line.858"></a>
<span class="sourceLineNo">859</span>                            xml.attribute(NAME_ATTR,a.getName());<a name="line.859"></a>
<span class="sourceLineNo">860</span>                            xml.closeTag(CONSORTIUM_TAG);<a name="line.860"></a>
<span class="sourceLineNo">861</span>                        } else {<a name="line.861"></a>
<span class="sourceLineNo">862</span>                            xml.openTag(PERSON_TAG);<a name="line.862"></a>
<span class="sourceLineNo">863</span>                            xml.attribute(NAME_ATTR,a.getName());<a name="line.863"></a>
<span class="sourceLineNo">864</span>                            xml.closeTag(PERSON_TAG);<a name="line.864"></a>
<span class="sourceLineNo">865</span>                        }<a name="line.865"></a>
<span class="sourceLineNo">866</span>                    }<a name="line.866"></a>
<span class="sourceLineNo">867</span>                }<a name="line.867"></a>
<span class="sourceLineNo">868</span>                xml.closeTag(EDITOR_LIST_TAG);<a name="line.868"></a>
<span class="sourceLineNo">869</span>            }<a name="line.869"></a>
<span class="sourceLineNo">870</span>            if (!auths.isEmpty()) {<a name="line.870"></a>
<span class="sourceLineNo">871</span>                xml.openTag(AUTHOR_LIST_TAG);<a name="line.871"></a>
<span class="sourceLineNo">872</span>                for (Iterator j = auths.iterator(); j.hasNext(); ) {<a name="line.872"></a>
<span class="sourceLineNo">873</span>                    DocRefAuthor a = (DocRefAuthor)j.next();<a name="line.873"></a>
<span class="sourceLineNo">874</span>                    if (a.isConsortium()) {<a name="line.874"></a>
<span class="sourceLineNo">875</span>                        xml.openTag(CONSORTIUM_TAG);<a name="line.875"></a>
<span class="sourceLineNo">876</span>                        xml.attribute(NAME_ATTR,a.getName());<a name="line.876"></a>
<span class="sourceLineNo">877</span>                        xml.closeTag(CONSORTIUM_TAG);<a name="line.877"></a>
<span class="sourceLineNo">878</span>                    } else {<a name="line.878"></a>
<span class="sourceLineNo">879</span>                        xml.openTag(PERSON_TAG);<a name="line.879"></a>
<span class="sourceLineNo">880</span>                        xml.attribute(NAME_ATTR,a.getName());<a name="line.880"></a>
<span class="sourceLineNo">881</span>                        xml.closeTag(PERSON_TAG);<a name="line.881"></a>
<span class="sourceLineNo">882</span>                    }<a name="line.882"></a>
<span class="sourceLineNo">883</span>                }<a name="line.883"></a>
<span class="sourceLineNo">884</span>                xml.closeTag(AUTHOR_LIST_TAG);<a name="line.884"></a>
<span class="sourceLineNo">885</span>            }<a name="line.885"></a>
<span class="sourceLineNo">886</span>            <a name="line.886"></a>
<span class="sourceLineNo">887</span>            xml.openTag(LOCATOR_TAG);<a name="line.887"></a>
<span class="sourceLineNo">888</span>            xml.print(dr.getLocation());<a name="line.888"></a>
<span class="sourceLineNo">889</span>            xml.closeTag(LOCATOR_TAG);<a name="line.889"></a>
<span class="sourceLineNo">890</span>            <a name="line.890"></a>
<span class="sourceLineNo">891</span>            CrossRef cr = dr.getCrossref();<a name="line.891"></a>
<span class="sourceLineNo">892</span>            if (cr!=null) {<a name="line.892"></a>
<span class="sourceLineNo">893</span>                xml.openTag(DBXREF_TAG);<a name="line.893"></a>
<span class="sourceLineNo">894</span>                xml.attribute(TYPE_ATTR,cr.getDbname());<a name="line.894"></a>
<span class="sourceLineNo">895</span>                xml.attribute(ID_ATTR,cr.getAccession());<a name="line.895"></a>
<span class="sourceLineNo">896</span>                xml.attribute(KEY_ATTR,""+(key++));<a name="line.896"></a>
<span class="sourceLineNo">897</span>                if (!cr.getNoteSet().isEmpty()) {<a name="line.897"></a>
<span class="sourceLineNo">898</span>                    for (Iterator&lt;Note&gt; j = cr.getNoteSet().iterator(); j.hasNext(); ) {<a name="line.898"></a>
<span class="sourceLineNo">899</span>                        Note n = j.next();<a name="line.899"></a>
<span class="sourceLineNo">900</span>                        xml.openTag(PROPERTY_TAG);<a name="line.900"></a>
<span class="sourceLineNo">901</span>                        xml.attribute(TYPE_ATTR,n.getTerm().getName());<a name="line.901"></a>
<span class="sourceLineNo">902</span>                        xml.attribute(VALUE_ATTR,n.getValue());<a name="line.902"></a>
<span class="sourceLineNo">903</span>                        xml.closeTag(PROPERTY_TAG);<a name="line.903"></a>
<span class="sourceLineNo">904</span>                    }<a name="line.904"></a>
<span class="sourceLineNo">905</span>                }<a name="line.905"></a>
<span class="sourceLineNo">906</span>                xml.closeTag(DBXREF_TAG);<a name="line.906"></a>
<span class="sourceLineNo">907</span>            }<a name="line.907"></a>
<span class="sourceLineNo">908</span>            <a name="line.908"></a>
<span class="sourceLineNo">909</span>            xml.closeTag(CITATION_TAG);<a name="line.909"></a>
<span class="sourceLineNo">910</span>            <a name="line.910"></a>
<span class="sourceLineNo">911</span>            // RP<a name="line.911"></a>
<span class="sourceLineNo">912</span>            xml.openTag(RP_LINE_TAG);<a name="line.912"></a>
<span class="sourceLineNo">913</span>            xml.print(dr.getRemark());<a name="line.913"></a>
<span class="sourceLineNo">914</span>            xml.closeTag(RP_LINE_TAG);<a name="line.914"></a>
<span class="sourceLineNo">915</span>            // Print out ref position if present<a name="line.915"></a>
<span class="sourceLineNo">916</span>            if (rdr.getStart()!=null &amp;&amp; rdr.getEnd()!=null &amp;&amp; !rppat.matcher(dr.getRemark()).matches()) {<a name="line.916"></a>
<span class="sourceLineNo">917</span>                xml.openTag(RP_LINE_TAG);<a name="line.917"></a>
<span class="sourceLineNo">918</span>                xml.print("SEQUENCE OF "+rdr.getStart()+"-"+rdr.getEnd()+".");<a name="line.918"></a>
<span class="sourceLineNo">919</span>                xml.closeTag(RP_LINE_TAG);<a name="line.919"></a>
<span class="sourceLineNo">920</span>            }<a name="line.920"></a>
<span class="sourceLineNo">921</span>            <a name="line.921"></a>
<span class="sourceLineNo">922</span>            // RC<a name="line.922"></a>
<span class="sourceLineNo">923</span>            boolean rcOpened = false;<a name="line.923"></a>
<span class="sourceLineNo">924</span>            Integer rank = new Integer(rdr.getRank());<a name="line.924"></a>
<span class="sourceLineNo">925</span>            if (speciesRecs.get(rank)!=null) {<a name="line.925"></a>
<span class="sourceLineNo">926</span>                if (!rcOpened) {<a name="line.926"></a>
<span class="sourceLineNo">927</span>                    xml.openTag(RC_LINE_TAG);<a name="line.927"></a>
<span class="sourceLineNo">928</span>                    rcOpened = true;<a name="line.928"></a>
<span class="sourceLineNo">929</span>                }<a name="line.929"></a>
<span class="sourceLineNo">930</span>                for (Iterator j = ((List)speciesRecs.get(rank)).iterator(); j.hasNext(); ) {<a name="line.930"></a>
<span class="sourceLineNo">931</span>                    xml.openTag(RC_SPECIES_TAG);<a name="line.931"></a>
<span class="sourceLineNo">932</span>                    xml.print((String)j.next());<a name="line.932"></a>
<span class="sourceLineNo">933</span>                    xml.closeTag(RC_SPECIES_TAG);<a name="line.933"></a>
<span class="sourceLineNo">934</span>                }<a name="line.934"></a>
<span class="sourceLineNo">935</span>            }<a name="line.935"></a>
<span class="sourceLineNo">936</span>            if (strainRecs.get(rank)!=null) {<a name="line.936"></a>
<span class="sourceLineNo">937</span>                if (!rcOpened) {<a name="line.937"></a>
<span class="sourceLineNo">938</span>                    xml.openTag(RC_LINE_TAG);<a name="line.938"></a>
<span class="sourceLineNo">939</span>                    rcOpened = true;<a name="line.939"></a>
<span class="sourceLineNo">940</span>                }<a name="line.940"></a>
<span class="sourceLineNo">941</span>                for (Iterator j = ((List)strainRecs.get(rank)).iterator(); j.hasNext(); ) {<a name="line.941"></a>
<span class="sourceLineNo">942</span>                    xml.openTag(RC_STRAIN_TAG);<a name="line.942"></a>
<span class="sourceLineNo">943</span>                    xml.print((String)j.next());<a name="line.943"></a>
<span class="sourceLineNo">944</span>                    xml.closeTag(RC_STRAIN_TAG);<a name="line.944"></a>
<span class="sourceLineNo">945</span>                }<a name="line.945"></a>
<span class="sourceLineNo">946</span>            }<a name="line.946"></a>
<span class="sourceLineNo">947</span>            if (tissueRecs.get(rank)!=null) {<a name="line.947"></a>
<span class="sourceLineNo">948</span>                if (!rcOpened) {<a name="line.948"></a>
<span class="sourceLineNo">949</span>                    xml.openTag(RC_LINE_TAG);<a name="line.949"></a>
<span class="sourceLineNo">950</span>                    rcOpened = true;<a name="line.950"></a>
<span class="sourceLineNo">951</span>                }<a name="line.951"></a>
<span class="sourceLineNo">952</span>                for (Iterator j = ((List)tissueRecs.get(rank)).iterator(); j.hasNext(); ) {<a name="line.952"></a>
<span class="sourceLineNo">953</span>                    xml.openTag(RC_TISSUE_TAG);<a name="line.953"></a>
<span class="sourceLineNo">954</span>                    xml.print((String)j.next());<a name="line.954"></a>
<span class="sourceLineNo">955</span>                    xml.closeTag(RC_TISSUE_TAG);<a name="line.955"></a>
<span class="sourceLineNo">956</span>                }<a name="line.956"></a>
<span class="sourceLineNo">957</span>            }<a name="line.957"></a>
<span class="sourceLineNo">958</span>            if (transpRecs.get(rank)!=null) {<a name="line.958"></a>
<span class="sourceLineNo">959</span>                if (!rcOpened) {<a name="line.959"></a>
<span class="sourceLineNo">960</span>                    xml.openTag(RC_LINE_TAG);<a name="line.960"></a>
<span class="sourceLineNo">961</span>                    rcOpened = true;<a name="line.961"></a>
<span class="sourceLineNo">962</span>                }<a name="line.962"></a>
<span class="sourceLineNo">963</span>                for (Iterator j = ((List)transpRecs.get(rank)).iterator(); j.hasNext(); ) {<a name="line.963"></a>
<span class="sourceLineNo">964</span>                    xml.openTag(RC_TRANSP_TAG);<a name="line.964"></a>
<span class="sourceLineNo">965</span>                    xml.print((String)j.next());<a name="line.965"></a>
<span class="sourceLineNo">966</span>                    xml.closeTag(RC_TRANSP_TAG);<a name="line.966"></a>
<span class="sourceLineNo">967</span>                }<a name="line.967"></a>
<span class="sourceLineNo">968</span>            }<a name="line.968"></a>
<span class="sourceLineNo">969</span>            if (plasmidRecs.get(rank)!=null) {<a name="line.969"></a>
<span class="sourceLineNo">970</span>                if (!rcOpened) {<a name="line.970"></a>
<span class="sourceLineNo">971</span>                    xml.openTag(RC_LINE_TAG);<a name="line.971"></a>
<span class="sourceLineNo">972</span>                    rcOpened = true;<a name="line.972"></a>
<span class="sourceLineNo">973</span>                }<a name="line.973"></a>
<span class="sourceLineNo">974</span>                for (Iterator j = ((List)plasmidRecs.get(rank)).iterator(); j.hasNext(); ) {<a name="line.974"></a>
<span class="sourceLineNo">975</span>                    xml.openTag(RC_PLASMID_TAG);<a name="line.975"></a>
<span class="sourceLineNo">976</span>                    xml.print((String)j.next());<a name="line.976"></a>
<span class="sourceLineNo">977</span>                    xml.closeTag(RC_PLASMID_TAG);<a name="line.977"></a>
<span class="sourceLineNo">978</span>                }<a name="line.978"></a>
<span class="sourceLineNo">979</span>            }<a name="line.979"></a>
<span class="sourceLineNo">980</span>            if (rcOpened)<a name="line.980"></a>
<span class="sourceLineNo">981</span>                xml.closeTag(RC_LINE_TAG);<a name="line.981"></a>
<span class="sourceLineNo">982</span>            <a name="line.982"></a>
<span class="sourceLineNo">983</span>            xml.closeTag(REFERENCE_TAG);<a name="line.983"></a>
<span class="sourceLineNo">984</span>        }<a name="line.984"></a>
<span class="sourceLineNo">985</span>        <a name="line.985"></a>
<span class="sourceLineNo">986</span>        // comments<a name="line.986"></a>
<span class="sourceLineNo">987</span>        for (Iterator&lt;Comment&gt; i = rs.getComments().iterator(); i.hasNext(); ) {<a name="line.987"></a>
<span class="sourceLineNo">988</span>            // use UniProtCommentParser to convert each text comment from string to object<a name="line.988"></a>
<span class="sourceLineNo">989</span>            // do not print unconvertible ones (eg. no -!- on text)<a name="line.989"></a>
<span class="sourceLineNo">990</span>            Comment c = i.next();<a name="line.990"></a>
<span class="sourceLineNo">991</span>            if (UniProtCommentParser.isParseable(c)) {<a name="line.991"></a>
<span class="sourceLineNo">992</span>                // otherwise parse and display appropriately<a name="line.992"></a>
<span class="sourceLineNo">993</span>                UniProtCommentParser ucp = new UniProtCommentParser();<a name="line.993"></a>
<span class="sourceLineNo">994</span>                try {<a name="line.994"></a>
<span class="sourceLineNo">995</span>                    ucp.parseComment(c);<a name="line.995"></a>
<span class="sourceLineNo">996</span>                } catch (ParseException ce) {<a name="line.996"></a>
<span class="sourceLineNo">997</span>                    IOException e = new IOException("Failed to parse comment when outputting");<a name="line.997"></a>
<span class="sourceLineNo">998</span>                    e.initCause(ce);<a name="line.998"></a>
<span class="sourceLineNo">999</span>                    throw e;<a name="line.999"></a>
<span class="sourceLineNo">1000</span>                }<a name="line.1000"></a>
<span class="sourceLineNo">1001</span>                String type = ucp.getCommentType();<a name="line.1001"></a>
<span class="sourceLineNo">1002</span>                String xtype = type.toLowerCase(); // uniprotxml requires lower case<a name="line.1002"></a>
<span class="sourceLineNo">1003</span>                if (type.equalsIgnoreCase(UniProtCommentParser.PTM)) xtype = "posttranslational modification";<a name="line.1003"></a>
<span class="sourceLineNo">1004</span>                else if (type.equalsIgnoreCase(UniProtCommentParser.DATABASE)) xtype = "online information";<a name="line.1004"></a>
<span class="sourceLineNo">1005</span>                <a name="line.1005"></a>
<span class="sourceLineNo">1006</span>                xml.openTag(COMMENT_TAG);<a name="line.1006"></a>
<span class="sourceLineNo">1007</span>                xml.attribute(TYPE_ATTR,xtype);<a name="line.1007"></a>
<span class="sourceLineNo">1008</span>                <a name="line.1008"></a>
<span class="sourceLineNo">1009</span>                // database comment<a name="line.1009"></a>
<span class="sourceLineNo">1010</span>                if (type.equalsIgnoreCase(UniProtCommentParser.DATABASE)) {<a name="line.1010"></a>
<span class="sourceLineNo">1011</span>                    xml.attribute(NAME_ATTR,ucp.getDatabaseName());<a name="line.1011"></a>
<span class="sourceLineNo">1012</span>                    <a name="line.1012"></a>
<span class="sourceLineNo">1013</span>                    xml.openTag(COMMENT_LINK_TAG);<a name="line.1013"></a>
<span class="sourceLineNo">1014</span>                    xml.attribute(COMMENT_LINK_URI_ATTR,ucp.getUri());<a name="line.1014"></a>
<span class="sourceLineNo">1015</span>                    xml.closeTag(COMMENT_LINK_TAG);<a name="line.1015"></a>
<span class="sourceLineNo">1016</span>                }<a name="line.1016"></a>
<span class="sourceLineNo">1017</span>                // mass spec<a name="line.1017"></a>
<span class="sourceLineNo">1018</span>                else if (type.equalsIgnoreCase(UniProtCommentParser.MASS_SPECTROMETRY)) {<a name="line.1018"></a>
<span class="sourceLineNo">1019</span>                    xml.attribute(COMMENT_MASS_ATTR,""+ucp.getMolecularWeight());<a name="line.1019"></a>
<span class="sourceLineNo">1020</span>                    if (ucp.getMolWeightError()!=null) xml.attribute(COMMENT_ERROR_ATTR,""+ucp.getMolWeightError());<a name="line.1020"></a>
<span class="sourceLineNo">1021</span>                    xml.attribute(COMMENT_METHOD_ATTR,""+ucp.getMolWeightMethod());<a name="line.1021"></a>
<span class="sourceLineNo">1022</span>                    <a name="line.1022"></a>
<span class="sourceLineNo">1023</span>                    xml.openTag(LOCATION_TAG);<a name="line.1023"></a>
<span class="sourceLineNo">1024</span>                    xml.openTag(LOCATION_BEGIN_TAG);<a name="line.1024"></a>
<span class="sourceLineNo">1025</span>                    xml.attribute(LOCATION_POSITION_ATTR,""+ucp.getMolWeightRangeStart());<a name="line.1025"></a>
<span class="sourceLineNo">1026</span>                    xml.closeTag(LOCATION_BEGIN_TAG);<a name="line.1026"></a>
<span class="sourceLineNo">1027</span>                    xml.openTag(LOCATION_END_TAG);<a name="line.1027"></a>
<span class="sourceLineNo">1028</span>                    xml.attribute(LOCATION_POSITION_ATTR,""+ucp.getMolWeightRangeEnd());<a name="line.1028"></a>
<span class="sourceLineNo">1029</span>                    xml.closeTag(LOCATION_END_TAG);<a name="line.1029"></a>
<span class="sourceLineNo">1030</span>                    xml.closeTag(LOCATION_TAG);<a name="line.1030"></a>
<span class="sourceLineNo">1031</span>                }<a name="line.1031"></a>
<span class="sourceLineNo">1032</span>                // interaction<a name="line.1032"></a>
<span class="sourceLineNo">1033</span>                else if (type.equalsIgnoreCase(UniProtCommentParser.INTERACTION)) {<a name="line.1033"></a>
<span class="sourceLineNo">1034</span>                    // UniProt flat allows for multiple interactions per comment, but<a name="line.1034"></a>
<span class="sourceLineNo">1035</span>                    // UniProtXML only allows for a single one. So, we have to open/close<a name="line.1035"></a>
<span class="sourceLineNo">1036</span>                    // and write additional comments as necessary.<a name="line.1036"></a>
<span class="sourceLineNo">1037</span>                    for (Iterator j = ucp.getInteractions().iterator(); j.hasNext(); ) {<a name="line.1037"></a>
<span class="sourceLineNo">1038</span>                        // process comment<a name="line.1038"></a>
<span class="sourceLineNo">1039</span>                        Interaction interact = (Interaction)j.next();<a name="line.1039"></a>
<span class="sourceLineNo">1040</span>                        <a name="line.1040"></a>
<span class="sourceLineNo">1041</span>                        xml.openTag(COMMENT_INTERACTANT_TAG);<a name="line.1041"></a>
<span class="sourceLineNo">1042</span>                        xml.attribute(COMMENT_INTERACT_INTACT_ATTR,interact.getFirstIntActID());<a name="line.1042"></a>
<span class="sourceLineNo">1043</span>                        xml.closeTag(COMMENT_INTERACTANT_TAG);<a name="line.1043"></a>
<span class="sourceLineNo">1044</span>                        <a name="line.1044"></a>
<span class="sourceLineNo">1045</span>                        xml.openTag(COMMENT_INTERACTANT_TAG);<a name="line.1045"></a>
<span class="sourceLineNo">1046</span>                        xml.attribute(COMMENT_INTERACT_INTACT_ATTR,interact.getSecondIntActID());<a name="line.1046"></a>
<span class="sourceLineNo">1047</span>                        xml.openTag(ID_TAG);<a name="line.1047"></a>
<span class="sourceLineNo">1048</span>                        xml.print(interact.getID());<a name="line.1048"></a>
<span class="sourceLineNo">1049</span>                        xml.closeTag(ID_TAG);<a name="line.1049"></a>
<span class="sourceLineNo">1050</span>                        if (interact.getLabel()!=null) {<a name="line.1050"></a>
<span class="sourceLineNo">1051</span>                            xml.openTag(COMMENT_INTERACT_LABEL_TAG);<a name="line.1051"></a>
<span class="sourceLineNo">1052</span>                            xml.print(interact.getLabel());<a name="line.1052"></a>
<span class="sourceLineNo">1053</span>                            xml.closeTag(COMMENT_INTERACT_LABEL_TAG);<a name="line.1053"></a>
<span class="sourceLineNo">1054</span>                        }<a name="line.1054"></a>
<span class="sourceLineNo">1055</span>                        xml.closeTag(COMMENT_INTERACTANT_TAG);<a name="line.1055"></a>
<span class="sourceLineNo">1056</span>                        <a name="line.1056"></a>
<span class="sourceLineNo">1057</span>                        xml.openTag(COMMENT_ORGANISMS_TAG);<a name="line.1057"></a>
<span class="sourceLineNo">1058</span>                        xml.print(interact.isOrganismsDiffer()?"true":"false");<a name="line.1058"></a>
<span class="sourceLineNo">1059</span>                        xml.closeTag(COMMENT_ORGANISMS_TAG);<a name="line.1059"></a>
<span class="sourceLineNo">1060</span>                        <a name="line.1060"></a>
<span class="sourceLineNo">1061</span>                        xml.openTag(COMMENT_EXPERIMENTS_TAG);<a name="line.1061"></a>
<span class="sourceLineNo">1062</span>                        xml.print(""+interact.getNumberExperiments());<a name="line.1062"></a>
<span class="sourceLineNo">1063</span>                        xml.closeTag(COMMENT_EXPERIMENTS_TAG);<a name="line.1063"></a>
<span class="sourceLineNo">1064</span>                        <a name="line.1064"></a>
<span class="sourceLineNo">1065</span>                        // if has next, close and open next comment tag<a name="line.1065"></a>
<span class="sourceLineNo">1066</span>                        if (j.hasNext()) {<a name="line.1066"></a>
<span class="sourceLineNo">1067</span>                            xml.closeTag(COMMENT_TAG);<a name="line.1067"></a>
<span class="sourceLineNo">1068</span>                            xml.openTag(COMMENT_TAG);<a name="line.1068"></a>
<span class="sourceLineNo">1069</span>                            xml.attribute(TYPE_ATTR,xtype);<a name="line.1069"></a>
<span class="sourceLineNo">1070</span>                        }<a name="line.1070"></a>
<span class="sourceLineNo">1071</span>                    }<a name="line.1071"></a>
<span class="sourceLineNo">1072</span>                }<a name="line.1072"></a>
<span class="sourceLineNo">1073</span>                // alternative products<a name="line.1073"></a>
<span class="sourceLineNo">1074</span>                else if (type.equalsIgnoreCase(UniProtCommentParser.ALTERNATIVE_PRODUCTS)) {<a name="line.1074"></a>
<span class="sourceLineNo">1075</span>                    for (Iterator j = ucp.getEvents().iterator(); j.hasNext(); ) {<a name="line.1075"></a>
<span class="sourceLineNo">1076</span>                        Event event = (Event)j.next();<a name="line.1076"></a>
<span class="sourceLineNo">1077</span>                        xml.openTag(COMMENT_EVENT_TAG);<a name="line.1077"></a>
<span class="sourceLineNo">1078</span>                        xml.attribute(TYPE_ATTR,event.getType().toLowerCase()); // uniprotxml requires lowercase<a name="line.1078"></a>
<span class="sourceLineNo">1079</span>                        xml.closeTag(COMMENT_EVENT_TAG);<a name="line.1079"></a>
<span class="sourceLineNo">1080</span>                    }<a name="line.1080"></a>
<span class="sourceLineNo">1081</span>                    for (Iterator j = ucp.getIsoforms().iterator(); j.hasNext(); ) {<a name="line.1081"></a>
<span class="sourceLineNo">1082</span>                        Isoform isoform = (Isoform)j.next();<a name="line.1082"></a>
<span class="sourceLineNo">1083</span>                        xml.openTag(COMMENT_ISOFORM_TAG);<a name="line.1083"></a>
<span class="sourceLineNo">1084</span>                        for (Iterator k = isoform.getIsoIDs().iterator(); k.hasNext(); ) {<a name="line.1084"></a>
<span class="sourceLineNo">1085</span>                            xml.openTag(ID_TAG);<a name="line.1085"></a>
<span class="sourceLineNo">1086</span>                            xml.print((String)k.next());<a name="line.1086"></a>
<span class="sourceLineNo">1087</span>                            xml.closeTag(ID_TAG);<a name="line.1087"></a>
<span class="sourceLineNo">1088</span>                        }<a name="line.1088"></a>
<span class="sourceLineNo">1089</span>                        for (Iterator k = isoform.getNames().iterator(); k.hasNext(); ) {<a name="line.1089"></a>
<span class="sourceLineNo">1090</span>                            xml.openTag(NAME_TAG);<a name="line.1090"></a>
<span class="sourceLineNo">1091</span>                            xml.print((String)k.next());<a name="line.1091"></a>
<span class="sourceLineNo">1092</span>                            xml.closeTag(NAME_TAG);<a name="line.1092"></a>
<span class="sourceLineNo">1093</span>                        }<a name="line.1093"></a>
<span class="sourceLineNo">1094</span>                        xml.openTag(SEQUENCE_TAG);<a name="line.1094"></a>
<span class="sourceLineNo">1095</span>                        xml.attribute(TYPE_ATTR,isoform.getSequenceType().toLowerCase());<a name="line.1095"></a>
<span class="sourceLineNo">1096</span>                        if (isoform.getSequenceType().equalsIgnoreCase("Described")) {<a name="line.1096"></a>
<span class="sourceLineNo">1097</span>                            xml.attribute(REF_ATTR,isoform.getSequenceRef());<a name="line.1097"></a>
<span class="sourceLineNo">1098</span>                        }<a name="line.1098"></a>
<span class="sourceLineNo">1099</span>                        xml.closeTag(SEQUENCE_TAG);<a name="line.1099"></a>
<span class="sourceLineNo">1100</span>                        xml.openTag(NOTE_TAG);<a name="line.1100"></a>
<span class="sourceLineNo">1101</span>                        xml.print(isoform.getNote());<a name="line.1101"></a>
<span class="sourceLineNo">1102</span>                        xml.closeTag(NOTE_TAG);<a name="line.1102"></a>
<span class="sourceLineNo">1103</span>                        xml.closeTag(COMMENT_ISOFORM_TAG);<a name="line.1103"></a>
<span class="sourceLineNo">1104</span>                    }<a name="line.1104"></a>
<span class="sourceLineNo">1105</span>                }<a name="line.1105"></a>
<span class="sourceLineNo">1106</span>                // biophysicoblahblah stuff<a name="line.1106"></a>
<span class="sourceLineNo">1107</span>                else if (type.equalsIgnoreCase(UniProtCommentParser.BIOPHYSICOCHEMICAL_PROPERTIES)) {<a name="line.1107"></a>
<span class="sourceLineNo">1108</span>                    if (ucp.getAbsorptionNote()!=null) {<a name="line.1108"></a>
<span class="sourceLineNo">1109</span>                        xml.openTag(COMMENT_ABSORPTION_TAG);<a name="line.1109"></a>
<span class="sourceLineNo">1110</span>                        xml.openTag(COMMENT_ABS_MAX_TAG);<a name="line.1110"></a>
<span class="sourceLineNo">1111</span>                        xml.print(ucp.getAbsorptionMax());<a name="line.1111"></a>
<span class="sourceLineNo">1112</span>                        xml.closeTag(COMMENT_ABS_MAX_TAG);<a name="line.1112"></a>
<span class="sourceLineNo">1113</span>                        xml.openTag(TEXT_TAG);<a name="line.1113"></a>
<span class="sourceLineNo">1114</span>                        xml.print(ucp.getAbsorptionNote());<a name="line.1114"></a>
<span class="sourceLineNo">1115</span>                        xml.closeTag(TEXT_TAG);<a name="line.1115"></a>
<span class="sourceLineNo">1116</span>                        xml.closeTag(COMMENT_ABSORPTION_TAG);<a name="line.1116"></a>
<span class="sourceLineNo">1117</span>                    }<a name="line.1117"></a>
<span class="sourceLineNo">1118</span>                    if (ucp.getKineticsNote()!=null) {<a name="line.1118"></a>
<span class="sourceLineNo">1119</span>                        xml.openTag(COMMENT_KINETICS_TAG);<a name="line.1119"></a>
<span class="sourceLineNo">1120</span>                        for (Iterator j = ucp.getKMs().iterator(); j.hasNext(); ) {<a name="line.1120"></a>
<span class="sourceLineNo">1121</span>                            xml.openTag(COMMENT_KIN_KM_TAG);<a name="line.1121"></a>
<span class="sourceLineNo">1122</span>                            xml.print((String)j.next());<a name="line.1122"></a>
<span class="sourceLineNo">1123</span>                            xml.closeTag(COMMENT_KIN_KM_TAG);<a name="line.1123"></a>
<span class="sourceLineNo">1124</span>                        }<a name="line.1124"></a>
<span class="sourceLineNo">1125</span>                        for (Iterator j = ucp.getVMaxes().iterator(); j.hasNext(); ) {<a name="line.1125"></a>
<span class="sourceLineNo">1126</span>                            xml.openTag(COMMENT_KIN_VMAX_TAG);<a name="line.1126"></a>
<span class="sourceLineNo">1127</span>                            xml.print((String)j.next());<a name="line.1127"></a>
<span class="sourceLineNo">1128</span>                            xml.closeTag(COMMENT_KIN_VMAX_TAG);<a name="line.1128"></a>
<span class="sourceLineNo">1129</span>                        }<a name="line.1129"></a>
<span class="sourceLineNo">1130</span>                        xml.openTag(TEXT_TAG);<a name="line.1130"></a>
<span class="sourceLineNo">1131</span>                        xml.print(ucp.getKineticsNote());<a name="line.1131"></a>
<span class="sourceLineNo">1132</span>                        xml.closeTag(TEXT_TAG);<a name="line.1132"></a>
<span class="sourceLineNo">1133</span>                        xml.closeTag(COMMENT_KINETICS_TAG);<a name="line.1133"></a>
<span class="sourceLineNo">1134</span>                    }<a name="line.1134"></a>
<span class="sourceLineNo">1135</span>                    if (ucp.getPHDependence()!=null) {<a name="line.1135"></a>
<span class="sourceLineNo">1136</span>                        xml.openTag(COMMENT_PH_TAG);<a name="line.1136"></a>
<span class="sourceLineNo">1137</span>                        xml.print(ucp.getPHDependence());<a name="line.1137"></a>
<span class="sourceLineNo">1138</span>                        xml.closeTag(COMMENT_PH_TAG);<a name="line.1138"></a>
<span class="sourceLineNo">1139</span>                    }<a name="line.1139"></a>
<span class="sourceLineNo">1140</span>                    if (ucp.getRedoxPotential()!=null) {<a name="line.1140"></a>
<span class="sourceLineNo">1141</span>                        xml.openTag(COMMENT_REDOX_TAG);<a name="line.1141"></a>
<span class="sourceLineNo">1142</span>                        xml.print(ucp.getRedoxPotential());<a name="line.1142"></a>
<span class="sourceLineNo">1143</span>                        xml.closeTag(COMMENT_REDOX_TAG);<a name="line.1143"></a>
<span class="sourceLineNo">1144</span>                    }<a name="line.1144"></a>
<span class="sourceLineNo">1145</span>                    if (ucp.getTemperatureDependence()!=null) {<a name="line.1145"></a>
<span class="sourceLineNo">1146</span>                        xml.openTag(COMMENT_TEMPERATURE_TAG);<a name="line.1146"></a>
<span class="sourceLineNo">1147</span>                        xml.print(ucp.getTemperatureDependence());<a name="line.1147"></a>
<span class="sourceLineNo">1148</span>                        xml.closeTag(COMMENT_TEMPERATURE_TAG);<a name="line.1148"></a>
<span class="sourceLineNo">1149</span>                    }<a name="line.1149"></a>
<span class="sourceLineNo">1150</span>                }<a name="line.1150"></a>
<span class="sourceLineNo">1151</span>                // all other comments<a name="line.1151"></a>
<span class="sourceLineNo">1152</span>                else {<a name="line.1152"></a>
<span class="sourceLineNo">1153</span>                    xml.openTag(TEXT_TAG);<a name="line.1153"></a>
<span class="sourceLineNo">1154</span>                    xml.print(ucp.getText());<a name="line.1154"></a>
<span class="sourceLineNo">1155</span>                    xml.closeTag(TEXT_TAG);<a name="line.1155"></a>
<span class="sourceLineNo">1156</span>                }<a name="line.1156"></a>
<span class="sourceLineNo">1157</span>                <a name="line.1157"></a>
<span class="sourceLineNo">1158</span>                // finish comment up<a name="line.1158"></a>
<span class="sourceLineNo">1159</span>                if (ucp.getNote()!=null) {<a name="line.1159"></a>
<span class="sourceLineNo">1160</span>                    xml.openTag(NOTE_TAG);<a name="line.1160"></a>
<span class="sourceLineNo">1161</span>                    xml.print(ucp.getNote());<a name="line.1161"></a>
<span class="sourceLineNo">1162</span>                    xml.closeTag(NOTE_TAG);<a name="line.1162"></a>
<span class="sourceLineNo">1163</span>                }<a name="line.1163"></a>
<span class="sourceLineNo">1164</span>                <a name="line.1164"></a>
<span class="sourceLineNo">1165</span>                xml.closeTag(COMMENT_TAG);<a name="line.1165"></a>
<span class="sourceLineNo">1166</span>            }<a name="line.1166"></a>
<span class="sourceLineNo">1167</span>        }<a name="line.1167"></a>
<span class="sourceLineNo">1168</span>        <a name="line.1168"></a>
<span class="sourceLineNo">1169</span>        // xrefs<a name="line.1169"></a>
<span class="sourceLineNo">1170</span>        for (Iterator&lt;RankedCrossRef&gt; i = rs.getRankedCrossRefs().iterator(); i.hasNext(); ) {<a name="line.1170"></a>
<span class="sourceLineNo">1171</span>            RankedCrossRef rcr = i.next();<a name="line.1171"></a>
<span class="sourceLineNo">1172</span>            CrossRef cr = rcr.getCrossRef();<a name="line.1172"></a>
<span class="sourceLineNo">1173</span>            <a name="line.1173"></a>
<span class="sourceLineNo">1174</span>            xml.openTag(DBXREF_TAG);<a name="line.1174"></a>
<span class="sourceLineNo">1175</span>            String dbname = cr.getDbname();<a name="line.1175"></a>
<span class="sourceLineNo">1176</span>            xml.attribute(TYPE_ATTR,dbname);<a name="line.1176"></a>
<span class="sourceLineNo">1177</span>            xml.attribute(ID_ATTR,cr.getAccession());<a name="line.1177"></a>
<span class="sourceLineNo">1178</span>            xml.attribute(KEY_ATTR,""+(key++));<a name="line.1178"></a>
<span class="sourceLineNo">1179</span>            if (!cr.getNoteSet().isEmpty()) {<a name="line.1179"></a>
<span class="sourceLineNo">1180</span>                int acccount = 2;<a name="line.1180"></a>
<span class="sourceLineNo">1181</span>                for (Iterator&lt;Note&gt; j = cr.getNoteSet().iterator(); j.hasNext(); ) {<a name="line.1181"></a>
<span class="sourceLineNo">1182</span>                    Note n = j.next();<a name="line.1182"></a>
<span class="sourceLineNo">1183</span>                    if (n.getTerm().equals(Terms.getAdditionalAccessionTerm()) &amp;&amp; !n.getValue().equals("-")) {<a name="line.1183"></a>
<span class="sourceLineNo">1184</span>                        xml.openTag(PROPERTY_TAG);<a name="line.1184"></a>
<span class="sourceLineNo">1185</span>                        String name = n.getTerm().getName();<a name="line.1185"></a>
<span class="sourceLineNo">1186</span>                        if (acccount==2) {<a name="line.1186"></a>
<span class="sourceLineNo">1187</span>                            // SECONDARY IDENTIFIER<a name="line.1187"></a>
<span class="sourceLineNo">1188</span>                            if (dbname.equalsIgnoreCase("HIV") ||<a name="line.1188"></a>
<span class="sourceLineNo">1189</span>                                    dbname.equalsIgnoreCase("INTERPRO") ||<a name="line.1189"></a>
<span class="sourceLineNo">1190</span>                                    dbname.equalsIgnoreCase("PANTHER") ||<a name="line.1190"></a>
<span class="sourceLineNo">1191</span>                                    dbname.equalsIgnoreCase("PFAM") ||<a name="line.1191"></a>
<span class="sourceLineNo">1192</span>                                    dbname.equalsIgnoreCase("PIR") ||<a name="line.1192"></a>
<span class="sourceLineNo">1193</span>                                    dbname.equalsIgnoreCase("PRINTS") ||<a name="line.1193"></a>
<span class="sourceLineNo">1194</span>                                    dbname.equalsIgnoreCase("PRODOM") ||<a name="line.1194"></a>
<span class="sourceLineNo">1195</span>                                    dbname.equalsIgnoreCase("REBASE") ||<a name="line.1195"></a>
<span class="sourceLineNo">1196</span>                                    dbname.equalsIgnoreCase("SMART") ||<a name="line.1196"></a>
<span class="sourceLineNo">1197</span>                                    dbname.equalsIgnoreCase("TIGRFAMS")) {<a name="line.1197"></a>
<span class="sourceLineNo">1198</span>                                // the secondary identifier is the entry name.<a name="line.1198"></a>
<span class="sourceLineNo">1199</span>                                name = "entry name";<a name="line.1199"></a>
<span class="sourceLineNo">1200</span>                            } else if (dbname.equalsIgnoreCase("PDB")) {<a name="line.1200"></a>
<span class="sourceLineNo">1201</span>                                // the secondary identifier is the structure determination method, which is controlled vocabulary that currently includes: X-ray(for X-ray crystallography), NMR(for NMR spectroscopy), EM(for electron microscopy and cryo-electron diffraction), Fiber(for fiber diffraction), IR(for infrared spectroscopy), Model(for predicted models) and Neutron(for neutron diffraction).<a name="line.1201"></a>
<span class="sourceLineNo">1202</span>                                name = "structure determination method";<a name="line.1202"></a>
<span class="sourceLineNo">1203</span>                            } else if (dbname.equalsIgnoreCase("DICTYBASE") ||<a name="line.1203"></a>
<span class="sourceLineNo">1204</span>                                    dbname.equalsIgnoreCase("ECOGENE") ||<a name="line.1204"></a>
<span class="sourceLineNo">1205</span>                                    dbname.equalsIgnoreCase("FLYBASE") ||<a name="line.1205"></a>
<span class="sourceLineNo">1206</span>                                    dbname.equalsIgnoreCase("HGNC") ||<a name="line.1206"></a>
<span class="sourceLineNo">1207</span>                                    dbname.equalsIgnoreCase("MGI") ||<a name="line.1207"></a>
<span class="sourceLineNo">1208</span>                                    dbname.equalsIgnoreCase("RGD") ||<a name="line.1208"></a>
<span class="sourceLineNo">1209</span>                                    dbname.equalsIgnoreCase("SGD") ||<a name="line.1209"></a>
<span class="sourceLineNo">1210</span>                                    dbname.equalsIgnoreCase("STYGENE") ||<a name="line.1210"></a>
<span class="sourceLineNo">1211</span>                                    dbname.equalsIgnoreCase("SUBTILIST") ||<a name="line.1211"></a>
<span class="sourceLineNo">1212</span>                                    dbname.equalsIgnoreCase("WORMBASE") ||<a name="line.1212"></a>
<span class="sourceLineNo">1213</span>                                    dbname.equalsIgnoreCase("ZFIN")) {<a name="line.1213"></a>
<span class="sourceLineNo">1214</span>                                // the secondary identifier is the gene designation. If the gene designation is not available, a dash('-') is used.<a name="line.1214"></a>
<span class="sourceLineNo">1215</span>                                name = "gene designation";<a name="line.1215"></a>
<span class="sourceLineNo">1216</span>                            } else if (dbname.equalsIgnoreCase("GO")) {<a name="line.1216"></a>
<span class="sourceLineNo">1217</span>                                // the second identifier is a 1-letter abbreviation for one of the 3 ontology aspects, separated from the GO term by a column. If the term is longer than 46 characters, the first 43 characters are indicated followed by 3 dots('...'). The abbreviations for the 3 distinct aspects of the ontology are P(biological Process), F(molecular Function), and C(cellular Component).<a name="line.1217"></a>
<span class="sourceLineNo">1218</span>                                name = "term";<a name="line.1218"></a>
<span class="sourceLineNo">1219</span>                            } else if (dbname.equalsIgnoreCase("HAMAP")) {<a name="line.1219"></a>
<span class="sourceLineNo">1220</span>                                // the secondary identifier indicates if a domain is 'atypical' and/or 'fused', otherwise the field is empty('-').<a name="line.1220"></a>
<span class="sourceLineNo">1221</span>                                name = "domain";<a name="line.1221"></a>
<span class="sourceLineNo">1222</span>                            } else if (dbname.equalsIgnoreCase("ECO2DBASE")) {<a name="line.1222"></a>
<span class="sourceLineNo">1223</span>                                // the secondary identifier is the latest release number or edition of the database that has been used to derive the cross-reference.<a name="line.1223"></a>
<span class="sourceLineNo">1224</span>                                name = "release number";<a name="line.1224"></a>
<span class="sourceLineNo">1225</span>                            } else if (dbname.equalsIgnoreCase("SWISS-2DPAGE") ||<a name="line.1225"></a>
<span class="sourceLineNo">1226</span>                                    dbname.equalsIgnoreCase("HSC-2DPAGE")) {<a name="line.1226"></a>
<span class="sourceLineNo">1227</span>                                // the secondary identifier is the species or tissue of origin.<a name="line.1227"></a>
<span class="sourceLineNo">1228</span>                                name = "organism name";<a name="line.1228"></a>
<span class="sourceLineNo">1229</span>                            } else if (dbname.equalsIgnoreCase("ENSEMBL")) {<a name="line.1229"></a>
<span class="sourceLineNo">1230</span>                                // the secondary identifier is the species of origin.<a name="line.1230"></a>
<span class="sourceLineNo">1231</span>                                name = "organism name";<a name="line.1231"></a>
<span class="sourceLineNo">1232</span>                            } else if (dbname.equalsIgnoreCase("PIRSF")) {<a name="line.1232"></a>
<span class="sourceLineNo">1233</span>                                // the secondary identifier is the protein family name.<a name="line.1233"></a>
<span class="sourceLineNo">1234</span>                                name = "protein family name";<a name="line.1234"></a>
<span class="sourceLineNo">1235</span>                            } else if (dbname.equalsIgnoreCase("AARHUS") ||<a name="line.1235"></a>
<span class="sourceLineNo">1236</span>                                    dbname.equalsIgnoreCase("GHENT-2DPAGE")) {<a name="line.1236"></a>
<span class="sourceLineNo">1237</span>                                // the secondary identifier is either 'IEF' (for isoelectric focusing) or 'NEPHGE' (for non-equilibrium pH gradient electrophoresis).<a name="line.1237"></a>
<span class="sourceLineNo">1238</span>                                name = "secondary identifier";<a name="line.1238"></a>
<span class="sourceLineNo">1239</span>                            } else if (dbname.equalsIgnoreCase("WORMPEP")) {<a name="line.1239"></a>
<span class="sourceLineNo">1240</span>                                // the secondary identifier is a number attributed by the C.elegans genome-sequencing project to that protein.<a name="line.1240"></a>
<span class="sourceLineNo">1241</span>                                name = "C.elegans number";<a name="line.1241"></a>
<span class="sourceLineNo">1242</span>                            } else if (dbname.equalsIgnoreCase("AGD") ||<a name="line.1242"></a>
<span class="sourceLineNo">1243</span>                                    dbname.equalsIgnoreCase("ANU-2DPAGE") ||<a name="line.1243"></a>
<span class="sourceLineNo">1244</span>                                    dbname.equalsIgnoreCase("COMPLUYEAST-2DPAGE") ||<a name="line.1244"></a>
<span class="sourceLineNo">1245</span>                                    dbname.equalsIgnoreCase("ECHOBASE") ||<a name="line.1245"></a>
<span class="sourceLineNo">1246</span>                                    dbname.equalsIgnoreCase("GENEDB_SPOMBE") ||<a name="line.1246"></a>
<span class="sourceLineNo">1247</span>                                    dbname.equalsIgnoreCase("GERMONLINE") ||<a name="line.1247"></a>
<span class="sourceLineNo">1248</span>                                    dbname.equalsIgnoreCase("GLYCOSUITEDB") ||<a name="line.1248"></a>
<span class="sourceLineNo">1249</span>                                    dbname.equalsIgnoreCase("GRAMENE") ||<a name="line.1249"></a>
<span class="sourceLineNo">1250</span>                                    dbname.equalsIgnoreCase("H-INVDB") ||<a name="line.1250"></a>
<span class="sourceLineNo">1251</span>                                    dbname.equalsIgnoreCase("INTACT") ||<a name="line.1251"></a>
<span class="sourceLineNo">1252</span>                                    dbname.equalsIgnoreCase("LEGIOLIST") ||<a name="line.1252"></a>
<span class="sourceLineNo">1253</span>                                    dbname.equalsIgnoreCase("LEPROMA") ||<a name="line.1253"></a>
<span class="sourceLineNo">1254</span>                                    dbname.equalsIgnoreCase("LISTILIST") ||<a name="line.1254"></a>
<span class="sourceLineNo">1255</span>                                    dbname.equalsIgnoreCase("MAIZEDB") ||<a name="line.1255"></a>
<span class="sourceLineNo">1256</span>                                    dbname.equalsIgnoreCase("MEROPS") ||<a name="line.1256"></a>
<span class="sourceLineNo">1257</span>                                    dbname.equalsIgnoreCase("MIM") ||<a name="line.1257"></a>
<span class="sourceLineNo">1258</span>                                    dbname.equalsIgnoreCase("MYPULIST") ||<a name="line.1258"></a>
<span class="sourceLineNo">1259</span>                                    dbname.equalsIgnoreCase("OGP") ||<a name="line.1259"></a>
<span class="sourceLineNo">1260</span>                                    dbname.equalsIgnoreCase("PHCI-2DPAGE") ||<a name="line.1260"></a>
<span class="sourceLineNo">1261</span>                                    dbname.equalsIgnoreCase("PHOSSITE") ||<a name="line.1261"></a>
<span class="sourceLineNo">1262</span>                                    dbname.equalsIgnoreCase("PHOTOLIST") ||<a name="line.1262"></a>
<span class="sourceLineNo">1263</span>                                    dbname.equalsIgnoreCase("PMMA-2DPAGE") ||<a name="line.1263"></a>
<span class="sourceLineNo">1264</span>                                    dbname.equalsIgnoreCase("RAT-HEART-2DPAGE") ||<a name="line.1264"></a>
<span class="sourceLineNo">1265</span>                                    dbname.equalsIgnoreCase("REACTOME") ||<a name="line.1265"></a>
<span class="sourceLineNo">1266</span>                                    dbname.equalsIgnoreCase("SAGALIST") ||<a name="line.1266"></a>
<span class="sourceLineNo">1267</span>                                    dbname.equalsIgnoreCase("SIENA-2DPAGE") ||<a name="line.1267"></a>
<span class="sourceLineNo">1268</span>                                    dbname.equalsIgnoreCase("TAIR") ||<a name="line.1268"></a>
<span class="sourceLineNo">1269</span>                                    dbname.equalsIgnoreCase("TIGR") ||<a name="line.1269"></a>
<span class="sourceLineNo">1270</span>                                    dbname.equalsIgnoreCase("TRANSFAC") ||<a name="line.1270"></a>
<span class="sourceLineNo">1271</span>                                    dbname.equalsIgnoreCase("TUBERCULIST")) {<a name="line.1271"></a>
<span class="sourceLineNo">1272</span>                                // the secondary identifier is not used and a dash('-') is stored in that field.<a name="line.1272"></a>
<span class="sourceLineNo">1273</span>                                // should never get here - I hope!<a name="line.1273"></a>
<span class="sourceLineNo">1274</span>                            } else if (dbname.equalsIgnoreCase("HSSP")) {<a name="line.1274"></a>
<span class="sourceLineNo">1275</span>                                // the secondary identifier is the entry name of the PDB structure related to that of the entry in which the HSSP cross-reference is present.<a name="line.1275"></a>
<span class="sourceLineNo">1276</span>                                name = "entry name";<a name="line.1276"></a>
<span class="sourceLineNo">1277</span>                            } else if (dbname.equalsIgnoreCase("GENEFARM")) {<a name="line.1277"></a>
<span class="sourceLineNo">1278</span>                                // the secondary identifier is the gene family identifier. If the gene family identifier is not available, a dash('-') is used.<a name="line.1278"></a>
<span class="sourceLineNo">1279</span>                                name = "gene family";<a name="line.1279"></a>
<span class="sourceLineNo">1280</span>                            } else if (dbname.equalsIgnoreCase("SMR")) {<a name="line.1280"></a>
<span class="sourceLineNo">1281</span>                                // the secondary identifier indicates the range(s) relevant to the structure model(s).<a name="line.1281"></a>
<span class="sourceLineNo">1282</span>                                name = "range";<a name="line.1282"></a>
<span class="sourceLineNo">1283</span>                            } else if (dbname.equalsIgnoreCase("EMBL") ||<a name="line.1283"></a>
<span class="sourceLineNo">1284</span>                                    dbname.equalsIgnoreCase("DDBJ") ||<a name="line.1284"></a>
<span class="sourceLineNo">1285</span>                                    dbname.equalsIgnoreCase("GENBANK")) {<a name="line.1285"></a>
<span class="sourceLineNo">1286</span>                                // PROTEIN_ID; STATUS_IDENTIFIER; MOLECULE_TYPE<a name="line.1286"></a>
<span class="sourceLineNo">1287</span>                                name = "protein id";<a name="line.1287"></a>
<span class="sourceLineNo">1288</span>                            } else if (dbname.equalsIgnoreCase("PROSITE")) {<a name="line.1288"></a>
<span class="sourceLineNo">1289</span>                                // ENTRY_NAME; STATUS.<a name="line.1289"></a>
<span class="sourceLineNo">1290</span>                                name = "entry name";<a name="line.1290"></a>
<span class="sourceLineNo">1291</span>                            }<a name="line.1291"></a>
<span class="sourceLineNo">1292</span>                        } else if (acccount==3) {<a name="line.1292"></a>
<span class="sourceLineNo">1293</span>                            // TERTIARY IDENTIFIER<a name="line.1293"></a>
<span class="sourceLineNo">1294</span>                            if (dbname.equalsIgnoreCase("HAMAP") ||<a name="line.1294"></a>
<span class="sourceLineNo">1295</span>                                    dbname.equalsIgnoreCase("PANTHER") ||<a name="line.1295"></a>
<span class="sourceLineNo">1296</span>                                    dbname.equalsIgnoreCase("PFAM") ||<a name="line.1296"></a>
<span class="sourceLineNo">1297</span>                                    dbname.equalsIgnoreCase("PIRSF") ||<a name="line.1297"></a>
<span class="sourceLineNo">1298</span>                                    dbname.equalsIgnoreCase("PRODOM") ||<a name="line.1298"></a>
<span class="sourceLineNo">1299</span>                                    dbname.equalsIgnoreCase("SMART") ||<a name="line.1299"></a>
<span class="sourceLineNo">1300</span>                                    dbname.equalsIgnoreCase("TIGRFAMS")) {<a name="line.1300"></a>
<span class="sourceLineNo">1301</span>                                // the tertiary identifier is the number of hits found in the sequence.<a name="line.1301"></a>
<span class="sourceLineNo">1302</span>                                name = "number of hits";<a name="line.1302"></a>
<span class="sourceLineNo">1303</span>                            } else if (dbname.equalsIgnoreCase("GO")) {<a name="line.1303"></a>
<span class="sourceLineNo">1304</span>                                // the tertiary identifier is a 3-character GO evidence code. The meaning of the evidence codes is: IDA=inferred from direct assay, IMP=inferred from mutant phenotype, IGI=inferred from genetic interaction, IPI=inferred from physical interaction, IEP=inferred from expression pattern, TAS=traceable author statement, NAS=non-traceable author statement, IC=inferred by curator, ISS=inferred from sequence or structural similarity.<a name="line.1304"></a>
<span class="sourceLineNo">1305</span>                                name = "evidence";<a name="line.1305"></a>
<span class="sourceLineNo">1306</span>                            } else if (dbname.equalsIgnoreCase("PDB")) {<a name="line.1306"></a>
<span class="sourceLineNo">1307</span>                                // the tertiary identifier indicates the chain(s) and the corresponding range, of which the structure has been determined. If the range is unknown, a dash is given rather than the range positions(e.g. 'A/B=-.'), if the chains and the range is unknown, a dash is used.<a name="line.1307"></a>
<span class="sourceLineNo">1308</span>                                name = "chains";<a name="line.1308"></a>
<span class="sourceLineNo">1309</span>                            } else if (dbname.equalsIgnoreCase("EMBL") ||<a name="line.1309"></a>
<span class="sourceLineNo">1310</span>                                    dbname.equalsIgnoreCase("DDBJ") ||<a name="line.1310"></a>
<span class="sourceLineNo">1311</span>                                    dbname.equalsIgnoreCase("GENBANK")) {<a name="line.1311"></a>
<span class="sourceLineNo">1312</span>                                // PROTEIN_ID; STATUS_IDENTIFIER; MOLECULE_TYPE<a name="line.1312"></a>
<span class="sourceLineNo">1313</span>                                name = "status identifier";<a name="line.1313"></a>
<span class="sourceLineNo">1314</span>                            } else if (dbname.equalsIgnoreCase("PROSITE")) {<a name="line.1314"></a>
<span class="sourceLineNo">1315</span>                                // ENTRY_NAME; STATUS.<a name="line.1315"></a>
<span class="sourceLineNo">1316</span>                                name = "status";<a name="line.1316"></a>
<span class="sourceLineNo">1317</span>                            }<a name="line.1317"></a>
<span class="sourceLineNo">1318</span>                        } else {<a name="line.1318"></a>
<span class="sourceLineNo">1319</span>                            // QUATERNARY AND ADDITIONAL<a name="line.1319"></a>
<span class="sourceLineNo">1320</span>                            if (dbname.equalsIgnoreCase("EMBL") ||<a name="line.1320"></a>
<span class="sourceLineNo">1321</span>                                    dbname.equalsIgnoreCase("DDBJ") ||<a name="line.1321"></a>
<span class="sourceLineNo">1322</span>                                    dbname.equalsIgnoreCase("GENBANK")) {<a name="line.1322"></a>
<span class="sourceLineNo">1323</span>                                // PROTEIN_ID; STATUS_IDENTIFIER; MOLECULE_TYPE<a name="line.1323"></a>
<span class="sourceLineNo">1324</span>                                name = "molecule type";<a name="line.1324"></a>
<span class="sourceLineNo">1325</span>                            }<a name="line.1325"></a>
<span class="sourceLineNo">1326</span>                        }<a name="line.1326"></a>
<span class="sourceLineNo">1327</span>                        xml.attribute(TYPE_ATTR,name);<a name="line.1327"></a>
<span class="sourceLineNo">1328</span>                        xml.attribute(VALUE_ATTR,n.getValue());<a name="line.1328"></a>
<span class="sourceLineNo">1329</span>                        xml.closeTag(PROPERTY_TAG);<a name="line.1329"></a>
<span class="sourceLineNo">1330</span>                        acccount++;<a name="line.1330"></a>
<span class="sourceLineNo">1331</span>                    }<a name="line.1331"></a>
<span class="sourceLineNo">1332</span>                }<a name="line.1332"></a>
<span class="sourceLineNo">1333</span>            }<a name="line.1333"></a>
<span class="sourceLineNo">1334</span>            xml.closeTag(DBXREF_TAG);<a name="line.1334"></a>
<span class="sourceLineNo">1335</span>        }<a name="line.1335"></a>
<span class="sourceLineNo">1336</span>        <a name="line.1336"></a>
<span class="sourceLineNo">1337</span>        // protein exists<a name="line.1337"></a>
<span class="sourceLineNo">1338</span>        xml.openTag(PROTEIN_EXISTS_TAG);<a name="line.1338"></a>
<span class="sourceLineNo">1339</span>        xml.attribute(TYPE_ATTR,proteinExists);<a name="line.1339"></a>
<span class="sourceLineNo">1340</span>        xml.closeTag(PROTEIN_EXISTS_TAG);<a name="line.1340"></a>
<span class="sourceLineNo">1341</span>        <a name="line.1341"></a>
<span class="sourceLineNo">1342</span>        // keywords<a name="line.1342"></a>
<span class="sourceLineNo">1343</span>        for (Iterator j = kws.iterator(); j.hasNext(); ) {<a name="line.1343"></a>
<span class="sourceLineNo">1344</span>            ComparableTerm t = (ComparableTerm)j.next();<a name="line.1344"></a>
<span class="sourceLineNo">1345</span>            xml.openTag(KEYWORD_TAG);<a name="line.1345"></a>
<span class="sourceLineNo">1346</span>            xml.attribute(ID_ATTR,t.getIdentifier());<a name="line.1346"></a>
<span class="sourceLineNo">1347</span>            xml.print(t.getName());<a name="line.1347"></a>
<span class="sourceLineNo">1348</span>            xml.closeTag(KEYWORD_TAG);<a name="line.1348"></a>
<span class="sourceLineNo">1349</span>        }<a name="line.1349"></a>
<span class="sourceLineNo">1350</span>        <a name="line.1350"></a>
<span class="sourceLineNo">1351</span>        // features<a name="line.1351"></a>
<span class="sourceLineNo">1352</span>        for (Iterator i = rs.getFeatureSet().iterator(); i.hasNext(); ) {<a name="line.1352"></a>
<span class="sourceLineNo">1353</span>            RichFeature f = (RichFeature)i.next();<a name="line.1353"></a>
<span class="sourceLineNo">1354</span>            String descr = null;<a name="line.1354"></a>
<span class="sourceLineNo">1355</span>            String ftid = null;<a name="line.1355"></a>
<span class="sourceLineNo">1356</span>            String ref = null;<a name="line.1356"></a>
<span class="sourceLineNo">1357</span>            String status = null;<a name="line.1357"></a>
<span class="sourceLineNo">1358</span>            String original = null;<a name="line.1358"></a>
<span class="sourceLineNo">1359</span>            String locseq = null;<a name="line.1359"></a>
<span class="sourceLineNo">1360</span>            List variation = new ArrayList();<a name="line.1360"></a>
<span class="sourceLineNo">1361</span>            for (Iterator&lt;Note&gt; j = f.getNoteSet().iterator(); j.hasNext(); ) {<a name="line.1361"></a>
<span class="sourceLineNo">1362</span>                Note n = j.next();<a name="line.1362"></a>
<span class="sourceLineNo">1363</span>                if (n.getTerm().equals(Terms.getFTIdTerm())) ftid = n.getValue();<a name="line.1363"></a>
<span class="sourceLineNo">1364</span>                else if (n.getTerm().equals(Terms.getFeatureDescTerm())) descr = n.getValue();<a name="line.1364"></a>
<span class="sourceLineNo">1365</span>                else if (n.getTerm().equals(Terms.getFeatureStatusTerm())) status = n.getValue();<a name="line.1365"></a>
<span class="sourceLineNo">1366</span>                else if (n.getTerm().equals(Terms.getFeatureRefTerm())) ref = n.getValue();<a name="line.1366"></a>
<span class="sourceLineNo">1367</span>                else if (n.getTerm().equals(Terms.getFeatureOriginalTerm())) original = n.getValue();<a name="line.1367"></a>
<span class="sourceLineNo">1368</span>                else if (n.getTerm().equals(Terms.getFeatureVariationTerm())) variation.add(n.getValue());<a name="line.1368"></a>
<span class="sourceLineNo">1369</span>                else if (n.getTerm().equals(Terms.getLocationSequenceTerm())) locseq = n.getValue();<a name="line.1369"></a>
<span class="sourceLineNo">1370</span>            }<a name="line.1370"></a>
<span class="sourceLineNo">1371</span>            <a name="line.1371"></a>
<span class="sourceLineNo">1372</span>            xml.openTag(FEATURE_TAG);<a name="line.1372"></a>
<span class="sourceLineNo">1373</span>            <a name="line.1373"></a>
<span class="sourceLineNo">1374</span>            xml.attribute(TYPE_ATTR,f.getTypeTerm().getName()); // TODO : need to translate from UniProt flatfile format names?<a name="line.1374"></a>
<span class="sourceLineNo">1375</span>            if (ftid!=null) xml.attribute(ID_ATTR,ftid);<a name="line.1375"></a>
<span class="sourceLineNo">1376</span>            if (descr!=null) xml.attribute(FEATURE_DESC_ATTR,descr);<a name="line.1376"></a>
<span class="sourceLineNo">1377</span>            if (ref!=null) xml.attribute(REF_ATTR,ref);<a name="line.1377"></a>
<span class="sourceLineNo">1378</span>            if (status!=null) xml.attribute(STATUS_ATTR,status);<a name="line.1378"></a>
<span class="sourceLineNo">1379</span>            if (original!=null) {<a name="line.1379"></a>
<span class="sourceLineNo">1380</span>                xml.openTag(FEATURE_ORIGINAL_TAG);<a name="line.1380"></a>
<span class="sourceLineNo">1381</span>                xml.print(original.trim());<a name="line.1381"></a>
<span class="sourceLineNo">1382</span>                xml.closeTag(FEATURE_ORIGINAL_TAG);<a name="line.1382"></a>
<span class="sourceLineNo">1383</span>            }<a name="line.1383"></a>
<span class="sourceLineNo">1384</span>            for (Iterator j = variation.iterator(); j.hasNext(); ) {<a name="line.1384"></a>
<span class="sourceLineNo">1385</span>                xml.openTag(FEATURE_VARIATION_TAG);<a name="line.1385"></a>
<span class="sourceLineNo">1386</span>                xml.print(((String)j.next()).trim());<a name="line.1386"></a>
<span class="sourceLineNo">1387</span>                xml.closeTag(FEATURE_VARIATION_TAG);<a name="line.1387"></a>
<span class="sourceLineNo">1388</span>            }<a name="line.1388"></a>
<span class="sourceLineNo">1389</span>            <a name="line.1389"></a>
<span class="sourceLineNo">1390</span>            xml.openTag(LOCATION_TAG);<a name="line.1390"></a>
<span class="sourceLineNo">1391</span>            if (locseq!=null) xml.attribute(LOCATION_SEQ_ATTR,locseq.trim());<a name="line.1391"></a>
<span class="sourceLineNo">1392</span>            RichLocation rl = (RichLocation)f.getLocation();<a name="line.1392"></a>
<span class="sourceLineNo">1393</span>            if (rl.getMinPosition().equals(rl.getMaxPosition())) {<a name="line.1393"></a>
<span class="sourceLineNo">1394</span>                // point position<a name="line.1394"></a>
<span class="sourceLineNo">1395</span>                xml.openTag(LOCATION_POSITION_TAG);<a name="line.1395"></a>
<span class="sourceLineNo">1396</span>                if (rl.getMinPosition().getFuzzyStart() || rl.getMaxPosition().getFuzzyStart()) xml.attribute(STATUS_ATTR,"less than");<a name="line.1396"></a>
<span class="sourceLineNo">1397</span>                else if (rl.getMinPosition().getFuzzyEnd() || rl.getMaxPosition().getFuzzyEnd()) xml.attribute(STATUS_ATTR,"greater than");<a name="line.1397"></a>
<span class="sourceLineNo">1398</span>                xml.attribute(LOCATION_POSITION_ATTR,""+rl.getMin());<a name="line.1398"></a>
<span class="sourceLineNo">1399</span>                xml.closeTag(LOCATION_POSITION_TAG);<a name="line.1399"></a>
<span class="sourceLineNo">1400</span>            } else {<a name="line.1400"></a>
<span class="sourceLineNo">1401</span>                // range position<a name="line.1401"></a>
<span class="sourceLineNo">1402</span>                // begin<a name="line.1402"></a>
<span class="sourceLineNo">1403</span>                xml.openTag(LOCATION_BEGIN_TAG);<a name="line.1403"></a>
<span class="sourceLineNo">1404</span>                Position begin = rl.getMinPosition();<a name="line.1404"></a>
<span class="sourceLineNo">1405</span>                if (begin.getFuzzyStart()) xml.attribute(STATUS_ATTR,"less than");<a name="line.1405"></a>
<span class="sourceLineNo">1406</span>                else if (begin.getFuzzyEnd()) xml.attribute(STATUS_ATTR,"greater than");<a name="line.1406"></a>
<span class="sourceLineNo">1407</span>                xml.attribute(LOCATION_POSITION_ATTR,""+begin.getStart());<a name="line.1407"></a>
<span class="sourceLineNo">1408</span>                xml.closeTag(LOCATION_BEGIN_TAG);<a name="line.1408"></a>
<span class="sourceLineNo">1409</span>                // end<a name="line.1409"></a>
<span class="sourceLineNo">1410</span>                xml.openTag(LOCATION_END_TAG);<a name="line.1410"></a>
<span class="sourceLineNo">1411</span>                Position end = rl.getMaxPosition();<a name="line.1411"></a>
<span class="sourceLineNo">1412</span>                if (end.getFuzzyStart()) xml.attribute(STATUS_ATTR,"less than");<a name="line.1412"></a>
<span class="sourceLineNo">1413</span>                else if (end.getFuzzyEnd()) xml.attribute(STATUS_ATTR,"greater than");<a name="line.1413"></a>
<span class="sourceLineNo">1414</span>                xml.attribute(LOCATION_POSITION_ATTR,""+end.getEnd());<a name="line.1414"></a>
<span class="sourceLineNo">1415</span>                xml.closeTag(LOCATION_END_TAG);<a name="line.1415"></a>
<span class="sourceLineNo">1416</span>            }<a name="line.1416"></a>
<span class="sourceLineNo">1417</span>            xml.closeTag(LOCATION_TAG);<a name="line.1417"></a>
<span class="sourceLineNo">1418</span>            <a name="line.1418"></a>
<span class="sourceLineNo">1419</span>            xml.closeTag(FEATURE_TAG);<a name="line.1419"></a>
<span class="sourceLineNo">1420</span>        }<a name="line.1420"></a>
<span class="sourceLineNo">1421</span>        <a name="line.1421"></a>
<span class="sourceLineNo">1422</span>        // evidence<a name="line.1422"></a>
<span class="sourceLineNo">1423</span>        for (Iterator i = evidenceIDs.iterator(); i.hasNext(); ) {<a name="line.1423"></a>
<span class="sourceLineNo">1424</span>            Integer evidenceID = (Integer)i.next();<a name="line.1424"></a>
<span class="sourceLineNo">1425</span>            String cat = (String)evcats.get(evidenceID);<a name="line.1425"></a>
<span class="sourceLineNo">1426</span>            String type = (String)evtypes.get(evidenceID);<a name="line.1426"></a>
<span class="sourceLineNo">1427</span>            String date = (String)evdates.get(evidenceID);<a name="line.1427"></a>
<span class="sourceLineNo">1428</span>            String attr = (String)evattrs.get(evidenceID);<a name="line.1428"></a>
<span class="sourceLineNo">1429</span>            <a name="line.1429"></a>
<span class="sourceLineNo">1430</span>            xml.openTag(EVIDENCE_TAG);<a name="line.1430"></a>
<span class="sourceLineNo">1431</span>            xml.attribute(KEY_ATTR,""+(key++));<a name="line.1431"></a>
<span class="sourceLineNo">1432</span>            xml.attribute(EVIDENCE_CATEGORY_ATTR,cat);<a name="line.1432"></a>
<span class="sourceLineNo">1433</span>            xml.attribute(EVIDENCE_DATE_ATTR,date);<a name="line.1433"></a>
<span class="sourceLineNo">1434</span>            xml.attribute(TYPE_ATTR,type);<a name="line.1434"></a>
<span class="sourceLineNo">1435</span>            if (attr!=null) xml.attribute(EVIDENCE_ATTRIBUTE_ATTR,attr);<a name="line.1435"></a>
<span class="sourceLineNo">1436</span>            xml.closeTag(EVIDENCE_TAG);<a name="line.1436"></a>
<span class="sourceLineNo">1437</span>        }<a name="line.1437"></a>
<span class="sourceLineNo">1438</span>        <a name="line.1438"></a>
<span class="sourceLineNo">1439</span>        // sequence<a name="line.1439"></a>
<span class="sourceLineNo">1440</span>        int mw = 0;<a name="line.1440"></a>
<span class="sourceLineNo">1441</span>        try {<a name="line.1441"></a>
<span class="sourceLineNo">1442</span>            mw = (int)MassCalc.getMolecularWeight(rs);<a name="line.1442"></a>
<span class="sourceLineNo">1443</span>        } catch (IllegalSymbolException e) {<a name="line.1443"></a>
<span class="sourceLineNo">1444</span>            throw new RuntimeException("Found illegal symbol", e);<a name="line.1444"></a>
<span class="sourceLineNo">1445</span>        }<a name="line.1445"></a>
<span class="sourceLineNo">1446</span>        CRC64Checksum crc = new CRC64Checksum();<a name="line.1446"></a>
<span class="sourceLineNo">1447</span>        String seqstr = rs.seqString();<a name="line.1447"></a>
<span class="sourceLineNo">1448</span>        crc.update(seqstr.getBytes(),0,seqstr.length());<a name="line.1448"></a>
<span class="sourceLineNo">1449</span>        xml.openTag(SEQUENCE_TAG);<a name="line.1449"></a>
<span class="sourceLineNo">1450</span>        xml.attribute(SEQUENCE_VERSION_ATTR,""+rs.getVersion());<a name="line.1450"></a>
<span class="sourceLineNo">1451</span>        xml.attribute(SEQUENCE_LENGTH_ATTR,""+rs.length());<a name="line.1451"></a>
<span class="sourceLineNo">1452</span>        xml.attribute(SEQUENCE_MASS_ATTR,""+mw);<a name="line.1452"></a>
<span class="sourceLineNo">1453</span>        xml.attribute(SEQUENCE_CHECKSUM_ATTR,""+crc);<a name="line.1453"></a>
<span class="sourceLineNo">1454</span>        xml.attribute(SEQUENCE_MODIFIED_ATTR,(udat==null?cdat:udat)); // sequence update<a name="line.1454"></a>
<span class="sourceLineNo">1455</span>        String[] lines = StringTools.wordWrap(rs.seqString(), "\\s+", this.getLineWidth());<a name="line.1455"></a>
<span class="sourceLineNo">1456</span>        for (int i = 0; i &lt; lines.length; i ++) xml.println(lines[i]);<a name="line.1456"></a>
<span class="sourceLineNo">1457</span>        xml.closeTag(SEQUENCE_TAG);<a name="line.1457"></a>
<span class="sourceLineNo">1458</span>        <a name="line.1458"></a>
<span class="sourceLineNo">1459</span>        // close entry<a name="line.1459"></a>
<span class="sourceLineNo">1460</span>        xml.closeTag(ENTRY_TAG);<a name="line.1460"></a>
<span class="sourceLineNo">1461</span>        <a name="line.1461"></a>
<span class="sourceLineNo">1462</span>        // copyright (if present)<a name="line.1462"></a>
<span class="sourceLineNo">1463</span>        if (copyright!=null) {<a name="line.1463"></a>
<span class="sourceLineNo">1464</span>            xml.openTag(COPYRIGHT_TAG);<a name="line.1464"></a>
<span class="sourceLineNo">1465</span>            xml.println(copyright);<a name="line.1465"></a>
<span class="sourceLineNo">1466</span>            xml.closeTag(COPYRIGHT_TAG);<a name="line.1466"></a>
<span class="sourceLineNo">1467</span>        }<a name="line.1467"></a>
<span class="sourceLineNo">1468</span>        <a name="line.1468"></a>
<span class="sourceLineNo">1469</span>        pw.flush();<a name="line.1469"></a>
<span class="sourceLineNo">1470</span>    }<a name="line.1470"></a>
<span class="sourceLineNo">1471</span>    <a name="line.1471"></a>
<span class="sourceLineNo">1472</span>    /**<a name="line.1472"></a>
<span class="sourceLineNo">1473</span>     * {@inheritDoc}<a name="line.1473"></a>
<span class="sourceLineNo">1474</span>     */<a name="line.1474"></a>
<span class="sourceLineNo">1475</span>    public String getDefaultFormat() {<a name="line.1475"></a>
<span class="sourceLineNo">1476</span>        return UNIPROTXML_FORMAT;<a name="line.1476"></a>
<span class="sourceLineNo">1477</span>    }<a name="line.1477"></a>
<span class="sourceLineNo">1478</span>    <a name="line.1478"></a>
<span class="sourceLineNo">1479</span>    // SAX event handler for parsing http://www.ebi.uniprot.org/support/docs/uniprot.xsd<a name="line.1479"></a>
<span class="sourceLineNo">1480</span>    private class UniProtXMLHandler extends DefaultHandler {<a name="line.1480"></a>
<span class="sourceLineNo">1481</span>        <a name="line.1481"></a>
<span class="sourceLineNo">1482</span>        private RichSequenceFormat parent;<a name="line.1482"></a>
<span class="sourceLineNo">1483</span>        private SymbolTokenization symParser;<a name="line.1483"></a>
<span class="sourceLineNo">1484</span>        private RichSeqIOListener rlistener;<a name="line.1484"></a>
<span class="sourceLineNo">1485</span>        private Namespace ns;<a name="line.1485"></a>
<span class="sourceLineNo">1486</span>        private StringBuffer m_currentString;<a name="line.1486"></a>
<span class="sourceLineNo">1487</span>        <a name="line.1487"></a>
<span class="sourceLineNo">1488</span>        private NCBITaxon tax;<a name="line.1488"></a>
<span class="sourceLineNo">1489</span>        private RichFeature.Template templ;<a name="line.1489"></a>
<span class="sourceLineNo">1490</span>        private StringBuffer proteinDesc;<a name="line.1490"></a>
<span class="sourceLineNo">1491</span>        private boolean firstNameInProteinGroup;<a name="line.1491"></a>
<span class="sourceLineNo">1492</span>        private boolean firstDomainInProteinGroup;<a name="line.1492"></a>
<span class="sourceLineNo">1493</span>        private boolean firstComponentInProteinGroup;<a name="line.1493"></a>
<span class="sourceLineNo">1494</span>        private int currGene;<a name="line.1494"></a>
<span class="sourceLineNo">1495</span>        private String geneNameClass;<a name="line.1495"></a>
<span class="sourceLineNo">1496</span>        private String organismNameClass;<a name="line.1496"></a>
<span class="sourceLineNo">1497</span>        private Map currNames = new TreeMap();<a name="line.1497"></a>
<span class="sourceLineNo">1498</span>        private StringBuffer organelleDesc;<a name="line.1498"></a>
<span class="sourceLineNo">1499</span>        private List currDBXrefs = new ArrayList();<a name="line.1499"></a>
<span class="sourceLineNo">1500</span>        private List currComments = new ArrayList();<a name="line.1500"></a>
<span class="sourceLineNo">1501</span>        private String currRefLocation;<a name="line.1501"></a>
<span class="sourceLineNo">1502</span>        private List currRefAuthors;<a name="line.1502"></a>
<span class="sourceLineNo">1503</span>        private String currRefTitle;<a name="line.1503"></a>
<span class="sourceLineNo">1504</span>        private int currRefStart;<a name="line.1504"></a>
<span class="sourceLineNo">1505</span>        private int currRefEnd;<a name="line.1505"></a>
<span class="sourceLineNo">1506</span>        private int currRefRank;<a name="line.1506"></a>
<span class="sourceLineNo">1507</span>        private String currPersonIs;<a name="line.1507"></a>
<span class="sourceLineNo">1508</span>        private int currRCID;<a name="line.1508"></a>
<span class="sourceLineNo">1509</span>        private int currEvID;<a name="line.1509"></a>
<span class="sourceLineNo">1510</span>        private String currKWID;<a name="line.1510"></a>
<span class="sourceLineNo">1511</span>        private UniProtCommentParser currUCParser;<a name="line.1511"></a>
<span class="sourceLineNo">1512</span>        private Interaction currUCParserInteract;<a name="line.1512"></a>
<span class="sourceLineNo">1513</span>        private Event currUCParserEvent;<a name="line.1513"></a>
<span class="sourceLineNo">1514</span>        private Isoform currUCParserIsoform;<a name="line.1514"></a>
<span class="sourceLineNo">1515</span>        private String currLocIsFor;<a name="line.1515"></a>
<span class="sourceLineNo">1516</span>        private String currTextIsFor;<a name="line.1516"></a>
<span class="sourceLineNo">1517</span>        private String currNoteIsFor;<a name="line.1517"></a>
<span class="sourceLineNo">1518</span>        private String currSeqIsFor;<a name="line.1518"></a>
<span class="sourceLineNo">1519</span>        private String currIDIsFor;<a name="line.1519"></a>
<span class="sourceLineNo">1520</span>        private String currNameIsFor;<a name="line.1520"></a>
<span class="sourceLineNo">1521</span>        private int interactantCount;<a name="line.1521"></a>
<span class="sourceLineNo">1522</span>        private StringBuffer currLocStr;<a name="line.1522"></a>
<span class="sourceLineNo">1523</span>        private int featNoteRank;<a name="line.1523"></a>
<span class="sourceLineNo">1524</span>        <a name="line.1524"></a>
<span class="sourceLineNo">1525</span>        // construct a new handler that will populate the given list of sequences<a name="line.1525"></a>
<span class="sourceLineNo">1526</span>        private UniProtXMLHandler(RichSequenceFormat parent,<a name="line.1526"></a>
<span class="sourceLineNo">1527</span>                SymbolTokenization symParser,<a name="line.1527"></a>
<span class="sourceLineNo">1528</span>                RichSeqIOListener rlistener,<a name="line.1528"></a>
<span class="sourceLineNo">1529</span>                Namespace ns) {<a name="line.1529"></a>
<span class="sourceLineNo">1530</span>            this.parent = parent;<a name="line.1530"></a>
<span class="sourceLineNo">1531</span>            this.symParser = symParser;<a name="line.1531"></a>
<span class="sourceLineNo">1532</span>            this.rlistener = rlistener;<a name="line.1532"></a>
<span class="sourceLineNo">1533</span>            this.ns = ns;<a name="line.1533"></a>
<span class="sourceLineNo">1534</span>            this.m_currentString = new StringBuffer();<a name="line.1534"></a>
<span class="sourceLineNo">1535</span>        }<a name="line.1535"></a>
<span class="sourceLineNo">1536</span>        <a name="line.1536"></a>
<span class="sourceLineNo">1537</span>        <a name="line.1537"></a>
<span class="sourceLineNo">1538</span>        // process an opening tag<a name="line.1538"></a>
<span class="sourceLineNo">1539</span>        @Override<a name="line.1539"></a>
<span class="sourceLineNo">1540</span>        public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {<a name="line.1540"></a>
<span class="sourceLineNo">1541</span>            <a name="line.1541"></a>
<span class="sourceLineNo">1542</span>            if (qName.equals(ENTRY_TAG)) {<a name="line.1542"></a>
<span class="sourceLineNo">1543</span>                try {<a name="line.1543"></a>
<span class="sourceLineNo">1544</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1544"></a>
<span class="sourceLineNo">1545</span>                        String name = attributes.getQName(i);<a name="line.1545"></a>
<span class="sourceLineNo">1546</span>                        String val = attributes.getValue(i);<a name="line.1546"></a>
<span class="sourceLineNo">1547</span>                        if (name.equals(ENTRY_NAMESPACE_ATTR) &amp;&amp; this.ns==null) ns=(Namespace)RichObjectFactory.getObject(SimpleNamespace.class,new Object[]{val});<a name="line.1547"></a>
<span class="sourceLineNo">1548</span>                        else if (name.equals(ENTRY_VERSION_ATTR)) rlistener.addSequenceProperty(Terms.getRelAnnotatedTerm(), val);<a name="line.1548"></a>
<span class="sourceLineNo">1549</span>                        else if (name.equals(ENTRY_CREATED_ATTR)) rlistener.addSequenceProperty(Terms.getDateCreatedTerm(), val);<a name="line.1549"></a>
<span class="sourceLineNo">1550</span>                        else if (name.equals(ENTRY_UPDATED_ATTR)) rlistener.addSequenceProperty(Terms.getDateAnnotatedTerm(), val);<a name="line.1550"></a>
<span class="sourceLineNo">1551</span>                    }<a name="line.1551"></a>
<span class="sourceLineNo">1552</span>                    if (this.ns==null) ns=RichObjectFactory.getDefaultNamespace();<a name="line.1552"></a>
<span class="sourceLineNo">1553</span>                    rlistener.setNamespace(ns);<a name="line.1553"></a>
<span class="sourceLineNo">1554</span>                } catch (ParseException e) {<a name="line.1554"></a>
<span class="sourceLineNo">1555</span>                    throw new SAXException(e);<a name="line.1555"></a>
<span class="sourceLineNo">1556</span>                }<a name="line.1556"></a>
<span class="sourceLineNo">1557</span>                this.currNameIsFor = "ENTRY";<a name="line.1557"></a>
<span class="sourceLineNo">1558</span>                this.currSeqIsFor = "ENTRY";<a name="line.1558"></a>
<span class="sourceLineNo">1559</span>                this.currGene = 0;<a name="line.1559"></a>
<span class="sourceLineNo">1560</span>                this.currNames.clear();<a name="line.1560"></a>
<span class="sourceLineNo">1561</span>                this.currRefRank = 0;<a name="line.1561"></a>
<span class="sourceLineNo">1562</span>                this.currRCID = 0;<a name="line.1562"></a>
<span class="sourceLineNo">1563</span>                this.currEvID = 0;<a name="line.1563"></a>
<span class="sourceLineNo">1564</span>            }<a name="line.1564"></a>
<span class="sourceLineNo">1565</span>            <a name="line.1565"></a>
<span class="sourceLineNo">1566</span>            else if (qName.equals(PROTEIN_TAG)) {<a name="line.1566"></a>
<span class="sourceLineNo">1567</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1567"></a>
<span class="sourceLineNo">1568</span>                    String name = attributes.getQName(i).trim();<a name="line.1568"></a>
<span class="sourceLineNo">1569</span>                    String val = attributes.getValue(i).trim();<a name="line.1569"></a>
<span class="sourceLineNo">1570</span>                    try {<a name="line.1570"></a>
<span class="sourceLineNo">1571</span>                        if (name.equals(TYPE_ATTR)) rlistener.addSequenceProperty(Terms.getProteinTypeTerm(),val);<a name="line.1571"></a>
<span class="sourceLineNo">1572</span>                    } catch (ParseException e) {<a name="line.1572"></a>
<span class="sourceLineNo">1573</span>                        throw new SAXException(e);<a name="line.1573"></a>
<span class="sourceLineNo">1574</span>                    }<a name="line.1574"></a>
<span class="sourceLineNo">1575</span>                }<a name="line.1575"></a>
<span class="sourceLineNo">1576</span>                this.proteinDesc = new StringBuffer();<a name="line.1576"></a>
<span class="sourceLineNo">1577</span>                this.currNameIsFor = "PROTEIN";<a name="line.1577"></a>
<span class="sourceLineNo">1578</span>                this.firstNameInProteinGroup = true;<a name="line.1578"></a>
<span class="sourceLineNo">1579</span>                this.firstDomainInProteinGroup = true;<a name="line.1579"></a>
<span class="sourceLineNo">1580</span>                this.firstComponentInProteinGroup = true;<a name="line.1580"></a>
<span class="sourceLineNo">1581</span>            } else if (qName.equals(DOMAIN_TAG)) {<a name="line.1581"></a>
<span class="sourceLineNo">1582</span>                if (!this.firstComponentInProteinGroup) proteinDesc.append("]");<a name="line.1582"></a>
<span class="sourceLineNo">1583</span>                if (this.firstDomainInProteinGroup) proteinDesc.append(" ["+Terms.CONTAINS_PREFIX);<a name="line.1583"></a>
<span class="sourceLineNo">1584</span>                else proteinDesc.append(";");<a name="line.1584"></a>
<span class="sourceLineNo">1585</span>                this.firstDomainInProteinGroup = false;<a name="line.1585"></a>
<span class="sourceLineNo">1586</span>                this.firstNameInProteinGroup = true;<a name="line.1586"></a>
<span class="sourceLineNo">1587</span>            } else if (qName.equals(COMPONENT_TAG)) {<a name="line.1587"></a>
<span class="sourceLineNo">1588</span>                if (!this.firstDomainInProteinGroup) proteinDesc.append("]");<a name="line.1588"></a>
<span class="sourceLineNo">1589</span>                if (this.firstComponentInProteinGroup) proteinDesc.append(" ["+Terms.INCLUDES_PREFIX);<a name="line.1589"></a>
<span class="sourceLineNo">1590</span>                else proteinDesc.append(";");<a name="line.1590"></a>
<span class="sourceLineNo">1591</span>                this.firstComponentInProteinGroup = false;<a name="line.1591"></a>
<span class="sourceLineNo">1592</span>                this.firstNameInProteinGroup = true;<a name="line.1592"></a>
<span class="sourceLineNo">1593</span>            }<a name="line.1593"></a>
<span class="sourceLineNo">1594</span>            <a name="line.1594"></a>
<span class="sourceLineNo">1595</span>            else if (qName.equals(GENE_TAG)) {<a name="line.1595"></a>
<span class="sourceLineNo">1596</span>                this.currGene++;<a name="line.1596"></a>
<span class="sourceLineNo">1597</span>                this.currNameIsFor="GENE";<a name="line.1597"></a>
<span class="sourceLineNo">1598</span>            }<a name="line.1598"></a>
<span class="sourceLineNo">1599</span>            <a name="line.1599"></a>
<span class="sourceLineNo">1600</span>            else if (qName.equals(NAME_TAG)) {<a name="line.1600"></a>
<span class="sourceLineNo">1601</span>                if (this.currNameIsFor.equals("GENE")) {<a name="line.1601"></a>
<span class="sourceLineNo">1602</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1602"></a>
<span class="sourceLineNo">1603</span>                        String name = attributes.getQName(i);<a name="line.1603"></a>
<span class="sourceLineNo">1604</span>                        String val = attributes.getValue(i);<a name="line.1604"></a>
<span class="sourceLineNo">1605</span>                        if (name.equals(TYPE_ATTR)) this.geneNameClass=val;<a name="line.1605"></a>
<span class="sourceLineNo">1606</span>                    }<a name="line.1606"></a>
<span class="sourceLineNo">1607</span>                }<a name="line.1607"></a>
<span class="sourceLineNo">1608</span>                <a name="line.1608"></a>
<span class="sourceLineNo">1609</span>                else if (this.currNameIsFor.equals("ORGANISM")) {<a name="line.1609"></a>
<span class="sourceLineNo">1610</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1610"></a>
<span class="sourceLineNo">1611</span>                        String name = attributes.getQName(i);<a name="line.1611"></a>
<span class="sourceLineNo">1612</span>                        String val = attributes.getValue(i);<a name="line.1612"></a>
<span class="sourceLineNo">1613</span>                        if (name.equals(TYPE_ATTR)) this.organismNameClass=val;<a name="line.1613"></a>
<span class="sourceLineNo">1614</span>                    }<a name="line.1614"></a>
<span class="sourceLineNo">1615</span>                }<a name="line.1615"></a>
<span class="sourceLineNo">1616</span>            }<a name="line.1616"></a>
<span class="sourceLineNo">1617</span>            <a name="line.1617"></a>
<span class="sourceLineNo">1618</span>            else if (qName.equals(ORGANISM_TAG)) {<a name="line.1618"></a>
<span class="sourceLineNo">1619</span>                this.currNameIsFor="ORGANISM";<a name="line.1619"></a>
<span class="sourceLineNo">1620</span>            }<a name="line.1620"></a>
<span class="sourceLineNo">1621</span>            <a name="line.1621"></a>
<span class="sourceLineNo">1622</span>            else if (qName.equals(DBXREF_TAG)) {<a name="line.1622"></a>
<span class="sourceLineNo">1623</span>                if (this.currNameIsFor.equals("ORGANISM")) {<a name="line.1623"></a>
<span class="sourceLineNo">1624</span>                    Integer taxID = null;<a name="line.1624"></a>
<span class="sourceLineNo">1625</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1625"></a>
<span class="sourceLineNo">1626</span>                        String name = attributes.getQName(i);<a name="line.1626"></a>
<span class="sourceLineNo">1627</span>                        String val = attributes.getValue(i);<a name="line.1627"></a>
<span class="sourceLineNo">1628</span>                        if (name.equals(ID_ATTR)) taxID = Integer.valueOf(val);<a name="line.1628"></a>
<span class="sourceLineNo">1629</span>                    }<a name="line.1629"></a>
<span class="sourceLineNo">1630</span>                    try {<a name="line.1630"></a>
<span class="sourceLineNo">1631</span>                        tax = (NCBITaxon)RichObjectFactory.getObject(SimpleNCBITaxon.class, new Object[]{taxID});<a name="line.1631"></a>
<span class="sourceLineNo">1632</span>                        rlistener.setTaxon(tax);<a name="line.1632"></a>
<span class="sourceLineNo">1633</span>                        for (Iterator j = currNames.keySet().iterator(); j.hasNext(); ) {<a name="line.1633"></a>
<span class="sourceLineNo">1634</span>                            String nameClass = (String)j.next();<a name="line.1634"></a>
<span class="sourceLineNo">1635</span>                            Set nameSet = (Set)this.currNames.get(nameClass);<a name="line.1635"></a>
<span class="sourceLineNo">1636</span>                            try {<a name="line.1636"></a>
<span class="sourceLineNo">1637</span>                                for (Iterator k = nameSet.iterator(); k.hasNext(); ) {<a name="line.1637"></a>
<span class="sourceLineNo">1638</span>                                    String name = (String)k.next();<a name="line.1638"></a>
<span class="sourceLineNo">1639</span>                                    tax.addName(nameClass,name);<a name="line.1639"></a>
<span class="sourceLineNo">1640</span>                                }<a name="line.1640"></a>
<span class="sourceLineNo">1641</span>                            } catch (ChangeVetoException ce) {<a name="line.1641"></a>
<span class="sourceLineNo">1642</span>                                throw new ParseException(ce);<a name="line.1642"></a>
<span class="sourceLineNo">1643</span>                            }<a name="line.1643"></a>
<span class="sourceLineNo">1644</span>                        }<a name="line.1644"></a>
<span class="sourceLineNo">1645</span>                    } catch (ParseException e) {<a name="line.1645"></a>
<span class="sourceLineNo">1646</span>                        throw new SAXException(e);<a name="line.1646"></a>
<span class="sourceLineNo">1647</span>                    }<a name="line.1647"></a>
<span class="sourceLineNo">1648</span>                    this.currNames.clear();<a name="line.1648"></a>
<span class="sourceLineNo">1649</span>                }<a name="line.1649"></a>
<span class="sourceLineNo">1650</span>                <a name="line.1650"></a>
<span class="sourceLineNo">1651</span>                else {<a name="line.1651"></a>
<span class="sourceLineNo">1652</span>                    String type = null;<a name="line.1652"></a>
<span class="sourceLineNo">1653</span>                    String id = null;<a name="line.1653"></a>
<span class="sourceLineNo">1654</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1654"></a>
<span class="sourceLineNo">1655</span>                        String name = attributes.getQName(i);<a name="line.1655"></a>
<span class="sourceLineNo">1656</span>                        String val = attributes.getValue(i);<a name="line.1656"></a>
<span class="sourceLineNo">1657</span>                        if (name.equals(ID_ATTR)) id = val;<a name="line.1657"></a>
<span class="sourceLineNo">1658</span>                        else if (name.equals(TYPE_ATTR)) type = val;<a name="line.1658"></a>
<span class="sourceLineNo">1659</span>                    }<a name="line.1659"></a>
<span class="sourceLineNo">1660</span>                    CrossRef dbx = (CrossRef)RichObjectFactory.getObject(SimpleCrossRef.class,new Object[]{type, id, new Integer(0)});<a name="line.1660"></a>
<span class="sourceLineNo">1661</span>                    this.currDBXrefs.add(dbx);<a name="line.1661"></a>
<span class="sourceLineNo">1662</span>                }<a name="line.1662"></a>
<span class="sourceLineNo">1663</span>            } else if (qName.equals(PROPERTY_TAG)) {<a name="line.1663"></a>
<span class="sourceLineNo">1664</span>                String id = null;<a name="line.1664"></a>
<span class="sourceLineNo">1665</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1665"></a>
<span class="sourceLineNo">1666</span>                    String name = attributes.getQName(i);<a name="line.1666"></a>
<span class="sourceLineNo">1667</span>                    String val = attributes.getValue(i);<a name="line.1667"></a>
<span class="sourceLineNo">1668</span>                    if (name.equals(VALUE_ATTR)) id = val;<a name="line.1668"></a>
<span class="sourceLineNo">1669</span>                }<a name="line.1669"></a>
<span class="sourceLineNo">1670</span>                Note note = new SimpleNote(Terms.getAdditionalAccessionTerm(),id,1);<a name="line.1670"></a>
<span class="sourceLineNo">1671</span>                try {<a name="line.1671"></a>
<span class="sourceLineNo">1672</span>                    int last = this.currDBXrefs.size();<a name="line.1672"></a>
<span class="sourceLineNo">1673</span>                    ((CrossRef)this.currDBXrefs.get(last-1)).getRichAnnotation().addNote(note);<a name="line.1673"></a>
<span class="sourceLineNo">1674</span>                } catch (ChangeVetoException ce) {<a name="line.1674"></a>
<span class="sourceLineNo">1675</span>                    SAXException pe = new SAXException("Could not annotate identifier terms");<a name="line.1675"></a>
<span class="sourceLineNo">1676</span>                    pe.initCause(ce);<a name="line.1676"></a>
<span class="sourceLineNo">1677</span>                    throw pe;<a name="line.1677"></a>
<span class="sourceLineNo">1678</span>                }<a name="line.1678"></a>
<span class="sourceLineNo">1679</span>            }<a name="line.1679"></a>
<span class="sourceLineNo">1680</span>            <a name="line.1680"></a>
<span class="sourceLineNo">1681</span>            else if (qName.equals(GENELOCATION_TAG)) {<a name="line.1681"></a>
<span class="sourceLineNo">1682</span>                this.currNameIsFor = "ORGANELLE";<a name="line.1682"></a>
<span class="sourceLineNo">1683</span>                this.organelleDesc = new StringBuffer();<a name="line.1683"></a>
<span class="sourceLineNo">1684</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1684"></a>
<span class="sourceLineNo">1685</span>                    String name = attributes.getQName(i);<a name="line.1685"></a>
<span class="sourceLineNo">1686</span>                    String val = attributes.getValue(i);<a name="line.1686"></a>
<span class="sourceLineNo">1687</span>                    if (name.equals(TYPE_ATTR)) {<a name="line.1687"></a>
<span class="sourceLineNo">1688</span>                        val = val.toUpperCase().charAt(0)+val.substring(1); // init caps for flat format<a name="line.1688"></a>
<span class="sourceLineNo">1689</span>                        if (!val.equals("Plasmid")) this.organelleDesc.append(val);<a name="line.1689"></a>
<span class="sourceLineNo">1690</span>                    }<a name="line.1690"></a>
<span class="sourceLineNo">1691</span>                }<a name="line.1691"></a>
<span class="sourceLineNo">1692</span>            }<a name="line.1692"></a>
<span class="sourceLineNo">1693</span>            <a name="line.1693"></a>
<span class="sourceLineNo">1694</span>            else if (qName.equals(REFERENCE_TAG) &amp;&amp; !this.parent.getElideReferences()) {<a name="line.1694"></a>
<span class="sourceLineNo">1695</span>                this.currRefLocation = null;<a name="line.1695"></a>
<span class="sourceLineNo">1696</span>                this.currRefAuthors = new ArrayList();<a name="line.1696"></a>
<span class="sourceLineNo">1697</span>                this.currRefTitle = null;<a name="line.1697"></a>
<span class="sourceLineNo">1698</span>                this.currDBXrefs.clear();<a name="line.1698"></a>
<span class="sourceLineNo">1699</span>                this.currComments.clear();<a name="line.1699"></a>
<span class="sourceLineNo">1700</span>                this.currRefRank++;<a name="line.1700"></a>
<span class="sourceLineNo">1701</span>                this.currRefStart = -999;<a name="line.1701"></a>
<span class="sourceLineNo">1702</span>                this.currRefEnd = -999;<a name="line.1702"></a>
<span class="sourceLineNo">1703</span>            } else if (qName.equals(CITATION_TAG) &amp;&amp; !this.parent.getElideReferences()) {<a name="line.1703"></a>
<span class="sourceLineNo">1704</span>                StringBuffer currRef = new StringBuffer();<a name="line.1704"></a>
<span class="sourceLineNo">1705</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1705"></a>
<span class="sourceLineNo">1706</span>                    String name = attributes.getQName(i);<a name="line.1706"></a>
<span class="sourceLineNo">1707</span>                    String val = attributes.getValue(i);<a name="line.1707"></a>
<span class="sourceLineNo">1708</span>                    // combine everything except type into a fake reference to use if locator is a no-show<a name="line.1708"></a>
<span class="sourceLineNo">1709</span>                    if (!name.equals(TYPE_ATTR)) {<a name="line.1709"></a>
<span class="sourceLineNo">1710</span>                        if (currRef.length()&gt;0) currRef.append(" ");<a name="line.1710"></a>
<span class="sourceLineNo">1711</span>                        currRef.append(val);<a name="line.1711"></a>
<span class="sourceLineNo">1712</span>                    }<a name="line.1712"></a>
<span class="sourceLineNo">1713</span>                }<a name="line.1713"></a>
<span class="sourceLineNo">1714</span>                this.currRefLocation = currRef.toString();<a name="line.1714"></a>
<span class="sourceLineNo">1715</span>            } else if (qName.equals(EDITOR_LIST_TAG)) {<a name="line.1715"></a>
<span class="sourceLineNo">1716</span>                this.currPersonIs = "EDITOR";<a name="line.1716"></a>
<span class="sourceLineNo">1717</span>            } else if (qName.equals(AUTHOR_LIST_TAG)) {<a name="line.1717"></a>
<span class="sourceLineNo">1718</span>                this.currPersonIs = "AUTHOR";<a name="line.1718"></a>
<span class="sourceLineNo">1719</span>            }  else if (qName.equals(PERSON_TAG)) {<a name="line.1719"></a>
<span class="sourceLineNo">1720</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1720"></a>
<span class="sourceLineNo">1721</span>                    String name = attributes.getQName(i);<a name="line.1721"></a>
<span class="sourceLineNo">1722</span>                    String val = attributes.getValue(i);<a name="line.1722"></a>
<span class="sourceLineNo">1723</span>                    if (name.equals(NAME_ATTR)) {<a name="line.1723"></a>
<span class="sourceLineNo">1724</span>                        if (this.currPersonIs.equals("AUTHOR")) currRefAuthors.add(new SimpleDocRefAuthor(val, false, false));<a name="line.1724"></a>
<span class="sourceLineNo">1725</span>                        else if (this.currPersonIs.equals("EDITOR")) currRefAuthors.add(new SimpleDocRefAuthor(val, false, true));<a name="line.1725"></a>
<span class="sourceLineNo">1726</span>                    }<a name="line.1726"></a>
<span class="sourceLineNo">1727</span>                }<a name="line.1727"></a>
<span class="sourceLineNo">1728</span>            } else if (qName.equals(CONSORTIUM_TAG)) {<a name="line.1728"></a>
<span class="sourceLineNo">1729</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1729"></a>
<span class="sourceLineNo">1730</span>                    String name = attributes.getQName(i);<a name="line.1730"></a>
<span class="sourceLineNo">1731</span>                    String val = attributes.getValue(i);<a name="line.1731"></a>
<span class="sourceLineNo">1732</span>                    if (name.equals(NAME_ATTR)) {<a name="line.1732"></a>
<span class="sourceLineNo">1733</span>                        if (this.currPersonIs.equals("AUTHOR")) currRefAuthors.add(new SimpleDocRefAuthor(val, true, false));<a name="line.1733"></a>
<span class="sourceLineNo">1734</span>                        else if (this.currPersonIs.equals("EDITOR")) currRefAuthors.add(new SimpleDocRefAuthor(val, true, true));<a name="line.1734"></a>
<span class="sourceLineNo">1735</span>                    }<a name="line.1735"></a>
<span class="sourceLineNo">1736</span>                }<a name="line.1736"></a>
<span class="sourceLineNo">1737</span>            }  else if (qName.equals(RC_LINE_TAG)) {<a name="line.1737"></a>
<span class="sourceLineNo">1738</span>                this.currRCID++;<a name="line.1738"></a>
<span class="sourceLineNo">1739</span>            }<a name="line.1739"></a>
<span class="sourceLineNo">1740</span>            <a name="line.1740"></a>
<span class="sourceLineNo">1741</span>            else if (qName.equals(PROTEIN_EXISTS_TAG)) {<a name="line.1741"></a>
<span class="sourceLineNo">1742</span>                try {<a name="line.1742"></a>
<span class="sourceLineNo">1743</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1743"></a>
<span class="sourceLineNo">1744</span>                        String name = attributes.getQName(i);<a name="line.1744"></a>
<span class="sourceLineNo">1745</span>                        String val = attributes.getValue(i);<a name="line.1745"></a>
<span class="sourceLineNo">1746</span>                        if (name.equals(TYPE_ATTR)) rlistener.addSequenceProperty(Terms.getProteinExistsTerm(),val);<a name="line.1746"></a>
<span class="sourceLineNo">1747</span>                    }<a name="line.1747"></a>
<span class="sourceLineNo">1748</span>                } catch (ParseException e) {<a name="line.1748"></a>
<span class="sourceLineNo">1749</span>                    SAXException pe = new SAXException("Could not annotate protein exists terms");<a name="line.1749"></a>
<span class="sourceLineNo">1750</span>                    pe.initCause(e);<a name="line.1750"></a>
<span class="sourceLineNo">1751</span>                    throw pe;<a name="line.1751"></a>
<span class="sourceLineNo">1752</span>                }<a name="line.1752"></a>
<span class="sourceLineNo">1753</span>            }<a name="line.1753"></a>
<span class="sourceLineNo">1754</span>            <a name="line.1754"></a>
<span class="sourceLineNo">1755</span>            else if (qName.equals(KEYWORD_TAG)) {<a name="line.1755"></a>
<span class="sourceLineNo">1756</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1756"></a>
<span class="sourceLineNo">1757</span>                    String name = attributes.getQName(i);<a name="line.1757"></a>
<span class="sourceLineNo">1758</span>                    String val = attributes.getValue(i);<a name="line.1758"></a>
<span class="sourceLineNo">1759</span>                    if (name.equals(ID_ATTR)) this.currKWID = val;<a name="line.1759"></a>
<span class="sourceLineNo">1760</span>                }<a name="line.1760"></a>
<span class="sourceLineNo">1761</span>            }<a name="line.1761"></a>
<span class="sourceLineNo">1762</span>            <a name="line.1762"></a>
<span class="sourceLineNo">1763</span>            else if (qName.equals(EVIDENCE_TAG)) {<a name="line.1763"></a>
<span class="sourceLineNo">1764</span>                this.currEvID++;<a name="line.1764"></a>
<span class="sourceLineNo">1765</span>                try {<a name="line.1765"></a>
<span class="sourceLineNo">1766</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1766"></a>
<span class="sourceLineNo">1767</span>                        String name = attributes.getQName(i);<a name="line.1767"></a>
<span class="sourceLineNo">1768</span>                        String val = attributes.getValue(i);<a name="line.1768"></a>
<span class="sourceLineNo">1769</span>                        if (name.equals(EVIDENCE_CATEGORY_ATTR)) rlistener.addSequenceProperty(Terms.getEvidenceCategoryTerm(),val);<a name="line.1769"></a>
<span class="sourceLineNo">1770</span>                        else if (name.equals(EVIDENCE_DATE_ATTR)) rlistener.addSequenceProperty(Terms.getEvidenceDateTerm(),val);<a name="line.1770"></a>
<span class="sourceLineNo">1771</span>                        else if (name.equals(TYPE_ATTR)) rlistener.addSequenceProperty(Terms.getEvidenceTypeTerm(),val);<a name="line.1771"></a>
<span class="sourceLineNo">1772</span>                        else if (name.equals(EVIDENCE_ATTRIBUTE_ATTR)) rlistener.addSequenceProperty(Terms.getEvidenceAttrTerm(),val);<a name="line.1772"></a>
<span class="sourceLineNo">1773</span>                    }<a name="line.1773"></a>
<span class="sourceLineNo">1774</span>                } catch (ParseException e) {<a name="line.1774"></a>
<span class="sourceLineNo">1775</span>                    SAXException pe = new SAXException("Could not annotate evidence terms");<a name="line.1775"></a>
<span class="sourceLineNo">1776</span>                    pe.initCause(e);<a name="line.1776"></a>
<span class="sourceLineNo">1777</span>                    throw pe;<a name="line.1777"></a>
<span class="sourceLineNo">1778</span>                }<a name="line.1778"></a>
<span class="sourceLineNo">1779</span>            }<a name="line.1779"></a>
<span class="sourceLineNo">1780</span>            <a name="line.1780"></a>
<span class="sourceLineNo">1781</span>            else if (qName.equals(LOCATION_TAG)) {<a name="line.1781"></a>
<span class="sourceLineNo">1782</span>                this.currLocStr = new StringBuffer();<a name="line.1782"></a>
<span class="sourceLineNo">1783</span>                if (this.currLocIsFor.equals("FEATURE")) {<a name="line.1783"></a>
<span class="sourceLineNo">1784</span>                    try {<a name="line.1784"></a>
<span class="sourceLineNo">1785</span>                        for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1785"></a>
<span class="sourceLineNo">1786</span>                            String name = attributes.getQName(i);<a name="line.1786"></a>
<span class="sourceLineNo">1787</span>                            String val = attributes.getValue(i);<a name="line.1787"></a>
<span class="sourceLineNo">1788</span>                            if (name.equals(LOCATION_SEQ_ATTR)) {<a name="line.1788"></a>
<span class="sourceLineNo">1789</span>                                Note note = new SimpleNote(Terms.getLocationSequenceTerm(), val, this.featNoteRank++);<a name="line.1789"></a>
<span class="sourceLineNo">1790</span>                                ((RichAnnotation)templ.annotation).addNote(note);<a name="line.1790"></a>
<span class="sourceLineNo">1791</span>                            }<a name="line.1791"></a>
<span class="sourceLineNo">1792</span>                        }<a name="line.1792"></a>
<span class="sourceLineNo">1793</span>                    } catch (ChangeVetoException e) {<a name="line.1793"></a>
<span class="sourceLineNo">1794</span>                        SAXException pe = new SAXException("Could not create location terms");<a name="line.1794"></a>
<span class="sourceLineNo">1795</span>                        pe.initCause(e);<a name="line.1795"></a>
<span class="sourceLineNo">1796</span>                        throw pe;<a name="line.1796"></a>
<span class="sourceLineNo">1797</span>                    }<a name="line.1797"></a>
<span class="sourceLineNo">1798</span>                }<a name="line.1798"></a>
<span class="sourceLineNo">1799</span>            } else if (qName.equals(LOCATION_BEGIN_TAG) || qName.equals(LOCATION_END_TAG) || qName.equals(LOCATION_POSITION_TAG)) {<a name="line.1799"></a>
<span class="sourceLineNo">1800</span>                StringBuffer pos = new StringBuffer();<a name="line.1800"></a>
<span class="sourceLineNo">1801</span>                pos.append(" "); // space between start and end<a name="line.1801"></a>
<span class="sourceLineNo">1802</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1802"></a>
<span class="sourceLineNo">1803</span>                    String name = attributes.getQName(i);<a name="line.1803"></a>
<span class="sourceLineNo">1804</span>                    String val = attributes.getValue(i);<a name="line.1804"></a>
<span class="sourceLineNo">1805</span>                    if (name.equals(STATUS_ATTR)) {<a name="line.1805"></a>
<span class="sourceLineNo">1806</span>                        if (val.equals("less than")) pos.append("&lt;");<a name="line.1806"></a>
<span class="sourceLineNo">1807</span>                        else if (val.equals("greater than")) pos.append("&gt;");<a name="line.1807"></a>
<span class="sourceLineNo">1808</span>                    } else if (name.equals(LOCATION_POSITION_ATTR)) {<a name="line.1808"></a>
<span class="sourceLineNo">1809</span>                        pos.append(val);<a name="line.1809"></a>
<span class="sourceLineNo">1810</span>                    }<a name="line.1810"></a>
<span class="sourceLineNo">1811</span>                }<a name="line.1811"></a>
<span class="sourceLineNo">1812</span>                this.currLocStr.append(pos.toString());<a name="line.1812"></a>
<span class="sourceLineNo">1813</span>                if (qName.equals(LOCATION_POSITION_TAG)) currLocStr.append(pos.toString()); // fake it as begin=end<a name="line.1813"></a>
<span class="sourceLineNo">1814</span>            }<a name="line.1814"></a>
<span class="sourceLineNo">1815</span>            <a name="line.1815"></a>
<span class="sourceLineNo">1816</span>            else if (qName.equals(FEATURE_TAG) &amp;&amp; !this.parent.getElideFeatures()) {<a name="line.1816"></a>
<span class="sourceLineNo">1817</span>                this.featNoteRank = 1;<a name="line.1817"></a>
<span class="sourceLineNo">1818</span>                templ = new RichFeature.Template();<a name="line.1818"></a>
<span class="sourceLineNo">1819</span>                templ.annotation = new SimpleRichAnnotation();<a name="line.1819"></a>
<span class="sourceLineNo">1820</span>                templ.sourceTerm = Terms.getUniProtXMLTerm();<a name="line.1820"></a>
<span class="sourceLineNo">1821</span>                templ.featureRelationshipSet = new TreeSet();<a name="line.1821"></a>
<span class="sourceLineNo">1822</span>                templ.rankedCrossRefs = new TreeSet();<a name="line.1822"></a>
<span class="sourceLineNo">1823</span>                try {<a name="line.1823"></a>
<span class="sourceLineNo">1824</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1824"></a>
<span class="sourceLineNo">1825</span>                        String name = attributes.getQName(i);<a name="line.1825"></a>
<span class="sourceLineNo">1826</span>                        String val = attributes.getValue(i);<a name="line.1826"></a>
<span class="sourceLineNo">1827</span>                        if (name.equals(TYPE_ATTR)) {<a name="line.1827"></a>
<span class="sourceLineNo">1828</span>                            templ.typeTerm = RichObjectFactory.getDefaultOntology().getOrCreateTerm(val);<a name="line.1828"></a>
<span class="sourceLineNo">1829</span>                        } else if (name.equals(ID_ATTR)) {<a name="line.1829"></a>
<span class="sourceLineNo">1830</span>                            Note note = new SimpleNote(Terms.getFTIdTerm(), val, this.featNoteRank++);<a name="line.1830"></a>
<span class="sourceLineNo">1831</span>                            ((RichAnnotation)templ.annotation).addNote(note);<a name="line.1831"></a>
<span class="sourceLineNo">1832</span>                        } else if (name.equals(FEATURE_DESC_ATTR)) {<a name="line.1832"></a>
<span class="sourceLineNo">1833</span>                            Note note = new SimpleNote(Terms.getFeatureDescTerm(), val, this.featNoteRank++);<a name="line.1833"></a>
<span class="sourceLineNo">1834</span>                            ((RichAnnotation)templ.annotation).addNote(note);<a name="line.1834"></a>
<span class="sourceLineNo">1835</span>                        } else if (name.equals(STATUS_ATTR)) {<a name="line.1835"></a>
<span class="sourceLineNo">1836</span>                            Note note = new SimpleNote(Terms.getFeatureStatusTerm(), val, this.featNoteRank++);<a name="line.1836"></a>
<span class="sourceLineNo">1837</span>                            ((RichAnnotation)templ.annotation).addNote(note);<a name="line.1837"></a>
<span class="sourceLineNo">1838</span>                        } else if (name.equals(REF_ATTR)) {<a name="line.1838"></a>
<span class="sourceLineNo">1839</span>                            Note note = new SimpleNote(Terms.getFeatureRefTerm(), val, this.featNoteRank++);<a name="line.1839"></a>
<span class="sourceLineNo">1840</span>                            ((RichAnnotation)templ.annotation).addNote(note);<a name="line.1840"></a>
<span class="sourceLineNo">1841</span>                        }<a name="line.1841"></a>
<span class="sourceLineNo">1842</span>                    }<a name="line.1842"></a>
<span class="sourceLineNo">1843</span>                } catch (ChangeVetoException e) {<a name="line.1843"></a>
<span class="sourceLineNo">1844</span>                    SAXException pe = new SAXException("Could not create location terms");<a name="line.1844"></a>
<span class="sourceLineNo">1845</span>                    pe.initCause(e);<a name="line.1845"></a>
<span class="sourceLineNo">1846</span>                    throw pe;<a name="line.1846"></a>
<span class="sourceLineNo">1847</span>                }<a name="line.1847"></a>
<span class="sourceLineNo">1848</span>                this.currLocStr = new StringBuffer();<a name="line.1848"></a>
<span class="sourceLineNo">1849</span>                this.currLocIsFor = "FEATURE";<a name="line.1849"></a>
<span class="sourceLineNo">1850</span>            }<a name="line.1850"></a>
<span class="sourceLineNo">1851</span>            <a name="line.1851"></a>
<span class="sourceLineNo">1852</span>            else if (qName.equals(COMMENT_TAG)) {<a name="line.1852"></a>
<span class="sourceLineNo">1853</span>                this.currUCParser = new UniProtCommentParser();<a name="line.1853"></a>
<span class="sourceLineNo">1854</span>                this.currUCParser.setInteractions(new ArrayList());<a name="line.1854"></a>
<span class="sourceLineNo">1855</span>                this.currUCParser.setEvents(new ArrayList());<a name="line.1855"></a>
<span class="sourceLineNo">1856</span>                this.currUCParser.setIsoforms(new ArrayList());<a name="line.1856"></a>
<span class="sourceLineNo">1857</span>                this.currUCParser.setKMs(new ArrayList());<a name="line.1857"></a>
<span class="sourceLineNo">1858</span>                this.currUCParser.setVMaxes(new ArrayList());<a name="line.1858"></a>
<span class="sourceLineNo">1859</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1859"></a>
<span class="sourceLineNo">1860</span>                    String name = attributes.getQName(i).trim();<a name="line.1860"></a>
<span class="sourceLineNo">1861</span>                    String val = attributes.getValue(i).trim();<a name="line.1861"></a>
<span class="sourceLineNo">1862</span>                    if (name.equals(TYPE_ATTR)) {<a name="line.1862"></a>
<span class="sourceLineNo">1863</span>                        String type = val.toUpperCase(); // easier to check this way, plus flat uniprot requires it<a name="line.1863"></a>
<span class="sourceLineNo">1864</span>                        if (type.equals("POSTTRANSLATIONAL MODIFICATION")) type="PTM";<a name="line.1864"></a>
<span class="sourceLineNo">1865</span>                        else if (type.equals("ONLINE INFORMATION")) type="DATABASE";<a name="line.1865"></a>
<span class="sourceLineNo">1866</span>                        currUCParser.setCommentType(type);<a name="line.1866"></a>
<span class="sourceLineNo">1867</span>                    } else if (name.equals(COMMENT_MASS_ATTR)) this.currUCParser.setMolecularWeight(Integer.parseInt(val));<a name="line.1867"></a>
<span class="sourceLineNo">1868</span>                    else if (name.equals(COMMENT_ERROR_ATTR)) this.currUCParser.setMolWeightError(Integer.valueOf(val));<a name="line.1868"></a>
<span class="sourceLineNo">1869</span>                    else if (name.equals(COMMENT_METHOD_ATTR)) this.currUCParser.setMolWeightMethod(val);<a name="line.1869"></a>
<span class="sourceLineNo">1870</span>                    else if (name.equals(NAME_ATTR)) this.currUCParser.setDatabaseName(val);<a name="line.1870"></a>
<span class="sourceLineNo">1871</span>                }<a name="line.1871"></a>
<span class="sourceLineNo">1872</span>                this.currLocIsFor="COMMENT";<a name="line.1872"></a>
<span class="sourceLineNo">1873</span>                this.currTextIsFor="COMMENT";<a name="line.1873"></a>
<span class="sourceLineNo">1874</span>                this.currNoteIsFor="COMMENT";<a name="line.1874"></a>
<span class="sourceLineNo">1875</span>                this.interactantCount = 0;<a name="line.1875"></a>
<span class="sourceLineNo">1876</span>            } else if (qName.equals(COMMENT_ABSORPTION_TAG)) {<a name="line.1876"></a>
<span class="sourceLineNo">1877</span>                this.currTextIsFor="ABSORPTION";<a name="line.1877"></a>
<span class="sourceLineNo">1878</span>            } else if (qName.equals(COMMENT_KINETICS_TAG)) {<a name="line.1878"></a>
<span class="sourceLineNo">1879</span>                this.currTextIsFor="KINETICS";<a name="line.1879"></a>
<span class="sourceLineNo">1880</span>            } else if (qName.equals(COMMENT_LINK_TAG)) {<a name="line.1880"></a>
<span class="sourceLineNo">1881</span>                this.currTextIsFor="KINETICS";<a name="line.1881"></a>
<span class="sourceLineNo">1882</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1882"></a>
<span class="sourceLineNo">1883</span>                    String name = attributes.getQName(i);<a name="line.1883"></a>
<span class="sourceLineNo">1884</span>                    String val = attributes.getValue(i);<a name="line.1884"></a>
<span class="sourceLineNo">1885</span>                    if (name.equals(COMMENT_LINK_URI_ATTR)) this.currUCParser.setUri(val);<a name="line.1885"></a>
<span class="sourceLineNo">1886</span>                }<a name="line.1886"></a>
<span class="sourceLineNo">1887</span>            } else if (qName.equals(COMMENT_EVENT_TAG)) {<a name="line.1887"></a>
<span class="sourceLineNo">1888</span>                this.currUCParserEvent = new Event();<a name="line.1888"></a>
<span class="sourceLineNo">1889</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1889"></a>
<span class="sourceLineNo">1890</span>                    String name = attributes.getQName(i);<a name="line.1890"></a>
<span class="sourceLineNo">1891</span>                    String val = attributes.getValue(i);<a name="line.1891"></a>
<span class="sourceLineNo">1892</span>                    if (name.equals(TYPE_ATTR)) {<a name="line.1892"></a>
<span class="sourceLineNo">1893</span>                        val = val.toUpperCase().charAt(0)+val.substring(1); // make first letter upper case for flat uniprot<a name="line.1893"></a>
<span class="sourceLineNo">1894</span>                        this.currUCParserEvent.setType(val);<a name="line.1894"></a>
<span class="sourceLineNo">1895</span>                    }<a name="line.1895"></a>
<span class="sourceLineNo">1896</span>                }<a name="line.1896"></a>
<span class="sourceLineNo">1897</span>                currUCParser.getEvents().add(currUCParserEvent);<a name="line.1897"></a>
<span class="sourceLineNo">1898</span>            } else if (qName.equals(COMMENT_ISOFORM_TAG)) {<a name="line.1898"></a>
<span class="sourceLineNo">1899</span>                this.currUCParserIsoform = new Isoform();<a name="line.1899"></a>
<span class="sourceLineNo">1900</span>                this.currUCParser.getIsoforms().add(currUCParserIsoform);<a name="line.1900"></a>
<span class="sourceLineNo">1901</span>                this.currUCParserEvent.setNamedIsoforms(this.currUCParser.getIsoforms().size());<a name="line.1901"></a>
<span class="sourceLineNo">1902</span>                this.currNameIsFor="ISOFORM";<a name="line.1902"></a>
<span class="sourceLineNo">1903</span>                this.currNoteIsFor="ISOFORM";<a name="line.1903"></a>
<span class="sourceLineNo">1904</span>                this.currSeqIsFor="ISOFORM";<a name="line.1904"></a>
<span class="sourceLineNo">1905</span>                this.currIDIsFor="ISOFORM";<a name="line.1905"></a>
<span class="sourceLineNo">1906</span>            } else if (qName.equals(COMMENT_INTERACTANT_TAG)) {<a name="line.1906"></a>
<span class="sourceLineNo">1907</span>                this.currIDIsFor="INTERACTION";<a name="line.1907"></a>
<span class="sourceLineNo">1908</span>                this.interactantCount++;<a name="line.1908"></a>
<span class="sourceLineNo">1909</span>                for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1909"></a>
<span class="sourceLineNo">1910</span>                    String name = attributes.getQName(i);<a name="line.1910"></a>
<span class="sourceLineNo">1911</span>                    String val = attributes.getValue(i);<a name="line.1911"></a>
<span class="sourceLineNo">1912</span>                    if (name.equals(COMMENT_INTERACT_INTACT_ATTR)) {<a name="line.1912"></a>
<span class="sourceLineNo">1913</span>                        if (this.interactantCount%2==1) {<a name="line.1913"></a>
<span class="sourceLineNo">1914</span>                            this.currUCParserInteract = new Interaction();<a name="line.1914"></a>
<span class="sourceLineNo">1915</span>                            this.currUCParserInteract.setFirstIntActID(val);<a name="line.1915"></a>
<span class="sourceLineNo">1916</span>                            this.currUCParser.getInteractions().add(this.currUCParserInteract);<a name="line.1916"></a>
<span class="sourceLineNo">1917</span>                        }<a name="line.1917"></a>
<span class="sourceLineNo">1918</span>                        else this.currUCParserInteract.setSecondIntActID(val);<a name="line.1918"></a>
<span class="sourceLineNo">1919</span>                    }<a name="line.1919"></a>
<span class="sourceLineNo">1920</span>                }<a name="line.1920"></a>
<span class="sourceLineNo">1921</span>            }<a name="line.1921"></a>
<span class="sourceLineNo">1922</span>            <a name="line.1922"></a>
<span class="sourceLineNo">1923</span>            else if (qName.equals(SEQUENCE_TAG)) {<a name="line.1923"></a>
<span class="sourceLineNo">1924</span>                if (this.currSeqIsFor.equals("ENTRY")) {<a name="line.1924"></a>
<span class="sourceLineNo">1925</span>                    try {<a name="line.1925"></a>
<span class="sourceLineNo">1926</span>                        for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1926"></a>
<span class="sourceLineNo">1927</span>                            String name = attributes.getQName(i);<a name="line.1927"></a>
<span class="sourceLineNo">1928</span>                            String val = attributes.getValue(i);<a name="line.1928"></a>
<span class="sourceLineNo">1929</span>                            if (name.equals(SEQUENCE_MODIFIED_ATTR)) {<a name="line.1929"></a>
<span class="sourceLineNo">1930</span>                                rlistener.addSequenceProperty(Terms.getDateUpdatedTerm(),val);<a name="line.1930"></a>
<span class="sourceLineNo">1931</span>                            }<a name="line.1931"></a>
<span class="sourceLineNo">1932</span>                            else if (name.equals(SEQUENCE_VERSION_ATTR))<a name="line.1932"></a>
<span class="sourceLineNo">1933</span>                                rlistener.setVersion(Integer.parseInt(val));<a name="line.1933"></a>
<span class="sourceLineNo">1934</span>                        }<a name="line.1934"></a>
<span class="sourceLineNo">1935</span>                    } catch (ParseException e) {<a name="line.1935"></a>
<span class="sourceLineNo">1936</span>                        SAXException pe = new SAXException("Could not set sequence properties");<a name="line.1936"></a>
<span class="sourceLineNo">1937</span>                        pe.initCause(e);<a name="line.1937"></a>
<span class="sourceLineNo">1938</span>                        throw pe;<a name="line.1938"></a>
<span class="sourceLineNo">1939</span>                    }<a name="line.1939"></a>
<span class="sourceLineNo">1940</span>                }<a name="line.1940"></a>
<span class="sourceLineNo">1941</span>                <a name="line.1941"></a>
<span class="sourceLineNo">1942</span>                else if (this.currSeqIsFor.equals("ISOFORM")) {<a name="line.1942"></a>
<span class="sourceLineNo">1943</span>                    for (int i = 0; i &lt; attributes.getLength(); i++) {<a name="line.1943"></a>
<span class="sourceLineNo">1944</span>                        String name = attributes.getQName(i);<a name="line.1944"></a>
<span class="sourceLineNo">1945</span>                        String val = attributes.getValue(i);<a name="line.1945"></a>
<span class="sourceLineNo">1946</span>                        if (name.equals(TYPE_ATTR)) {<a name="line.1946"></a>
<span class="sourceLineNo">1947</span>                            val = val.toUpperCase().charAt(0)+val.substring(1); // init caps for flat uniprot<a name="line.1947"></a>
<span class="sourceLineNo">1948</span>                            this.currUCParserIsoform.setSequenceType(val);<a name="line.1948"></a>
<span class="sourceLineNo">1949</span>                        } else if (name.equals(REF_ATTR)) {<a name="line.1949"></a>
<span class="sourceLineNo">1950</span>                            this.currUCParserIsoform.setSequenceRef(val);<a name="line.1950"></a>
<span class="sourceLineNo">1951</span>                        }<a name="line.1951"></a>
<span class="sourceLineNo">1952</span>                    }<a name="line.1952"></a>
<span class="sourceLineNo">1953</span>                }<a name="line.1953"></a>
<span class="sourceLineNo">1954</span>            }<a name="line.1954"></a>
<span class="sourceLineNo">1955</span>        }<a name="line.1955"></a>
<span class="sourceLineNo">1956</span>        <a name="line.1956"></a>
<span class="sourceLineNo">1957</span>        // process a closing tag - we will have read the text already<a name="line.1957"></a>
<span class="sourceLineNo">1958</span>        @Override<a name="line.1958"></a>
<span class="sourceLineNo">1959</span>        public void endElement(String uri, String localName, String qName) throws SAXException {<a name="line.1959"></a>
<span class="sourceLineNo">1960</span>            String val = this.m_currentString.toString().trim();<a name="line.1960"></a>
<span class="sourceLineNo">1961</span>            <a name="line.1961"></a>
<span class="sourceLineNo">1962</span>            try {<a name="line.1962"></a>
<span class="sourceLineNo">1963</span>                if (qName.equals(COPYRIGHT_TAG)) {<a name="line.1963"></a>
<span class="sourceLineNo">1964</span>                    rlistener.addSequenceProperty(Terms.getCopyrightTerm(),val);<a name="line.1964"></a>
<span class="sourceLineNo">1965</span>                }<a name="line.1965"></a>
<span class="sourceLineNo">1966</span>                <a name="line.1966"></a>
<span class="sourceLineNo">1967</span>                else if (qName.equals(ACCESSION_TAG)) {<a name="line.1967"></a>
<span class="sourceLineNo">1968</span>                    rlistener.setAccession(val);<a name="line.1968"></a>
<span class="sourceLineNo">1969</span>                } else if (qName.equals(NAME_TAG)) {<a name="line.1969"></a>
<span class="sourceLineNo">1970</span>                    if (this.currNameIsFor.equals("ENTRY")) rlistener.setName(val);<a name="line.1970"></a>
<span class="sourceLineNo">1971</span>                    <a name="line.1971"></a>
<span class="sourceLineNo">1972</span>                    else if (this.currNameIsFor.equals("PROTEIN")) {<a name="line.1972"></a>
<span class="sourceLineNo">1973</span>                        if (this.firstNameInProteinGroup) {<a name="line.1973"></a>
<span class="sourceLineNo">1974</span>                            proteinDesc.append(" ");<a name="line.1974"></a>
<span class="sourceLineNo">1975</span>                            proteinDesc.append(val);<a name="line.1975"></a>
<span class="sourceLineNo">1976</span>                        } else {<a name="line.1976"></a>
<span class="sourceLineNo">1977</span>                            proteinDesc.append(" (");<a name="line.1977"></a>
<span class="sourceLineNo">1978</span>                            proteinDesc.append(val);<a name="line.1978"></a>
<span class="sourceLineNo">1979</span>                            proteinDesc.append(")");<a name="line.1979"></a>
<span class="sourceLineNo">1980</span>                        }<a name="line.1980"></a>
<span class="sourceLineNo">1981</span>                        this.firstNameInProteinGroup = false;<a name="line.1981"></a>
<span class="sourceLineNo">1982</span>                    }<a name="line.1982"></a>
<span class="sourceLineNo">1983</span>                    <a name="line.1983"></a>
<span class="sourceLineNo">1984</span>                    else if (this.currNameIsFor.equals("GENE")) {<a name="line.1984"></a>
<span class="sourceLineNo">1985</span>                        if (this.geneNameClass.equals(Terms.GENENAME_KEY)) rlistener.addSequenceProperty(Terms.getGeneNameTerm(), this.currGene+":"+val);<a name="line.1985"></a>
<span class="sourceLineNo">1986</span>                        else if (this.geneNameClass.equals(Terms.GENESYNONYM_KEY)) rlistener.addSequenceProperty(Terms.getGeneSynonymTerm(), this.currGene+":"+val);<a name="line.1986"></a>
<span class="sourceLineNo">1987</span>                        else if (this.geneNameClass.equals(Terms.ORDLOCNAME_KEY)) rlistener.addSequenceProperty(Terms.getOrderedLocusNameTerm(), this.currGene+":"+val);<a name="line.1987"></a>
<span class="sourceLineNo">1988</span>                        else if (this.geneNameClass.equals(Terms.ORFNAME_KEY)) rlistener.addSequenceProperty(Terms.getORFNameTerm(), this.currGene+":"+val);<a name="line.1988"></a>
<span class="sourceLineNo">1989</span>                    }<a name="line.1989"></a>
<span class="sourceLineNo">1990</span>                    <a name="line.1990"></a>
<span class="sourceLineNo">1991</span>                    else if (this.currNameIsFor.equals("ORGANISM")) {<a name="line.1991"></a>
<span class="sourceLineNo">1992</span>                        String ournameclass = NCBITaxon.COMMON;<a name="line.1992"></a>
<span class="sourceLineNo">1993</span>                        if (this.organismNameClass.equals(Terms.ABBREV_NAME_KEY)) ournameclass = NCBITaxon.ACRONYM;<a name="line.1993"></a>
<span class="sourceLineNo">1994</span>                        else if (this.organismNameClass.equals(Terms.FULL_NAME_KEY)) ournameclass = NCBITaxon.EQUIVALENT;<a name="line.1994"></a>
<span class="sourceLineNo">1995</span>                        else if (this.organismNameClass.equals(Terms.SCIENTIFIC_NAME_KEY)) ournameclass = NCBITaxon.SCIENTIFIC;<a name="line.1995"></a>
<span class="sourceLineNo">1996</span>                        else if (this.organismNameClass.equals(Terms.SYNONYM_NAME_KEY)) ournameclass = NCBITaxon.SYNONYM;<a name="line.1996"></a>
<span class="sourceLineNo">1997</span>                        if (!this.currNames.containsKey(ournameclass)) this.currNames.put(ournameclass,new TreeSet());<a name="line.1997"></a>
<span class="sourceLineNo">1998</span>                        ((Set)this.currNames.get(ournameclass)).add(val);<a name="line.1998"></a>
<span class="sourceLineNo">1999</span>                    }<a name="line.1999"></a>
<span class="sourceLineNo">2000</span>                    <a name="line.2000"></a>
<span class="sourceLineNo">2001</span>                    else if (this.currNameIsFor.equals("ORGANELLE")) {<a name="line.2001"></a>
<span class="sourceLineNo">2002</span>                        this.organelleDesc.append(", Plasmid ");<a name="line.2002"></a>
<span class="sourceLineNo">2003</span>                        this.organelleDesc.append(val);<a name="line.2003"></a>
<span class="sourceLineNo">2004</span>                    }<a name="line.2004"></a>
<span class="sourceLineNo">2005</span>                    <a name="line.2005"></a>
<span class="sourceLineNo">2006</span>                    else if (this.currNameIsFor.equals("ISOFORM")) {<a name="line.2006"></a>
<span class="sourceLineNo">2007</span>                        this.currUCParserIsoform.getNames().add(val);<a name="line.2007"></a>
<span class="sourceLineNo">2008</span>                    }<a name="line.2008"></a>
<span class="sourceLineNo">2009</span>                }<a name="line.2009"></a>
<span class="sourceLineNo">2010</span>                <a name="line.2010"></a>
<span class="sourceLineNo">2011</span>                else if (qName.equals(PROTEIN_TAG)) {<a name="line.2011"></a>
<span class="sourceLineNo">2012</span>                    if (!this.firstDomainInProteinGroup || !this.firstComponentInProteinGroup) this.proteinDesc.append("]");<a name="line.2012"></a>
<span class="sourceLineNo">2013</span>                    this.proteinDesc.append(".");<a name="line.2013"></a>
<span class="sourceLineNo">2014</span>                    rlistener.setDescription(this.proteinDesc.toString());<a name="line.2014"></a>
<span class="sourceLineNo">2015</span>                }<a name="line.2015"></a>
<span class="sourceLineNo">2016</span>                <a name="line.2016"></a>
<span class="sourceLineNo">2017</span>                else if (qName.equals(ORGANISM_TAG)) {<a name="line.2017"></a>
<span class="sourceLineNo">2018</span>                    this.currNameIsFor="";<a name="line.2018"></a>
<span class="sourceLineNo">2019</span>                }<a name="line.2019"></a>
<span class="sourceLineNo">2020</span>                <a name="line.2020"></a>
<span class="sourceLineNo">2021</span>                else if (qName.equals(GENELOCATION_TAG)) {<a name="line.2021"></a>
<span class="sourceLineNo">2022</span>                    String total = this.organelleDesc.toString().substring(3); // chomp leading ", "<a name="line.2022"></a>
<span class="sourceLineNo">2023</span>                    int lastComma = total.lastIndexOf(',');<a name="line.2023"></a>
<span class="sourceLineNo">2024</span>                    if (lastComma&gt;-1) {<a name="line.2024"></a>
<span class="sourceLineNo">2025</span>                        this.organelleDesc.insert(lastComma+1," and");<a name="line.2025"></a>
<span class="sourceLineNo">2026</span>                        total = this.organelleDesc.toString();<a name="line.2026"></a>
<span class="sourceLineNo">2027</span>                    }<a name="line.2027"></a>
<span class="sourceLineNo">2028</span>                    rlistener.addSequenceProperty(Terms.getOrganelleTerm(), total);<a name="line.2028"></a>
<span class="sourceLineNo">2029</span>                }<a name="line.2029"></a>
<span class="sourceLineNo">2030</span>                <a name="line.2030"></a>
<span class="sourceLineNo">2031</span>                else if (qName.equals(RC_SPECIES_TAG)) {<a name="line.2031"></a>
<span class="sourceLineNo">2032</span>                    rlistener.addSequenceProperty(Terms.getSpeciesTerm(), this.currRCID+":"+val);<a name="line.2032"></a>
<span class="sourceLineNo">2033</span>                } else if (qName.equals(RC_TISSUE_TAG)) {<a name="line.2033"></a>
<span class="sourceLineNo">2034</span>                    rlistener.addSequenceProperty(Terms.getTissueTerm(), this.currRCID+":"+val);<a name="line.2034"></a>
<span class="sourceLineNo">2035</span>                } else if (qName.equals(RC_TRANSP_TAG)) {<a name="line.2035"></a>
<span class="sourceLineNo">2036</span>                    rlistener.addSequenceProperty(Terms.getTransposonTerm(), this.currRCID+":"+val);<a name="line.2036"></a>
<span class="sourceLineNo">2037</span>                } else if (qName.equals(RC_PLASMID_TAG)) {<a name="line.2037"></a>
<span class="sourceLineNo">2038</span>                    rlistener.addSequenceProperty(Terms.getPlasmidTerm(), this.currRCID+":"+val);<a name="line.2038"></a>
<span class="sourceLineNo">2039</span>                }<a name="line.2039"></a>
<span class="sourceLineNo">2040</span>                <a name="line.2040"></a>
<span class="sourceLineNo">2041</span>                else if (qName.equals(TITLE_TAG)) {<a name="line.2041"></a>
<span class="sourceLineNo">2042</span>                    this.currRefTitle = val;<a name="line.2042"></a>
<span class="sourceLineNo">2043</span>                } else if (qName.equals(LOCATOR_TAG)) {<a name="line.2043"></a>
<span class="sourceLineNo">2044</span>                    this.currRefLocation = val;<a name="line.2044"></a>
<span class="sourceLineNo">2045</span>                } else if (qName.equals(RP_LINE_TAG)) {<a name="line.2045"></a>
<span class="sourceLineNo">2046</span>                    this.currComments.add(val);<a name="line.2046"></a>
<span class="sourceLineNo">2047</span>                    // Try to use it to find the location of the reference, if we have one.<a name="line.2047"></a>
<span class="sourceLineNo">2048</span>                    Matcher m = rppat.matcher(val);<a name="line.2048"></a>
<span class="sourceLineNo">2049</span>                    if (m.matches()) {<a name="line.2049"></a>
<span class="sourceLineNo">2050</span>                        this.currRefStart = Integer.parseInt(m.group(1));<a name="line.2050"></a>
<span class="sourceLineNo">2051</span>                        this.currRefEnd = Integer.parseInt(m.group(2));<a name="line.2051"></a>
<span class="sourceLineNo">2052</span>                    }<a name="line.2052"></a>
<span class="sourceLineNo">2053</span>                } else if (qName.equals(REFERENCE_TAG) &amp;&amp; !this.parent.getElideReferences()) {<a name="line.2053"></a>
<span class="sourceLineNo">2054</span>                    // do the crossrefs<a name="line.2054"></a>
<span class="sourceLineNo">2055</span>                    CrossRef useForDocRef = null;<a name="line.2055"></a>
<span class="sourceLineNo">2056</span>                    for (Iterator j = this.currDBXrefs.iterator(); j.hasNext();) {<a name="line.2056"></a>
<span class="sourceLineNo">2057</span>                        CrossRef dbx = (CrossRef)j.next();<a name="line.2057"></a>
<span class="sourceLineNo">2058</span>                        RankedCrossRef rdbx = new SimpleRankedCrossRef(dbx,0);<a name="line.2058"></a>
<span class="sourceLineNo">2059</span>                        rlistener.setRankedCrossRef(rdbx);<a name="line.2059"></a>
<span class="sourceLineNo">2060</span>                        if (useForDocRef==null) useForDocRef = dbx;<a name="line.2060"></a>
<span class="sourceLineNo">2061</span>                        else {<a name="line.2061"></a>
<span class="sourceLineNo">2062</span>                            // medline gets priority, then pubmed - if multiple, use last<a name="line.2062"></a>
<span class="sourceLineNo">2063</span>                            if (dbx.getDbname().equalsIgnoreCase(Terms.MEDLINE_KEY) || <a name="line.2063"></a>
<span class="sourceLineNo">2064</span>                                    (dbx.getDbname().equalsIgnoreCase(Terms.PUBMED_KEY) &amp;&amp; <a name="line.2064"></a>
<span class="sourceLineNo">2065</span>                                    !useForDocRef.getDbname().equalsIgnoreCase(Terms.MEDLINE_KEY))) {<a name="line.2065"></a>
<span class="sourceLineNo">2066</span>                                useForDocRef = dbx;<a name="line.2066"></a>
<span class="sourceLineNo">2067</span>                            }<a name="line.2067"></a>
<span class="sourceLineNo">2068</span>                        }<a name="line.2068"></a>
<span class="sourceLineNo">2069</span>                    }<a name="line.2069"></a>
<span class="sourceLineNo">2070</span>                    // do the comment - can only be one in this object model<a name="line.2070"></a>
<span class="sourceLineNo">2071</span>                    String currRefRemark = null;<a name="line.2071"></a>
<span class="sourceLineNo">2072</span>                    if (currComments.size()&gt;0) currRefRemark = (String)currComments.iterator().next();<a name="line.2072"></a>
<span class="sourceLineNo">2073</span>                    // create the docref object<a name="line.2073"></a>
<span class="sourceLineNo">2074</span>                    try {<a name="line.2074"></a>
<span class="sourceLineNo">2075</span>                        DocRef dr = (DocRef)RichObjectFactory.getObject(SimpleDocRef.class,new Object[]{currRefAuthors,currRefLocation,currRefTitle});<a name="line.2075"></a>
<span class="sourceLineNo">2076</span>                        // assign the pubmed or medline to the docref - medline gets priority<a name="line.2076"></a>
<span class="sourceLineNo">2077</span>                        if (useForDocRef!=null) dr.setCrossref(useForDocRef);<a name="line.2077"></a>
<span class="sourceLineNo">2078</span>                        // assign the remarks<a name="line.2078"></a>
<span class="sourceLineNo">2079</span>                        dr.setRemark(currRefRemark);<a name="line.2079"></a>
<span class="sourceLineNo">2080</span>                        // assign the docref to the bioentry<a name="line.2080"></a>
<span class="sourceLineNo">2081</span>                        RankedDocRef rdr = new SimpleRankedDocRef(dr,<a name="line.2081"></a>
<span class="sourceLineNo">2082</span>                                (currRefStart != -999 ? new Integer(currRefStart) : null),<a name="line.2082"></a>
<span class="sourceLineNo">2083</span>                                (currRefEnd != -999 ? new Integer(currRefEnd) : null),<a name="line.2083"></a>
<span class="sourceLineNo">2084</span>                                currRefRank);<a name="line.2084"></a>
<span class="sourceLineNo">2085</span>                        rlistener.setRankedDocRef(rdr);<a name="line.2085"></a>
<span class="sourceLineNo">2086</span>                    } catch (ChangeVetoException e) {<a name="line.2086"></a>
<span class="sourceLineNo">2087</span>                        throw new ParseException(e);<a name="line.2087"></a>
<span class="sourceLineNo">2088</span>                    }<a name="line.2088"></a>
<span class="sourceLineNo">2089</span>                    currDBXrefs.clear();<a name="line.2089"></a>
<span class="sourceLineNo">2090</span>                    currComments.clear();<a name="line.2090"></a>
<span class="sourceLineNo">2091</span>                }<a name="line.2091"></a>
<span class="sourceLineNo">2092</span>                <a name="line.2092"></a>
<span class="sourceLineNo">2093</span>                // keywords<a name="line.2093"></a>
<span class="sourceLineNo">2094</span>                else if (qName.equals(KEYWORD_TAG)) {<a name="line.2094"></a>
<span class="sourceLineNo">2095</span>                    // create and persist term<a name="line.2095"></a>
<span class="sourceLineNo">2096</span>                    ComparableTerm t = Terms.getUniprotKWOnto().getOrCreateTerm(val);<a name="line.2096"></a>
<span class="sourceLineNo">2097</span>                    try {<a name="line.2097"></a>
<span class="sourceLineNo">2098</span>                        t.setIdentifier(currKWID);<a name="line.2098"></a>
<span class="sourceLineNo">2099</span>                    } catch (ChangeVetoException e) {<a name="line.2099"></a>
<span class="sourceLineNo">2100</span>                        throw new ParseException(e);<a name="line.2100"></a>
<span class="sourceLineNo">2101</span>                    }<a name="line.2101"></a>
<span class="sourceLineNo">2102</span>                    rlistener.addSequenceProperty(Terms.getKeywordTerm(), val);<a name="line.2102"></a>
<span class="sourceLineNo">2103</span>                }<a name="line.2103"></a>
<span class="sourceLineNo">2104</span>                <a name="line.2104"></a>
<span class="sourceLineNo">2105</span>                else if (qName.equals(LOCATION_TAG)) {<a name="line.2105"></a>
<span class="sourceLineNo">2106</span>                    if (currLocIsFor.equals("FEATURE")) {<a name="line.2106"></a>
<span class="sourceLineNo">2107</span>                        templ.location = UniProtLocationParser.parseLocation(currLocStr.toString());<a name="line.2107"></a>
<span class="sourceLineNo">2108</span>                    } else if (currLocIsFor.equals("COMMENT")) {<a name="line.2108"></a>
<span class="sourceLineNo">2109</span>                        Location l = UniProtLocationParser.parseLocation(currLocStr.toString());<a name="line.2109"></a>
<span class="sourceLineNo">2110</span>                        this.currUCParser.setMolWeightRangeStart(l.getMin());<a name="line.2110"></a>
<span class="sourceLineNo">2111</span>                        this.currUCParser.setMolWeightRangeEnd(l.getMax());<a name="line.2111"></a>
<span class="sourceLineNo">2112</span>                    }<a name="line.2112"></a>
<span class="sourceLineNo">2113</span>                }<a name="line.2113"></a>
<span class="sourceLineNo">2114</span>                <a name="line.2114"></a>
<span class="sourceLineNo">2115</span>                else if (qName.equals(FEATURE_TAG)) {<a name="line.2115"></a>
<span class="sourceLineNo">2116</span>                    // start the feature from the template we built<a name="line.2116"></a>
<span class="sourceLineNo">2117</span>                    rlistener.startFeature(templ);<a name="line.2117"></a>
<span class="sourceLineNo">2118</span>                    // end the feature<a name="line.2118"></a>
<span class="sourceLineNo">2119</span>                    rlistener.endFeature();<a name="line.2119"></a>
<span class="sourceLineNo">2120</span>                } else if (qName.equals(FEATURE_ORIGINAL_TAG)) {<a name="line.2120"></a>
<span class="sourceLineNo">2121</span>                    try {<a name="line.2121"></a>
<span class="sourceLineNo">2122</span>                        Note note = new SimpleNote(Terms.getFeatureOriginalTerm(), val, featNoteRank++);<a name="line.2122"></a>
<span class="sourceLineNo">2123</span>                        ((RichAnnotation)templ.annotation).addNote(note);<a name="line.2123"></a>
<span class="sourceLineNo">2124</span>                    } catch (ChangeVetoException e) {<a name="line.2124"></a>
<span class="sourceLineNo">2125</span>                        SAXException pe = new SAXException("Could not create location terms");<a name="line.2125"></a>
<span class="sourceLineNo">2126</span>                        pe.initCause(e);<a name="line.2126"></a>
<span class="sourceLineNo">2127</span>                        throw pe;<a name="line.2127"></a>
<span class="sourceLineNo">2128</span>                    }<a name="line.2128"></a>
<span class="sourceLineNo">2129</span>                } else if (qName.equals(FEATURE_VARIATION_TAG)) {<a name="line.2129"></a>
<span class="sourceLineNo">2130</span>                    try {<a name="line.2130"></a>
<span class="sourceLineNo">2131</span>                        Note note = new SimpleNote(Terms.getFeatureVariationTerm(), val, featNoteRank++);<a name="line.2131"></a>
<span class="sourceLineNo">2132</span>                        ((RichAnnotation)templ.annotation).addNote(note);<a name="line.2132"></a>
<span class="sourceLineNo">2133</span>                    } catch (ChangeVetoException e) {<a name="line.2133"></a>
<span class="sourceLineNo">2134</span>                        SAXException pe = new SAXException("Could not create location terms");<a name="line.2134"></a>
<span class="sourceLineNo">2135</span>                        pe.initCause(e);<a name="line.2135"></a>
<span class="sourceLineNo">2136</span>                        throw pe;<a name="line.2136"></a>
<span class="sourceLineNo">2137</span>                    }<a name="line.2137"></a>
<span class="sourceLineNo">2138</span>                }<a name="line.2138"></a>
<span class="sourceLineNo">2139</span>                <a name="line.2139"></a>
<span class="sourceLineNo">2140</span>                else if (qName.equals(COMMENT_TAG)) {<a name="line.2140"></a>
<span class="sourceLineNo">2141</span>                    rlistener.setComment(currUCParser.generate());<a name="line.2141"></a>
<span class="sourceLineNo">2142</span>                } else if (qName.equals(TEXT_TAG)) {<a name="line.2142"></a>
<span class="sourceLineNo">2143</span>                    if (this.currTextIsFor.equals("COMMENT")) currUCParser.setText(val);<a name="line.2143"></a>
<span class="sourceLineNo">2144</span>                    else if (this.currTextIsFor.equals("ABSORPTION")) currUCParser.setAbsorptionNote(val);<a name="line.2144"></a>
<span class="sourceLineNo">2145</span>                    else if (this.currTextIsFor.equals("KINETICS")) currUCParser.setKineticsNote(val);<a name="line.2145"></a>
<span class="sourceLineNo">2146</span>                } else if (qName.equals(COMMENT_ABS_MAX_TAG)) {<a name="line.2146"></a>
<span class="sourceLineNo">2147</span>                    currUCParser.setAbsorptionMax(val);<a name="line.2147"></a>
<span class="sourceLineNo">2148</span>                } else if (qName.equals(COMMENT_KIN_KM_TAG)) {<a name="line.2148"></a>
<span class="sourceLineNo">2149</span>                    currUCParser.getKMs().add(val);<a name="line.2149"></a>
<span class="sourceLineNo">2150</span>                } else if (qName.equals(COMMENT_KIN_VMAX_TAG)) {<a name="line.2150"></a>
<span class="sourceLineNo">2151</span>                    currUCParser.getVMaxes().add(val);<a name="line.2151"></a>
<span class="sourceLineNo">2152</span>                } else if (qName.equals(COMMENT_PH_TAG)) {<a name="line.2152"></a>
<span class="sourceLineNo">2153</span>                    currUCParser.setPHDependence(val);<a name="line.2153"></a>
<span class="sourceLineNo">2154</span>                } else if (qName.equals(COMMENT_REDOX_TAG)) {<a name="line.2154"></a>
<span class="sourceLineNo">2155</span>                    currUCParser.setRedoxPotential(val);<a name="line.2155"></a>
<span class="sourceLineNo">2156</span>                } else if (qName.equals(COMMENT_TEMPERATURE_TAG)) {<a name="line.2156"></a>
<span class="sourceLineNo">2157</span>                    currUCParser.setTemperatureDependence(val);<a name="line.2157"></a>
<span class="sourceLineNo">2158</span>                } else if (qName.equals(COMMENT_ORGANISMS_TAG)) {<a name="line.2158"></a>
<span class="sourceLineNo">2159</span>                    if (val.equalsIgnoreCase("true")) currUCParserInteract.setOrganismsDiffer(true);<a name="line.2159"></a>
<span class="sourceLineNo">2160</span>                    else currUCParserInteract.setOrganismsDiffer(false);<a name="line.2160"></a>
<span class="sourceLineNo">2161</span>                } else if (qName.equals(COMMENT_EXPERIMENTS_TAG)) {<a name="line.2161"></a>
<span class="sourceLineNo">2162</span>                    currUCParserInteract.setNumberExperiments(Integer.parseInt(val));<a name="line.2162"></a>
<span class="sourceLineNo">2163</span>                } else if (qName.equals(NOTE_TAG)) {<a name="line.2163"></a>
<span class="sourceLineNo">2164</span>                    if (currNoteIsFor.equals("COMMENT")) currUCParser.setNote(val);<a name="line.2164"></a>
<span class="sourceLineNo">2165</span>                    else if (currNoteIsFor.equals("ISOFORM")) currUCParser.setNote(val);<a name="line.2165"></a>
<span class="sourceLineNo">2166</span>                } else if (qName.equals(COMMENT_EVENT_TAG)) {<a name="line.2166"></a>
<span class="sourceLineNo">2167</span>                    currUCParserEvent.setComment(val);<a name="line.2167"></a>
<span class="sourceLineNo">2168</span>                } else if (qName.equals(COMMENT_ISOFORM_TAG)) {<a name="line.2168"></a>
<span class="sourceLineNo">2169</span>                    this.currSeqIsFor = "ENTRY";<a name="line.2169"></a>
<span class="sourceLineNo">2170</span>                    this.currNoteIsFor = "COMMENT";<a name="line.2170"></a>
<span class="sourceLineNo">2171</span>                } else if (qName.equals(ID_TAG)) {<a name="line.2171"></a>
<span class="sourceLineNo">2172</span>                    if (currIDIsFor.equals("ISOFORM")) currUCParserIsoform.getIsoIDs().add(val);<a name="line.2172"></a>
<span class="sourceLineNo">2173</span>                    else if (currIDIsFor.equals("INTERACTION")) currUCParserInteract.setID(val);<a name="line.2173"></a>
<span class="sourceLineNo">2174</span>                } else if (qName.equals(COMMENT_INTERACT_LABEL_TAG)) {<a name="line.2174"></a>
<span class="sourceLineNo">2175</span>                    currUCParserInteract.setLabel(val);<a name="line.2175"></a>
<span class="sourceLineNo">2176</span>                }<a name="line.2176"></a>
<span class="sourceLineNo">2177</span>                <a name="line.2177"></a>
<span class="sourceLineNo">2178</span>                else if (qName.equals(SEQUENCE_TAG)) {<a name="line.2178"></a>
<span class="sourceLineNo">2179</span>                    if (this.currSeqIsFor.equals("ENTRY") &amp;&amp; !this.parent.getElideSymbols()) {<a name="line.2179"></a>
<span class="sourceLineNo">2180</span>                        try {<a name="line.2180"></a>
<span class="sourceLineNo">2181</span>                            SymbolList sl = new SimpleSymbolList(symParser,<a name="line.2181"></a>
<span class="sourceLineNo">2182</span>                                    val.replaceAll("\\s+","").replaceAll("[\\.|~]","-"));<a name="line.2182"></a>
<span class="sourceLineNo">2183</span>                            rlistener.addSymbols(symParser.getAlphabet(),<a name="line.2183"></a>
<span class="sourceLineNo">2184</span>                                    (Symbol[])(sl.toList().toArray(new Symbol[0])),<a name="line.2184"></a>
<span class="sourceLineNo">2185</span>                                    0, sl.length());<a name="line.2185"></a>
<span class="sourceLineNo">2186</span>                        } catch (Exception e) {<a name="line.2186"></a>
<span class="sourceLineNo">2187</span>                            throw new ParseException(e);<a name="line.2187"></a>
<span class="sourceLineNo">2188</span>                        }<a name="line.2188"></a>
<span class="sourceLineNo">2189</span>                    }<a name="line.2189"></a>
<span class="sourceLineNo">2190</span>                }<a name="line.2190"></a>
<span class="sourceLineNo">2191</span>                <a name="line.2191"></a>
<span class="sourceLineNo">2192</span>                else if (qName.equals(ENTRY_TAG)) {<a name="line.2192"></a>
<span class="sourceLineNo">2193</span>                    // do the comments<a name="line.2193"></a>
<span class="sourceLineNo">2194</span>                    for (Iterator j = currComments.iterator(); j.hasNext();) {<a name="line.2194"></a>
<span class="sourceLineNo">2195</span>                        rlistener.setComment((String)j.next());<a name="line.2195"></a>
<span class="sourceLineNo">2196</span>                    }<a name="line.2196"></a>
<span class="sourceLineNo">2197</span>                    // do the crossrefs<a name="line.2197"></a>
<span class="sourceLineNo">2198</span>                    for (Iterator j = currDBXrefs.iterator(); j.hasNext();) {<a name="line.2198"></a>
<span class="sourceLineNo">2199</span>                        CrossRef dbx = (CrossRef)j.next();<a name="line.2199"></a>
<span class="sourceLineNo">2200</span>                        RankedCrossRef rdbx = new SimpleRankedCrossRef(dbx, 0);<a name="line.2200"></a>
<span class="sourceLineNo">2201</span>                        rlistener.setRankedCrossRef(rdbx);<a name="line.2201"></a>
<span class="sourceLineNo">2202</span>                    }<a name="line.2202"></a>
<span class="sourceLineNo">2203</span>                    // end the sequence<a name="line.2203"></a>
<span class="sourceLineNo">2204</span>                    currComments.clear();<a name="line.2204"></a>
<span class="sourceLineNo">2205</span>                    currDBXrefs.clear();<a name="line.2205"></a>
<span class="sourceLineNo">2206</span>                }<a name="line.2206"></a>
<span class="sourceLineNo">2207</span>                <a name="line.2207"></a>
<span class="sourceLineNo">2208</span>            } catch (ParseException e) {<a name="line.2208"></a>
<span class="sourceLineNo">2209</span>                throw new SAXException(e);<a name="line.2209"></a>
<span class="sourceLineNo">2210</span>            }<a name="line.2210"></a>
<span class="sourceLineNo">2211</span>            <a name="line.2211"></a>
<span class="sourceLineNo">2212</span>            // drop old string<a name="line.2212"></a>
<span class="sourceLineNo">2213</span>            this.m_currentString.setLength(0);<a name="line.2213"></a>
<span class="sourceLineNo">2214</span>        }<a name="line.2214"></a>
<span class="sourceLineNo">2215</span>        <a name="line.2215"></a>
<span class="sourceLineNo">2216</span>        // process text inside tags<a name="line.2216"></a>
<span class="sourceLineNo">2217</span>        @Override<a name="line.2217"></a>
<span class="sourceLineNo">2218</span>        public void characters(char[] ch, int start, int length) {<a name="line.2218"></a>
<span class="sourceLineNo">2219</span>            this.m_currentString.append(ch, start, length);<a name="line.2219"></a>
<span class="sourceLineNo">2220</span>        }<a name="line.2220"></a>
<span class="sourceLineNo">2221</span>    }<a name="line.2221"></a>
<span class="sourceLineNo">2222</span>}<a name="line.2222"></a>
<span class="sourceLineNo">2223</span><a name="line.2223"></a>




























































</pre>
</div>
</body>
</html>
