<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_131) on Thu Oct 19 12:07:58 PDT 2017 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Sequence (biojava 4.2.9 API)</title>
<meta name="date" content="2017-10-19">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
</head>
<body>
<script type="text/javascript"><!--
    try {
        if (location.href.indexOf('is-external=true') == -1) {
            parent.document.title="Sequence (biojava 4.2.9 API)";
        }
    }
    catch(err) {
    }
//-->
var methods = {"i0":6,"i1":6,"i2":6,"i3":6,"i4":6,"i5":6,"i6":6,"i7":6,"i8":6,"i9":6};
var tabs = {65535:["t0","All Methods"],2:["t2","Instance Methods"],4:["t3","Abstract Methods"]};
var altColor = "altColor";
var rowColor = "rowColor";
var tableTab = "tableTab";
var activeTableTab = "activeTableTab";
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar.top">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.top" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.top.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Sequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html" title="class in org.biojava.nbio.core.sequence.template"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/template/Sequence.html" target="_top">Frames</a></li>
<li><a href="Sequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.core.sequence.template</div>
<h2 title="Interface Sequence" class="title">Interface Sequence&lt;C extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt><span class="paramLabel">Type Parameters:</span></dt>
<dd><code>C</code> - Compound a Sequence holds</dd>
</dl>
<dl>
<dt>All Superinterfaces:</dt>
<dd><a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;C&gt;</dd>
</dl>
<dl>
<dt>All Known Subinterfaces:</dt>
<dd><a href="../../../../../../org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence</a>&lt;S,C&gt;, <a href="../../../../../../org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence</a>&lt;S,C&gt;, <a href="../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader</a>&lt;C&gt;, <a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;C&gt;, <a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;C&gt;</dd>
</dl>
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">ArrayListProxySequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">ArrayListSequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ComplementSequenceView</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">GenbankProxySequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">JoiningSequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ReversedSequenceView</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">RnaSequenceView</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava.nbio.core.sequence.loader">SequenceFileProxyLoader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template">SequenceProxyView</a>, <a href="../../../../../../org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">SingleCompoundSequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">StringProxySequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">TwoBitSequenceReader</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">UniprotProxySequenceReader</a></dd>
</dl>
<hr>
<br>
<pre>public interface <a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.39">Sequence</a>&lt;C extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;
extends <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;C&gt;, <a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a></pre>
<div class="block">Main interface for defining a collection of Compounds and accessing them
 using biological indexes</div>
<dl>
<dt><span class="simpleTagLabel">Author:</span></dt>
<dd>Richard Holland, Andy Yates, Scooter Willis</dd>
</dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method.summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="memberSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span id="t0" class="activeTableTab"><span>All Methods</span><span class="tabEnd">&nbsp;</span></span><span id="t2" class="tableTab"><span><a href="javascript:show(2);">Instance Methods</a></span><span class="tabEnd">&nbsp;</span></span><span id="t3" class="tableTab"><span><a href="javascript:show(4);">Abstract Methods</a></span><span class="tabEnd">&nbsp;</span></span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr id="i0" class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#countCompounds-C...-">countCompounds</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>...&nbsp;compounds)</code>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</td>
</tr>
<tr id="i1" class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getAsList--">getAsList</a></span>()</code>
<div class="block">Returns the Sequence as a List of compounds</div>
</td>
</tr>
<tr id="i2" class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getCompoundAt-int-">getCompoundAt</a></span>(int&nbsp;position)</code>
<div class="block">Returns the Compound at the given biological index</div>
</td>
</tr>
<tr id="i3" class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getCompoundSet--">getCompoundSet</a></span>()</code>
<div class="block">Gets the compound set used to back this Sequence</div>
</td>
</tr>
<tr id="i4" class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getIndexOf-C-">getIndexOf</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&nbsp;compound)</code>
<div class="block">Scans through the Sequence looking for the first occurrence of the given
 compound</div>
</td>
</tr>
<tr id="i5" class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getInverse--">getInverse</a></span>()</code>
<div class="block">Does the <em>right thing</em> to get the inverse of the current
 Sequence.</div>
</td>
</tr>
<tr id="i6" class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLastIndexOf-C-">getLastIndexOf</a></span>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&nbsp;compound)</code>
<div class="block">Scans through the Sequence looking for the last occurrence of the given
 compound</div>
</td>
</tr>
<tr id="i7" class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLength--">getLength</a></span>()</code>
<div class="block">Returns the length of the Sequence</div>
</td>
</tr>
<tr id="i8" class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getSequenceAsString--">getSequenceAsString</a></span>()</code>
<div class="block">Returns the String representation of the Sequence</div>
</td>
</tr>
<tr id="i9" class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;</code></td>
<td class="colLast"><code><span class="memberNameLink"><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getSubSequence-java.lang.Integer-java.lang.Integer-">getSubSequence</a></span>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;start,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>
<div class="block">Returns a portion of the sequence from the different positions.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.java.lang.Iterable">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true#forEach-java.util.function.Consumer-" title="class or interface in java.lang">forEach</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true#iterator--" title="class or interface in java.lang">iterator</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true#spliterator--" title="class or interface in java.lang">spliterator</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods.inherited.from.class.org.biojava.nbio.core.sequence.template.Accessioned">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;org.biojava.nbio.core.sequence.template.<a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a></h3>
<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html#getAccession--">getAccession</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method.detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getLength--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLength</h4>
<pre>int&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.44">getLength</a>()</pre>
<div class="block">Returns the length of the Sequence</div>
</li>
</ul>
<a name="getCompoundAt-int-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompoundAt</h4>
<pre><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.52">getCompoundAt</a>(int&nbsp;position)</pre>
<div class="block">Returns the Compound at the given biological index</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>position</code> - Biological index (1 to n)</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>Compound at the specified position</dd>
</dl>
</li>
</ul>
<a name="getIndexOf-org.biojava.nbio.core.sequence.template.Compound-">
<!--   -->
</a><a name="getIndexOf-C-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIndexOf</h4>
<pre>int&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.61">getIndexOf</a>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&nbsp;compound)</pre>
<div class="block">Scans through the Sequence looking for the first occurrence of the given
 compound</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>compound</code> - Compounds to look for</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>Index of the first position of the compound in the sequence (1 to n)</dd>
</dl>
</li>
</ul>
<a name="getLastIndexOf-org.biojava.nbio.core.sequence.template.Compound-">
<!--   -->
</a><a name="getLastIndexOf-C-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLastIndexOf</h4>
<pre>int&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.70">getLastIndexOf</a>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&nbsp;compound)</pre>
<div class="block">Scans through the Sequence looking for the last occurrence of the given
 compound</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>compound</code> - Compounds to look for</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>Index of the last position of the compound in the sequence (1 to n)</dd>
</dl>
</li>
</ul>
<a name="getSequenceAsString--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAsString</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.75">getSequenceAsString</a>()</pre>
<div class="block">Returns the String representation of the Sequence</div>
</li>
</ul>
<a name="getAsList--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsList</h4>
<pre><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.80">getAsList</a>()</pre>
<div class="block">Returns the Sequence as a List of compounds</div>
</li>
</ul>
<a name="getSubSequence-java.lang.Integer-java.lang.Integer-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSubSequence</h4>
<pre><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.90">getSubSequence</a>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;start,
                               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</pre>
<div class="block">Returns a portion of the sequence from the different positions. This is
 indexed from 1</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>start</code> - Biological index start; must be greater than 0</dd>
<dd><code>end</code> - Biological end; must be less than length + 1</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>A SequenceView of the offset</dd>
</dl>
</li>
</ul>
<a name="getCompoundSet--">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompoundSet</h4>
<pre><a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.95">getCompoundSet</a>()</pre>
<div class="block">Gets the compound set used to back this Sequence</div>
</li>
</ul>
<a name="countCompounds-org.biojava.nbio.core.sequence.template.Compound:A-">
<!--   -->
</a><a name="countCompounds-C...-">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>countCompounds</h4>
<pre>int&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.103">countCompounds</a>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>...&nbsp;compounds)</pre>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
<dl>
<dt><span class="paramLabel">Parameters:</span></dt>
<dd><code>compounds</code> - Vargs of the compounds to count</dd>
<dt><span class="returnLabel">Returns:</span></dt>
<dd>Number of times a compound was found</dd>
</dl>
</li>
</ul>
<a name="getInverse--">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getInverse</h4>
<pre><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="type parameter in Sequence">C</a>&gt;&nbsp;<a href="../../../../../../src-html/org/biojava/nbio/core/sequence/template/Sequence.html#line.110">getInverse</a>()</pre>
<div class="block">Does the <em>right thing</em> to get the inverse of the current
 Sequence. This means either reversing the Sequence and optionally
 complementing the Sequence.</div>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar.bottom">
<!--   -->
</a>
<div class="skipNav"><a href="#skip.navbar.bottom" title="Skip navigation links">Skip navigation links</a></div>
<a name="navbar.bottom.firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/Sequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template"><span class="typeNameLink">Prev&nbsp;Class</span></a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceMixin.html" title="class in org.biojava.nbio.core.sequence.template"><span class="typeNameLink">Next&nbsp;Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/template/Sequence.html" target="_top">Frames</a></li>
<li><a href="Sequence.html" target="_top">No&nbsp;Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All&nbsp;Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method.detail">Method</a></li>
</ul>
</div>
<a name="skip.navbar.bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2017 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
