<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.bio.seq.io;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.io.BufferedReader;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.io.File;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.io.FileNotFoundException;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.io.FileReader;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.io.IOException;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.io.InputStream;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.io.OutputStream;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import java.io.PrintStream;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import java.util.regex.Pattern;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import java.util.regex.PatternSyntaxException;<a name="line.33"></a>
<span class="sourceLineNo">034</span><a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.bio.BioError;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.bio.BioException;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.bio.alignment.Alignment;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.bio.seq.DNATools;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.bio.seq.NucleotideTools;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.bio.seq.ProteinTools;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.bio.seq.RNATools;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.bio.seq.Sequence;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import org.biojava.bio.seq.SequenceIterator;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import org.biojava.bio.seq.db.HashSequenceDB;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import org.biojava.bio.seq.db.IDMaker;<a name="line.45"></a>
<span class="sourceLineNo">046</span>import org.biojava.bio.seq.db.SequenceDB;<a name="line.46"></a>
<span class="sourceLineNo">047</span>import org.biojava.bio.symbol.Alphabet;<a name="line.47"></a>
<span class="sourceLineNo">048</span>import org.biojava.bio.symbol.FiniteAlphabet;<a name="line.48"></a>
<span class="sourceLineNo">049</span>import org.biojava.bio.symbol.IllegalSymbolException;<a name="line.49"></a>
<span class="sourceLineNo">050</span>import org.biojava.utils.AssertionFailure;<a name="line.50"></a>
<span class="sourceLineNo">051</span>import org.biojava.utils.ChangeVetoException;<a name="line.51"></a>
<span class="sourceLineNo">052</span><a name="line.52"></a>
<span class="sourceLineNo">053</span>/**<a name="line.53"></a>
<span class="sourceLineNo">054</span> * A set of convenience methods for handling common file formats.<a name="line.54"></a>
<span class="sourceLineNo">055</span> *<a name="line.55"></a>
<span class="sourceLineNo">056</span> * @author Thomas Down<a name="line.56"></a>
<span class="sourceLineNo">057</span> * @author Mark Schreiber<a name="line.57"></a>
<span class="sourceLineNo">058</span> * @author Nimesh Singh<a name="line.58"></a>
<span class="sourceLineNo">059</span> * @author Matthew Pocock<a name="line.59"></a>
<span class="sourceLineNo">060</span> * @author Keith James<a name="line.60"></a>
<span class="sourceLineNo">061</span> * @since 1.1<a name="line.61"></a>
<span class="sourceLineNo">062</span> * @deprecated use org.biojavax.bio.seq.RichSequence.IOTools<a name="line.62"></a>
<span class="sourceLineNo">063</span> */<a name="line.63"></a>
<span class="sourceLineNo">064</span>public final class SeqIOTools  {<a name="line.64"></a>
<span class="sourceLineNo">065</span>    private static SequenceBuilderFactory _emblBuilderFactory;<a name="line.65"></a>
<span class="sourceLineNo">066</span>    private static SequenceBuilderFactory _genbankBuilderFactory;<a name="line.66"></a>
<span class="sourceLineNo">067</span>    private static SequenceBuilderFactory _genpeptBuilderFactory;<a name="line.67"></a>
<span class="sourceLineNo">068</span>    private static SequenceBuilderFactory _swissprotBuilderFactory;<a name="line.68"></a>
<span class="sourceLineNo">069</span>    private static SequenceBuilderFactory _fastaBuilderFactory;<a name="line.69"></a>
<span class="sourceLineNo">070</span><a name="line.70"></a>
<span class="sourceLineNo">071</span>    /**<a name="line.71"></a>
<span class="sourceLineNo">072</span>     * This can't be instantiated.<a name="line.72"></a>
<span class="sourceLineNo">073</span>     */<a name="line.73"></a>
<span class="sourceLineNo">074</span>    private SeqIOTools() {<a name="line.74"></a>
<span class="sourceLineNo">075</span>    }<a name="line.75"></a>
<span class="sourceLineNo">076</span><a name="line.76"></a>
<span class="sourceLineNo">077</span>    /**<a name="line.77"></a>
<span class="sourceLineNo">078</span>     * Get a default SequenceBuilderFactory for handling EMBL<a name="line.78"></a>
<span class="sourceLineNo">079</span>     * files.<a name="line.79"></a>
<span class="sourceLineNo">080</span>     * @return a &lt;CODE&gt;SmartSequenceBuilder.FACTORY&lt;/CODE&gt;<a name="line.80"></a>
<span class="sourceLineNo">081</span>     */<a name="line.81"></a>
<span class="sourceLineNo">082</span>    public static SequenceBuilderFactory getEmblBuilderFactory() {<a name="line.82"></a>
<span class="sourceLineNo">083</span>        if (_emblBuilderFactory == null) {<a name="line.83"></a>
<span class="sourceLineNo">084</span>            _emblBuilderFactory =<a name="line.84"></a>
<span class="sourceLineNo">085</span>                new EmblProcessor.Factory(SmartSequenceBuilder.FACTORY);<a name="line.85"></a>
<span class="sourceLineNo">086</span>        }<a name="line.86"></a>
<span class="sourceLineNo">087</span>        return _emblBuilderFactory;<a name="line.87"></a>
<span class="sourceLineNo">088</span>    }<a name="line.88"></a>
<span class="sourceLineNo">089</span><a name="line.89"></a>
<span class="sourceLineNo">090</span>    /**<a name="line.90"></a>
<span class="sourceLineNo">091</span>     * Iterate over the sequences in an EMBL-format stream.<a name="line.91"></a>
<span class="sourceLineNo">092</span>     * @param br A reader for the EMBL source or file<a name="line.92"></a>
<span class="sourceLineNo">093</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.93"></a>
<span class="sourceLineNo">094</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.94"></a>
<span class="sourceLineNo">095</span>     */<a name="line.95"></a>
<span class="sourceLineNo">096</span>    public static SequenceIterator readEmbl(BufferedReader br) {<a name="line.96"></a>
<span class="sourceLineNo">097</span>        return new StreamReader(br,<a name="line.97"></a>
<span class="sourceLineNo">098</span>                                new EmblLikeFormat(),<a name="line.98"></a>
<span class="sourceLineNo">099</span>                                getDNAParser(),<a name="line.99"></a>
<span class="sourceLineNo">100</span>                                getEmblBuilderFactory());<a name="line.100"></a>
<span class="sourceLineNo">101</span>    }<a name="line.101"></a>
<span class="sourceLineNo">102</span><a name="line.102"></a>
<span class="sourceLineNo">103</span>    /**<a name="line.103"></a>
<span class="sourceLineNo">104</span>     * Iterate over the sequences in an EMBL-format stream, but for RNA.<a name="line.104"></a>
<span class="sourceLineNo">105</span>     * @param br A reader for the EMBL source or file<a name="line.105"></a>
<span class="sourceLineNo">106</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.106"></a>
<span class="sourceLineNo">107</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.107"></a>
<span class="sourceLineNo">108</span>     */<a name="line.108"></a>
<span class="sourceLineNo">109</span>    public static SequenceIterator readEmblRNA(BufferedReader br) {<a name="line.109"></a>
<span class="sourceLineNo">110</span>        return new StreamReader(br,<a name="line.110"></a>
<span class="sourceLineNo">111</span>                                new EmblLikeFormat(),<a name="line.111"></a>
<span class="sourceLineNo">112</span>                                getRNAParser(),<a name="line.112"></a>
<span class="sourceLineNo">113</span>                                getEmblBuilderFactory());<a name="line.113"></a>
<span class="sourceLineNo">114</span>    }<a name="line.114"></a>
<span class="sourceLineNo">115</span><a name="line.115"></a>
<span class="sourceLineNo">116</span>    /**<a name="line.116"></a>
<span class="sourceLineNo">117</span>     * Iterate over the sequences in an EMBL-format stream.<a name="line.117"></a>
<span class="sourceLineNo">118</span>     * @param br A reader for the EMBL source or file<a name="line.118"></a>
<span class="sourceLineNo">119</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.119"></a>
<span class="sourceLineNo">120</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.120"></a>
<span class="sourceLineNo">121</span>     */<a name="line.121"></a>
<span class="sourceLineNo">122</span>    public static SequenceIterator readEmblNucleotide(BufferedReader br) {<a name="line.122"></a>
<span class="sourceLineNo">123</span>        return new StreamReader(br,<a name="line.123"></a>
<span class="sourceLineNo">124</span>                                new EmblLikeFormat(),<a name="line.124"></a>
<span class="sourceLineNo">125</span>                                getNucleotideParser(),<a name="line.125"></a>
<span class="sourceLineNo">126</span>                                getEmblBuilderFactory());<a name="line.126"></a>
<span class="sourceLineNo">127</span>    }<a name="line.127"></a>
<span class="sourceLineNo">128</span><a name="line.128"></a>
<span class="sourceLineNo">129</span>    /**<a name="line.129"></a>
<span class="sourceLineNo">130</span>     * Get a default SequenceBuilderFactory for handling GenBank<a name="line.130"></a>
<span class="sourceLineNo">131</span>     * files.<a name="line.131"></a>
<span class="sourceLineNo">132</span>     * @return a &lt;code&gt;SmartSequenceBuilder.FACTORY&lt;/code&gt;<a name="line.132"></a>
<span class="sourceLineNo">133</span>     */<a name="line.133"></a>
<span class="sourceLineNo">134</span>    public static SequenceBuilderFactory getGenbankBuilderFactory() {<a name="line.134"></a>
<span class="sourceLineNo">135</span>        if (_genbankBuilderFactory == null) {<a name="line.135"></a>
<span class="sourceLineNo">136</span>            _genbankBuilderFactory =<a name="line.136"></a>
<span class="sourceLineNo">137</span>                new GenbankProcessor.Factory(SmartSequenceBuilder.FACTORY);<a name="line.137"></a>
<span class="sourceLineNo">138</span>        }<a name="line.138"></a>
<span class="sourceLineNo">139</span>        return _genbankBuilderFactory;<a name="line.139"></a>
<span class="sourceLineNo">140</span>    }<a name="line.140"></a>
<span class="sourceLineNo">141</span><a name="line.141"></a>
<span class="sourceLineNo">142</span>    /**<a name="line.142"></a>
<span class="sourceLineNo">143</span>     * Iterate over the sequences in an Genbank-format stream.<a name="line.143"></a>
<span class="sourceLineNo">144</span>     * @param br A reader for the Genbank source or file<a name="line.144"></a>
<span class="sourceLineNo">145</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.145"></a>
<span class="sourceLineNo">146</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.146"></a>
<span class="sourceLineNo">147</span>     */<a name="line.147"></a>
<span class="sourceLineNo">148</span>    public static SequenceIterator readGenbank(BufferedReader br) {<a name="line.148"></a>
<span class="sourceLineNo">149</span>        return new StreamReader(br,<a name="line.149"></a>
<span class="sourceLineNo">150</span>                                new GenbankFormat(),<a name="line.150"></a>
<span class="sourceLineNo">151</span>                                getDNAParser(),<a name="line.151"></a>
<span class="sourceLineNo">152</span>                                getGenbankBuilderFactory());<a name="line.152"></a>
<span class="sourceLineNo">153</span>    }<a name="line.153"></a>
<span class="sourceLineNo">154</span>    <a name="line.154"></a>
<span class="sourceLineNo">155</span>    /**<a name="line.155"></a>
<span class="sourceLineNo">156</span>     * Iterate over the sequences in an GenbankXML-format stream.<a name="line.156"></a>
<span class="sourceLineNo">157</span>     * @param br A reader for the GenbanXML source or file<a name="line.157"></a>
<span class="sourceLineNo">158</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.158"></a>
<span class="sourceLineNo">159</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.159"></a>
<span class="sourceLineNo">160</span>     */    <a name="line.160"></a>
<span class="sourceLineNo">161</span>    public static SequenceIterator readGenbankXml( BufferedReader br )<a name="line.161"></a>
<span class="sourceLineNo">162</span>    {<a name="line.162"></a>
<span class="sourceLineNo">163</span>        return new StreamReader( br,<a name="line.163"></a>
<span class="sourceLineNo">164</span>                                 new GenbankXmlFormat(),<a name="line.164"></a>
<span class="sourceLineNo">165</span>                                 getDNAParser(),<a name="line.165"></a>
<span class="sourceLineNo">166</span>                                 getGenbankBuilderFactory() );<a name="line.166"></a>
<span class="sourceLineNo">167</span>    }<a name="line.167"></a>
<span class="sourceLineNo">168</span><a name="line.168"></a>
<span class="sourceLineNo">169</span>    /**<a name="line.169"></a>
<span class="sourceLineNo">170</span>    * Get a default SequenceBuilderFactory for handling Genpept<a name="line.170"></a>
<span class="sourceLineNo">171</span>    * files.<a name="line.171"></a>
<span class="sourceLineNo">172</span>    * @return a &lt;code&gt;SmartSequenceBuilder.FACTORY&lt;/code&gt;<a name="line.172"></a>
<span class="sourceLineNo">173</span>    */<a name="line.173"></a>
<span class="sourceLineNo">174</span>    public static SequenceBuilderFactory getGenpeptBuilderFactory() {<a name="line.174"></a>
<span class="sourceLineNo">175</span>        if (_genpeptBuilderFactory == null) {<a name="line.175"></a>
<span class="sourceLineNo">176</span>            _genpeptBuilderFactory =<a name="line.176"></a>
<span class="sourceLineNo">177</span>                new GenbankProcessor.Factory(SmartSequenceBuilder.FACTORY);<a name="line.177"></a>
<span class="sourceLineNo">178</span>        }<a name="line.178"></a>
<span class="sourceLineNo">179</span>        return _genpeptBuilderFactory;<a name="line.179"></a>
<span class="sourceLineNo">180</span>    }<a name="line.180"></a>
<span class="sourceLineNo">181</span><a name="line.181"></a>
<span class="sourceLineNo">182</span>    /**<a name="line.182"></a>
<span class="sourceLineNo">183</span>    * Iterate over the sequences in an Genpept-format stream.<a name="line.183"></a>
<span class="sourceLineNo">184</span>    * @param br A reader for the Genpept source or file<a name="line.184"></a>
<span class="sourceLineNo">185</span>    * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.185"></a>
<span class="sourceLineNo">186</span>    * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.186"></a>
<span class="sourceLineNo">187</span>    */<a name="line.187"></a>
<span class="sourceLineNo">188</span>    public static SequenceIterator readGenpept(BufferedReader br) {<a name="line.188"></a>
<span class="sourceLineNo">189</span>        return new StreamReader(br,<a name="line.189"></a>
<span class="sourceLineNo">190</span>                                new GenbankFormat(),<a name="line.190"></a>
<span class="sourceLineNo">191</span>                                getProteinParser(),<a name="line.191"></a>
<span class="sourceLineNo">192</span>                                getGenpeptBuilderFactory());<a name="line.192"></a>
<span class="sourceLineNo">193</span>    }<a name="line.193"></a>
<span class="sourceLineNo">194</span><a name="line.194"></a>
<span class="sourceLineNo">195</span>    /**<a name="line.195"></a>
<span class="sourceLineNo">196</span>     * Get a default SequenceBuilderFactory for handling Swissprot<a name="line.196"></a>
<span class="sourceLineNo">197</span>     * files.<a name="line.197"></a>
<span class="sourceLineNo">198</span>     * @return a &lt;code&gt;SmartSequenceBuilder.FACTORY&lt;/code&gt;<a name="line.198"></a>
<span class="sourceLineNo">199</span>     */<a name="line.199"></a>
<span class="sourceLineNo">200</span>    public static SequenceBuilderFactory getSwissprotBuilderFactory() {<a name="line.200"></a>
<span class="sourceLineNo">201</span>        if (_swissprotBuilderFactory == null) {<a name="line.201"></a>
<span class="sourceLineNo">202</span>            _swissprotBuilderFactory =<a name="line.202"></a>
<span class="sourceLineNo">203</span>                new SwissprotProcessor.Factory(SmartSequenceBuilder.FACTORY);<a name="line.203"></a>
<span class="sourceLineNo">204</span>        }<a name="line.204"></a>
<span class="sourceLineNo">205</span>        return _swissprotBuilderFactory;<a name="line.205"></a>
<span class="sourceLineNo">206</span>    }<a name="line.206"></a>
<span class="sourceLineNo">207</span><a name="line.207"></a>
<span class="sourceLineNo">208</span>    /**<a name="line.208"></a>
<span class="sourceLineNo">209</span>     * Iterate over the sequences in an Swissprot-format stream.<a name="line.209"></a>
<span class="sourceLineNo">210</span>    * @param br A reader for the Swissprot source or file<a name="line.210"></a>
<span class="sourceLineNo">211</span>    * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.211"></a>
<span class="sourceLineNo">212</span>    * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file     <a name="line.212"></a>
<span class="sourceLineNo">213</span>     */<a name="line.213"></a>
<span class="sourceLineNo">214</span>    public static SequenceIterator readSwissprot(BufferedReader br) {<a name="line.214"></a>
<span class="sourceLineNo">215</span>        return new StreamReader(br,<a name="line.215"></a>
<span class="sourceLineNo">216</span>                                new EmblLikeFormat(),<a name="line.216"></a>
<span class="sourceLineNo">217</span>                                getProteinParser(),<a name="line.217"></a>
<span class="sourceLineNo">218</span>                                getSwissprotBuilderFactory());<a name="line.218"></a>
<span class="sourceLineNo">219</span>    }<a name="line.219"></a>
<span class="sourceLineNo">220</span><a name="line.220"></a>
<span class="sourceLineNo">221</span>    /**<a name="line.221"></a>
<span class="sourceLineNo">222</span>     * Get a default SequenceBuilderFactory for handling FASTA<a name="line.222"></a>
<span class="sourceLineNo">223</span>     * files.<a name="line.223"></a>
<span class="sourceLineNo">224</span>     * @return a &lt;code&gt;SmartSequenceBuilder.FACTORY&lt;/code&gt;<a name="line.224"></a>
<span class="sourceLineNo">225</span>     */<a name="line.225"></a>
<span class="sourceLineNo">226</span>    public static SequenceBuilderFactory getFastaBuilderFactory() {<a name="line.226"></a>
<span class="sourceLineNo">227</span>        if (_fastaBuilderFactory == null) {<a name="line.227"></a>
<span class="sourceLineNo">228</span>            _fastaBuilderFactory = new FastaDescriptionLineParser.Factory(<a name="line.228"></a>
<span class="sourceLineNo">229</span>                        SmartSequenceBuilder.FACTORY);<a name="line.229"></a>
<span class="sourceLineNo">230</span>        }<a name="line.230"></a>
<span class="sourceLineNo">231</span>        return _fastaBuilderFactory;<a name="line.231"></a>
<span class="sourceLineNo">232</span>    }<a name="line.232"></a>
<span class="sourceLineNo">233</span><a name="line.233"></a>
<span class="sourceLineNo">234</span>    /**<a name="line.234"></a>
<span class="sourceLineNo">235</span>     * Read a fasta file.<a name="line.235"></a>
<span class="sourceLineNo">236</span>     *<a name="line.236"></a>
<span class="sourceLineNo">237</span>     * @param br    the BufferedReader to read data from<a name="line.237"></a>
<span class="sourceLineNo">238</span>     * @param sTok  a SymbolTokenization that understands the sequences<a name="line.238"></a>
<span class="sourceLineNo">239</span>     * @return      a SequenceIterator over each sequence in the fasta file<a name="line.239"></a>
<span class="sourceLineNo">240</span>     */<a name="line.240"></a>
<span class="sourceLineNo">241</span>    public static SequenceIterator readFasta(<a name="line.241"></a>
<span class="sourceLineNo">242</span>            BufferedReader br, SymbolTokenization sTok)<a name="line.242"></a>
<span class="sourceLineNo">243</span>    {<a name="line.243"></a>
<span class="sourceLineNo">244</span>      return new StreamReader(br,<a name="line.244"></a>
<span class="sourceLineNo">245</span>                              new FastaFormat(),<a name="line.245"></a>
<span class="sourceLineNo">246</span>                              sTok,<a name="line.246"></a>
<span class="sourceLineNo">247</span>                              getFastaBuilderFactory());<a name="line.247"></a>
<span class="sourceLineNo">248</span>    }<a name="line.248"></a>
<span class="sourceLineNo">249</span><a name="line.249"></a>
<span class="sourceLineNo">250</span>    /**<a name="line.250"></a>
<span class="sourceLineNo">251</span>     * Read a fasta file using a custom type of SymbolList. For example,<a name="line.251"></a>
<span class="sourceLineNo">252</span>     * use SmartSequenceBuilder.FACTORY to emulate readFasta(BufferedReader,<a name="line.252"></a>
<span class="sourceLineNo">253</span>     * SymbolTokenization) and SmartSequenceBuilder.BIT_PACKED to force all<a name="line.253"></a>
<span class="sourceLineNo">254</span>     * symbols to be encoded using bit-packing.<a name="line.254"></a>
<span class="sourceLineNo">255</span>     * @param br the BufferedReader to read data from<a name="line.255"></a>
<span class="sourceLineNo">256</span>     * @param sTok a SymbolTokenization that understands the sequences<a name="line.256"></a>
<span class="sourceLineNo">257</span>     * @param seqFactory a factory used to build a SymbolList<a name="line.257"></a>
<span class="sourceLineNo">258</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.258"></a>
<span class="sourceLineNo">259</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.259"></a>
<span class="sourceLineNo">260</span>     */<a name="line.260"></a>
<span class="sourceLineNo">261</span>    public static SequenceIterator readFasta(<a name="line.261"></a>
<span class="sourceLineNo">262</span>            BufferedReader br,<a name="line.262"></a>
<span class="sourceLineNo">263</span>            SymbolTokenization sTok,<a name="line.263"></a>
<span class="sourceLineNo">264</span>            SequenceBuilderFactory seqFactory)<a name="line.264"></a>
<span class="sourceLineNo">265</span>    {<a name="line.265"></a>
<span class="sourceLineNo">266</span>      return new StreamReader(<a name="line.266"></a>
<span class="sourceLineNo">267</span>              br,<a name="line.267"></a>
<span class="sourceLineNo">268</span>              new FastaFormat(),<a name="line.268"></a>
<span class="sourceLineNo">269</span>              sTok,<a name="line.269"></a>
<span class="sourceLineNo">270</span>              new FastaDescriptionLineParser.Factory(seqFactory));<a name="line.270"></a>
<span class="sourceLineNo">271</span>    }<a name="line.271"></a>
<span class="sourceLineNo">272</span><a name="line.272"></a>
<span class="sourceLineNo">273</span>    /**<a name="line.273"></a>
<span class="sourceLineNo">274</span>     * Iterate over the sequences in an FASTA-format stream of DNA sequences.<a name="line.274"></a>
<span class="sourceLineNo">275</span>     * @param br the BufferedReader to read data from<a name="line.275"></a>
<span class="sourceLineNo">276</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.276"></a>
<span class="sourceLineNo">277</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.277"></a>
<span class="sourceLineNo">278</span>     */<a name="line.278"></a>
<span class="sourceLineNo">279</span>    public static SequenceIterator readFastaDNA(BufferedReader br) {<a name="line.279"></a>
<span class="sourceLineNo">280</span>        return new StreamReader(br,<a name="line.280"></a>
<span class="sourceLineNo">281</span>                                new FastaFormat(),<a name="line.281"></a>
<span class="sourceLineNo">282</span>                                getDNAParser(),<a name="line.282"></a>
<span class="sourceLineNo">283</span>                                getFastaBuilderFactory());<a name="line.283"></a>
<span class="sourceLineNo">284</span>    }<a name="line.284"></a>
<span class="sourceLineNo">285</span><a name="line.285"></a>
<span class="sourceLineNo">286</span>    /**<a name="line.286"></a>
<span class="sourceLineNo">287</span>     * Iterate over the sequences in an FASTA-format stream of RNA sequences.<a name="line.287"></a>
<span class="sourceLineNo">288</span>     * @param br the BufferedReader to read data from<a name="line.288"></a>
<span class="sourceLineNo">289</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.289"></a>
<span class="sourceLineNo">290</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.290"></a>
<span class="sourceLineNo">291</span>     */<a name="line.291"></a>
<span class="sourceLineNo">292</span>    public static SequenceIterator readFastaRNA(BufferedReader br) {<a name="line.292"></a>
<span class="sourceLineNo">293</span>        return new StreamReader(br,<a name="line.293"></a>
<span class="sourceLineNo">294</span>                                new FastaFormat(),<a name="line.294"></a>
<span class="sourceLineNo">295</span>                                getRNAParser(),<a name="line.295"></a>
<span class="sourceLineNo">296</span>                                getFastaBuilderFactory());<a name="line.296"></a>
<span class="sourceLineNo">297</span>    }<a name="line.297"></a>
<span class="sourceLineNo">298</span><a name="line.298"></a>
<span class="sourceLineNo">299</span>    /**<a name="line.299"></a>
<span class="sourceLineNo">300</span>     * Iterate over the sequences in an FASTA-format stream of Protein sequences.<a name="line.300"></a>
<span class="sourceLineNo">301</span>     * @param br the BufferedReader to read data from<a name="line.301"></a>
<span class="sourceLineNo">302</span>     * @return a &lt;CODE&gt;SequenceIterator&lt;/CODE&gt; that iterates over each<a name="line.302"></a>
<span class="sourceLineNo">303</span>     * &lt;CODE&gt;Sequence&lt;/CODE&gt; in the file<a name="line.303"></a>
<span class="sourceLineNo">304</span>     */<a name="line.304"></a>
<span class="sourceLineNo">305</span>    public static SequenceIterator readFastaProtein(BufferedReader br) {<a name="line.305"></a>
<span class="sourceLineNo">306</span>        return new StreamReader(br,<a name="line.306"></a>
<span class="sourceLineNo">307</span>                                new FastaFormat(),<a name="line.307"></a>
<span class="sourceLineNo">308</span>                                getProteinParser(),<a name="line.308"></a>
<span class="sourceLineNo">309</span>                                getFastaBuilderFactory());<a name="line.309"></a>
<span class="sourceLineNo">310</span>    }<a name="line.310"></a>
<span class="sourceLineNo">311</span><a name="line.311"></a>
<span class="sourceLineNo">312</span>    /**<a name="line.312"></a>
<span class="sourceLineNo">313</span>     * Create a sequence database from a fasta file provided as an<a name="line.313"></a>
<span class="sourceLineNo">314</span>     * input stream.  Note this somewhat duplicates functionality in<a name="line.314"></a>
<span class="sourceLineNo">315</span>     * the readFastaDNA and readFastaProtein methods but uses a stream<a name="line.315"></a>
<span class="sourceLineNo">316</span>     * rather than a reader and returns a SequenceDB rather than a<a name="line.316"></a>
<span class="sourceLineNo">317</span>     * SequenceIterator. If the returned DB is likely to be large then<a name="line.317"></a>
<span class="sourceLineNo">318</span>     * the above mentioned methods should be used.<a name="line.318"></a>
<span class="sourceLineNo">319</span>     * @return a &lt;code&gt;SequenceDB&lt;/code&gt; containing all the &lt;code&gt;Sequences&lt;/code&gt;<a name="line.319"></a>
<span class="sourceLineNo">320</span>     * in the file.<a name="line.320"></a>
<span class="sourceLineNo">321</span>     * @since 1.2<a name="line.321"></a>
<span class="sourceLineNo">322</span>     * @param seqFile The file containg the fasta formatted sequences<a name="line.322"></a>
<span class="sourceLineNo">323</span>     * @param alpha The &lt;code&gt;Alphabet&lt;/code&gt; of the sequence, ie DNA, RNA etc<a name="line.323"></a>
<span class="sourceLineNo">324</span>     * @throws BioException if problems occur during reading of the<a name="line.324"></a>
<span class="sourceLineNo">325</span>     * stream.<a name="line.325"></a>
<span class="sourceLineNo">326</span>     */<a name="line.326"></a>
<span class="sourceLineNo">327</span>    public static SequenceDB readFasta(InputStream seqFile, Alphabet alpha)<a name="line.327"></a>
<span class="sourceLineNo">328</span>        throws BioException {<a name="line.328"></a>
<span class="sourceLineNo">329</span>        HashSequenceDB db = new HashSequenceDB(IDMaker.byName);<a name="line.329"></a>
<span class="sourceLineNo">330</span>        SequenceBuilderFactory sbFact =<a name="line.330"></a>
<span class="sourceLineNo">331</span>            new FastaDescriptionLineParser.Factory(SmartSequenceBuilder.FACTORY);<a name="line.331"></a>
<span class="sourceLineNo">332</span>        FastaFormat fFormat = new FastaFormat();<a name="line.332"></a>
<span class="sourceLineNo">333</span>        for (SequenceIterator seqI = new StreamReader(seqFile,<a name="line.333"></a>
<span class="sourceLineNo">334</span>                                                      fFormat,<a name="line.334"></a>
<span class="sourceLineNo">335</span>                                                      alpha.getTokenization("token"),<a name="line.335"></a>
<span class="sourceLineNo">336</span>                                                      sbFact);seqI.hasNext();) {<a name="line.336"></a>
<span class="sourceLineNo">337</span>            Sequence seq = seqI.nextSequence();<a name="line.337"></a>
<span class="sourceLineNo">338</span>            try {<a name="line.338"></a>
<span class="sourceLineNo">339</span>                db.addSequence(seq);<a name="line.339"></a>
<span class="sourceLineNo">340</span>            } catch (ChangeVetoException cve) {<a name="line.340"></a>
<span class="sourceLineNo">341</span>                throw new AssertionFailure(<a name="line.341"></a>
<span class="sourceLineNo">342</span>                  "Could not successfully add sequence "<a name="line.342"></a>
<span class="sourceLineNo">343</span>                  + seq.getName()<a name="line.343"></a>
<span class="sourceLineNo">344</span>                  + " to sequence database",<a name="line.344"></a>
<span class="sourceLineNo">345</span>                  cve);<a name="line.345"></a>
<span class="sourceLineNo">346</span>            }<a name="line.346"></a>
<span class="sourceLineNo">347</span>        }<a name="line.347"></a>
<span class="sourceLineNo">348</span>        return db;<a name="line.348"></a>
<span class="sourceLineNo">349</span>    }<a name="line.349"></a>
<span class="sourceLineNo">350</span><a name="line.350"></a>
<span class="sourceLineNo">351</span>    /**<a name="line.351"></a>
<span class="sourceLineNo">352</span>     * Write a sequenceDB to an output stream in fasta format.<a name="line.352"></a>
<span class="sourceLineNo">353</span>     * @since 1.2<a name="line.353"></a>
<span class="sourceLineNo">354</span>     * @param os the stream to write the fasta formatted data to.<a name="line.354"></a>
<span class="sourceLineNo">355</span>     * @param db the database of &lt;code&gt;Sequence&lt;/code&gt;s to write<a name="line.355"></a>
<span class="sourceLineNo">356</span>     * @throws IOException if there was an error while writing.<a name="line.356"></a>
<span class="sourceLineNo">357</span>     */<a name="line.357"></a>
<span class="sourceLineNo">358</span>    public static void writeFasta(OutputStream os, SequenceDB db)<a name="line.358"></a>
<span class="sourceLineNo">359</span>        throws IOException {<a name="line.359"></a>
<span class="sourceLineNo">360</span>        StreamWriter sw = new StreamWriter(os,new FastaFormat());<a name="line.360"></a>
<span class="sourceLineNo">361</span>        sw.writeStream(db.sequenceIterator());<a name="line.361"></a>
<span class="sourceLineNo">362</span>    }<a name="line.362"></a>
<span class="sourceLineNo">363</span><a name="line.363"></a>
<span class="sourceLineNo">364</span>    /**<a name="line.364"></a>
<span class="sourceLineNo">365</span>     * Writes sequences from a SequenceIterator to an OutputStream in<a name="line.365"></a>
<span class="sourceLineNo">366</span>     * Fasta Format.  This makes for a useful format filter where a<a name="line.366"></a>
<span class="sourceLineNo">367</span>     * StreamReader can be sent to the StreamWriter after formatting.<a name="line.367"></a>
<span class="sourceLineNo">368</span>     * <a name="line.368"></a>
<span class="sourceLineNo">369</span>     * @since 1.2<a name="line.369"></a>
<span class="sourceLineNo">370</span>     * @param os The stream to write fasta formatted data to<a name="line.370"></a>
<span class="sourceLineNo">371</span>     * @param in The source of input &lt;code&gt;Sequences&lt;/code&gt;<a name="line.371"></a>
<span class="sourceLineNo">372</span>     * @throws IOException if there was an error while writing.<a name="line.372"></a>
<span class="sourceLineNo">373</span>     */<a name="line.373"></a>
<span class="sourceLineNo">374</span>    public static void writeFasta(OutputStream os, SequenceIterator in)<a name="line.374"></a>
<span class="sourceLineNo">375</span>        throws IOException {<a name="line.375"></a>
<span class="sourceLineNo">376</span>        StreamWriter sw = new StreamWriter(os,new FastaFormat());<a name="line.376"></a>
<span class="sourceLineNo">377</span>        sw.writeStream(in);<a name="line.377"></a>
<span class="sourceLineNo">378</span>    }<a name="line.378"></a>
<span class="sourceLineNo">379</span><a name="line.379"></a>
<span class="sourceLineNo">380</span>    /**<a name="line.380"></a>
<span class="sourceLineNo">381</span>     * Writes a single Sequence to an OutputStream in Fasta format.<a name="line.381"></a>
<span class="sourceLineNo">382</span>     *<a name="line.382"></a>
<span class="sourceLineNo">383</span>     * @param os  the OutputStream.<a name="line.383"></a>
<span class="sourceLineNo">384</span>     * @param seq  the Sequence.<a name="line.384"></a>
<span class="sourceLineNo">385</span>     * @throws IOException if there was an error while writing.<a name="line.385"></a>
<span class="sourceLineNo">386</span>     */<a name="line.386"></a>
<span class="sourceLineNo">387</span>    public static void writeFasta(OutputStream os, Sequence seq)<a name="line.387"></a>
<span class="sourceLineNo">388</span>        throws IOException {<a name="line.388"></a>
<span class="sourceLineNo">389</span>        writeFasta(os, new SingleSeqIterator(seq));<a name="line.389"></a>
<span class="sourceLineNo">390</span>    }<a name="line.390"></a>
<span class="sourceLineNo">391</span><a name="line.391"></a>
<span class="sourceLineNo">392</span>    /**<a name="line.392"></a>
<span class="sourceLineNo">393</span>     * Writes a stream of Sequences to an OutputStream in EMBL format.<a name="line.393"></a>
<span class="sourceLineNo">394</span>     *<a name="line.394"></a>
<span class="sourceLineNo">395</span>     * @param os the OutputStream.<a name="line.395"></a>
<span class="sourceLineNo">396</span>     * @param in a SequenceIterator.<a name="line.396"></a>
<span class="sourceLineNo">397</span>     * @exception IOException if there was an error while writing.<a name="line.397"></a>
<span class="sourceLineNo">398</span>     */<a name="line.398"></a>
<span class="sourceLineNo">399</span>    public static void writeEmbl(OutputStream os, SequenceIterator in)<a name="line.399"></a>
<span class="sourceLineNo">400</span>        throws IOException {<a name="line.400"></a>
<span class="sourceLineNo">401</span>        StreamWriter sw = new StreamWriter(os, new EmblLikeFormat());<a name="line.401"></a>
<span class="sourceLineNo">402</span>        sw.writeStream(in);<a name="line.402"></a>
<span class="sourceLineNo">403</span>    }<a name="line.403"></a>
<span class="sourceLineNo">404</span><a name="line.404"></a>
<span class="sourceLineNo">405</span>    /**<a name="line.405"></a>
<span class="sourceLineNo">406</span>     * Writes a single Sequence to an OutputStream in EMBL format.<a name="line.406"></a>
<span class="sourceLineNo">407</span>     *<a name="line.407"></a>
<span class="sourceLineNo">408</span>     * @param os  the OutputStream.<a name="line.408"></a>
<span class="sourceLineNo">409</span>     * @param seq  the Sequence.<a name="line.409"></a>
<span class="sourceLineNo">410</span>     * @throws IOException if there was an error while writing.<a name="line.410"></a>
<span class="sourceLineNo">411</span>     */<a name="line.411"></a>
<span class="sourceLineNo">412</span>    public static void writeEmbl(OutputStream os, Sequence seq) throws IOException {<a name="line.412"></a>
<span class="sourceLineNo">413</span>        writeEmbl(os, new SingleSeqIterator(seq));<a name="line.413"></a>
<span class="sourceLineNo">414</span>    }<a name="line.414"></a>
<span class="sourceLineNo">415</span><a name="line.415"></a>
<span class="sourceLineNo">416</span>    /**<a name="line.416"></a>
<span class="sourceLineNo">417</span>     * Writes a stream of Sequences to an OutputStream in SwissProt<a name="line.417"></a>
<span class="sourceLineNo">418</span>     * format.<a name="line.418"></a>
<span class="sourceLineNo">419</span>     * @param os the OutputStream.<a name="line.419"></a>
<span class="sourceLineNo">420</span>     * @param in a SequenceIterator.<a name="line.420"></a>
<span class="sourceLineNo">421</span>     * @throws org.biojava.bio.BioException if the &lt;CODE&gt;Sequence&lt;/CODE&gt; cannot be converted to SwissProt<a name="line.421"></a>
<span class="sourceLineNo">422</span>     * format<a name="line.422"></a>
<span class="sourceLineNo">423</span>     * @exception IOException if there was an error while writing.<a name="line.423"></a>
<span class="sourceLineNo">424</span>     */<a name="line.424"></a>
<span class="sourceLineNo">425</span>    public static void writeSwissprot(OutputStream os, SequenceIterator in)<a name="line.425"></a>
<span class="sourceLineNo">426</span>        throws IOException, BioException {<a name="line.426"></a>
<span class="sourceLineNo">427</span>        SequenceFormat former = new EmblLikeFormat();<a name="line.427"></a>
<span class="sourceLineNo">428</span>        PrintStream ps = new PrintStream(os);<a name="line.428"></a>
<span class="sourceLineNo">429</span>        while (in.hasNext()) {<a name="line.429"></a>
<span class="sourceLineNo">430</span>            former.writeSequence(in.nextSequence(), ps);<a name="line.430"></a>
<span class="sourceLineNo">431</span>        }<a name="line.431"></a>
<span class="sourceLineNo">432</span>    }<a name="line.432"></a>
<span class="sourceLineNo">433</span><a name="line.433"></a>
<span class="sourceLineNo">434</span>    /**<a name="line.434"></a>
<span class="sourceLineNo">435</span>     * Writes a single Sequence to an OutputStream in SwissProt format.<a name="line.435"></a>
<span class="sourceLineNo">436</span>     * @param os the OutputStream.<a name="line.436"></a>
<span class="sourceLineNo">437</span>     * @param seq the Sequence.<a name="line.437"></a>
<span class="sourceLineNo">438</span>     * @throws org.biojava.bio.BioException if the &lt;CODE&gt;Sequence&lt;/CODE&gt; cannot be written to SwissProt format<a name="line.438"></a>
<span class="sourceLineNo">439</span>     * @throws IOException if there was an error while writing.<a name="line.439"></a>
<span class="sourceLineNo">440</span>     */<a name="line.440"></a>
<span class="sourceLineNo">441</span>    public static void writeSwissprot(OutputStream os, Sequence seq)<a name="line.441"></a>
<span class="sourceLineNo">442</span>        throws IOException, BioException {<a name="line.442"></a>
<span class="sourceLineNo">443</span>        writeSwissprot(os, new SingleSeqIterator(seq));<a name="line.443"></a>
<span class="sourceLineNo">444</span>    }<a name="line.444"></a>
<span class="sourceLineNo">445</span><a name="line.445"></a>
<span class="sourceLineNo">446</span>    /**<a name="line.446"></a>
<span class="sourceLineNo">447</span>     * Writes a stream of Sequences to an OutputStream in Genpept<a name="line.447"></a>
<span class="sourceLineNo">448</span>     * format.<a name="line.448"></a>
<span class="sourceLineNo">449</span>     * @param os the OutputStream.<a name="line.449"></a>
<span class="sourceLineNo">450</span>     * @param in a SequenceIterator.<a name="line.450"></a>
<span class="sourceLineNo">451</span>     * @throws org.biojava.bio.BioException if the &lt;CODE&gt;Sequence&lt;/CODE&gt; cannot be written to Genpept format<a name="line.451"></a>
<span class="sourceLineNo">452</span>     * @exception IOException if there was an error while writing.<a name="line.452"></a>
<span class="sourceLineNo">453</span>     */<a name="line.453"></a>
<span class="sourceLineNo">454</span>    public static void writeGenpept(OutputStream os, SequenceIterator in)<a name="line.454"></a>
<span class="sourceLineNo">455</span>        throws IOException, BioException {<a name="line.455"></a>
<span class="sourceLineNo">456</span>        SequenceFormat former = new GenpeptFormat();<a name="line.456"></a>
<span class="sourceLineNo">457</span>        PrintStream ps = new PrintStream(os);<a name="line.457"></a>
<span class="sourceLineNo">458</span>        while (in.hasNext()) {<a name="line.458"></a>
<span class="sourceLineNo">459</span>            former.writeSequence(in.nextSequence(), ps);<a name="line.459"></a>
<span class="sourceLineNo">460</span>        }<a name="line.460"></a>
<span class="sourceLineNo">461</span>    }<a name="line.461"></a>
<span class="sourceLineNo">462</span><a name="line.462"></a>
<span class="sourceLineNo">463</span>    /**<a name="line.463"></a>
<span class="sourceLineNo">464</span>     * Writes a single Sequence to an OutputStream in Genpept format.<a name="line.464"></a>
<span class="sourceLineNo">465</span>     * @param os the OutputStream.<a name="line.465"></a>
<span class="sourceLineNo">466</span>     * @param seq the Sequence.<a name="line.466"></a>
<span class="sourceLineNo">467</span>     * @throws org.biojava.bio.BioException if the &lt;CODE&gt;Sequence&lt;/CODE&gt; cannot be written to Genpept format<a name="line.467"></a>
<span class="sourceLineNo">468</span>     * @throws IOException if there was an error while writing.<a name="line.468"></a>
<span class="sourceLineNo">469</span>     */<a name="line.469"></a>
<span class="sourceLineNo">470</span>    public static void writeGenpept(OutputStream os, Sequence seq)<a name="line.470"></a>
<span class="sourceLineNo">471</span>    throws IOException, BioException {<a name="line.471"></a>
<span class="sourceLineNo">472</span>      writeGenpept(os, new SingleSeqIterator(seq));<a name="line.472"></a>
<span class="sourceLineNo">473</span>    }<a name="line.473"></a>
<span class="sourceLineNo">474</span><a name="line.474"></a>
<span class="sourceLineNo">475</span>    /**<a name="line.475"></a>
<span class="sourceLineNo">476</span>     * Writes a stream of Sequences to an OutputStream in Genbank<a name="line.476"></a>
<span class="sourceLineNo">477</span>     * format.<a name="line.477"></a>
<span class="sourceLineNo">478</span>     *<a name="line.478"></a>
<span class="sourceLineNo">479</span>     * @param os the OutputStream.<a name="line.479"></a>
<span class="sourceLineNo">480</span>     * @param in a SequenceIterator.<a name="line.480"></a>
<span class="sourceLineNo">481</span>     * @exception IOException if there was an error while writing.<a name="line.481"></a>
<span class="sourceLineNo">482</span>     */<a name="line.482"></a>
<span class="sourceLineNo">483</span>    public static void writeGenbank(OutputStream os, SequenceIterator in)<a name="line.483"></a>
<span class="sourceLineNo">484</span>        throws IOException {<a name="line.484"></a>
<span class="sourceLineNo">485</span>        StreamWriter sw = new StreamWriter(os, new GenbankFormat());<a name="line.485"></a>
<span class="sourceLineNo">486</span>        sw.writeStream(in);<a name="line.486"></a>
<span class="sourceLineNo">487</span>    }<a name="line.487"></a>
<span class="sourceLineNo">488</span><a name="line.488"></a>
<span class="sourceLineNo">489</span>    /**<a name="line.489"></a>
<span class="sourceLineNo">490</span>     * Writes a single Sequence to an OutputStream in Genbank format.<a name="line.490"></a>
<span class="sourceLineNo">491</span>     *<a name="line.491"></a>
<span class="sourceLineNo">492</span>     * @param os  the OutputStream.<a name="line.492"></a>
<span class="sourceLineNo">493</span>     * @param seq  the Sequence.<a name="line.493"></a>
<span class="sourceLineNo">494</span>     * @throws IOException if there was an error while writing.<a name="line.494"></a>
<span class="sourceLineNo">495</span>     */<a name="line.495"></a>
<span class="sourceLineNo">496</span>    public static void writeGenbank(OutputStream os, Sequence seq)<a name="line.496"></a>
<span class="sourceLineNo">497</span>        throws IOException {<a name="line.497"></a>
<span class="sourceLineNo">498</span>        writeGenbank(os, new SingleSeqIterator(seq));<a name="line.498"></a>
<span class="sourceLineNo">499</span>    }<a name="line.499"></a>
<span class="sourceLineNo">500</span><a name="line.500"></a>
<span class="sourceLineNo">501</span>   /**<a name="line.501"></a>
<span class="sourceLineNo">502</span>     * &lt;code&gt;identifyFormat&lt;/code&gt; performs a case-insensitive mapping<a name="line.502"></a>
<span class="sourceLineNo">503</span>     * of a pair of common sequence format name (such as 'embl',<a name="line.503"></a>
<span class="sourceLineNo">504</span>     * 'genbank' or 'fasta') and alphabet name (such as 'dna', 'rna',<a name="line.504"></a>
<span class="sourceLineNo">505</span>     * 'protein', 'aa') to an integer. The value returned will be one<a name="line.505"></a>
<span class="sourceLineNo">506</span>     * of the public static final fields in<a name="line.506"></a>
<span class="sourceLineNo">507</span>     * &lt;code&gt;SeqIOConstants&lt;/code&gt;, or a bitwise-or combination of<a name="line.507"></a>
<span class="sourceLineNo">508</span>     * them. The method will reject known illegal combinations of<a name="line.508"></a>
<span class="sourceLineNo">509</span>     * format and alphabet (such as swissprot + dna) by throwing an<a name="line.509"></a>
<span class="sourceLineNo">510</span>     * &lt;code&gt;IllegalArgumentException&lt;/code&gt;. It will return the<a name="line.510"></a>
<span class="sourceLineNo">511</span>     * &lt;code&gt;SeqIOConstants.UNKNOWN&lt;/code&gt; value when either format or<a name="line.511"></a>
<span class="sourceLineNo">512</span>     * alphabet are unknown.<a name="line.512"></a>
<span class="sourceLineNo">513</span>     *<a name="line.513"></a>
<span class="sourceLineNo">514</span>     * @param formatName a &lt;code&gt;String&lt;/code&gt;.<a name="line.514"></a>
<span class="sourceLineNo">515</span>     * @param alphabetName a &lt;code&gt;String&lt;/code&gt;.<a name="line.515"></a>
<span class="sourceLineNo">516</span>     *<a name="line.516"></a>
<span class="sourceLineNo">517</span>     * @return an &lt;code&gt;int&lt;/code&gt;.<a name="line.517"></a>
<span class="sourceLineNo">518</span>     */<a name="line.518"></a>
<span class="sourceLineNo">519</span>    public static int identifyFormat(String formatName, String alphabetName) {<a name="line.519"></a>
<span class="sourceLineNo">520</span>        int format, alpha;<a name="line.520"></a>
<span class="sourceLineNo">521</span>        if (formatName.equalsIgnoreCase("raw")) {<a name="line.521"></a>
<span class="sourceLineNo">522</span>            format = SeqIOConstants.RAW;<a name="line.522"></a>
<span class="sourceLineNo">523</span>        }<a name="line.523"></a>
<span class="sourceLineNo">524</span>        else if (formatName.equalsIgnoreCase("fasta")) {<a name="line.524"></a>
<span class="sourceLineNo">525</span>            format = SeqIOConstants.FASTA;<a name="line.525"></a>
<span class="sourceLineNo">526</span>        }<a name="line.526"></a>
<span class="sourceLineNo">527</span>        else if (formatName.equalsIgnoreCase("nbrf")) {<a name="line.527"></a>
<span class="sourceLineNo">528</span>            format = SeqIOConstants.NBRF;<a name="line.528"></a>
<span class="sourceLineNo">529</span>        }<a name="line.529"></a>
<span class="sourceLineNo">530</span>        else if (formatName.equalsIgnoreCase("ig")) {<a name="line.530"></a>
<span class="sourceLineNo">531</span>            format = SeqIOConstants.IG;<a name="line.531"></a>
<span class="sourceLineNo">532</span>        }<a name="line.532"></a>
<span class="sourceLineNo">533</span>        else if (formatName.equalsIgnoreCase("embl")) {<a name="line.533"></a>
<span class="sourceLineNo">534</span>            format = SeqIOConstants.EMBL;<a name="line.534"></a>
<span class="sourceLineNo">535</span>        }<a name="line.535"></a>
<span class="sourceLineNo">536</span>        else if (formatName.equalsIgnoreCase("swissprot") ||<a name="line.536"></a>
<span class="sourceLineNo">537</span>                 formatName.equalsIgnoreCase("swiss")) {<a name="line.537"></a>
<span class="sourceLineNo">538</span>            if (alphabetName.equalsIgnoreCase("aa") ||<a name="line.538"></a>
<span class="sourceLineNo">539</span>                alphabetName.equalsIgnoreCase("protein")) {<a name="line.539"></a>
<span class="sourceLineNo">540</span>                return SeqIOConstants.SWISSPROT;<a name="line.540"></a>
<span class="sourceLineNo">541</span>            } else {<a name="line.541"></a>
<span class="sourceLineNo">542</span>                throw new IllegalArgumentException("Illegal format and alphabet "<a name="line.542"></a>
<span class="sourceLineNo">543</span>                                                   + "combination "<a name="line.543"></a>
<span class="sourceLineNo">544</span>                                                   + formatName<a name="line.544"></a>
<span class="sourceLineNo">545</span>                                                   + " + "<a name="line.545"></a>
<span class="sourceLineNo">546</span>                                                   + alphabetName);<a name="line.546"></a>
<span class="sourceLineNo">547</span>            }<a name="line.547"></a>
<span class="sourceLineNo">548</span>        } else if (formatName.equalsIgnoreCase("genbank")) {<a name="line.548"></a>
<span class="sourceLineNo">549</span>            format = SeqIOConstants.GENBANK;<a name="line.549"></a>
<span class="sourceLineNo">550</span>        } else if (formatName.equalsIgnoreCase("genpept")) {<a name="line.550"></a>
<span class="sourceLineNo">551</span>            if (alphabetName.equalsIgnoreCase("aa") ||<a name="line.551"></a>
<span class="sourceLineNo">552</span>                alphabetName.equalsIgnoreCase("protein")) {<a name="line.552"></a>
<span class="sourceLineNo">553</span>                return SeqIOConstants.GENPEPT;<a name="line.553"></a>
<span class="sourceLineNo">554</span>            } else {<a name="line.554"></a>
<span class="sourceLineNo">555</span>                throw new IllegalArgumentException("Illegal format and alphabet "<a name="line.555"></a>
<span class="sourceLineNo">556</span>                                                   + "combination "<a name="line.556"></a>
<span class="sourceLineNo">557</span>                                                   + formatName<a name="line.557"></a>
<span class="sourceLineNo">558</span>                                                   + " + "<a name="line.558"></a>
<span class="sourceLineNo">559</span>                                                   + alphabetName);<a name="line.559"></a>
<span class="sourceLineNo">560</span>            }<a name="line.560"></a>
<span class="sourceLineNo">561</span>        } else if (formatName.equalsIgnoreCase("refseq")) {<a name="line.561"></a>
<span class="sourceLineNo">562</span>            format = SeqIOConstants.REFSEQ;<a name="line.562"></a>
<span class="sourceLineNo">563</span>        } else if (formatName.equalsIgnoreCase("gcg")) {<a name="line.563"></a>
<span class="sourceLineNo">564</span>            format = SeqIOConstants.GCG;<a name="line.564"></a>
<span class="sourceLineNo">565</span>        } else if (formatName.equalsIgnoreCase("gff")) {<a name="line.565"></a>
<span class="sourceLineNo">566</span>            format = SeqIOConstants.GFF;<a name="line.566"></a>
<span class="sourceLineNo">567</span>        }<a name="line.567"></a>
<span class="sourceLineNo">568</span>        else if (formatName.equalsIgnoreCase("pdb")) {<a name="line.568"></a>
<span class="sourceLineNo">569</span>            if (alphabetName.equalsIgnoreCase("aa") ||<a name="line.569"></a>
<span class="sourceLineNo">570</span>                alphabetName.equalsIgnoreCase("protein")) {<a name="line.570"></a>
<span class="sourceLineNo">571</span>                return SeqIOConstants.PDB;<a name="line.571"></a>
<span class="sourceLineNo">572</span>            } else {<a name="line.572"></a>
<span class="sourceLineNo">573</span>                throw new IllegalArgumentException("Illegal format and alphabet "<a name="line.573"></a>
<span class="sourceLineNo">574</span>                                                   + "combination "<a name="line.574"></a>
<span class="sourceLineNo">575</span>                                                   + formatName<a name="line.575"></a>
<span class="sourceLineNo">576</span>                                                   + " + "<a name="line.576"></a>
<span class="sourceLineNo">577</span>                                                   + alphabetName);<a name="line.577"></a>
<span class="sourceLineNo">578</span>            }<a name="line.578"></a>
<span class="sourceLineNo">579</span>        } else if (formatName.equalsIgnoreCase("phred")) {<a name="line.579"></a>
<span class="sourceLineNo">580</span>            if (alphabetName.equalsIgnoreCase("dna")) {<a name="line.580"></a>
<span class="sourceLineNo">581</span>                return SeqIOConstants.PHRED;<a name="line.581"></a>
<span class="sourceLineNo">582</span>            } else {<a name="line.582"></a>
<span class="sourceLineNo">583</span>                throw new IllegalArgumentException("Illegal format and alphabet "<a name="line.583"></a>
<span class="sourceLineNo">584</span>                                                   + "combination "<a name="line.584"></a>
<span class="sourceLineNo">585</span>                                                   + formatName<a name="line.585"></a>
<span class="sourceLineNo">586</span>                                                   + " + "<a name="line.586"></a>
<span class="sourceLineNo">587</span>                                                   + alphabetName);<a name="line.587"></a>
<span class="sourceLineNo">588</span>            }<a name="line.588"></a>
<span class="sourceLineNo">589</span>        } else if (formatName.equalsIgnoreCase("clustal")) {<a name="line.589"></a>
<span class="sourceLineNo">590</span>            format = AlignIOConstants.CLUSTAL;<a name="line.590"></a>
<span class="sourceLineNo">591</span>        } else if (formatName.equalsIgnoreCase("msf")) {<a name="line.591"></a>
<span class="sourceLineNo">592</span>            format = AlignIOConstants.MSF;<a name="line.592"></a>
<span class="sourceLineNo">593</span>        }<a name="line.593"></a>
<span class="sourceLineNo">594</span>        else {<a name="line.594"></a>
<span class="sourceLineNo">595</span>            return SeqIOConstants.UNKNOWN;<a name="line.595"></a>
<span class="sourceLineNo">596</span>        }<a name="line.596"></a>
<span class="sourceLineNo">597</span><a name="line.597"></a>
<span class="sourceLineNo">598</span>        if (alphabetName.equalsIgnoreCase("dna")) {<a name="line.598"></a>
<span class="sourceLineNo">599</span>            alpha = SeqIOConstants.DNA;<a name="line.599"></a>
<span class="sourceLineNo">600</span>        } else if (alphabetName.equalsIgnoreCase("rna")) {<a name="line.600"></a>
<span class="sourceLineNo">601</span>            alpha = SeqIOConstants.RNA;<a name="line.601"></a>
<span class="sourceLineNo">602</span>        } else if (alphabetName.equalsIgnoreCase("aa") ||<a name="line.602"></a>
<span class="sourceLineNo">603</span>                 alphabetName.equalsIgnoreCase("protein")) {<a name="line.603"></a>
<span class="sourceLineNo">604</span>            alpha = SeqIOConstants.AA;<a name="line.604"></a>
<span class="sourceLineNo">605</span>        } else {<a name="line.605"></a>
<span class="sourceLineNo">606</span>            return SeqIOConstants.UNKNOWN;<a name="line.606"></a>
<span class="sourceLineNo">607</span>        }<a name="line.607"></a>
<span class="sourceLineNo">608</span><a name="line.608"></a>
<span class="sourceLineNo">609</span>        return (format | alpha);<a name="line.609"></a>
<span class="sourceLineNo">610</span>    }<a name="line.610"></a>
<span class="sourceLineNo">611</span><a name="line.611"></a>
<span class="sourceLineNo">612</span>    /**<a name="line.612"></a>
<span class="sourceLineNo">613</span>     * &lt;code&gt;getSequenceFormat&lt;/code&gt; accepts a value which represents<a name="line.613"></a>
<span class="sourceLineNo">614</span>     * a sequence format and returns the relevant<a name="line.614"></a>
<span class="sourceLineNo">615</span>     * &lt;code&gt;SequenceFormat&lt;/code&gt; object.<a name="line.615"></a>
<span class="sourceLineNo">616</span>     *<a name="line.616"></a>
<span class="sourceLineNo">617</span>     * @param identifier an &lt;code&gt;int&lt;/code&gt; which represents a binary<a name="line.617"></a>
<span class="sourceLineNo">618</span>     * value with bits set according to the scheme described in<a name="line.618"></a>
<span class="sourceLineNo">619</span>     * &lt;code&gt;SeqIOConstants&lt;/code&gt;.<a name="line.619"></a>
<span class="sourceLineNo">620</span>     *<a name="line.620"></a>
<span class="sourceLineNo">621</span>     * @return a &lt;code&gt;SequenceFormat&lt;/code&gt;.<a name="line.621"></a>
<span class="sourceLineNo">622</span>     *<a name="line.622"></a>
<span class="sourceLineNo">623</span>     * @exception BioException if an error occurs.<a name="line.623"></a>
<span class="sourceLineNo">624</span>     */<a name="line.624"></a>
<span class="sourceLineNo">625</span>    public static SequenceFormat getSequenceFormat(int identifier)<a name="line.625"></a>
<span class="sourceLineNo">626</span>        throws BioException {<a name="line.626"></a>
<span class="sourceLineNo">627</span><a name="line.627"></a>
<span class="sourceLineNo">628</span>        // Mask the sequence format bytes<a name="line.628"></a>
<span class="sourceLineNo">629</span>        int alphaType = identifier &amp; (~ 0xffff);<a name="line.629"></a>
<span class="sourceLineNo">630</span>        if (alphaType == 0)<a name="line.630"></a>
<span class="sourceLineNo">631</span>            throw new IllegalArgumentException("No alphabet was set in the identifier");<a name="line.631"></a>
<span class="sourceLineNo">632</span><a name="line.632"></a>
<span class="sourceLineNo">633</span>        // Mask alphabet bytes<a name="line.633"></a>
<span class="sourceLineNo">634</span>        int formatType = identifier &amp; (~ 0xffff0000);<a name="line.634"></a>
<span class="sourceLineNo">635</span>        if (formatType == 0)<a name="line.635"></a>
<span class="sourceLineNo">636</span>            throw new IllegalArgumentException("No format was set in the identifier");<a name="line.636"></a>
<span class="sourceLineNo">637</span><a name="line.637"></a>
<span class="sourceLineNo">638</span>        switch (identifier) {<a name="line.638"></a>
<span class="sourceLineNo">639</span>            case SeqIOConstants.FASTA_DNA:<a name="line.639"></a>
<span class="sourceLineNo">640</span>            case SeqIOConstants.FASTA_RNA:<a name="line.640"></a>
<span class="sourceLineNo">641</span>            case SeqIOConstants.FASTA_AA:<a name="line.641"></a>
<span class="sourceLineNo">642</span>                return new FastaFormat();<a name="line.642"></a>
<span class="sourceLineNo">643</span>            case SeqIOConstants.EMBL_DNA:<a name="line.643"></a>
<span class="sourceLineNo">644</span>            case SeqIOConstants.EMBL_RNA:<a name="line.644"></a>
<span class="sourceLineNo">645</span>                return new EmblLikeFormat();<a name="line.645"></a>
<span class="sourceLineNo">646</span>            case SeqIOConstants.GENBANK_DNA:<a name="line.646"></a>
<span class="sourceLineNo">647</span>            case SeqIOConstants.GENBANK_RNA:<a name="line.647"></a>
<span class="sourceLineNo">648</span>                return new GenbankFormat();<a name="line.648"></a>
<span class="sourceLineNo">649</span>            case SeqIOConstants.SWISSPROT:<a name="line.649"></a>
<span class="sourceLineNo">650</span>                return new EmblLikeFormat();<a name="line.650"></a>
<span class="sourceLineNo">651</span>            default:<a name="line.651"></a>
<span class="sourceLineNo">652</span>                throw new BioException("No SequenceFormat available for "<a name="line.652"></a>
<span class="sourceLineNo">653</span>                                       + "format/alphabet identifier '"<a name="line.653"></a>
<span class="sourceLineNo">654</span>                                       + identifier<a name="line.654"></a>
<span class="sourceLineNo">655</span>                                       + "'");<a name="line.655"></a>
<span class="sourceLineNo">656</span>        }<a name="line.656"></a>
<span class="sourceLineNo">657</span>    }<a name="line.657"></a>
<span class="sourceLineNo">658</span><a name="line.658"></a>
<span class="sourceLineNo">659</span>    /**<a name="line.659"></a>
<span class="sourceLineNo">660</span>     * &lt;code&gt;getBuilderFactory&lt;/code&gt; accepts a value which represents<a name="line.660"></a>
<span class="sourceLineNo">661</span>     * a sequence format and returns the relevant<a name="line.661"></a>
<span class="sourceLineNo">662</span>     * &lt;code&gt;SequenceBuilderFactory&lt;/code&gt; object.<a name="line.662"></a>
<span class="sourceLineNo">663</span>     *<a name="line.663"></a>
<span class="sourceLineNo">664</span>     * @param identifier an &lt;code&gt;int&lt;/code&gt; which represents a binary<a name="line.664"></a>
<span class="sourceLineNo">665</span>     * value with bits set according to the scheme described in<a name="line.665"></a>
<span class="sourceLineNo">666</span>     * &lt;code&gt;SeqIOConstants&lt;/code&gt;.<a name="line.666"></a>
<span class="sourceLineNo">667</span>     *<a name="line.667"></a>
<span class="sourceLineNo">668</span>     * @return a &lt;code&gt;SequenceBuilderFactory&lt;/code&gt;.<a name="line.668"></a>
<span class="sourceLineNo">669</span>     *<a name="line.669"></a>
<span class="sourceLineNo">670</span>     * @exception BioException if an error occurs.<a name="line.670"></a>
<span class="sourceLineNo">671</span>     */<a name="line.671"></a>
<span class="sourceLineNo">672</span>    public static SequenceBuilderFactory getBuilderFactory(int identifier)<a name="line.672"></a>
<span class="sourceLineNo">673</span>        throws BioException {<a name="line.673"></a>
<span class="sourceLineNo">674</span><a name="line.674"></a>
<span class="sourceLineNo">675</span>        // Mask the sequence format bytes<a name="line.675"></a>
<span class="sourceLineNo">676</span>        int alphaType = identifier &amp; (~ 0xffff);<a name="line.676"></a>
<span class="sourceLineNo">677</span>        if (alphaType == 0)<a name="line.677"></a>
<span class="sourceLineNo">678</span>            throw new IllegalArgumentException("No alphabet was set in the identifier");<a name="line.678"></a>
<span class="sourceLineNo">679</span><a name="line.679"></a>
<span class="sourceLineNo">680</span>        // Mask alphabet bytes<a name="line.680"></a>
<span class="sourceLineNo">681</span>        int formatType = identifier &amp; (~ 0xffff0000);<a name="line.681"></a>
<span class="sourceLineNo">682</span>        if (formatType == 0)<a name="line.682"></a>
<span class="sourceLineNo">683</span>            throw new IllegalArgumentException("No format was set in the identifier");<a name="line.683"></a>
<span class="sourceLineNo">684</span><a name="line.684"></a>
<span class="sourceLineNo">685</span>        switch (identifier) {<a name="line.685"></a>
<span class="sourceLineNo">686</span>            case SeqIOConstants.FASTA_DNA:<a name="line.686"></a>
<span class="sourceLineNo">687</span>            case SeqIOConstants.FASTA_RNA:<a name="line.687"></a>
<span class="sourceLineNo">688</span>            case SeqIOConstants.FASTA_AA:<a name="line.688"></a>
<span class="sourceLineNo">689</span>                return getFastaBuilderFactory();<a name="line.689"></a>
<span class="sourceLineNo">690</span>            case SeqIOConstants.EMBL_DNA:<a name="line.690"></a>
<span class="sourceLineNo">691</span>                    return getEmblBuilderFactory();<a name="line.691"></a>
<span class="sourceLineNo">692</span>            case SeqIOConstants.GENBANK_DNA:<a name="line.692"></a>
<span class="sourceLineNo">693</span>                return getGenbankBuilderFactory();<a name="line.693"></a>
<span class="sourceLineNo">694</span>            case SeqIOConstants.SWISSPROT:<a name="line.694"></a>
<span class="sourceLineNo">695</span>                return getSwissprotBuilderFactory();<a name="line.695"></a>
<span class="sourceLineNo">696</span>            case SeqIOConstants.GENPEPT:<a name="line.696"></a>
<span class="sourceLineNo">697</span>                return getGenpeptBuilderFactory();<a name="line.697"></a>
<span class="sourceLineNo">698</span>            default:<a name="line.698"></a>
<span class="sourceLineNo">699</span>                throw new BioException("No SequenceBuilderFactory available for "<a name="line.699"></a>
<span class="sourceLineNo">700</span>                                       + "format/alphabet identifier '"<a name="line.700"></a>
<span class="sourceLineNo">701</span>                                       + identifier<a name="line.701"></a>
<span class="sourceLineNo">702</span>                                       + "'");<a name="line.702"></a>
<span class="sourceLineNo">703</span>        }<a name="line.703"></a>
<span class="sourceLineNo">704</span>    }<a name="line.704"></a>
<span class="sourceLineNo">705</span><a name="line.705"></a>
<span class="sourceLineNo">706</span>    /**<a name="line.706"></a>
<span class="sourceLineNo">707</span>     * &lt;code&gt;getAlphabet&lt;/code&gt; accepts a value which represents a<a name="line.707"></a>
<span class="sourceLineNo">708</span>     * sequence format and returns the relevant<a name="line.708"></a>
<span class="sourceLineNo">709</span>     * &lt;code&gt;FiniteAlphabet&lt;/code&gt; object.<a name="line.709"></a>
<span class="sourceLineNo">710</span>     *<a name="line.710"></a>
<span class="sourceLineNo">711</span>     * @param identifier an &lt;code&gt;int&lt;/code&gt; which represents a binary<a name="line.711"></a>
<span class="sourceLineNo">712</span>     * value with bits set according to the scheme described in<a name="line.712"></a>
<span class="sourceLineNo">713</span>     * &lt;code&gt;SeqIOConstants&lt;/code&gt;.<a name="line.713"></a>
<span class="sourceLineNo">714</span>     *<a name="line.714"></a>
<span class="sourceLineNo">715</span>     * @return a &lt;code&gt;FiniteAlphabet&lt;/code&gt;.<a name="line.715"></a>
<span class="sourceLineNo">716</span>     *<a name="line.716"></a>
<span class="sourceLineNo">717</span>     * @exception BioException if an error occurs.<a name="line.717"></a>
<span class="sourceLineNo">718</span>     */<a name="line.718"></a>
<span class="sourceLineNo">719</span>    public static FiniteAlphabet getAlphabet(int identifier)<a name="line.719"></a>
<span class="sourceLineNo">720</span>        throws BioException {<a name="line.720"></a>
<span class="sourceLineNo">721</span><a name="line.721"></a>
<span class="sourceLineNo">722</span>        // Mask the sequence format bytes<a name="line.722"></a>
<span class="sourceLineNo">723</span>        int alphaType = identifier &amp; (~ 0xffff);<a name="line.723"></a>
<span class="sourceLineNo">724</span>        if (alphaType == 0)<a name="line.724"></a>
<span class="sourceLineNo">725</span>            throw new IllegalArgumentException("No alphabet was set in the identifier");<a name="line.725"></a>
<span class="sourceLineNo">726</span><a name="line.726"></a>
<span class="sourceLineNo">727</span>        switch (alphaType) {<a name="line.727"></a>
<span class="sourceLineNo">728</span>            case SeqIOConstants.DNA:<a name="line.728"></a>
<span class="sourceLineNo">729</span>                return DNATools.getDNA();<a name="line.729"></a>
<span class="sourceLineNo">730</span>            case SeqIOConstants.RNA:<a name="line.730"></a>
<span class="sourceLineNo">731</span>                return RNATools.getRNA();<a name="line.731"></a>
<span class="sourceLineNo">732</span>            case SeqIOConstants.AA:<a name="line.732"></a>
<span class="sourceLineNo">733</span>                return ProteinTools.getTAlphabet();<a name="line.733"></a>
<span class="sourceLineNo">734</span>            default:<a name="line.734"></a>
<span class="sourceLineNo">735</span>                throw new BioException("No FiniteAlphabet available for "<a name="line.735"></a>
<span class="sourceLineNo">736</span>                                       + "alphabet identifier '"<a name="line.736"></a>
<span class="sourceLineNo">737</span>                                       + identifier<a name="line.737"></a>
<span class="sourceLineNo">738</span>                                       + "'");<a name="line.738"></a>
<span class="sourceLineNo">739</span>        }<a name="line.739"></a>
<span class="sourceLineNo">740</span>    }<a name="line.740"></a>
<span class="sourceLineNo">741</span><a name="line.741"></a>
<span class="sourceLineNo">742</span>    //<a name="line.742"></a>
<span class="sourceLineNo">743</span>    // The following methods provide an alternate interface for<a name="line.743"></a>
<span class="sourceLineNo">744</span>    // reading and writing sequences and alignments. (Nimesh Singh).<a name="line.744"></a>
<span class="sourceLineNo">745</span>    //<a name="line.745"></a>
<span class="sourceLineNo">746</span>    //<a name="line.746"></a>
<span class="sourceLineNo">747</span><a name="line.747"></a>
<span class="sourceLineNo">748</span>    /**<a name="line.748"></a>
<span class="sourceLineNo">749</span>     * Attempts to guess the filetype of a file given the name.  For<a name="line.749"></a>
<span class="sourceLineNo">750</span>     * use with the functions below that take an int fileType as a<a name="line.750"></a>
<span class="sourceLineNo">751</span>     * parameter. EMBL and Genbank files are assumed to contain DNA<a name="line.751"></a>
<span class="sourceLineNo">752</span>     * sequence.<a name="line.752"></a>
<span class="sourceLineNo">753</span>     * @deprecated because there is no standard file naming convention<a name="line.753"></a>
<span class="sourceLineNo">754</span>     * and guessing by file name is inherantly error prone and bad.<a name="line.754"></a>
<span class="sourceLineNo">755</span>     * @param seqFile the &lt;CODE&gt;File&lt;/CODE&gt; to read from.<a name="line.755"></a>
<span class="sourceLineNo">756</span>     * @throws java.io.IOException if &lt;CODE&gt;seqFile&lt;/CODE&gt; cannot be read<a name="line.756"></a>
<span class="sourceLineNo">757</span>     * @throws java.io.FileNotFoundException if &lt;CODE&gt;seqFile&lt;/CODE&gt; cannot be found<a name="line.757"></a>
<span class="sourceLineNo">758</span>     * @return a value that describes the file type.<a name="line.758"></a>
<span class="sourceLineNo">759</span>     */<a name="line.759"></a>
<span class="sourceLineNo">760</span>    public static int guessFileType(File seqFile)<a name="line.760"></a>
<span class="sourceLineNo">761</span>        throws IOException, FileNotFoundException {<a name="line.761"></a>
<span class="sourceLineNo">762</span>        //First tries by matching an extension<a name="line.762"></a>
<span class="sourceLineNo">763</span>        String fileName = seqFile.getName();<a name="line.763"></a>
<span class="sourceLineNo">764</span>        try {<a name="line.764"></a>
<span class="sourceLineNo">765</span>            if (Pattern.matches(".*\\u002eem.*", fileName)) {<a name="line.765"></a>
<span class="sourceLineNo">766</span>                return SeqIOConstants.EMBL_DNA;<a name="line.766"></a>
<span class="sourceLineNo">767</span>            }<a name="line.767"></a>
<span class="sourceLineNo">768</span>            else if (Pattern.matches(".*\\u002edat.*", fileName)) {<a name="line.768"></a>
<span class="sourceLineNo">769</span>                return SeqIOConstants.EMBL_DNA;<a name="line.769"></a>
<span class="sourceLineNo">770</span>            }<a name="line.770"></a>
<span class="sourceLineNo">771</span>            else if (Pattern.matches(".*\\u002egb.*", fileName)) {<a name="line.771"></a>
<span class="sourceLineNo">772</span>                return SeqIOConstants.GENBANK_DNA;<a name="line.772"></a>
<span class="sourceLineNo">773</span>            }<a name="line.773"></a>
<span class="sourceLineNo">774</span>            else if (Pattern.matches(".*\\u002esp.*", fileName)) {<a name="line.774"></a>
<span class="sourceLineNo">775</span>                return SeqIOConstants.SWISSPROT;<a name="line.775"></a>
<span class="sourceLineNo">776</span>            }<a name="line.776"></a>
<span class="sourceLineNo">777</span>            else if (Pattern.matches(".*\\u002egp.*", fileName)) {<a name="line.777"></a>
<span class="sourceLineNo">778</span>                return SeqIOConstants.GENPEPT;<a name="line.778"></a>
<span class="sourceLineNo">779</span>            }<a name="line.779"></a>
<span class="sourceLineNo">780</span>            else if (Pattern.matches(".*\\u002efa.*", fileName)) {<a name="line.780"></a>
<span class="sourceLineNo">781</span>                return guessFastaType(seqFile);<a name="line.781"></a>
<span class="sourceLineNo">782</span>            }<a name="line.782"></a>
<span class="sourceLineNo">783</span>            else if (Pattern.matches(".*\\u002emsf.*", fileName)) {<a name="line.783"></a>
<span class="sourceLineNo">784</span>                return guessMsfType(seqFile);<a name="line.784"></a>
<span class="sourceLineNo">785</span>            }<a name="line.785"></a>
<span class="sourceLineNo">786</span>        } catch (PatternSyntaxException e) {<a name="line.786"></a>
<span class="sourceLineNo">787</span>            throw new BioError("Internal error in SeqIOTools", e);<a name="line.787"></a>
<span class="sourceLineNo">788</span>        }<a name="line.788"></a>
<span class="sourceLineNo">789</span><a name="line.789"></a>
<span class="sourceLineNo">790</span>        //Reads the file to guess based on content<a name="line.790"></a>
<span class="sourceLineNo">791</span>        BufferedReader br = new BufferedReader(new FileReader(seqFile));<a name="line.791"></a>
<span class="sourceLineNo">792</span>        String line1 = br.readLine();<a name="line.792"></a>
<span class="sourceLineNo">793</span>        br.close();<a name="line.793"></a>
<span class="sourceLineNo">794</span><a name="line.794"></a>
<span class="sourceLineNo">795</span>        if (line1.startsWith("&gt;")) {<a name="line.795"></a>
<span class="sourceLineNo">796</span>            return guessFastaType(seqFile);<a name="line.796"></a>
<span class="sourceLineNo">797</span>        }<a name="line.797"></a>
<span class="sourceLineNo">798</span>        else if (line1.startsWith("PileUp")) {<a name="line.798"></a>
<span class="sourceLineNo">799</span>            return guessMsfType(seqFile);<a name="line.799"></a>
<span class="sourceLineNo">800</span>        }<a name="line.800"></a>
<span class="sourceLineNo">801</span>        else if (line1.startsWith("!!AA_MULTIPLE_ALIGNMENT")) {<a name="line.801"></a>
<span class="sourceLineNo">802</span>            return AlignIOConstants.MSF_AA;<a name="line.802"></a>
<span class="sourceLineNo">803</span>        }<a name="line.803"></a>
<span class="sourceLineNo">804</span>        else if (line1.startsWith("!!NA_MULTIPLE_ALIGNMENT")) {<a name="line.804"></a>
<span class="sourceLineNo">805</span>            return AlignIOConstants.MSF_DNA;<a name="line.805"></a>
<span class="sourceLineNo">806</span>        }<a name="line.806"></a>
<span class="sourceLineNo">807</span>        else if (line1.startsWith("ID")) {<a name="line.807"></a>
<span class="sourceLineNo">808</span>            for (int i = 0; i &lt; line1.length(); i++) {<a name="line.808"></a>
<span class="sourceLineNo">809</span>                if (Character.toUpperCase(line1.charAt(i)) == 'P' &amp;&amp;<a name="line.809"></a>
<span class="sourceLineNo">810</span>                    Character.toUpperCase(line1.charAt(i+1)) == 'R' &amp;&amp;<a name="line.810"></a>
<span class="sourceLineNo">811</span>                    Character.toUpperCase(line1.charAt(i+2)) == 'T') {<a name="line.811"></a>
<span class="sourceLineNo">812</span>                    return SeqIOConstants.SWISSPROT;<a name="line.812"></a>
<span class="sourceLineNo">813</span>                }<a name="line.813"></a>
<span class="sourceLineNo">814</span>            }<a name="line.814"></a>
<span class="sourceLineNo">815</span>            return SeqIOConstants.EMBL_DNA;<a name="line.815"></a>
<span class="sourceLineNo">816</span>        }<a name="line.816"></a>
<span class="sourceLineNo">817</span>        else if (line1.toUpperCase().startsWith("LOCUS")) {<a name="line.817"></a>
<span class="sourceLineNo">818</span>            for (int i = 0; i &lt; line1.length(); i++) {<a name="line.818"></a>
<span class="sourceLineNo">819</span>                if (Character.toUpperCase(line1.charAt(i)) == 'A' &amp;&amp;<a name="line.819"></a>
<span class="sourceLineNo">820</span>                    Character.toUpperCase(line1.charAt(i+1)) == 'A') {<a name="line.820"></a>
<span class="sourceLineNo">821</span>                    return SeqIOConstants.GENPEPT;<a name="line.821"></a>
<span class="sourceLineNo">822</span>                }<a name="line.822"></a>
<span class="sourceLineNo">823</span>            }<a name="line.823"></a>
<span class="sourceLineNo">824</span>            return SeqIOConstants.GENBANK_DNA;<a name="line.824"></a>
<span class="sourceLineNo">825</span>        }<a name="line.825"></a>
<span class="sourceLineNo">826</span>        else if (line1.length() &gt;= 45 &amp;&amp;<a name="line.826"></a>
<span class="sourceLineNo">827</span>                 line1.substring(19, 45).equalsIgnoreCase("GENETIC SEQUENCE DATA BANK")) {<a name="line.827"></a>
<span class="sourceLineNo">828</span>            return guessGenType(fileName);<a name="line.828"></a>
<span class="sourceLineNo">829</span>        }<a name="line.829"></a>
<span class="sourceLineNo">830</span>        else {<a name="line.830"></a>
<span class="sourceLineNo">831</span>            return SeqIOConstants.UNKNOWN;<a name="line.831"></a>
<span class="sourceLineNo">832</span>        }<a name="line.832"></a>
<span class="sourceLineNo">833</span>    }<a name="line.833"></a>
<span class="sourceLineNo">834</span><a name="line.834"></a>
<span class="sourceLineNo">835</span>    /**<a name="line.835"></a>
<span class="sourceLineNo">836</span>     * Attempts to retrieve the most appropriate<a name="line.836"></a>
<span class="sourceLineNo">837</span>     * &lt;code&gt;SequenceBuilder&lt;/code&gt; object for some combination of<a name="line.837"></a>
<span class="sourceLineNo">838</span>     * &lt;code&gt;Alphabet&lt;/code&gt; and &lt;code&gt;SequenceFormat&lt;/code&gt;<a name="line.838"></a>
<span class="sourceLineNo">839</span>     *<a name="line.839"></a>
<span class="sourceLineNo">840</span>     * @param format currently supports &lt;code&gt;FastaFormat&lt;/code&gt;,<a name="line.840"></a>
<span class="sourceLineNo">841</span>     * &lt;code&gt;GenbankFormat&lt;/code&gt;, &lt;code&gt;EmblLikeFormat&lt;/code&gt;<a name="line.841"></a>
<span class="sourceLineNo">842</span>     * @param alpha currently only supports the DNA and Protein<a name="line.842"></a>
<span class="sourceLineNo">843</span>     * alphabets<a name="line.843"></a>
<span class="sourceLineNo">844</span>     *<a name="line.844"></a>
<span class="sourceLineNo">845</span>     * @return the &lt;code&gt;SequenceBuilderFactory&lt;/code&gt;<a name="line.845"></a>
<span class="sourceLineNo">846</span>     *<a name="line.846"></a>
<span class="sourceLineNo">847</span>     * @throws BioException if the combination of alpha and format is<a name="line.847"></a>
<span class="sourceLineNo">848</span>     * unrecognized.<a name="line.848"></a>
<span class="sourceLineNo">849</span>     *<a name="line.849"></a>
<span class="sourceLineNo">850</span>     * @deprecated as this essentially duplicates the operation<a name="line.850"></a>
<span class="sourceLineNo">851</span>     * available in the method &lt;code&gt;identifyBuilderFactory&lt;/code&gt;.<a name="line.851"></a>
<span class="sourceLineNo">852</span>     */<a name="line.852"></a>
<span class="sourceLineNo">853</span>    public static SequenceBuilderFactory formatToFactory(SequenceFormat format,<a name="line.853"></a>
<span class="sourceLineNo">854</span>                                                         Alphabet alpha)<a name="line.854"></a>
<span class="sourceLineNo">855</span>        throws BioException {<a name="line.855"></a>
<span class="sourceLineNo">856</span><a name="line.856"></a>
<span class="sourceLineNo">857</span>        if ((format instanceof FastaFormat) &amp;&amp;<a name="line.857"></a>
<span class="sourceLineNo">858</span>           (alpha == DNATools.getDNA() ||<a name="line.858"></a>
<span class="sourceLineNo">859</span>            alpha == ProteinTools.getAlphabet())) {<a name="line.859"></a>
<span class="sourceLineNo">860</span><a name="line.860"></a>
<span class="sourceLineNo">861</span>            return getFastaBuilderFactory();<a name="line.861"></a>
<span class="sourceLineNo">862</span>        }<a name="line.862"></a>
<span class="sourceLineNo">863</span>        else if (format instanceof GenbankFormat &amp;&amp;<a name="line.863"></a>
<span class="sourceLineNo">864</span>                alpha == DNATools.getDNA()) {<a name="line.864"></a>
<span class="sourceLineNo">865</span><a name="line.865"></a>
<span class="sourceLineNo">866</span>            return getGenbankBuilderFactory();<a name="line.866"></a>
<span class="sourceLineNo">867</span>        }<a name="line.867"></a>
<span class="sourceLineNo">868</span>        else if (format instanceof GenbankFormat &amp;&amp;<a name="line.868"></a>
<span class="sourceLineNo">869</span>                 alpha == ProteinTools.getAlphabet()) {<a name="line.869"></a>
<span class="sourceLineNo">870</span>            return getGenpeptBuilderFactory();<a name="line.870"></a>
<span class="sourceLineNo">871</span>        }<a name="line.871"></a>
<span class="sourceLineNo">872</span>        else if (format instanceof EmblLikeFormat &amp;&amp;<a name="line.872"></a>
<span class="sourceLineNo">873</span>                 alpha == DNATools.getDNA()){<a name="line.873"></a>
<span class="sourceLineNo">874</span>            return getEmblBuilderFactory();<a name="line.874"></a>
<span class="sourceLineNo">875</span>        }<a name="line.875"></a>
<span class="sourceLineNo">876</span>        else if (format instanceof EmblLikeFormat &amp;&amp;<a name="line.876"></a>
<span class="sourceLineNo">877</span>                 alpha == ProteinTools.getAlphabet()) {<a name="line.877"></a>
<span class="sourceLineNo">878</span>            return getSwissprotBuilderFactory();<a name="line.878"></a>
<span class="sourceLineNo">879</span>        }<a name="line.879"></a>
<span class="sourceLineNo">880</span>        else {<a name="line.880"></a>
<span class="sourceLineNo">881</span>            throw new BioException("Unknown combination of"<a name="line.881"></a>
<span class="sourceLineNo">882</span>                                   + " Alphabet and Format");<a name="line.882"></a>
<span class="sourceLineNo">883</span>      }<a name="line.883"></a>
<span class="sourceLineNo">884</span>    }<a name="line.884"></a>
<span class="sourceLineNo">885</span><a name="line.885"></a>
<span class="sourceLineNo">886</span>    /**<a name="line.886"></a>
<span class="sourceLineNo">887</span>     * Reads a file with the specified format and alphabet<a name="line.887"></a>
<span class="sourceLineNo">888</span>     * @param formatName the name of the format eg genbank or<a name="line.888"></a>
<span class="sourceLineNo">889</span>     * swissprot (case insensitive)<a name="line.889"></a>
<span class="sourceLineNo">890</span>     * @param alphabetName the name of the alphabet eg dna or rna or<a name="line.890"></a>
<span class="sourceLineNo">891</span>     * protein (case insensitive)<a name="line.891"></a>
<span class="sourceLineNo">892</span>     * @param br a BufferedReader for the input<a name="line.892"></a>
<span class="sourceLineNo">893</span>     * @return either an Alignment object or a SequenceIterator<a name="line.893"></a>
<span class="sourceLineNo">894</span>     * (depending on the format read)<a name="line.894"></a>
<span class="sourceLineNo">895</span>     * @throws BioException if an error occurs while reading or a<a name="line.895"></a>
<span class="sourceLineNo">896</span>     * unrecognized format, alphabet combination is used (eg swissprot<a name="line.896"></a>
<span class="sourceLineNo">897</span>     * and DNA).<a name="line.897"></a>
<span class="sourceLineNo">898</span>     *<a name="line.898"></a>
<span class="sourceLineNo">899</span>     * @since 1.3<a name="line.899"></a>
<span class="sourceLineNo">900</span>     */<a name="line.900"></a>
<span class="sourceLineNo">901</span>    public static Object fileToBiojava(String formatName,<a name="line.901"></a>
<span class="sourceLineNo">902</span>                                       String alphabetName,<a name="line.902"></a>
<span class="sourceLineNo">903</span>                                       BufferedReader br)<a name="line.903"></a>
<span class="sourceLineNo">904</span>        throws BioException {<a name="line.904"></a>
<span class="sourceLineNo">905</span><a name="line.905"></a>
<span class="sourceLineNo">906</span>        int fileType = identifyFormat(formatName, alphabetName);<a name="line.906"></a>
<span class="sourceLineNo">907</span><a name="line.907"></a>
<span class="sourceLineNo">908</span>        return fileToBiojava(fileType, br);<a name="line.908"></a>
<span class="sourceLineNo">909</span>    }<a name="line.909"></a>
<span class="sourceLineNo">910</span><a name="line.910"></a>
<span class="sourceLineNo">911</span>    /**<a name="line.911"></a>
<span class="sourceLineNo">912</span>     * Reads a file and returns the corresponding Biojava object. You<a name="line.912"></a>
<span class="sourceLineNo">913</span>     * need to cast it as an Alignment or a SequenceIterator as<a name="line.913"></a>
<span class="sourceLineNo">914</span>     * appropriate.<a name="line.914"></a>
<span class="sourceLineNo">915</span>     * @param fileType a value that describes the file type<a name="line.915"></a>
<span class="sourceLineNo">916</span>     * @param br the reader for the input<a name="line.916"></a>
<span class="sourceLineNo">917</span>     * @throws org.biojava.bio.BioException if the file cannot be parsed<a name="line.917"></a>
<span class="sourceLineNo">918</span>     * @return either a &lt;code&gt;SequenceIterator&lt;/code&gt; if the file type is a <a name="line.918"></a>
<span class="sourceLineNo">919</span>     * sequence file, or a &lt;code&gt;Alignment&lt;/code&gt; if the file is a sequence<a name="line.919"></a>
<span class="sourceLineNo">920</span>     * alignment.<a name="line.920"></a>
<span class="sourceLineNo">921</span>     */<a name="line.921"></a>
<span class="sourceLineNo">922</span>    public static Object fileToBiojava(int fileType, BufferedReader br)<a name="line.922"></a>
<span class="sourceLineNo">923</span>        throws BioException {<a name="line.923"></a>
<span class="sourceLineNo">924</span><a name="line.924"></a>
<span class="sourceLineNo">925</span>        // Mask the sequence format bytes<a name="line.925"></a>
<span class="sourceLineNo">926</span>        int alphaType = fileType &amp; (~ 0xffff);<a name="line.926"></a>
<span class="sourceLineNo">927</span>        if (alphaType == 0)<a name="line.927"></a>
<span class="sourceLineNo">928</span>            throw new IllegalArgumentException("No alphabet was set in the identifier");<a name="line.928"></a>
<span class="sourceLineNo">929</span><a name="line.929"></a>
<span class="sourceLineNo">930</span>        // Mask alphabet bytes<a name="line.930"></a>
<span class="sourceLineNo">931</span>        int formatType = fileType &amp; (~ 0xffff0000);<a name="line.931"></a>
<span class="sourceLineNo">932</span>        if (formatType == 0)<a name="line.932"></a>
<span class="sourceLineNo">933</span>            throw new IllegalArgumentException("No format was set in the identifier");<a name="line.933"></a>
<span class="sourceLineNo">934</span><a name="line.934"></a>
<span class="sourceLineNo">935</span>        switch (fileType) {<a name="line.935"></a>
<span class="sourceLineNo">936</span>            case AlignIOConstants.MSF_DNA:<a name="line.936"></a>
<span class="sourceLineNo">937</span>            case AlignIOConstants.MSF_AA:<a name="line.937"></a>
<span class="sourceLineNo">938</span>            case AlignIOConstants.FASTA_DNA:<a name="line.938"></a>
<span class="sourceLineNo">939</span>            case AlignIOConstants.FASTA_AA:<a name="line.939"></a>
<span class="sourceLineNo">940</span>                return fileToAlign(fileType, br);<a name="line.940"></a>
<span class="sourceLineNo">941</span>            case SeqIOConstants.FASTA_DNA:<a name="line.941"></a>
<span class="sourceLineNo">942</span>            case SeqIOConstants.FASTA_AA:<a name="line.942"></a>
<span class="sourceLineNo">943</span>            case SeqIOConstants.EMBL_DNA:<a name="line.943"></a>
<span class="sourceLineNo">944</span>            case SeqIOConstants.GENBANK_DNA:<a name="line.944"></a>
<span class="sourceLineNo">945</span>            case SeqIOConstants.SWISSPROT:<a name="line.945"></a>
<span class="sourceLineNo">946</span>            case SeqIOConstants.GENPEPT:<a name="line.946"></a>
<span class="sourceLineNo">947</span>                return fileToSeq(fileType, br);<a name="line.947"></a>
<span class="sourceLineNo">948</span>            default:<a name="line.948"></a>
<span class="sourceLineNo">949</span>                throw new BioException("Unknown file type '"<a name="line.949"></a>
<span class="sourceLineNo">950</span>                                       + fileType<a name="line.950"></a>
<span class="sourceLineNo">951</span>                                       + "'");<a name="line.951"></a>
<span class="sourceLineNo">952</span>        }<a name="line.952"></a>
<span class="sourceLineNo">953</span>    }<a name="line.953"></a>
<span class="sourceLineNo">954</span><a name="line.954"></a>
<span class="sourceLineNo">955</span>    /**<a name="line.955"></a>
<span class="sourceLineNo">956</span>     * Writes a Biojava &lt;code&gt;SequenceIterator&lt;/code&gt;,<a name="line.956"></a>
<span class="sourceLineNo">957</span>     * &lt;code&gt;SequenceDB&lt;/code&gt;, &lt;code&gt;Sequence&lt;/code&gt; or &lt;code&gt;Aligment&lt;/code&gt;<a name="line.957"></a>
<span class="sourceLineNo">958</span>     * to an &lt;code&gt;OutputStream&lt;/code&gt;<a name="line.958"></a>
<span class="sourceLineNo">959</span>     *<a name="line.959"></a>
<span class="sourceLineNo">960</span>     * @param formatName eg fasta, GenBank (case insensitive)<a name="line.960"></a>
<span class="sourceLineNo">961</span>     * @param alphabetName eg DNA, RNA (case insensititve)<a name="line.961"></a>
<span class="sourceLineNo">962</span>     * @param os where to write to<a name="line.962"></a>
<span class="sourceLineNo">963</span>     * @param biojava the object to write<a name="line.963"></a>
<span class="sourceLineNo">964</span>     * @throws BioException problems getting data from the biojava object.<a name="line.964"></a>
<span class="sourceLineNo">965</span>     * @throws IOException if there are IO problems<a name="line.965"></a>
<span class="sourceLineNo">966</span>     * @throws IllegalSymbolException a Symbol cannot be parsed<a name="line.966"></a>
<span class="sourceLineNo">967</span>     */<a name="line.967"></a>
<span class="sourceLineNo">968</span>    public static void biojavaToFile(String formatName, String alphabetName,<a name="line.968"></a>
<span class="sourceLineNo">969</span>                                     OutputStream os, Object biojava)<a name="line.969"></a>
<span class="sourceLineNo">970</span>    throws BioException, IOException, IllegalSymbolException{<a name="line.970"></a>
<span class="sourceLineNo">971</span>      int fileType = identifyFormat(formatName,alphabetName);<a name="line.971"></a>
<span class="sourceLineNo">972</span>      biojavaToFile(fileType, os, biojava);<a name="line.972"></a>
<span class="sourceLineNo">973</span>    }<a name="line.973"></a>
<span class="sourceLineNo">974</span><a name="line.974"></a>
<span class="sourceLineNo">975</span>    /**<a name="line.975"></a>
<span class="sourceLineNo">976</span>     * Converts a Biojava object to the given filetype.<a name="line.976"></a>
<span class="sourceLineNo">977</span>     * @param fileType a value that describes the type of sequence file<a name="line.977"></a>
<span class="sourceLineNo">978</span>     * @param os the stream to write the formatted results to<a name="line.978"></a>
<span class="sourceLineNo">979</span>     * @param biojava a &lt;code&gt;SequenceIterator&lt;/code&gt;, &lt;code&gt;SequenceDB&lt;/code&gt;, <a name="line.979"></a>
<span class="sourceLineNo">980</span>     * &lt;code&gt;Sequence&lt;/code&gt;, or &lt;code&gt;Alignment&lt;/code&gt;<a name="line.980"></a>
<span class="sourceLineNo">981</span>     * @throws org.biojava.bio.BioException if &lt;code&gt;biojava&lt;/code&gt; cannot be <a name="line.981"></a>
<span class="sourceLineNo">982</span>     * converted to that format.<a name="line.982"></a>
<span class="sourceLineNo">983</span>     * @throws java.io.IOException if the output cannot be written to <a name="line.983"></a>
<span class="sourceLineNo">984</span>     * &lt;code&gt;os&lt;/code&gt;<a name="line.984"></a>
<span class="sourceLineNo">985</span>     * @throws org.biojava.bio.symbol.IllegalSymbolException if &lt;code&gt;biojava<a name="line.985"></a>
<span class="sourceLineNo">986</span>     * &lt;/code&gt; contains a &lt;code&gt;Symbol&lt;/code&gt; that cannot be understood by the<a name="line.986"></a>
<span class="sourceLineNo">987</span>     * parser.<a name="line.987"></a>
<span class="sourceLineNo">988</span>     */<a name="line.988"></a>
<span class="sourceLineNo">989</span>    public static void biojavaToFile(int fileType, OutputStream os,<a name="line.989"></a>
<span class="sourceLineNo">990</span>                                     Object biojava)<a name="line.990"></a>
<span class="sourceLineNo">991</span>        throws BioException, IOException, IllegalSymbolException {<a name="line.991"></a>
<span class="sourceLineNo">992</span>        switch (fileType) {<a name="line.992"></a>
<span class="sourceLineNo">993</span>            case AlignIOConstants.MSF_DNA:<a name="line.993"></a>
<span class="sourceLineNo">994</span>            case AlignIOConstants.MSF_AA:<a name="line.994"></a>
<span class="sourceLineNo">995</span>            case AlignIOConstants.FASTA_DNA:<a name="line.995"></a>
<span class="sourceLineNo">996</span>            case AlignIOConstants.FASTA_AA:<a name="line.996"></a>
<span class="sourceLineNo">997</span>                alignToFile(fileType, os, (Alignment) biojava);<a name="line.997"></a>
<span class="sourceLineNo">998</span>                break;<a name="line.998"></a>
<span class="sourceLineNo">999</span>            case SeqIOConstants.FASTA_DNA:<a name="line.999"></a>
<span class="sourceLineNo">1000</span>            case SeqIOConstants.FASTA_AA:<a name="line.1000"></a>
<span class="sourceLineNo">1001</span>            case SeqIOConstants.EMBL_DNA:<a name="line.1001"></a>
<span class="sourceLineNo">1002</span>            case SeqIOConstants.GENBANK_DNA:<a name="line.1002"></a>
<span class="sourceLineNo">1003</span>            case SeqIOConstants.SWISSPROT:<a name="line.1003"></a>
<span class="sourceLineNo">1004</span>            case SeqIOConstants.GENPEPT:<a name="line.1004"></a>
<span class="sourceLineNo">1005</span>                if(biojava instanceof SequenceDB){<a name="line.1005"></a>
<span class="sourceLineNo">1006</span>                  seqToFile(fileType, os, ((SequenceDB)biojava).sequenceIterator());<a name="line.1006"></a>
<span class="sourceLineNo">1007</span>                }else if(biojava instanceof Sequence){<a name="line.1007"></a>
<span class="sourceLineNo">1008</span>                  seqToFile(fileType, os, new SingleSeqIterator((Sequence)biojava));<a name="line.1008"></a>
<span class="sourceLineNo">1009</span>                }else{<a name="line.1009"></a>
<span class="sourceLineNo">1010</span>                  seqToFile(fileType, os, (SequenceIterator) biojava);<a name="line.1010"></a>
<span class="sourceLineNo">1011</span>                }<a name="line.1011"></a>
<span class="sourceLineNo">1012</span>                break;<a name="line.1012"></a>
<span class="sourceLineNo">1013</span>            default:<a name="line.1013"></a>
<span class="sourceLineNo">1014</span>                throw new BioException("Unknown file type '"<a name="line.1014"></a>
<span class="sourceLineNo">1015</span>                                       + fileType<a name="line.1015"></a>
<span class="sourceLineNo">1016</span>                                       + "'");<a name="line.1016"></a>
<span class="sourceLineNo">1017</span>        }<a name="line.1017"></a>
<span class="sourceLineNo">1018</span>    }<a name="line.1018"></a>
<span class="sourceLineNo">1019</span><a name="line.1019"></a>
<span class="sourceLineNo">1020</span>    /**<a name="line.1020"></a>
<span class="sourceLineNo">1021</span>     * Helper function for guessFileName.<a name="line.1021"></a>
<span class="sourceLineNo">1022</span>     */<a name="line.1022"></a>
<span class="sourceLineNo">1023</span>    private static int guessFastaType(File seqFile)<a name="line.1023"></a>
<span class="sourceLineNo">1024</span>        throws IOException, FileNotFoundException {<a name="line.1024"></a>
<span class="sourceLineNo">1025</span>        BufferedReader br = new BufferedReader(new FileReader(seqFile));<a name="line.1025"></a>
<span class="sourceLineNo">1026</span>        String line = br.readLine();<a name="line.1026"></a>
<span class="sourceLineNo">1027</span>        line = br.readLine();<a name="line.1027"></a>
<span class="sourceLineNo">1028</span>        br.close();<a name="line.1028"></a>
<span class="sourceLineNo">1029</span>        for (int i = 0; i &lt; line.length(); i++) {<a name="line.1029"></a>
<span class="sourceLineNo">1030</span>            if (Character.toUpperCase(line.charAt(i)) == 'F' ||<a name="line.1030"></a>
<span class="sourceLineNo">1031</span>                Character.toUpperCase(line.charAt(i)) == 'L' ||<a name="line.1031"></a>
<span class="sourceLineNo">1032</span>                Character.toUpperCase(line.charAt(i)) == 'I' ||<a name="line.1032"></a>
<span class="sourceLineNo">1033</span>                Character.toUpperCase(line.charAt(i)) == 'P' ||<a name="line.1033"></a>
<span class="sourceLineNo">1034</span>                Character.toUpperCase(line.charAt(i)) == 'Q' ||<a name="line.1034"></a>
<span class="sourceLineNo">1035</span>                Character.toUpperCase(line.charAt(i)) == 'E') {<a name="line.1035"></a>
<span class="sourceLineNo">1036</span>                return SeqIOConstants.FASTA_AA;<a name="line.1036"></a>
<span class="sourceLineNo">1037</span>            }<a name="line.1037"></a>
<span class="sourceLineNo">1038</span>        }<a name="line.1038"></a>
<span class="sourceLineNo">1039</span><a name="line.1039"></a>
<span class="sourceLineNo">1040</span>        return SeqIOConstants.FASTA_DNA;<a name="line.1040"></a>
<span class="sourceLineNo">1041</span>    }<a name="line.1041"></a>
<span class="sourceLineNo">1042</span><a name="line.1042"></a>
<span class="sourceLineNo">1043</span>    private static SymbolTokenization getDNAParser() {<a name="line.1043"></a>
<span class="sourceLineNo">1044</span>        try {<a name="line.1044"></a>
<span class="sourceLineNo">1045</span>            return DNATools.getDNA().getTokenization("token");<a name="line.1045"></a>
<span class="sourceLineNo">1046</span>        } catch (BioException ex) {<a name="line.1046"></a>
<span class="sourceLineNo">1047</span>            throw new BioError("Assertion failing:"<a name="line.1047"></a>
<span class="sourceLineNo">1048</span>                               + " Couldn't get DNA token parser",ex);<a name="line.1048"></a>
<span class="sourceLineNo">1049</span>        }<a name="line.1049"></a>
<span class="sourceLineNo">1050</span>    }<a name="line.1050"></a>
<span class="sourceLineNo">1051</span><a name="line.1051"></a>
<span class="sourceLineNo">1052</span>    private static SymbolTokenization getRNAParser() {<a name="line.1052"></a>
<span class="sourceLineNo">1053</span>        try {<a name="line.1053"></a>
<span class="sourceLineNo">1054</span>            return RNATools.getRNA().getTokenization("token");<a name="line.1054"></a>
<span class="sourceLineNo">1055</span>        } catch (BioException ex) {<a name="line.1055"></a>
<span class="sourceLineNo">1056</span>            throw new BioError("Assertion failing:"<a name="line.1056"></a>
<span class="sourceLineNo">1057</span>                               + " Couldn't get RNA token parser",ex);<a name="line.1057"></a>
<span class="sourceLineNo">1058</span>        }<a name="line.1058"></a>
<span class="sourceLineNo">1059</span>    }<a name="line.1059"></a>
<span class="sourceLineNo">1060</span><a name="line.1060"></a>
<span class="sourceLineNo">1061</span>    private static SymbolTokenization getNucleotideParser() {<a name="line.1061"></a>
<span class="sourceLineNo">1062</span>        try {<a name="line.1062"></a>
<span class="sourceLineNo">1063</span>            return NucleotideTools.getNucleotide().getTokenization("token");<a name="line.1063"></a>
<span class="sourceLineNo">1064</span>        } catch (BioException ex) {<a name="line.1064"></a>
<span class="sourceLineNo">1065</span>            throw new BioError("Assertion failing:"<a name="line.1065"></a>
<span class="sourceLineNo">1066</span>                               + " Couldn't get nucleotide token parser",ex);<a name="line.1066"></a>
<span class="sourceLineNo">1067</span>        }<a name="line.1067"></a>
<span class="sourceLineNo">1068</span>    }<a name="line.1068"></a>
<span class="sourceLineNo">1069</span><a name="line.1069"></a>
<span class="sourceLineNo">1070</span>    private static SymbolTokenization getProteinParser() {<a name="line.1070"></a>
<span class="sourceLineNo">1071</span>        try {<a name="line.1071"></a>
<span class="sourceLineNo">1072</span>            return ProteinTools.getTAlphabet().getTokenization("token");<a name="line.1072"></a>
<span class="sourceLineNo">1073</span>        } catch (BioException ex) {<a name="line.1073"></a>
<span class="sourceLineNo">1074</span>            throw new BioError("Assertion failing:"<a name="line.1074"></a>
<span class="sourceLineNo">1075</span>                               + " Couldn't get PROTEIN token parser",ex);<a name="line.1075"></a>
<span class="sourceLineNo">1076</span>        }<a name="line.1076"></a>
<span class="sourceLineNo">1077</span>    }<a name="line.1077"></a>
<span class="sourceLineNo">1078</span><a name="line.1078"></a>
<span class="sourceLineNo">1079</span>    /**<a name="line.1079"></a>
<span class="sourceLineNo">1080</span>     * Helper function for guessFileName.<a name="line.1080"></a>
<span class="sourceLineNo">1081</span>     */<a name="line.1081"></a>
<span class="sourceLineNo">1082</span>    private static int guessMsfType(File seqFile)<a name="line.1082"></a>
<span class="sourceLineNo">1083</span>        throws IOException, FileNotFoundException {<a name="line.1083"></a>
<span class="sourceLineNo">1084</span>        BufferedReader br = new BufferedReader(new FileReader(seqFile));<a name="line.1084"></a>
<span class="sourceLineNo">1085</span>        String line = br.readLine();<a name="line.1085"></a>
<span class="sourceLineNo">1086</span>        if (line.startsWith("!!NA_MULTIPLE_ALIGNMENT")) {<a name="line.1086"></a>
<span class="sourceLineNo">1087</span>            return AlignIOConstants.MSF_DNA;<a name="line.1087"></a>
<span class="sourceLineNo">1088</span>        }<a name="line.1088"></a>
<span class="sourceLineNo">1089</span>        else if (line.startsWith("!!AA_MULTIPLE_ALIGNMENT")) {<a name="line.1089"></a>
<span class="sourceLineNo">1090</span>            return AlignIOConstants.MSF_AA;<a name="line.1090"></a>
<span class="sourceLineNo">1091</span>        }<a name="line.1091"></a>
<span class="sourceLineNo">1092</span>        else {<a name="line.1092"></a>
<span class="sourceLineNo">1093</span>            while (line.indexOf("Type: ") == -1) {<a name="line.1093"></a>
<span class="sourceLineNo">1094</span>                line = br.readLine();<a name="line.1094"></a>
<span class="sourceLineNo">1095</span>            }<a name="line.1095"></a>
<span class="sourceLineNo">1096</span>            br.close();<a name="line.1096"></a>
<span class="sourceLineNo">1097</span>            int typeIndex = line.indexOf("Type: ") + 6;<a name="line.1097"></a>
<span class="sourceLineNo">1098</span>            if (line.substring(typeIndex).startsWith("N")) {<a name="line.1098"></a>
<span class="sourceLineNo">1099</span>                return AlignIOConstants.MSF_DNA;<a name="line.1099"></a>
<span class="sourceLineNo">1100</span>            }<a name="line.1100"></a>
<span class="sourceLineNo">1101</span>            else if (line.substring(typeIndex).startsWith("P")) {<a name="line.1101"></a>
<span class="sourceLineNo">1102</span>                return AlignIOConstants.MSF_AA;<a name="line.1102"></a>
<span class="sourceLineNo">1103</span>            }<a name="line.1103"></a>
<span class="sourceLineNo">1104</span>            else {<a name="line.1104"></a>
<span class="sourceLineNo">1105</span>                return AlignIOConstants.UNKNOWN;<a name="line.1105"></a>
<span class="sourceLineNo">1106</span>            }<a name="line.1106"></a>
<span class="sourceLineNo">1107</span>        }<a name="line.1107"></a>
<span class="sourceLineNo">1108</span>    }<a name="line.1108"></a>
<span class="sourceLineNo">1109</span><a name="line.1109"></a>
<span class="sourceLineNo">1110</span>    /**<a name="line.1110"></a>
<span class="sourceLineNo">1111</span>     * Helper function for guessFileName.<a name="line.1111"></a>
<span class="sourceLineNo">1112</span>     */<a name="line.1112"></a>
<span class="sourceLineNo">1113</span>    private static int guessGenType(String fileName)<a name="line.1113"></a>
<span class="sourceLineNo">1114</span>        throws IOException, FileNotFoundException {<a name="line.1114"></a>
<span class="sourceLineNo">1115</span>        BufferedReader br = new BufferedReader(new FileReader(fileName));<a name="line.1115"></a>
<span class="sourceLineNo">1116</span>        String line = br.readLine();<a name="line.1116"></a>
<span class="sourceLineNo">1117</span>        while (line.indexOf("LOCUS") == -1) {<a name="line.1117"></a>
<span class="sourceLineNo">1118</span>            line = br.readLine();<a name="line.1118"></a>
<span class="sourceLineNo">1119</span>        }<a name="line.1119"></a>
<span class="sourceLineNo">1120</span>        br.close();<a name="line.1120"></a>
<span class="sourceLineNo">1121</span>        for (int i = 0; i &lt; line.length(); i++) {<a name="line.1121"></a>
<span class="sourceLineNo">1122</span>            if (Character.toUpperCase(line.charAt(i)) == 'A' &amp;&amp;<a name="line.1122"></a>
<span class="sourceLineNo">1123</span>                Character.toUpperCase(line.charAt(i+1)) == 'A') {<a name="line.1123"></a>
<span class="sourceLineNo">1124</span>                    return SeqIOConstants.GENPEPT;<a name="line.1124"></a>
<span class="sourceLineNo">1125</span>            }<a name="line.1125"></a>
<span class="sourceLineNo">1126</span>        }<a name="line.1126"></a>
<span class="sourceLineNo">1127</span>        return SeqIOConstants.GENBANK_DNA;<a name="line.1127"></a>
<span class="sourceLineNo">1128</span>    }<a name="line.1128"></a>
<span class="sourceLineNo">1129</span><a name="line.1129"></a>
<span class="sourceLineNo">1130</span>    /**<a name="line.1130"></a>
<span class="sourceLineNo">1131</span>     * Converts a file to an Biojava alignment.<a name="line.1131"></a>
<span class="sourceLineNo">1132</span>     */<a name="line.1132"></a>
<span class="sourceLineNo">1133</span>    private static Alignment fileToAlign(int fileType, BufferedReader br)<a name="line.1133"></a>
<span class="sourceLineNo">1134</span>        throws BioException {<a name="line.1134"></a>
<span class="sourceLineNo">1135</span>        switch(fileType) {<a name="line.1135"></a>
<span class="sourceLineNo">1136</span>            case AlignIOConstants.MSF_DNA:<a name="line.1136"></a>
<span class="sourceLineNo">1137</span>            case AlignIOConstants.MSF_AA:<a name="line.1137"></a>
<span class="sourceLineNo">1138</span>                return (new MSFAlignmentFormat()).read(br);<a name="line.1138"></a>
<span class="sourceLineNo">1139</span>            case AlignIOConstants.FASTA_DNA:<a name="line.1139"></a>
<span class="sourceLineNo">1140</span>            case AlignIOConstants.FASTA_AA:<a name="line.1140"></a>
<span class="sourceLineNo">1141</span>                return (new FastaAlignmentFormat()).read(br);<a name="line.1141"></a>
<span class="sourceLineNo">1142</span>            default:<a name="line.1142"></a>
<span class="sourceLineNo">1143</span>                throw new BioException("Unknown file type '"<a name="line.1143"></a>
<span class="sourceLineNo">1144</span>                                       + fileType<a name="line.1144"></a>
<span class="sourceLineNo">1145</span>                                       + "'");<a name="line.1145"></a>
<span class="sourceLineNo">1146</span>        }<a name="line.1146"></a>
<span class="sourceLineNo">1147</span>    }<a name="line.1147"></a>
<span class="sourceLineNo">1148</span><a name="line.1148"></a>
<span class="sourceLineNo">1149</span>    /**<a name="line.1149"></a>
<span class="sourceLineNo">1150</span>     * Converts a file to a Biojava sequence.<a name="line.1150"></a>
<span class="sourceLineNo">1151</span>     */<a name="line.1151"></a>
<span class="sourceLineNo">1152</span>    private static SequenceIterator fileToSeq(int fileType,<a name="line.1152"></a>
<span class="sourceLineNo">1153</span>                                              BufferedReader br)<a name="line.1153"></a>
<span class="sourceLineNo">1154</span>        throws BioException {<a name="line.1154"></a>
<span class="sourceLineNo">1155</span>        switch (fileType) {<a name="line.1155"></a>
<span class="sourceLineNo">1156</span>            case SeqIOConstants.FASTA_DNA:<a name="line.1156"></a>
<span class="sourceLineNo">1157</span>                return SeqIOTools.readFastaDNA(br);<a name="line.1157"></a>
<span class="sourceLineNo">1158</span>            case SeqIOConstants.FASTA_AA:<a name="line.1158"></a>
<span class="sourceLineNo">1159</span>                return SeqIOTools.readFastaProtein(br);<a name="line.1159"></a>
<span class="sourceLineNo">1160</span>            case SeqIOConstants.EMBL_DNA:<a name="line.1160"></a>
<span class="sourceLineNo">1161</span>                return SeqIOTools.readEmbl(br);<a name="line.1161"></a>
<span class="sourceLineNo">1162</span>            case SeqIOConstants.GENBANK_DNA:<a name="line.1162"></a>
<span class="sourceLineNo">1163</span>                return SeqIOTools.readGenbank(br);<a name="line.1163"></a>
<span class="sourceLineNo">1164</span>            case SeqIOConstants.SWISSPROT:<a name="line.1164"></a>
<span class="sourceLineNo">1165</span>                return SeqIOTools.readSwissprot(br);<a name="line.1165"></a>
<span class="sourceLineNo">1166</span>            case SeqIOConstants.GENPEPT:<a name="line.1166"></a>
<span class="sourceLineNo">1167</span>                return SeqIOTools.readGenpept(br);<a name="line.1167"></a>
<span class="sourceLineNo">1168</span>            default:<a name="line.1168"></a>
<span class="sourceLineNo">1169</span>                throw new BioException("Unknown file type '"<a name="line.1169"></a>
<span class="sourceLineNo">1170</span>                                       + fileType<a name="line.1170"></a>
<span class="sourceLineNo">1171</span>                                       + "'");<a name="line.1171"></a>
<span class="sourceLineNo">1172</span>        }<a name="line.1172"></a>
<span class="sourceLineNo">1173</span>    }<a name="line.1173"></a>
<span class="sourceLineNo">1174</span><a name="line.1174"></a>
<span class="sourceLineNo">1175</span>    /**<a name="line.1175"></a>
<span class="sourceLineNo">1176</span>     * Converts a Biojava alignment to the given filetype.<a name="line.1176"></a>
<span class="sourceLineNo">1177</span>     */<a name="line.1177"></a>
<span class="sourceLineNo">1178</span>    private static void alignToFile(int fileType, OutputStream os,<a name="line.1178"></a>
<span class="sourceLineNo">1179</span>                                    Alignment align)<a name="line.1179"></a>
<span class="sourceLineNo">1180</span>        throws BioException, IllegalSymbolException {<a name="line.1180"></a>
<span class="sourceLineNo">1181</span>        switch(fileType) {<a name="line.1181"></a>
<span class="sourceLineNo">1182</span>            case AlignIOConstants.MSF_DNA:<a name="line.1182"></a>
<span class="sourceLineNo">1183</span>                (new MSFAlignmentFormat()).writeDna(os, align);<a name="line.1183"></a>
<span class="sourceLineNo">1184</span>                break;<a name="line.1184"></a>
<span class="sourceLineNo">1185</span>            case AlignIOConstants.MSF_AA:<a name="line.1185"></a>
<span class="sourceLineNo">1186</span>                (new MSFAlignmentFormat()).writeProtein(os, align);<a name="line.1186"></a>
<span class="sourceLineNo">1187</span>                break;<a name="line.1187"></a>
<span class="sourceLineNo">1188</span>            case AlignIOConstants.FASTA_DNA:<a name="line.1188"></a>
<span class="sourceLineNo">1189</span>                (new FastaAlignmentFormat()).writeDna(os, align);<a name="line.1189"></a>
<span class="sourceLineNo">1190</span>                break;<a name="line.1190"></a>
<span class="sourceLineNo">1191</span>            case AlignIOConstants.FASTA_AA:<a name="line.1191"></a>
<span class="sourceLineNo">1192</span>                (new FastaAlignmentFormat()).writeProtein(os, align);<a name="line.1192"></a>
<span class="sourceLineNo">1193</span>                break;<a name="line.1193"></a>
<span class="sourceLineNo">1194</span>            default:<a name="line.1194"></a>
<span class="sourceLineNo">1195</span>                throw new BioException("Unknown file type '"<a name="line.1195"></a>
<span class="sourceLineNo">1196</span>                                       + fileType<a name="line.1196"></a>
<span class="sourceLineNo">1197</span>                                       + "'");<a name="line.1197"></a>
<span class="sourceLineNo">1198</span>        }<a name="line.1198"></a>
<span class="sourceLineNo">1199</span>    }<a name="line.1199"></a>
<span class="sourceLineNo">1200</span><a name="line.1200"></a>
<span class="sourceLineNo">1201</span>    /**<a name="line.1201"></a>
<span class="sourceLineNo">1202</span>     * Converts a Biojava sequence to the given filetype.<a name="line.1202"></a>
<span class="sourceLineNo">1203</span>     */<a name="line.1203"></a>
<span class="sourceLineNo">1204</span>    private static void seqToFile(int fileType, OutputStream os,<a name="line.1204"></a>
<span class="sourceLineNo">1205</span>                                  SequenceIterator seq)<a name="line.1205"></a>
<span class="sourceLineNo">1206</span>        throws IOException, BioException {<a name="line.1206"></a>
<span class="sourceLineNo">1207</span>        switch (fileType) {<a name="line.1207"></a>
<span class="sourceLineNo">1208</span>            case SeqIOConstants.FASTA_DNA:<a name="line.1208"></a>
<span class="sourceLineNo">1209</span>            case SeqIOConstants.FASTA_AA:<a name="line.1209"></a>
<span class="sourceLineNo">1210</span>                SeqIOTools.writeFasta(os, seq);<a name="line.1210"></a>
<span class="sourceLineNo">1211</span>                break;<a name="line.1211"></a>
<span class="sourceLineNo">1212</span>            case SeqIOConstants.EMBL_DNA:<a name="line.1212"></a>
<span class="sourceLineNo">1213</span>                SeqIOTools.writeEmbl(os, seq);<a name="line.1213"></a>
<span class="sourceLineNo">1214</span>                break;<a name="line.1214"></a>
<span class="sourceLineNo">1215</span>            case SeqIOConstants.SWISSPROT:<a name="line.1215"></a>
<span class="sourceLineNo">1216</span>                SeqIOTools.writeSwissprot(os, seq);<a name="line.1216"></a>
<span class="sourceLineNo">1217</span>                break;<a name="line.1217"></a>
<span class="sourceLineNo">1218</span>            case SeqIOConstants.GENBANK_DNA:<a name="line.1218"></a>
<span class="sourceLineNo">1219</span>                SeqIOTools.writeGenbank(os, seq);<a name="line.1219"></a>
<span class="sourceLineNo">1220</span>                break;<a name="line.1220"></a>
<span class="sourceLineNo">1221</span>            case SeqIOConstants.GENPEPT:<a name="line.1221"></a>
<span class="sourceLineNo">1222</span>                SeqIOTools.writeGenpept(os, seq);<a name="line.1222"></a>
<span class="sourceLineNo">1223</span>                break;<a name="line.1223"></a>
<span class="sourceLineNo">1224</span>            default:<a name="line.1224"></a>
<span class="sourceLineNo">1225</span>                throw new BioException("Unknown file type '"<a name="line.1225"></a>
<span class="sourceLineNo">1226</span>                                       + fileType<a name="line.1226"></a>
<span class="sourceLineNo">1227</span>                                       + "'");<a name="line.1227"></a>
<span class="sourceLineNo">1228</span>        }<a name="line.1228"></a>
<span class="sourceLineNo">1229</span>    }<a name="line.1229"></a>
<span class="sourceLineNo">1230</span><a name="line.1230"></a>
<span class="sourceLineNo">1231</span>    private static final class SingleSeqIterator<a name="line.1231"></a>
<span class="sourceLineNo">1232</span>        implements SequenceIterator {<a name="line.1232"></a>
<span class="sourceLineNo">1233</span>        private Sequence seq;<a name="line.1233"></a>
<span class="sourceLineNo">1234</span>        SingleSeqIterator(Sequence seq) {<a name="line.1234"></a>
<span class="sourceLineNo">1235</span>            this.seq = seq;<a name="line.1235"></a>
<span class="sourceLineNo">1236</span>        }<a name="line.1236"></a>
<span class="sourceLineNo">1237</span><a name="line.1237"></a>
<span class="sourceLineNo">1238</span>        public boolean hasNext() {<a name="line.1238"></a>
<span class="sourceLineNo">1239</span>            return seq != null;<a name="line.1239"></a>
<span class="sourceLineNo">1240</span>        }<a name="line.1240"></a>
<span class="sourceLineNo">1241</span><a name="line.1241"></a>
<span class="sourceLineNo">1242</span>        public Sequence nextSequence() {<a name="line.1242"></a>
<span class="sourceLineNo">1243</span>            Sequence seq = this.seq;<a name="line.1243"></a>
<span class="sourceLineNo">1244</span>            this.seq = null;<a name="line.1244"></a>
<span class="sourceLineNo">1245</span>            return seq;<a name="line.1245"></a>
<span class="sourceLineNo">1246</span>        }<a name="line.1246"></a>
<span class="sourceLineNo">1247</span>    }<a name="line.1247"></a>
<span class="sourceLineNo">1248</span>}<a name="line.1248"></a>




























































</pre>
</div>
</body>
</html>
