<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>AminoAcid (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.structure, interface: AminoAcid">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/AminoAcid.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html#class">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Summary:</p>
<ul>
<li>Nested</li>
<li><a href="#field-summary">Field</a></li>
<li>Constr</li>
<li><a href="#method-summary">Method</a></li>
</ul>
</li>
<li>
<p>Detail:</p>
<ul>
<li><a href="#field-detail">Field</a></li>
<li>Constr</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field-summary">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li><a href="#field-detail">Field</a>&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.structure</a></div>
<h1 title="Interface AminoAcid" class="title">Interface AminoAcid</h1>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>All Superinterfaces:</dt>
<dd><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></code>, <code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html" title="class or interface in java.io" class="external-link">Serializable</a></code></dd>
</dl>
<dl class="notes">
<dt>All Known Implementing Classes:</dt>
<dd><code><a href="AminoAcidImpl.html" title="class in org.biojava.nbio.structure">AminoAcidImpl</a></code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public interface </span><span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-49">AminoAcid</a></span><span class="extends-implements">
extends <a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></span></div>
<div class="block"><p>
 A <a href="Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a> that represents an AminoAcid.
 </p>

 <p>
 In PDB files information on AminoAcids can be observed in the SEQRES and in the ATOM records.
 Since frequently coordinates for some of the amino acids are missing, during parsing of the PDB
 files the SEQRES and the ATOM records are aligned and a whenever possible the AminoAcid objects
 are combined. Access to the SEQRES and ATOM sequence is possible through the <a href="Chain.html" title="interface in org.biojava.nbio.structure"><code>Chain</code></a> object.
 It is possible to distinguish between SEQRES and ATOM derived AminoAcids by <a href="#getRecordType()"><code>getRecordType()</code></a>.
 </p>

<p>
  AminoAcid inherits most from <a href="HetatomImpl.html" title="class in org.biojava.nbio.structure"><code>HetatomImpl</code></a>.  Adds a few AminoAcid
  specific methods.
  </p></div>
<dl class="notes">
<dt>Since:</dt>
<dd>1.4</dd>
<dt>Version:</dt>
<dd>%I% %G%</dd>
<dt>Author:</dt>
<dd>Andreas Prlic</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- =========== FIELD SUMMARY =========== -->
<li>
<section class="field-summary" id="field-summary">
<h2>Field Summary</h2>
<div class="caption"><span>Fields</span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Field</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><code><a href="#ATOMRECORD" class="member-name-link">ATOMRECORD</a></code></div>
<div class="col-last even-row-color">
<div class="block">Field to distinguish AminoAcids that have been created from SEQRES records and ATOM records.</div>
</div>
<div class="col-first odd-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><code><a href="#SEQRESRECORD" class="member-name-link">SEQRESRECORD</a></code></div>
<div class="col-last odd-row-color">
<div class="block">Field to distinguish AminoAcids that have been created from SEQRES records and ATOM records.</div>
</div>
</div>
<div class="inherited-list">
<h3 id="fields-inherited-from-class-org.biojava.nbio.structure.Group">Fields inherited from interface&nbsp;org.biojava.nbio.structure.<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></h3>
<code><a href="Group.html#SEC_STRUC">SEC_STRUC</a></code></div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab3" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab3', 3)" class="table-tab">Abstract Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel" aria-labelledby="method-summary-table-tab0">
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html" title="class or interface in java.lang" class="external-link">Character</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getAminoType()" class="member-name-link">getAminoType</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the name of the AA, in single letter code.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getC()" class="member-name-link">getC</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get C atom.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getCA()" class="member-name-link">getCA</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get CA atom.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getCB()" class="member-name-link">getCB</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get CB atom.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getN()" class="member-name-link">getN</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get N atom.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getO()" class="member-name-link">getO</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get O atom.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getRecordType()" class="member-name-link">getRecordType</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Allows to distinguish between amino acids that are provided
 as ATOM records and a SEQRES records.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setAminoType(java.lang.Character)" class="member-name-link">setAminoType</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html" title="class or interface in java.lang" class="external-link">Character</a>&nbsp;aa)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Set the name of the AA, in single letter code .</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setRecordType(java.lang.String)" class="member-name-link">setRecordType</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;recordName)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Allows to distinguish between amino acids that are provided
 as ATOM records and a SEQRES records.</div>
</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-org.biojava.nbio.structure.Group">Methods inherited from interface&nbsp;org.biojava.nbio.structure.<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></h3>
<code><a href="Group.html#addAltLoc(org.biojava.nbio.structure.Group)">addAltLoc</a>, <a href="Group.html#addAtom(org.biojava.nbio.structure.Atom)">addAtom</a>, <a href="Group.html#clearAtoms()">clearAtoms</a>, <a href="Group.html#clone()">clone</a>, <a href="Group.html#getAltLocGroup(java.lang.Character)">getAltLocGroup</a>, <a href="Group.html#getAltLocs()">getAltLocs</a>, <a href="Group.html#getAtom(int)">getAtom</a>, <a href="Group.html#getAtom(java.lang.String)">getAtom</a>, <a href="Group.html#getAtoms()">getAtoms</a>, <a href="Group.html#getChain()">getChain</a>, <a href="Group.html#getChainId()">getChainId</a>, <a href="Group.html#getChemComp()">getChemComp</a>, <a href="Group.html#getPDBName()">getPDBName</a>, <a href="Group.html#getProperties()">getProperties</a>, <a href="Group.html#getProperty(java.lang.String)">getProperty</a>, <a href="Group.html#getResidueNumber()">getResidueNumber</a>, <a href="Group.html#getType()">getType</a>, <a href="Group.html#has3D()">has3D</a>, <a href="Group.html#hasAltLoc()">hasAltLoc</a>, <a href="Group.html#hasAminoAtoms()">hasAminoAtoms</a>, <a href="Group.html#hasAtom(java.lang.String)">hasAtom</a>, <a href="Group.html#isAminoAcid()">isAminoAcid</a>, <a href="Group.html#isHetAtomInFile()">isHetAtomInFile</a>, <a href="Group.html#isNucleotide()">isNucleotide</a>, <a href="Group.html#isPolymeric()">isPolymeric</a>, <a href="Group.html#isWater()">isWater</a>, <a href="Group.html#iterator()">iterator</a>, <a href="Group.html#setAtoms(java.util.List)">setAtoms</a>, <a href="Group.html#setChain(org.biojava.nbio.structure.Chain)">setChain</a>, <a href="Group.html#setChemComp(org.biojava.nbio.structure.chem.ChemComp)">setChemComp</a>, <a href="Group.html#setHetAtomInFile(boolean)">setHetAtomInFile</a>, <a href="Group.html#setPDBFlag(boolean)">setPDBFlag</a>, <a href="Group.html#setPDBName(java.lang.String)">setPDBName</a>, <a href="Group.html#setProperties(java.util.Map)">setProperties</a>, <a href="Group.html#setProperty(java.lang.String,java.lang.Object)">setProperty</a>, <a href="Group.html#setResidueNumber(java.lang.String,java.lang.Integer,java.lang.Character)">setResidueNumber</a>, <a href="Group.html#setResidueNumber(org.biojava.nbio.structure.ResidueNumber)">setResidueNumber</a>, <a href="Group.html#size()">size</a>, <a href="Group.html#toSDF()">toSDF</a>, <a href="Group.html#trimToSize()">trimToSize</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ============ FIELD DETAIL =========== -->
<li>
<section class="field-details" id="field-detail">
<h2>Field Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="ATOMRECORD">
<h3>ATOMRECORD</h3>
<div class="member-signature"><span class="modifiers">static final</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-55">ATOMRECORD</a></span></div>
<div class="block">Field to distinguish AminoAcids that have been created from SEQRES records and ATOM records.</div>
<dl class="notes">
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="../../../../constant-values.html#org.biojava.nbio.structure.AminoAcid.ATOMRECORD">Constant Field Values</a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="SEQRESRECORD">
<h3>SEQRESRECORD</h3>
<div class="member-signature"><span class="modifiers">static final</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-61">SEQRESRECORD</a></span></div>
<div class="block">Field to distinguish AminoAcids that have been created from SEQRES records and ATOM records.</div>
<dl class="notes">
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="../../../../constant-values.html#org.biojava.nbio.structure.AminoAcid.SEQRESRECORD">Constant Field Values</a></li>
</ul>
</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="getN()">
<h3>getN</h3>
<div class="member-signature"><span class="return-type"><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-68">getN</a></span>()</div>
<div class="block">Get N atom.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>an Atom object or null if N atom does not exist</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCA()">
<h3>getCA</h3>
<div class="member-signature"><span class="return-type"><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-74">getCA</a></span>()</div>
<div class="block">Get CA atom.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>an Atom object or null if CA atom does not exist</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getC()">
<h3>getC</h3>
<div class="member-signature"><span class="return-type"><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-80">getC</a></span>()</div>
<div class="block">Get C atom.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>an Atom object or null if C atom does not exist</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getO()">
<h3>getO</h3>
<div class="member-signature"><span class="return-type"><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-86">getO</a></span>()</div>
<div class="block">Get O atom.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>an Atom object or null if O atom does not exist</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCB()">
<h3>getCB</h3>
<div class="member-signature"><span class="return-type"><a href="Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-92">getCB</a></span>()</div>
<div class="block">Get CB atom.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>an Atom object or null if CB atom does not exist</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAminoType()">
<h3>getAminoType</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html" title="class or interface in java.lang" class="external-link">Character</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-100">getAminoType</a></span>()</div>
<div class="block">Returns the name of the AA, in single letter code.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>a Character object representing the amino type value</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list-long">
<li><a href="#setAminoType(java.lang.Character)"><code>setAminoType(java.lang.Character)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setAminoType(java.lang.Character)">
<h3>setAminoType</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-108">setAminoType</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html" title="class or interface in java.lang" class="external-link">Character</a>&nbsp;aa)</span></div>
<div class="block">Set the name of the AA, in single letter code .</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>aa</code> - a Character object specifying the amino type value</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getAminoType()"><code>getAminoType()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setRecordType(java.lang.String)">
<h3>setRecordType</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-116">setRecordType</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;recordName)</span></div>
<div class="block">Allows to distinguish between amino acids that are provided
 as ATOM records and a SEQRES records.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>recordName</code> - either ATOMRECORD or SEQRESRECORD</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getRecordType()"><code>getRecordType()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getRecordType()">
<h3>getRecordType</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/AminoAcid.html#line-124">getRecordType</a></span>()</div>
<div class="block">Allows to distinguish between amino acids that are provided
 as ATOM records and a SEQRES records.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the origin of this amino acid (ATOM or SEQRES records)</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setRecordType(java.lang.String)"><code>setRecordType(String)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
