<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Chain (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.structure, interface: Chain">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../script.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/Chain.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html#class">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Summary:</p>
<ul>
<li>Nested</li>
<li>Field</li>
<li>Constr</li>
<li><a href="#method-summary">Method</a></li>
</ul>
</li>
<li>
<p>Detail:</p>
<ul>
<li>Field</li>
<li>Constr</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.structure</a></div>
<h1 title="Interface Chain" class="title">Interface Chain</h1>
</div>
<section class="class-description" id="class-description">
<dl class="notes">
<dt>All Superinterfaces:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html" title="class or interface in java.io" class="external-link">Serializable</a></code></dd>
</dl>
<dl class="notes">
<dt>All Known Implementing Classes:</dt>
<dd><code><a href="ChainImpl.html" title="class in org.biojava.nbio.structure">ChainImpl</a></code></dd>
</dl>
<hr>
<div class="type-signature"><span class="modifiers">public interface </span><span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-51">Chain</a></span><span class="extends-implements">
extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html" title="class or interface in java.io" class="external-link">Serializable</a></span></div>
<div class="block"><p>
 Defines the interface for a Chain. A Chain corresponds to a Chain in a PDB file.
 A chain consists of a list of <a href="Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a> objects. A Group can either be
 an <a href="AminoAcid.html" title="interface in org.biojava.nbio.structure"><code>AminoAcid</code></a>, <a href="HetatomImpl.html" title="class in org.biojava.nbio.structure"><code>Hetatom</code></a> or <a href="NucleotideImpl.html" title="class in org.biojava.nbio.structure"><code>Nucleotide</code></a>.
 </p>

 <p>
 The BioJava API provides access to both the ATOM and SEQRES records in a PDB file.
 During parsing of a PDB file it aligns the ATOM and SEQRES groups and joins them.
 The SEQRES sequence can be accessed via  <a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a> and the
 ATOM groups via <a href="#getAtomGroups()"><code>getAtomGroups()</code></a>. Groups that have been observed
 (i.e. they are in the ATOM records) can be detected by <a href="Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a>.has3D()
  </p></div>
<dl class="notes">
<dt>Since:</dt>
<dd>1.4</dd>
<dt>Version:</dt>
<dd>%I% %G%</dd>
<dt>Author:</dt>
<dd>Andreas Prlic</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab2" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab2', 3)" class="table-tab">Instance Methods</button><button id="method-summary-table-tab3" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab3', 3)" class="table-tab">Abstract Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel" aria-labelledby="method-summary-table-tab0">
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#addGroup(org.biojava.nbio.structure.Group)" class="member-name-link">addGroup</a><wbr>(<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Add a group to the list of ATOM record group of this chain.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#clone()" class="member-name-link">clone</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns an identical copy of this Chain.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getAtomGroup(int)" class="member-name-link">getAtomGroup</a><wbr>(int&nbsp;position)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Return the Group at given position,
 from within Groups with observed density in the chain, i.e.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getAtomGroups()" class="member-name-link">getAtomGroups</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Return all Groups with observed density in the chain, i.e.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getAtomGroups(org.biojava.nbio.structure.GroupType)" class="member-name-link">getAtomGroups</a><wbr>(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Return a List of all (observed) Groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getAtomLength()" class="member-name-link">getAtomLength</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the number of Groups with observed density in the chain, i.e.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getAtomSequence()" class="member-name-link">getAtomSequence</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the sequence of amino acids as it has been provided in the ATOM records.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="../core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a><wbr>&lt;?&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getBJSequence()" class="member-name-link">getBJSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getEntityInfo()" class="member-name-link">getEntityInfo</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the EntityInfo for this chain.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="EntityType.html" title="enum in org.biojava.nbio.structure">EntityType</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getEntityType()" class="member-name-link">getEntityType</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the EntityType of this chain.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)" class="member-name-link">getGroupByPDB</a><wbr>(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get a group by its PDB residue numbering.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber)" class="member-name-link">getGroupsByPDB</a><wbr>(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
 <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber,boolean)" class="member-name-link">getGroupsByPDB</a><wbr>(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
 <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
 boolean&nbsp;ignoreMissing)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getId()" class="member-name-link">getId</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getName()" class="member-name-link">getName</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get the 'public' authId (chain ID in PDB file)</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getPredominantGroupType()" class="member-name-link">getPredominantGroupType</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Get the predominant <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> for a given Chain, following these
 rules:
 
 if the ratio of number of residues of a certain
 <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> to total non-water residues is above the threshold
 <a href="StructureTools.html#RATIO_RESIDUES_TO_TOTAL">0.95</a>, then that <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> is
 returned if there is no <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> that is above the
 threshold then the <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> with most members is chosen, logging
 it
 </div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getSeqMisMatches()" class="member-name-link">getSeqMisMatches</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Gets annotated sequence mismatches for this chain.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getSeqResGroup(int)" class="member-name-link">getSeqResGroup</a><wbr>(int&nbsp;position)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Return the Group at given position,
 from within groups in the SEQRES records of the chain, i.e.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getSeqResGroups()" class="member-name-link">getSeqResGroups</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns a list of all groups in SEQRES records of the chain, i.e.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getSeqResGroups(org.biojava.nbio.structure.GroupType)" class="member-name-link">getSeqResGroups</a><wbr>(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns a List of all SEQRES groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getSeqResLength()" class="member-name-link">getSeqResLength</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the number of groups in the SEQRES records of the chain, i.e.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getSeqResSequence()" class="member-name-link">getSeqResSequence</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the PDB SEQRES sequence as a one-letter sequence string.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#getStructure()" class="member-name-link">getStructure</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns the parent Structure of this chain.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>boolean</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#isNucleicAcid()" class="member-name-link">isNucleicAcid</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Tell whether given chain is DNA or RNA</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>boolean</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#isProtein()" class="member-name-link">isProtein</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Tell whether given chain is a protein chain</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>boolean</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#isPureNonPolymer()" class="member-name-link">isPureNonPolymer</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Returns true if the given chain is composed of non-polymeric (including water) groups only.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>boolean</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#isWaterOnly()" class="member-name-link">isWaterOnly</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Tests if a chain is consisting of water molecules only</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setAtomGroups(java.util.List)" class="member-name-link">setAtomGroups</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groups)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Set all Groups with observed density in the chain, i.e.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setEntityInfo(org.biojava.nbio.structure.EntityInfo)" class="member-name-link">setEntityInfo</a><wbr>(<a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a>&nbsp;entityInfo)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Sets the Entity information</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setId(java.lang.String)" class="member-name-link">setId</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;asymId)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Set the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setName(java.lang.String)" class="member-name-link">setName</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;authId)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Set the 'public' authId (chain ID in PDB file)</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setSeqMisMatches(java.util.List)" class="member-name-link">setSeqMisMatches</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;seqMisMatches)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Sets annotated sequence mismatches for this chain.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setSeqResGroups(java.util.List)" class="member-name-link">setSeqResGroups</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;seqResGroups)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Sets the list of SeqResGroups for this chain.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code>void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#setStructure(org.biojava.nbio.structure.Structure)" class="member-name-link">setStructure</a><wbr>(<a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Sets the back-reference to its parent Structure.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#toMMCIF()" class="member-name-link">toMMCIF</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Convert this Chain to a String in mmCIF format</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#toPDB()" class="member-name-link">toPDB</a>()</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">
<div class="block">Convert this Chain to a String in PDB format</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3"><code><a href="#toString()" class="member-name-link">toString</a>()</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab2 method-summary-table-tab3">&nbsp;</div>
</div>
</div>
</div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="clone()">
<h3>clone</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-57">clone</a></span>()</div>
<div class="block">Returns an identical copy of this Chain.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>an identical copy of this Chain</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="addGroup(org.biojava.nbio.structure.Group)">
<h3>addGroup</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-65">addGroup</a></span><wbr><span class="parameters">(<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&nbsp;group)</span></div>
<div class="block">Add a group to the list of ATOM record group of this chain.
 To add SEQRES records a more complex alignment between ATOM and SEQRES residues
 is required, please see SeqRes2AtomAligner for more details on that.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>group</code> - a Group object</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getId()">
<h3>getId</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-74">getId</a></span>()</div>
<div class="block">Get the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the asymId</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setId(java.lang.String)"><code>setId(String)</code></a></li>
<li><a href="#getName()"><code>getName()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setId(java.lang.String)">
<h3>setId</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-82">setId</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;asymId)</span></div>
<div class="block">Set the 'private' asymId (internal chain IDs in mmCif) for this chain.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>asymId</code> - the internal chain Id</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setName(java.lang.String)">
<h3>setName</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-91">setName</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;authId)</span></div>
<div class="block">Set the 'public' authId (chain ID in PDB file)</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>authId</code> - the 'public' authId (chain ID in PDB file)</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getId()"><code>getId()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getName()">
<h3>getName</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-99">getName</a></span>()</div>
<div class="block">Get the 'public' authId (chain ID in PDB file)</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the authId for this chain.</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getId()"><code>getId()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtomGroup(int)">
<h3>getAtomGroup</h3>
<div class="member-signature"><span class="return-type"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-112">getAtomGroup</a></span><wbr><span class="parameters">(int&nbsp;position)</span></div>
<div class="block">Return the Group at given position,
 from within Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMS (including waters) records.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>position</code> - an int</dd>
<dt>Returns:</dt>
<dd>a Group object</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getAtomLength()"><code>getAtomLength()</code></a></li>
<li><a href="#getAtomGroups()"><code>getAtomGroups()</code></a></li>
<li><a href="#getSeqResGroup(int)"><code>getSeqResGroup(int)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSeqResGroup(int)">
<h3>getSeqResGroup</h3>
<div class="member-signature"><span class="return-type"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-124">getSeqResGroup</a></span><wbr><span class="parameters">(int&nbsp;position)</span></div>
<div class="block">Return the Group at given position,
 from within groups in the SEQRES records of the chain, i.e.
 the aminoacids/nucleotides in the construct.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>position</code> - an int</dd>
<dt>Returns:</dt>
<dd>a Group object</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getSeqResLength()"><code>getSeqResLength()</code></a></li>
<li><a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a></li>
<li><a href="#getAtomGroup(int)"><code>getAtomGroup(int)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtomGroups()">
<h3>getAtomGroups</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-136">getAtomGroups</a></span>()</div>
<div class="block">Return all Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMS (including waters) records.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>a List object representing the Groups of this Chain.</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setAtomGroups(java.util.List)"><code>setAtomGroups(List)</code></a></li>
<li><a href="#getAtomLength()"><code>getAtomLength()</code></a></li>
<li><a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setAtomGroups(java.util.List)">
<h3>setAtomGroups</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-144">setAtomGroups</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;groups)</span></div>
<div class="block">Set all Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMs (including waters) records.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>groups</code> - a List object representing the Groups of this Chain.</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getAtomGroups()"><code>getAtomGroups()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtomGroups(org.biojava.nbio.structure.GroupType)">
<h3>getAtomGroups</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-155">getAtomGroups</a></span><wbr><span class="parameters">(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</span></div>
<div class="block">Return a List of all (observed) Groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.
 Note that if a standard aminoacid appears as a HETATM (because it is part of a ligand) then
 it is still considered as <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a> and not as <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a>.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>type</code> - GroupType</dd>
<dt>Returns:</dt>
<dd>a List object</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setAtomGroups(java.util.List)"><code>setAtomGroups(List)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGroupByPDB(org.biojava.nbio.structure.ResidueNumber)">
<h3>getGroupByPDB</h3>
<div class="member-signature"><span class="return-type"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-166">getGroupByPDB</a></span><wbr><span class="parameters">(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;resNum)</span>
             throws <span class="exceptions"><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span></div>
<div class="block">Get a group by its PDB residue numbering. If the PDB residue number is not known,
 throws a StructureException.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>resNum</code> - the PDB residue number of the group</dd>
<dt>Returns:</dt>
<dd>the matching group</dd>
<dt>Throws:</dt>
<dd><code><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber)">
<h3>getGroupsByPDB</h3>
<div class="member-signature"><span class="return-type"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-176">getGroupsByPDB</a></span><wbr><span class="parameters">(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
 <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd)</span>
                throws <span class="exceptions"><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span></div>
<div class="block">Get all groups that are located between two PDB residue numbers.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>pdbresnumStart</code> - PDB residue number of start. If null, defaults to the chain start.</dd>
<dd><code>pdbresnumEnd</code> - PDB residue number of end. If null, defaults to the chain end.</dd>
<dt>Returns:</dt>
<dd>Groups in between. or throws a StructureException if either start or end can not be found,</dd>
<dt>Throws:</dt>
<dd><code><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getGroupsByPDB(org.biojava.nbio.structure.ResidueNumber,org.biojava.nbio.structure.ResidueNumber,boolean)">
<h3>getGroupsByPDB</h3>
<div class="member-signature"><span class="return-type"><a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-192">getGroupsByPDB</a></span><wbr><span class="parameters">(<a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumStart,
 <a href="ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>&nbsp;pdbresnumEnd,
 boolean&nbsp;ignoreMissing)</span>
                throws <span class="exceptions"><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></span></div>
<div class="block">Get all groups that are located between two PDB residue numbers. In contrast to getGroupsByPDB
 this method call ignores if the exact outer groups are not found. This is useful e.g. when requesting the range
 of groups as specified by the DBREF records - these frequently are rather inaccurate.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>pdbresnumStart</code> - PDB residue number of start. If null, defaults to the chain start.</dd>
<dd><code>pdbresnumEnd</code> - PDB residue number of end. If null, defaults to the chain end.</dd>
<dd><code>ignoreMissing</code> - ignore missing groups in this range.</dd>
<dt>Returns:</dt>
<dd>Groups in between. or throws a StructureException if either start or end can not be found,</dd>
<dt>Throws:</dt>
<dd><code><a href="StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtomLength()">
<h3>getAtomLength</h3>
<div class="member-signature"><span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-204">getAtomLength</a></span>()</div>
<div class="block">Returns the number of Groups with observed density in the chain, i.e.
 those with coordinates in ATOM and HETATMs (including waters) records</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the length</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getAtomGroup(int)"><code>getAtomGroup(int)</code></a></li>
<li><a href="#getAtomGroups()"><code>getAtomGroups()</code></a></li>
<li><a href="#getSeqResLength()"><code>getSeqResLength()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSeqResLength()">
<h3>getSeqResLength</h3>
<div class="member-signature"><span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-215">getSeqResLength</a></span>()</div>
<div class="block">Returns the number of groups in the SEQRES records of the chain, i.e.
 the number of aminoacids/nucleotides in the construct</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the length</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getSeqResGroup(int)"><code>getSeqResGroup(int)</code></a></li>
<li><a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a></li>
<li><a href="#getAtomLength()"><code>getAtomLength()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setEntityInfo(org.biojava.nbio.structure.EntityInfo)">
<h3>setEntityInfo</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-222">setEntityInfo</a></span><wbr><span class="parameters">(<a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a>&nbsp;entityInfo)</span></div>
<div class="block">Sets the Entity information</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>entityInfo</code> - the EntityInfo</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getEntityInfo()"><code>getEntityInfo()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getEntityInfo()">
<h3>getEntityInfo</h3>
<div class="member-signature"><span class="return-type"><a href="EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-230">getEntityInfo</a></span>()</div>
<div class="block">Returns the EntityInfo for this chain.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the EntityInfo object</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setEntityInfo(org.biojava.nbio.structure.EntityInfo)"><code>setEntityInfo(EntityInfo)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toString()">
<h3>toString</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-232">toString</a></span>()</div>
<dl class="notes">
<dt>Overrides:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a></code>&nbsp;in class&nbsp;<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getBJSequence()">
<h3>getBJSequence</h3>
<div class="member-signature"><span class="return-type"><a href="../core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;?&gt;</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-240">getBJSequence</a></span>()</div>
<div class="block">Converts the SEQRES groups of a Chain to a Biojava Sequence object.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the SEQRES groups of the Chain as a Sequence object.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getAtomSequence()">
<h3>getAtomSequence</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-247">getAtomSequence</a></span>()</div>
<div class="block">Returns the sequence of amino acids as it has been provided in the ATOM records.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>amino acid sequence as string</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getSeqResSequence()"><code>getSeqResSequence()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSeqResSequence()">
<h3>getSeqResSequence</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-255">getSeqResSequence</a></span>()</div>
<div class="block">Returns the PDB SEQRES sequence as a one-letter sequence string.
 Non-standard residues are represented by an "X".</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>one-letter PDB SEQRES sequence as string</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getAtomSequence()"><code>getAtomSequence()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSeqResGroups(org.biojava.nbio.structure.GroupType)">
<h3>getSeqResGroups</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-264">getSeqResGroups</a></span><wbr><span class="parameters">(<a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a>&nbsp;type)</span></div>
<div class="block">Returns a List of all SEQRES groups of a special type, one of: <a href="GroupType.html#AMINOACID"><code>GroupType.AMINOACID</code></a>,
 <a href="GroupType.html#HETATM"><code>GroupType.HETATM</code></a> or <a href="GroupType.html#NUCLEOTIDE"><code>GroupType.NUCLEOTIDE</code></a>.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>type</code> - a GroupType</dd>
<dt>Returns:</dt>
<dd>an List object</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setSeqResGroups(java.util.List)"><code>setSeqResGroups(List)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSeqResGroups()">
<h3>getSeqResGroups</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-274">getSeqResGroups</a></span>()</div>
<div class="block">Returns a list of all groups in SEQRES records of the chain, i.e.
 the aminoacids/nucleotides in the construct.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>a List of all Group objects of this chain</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setSeqResGroups(java.util.List)"><code>setSeqResGroups(List)</code></a></li>
<li><a href="#getSeqResLength()"><code>getSeqResLength()</code></a></li>
<li><a href="#getAtomGroups()"><code>getAtomGroups()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setSeqResGroups(java.util.List)">
<h3>setSeqResGroups</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-282">setSeqResGroups</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;seqResGroups)</span></div>
<div class="block">Sets the list of SeqResGroups for this chain.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seqResGroups</code> - a List of Group objects that from the SEQRES groups of this chain.</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getSeqResGroups()"><code>getSeqResGroups()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="setStructure(org.biojava.nbio.structure.Structure)">
<h3>setStructure</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-289">setStructure</a></span><wbr><span class="parameters">(<a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;parent)</span></div>
<div class="block">Sets the back-reference to its parent Structure.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>parent</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getStructure()">
<h3>getStructure</h3>
<div class="member-signature"><span class="return-type"><a href="Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-297">getStructure</a></span>()</div>
<div class="block">Returns the parent Structure of this chain.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the parent Structure object</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#setStructure(org.biojava.nbio.structure.Structure)"><code>setStructure(Structure)</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="toPDB()">
<h3>toPDB</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-303">toPDB</a></span>()</div>
<div class="block">Convert this Chain to a String in PDB format</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="toMMCIF()">
<h3>toMMCIF</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-309">toMMCIF</a></span>()</div>
<div class="block">Convert this Chain to a String in mmCIF format</div>
<dl class="notes">
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="setSeqMisMatches(java.util.List)">
<h3>setSeqMisMatches</h3>
<div class="member-signature"><span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-317">setSeqMisMatches</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;&nbsp;seqMisMatches)</span></div>
<div class="block">Sets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>seqMisMatches</code> - </dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getSeqMisMatches()">
<h3>getSeqMisMatches</h3>
<div class="member-signature"><span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a>&gt;</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-324">getSeqMisMatches</a></span>()</div>
<div class="block">Gets annotated sequence mismatches for this chain. This is based on the STRUCT_REF_SEQ_DIF mmCif category</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>a list of sequence mismatches (or null if none found)</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getEntityType()">
<h3>getEntityType</h3>
<div class="member-signature"><span class="return-type"><a href="EntityType.html" title="enum in org.biojava.nbio.structure">EntityType</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-331">getEntityType</a></span>()</div>
<div class="block">Returns the EntityType of this chain. Equivalent to getEntityInfo().getType()</div>
<dl class="notes">
<dt>Returns:</dt>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="EntityType.html" title="enum in org.biojava.nbio.structure"><code>EntityType</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="isWaterOnly()">
<h3>isWaterOnly</h3>
<div class="member-signature"><span class="return-type">boolean</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-338">isWaterOnly</a></span>()</div>
<div class="block">Tests if a chain is consisting of water molecules only</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>true if there are only solvent molecules in this chain.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="isPureNonPolymer()">
<h3>isPureNonPolymer</h3>
<div class="member-signature"><span class="return-type">boolean</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-345">isPureNonPolymer</a></span>()</div>
<div class="block">Returns true if the given chain is composed of non-polymeric (including water) groups only.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>true if only non-polymeric groups in this chain.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getPredominantGroupType()">
<h3>getPredominantGroupType</h3>
<div class="member-signature"><span class="return-type"><a href="GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a></span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-366">getPredominantGroupType</a></span>()</div>
<div class="block">Get the predominant <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> for a given Chain, following these
 rules:
 <ul>
 <li>if the ratio of number of residues of a certain
 <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> to total non-water residues is above the threshold
 <a href="StructureTools.html#RATIO_RESIDUES_TO_TOTAL">0.95</a>, then that <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> is
 returned</li> <li>if there is no <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> that is above the
 threshold then the <a href="GroupType.html" title="enum in org.biojava.nbio.structure"><code>GroupType</code></a> with most members is chosen, logging
 it</li>
 </ul>
 <p>
 See also <a href="chem/ChemComp.html#getPolymerType()"><code>ChemComp.getPolymerType()</code></a> and
 <a href="chem/ChemComp.html#getResidueType()"><code>ChemComp.getResidueType()</code></a> which follow the PDB chemical component
 dictionary and provide a much more accurate description of groups and
 their linking.</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>the predominant group type</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="isProtein()">
<h3>isProtein</h3>
<div class="member-signature"><span class="return-type">boolean</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-374">isProtein</a></span>()</div>
<div class="block">Tell whether given chain is a protein chain</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>true if protein, false if nucleotide or ligand</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getPredominantGroupType()"><code>getPredominantGroupType()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="isNucleicAcid()">
<h3>isNucleicAcid</h3>
<div class="member-signature"><span class="return-type">boolean</span>&nbsp;<span class="element-name"><a href="../../../../src-html/org/biojava/nbio/structure/Chain.html#line-382">isNucleicAcid</a></span>()</div>
<div class="block">Tell whether given chain is DNA or RNA</div>
<dl class="notes">
<dt>Returns:</dt>
<dd>true if nucleic acid, false if protein or ligand</dd>
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="#getPredominantGroupType()"><code>getPredominantGroupType()</code></a></li>
</ul>
</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
