<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>ChromosomeMappingTools (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="declaration: package: org.biojava.nbio.genome.util, class: ChromosomeMappingTools">
<meta name="generator" content="javadoc/ClassWriterImpl">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-declaration-page">
<script type="text/javascript">var pathtoroot = "../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../index.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="nav-bar-cell1-rev">Class</li>
<li><a href="class-use/ChromosomeMappingTools.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html#class">Help</a></li>
</ul>
<ul class="sub-nav-list-small">
<li>
<p>Summary:</p>
<ul>
<li>Nested</li>
<li><a href="#field-summary">Field</a></li>
<li><a href="#constructor-summary">Constr</a></li>
<li><a href="#method-summary">Method</a></li>
</ul>
</li>
<li>
<p>Detail:</p>
<ul>
<li><a href="#field-detail">Field</a></li>
<li><a href="#constructor-detail">Constr</a></li>
<li><a href="#method-detail">Method</a></li>
</ul>
</li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list">
<ul class="sub-nav-list">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#field-summary">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor-summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-summary">Method</a></li>
</ul>
<ul class="sub-nav-list">
<li>Detail:&nbsp;</li>
<li><a href="#field-detail">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor-detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method-detail">Method</a></li>
</ul>
</div>
<div class="nav-list-search"><a href="../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="sub-title"><span class="package-label-in-type">Package</span>&nbsp;<a href="package-summary.html">org.biojava.nbio.genome.util</a></div>
<h1 title="Class ChromosomeMappingTools" class="title">Class ChromosomeMappingTools</h1>
</div>
<div class="inheritance" title="Inheritance Tree"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">java.lang.Object</a>
<div class="inheritance">org.biojava.nbio.genome.util.ChromosomeMappingTools</div>
</div>
<section class="class-description" id="class-description">
<hr>
<div class="type-signature"><span class="modifiers">public class </span><span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-43">ChromosomeMappingTools</a></span>
<span class="extends-implements">extends <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></span></div>
<div class="block">A class that can map chromosomal positions to mRNA (coding sequence) positions.</div>
<dl class="notes">
<dt>Author:</dt>
<dd>Andreas Prlic</dd>
</dl>
</section>
<section class="summary">
<ul class="summary-list">
<!-- =========== FIELD SUMMARY =========== -->
<li>
<section class="field-summary" id="field-summary">
<h2>Field Summary</h2>
<div class="caption"><span>Fields</span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Field</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><code><a href="#CDS" class="member-name-link">CDS</a></code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static final <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><code><a href="#CHROMOSOME" class="member-name-link">CHROMOSOME</a></code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<li>
<section class="constructor-summary" id="constructor-summary">
<h2>Constructor Summary</h2>
<div class="caption"><span>Constructors</span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-constructor-name even-row-color"><code><a href="#%3Cinit%3E()" class="member-name-link">ChromosomeMappingTools</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<!-- ========== METHOD SUMMARY =========== -->
<li>
<section class="method-summary" id="method-summary">
<h2>Method Summary</h2>
<div id="method-summary-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="method-summary-table-tab0" role="tab" aria-selected="true" aria-controls="method-summary-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table', 3)" class="active-table-tab">All Methods</button><button id="method-summary-table-tab1" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab1', 3)" class="table-tab">Static Methods</button><button id="method-summary-table-tab4" role="tab" aria-selected="false" aria-controls="method-summary-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('method-summary-table', 'method-summary-table-tab4', 3)" class="table-tab">Concrete Methods</button></div>
<div id="method-summary-table.tabpanel" role="tabpanel" aria-labelledby="method-summary-table-tab0">
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#formatExonStructure(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">formatExonStructure</a><wbr>(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Pretty print the details of a GeneChromosomePosition to a String</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSExonRanges(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getCDSExonRanges</a><wbr>(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Extracts the exon boundaries in CDS coordinates.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSLength(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getCDSLength</a><wbr>(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get the length of the CDS in nucleotides.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSLengthForward(java.util.List,java.util.List,int,int)" class="member-name-link">getCDSLengthForward</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get the length of the coding sequence</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSLengthReverse(java.util.List,java.util.List,int,int)" class="member-name-link">getCDSLengthReverse</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get the length of the coding sequence</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSPosForChromosomeCoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getCDSPosForChromosomeCoordinate</a><wbr>(int&nbsp;coordinate,
 <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">I have a genomic coordinate, where is it on the mRNA</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static int</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSPosForward(int,java.util.List,java.util.List,int,int)" class="member-name-link">getCDSPosForward</a><wbr>(int&nbsp;chromPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the forward DNA strand.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static int</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSPosReverse(int,java.util.List,java.util.List,int,int)" class="member-name-link">getCDSPosReverse</a><wbr>(int&nbsp;chromPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the reverse DNA strand.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getCDSRegions(java.util.List,java.util.List,int,int)" class="member-name-link">getCDSRegions</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;origExonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;origExonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Extracts the exons boundaries in CDS coordinates corresponding to the forward DNA strand.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getChromosomalRangesForCDS(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getChromosomalRangesForCDS</a><wbr>(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Extracts the boundaries of the coding regions in chromosomal coordinates</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getChromosomePosForCDScoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getChromosomePosForCDScoordinate</a><wbr>(int&nbsp;cdsNucleotidePosition,
 <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Maps the position of a CDS nucleotide back to the genome</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getChromPosForward(int,java.util.List,java.util.List,int,int)" class="member-name-link">getChromPosForward</a><wbr>(int&nbsp;cdsPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get the CDS position mapped onto the chromosome position</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getChromPosReverse(int,java.util.List,java.util.List,int,int)" class="member-name-link">getChromPosReverse</a><wbr>(int&nbsp;cdsPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Get the CDS position mapped on the chromosome position</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,java.lang.String,java.util.List,java.util.List,int,int,java.lang.Character)" class="member-name-link">getTranscriptDNASequence</a><wbr>(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromosome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html" title="class or interface in java.lang" class="external-link">Character</a>&nbsp;orientation)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
</div>
<div class="col-first even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static <a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></div>
<div class="col-second even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)" class="member-name-link">getTranscriptDNASequence</a><wbr>(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
 <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;gcp)</code></div>
<div class="col-last even-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
</div>
<div class="col-first odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code>static void</code></div>
<div class="col-second odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4"><code><a href="#setCoordinateSystem(int)" class="member-name-link">setCoordinateSystem</a><wbr>(int&nbsp;baseInt)</code></div>
<div class="col-last odd-row-color method-summary-table method-summary-table-tab1 method-summary-table-tab4">&nbsp;</div>
</div>
</div>
</div>
<div class="inherited-list">
<h3 id="methods-inherited-from-class-java.lang.Object">Methods inherited from class&nbsp;java.lang.<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a></h3>
<code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#clone()" title="class or interface in java.lang" class="external-link">clone</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)" title="class or interface in java.lang" class="external-link">equals</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#finalize()" title="class or interface in java.lang" class="external-link">finalize</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#getClass()" title="class or interface in java.lang" class="external-link">getClass</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#hashCode()" title="class or interface in java.lang" class="external-link">hashCode</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notify()" title="class or interface in java.lang" class="external-link">notify</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#notifyAll()" title="class or interface in java.lang" class="external-link">notifyAll</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#toString()" title="class or interface in java.lang" class="external-link">toString</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait()" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long)" title="class or interface in java.lang" class="external-link">wait</a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#wait(long,int)" title="class or interface in java.lang" class="external-link">wait</a></code></div>
</section>
</li>
</ul>
</section>
<section class="details">
<ul class="details-list">
<!-- ============ FIELD DETAIL =========== -->
<li>
<section class="field-details" id="field-detail">
<h2>Field Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="CHROMOSOME">
<h3>CHROMOSOME</h3>
<div class="member-signature"><span class="modifiers">public static final</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-49">CHROMOSOME</a></span></div>
<dl class="notes">
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="../../../../../constant-values.html#org.biojava.nbio.genome.util.ChromosomeMappingTools.CHROMOSOME">Constant Field Values</a></li>
</ul>
</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="CDS">
<h3>CDS</h3>
<div class="member-signature"><span class="modifiers">public static final</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-50">CDS</a></span></div>
<dl class="notes">
<dt>See Also:</dt>
<dd>
<ul class="tag-list">
<li><a href="../../../../../constant-values.html#org.biojava.nbio.genome.util.ChromosomeMappingTools.CDS">Constant Field Values</a></li>
</ul>
</dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<li>
<section class="constructor-details" id="constructor-detail">
<h2>Constructor Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="&lt;init&gt;()">
<h3>ChromosomeMappingTools</h3>
<div class="member-signature"><span class="modifiers">public</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-43">ChromosomeMappingTools</a></span>()</div>
</section>
</li>
</ul>
</section>
</li>
<!-- ============ METHOD DETAIL ========== -->
<li>
<section class="method-details" id="method-detail">
<h2>Method Details</h2>
<ul class="member-list">
<li>
<section class="detail" id="setCoordinateSystem(int)">
<h3>setCoordinateSystem</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">void</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-53">setCoordinateSystem</a></span><wbr><span class="parameters">(int&nbsp;baseInt)</span></div>
</section>
</li>
<li>
<section class="detail" id="formatExonStructure(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<h3>formatExonStructure</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-63">formatExonStructure</a></span><wbr><span class="parameters">(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</span></div>
<div class="block">Pretty print the details of a GeneChromosomePosition to a String</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromosomePosition</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSLength(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<h3>getCDSLength</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-210">getCDSLength</a></span><wbr><span class="parameters">(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</span></div>
<div class="block">Get the length of the CDS in nucleotides.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromPos</code> - </dd>
<dt>Returns:</dt>
<dd>length of the CDS in nucleotides.</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getChromosomePosForCDScoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<h3>getChromosomePosForCDScoordinate</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-232">getChromosomePosForCDScoordinate</a></span><wbr><span class="parameters">(int&nbsp;cdsNucleotidePosition,
 <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</span></div>
<div class="block">Maps the position of a CDS nucleotide back to the genome</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>cdsNucleotidePosition</code> - </dd>
<dt>Returns:</dt>
<dd>a ChromPos object</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getChromPosReverse(int,java.util.List,java.util.List,int,int)">
<h3>getChromPosReverse</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-277">getChromPosReverse</a></span><wbr><span class="parameters">(int&nbsp;cdsPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</span></div>
<div class="block">Get the CDS position mapped on the chromosome position</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getChromPosForward(int,java.util.List,java.util.List,int,int)">
<h3>getChromPosForward</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-415">getChromPosForward</a></span><wbr><span class="parameters">(int&nbsp;cdsPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</span></div>
<div class="block">Get the CDS position mapped onto the chromosome position</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSLengthReverse(java.util.List,java.util.List,int,int)">
<h3>getCDSLengthReverse</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-501">getCDSLengthReverse</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</span></div>
<div class="block">Get the length of the coding sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSLengthForward(java.util.List,java.util.List,int,int)">
<h3>getCDSLengthForward</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-548">getCDSLengthForward</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</span></div>
<div class="block">Get the length of the coding sequence</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>exonStarts</code> - </dd>
<dd><code>exonEnds</code> - </dd>
<dd><code>cdsStart</code> - </dd>
<dd><code>cdsEnd</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSExonRanges(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<h3>getCDSExonRanges</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-577">getCDSExonRanges</a></span><wbr><span class="parameters">(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</span></div>
<div class="block">Extracts the exon boundaries in CDS coordinates. (needs to be divided by 3 to get AA positions)</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromPos</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getChromosomalRangesForCDS(org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<h3>getChromosomalRangesForCDS</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-588">getChromosomalRangesForCDS</a></span><wbr><span class="parameters">(<a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromPos)</span></div>
<div class="block">Extracts the boundaries of the coding regions in chromosomal coordinates</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromPos</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSPosForChromosomeCoordinate(int,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<h3>getCDSPosForChromosomeCoordinate</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-779">getCDSPosForChromosomeCoordinate</a></span><wbr><span class="parameters">(int&nbsp;coordinate,
 <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;chromosomePosition)</span></div>
<div class="block">I have a genomic coordinate, where is it on the mRNA</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>coordinate</code> - </dd>
<dd><code>chromosomePosition</code> - </dd>
<dt>Returns:</dt>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSPosForward(int,java.util.List,java.util.List,int,int)">
<h3>getCDSPosForward</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-809">getCDSPosForward</a></span><wbr><span class="parameters">(int&nbsp;chromPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</span></div>
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the forward DNA strand.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromPos</code> - The genetic coordinate on a chromosome</dd>
<dd><code>exonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>exonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dt>Returns:</dt>
<dd>the position of the nucleotide base on the mRNA sequence corresponding to the input genetic coordinate (base 1)</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSPosReverse(int,java.util.List,java.util.List,int,int)">
<h3>getCDSPosReverse</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type">int</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-856">getCDSPosReverse</a></span><wbr><span class="parameters">(int&nbsp;chromPos,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</span></div>
<div class="block">Converts the genetic coordinate to the position of the nucleotide on the mRNA sequence for a gene
 living on the reverse DNA strand.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromPos</code> - The genetic coordinate on a chromosome</dd>
<dd><code>exonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>exonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dt>Returns:</dt>
<dd>the position of the nucleotide base on the mRNA sequence corresponding to the input genetic coordinate (base 1)</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getCDSRegions(java.util.List,java.util.List,int,int)">
<h3>getCDSRegions</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;com.google.common.collect.Range&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-899">getCDSRegions</a></span><wbr><span class="parameters">(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;origExonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;origExonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd)</span></div>
<div class="block">Extracts the exons boundaries in CDS coordinates corresponding to the forward DNA strand.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>origExonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>origExonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dt>Returns:</dt>
<dd>the list of genetic positions corresponding to the exons boundaries in CDS coordinates</dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,org.biojava.nbio.genome.parsers.genename.GeneChromosomePosition)">
<h3>getTranscriptDNASequence</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-939">getTranscriptDNASequence</a></span><wbr><span class="parameters">(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
 <a href="../parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a>&nbsp;gcp)</span>
                                            throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></span></div>
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>twoBitFacade</code> - the facade that provide an access to a 2bit file</dd>
<dd><code>gcp</code> - The container with chromosomal positions</dd>
<dt>Returns:</dt>
<dd>the DNA sequence transcribed from the input genetic coordinates</dd>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></code></dd>
</dl>
</section>
</li>
<li>
<section class="detail" id="getTranscriptDNASequence(org.biojava.nbio.genome.parsers.twobit.TwoBitFacade,java.lang.String,java.util.List,java.util.List,int,int,java.lang.Character)">
<h3>getTranscriptDNASequence</h3>
<div class="member-signature"><span class="modifiers">public static</span>&nbsp;<span class="return-type"><a href="../../core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span>&nbsp;<span class="element-name"><a href="../../../../../src-html/org/biojava/nbio/genome/util/ChromosomeMappingTools.html#line-955">getTranscriptDNASequence</a></span><wbr><span class="parameters">(<a href="../parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a>&nbsp;twoBitFacade,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;chromosome,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonStarts,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;exonEnds,
 int&nbsp;cdsStart,
 int&nbsp;cdsEnd,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Character.html" title="class or interface in java.lang" class="external-link">Character</a>&nbsp;orientation)</span>
                                            throws <span class="exceptions"><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></span></div>
<div class="block">Extracts the DNA sequence transcribed from the input genetic coordinates.</div>
<dl class="notes">
<dt>Parameters:</dt>
<dd><code>chromosome</code> - the name of the chromosome</dd>
<dd><code>exonStarts</code> - The list holding the genetic coordinates pointing to the start positions of the exons (including UTR regions)</dd>
<dd><code>exonEnds</code> - The list holding the genetic coordinates pointing to the end positions of the exons (including UTR regions)</dd>
<dd><code>cdsStart</code> - The start position of a coding region</dd>
<dd><code>cdsEnd</code> - The end position of a coding region</dd>
<dd><code>orientation</code> - The orientation of the strand where the gene is living</dd>
<dt>Returns:</dt>
<dd>the DNA sequence transcribed from the input genetic coordinates</dd>
<dt>Throws:</dt>
<dd><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Exception.html" title="class or interface in java.lang" class="external-link">Exception</a></code></dd>
</dl>
</section>
</li>
</ul>
</section>
</li>
</ul>
</section>
<!-- ========= END OF CLASS DATA ========= -->
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
