<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:34 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>AbstractSequence (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="AbstractSequence (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AbstractSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava.nbio.core.sequence.template"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/template/AbstractSequence.html" target="_top">Frames</a></li>
<li><a href="AbstractSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.core.sequence.template</div>
<h2 title="Class AbstractSequence" class="title">Class AbstractSequence&lt;C extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.core.sequence.template.AbstractSequence&lt;C&gt;</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl><dt><span class="strong">Type Parameters:</span></dt><dd><code>C</code> - </dd></dl>
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;C&gt;, <a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</dd>
</dl>
<dl>
<dt>Direct Known Subclasses:</dt>
<dd><a href="../../../../../../org/biojava/nbio/core/sequence/BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>, <a href="../../../../../../org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></dd>
</dl>
<hr>
<br>
<pre>public abstract class <span class="strong">AbstractSequence&lt;C extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</span>
extends <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>
implements <a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;</pre>
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== NESTED CLASS SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="nested_class_summary">
<!--   -->
</a>
<h3>Nested Class Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Nested Class Summary table, listing nested classes, and an explanation">
<caption><span>Nested Classes</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></strong></code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#AbstractSequence()">AbstractSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#AbstractSequence(org.biojava.nbio.core.sequence.template.SequenceReader,%20org.biojava.nbio.core.sequence.template.CompoundSet)">AbstractSequence</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
                <a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code>
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence data and the location
 of the sequence data.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#AbstractSequence(java.lang.String,%20org.biojava.nbio.core.sequence.template.CompoundSet)">AbstractSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
                <a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code>
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code>
<div class="block">Add a feature to this sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addFeature(int,%20int,%20org.biojava.nbio.core.sequence.features.FeatureInterface)">addFeature</a></strong>(int&nbsp;bioStart,
          int&nbsp;bioEnd,
          <a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code>
<div class="block">Method to help set the proper details for a feature as it relates to a sequence
 where the feature needs to have a location on the sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#addNote(java.lang.String)">addNote</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;note)</code>
<div class="block">Add notes about this sequence that will get exported for GFF3</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#countCompounds(C...)">countCompounds</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</code>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAccession()">getAccession</a></strong>()</code>
<div class="block">Returns the AccessionID this location is currently bound with</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAnnotationType()">getAnnotationType</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getAsList()">getAsList</a></strong>()</code>
<div class="block">Returns the Sequence as a List of compounds</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioBegin()">getBioBegin</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getBioEnd()">getBioEnd</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundAt(int)">getCompoundAt</a></strong>(int&nbsp;position)</code>
<div class="block">Returns the Compound at the given biological index</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getCompoundSet()">getCompoundSet</a></strong>()</code>
<div class="block">Gets the compound set used to back this Sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.features">DatabaseReferenceInterface</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDatabaseReferences()">getDatabaseReferences</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getDescription()">getDescription</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureRetriever.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureRetriever</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatureRetriever()">getFeatureRetriever</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures()">getFeatures</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(int)">getFeatures</a></strong>(int&nbsp;bioSequencePosition)</code>
<div class="block">Return features at a sequence position</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeatures(java.lang.String,%20int)">getFeatures</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;featureType,
           int&nbsp;bioSequencePosition)</code>
<div class="block">Return features at a sequence position by type</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesByType(java.lang.String)">getFeaturesByType</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeaturesKeyWordInterface</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getFeaturesKeyWord()">getFeaturesKeyWord</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getIndexOf(C)">getIndexOf</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</code>
<div class="block">Scans through the Sequence looking for the first occurrence of the given
 compound</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getInverse()">getInverse</a></strong>()</code>
<div class="block">Does the <em>right thing</em> to get the inverse of the current
 Sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLastIndexOf(C)">getLastIndexOf</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</code>
<div class="block">Scans through the Sequence looking for the last occurrence of the given
 compound</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getLength()">getLength</a></strong>()</code>
<div class="block">Returns the length of the Sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getNotesList()">getNotesList</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getOriginalHeader()">getOriginalHeader</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getParentSequence()">getParentSequence</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getProxySequenceReader()">getProxySequenceReader</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString()">getSequenceAsString</a></strong>()</code>
<div class="block">Default case is to assume strand is positive because only CDSSequence can be either positive or negative Strand.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceAsString(java.lang.Integer,%20java.lang.Integer,%20org.biojava.nbio.core.sequence.Strand)">getSequenceAsString</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioStart,
                   <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd,
                   <a href="../../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSequenceScore()">getSequenceScore</a></strong>()</code>
<div class="block">Provide place holder for a metric that indicate a score associated with the sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSource()">getSource</a></strong>()</code>
<div class="block">Added support for the source of this sequence for GFF3 export
 If a sub sequence doesn't have  source then check for parent source</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioStart,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</code>
<div class="block">Returns a portion of the sequence from the different positions.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence">TaxonomyID</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getTaxonomy()">getTaxonomy</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getUserCollection()">getUserCollection</a></strong>()</code>
<div class="block">Provided for convince if the developer needs to associate data with a sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#iterator()">iterator</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">removeFeature</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</code>
<div class="block">Remove a feature from the sequence</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#removeNote(java.lang.String)">removeNote</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;note)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAccession(org.biojava.nbio.core.sequence.AccessionID)">setAccession</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setAnnotationType(org.biojava.nbio.core.sequence.template.AbstractSequence.AnnotationType)">setAnnotationType</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;annotationType)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioBegin(java.lang.Integer)">setBioBegin</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;begin)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setBioEnd(java.lang.Integer)">setBioEnd</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setCompoundSet(org.biojava.nbio.core.sequence.template.CompoundSet)">setCompoundSet</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDatabaseReferences(org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface)">setDatabaseReferences</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;databaseReferences)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setDescription(java.lang.String)">setDescription</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeatureRetriever(org.biojava.nbio.core.sequence.features.FeatureRetriever)">setFeatureRetriever</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureRetriever.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureRetriever</a>&nbsp;featureRetriever)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setFeaturesKeyWord(org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface)">setFeaturesKeyWord</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;featuresKeyWord)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setNotesList(java.util.ArrayList)">setNotesList</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;notesList)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setOriginalHeader(java.lang.String)">setOriginalHeader</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;originalHeader)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setParentSequence(org.biojava.nbio.core.sequence.template.AbstractSequence)">setParentSequence</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;parentSequence)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setProxySequenceReader(org.biojava.nbio.core.sequence.template.SequenceReader)">setProxySequenceReader</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader)</code>
<div class="block">Very important method that allows external mappings of sequence data and features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSequenceScore(java.lang.Double)">setSequenceScore</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;sequenceScore)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setSource(java.lang.String)">setSource</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;source)</code>
<div class="block">Added support for the source of this sequence for GFF3 export</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setTaxonomy(org.biojava.nbio.core.sequence.TaxonomyID)">setTaxonomy</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence">TaxonomyID</a>&nbsp;taxonomy)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#setUserCollection(java.util.Collection)">setUserCollection</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&gt;&nbsp;userCollection)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="AbstractSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AbstractSequence</h4>
<pre>public&nbsp;AbstractSequence()</pre>
</li>
</ul>
<a name="AbstractSequence(java.lang.String, org.biojava.nbio.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>AbstractSequence</h4>
<pre>public&nbsp;AbstractSequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqString,
                <a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)
                 throws <a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></pre>
<div class="block">Create a Sequence from a simple string where the values should be found in compoundSet</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>seqString</code> - </dd><dd><code>compoundSet</code> - </dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></code></dd></dl>
</li>
</ul>
<a name="AbstractSequence(org.biojava.nbio.core.sequence.template.SequenceReader, org.biojava.nbio.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>AbstractSequence</h4>
<pre>public&nbsp;AbstractSequence(<a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader,
                <a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</pre>
<div class="block">A ProxySequenceReader allows abstraction of both the storage of the sequence data and the location
 of the sequence data. A variety of use cases are possible. A ProxySequenceReader that knows the offset and of the sequence in
 a large fasta file. A ProxySequenceReader that can pull Sequence data from UniProt, NCBI or a custom database.
 If the ProxySequenceReader implements various interfaces then the sequence will set those interfaces so that calls to
 various methods will be valid.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>proxyLoader</code> - </dd><dd><code>compoundSet</code> - </dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="setProxySequenceReader(org.biojava.nbio.core.sequence.template.SequenceReader)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setProxySequenceReader</h4>
<pre>public&nbsp;void&nbsp;setProxySequenceReader(<a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;proxyLoader)</pre>
<div class="block">Very important method that allows external mappings of sequence data and features. This method
 will gain additional interface inspection that allows external data sources with knowledge
 of features for a sequence to be supported.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>proxyLoader</code> - </dd></dl>
</li>
</ul>
<a name="getProxySequenceReader()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getProxySequenceReader</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getProxySequenceReader()</pre>
</li>
</ul>
<a name="getBioBegin()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioBegin</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;getBioBegin()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the bioBegin</dd></dl>
</li>
</ul>
<a name="setBioBegin(java.lang.Integer)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBioBegin</h4>
<pre>public&nbsp;void&nbsp;setBioBegin(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;begin)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>bioBegin</code> - the bioBegin to set</dd></dl>
</li>
</ul>
<a name="getBioEnd()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getBioEnd</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;getBioEnd()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the bioEnd</dd></dl>
</li>
</ul>
<a name="setBioEnd(java.lang.Integer)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setBioEnd</h4>
<pre>public&nbsp;void&nbsp;setBioEnd(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;end)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>bioEnd</code> - the bioEnd to set</dd></dl>
</li>
</ul>
<a name="getUserCollection()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getUserCollection</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&gt;&nbsp;getUserCollection()</pre>
<div class="block">Provided for convince if the developer needs to associate data with a sequence</div>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="setUserCollection(java.util.Collection)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setUserCollection</h4>
<pre>public&nbsp;void&nbsp;setUserCollection(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&gt;&nbsp;userCollection)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>userCollection</code> - </dd></dl>
</li>
</ul>
<a name="getAnnotationType()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAnnotationType</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;getAnnotationType()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the annotation</dd></dl>
</li>
</ul>
<a name="setAnnotationType(org.biojava.nbio.core.sequence.template.AbstractSequence.AnnotationType)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAnnotationType</h4>
<pre>public&nbsp;void&nbsp;setAnnotationType(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a>&nbsp;annotationType)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>annotation</code> - the annotation to set</dd></dl>
</li>
</ul>
<a name="getDescription()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDescription</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getDescription()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the description</dd></dl>
</li>
</ul>
<a name="setDescription(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDescription</h4>
<pre>public&nbsp;void&nbsp;setDescription(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;description)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>description</code> - the description to set</dd></dl>
</li>
</ul>
<a name="getOriginalHeader()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getOriginalHeader</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getOriginalHeader()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the originalHeader</dd></dl>
</li>
</ul>
<a name="setOriginalHeader(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setOriginalHeader</h4>
<pre>public&nbsp;void&nbsp;setOriginalHeader(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;originalHeader)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>originalHeader</code> - the originalHeader to set</dd></dl>
</li>
</ul>
<a name="getParentSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getParentSequence</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;getParentSequence()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the parentSequence</dd></dl>
</li>
</ul>
<a name="setParentSequence(org.biojava.nbio.core.sequence.template.AbstractSequence)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setParentSequence</h4>
<pre>public&nbsp;void&nbsp;setParentSequence(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;?&gt;&nbsp;parentSequence)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>parentSequence</code> - the parentSequence to set</dd></dl>
</li>
</ul>
<a name="getSource()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSource</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSource()</pre>
<div class="block">Added support for the source of this sequence for GFF3 export
 If a sub sequence doesn't have  source then check for parent source</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the source</dd></dl>
</li>
</ul>
<a name="setSource(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSource</h4>
<pre>public&nbsp;void&nbsp;setSource(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;source)</pre>
<div class="block">Added support for the source of this sequence for GFF3 export</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>source</code> - the source to set</dd></dl>
</li>
</ul>
<a name="addNote(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addNote</h4>
<pre>public&nbsp;void&nbsp;addNote(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;note)</pre>
<div class="block">Add notes about this sequence that will get exported for GFF3</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>note</code> - </dd></dl>
</li>
</ul>
<a name="removeNote(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeNote</h4>
<pre>public&nbsp;void&nbsp;removeNote(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;note)</pre>
</li>
</ul>
<a name="getNotesList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getNotesList</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;getNotesList()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the notesList</dd></dl>
</li>
</ul>
<a name="setNotesList(java.util.ArrayList)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setNotesList</h4>
<pre>public&nbsp;void&nbsp;setNotesList(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html?is-external=true" title="class or interface in java.util">ArrayList</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;&nbsp;notesList)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>notesList</code> - the notesList to set</dd></dl>
</li>
</ul>
<a name="getSequenceScore()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceScore</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;getSequenceScore()</pre>
<div class="block">Provide place holder for a metric that indicate a score associated with the sequence</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>the sequenceScore</dd></dl>
</li>
</ul>
<a name="setSequenceScore(java.lang.Double)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setSequenceScore</h4>
<pre>public&nbsp;void&nbsp;setSequenceScore(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Double.html?is-external=true" title="class or interface in java.lang">Double</a>&nbsp;sequenceScore)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>sequenceScore</code> - the sequenceScore to set</dd></dl>
</li>
</ul>
<a name="getFeatures(java.lang.String, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeatures(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;featureType,
                                                        int&nbsp;bioSequencePosition)</pre>
<div class="block">Return features at a sequence position by type</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>featureType</code> - </dd><dd><code>bioSequencePosition</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getFeatures(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeatures(int&nbsp;bioSequencePosition)</pre>
<div class="block">Return features at a sequence position</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>featureType</code> - </dd><dd><code>bioSequencePosition</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getFeatures()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatures</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeatures()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="addFeature(int, int, org.biojava.nbio.core.sequence.features.FeatureInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addFeature</h4>
<pre>public&nbsp;void&nbsp;addFeature(int&nbsp;bioStart,
              int&nbsp;bioEnd,
              <a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</pre>
<div class="block">Method to help set the proper details for a feature as it relates to a sequence
 where the feature needs to have a location on the sequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>bioStart</code> - </dd><dd><code>bioEnd</code> - </dd><dd><code>feature</code> - </dd></dl>
</li>
</ul>
<a name="addFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>addFeature</h4>
<pre>public&nbsp;void&nbsp;addFeature(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</pre>
<div class="block">Add a feature to this sequence. The feature will be added to the collection where the order is start position and if more than
 one feature at the same start position then longest is added first. This helps on doing feature layout for displaying features
 in SequenceFeaturePanel</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>feature</code> - </dd></dl>
</li>
</ul>
<a name="removeFeature(org.biojava.nbio.core.sequence.features.FeatureInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>removeFeature</h4>
<pre>public&nbsp;void&nbsp;removeFeature(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;feature)</pre>
<div class="block">Remove a feature from the sequence</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>feature</code> - </dd></dl>
</li>
</ul>
<a name="getFeaturesByType(java.lang.String)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeaturesByType</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;,<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&gt;&nbsp;getFeaturesByType(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;type)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>type</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getFeaturesKeyWord()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeaturesKeyWord</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;getFeaturesKeyWord()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the featuresKeyWord</dd></dl>
</li>
</ul>
<a name="setFeaturesKeyWord(org.biojava.nbio.core.sequence.features.FeaturesKeyWordInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFeaturesKeyWord</h4>
<pre>public&nbsp;void&nbsp;setFeaturesKeyWord(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeaturesKeyWordInterface</a>&nbsp;featuresKeyWord)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>featuresKeyWord</code> - the featuresKeyWord to set</dd></dl>
</li>
</ul>
<a name="getDatabaseReferences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getDatabaseReferences</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;getDatabaseReferences()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the databaseReferences</dd></dl>
</li>
</ul>
<a name="setDatabaseReferences(org.biojava.nbio.core.sequence.features.DatabaseReferenceInterface)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setDatabaseReferences</h4>
<pre>public&nbsp;void&nbsp;setDatabaseReferences(<a href="../../../../../../org/biojava/nbio/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.features">DatabaseReferenceInterface</a>&nbsp;databaseReferences)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>databaseReferences</code> - the databaseReferences to set</dd></dl>
</li>
</ul>
<a name="getFeatureRetriever()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getFeatureRetriever</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureRetriever.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureRetriever</a>&nbsp;getFeatureRetriever()</pre>
</li>
</ul>
<a name="setFeatureRetriever(org.biojava.nbio.core.sequence.features.FeatureRetriever)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFeatureRetriever</h4>
<pre>public&nbsp;void&nbsp;setFeatureRetriever(<a href="../../../../../../org/biojava/nbio/core/sequence/features/FeatureRetriever.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureRetriever</a>&nbsp;featureRetriever)</pre>
</li>
</ul>
<a name="getAccession()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAccession</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;getAccession()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html#getAccession()">Accessioned</a></code></strong></div>
<div class="block">Returns the AccessionID this location is currently bound with</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html#getAccession()">getAccession</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>the accession</dd></dl>
</li>
</ul>
<a name="setAccession(org.biojava.nbio.core.sequence.AccessionID)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setAccession</h4>
<pre>public&nbsp;void&nbsp;setAccession(<a href="../../../../../../org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a>&nbsp;accession)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>accession</code> - the accession to set</dd></dl>
</li>
</ul>
<a name="getTaxonomy()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTaxonomy</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence">TaxonomyID</a>&nbsp;getTaxonomy()</pre>
<dl><dt><span class="strong">Returns:</span></dt><dd>the species</dd></dl>
</li>
</ul>
<a name="setTaxonomy(org.biojava.nbio.core.sequence.TaxonomyID)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setTaxonomy</h4>
<pre>public&nbsp;void&nbsp;setTaxonomy(<a href="../../../../../../org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence">TaxonomyID</a>&nbsp;taxonomy)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>species</code> - the species to set</dd></dl>
</li>
</ul>
<a name="getCompoundSet()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompoundSet</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getCompoundSet()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getCompoundSet()">Sequence</a></code></strong></div>
<div class="block">Gets the compound set used to back this Sequence</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getCompoundSet()">getCompoundSet</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
</dl>
</li>
</ul>
<a name="setCompoundSet(org.biojava.nbio.core.sequence.template.CompoundSet)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setCompoundSet</h4>
<pre>public&nbsp;void&nbsp;setCompoundSet(<a href="../../../../../../org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;compoundSet)</pre>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="getSequenceAsString(java.lang.Integer, java.lang.Integer, org.biojava.nbio.core.sequence.Strand)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAsString</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSequenceAsString(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioStart,
                         <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd,
                         <a href="../../../../../../org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a>&nbsp;strand)</pre>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>begin</code> - </dd><dd><code>end</code> - </dd><dd><code>strand</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getSequenceAsString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequenceAsString</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getSequenceAsString()</pre>
<div class="block">Default case is to assume strand is positive because only CDSSequence can be either positive or negative Strand.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getSequenceAsString()">getSequenceAsString</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getAsList()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getAsList</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getAsList()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getAsList()">Sequence</a></code></strong></div>
<div class="block">Returns the Sequence as a List of compounds</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getAsList()">getAsList</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getCompoundAt(int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getCompoundAt</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;getCompoundAt(int&nbsp;position)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getCompoundAt(int)">Sequence</a></code></strong></div>
<div class="block">Returns the Compound at the given biological index</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getCompoundAt(int)">getCompoundAt</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>position</code> - The 1-indexed position of the amino acid</dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getIndexOf(org.biojava.nbio.core.sequence.template.Compound)">
<!--   -->
</a><a name="getIndexOf(C)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getIndexOf</h4>
<pre>public&nbsp;int&nbsp;getIndexOf(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getIndexOf(C)">Sequence</a></code></strong></div>
<div class="block">Scans through the Sequence looking for the first occurrence of the given
 compound</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getIndexOf(C)">getIndexOf</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>compound</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>The first index of compound in this sequence (1-based)</dd></dl>
</li>
</ul>
<a name="getLastIndexOf(org.biojava.nbio.core.sequence.template.Compound)">
<!--   -->
</a><a name="getLastIndexOf(C)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLastIndexOf</h4>
<pre>public&nbsp;int&nbsp;getLastIndexOf(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&nbsp;compound)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLastIndexOf(C)">Sequence</a></code></strong></div>
<div class="block">Scans through the Sequence looking for the last occurrence of the given
 compound</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLastIndexOf(C)">getLastIndexOf</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>compound</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd>The last index of compound in this sequence (1-based)</dd></dl>
</li>
</ul>
<a name="getLength()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getLength</h4>
<pre>public&nbsp;int&nbsp;getLength()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLength()">Sequence</a></code></strong></div>
<div class="block">Returns the length of the Sequence</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getLength()">getLength</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getSubSequence(java.lang.Integer, java.lang.Integer)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSubSequence</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getSubSequence(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioStart,
                             <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;bioEnd)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">Sequence</a></code></strong></div>
<div class="block">Returns a portion of the sequence from the different positions. This is
 indexed from 1</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getSubSequence(java.lang.Integer,%20java.lang.Integer)">getSubSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>bioStart</code> - </dd><dd><code>bioEnd</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="iterator()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>iterator</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html?is-external=true" title="class or interface in java.util">Iterator</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;iterator()</pre>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true#iterator()" title="class or interface in java.lang">iterator</a></code>&nbsp;in interface&nbsp;<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Iterable.html?is-external=true" title="class or interface in java.lang">Iterable</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="countCompounds(org.biojava.nbio.core.sequence.template.Compound[])">
<!--   -->
</a><a name="countCompounds(C...)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>countCompounds</h4>
<pre>public&nbsp;int&nbsp;countCompounds(<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>...&nbsp;compounds)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#countCompounds(C...)">Sequence</a></code></strong></div>
<div class="block">Returns the number of times we found a compound in the Sequence</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#countCompounds(C...)">countCompounds</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>compounds</code> - </dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
<a name="getInverse()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getInverse</h4>
<pre>public&nbsp;<a href="../../../../../../org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a>&gt;&nbsp;getInverse()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getInverse()">Sequence</a></code></strong></div>
<div class="block">Does the <em>right thing</em> to get the inverse of the current
 Sequence. This means either reversing the Sequence and optionally
 complementing the Sequence.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html#getInverse()">getInverse</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="type parameter in AbstractSequence">C</a> extends <a href="../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</code></dd>
<dt><span class="strong">Returns:</span></dt><dd></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/AbstractSequence.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava.nbio.core.sequence.template"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/template/AbstractSequence.html" target="_top">Frames</a></li>
<li><a href="AbstractSequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li><a href="#nested_class_summary">Nested</a>&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
