<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created on 01-21-2010<a name="line.20"></a>
<span class="sourceLineNo">021</span> */<a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.nbio.core.sequence.io;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import org.biojava.nbio.core.sequence.DNASequence;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.core.sequence.RNASequence;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.sequence.compound.DNACompoundSet;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.sequence.compound.NucleotideCompound;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.sequence.compound.RNACompoundSet;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.nbio.core.sequence.template.AbstractSequence;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.slf4j.Logger;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.slf4j.LoggerFactory;<a name="line.34"></a>
<span class="sourceLineNo">035</span><a name="line.35"></a>
<span class="sourceLineNo">036</span>import java.io.File;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import java.io.FileInputStream;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import java.io.InputStream;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import java.util.LinkedHashMap;<a name="line.39"></a>
<span class="sourceLineNo">040</span><a name="line.40"></a>
<span class="sourceLineNo">041</span>/**<a name="line.41"></a>
<span class="sourceLineNo">042</span> *<a name="line.42"></a>
<span class="sourceLineNo">043</span> * @author Scooter Willis &lt;willishf at gmail dot com&gt;<a name="line.43"></a>
<span class="sourceLineNo">044</span> */<a name="line.44"></a>
<span class="sourceLineNo">045</span>public class GenbankReaderHelper {<a name="line.45"></a>
<span class="sourceLineNo">046</span><a name="line.46"></a>
<span class="sourceLineNo">047</span>        private final static Logger logger = LoggerFactory.getLogger(GenbankReaderHelper.class);<a name="line.47"></a>
<span class="sourceLineNo">048</span><a name="line.48"></a>
<span class="sourceLineNo">049</span>        /**<a name="line.49"></a>
<span class="sourceLineNo">050</span>         * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects<a name="line.50"></a>
<span class="sourceLineNo">051</span>         * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested<a name="line.51"></a>
<span class="sourceLineNo">052</span>         * in one sequence based on accession id.<a name="line.52"></a>
<span class="sourceLineNo">053</span>         * @param file<a name="line.53"></a>
<span class="sourceLineNo">054</span>         * @param lazySequenceLoad<a name="line.54"></a>
<span class="sourceLineNo">055</span>         * @return<a name="line.55"></a>
<span class="sourceLineNo">056</span>         * @throws Exception<a name="line.56"></a>
<span class="sourceLineNo">057</span>         */<a name="line.57"></a>
<span class="sourceLineNo">058</span>        public static LinkedHashMap&lt;String, DNASequence&gt; readGenbankDNASequence(File file, boolean lazySequenceLoad) throws Exception {<a name="line.58"></a>
<span class="sourceLineNo">059</span>                if (!lazySequenceLoad) {<a name="line.59"></a>
<span class="sourceLineNo">060</span>                        return readGenbankDNASequence(file);<a name="line.60"></a>
<span class="sourceLineNo">061</span>                }<a name="line.61"></a>
<span class="sourceLineNo">062</span><a name="line.62"></a>
<span class="sourceLineNo">063</span>                GenbankReader&lt;DNASequence, NucleotideCompound&gt; GenbankProxyReader =<a name="line.63"></a>
<span class="sourceLineNo">064</span>                                new GenbankReader&lt;DNASequence, NucleotideCompound&gt;(<a name="line.64"></a>
<span class="sourceLineNo">065</span>                                                file,<a name="line.65"></a>
<span class="sourceLineNo">066</span>                                                new GenericGenbankHeaderParser&lt;DNASequence, NucleotideCompound&gt;(),<a name="line.66"></a>
<span class="sourceLineNo">067</span>                                                new FileProxyDNASequenceCreator(<a name="line.67"></a>
<span class="sourceLineNo">068</span>                                                                file,<a name="line.68"></a>
<span class="sourceLineNo">069</span>                                                                DNACompoundSet.getDNACompoundSet(),<a name="line.69"></a>
<span class="sourceLineNo">070</span>                                                                new GenbankSequenceParser&lt;AbstractSequence&lt;NucleotideCompound&gt;, NucleotideCompound&gt;()<a name="line.70"></a>
<span class="sourceLineNo">071</span>                                                        )<a name="line.71"></a>
<span class="sourceLineNo">072</span>                                        );<a name="line.72"></a>
<span class="sourceLineNo">073</span>                return GenbankProxyReader.process();<a name="line.73"></a>
<span class="sourceLineNo">074</span><a name="line.74"></a>
<span class="sourceLineNo">075</span>        }<a name="line.75"></a>
<span class="sourceLineNo">076</span><a name="line.76"></a>
<span class="sourceLineNo">077</span>        /**<a name="line.77"></a>
<span class="sourceLineNo">078</span>         * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects<a name="line.78"></a>
<span class="sourceLineNo">079</span>         * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested<a name="line.79"></a>
<span class="sourceLineNo">080</span>         * in one sequence based on accession id.<a name="line.80"></a>
<span class="sourceLineNo">081</span>         * @param file<a name="line.81"></a>
<span class="sourceLineNo">082</span>         * @param lazySequenceLoad<a name="line.82"></a>
<span class="sourceLineNo">083</span>         * @return<a name="line.83"></a>
<span class="sourceLineNo">084</span>         * @throws Exception<a name="line.84"></a>
<span class="sourceLineNo">085</span>         */<a name="line.85"></a>
<span class="sourceLineNo">086</span>        public static LinkedHashMap&lt;String, ProteinSequence&gt; readGenbankProteinSequence(File file, boolean lazySequenceLoad) throws Exception {<a name="line.86"></a>
<span class="sourceLineNo">087</span>                if (!lazySequenceLoad) {<a name="line.87"></a>
<span class="sourceLineNo">088</span>                        return readGenbankProteinSequence(file);<a name="line.88"></a>
<span class="sourceLineNo">089</span>                }<a name="line.89"></a>
<span class="sourceLineNo">090</span><a name="line.90"></a>
<span class="sourceLineNo">091</span>                GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; GenbankProxyReader =<a name="line.91"></a>
<span class="sourceLineNo">092</span>                                new GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.92"></a>
<span class="sourceLineNo">093</span>                                                file,<a name="line.93"></a>
<span class="sourceLineNo">094</span>                                                new GenericGenbankHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),<a name="line.94"></a>
<span class="sourceLineNo">095</span>                                                new FileProxyProteinSequenceCreator(<a name="line.95"></a>
<span class="sourceLineNo">096</span>                                                                file,<a name="line.96"></a>
<span class="sourceLineNo">097</span>                                                                AminoAcidCompoundSet.getAminoAcidCompoundSet(),<a name="line.97"></a>
<span class="sourceLineNo">098</span>                                                                new GenbankSequenceParser&lt;AbstractSequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;()<a name="line.98"></a>
<span class="sourceLineNo">099</span>                                                        )<a name="line.99"></a>
<span class="sourceLineNo">100</span>                                        );<a name="line.100"></a>
<span class="sourceLineNo">101</span>                return GenbankProxyReader.process();<a name="line.101"></a>
<span class="sourceLineNo">102</span><a name="line.102"></a>
<span class="sourceLineNo">103</span>        }<a name="line.103"></a>
<span class="sourceLineNo">104</span><a name="line.104"></a>
<span class="sourceLineNo">105</span>        /**<a name="line.105"></a>
<span class="sourceLineNo">106</span>         * Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects<a name="line.106"></a>
<span class="sourceLineNo">107</span>         * that can in the future read the sequence from the disk. This allows the loading of large Genbank files where you are only interested<a name="line.107"></a>
<span class="sourceLineNo">108</span>         * in one sequence based on accession id.<a name="line.108"></a>
<span class="sourceLineNo">109</span>         * @param file<a name="line.109"></a>
<span class="sourceLineNo">110</span>         * @param lazySequenceLoad<a name="line.110"></a>
<span class="sourceLineNo">111</span>         * @return<a name="line.111"></a>
<span class="sourceLineNo">112</span>         * @throws Exception<a name="line.112"></a>
<span class="sourceLineNo">113</span>         */<a name="line.113"></a>
<span class="sourceLineNo">114</span>        public static LinkedHashMap&lt;String, RNASequence&gt; readGenbankRNASequence(File file, boolean lazySequenceLoad) throws Exception {<a name="line.114"></a>
<span class="sourceLineNo">115</span>                if (!lazySequenceLoad) {<a name="line.115"></a>
<span class="sourceLineNo">116</span>                        return readGenbankRNASequence(file);<a name="line.116"></a>
<span class="sourceLineNo">117</span>                }<a name="line.117"></a>
<span class="sourceLineNo">118</span><a name="line.118"></a>
<span class="sourceLineNo">119</span>                GenbankReader&lt;RNASequence, NucleotideCompound&gt; GenbankProxyReader =<a name="line.119"></a>
<span class="sourceLineNo">120</span>                                new GenbankReader&lt;RNASequence, NucleotideCompound&gt;(<a name="line.120"></a>
<span class="sourceLineNo">121</span>                                                file,<a name="line.121"></a>
<span class="sourceLineNo">122</span>                                                new GenericGenbankHeaderParser&lt;RNASequence, NucleotideCompound&gt;(),<a name="line.122"></a>
<span class="sourceLineNo">123</span>                                                new FileProxyRNASequenceCreator(<a name="line.123"></a>
<span class="sourceLineNo">124</span>                                                                file,<a name="line.124"></a>
<span class="sourceLineNo">125</span>                                                                RNACompoundSet.getRNACompoundSet(),<a name="line.125"></a>
<span class="sourceLineNo">126</span>                                                                new GenbankSequenceParser&lt;AbstractSequence&lt;NucleotideCompound&gt;, NucleotideCompound&gt;()<a name="line.126"></a>
<span class="sourceLineNo">127</span>                                                        )<a name="line.127"></a>
<span class="sourceLineNo">128</span>                                        );<a name="line.128"></a>
<span class="sourceLineNo">129</span>                return GenbankProxyReader.process();<a name="line.129"></a>
<span class="sourceLineNo">130</span><a name="line.130"></a>
<span class="sourceLineNo">131</span>        }<a name="line.131"></a>
<span class="sourceLineNo">132</span><a name="line.132"></a>
<span class="sourceLineNo">133</span>        /**<a name="line.133"></a>
<span class="sourceLineNo">134</span>         * Read a Genbank file containing amino acids with setup that would handle most<a name="line.134"></a>
<span class="sourceLineNo">135</span>         * cases.<a name="line.135"></a>
<span class="sourceLineNo">136</span>         *<a name="line.136"></a>
<span class="sourceLineNo">137</span>         * @param file<a name="line.137"></a>
<span class="sourceLineNo">138</span>         * @return<a name="line.138"></a>
<span class="sourceLineNo">139</span>         * @throws Exception<a name="line.139"></a>
<span class="sourceLineNo">140</span>         */<a name="line.140"></a>
<span class="sourceLineNo">141</span>        public static LinkedHashMap&lt;String, ProteinSequence&gt; readGenbankProteinSequence(<a name="line.141"></a>
<span class="sourceLineNo">142</span>                        File file) throws Exception {<a name="line.142"></a>
<span class="sourceLineNo">143</span>                FileInputStream inStream = new FileInputStream(file);<a name="line.143"></a>
<span class="sourceLineNo">144</span>                LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequences = readGenbankProteinSequence(inStream);<a name="line.144"></a>
<span class="sourceLineNo">145</span>                inStream.close();<a name="line.145"></a>
<span class="sourceLineNo">146</span>                return proteinSequences;<a name="line.146"></a>
<span class="sourceLineNo">147</span>        }<a name="line.147"></a>
<span class="sourceLineNo">148</span><a name="line.148"></a>
<span class="sourceLineNo">149</span>        /**<a name="line.149"></a>
<span class="sourceLineNo">150</span>         * Read a Genbank file containing amino acids with setup that would handle most<a name="line.150"></a>
<span class="sourceLineNo">151</span>         * cases. User is responsible for closing InputStream because you opened it<a name="line.151"></a>
<span class="sourceLineNo">152</span>         *<a name="line.152"></a>
<span class="sourceLineNo">153</span>         * @param inStream<a name="line.153"></a>
<span class="sourceLineNo">154</span>         * @return<a name="line.154"></a>
<span class="sourceLineNo">155</span>         * @throws Exception<a name="line.155"></a>
<span class="sourceLineNo">156</span>         */<a name="line.156"></a>
<span class="sourceLineNo">157</span>        public static LinkedHashMap&lt;String, ProteinSequence&gt; readGenbankProteinSequence(<a name="line.157"></a>
<span class="sourceLineNo">158</span>                        InputStream inStream) throws Exception {<a name="line.158"></a>
<span class="sourceLineNo">159</span>                GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt; GenbankReader = new GenbankReader&lt;ProteinSequence, AminoAcidCompound&gt;(<a name="line.159"></a>
<span class="sourceLineNo">160</span>                                inStream,<a name="line.160"></a>
<span class="sourceLineNo">161</span>                                new GenericGenbankHeaderParser&lt;ProteinSequence, AminoAcidCompound&gt;(),<a name="line.161"></a>
<span class="sourceLineNo">162</span>                                new ProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet()));<a name="line.162"></a>
<span class="sourceLineNo">163</span>                return GenbankReader.process();<a name="line.163"></a>
<span class="sourceLineNo">164</span>        }<a name="line.164"></a>
<span class="sourceLineNo">165</span><a name="line.165"></a>
<span class="sourceLineNo">166</span>        /**<a name="line.166"></a>
<span class="sourceLineNo">167</span>         * Read a Genbank DNA sequence<a name="line.167"></a>
<span class="sourceLineNo">168</span>         * @param inStream<a name="line.168"></a>
<span class="sourceLineNo">169</span>         * @return<a name="line.169"></a>
<span class="sourceLineNo">170</span>         * @throws Exception<a name="line.170"></a>
<span class="sourceLineNo">171</span>         */<a name="line.171"></a>
<span class="sourceLineNo">172</span>        public static LinkedHashMap&lt;String, DNASequence&gt; readGenbankDNASequence(<a name="line.172"></a>
<span class="sourceLineNo">173</span>                        InputStream inStream) throws Exception {<a name="line.173"></a>
<span class="sourceLineNo">174</span>                GenbankReader&lt;DNASequence, NucleotideCompound&gt; GenbankReader = new GenbankReader&lt;DNASequence, NucleotideCompound&gt;(<a name="line.174"></a>
<span class="sourceLineNo">175</span>                                inStream,<a name="line.175"></a>
<span class="sourceLineNo">176</span>                                new GenericGenbankHeaderParser&lt;DNASequence, NucleotideCompound&gt;(),<a name="line.176"></a>
<span class="sourceLineNo">177</span>                                new DNASequenceCreator(DNACompoundSet.getDNACompoundSet()));<a name="line.177"></a>
<span class="sourceLineNo">178</span>                return GenbankReader.process();<a name="line.178"></a>
<span class="sourceLineNo">179</span>        }<a name="line.179"></a>
<span class="sourceLineNo">180</span><a name="line.180"></a>
<span class="sourceLineNo">181</span>        /**<a name="line.181"></a>
<span class="sourceLineNo">182</span>         *<a name="line.182"></a>
<span class="sourceLineNo">183</span>         * @param file<a name="line.183"></a>
<span class="sourceLineNo">184</span>         * @return<a name="line.184"></a>
<span class="sourceLineNo">185</span>         * @throws Exception<a name="line.185"></a>
<span class="sourceLineNo">186</span>         */<a name="line.186"></a>
<span class="sourceLineNo">187</span>        public static LinkedHashMap&lt;String, DNASequence&gt; readGenbankDNASequence(<a name="line.187"></a>
<span class="sourceLineNo">188</span>                        File file) throws Exception {<a name="line.188"></a>
<span class="sourceLineNo">189</span>                FileInputStream inStream = new FileInputStream(file);<a name="line.189"></a>
<span class="sourceLineNo">190</span>                LinkedHashMap&lt;String, DNASequence&gt; dnaSequences = readGenbankDNASequence(inStream);<a name="line.190"></a>
<span class="sourceLineNo">191</span>                inStream.close();<a name="line.191"></a>
<span class="sourceLineNo">192</span>                return dnaSequences;<a name="line.192"></a>
<span class="sourceLineNo">193</span>        }<a name="line.193"></a>
<span class="sourceLineNo">194</span>        /**<a name="line.194"></a>
<span class="sourceLineNo">195</span>         * Read a Genbank RNA sequence<a name="line.195"></a>
<span class="sourceLineNo">196</span>         * @param inStream<a name="line.196"></a>
<span class="sourceLineNo">197</span>         * @return<a name="line.197"></a>
<span class="sourceLineNo">198</span>         * @throws Exception<a name="line.198"></a>
<span class="sourceLineNo">199</span>         */<a name="line.199"></a>
<span class="sourceLineNo">200</span>        public static LinkedHashMap&lt;String, RNASequence&gt; readGenbankRNASequence(<a name="line.200"></a>
<span class="sourceLineNo">201</span>                        InputStream inStream) throws Exception {<a name="line.201"></a>
<span class="sourceLineNo">202</span>                GenbankReader&lt;RNASequence, NucleotideCompound&gt; GenbankReader = new GenbankReader&lt;RNASequence, NucleotideCompound&gt;(<a name="line.202"></a>
<span class="sourceLineNo">203</span>                                inStream,<a name="line.203"></a>
<span class="sourceLineNo">204</span>                                new GenericGenbankHeaderParser&lt;RNASequence, NucleotideCompound&gt;(),<a name="line.204"></a>
<span class="sourceLineNo">205</span>                                new RNASequenceCreator(RNACompoundSet.getRNACompoundSet()));<a name="line.205"></a>
<span class="sourceLineNo">206</span>                return GenbankReader.process();<a name="line.206"></a>
<span class="sourceLineNo">207</span>        }<a name="line.207"></a>
<span class="sourceLineNo">208</span><a name="line.208"></a>
<span class="sourceLineNo">209</span>        /**<a name="line.209"></a>
<span class="sourceLineNo">210</span>         *<a name="line.210"></a>
<span class="sourceLineNo">211</span>         * @param file<a name="line.211"></a>
<span class="sourceLineNo">212</span>         * @return<a name="line.212"></a>
<span class="sourceLineNo">213</span>         * @throws Exception<a name="line.213"></a>
<span class="sourceLineNo">214</span>         */<a name="line.214"></a>
<span class="sourceLineNo">215</span>        public static LinkedHashMap&lt;String, RNASequence&gt; readGenbankRNASequence(<a name="line.215"></a>
<span class="sourceLineNo">216</span>                        File file) throws Exception {<a name="line.216"></a>
<span class="sourceLineNo">217</span>                FileInputStream inStream = new FileInputStream(file);<a name="line.217"></a>
<span class="sourceLineNo">218</span>                LinkedHashMap&lt;String, RNASequence&gt; rnaSequences = readGenbankRNASequence(inStream);<a name="line.218"></a>
<span class="sourceLineNo">219</span>                inStream.close();<a name="line.219"></a>
<span class="sourceLineNo">220</span>                return rnaSequences;<a name="line.220"></a>
<span class="sourceLineNo">221</span>        }<a name="line.221"></a>
<span class="sourceLineNo">222</span><a name="line.222"></a>
<span class="sourceLineNo">223</span>        public static void main(String[] args) throws Exception {<a name="line.223"></a>
<span class="sourceLineNo">224</span><a name="line.224"></a>
<span class="sourceLineNo">225</span>                LinkedHashMap&lt;String, DNASequence&gt; dnaSequences = GenbankReaderHelper.readGenbankDNASequence(new File("src/test/resources/NM_000266.gb"), true);<a name="line.225"></a>
<span class="sourceLineNo">226</span>                for (DNASequence sequence : dnaSequences.values()) {<a name="line.226"></a>
<span class="sourceLineNo">227</span>                        logger.info("DNA Sequence: {}", sequence.getRNASequence().getProteinSequence().getSequenceAsString());<a name="line.227"></a>
<span class="sourceLineNo">228</span>                }<a name="line.228"></a>
<span class="sourceLineNo">229</span><a name="line.229"></a>
<span class="sourceLineNo">230</span>                LinkedHashMap&lt;String, ProteinSequence&gt; proteinSequences = GenbankReaderHelper.readGenbankProteinSequence(new File("src/test/resources/BondFeature.gb"), true);<a name="line.230"></a>
<span class="sourceLineNo">231</span>                for (ProteinSequence sequence : proteinSequences.values()) {<a name="line.231"></a>
<span class="sourceLineNo">232</span>                        logger.info("Protein Sequence: {}", sequence.getSequenceAsString());<a name="line.232"></a>
<span class="sourceLineNo">233</span>                }<a name="line.233"></a>
<span class="sourceLineNo">234</span>        }<a name="line.234"></a>
<span class="sourceLineNo">235</span>}<a name="line.235"></a>




























































</pre>
</div>
</body>
</html>
