<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/**<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> * Created by Spencer Bliven<a name="line.20"></a>
<span class="sourceLineNo">021</span> *<a name="line.21"></a>
<span class="sourceLineNo">022</span> */<a name="line.22"></a>
<span class="sourceLineNo">023</span>package org.biojava.nbio.structure.io;<a name="line.23"></a>
<span class="sourceLineNo">024</span><a name="line.24"></a>
<span class="sourceLineNo">025</span>import org.biojava.nbio.alignment.Alignments;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import org.biojava.nbio.alignment.SimpleGapPenalty;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import org.biojava.nbio.core.alignment.matrices.SimpleSubstitutionMatrix;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import org.biojava.nbio.core.alignment.template.AlignedSequence;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import org.biojava.nbio.core.alignment.template.SequencePair;<a name="line.30"></a>
<span class="sourceLineNo">031</span>import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;<a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.nbio.structure.*;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.nbio.core.exceptions.CompoundNotFoundException;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.biojava.nbio.core.sequence.ProteinSequence;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.nbio.core.sequence.template.CompoundSet;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.slf4j.Logger;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.slf4j.LoggerFactory;<a name="line.39"></a>
<span class="sourceLineNo">040</span><a name="line.40"></a>
<span class="sourceLineNo">041</span>import java.io.InputStreamReader;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import java.util.Arrays;<a name="line.42"></a>
<span class="sourceLineNo">043</span>import java.util.HashMap;<a name="line.43"></a>
<span class="sourceLineNo">044</span>import java.util.List;<a name="line.44"></a>
<span class="sourceLineNo">045</span>import java.util.Map;<a name="line.45"></a>
<span class="sourceLineNo">046</span><a name="line.46"></a>
<span class="sourceLineNo">047</span><a name="line.47"></a>
<span class="sourceLineNo">048</span>/**<a name="line.48"></a>
<span class="sourceLineNo">049</span> * A utility class with methods for matching ProteinSequences with<a name="line.49"></a>
<span class="sourceLineNo">050</span> * Structures.<a name="line.50"></a>
<span class="sourceLineNo">051</span> * @author Spencer Bliven<a name="line.51"></a>
<span class="sourceLineNo">052</span> *<a name="line.52"></a>
<span class="sourceLineNo">053</span> */<a name="line.53"></a>
<span class="sourceLineNo">054</span>public class StructureSequenceMatcher {<a name="line.54"></a>
<span class="sourceLineNo">055</span><a name="line.55"></a>
<span class="sourceLineNo">056</span>        private static final Logger logger = LoggerFactory.getLogger(StructureSequenceMatcher.class);<a name="line.56"></a>
<span class="sourceLineNo">057</span><a name="line.57"></a>
<span class="sourceLineNo">058</span>        /**<a name="line.58"></a>
<span class="sourceLineNo">059</span>         * Get a substructure of {@code wholeStructure} containing only the {@link Group Groups} that are included in<a name="line.59"></a>
<span class="sourceLineNo">060</span>         * {@code sequence}. The resulting structure will contain only {@code ATOM} residues; the SEQ-RES will be empty.<a name="line.60"></a>
<span class="sourceLineNo">061</span>         * The {@link Chain Chains} of the Structure will be new instances (cloned), but the {@link Group Groups} will not.<a name="line.61"></a>
<span class="sourceLineNo">062</span>         * @param sequence The input protein sequence<a name="line.62"></a>
<span class="sourceLineNo">063</span>         * @param wholeStructure The structure from which to take a substructure<a name="line.63"></a>
<span class="sourceLineNo">064</span>         * @return The resulting structure<a name="line.64"></a>
<span class="sourceLineNo">065</span>         * @throws StructureException<a name="line.65"></a>
<span class="sourceLineNo">066</span>         * @see {@link #matchSequenceToStructure(ProteinSequence, Structure)}<a name="line.66"></a>
<span class="sourceLineNo">067</span>         */<a name="line.67"></a>
<span class="sourceLineNo">068</span>        public static Structure getSubstructureMatchingProteinSequence(ProteinSequence sequence, Structure wholeStructure) {<a name="line.68"></a>
<span class="sourceLineNo">069</span>                ResidueNumber[] rns = matchSequenceToStructure(sequence, wholeStructure);<a name="line.69"></a>
<span class="sourceLineNo">070</span>                Structure structure = wholeStructure.clone();<a name="line.70"></a>
<span class="sourceLineNo">071</span>                structure.getChains().clear();<a name="line.71"></a>
<span class="sourceLineNo">072</span>//              structure.getHetGroups().clear();<a name="line.72"></a>
<span class="sourceLineNo">073</span>                Chain currentChain = null;<a name="line.73"></a>
<span class="sourceLineNo">074</span>                for (ResidueNumber rn : rns) {<a name="line.74"></a>
<span class="sourceLineNo">075</span>                        if (rn == null) continue;<a name="line.75"></a>
<span class="sourceLineNo">076</span>                        Group group; // note that we don't clone<a name="line.76"></a>
<span class="sourceLineNo">077</span>                        try {<a name="line.77"></a>
<span class="sourceLineNo">078</span>                                group = StructureTools.getGroupByPDBResidueNumber(wholeStructure, rn);<a name="line.78"></a>
<span class="sourceLineNo">079</span>                        } catch (StructureException e) {<a name="line.79"></a>
<span class="sourceLineNo">080</span>                                throw new IllegalArgumentException("Could not find residue " + rn + " in structure", e);<a name="line.80"></a>
<span class="sourceLineNo">081</span>                        }<a name="line.81"></a>
<span class="sourceLineNo">082</span>                        Chain chain = new ChainImpl();<a name="line.82"></a>
<span class="sourceLineNo">083</span>                        chain.setChainID(group.getChainId());<a name="line.83"></a>
<span class="sourceLineNo">084</span>                        if (currentChain == null || !currentChain.getChainID().equals(chain.getChainID())) {<a name="line.84"></a>
<span class="sourceLineNo">085</span>                                structure.addChain(chain);<a name="line.85"></a>
<span class="sourceLineNo">086</span>                                chain.setCompound(group.getChain().getCompound());<a name="line.86"></a>
<span class="sourceLineNo">087</span>                                chain.setStructure(structure);<a name="line.87"></a>
<span class="sourceLineNo">088</span>                                chain.setSwissprotId(group.getChain().getSwissprotId());<a name="line.88"></a>
<span class="sourceLineNo">089</span>                                chain.setInternalChainID(group.getChain().getInternalChainID());<a name="line.89"></a>
<span class="sourceLineNo">090</span>                                chain.setId(group.getChain().getId());<a name="line.90"></a>
<span class="sourceLineNo">091</span>                                currentChain = chain;<a name="line.91"></a>
<span class="sourceLineNo">092</span>                        }<a name="line.92"></a>
<span class="sourceLineNo">093</span>                        currentChain.addGroup(group);<a name="line.93"></a>
<span class="sourceLineNo">094</span>                }<a name="line.94"></a>
<span class="sourceLineNo">095</span>                return structure;<a name="line.95"></a>
<span class="sourceLineNo">096</span>        }<a name="line.96"></a>
<span class="sourceLineNo">097</span><a name="line.97"></a>
<span class="sourceLineNo">098</span>        /**<a name="line.98"></a>
<span class="sourceLineNo">099</span>         * Generates a ProteinSequence corresponding to the sequence of struct,<a name="line.99"></a>
<span class="sourceLineNo">100</span>         * and maintains a mapping from the sequence back to the original groups.<a name="line.100"></a>
<span class="sourceLineNo">101</span>         *<a name="line.101"></a>
<span class="sourceLineNo">102</span>         * Chains are appended to one another. 'X' is used for heteroatoms.<a name="line.102"></a>
<span class="sourceLineNo">103</span>         *<a name="line.103"></a>
<span class="sourceLineNo">104</span>         * @param struct Input structure<a name="line.104"></a>
<span class="sourceLineNo">105</span>         * @param groupIndexPosition An empty map, which will be populated with<a name="line.105"></a>
<span class="sourceLineNo">106</span>         *  (residue index in returned ProteinSequence) -&gt; (Group within struct)<a name="line.106"></a>
<span class="sourceLineNo">107</span>         * @return A ProteinSequence with the full sequence of struct. Chains are<a name="line.107"></a>
<span class="sourceLineNo">108</span>         *  concatenated in the same order as the input structures<a name="line.108"></a>
<span class="sourceLineNo">109</span>         *<a name="line.109"></a>
<span class="sourceLineNo">110</span>         * @see {@link SeqRes2AtomAligner#getFullAtomSequence(List, Map)}, which<a name="line.110"></a>
<span class="sourceLineNo">111</span>         *      does the heavy lifting.<a name="line.111"></a>
<span class="sourceLineNo">112</span>         *<a name="line.112"></a>
<span class="sourceLineNo">113</span>         */<a name="line.113"></a>
<span class="sourceLineNo">114</span>        public static ProteinSequence getProteinSequenceForStructure(Structure struct, Map&lt;Integer,Group&gt; groupIndexPosition ) {<a name="line.114"></a>
<span class="sourceLineNo">115</span><a name="line.115"></a>
<span class="sourceLineNo">116</span>                if( groupIndexPosition != null) {<a name="line.116"></a>
<span class="sourceLineNo">117</span>                        groupIndexPosition.clear();<a name="line.117"></a>
<span class="sourceLineNo">118</span>                }<a name="line.118"></a>
<span class="sourceLineNo">119</span><a name="line.119"></a>
<span class="sourceLineNo">120</span>                StringBuilder seqStr = new StringBuilder();<a name="line.120"></a>
<span class="sourceLineNo">121</span><a name="line.121"></a>
<span class="sourceLineNo">122</span>                for(Chain chain : struct.getChains()) {<a name="line.122"></a>
<span class="sourceLineNo">123</span>                        List&lt;Group&gt; groups = chain.getAtomGroups();<a name="line.123"></a>
<span class="sourceLineNo">124</span>                        Map&lt;Integer,Integer&gt; chainIndexPosition = new HashMap&lt;Integer, Integer&gt;();<a name="line.124"></a>
<span class="sourceLineNo">125</span>                        int prevLen = seqStr.length();<a name="line.125"></a>
<span class="sourceLineNo">126</span><a name="line.126"></a>
<span class="sourceLineNo">127</span>                        // get the sequence for this chain<a name="line.127"></a>
<span class="sourceLineNo">128</span>                        String chainSeq = SeqRes2AtomAligner.getFullAtomSequence(groups, chainIndexPosition, false);<a name="line.128"></a>
<span class="sourceLineNo">129</span>                        seqStr.append(chainSeq);<a name="line.129"></a>
<span class="sourceLineNo">130</span><a name="line.130"></a>
<span class="sourceLineNo">131</span><a name="line.131"></a>
<span class="sourceLineNo">132</span>                        // fix up the position to include previous chains, and map the value back to a Group<a name="line.132"></a>
<span class="sourceLineNo">133</span>                        for(Integer seqIndex : chainIndexPosition.keySet()) {<a name="line.133"></a>
<span class="sourceLineNo">134</span>                                Integer groupIndex = chainIndexPosition.get(seqIndex);<a name="line.134"></a>
<span class="sourceLineNo">135</span>                                groupIndexPosition.put(prevLen + seqIndex, groups.get(groupIndex));<a name="line.135"></a>
<span class="sourceLineNo">136</span>                        }<a name="line.136"></a>
<span class="sourceLineNo">137</span>                }<a name="line.137"></a>
<span class="sourceLineNo">138</span><a name="line.138"></a>
<span class="sourceLineNo">139</span>                ProteinSequence s = null;<a name="line.139"></a>
<span class="sourceLineNo">140</span>                try {<a name="line.140"></a>
<span class="sourceLineNo">141</span>                        s = new ProteinSequence(seqStr.toString());<a name="line.141"></a>
<span class="sourceLineNo">142</span>                } catch (CompoundNotFoundException e) {<a name="line.142"></a>
<span class="sourceLineNo">143</span>                        // I believe this can't happen, please correct this if I'm wrong - JD 2014-10-24<a name="line.143"></a>
<span class="sourceLineNo">144</span>                        // we can log an error if it does, it would mean there's a bad bug somewhere<a name="line.144"></a>
<span class="sourceLineNo">145</span>                        logger.error("Could not create protein sequence, unknown compounds in string: {}", e.getMessage());<a name="line.145"></a>
<span class="sourceLineNo">146</span>                }<a name="line.146"></a>
<span class="sourceLineNo">147</span><a name="line.147"></a>
<span class="sourceLineNo">148</span>                return s;<a name="line.148"></a>
<span class="sourceLineNo">149</span>        }<a name="line.149"></a>
<span class="sourceLineNo">150</span><a name="line.150"></a>
<span class="sourceLineNo">151</span>        /**<a name="line.151"></a>
<span class="sourceLineNo">152</span>         * Given a sequence and the corresponding Structure, get the ResidueNumber<a name="line.152"></a>
<span class="sourceLineNo">153</span>         * for each residue in the sequence.<a name="line.153"></a>
<span class="sourceLineNo">154</span>         *<a name="line.154"></a>
<span class="sourceLineNo">155</span>         * &lt;p&gt;Smith-Waterman alignment is used to match the sequences. Residues<a name="line.155"></a>
<span class="sourceLineNo">156</span>         * in the sequence but not the structure or mismatched between sequence<a name="line.156"></a>
<span class="sourceLineNo">157</span>         * and structure will have a null atom, while<a name="line.157"></a>
<span class="sourceLineNo">158</span>         * residues in the structure but not the sequence are ignored with a warning.<a name="line.158"></a>
<span class="sourceLineNo">159</span>         * @param seq The protein sequence. Should match the sequence of struct very<a name="line.159"></a>
<span class="sourceLineNo">160</span>         *      closely.<a name="line.160"></a>
<span class="sourceLineNo">161</span>         * @param struct The corresponding protein structure<a name="line.161"></a>
<span class="sourceLineNo">162</span>         * @return A list of ResidueNumbers of the same length as seq, containing<a name="line.162"></a>
<span class="sourceLineNo">163</span>         *  either the corresponding residue or null.<a name="line.163"></a>
<span class="sourceLineNo">164</span>         */<a name="line.164"></a>
<span class="sourceLineNo">165</span>        public static ResidueNumber[] matchSequenceToStructure(ProteinSequence seq, Structure struct) {<a name="line.165"></a>
<span class="sourceLineNo">166</span><a name="line.166"></a>
<span class="sourceLineNo">167</span>                //1. Create ProteinSequence for struct while remembering to which group each residue corresponds<a name="line.167"></a>
<span class="sourceLineNo">168</span>                Map&lt;Integer,Group&gt; atomIndexPosition   = new HashMap&lt;Integer, Group&gt;();<a name="line.168"></a>
<span class="sourceLineNo">169</span><a name="line.169"></a>
<span class="sourceLineNo">170</span>                ProteinSequence structSeq = getProteinSequenceForStructure(struct,atomIndexPosition);<a name="line.170"></a>
<span class="sourceLineNo">171</span><a name="line.171"></a>
<span class="sourceLineNo">172</span>                // TODO This should really be semi-global alignment, though global for either sequence OR structure (we don't know which)<a name="line.172"></a>
<span class="sourceLineNo">173</span>                //2. Run Smith-Waterman to get the alignment<a name="line.173"></a>
<span class="sourceLineNo">174</span>                // Identity substitution matrix with +1 for match, -1 for mismatch<a name="line.174"></a>
<span class="sourceLineNo">175</span>                // TODO<a name="line.175"></a>
<span class="sourceLineNo">176</span>                SubstitutionMatrix&lt;AminoAcidCompound&gt; matrix =<a name="line.176"></a>
<span class="sourceLineNo">177</span>                        new SimpleSubstitutionMatrix&lt;AminoAcidCompound&gt;(<a name="line.177"></a>
<span class="sourceLineNo">178</span>                                        AminoAcidCompoundSet.getAminoAcidCompoundSet(),<a name="line.178"></a>
<span class="sourceLineNo">179</span>                                        (short)1, (short)-1 );<a name="line.179"></a>
<span class="sourceLineNo">180</span>                matrix = new SimpleSubstitutionMatrix&lt;AminoAcidCompound&gt;(<a name="line.180"></a>
<span class="sourceLineNo">181</span>                                AminoAcidCompoundSet.getAminoAcidCompoundSet(),<a name="line.181"></a>
<span class="sourceLineNo">182</span>                                new InputStreamReader(<a name="line.182"></a>
<span class="sourceLineNo">183</span>                                                SimpleSubstitutionMatrix.class.getResourceAsStream("/matrices/blosum100.txt")),<a name="line.183"></a>
<span class="sourceLineNo">184</span>                "blosum100");<a name="line.184"></a>
<span class="sourceLineNo">185</span>                SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; pair =<a name="line.185"></a>
<span class="sourceLineNo">186</span>                        Alignments.getPairwiseAlignment(seq, structSeq,<a name="line.186"></a>
<span class="sourceLineNo">187</span>                                        PairwiseSequenceAlignerType.GLOBAL, new SimpleGapPenalty(), matrix);<a name="line.187"></a>
<span class="sourceLineNo">188</span><a name="line.188"></a>
<span class="sourceLineNo">189</span>                //System.out.print(pair.toString());<a name="line.189"></a>
<span class="sourceLineNo">190</span><a name="line.190"></a>
<span class="sourceLineNo">191</span>                //3. Convert the alignment back to Atoms<a name="line.191"></a>
<span class="sourceLineNo">192</span>                AlignedSequence&lt;ProteinSequence,AminoAcidCompound&gt; alignedSeq = pair.getQuery();<a name="line.192"></a>
<span class="sourceLineNo">193</span>                AlignedSequence&lt;ProteinSequence,AminoAcidCompound&gt; alignedStruct = pair.getTarget();<a name="line.193"></a>
<span class="sourceLineNo">194</span><a name="line.194"></a>
<span class="sourceLineNo">195</span><a name="line.195"></a>
<span class="sourceLineNo">196</span>                assert(alignedSeq.getLength() == alignedStruct.getLength());<a name="line.196"></a>
<span class="sourceLineNo">197</span><a name="line.197"></a>
<span class="sourceLineNo">198</span>//              System.out.println(pair.toString(80));<a name="line.198"></a>
<span class="sourceLineNo">199</span>//              System.out.format("%d/min{%d,%d}\n", pair.getNumIdenticals(),<a name="line.199"></a>
<span class="sourceLineNo">200</span>//                              alignedSeq.getLength()-alignedSeq.getNumGaps(),<a name="line.200"></a>
<span class="sourceLineNo">201</span>//                              alignedStruct.getLength()-alignedStruct.getNumGaps());<a name="line.201"></a>
<span class="sourceLineNo">202</span><a name="line.202"></a>
<span class="sourceLineNo">203</span>                ResidueNumber[] ca = new ResidueNumber[seq.getLength()];<a name="line.203"></a>
<span class="sourceLineNo">204</span><a name="line.204"></a>
<span class="sourceLineNo">205</span>                for( int pos = alignedSeq.getStart().getPosition(); pos &lt;= alignedSeq.getEnd().getPosition(); pos++ ) { // 1-indexed<a name="line.205"></a>
<span class="sourceLineNo">206</span>                        //skip missing residues from sequence. These probably represent alignment errors<a name="line.206"></a>
<span class="sourceLineNo">207</span>                        if(alignedSeq.isGap(pos)) {<a name="line.207"></a>
<span class="sourceLineNo">208</span>                                int structIndex = alignedStruct.getSequenceIndexAt(pos)-1;<a name="line.208"></a>
<span class="sourceLineNo">209</span>                                assert(structIndex &gt; 0);//should be defined since seq gap<a name="line.209"></a>
<span class="sourceLineNo">210</span><a name="line.210"></a>
<span class="sourceLineNo">211</span>                                Group g = atomIndexPosition.get(structIndex);<a name="line.211"></a>
<span class="sourceLineNo">212</span><a name="line.212"></a>
<span class="sourceLineNo">213</span>                                logger.warn("Chain {} residue {} in the Structure {} has no corresponding amino acid in the sequence.",<a name="line.213"></a>
<span class="sourceLineNo">214</span>                                                g.getChainId(),<a name="line.214"></a>
<span class="sourceLineNo">215</span>                                                g.getResidueNumber().toString(),<a name="line.215"></a>
<span class="sourceLineNo">216</span>                                                g.getChain().getStructure().getPDBCode() );<a name="line.216"></a>
<span class="sourceLineNo">217</span>                                continue;<a name="line.217"></a>
<span class="sourceLineNo">218</span>                        }<a name="line.218"></a>
<span class="sourceLineNo">219</span><a name="line.219"></a>
<span class="sourceLineNo">220</span>                        if(! alignedStruct.isGap(pos) ) {<a name="line.220"></a>
<span class="sourceLineNo">221</span>                                int seqIndex = alignedSeq.getSequenceIndexAt(pos)-1;//1-indexed<a name="line.221"></a>
<span class="sourceLineNo">222</span>                                int structIndex = alignedStruct.getSequenceIndexAt(pos)-1;//1-indexed<a name="line.222"></a>
<span class="sourceLineNo">223</span>                                Group g = atomIndexPosition.get(structIndex);<a name="line.223"></a>
<span class="sourceLineNo">224</span><a name="line.224"></a>
<span class="sourceLineNo">225</span>                                assert(0&lt;=seqIndex &amp;&amp; seqIndex &lt; ca.length);<a name="line.225"></a>
<span class="sourceLineNo">226</span><a name="line.226"></a>
<span class="sourceLineNo">227</span>                                ca[seqIndex] = g.getResidueNumber(); //remains null for gaps<a name="line.227"></a>
<span class="sourceLineNo">228</span>                        }<a name="line.228"></a>
<span class="sourceLineNo">229</span>                }<a name="line.229"></a>
<span class="sourceLineNo">230</span>                return ca;<a name="line.230"></a>
<span class="sourceLineNo">231</span>        }<a name="line.231"></a>
<span class="sourceLineNo">232</span><a name="line.232"></a>
<span class="sourceLineNo">233</span><a name="line.233"></a>
<span class="sourceLineNo">234</span>        /**<a name="line.234"></a>
<span class="sourceLineNo">235</span>         * Removes all gaps ('-') from a protein sequence<a name="line.235"></a>
<span class="sourceLineNo">236</span>         * @param gapped<a name="line.236"></a>
<span class="sourceLineNo">237</span>         * @return<a name="line.237"></a>
<span class="sourceLineNo">238</span>         */<a name="line.238"></a>
<span class="sourceLineNo">239</span>        public static ProteinSequence removeGaps(ProteinSequence gapped) {<a name="line.239"></a>
<span class="sourceLineNo">240</span>                final String[] gapStrings = {"-","."};<a name="line.240"></a>
<span class="sourceLineNo">241</span><a name="line.241"></a>
<span class="sourceLineNo">242</span>                StringBuilder seq = new StringBuilder();<a name="line.242"></a>
<span class="sourceLineNo">243</span><a name="line.243"></a>
<span class="sourceLineNo">244</span>                CompoundSet&lt;AminoAcidCompound&gt; aaSet = gapped.getCompoundSet();<a name="line.244"></a>
<span class="sourceLineNo">245</span>                AminoAcidCompound[] gaps = new AminoAcidCompound[gapStrings.length];<a name="line.245"></a>
<span class="sourceLineNo">246</span><a name="line.246"></a>
<span class="sourceLineNo">247</span>                for(int i=0;i&lt;gapStrings.length;i++) {<a name="line.247"></a>
<span class="sourceLineNo">248</span>                        gaps[i] = aaSet.getCompoundForString(gapStrings[i]);<a name="line.248"></a>
<span class="sourceLineNo">249</span>                }<a name="line.249"></a>
<span class="sourceLineNo">250</span><a name="line.250"></a>
<span class="sourceLineNo">251</span>                for(int i=1; i&lt;=gapped.getLength();i++) { //1-indexed<a name="line.251"></a>
<span class="sourceLineNo">252</span>                        AminoAcidCompound aa = gapped.getCompoundAt(i);<a name="line.252"></a>
<span class="sourceLineNo">253</span>                        boolean isGap = false;<a name="line.253"></a>
<span class="sourceLineNo">254</span>                        for(AminoAcidCompound gap : gaps) {<a name="line.254"></a>
<span class="sourceLineNo">255</span>                                if( aa.equals(gap)) {<a name="line.255"></a>
<span class="sourceLineNo">256</span>                                        isGap = true;<a name="line.256"></a>
<span class="sourceLineNo">257</span>                                        break;<a name="line.257"></a>
<span class="sourceLineNo">258</span>                                }<a name="line.258"></a>
<span class="sourceLineNo">259</span>                        }<a name="line.259"></a>
<span class="sourceLineNo">260</span>                        if(!isGap) {<a name="line.260"></a>
<span class="sourceLineNo">261</span>                                seq.append(aa.getShortName());<a name="line.261"></a>
<span class="sourceLineNo">262</span>                        }<a name="line.262"></a>
<span class="sourceLineNo">263</span>                }<a name="line.263"></a>
<span class="sourceLineNo">264</span><a name="line.264"></a>
<span class="sourceLineNo">265</span>                ProteinSequence ungapped = null;<a name="line.265"></a>
<span class="sourceLineNo">266</span>                try {<a name="line.266"></a>
<span class="sourceLineNo">267</span>                        ungapped = new ProteinSequence(seq.toString());<a name="line.267"></a>
<span class="sourceLineNo">268</span>                } catch (CompoundNotFoundException e) {<a name="line.268"></a>
<span class="sourceLineNo">269</span>                        // this can't happen, if it does there's a bug somewhere<a name="line.269"></a>
<span class="sourceLineNo">270</span>                        logger.error("Could not create ungapped protein sequence, found unknown compounds: {}. This is most likely a bug.", e.getMessage());<a name="line.270"></a>
<span class="sourceLineNo">271</span>                }<a name="line.271"></a>
<span class="sourceLineNo">272</span><a name="line.272"></a>
<span class="sourceLineNo">273</span>                return ungapped;<a name="line.273"></a>
<span class="sourceLineNo">274</span>        }<a name="line.274"></a>
<span class="sourceLineNo">275</span><a name="line.275"></a>
<span class="sourceLineNo">276</span>        /**<a name="line.276"></a>
<span class="sourceLineNo">277</span>         * Creates a new list consisting of all columns of gapped where no row<a name="line.277"></a>
<span class="sourceLineNo">278</span>         * contained a null value.<a name="line.278"></a>
<span class="sourceLineNo">279</span>         *<a name="line.279"></a>
<span class="sourceLineNo">280</span>         * Here, "row" refers to the first index and "column" to the second, eg<a name="line.280"></a>
<span class="sourceLineNo">281</span>         * gapped.get(row).get(column)<a name="line.281"></a>
<span class="sourceLineNo">282</span>         * @param gapped A rectangular matrix containing null to mark gaps<a name="line.282"></a>
<span class="sourceLineNo">283</span>         * @return A new List without columns containing nulls<a name="line.283"></a>
<span class="sourceLineNo">284</span>         */<a name="line.284"></a>
<span class="sourceLineNo">285</span>        public static &lt;T&gt; T[][] removeGaps(final T[][] gapped) {<a name="line.285"></a>
<span class="sourceLineNo">286</span>                if(gapped == null ) return null;<a name="line.286"></a>
<span class="sourceLineNo">287</span>                if(gapped.length &lt; 1) return Arrays.copyOf(gapped, gapped.length);<a name="line.287"></a>
<span class="sourceLineNo">288</span><a name="line.288"></a>
<span class="sourceLineNo">289</span>                final int nProts = gapped.length;<a name="line.289"></a>
<span class="sourceLineNo">290</span>                final int protLen = gapped[0].length; // length of gapped proteins<a name="line.290"></a>
<span class="sourceLineNo">291</span><a name="line.291"></a>
<span class="sourceLineNo">292</span>                // Verify that input is rectangular<a name="line.292"></a>
<span class="sourceLineNo">293</span>                for(int i=0;i&lt;nProts;i++) {<a name="line.293"></a>
<span class="sourceLineNo">294</span>                        if(gapped[i].length != protLen) {<a name="line.294"></a>
<span class="sourceLineNo">295</span>                                throw new IllegalArgumentException(String.format(<a name="line.295"></a>
<span class="sourceLineNo">296</span>                                                "Expected a rectangular array, but row 0 has %d elements " +<a name="line.296"></a>
<span class="sourceLineNo">297</span>                                                "while row %d has %d.", protLen,i,gapped[i].length));<a name="line.297"></a>
<span class="sourceLineNo">298</span><a name="line.298"></a>
<span class="sourceLineNo">299</span>                        }<a name="line.299"></a>
<span class="sourceLineNo">300</span>                }<a name="line.300"></a>
<span class="sourceLineNo">301</span><a name="line.301"></a>
<span class="sourceLineNo">302</span>                // determine where gaps exist in any structures<a name="line.302"></a>
<span class="sourceLineNo">303</span>                boolean[] isGap = new boolean[protLen];<a name="line.303"></a>
<span class="sourceLineNo">304</span>                int gaps = 0;<a name="line.304"></a>
<span class="sourceLineNo">305</span>                for(int j=0;j&lt;protLen;j++) {<a name="line.305"></a>
<span class="sourceLineNo">306</span>                        for(int i=0;i&lt;nProts;i++) {<a name="line.306"></a>
<span class="sourceLineNo">307</span>                                if(gapped[i][j] == null ) {<a name="line.307"></a>
<span class="sourceLineNo">308</span>                                        isGap[j] = true;<a name="line.308"></a>
<span class="sourceLineNo">309</span>                                        gaps++;<a name="line.309"></a>
<span class="sourceLineNo">310</span>                                        break; //go to next position<a name="line.310"></a>
<span class="sourceLineNo">311</span>                                }<a name="line.311"></a>
<span class="sourceLineNo">312</span>                        }<a name="line.312"></a>
<span class="sourceLineNo">313</span>                }<a name="line.313"></a>
<span class="sourceLineNo">314</span><a name="line.314"></a>
<span class="sourceLineNo">315</span>                // Create ungapped array<a name="line.315"></a>
<span class="sourceLineNo">316</span>                T[][] ungapped = Arrays.copyOf(gapped,nProts);<a name="line.316"></a>
<span class="sourceLineNo">317</span>                final int ungappedLen = protLen-gaps;<a name="line.317"></a>
<span class="sourceLineNo">318</span>                for(int i=0;i&lt;nProts;i++) {<a name="line.318"></a>
<span class="sourceLineNo">319</span>                        ungapped[i] = Arrays.copyOf(gapped[i],ungappedLen);<a name="line.319"></a>
<span class="sourceLineNo">320</span>                        int k = 0;<a name="line.320"></a>
<span class="sourceLineNo">321</span>                        for(int j=0;j&lt;protLen;j++) {<a name="line.321"></a>
<span class="sourceLineNo">322</span>                                if(!isGap[j]) { //skip gaps<a name="line.322"></a>
<span class="sourceLineNo">323</span>                                        assert(gapped[i][j] != null);<a name="line.323"></a>
<span class="sourceLineNo">324</span>                                        ungapped[i][k] = gapped[i][j];<a name="line.324"></a>
<span class="sourceLineNo">325</span>                                        k++;<a name="line.325"></a>
<span class="sourceLineNo">326</span>                                }<a name="line.326"></a>
<span class="sourceLineNo">327</span>                        }<a name="line.327"></a>
<span class="sourceLineNo">328</span>                        assert(k == ungappedLen);<a name="line.328"></a>
<span class="sourceLineNo">329</span>                }<a name="line.329"></a>
<span class="sourceLineNo">330</span><a name="line.330"></a>
<span class="sourceLineNo">331</span>                return ungapped;<a name="line.331"></a>
<span class="sourceLineNo">332</span>        }<a name="line.332"></a>
<span class="sourceLineNo">333</span>}<a name="line.333"></a>




























































</pre>
</div>
</body>
</html>
