<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:49 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Interface org.biojava.nbio.core.alignment.template.SubstitutionMatrix (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Interface org.biojava.nbio.core.alignment.template.SubstitutionMatrix (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/alignment/template/class-use/SubstitutionMatrix.html" target="_top">Frames</a></li>
<li><a href="SubstitutionMatrix.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Interface org.biojava.nbio.core.alignment.template.SubstitutionMatrix" class="title">Uses of Interface<br>org.biojava.nbio.core.alignment.template.SubstitutionMatrix</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.alignment">org.biojava.nbio.alignment</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.alignment.routines">org.biojava.nbio.alignment.routines</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.alignment.template">org.biojava.nbio.alignment.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.alignment">org.biojava.nbio.core.alignment</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.alignment.matrices">org.biojava.nbio.core.alignment.matrices</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.alignment.template">org.biojava.nbio.core.alignment.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.phylo">org.biojava.nbio.phylo</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/alignment/package-summary.html">org.biojava.nbio.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/alignment/package-summary.html">org.biojava.nbio.alignment</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getAllPairsAlignments(java.util.List,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsAlignments</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences,
                     <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                     <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                     <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which computes a sequence alignment for all <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer</a>&lt;S,C&gt;&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getAllPairsScorers(java.util.List,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScorers</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences,
                  <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                  <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which sets up a sequence pair scorer for all <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>double[]</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getAllPairsScores(java.util.List,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceScorerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getAllPairsScores</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;S&gt;&nbsp;sequences,
                 <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a>&nbsp;type,
                 <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                 <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which computes a sequence pair score for all <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pairs in the given <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util"><code>List</code></a>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner</a>&lt;S,C&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getPairwiseAligner(S,%20S,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAligner</a></strong>(S&nbsp;query,
                  S&nbsp;target,
                  <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                  <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                  <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which constructs a pairwise sequence aligner.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;S,C&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getPairwiseAlignment(S,%20S,%20org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getPairwiseAlignment</a></strong>(S&nbsp;query,
                    S&nbsp;target,
                    <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a>&nbsp;type,
                    <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                    <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method which computes a sequence alignment for the given <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> pair.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;S extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;C&gt;,C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br><a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;S,C&gt;</code></td>
<td class="colLast"><span class="strong">Alignments.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/Alignments.html#getProgressiveAlignment(org.biojava.nbio.alignment.GuideTree,%20org.biojava.nbio.alignment.Alignments.ProfileProfileAlignerType,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">getProgressiveAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree</a>&lt;S,C&gt;&nbsp;tree,
                       <a href="../../../../../../../org/biojava/nbio/alignment/Alignments.ProfileProfileAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.ProfileProfileAlignerType</a>&nbsp;type,
                       <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                       <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;C&gt;&nbsp;subMatrix)</code>
<div class="block">Factory method to run the profile-profile alignments of a progressive multiple sequence alignment concurrently.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/alignment/package-summary.html">org.biojava.nbio.alignment</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/NeedlemanWunsch.html#NeedlemanWunsch(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">NeedlemanWunsch</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/NeedlemanWunsch.html" title="type parameter in NeedlemanWunsch">S</a>&nbsp;query,
               <a href="../../../../../../../org/biojava/nbio/alignment/NeedlemanWunsch.html" title="type parameter in NeedlemanWunsch">S</a>&nbsp;target,
               <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
               <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/NeedlemanWunsch.html" title="type parameter in NeedlemanWunsch">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a pairwise global sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html#SimpleProfileProfileAligner(java.util.concurrent.Future,%20java.util.concurrent.Future,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">SimpleProfileProfileAligner</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&gt;&nbsp;query,
                           <a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&gt;&nbsp;target,
                           <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                           <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment run concurrently.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html#SimpleProfileProfileAligner(java.util.concurrent.Future,%20org.biojava.nbio.core.alignment.template.Profile,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">SimpleProfileProfileAligner</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&gt;&nbsp;query,
                           <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;target,
                           <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                           <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment run concurrently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html#SimpleProfileProfileAligner(org.biojava.nbio.core.alignment.template.Profile,%20java.util.concurrent.Future,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">SimpleProfileProfileAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;query,
                           <a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&gt;&nbsp;target,
                           <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                           <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment run concurrently.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html#SimpleProfileProfileAligner(org.biojava.nbio.core.alignment.template.Profile,%20org.biojava.nbio.core.alignment.template.Profile,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">SimpleProfileProfileAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;query,
                           <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;target,
                           <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                           <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="type parameter in SimpleProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SmithWaterman.html#SmithWaterman(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">SmithWaterman</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/SmithWaterman.html" title="type parameter in SmithWaterman">S</a>&nbsp;query,
             <a href="../../../../../../../org/biojava/nbio/alignment/SmithWaterman.html" title="type parameter in SmithWaterman">S</a>&nbsp;target,
             <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
             <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SmithWaterman.html" title="type parameter in SmithWaterman">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a pairwise local sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/SubstitutionMatrixScorer.html#SubstitutionMatrixScorer(org.biojava.nbio.core.alignment.template.SequencePair,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">SubstitutionMatrixScorer</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="type parameter in SubstitutionMatrixScorer">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="type parameter in SubstitutionMatrixScorer">C</a>&gt;&nbsp;pair,
                        <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="type parameter in SubstitutionMatrixScorer">C</a>&gt;&nbsp;matrix)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.alignment.routines">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/alignment/routines/package-summary.html">org.biojava.nbio.alignment.routines</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html#AnchoredPairwiseSequenceAligner(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">AnchoredPairwiseSequenceAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">S</a>&nbsp;query,
                               <a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">S</a>&nbsp;target,
                               <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                               <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a pairwise global sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html#AnchoredPairwiseSequenceAligner(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix,%20int[])">AnchoredPairwiseSequenceAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">S</a>&nbsp;query,
                               <a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">S</a>&nbsp;target,
                               <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                               <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="type parameter in AnchoredPairwiseSequenceAligner">C</a>&gt;&nbsp;subMatrix,
                               int[]&nbsp;anchors)</code>
<div class="block">Prepares for a pairwise global sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html#GuanUberbacher(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">GuanUberbacher</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">S</a>&nbsp;query,
              <a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">S</a>&nbsp;target,
              <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
              <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a pairwise global sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html#GuanUberbacher(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix,%20int)">GuanUberbacher</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">S</a>&nbsp;query,
              <a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">S</a>&nbsp;target,
              <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
              <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="type parameter in GuanUberbacher">C</a>&gt;&nbsp;subMatrix,
              int&nbsp;cutsPerSection)</code>
<div class="block">Prepares for a pairwise global sequence alignment.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a> that return <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="type parameter in AbstractMatrixAligner">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">AbstractMatrixAligner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html#getSubstitutionMatrix()">getSubstitutionMatrix</a></strong>()</code>
<div class="block">Returns the substitution matrix.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AbstractMatrixAligner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html#setSubstitutionMatrix(org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">setSubstitutionMatrix</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="type parameter in AbstractMatrixAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Sets the substitution matrix.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/alignment/template/package-summary.html">org.biojava.nbio.alignment.template</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html#AbstractMatrixAligner(org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">AbstractMatrixAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                     <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="type parameter in AbstractMatrixAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for an alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html#AbstractMatrixAligner(org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix,%20boolean)">AbstractMatrixAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                     <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="type parameter in AbstractMatrixAligner">C</a>&gt;&nbsp;subMatrix,
                     boolean&nbsp;local)</code>
<div class="block">Prepares for an alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html#AbstractPairwiseSequenceAligner(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">AbstractPairwiseSequenceAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">S</a>&nbsp;query,
                               <a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">S</a>&nbsp;target,
                               <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                               <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a pairwise global sequence alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html#AbstractPairwiseSequenceAligner(S,%20S,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix,%20boolean)">AbstractPairwiseSequenceAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">S</a>&nbsp;query,
                               <a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">S</a>&nbsp;target,
                               <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                               <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="type parameter in AbstractPairwiseSequenceAligner">C</a>&gt;&nbsp;subMatrix,
                               boolean&nbsp;local)</code>
<div class="block">Prepares for a pairwise sequence alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html#AbstractProfileProfileAligner(java.util.concurrent.Future,%20java.util.concurrent.Future,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">AbstractProfileProfileAligner</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&gt;&nbsp;query,
                             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&gt;&nbsp;target,
                             <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                             <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment run concurrently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html#AbstractProfileProfileAligner(java.util.concurrent.Future,%20org.biojava.nbio.core.alignment.template.Profile,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">AbstractProfileProfileAligner</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&gt;&nbsp;query,
                             <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;target,
                             <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                             <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment run concurrently.</div>
</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html#AbstractProfileProfileAligner(org.biojava.nbio.core.alignment.template.Profile,%20java.util.concurrent.Future,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">AbstractProfileProfileAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;query,
                             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html?is-external=true" title="class or interface in java.util.concurrent">Future</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&gt;&nbsp;target,
                             <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                             <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment run concurrently.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html#AbstractProfileProfileAligner(org.biojava.nbio.core.alignment.template.Profile,%20org.biojava.nbio.core.alignment.template.Profile,%20org.biojava.nbio.alignment.template.GapPenalty,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">AbstractProfileProfileAligner</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;query,
                             <a href="../../../../../../../org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">S</a>,<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;target,
                             <a href="../../../../../../../org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a>&nbsp;gapPenalty,
                             <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="type parameter in AbstractProfileProfileAligner">C</a>&gt;&nbsp;subMatrix)</code>
<div class="block">Prepares for a profile-profile alignment.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.alignment">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../../../../../org/biojava/nbio/core/alignment/package-summary.html">org.biojava.nbio.core.alignment</a> declared as <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SimpleProfile.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/SimpleProfile.html#matrix">matrix</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.alignment.matrices">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/package-summary.html">org.biojava.nbio.core.alignment.matrices</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/package-summary.html">org.biojava.nbio.core.alignment.matrices</a> that implement <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/ScaledSubstitutionMatrix.html" title="class in org.biojava.nbio.core.alignment.matrices">ScaledSubstitutionMatrix</a></strong></code>
<div class="block">The biojava-alignment module represents substitution matrices with short
 values.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SimpleSubstitutionMatrix.html" title="class in org.biojava.nbio.core.alignment.matrices">SimpleSubstitutionMatrix</a>&lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;</strong></code>
<div class="block">Implements a data structure which holds the score (penalty or bonus) given during alignment for the exchange of one
 <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> in a sequence for another.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/package-summary.html">org.biojava.nbio.core.alignment.matrices</a> that return <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getAminoAcidSubstitutionMatrix(java.lang.String)">getAminoAcidSubstitutionMatrix</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name)</code>
<div class="block">Returns a substitution matrix for <a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>amino acids</code></a> given by the name <code>name</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum100()">getBlosum100</a></strong>()</code>
<div class="block">Returns Blosum 100 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum30()">getBlosum30</a></strong>()</code>
<div class="block">Returns Blosum 30 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum35()">getBlosum35</a></strong>()</code>
<div class="block">Returns Blosum 35 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum40()">getBlosum40</a></strong>()</code>
<div class="block">Returns Blosum 40 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum45()">getBlosum45</a></strong>()</code>
<div class="block">Returns Blosum 45 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum50()">getBlosum50</a></strong>()</code>
<div class="block">Returns Blosum 50 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum55()">getBlosum55</a></strong>()</code>
<div class="block">Returns Blosum 55 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum60()">getBlosum60</a></strong>()</code>
<div class="block">Returns Blosum 60 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum62()">getBlosum62</a></strong>()</code>
<div class="block">Returns Blosum 62 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SimpleSubstitutionMatrix.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SimpleSubstitutionMatrix.html#getBlosum62()">getBlosum62</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum65()">getBlosum65</a></strong>()</code>
<div class="block">Returns Blosum 65 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum70()">getBlosum70</a></strong>()</code>
<div class="block">Returns Blosum 70 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum75()">getBlosum75</a></strong>()</code>
<div class="block">Returns Blosum 75 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum80()">getBlosum80</a></strong>()</code>
<div class="block">Returns Blosum 80 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum85()">getBlosum85</a></strong>()</code>
<div class="block">Returns Blosum 85 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getBlosum90()">getBlosum90</a></strong>()</code>
<div class="block">Returns Blosum 90 matrix by Henikoff & Henikoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getGonnet250()">getGonnet250</a></strong>()</code>
<div class="block">Returns PAM 250 matrix by Gonnet, Cohen & Benner</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getIdentity()">getIdentity</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">DefaultAAIndexProvider.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/DefaultAAIndexProvider.html#getMatrix(java.lang.String)">getMatrix</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;matrixName)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">AAIndexProvider.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/AAIndexProvider.html#getMatrix(java.lang.String)">getMatrix</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;matrixName)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getMatrixFromAAINDEX(java.lang.String)">getMatrixFromAAINDEX</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;matrixName)</code>
<div class="block">Returns any matrix from the AAINDEX database file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getNuc4_2()">getNuc4_2</a></strong>()</code>
<div class="block">Returns Nuc 4.2 matrix by Lowe
 Only the first nucleotide sequence to align can contain ambiguous nucleotides</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getNuc4_4()">getNuc4_4</a></strong>()</code>
<div class="block">Returns Nuc 4.4 matrix by Lowe
 Both of the nucleotide sequences to align can contain ambiguous nucleotides</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrixHelper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html#getPAM250()">getPAM250</a></strong>()</code>
<div class="block">Returns PAM 250 matrix by Dayhoff</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SimpleSubstitutionMatrix.html" title="type parameter in SimpleSubstitutionMatrix">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SimpleSubstitutionMatrix.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/SimpleSubstitutionMatrix.html#normalizeMatrix(short)">normalizeMatrix</a></strong>(short&nbsp;scale)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">ScaledSubstitutionMatrix.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/ScaledSubstitutionMatrix.html#normalizeMatrix(short)">normalizeMatrix</a></strong>(short&nbsp;scale)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/package-summary.html">org.biojava.nbio.core.alignment.matrices</a> that return types with arguments of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">AAIndexFileParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/matrices/AAIndexFileParser.html#getMatrices()">getMatrices</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.alignment.template">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/core/alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/core/alignment/template/package-summary.html">org.biojava.nbio.core.alignment.template</a> that return <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="type parameter in SubstitutionMatrix">C</a>&gt;</code></td>
<td class="colLast"><span class="strong">SubstitutionMatrix.</span><code><strong><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html#normalizeMatrix(short)">normalizeMatrix</a></strong>(short&nbsp;scale)</code>
<div class="block">Rescales the matrix so that to <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html#getMaxValue()"><code>getMaxValue()</code></a> - <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html#getMinValue()"><code>getMinValue()</code></a> = scale.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.phylo">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/phylo/package-summary.html">org.biojava.nbio.phylo</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/phylo/package-summary.html">org.biojava.nbio.phylo</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#dissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">dissimilarityScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa,
                  <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code>
<div class="block">The dissimilarity score is the additive inverse of the similarity score
 (sum of scores) between two aligned sequences using a substitution model
 (Substitution Matrix).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static &lt;C extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;D&gt;,D extends <a href="../../../../../../../org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a>&gt;&nbsp;<br>org.forester.evoinference.matrix.distance.DistanceMatrix</code></td>
<td class="colLast"><span class="strong">DistanceMatrixCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/phylo/DistanceMatrixCalculator.html#fractionalDissimilarityScore(org.biojava.nbio.core.sequence.MultipleSequenceAlignment,%20org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">fractionalDissimilarityScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a>&lt;C,D&gt;&nbsp;msa,
                            <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;D&gt;&nbsp;M)</code>
<div class="block">The fractional dissimilarity score (Ds) is a relative measure of the
 dissimilarity between two aligned sequences.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.ce">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> declared as <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">CeParameters.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeParameters.html#substitutionMatrix">substitutionMatrix</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that return <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></td>
<td class="colLast"><span class="strong">CeParameters.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeParameters.html#getSubstitutionMatrix()">getSubstitutionMatrix</a></strong>()</code>
<div class="block">Sets the  substitution matrix to be used for influencing the alignment with sequence conservation information.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeParameters.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeParameters.html#setSubstitutionMatrix(org.biojava.nbio.core.alignment.template.SubstitutionMatrix)">setSubstitutionMatrix</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;substitutionMatrix)</code>
<div class="block">Sets the  substitution matrix to be used for influencing the alignment with sequence conservation information.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/core/alignment/template/class-use/SubstitutionMatrix.html" target="_top">Frames</a></li>
<li><a href="SubstitutionMatrix.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
