<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:50 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Class org.biojava.nbio.core.sequence.DNASequence (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.biojava.nbio.core.sequence.DNASequence (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/class-use/DNASequence.html" target="_top">Frames</a></li>
<li><a href="DNASequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.core.sequence.DNASequence" class="title">Uses of Class<br>org.biojava.nbio.core.sequence.DNASequence</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence">org.biojava.nbio.core.sequence</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome">org.biojava.nbio.genome</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome.parsers.geneid">org.biojava.nbio.genome.parsers.geneid</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.genome.parsers.gff">org.biojava.nbio.genome.parsers.gff</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.sequencing.io.fastq">org.biojava.nbio.sequencing.io.fastq</a></td>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.core.sequence">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing subclasses, and an explanation">
<caption><span>Subclasses of <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></strong></code>
<div class="block">Represents a exon or coding sequence in a gene.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></strong></code>
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></strong></code>
<div class="block">A gene contains a collection of Exon sequences</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></strong></code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></strong></code>
<div class="block">Used to map the start codon feature on a gene</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></strong></code>
<div class="block">Used to map the stop codon sequence on a gene</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>class&nbsp;</code></td>
<td class="colLast"><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></strong></code>
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing fields, and an explanation">
<caption><span>Fields in <a href="../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> declared as <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Field and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">StopCodonSequence.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/StopCodonSequence.html#parentGeneSequence">parentGeneSequence</a></strong></code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">StartCodonSequence.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/StartCodonSequence.html#parentGeneSequence">parentGeneSequence</a></strong></code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/core/sequence/package-summary.html">org.biojava.nbio.core.sequence</a> that return <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">TranscriptSequence.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/TranscriptSequence.html#getDNACodingSequence()">getDNACodingSequence</a></strong>()</code>
<div class="block">Get the stitched together CDS sequences then maps to the cDNA</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">GeneSequence.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/GeneSequence.html#getSequence5PrimeTo3Prime()">getSequence5PrimeTo3Prime</a></strong>()</code>
<div class="block">Try to give method clarity where you want a DNASequence coding in the 5' to 3' direction
 Returns the DNASequence representative of the 5' and 3' reading based on strand</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> that return types with arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">FastaReaderHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.File)">readFastaDNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">FastaReaderHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.File,%20boolean)">readFastaDNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                    boolean&nbsp;lazySequenceLoad)</code>
<div class="block">Selecting lazySequenceLoad=true will parse the FASTA file and figure out the accessionid and offsets and return sequence objects
 that can in the future read the sequence from the disk.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">FastaReaderHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaReaderHelper.html#readFastaDNASequence(java.io.InputStream)">readFastaDNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>
<div class="block">Read a fasta DNA sequence</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">GenbankReaderHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html#readGenbankDNASequence(java.io.File)">readGenbankDNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">GenbankReaderHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html#readGenbankDNASequence(java.io.File,%20boolean)">readGenbankDNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                      boolean&nbsp;lazySequenceLoad)</code>
<div class="block">Selecting lazySequenceLoad=true will parse the Genbank file and figure out the accessionid and offsets and return sequence objects
 that can in the future read the sequence from the disk.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">GenbankReaderHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html#readGenbankDNASequence(java.io.InputStream)">readGenbankDNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;inStream)</code>
<div class="block">Read a Genbank DNA sequence</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a> with type arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FastaWriterHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaWriterHelper.html#writeNucleotideSequence(java.io.File,%20java.util.Collection)">writeNucleotideSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code>
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GenbankWriterHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeNucleotideSequence(java.io.File,%20java.util.Collection)">writeNucleotideSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code>
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">FastaWriterHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/FastaWriterHelper.html#writeNucleotideSequence(java.io.OutputStream,%20java.util.Collection)">writeNucleotideSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
                       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code>
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GenbankWriterHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeNucleotideSequence(java.io.OutputStream,%20java.util.Collection)">writeNucleotideSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
                       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences)</code>
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GenbankWriterHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html#writeNucleotideSequence(java.io.OutputStream,%20java.util.Collection,%20java.lang.String)">writeNucleotideSequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/OutputStream.html?is-external=true" title="class or interface in java.io">OutputStream</a>&nbsp;outputStream,
                       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequences,
                       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;seqType)</code>
<div class="block">Write a collection of NucleotideSequences to a file</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../../../../org/biojava/nbio/genome/package-summary.html">org.biojava.nbio.genome</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../org/biojava/nbio/genome/package-summary.html">org.biojava.nbio.genome</a> with type arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">GeneFeatureHelper.</span><code><strong><a href="../../../../../../org/biojava/nbio/genome/GeneFeatureHelper.html#getChromosomeSequenceFromDNASequence(java.util.LinkedHashMap)">getChromosomeSequenceFromDNASequence</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;dnaSequenceList)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome.parsers.geneid">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../../../../org/biojava/nbio/genome/parsers/geneid/package-summary.html">org.biojava.nbio.genome.parsers.geneid</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/genome/parsers/geneid/package-summary.html">org.biojava.nbio.genome.parsers.geneid</a> that return types with arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/util/LinkedHashMap.html?is-external=true" title="class or interface in java.util">LinkedHashMap</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;</code></td>
<td class="colLast"><span class="strong">GeneIDXMLReader.</span><code><strong><a href="../../../../../../org/biojava/nbio/genome/parsers/geneid/GeneIDXMLReader.html#getDNACodingSequences()">getDNACodingSequences</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.genome.parsers.gff">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../../../../org/biojava/nbio/genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a> with parameters of type <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">FeatureList.</span><code><strong><a href="../../../../../../org/biojava/nbio/genome/parsers/gff/FeatureList.html#splice(org.biojava.nbio.core.sequence.DNASequence)">splice</a></strong>(<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&nbsp;sequence)</code>
<div class="block">Concatenate successive portions of the specified sequence
 using the feature locations in the list.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../org/biojava/nbio/genome/parsers/gff/package-summary.html">org.biojava.nbio.genome.parsers.gff</a> with type arguments of type <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">GCStats.</span><code><strong><a href="../../../../../../org/biojava/nbio/genome/parsers/gff/GCStats.html#getGCStats(java.util.Collection)">getGCStats</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Collection.html?is-external=true" title="class or interface in java.util">Collection</a>&lt;<a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a>&gt;&nbsp;sequences)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.sequencing.io.fastq">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a> in <a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/package-summary.html">org.biojava.nbio.sequencing.io.fastq</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/package-summary.html">org.biojava.nbio.sequencing.io.fastq</a> that return <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">FastqTools.</span><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequence(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequence</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> from the specified FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">FastqTools.</span><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequenceWithErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequenceWithErrorProbabilities</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with error probabilities from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">FastqTools.</span><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequenceWithQualityScores(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequenceWithQualityScores</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores from the specified
 FASTQ formatted sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></code></td>
<td class="colLast"><span class="strong">FastqTools.</span><code><strong><a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/FastqTools.html#createDNASequenceWithQualityScoresAndErrorProbabilities(org.biojava.nbio.sequencing.io.fastq.Fastq)">createDNASequenceWithQualityScoresAndErrorProbabilities</a></strong>(<a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/Fastq.html" title="class in org.biojava.nbio.sequencing.io.fastq">Fastq</a>&nbsp;fastq)</code>
<div class="block">Create and return a new <a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>DNASequence</code></a> with quality scores and error probabilities from the
 specified FASTQ formatted sequence.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/class-use/DNASequence.html" target="_top">Frames</a></li>
<li><a href="DNASequence.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
