<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html lang="en">
<head>
<title>Source code</title>
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<div class="sourceContainer">
<pre><span class="sourceLineNo">001</span>/*<a name="line.1"></a>
<span class="sourceLineNo">002</span> *                    BioJava development code<a name="line.2"></a>
<span class="sourceLineNo">003</span> *<a name="line.3"></a>
<span class="sourceLineNo">004</span> * This code may be freely distributed and modified under the<a name="line.4"></a>
<span class="sourceLineNo">005</span> * terms of the GNU Lesser General Public Licence.  This should<a name="line.5"></a>
<span class="sourceLineNo">006</span> * be distributed with the code.  If you do not have a copy,<a name="line.6"></a>
<span class="sourceLineNo">007</span> * see:<a name="line.7"></a>
<span class="sourceLineNo">008</span> *<a name="line.8"></a>
<span class="sourceLineNo">009</span> *      http://www.gnu.org/copyleft/lesser.html<a name="line.9"></a>
<span class="sourceLineNo">010</span> *<a name="line.10"></a>
<span class="sourceLineNo">011</span> * Copyright for this code is held jointly by the individual<a name="line.11"></a>
<span class="sourceLineNo">012</span> * authors.  These should be listed in @author doc comments.<a name="line.12"></a>
<span class="sourceLineNo">013</span> *<a name="line.13"></a>
<span class="sourceLineNo">014</span> * For more information on the BioJava project and its aims,<a name="line.14"></a>
<span class="sourceLineNo">015</span> * or to join the biojava-l mailing list, visit the home page<a name="line.15"></a>
<span class="sourceLineNo">016</span> * at:<a name="line.16"></a>
<span class="sourceLineNo">017</span> *<a name="line.17"></a>
<span class="sourceLineNo">018</span> *      http://www.biojava.org/<a name="line.18"></a>
<span class="sourceLineNo">019</span> *<a name="line.19"></a>
<span class="sourceLineNo">020</span> */<a name="line.20"></a>
<span class="sourceLineNo">021</span><a name="line.21"></a>
<span class="sourceLineNo">022</span>package org.biojava.bio.program.abi;<a name="line.22"></a>
<span class="sourceLineNo">023</span><a name="line.23"></a>
<span class="sourceLineNo">024</span>import java.io.File;<a name="line.24"></a>
<span class="sourceLineNo">025</span>import java.io.IOException;<a name="line.25"></a>
<span class="sourceLineNo">026</span>import java.io.InputStream;<a name="line.26"></a>
<span class="sourceLineNo">027</span>import java.io.Serializable;<a name="line.27"></a>
<span class="sourceLineNo">028</span>import java.util.ArrayList;<a name="line.28"></a>
<span class="sourceLineNo">029</span>import java.util.List;<a name="line.29"></a>
<span class="sourceLineNo">030</span>import java.util.Map;<a name="line.30"></a>
<span class="sourceLineNo">031</span><a name="line.31"></a>
<span class="sourceLineNo">032</span>import org.biojava.bio.BioError;<a name="line.32"></a>
<span class="sourceLineNo">033</span>import org.biojava.bio.chromatogram.AbstractChromatogram;<a name="line.33"></a>
<span class="sourceLineNo">034</span>import org.biojava.bio.chromatogram.Chromatogram;<a name="line.34"></a>
<span class="sourceLineNo">035</span>import org.biojava.bio.chromatogram.UnsupportedChromatogramFormatException;<a name="line.35"></a>
<span class="sourceLineNo">036</span>import org.biojava.bio.seq.DNATools;<a name="line.36"></a>
<span class="sourceLineNo">037</span>import org.biojava.bio.symbol.AtomicSymbol;<a name="line.37"></a>
<span class="sourceLineNo">038</span>import org.biojava.bio.symbol.IllegalAlphabetException;<a name="line.38"></a>
<span class="sourceLineNo">039</span>import org.biojava.bio.symbol.IllegalSymbolException;<a name="line.39"></a>
<span class="sourceLineNo">040</span>import org.biojava.bio.symbol.IntegerAlphabet;<a name="line.40"></a>
<span class="sourceLineNo">041</span>import org.biojava.bio.symbol.Symbol;<a name="line.41"></a>
<span class="sourceLineNo">042</span>import org.biojava.utils.SmallMap;<a name="line.42"></a>
<span class="sourceLineNo">043</span><a name="line.43"></a>
<span class="sourceLineNo">044</span><a name="line.44"></a>
<span class="sourceLineNo">045</span>/**<a name="line.45"></a>
<span class="sourceLineNo">046</span> * An implementation of {@link org.biojava.bio.chromatogram.Chromatogram} to<a name="line.46"></a>
<span class="sourceLineNo">047</span> * encapulsulate chromatogram data extracted from the files produced by ABI<a name="line.47"></a>
<span class="sourceLineNo">048</span> * sequencers, such as the the 377 and the 3700.  The format was described by<a name="line.48"></a>
<span class="sourceLineNo">049</span> * Clark Tibbetts in his paper "Raw Data File Formats, and the Digital and<a name="line.49"></a>
<span class="sourceLineNo">050</span> * Analog Raw Data Streams of the ABI PRISM 377 DNA Sequencer."  Available<a name="line.50"></a>
<span class="sourceLineNo">051</span> * online &lt;kbd&gt;&lt;a href="http://www-2.cs.cmu.edu/afs/cs/project/genome/WWW/Papers/clark.html"&gt;<a name="line.51"></a>
<span class="sourceLineNo">052</span> * http://www-2.cs.cmu.edu/afs/cs/project/genome/WWW/Papers/clark.html&lt;/a&gt;&lt;/kbd&gt;<a name="line.52"></a>
<span class="sourceLineNo">053</span> *<a name="line.53"></a>
<span class="sourceLineNo">054</span> * @author Rhett Sutphin (&lt;a href="http://genome.uiowa.edu/"&gt;UI CBCB&lt;/a&gt;)<a name="line.54"></a>
<span class="sourceLineNo">055</span> * @author Richard Holland<a name="line.55"></a>
<span class="sourceLineNo">056</span> * @see ABIFParser<a name="line.56"></a>
<span class="sourceLineNo">057</span> */<a name="line.57"></a>
<span class="sourceLineNo">058</span>public class ABIFChromatogram extends AbstractChromatogram implements Serializable {<a name="line.58"></a>
<span class="sourceLineNo">059</span>    public ABIFChromatogram() {<a name="line.59"></a>
<span class="sourceLineNo">060</span>        super();<a name="line.60"></a>
<span class="sourceLineNo">061</span>    }<a name="line.61"></a>
<span class="sourceLineNo">062</span><a name="line.62"></a>
<span class="sourceLineNo">063</span>    /** Create a new ABIF object from a file.<a name="line.63"></a>
<span class="sourceLineNo">064</span>     *  &lt;p&gt;<a name="line.64"></a>
<span class="sourceLineNo">065</span>     *  This method is more efficent than {@link #create(InputStream)}.<a name="line.65"></a>
<span class="sourceLineNo">066</span>     *  &lt;/p&gt;<a name="line.66"></a>
<span class="sourceLineNo">067</span>     */<a name="line.67"></a>
<span class="sourceLineNo">068</span>    public static ABIFChromatogram create(File f)<a name="line.68"></a>
<span class="sourceLineNo">069</span>    throws IOException, UnsupportedChromatogramFormatException {<a name="line.69"></a>
<span class="sourceLineNo">070</span>        ABIFChromatogram newOne = new ABIFChromatogram();<a name="line.70"></a>
<span class="sourceLineNo">071</span>        newOne.load(f);<a name="line.71"></a>
<span class="sourceLineNo">072</span>        return newOne;<a name="line.72"></a>
<span class="sourceLineNo">073</span>    }<a name="line.73"></a>
<span class="sourceLineNo">074</span><a name="line.74"></a>
<span class="sourceLineNo">075</span>    /**<a name="line.75"></a>
<span class="sourceLineNo">076</span>     * Create a new ABIF object from a stream of bytes.<a name="line.76"></a>
<span class="sourceLineNo">077</span>     * &lt;p&gt;<a name="line.77"></a>
<span class="sourceLineNo">078</span>     * Due to the non-single-pass design of the ABI format, this method will<a name="line.78"></a>
<span class="sourceLineNo">079</span>     * wrap the InputStream in an {@link org.biojava.utils.io.CachingInputStream}.<a name="line.79"></a>
<span class="sourceLineNo">080</span>     * For this reason, {@link #create(File)} should be preferred.<a name="line.80"></a>
<span class="sourceLineNo">081</span>     * &lt;/p&gt;<a name="line.81"></a>
<span class="sourceLineNo">082</span>     * @param in the stream from which to read<a name="line.82"></a>
<span class="sourceLineNo">083</span>     * @return a new ABIFChromatogram object<a name="line.83"></a>
<span class="sourceLineNo">084</span>     * @throws IOException if there is a problem with the underlying stream<a name="line.84"></a>
<span class="sourceLineNo">085</span>     */<a name="line.85"></a>
<span class="sourceLineNo">086</span>    public static ABIFChromatogram create(InputStream in)<a name="line.86"></a>
<span class="sourceLineNo">087</span>    throws IOException, UnsupportedChromatogramFormatException {<a name="line.87"></a>
<span class="sourceLineNo">088</span>        ABIFChromatogram newOne = new ABIFChromatogram();<a name="line.88"></a>
<span class="sourceLineNo">089</span>        newOne.load(in);<a name="line.89"></a>
<span class="sourceLineNo">090</span>        return newOne;<a name="line.90"></a>
<span class="sourceLineNo">091</span>    }<a name="line.91"></a>
<span class="sourceLineNo">092</span><a name="line.92"></a>
<span class="sourceLineNo">093</span>    protected ABIFChromatogram load(File f)<a name="line.93"></a>
<span class="sourceLineNo">094</span>    throws IOException, UnsupportedChromatogramFormatException {<a name="line.94"></a>
<span class="sourceLineNo">095</span>        new Parser(f);<a name="line.95"></a>
<span class="sourceLineNo">096</span>        return this;<a name="line.96"></a>
<span class="sourceLineNo">097</span>    }<a name="line.97"></a>
<span class="sourceLineNo">098</span><a name="line.98"></a>
<span class="sourceLineNo">099</span>    protected ABIFChromatogram load(InputStream in)<a name="line.99"></a>
<span class="sourceLineNo">100</span>    throws IOException, UnsupportedChromatogramFormatException {<a name="line.100"></a>
<span class="sourceLineNo">101</span>        new Parser(in);<a name="line.101"></a>
<span class="sourceLineNo">102</span>        return this;<a name="line.102"></a>
<span class="sourceLineNo">103</span>    }<a name="line.103"></a>
<span class="sourceLineNo">104</span><a name="line.104"></a>
<span class="sourceLineNo">105</span>    protected AbstractChromatogram reverseComplementInstance() {<a name="line.105"></a>
<span class="sourceLineNo">106</span>        return new ABIFChromatogram();<a name="line.106"></a>
<span class="sourceLineNo">107</span>    }<a name="line.107"></a>
<span class="sourceLineNo">108</span><a name="line.108"></a>
<span class="sourceLineNo">109</span>    /**<a name="line.109"></a>
<span class="sourceLineNo">110</span>     * An extension of {@link ABIFParser} that reads the particular fields from<a name="line.110"></a>
<span class="sourceLineNo">111</span>     * the ABIF that contain the chromatogram data and initializes the fields<a name="line.111"></a>
<span class="sourceLineNo">112</span>     * in its enclosing &lt;code&gt;ABIFChromatogram&lt;/code&gt; instance.<a name="line.112"></a>
<span class="sourceLineNo">113</span>     */<a name="line.113"></a>
<span class="sourceLineNo">114</span>    protected class Parser extends ABIFParser {<a name="line.114"></a>
<span class="sourceLineNo">115</span>        public Parser(InputStream in)<a name="line.115"></a>
<span class="sourceLineNo">116</span>        throws IOException, UnsupportedChromatogramFormatException {<a name="line.116"></a>
<span class="sourceLineNo">117</span>            super(in);<a name="line.117"></a>
<span class="sourceLineNo">118</span>            parse();<a name="line.118"></a>
<span class="sourceLineNo">119</span>        }<a name="line.119"></a>
<span class="sourceLineNo">120</span><a name="line.120"></a>
<span class="sourceLineNo">121</span>        public Parser(File f)<a name="line.121"></a>
<span class="sourceLineNo">122</span>        throws IOException, UnsupportedChromatogramFormatException {<a name="line.122"></a>
<span class="sourceLineNo">123</span>            super(f);<a name="line.123"></a>
<span class="sourceLineNo">124</span>            parse();<a name="line.124"></a>
<span class="sourceLineNo">125</span>        }<a name="line.125"></a>
<span class="sourceLineNo">126</span><a name="line.126"></a>
<span class="sourceLineNo">127</span>        private final void parse()<a name="line.127"></a>
<span class="sourceLineNo">128</span>        throws IOException, UnsupportedChromatogramFormatException {<a name="line.128"></a>
<span class="sourceLineNo">129</span>            // read filter-wheel-order tag<a name="line.129"></a>
<span class="sourceLineNo">130</span>            char[] fwo_ = new char[4];<a name="line.130"></a>
<span class="sourceLineNo">131</span>            ABIFParser.TaggedDataRecord fwoRec = getDataRecord("FWO_", 1);<a name="line.131"></a>
<span class="sourceLineNo">132</span>            if (fwoRec == null)<a name="line.132"></a>
<span class="sourceLineNo">133</span>                throw new UnsupportedChromatogramFormatException("No FWO_ (1) record in ABIF file, therefore no trace data");<a name="line.133"></a>
<span class="sourceLineNo">134</span>            fwo_[0] = (char) ( (fwoRec.dataRecord &gt;&gt;&gt; 24) &amp; 0xff );<a name="line.134"></a>
<span class="sourceLineNo">135</span>            fwo_[1] = (char) ( (fwoRec.dataRecord &gt;&gt;&gt; 16) &amp; 0xff );<a name="line.135"></a>
<span class="sourceLineNo">136</span>            fwo_[2] = (char) ( (fwoRec.dataRecord &gt;&gt;&gt; 8 ) &amp; 0xff );<a name="line.136"></a>
<span class="sourceLineNo">137</span>            fwo_[3] = (char) ( (fwoRec.dataRecord       ) &amp; 0xff );<a name="line.137"></a>
<span class="sourceLineNo">138</span><a name="line.138"></a>
<span class="sourceLineNo">139</span>            Symbol sym;<a name="line.139"></a>
<span class="sourceLineNo">140</span>            clearTraces();<a name="line.140"></a>
<span class="sourceLineNo">141</span>            for (int i = 0 ; i &lt; 4 ; i++) {<a name="line.141"></a>
<span class="sourceLineNo">142</span>                <a name="line.142"></a>
<span class="sourceLineNo">143</span>                try {<a name="line.143"></a>
<span class="sourceLineNo">144</span>                    sym = ABIFParser.decodeDNAToken(fwo_[i]);<a name="line.144"></a>
<span class="sourceLineNo">145</span>                } catch (IllegalSymbolException ise) {<a name="line.145"></a>
<span class="sourceLineNo">146</span>                    throw new UnsupportedChromatogramFormatException("An unexpected character (" + fwo_[i] +") was found in the FWO_ tag.  Parsing cannot continue.");<a name="line.146"></a>
<span class="sourceLineNo">147</span>                }<a name="line.147"></a>
<span class="sourceLineNo">148</span>                if (!(sym instanceof AtomicSymbol)) {<a name="line.148"></a>
<span class="sourceLineNo">149</span>                    throw new UnsupportedChromatogramFormatException("An unexpected character (" + fwo_[i] +") was found in the FWO_ tag.  Parsing cannot continue.");<a name="line.149"></a>
<span class="sourceLineNo">150</span>                }<a name="line.150"></a>
<span class="sourceLineNo">151</span>                parseTrace((AtomicSymbol) sym, i+9);<a name="line.151"></a>
<span class="sourceLineNo">152</span>            }<a name="line.152"></a>
<span class="sourceLineNo">153</span>            parseBaseCalls();<a name="line.153"></a>
<span class="sourceLineNo">154</span>            getDataAccess().finishedReading();<a name="line.154"></a>
<span class="sourceLineNo">155</span>        }<a name="line.155"></a>
<span class="sourceLineNo">156</span><a name="line.156"></a>
<span class="sourceLineNo">157</span>        private void parseTrace(AtomicSymbol sym, int whichData) throws IOException, UnsupportedChromatogramFormatException {<a name="line.157"></a>
<span class="sourceLineNo">158</span>            TaggedDataRecord dataPtr = getDataRecord("DATA", whichData);<a name="line.158"></a>
<span class="sourceLineNo">159</span>            if (dataPtr.numberOfElements &gt; Integer.MAX_VALUE)<a name="line.159"></a>
<span class="sourceLineNo">160</span>                throw new UnsupportedChromatogramFormatException("Chromatogram has more than " + Integer.MAX_VALUE + " trace samples -- can't handle it");<a name="line.160"></a>
<span class="sourceLineNo">161</span>            int count = (int) dataPtr.numberOfElements;<a name="line.161"></a>
<span class="sourceLineNo">162</span>            int[] trace = new int[count];<a name="line.162"></a>
<span class="sourceLineNo">163</span>            int max = -1;<a name="line.163"></a>
<span class="sourceLineNo">164</span>            setBits(8*dataPtr.elementLength);<a name="line.164"></a>
<span class="sourceLineNo">165</span>            if (dataPtr.elementLength == 2) {<a name="line.165"></a>
<span class="sourceLineNo">166</span>                byte[] shortArray = dataPtr.offsetData;<a name="line.166"></a>
<span class="sourceLineNo">167</span>                int i = 0;<a name="line.167"></a>
<span class="sourceLineNo">168</span>                for (int s = 0; s &lt; shortArray.length; s += 2) {<a name="line.168"></a>
<span class="sourceLineNo">169</span>                    trace[i] =  ((short)((shortArray[s] &lt;&lt; 8) | (shortArray[s + 1] &amp; 0xff))) &amp; 0xffff;<a name="line.169"></a>
<span class="sourceLineNo">170</span>                    max = Math.max(trace[i++], max);<a name="line.170"></a>
<span class="sourceLineNo">171</span>                }<a name="line.171"></a>
<span class="sourceLineNo">172</span>            }<a name="line.172"></a>
<span class="sourceLineNo">173</span>            else if (dataPtr.elementLength == 1) {<a name="line.173"></a>
<span class="sourceLineNo">174</span>                byte[] byteArray = dataPtr.offsetData;<a name="line.174"></a>
<span class="sourceLineNo">175</span>                for (int i = 0; i &lt; byteArray.length; i++) {<a name="line.175"></a>
<span class="sourceLineNo">176</span>                    trace[i] = byteArray[i] &amp; 0xff;<a name="line.176"></a>
<span class="sourceLineNo">177</span>                    max = Math.max(trace[i], max);<a name="line.177"></a>
<span class="sourceLineNo">178</span>                }<a name="line.178"></a>
<span class="sourceLineNo">179</span>            }<a name="line.179"></a>
<span class="sourceLineNo">180</span>            else {<a name="line.180"></a>
<span class="sourceLineNo">181</span>                throw new UnsupportedChromatogramFormatException("Only 8- and 16-bit trace samples are supported");<a name="line.181"></a>
<span class="sourceLineNo">182</span>            }<a name="line.182"></a>
<span class="sourceLineNo">183</span>            <a name="line.183"></a>
<span class="sourceLineNo">184</span>            try {<a name="line.184"></a>
<span class="sourceLineNo">185</span>                setTrace(sym, trace, max);<a name="line.185"></a>
<span class="sourceLineNo">186</span>            } catch (IllegalSymbolException ise) {<a name="line.186"></a>
<span class="sourceLineNo">187</span>                throw new BioError("Can't happen", ise);<a name="line.187"></a>
<span class="sourceLineNo">188</span>            }<a name="line.188"></a>
<span class="sourceLineNo">189</span>        }<a name="line.189"></a>
<span class="sourceLineNo">190</span><a name="line.190"></a>
<span class="sourceLineNo">191</span>        private void parseBaseCalls() throws IOException, UnsupportedChromatogramFormatException {<a name="line.191"></a>
<span class="sourceLineNo">192</span>            // do offsets, then call letters<a name="line.192"></a>
<span class="sourceLineNo">193</span>            // offsets are in PLOC1 (we'll use the possibly-edited stream)<a name="line.193"></a>
<span class="sourceLineNo">194</span>            TaggedDataRecord offsetsPtr = getDataRecord("PLOC", 1);<a name="line.194"></a>
<span class="sourceLineNo">195</span>            // call letters are int PBAS1<a name="line.195"></a>
<span class="sourceLineNo">196</span>            TaggedDataRecord basesPtr = getDataRecord("PBAS", 1);<a name="line.196"></a>
<span class="sourceLineNo">197</span>            // these should be equal, but just in case...<a name="line.197"></a>
<span class="sourceLineNo">198</span>            if (offsetsPtr.numberOfElements != basesPtr.numberOfElements)<a name="line.198"></a>
<span class="sourceLineNo">199</span>                throw new BioError("PLOC and PBAS are different lengths.  Can't proceed.");<a name="line.199"></a>
<span class="sourceLineNo">200</span>            if (offsetsPtr.numberOfElements &gt; Integer.MAX_VALUE)<a name="line.200"></a>
<span class="sourceLineNo">201</span>                throw new UnsupportedChromatogramFormatException("Chromatogram has more than " + Integer.MAX_VALUE + " base calls -- can't handle it");<a name="line.201"></a>
<span class="sourceLineNo">202</span>            int count = (int) offsetsPtr.numberOfElements;<a name="line.202"></a>
<span class="sourceLineNo">203</span>            // the list of called bases<a name="line.203"></a>
<span class="sourceLineNo">204</span>            List dna = new ArrayList(count);<a name="line.204"></a>
<span class="sourceLineNo">205</span>            // the list of offsets<a name="line.205"></a>
<span class="sourceLineNo">206</span>            List offsets = new ArrayList(count);<a name="line.206"></a>
<span class="sourceLineNo">207</span>            // start reading offsets, creating SimpleBaseCalls along the way<a name="line.207"></a>
<span class="sourceLineNo">208</span>            if (offsetsPtr.elementLength == 2) {<a name="line.208"></a>
<span class="sourceLineNo">209</span>                byte[] shortArray = offsetsPtr.offsetData;<a name="line.209"></a>
<span class="sourceLineNo">210</span>                IntegerAlphabet integerAlphabet = IntegerAlphabet.getInstance();<a name="line.210"></a>
<span class="sourceLineNo">211</span>                for (int s = 0; s &lt; shortArray.length; s += 2) {<a name="line.211"></a>
<span class="sourceLineNo">212</span>                    offsets.add(integerAlphabet.getSymbol(((short)((shortArray[s] &lt;&lt; 8) | (shortArray[s + 1] &amp; 0xff))) &amp; 0xffff));<a name="line.212"></a>
<span class="sourceLineNo">213</span>                }<a name="line.213"></a>
<span class="sourceLineNo">214</span>            }<a name="line.214"></a>
<span class="sourceLineNo">215</span>            else if (offsetsPtr.elementLength == 1) {<a name="line.215"></a>
<span class="sourceLineNo">216</span>                byte[] byteArray = offsetsPtr.offsetData;<a name="line.216"></a>
<span class="sourceLineNo">217</span>                IntegerAlphabet integerAlphabet = IntegerAlphabet.getInstance();<a name="line.217"></a>
<span class="sourceLineNo">218</span>                for (int i = 0 ; i &lt; byteArray.length; i++) {<a name="line.218"></a>
<span class="sourceLineNo">219</span>                    offsets.add(integerAlphabet.getSymbol(byteArray[i] &amp; 0xff));<a name="line.219"></a>
<span class="sourceLineNo">220</span>                }<a name="line.220"></a>
<span class="sourceLineNo">221</span>            }<a name="line.221"></a>
<span class="sourceLineNo">222</span>            else {<a name="line.222"></a>
<span class="sourceLineNo">223</span>                throw new IllegalStateException("Only 8- and 16-bit trace samples are supported");<a name="line.223"></a>
<span class="sourceLineNo">224</span>            }<a name="line.224"></a>
<span class="sourceLineNo">225</span><a name="line.225"></a>
<span class="sourceLineNo">226</span>            // then read the base calls<a name="line.226"></a>
<span class="sourceLineNo">227</span>            try {<a name="line.227"></a>
<span class="sourceLineNo">228</span>                byte[] byteArray = basesPtr.offsetData;<a name="line.228"></a>
<span class="sourceLineNo">229</span>                for (int i = 0; i &lt; byteArray.length; i++) {<a name="line.229"></a>
<span class="sourceLineNo">230</span>                    dna.add(ABIFParser.decodeDNAToken((char) byteArray[i]));<a name="line.230"></a>
<span class="sourceLineNo">231</span>                }<a name="line.231"></a>
<span class="sourceLineNo">232</span>            } catch (IllegalSymbolException ise) {<a name="line.232"></a>
<span class="sourceLineNo">233</span>                throw new BioError("Can't happen", ise);<a name="line.233"></a>
<span class="sourceLineNo">234</span>            }<a name="line.234"></a>
<span class="sourceLineNo">235</span>            // create the base call alignment and set it<a name="line.235"></a>
<span class="sourceLineNo">236</span>            try {<a name="line.236"></a>
<span class="sourceLineNo">237</span>                Map baseCalls = new SmallMap(2);<a name="line.237"></a>
<span class="sourceLineNo">238</span>                baseCalls.put(Chromatogram.DNA, createImmutableSymbolList(DNATools.getDNA(), dna));<a name="line.238"></a>
<span class="sourceLineNo">239</span>                baseCalls.put(Chromatogram.OFFSETS, createImmutableSymbolList(IntegerAlphabet.getInstance(), offsets));<a name="line.239"></a>
<span class="sourceLineNo">240</span>                setBaseCallAlignment(createImmutableAlignment(baseCalls));<a name="line.240"></a>
<span class="sourceLineNo">241</span>            } catch (IllegalAlphabetException iae) {<a name="line.241"></a>
<span class="sourceLineNo">242</span>                throw new BioError("Can't happen", iae);<a name="line.242"></a>
<span class="sourceLineNo">243</span>            } catch (IllegalSymbolException ise) {<a name="line.243"></a>
<span class="sourceLineNo">244</span>                throw new BioError("Can't happen", ise);<a name="line.244"></a>
<span class="sourceLineNo">245</span>            }<a name="line.245"></a>
<span class="sourceLineNo">246</span>        }<a name="line.246"></a>
<span class="sourceLineNo">247</span>    }<a name="line.247"></a>
<span class="sourceLineNo">248</span>}<a name="line.248"></a>




























































</pre>
</div>
</body>
</html>
