<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:38 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>FastaStructureParser (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FastaStructureParser (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaStructureParser.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html" title="class in org.biojava.nbio.structure.io"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/io/FastaStructureParser.html" target="_top">Frames</a></li>
<li><a href="FastaStructureParser.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">org.biojava.nbio.structure.io</div>
<h2 title="Class FastaStructureParser" class="title">Class FastaStructureParser</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li>org.biojava.nbio.structure.io.FastaStructureParser</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<hr>
<br>
<pre>public class <span class="strong">FastaStructureParser</span>
extends <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></pre>
<div class="block">Reads a protein sequence from a fasta file and attempts to match it to a
 3D structure. Any gaps ('-') in the fasta file are preserved as null atoms in
 the output, allowing structural alignments to be read from fasta files.

 <p>Structures are loaded from an AtomCache. For this to work, the accession
 for each protein should be parsed from the fasta header line into a form
 understood by <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html#getStructure(java.lang.String)"><code>AtomCache.getStructure(String)</code></a>.

 <p>Lowercase letters are sometimes used to specify unaligned residues.
 This information can be preserved by using a CasePreservingSequenceCreator,
 which allows the case of residues to be accessed through the
 <a href="../../../../../org/biojava/nbio/core/sequence/template/AbstractSequence.html#getUserCollection()"><code>AbstractSequence.getUserCollection()</code></a> method.</div>
<dl><dt><span class="strong">Author:</span></dt>
  <dd>Spencer Bliven</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser(org.biojava.nbio.core.sequence.io.FastaReader,%20org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></strong>(<a href="../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;reader,
                    <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser(java.io.File,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,%20org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#FastaStructureParser(java.io.InputStream,%20org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface,%20org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface,%20org.biojava.nbio.structure.align.util.AtomCache)">FastaStructureParser</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;is,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</code>&nbsp;</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getAccessions()">getAccessions</a></strong>()</code>
<div class="block">Gets the protein accessions mapped from the Fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[][]</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getResidues()">getResidues</a></strong>()</code>
<div class="block">For each residue in the fasta file, return the ResidueNumber in the
 corresponding structure.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getSequences()">getSequences</a></strong>()</code>
<div class="block">Gets the protein sequences read from the Fasta file.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>[]</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getStructures()">getStructures</a></strong>()</code>
<div class="block">Gets the protein structures mapped from the Fasta file.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#process()">process</a></strong>()</code>
<div class="block">Parses the fasta file and loads it into memory.</div>
</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="FastaStructureParser(java.io.InputStream, org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface, org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface, org.biojava.nbio.structure.align.util.AtomCache)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FastaStructureParser</h4>
<pre>public&nbsp;FastaStructureParser(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html?is-external=true" title="class or interface in java.io">InputStream</a>&nbsp;is,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</pre>
</li>
</ul>
<a name="FastaStructureParser(java.io.File, org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface, org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface, org.biojava.nbio.structure.align.util.AtomCache)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FastaStructureParser</h4>
<pre>public&nbsp;FastaStructureParser(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;file,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;headerParser,
                    <a href="../../../../../org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;sequenceCreator,
                    <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)
                     throws <a href="http://docs.oracle.com/javase/7/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io">FileNotFoundException</a></pre>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/FileNotFoundException.html?is-external=true" title="class or interface in java.io">FileNotFoundException</a></code></dd></dl>
</li>
</ul>
<a name="FastaStructureParser(org.biojava.nbio.core.sequence.io.FastaReader, org.biojava.nbio.structure.align.util.AtomCache)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FastaStructureParser</h4>
<pre>public&nbsp;FastaStructureParser(<a href="../../../../../org/biojava/nbio/core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader</a>&lt;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<a href="../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;reader,
                    <a href="../../../../../org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a>&nbsp;cache)</pre>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="process()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>process</h4>
<pre>public&nbsp;void&nbsp;process()
             throws <a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                    <a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></pre>
<div class="block">Parses the fasta file and loads it into memory.

 Information can be subsequently accessed through
 <a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getSequences()"><code>getSequences()</code></a>,
 <a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getStructures()"><code>getStructures()</code></a>,
 <a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getResidues()"><code>getResidues()</code></a>, and
 <a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#getAccessions()"><code>getAccessions()</code></a>.</div>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/7/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../../org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></code></dd></dl>
</li>
</ul>
<a name="getSequences()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getSequences</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>[]&nbsp;getSequences()</pre>
<div class="block">Gets the protein sequences read from the Fasta file.
 Returns null if <a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#process()"><code>process()</code></a> has not been called.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>An array ProteinSequences from
  parsing the fasta file, or null if process() hasn't been called.</dd></dl>
</li>
</ul>
<a name="getStructures()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getStructures</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>[]&nbsp;getStructures()</pre>
<div class="block">Gets the protein structures mapped from the Fasta file.
 Returns null if <a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#process()"><code>process()</code></a> has not been called.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>An array of Structures for each protein
  in the fasta file, or null if process() hasn't been called.</dd></dl>
</li>
</ul>
<a name="getResidues()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getResidues</h4>
<pre>public&nbsp;<a href="../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[][]&nbsp;getResidues()</pre>
<div class="block">For each residue in the fasta file, return the ResidueNumber in the
 corresponding structure. If the residue cannot be found in the structure,
 that entry will be null. This can happen if that residue was not included
 in the PDB file (eg disordered residues), if the fasta sequence does not
 match the PDB sequence, or if errors occur during the matching process.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>A 2D array of ResidueNumbers, or null if process() hasn't been called.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../../org/biojava/nbio/structure/io/StructureSequenceMatcher.html#matchSequenceToStructure(org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.structure.Structure)"><code>StructureSequenceMatcher.matchSequenceToStructure(ProteinSequence, Structure)</code></a></dd></dl>
</li>
</ul>
<a name="getAccessions()">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getAccessions</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>[]&nbsp;getAccessions()</pre>
<div class="block">Gets the protein accessions mapped from the Fasta file.
 Returns null if <a href="../../../../../org/biojava/nbio/structure/io/FastaStructureParser.html#process()"><code>process()</code></a> has not been called.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>An array of Structures for each protein
  in the fasta file, or null if process() hasn't been called.</dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaStructureParser.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../org/biojava/nbio/structure/io/FileConvert.html" title="class in org.biojava.nbio.structure.io"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?org/biojava/nbio/structure/io/FastaStructureParser.html" target="_top">Frames</a></li>
<li><a href="FastaStructureParser.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
