<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>All Classes and Interfaces (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="class index">
<meta name="generator" content="javadoc/AllClassesIndexWriter">
<link rel="stylesheet" type="text/css" href="stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="script.js"></script>
<script type="text/javascript" src="script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="script-dir/jquery-ui.min.js"></script>
</head>
<body class="all-classes-index-page">
<script type="text/javascript">var pathtoroot = "./";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="index.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="overview-tree.html">Tree</a></li>
<li><a href="deprecated-list.html">Deprecated</a></li>
<li><a href="index-all.html">Index</a></li>
<li><a href="help-doc.html#all-classes">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="All Classes and Interfaces" class="title">All Classes and Interfaces</h1>
</div>
<div id="all-classes-table">
<div class="table-tabs" role="tablist" aria-orientation="horizontal"><button id="all-classes-table-tab0" role="tab" aria-selected="true" aria-controls="all-classes-table.tabpanel" tabindex="0" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table', 2)" class="active-table-tab">All Classes and Interfaces</button><button id="all-classes-table-tab1" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab1', 2)" class="table-tab">Interfaces</button><button id="all-classes-table-tab2" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab2', 2)" class="table-tab">Classes</button><button id="all-classes-table-tab3" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab3', 2)" class="table-tab">Enums</button><button id="all-classes-table-tab5" role="tab" aria-selected="false" aria-controls="all-classes-table.tabpanel" tabindex="-1" onkeydown="switchTab(event)" onclick="show('all-classes-table', 'all-classes-table-tab5', 2)" class="table-tab">Exception Classes</button></div>
<div id="all-classes-table.tabpanel" role="tabpanel" aria-labelledby="all-classes-table-tab0">
<div class="summary-table two-column-summary">
<div class="table-header col-first">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/matrices/AAindexFactory.html" title="class in org.biojava.nbio.core.alignment.matrices">AAindexFactory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Factory class to get Providers for substitution matrices that are provided by
 the AAINDEX database.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/matrices/AAIndexFileParser.html" title="class in org.biojava.nbio.core.alignment.matrices">AAIndexFileParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/matrices/AAIndexProvider.html" title="interface in org.biojava.nbio.core.alignment.matrices">AAIndexProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/ABITrace.html" title="class in org.biojava.nbio.core.sequence.io">ABITrace</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Title: ABITrace</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/ABITracerCompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">ABITracerCompoundSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/AboutDialog.html" title="class in org.biojava.nbio.structure.align.gui">AboutDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An Abstract Class to generalize the visualization of AFP and
 MultipleAlignment structure alignments in Jmol.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/utils/AbstractAnnotation.html" title="class in org.biojava.nbio.ontology.utils">AbstractAnnotation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class to ease the problem of implementing an Annotation to that of
 providing an apropreate implementation of Map.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/AbstractCifFileSupplier.html" title="class in org.biojava.nbio.structure.io.cif">AbstractCifFileSupplier&lt;S&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Convert a BioJava object to a CifFile.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/AbstractCifFileSupplier.WrappedAtom.html" title="class in org.biojava.nbio.structure.io.cif">AbstractCifFileSupplier.WrappedAtom</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Wrapped atoms represent individual atoms enriched with model- and chain-level information.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/AbstractCompound.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The details of a Compound</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/AbstractCompoundSet.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompoundSet&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/AbstractCompoundTranslator.html" title="class in org.biojava.nbio.core.sequence.template">AbstractCompoundTranslator&lt;F,<wbr>T&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/features/AbstractFeature.html" title="class in org.biojava.nbio.core.sequence.features">AbstractFeature&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/template/AbstractLocation.html" title="class in org.biojava.nbio.core.sequence.location.template">AbstractLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Base abstraction of a location which encodes for the majority of important
 features about a location such as the start, end and strand</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/template/AbstractMatrixAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractMatrixAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements common code for an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/AbstractNucleotideCompoundSet.html" title="class in org.biojava.nbio.core.sequence.template">AbstractNucleotideCompoundSet&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/template/AbstractPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractPairwiseSequenceAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements common code for an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/template/AbstractProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">AbstractProfileProfileAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements common code for an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/reference/AbstractReference.html" title="class in org.biojava.nbio.core.sequence.reference">AbstractReference</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/template/AbstractScorer.html" title="class in org.biojava.nbio.alignment.template">AbstractScorer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements common code for algorithms which compute a score.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/AbstractScoresCache.html" title="class in org.biojava.nbio.structure.align.multiple">AbstractScoresCache</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Abstact implementation of the <a href="org/biojava/nbio/structure/align/multiple/ScoresCache.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>ScoresCache</code></a> with the shared code used
 in all objects with a variables cache.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/AbstractSequence.html" title="class in org.biojava.nbio.core.sequence.template">AbstractSequence&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The base class for DNA, RNA and Protein sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/template/AbstractSequence.AnnotationType.html" title="enum in org.biojava.nbio.core.sequence.template">AbstractSequence.AnnotationType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/AbstractStructureAlignment.html" title="class in org.biojava.nbio.structure.align">AbstractStructureAlignment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/AbstractTerm.html" title="class in org.biojava.nbio.ontology">AbstractTerm</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Abstract implementation of term

 This provides basic change-forwarding functionality from
                the annotation and ontology properties.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/AbstractUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">AbstractUserArgumentProcessor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Base class for a new structure alignment CLI.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/location/template/AccesionedLocation.html" title="interface in org.biojava.nbio.core.sequence.location.template">AccesionedLocation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A location which is bound to an AccessionID.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/Accessioned.html" title="interface in org.biojava.nbio.core.sequence.template">Accessioned</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Indicates an entity is accessioned</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/AccessionID.html" title="class in org.biojava.nbio.core.sequence">AccessionID</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used in Sequences as the unique identifier.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/model/AFP.html" title="class in org.biojava.nbio.structure.align.model">AFP</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to represent a FATCAT aligned fragment pair (AFP)</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html" title="class in org.biojava.nbio.structure.align.util">AFPAlignmentDisplay</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPCalculator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that performs calculations on AFPChains</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A bean to contain the core of a structure alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/AFPChainCoordManager.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">AFPChainCoordManager</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPChainer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to chain AFPs to an alignment</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/xml/AFPChainFlipper.html" title="class in org.biojava.nbio.structure.align.xml">AFPChainFlipper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/AFPChainScorer.html" title="class in org.biojava.nbio.structure.align.util">AFPChainScorer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/model/AfpChainWriter.html" title="class in org.biojava.nbio.structure.align.model">AfpChainWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to convert the data in an AfpChain object to various String outputs.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html" title="class in org.biojava.nbio.structure.align.xml">AFPChainXMLConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html" title="class in org.biojava.nbio.structure.align.xml">AFPChainXMLParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/AFPFromFasta.html" title="class in demo">AFPFromFasta</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demo displaying a structural alignment from a FASTA file using <a href="org/biojava/nbio/structure/io/FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io"><code>FastaAFPChainConverter</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPOptimizer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/AFPPostProcessor.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">AFPPostProcessor</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">does post processing after alignment chaingin</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/AFPTwister.html" title="class in org.biojava.nbio.structure.align">AFPTwister</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/stats/AgScore.html" title="class in org.biojava.nbio.survival.cox.stats">AgScore</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/helper/AligMatEl.html" title="class in org.biojava.nbio.structure.align.helper">AligMatEl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/pairwise/Alignable.html" title="interface in org.biojava.nbio.structure.align.pairwise">Alignable</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/AlignedPosition.html" title="class in org.biojava.nbio.structure.gui.util">AlignedPosition</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">AlignedSequence&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure for a <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/alignment/template/AlignedSequence.Step.html" title="enum in org.biojava.nbio.core.alignment.template">AlignedSequence.Step</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Defines an alignment step in order to pass alignment information from an Aligner to a constructor.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template">Aligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an algorithm which computes an alignment <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a> from a list of <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Static utility to construct alignment routines from a common library of methods.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Anchor.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Anchor</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Compounds in query and target sequences that must align</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Anchor.QueryIndexComparator.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Anchor.QueryIndexComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Cut.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Cut</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Defines a 'cut' row for divide-and-conquer alignment in which a new anchor is found.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Last.html" title="enum in org.biojava.nbio.alignment.routines">AlignerHelper.Last</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Define a traceback pointer for the three edit operations: substitution (match/replacement of a query compound
 with a target compound), deletion (removal of a query compound leaving a gap in the target sequence), and
 insertion (addition of a target compound opening a gap in the query sequence).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/routines/AlignerHelper.Subproblem.html" title="class in org.biojava.nbio.alignment.routines">AlignerHelper.Subproblem</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Alignment subproblem.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/AlignmentCalc.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentCalc</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that obtains two structures via DAS and aligns them
  This is done in a separate thread.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/AlignmentCalc.html" title="class in org.biojava.nbio.structure.gui.util">AlignmentCalc</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that obtains two structures via DAS and aligns them
  This is done in a separate thread.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/gui/AlignmentCalculationRunnable.html" title="interface in org.biojava.nbio.structure.align.gui">AlignmentCalculationRunnable</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/AlignmentGui.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentGui</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JFrame that allows to trigger a pairwise structure alignment,
 either from files in a directory,
 or after manual upload.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/AlignmentGuiDemo.html" title="class in demo">AlignmentGuiDemo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Get an instance of the two alignment GUIs.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/gui/events/AlignmentPositionListener.html" title="interface in org.biojava.nbio.structure.gui.events">AlignmentPositionListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/AlignmentProgressListener.html" title="class in org.biojava.nbio.structure.align.pairwise">AlignmentProgressListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to listen to progress of the structure alignment calculations</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/AlignmentResult.html" title="class in org.biojava.nbio.structure.align.pairwise">AlignmentResult</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to track the alignment results in a flat file</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/Alignments.html" title="class in org.biojava.nbio.alignment">Alignments</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Static utility to easily run alignment routines.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/alignment/Alignments.PairInProfileScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairInProfileScorerType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of implemented sequence pair in a profile scoring routines.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/alignment/Alignments.PairwiseSequenceAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceAlignerType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of implemented pairwise sequence alignment routines.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/alignment/Alignments.PairwiseSequenceScorerType.html" title="enum in org.biojava.nbio.alignment">Alignments.PairwiseSequenceScorerType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of implemented pairwise sequence scoring routines.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/alignment/Alignments.ProfileProfileAlignerType.html" title="enum in org.biojava.nbio.alignment">Alignments.ProfileProfileAlignerType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of implemented profile-profile alignment routines.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/alignment/Alignments.RefinerType.html" title="enum in org.biojava.nbio.alignment">Alignments.RefinerType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of implemented profile refinement routines.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/AlignmentTextPanel.html" title="class in org.biojava.nbio.structure.align.gui">AlignmentTextPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/AlignmentTools.html" title="class in org.biojava.nbio.structure.align.util">AlignmentTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Methods for analyzing and manipulating AFPChains and for
 other pairwise alignment utilities.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/AlignmentTools.IdentityMap.html" title="class in org.biojava.nbio.structure.align.util">AlignmentTools.IdentityMap&lt;K&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A <code>Map&lt;K,V&gt;</code> can be viewed as a function from K to V.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/AligNPE.html" title="class in org.biojava.nbio.structure.align.pairwise">AligNPE</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/helper/AlignUtils.html" title="class in org.biojava.nbio.structure.align.helper">AlignUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Low level helper methods for CE and FATCAT algorithms.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">AligPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JPanel that can display an AFPChain in a nice way and interact with Jmol.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/AligPanelMouseMotionListener.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">AligPanelMouseMotionListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/webstart/AligUIManager.html" title="class in org.biojava.nbio.structure.align.webstart">AligUIManager</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/AllChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">AllChemCompProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ChemComp provider that downloads and caches the components.cif file from the wwPDB site.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/ontology/AlreadyExistsException.html" title="class in org.biojava.nbio.ontology">AlreadyExistsException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">Thrown to indicate that a term or triple can't be added to an ontology
 because it is already present.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/AltAligComparator.html" title="class in org.biojava.nbio.structure.align.pairwise">AltAligComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A comparator to sort AlternativeAlignments based on their number of equivalent residues
 and RMSD.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/AlternativeAlignment.html" title="class in org.biojava.nbio.structure.align.pairwise">AlternativeAlignment</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a class which handles one possible (alternative) solution.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/AlternativeAlignmentFrame.html" title="class in org.biojava.nbio.structure.gui.util">AlternativeAlignmentFrame</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">a frame showing the alternative alignments, which are the result of a structure superimposition</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/AmbiguityDNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AmbiguityDNACompoundSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/AmbiguityDNARNAHybridCompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AmbiguityDNARNAHybridCompoundSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Ambiguity set for hybrid DNA/RNA sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/AmbiguityRNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AmbiguityRNACompoundSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/AminoAcid.html" title="interface in org.biojava.nbio.structure">AminoAcid</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">
 A <a href="org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a> that represents an AminoAcid.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/AminoAcidComposition.html" title="class in org.biojava.nbio.aaproperties.xml">AminoAcidComposition</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/AminoAcidCompositionTable.html" title="class in org.biojava.nbio.aaproperties.xml">AminoAcidCompositionTable</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to describe an Amino Acid.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompoundSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Set of proteinogenic amino acids.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/AminoAcidImpl.html" title="class in org.biojava.nbio.structure">AminoAcidImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">AminoAcid inherits most from Hetatom.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/AminoAcidProperties.html" title="class in org.biojava.nbio.aaproperties">AminoAcidProperties</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class provides the protein properties at the level of individual amino acids.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/routines/AnchoredPairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.routines">AnchoredPairwiseSequenceAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This algorithm uses a divide-and-conquer approach to find optimal pairwise global sequence alignments (from the
 first until the last <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>) with the restriction that any alignment produced
 will connect the query sequence to the target sequence at the <em>anchors</em>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/AngleOrderDetectorPlus.html" title="class in org.biojava.nbio.structure.symmetry.internal">AngleOrderDetectorPlus</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Guesses an order of rotational symmetry from the angle.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/utils/Annotatable.html" title="interface in org.biojava.nbio.ontology.utils">Annotatable</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Indicates that an object has an associated annotation.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/utils/Annotation.html" title="interface in org.biojava.nbio.ontology.utils">Annotation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">
 Arbitrary annotation associated with one or more objects.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/App.html" title="class in org.biojava.nbio.genome">App</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Hello world!</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/App.html" title="class in org.biojava.nbio.ontology">App</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Hello world!</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/loader/ArrayListProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">ArrayListProxySequenceReader&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/ArrayListSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">ArrayListSequenceReader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores a Sequence as a collection of compounds in an ArrayList</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/asa/AsaCalculator.html" title="class in org.biojava.nbio.structure.asa">AsaCalculator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class to calculate Accessible Surface Areas based on
 the rolling ball algorithm by Shrake and Rupley.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/ontology/utils/AssertionFailure.html" title="class in org.biojava.nbio.ontology.utils">AssertionFailure</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab5">
<div class="block">An unchecked exception representing an Assertion failure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/Astral.html" title="class in org.biojava.nbio.structure.scop">Astral</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides programmatic access to ASTRAL representative sets.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/scop/Astral.AstralSet.html" title="enum in org.biojava.nbio.structure.scop">Astral.AstralSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">An ASTRAL sequence-identity cutoff with an identifier such as:</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A simple interface for an Atom.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/AtomCache.html" title="class in org.biojava.nbio.structure.align.util">AtomCache</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class that provides easy access to Structure objects.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/AtomContact.html" title="class in org.biojava.nbio.structure.contact">AtomContact</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A pair of atoms that are in contact</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/AtomContactSet.html" title="class in org.biojava.nbio.structure.contact">AtomContactSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A set of atom-atom contacts to hold the results of intra and inter-chain contact calculations</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/AtomIdentifier.html" title="class in org.biojava.nbio.structure.contact">AtomIdentifier</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/AtomImpl.html" title="class in org.biojava.nbio.structure">AtomImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation of an Atom of a PDB file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/AtomInfo.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AtomInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class uniquely describes an atom</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/AtomInfoParser.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AtomInfoParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/AtomIterator.html" title="class in org.biojava.nbio.structure">AtomIterator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">an iterator over all atoms of a structure / group.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/AtomPositionMap.html" title="class in org.biojava.nbio.structure">AtomPositionMap</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A map from <a href="org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure"><code>ResidueNumbers</code></a> to ATOM record positions in a PDB file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/AtomPositionMap.GroupMatcher.html" title="interface in org.biojava.nbio.structure">AtomPositionMap.GroupMatcher</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Used as a Predicate to indicate whether a particular Atom should be mapped</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/AugmentedResidueRange.html" title="class in org.biojava.nbio.structure">AugmentedResidueRange</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by douglas on 1/23/15.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/Author.html" title="class in org.biojava.nbio.structure">Author</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Describes author attributes for author information in a PDB file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/gui/autosuggest/AutoSuggestProvider.html" title="interface in org.biojava.nbio.structure.align.gui.autosuggest">AutoSuggestProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A class that provides auto-completion suggestions for JAutoSuggest</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/axis/AxisAligner.html" title="class in org.biojava.nbio.structure.symmetry.axis">AxisAligner</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/basepairs/BasePairParameters.html" title="class in org.biojava.nbio.structure.basepairs">BasePairParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This module calculates the el Hassan-Calladine Base Pairing and Base-pair Step Parameters for any nucleic
 acid containing structure that has the information about the core base-pair rings.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/BasicSequence.html" title="class in org.biojava.nbio.core.sequence">BasicSequence&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Bare bones version of the Sequence object to be used sparingly.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/BcifFileReader.html" title="class in org.biojava.nbio.structure.io">BcifFileReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parse binary Cif files and provide capabilities to store them locally.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/BerkeleyScopInstallation.html" title="class in org.biojava.nbio.structure.scop">BerkeleyScopInstallation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">SCOPe:

 The Structural Classification of Proteins (extended) at Berkeley Lab and UC Berkeley
 (<a href="http://scop.berkeley.edu/">http://scop.berkeley.edu/</a>).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/BetaBridge.html" title="class in org.biojava.nbio.structure.secstruc">BetaBridge</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Container that represents a beta Bridge between two residues.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/BioAssemblyIdentifier.html" title="class in org.biojava.nbio.structure">BioAssemblyIdentifier</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/quaternary/BioAssemblyInfo.html" title="class in org.biojava.nbio.structure.quaternary">BioAssemblyInfo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Representation of a Biological Assembly annotation as provided by the PDB.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/quaternary/BioAssemblyTools.html" title="class in org.biojava.nbio.structure.quaternary">BioAssemblyTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/BioJavaAADemo.html" title="class in demo">BioJavaAADemo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by andreas on 8/9/14.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/BiojavaJmol.html" title="class in org.biojava.nbio.structure.gui">BiojavaJmol</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides a simple GUI for Jmol</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/BioJavaStructureAlignment.html" title="class in org.biojava.nbio.structure.align">BioJavaStructureAlignment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Wrapper for the BioJava Structure Alignment Implementation</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/quaternary/BiologicalAssemblyBuilder.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyBuilder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Reconstructs the quaternary structure of a protein from an asymmetric unit</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/quaternary/BiologicalAssemblyTransformation.html" title="class in org.biojava.nbio.structure.quaternary">BiologicalAssemblyTransformation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The transformation needed for generation of biological assemblies
 from the contents of a PDB/mmCIF file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/BitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An implementation of the popular bit encodings.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/BitSequenceReader.BitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">BitSequenceReader.BitArrayWorker&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The logic of working with a bit has been separated out into this class
 to help developers create the bit data structures without having to
 put the code into an intermediate format and to also use the format
 without the need to copy this code.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/ws/alignment/qblast/BlastAlignmentParameterEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastAlignmentParameterEnum</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Alignment request parameters accepted by QBlast service.<br/>
 Not all are mandatory.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/utils/BlastClustReader.html" title="class in org.biojava.nbio.structure.symmetry.utils">BlastClustReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/blast/BlastHitBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastHitBuilder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Designed by Paolo Pavan.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/homology/BlastHomologyHits.html" title="class in org.biojava.nbio.genome.homology">BlastHomologyHits</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/blast/BlastHspBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastHspBuilder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Designed by Paolo Pavan.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/alignment/qblast/BlastJob.html" title="class in org.biojava.nbio.ws.alignment.qblast">BlastJob</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Information about QBlast search job</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/ws/alignment/qblast/BlastMatrixEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastMatrixEnum</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enum representing matrices supported by QBlast</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/ws/alignment/qblast/BlastOutputAlignmentFormatEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastOutputAlignmentFormatEnum</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enum representing available output alignment types.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/ws/alignment/qblast/BlastOutputFormatEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastOutputFormatEnum</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enum representing available output formats.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/ws/alignment/qblast/BlastOutputParameterEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastOutputParameterEnum</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Output parameters accepted by QBlast service.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/ws/alignment/qblast/BlastProgramEnum.html" title="enum in org.biojava.nbio.ws.alignment.qblast">BlastProgramEnum</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enum representing available blast programs.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/blast/BlastResult.html" title="class in org.biojava.nbio.core.search.io.blast">BlastResult</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class models a Blast/Blast plus result.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/blast/BlastResultBuilder.html" title="class in org.biojava.nbio.core.search.io.blast">BlastResultBuilder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Designed by Paolo Pavan.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/blast/BlastTabularParser.html" title="class in org.biojava.nbio.core.search.io.blast">BlastTabularParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Designed by Paolo Pavan.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/blast/BlastXMLParser.html" title="class in org.biojava.nbio.core.search.io.blast">BlastXMLParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Re-designed by Paolo Pavan on the footprint of:
 org.biojava.nbio.genome.query.BlastXMLQuery by Scooter Willis 

 You may want to find my contacts on Github and LinkedIn for code info
 or discuss major changes.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/query/BlastXMLQuery.html" title="class in org.biojava.nbio.genome.query">BlastXMLQuery</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple">Block</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A Block is a Data Structure that stores aligned positions of a
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> with the condition that residues are in a
 sequential order.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/BlockImpl.html" title="class in org.biojava.nbio.structure.align.multiple">BlockImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">General implementation of a <a href="org/biojava/nbio/structure/align/multiple/Block.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>Block</code></a> that supports any type of
 sequential alignment with gaps.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/multiple/BlockSet.html" title="interface in org.biojava.nbio.structure.align.multiple">BlockSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A BlockSet is a Data Structure to store a flexible alignment part of a
 multiple alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/BlockSetImpl.html" title="class in org.biojava.nbio.structure.align.multiple">BlockSetImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A general implementation of a BlockSet to store a flexible part of a multiple
 alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/Bond.html" title="interface in org.biojava.nbio.structure">Bond</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A simple bond -- it stores information about two atoms as well as information
 about its bond order.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/BondImpl.html" title="class in org.biojava.nbio.structure">BondImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A simple bond -- it stores information about two atoms as well as information
 about its bond order.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/BondMaker.html" title="class in org.biojava.nbio.structure.io">BondMaker</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Adds polymer bonds for peptides and nucleotides based on distance cutoffs and
 intra-group (residue) bonds based on data from the Chemical Component Dictionary
 to the Structure object.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/BondType.html" title="enum in org.biojava.nbio.structure">BondType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Work in progress - NOT final!</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/BoundingBox.html" title="class in org.biojava.nbio.structure.contact">BoundingBox</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A bounding box for short cutting some geometrical calculations.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/xtal/BravaisLattice.html" title="enum in org.biojava.nbio.structure.xtal">BravaisLattice</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">An enum to represent the 7 Bravais lattices</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/secstruc/BridgeType.html" title="enum in org.biojava.nbio.structure.secstruc">BridgeType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">A bridge is formed by two non-overlapping stretches of three residues each
 (i-1,i,i+1) and (j-1,j,j+1), where i&lt;j.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/webstart/BrowserOpener.html" title="class in org.biojava.nbio.structure.align.webstart">BrowserOpener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/BufferedReaderBytesRead.html" title="class in org.biojava.nbio.core.sequence.io">BufferedReaderBytesRead</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Need to keep track of actual bytes read and take advantage of buffered reader
 performance.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/C2RotationSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">C2RotationSolver</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/CAConverter.html" title="class in org.biojava.nbio.structure.io">CAConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Converts full atom representations to Calpha only ones.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/Calc.html" title="class in org.biojava.nbio.structure">Calc</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility operations on Atoms, AminoAcids, Matrices, Point3d, etc.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/CalcPoint.html" title="class in org.biojava.nbio.structure.geometry">CalcPoint</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility operations on Point3d.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceAligner.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a concurrency wrapper for a <a href="org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceAligner</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/template/CallablePairwiseSequenceScorer.html" title="class in org.biojava.nbio.alignment.template">CallablePairwiseSequenceScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a concurrency wrapper for a <a href="org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template"><code>PairwiseSequenceScorer</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/template/CallableProfileProfileAligner.html" title="class in org.biojava.nbio.alignment.template">CallableProfileProfileAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a concurrency wrapper for a <a href="org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template"><code>ProfileProfileAligner</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/CallableStructureAlignment.html" title="class in org.biojava.nbio.structure.align">CallableStructureAlignment</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Simple Callable Class that calculates a pairwise alignment in a different
 thread, so that multiple pairwise alignments can be run in parallel
 (examples: all-to-all alignments, DB search alignments).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/quaternary/CartesianProduct.html" title="class in org.biojava.nbio.structure.quaternary">CartesianProduct&lt;T&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A cartesian product between two lists A and B is the set of all ordered pairs
 of the elements of both sets.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/CaseFreeAminoAcidCompoundSet.html" title="class in org.biojava.nbio.aaproperties.xml">CaseFreeAminoAcidCompoundSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Set of proteinogenic amino acids.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/transcription/CaseInsensitiveCompound.html" title="class in org.biojava.nbio.core.sequence.transcription">CaseInsensitiveCompound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Attempts to wrap compounds so it is possible to view them
 in a case insensitive manner</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/CasePreservingProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">CasePreservingProteinSequenceCreator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sequence creator which preserves the case of its input string in
 the user collection of the returned ProteinSequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/cath/CathCategory.html" title="enum in org.biojava.nbio.structure.cath">CathCategory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">The categories found within CATH.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath">CathDatabase</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">General API for interacting with CATH.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cath/CathDomain.html" title="class in org.biojava.nbio.structure.cath">CathDomain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class which represents a single CATH domain.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cath/CathFactory.html" title="class in org.biojava.nbio.structure.cath">CathFactory</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Controls global <a href="org/biojava/nbio/structure/cath/CathDatabase.html" title="interface in org.biojava.nbio.structure.cath"><code>CathDatabases</code></a> being used.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cath/CathFragment.html" title="class in org.biojava.nbio.structure.cath">CathFragment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cath/CathInstallation.html" title="class in org.biojava.nbio.structure.cath">CathInstallation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cath/CathNode.html" title="class in org.biojava.nbio.structure.cath">CathNode</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents a node in the CATH hierarchy.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cath/CathSegment.html" title="class in org.biojava.nbio.structure.cath">CathSegment</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/CDSComparator.html" title="class in org.biojava.nbio.core.sequence">CDSComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/CDSSequence.html" title="class in org.biojava.nbio.core.sequence">CDSSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents a exon or coding sequence in a gene.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is based on the original Combinatorial Extension (CE) source code from 2003 or 2004 (CE version 2.3),
 as has been originally developed by I.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html" title="class in org.biojava.nbio.structure.align.ce">CeCalculatorEnhanced</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is based on the original Combinatorial Extension (CE) source code from 2003 or 2004 (CE version 2.3),
 as has been originally developed by I.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeCPMain.html" title="class in org.biojava.nbio.structure.align.ce">CeCPMain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A wrapper for <a href="org/biojava/nbio/structure/align/ce/CeMain.html" title="class in org.biojava.nbio.structure.align.ce"><code>CeMain</code></a> which sets default parameters to be appropriate for finding
 circular permutations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeCPMain.CPRange.html" title="class in org.biojava.nbio.structure.align.ce">CeCPMain.CPRange</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tiny wrapper for the disallowed regions of an alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides parameters to <a href="org/biojava/nbio/structure/align/ce/CeCPMain.html" title="class in org.biojava.nbio.structure.align.ce"><code>CeCPMain</code></a></div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/align/ce/CECPParameters.DuplicationHint.html" title="enum in org.biojava.nbio.structure.align.ce">CECPParameters.DuplicationHint</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeCPUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">CeCPUserArgumentProcessor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeMain.html" title="class in org.biojava.nbio.structure.align.ce">CeMain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The main class of the Java implementation of the Combinatorial Extension Algorithm (CE),
 as has been originally developed by I.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/figure/CensorStatus.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">CensorStatus</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/survival/kaplanmeier/figure/CensorStatusSelect.html" title="interface in org.biojava.nbio.survival.kaplanmeier.figure">CensorStatusSelect</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeParameters.html" title="class in org.biojava.nbio.structure.align.ce">CeParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains the parameters that can be sent to CE</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/align/ce/CeParameters.ScoringStrategy.html" title="enum in org.biojava.nbio.structure.align.ce">CeParameters.ScoringStrategy</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeSideChainMain.html" title="class in org.biojava.nbio.structure.align.ce">CeSideChainMain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeSideChainUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">CeSideChainUserArgumentProcessor</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/CeSymm.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymm</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Identify the symmetries in a structure by running an alignment of the
 structure against itself disabling the diagonal of the identity alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/CeSymmIterative.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmIterative</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Iterative version of CeSymm that aims at identifying all symmetry axis of a
 structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.html" title="class in org.biojava.nbio.structure.symmetry.internal">CESymmParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides parameters to <a href="org/biojava/nbio/structure/symmetry/internal/CeSymm.html" title="class in org.biojava.nbio.structure.symmetry.internal"><code>CeSymm</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.OrderDetectorMethod.html" title="enum in org.biojava.nbio.structure.symmetry.internal">CESymmParameters.OrderDetectorMethod</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.RefineMethod.html" title="enum in org.biojava.nbio.structure.symmetry.internal">CESymmParameters.RefineMethod</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/symmetry/internal/CESymmParameters.SymmetryType.html" title="enum in org.biojava.nbio.structure.symmetry.internal">CESymmParameters.SymmetryType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">The internal symmetry detection can be divided into two types: CLOSE:
 includes the circular and dihedral symmetries, and OPEN: includes the
 helical and protein repeats symmetries.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This Class stores all the relevant information of an internal symmetry result
 obtained with CeSymm.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.ce">CeUserArgumentProcessor</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Process the arguments from command line</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/CeUserArgumentProcessor.CeStartupParams.html" title="class in org.biojava.nbio.structure.align.ce">CeUserArgumentProcessor.CeStartupParams</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure">Chain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">
 Defines the interface for a Chain.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ChainImpl.html" title="class in org.biojava.nbio.structure">ChainImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Chain in a PDB file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/misc/ChainSignature.html" title="class in org.biojava.nbio.structure.symmetry.misc">ChainSignature</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/survival/data/ChangeValue.html" title="interface in org.biojava.nbio.survival.data">ChangeValue</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/ChargeAdder.html" title="class in org.biojava.nbio.structure.io">ChargeAdder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to add appropriate charge information to a structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ChemComp.html" title="class in org.biojava.nbio.structure.chem">ChemComp</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Properties of a chemical component.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ChemCompAtom.html" title="class in org.biojava.nbio.structure.chem">ChemCompAtom</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Properties of an atom of a chemical component.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ChemCompBond.html" title="class in org.biojava.nbio.structure.chem">ChemCompBond</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Properties of a bond in a chemical component.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/cif/ChemCompConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">ChemCompConsumer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Create the <a href="org/biojava/nbio/structure/chem/ChemicalComponentDictionary.html" title="class in org.biojava.nbio.structure.chem"><code>ChemicalComponentDictionary</code></a> from CIF data.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/ChemCompConsumerImpl.html" title="class in org.biojava.nbio.structure.io.cif">ChemCompConsumerImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Consumes a CCD file to create the <a href="org/biojava/nbio/structure/chem/ChemicalComponentDictionary.html" title="class in org.biojava.nbio.structure.chem"><code>ChemicalComponentDictionary</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/ChemCompConverter.html" title="class in org.biojava.nbio.structure.io.cif">ChemCompConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Convert CifFiles to chem comps.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ChemCompDescriptor.html" title="class in org.biojava.nbio.structure.chem">ChemCompDescriptor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Properties of the chemical component descriptor.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/ChemCompDistribution.html" title="class in demo">ChemCompDistribution</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ChemCompGroupFactory.html" title="class in org.biojava.nbio.structure.chem">ChemCompGroupFactory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/chem/ChemCompProvider.html" title="interface in org.biojava.nbio.structure.chem">ChemCompProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface that is implemented by all classes that can provide <a href="org/biojava/nbio/structure/chem/ChemComp.html" title="class in org.biojava.nbio.structure.chem"><code>ChemComp</code></a> definitions.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ChemCompTools.html" title="class in org.biojava.nbio.structure.chem">ChemCompTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ChemicalComponentDictionary.html" title="class in org.biojava.nbio.structure.chem">ChemicalComponentDictionary</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A representation of the Chemical Component Dictionary.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/stats/ChiSq.html" title="class in org.biojava.nbio.survival.cox.stats">ChiSq</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/stats/Cholesky2.html" title="class in org.biojava.nbio.survival.cox.stats">Cholesky2</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/jama/CholeskyDecomposition.html" title="class in org.biojava.nbio.structure.jama">CholeskyDecomposition</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Cholesky Decomposition.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/ChooseDirAction.html" title="class in org.biojava.nbio.structure.align.gui">ChooseDirAction</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Ask the user to provide a directory containting PDB files.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/util/ChromosomeMappingTools.html" title="class in org.biojava.nbio.genome.util">ChromosomeMappingTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that can map chromosomal positions to mRNA (coding sequence) positions.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/ChromosomeSequence.html" title="class in org.biojava.nbio.core.sequence">ChromosomeSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ChromosomeSequence is a DNASequence but keeps track of geneSequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/genename/ChromPos.html" title="class in org.biojava.nbio.genome.parsers.genename">ChromPos</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by ap3 on 27/10/2014.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/stats/Chsolve2.html" title="class in org.biojava.nbio.survival.cox.stats">Chsolve2</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/cif/CifBean.html" title="interface in org.biojava.nbio.structure.io.cif">CifBean</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Flag for BioJava beans that resemble categories defined by the mmCIF schema.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/CifChainSupplierImpl.html" title="class in org.biojava.nbio.structure.io.cif">CifChainSupplierImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Convert a chain to a <code>CifFile</code>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/cif/CifFileConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">CifFileConsumer&lt;S&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a rather generic interface which allows to populate some data structure with data parsed from a CIF file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/CifFileReader.html" title="class in org.biojava.nbio.structure.io">CifFileReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parse text Cif files and provide capabilities to store them locally.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/cif/CifFileSupplier.html" title="interface in org.biojava.nbio.structure.io.cif">CifFileSupplier&lt;S&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Create a CifFile instance for a given container of structure data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/cif/CifStructureConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">CifStructureConsumer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines the categories to consume during CIF parsing.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/CifStructureConsumerImpl.html" title="class in org.biojava.nbio.structure.io.cif">CifStructureConsumerImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An implementation of a CifFileConsumer for BioJava.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/CifStructureConverter.html" title="class in org.biojava.nbio.structure.io.cif">CifStructureConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Convert BioJava structures to CifFiles and vice versa.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/CifStructureSupplierImpl.html" title="class in org.biojava.nbio.structure.io.cif">CifStructureSupplierImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Convert a structure to a CifFile.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/util/ClasspathResource.html" title="class in org.biojava.nbio.core.sequence.io.util">ClasspathResource</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This object represents a classpath resource on the local system.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/ClinicalMetaDataOutcome.html" title="class in org.biojava.nbio.survival.kaplanmeier.metadata">ClinicalMetaDataOutcome</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/CliTools.html" title="class in org.biojava.nbio.structure.align.util">CliTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utilities for autoconfiguring javabeans based on command line arguments.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ClusterAltAligs.html" title="class in org.biojava.nbio.structure.align">ClusterAltAligs</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that clusters alternative alignments according to their
 similarity.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/ClusterDomains.html" title="class in org.biojava.nbio.structure.domain.pdp">ClusterDomains</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/CodonCompound.html" title="class in org.biojava.nbio.core.sequence.compound">CodonCompound</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Define a codon</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/CollectionTools.html" title="class in org.biojava.nbio.structure.align.util">CollectionTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utilities for working with collections.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/ColorConverter.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">ColorConverter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/gui/util/color/ColorInterpolator.html" title="interface in org.biojava.nbio.structure.gui.util.color">ColorInterpolator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/ColorUtils.html" title="class in org.biojava.nbio.structure.gui.util.color">ColorUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/CommandPrompt.html" title="class in org.biojava.nbio.aaproperties">CommandPrompt</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/aaproperties/CommandPrompt.PropertyName.html" title="enum in org.biojava.nbio.aaproperties">CommandPrompt.PropertyName</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/data/CompactCharSequence.html" title="class in org.biojava.nbio.survival.data">CompactCharSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">http://www.javamex.com/tutorials/memory/ascii_charsequence.shtml</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/phylo/Comparison.html" title="class in org.biojava.nbio.phylo">Comparison</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class provides static methods for the calculation of the percentage of
 identity between two aligned sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/ComplementCompound.html" title="interface in org.biojava.nbio.core.sequence.template">ComplementCompound</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/views/ComplementSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ComplementSequenceView&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">For a given sequence this class will create a view over the top of it
 and for every request the code will return the complement of the underlying
 base e.g. base A will become base T</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/Component.html" title="class in org.biojava.nbio.protmod">Component</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">contains information about a certain Component.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/io/ComponentXMLConverter.html" title="class in org.biojava.nbio.protmod.io">ComponentXMLConverter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template">Compound</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/core/exceptions/CompoundNotFoundException.html" title="class in org.biojava.nbio.core.exceptions">CompoundNotFoundException</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab5">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/CompoundSet.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundSet&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/CompoundTranslator.html" title="interface in org.biojava.nbio.core.sequence.template">CompoundTranslator&lt;F,<wbr>T&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/ConcurrencyTools.html" title="class in org.biojava.nbio.core.util">ConcurrencyTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Static utility to easily share a thread pool for concurrent/parallel/lazy execution.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/ConfigPDBInstallPanel.html" title="class in org.biojava.nbio.structure.align.gui">ConfigPDBInstallPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce">ConfigStrucAligParams</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/structure/align/util/ConfigurationException.html" title="class in org.biojava.nbio.structure.align.util">ConfigurationException</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab5">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/webstart/ConfigXMLHandler.html" title="class in org.biojava.nbio.structure.align.webstart">ConfigXMLHandler</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">XML content handler for serialisation of RegistryConfiguration class</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/Constraints.html" title="class in org.biojava.nbio.aaproperties">Constraints</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is used to support the implementation of properties stated in IPeptideProperties.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/Contact.html" title="class in org.biojava.nbio.structure.contact">Contact</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A simple class to store contacts in the form of pairs of indices and a distance associated to them.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/gui/util/color/ContinuousColorMapper.html" title="interface in org.biojava.nbio.structure.gui.util.color">ContinuousColorMapper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Provides a mapping between real numbers and Colors.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/ContinuousColorMapperTransform.html" title="class in org.biojava.nbio.structure.gui.util.color">ContinuousColorMapperTransform</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Maps colors by performing a transform of the input data and then passing
 the transformed value to a ContinuousColorMapper for rendering.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Charge.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Charge</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Hydrophobicity.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Hydrophobicity</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2NormalizedVanDerWaalsVolume.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2NormalizedVanDerWaalsVolume</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Polarity.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Polarity</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2Polarizability.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2Polarizability</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2SecondaryStructure.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2SecondaryStructure</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convert2SolventAccessibility.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convert2SolventAccessibility</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/convertor/Convertor.html" title="class in org.biojava.nbio.aaproperties.profeat.convertor">Convertor</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/CookBook.html" title="class in demo">CookBook</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/CookbookMSA.html" title="class in demo">CookbookMSA</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/CoordManager.html" title="class in org.biojava.nbio.structure.gui.util">CoordManager</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">a class that manages the conversion of sequence coordinate system to
 JPanel drawing coordinates</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/util/CoreSuperimposer.html" title="class in org.biojava.nbio.structure.align.multiple.util">CoreSuperimposer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Superimposes the core aligned residues of every structure in a
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> onto a reference structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxCC.html" title="class in org.biojava.nbio.survival.cox">CoxCC</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxCoefficient.html" title="class in org.biojava.nbio.survival.cox">CoxCoefficient</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/survival/cox/comparators/CoxComparatorInterface.html" title="interface in org.biojava.nbio.survival.cox.comparators">CoxComparatorInterface</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxHelper.html" title="class in org.biojava.nbio.survival.cox">CoxHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The CoxHelper class is provided to start with a tab delimited file in a similar process in R and return the results as a CoxInfo class.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxInfo.html" title="class in org.biojava.nbio.survival.cox">CoxInfo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Holds the results of a cox analysis where calling dump(), toString() will give an output similar to R</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxMart.html" title="class in org.biojava.nbio.survival.cox">CoxMart</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/survival/cox/CoxMethod.html" title="enum in org.biojava.nbio.survival.cox">CoxMethod</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxR.html" title="class in org.biojava.nbio.survival.cox">CoxR</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is a port of the R survival code used for doing Cox Regression.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxScore.html" title="class in org.biojava.nbio.survival.cox">CoxScore</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/CoxVariables.html" title="class in org.biojava.nbio.survival.cox">CoxVariables</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/comparators/CoxVariablesOverallModelFitComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">CoxVariablesOverallModelFitComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/comparators/CoxVariablesVariableComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">CoxVariablesVariableComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/CRC64Checksum.html" title="class in org.biojava.nbio.core.util">CRC64Checksum</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility class that calculates a CRC64 checksum on a stream of bytes.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/CrystalBuilder.html" title="class in org.biojava.nbio.structure.xtal">CrystalBuilder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class containing methods to find interfaces in a given crystallographic Structure by
 reconstructing the crystal lattice through application of symmetry operators</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/CrystalCell.html" title="class in org.biojava.nbio.structure.xtal">CrystalCell</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A crystal cell's parameters.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/CrystalTransform.html" title="class in org.biojava.nbio.structure.xtal">CrystalTransform</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Representation of a transformation in a crystal:
 - a transformation id (each of the transformations in a space group, 0 to m)
 - a crystal translation
 The transformation matrix in crystal basis is stored, representing the basic
 transformation together with the crystal translation.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/Cut.html" title="class in org.biojava.nbio.structure.domain.pdp">Cut</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/CutDomain.html" title="class in org.biojava.nbio.structure.domain.pdp">CutDomain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/CutSites.html" title="class in org.biojava.nbio.structure.domain.pdp">CutSites</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/CutValues.html" title="class in org.biojava.nbio.structure.domain.pdp">CutValues</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/cytoband/Cytoband.html" title="class in org.biojava.nbio.genome.parsers.cytoband">Cytoband</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/cytoband/CytobandParser.html" title="class in org.biojava.nbio.genome.parsers.cytoband">CytobandParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parses the cytoband (karyotype) file from UCSC.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/DatabasePDBRevRecord.html" title="class in org.biojava.nbio.structure">DatabasePDBRevRecord</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents revision records for use by <a href="org/biojava/nbio/structure/PDBHeader.html" title="class in org.biojava.nbio.structure"><code>PDBHeader</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/features/DatabaseReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.features">DatabaseReferenceInterface</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/phosphosite/Dataset.html" title="class in org.biojava.nbio.phosphosite">Dataset</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Phosphosite is available under the PhosphoSitePlus® is licensed under Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License and is freely available for non-commercial purposes from

 http://www.phosphosite.org/staticDownloads.do

 Please acknowledge PhosphoSitePlus®, www.phosphosite.org" at appropriate locations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/DataSource.html" title="enum in org.biojava.nbio.core.sequence">DataSource</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">
 GenBank                           gi|gi-number|gb|accession|locus
 ENA Data Library                  gi|gi-number|emb|accession|locus
 DDBJ, DNA Database of Japan       gi|gi-number|dbj|accession|locus
 NBRF PIR                          pir||entry
 Protein Research Foundation       prf||name
 SWISS-PROT UNIPROT                sp|accession|name
 Brookhaven Protein Data Bank (1)  pdb|entry|chain
 Brookhaven Protein Data Bank (2)  entry:chain|PDBID|CHAIN|SEQUENCE
 Patents                           pat|country|number
 GenInfo Backbone Id               bbs|number
 General database identifier       gnl|database|identifier
 NCBI Reference Sequence           ref|accession|locus
 Local Sequence identifier         lcl|identifier</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/DBRef.html" title="class in org.biojava.nbio.structure">DBRef</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to represent database cross references.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features">DBReferenceInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/matrices/DefaultAAIndexProvider.html" title="class in org.biojava.nbio.core.alignment.matrices">DefaultAAIndexProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The default provider for AAINDEX loads substitution matrices from the AAINDEX file in the resources directory</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/autosuggest/DefaultAutoSuggestProvider.html" title="class in org.biojava.nbio.structure.align.gui.autosuggest">DefaultAutoSuggestProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/DefaultMatrixMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">DefaultMatrixMapper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Color Mapper which mimics the default coloring of JMatrixPanel pixels.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/DefaultOps.html" title="class in org.biojava.nbio.ontology">DefaultOps</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Default implementation of OntologyOps.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoAlignmentFromFasta.html" title="class in demo">DemoAlignmentFromFasta</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demo of how to use the <a href="org/biojava/nbio/structure/io/FastaStructureParser.html" title="class in org.biojava.nbio.structure.io"><code>FastaStructureParser</code></a> class to read protein
 structures from a FASTA file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoAlignProteins.html" title="class in demo">DemoAlignProteins</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoAsa.html" title="class in demo">DemoAsa</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoAtomCache.html" title="class in demo">DemoAtomCache</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Example of how to load PDB files using the AtomCache class.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoBerkeleyScop.html" title="class in demo">DemoBerkeleyScop</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A demo for how to use the Berkeley version of SCOP instead of the default UK-SCOP</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoBioAssemblies.html" title="class in demo">DemoBioAssemblies</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoCATH.html" title="class in demo">DemoCATH</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An example for how to access CATH data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoCE.html" title="class in demo">DemoCE</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Example of how to run a structure alignment using the CE algorithm.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoCeSymm.html" title="class in demo">DemoCeSymm</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Quick demo of how to call CE-Symm programmatically.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoChangeChemCompProvider.html" title="class in demo">DemoChangeChemCompProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This demo shows how to use an alternative ChemCompProvider.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoCommandLineStartup.html" title="class in demo">DemoCommandLineStartup</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoContacts.html" title="class in demo">DemoContacts</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoCrystalInterfaces.html" title="class in demo">DemoCrystalInterfaces</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoDistanceTree.html" title="class in demo">DemoDistanceTree</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This demo contains the CookBook example to create a phylogenetic tree from a
 multiple sequence alignment (MSA).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoDomainsplit.html" title="class in demo">DemoDomainsplit</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoFATCAT.html" title="class in demo">DemoFATCAT</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoLoadSecStruc.html" title="class in demo">DemoLoadSecStruc</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demonstration of how to load a Structure with the SS information, either from
 the PDB file annotation (Author's assignment) or from the DSSP file in the
 PDB servers (DSSP assignment).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoLoadStructure.html" title="class in demo">DemoLoadStructure</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Example for how to load protein structures (from PDB files).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoLoadSubstMax.html" title="class in demo">DemoLoadSubstMax</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoMMCIFReader.html" title="class in demo">DemoMMCIFReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An example of how to read MMcif files</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoMmcifToPdbConverter.html" title="class in demo">DemoMmcifToPdbConverter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An example of how to convert mmCIF file to PDB file</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoMmtfWriter.html" title="class in demo">DemoMmtfWriter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoMultipleMC.html" title="class in demo">DemoMultipleMC</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demo for running the MultipleMC Algorithm on a protein family.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoNWALignment.html" title="class in demo">DemoNWALignment</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoQsAlign.html" title="class in demo">DemoQsAlign</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demo on how to use programatically <a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary"><code>QsAlign</code></a> for the alignment of
 quaternary structures.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoQuatSymmetryJmol.html" title="class in demo">DemoQuatSymmetryJmol</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This demo shows how to display the <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryResults</code></a> of a
 structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoRotationAxis.html" title="class in demo">DemoRotationAxis</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A demo for how to use <a href="org/biojava/nbio/structure/align/util/RotationAxis.html" title="class in org.biojava.nbio.structure.align.util"><code>RotationAxis</code></a> to display the rotation for an
 alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoSCOP.html" title="class in demo">DemoSCOP</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class demonstrating the use of the SCOP parsing tools</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoSecStrucCalc.html" title="class in demo">DemoSecStrucCalc</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demonstration on how to use the Secondary Structure Prediction (DSSP)
 implementation in BioJava and obtain different SS representations and
 outputs.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoShowBiolAssembly.html" title="class in demo">DemoShowBiolAssembly</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoShowCATHDomain.html" title="class in demo">DemoShowCATHDomain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoShowLargeAssembly.html" title="class in demo">DemoShowLargeAssembly</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoSixFrameTranslation.html" title="class in demo">DemoSixFrameTranslation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by andreas on 8/10/15.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoStructureFromFasta.html" title="class in demo">DemoStructureFromFasta</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demo of how to use the <a href="org/biojava/nbio/structure/io/FastaStructureParser.html" title="class in org.biojava.nbio.structure.io"><code>FastaStructureParser</code></a> class to read protein
 structures from a FASTA file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoSW3DAligner.html" title="class in demo">DemoSW3DAligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/DemoSymmetry.html" title="class in demo">DemoSymmetry</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A demo on how to use the quaternary symmetry detection algorithms.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/DiscreteQuantizerInterface.html" title="interface in org.biojava.nbio.survival.kaplanmeier.metadata">DiscreteQuantizerInterface</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/DisplayAFP.html" title="class in org.biojava.nbio.structure.align.gui">DisplayAFP</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class for visualistion of structure alignments</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/DistanceBox.html" title="class in org.biojava.nbio.structure.symmetry.geometry">DistanceBox&lt;T&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/phylo/DistanceMatrixCalculator.html" title="class in org.biojava.nbio.phylo">DistanceMatrixCalculator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The DistanceMatrixCalculator methods generate a <code>DistanceMatrix</code> from a
 <a href="org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence"><code>MultipleSequenceAlignment</code></a> or other indirect distance infomation (RMSD).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/phylo/DistanceTreeEvaluator.html" title="class in org.biojava.nbio.phylo">DistanceTreeEvaluator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Check the accuracy of a Distance Tree by least squares error (LSE) of the
 Tree branch lengths and the original Distance Matrix.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/DNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">DNACompoundSet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/DNASequence.html" title="class in org.biojava.nbio.core.sequence">DNASequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is class should model the attributes associated with a DNA sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/DNASequence.DNAType.html" title="enum in org.biojava.nbio.core.sequence">DNASequence.DNAType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">The type of DNA sequence</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/DNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">DNASequenceCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A helper class that allows different ways to read a string and create a DNA sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/transcription/DNAToRNATranslator.html" title="class in org.biojava.nbio.core.sequence.transcription">DNAToRNATranslator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Performs the first stage of transcription by going from DNA to RNA.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/Domain.html" title="class in org.biojava.nbio.structure.domain.pdp">Domain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/domain/DomainProvider.html" title="interface in org.biojava.nbio.structure.domain">DomainProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Decomposes a structure from the PDB into representative domains</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/DotPlotPanel.html" title="class in org.biojava.nbio.structure.align.gui">DotPlotPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Displays the dot plot trace for an alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/DownloadChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">DownloadChemCompProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This provider of chemical components can download and cache chemical component definition files from the RCSB PDB web
 site.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/DSSPParser.html" title="class in org.biojava.nbio.structure.secstruc">DSSPParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class to parse a DSSP file (output of the DSSP program),
 that contains the secondary structure assignment of a structure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/ecod/EcodDatabase.html" title="interface in org.biojava.nbio.structure.ecod">EcodDatabase</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">General API for interacting with ECOD.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ecod/EcodDomain.html" title="class in org.biojava.nbio.structure.ecod">EcodDomain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An EcodDomain contains all the information of the ECOD database: id,
 classification groups (from higher to lower in the tree: X,H,T,F), PDB code,
 chain, residue ranges and status (manual or automatic classification).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ecod/EcodFactory.html" title="class in org.biojava.nbio.structure.ecod">EcodFactory</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Controls global <a href="org/biojava/nbio/structure/ecod/EcodDatabase.html" title="interface in org.biojava.nbio.structure.ecod"><code>EcodDatabases</code></a> being used.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ecod/EcodInstallation.html" title="class in org.biojava.nbio.structure.ecod">EcodInstallation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides access to the Evolutionary Classification of Protein Domains (ECOD).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ecod/EcodInstallation.EcodParser.html" title="class in org.biojava.nbio.structure.ecod">EcodInstallation.EcodParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/edits/Edit.html" title="interface in org.biojava.nbio.core.sequence.edits">Edit&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for carrying out edit operations on a Sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/edits/Edit.AbstractEdit.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.AbstractEdit&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Abstract class which defines all edit operations as a call to discover
 what 5' and 3' ends of an editing Sequence should be joined together
 with a target Sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/edits/Edit.Delete.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Delete&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation which allows for the deletion of bases from a Sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/edits/Edit.Insert.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Insert&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Edit implementation which allows us to insert a base at any position
 in a Sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/edits/Edit.Substitute.html" title="class in org.biojava.nbio.core.sequence.edits">Edit.Substitute&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Allows for the substitution of bases into an existing Sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/jama/EigenvalueDecomposition.html" title="class in org.biojava.nbio.structure.jama">EigenvalueDecomposition</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Eigenvalues and eigenvectors of a real matrix.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/Element.html" title="class in org.biojava.nbio.aaproperties.xml">Element</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">One way to model the elements</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/Element.html" title="enum in org.biojava.nbio.structure">Element</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Element is an enumeration of the elements of the periodic table.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/ElementTable.html" title="class in org.biojava.nbio.aaproperties.xml">ElementTable</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/ElementType.html" title="enum in org.biojava.nbio.structure">ElementType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">ElementType is an enumeration of the types of elements found in the periodic table.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/embl/EmblId.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblId</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class contains the processed data of embl file
 Primary accession number
 Sequence version number
 Topology: 'circular' or 'linear'
 Molecule type
 Data class
 Taxonomic division
 Sequence length</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/embl/EmblReader.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class should process the data of embl file</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/embl/EmblRecord.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblRecord</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">this class contains the parsed data of embl file</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/embl/EmblReference.html" title="class in org.biojava.nbio.core.sequence.io.embl">EmblReference</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class contains the processed data of embl file that
 contains the referenceNumber, referenceComment, referencePosition
 referenceCrossReference, referenceGroup, referenceAuthor
 referenceTitle, referenceLocation</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/EntityFinder.html" title="class in org.biojava.nbio.structure.io">EntityFinder</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Heuristical finding of Entities (called Compounds in legacy PDB format)
 in a given Structure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/EntityInfo.html" title="class in org.biojava.nbio.structure">EntityInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An object to contain the info from the PDB header for a Molecule.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/EntityType.html" title="enum in org.biojava.nbio.structure">EntityType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">The type of entity (polymer, non-polymer, water, macrolide)
 as defined in the mmCIF dictionary: <a href="http://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v40.dic/Items/_entity.type.html"></a></div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/Equals.html" title="class in org.biojava.nbio.core.util">Equals</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A set of helper methods which return true if the two parameters are
 equal to each other.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/ExonComparator.html" title="class in org.biojava.nbio.core.sequence">ExonComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Sort Exon where it is a little confusing if exons should always be ordered left to right
 where a negative stranded gene should go the other direction.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/ExonSequence.html" title="class in org.biojava.nbio.core.sequence">ExonSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A gene contains a collection of Exon sequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/ExperimentalTechnique.html" title="enum in org.biojava.nbio.structure">ExperimentalTechnique</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">An enum to represent the experimental technique of a PDB structure</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/figure/ExpressionFigure.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">ExpressionFigure</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/FastaAFPChainConverter.html" title="class in org.biojava.nbio.structure.io">FastaAFPChainConverter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A collection of static utilities to convert between <a href="org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model"><code>AFPChains</code></a> and FastaSequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FastaGeneWriter.html" title="class in org.biojava.nbio.core.sequence.io">FastaGeneWriter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Gene sequence has a Positive or Negative Strand where we want to write out to a stream the 5 to 3 prime version.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/io/template/FastaHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">FastaHeaderFormatInterface&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FastaReader.html" title="class in org.biojava.nbio.core.sequence.io">FastaReader&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Use FastaReaderHelper as an example of how to use this class where FastaReaderHelper should be the
 primary class used to read Fasta files</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FastaReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io">FastaReaderHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/data/sequence/FastaSequence.html" title="class in org.biojava.nbio.data.sequence">FastaSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A FASTA formatted sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FastaSequenceParser.html" title="class in org.biojava.nbio.core.sequence.io">FastaSequenceParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to parse a stream of a fasta file to get the sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FastaStreamer.html" title="class in org.biojava.nbio.core.sequence.io">FastaStreamer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Read from a FASTA file (or gzipped FASTA file) and create a Java stream of <a href="org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence"><code>ProteinSequence</code></a> objects
 for use in a functional programming paradigm.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/FastaStructureParser.html" title="class in org.biojava.nbio.structure.io">FastaStructureParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Reads a protein sequence from a fasta file and attempts to match it to a
 3D structure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FastaWriter.html" title="class in org.biojava.nbio.core.sequence.io">FastaWriter&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The FastaWriter writes a collection of sequences to an outputStream.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FastaWriterHelper.html" title="class in org.biojava.nbio.core.sequence.io">FastaWriterHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The class that should be used to write out fasta file of a sequence collection</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/Fastq.html" title="class in org.biojava.nbio.genome.io.fastq">Fastq</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">FASTQ formatted sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/FastqBuilder.html" title="class in org.biojava.nbio.genome.io.fastq">FastqBuilder</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Fluent builder API for creating FASTQ formatted sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/genome/io/fastq/FastqReader.html" title="interface in org.biojava.nbio.genome.io.fastq">FastqReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Reader for FASTQ formatted sequences.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/FastqTools.html" title="class in org.biojava.nbio.genome.io.fastq">FastqTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility methods for FASTQ formatted sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html" title="enum in org.biojava.nbio.genome.io.fastq">FastqVariant</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">FASTQ sequence format variant.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/genome/io/fastq/FastqWriter.html" title="interface in org.biojava.nbio.genome.io.fastq">FastqWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Writer for FASTQ formatted sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/FatCat.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCat</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/FatCatAligner.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatAligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that does calculations on an AFPChain</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCatFlexible</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/FatCatRigid.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCatRigid</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/FatCatUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.fatcat">FatCatUserArgumentProcessor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/FCAlignHelper.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FCAlignHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/Feature.html" title="class in org.biojava.nbio.genome.parsers.gff">Feature</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Feature corresponds to a single row in a GFF file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/features/FeatureDbReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features">FeatureDbReferenceInfo&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">It is <a href="org/biojava/nbio/core/sequence/features/DBReferenceInfo.html" title="class in org.biojava.nbio.core.sequence.features"><code>DBReferenceInfo</code></a> which implements <a href="org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features"><code>FeatureInterface</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/FeatureHelper.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/genome/parsers/gff/FeatureI.html" title="interface in org.biojava.nbio.genome.parsers.gff">FeatureI</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A feature on a sequence (for example, an exon or a gene), defined by a location
 and a set of attributes encoded as key/value pairs.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/features/FeatureInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureInterface&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface class to handle describing arbitrary features.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/FeatureList.html" title="class in org.biojava.nbio.genome.parsers.gff">FeatureList</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A list of FeatureI objects implemented using a Java ArrayList; corresponds to a GFF file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/features/FeatureRetriever.html" title="interface in org.biojava.nbio.core.sequence.features">FeatureRetriever&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/features/FeaturesKeyWordInterface.html" title="interface in org.biojava.nbio.core.sequence.features">FeaturesKeyWordInterface</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/FileConvert.html" title="class in org.biojava.nbio.structure.io">FileConvert</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Methods to convert a structure object into different file formats.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/FileDownloadUtils.html" title="class in org.biojava.nbio.core.util">FileDownloadUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/util/FileDownloadUtils.Hash.html" title="enum in org.biojava.nbio.core.util">FileDownloadUtils.Hash</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/FileParsingParameters.html" title="class in org.biojava.nbio.structure.io">FileParsingParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that configures parameters that can be sent to the PDB file parsers

 
  <a href="org/biojava/nbio/structure/io/FileParsingParameters.html#setParseCAOnly(boolean)"><code>FileParsingParameters.setParseCAOnly(boolean)</code></a> - parse only the Atom records for C-alpha atoms
  <a href="org/biojava/nbio/structure/io/FileParsingParameters.html#setParseSecStruc(boolean)"><code>FileParsingParameters.setParseSecStruc(boolean)</code></a> - a flag if the secondary structure information from the PDB file (author's assignment) should be parsed.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FileProxyDNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">FileProxyDNASequenceCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the stream
 the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FileProxyProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">FileProxyProteinSequenceCreator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the stream
 the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/FileProxyRNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">FileProxyRNASequenceCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is a good example of using the SequenceCreatorInterface where during parsing of the stream
 the sequence and the offset index are passed to create a Protein sequence that will be loaded in lazily.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/FlatFileCache.html" title="class in org.biojava.nbio.core.util">FlatFileCache</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides a cache for storing multiple small files in memory.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/phylo/ForesterWrapper.html" title="class in org.biojava.nbio.phylo">ForesterWrapper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class contains wrapper methods for communication between BioJava and
 forester (e.g, Data Structure conversion).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Four bit encoding of the bit formats.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/FourBitSequenceReader.FourBitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">FourBitSequenceReader.FourBitArrayWorker&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A four bit per compound implementation of the bit array worker code.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/FractionalIdentityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityInProfileScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/FractionalIdentityScorer.html" title="class in org.biojava.nbio.alignment">FractionalIdentityScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/FractionalSimilarityInProfileScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityInProfileScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair picked from an alignment
 <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/FractionalSimilarityScorer.html" title="class in org.biojava.nbio.alignment">FractionalSimilarityScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements an algorithm which computes a score for a sequence alignment pair.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/FragmentJoiner.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentJoiner</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Joins the initial Fragments together to larger Fragments</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/FragmentPair.html" title="class in org.biojava.nbio.structure.align.pairwise">FragmentPair</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">a pair of fragments of two protein structures</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/transcription/Frame.html" title="enum in org.biojava.nbio.core.sequence.transcription">Frame</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Indicates a way of translating a sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/FuzzyPoint.html" title="class in org.biojava.nbio.core.sequence.location">FuzzyPoint</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation for resolving fuzzy locations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/helper/GapArray.html" title="class in org.biojava.nbio.structure.align.helper">GapArray</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/GapPenalty.html" title="interface in org.biojava.nbio.alignment.template">GapPenalty</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure for the gap penalties used during a sequence alignment routine.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/alignment/template/GapPenalty.Type.html" title="enum in org.biojava.nbio.alignment.template">GapPenalty.Type</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Defines the possible types of gap penalties.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/GCStats.html" title="class in org.biojava.nbio.genome.parsers.gff">GCStats</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/io/template/GenbankHeaderFormatInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">GenbankHeaderFormatInterface&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/loader/GenbankProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">GenbankProxySequenceReader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenbankReader.html" title="class in org.biojava.nbio.core.sequence.io">GenbankReader&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Use <a href="org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> as an example of how to use this class where <a href="org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io"><code>GenbankReaderHelper</code></a> should be the
 primary class used to read Genbank files</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenbankReaderHelper.html" title="class in org.biojava.nbio.core.sequence.io">GenbankReaderHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/reference/GenbankReference.html" title="class in org.biojava.nbio.core.sequence.reference">GenbankReference</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">For Genbank format file only.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenbankSequenceParser.html" title="class in org.biojava.nbio.core.sequence.io">GenbankSequenceParser&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenbankWriter.html" title="class in org.biojava.nbio.core.sequence.io">GenbankWriter&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenbankWriterHelper.html" title="class in org.biojava.nbio.core.sequence.io">GenbankWriterHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The class that should be used to write out genbank file of a sequence
 collection</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/genename/GeneChromosomePosition.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePosition</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/genename/GeneChromosomePositionParser.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneChromosomePositionParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A parser that parses a file from the UCSC genome browser that contains mapping of gene name to chromosome positions</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/GeneFeatureHelper.html" title="class in org.biojava.nbio.genome">GeneFeatureHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/GeneIDGFF2Reader.html" title="class in org.biojava.nbio.genome.parsers.gff">GeneIDGFF2Reader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">http://www.bioperl.org/wiki/GTF
 Read and write FeatureLists as GFF/GTF formatted files.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/geneid/GeneIDXMLReader.html" title="class in org.biojava.nbio.genome.parsers.geneid">GeneIDXMLReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/GeneMarkGTFReader.html" title="class in org.biojava.nbio.genome.parsers.gff">GeneMarkGTFReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">http://www.bioperl.org/wiki/GTF
 Read and write FeatureLists as GFF/GTF formatted files.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/genename/GeneName.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneName</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A simple bean that contains gene name information as available from www.genenames.org</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/genename/GeneNamesParser.html" title="class in org.biojava.nbio.genome.parsers.genename">GeneNamesParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parses a file from the www.genenames.org website that contains a mapping of human gene names to other databases</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderFormat&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">We store the original header if the sequence is parsed from a fasta file and will use that exact
 sequence if we write out the sequences to a fasta file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenericFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericFastaHeaderParser&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The default fasta header parser where some headers are well defined based on the source
 database which allows us to set the source of the protein sequence and the identifier
 that can be used in future implementations to load features from external sources

 If the user has a custom header with local data then they can create their own implementation
 of a FastaHeaderParserInterface</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderFormat&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenericGenbankHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">GenericGenbankHeaderParser&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/GenericInsdcHeaderFormat.html" title="class in org.biojava.nbio.core.sequence.io">GenericInsdcHeaderFormat&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/GeneSequence.html" title="class in org.biojava.nbio.core.sequence">GeneSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/GetDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">GetDistanceMatrix</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/homology/GFF3FromUniprotBlastHits.html" title="class in org.biojava.nbio.genome.homology">GFF3FromUniprotBlastHits</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/GFF3Reader.html" title="class in org.biojava.nbio.genome.parsers.gff">GFF3Reader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">http://www.bioperl.org/wiki/GTF
 Read and write FeatureLists as GFF/GTF formatted files.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/GFF3Writer.html" title="class in org.biojava.nbio.genome.parsers.gff">GFF3Writer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/io/GOParser.html" title="class in org.biojava.nbio.ontology.io">GOParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Simple parser for the Gene Ontology (GO) flatfile format.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/Gotoh.html" title="class in org.biojava.nbio.structure.align.pairwise">Gotoh</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">a class to perform Gotoh algorithm</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/GradientMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">GradientMapper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Maps a set of real values onto a gradient.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/GradientPanel.html" title="class in org.biojava.nbio.structure.gui.util.color">GradientPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/GraphComponentOrderDetector.html" title="class in org.biojava.nbio.structure.symmetry.internal">GraphComponentOrderDetector</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The GraphOrderDetector transforms the self-alignment into a Graph and
 extracts its maximally connected Components.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/GraphComponentRefiner.html" title="class in org.biojava.nbio.structure.symmetry.internal">GraphComponentRefiner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The GraphRefiner transforms the self-alignment into a Graph and extracts its
 maximally connected Components.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/Grid.html" title="class in org.biojava.nbio.structure.contact">Grid</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A grid to be used for calculating atom contacts through a spatial hashing algorithm.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/GridCell.html" title="class in org.biojava.nbio.structure.contact">GridCell</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A grid cell to be used in contact calculation via spatial hashing algorithm.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">This is the data structure for a single Group of atoms.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/asa/GroupAsa.html" title="class in org.biojava.nbio.structure.asa">GroupAsa</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to store the results of ASA calculations, it can
 hold ASA values per atom present in <a href="org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure"><code>Group</code></a></div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/GroupContact.html" title="class in org.biojava.nbio.structure.contact">GroupContact</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A pair of residues that are in contact</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/GroupContactSet.html" title="class in org.biojava.nbio.structure.contact">GroupContactSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A set of residue-residue contacts.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/GroupIterator.html" title="class in org.biojava.nbio.structure">GroupIterator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An iterator over all groups of a structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/figure/GroupResults.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">GroupResults</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/GroupToSDF.html" title="class in org.biojava.nbio.structure.io">GroupToSDF</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/GroupType.html" title="enum in org.biojava.nbio.structure">GroupType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">This contains basic categories for Group types.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/routines/GuanUberbacher.html" title="class in org.biojava.nbio.alignment.routines">GuanUberbacher&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Guan and Uberbacher defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/GuideTree.html" title="class in org.biojava.nbio.alignment">GuideTree&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a data structure for a guide tree used during progressive multiple sequence alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/GuideTreeNode.html" title="interface in org.biojava.nbio.alignment.template">GuideTreeNode&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure for the node in a guide tree used during progressive multiple sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/GuiWrapper.html" title="class in org.biojava.nbio.structure.align.ce">GuiWrapper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to wrap some of the strucutre.gui classes using Reflection</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/Hashcoder.html" title="class in org.biojava.nbio.core.util">Hashcoder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains helper methods for generating a HashCode without having to resort to
 the commons lang hashcode builders.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/HBond.html" title="class in org.biojava.nbio.structure.secstruc">HBond</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Container that represents a hidrogen bond.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/data/HeaderInfo.html" title="class in org.biojava.nbio.survival.data">HeaderInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/HelicalRepeatUnit.html" title="class in org.biojava.nbio.structure.symmetry.core">HelicalRepeatUnit</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/Helix.html" title="class in org.biojava.nbio.structure.symmetry.core">Helix</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/axis/HelixAxisAligner.html" title="class in org.biojava.nbio.structure.symmetry.axis">HelixAxisAligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/HelixExtender.html" title="class in org.biojava.nbio.structure.symmetry.core">HelixExtender</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/HelixLayers.html" title="class in org.biojava.nbio.structure.symmetry.core">HelixLayers</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/HelixSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">HelixSolver</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/proteincomparisontool/HelloWorld.html" title="class in org.biojava.nbio.proteincomparisontool">HelloWorld</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A dummy class that does nothing.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/HelpDialog.html" title="class in org.biojava.nbio.structure.align.gui">HelpDialog</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/HetatomImpl.html" title="class in org.biojava.nbio.structure">HetatomImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Generic Implementation of a Group interface.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/HetatomImpl.PerformanceBehavior.html" title="enum in org.biojava.nbio.structure">HetatomImpl.PerformanceBehavior</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Behaviors for how to balance memory vs. performance.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/HierarchicalClusterer.html" title="interface in org.biojava.nbio.alignment.template">HierarchicalClusterer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a clustering algorithm that converts a distance matrix into a tree.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/Hit.html" title="class in org.biojava.nbio.core.search.io">Hit</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class models a search Hit.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/HmmerDemo.html" title="class in demo">HmmerDemo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The cookbook recipe for how to request Pfam annotations for a protein sequence using the Hmmer3 service</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/hmmer/HmmerDomain.html" title="class in org.biojava.nbio.ws.hmmer">HmmerDomain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides the details of a domain hit</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/hmmer/HmmerResult.html" title="class in org.biojava.nbio.ws.hmmer">HmmerResult</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The results of a Hmmer search for a single sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ws/hmmer/HmmerScan.html" title="interface in org.biojava.nbio.ws.hmmer">HmmerScan</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for performing Hmmscans on sequences.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/Hsp.html" title="class in org.biojava.nbio.core.search.io">Hsp&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class models a search Hsp.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/HSVColorSpace.html" title="class in org.biojava.nbio.structure.gui.util.color">HSVColorSpace</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/IcosahedralSampler.html" title="class in org.biojava.nbio.structure.symmetry.geometry">IcosahedralSampler</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents an even coverage of quaternion space by 60 points.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/Icosahedron.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Icosahedron</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/helper/IdxComparator.html" title="class in org.biojava.nbio.structure.align.helper">IdxComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/IlluminaFastqReader.html" title="class in org.biojava.nbio.genome.io.fastq">IlluminaFastqReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Reader for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_ILLUMINA"><code>FastqVariant.FASTQ_ILLUMINA</code></a> formatted sequences.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/IlluminaFastqWriter.html" title="class in org.biojava.nbio.genome.io.fastq">IlluminaFastqWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Writer for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_ILLUMINA"><code>FastqVariant.FASTQ_ILLUMINA</code></a> formatted sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/helper/IndexPair.html" title="class in org.biojava.nbio.structure.align.helper">IndexPair</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/InputStreamProvider.html" title="class in org.biojava.nbio.core.util">InputStreamProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides an InputStream from a File.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A collection of locations which are used whenever we work with INSDC; some
 of which could be deprecated (from INSDC's point of view) yet appear
 in records.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.BondLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.BondLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to represent bond locations equivalent to bond(7,8) or bond(7).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.GroupLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.GroupLocation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Deprecated in INSDC yet still appears; equivalent to the order()
 directive except no 5' to 3' ordering is defined.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.OneOfLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.OneOfLocation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Deprecated in INSDC; refers to a set of locations of which one
 location could be valid e.g. one-of(location, location, location).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/InsdcLocations.OrderLocation.html" title="class in org.biojava.nbio.core.sequence.location">InsdcLocations.OrderLocation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to describe a 5' to 3' ordering but no firm assurance it is correct</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/InsdcParser.html" title="class in org.biojava.nbio.core.sequence.location">InsdcParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parser for working with INSDC style locations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/IntegerOntology.html" title="class in org.biojava.nbio.ontology">IntegerOntology</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/InterfaceFinder.html" title="class in org.biojava.nbio.structure.contact">InterfaceFinder</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class containing methods to find interfaces in a given structure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/IntronSequence.html" title="class in org.biojava.nbio.core.sequence">IntronSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/ontology/InvalidTermException.html" title="class in org.biojava.nbio.ontology">InvalidTermException</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab5">
<div class="block">Thrown to indicate that an ontology term is not acceptable or
 appropriate in a given context</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/util/IOUtils.html" title="class in org.biojava.nbio.core.sequence.io.util">IOUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/io/util/IOUtils.ReaderProcessor.html" title="interface in org.biojava.nbio.core.sequence.io.util">IOUtils.ReaderProcessor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Closure interface used when working with
 <a href="org/biojava/nbio/core/sequence/io/util/IOUtils.html#processReader(java.io.BufferedReader,org.biojava.nbio.core.sequence.io.util.IOUtils.ReaderProcessor)"><code>IOUtils.processReader(BufferedReader, ReaderProcessor)</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/aaproperties/IPeptideProperties.html" title="interface in org.biojava.nbio.aaproperties">IPeptideProperties</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">An interface to generate some basic physico-chemical properties of protein sequences.<br/>
 The following properties could be generated:</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.html" title="interface in org.biojava.nbio.aaproperties.profeat">IProfeatProperties</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.ATTRIBUTE.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.ATTRIBUTE</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enumeration of the seven different attributes</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.DISTRIBUTION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.DISTRIBUTION</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enumeration of the distribution for the first, first 25%, first 50%, first 75% and 100% of the grouping</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.GROUPING.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.GROUPING</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enumeration of the three different groupings for each attributes</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/aaproperties/profeat/IProfeatProperties.TRANSITION.html" title="enum in org.biojava.nbio.aaproperties.profeat">IProfeatProperties.TRANSITION</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enumeration of the transition between groupA and groupB</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/Isotope.html" title="class in org.biojava.nbio.aaproperties.xml">Isotope</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/IUPACParser.html" title="class in org.biojava.nbio.core.sequence.io">IUPACParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Available translations

 
 1 - UNIVERSAL
 2 - VERTEBRATE_MITOCHONDRIAL
 3 - YEAST_MITOCHONDRIAL
 4 - MOLD_MITOCHONDRIAL
 5 - INVERTEBRATE_MITOCHONDRIAL
 6 - CILIATE_NUCLEAR
 9 - ECHINODERM_MITOCHONDRIAL
 10 - EUPLOTID_NUCLEAR
 11 - BACTERIAL
 12 - ALTERNATIVE_YEAST_NUCLEAR
 13 - ASCIDIAN_MITOCHONDRIAL
 14 - FLATWORM_MITOCHONDRIAL
 15 - BLEPHARISMA_MACRONUCLEAR
 16 - 2CHLOROPHYCEAN_MITOCHONDRIAL
 21 - TREMATODE_MITOCHONDRIAL
 23 - SCENEDESMUS_MITOCHONDRIAL
 

 Taken from <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi?mode=c">NCBI</a> with slight modification and put into the classpath resource.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava.nbio.core.sequence.io">IUPACParser.IUPACTable</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Holds the concept of a codon table from the IUPAC format</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/autosuggest/JAutoSuggest.html" title="class in org.biojava.nbio.structure.align.gui.autosuggest">JAutoSuggest</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JTextField that can make suggestions for auto-complete.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/JMatrixPanel.html" title="class in org.biojava.nbio.structure.gui">JMatrixPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">a JPanel that can display a difference of distance matrix and paths that have been
 taken for the alignment</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/events/JmolAlignedPositionListener.html" title="class in org.biojava.nbio.structure.gui.events">JmolAlignedPositionListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/JmolPanel.html" title="class in org.biojava.nbio.structure.align.gui.jmol">JmolPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/JmolPanel.JmolLoggerAdapter.html" title="class in org.biojava.nbio.structure.align.gui.jmol">JmolPanel.JmolLoggerAdapter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGenerator.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGenerator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorC1.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorC1</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorCn.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorCn</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorDn.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorDn</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorH.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorH</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorI.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorI</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorO.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorO</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorPointGroup.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorPointGroup</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/jmolScript/JmolSymmetryScriptGeneratorT.html" title="class in org.biojava.nbio.structure.symmetry.jmolScript">JmolSymmetryScriptGeneratorT</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/JmolTools.html" title="class in org.biojava.nbio.structure.align.gui.jmol">JmolTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/JmolViewerImpl.html" title="class in org.biojava.nbio.structure.gui">JmolViewerImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/webstart/JNLPProxy.html" title="class in org.biojava.nbio.structure.align.webstart">JNLPProxy</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/JoiningSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">JoiningSequenceReader&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This reader actually proxies onto multiple types of sequence in order
 to allow a number of sequence objects to act as if they are one sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/helper/JointFragments.html" title="class in org.biojava.nbio.structure.align.helper">JointFragments</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class that defines which set of atoms are considered
 to be on equivalent positions.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/JournalArticle.html" title="class in org.biojava.nbio.structure">JournalArticle</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">PDB-specific</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/JPrintPanel.html" title="class in org.biojava.nbio.structure.align.gui">JPrintPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/Jronn.html" title="class in org.biojava.nbio.ronn">Jronn</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class gives public API to RONN functions.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/Jronn.Range.html" title="class in org.biojava.nbio.ronn">Jronn.Range</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Holder for the ranges, contain pointers to starting and ending position
 on the sequence which comprises a disordered region.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/figure/KaplanMeierFigure.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">KaplanMeierFigure</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/utils/KeyedWeakReference.html" title="class in org.biojava.nbio.ontology.utils">KeyedWeakReference</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Subclass of WeakReference which includes and extra field (the key)
 which can be used to help cleanup once this reference has been
 enqueued.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/figure/KMFigureInfo.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">KMFigureInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/Ladder.html" title="class in org.biojava.nbio.structure.secstruc">Ladder</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Ladder is a set of one or more consecutive bridges of identical type.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/LightweightProfile.html" title="interface in org.biojava.nbio.core.sequence.template">LightweightProfile&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a minimal data structure for reading and writing a sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/template/LightweightProfile.StringFormat.html" title="enum in org.biojava.nbio.core.sequence.template">LightweightProfile.StringFormat</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of output formats.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/LinearColorInterpolator.html" title="class in org.biojava.nbio.structure.gui.util.color">LinearColorInterpolator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/gui/util/color/LinearColorInterpolator.InterpolationDirection.html" title="enum in org.biojava.nbio.structure.gui.util.color">LinearColorInterpolator.InterpolationDirection</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.html" title="class in org.biojava.nbio.structure.io">LocalPDBDirectory</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Superclass for classes which download and interact with the PDB's FTP server,
 specifically <a href="org/biojava/nbio/structure/io/PDBFileReader.html" title="class in org.biojava.nbio.structure.io"><code>PDBFileReader</code></a> and <a href="org/biojava/nbio/structure/io/CifFileReader.html" title="class in org.biojava.nbio.structure.io"><code>CifFileReader</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.FetchBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.FetchBehavior</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Controls when the class should fetch files from the ftp server</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/io/LocalPDBDirectory.ObsoleteBehavior.html" title="enum in org.biojava.nbio.structure.io">LocalPDBDirectory.ObsoleteBehavior</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Behaviors for when an obsolete structure is requested.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/LocalProteinDomainParser.html" title="class in org.biojava.nbio.structure.domain">LocalProteinDomainParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Protein Domain Parser is a an algorithm that attempts at assigning domains for 3D protein structures.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/scop/LocalScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">LocalScopDatabase</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Classes which implement ScopDatabase in a way which allows them to serve queries
 without accessing the internet should implement this interface instead.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/location/template/Location.html" title="interface in org.biojava.nbio.core.sequence.location.template">Location</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Sets of integers used to represent the location of features on sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/Location.html" title="class in org.biojava.nbio.genome.parsers.gff">Location</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A location on a sequence.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/template/Location.Tools.html" title="class in org.biojava.nbio.core.sequence.location.template">Location.Tools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Helper methods for use with the Location classes.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/LocationHelper.html" title="class in org.biojava.nbio.core.sequence.location">LocationHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Helper methods for use with the Location classes.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/gff/LocIterator.html" title="class in org.biojava.nbio.genome.parsers.gff">LocIterator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Move a sliding window over a Location.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/LogColorMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">LogColorMapper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Performs a log10 transform on input before passing the values off to another
 colorMapper.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/jama/LUDecomposition.html" title="class in org.biojava.nbio.structure.jama">LUDecomposition</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">LU Decomposition.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/alignment/qblast/MapToStringTransformer.html" title="class in org.biojava.nbio.ws.alignment.qblast">MapToStringTransformer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Transforms Map to String.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/jama/Maths.html" title="class in org.biojava.nbio.structure.jama">Maths</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/Matrices.html" title="class in org.biojava.nbio.structure.geometry">Matrices</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Matrices contains static methods to operate and transform matrices used in 3D
 geometry (transformation matrices and rotation matrices).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Jama = Java Matrix class.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/matrix/Matrix.html" title="class in org.biojava.nbio.survival.cox.matrix">Matrix</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">***********************************************************************
 Compilation: javac Matrix.java Execution: java Matrix

 A bare-bones collection of static methods for manipulating matrices.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/MatrixAligner.html" title="interface in org.biojava.nbio.alignment.template">MatrixAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> which builds a score matrix during computation.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/ce/MatrixListener.html" title="interface in org.biojava.nbio.structure.align.ce">MatrixListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/comparators/MeanModelComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">MeanModelComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/MeanQuantizer.html" title="class in org.biojava.nbio.survival.kaplanmeier.metadata">MeanQuantizer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MemoryMonitor.html" title="class in org.biojava.nbio.structure.align.gui">MemoryMonitor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tracks Memory allocated and used, displayed in graph form.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MenuCreator.html" title="class in org.biojava.nbio.structure.align.gui">MenuCreator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Create the menus for structure alignment GUI windows (JFrames).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/MenuCreator.html" title="class in org.biojava.nbio.structure.gui.util">MenuCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Create the menu for BiojavaJmol</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MenuCreator.DotPlotListener.html" title="class in org.biojava.nbio.structure.align.gui">MenuCreator.DotPlotListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Creates a frame to display a DotPlotPanel.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/exceptions/Messages.html" title="class in org.biojava.nbio.core.exceptions">Messages</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/metadata/MetaDataInfo.html" title="class in org.biojava.nbio.survival.kaplanmeier.metadata">MetaDataInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/cif/MetalBondConsumer.html" title="interface in org.biojava.nbio.structure.io.cif">MetalBondConsumer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Consume metal bond data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/MetalBondConsumerImpl.html" title="class in org.biojava.nbio.structure.io.cif">MetalBondConsumerImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by andreas on 6/9/16.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/cif/MetalBondConverter.html" title="class in org.biojava.nbio.structure.io.cif">MetalBondConverter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by andreas on 6/6/16.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/MetalBondDistance.html" title="class in org.biojava.nbio.structure.chem">MetalBondDistance</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A bean that contains cutoffs for correctly detecting metal bonds.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/basepairs/MismatchedBasePairParameters.html" title="class in org.biojava.nbio.structure.basepairs">MismatchedBasePairParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class allows for finding inter-strand base pairs that are not necessarily canonical Watson-Crick pairs.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/mmtf/MmtfActions.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfActions</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class of functions for reading and writing Biojava structures using MMTF</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/MMTFFileReader.html" title="class in org.biojava.nbio.structure.io">MMTFFileReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to read MMTF files and cache them locally.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/mmtf/MmtfStructureReader.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfStructureReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A biojava specific structure inflator for MMTF.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/mmtf/MmtfStructureWriter.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfStructureWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class to take Biojava structure data and covert to the DataApi for encoding.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/mmtf/MmtfSummaryDataBean.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfSummaryDataBean</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class to store the summary data for a given structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/mmtf/MmtfUtils.html" title="class in org.biojava.nbio.structure.io.mmtf">MmtfUtils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utils class of functions needed for Biojava to read and write to mmtf.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/Model.html" title="class in org.biojava.nbio.structure">Model</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An internal utility class for StructureImpl to make it easier to manage poly and nonpoly chains.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/ModelLoader.html" title="class in org.biojava.nbio.ronn">ModelLoader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class that loads data from the model files into <a href="org/biojava/nbio/ronn/ModelLoader.Model.html" title="class in org.biojava.nbio.ronn"><code>ModelLoader.Model</code></a> objects</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/ModelLoader.Model.html" title="class in org.biojava.nbio.ronn">ModelLoader.Model</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represent a RONN model</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/ModelLoader.Threshold.html" title="class in org.biojava.nbio.ronn">ModelLoader.Threshold</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents a Threshold</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/protmod/ModificationCategory.html" title="enum in org.biojava.nbio.protmod">ModificationCategory</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">define modification categories.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/protmod/ModificationCondition.html" title="interface in org.biojava.nbio.protmod">ModificationCondition</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Conditions of a protein modification, e.g. components and atoms.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/ModificationConditionImpl.html" title="class in org.biojava.nbio.protmod">ModificationConditionImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/ModificationLinkage.html" title="class in org.biojava.nbio.protmod">ModificationLinkage</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/protmod/ModificationOccurrenceType.html" title="enum in org.biojava.nbio.protmod">ModificationOccurrenceType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/ModifiedAminoAcidCompoundSet.html" title="class in org.biojava.nbio.aaproperties.xml">ModifiedAminoAcidCompoundSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/protmod/structure/ModifiedCompound.html" title="interface in org.biojava.nbio.protmod.structure">ModifiedCompound</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Root interface for all modifications in structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/structure/ModifiedCompoundImpl.html" title="class in org.biojava.nbio.protmod.structure">ModifiedCompoundImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/io/ModifiedCompoundXMLConverter.html" title="class in org.biojava.nbio.protmod.io">ModifiedCompoundXMLConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/MomentsOfInertia.html" title="class in org.biojava.nbio.structure.geometry">MomentsOfInertia</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The moment of inertia, otherwise known as the angular mass or rotational
 inertia, of a rigid body determines the torque needed for a desired angular
 acceleration about a rotational axis.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/geometry/MomentsOfInertia.SymmetryClass.html" title="enum in org.biojava.nbio.structure.geometry">MomentsOfInertia.SymmetryClass</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A MultipleAlignment is a Data Structure to store the core information of a
 multiple structure alignment, as a return type.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MultipleAlignmentCalc.html" title="class in org.biojava.nbio.structure.align.gui">MultipleAlignmentCalc</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that obtains structures via DAS and aligns them.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAlignmentCoordManager.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleAlignmentCoordManager</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Generalization of the Coodinate Manager to include an arbitrary number of
 sequences (lines) for MultipleAlignment visualization.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentDisplay.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentDisplay</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility functions to generalize the visualization of MultipleAlignments in
 molecular viewers.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignmentEnsemble</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A MultipleAlignmentEnsemble is a collection of <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s
 that share the same structures (Atoms) and creation properties (algorithm,
 version, creation time, etc.).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html" title="class in org.biojava.nbio.structure.align.multiple">MultipleAlignmentEnsembleImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A general implementation of a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignmentEnsemble</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MultipleAlignmentGUI.html" title="class in org.biojava.nbio.structure.align.gui">MultipleAlignmentGUI</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JFrame that allows to trigger a multiple structure alignment,
 either from files in a directory or after manual upload.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentImpl.html" title="class in org.biojava.nbio.structure.align.multiple">MultipleAlignmentImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A general implementation of a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides a 3D visualization Frame in Jmol for
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MultipleAlignmentJmolDisplay.html" title="class in org.biojava.nbio.structure.align.gui">MultipleAlignmentJmolDisplay</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility Class that provides helper methods for the visualization of
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentScorer.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentScorer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility class for calculating common scores of <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>s.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentTools.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility functions for working with <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleAlignmentWriter.html" title="class in org.biojava.nbio.structure.align.multiple.util">MultipleAlignmentWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class contains functions for the conversion of <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>
 to various String outputs.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/xml/MultipleAlignmentXMLConverter.html" title="class in org.biojava.nbio.structure.align.xml">MultipleAlignmentXMLConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Helper methods to convert all the hierarchy levels of a MultipleAlignment
 into an XML format.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/xml/MultipleAlignmentXMLParser.html" title="class in org.biojava.nbio.structure.align.xml">MultipleAlignmentXMLParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parse an XML file representing a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsemble.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignmentEnsemble</code></a>, so
 that the original alignment can be recovered.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleAligPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JPanel that can display the sequence alignment of a
 <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> in a nice way and interact with Jmol by
 selecting the aligned atoms of the  sequence selection.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanelMouseMotionListener.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleAligPanelMouseMotionListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Mouse Motion Listener for the <a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel"><code>MultipleAligPanel</code></a>,
 which provides methods to obtain positions of the mouse
 and connect them to the sequence alignment positions using
 the information in <a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAlignmentCoordManager.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel"><code>MultipleAlignmentCoordManager</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/mc/MultipleMcMain.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcMain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Main class of the Java implementation of the Combinatorial Extension -
 Monte Carlo (CEMC) Algorithm,
 as it was originally described by C.Guda, E.D.Scheeff, P.E.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/mc/MultipleMcOptimizer.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcOptimizer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class takes a MultipleAlignment seed previously generated and runs a
 Monte Carlo optimization in order to improve the overall score and highlight
 common structural motifs.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/mc/MultipleMcParameters.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains the parameters to be sent to the MC optimization.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a minimal data structure for reading and writing a sequence alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleStatusDisplay.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">MultipleStatusDisplay</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class provides information of the selected positions in the
 <a href="org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel"><code>MultipleAligPanel</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/MultipleStructureAligner.html" title="interface in org.biojava.nbio.structure.align">MultipleStructureAligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for the Multiple Structure Alignment Algorithms.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/multiple/util/MultipleSuperimposer.html" title="interface in org.biojava.nbio.structure.align.multiple.util">MultipleSuperimposer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for Multiple Alignment superposition algorithms.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/MutableAlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template">MutableAlignedSequence&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a mutable (editable) data structure for an <a href="org/biojava/nbio/core/alignment/template/AlignedSequence.html" title="interface in org.biojava.nbio.core.alignment.template"><code>AlignedSequence</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/MutableProfile.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfile&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a mutable (editable) data structure for a <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/MutableProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableProfilePair&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a mutable (editable) data structure for a <a href="org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template"><code>ProfilePair</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/MutableSequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">MutableSequencePair&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a mutable (editable) data structure for the results of pairwise sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/Mutator.html" title="class in org.biojava.nbio.structure">Mutator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that can change one amino acid to another.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MyAlignmentLoadListener.html" title="class in org.biojava.nbio.structure.align.gui">MyAlignmentLoadListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Loads an alignment in an XML format and displays its content in a
 new Jmol panel.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MyDistMaxListener.html" title="class in org.biojava.nbio.structure.align.gui">MyDistMaxListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Shows the interatomic Distance Matrices of all the Structures aligned in different Frames.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MyExportListener.html" title="class in org.biojava.nbio.structure.align.gui">MyExportListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/MyJmolStatusListener.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MyJmolStatusListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MyOpenPdbFileListener.html" title="class in org.biojava.nbio.structure.align.gui">MyOpenPdbFileListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MySaveFileListener.html" title="class in org.biojava.nbio.structure.align.gui">MySaveFileListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Save an alignment to a specified File by the user.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/MyTableRowSorter.html" title="class in org.biojava.nbio.structure.align.gui">MyTableRowSorter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/MyValidationEventHandler.html" title="class in org.biojava.nbio.aaproperties.xml">MyValidationEventHandler</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/Name2Count.html" title="class in org.biojava.nbio.aaproperties.xml">Name2Count</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastAlignmentProperties.html" title="class in org.biojava.nbio.ws.alignment.qblast">NCBIQBlastAlignmentProperties</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class wraps a QBlast search request parameter <code>Map</code> by adding several convenient parameter addition
 methods.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastOutputProperties.html" title="class in org.biojava.nbio.ws.alignment.qblast">NCBIQBlastOutputProperties</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class wraps a QBlast output parameter <code>Map</code> by adding several convenient parameter addition methods.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastService.html" title="class in org.biojava.nbio.ws.alignment.qblast">NCBIQBlastService</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides a simple way of submitting BLAST request to the QBlast service at NCBI.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/NCBIQBlastServiceDemo.html" title="class in demo">NCBIQBlastServiceDemo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A simple demo showing <a href="org/biojava/nbio/ws/alignment/qblast/NCBIQBlastService.html" title="class in org.biojava.nbio.ws.alignment.qblast"><code>NCBIQBlastService</code></a> usage</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/NeedlemanWunsch.html" title="class in org.biojava.nbio.alignment">NeedlemanWunsch&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Needleman and Wunsch defined an algorithm for pairwise global sequence alignments (from the first until the last
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> of each <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound">NucleotideCompound</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/NucleotideImpl.html" title="class in org.biojava.nbio.structure">NucleotideImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A nucleotide group is almost the same as a Hetatm group.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/NullOutputStream.html" title="class in org.biojava.nbio.ronn">NullOutputStream</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The stream that void its input</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/figure/NumbersAtRiskPanel.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">NumbersAtRiskPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/obo/OboFileEventListener.html" title="interface in org.biojava.nbio.ontology.obo">OboFileEventListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">an interface for events that occur during parsing of .obo files</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/obo/OboFileHandler.html" title="class in org.biojava.nbio.ontology.obo">OboFileHandler</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A  file handler for .obo files</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/obo/OboFileParser.html" title="class in org.biojava.nbio.ontology.obo">OboFileParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to parse the content of an OBO file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/obo/OboFileParser.SOPair.html" title="class in org.biojava.nbio.ontology.obo">OboFileParser.SOPair</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/io/OboParser.html" title="class in org.biojava.nbio.ontology.io">OboParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parses an OBO file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/Octahedron.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Octahedron</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/Ontology.html" title="interface in org.biojava.nbio.ontology">Ontology</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">An ontology.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/Ontology.Impl.html" title="class in org.biojava.nbio.ontology">Ontology.Impl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A basic in-memory implementation of an ontology</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/ontology/OntologyException.html" title="class in org.biojava.nbio.ontology">OntologyException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">Thrown to indicate an error in an Ontology object</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/OntologyFactory.html" title="interface in org.biojava.nbio.ontology">OntologyFactory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A factory for Ontology instances.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/OntologyOps.html" title="interface in org.biojava.nbio.ontology">OntologyOps</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">This is an interface for optimizing ontology operators.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/OntologyTerm.html" title="interface in org.biojava.nbio.ontology">OntologyTerm</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A term in an ontology which identifies another ontology.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/OntologyTerm.Impl.html" title="class in org.biojava.nbio.ontology">OntologyTerm.Impl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Simple in-memory implementation of a remote ontology term.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/OntoTools.html" title="class in org.biojava.nbio.ontology">OntoTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Tools for manipulating ontologies.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/quaternary/OperatorResolver.html" title="class in org.biojava.nbio.structure.quaternary">OperatorResolver</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to resolve the operators for transformations</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/OptimalCECPMain.html" title="class in org.biojava.nbio.structure.align.ce">OptimalCECPMain</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A wrapper for <a href="org/biojava/nbio/structure/align/ce/CeMain.html" title="class in org.biojava.nbio.structure.align.ce"><code>CeMain</code></a> which sets default parameters to be appropriate for finding
 circular permutations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/OptimalCECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">OptimalCECPParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains the parameters that can be sent to CE</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/symmetry/internal/OrderDetector.html" title="interface in org.biojava.nbio.structure.symmetry.internal">OrderDetector</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A method to decide the order of symmetry (number of subunits)
 given a structure self-alignment, calculated by CE-Symm.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/quaternary/OrderedPair.html" title="class in org.biojava.nbio.structure.quaternary">OrderedPair&lt;T&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An ordered pair represents a component of a cartesian product.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/ORonn.html" title="class in org.biojava.nbio.ronn">ORonn</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Fully re-factored and enhanced version of RONN.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/ORonnModel.html" title="class in org.biojava.nbio.ronn">ORonnModel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Fully re-factored version of RONN model.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/query/OutputHitsGFF.html" title="class in org.biojava.nbio.genome.query">OutputHitsGFF</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/Pair.html" title="class in org.biojava.nbio.structure.contact">Pair&lt;T&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Pair of objects.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/PairInProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">PairInProfileScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an algorithm which computes a score for a sequence alignment pair picked from an alignment <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/PairwiseSequenceAligner.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>s.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/PairwiseSequenceScorer.html" title="interface in org.biojava.nbio.alignment.template">PairwiseSequenceScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an algorithm which computes a score for a pair of sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/ParameterGUI.html" title="class in org.biojava.nbio.structure.align.gui">ParameterGUI</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">UI for <a href="org/biojava/nbio/structure/align/ce/ConfigStrucAligParams.html" title="interface in org.biojava.nbio.structure.align.ce"><code>ConfigStrucAligParams</code></a>, for the AlignmentGUI.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/ParseFastaFileDemo.html" title="class in demo">ParseFastaFileDemo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by andreas on 6/17/15.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/ParseGO.html" title="class in demo">ParseGO</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/genome/io/fastq/ParseListener.html" title="interface in org.biojava.nbio.genome.io.fastq">ParseListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Low-level event based parser callback.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/core/exceptions/ParserException.html" title="class in org.biojava.nbio.core.exceptions">ParserException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">General abstraction of different parsing errors</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/PartitionRefiner.html" title="interface in org.biojava.nbio.alignment.template">PartitionRefiner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an algorithm which computes a new alignment profile by splitting a current alignment and realigning.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/PassthroughIdentifier.html" title="class in org.biojava.nbio.structure">PassthroughIdentifier</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A stub StructureIdentifier, representing the full structure in all cases.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/PDBBioAssemblyParser.html" title="class in org.biojava.nbio.structure.io">PDBBioAssemblyParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parses REMARK 350 records in a PDB file and creates transformations to
 construct the quaternary structure of a protein from an asymmetric unit</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/PDBCrystallographicInfo.html" title="class in org.biojava.nbio.structure">PDBCrystallographicInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to hold crystallographic information about a PDB structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/PDBDirPanel.html" title="class in org.biojava.nbio.structure.gui.util">PDBDirPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to define where a structure for the alignment is coming from</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/PDBFileParser.html" title="class in org.biojava.nbio.structure.io">PDBFileParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class implements the actual PDB file parsing.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/PDBFileReader.html" title="class in org.biojava.nbio.structure.io">PDBFileReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The wrapper class for parsing a PDB file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/PDBHeader.html" title="class in org.biojava.nbio.structure">PDBHeader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that contains PDB Header information.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/PdbId.html" title="class in org.biojava.nbio.structure">PdbId</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A wrapper class for the PDB identifier.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/client/PdbPair.html" title="class in org.biojava.nbio.structure.align.client">PdbPair</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A pair for structure alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/xml/PdbPairsMessage.html" title="class in org.biojava.nbio.structure.align.xml">PdbPairsMessage</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/xml/PdbPairXMLConverter.html" title="class in org.biojava.nbio.structure.align.xml">PdbPairXMLConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/structure/io/PDBParseException.html" title="class in org.biojava.nbio.structure.io">PDBParseException</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab5">
<div class="block">An exception during the parsing of a PDB file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/PDBRecord.html" title="interface in org.biojava.nbio.structure">PDBRecord</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">An interface implemented by all classes that represent PDB records.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/PDBServerPanel.html" title="class in org.biojava.nbio.structure.gui.util">PDBServerPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to define where a structure for the alignment is coming from</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/PDBStatus.html" title="class in org.biojava.nbio.structure">PDBStatus</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Methods for getting the status of a PDB file (current, removed, unreleased)
 and for accessing different versions of the structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/PDBStatus.Status.html" title="enum in org.biojava.nbio.structure">PDBStatus.Status</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Represents a simplified 3 state status of PDB IDs.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/util/PDBTemporaryStorageUtils.html" title="class in org.biojava.nbio.structure.io.util">PDBTemporaryStorageUtils</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block"><b>Internal use only.</b></div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/util/PDBTemporaryStorageUtils.LinkRecord.html" title="class in org.biojava.nbio.structure.io.util">PDBTemporaryStorageUtils.LinkRecord</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Temporary data storage for LINK records.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/PDBUploadPanel.html" title="class in org.biojava.nbio.structure.gui.util">PDBUploadPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JPanel to upload 2 custom PDB files.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/PDPDistanceMatrix.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPDistanceMatrix</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/PDPParameters.html" title="class in org.biojava.nbio.structure.domain.pdp">PDPParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/PeptideProperties.html" title="class in org.biojava.nbio.aaproperties">PeptideProperties</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is an adaptor class which enable the ease of generating protein properties.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/aaproperties/PeptideProperties.SingleLetterAACode.html" title="enum in org.biojava.nbio.aaproperties">PeptideProperties.SingleLetterAACode</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enumeration of 20 standard amino acid code</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/PeptidePropertiesImpl.html" title="class in org.biojava.nbio.aaproperties">PeptidePropertiesImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class contains the actual implementation of IPeptideProperties and is wrapped around by PeptideProperties for ease of use.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/utils/PermutationGenerator.html" title="class in org.biojava.nbio.structure.symmetry.utils">PermutationGenerator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/PermutationGroup.html" title="class in org.biojava.nbio.structure.symmetry.core">PermutationGroup</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/PlainFastaHeaderParser.html" title="class in org.biojava.nbio.core.sequence.io">PlainFastaHeaderParser&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The plain fasta header takes everything in the header as a single entity.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/location/template/Point.html" title="interface in org.biojava.nbio.core.sequence.location.template">Point</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Holds a single point part of a location</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/location/template/Point.Resolver.html" title="interface in org.biojava.nbio.core.sequence.location.template">Point.Resolver&lt;T&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Used to resolve a position about a point</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/symmetry/geometry/Polyhedron.html" title="interface in org.biojava.nbio.structure.symmetry.geometry">Polyhedron</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/chem/PolymerType.html" title="enum in org.biojava.nbio.structure.chem">PolymerType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enumerates the classification of polymers.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/utils/PowerSet.html" title="class in org.biojava.nbio.structure.symmetry.utils">PowerSet&lt;T&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">In mathematics, the power set (or powerset) of any set S, written P(S), is
 the set of all subsets of S, including the empty set and S itself.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="demo/PredictDisorder.html" title="class in demo">PredictDisorder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation of XMLWriter which emits nicely formatted documents
 to a PrintWriter.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/Prism.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Prism</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/ProfeatProperties.html" title="class in org.biojava.nbio.aaproperties.profeat">ProfeatProperties</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is an adaptor class which enable the ease of generating profeat properties.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/profeat/ProfeatPropertiesImpl.html" title="class in org.biojava.nbio.aaproperties.profeat">ProfeatPropertiesImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template">Profile&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure for the results of sequence alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/alignment/template/Profile.StringFormat.html" title="enum in org.biojava.nbio.core.alignment.template">Profile.StringFormat</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">List of output formats.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/ProfilePair.html" title="interface in org.biojava.nbio.core.alignment.template">ProfilePair&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure for the results of the alignment of a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/ProfileProfileAligner.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of profiles.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/ProfileProfileScorer.html" title="interface in org.biojava.nbio.alignment.template">ProfileProfileScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an algorithm which computes a score for a pairing of alignment profiles.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/ProfileView.html" title="interface in org.biojava.nbio.core.alignment.template">ProfileView&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure for a view of sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/misc/ProteinComplexSignature.html" title="class in org.biojava.nbio.structure.symmetry.misc">ProteinComplexSignature</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/util/ProteinMappingTools.html" title="class in org.biojava.nbio.genome.util">ProteinMappingTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/protmod/ProteinModification.html" title="interface in org.biojava.nbio.protmod">ProteinModification</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">This interface defines information about a specific protein
 modification.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/structure/ProteinModificationIdentifier.html" title="class in org.biojava.nbio.protmod.structure">ProteinModificationIdentifier</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Identify attachment modification in a 3-D structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/ProteinModificationImpl.html" title="class in org.biojava.nbio.protmod">ProteinModificationImpl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class contains information about a specific protein
 modification.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/ProteinModificationImpl.Builder.html" title="class in org.biojava.nbio.protmod">ProteinModificationImpl.Builder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Uses Builder pattern to build a ProteinModification.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/ProteinModificationRegistry.html" title="class in org.biojava.nbio.protmod">ProteinModificationRegistry</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class serves as a instance registry by maintaining
 a pool of ProteinModification instances.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/io/ProteinModificationXmlReader.html" title="class in org.biojava.nbio.protmod.io">ProteinModificationXmlReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The representation of a ProteinSequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/ProteinSequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">ProteinSequenceCreator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to create a ProteinSequence from a String to allow for details
 about the location of the sequence etc.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/ProxySequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">ProxySequenceReader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/jama/QRDecomposition.html" title="class in org.biojava.nbio.structure.jama">QRDecomposition</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">QR Decomposition.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlign</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Quaternary Structure Alignment (QS-Align).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/quaternary/QsAlignParameters.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The parameter bean for the <a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary"><code>QsAlign</code></a> algorithm.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/quaternary/QsAlignResult.html" title="class in org.biojava.nbio.structure.align.quaternary">QsAlignResult</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Result of a Quaternary Structure Alignment <a href="org/biojava/nbio/structure/align/quaternary/QsAlign.html" title="class in org.biojava.nbio.structure.align.quaternary"><code>QsAlign</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/align/quaternary/QsRelation.html" title="enum in org.biojava.nbio.structure.align.quaternary">QsRelation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">The Quaternary Structure Relation describes the pairwise relation between two
 quaternary structures.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/features/Qualifier.html" title="class in org.biojava.nbio.core.sequence.features">Qualifier</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/features/QualityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QualityFeature&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/features/QuantityFeature.html" title="class in org.biojava.nbio.core.sequence.features">QuantityFeature&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/QuatSuperpositionScorer.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSuperpositionScorer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryDetector</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Detects the symmetry (global, pseudo, internal and local) of protein
 structures.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryParameters.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The QuatSymmetryParameters specify the options used for the detection of
 quaternary symmetry in structures using the <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryResults.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryResults</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Holds the results of quaternary symmetry perception obtained with
 <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryScores.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetryScores</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetrySolver.html" title="interface in org.biojava.nbio.structure.symmetry.core">QuatSymmetrySolver</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetrySubunits.html" title="class in org.biojava.nbio.structure.symmetry.core">QuatSymmetrySubunits</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A bean to represent information about the set of <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a>s being
 considered for symmetry detection.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/RasmolCommandListener.html" title="class in org.biojava.nbio.structure.align.gui.jmol">RasmolCommandListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">a utility class that listens to Ramsol script commands in the @link <a href="org/biojava/nbio/structure/gui/BiojavaJmol.html" title="class in org.biojava.nbio.structure.gui"><code>BiojavaJmol</code></a> class</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/RectangularPrism.html" title="class in org.biojava.nbio.structure.symmetry.geometry">RectangularPrism</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ReducedChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">ReducedChemCompProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Unlike the <a href="org/biojava/nbio/structure/chem/DownloadChemCompProvider.html" title="class in org.biojava.nbio.structure.chem"><code>DownloadChemCompProvider</code></a>, this  <a href="org/biojava/nbio/structure/chem/ChemCompProvider.html" title="interface in org.biojava.nbio.structure.chem"><code>ChemCompProvider</code></a> does not download any chem comp
 definitions.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/reference/ReferenceInterface.html" title="interface in org.biojava.nbio.core.sequence.reference">ReferenceInterface</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/multiple/util/ReferenceSuperimposer.html" title="class in org.biojava.nbio.structure.align.multiple.util">ReferenceSuperimposer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Superimposes each structure in a <a href="org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> onto a reference
 structure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/structure/symmetry/internal/RefinerFailedException.html" title="class in org.biojava.nbio.structure.symmetry.internal">RefinerFailedException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">Refinement of the self-alignment failed.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ws/hmmer/RemoteHmmerScan.html" title="class in org.biojava.nbio.ws.hmmer">RemoteHmmerScan</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Makes remote calls to the HMMER web service at the EBI web site and returns Pfam domain annotations for an input protein sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentOutputProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentOutputProperties</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">RemotePairwiseAlignmentOutputProperties: the simplest representation of an object capable of holding
 output formatting informations to be fed to a RemotePairwiseAlignmentService-implemented object.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentProperties.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentProperties</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">RemotePairwiseAlignmentProperties is a interface that contains the barest of
 methods for setting and getting Alignment properties.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ws/alignment/RemotePairwiseAlignmentService.html" title="interface in org.biojava.nbio.ws.alignment">RemotePairwiseAlignmentService</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">This interface specifies minimal information needed to execute a pairwise alignment on a remote service.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/RemoteTerm.html" title="interface in org.biojava.nbio.ontology">RemoteTerm</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A term in another ontology.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/RemoteTerm.Impl.html" title="class in org.biojava.nbio.ontology">RemoteTerm.Impl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Simple in-memory implementation of a remote ontology term.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/gui/RenderStyle.html" title="enum in org.biojava.nbio.structure.gui">RenderStyle</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/RescoreRefiner.html" title="interface in org.biojava.nbio.alignment.template">RescoreRefiner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an algorithm which computes a new alignment profile by rescoring all pairs in an alignment and
 realigning.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/ResidualsCoxph.html" title="class in org.biojava.nbio.survival.cox">ResidualsCoxph</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/survival/cox/ResidualsCoxph.Type.html" title="enum in org.biojava.nbio.survival.cox">ResidualsCoxph.Type</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/ResidueGroup.html" title="class in org.biojava.nbio.structure.symmetry.internal">ResidueGroup</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A ResidueGroup is a set of residues that are part of a maximally connected
 component of the self-Alignment Graph in symmetry analysis.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Everything that is needed to uniquely describe a residue position</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ResidueRange.html" title="class in org.biojava.nbio.structure">ResidueRange</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A chainName, a start residue, and an end residue.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/ResidueRangeAndLength.html" title="class in org.biojava.nbio.structure">ResidueRangeAndLength</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A chain, a start residue, and an end residue.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/chem/ResidueType.html" title="enum in org.biojava.nbio.structure.chem">ResidueType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Enumerates the possible classifications of residues.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/ResourceManager.html" title="class in org.biojava.nbio.structure.align.util">ResourceManager</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that manages the Strings that are defined in the spice.properties file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/Result.html" title="class in org.biojava.nbio.core.search.io">Result</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class models a search result.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/search/io/ResultFactory.html" title="interface in org.biojava.nbio.core.search.io">ResultFactory</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Designed by Paolo Pavan.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/views/ReversedSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">ReversedSequenceView&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">For a given sequence this class will return the base at the reversed
 position i.e. in a sequence of size 10, if you request base 2 you will get
 back the base at position 9.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/RiskInfo.html" title="class in org.biojava.nbio.survival.cox">RiskInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/compound/RNACompoundSet.html" title="class in org.biojava.nbio.core.sequence.compound">RNACompoundSet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence">RNASequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">RNASequence where RNACompoundSet are the allowed values</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/io/RNASequenceCreator.html" title="class in org.biojava.nbio.core.sequence.io">RNASequenceCreator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to create a RNA sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/views/RnaSequenceView.html" title="class in org.biojava.nbio.core.sequence.views">RnaSequenceView</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Attempts to do on the fly translation of RNA by not requesting the compounds
 until asked.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/transcription/RNAToAminoAcidTranslator.html" title="class in org.biojava.nbio.core.sequence.transcription">RNAToAminoAcidTranslator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Takes a <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> of <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>NucleotideCompound</code></a> which should represent
 an RNA sequence (<a href="org/biojava/nbio/core/sequence/RNASequence.html" title="class in org.biojava.nbio.core.sequence"><code>RNASequence</code></a> is good for this) and returns a list of
 <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> which hold <a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>AminoAcidCompound</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/RonnConstraint.html" title="class in org.biojava.nbio.ronn">RonnConstraint</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A collection of various constrain values used by RONN</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/ronn/RonnConstraint.Threshold.html" title="enum in org.biojava.nbio.ronn">RonnConstraint.Threshold</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/Rotation.html" title="class in org.biojava.nbio.structure.symmetry.core">Rotation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/RotationAxis.html" title="class in org.biojava.nbio.structure.align.util">RotationAxis</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Calculates the rotation axis for an alignment</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/axis/RotationAxisAligner.html" title="class in org.biojava.nbio.structure.symmetry.axis">RotationAxisAligner</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/RotationGroup.html" title="class in org.biojava.nbio.structure.symmetry.core">RotationGroup</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">See <a href="http://en.wikipedia.org/wiki/Rotation_group_SO(3)">http://en.wikipedia.org/wiki/Rotation_group_SO(3)</a></div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/RotationSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">RotationSolver</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/SangerFastqReader.html" title="class in org.biojava.nbio.genome.io.fastq">SangerFastqReader</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Reader for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SANGER"><code>FastqVariant.FASTQ_SANGER</code></a> formatted sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/SangerFastqWriter.html" title="class in org.biojava.nbio.genome.io.fastq">SangerFastqWriter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Writer for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SANGER"><code>FastqVariant.FASTQ_SANGER</code></a> formatted sequences.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/ScaleableMatrixPanel.html" title="class in org.biojava.nbio.structure.gui">ScaleableMatrixPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JPanel that can display the underlying distance matrix
 data of the protein structure alignment algorithm.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/matrices/ScaledSubstitutionMatrix.html" title="class in org.biojava.nbio.core.alignment.matrices">ScaledSubstitutionMatrix</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The biojava-alignment module represents substitution matrices with short
 values.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/xml/SchemaGenerator.html" title="class in org.biojava.nbio.aaproperties.xml">SchemaGenerator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/autosuggest/SCOPAutoSuggestProvider.html" title="class in org.biojava.nbio.structure.align.gui.autosuggest">SCOPAutoSuggestProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/scop/ScopCategory.html" title="enum in org.biojava.nbio.structure.scop">ScopCategory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">The various categories provided by SCOP.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/scop/ScopDatabase.html" title="interface in org.biojava.nbio.structure.scop">ScopDatabase</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">General API how to interact with SCOP</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/ScopDescription.html" title="class in org.biojava.nbio.structure.scop">ScopDescription</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains data from
 dir.des.scop.txt_1.75</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/ScopDomain.html" title="class in org.biojava.nbio.structure.scop">ScopDomain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Container for the information for a domain.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/ScopFactory.html" title="class in org.biojava.nbio.structure.scop">ScopFactory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Controls the global ScopDatabase being used.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/ScopInstallation.html" title="class in org.biojava.nbio.structure.scop">ScopInstallation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class provides access to the SCOP protein structure classification.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/ScopInstallationInstance.html" title="class in org.biojava.nbio.structure.gui.util">ScopInstallationInstance</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/structure/scop/ScopIOException.html" title="class in org.biojava.nbio.structure.scop">ScopIOException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">Indicates that an I/O error occurred with SCOP lazy initialization.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/ScopMirror.html" title="class in org.biojava.nbio.structure.scop">ScopMirror</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Helper class to store paths to the four SCOP files

 The string "%s" is replaced with the version number.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/scop/ScopNode.html" title="class in org.biojava.nbio.structure.scop">ScopNode</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/ScopSelectPanel.html" title="class in org.biojava.nbio.structure.gui.util">ScopSelectPanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/alignment/template/Scorer.html" title="interface in org.biojava.nbio.alignment.template">Scorer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines an algorithm which computes a score.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/multiple/ScoresCache.html" title="interface in org.biojava.nbio.structure.align.multiple">ScoresCache</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for classes which implement a temporary cache for various numeric
 scores, e.g.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/search/io/SearchIO.html" title="class in org.biojava.nbio.core.search.io">SearchIO</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Designed by Paolo Pavan.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/SecStrucCalc.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucCalc</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Calculate and assign the secondary structure (SS) to the
 Groups of a Structure object.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/SecStrucElement.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucElement</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A secondary structure element (SSE) is an object representing a block of
 sequential residues that share the same secondary structure type.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/SecStrucInfo.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucInfo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Container for the secondary structure information of a single residue.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/SecStrucState.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucState</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class extends the basic container for secondary structure annotation,
 including all the information used in the DSSP algorithm.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/secstruc/SecStrucTools.html" title="class in org.biojava.nbio.structure.secstruc">SecStrucTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class contains methods for obtaining and converting secondary structure
 information from BioJava <a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a>s.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/secstruc/SecStrucType.html" title="enum in org.biojava.nbio.structure.secstruc">SecStrucType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">This enum contains all of the secondary structure types found in the DSSP
 output.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/Segment.html" title="class in org.biojava.nbio.structure.domain.pdp">Segment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/SegmentComparator.html" title="class in org.biojava.nbio.structure.domain.pdp">SegmentComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/gui/Selection.html" title="interface in org.biojava.nbio.structure.gui">Selection</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/SelectionImpl.html" title="class in org.biojava.nbio.structure.gui">SelectionImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/SelectMultiplePanel.html" title="class in org.biojava.nbio.structure.gui.util">SelectMultiplePanel</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Text Panel that allows the user to specify multiple structure
 identifiers, space separated.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/SelectPDBPanel.html" title="class in org.biojava.nbio.structure.align.gui">SelectPDBPanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Panel that allows user to specify PDB and chain ID, as well as sub-ranges</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/SeqMisMatch.html" title="interface in org.biojava.nbio.structure">SeqMisMatch</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Created by andreas on 9/11/15.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/SeqMisMatchImpl.html" title="class in org.biojava.nbio.structure">SeqMisMatchImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by andreas on 9/11/15.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/SeqRes2AtomAligner.html" title="class in org.biojava.nbio.structure.io">SeqRes2AtomAligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Aligns the SEQRES residues to the ATOM residues.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Main interface for defining a collection of Compounds and accessing them
 using biological indexes</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/SequenceAsStringHelper.html" title="class in org.biojava.nbio.core.sequence.storage">SequenceAsStringHelper&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is a common method that can be used across multiple storage/proxy implementations to
 handle Negative strand and other interesting elements of sequence data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/SequenceComparator.html" title="class in org.biojava.nbio.core.sequence">SequenceComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to sort sequences in ascending order of  bioBegin property.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/io/template/SequenceCreatorInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceCreatorInterface&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/SequenceDisplay.html" title="class in org.biojava.nbio.structure.gui">SequenceDisplay</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sequence display that can show the results of a protein structure alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/loader/SequenceFileProxyLoader.html" title="class in org.biojava.nbio.core.sequence.loader">SequenceFileProxyLoader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class represents the storage container of a sequence stored in a fasta file where
 the initial parsing of the file we store the offset and length of the sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/SequenceFunctionOrderDetector.html" title="class in org.biojava.nbio.structure.symmetry.internal">SequenceFunctionOrderDetector</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Calls Spencer's method for determining order.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html" title="class in org.biojava.nbio.structure.symmetry.internal">SequenceFunctionRefiner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Creates a refined alignment with the CE-Symm alternative self-alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/io/template/SequenceHeaderParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceHeaderParserInterface&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/SequenceLocation.html" title="class in org.biojava.nbio.core.sequence.location">SequenceLocation&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A location in a sequence that keeps a reference to its parent sequence</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/SequenceMixin.html" title="class in org.biojava.nbio.core.sequence.template">SequenceMixin</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides a set of static methods to be used as static imports when needed
 across multiple Sequence implementations but inheritance gets in the way.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/SequenceMixin.SequenceIterator.html" title="class in org.biojava.nbio.core.sequence.template">SequenceMixin.SequenceIterator&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A basic sequence iterator which iterates over the given Sequence by
 biological index.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/SequenceMouseListener.html" title="class in org.biojava.nbio.structure.gui.util">SequenceMouseListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">a mouse listener for the AbstractChainRenderer class
 it listens to all mouse events and triggeres appropriate
 SequenceListener and FeatureListener events</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/SequenceOptimizationHints.html" title="class in org.biojava.nbio.core.sequence">SequenceOptimizationHints</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A static class that provides optimization hints for memory or performance handling of sequence data.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/SequenceOptimizationHints.SequenceCollection.html" title="enum in org.biojava.nbio.core.sequence">SequenceOptimizationHints.SequenceCollection</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/SequenceOptimizationHints.SequenceUsage.html" title="enum in org.biojava.nbio.core.sequence">SequenceOptimizationHints.SequenceUsage</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure for the results of pairwise sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/io/template/SequenceParserInterface.html" title="interface in org.biojava.nbio.core.sequence.io.template">SequenceParserInterface</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/template/SequenceProxyView.html" title="class in org.biojava.nbio.core.sequence.template">SequenceProxyView&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/SequenceReader.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceReader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/SequenceScalePanel.html" title="class in org.biojava.nbio.structure.gui.util">SequenceScalePanel</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that draws a Sequence as a rectangle, a scale display over it.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/SequenceTools.html" title="class in org.biojava.nbio.core.util">SequenceTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/data/sequence/SequenceUtil.html" title="class in org.biojava.nbio.data.sequence">SequenceUtil</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility class for operations on sequences</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/template/SequenceView.html" title="interface in org.biojava.nbio.core.sequence.template">SequenceView&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/SerializableCache.html" title="class in org.biojava.nbio.structure.domain">SerializableCache&lt;K,<wbr>V&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides all that is necessary to create a Serializable Cache</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/domain/pdp/ShortSegmentRemover.html" title="class in org.biojava.nbio.structure.domain.pdp">ShortSegmentRemover</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/ShowPDBIDListener.html" title="class in org.biojava.nbio.structure.align.gui">ShowPDBIDListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/ShowStructureInJmol.html" title="class in demo">ShowStructureInJmol</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Demo how to load and display a structure in Jmol</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/sifts/SiftsChainEntry.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsChainEntry</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An entry in the chain-level SIFTS mapping between UniProt and the PDB.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/sifts/SiftsChainToUniprotMapping.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsChainToUniprotMapping</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A mapping between UniProt entries and PDB chains.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/sifts/SiftsEntity.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsEntity</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/sifts/SiftsMappingProvider.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsMappingProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/sifts/SiftsResidue.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsResidue</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/sifts/SiftsSegment.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsSegment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/sifts/SiftsXMLParser.html" title="class in org.biojava.nbio.structure.io.sifts">SiftsXMLParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/SigEva.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">SigEva</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/SimpleAlignedSequence.html" title="class in org.biojava.nbio.core.alignment">SimpleAlignedSequence&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a data structure for a <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> within an alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/SimpleGapPenalty.html" title="class in org.biojava.nbio.alignment">SimpleGapPenalty</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a data structure for the gap penalties used during a sequence alignment routine.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/SimpleLocation.html" title="class in org.biojava.nbio.core.sequence.location">SimpleLocation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Very basic implementation of the Location interface which defines a series
 of simple constructors.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/location/SimplePoint.html" title="class in org.biojava.nbio.core.sequence.location">SimplePoint</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Basic implementation of the Point interface.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/SimpleProfile.html" title="class in org.biojava.nbio.core.alignment">SimpleProfile&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a data structure for the results of sequence alignment.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/SimpleProfilePair.html" title="class in org.biojava.nbio.core.alignment">SimpleProfilePair&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a data structure for the results of the alignment of a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/SimpleProfileProfileAligner.html" title="class in org.biojava.nbio.alignment">SimpleProfileProfileAligner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a simple (naive) <a href="org/biojava/nbio/alignment/template/Aligner.html" title="interface in org.biojava.nbio.alignment.template"><code>Aligner</code></a> for a pair of <a href="org/biojava/nbio/core/alignment/template/Profile.html" title="interface in org.biojava.nbio.core.alignment.template"><code>Profile</code></a>s.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/SimpleSequencePair.html" title="class in org.biojava.nbio.core.alignment">SimpleSequencePair&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a data structure for the results of pairwise sequence alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/matrices/SimpleSubstitutionMatrix.html" title="class in org.biojava.nbio.core.alignment.matrices">SimpleSubstitutionMatrix&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implements a data structure which holds the score (penalty or bonus) given during alignment for the exchange of one
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> in a sequence for another.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/SingleCompoundSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">SingleCompoundSequenceReader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An implementation of the SequenceReader interface which for every
 call will return only 1 compound (given to it during construction; a String
 is also valid but will require a CompoundSet).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/SingleLinkageClusterer.html" title="class in org.biojava.nbio.core.util">SingleLinkageClusterer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An implementation of a single linkage clusterer

 See http://en.wikipedia.org/wiki/Single-linkage_clustering</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/jama/SingularValueDecomposition.html" title="class in org.biojava.nbio.structure.jama">SingularValueDecomposition</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Singular Value Decomposition.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/phosphosite/Site.html" title="class in org.biojava.nbio.phosphosite">Site</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Created by ap3 on 31/10/2014.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/Site.html" title="class in org.biojava.nbio.structure">Site</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Holds the data of sites presented in PDB files.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/utils/SmallAnnotation.html" title="class in org.biojava.nbio.ontology.utils">SmallAnnotation</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Annotation that is optimized for memory usage.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/utils/SmallMap.html" title="class in org.biojava.nbio.ontology.utils">SmallMap</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Lightweight implementation of Map which uses little memory to store a
 small number of mappings, at the expense of scalability.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/SmithWaterman.html" title="class in org.biojava.nbio.alignment">SmithWaterman&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Smith and Waterman defined an algorithm for pairwise local sequence alignments (best match of sections from each
 <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a>).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html" title="class in org.biojava.nbio.structure.align.seq">SmithWaterman3Daligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Provides a 3D superimposition of two structures based on their sequence
 alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/seq/SmithWaterman3DParameters.html" title="class in org.biojava.nbio.structure.align.seq">SmithWaterman3DParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/seq/SmithWatermanUserArgumentProcessor.html" title="class in org.biojava.nbio.structure.align.seq">SmithWatermanUserArgumentProcessor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/seq/SmithWatermanUserArgumentProcessor.SmithWatermanStartupParams.html" title="class in org.biojava.nbio.structure.align.seq">SmithWatermanUserArgumentProcessor.SmithWatermanStartupParams</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/SoftHashMap.html" title="class in org.biojava.nbio.core.util">SoftHashMap&lt;K,<wbr>V&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An in memory cache using soft references.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/SolexaFastqReader.html" title="class in org.biojava.nbio.genome.io.fastq">SolexaFastqReader</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Reader for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SOLEXA"><code>FastqVariant.FASTQ_SOLEXA</code></a> formatted sequences.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/io/fastq/SolexaFastqWriter.html" title="class in org.biojava.nbio.genome.io.fastq">SolexaFastqWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Writer for <a href="org/biojava/nbio/genome/io/fastq/FastqVariant.html#FASTQ_SOLEXA"><code>FastqVariant.FASTQ_SOLEXA</code></a> formatted sequences.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/SpaceGroup.html" title="class in org.biojava.nbio.structure.xtal">SpaceGroup</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A crystallographic space group.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/io/SpaceGroupMapElements.html" title="class in org.biojava.nbio.structure.xtal.io">SpaceGroupMapElements</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/io/SpaceGroupMapRoot.html" title="class in org.biojava.nbio.structure.xtal.io">SpaceGroupMapRoot</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/math/SparseSquareMatrix.html" title="class in org.biojava.nbio.structure.math">SparseSquareMatrix</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sparse, square matrix, implementing using two arrays of sparse
  vectors, one representation for the rows and one for the columns.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/math/SparseVector.html" title="class in org.biojava.nbio.structure.math">SparseVector</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sparse vector, implemented using a symbol table.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/SphereSampler.html" title="class in org.biojava.nbio.structure.symmetry.geometry">SphereSampler</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Sample possible orientations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/util/SplitFasta.html" title="class in org.biojava.nbio.genome.util">SplitFasta</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility to write each Fasta entry to a unique file</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/util/color/SqrtColorMapper.html" title="class in org.biojava.nbio.structure.gui.util.color">SqrtColorMapper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Performs a sqrt transform on input before passing the values off to another
 colorMapper.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/SSBondImpl.html" title="class in org.biojava.nbio.structure.io">SSBondImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A simple bean to store disulfide bridge information, the SSBOND records in the PDB files.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/genome/parsers/cytoband/StainType.html" title="enum in org.biojava.nbio.genome.parsers.cytoband">StainType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/StandardAminoAcid.html" title="class in org.biojava.nbio.structure">StandardAminoAcid</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides a set of standard amino acids.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/StandardRescoreRefiner.html" title="class in org.biojava.nbio.alignment">StandardRescoreRefiner&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/StartCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StartCodonSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to map the start codon feature on a gene</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/ce/StartupParameters.html" title="class in org.biojava.nbio.structure.align.ce">StartupParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A simple bean that contains the parameters that can get set at startup</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/utils/StaticMemberPlaceHolder.html" title="class in org.biojava.nbio.ontology.utils">StaticMemberPlaceHolder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html" title="class in org.biojava.nbio.structure.align.gui.aligpanel">StatusDisplay</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/matrix/StdArrayIO.html" title="class in org.biojava.nbio.survival.cox.matrix">StdArrayIO</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block"><i>Standard array IO</i>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/io/StockholmFileAnnotation.html" title="class in org.biojava.nbio.alignment.io">StockholmFileAnnotation</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores all the content parsed from the #=GF lines</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/io/StockholmFileAnnotation.StockholmFileAnnotationReference.html" title="class in org.biojava.nbio.alignment.io">StockholmFileAnnotation.StockholmFileAnnotationReference</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/io/StockholmFileParser.html" title="class in org.biojava.nbio.alignment.io">StockholmFileParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stockholm file parser.<br>
 for more information about the format refer to
 
 <a href="ftp://ftp.sanger.ac.uk/pub/databases/Pfam/current_release/userman.txt">ftp://ftp.sanger.ac.uk/pub/databases
 /Pfam/current_release/userman.txt</a>.
 <a href="ftp://ftp.sanger.ac.uk/pub/databases/Rfam/CURRENT/USERMAN">ftp://ftp.sanger.ac.uk/pub/databases/Rfam/CURRENT
 /USERMAN</a>.
 <a href="http://sonnhammer.sbc.su.se/Stockholm.html">http://sonnhammer.sbc.su.se/Stockholm.html</a>.
 </div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/io/StockholmStructure.html" title="class in org.biojava.nbio.alignment.io">StockholmStructure</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Stores all the content of a Stockholm file.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/io/StockholmStructure.DatabaseReference.html" title="class in org.biojava.nbio.alignment.io">StockholmStructure.DatabaseReference</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/Stoichiometry.html" title="class in org.biojava.nbio.structure.symmetry.core">Stoichiometry</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility object that describes Stoichiometry (composition of a protein assembly),
 determined via clustering procedure <a href="org/biojava/nbio/structure/cluster/SubunitClusterer.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitClusterer</code></a>,
 and implements human-readable representation using various strategies.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/symmetry/core/Stoichiometry.StringOverflowStrategy.html" title="enum in org.biojava.nbio.structure.symmetry.core">Stoichiometry.StringOverflowStrategy</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">What to do when the number of <a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitCluster</code></a> exceeds the length of the alphabet.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/StopCodonSequence.html" title="class in org.biojava.nbio.core.sequence">StopCodonSequence</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to map the stop codon sequence on a gene</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/core/sequence/Strand.html" title="enum in org.biojava.nbio.core.sequence">Strand</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">Provides a way of representing the strand of a sequence, location
 hit or feature.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/StrataInfo.html" title="class in org.biojava.nbio.survival.cox">StrataInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Information needed to represent a survival curve</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/pairwise/StrCompAlignment.html" title="class in org.biojava.nbio.structure.align.pairwise">StrCompAlignment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/genome/io/fastq/StreamListener.html" title="interface in org.biojava.nbio.genome.io.fastq">StreamListener</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Event based parser callback.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/StringManipulationHelper.html" title="class in org.biojava.nbio.core.util">StringManipulationHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class for common <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link"><code>String</code></a> manipulation tasks.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/loader/StringProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">StringProxySequenceReader&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">An example of a ProxySequenceReader that is created from a String.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/StrucAligParameters.html" title="class in org.biojava.nbio.structure.align">StrucAligParameters</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that contains all the parameters of the structure alignment algorithm.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for a structure object.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/StructureAlignment.html" title="interface in org.biojava.nbio.structure.align">StructureAlignment</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html" title="class in org.biojava.nbio.structure.align.gui">StructureAlignmentDisplay</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/StructureAlignmentFactory.html" title="class in org.biojava.nbio.structure.align">StructureAlignmentFactory</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides a simple GUI for Jmol</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/fatcat/calc/StructureAlignmentOptimizer.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">StructureAlignmentOptimizer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/structure/StructureAtom.html" title="class in org.biojava.nbio.protmod.structure">StructureAtom</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Everything that is needed to uniquely describe a atom.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/structure/StructureAtomLinkage.html" title="class in org.biojava.nbio.protmod.structure">StructureAtomLinkage</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/io/StructureAtomXMLConverter.html" title="class in org.biojava.nbio.protmod.io">StructureAtomXMLConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/structure/StructureException.html" title="class in org.biojava.nbio.structure">StructureException</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab5">
<div class="block">An exception during the parsing of a PDB file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/io/StructureFiletype.html" title="enum in org.biojava.nbio.structure.io">StructureFiletype</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">An enum of supported file formats.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/structure/StructureGroup.html" title="class in org.biojava.nbio.protmod.structure">StructureGroup</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Information of a group (residue or ligand) involved in a modification.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/io/StructureGroupXMLConverter.html" title="class in org.biojava.nbio.protmod.io">StructureGroupXMLConverter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/StructureIdentifier.html" title="interface in org.biojava.nbio.structure">StructureIdentifier</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">An identifier that <em>uniquely</em> identifies a whole <a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> or
 arbitrary substructure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/StructureImpl.html" title="class in org.biojava.nbio.structure">StructureImpl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation of a PDB Structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/StructureInterface.html" title="class in org.biojava.nbio.structure.contact">StructureInterface</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">An interface between 2 molecules (2 sets of atoms).</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/StructureInterfaceCluster.html" title="class in org.biojava.nbio.structure.contact">StructureInterfaceCluster</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/contact/StructureInterfaceList.html" title="class in org.biojava.nbio.structure.contact">StructureInterfaceList</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A list of interfaces between 2 molecules (2 sets of atoms)</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/StructureIO.html" title="class in org.biojava.nbio.structure">StructureIO</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides static access methods for easy lookup of protein structure related components</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/StructureIOFile.html" title="interface in org.biojava.nbio.structure.io">StructureIOFile</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">StructureIOFile extends StructureProvider with methods specific to
  parsing files from the filesystem.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/StructureLoaderThread.html" title="class in org.biojava.nbio.structure.align.gui">StructureLoaderThread</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/client/StructureName.html" title="class in org.biojava.nbio.structure.align.client">StructureName</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class that makes working with names of structures, domains and ranges easier.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/align/client/StructureName.Source.html" title="enum in org.biojava.nbio.structure.align.client">StructureName.Source</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/StructurePairAligner.html" title="class in org.biojava.nbio.structure.align">StructurePairAligner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Perform a pairwise protein structure superimposition.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/gui/util/StructurePairSelector.html" title="interface in org.biojava.nbio.structure.gui.util">StructurePairSelector</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">To be implemented by JPanels that are part of the GUI to trigger structure aligmnents.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/io/StructureProvider.html" title="interface in org.biojava.nbio.structure.io">StructureProvider</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A class that can provide a protein structure object from somewhere.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/io/StructureSequenceMatcher.html" title="class in org.biojava.nbio.structure.io">StructureSequenceMatcher</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A utility class with methods for matching ProteinSequences with
 Structures.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/StructureTools.html" title="class in org.biojava.nbio.structure">StructureTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that provides some tool methods.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/protmod/structure/StructureUtil.html" title="class in org.biojava.nbio.protmod.structure">StructureUtil</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/gui/StructureViewer.html" title="interface in org.biojava.nbio.structure.gui">StructureViewer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/alignment/template/SubstitutionMatrix.html" title="interface in org.biojava.nbio.core.alignment.template">SubstitutionMatrix&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Defines a data structure which holds the score (penalty or bonus) given during alignment for the exchange of one
 <a href="org/biojava/nbio/core/sequence/template/Compound.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Compound</code></a> in a sequence for another.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/alignment/matrices/SubstitutionMatrixHelper.html" title="class in org.biojava.nbio.core.alignment.matrices">SubstitutionMatrixHelper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Static utility to access substitution matrices that come bundled with BioJava.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/alignment/SubstitutionMatrixScorer.html" title="class in org.biojava.nbio.alignment">SubstitutionMatrixScorer&lt;S,<wbr>C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Scores using a substitution matrix.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/SubstructureIdentifier.html" title="class in org.biojava.nbio.structure">SubstructureIdentifier</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is the canonical way to identify a part of a structure.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster">Subunit</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A Subunit consists of a set of residues from a Structure, which may
 correspond to an entire Chain, a Domain, or any subset or combination of
 residues from them.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster">SubunitCluster</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A SubunitCluster contains a set of equivalent <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetrySubunits.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetrySubunits</code></a>,
 the set of equivalent residues (EQR) between <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> and a
 <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> representative.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cluster/SubunitClusterer.html" title="class in org.biojava.nbio.structure.cluster">SubunitClusterer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The SubunitClusterer takes as input a collection of <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> and
 returns a collection of <a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitCluster</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/cluster/SubunitClustererMethod.html" title="enum in org.biojava.nbio.structure.cluster">SubunitClustererMethod</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">
<div class="block">The SubunitClustererMethod ennummerates all methods that can be used to
 cluster <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a> in the <a href="org/biojava/nbio/structure/cluster/SubunitCluster.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitCluster</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cluster/SubunitClustererParameters.html" title="class in org.biojava.nbio.structure.cluster">SubunitClustererParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">The SubunitClustererParameters specifies the options used for the clustering
 of the subunits in structures using the <a href="org/biojava/nbio/structure/cluster/SubunitClusterer.html" title="class in org.biojava.nbio.structure.cluster"><code>SubunitClusterer</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/cluster/SubunitExtractor.html" title="class in org.biojava.nbio.structure.cluster">SubunitExtractor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The SubunitExtractor extracts the information of each protein <a href="org/biojava/nbio/structure/Chain.html" title="interface in org.biojava.nbio.structure"><code>Chain</code></a>
 in a <a href="org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure"><code>Structure</code></a> and converts them into a List of <a href="org/biojava/nbio/structure/cluster/Subunit.html" title="class in org.biojava.nbio.structure.cluster"><code>Subunit</code></a>.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/geometry/SuperPosition.html" title="interface in org.biojava.nbio.structure.geometry">SuperPosition</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">The SuperPosition interface defines and documents the required methods for
 any superpostion algorithm implementation, so that the input and expected
 output are uniform.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/SuperPositionAbstract.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionAbstract</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The SuperPositionAbstract contains common code shared by all SuperPosition
 algorithm implementations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/SuperPositionQCP.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionQCP</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation of the Quaternion-Based Characteristic Polynomial algorithm
 for RMSD and Superposition calculations.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/SuperPositionQuat.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionQuat</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The SuperPositionQuat implements a quaternion based algorithm to superpose
 arrays of Points in 3D.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/SuperPositions.html" title="class in org.biojava.nbio.structure.geometry">SuperPositions</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">SuperPositions is a Class that provides static helper methods and an easy
 access to the whole family of <a href="org/biojava/nbio/structure/geometry/SuperPosition.html" title="interface in org.biojava.nbio.structure.geometry"><code>SuperPosition</code></a> algorithms.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/SuperPositionSVD.html" title="class in org.biojava.nbio.structure.geometry">SuperPositionSVD</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that calculates the superposition between two sets of points using an
 SVD Matrix Decomposition.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/SurvFitInfo.html" title="class in org.biojava.nbio.survival.cox">SurvFitInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Contains info for graphing km figures</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.html" title="class in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Ported from survfitKM.S When combining multiple entries with same time not
 sure how the weighting adds up</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.ConfLower.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.ConfLower</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.ConfType.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.ConfType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.Error.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.Error</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/survival/kaplanmeier/figure/SurvFitKM.Method.html" title="enum in org.biojava.nbio.survival.kaplanmeier.figure">SurvFitKM.Method</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/SurvivalInfo.html" title="class in org.biojava.nbio.survival.cox">SurvivalInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Data class to represent a single sample where time and event/censor status is required
 Additionally each variable and data associated with that variable.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/comparators/SurvivalInfoComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">SurvivalInfoComparator</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/SurvivalInfoHelper.html" title="class in org.biojava.nbio.survival.cox">SurvivalInfoHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used to work with SurvivalInfo</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/SurvivalInfoIndex.html" title="class in org.biojava.nbio.survival.cox">SurvivalInfoIndex</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Not used and probably should be deleted</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/comparators/SurvivalInfoValueComparator.html" title="class in org.biojava.nbio.survival.cox.comparators">SurvivalInfoValueComparator</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/math/SymbolTable.html" title="class in org.biojava.nbio.structure.math">SymbolTable&lt;Key,<wbr>Value&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Sorted symbol table implementation using a java.util.TreeMap.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Data Structure that stores all the symmetry axis that describe
 the symmetry of a structure.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/SymmetryAxes.Axis.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes.Axis</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents an axis of symmetry</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryCalc.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryCalc</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Calculates a symmetry analysis and displays the results.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryDisplay.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryDisplay</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class that provides visualizations methods for symmetry alignments.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryGui.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryGui</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A JFrame that allows to trigger a symmetry analysis, either from files
 in a directory or after manual upload
 Adapted from the AlignmentGui class in biojava.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/gui/SymmetryListener.html" title="class in org.biojava.nbio.structure.symmetry.gui">SymmetryListener</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Action Listener for the symmetry menu.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/symmetry/core/SymmetryPerceptionMethod.html" title="enum in org.biojava.nbio.structure.symmetry.core">SymmetryPerceptionMethod</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">Method used for symmetry perception in the <a href="org/biojava/nbio/structure/symmetry/core/QuatSymmetryDetector.html" title="class in org.biojava.nbio.structure.symmetry.core"><code>QuatSymmetryDetector</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/symmetry/internal/SymmetryRefiner.html" title="interface in org.biojava.nbio.structure.symmetry.internal">SymmetryRefiner</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Interface for all symmetry refinement implementations.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html" title="class in org.biojava.nbio.structure.symmetry.utils">SymmetryTools</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Utility methods for symmetry (quaternary and internal) detection and result
 manipulation.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/internal/SymmOptimizer.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmOptimizer</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Optimizes a symmetry alignment by a Monte Carlo score optimization of the
 repeat multiple alignment.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/SymoplibParser.html" title="class in org.biojava.nbio.structure.xtal">SymoplibParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class containing static methods to parse the symop.lib file from the
 CCP4 package.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/SynchronizedOutFile.html" title="class in org.biojava.nbio.structure.align.util">SynchronizedOutFile</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/Synonym.html" title="class in org.biojava.nbio.ontology">Synonym</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/core/SystematicSolver.html" title="class in org.biojava.nbio.structure.symmetry.core">SystematicSolver</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/gui/SystemInfo.html" title="class in org.biojava.nbio.structure.align.gui">SystemInfo</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/io/TabDelimParser.html" title="class in org.biojava.nbio.ontology.io">TabDelimParser</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Parse tab-delimited ontology files into Ontology objects.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/sequence/transcription/Table.html" title="interface in org.biojava.nbio.core.sequence.transcription">Table</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">Provides a way of separating us from the specific <a href="org/biojava/nbio/core/sequence/io/IUPACParser.IUPACTable.html" title="class in org.biojava.nbio.core.sequence.io"><code>IUPACParser.IUPACTable</code></a> even
 though this is the only implementing class for the interface.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/transcription/Table.CaseInsensitiveTriplet.html" title="class in org.biojava.nbio.core.sequence.transcription">Table.CaseInsensitiveTriplet</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Class used to hold three nucleotides together and allow for equality
 to be assessed in a case insensitive manner.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/transcription/Table.Codon.html" title="class in org.biojava.nbio.core.sequence.transcription">Table.Codon</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Instance of a Codon which is 3 <a href="org/biojava/nbio/core/sequence/compound/NucleotideCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>NucleotideCompound</code></a>s, its
 corresponding <a href="org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound"><code>AminoAcidCompound</code></a> and if it is a start or stop codon.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/TaxonomyID.html" title="class in org.biojava.nbio.core.sequence">TaxonomyID</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sequence can be associated with a species or Taxonomy ID</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/Term.html" title="interface in org.biojava.nbio.ontology">Term</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">
<div class="block">A term in an ontology.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/Term.Impl.html" title="class in org.biojava.nbio.ontology">Term.Impl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Simple in-memory implementation of an ontology term.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/basepairs/TertiaryBasePairParameters.html" title="class in org.biojava.nbio.structure.basepairs">TertiaryBasePairParameters</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class also finds the base pairing and base-pair step parameters but has a broader definition
 of a base pair so that non-canonical-WC base pairs will be detected and reported.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/TestDNANeedlemanWunsch.html" title="class in demo">TestDNANeedlemanWunsch</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/symmetry/geometry/Tetrahedron.html" title="class in org.biojava.nbio.structure.symmetry.geometry">Tetrahedron</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/features/TextFeature.html" title="class in org.biojava.nbio.core.sequence.features">TextFeature&lt;S,<wbr>C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A implmentation of AbstractFeature</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ronn/Timer.html" title="class in org.biojava.nbio.ronn">Timer</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A simple timer, calculates the time interval between two events.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Used as a way of encapsulating the data structures required to parse DNA to a
 Protein sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.Builder.html" title="class in org.biojava.nbio.core.sequence.transcription">TranscriptionEngine.Builder</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class is the way to create a <a href="org/biojava/nbio/core/sequence/transcription/TranscriptionEngine.html" title="class in org.biojava.nbio.core.sequence.transcription"><code>TranscriptionEngine</code></a>.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/TranscriptSequence.html" title="class in org.biojava.nbio.core.sequence">TranscriptSequence</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is the sequence if you want to go from a gene sequence to a protein sequence.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/io/TransfAlgebraicAdapter.html" title="class in org.biojava.nbio.structure.xtal.io">TransfAlgebraicAdapter</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/structure/xtal/TransformType.html" title="enum in org.biojava.nbio.structure.xtal">TransformType</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab3">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab5"><a href="org/biojava/nbio/core/exceptions/TranslationException.html" title="class in org.biojava.nbio.core.exceptions">TranslationException</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab5">
<div class="block">Thrown from AbstractCompundTranslator</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/phylo/TreeConstructor.html" title="class in org.biojava.nbio.phylo">TreeConstructor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">The TreeConstructor uses the forester library to build different types of
 phylogenetic trees.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/phylo/TreeConstructorType.html" title="enum in org.biojava.nbio.phylo">TreeConstructorType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">The TreeConstructorType specifies the aligorithm used to construct the tree
 (clustering algorithm).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/io/TreeMapSpaceGroupWrapper.html" title="class in org.biojava.nbio.structure.xtal.io">TreeMapSpaceGroupWrapper</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab3"><a href="org/biojava/nbio/phylo/TreeType.html" title="enum in org.biojava.nbio.phylo">TreeType</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab3">
<div class="block">The TreeType specifies the optimization criteria used to generate the tree.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/Triple.html" title="interface in org.biojava.nbio.ontology">Triple</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">A triple in an ontology.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/Triple.Impl.html" title="class in org.biojava.nbio.ontology">Triple.Impl</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Basic in-memory implementation of a Triple in an ontology

 This can be used to implement Ontology.createTriple</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/twobit/TwoBitFacade.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitFacade</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A facade that makes it easier to work with a 2bit file.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/parsers/twobit/TwoBitParser.html" title="class in org.biojava.nbio.genome.parsers.twobit">TwoBitParser</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">downloaded from http://storage.bioinf.fbb.msu.ru/~roman/TwoBitParser.java

 Class is a parser of UCSC Genome Browser file format .2bit used to store
 nucleotide sequence information.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.html" title="class in org.biojava.nbio.core.sequence.storage">TwoBitSequenceReader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Implementation of the 2bit encoding.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/storage/TwoBitSequenceReader.TwoBitArrayWorker.html" title="class in org.biojava.nbio.core.sequence.storage">TwoBitSequenceReader.TwoBitArrayWorker&lt;C&gt;</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Extension of the BitArrayWorker which provides the 2bit implementation
 code.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/UncompressFile.html" title="class in demo">UncompressFile</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Uncompresses a single tarred or zipped file, writing output to stdandard out</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/UncompressInputStream.html" title="class in org.biojava.nbio.core.util">UncompressInputStream</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This class decompresses an input stream containing data compressed with
 the unix "compress" utility (LZC, a LZW variant).</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.html" title="class in org.biojava.nbio.core.sequence.loader">UniprotProxySequenceReader&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">Pass in a Uniprot ID and this ProxySequenceReader when passed to a ProteinSequence will get the sequence data and other data elements
 associated with the ProteinSequence by Uniprot.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/genome/uniprot/UniprotToFasta.html" title="class in org.biojava.nbio.genome.uniprot">UniprotToFasta</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/xtal/UnitCellBoundingBox.html" title="class in org.biojava.nbio.structure.xtal">UnitCellBoundingBox</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class to contain the BoundingBoxes of all polymeric molecules in a full unit cell.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/geometry/UnitQuaternions.html" title="class in org.biojava.nbio.structure.geometry">UnitQuaternions</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">UnitQuaternions is a static Class that contains methods for calculating and
 using unit quaternions.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/URLConnectionTools.html" title="class in org.biojava.nbio.structure.align.util">URLConnectionTools</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A class that takes care about opening URLConnections and sets the proper timeouts</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/URLIdentifier.html" title="class in org.biojava.nbio.structure">URLIdentifier</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Represents a structure loaded from a URL (including a file URL)</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/structure/align/ce/UserArgumentProcessor.html" title="interface in org.biojava.nbio.structure.align.ce">UserArgumentProcessor</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/util/UserConfiguration.html" title="class in org.biojava.nbio.structure.align.util">UserConfiguration</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">A container to persist config to the file system</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/aaproperties/Utils.html" title="class in org.biojava.nbio.aaproperties">Utils</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">This is a utility class that contains utility methods which will facilitates the coding of other methods</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/ontology/Variable.html" title="interface in org.biojava.nbio.ontology">Variable</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab1">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/Variable.Impl.html" title="class in org.biojava.nbio.ontology">Variable.Impl</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/WaldTest.html" title="class in org.biojava.nbio.survival.cox">WaldTest</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/cox/WaldTestInfo.html" title="class in org.biojava.nbio.survival.cox">WaldTestInfo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/ontology/utils/WeakValueHashMap.html" title="class in org.biojava.nbio.ontology.utils">WeakValueHashMap</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Map implementation which keeps weak references to values.</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="demo/WebStartClientDemo.html" title="class in demo">WebStartClientDemo</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/align/webstart/WebStartMain.html" title="class in org.biojava.nbio.structure.align.webstart">WebStartMain</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">&nbsp;</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/sequence/views/WindowedSequence.html" title="class in org.biojava.nbio.core.sequence.views">WindowedSequence&lt;C&gt;</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">A sliding window view of a sequence which does not implement any
 interfaces like <a href="org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template"><code>Sequence</code></a> because they do not fit how this works.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/survival/data/WorkSheet.html" title="class in org.biojava.nbio.survival.data">WorkSheet</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Need to handle very large spreadsheets of expression data so keep memory
 footprint low</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/gui/WrapLayout.html" title="class in org.biojava.nbio.structure.gui">WrapLayout</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab2">
<div class="block">FlowLayout subclass that fully supports wrapping of components.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/core/util/XMLHelper.html" title="class in org.biojava.nbio.core.util">XMLHelper</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">Helper methods to simplify boilerplate XML parsing code for  org.w3c.dom XML objects</div>
</div>
<div class="col-first even-row-color all-classes-table all-classes-table-tab1"><a href="org/biojava/nbio/core/util/XMLWriter.html" title="interface in org.biojava.nbio.core.util">XMLWriter</a></div>
<div class="col-last even-row-color all-classes-table all-classes-table-tab1">
<div class="block">Simple interface for building XML documents.</div>
</div>
<div class="col-first odd-row-color all-classes-table all-classes-table-tab2"><a href="org/biojava/nbio/structure/chem/ZipChemCompProvider.html" title="class in org.biojava.nbio.structure.chem">ZipChemCompProvider</a></div>
<div class="col-last odd-row-color all-classes-table all-classes-table-tab2">
<div class="block">This chemical component provider retrieves and caches chemical component definition files from a
 zip archive specified in its construction.</div>
</div>
</div>
</div>
</div>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
