<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) on Wed Jan 08 16:43:01 PST 2025 -->
<title>Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment (biojava 7.1.4 API)</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="dc.created" content="2025-01-08">
<meta name="description" content="use: package: org.biojava.nbio.structure.align.multiple, interface: MultipleAlignment">
<meta name="generator" content="javadoc/ClassUseWriter">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
<link rel="stylesheet" type="text/css" href="../../../../../../../script-dir/jquery-ui.min.css" title="Style">
<script type="text/javascript" src="../../../../../../../script.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-3.7.1.min.js"></script>
<script type="text/javascript" src="../../../../../../../script-dir/jquery-ui.min.js"></script>
</head>
<body class="class-use-page">
<script type="text/javascript">var pathtoroot = "../../../../../../../";
loadScripts(document, 'script');</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<div class="flex-box">
<header role="banner" class="flex-header">
<nav role="navigation">
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="top-nav" id="navbar-top"><button id="navbar-toggle-button" aria-controls="navbar-top" aria-expanded="false" aria-label="Toggle navigation links"><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span><span class="nav-bar-toggle-icon">&nbsp;</span></button>
<div class="skip-nav"><a href="#skip-navbar-top" title="Skip navigation links">Skip navigation links</a></div>
<ul id="navbar-top-firstrow" class="nav-list" title="Navigation">
<li><a href="../../../../../../../index.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">Class</a></li>
<li class="nav-bar-cell1-rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html#use">Help</a></li>
</ul>
</div>
<div class="sub-nav">
<div id="navbar-sub-list"></div>
<div class="nav-list-search"><a href="../../../../../../../search.html">SEARCH</a>
<input type="text" id="search-input" disabled placeholder="Search">
<input type="reset" id="reset-button" disabled value="reset">
</div>
</div>
<!-- ========= END OF TOP NAVBAR ========= -->
<span class="skip-nav" id="skip-navbar-top"></span></nav>
</header>
<div class="flex-content">
<main role="main">
<div class="header">
<h1 title="Uses of Interface org.biojava.nbio.structure.align.multiple.MultipleAlignment" class="title">Uses of Interface<br>org.biojava.nbio.structure.align.multiple.MultipleAlignment</h1>
</div>
<div class="caption"><span>Packages that use <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table two-column-summary">
<div class="table-header col-first">Package</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></div>
<div class="col-last even-row-color">
<div class="block">Classes for the alignment of structures.</div>
</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></div>
<div class="col-last odd-row-color">
<div class="block">Utility methods for better interaction with Jmol.</div>
</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.multiple.mc">org.biojava.nbio.structure.align.multiple.mc</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.multiple.util">org.biojava.nbio.structure.align.multiple.util</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.align.quaternary">org.biojava.nbio.structure.align.quaternary</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></div>
<div class="col-last even-row-color">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.cluster">org.biojava.nbio.structure.cluster</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<section class="class-uses">
<ul class="block-list">
<li>
<section class="detail" id="org.biojava.nbio.structure.align">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../package-summary.html">org.biojava.nbio.structure.align</a></h2>
<div class="caption"><span>Methods in <a href="../../package-summary.html">org.biojava.nbio.structure.align</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleStructureAligner.</span><code><a href="../../MultipleStructureAligner.html#align(java.util.List)" class="member-name-link">align</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></div>
<div class="col-last even-row-color">
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleStructureAligner.</span><code><a href="../../MultipleStructureAligner.html#align(java.util.List,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;params)</code></div>
<div class="col-last odd-row-color">
<div class="block">Run an alignment and also send a bean containing the parameters.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.gui">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h2>
<div class="caption"><span>Methods in <a href="../../gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../../gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">MultipleAlignmentJmol</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentJmolDisplay.</span><code><a href="../../gui/MultipleAlignmentJmolDisplay.html#display(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">display</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></div>
<div class="col-last even-row-color">
<div class="block">Display a MultipleAlignment with a JmolPanel.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/JMenuBar.html" title="class or interface in javax.swing" class="external-link">JMenuBar</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MenuCreator.</span><code><a href="../../gui/MenuCreator.html#getAlignmentPanelMenu(javax.swing.JFrame,java.awt.event.ActionListener,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getAlignmentPanelMenu</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/JFrame.html" title="class or interface in javax.swing" class="external-link">JFrame</a>&nbsp;frame,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/event/ActionListener.html" title="class or interface in java.awt.event" class="external-link">ActionListener</a>&nbsp;actionListener,
 <a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last odd-row-color">
<div class="block">Create the menu for the Alignment Panel representation of
 Structural Alignments.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/JMenuBar.html" title="class or interface in javax.swing" class="external-link">JMenuBar</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MenuCreator.</span><code><a href="../../gui/MenuCreator.html#getAlignmentTextMenu(javax.swing.JFrame,java.awt.event.ActionListener,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getAlignmentTextMenu</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/JFrame.html" title="class or interface in javax.swing" class="external-link">JFrame</a>&nbsp;frame,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/event/ActionListener.html" title="class or interface in java.awt.event" class="external-link">ActionListener</a>&nbsp;actionListener,
 <a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color">
<div class="block">Create the menu for the Text representations of Structural Alignments.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentJmolDisplay.</span><code><a href="../../gui/MultipleAlignmentJmolDisplay.html#getPDBresnum(int,org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">getPDBresnum</a><wbr>(int&nbsp;structNum,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code></div>
<div class="col-last odd-row-color">
<div class="block">Utility method used in the <a href="../../gui/jmol/MultipleAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol"><code>MultipleAlignmentJmol</code></a> Frame,
 when the aligned residues of a structure in the alignment have
 to be selected for formatting them (coloring and style).</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/JMenuItem.html" title="class or interface in javax.swing" class="external-link">JMenuItem</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MenuCreator.</span><code><a href="../../gui/MenuCreator.html#getSaveAlignmentMenuItem(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getSaveAlignmentMenuItem</a><wbr>(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/JMenuBar.html" title="class or interface in javax.swing" class="external-link">JMenuBar</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MenuCreator.</span><code><a href="../../gui/MenuCreator.html#initJmolMenu(javax.swing.JFrame,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol,org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">initJmolMenu</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/swing/JFrame.html" title="class or interface in javax.swing" class="external-link">JFrame</a>&nbsp;frame,
 <a href="../../gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;parent,
 <a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last odd-row-color">
<div class="block">Provide a JMenuBar that can be added to a JFrame containing
 a JmolPanel.</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentJmolDisplay.</span><code><a href="../../gui/MultipleAlignmentJmolDisplay.html#showAlignmentImage(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.lang.String)" class="member-name-link">showAlignmentImage</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&nbsp;result)</code></div>
<div class="col-last even-row-color">
<div class="block">Creates a new Frame with the String output representation of the
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentJmolDisplay.</span><code><a href="../../gui/MultipleAlignmentJmolDisplay.html#showMultipleAligmentPanel(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)" class="member-name-link">showMultipleAligmentPanel</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="../../gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></div>
<div class="col-last odd-row-color">
<div class="block"><span class="deprecated-label">Deprecated.</span>
<div class="deprecation-comment">Replace with </div>
</div>
</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentJmolDisplay.</span><code><a href="../../gui/MultipleAlignmentJmolDisplay.html#showMultipleAlignmentPanel(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)" class="member-name-link">showMultipleAlignmentPanel</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="../../gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code></div>
<div class="col-last even-row-color">
<div class="block">Creates a new Frame with the MultipleAlignment Sequence Panel.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../gui/MySaveFileListener.html#%3Cinit%3E(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">MySaveFileListener</a><wbr>(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructor to avoid checking which of the two is null before
 instantiating this class.</div>
</div>
<div class="col-first odd-row-color"><code>&nbsp;</code></div>
<div class="col-second odd-row-color"><code><a href="../../gui/MySaveFileListener.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">MySaveFileListener</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.gui.aligpanel">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h2>
<div class="caption"><span>Methods in <a href="../../gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAligPanel.</span><code><a href="../../gui/aligpanel/MultipleAligPanel.html#getMultipleAlignment()" class="member-name-link">getMultipleAlignment</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../../gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../gui/aligpanel/MultipleAligPanel.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)" class="member-name-link">MultipleAligPanel</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 <a href="../../gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jm)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructor using a MultipleAlignment.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.gui.jmol">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h2>
<div class="caption"><span>Methods in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentJmol.</span><code><a href="../../gui/jmol/MultipleAlignmentJmol.html#getMultipleAlignment()" class="member-name-link">getMultipleAlignment</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentJmol.</span><code><a href="../../gui/jmol/MultipleAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,org.jcolorbrewer.ColorBrewer,boolean)" class="member-name-link">getJmolString</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms,
 org.jcolorbrewer.ColorBrewer&nbsp;colorPalette,
 boolean&nbsp;colorByBlocks)</code></div>
<div class="col-last even-row-color">
<div class="block">Generate a Jmol command String that colors the aligned residues of every
 structure.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentJmol.</span><code><a href="../../gui/jmol/MultipleAlignmentJmol.html#getMultiBlockJmolString(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,org.jcolorbrewer.ColorBrewer,boolean)" class="member-name-link">getMultiBlockJmolString</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms,
 org.jcolorbrewer.ColorBrewer&nbsp;colorPalette,
 boolean&nbsp;colorByBlocks)</code></div>
<div class="col-last odd-row-color">
<div class="block">Colors every Block of the structures with a different color, following
 the palette.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../../gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../../gui/jmol/MultipleAlignmentJmol.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)" class="member-name-link">MultipleAlignmentJmol</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;rotatedAtoms)</code></div>
<div class="col-last even-row-color">
<div class="block">The constructor displays the Mutltiple Alignment in a new JmolPanel
 Frame.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.multiple">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a></h2>
<div class="caption"><span>Classes in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> that implement <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Class</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>class&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../MultipleAlignmentImpl.html" class="type-name-link" title="class in org.biojava.nbio.structure.align.multiple">MultipleAlignmentImpl</a></code></div>
<div class="col-last even-row-color">
<div class="block">A general implementation of a <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignment.</span><code><a href="../MultipleAlignment.html#clone()" class="member-name-link">clone</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Creates and returns an identical copy of this alignment, including a deep
 copy of all constituent BlockSets.</div>
</div>
<div class="col-first odd-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BlockSet.</span><code><a href="../BlockSet.html#getMultipleAlignment()" class="member-name-link">getMultipleAlignment</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the parent MultipleAlignment of the BlockSet.</div>
</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">BlockSetImpl.</span><code><a href="../BlockSetImpl.html#getMultipleAlignment()" class="member-name-link">getMultipleAlignment</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentEnsemble.</span><code><a href="../MultipleAlignmentEnsemble.html#getMultipleAlignment(int)" class="member-name-link">getMultipleAlignment</a><wbr>(int&nbsp;index)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the MultipleAlignments at the specified index
 in the ensemble.</div>
</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentEnsembleImpl.</span><code><a href="../MultipleAlignmentEnsembleImpl.html#getMultipleAlignment(int)" class="member-name-link">getMultipleAlignment</a><wbr>(int&nbsp;index)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> that return types with arguments of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentEnsemble.</span><code><a href="../MultipleAlignmentEnsemble.html#getMultipleAlignments()" class="member-name-link">getMultipleAlignments</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the List of MultipleAlignments in the ensemble.</div>
</div>
<div class="col-first odd-row-color"><code><a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentEnsembleImpl.</span><code><a href="../MultipleAlignmentEnsembleImpl.html#getMultipleAlignments()" class="member-name-link">getMultipleAlignments</a>()</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Methods in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentEnsemble.</span><code><a href="../MultipleAlignmentEnsemble.html#addMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">addMultipleAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">Add a new MultipleAlignment to the end of the ensemble and set its
 parent ensemble to this.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentEnsembleImpl.</span><code><a href="../MultipleAlignmentEnsembleImpl.html#addMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">addMultipleAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">BlockSet.</span><code><a href="../BlockSet.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">setMultipleAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;parent)</code></div>
<div class="col-last even-row-color">
<div class="block">Set the back-reference to its parent MultipleAlignment.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">BlockSetImpl.</span><code><a href="../BlockSetImpl.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">setMultipleAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;parent)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Method parameters in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> with type arguments of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentEnsemble.</span><code><a href="../MultipleAlignmentEnsemble.html#setMultipleAlignments(java.util.List)" class="member-name-link">setMultipleAlignments</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;&nbsp;alignments)</code></div>
<div class="col-last even-row-color">
<div class="block">Set the List of MultipleAlignments in the ensemble.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentEnsembleImpl.</span><code><a href="../MultipleAlignmentEnsembleImpl.html#setMultipleAlignments(java.util.List)" class="member-name-link">setMultipleAlignments</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&gt;&nbsp;alignments)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
</div>
<div class="caption"><span>Constructors in <a href="../package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../BlockSetImpl.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">BlockSetImpl</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructor.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.multiple.mc">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a></h2>
<div class="caption"><span>Methods in <a href="../mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleMcMain.</span><code><a href="../mc/MultipleMcMain.html#align(java.util.List)" class="member-name-link">align</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleMcMain.</span><code><a href="../mc/MultipleMcMain.html#align(java.util.List,java.lang.Object)" class="member-name-link">align</a><wbr>(<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;atomArrays,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html" title="class or interface in java.lang" class="external-link">Object</a>&nbsp;parameters)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleMcOptimizer.</span><code><a href="../mc/MultipleMcOptimizer.html#call()" class="member-name-link">call</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleMcOptimizer.</span><code><a href="../mc/MultipleMcOptimizer.html#optimize()" class="member-name-link">optimize</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</div>
</div>
<div class="caption"><span>Constructors in <a href="../mc/package-summary.html">org.biojava.nbio.structure.align.multiple.mc</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier</div>
<div class="table-header col-second">Constructor</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>&nbsp;</code></div>
<div class="col-second even-row-color"><code><a href="../mc/MultipleMcOptimizer.html#%3Cinit%3E(org.biojava.nbio.structure.align.multiple.MultipleAlignment,org.biojava.nbio.structure.align.multiple.mc.MultipleMcParameters,int)" class="member-name-link">MultipleMcOptimizer</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;seedAln,
 <a href="../mc/MultipleMcParameters.html" title="class in org.biojava.nbio.structure.align.multiple.mc">MultipleMcParameters</a>&nbsp;params,
 int&nbsp;reference)</code></div>
<div class="col-last even-row-color">
<div class="block">Constructor.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.multiple.util">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a></h2>
<div class="caption"><span>Methods in <a href="../util/package-summary.html">org.biojava.nbio.structure.align.multiple.util</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentScorer.</span><code><a href="../util/MultipleAlignmentScorer.html#calculateScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">calculateScores</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculates and puts the RMSD and the average TM-Score of the
 MultipleAlignment.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getAtomForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int,int)" class="member-name-link">getAtomForSequencePosition</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct,
 int&nbsp;str,
 int&nbsp;sequencePos)</code></div>
<div class="col-last odd-row-color">
<div class="block">Returns the Atom of the specified structure that is aligned in the
 sequence alignment position specified.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getAverageResidueDistances(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getAverageResidueDistances</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color">
<div class="block">The average residue distance Matrix contains the average distance from
 each residue to all other residues aligned with it.</div>
</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentScorer.</span><code><a href="../util/MultipleAlignmentScorer.html#getAvgTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getAvgTMScore</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculates the average TMScore of all the possible pairwise structure
 comparisons of the given alignment.</div>
</div>
<div class="col-first even-row-color"><code>static int</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getBlockForSequencePosition(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List,int)" class="member-name-link">getBlockForSequencePosition</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct,
 int&nbsp;sequencePos)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns the block number of a specified position in the sequence
 alignment, given the mapping from structure to function.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getBlockSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;ma)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getBlockSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)" class="member-name-link">getBlockSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate the sequence alignment Strings for the alignment Blocks in an
 alignment.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getEquivalentResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment,boolean)" class="member-name-link">getEquivalentResidues</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa,
 boolean&nbsp;coreOnly)</code></div>
<div class="col-last odd-row-color">
<div class="block">Convert an MSA into a matrix of equivalent residues.</div>
</div>
<div class="col-first even-row-color"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getHSDMTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getHSDMTree</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using
 dissimilarity scores (DS), based in SDM Substitution Matrix (ideal for
 distantly related proteins, structure-derived) and the Neighbor Joining
 algorithm from forester.</div>
</div>
<div class="col-first odd-row-color"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getKimuraTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getKimuraTree</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using Kimura
 distances and the Neighbor Joining algorithm from forester.</div>
</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentScorer.</span><code><a href="../util/MultipleAlignmentScorer.html#getMCScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,double,double,double)" class="member-name-link">getMCScore</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 double&nbsp;gapOpen,
 double&nbsp;gapExtension,
 double&nbsp;dCutoff)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculates the MC score, specific for the MultipleAlignment algorithm.</div>
</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentScorer.</span><code><a href="../util/MultipleAlignmentScorer.html#getRefRMSD(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int)" class="member-name-link">getRefRMSD</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 int&nbsp;ref)</code></div>
<div class="col-last odd-row-color">
<div class="block">/** Calculates the average RMSD from all structures to a reference s
 tructure, given a set of superimposed atoms.</div>
</div>
<div class="col-first even-row-color"><code>static double</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentScorer.</span><code><a href="../util/MultipleAlignmentScorer.html#getRefTMScore(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int)" class="member-name-link">getRefTMScore</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 int&nbsp;ref)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculates the average TMScore from all structures to a reference
 structure, given a set of superimposed atoms.</div>
</div>
<div class="col-first odd-row-color"><code>static double</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentScorer.</span><code><a href="../util/MultipleAlignmentScorer.html#getRMSD(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getRMSD</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculates the RMSD of all-to-all structure comparisons (distances) of
 the given MultipleAlignment.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../../jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getRMSDMatrix(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getRMSDMatrix</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate the RMSD matrix of a MultipleAlignment, that is, entry (i,j) of
 the matrix contains the RMSD between structures i and j.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentDisplay.</span><code><a href="../util/MultipleAlignmentDisplay.html#getRotatedAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getRotatedAtoms</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></div>
<div class="col-last odd-row-color">
<div class="block">New structures are downloaded if they were not cached in the alignment
 and they are entirely transformed here with the superposition information
 in the Multiple Alignment.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getSequenceAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)" class="member-name-link">getSequenceAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html" title="class or interface in java.lang" class="external-link">Integer</a>&gt;&nbsp;mapSeqToStruct)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculate the sequence alignment Strings for the whole alignment.</div>
</div>
<div class="col-first even-row-color"><code>static org.forester.phylogeny.Phylogeny</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#getStructuralTree(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">getStructuralTree</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last even-row-color">
<div class="block">Calculate a phylogenetic tree of the MultipleAlignment using RMSD
 distances and the Neighbor Joining algorithm from forester.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">CoreSuperimposer.</span><code><a href="../util/CoreSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">superimpose</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleSuperimposer.</span><code><a href="../util/MultipleSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">superimpose</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">Superimpose all structures from a <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a>.</div>
</div>
<div class="col-first odd-row-color"><code>void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">ReferenceSuperimposer.</span><code><a href="../util/ReferenceSuperimposer.html#superimpose(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">superimpose</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentWriter.</span><code><a href="../util/MultipleAlignmentWriter.html#to3DFormat(org.biojava.nbio.structure.align.multiple.MultipleAlignment,int,int)" class="member-name-link">to3DFormat</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment,
 int&nbsp;queryIndex,
 int&nbsp;templateIndex)</code></div>
<div class="col-last even-row-color">
<div class="block">Outputs a pairwise alignment in I-TASSER's 3D Format for target-template
 alignment. http://zhanglab.ccmb.med.umich.edu/I-TASSER/option4.html</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentWriter.</span><code><a href="../util/MultipleAlignmentWriter.html#toAlignedResidues(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toAlignedResidues</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln)</code></div>
<div class="col-last odd-row-color">
<div class="block">Converts the alignment to its simplest form: a list of groups of aligned
 residues.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentWriter.</span><code><a href="../util/MultipleAlignmentWriter.html#toFASTA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toFASTA</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a multiple sequence alignment
 String in FASTA format.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentWriter.</span><code><a href="../util/MultipleAlignmentWriter.html#toFatCat(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toFatCat</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color">
<div class="block">Converts the <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple"><code>MultipleAlignment</code></a> into a FatCat String format.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../../../Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#toMultimodelStructure(org.biojava.nbio.structure.align.multiple.MultipleAlignment,java.util.List)" class="member-name-link">toMultimodelStructure</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multAln,
 <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;&nbsp;transformedAtoms)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code>static <a href="../../../../core/sequence/MultipleSequenceAlignment.html" title="class in org.biojava.nbio.core.sequence">MultipleSequenceAlignment</a><wbr>&lt;<a href="../../../../core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>,<wbr><a href="../../../../core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#toProteinMSA(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toProteinMSA</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msta)</code></div>
<div class="col-last odd-row-color">
<div class="block">Convert a MultipleAlignment into a MultipleSequenceAlignment of AminoAcid
 residues.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html" title="class or interface in java.lang" class="external-link">String</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentWriter.</span><code><a href="../util/MultipleAlignmentWriter.html#toTransformMatrices(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">toTransformMatrices</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">Converts the transformation Matrices of the alignment into a String
 output.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html" title="class or interface in java.util" class="external-link">List</a><wbr>&lt;<a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&gt;</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">MultipleAlignmentTools.</span><code><a href="../util/MultipleAlignmentTools.html#transformAtoms(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">transformAtoms</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last odd-row-color">
<div class="block">Transforms atoms according to the superposition stored in the alignment.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.quaternary">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a></h2>
<div class="caption"><span>Methods in <a href="../../quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">QsAlignResult.</span><code><a href="../../quaternary/QsAlignResult.html#getAlignment()" class="member-name-link">getAlignment</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">The alignment that specifies the residue equivalencies of the equivalent
 Subunits.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../quaternary/package-summary.html">org.biojava.nbio.structure.align.quaternary</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">QsAlignResult.</span><code><a href="../../quaternary/QsAlignResult.html#setAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">setAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;alignment)</code></div>
<div class="col-last even-row-color">
<div class="block">The alignment that specifies the residue equivalencies of the equivalent
 Subunits.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.align.xml">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h2>
<div class="caption"><span>Methods in <a href="../../xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">MultipleAlignmentXMLConverter.</span><code><a href="../../xml/MultipleAlignmentXMLConverter.html#printXMLalignment(org.biojava.nbio.core.util.PrettyXMLWriter,org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">printXMLalignment</a><wbr>(<a href="../../../../core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.cluster">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a></h2>
<div class="caption"><span>Methods in <a href="../../../cluster/package-summary.html">org.biojava.nbio.structure.cluster</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SubunitCluster.</span><code><a href="../../../cluster/SubunitCluster.html#getMultipleAlignment()" class="member-name-link">getMultipleAlignment</a>()</code></div>
<div class="col-last even-row-color">
<div class="block">The multiple alignment is calculated from the equivalent residues in the
 SubunitCluster.</div>
</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.symmetry.internal">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h2>
<div class="caption"><span>Methods in <a href="../../../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeSymmResult.</span><code><a href="../../../symmetry/internal/CeSymmResult.html#getMultipleAlignment()" class="member-name-link">getMultipleAlignment</a>()</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmOptimizer.</span><code><a href="../../../symmetry/internal/SymmOptimizer.html#optimize()" class="member-name-link">optimize</a>()</code></div>
<div class="col-last odd-row-color">
<div class="block">Optimization method based in a Monte-Carlo approach.</div>
</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">GraphComponentRefiner.</span><code><a href="../../../symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">refine</a><wbr>(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
 <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;order)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
<div class="col-first odd-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SequenceFunctionRefiner.</span><code><a href="../../../symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">refine</a><wbr>(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
 <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;order)</code></div>
<div class="col-last odd-row-color">&nbsp;</div>
<div class="col-first even-row-color"><code><a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryRefiner.</span><code><a href="../../../symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D,int)" class="member-name-link">refine</a><wbr>(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
 <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
 int&nbsp;order)</code></div>
<div class="col-last even-row-color">
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../../symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">CeSymmResult.</span><code><a href="../../../symmetry/internal/CeSymmResult.html#setMultipleAlignment(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">setMultipleAlignment</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;multipleAlignment)</code></div>
<div class="col-last even-row-color">&nbsp;</div>
</div>
</section>
</li>
<li>
<section class="detail" id="org.biojava.nbio.structure.symmetry.utils">
<h2>Uses of <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a> in <a href="../../../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h2>
<div class="caption"><span>Methods in <a href="../../../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> that return <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../../../symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,org.biojava.nbio.structure.Atom%5B%5D)" class="member-name-link">fromAFP</a><wbr>(<a href="../../model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
 <a href="../../../Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code></div>
<div class="col-last even-row-color">
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</div>
<div class="col-first odd-row-color"><code>static <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../../../symmetry/utils/SymmetryTools.html#toFullAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)" class="member-name-link">toFullAlignment</a><wbr>(<a href="../../../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;symm)</code></div>
<div class="col-last odd-row-color">
<div class="block">Method that converts a repeats symmetric alignment into an alignment of
 whole structures.</div>
</div>
<div class="col-first even-row-color"><code>static <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../../../symmetry/utils/SymmetryTools.html#toRepeatsAlignment(org.biojava.nbio.structure.symmetry.internal.CeSymmResult)" class="member-name-link">toRepeatsAlignment</a><wbr>(<a href="../../../symmetry/internal/CeSymmResult.html" title="class in org.biojava.nbio.structure.symmetry.internal">CeSymmResult</a>&nbsp;result)</code></div>
<div class="col-last even-row-color">
<div class="block">Method that converts a symmetry alignment into an alignment of the
 repeats only, as new independent structures.</div>
</div>
</div>
<div class="caption"><span>Methods in <a href="../../../symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></span></div>
<div class="summary-table three-column-summary">
<div class="table-header col-first">Modifier and Type</div>
<div class="table-header col-second">Method</div>
<div class="table-header col-last">Description</div>
<div class="col-first even-row-color"><code>static void</code></div>
<div class="col-second even-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../../../symmetry/utils/SymmetryTools.html#updateSymmetryScores(org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">updateSymmetryScores</a><wbr>(<a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;symm)</code></div>
<div class="col-last even-row-color">
<div class="block">Update the scores (TM-score and RMSD) of a symmetry multiple alignment.</div>
</div>
<div class="col-first odd-row-color"><code>static void</code></div>
<div class="col-second odd-row-color"><span class="type-name-label">SymmetryTools.</span><code><a href="../../../symmetry/utils/SymmetryTools.html#updateSymmetryTransformation(org.biojava.nbio.structure.symmetry.internal.SymmetryAxes,org.biojava.nbio.structure.align.multiple.MultipleAlignment)" class="member-name-link">updateSymmetryTransformation</a><wbr>(<a href="../../../symmetry/internal/SymmetryAxes.html" title="class in org.biojava.nbio.structure.symmetry.internal">SymmetryAxes</a>&nbsp;axes,
 <a href="../MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code></div>
<div class="col-last odd-row-color">
<div class="block">Calculates the set of symmetry operation Matrices (transformations) of
 the new alignment, based on the symmetry relations in the SymmetryAxes
 object.</div>
</div>
</div>
</section>
</li>
</ul>
</section>
</main>
<footer role="contentinfo">
<hr>
<p class="legal-copy"><small>Copyright &#169; 2000&#x2013;2025 <a href="https://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</footer>
</div>
</div>
</body>
</html>
