<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.structure.io, class: EntityFinder">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> */</span>
<span class="source-line-no">021</span><span id="line-21">package org.biojava.nbio.structure.io;</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">import org.biojava.nbio.alignment.Alignments;</span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.alignment.SimpleGapPenalty;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.alignment.template.GapPenalty;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.alignment.template.PairwiseSequenceAligner;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.alignment.template.SequencePair;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.core.sequence.DNASequence;</span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">034</span><span id="line-34">import org.biojava.nbio.core.sequence.RNASequence;</span>
<span class="source-line-no">035</span><span id="line-35">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">036</span><span id="line-36">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">037</span><span id="line-37">import org.biojava.nbio.structure.*;</span>
<span class="source-line-no">038</span><span id="line-38">import org.slf4j.Logger;</span>
<span class="source-line-no">039</span><span id="line-39">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">040</span><span id="line-40"></span>
<span class="source-line-no">041</span><span id="line-41">import java.util.ArrayList;</span>
<span class="source-line-no">042</span><span id="line-42">import java.util.Collections;</span>
<span class="source-line-no">043</span><span id="line-43">import java.util.Comparator;</span>
<span class="source-line-no">044</span><span id="line-44">import java.util.HashMap;</span>
<span class="source-line-no">045</span><span id="line-45">import java.util.List;</span>
<span class="source-line-no">046</span><span id="line-46">import java.util.Set;</span>
<span class="source-line-no">047</span><span id="line-47">import java.util.TreeMap;</span>
<span class="source-line-no">048</span><span id="line-48">import java.util.TreeSet;</span>
<span class="source-line-no">049</span><span id="line-49"></span>
<span class="source-line-no">050</span><span id="line-50">/**</span>
<span class="source-line-no">051</span><span id="line-51"> * Heuristical finding of Entities (called Compounds in legacy PDB format)</span>
<span class="source-line-no">052</span><span id="line-52"> * in a given Structure. Entities are the groups of sequence identical NCS-related polymer chains</span>
<span class="source-line-no">053</span><span id="line-53"> * in the Structure.</span>
<span class="source-line-no">054</span><span id="line-54"> *</span>
<span class="source-line-no">055</span><span id="line-55"> * This is related to {@link SeqRes2AtomAligner} but it is intended for raw PDB/mmCIF files where</span>
<span class="source-line-no">056</span><span id="line-56"> * possibly no SEQRES is given.</span>
<span class="source-line-no">057</span><span id="line-57"> *</span>
<span class="source-line-no">058</span><span id="line-58"> * @author Jose Duarte</span>
<span class="source-line-no">059</span><span id="line-59"> */</span>
<span class="source-line-no">060</span><span id="line-60">public class EntityFinder {</span>
<span class="source-line-no">061</span><span id="line-61"></span>
<span class="source-line-no">062</span><span id="line-62">        private static final Logger logger = LoggerFactory.getLogger(EntityFinder.class);</span>
<span class="source-line-no">063</span><span id="line-63"></span>
<span class="source-line-no">064</span><span id="line-64">        /**</span>
<span class="source-line-no">065</span><span id="line-65">         * Above this ratio of mismatching residue types for same residue numbers we</span>
<span class="source-line-no">066</span><span id="line-66">         * consider the 2 chains to have mismatching residue numbers and warn about it</span>
<span class="source-line-no">067</span><span id="line-67">         */</span>
<span class="source-line-no">068</span><span id="line-68">        public static final double RATIO_GAPS_FOR_MISMATCH = 0.50;</span>
<span class="source-line-no">069</span><span id="line-69"></span>
<span class="source-line-no">070</span><span id="line-70">        /**</span>
<span class="source-line-no">071</span><span id="line-71">         * Identity value for 2 chains to be considered part of same entity</span>
<span class="source-line-no">072</span><span id="line-72">         */</span>
<span class="source-line-no">073</span><span id="line-73">        public static final double IDENTITY_THRESHOLD = 0.99999;</span>
<span class="source-line-no">074</span><span id="line-74"></span>
<span class="source-line-no">075</span><span id="line-75">        /**</span>
<span class="source-line-no">076</span><span id="line-76">         * Gap coverage value (num gaps over length of sequence) for each chain of the match:</span>
<span class="source-line-no">077</span><span id="line-77">         * 2 chains with more gap coverage than this value will not be considered part of the same entity</span>
<span class="source-line-no">078</span><span id="line-78">         */</span>
<span class="source-line-no">079</span><span id="line-79">        public static final double GAP_COVERAGE_THRESHOLD = 0.3;</span>
<span class="source-line-no">080</span><span id="line-80"></span>
<span class="source-line-no">081</span><span id="line-81"></span>
<span class="source-line-no">082</span><span id="line-82"></span>
<span class="source-line-no">083</span><span id="line-83"></span>
<span class="source-line-no">084</span><span id="line-84">        /**</span>
<span class="source-line-no">085</span><span id="line-85">         * Utility method that employs some heuristics to find the {@link EntityInfo}s</span>
<span class="source-line-no">086</span><span id="line-86">         * for the polymeric chains given in constructor.</span>
<span class="source-line-no">087</span><span id="line-87">         * To be used in case the information is missing in PDB/mmCIF file</span>
<span class="source-line-no">088</span><span id="line-88">         * @return</span>
<span class="source-line-no">089</span><span id="line-89">         */</span>
<span class="source-line-no">090</span><span id="line-90">        public static List&lt;EntityInfo&gt; findPolyEntities(List&lt;List&lt;Chain&gt;&gt; polyModels) {</span>
<span class="source-line-no">091</span><span id="line-91">                TreeMap&lt;String,EntityInfo&gt; chainIds2entities = findEntitiesFromAlignment(polyModels);</span>
<span class="source-line-no">092</span><span id="line-92"></span>
<span class="source-line-no">093</span><span id="line-93">                return findUniqueEntities(chainIds2entities);</span>
<span class="source-line-no">094</span><span id="line-94">        }</span>
<span class="source-line-no">095</span><span id="line-95"></span>
<span class="source-line-no">096</span><span id="line-96">        /**</span>
<span class="source-line-no">097</span><span id="line-97">         * Utility method to obtain a list of unique entities from the chainIds2entities map</span>
<span class="source-line-no">098</span><span id="line-98">         * @return</span>
<span class="source-line-no">099</span><span id="line-99">         */</span>
<span class="source-line-no">100</span><span id="line-100">        private static List&lt;EntityInfo&gt; findUniqueEntities(TreeMap&lt;String,EntityInfo&gt; chainIds2entities) {</span>
<span class="source-line-no">101</span><span id="line-101"></span>
<span class="source-line-no">102</span><span id="line-102">                List&lt;EntityInfo&gt; list = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">103</span><span id="line-103"></span>
<span class="source-line-no">104</span><span id="line-104">                for (EntityInfo cluster:chainIds2entities.values()) {</span>
<span class="source-line-no">105</span><span id="line-105">                        boolean present = false;</span>
<span class="source-line-no">106</span><span id="line-106">                        for (EntityInfo cl:list) {</span>
<span class="source-line-no">107</span><span id="line-107">                                if (cl==cluster) {</span>
<span class="source-line-no">108</span><span id="line-108">                                        present = true;</span>
<span class="source-line-no">109</span><span id="line-109">                                        break;</span>
<span class="source-line-no">110</span><span id="line-110">                                }</span>
<span class="source-line-no">111</span><span id="line-111">                        }</span>
<span class="source-line-no">112</span><span id="line-112">                        if (!present) list.add(cluster);</span>
<span class="source-line-no">113</span><span id="line-113">                }</span>
<span class="source-line-no">114</span><span id="line-114">                return list;</span>
<span class="source-line-no">115</span><span id="line-115">        }</span>
<span class="source-line-no">116</span><span id="line-116"></span>
<span class="source-line-no">117</span><span id="line-117">        /**</span>
<span class="source-line-no">118</span><span id="line-118">         * Given all chains of all models find entities for the nonpolymers and water chains within them,</span>
<span class="source-line-no">119</span><span id="line-119">         * assigning entity ids, types and descriptions to them. The result is written back to the passed entities List.</span>
<span class="source-line-no">120</span><span id="line-120">         * @param nonPolyModels</span>
<span class="source-line-no">121</span><span id="line-121">         * @param waterModels</span>
<span class="source-line-no">122</span><span id="line-122">         * @param entities</span>
<span class="source-line-no">123</span><span id="line-123">         */</span>
<span class="source-line-no">124</span><span id="line-124">        public static void createPurelyNonPolyEntities(List&lt;List&lt;Chain&gt;&gt; nonPolyModels, List&lt;List&lt;Chain&gt;&gt; waterModels, List&lt;EntityInfo&gt; entities) {</span>
<span class="source-line-no">125</span><span id="line-125"></span>
<span class="source-line-no">126</span><span id="line-126">                if (nonPolyModels.isEmpty()) return;</span>
<span class="source-line-no">127</span><span id="line-127"></span>
<span class="source-line-no">128</span><span id="line-128">                // let's find first the max entity id to assign entity ids to the newly found entities</span>
<span class="source-line-no">129</span><span id="line-129">                int maxMolId = 0;</span>
<span class="source-line-no">130</span><span id="line-130">                if (!entities.isEmpty()) {</span>
<span class="source-line-no">131</span><span id="line-131">                        maxMolId = Collections.max(entities, Comparator.comparingInt(EntityInfo::getMolId)).getMolId();</span>
<span class="source-line-no">132</span><span id="line-132">                }</span>
<span class="source-line-no">133</span><span id="line-133">                // we go one over the max</span>
<span class="source-line-no">134</span><span id="line-134">                int molId = maxMolId + 1;</span>
<span class="source-line-no">135</span><span id="line-135"></span>
<span class="source-line-no">136</span><span id="line-136">                if (!nonPolyModels.get(0).isEmpty()) {</span>
<span class="source-line-no">137</span><span id="line-137">                        List&lt;EntityInfo&gt; nonPolyEntities = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">138</span><span id="line-138"></span>
<span class="source-line-no">139</span><span id="line-139">                        for (List&lt;Chain&gt; model:nonPolyModels) {</span>
<span class="source-line-no">140</span><span id="line-140">                                for (Chain c: model) {</span>
<span class="source-line-no">141</span><span id="line-141">                                        // we assume there's only 1 group per non-poly chain</span>
<span class="source-line-no">142</span><span id="line-142">                                        String molecPdbName = c.getAtomGroup(0).getPDBName();</span>
<span class="source-line-no">143</span><span id="line-143"></span>
<span class="source-line-no">144</span><span id="line-144">                                        EntityInfo nonPolyEntity = findNonPolyEntityWithDescription(molecPdbName, nonPolyEntities);</span>
<span class="source-line-no">145</span><span id="line-145">                                        if (nonPolyEntity == null) {</span>
<span class="source-line-no">146</span><span id="line-146">                                                nonPolyEntity = new EntityInfo();</span>
<span class="source-line-no">147</span><span id="line-147">                                                nonPolyEntity.setDescription(molecPdbName);</span>
<span class="source-line-no">148</span><span id="line-148">                                                nonPolyEntity.setType(EntityType.NONPOLYMER);</span>
<span class="source-line-no">149</span><span id="line-149">                                                nonPolyEntity.setMolId(molId++);</span>
<span class="source-line-no">150</span><span id="line-150">                                                nonPolyEntities.add(nonPolyEntity);</span>
<span class="source-line-no">151</span><span id="line-151"></span>
<span class="source-line-no">152</span><span id="line-152">                                        }</span>
<span class="source-line-no">153</span><span id="line-153">                                        nonPolyEntity.addChain(c);</span>
<span class="source-line-no">154</span><span id="line-154">                                        c.setEntityInfo(nonPolyEntity);</span>
<span class="source-line-no">155</span><span id="line-155">                                }</span>
<span class="source-line-no">156</span><span id="line-156">                        }</span>
<span class="source-line-no">157</span><span id="line-157">                        entities.addAll(nonPolyEntities);</span>
<span class="source-line-no">158</span><span id="line-158">                }</span>
<span class="source-line-no">159</span><span id="line-159"></span>
<span class="source-line-no">160</span><span id="line-160">                if (!waterModels.get(0).isEmpty()) {</span>
<span class="source-line-no">161</span><span id="line-161">                        EntityInfo waterEntity = new EntityInfo();</span>
<span class="source-line-no">162</span><span id="line-162">                        waterEntity.setType(EntityType.WATER);</span>
<span class="source-line-no">163</span><span id="line-163">                        waterEntity.setDescription("water");</span>
<span class="source-line-no">164</span><span id="line-164">                        waterEntity.setMolId(molId);</span>
<span class="source-line-no">165</span><span id="line-165"></span>
<span class="source-line-no">166</span><span id="line-166">                        for (List&lt;Chain&gt; model:waterModels) {</span>
<span class="source-line-no">167</span><span id="line-167">                                for (Chain waterChain:model) {</span>
<span class="source-line-no">168</span><span id="line-168">                                        waterEntity.addChain(waterChain);</span>
<span class="source-line-no">169</span><span id="line-169">                                        waterChain.setEntityInfo(waterEntity);</span>
<span class="source-line-no">170</span><span id="line-170">                                }</span>
<span class="source-line-no">171</span><span id="line-171">                        }</span>
<span class="source-line-no">172</span><span id="line-172">                        entities.add(waterEntity);</span>
<span class="source-line-no">173</span><span id="line-173"></span>
<span class="source-line-no">174</span><span id="line-174">                }</span>
<span class="source-line-no">175</span><span id="line-175"></span>
<span class="source-line-no">176</span><span id="line-176">        }</span>
<span class="source-line-no">177</span><span id="line-177"></span>
<span class="source-line-no">178</span><span id="line-178">        private static EntityInfo findNonPolyEntityWithDescription(String description, List&lt;EntityInfo&gt; nonPolyEntities) {</span>
<span class="source-line-no">179</span><span id="line-179">                for (EntityInfo e:nonPolyEntities) {</span>
<span class="source-line-no">180</span><span id="line-180">                        if (e.getDescription().equals(description)) return e;</span>
<span class="source-line-no">181</span><span id="line-181">                }</span>
<span class="source-line-no">182</span><span id="line-182">                return null;</span>
<span class="source-line-no">183</span><span id="line-183">        }</span>
<span class="source-line-no">184</span><span id="line-184"></span>
<span class="source-line-no">185</span><span id="line-185"></span>
<span class="source-line-no">186</span><span id="line-186">        private static boolean areResNumbersAligned(Chain c1, Chain c2) {</span>
<span class="source-line-no">187</span><span id="line-187"></span>
<span class="source-line-no">188</span><span id="line-188">                boolean isC1prot = c1.isProtein();</span>
<span class="source-line-no">189</span><span id="line-189">                boolean isC2prot = c2.isProtein();</span>
<span class="source-line-no">190</span><span id="line-190"></span>
<span class="source-line-no">191</span><span id="line-191">                // different kind of chain: we won't try to align them</span>
<span class="source-line-no">192</span><span id="line-192">                if (isC1prot != isC2prot ) return false;</span>
<span class="source-line-no">193</span><span id="line-193"></span>
<span class="source-line-no">194</span><span id="line-194">                List&lt;Group&gt; c1AtomGroups = null;</span>
<span class="source-line-no">195</span><span id="line-195">                if (isC1prot) {</span>
<span class="source-line-no">196</span><span id="line-196">                        c1AtomGroups = c1.getAtomGroups(GroupType.AMINOACID);</span>
<span class="source-line-no">197</span><span id="line-197">                }</span>
<span class="source-line-no">198</span><span id="line-198">                else {</span>
<span class="source-line-no">199</span><span id="line-199">                        c1AtomGroups = c1.getAtomGroups(GroupType.NUCLEOTIDE);</span>
<span class="source-line-no">200</span><span id="line-200">                }</span>
<span class="source-line-no">201</span><span id="line-201"></span>
<span class="source-line-no">202</span><span id="line-202">                int countNonExisting = 0;</span>
<span class="source-line-no">203</span><span id="line-203"></span>
<span class="source-line-no">204</span><span id="line-204">                for (Group g1:c1AtomGroups) {</span>
<span class="source-line-no">205</span><span id="line-205">                        try {</span>
<span class="source-line-no">206</span><span id="line-206">                                Group g2 = c2.getGroupByPDB(g1.getResidueNumber());</span>
<span class="source-line-no">207</span><span id="line-207">                                if (!g2.getPDBName().equals(g1.getPDBName())) {</span>
<span class="source-line-no">208</span><span id="line-208">                                        logger.debug("Mismatch of residues between chains {},{} for residue number {}: {} {}",</span>
<span class="source-line-no">209</span><span id="line-209">                                                        c1.getId(),c2.getId(),g1.getResidueNumber(), g1.getPDBName(), g2.getPDBName());</span>
<span class="source-line-no">210</span><span id="line-210">                                        return false;</span>
<span class="source-line-no">211</span><span id="line-211">                                }</span>
<span class="source-line-no">212</span><span id="line-212">                        } catch (StructureException e) {</span>
<span class="source-line-no">213</span><span id="line-213">                                // the group doesn't exist (no density) in the chain, go on</span>
<span class="source-line-no">214</span><span id="line-214">                                countNonExisting++;</span>
<span class="source-line-no">215</span><span id="line-215">                        }</span>
<span class="source-line-no">216</span><span id="line-216">                }</span>
<span class="source-line-no">217</span><span id="line-217"></span>
<span class="source-line-no">218</span><span id="line-218">                if ((double)countNonExisting/(double)c1AtomGroups.size() &gt; RATIO_GAPS_FOR_MISMATCH) {</span>
<span class="source-line-no">219</span><span id="line-219">                        logger.debug("More than {} of the residues ({} out of {}) are not present in chain {} when comparing by residue numbers to chain {}.",</span>
<span class="source-line-no">220</span><span id="line-220">                                        RATIO_GAPS_FOR_MISMATCH, countNonExisting, c1AtomGroups.size(), c2.getId(), c1.getId());</span>
<span class="source-line-no">221</span><span id="line-221">                        return false;</span>
<span class="source-line-no">222</span><span id="line-222">                }</span>
<span class="source-line-no">223</span><span id="line-223"></span>
<span class="source-line-no">224</span><span id="line-224">                return true;</span>
<span class="source-line-no">225</span><span id="line-225">        }</span>
<span class="source-line-no">226</span><span id="line-226"></span>
<span class="source-line-no">227</span><span id="line-227"></span>
<span class="source-line-no">228</span><span id="line-228"></span>
<span class="source-line-no">229</span><span id="line-229">        private static TreeMap&lt;String,EntityInfo&gt; findEntitiesFromAlignment(List&lt;List&lt;Chain&gt;&gt; polyModels) {</span>
<span class="source-line-no">230</span><span id="line-230"></span>
<span class="source-line-no">231</span><span id="line-231">                TreeMap&lt;String, EntityInfo&gt; chainIds2entities = new TreeMap&lt;&gt;();</span>
<span class="source-line-no">232</span><span id="line-232"></span>
<span class="source-line-no">233</span><span id="line-233">                if (polyModels.isEmpty()) return chainIds2entities;</span>
<span class="source-line-no">234</span><span id="line-234"></span>
<span class="source-line-no">235</span><span id="line-235">                Set&lt;Integer&gt; polyChainIndices = new TreeSet&lt;&gt;();</span>
<span class="source-line-no">236</span><span id="line-236">                for (int i=0;i&lt;polyModels.get(0).size();i++) {</span>
<span class="source-line-no">237</span><span id="line-237">                        polyChainIndices.add(i);</span>
<span class="source-line-no">238</span><span id="line-238">                }</span>
<span class="source-line-no">239</span><span id="line-239"></span>
<span class="source-line-no">240</span><span id="line-240">                int molId = 1;</span>
<span class="source-line-no">241</span><span id="line-241"></span>
<span class="source-line-no">242</span><span id="line-242">                outer:</span>
<span class="source-line-no">243</span><span id="line-243">                        for (int i:polyChainIndices) {</span>
<span class="source-line-no">244</span><span id="line-244">                                Chain c1 = polyModels.get(0).get(i);</span>
<span class="source-line-no">245</span><span id="line-245">                                // here we use false, which means that X will be used for unknown compounds</span>
<span class="source-line-no">246</span><span id="line-246">                                String str1 = SeqRes2AtomAligner.getFullAtomSequence(c1.getAtomGroups(), new HashMap&lt;&gt;(), false);</span>
<span class="source-line-no">247</span><span id="line-247"></span>
<span class="source-line-no">248</span><span id="line-248">                                for (int j:polyChainIndices) {</span>
<span class="source-line-no">249</span><span id="line-249"></span>
<span class="source-line-no">250</span><span id="line-250">                                        if (j&lt;=i) continue;</span>
<span class="source-line-no">251</span><span id="line-251"></span>
<span class="source-line-no">252</span><span id="line-252">                                        Chain c2 = polyModels.get(0).get(j);</span>
<span class="source-line-no">253</span><span id="line-253">                                        String str2 = SeqRes2AtomAligner.getFullAtomSequence(c2.getAtomGroups(), new HashMap&lt;&gt;(), false);</span>
<span class="source-line-no">254</span><span id="line-254"></span>
<span class="source-line-no">255</span><span id="line-255">                                        int seq1Length = 0;</span>
<span class="source-line-no">256</span><span id="line-256">                                        int seq2Length = 0;</span>
<span class="source-line-no">257</span><span id="line-257"></span>
<span class="source-line-no">258</span><span id="line-258">                                        SequencePair&lt;?,?&gt; pair;</span>
<span class="source-line-no">259</span><span id="line-259">                                        if (isProteinSequence(str1) &amp;&amp; isProteinSequence(str2)) {</span>
<span class="source-line-no">260</span><span id="line-260">                                                ProteinSequence s1 = getProteinSequence(str1);</span>
<span class="source-line-no">261</span><span id="line-261">                                                ProteinSequence s2 = getProteinSequence(str2);</span>
<span class="source-line-no">262</span><span id="line-262">                                                seq1Length = s1.getLength();</span>
<span class="source-line-no">263</span><span id="line-263">                                                seq2Length = s2.getLength();</span>
<span class="source-line-no">264</span><span id="line-264"></span>
<span class="source-line-no">265</span><span id="line-265">                                                pair = alignProtein(s1,s2);</span>
<span class="source-line-no">266</span><span id="line-266"></span>
<span class="source-line-no">267</span><span id="line-267">                                        } else if (isDNASequence(str1) &amp;&amp; isDNASequence(str2)) {</span>
<span class="source-line-no">268</span><span id="line-268">                                                DNASequence s1 = getDNASequence(str1);</span>
<span class="source-line-no">269</span><span id="line-269">                                                DNASequence s2 = getDNASequence(str2);</span>
<span class="source-line-no">270</span><span id="line-270">                                                seq1Length = s1.getLength();</span>
<span class="source-line-no">271</span><span id="line-271">                                                seq2Length = s2.getLength();</span>
<span class="source-line-no">272</span><span id="line-272"></span>
<span class="source-line-no">273</span><span id="line-273">                                                pair = alignDNA(s1,s2);</span>
<span class="source-line-no">274</span><span id="line-274"></span>
<span class="source-line-no">275</span><span id="line-275">                                        } else if (isRNASequence(str1) &amp;&amp; isRNASequence(str2)) {</span>
<span class="source-line-no">276</span><span id="line-276">                                                RNASequence s1 = getRNASequence(str1);</span>
<span class="source-line-no">277</span><span id="line-277">                                                RNASequence s2 = getRNASequence(str2);</span>
<span class="source-line-no">278</span><span id="line-278">                                                seq1Length = s1.getLength();</span>
<span class="source-line-no">279</span><span id="line-279">                                                seq2Length = s2.getLength();</span>
<span class="source-line-no">280</span><span id="line-280"></span>
<span class="source-line-no">281</span><span id="line-281">                                                pair = alignRNA(s1,s2);</span>
<span class="source-line-no">282</span><span id="line-282"></span>
<span class="source-line-no">283</span><span id="line-283">                                        } else {</span>
<span class="source-line-no">284</span><span id="line-284">                                                logger.debug("Chains {},{} are either different kind of polymers or could not be recognized as protein or nucleotide polymers", c1.getId(), c2.getId());</span>
<span class="source-line-no">285</span><span id="line-285">                                                continue;</span>
<span class="source-line-no">286</span><span id="line-286">                                        }</span>
<span class="source-line-no">287</span><span id="line-287"></span>
<span class="source-line-no">288</span><span id="line-288">                                        int numGaps = getNumGaps(pair);</span>
<span class="source-line-no">289</span><span id="line-289">                                        int numGaps1 = getNumGapsQuery(pair);</span>
<span class="source-line-no">290</span><span id="line-290">                                        int numGaps2 = getNumGapsTarget(pair);</span>
<span class="source-line-no">291</span><span id="line-291"></span>
<span class="source-line-no">292</span><span id="line-292">                                        int nonGaps = pair.getLength() - numGaps;</span>
<span class="source-line-no">293</span><span id="line-293"></span>
<span class="source-line-no">294</span><span id="line-294">                                        double identity = (double)pair.getNumIdenticals()/(double)nonGaps;</span>
<span class="source-line-no">295</span><span id="line-295">                                        double gapCov1 = (double) numGaps1 / (double) seq1Length;</span>
<span class="source-line-no">296</span><span id="line-296">                                        double gapCov2 = (double) numGaps2 / (double) seq2Length;</span>
<span class="source-line-no">297</span><span id="line-297"></span>
<span class="source-line-no">298</span><span id="line-298">                                        logger.debug("Alignment for chain pair {},{}: identity: {}, gap coverage 1: {}, gap coverage 2: {}",</span>
<span class="source-line-no">299</span><span id="line-299">                                                        c1.getId(), c2.getId(), String.format("%4.2f",identity), String.format("%4.2f",gapCov1), String.format("%4.2f",gapCov2));</span>
<span class="source-line-no">300</span><span id="line-300">                                        logger.debug("\n{}", pair.toString(100));</span>
<span class="source-line-no">301</span><span id="line-301"></span>
<span class="source-line-no">302</span><span id="line-302">                                        if (identity &gt; IDENTITY_THRESHOLD &amp;&amp; gapCov1&lt;GAP_COVERAGE_THRESHOLD &amp;&amp; gapCov2&lt;GAP_COVERAGE_THRESHOLD) {</span>
<span class="source-line-no">303</span><span id="line-303">                                                if (    !chainIds2entities.containsKey(c1.getId()) &amp;&amp;</span>
<span class="source-line-no">304</span><span id="line-304">                                                                !chainIds2entities.containsKey(c2.getId())) {</span>
<span class="source-line-no">305</span><span id="line-305">                                                        logger.debug("Creating entity with chains {},{}",c1.getId(),c2.getId());</span>
<span class="source-line-no">306</span><span id="line-306"></span>
<span class="source-line-no">307</span><span id="line-307">                                                        EntityInfo ent = new EntityInfo();</span>
<span class="source-line-no">308</span><span id="line-308">                                                        ent.addChain(c1);</span>
<span class="source-line-no">309</span><span id="line-309">                                                        ent.addChain(c2);</span>
<span class="source-line-no">310</span><span id="line-310">                                                        ent.setMolId(molId++);</span>
<span class="source-line-no">311</span><span id="line-311">                                                        ent.setType(EntityType.POLYMER);</span>
<span class="source-line-no">312</span><span id="line-312">                                                        c1.setEntityInfo(ent);</span>
<span class="source-line-no">313</span><span id="line-313">                                                        c2.setEntityInfo(ent);</span>
<span class="source-line-no">314</span><span id="line-314">                                                        chainIds2entities.put(c1.getId(), ent);</span>
<span class="source-line-no">315</span><span id="line-315">                                                        chainIds2entities.put(c2.getId(), ent);</span>
<span class="source-line-no">316</span><span id="line-316"></span>
<span class="source-line-no">317</span><span id="line-317">                                                } else {</span>
<span class="source-line-no">318</span><span id="line-318">                                                        Chain chainToAdd;</span>
<span class="source-line-no">319</span><span id="line-319">                                                        EntityInfo ent = chainIds2entities.get(c1.getId());</span>
<span class="source-line-no">320</span><span id="line-320">                                                        if (ent==null) {</span>
<span class="source-line-no">321</span><span id="line-321">                                                                ent = chainIds2entities.get(c2.getId());</span>
<span class="source-line-no">322</span><span id="line-322">                                                                chainToAdd = c1;</span>
<span class="source-line-no">323</span><span id="line-323">                                                        } else {</span>
<span class="source-line-no">324</span><span id="line-324">                                                                chainToAdd = c2;</span>
<span class="source-line-no">325</span><span id="line-325">                                                        }</span>
<span class="source-line-no">326</span><span id="line-326">                                                        if (!chainIds2entities.containsKey(chainToAdd.getId())) {</span>
<span class="source-line-no">327</span><span id="line-327">                                                                logger.debug("Adding chain {} to entity {}", chainToAdd.getId(), ent.getId());</span>
<span class="source-line-no">328</span><span id="line-328">                                                                ent.addChain(chainToAdd);</span>
<span class="source-line-no">329</span><span id="line-329">                                                                chainToAdd.setEntityInfo(ent);</span>
<span class="source-line-no">330</span><span id="line-330">                                                                chainIds2entities.put(chainToAdd.getId(), ent);</span>
<span class="source-line-no">331</span><span id="line-331">                                                        }</span>
<span class="source-line-no">332</span><span id="line-332">                                                }</span>
<span class="source-line-no">333</span><span id="line-333">                                                if (!areResNumbersAligned(c1, c2)) {</span>
<span class="source-line-no">334</span><span id="line-334">                                                        logger.warn("Including 100% identical chains {},{} in same entity, although they have misaligned residue numbers",</span>
<span class="source-line-no">335</span><span id="line-335">                                                                        c1.getId(),c2.getId());</span>
<span class="source-line-no">336</span><span id="line-336">                                                }</span>
<span class="source-line-no">337</span><span id="line-337">                                        }</span>
<span class="source-line-no">338</span><span id="line-338"></span>
<span class="source-line-no">339</span><span id="line-339">                                        if (identity&gt;1) {</span>
<span class="source-line-no">340</span><span id="line-340">                                                logger.warn("Identity for chains {},{} above 1. {} identicals out of {} non-gap-aligned residues (identity {})",</span>
<span class="source-line-no">341</span><span id="line-341">                                                                c1.getId(),c2.getId(),pair.getNumIdenticals(),nonGaps,identity);</span>
<span class="source-line-no">342</span><span id="line-342">                                                logger.warn("\n"+pair.toString(100));</span>
<span class="source-line-no">343</span><span id="line-343">                                        }</span>
<span class="source-line-no">344</span><span id="line-344"></span>
<span class="source-line-no">345</span><span id="line-345">                                        if (chainIds2entities.size()==polyChainIndices.size()) // we've got all chains in entities</span>
<span class="source-line-no">346</span><span id="line-346">                                                break outer;</span>
<span class="source-line-no">347</span><span id="line-347">                                }</span>
<span class="source-line-no">348</span><span id="line-348">                        }</span>
<span class="source-line-no">349</span><span id="line-349"></span>
<span class="source-line-no">350</span><span id="line-350">                // anything not in an Entity will be its own Entity</span>
<span class="source-line-no">351</span><span id="line-351">                for (int i:polyChainIndices) {</span>
<span class="source-line-no">352</span><span id="line-352">                        Chain c = polyModels.get(0).get(i);</span>
<span class="source-line-no">353</span><span id="line-353">                        if (!chainIds2entities.containsKey(c.getId())) {</span>
<span class="source-line-no">354</span><span id="line-354">                                logger.debug("Creating a 1-member entity for chain {}",c.getId());</span>
<span class="source-line-no">355</span><span id="line-355"></span>
<span class="source-line-no">356</span><span id="line-356">                                EntityInfo ent = new EntityInfo();</span>
<span class="source-line-no">357</span><span id="line-357">                                ent.addChain(c);</span>
<span class="source-line-no">358</span><span id="line-358">                                ent.setMolId(molId++);</span>
<span class="source-line-no">359</span><span id="line-359">                                ent.setType(EntityType.POLYMER);</span>
<span class="source-line-no">360</span><span id="line-360">                                c.setEntityInfo(ent);</span>
<span class="source-line-no">361</span><span id="line-361"></span>
<span class="source-line-no">362</span><span id="line-362">                                chainIds2entities.put(c.getId(),ent);</span>
<span class="source-line-no">363</span><span id="line-363">                        }</span>
<span class="source-line-no">364</span><span id="line-364">                }</span>
<span class="source-line-no">365</span><span id="line-365"></span>
<span class="source-line-no">366</span><span id="line-366">                // now that we've found the entities for first model, let's do the same for the rest of the models</span>
<span class="source-line-no">367</span><span id="line-367"></span>
<span class="source-line-no">368</span><span id="line-368">                for (int i=1;i&lt;polyModels.size();i++) {</span>
<span class="source-line-no">369</span><span id="line-369">                        for (Chain chain : polyModels.get(i)) {</span>
<span class="source-line-no">370</span><span id="line-370">                                EntityInfo e = chainIds2entities.get(chain.getId());</span>
<span class="source-line-no">371</span><span id="line-371">                                chain.setEntityInfo(e);</span>
<span class="source-line-no">372</span><span id="line-372">                                e.addChain(chain);</span>
<span class="source-line-no">373</span><span id="line-373">                        }</span>
<span class="source-line-no">374</span><span id="line-374">                }</span>
<span class="source-line-no">375</span><span id="line-375"></span>
<span class="source-line-no">376</span><span id="line-376">                return chainIds2entities;</span>
<span class="source-line-no">377</span><span id="line-377">        }</span>
<span class="source-line-no">378</span><span id="line-378"></span>
<span class="source-line-no">379</span><span id="line-379">        private static SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; alignProtein(ProteinSequence s1, ProteinSequence s2) {</span>
<span class="source-line-no">380</span><span id="line-380">                SubstitutionMatrix&lt;AminoAcidCompound&gt; matrix = SubstitutionMatrixHelper.getIdentity();</span>
<span class="source-line-no">381</span><span id="line-381"></span>
<span class="source-line-no">382</span><span id="line-382">                GapPenalty penalty = new SimpleGapPenalty(8, 1);</span>
<span class="source-line-no">383</span><span id="line-383"></span>
<span class="source-line-no">384</span><span id="line-384">                PairwiseSequenceAligner&lt;ProteinSequence, AminoAcidCompound&gt; nw =</span>
<span class="source-line-no">385</span><span id="line-385">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);</span>
<span class="source-line-no">386</span><span id="line-386"></span>
<span class="source-line-no">387</span><span id="line-387">                return nw.getPair();</span>
<span class="source-line-no">388</span><span id="line-388">        }</span>
<span class="source-line-no">389</span><span id="line-389"></span>
<span class="source-line-no">390</span><span id="line-390">        private static SequencePair&lt;DNASequence, NucleotideCompound&gt; alignDNA(DNASequence s1, DNASequence s2) {</span>
<span class="source-line-no">391</span><span id="line-391">                SubstitutionMatrix&lt;NucleotideCompound&gt; matrix = SubstitutionMatrixHelper.getNuc4_4();</span>
<span class="source-line-no">392</span><span id="line-392"></span>
<span class="source-line-no">393</span><span id="line-393">                GapPenalty penalty = new SimpleGapPenalty(8, 1);</span>
<span class="source-line-no">394</span><span id="line-394"></span>
<span class="source-line-no">395</span><span id="line-395">                PairwiseSequenceAligner&lt;DNASequence, NucleotideCompound&gt; nw =</span>
<span class="source-line-no">396</span><span id="line-396">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);</span>
<span class="source-line-no">397</span><span id="line-397"></span>
<span class="source-line-no">398</span><span id="line-398">                return nw.getPair();</span>
<span class="source-line-no">399</span><span id="line-399">        }</span>
<span class="source-line-no">400</span><span id="line-400"></span>
<span class="source-line-no">401</span><span id="line-401">        private static SequencePair&lt;RNASequence, NucleotideCompound&gt; alignRNA(RNASequence s1, RNASequence s2) {</span>
<span class="source-line-no">402</span><span id="line-402">                SubstitutionMatrix&lt;NucleotideCompound&gt; matrix = SubstitutionMatrixHelper.getNuc4_4();</span>
<span class="source-line-no">403</span><span id="line-403"></span>
<span class="source-line-no">404</span><span id="line-404">                GapPenalty penalty = new SimpleGapPenalty(8, 1);</span>
<span class="source-line-no">405</span><span id="line-405"></span>
<span class="source-line-no">406</span><span id="line-406">                PairwiseSequenceAligner&lt;RNASequence, NucleotideCompound&gt; nw =</span>
<span class="source-line-no">407</span><span id="line-407">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.GLOBAL, penalty, matrix);</span>
<span class="source-line-no">408</span><span id="line-408"></span>
<span class="source-line-no">409</span><span id="line-409">                return nw.getPair();</span>
<span class="source-line-no">410</span><span id="line-410">        }</span>
<span class="source-line-no">411</span><span id="line-411"></span>
<span class="source-line-no">412</span><span id="line-412"></span>
<span class="source-line-no">413</span><span id="line-413"></span>
<span class="source-line-no">414</span><span id="line-414">        private static int getNumGaps(SequencePair&lt;?, ?&gt; pair) {</span>
<span class="source-line-no">415</span><span id="line-415">                int numGaps = 0;</span>
<span class="source-line-no">416</span><span id="line-416">                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {</span>
<span class="source-line-no">417</span><span id="line-417">                        if (pair.hasGap(alignmentIndex)) numGaps++;</span>
<span class="source-line-no">418</span><span id="line-418">                }</span>
<span class="source-line-no">419</span><span id="line-419">                return numGaps;</span>
<span class="source-line-no">420</span><span id="line-420">        }</span>
<span class="source-line-no">421</span><span id="line-421"></span>
<span class="source-line-no">422</span><span id="line-422">        private static int getNumGapsQuery(SequencePair&lt;?, ?&gt; pair) {</span>
<span class="source-line-no">423</span><span id="line-423">                int numGaps = 0;</span>
<span class="source-line-no">424</span><span id="line-424">                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {</span>
<span class="source-line-no">425</span><span id="line-425">                        if ("-".equals(pair.getCompoundInQueryAt(alignmentIndex).getShortName())) {</span>
<span class="source-line-no">426</span><span id="line-426">                                numGaps++;</span>
<span class="source-line-no">427</span><span id="line-427">                        }</span>
<span class="source-line-no">428</span><span id="line-428">                }</span>
<span class="source-line-no">429</span><span id="line-429">                return numGaps;</span>
<span class="source-line-no">430</span><span id="line-430">        }</span>
<span class="source-line-no">431</span><span id="line-431"></span>
<span class="source-line-no">432</span><span id="line-432">        private static int getNumGapsTarget(SequencePair&lt;?, ?&gt; pair) {</span>
<span class="source-line-no">433</span><span id="line-433">                int numGaps = 0;</span>
<span class="source-line-no">434</span><span id="line-434">                for (int alignmentIndex=1;alignmentIndex&lt;=pair.getLength();alignmentIndex++) {</span>
<span class="source-line-no">435</span><span id="line-435">                        if ("-".equals(pair.getCompoundInTargetAt(alignmentIndex).getShortName())) {</span>
<span class="source-line-no">436</span><span id="line-436">                                numGaps++;</span>
<span class="source-line-no">437</span><span id="line-437">                        }</span>
<span class="source-line-no">438</span><span id="line-438">                }</span>
<span class="source-line-no">439</span><span id="line-439">                return numGaps;</span>
<span class="source-line-no">440</span><span id="line-440">        }</span>
<span class="source-line-no">441</span><span id="line-441"></span>
<span class="source-line-no">442</span><span id="line-442">        private static boolean isProteinSequence(String str) {</span>
<span class="source-line-no">443</span><span id="line-443">                try {</span>
<span class="source-line-no">444</span><span id="line-444">                        new ProteinSequence(str);</span>
<span class="source-line-no">445</span><span id="line-445">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">446</span><span id="line-446"></span>
<span class="source-line-no">447</span><span id="line-447">                        return false;</span>
<span class="source-line-no">448</span><span id="line-448">                }</span>
<span class="source-line-no">449</span><span id="line-449">                return true;</span>
<span class="source-line-no">450</span><span id="line-450">        }</span>
<span class="source-line-no">451</span><span id="line-451"></span>
<span class="source-line-no">452</span><span id="line-452">        private static boolean isDNASequence(String str) {</span>
<span class="source-line-no">453</span><span id="line-453">                try {</span>
<span class="source-line-no">454</span><span id="line-454">                        new DNASequence(str);</span>
<span class="source-line-no">455</span><span id="line-455">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">456</span><span id="line-456"></span>
<span class="source-line-no">457</span><span id="line-457">                        return false;</span>
<span class="source-line-no">458</span><span id="line-458">                }</span>
<span class="source-line-no">459</span><span id="line-459">                return true;</span>
<span class="source-line-no">460</span><span id="line-460">        }</span>
<span class="source-line-no">461</span><span id="line-461"></span>
<span class="source-line-no">462</span><span id="line-462">        private static boolean isRNASequence(String str) {</span>
<span class="source-line-no">463</span><span id="line-463">                try {</span>
<span class="source-line-no">464</span><span id="line-464">                        new RNASequence(str);</span>
<span class="source-line-no">465</span><span id="line-465">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">466</span><span id="line-466"></span>
<span class="source-line-no">467</span><span id="line-467">                        return false;</span>
<span class="source-line-no">468</span><span id="line-468">                }</span>
<span class="source-line-no">469</span><span id="line-469">                return true;</span>
<span class="source-line-no">470</span><span id="line-470"></span>
<span class="source-line-no">471</span><span id="line-471">        }</span>
<span class="source-line-no">472</span><span id="line-472"></span>
<span class="source-line-no">473</span><span id="line-473">        /**</span>
<span class="source-line-no">474</span><span id="line-474">         * Returns the ProteinSequence or null if one can't be created</span>
<span class="source-line-no">475</span><span id="line-475">         * @param str</span>
<span class="source-line-no">476</span><span id="line-476">         * @return</span>
<span class="source-line-no">477</span><span id="line-477">         */</span>
<span class="source-line-no">478</span><span id="line-478">        private static ProteinSequence getProteinSequence(String str) {</span>
<span class="source-line-no">479</span><span id="line-479">                try {</span>
<span class="source-line-no">480</span><span id="line-480">                        ProteinSequence s = new ProteinSequence(str);</span>
<span class="source-line-no">481</span><span id="line-481">                        return s;</span>
<span class="source-line-no">482</span><span id="line-482">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">483</span><span id="line-483"></span>
<span class="source-line-no">484</span><span id="line-484">                        logger.error("Unexpected error when creating ProteinSequence",e);</span>
<span class="source-line-no">485</span><span id="line-485">                }</span>
<span class="source-line-no">486</span><span id="line-486">                return null;</span>
<span class="source-line-no">487</span><span id="line-487">        }</span>
<span class="source-line-no">488</span><span id="line-488"></span>
<span class="source-line-no">489</span><span id="line-489">        /**</span>
<span class="source-line-no">490</span><span id="line-490">         * Returns the DNASequence or null if one can't be created</span>
<span class="source-line-no">491</span><span id="line-491">         * @param str</span>
<span class="source-line-no">492</span><span id="line-492">         * @return</span>
<span class="source-line-no">493</span><span id="line-493">         */</span>
<span class="source-line-no">494</span><span id="line-494">        private static DNASequence getDNASequence(String str) {</span>
<span class="source-line-no">495</span><span id="line-495">                try {</span>
<span class="source-line-no">496</span><span id="line-496">                        DNASequence s = new DNASequence(str);</span>
<span class="source-line-no">497</span><span id="line-497">                        return s;</span>
<span class="source-line-no">498</span><span id="line-498">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">499</span><span id="line-499"></span>
<span class="source-line-no">500</span><span id="line-500">                        logger.error("Unexpected error when creating DNASequence ",e);</span>
<span class="source-line-no">501</span><span id="line-501">                }</span>
<span class="source-line-no">502</span><span id="line-502">                return null;</span>
<span class="source-line-no">503</span><span id="line-503">        }</span>
<span class="source-line-no">504</span><span id="line-504"></span>
<span class="source-line-no">505</span><span id="line-505">        /**</span>
<span class="source-line-no">506</span><span id="line-506">         * Returns the RNASequence or null if one can't be created</span>
<span class="source-line-no">507</span><span id="line-507">         * @param str</span>
<span class="source-line-no">508</span><span id="line-508">         * @return</span>
<span class="source-line-no">509</span><span id="line-509">         */</span>
<span class="source-line-no">510</span><span id="line-510">        private static RNASequence getRNASequence(String str) {</span>
<span class="source-line-no">511</span><span id="line-511">                try {</span>
<span class="source-line-no">512</span><span id="line-512">                        RNASequence s = new RNASequence(str);</span>
<span class="source-line-no">513</span><span id="line-513">                        return s;</span>
<span class="source-line-no">514</span><span id="line-514">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">515</span><span id="line-515"></span>
<span class="source-line-no">516</span><span id="line-516">                        logger.error("Unexpected error when creating RNASequence ",e);</span>
<span class="source-line-no">517</span><span id="line-517">                }</span>
<span class="source-line-no">518</span><span id="line-518">                return null;</span>
<span class="source-line-no">519</span><span id="line-519">        }</span>
<span class="source-line-no">520</span><span id="line-520"></span>
<span class="source-line-no">521</span><span id="line-521">}</span>




























































</pre>
</div>
</main>
</body>
</html>
