<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.structure.io, class: FastaAFPChainConverter">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/**</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on 2013-05-28</span>
<span class="source-line-no">021</span><span id="line-21"> * Created by Douglas Myers-Turnbull</span>
<span class="source-line-no">022</span><span id="line-22"> *</span>
<span class="source-line-no">023</span><span id="line-23"> * @since 3.0.6</span>
<span class="source-line-no">024</span><span id="line-24"> */</span>
<span class="source-line-no">025</span><span id="line-25">package org.biojava.nbio.structure.io;</span>
<span class="source-line-no">026</span><span id="line-26"></span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.alignment.template.AlignedSequence;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.alignment.template.SequencePair;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.structure.*;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.structure.align.model.AFPChain;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.structure.align.util.AlignmentTools;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.structure.align.xml.AFPChainXMLConverter;</span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">034</span><span id="line-34">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">035</span><span id="line-35">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">036</span><span id="line-36">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span>
<span class="source-line-no">037</span><span id="line-37">import org.biojava.nbio.core.sequence.io.CasePreservingProteinSequenceCreator;</span>
<span class="source-line-no">038</span><span id="line-38">import org.biojava.nbio.core.sequence.io.FastaReader;</span>
<span class="source-line-no">039</span><span id="line-39">import org.biojava.nbio.core.sequence.io.GenericFastaHeaderParser;</span>
<span class="source-line-no">040</span><span id="line-40">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</span>
<span class="source-line-no">041</span><span id="line-41">import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;</span>
<span class="source-line-no">042</span><span id="line-42">import org.biojava.nbio.core.sequence.template.Sequence;</span>
<span class="source-line-no">043</span><span id="line-43">import org.biojava.nbio.core.util.SequenceTools;</span>
<span class="source-line-no">044</span><span id="line-44">import org.slf4j.Logger;</span>
<span class="source-line-no">045</span><span id="line-45">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">046</span><span id="line-46"></span>
<span class="source-line-no">047</span><span id="line-47">import java.io.File;</span>
<span class="source-line-no">048</span><span id="line-48">import java.io.FileInputStream;</span>
<span class="source-line-no">049</span><span id="line-49">import java.io.IOException;</span>
<span class="source-line-no">050</span><span id="line-50">import java.io.InputStream;</span>
<span class="source-line-no">051</span><span id="line-51">import java.util.*;</span>
<span class="source-line-no">052</span><span id="line-52"></span>
<span class="source-line-no">053</span><span id="line-53">/**</span>
<span class="source-line-no">054</span><span id="line-54"> * A collection of static utilities to convert between {@link AFPChain AFPChains} and FastaSequences.</span>
<span class="source-line-no">055</span><span id="line-55"> *</span>
<span class="source-line-no">056</span><span id="line-56"> * @author dmyersturnbull</span>
<span class="source-line-no">057</span><span id="line-57"> * @see StructureSequenceMatcher</span>
<span class="source-line-no">058</span><span id="line-58"> * @see FastaStructureParser</span>
<span class="source-line-no">059</span><span id="line-59"> * @see SeqRes2AtomAligner</span>
<span class="source-line-no">060</span><span id="line-60"> */</span>
<span class="source-line-no">061</span><span id="line-61">public class FastaAFPChainConverter {</span>
<span class="source-line-no">062</span><span id="line-62"></span>
<span class="source-line-no">063</span><span id="line-63">        private final static Logger logger = LoggerFactory.getLogger(FastaAFPChainConverter.class);</span>
<span class="source-line-no">064</span><span id="line-64"></span>
<span class="source-line-no">065</span><span id="line-65"></span>
<span class="source-line-no">066</span><span id="line-66">        public static AFPChain cpFastaToAfpChain(String first, String second, Structure structure, int cpSite) throws StructureException, CompoundNotFoundException {</span>
<span class="source-line-no">067</span><span id="line-67">                ProteinSequence s1 = new ProteinSequence(first);</span>
<span class="source-line-no">068</span><span id="line-68">                s1.setUserCollection(getAlignedUserCollection(first));</span>
<span class="source-line-no">069</span><span id="line-69">                ProteinSequence s2 = new ProteinSequence(second);</span>
<span class="source-line-no">070</span><span id="line-70">                s2.setUserCollection(getAlignedUserCollection(second));</span>
<span class="source-line-no">071</span><span id="line-71">                return cpFastaToAfpChain(s1, s2, structure, cpSite);</span>
<span class="source-line-no">072</span><span id="line-72">        }</span>
<span class="source-line-no">073</span><span id="line-73"></span>
<span class="source-line-no">074</span><span id="line-74">        /**</span>
<span class="source-line-no">075</span><span id="line-75">         * Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site</span>
<span class="source-line-no">076</span><span id="line-76">         * &lt;code&gt;cpSite&lt;/code&gt; residues to the right.</span>
<span class="source-line-no">077</span><span id="line-77">         *</span>
<span class="source-line-no">078</span><span id="line-78">         * @param fastaFile A FASTA file containing exactly 2 sequences, the first unpermuted and the second permuted</span>
<span class="source-line-no">079</span><span id="line-79">         * @param cpSite</span>
<span class="source-line-no">080</span><span id="line-80">         *            The number of residues from the beginning of the sequence at which the circular permutation site occurs; can be positive or negative; values greater than the length of the sequence</span>
<span class="source-line-no">081</span><span id="line-81">         *            are acceptable</span>
<span class="source-line-no">082</span><span id="line-82">         * @throws IOException</span>
<span class="source-line-no">083</span><span id="line-83">         * @throws StructureException</span>
<span class="source-line-no">084</span><span id="line-84">         */</span>
<span class="source-line-no">085</span><span id="line-85">        public static AFPChain cpFastaToAfpChain(File fastaFile, Structure structure, int cpSite) throws IOException, StructureException {</span>
<span class="source-line-no">086</span><span id="line-86">                InputStream inStream = new FileInputStream(fastaFile);</span>
<span class="source-line-no">087</span><span id="line-87">                SequenceCreatorInterface&lt;AminoAcidCompound&gt; creator = new CasePreservingProteinSequenceCreator(AminoAcidCompoundSet.getAminoAcidCompoundSet());</span>
<span class="source-line-no">088</span><span id="line-88">                SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser = new GenericFastaHeaderParser&lt;&gt;();</span>
<span class="source-line-no">089</span><span id="line-89">                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;&gt;(inStream, headerParser, creator);</span>
<span class="source-line-no">090</span><span id="line-90">                Map&lt;String, ProteinSequence&gt; sequences = fastaReader.process();</span>
<span class="source-line-no">091</span><span id="line-91">                inStream.close();</span>
<span class="source-line-no">092</span><span id="line-92">                Iterator&lt;ProteinSequence&gt; iter = sequences.values().iterator();</span>
<span class="source-line-no">093</span><span id="line-93">                ProteinSequence first = iter.next();</span>
<span class="source-line-no">094</span><span id="line-94">                ProteinSequence second = iter.next();</span>
<span class="source-line-no">095</span><span id="line-95">                return cpFastaToAfpChain(first, second, structure, cpSite);</span>
<span class="source-line-no">096</span><span id="line-96">        }</span>
<span class="source-line-no">097</span><span id="line-97"></span>
<span class="source-line-no">098</span><span id="line-98">        /**</span>
<span class="source-line-no">099</span><span id="line-99">         * Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site</span>
<span class="source-line-no">100</span><span id="line-100">         * &lt;code&gt;cpSite&lt;/code&gt; residues to the right.</span>
<span class="source-line-no">101</span><span id="line-101">         *</span>
<span class="source-line-no">102</span><span id="line-102">         * @param first The unpermuted sequence</span>
<span class="source-line-no">103</span><span id="line-103">         * @param second The sequence permuted by cpSite</span>
<span class="source-line-no">104</span><span id="line-104">         * @param cpSite</span>
<span class="source-line-no">105</span><span id="line-105">         *            The number of residues from the beginning of the sequence at which the circular permutation site occurs; can be positive or negative; values greater than the length of the sequence</span>
<span class="source-line-no">106</span><span id="line-106">         *            are acceptable</span>
<span class="source-line-no">107</span><span id="line-107">         * @throws StructureException</span>
<span class="source-line-no">108</span><span id="line-108">         */</span>
<span class="source-line-no">109</span><span id="line-109">        public static AFPChain cpFastaToAfpChain(ProteinSequence first, ProteinSequence second, Structure structure, int cpSite)</span>
<span class="source-line-no">110</span><span id="line-110">                        throws StructureException {</span>
<span class="source-line-no">111</span><span id="line-111"></span>
<span class="source-line-no">112</span><span id="line-112">                if (structure == null) {</span>
<span class="source-line-no">113</span><span id="line-113">                        throw new IllegalArgumentException("The structure is null");</span>
<span class="source-line-no">114</span><span id="line-114">                }</span>
<span class="source-line-no">115</span><span id="line-115"></span>
<span class="source-line-no">116</span><span id="line-116">                if (first == null) {</span>
<span class="source-line-no">117</span><span id="line-117">                        throw new IllegalArgumentException("The sequence is null");</span>
<span class="source-line-no">118</span><span id="line-118">                }</span>
<span class="source-line-no">119</span><span id="line-119"></span>
<span class="source-line-no">120</span><span id="line-120">                // we need to find the ungapped CP site</span>
<span class="source-line-no">121</span><span id="line-121">                int gappedCpShift = 0;</span>
<span class="source-line-no">122</span><span id="line-122">                int ungappedCpShift = 0;</span>
<span class="source-line-no">123</span><span id="line-123">                while (ungappedCpShift &lt; Math.abs(cpSite)) {</span>
<span class="source-line-no">124</span><span id="line-124">                        char c;</span>
<span class="source-line-no">125</span><span id="line-125">                        try {</span>
<span class="source-line-no">126</span><span id="line-126">                                if (cpSite &lt;= 0) {</span>
<span class="source-line-no">127</span><span id="line-127">                                        c = second.getSequenceAsString().charAt(gappedCpShift);</span>
<span class="source-line-no">128</span><span id="line-128">                                } else {</span>
<span class="source-line-no">129</span><span id="line-129">                                        c = second.getSequenceAsString().charAt(first.getLength()-1 - gappedCpShift);</span>
<span class="source-line-no">130</span><span id="line-130">                                }</span>
<span class="source-line-no">131</span><span id="line-131">                        } catch (StringIndexOutOfBoundsException e) {</span>
<span class="source-line-no">132</span><span id="line-132">                                throw new IllegalArgumentException("CP site of " + cpSite + " is wrong");</span>
<span class="source-line-no">133</span><span id="line-133">                        }</span>
<span class="source-line-no">134</span><span id="line-134">                        if (c != '-') {</span>
<span class="source-line-no">135</span><span id="line-135">                                ungappedCpShift++;</span>
<span class="source-line-no">136</span><span id="line-136">                        }</span>
<span class="source-line-no">137</span><span id="line-137">                        gappedCpShift++;</span>
<span class="source-line-no">138</span><span id="line-138">                }</span>
<span class="source-line-no">139</span><span id="line-139"></span>
<span class="source-line-no">140</span><span id="line-140">                Atom[] ca1 = StructureTools.getRepresentativeAtomArray(structure);</span>
<span class="source-line-no">141</span><span id="line-141">                Atom[] ca2 =  StructureTools.getRepresentativeAtomArray(structure); // can't use cloneCAArray because it doesn't set parent group.chain.structure</span>
<span class="source-line-no">142</span><span id="line-142"></span>
<span class="source-line-no">143</span><span id="line-143">                ProteinSequence antipermuted = null;</span>
<span class="source-line-no">144</span><span id="line-144">                try {</span>
<span class="source-line-no">145</span><span id="line-145">                        antipermuted = new ProteinSequence(SequenceTools.permuteCyclic(second.getSequenceAsString(), gappedCpShift));</span>
<span class="source-line-no">146</span><span id="line-146">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">147</span><span id="line-147">                        // this can't happen, the original sequence comes from a ProteinSequence</span>
<span class="source-line-no">148</span><span id="line-148">                        logger.error("Unexpected error while creating protein sequence: {}. This is most likely a bug.",e.getMessage() );</span>
<span class="source-line-no">149</span><span id="line-149">                }</span>
<span class="source-line-no">150</span><span id="line-150"></span>
<span class="source-line-no">151</span><span id="line-151">                ResidueNumber[] residues = StructureSequenceMatcher.matchSequenceToStructure(first, structure);</span>
<span class="source-line-no">152</span><span id="line-152">                ResidueNumber[] antipermutedResidues = StructureSequenceMatcher.matchSequenceToStructure(antipermuted, structure);</span>
<span class="source-line-no">153</span><span id="line-153"></span>
<span class="source-line-no">154</span><span id="line-154">                ResidueNumber[] nonpermutedResidues = new ResidueNumber[antipermutedResidues.length];</span>
<span class="source-line-no">155</span><span id="line-155">                SequenceTools.permuteCyclic(antipermutedResidues, nonpermutedResidues, -gappedCpShift);</span>
<span class="source-line-no">156</span><span id="line-156"></span>
<span class="source-line-no">157</span><span id="line-157">                // nullify ResidueNumbers that have a lowercase sequence character</span>
<span class="source-line-no">158</span><span id="line-158">                if (first.getUserCollection() != null) {</span>
<span class="source-line-no">159</span><span id="line-159">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(first, residues);</span>
<span class="source-line-no">160</span><span id="line-160">                }</span>
<span class="source-line-no">161</span><span id="line-161">                if (second.getUserCollection() != null) {</span>
<span class="source-line-no">162</span><span id="line-162">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(second, nonpermutedResidues);</span>
<span class="source-line-no">163</span><span id="line-163">                }</span>
<span class="source-line-no">164</span><span id="line-164"></span>
<span class="source-line-no">165</span><span id="line-165">//              for (int i = 0; i &lt; residues.length; i++) {</span>
<span class="source-line-no">166</span><span id="line-166">//                      if (residues[i] == null) {</span>
<span class="source-line-no">167</span><span id="line-167">//                              System.out.print("=");</span>
<span class="source-line-no">168</span><span id="line-168">//                      } else {</span>
<span class="source-line-no">169</span><span id="line-169">//                              System.out.print(sequence.getSequenceAsString().charAt(i));</span>
<span class="source-line-no">170</span><span id="line-170">//                      }</span>
<span class="source-line-no">171</span><span id="line-171">//              }</span>
<span class="source-line-no">172</span><span id="line-172">//              System.out.println();</span>
<span class="source-line-no">173</span><span id="line-173">//              for (int i = 0; i &lt; residues.length; i++) {</span>
<span class="source-line-no">174</span><span id="line-174">//                      if (nonpermutedResidues[i] == null) {</span>
<span class="source-line-no">175</span><span id="line-175">//                              System.out.print("=");</span>
<span class="source-line-no">176</span><span id="line-176">//                      } else {</span>
<span class="source-line-no">177</span><span id="line-177">//                              System.out.print(second.getSequenceAsString().charAt(i));</span>
<span class="source-line-no">178</span><span id="line-178">//                      }</span>
<span class="source-line-no">179</span><span id="line-179">//              }</span>
<span class="source-line-no">180</span><span id="line-180">//              System.out.println();</span>
<span class="source-line-no">181</span><span id="line-181"></span>
<span class="source-line-no">182</span><span id="line-182">                return buildAlignment(ca1, ca2, residues, nonpermutedResidues);</span>
<span class="source-line-no">183</span><span id="line-183"></span>
<span class="source-line-no">184</span><span id="line-184">        }</span>
<span class="source-line-no">185</span><span id="line-185"></span>
<span class="source-line-no">186</span><span id="line-186">        /**</span>
<span class="source-line-no">187</span><span id="line-187">         * Reads the file {@code fastaFile}, expecting exactly two sequences which give a pairwise alignment. Uses this and two structures to create an AFPChain corresponding to the alignment. Uses a</span>
<span class="source-line-no">188</span><span id="line-188">         * {@link CasePreservingProteinSequenceCreator} and assumes that a residue is aligned if and only if it is given by an uppercase letter.</span>
<span class="source-line-no">189</span><span id="line-189">         *</span>
<span class="source-line-no">190</span><span id="line-190">         * @see #fastaToAfpChain(ProteinSequence, ProteinSequence, Structure, Structure)</span>
<span class="source-line-no">191</span><span id="line-191">         * @throws IOException</span>
<span class="source-line-no">192</span><span id="line-192">         * @throws StructureException</span>
<span class="source-line-no">193</span><span id="line-193">         */</span>
<span class="source-line-no">194</span><span id="line-194">        public static AFPChain fastaFileToAfpChain(File fastaFile, Structure structure1, Structure structure2)</span>
<span class="source-line-no">195</span><span id="line-195">                        throws IOException, StructureException {</span>
<span class="source-line-no">196</span><span id="line-196">                InputStream inStream = new FileInputStream(fastaFile);</span>
<span class="source-line-no">197</span><span id="line-197">                SequenceCreatorInterface&lt;AminoAcidCompound&gt; creator = new CasePreservingProteinSequenceCreator(</span>
<span class="source-line-no">198</span><span id="line-198">                                AminoAcidCompoundSet.getAminoAcidCompoundSet());</span>
<span class="source-line-no">199</span><span id="line-199">                SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser = new GenericFastaHeaderParser&lt;&gt;();</span>
<span class="source-line-no">200</span><span id="line-200">                FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; fastaReader = new FastaReader&lt;&gt;(</span>
<span class="source-line-no">201</span><span id="line-201">                                inStream, headerParser, creator);</span>
<span class="source-line-no">202</span><span id="line-202">                Map&lt;String, ProteinSequence&gt; sequences = fastaReader.process();</span>
<span class="source-line-no">203</span><span id="line-203">                inStream.close();</span>
<span class="source-line-no">204</span><span id="line-204">                return fastaToAfpChain(sequences, structure1, structure2);</span>
<span class="source-line-no">205</span><span id="line-205">        }</span>
<span class="source-line-no">206</span><span id="line-206"></span>
<span class="source-line-no">207</span><span id="line-207">        /**</span>
<span class="source-line-no">208</span><span id="line-208">         * Returns an AFPChain corresponding to the alignment between {@code structure1} and {@code structure2}, which is given by the gapped protein sequences {@code sequence1} and {@code sequence2}. The</span>
<span class="source-line-no">209</span><span id="line-209">         * sequences need not correspond to the entire structures, since local alignment is performed to match the sequences to structures.</span>
<span class="source-line-no">210</span><span id="line-210">         * @throws StructureException</span>
<span class="source-line-no">211</span><span id="line-211">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">212</span><span id="line-212">         */</span>
<span class="source-line-no">213</span><span id="line-213">        public static AFPChain fastaStringToAfpChain(String sequence1, String sequence2, Structure structure1,</span>
<span class="source-line-no">214</span><span id="line-214">                        Structure structure2) throws StructureException, CompoundNotFoundException {</span>
<span class="source-line-no">215</span><span id="line-215">                ProteinSequence seq1 = new ProteinSequence(sequence1);</span>
<span class="source-line-no">216</span><span id="line-216">                ProteinSequence seq2 = new ProteinSequence(sequence2);</span>
<span class="source-line-no">217</span><span id="line-217">                return fastaToAfpChain(seq1, seq2, structure1, structure2);</span>
<span class="source-line-no">218</span><span id="line-218">        }</span>
<span class="source-line-no">219</span><span id="line-219"></span>
<span class="source-line-no">220</span><span id="line-220">        /**</span>
<span class="source-line-no">221</span><span id="line-221">         * Uses two sequences each with a corresponding structure to create an AFPChain corresponding to the alignment. Provided only for convenience since FastaReaders return such maps.</span>
<span class="source-line-no">222</span><span id="line-222">         *</span>
<span class="source-line-no">223</span><span id="line-223">         * @param sequences</span>
<span class="source-line-no">224</span><span id="line-224">         *            A Map containing exactly two entries from sequence names as Strings to gapped ProteinSequences; the name is ignored</span>
<span class="source-line-no">225</span><span id="line-225">         * @see #fastaToAfpChain(ProteinSequence, ProteinSequence, Structure, Structure)</span>
<span class="source-line-no">226</span><span id="line-226">         * @throws StructureException</span>
<span class="source-line-no">227</span><span id="line-227">         */</span>
<span class="source-line-no">228</span><span id="line-228">        public static AFPChain fastaToAfpChain(Map&lt;String, ProteinSequence&gt; sequences, Structure structure1,</span>
<span class="source-line-no">229</span><span id="line-229">                        Structure structure2) throws StructureException {</span>
<span class="source-line-no">230</span><span id="line-230"></span>
<span class="source-line-no">231</span><span id="line-231">                if (sequences.size() != 2) {</span>
<span class="source-line-no">232</span><span id="line-232">                        throw new IllegalArgumentException("There must be exactly 2 sequences, but there were " + sequences.size());</span>
<span class="source-line-no">233</span><span id="line-233">                }</span>
<span class="source-line-no">234</span><span id="line-234"></span>
<span class="source-line-no">235</span><span id="line-235">                if (structure1 == null || structure2 == null) {</span>
<span class="source-line-no">236</span><span id="line-236">                        throw new IllegalArgumentException("A structure is null");</span>
<span class="source-line-no">237</span><span id="line-237">                }</span>
<span class="source-line-no">238</span><span id="line-238"></span>
<span class="source-line-no">239</span><span id="line-239">                List&lt;ProteinSequence&gt; seqs = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">240</span><span id="line-240">                List&lt;String&gt; names = new ArrayList&lt;&gt;(2);</span>
<span class="source-line-no">241</span><span id="line-241">                for (Map.Entry&lt;String, ProteinSequence&gt; entry : sequences.entrySet()) {</span>
<span class="source-line-no">242</span><span id="line-242">                        seqs.add(entry.getValue());</span>
<span class="source-line-no">243</span><span id="line-243">                        names.add(entry.getKey());</span>
<span class="source-line-no">244</span><span id="line-244">                }</span>
<span class="source-line-no">245</span><span id="line-245"></span>
<span class="source-line-no">246</span><span id="line-246">                return fastaToAfpChain(seqs.get(0), seqs.get(1), structure1, structure2);</span>
<span class="source-line-no">247</span><span id="line-247">        }</span>
<span class="source-line-no">248</span><span id="line-248"></span>
<span class="source-line-no">249</span><span id="line-249">        /**</span>
<span class="source-line-no">250</span><span id="line-250">         * TODO Write comment</span>
<span class="source-line-no">251</span><span id="line-251">         * @param sequence1</span>
<span class="source-line-no">252</span><span id="line-252">         * @param sequence2</span>
<span class="source-line-no">253</span><span id="line-253">         * @param structure1</span>
<span class="source-line-no">254</span><span id="line-254">         * @param structure2</span>
<span class="source-line-no">255</span><span id="line-255">         * @return</span>
<span class="source-line-no">256</span><span id="line-256">         * @throws StructureException</span>
<span class="source-line-no">257</span><span id="line-257">         * @throws CompoundNotFoundException</span>
<span class="source-line-no">258</span><span id="line-258">         */</span>
<span class="source-line-no">259</span><span id="line-259">        public static AFPChain fastaToAfpChain(String sequence1, String sequence2, Structure structure1,</span>
<span class="source-line-no">260</span><span id="line-260">                        Structure structure2) throws StructureException, CompoundNotFoundException {</span>
<span class="source-line-no">261</span><span id="line-261">                ProteinSequence s1 = new ProteinSequence(sequence1);</span>
<span class="source-line-no">262</span><span id="line-262">                s1.setUserCollection(getAlignedUserCollection(sequence1));</span>
<span class="source-line-no">263</span><span id="line-263">                ProteinSequence s2 = new ProteinSequence(sequence2);</span>
<span class="source-line-no">264</span><span id="line-264">                s2.setUserCollection(getAlignedUserCollection(sequence2));</span>
<span class="source-line-no">265</span><span id="line-265">                return fastaToAfpChain(s1, s2, structure1, structure2);</span>
<span class="source-line-no">266</span><span id="line-266">        }</span>
<span class="source-line-no">267</span><span id="line-267"></span>
<span class="source-line-no">268</span><span id="line-268">        /**</span>
<span class="source-line-no">269</span><span id="line-269">         * Returns an AFPChain corresponding to the alignment between {@code structure1} and {@code structure2}, which is given by the gapped protein sequences {@code sequence1} and {@code sequence2}. The</span>
<span class="source-line-no">270</span><span id="line-270">         * sequences need not correspond to the entire structures, since local alignment is performed to match the sequences to structures. Assumes that a residue is aligned if and only if it is given by</span>
<span class="source-line-no">271</span><span id="line-271">         * an uppercase letter.</span>
<span class="source-line-no">272</span><span id="line-272">         * @param sequence1 &lt;em&gt;Must&lt;/em&gt; have {@link ProteinSequence#getUserCollection()} set to document upper- and lower-case as aligned and unaligned; see {@link #getAlignedUserCollection(String)}</span>
<span class="source-line-no">273</span><span id="line-273">         * @throws StructureException</span>
<span class="source-line-no">274</span><span id="line-274">         */</span>
<span class="source-line-no">275</span><span id="line-275">        public static AFPChain fastaToAfpChain(ProteinSequence sequence1, ProteinSequence sequence2, Structure structure1,</span>
<span class="source-line-no">276</span><span id="line-276">                        Structure structure2) throws StructureException {</span>
<span class="source-line-no">277</span><span id="line-277"></span>
<span class="source-line-no">278</span><span id="line-278">                if (structure1 == null || structure2 == null) {</span>
<span class="source-line-no">279</span><span id="line-279">                        throw new IllegalArgumentException("A structure is null");</span>
<span class="source-line-no">280</span><span id="line-280">                }</span>
<span class="source-line-no">281</span><span id="line-281"></span>
<span class="source-line-no">282</span><span id="line-282">                if (sequence1 == null || sequence2 == null) {</span>
<span class="source-line-no">283</span><span id="line-283">                        throw new IllegalArgumentException("A sequence is null");</span>
<span class="source-line-no">284</span><span id="line-284">                }</span>
<span class="source-line-no">285</span><span id="line-285"></span>
<span class="source-line-no">286</span><span id="line-286">                Atom[] ca1 = StructureTools.getRepresentativeAtomArray(structure1);</span>
<span class="source-line-no">287</span><span id="line-287">                Atom[] ca2 = StructureTools.getRepresentativeAtomArray(structure2);</span>
<span class="source-line-no">288</span><span id="line-288"></span>
<span class="source-line-no">289</span><span id="line-289">                ResidueNumber[] residues1 = StructureSequenceMatcher.matchSequenceToStructure(sequence1, structure1);</span>
<span class="source-line-no">290</span><span id="line-290">                ResidueNumber[] residues2 = StructureSequenceMatcher.matchSequenceToStructure(sequence2, structure2);</span>
<span class="source-line-no">291</span><span id="line-291"></span>
<span class="source-line-no">292</span><span id="line-292">                // nullify ResidueNumbers that have a lowercase sequence character</span>
<span class="source-line-no">293</span><span id="line-293">                if (sequence1.getUserCollection() != null) {</span>
<span class="source-line-no">294</span><span id="line-294">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(sequence1, residues1);</span>
<span class="source-line-no">295</span><span id="line-295">                }</span>
<span class="source-line-no">296</span><span id="line-296">                if (sequence2.getUserCollection() != null) {</span>
<span class="source-line-no">297</span><span id="line-297">                        CasePreservingProteinSequenceCreator.setLowercaseToNull(sequence2, residues2);</span>
<span class="source-line-no">298</span><span id="line-298">                }</span>
<span class="source-line-no">299</span><span id="line-299"></span>
<span class="source-line-no">300</span><span id="line-300">                return buildAlignment(ca1, ca2, residues1, residues2);</span>
<span class="source-line-no">301</span><span id="line-301"></span>
<span class="source-line-no">302</span><span id="line-302">        }</span>
<span class="source-line-no">303</span><span id="line-303"></span>
<span class="source-line-no">304</span><span id="line-304">        /**</span>
<span class="source-line-no">305</span><span id="line-305">         * Provided only for convenience.</span>
<span class="source-line-no">306</span><span id="line-306">         *</span>
<span class="source-line-no">307</span><span id="line-307">         * @see #fastaToAfpChain(ProteinSequence, ProteinSequence, Structure, Structure)</span>
<span class="source-line-no">308</span><span id="line-308">         * @throws StructureException</span>
<span class="source-line-no">309</span><span id="line-309">         */</span>
<span class="source-line-no">310</span><span id="line-310">        public static AFPChain fastaToAfpChain(SequencePair&lt;Sequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt; alignment,</span>
<span class="source-line-no">311</span><span id="line-311">                        Structure structure1, Structure structure2) throws StructureException {</span>
<span class="source-line-no">312</span><span id="line-312">                List&lt;AlignedSequence&lt;Sequence&lt;AminoAcidCompound&gt;, AminoAcidCompound&gt;&gt; seqs = alignment.getAlignedSequences();</span>
<span class="source-line-no">313</span><span id="line-313">                StringBuilder sb1 = new StringBuilder();</span>
<span class="source-line-no">314</span><span id="line-314">                for (AminoAcidCompound a : seqs.get(0)) {</span>
<span class="source-line-no">315</span><span id="line-315">                        sb1.append(a.getBase());</span>
<span class="source-line-no">316</span><span id="line-316">                }</span>
<span class="source-line-no">317</span><span id="line-317">                try {</span>
<span class="source-line-no">318</span><span id="line-318">                        ProteinSequence seq1 = new ProteinSequence(sb1.toString());</span>
<span class="source-line-no">319</span><span id="line-319">                        StringBuilder sb2 = new StringBuilder();</span>
<span class="source-line-no">320</span><span id="line-320">                        for (AminoAcidCompound a : seqs.get(1)) {</span>
<span class="source-line-no">321</span><span id="line-321">                                sb1.append(a.getBase());</span>
<span class="source-line-no">322</span><span id="line-322">                        }</span>
<span class="source-line-no">323</span><span id="line-323">                        ProteinSequence seq2 = new ProteinSequence(sb2.toString());</span>
<span class="source-line-no">324</span><span id="line-324">                        LinkedHashMap&lt;String, ProteinSequence&gt; map = new LinkedHashMap&lt;&gt;();</span>
<span class="source-line-no">325</span><span id="line-325">                        map.put(structure1.getName(), seq1);</span>
<span class="source-line-no">326</span><span id="line-326">                        map.put(structure2.getName(), seq2);</span>
<span class="source-line-no">327</span><span id="line-327">                        return fastaToAfpChain(map, structure1, structure2);</span>
<span class="source-line-no">328</span><span id="line-328">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">329</span><span id="line-329">                        logger.error("Unexpected error while creating protein sequences: {}. This is most likely a bug.",e.getMessage());</span>
<span class="source-line-no">330</span><span id="line-330">                        return null;</span>
<span class="source-line-no">331</span><span id="line-331">                }</span>
<span class="source-line-no">332</span><span id="line-332">        }</span>
<span class="source-line-no">333</span><span id="line-333"></span>
<span class="source-line-no">334</span><span id="line-334">        /**</span>
<span class="source-line-no">335</span><span id="line-335">         * Builds an {@link AFPChain} from already-matched arrays of atoms and residues.</span>
<span class="source-line-no">336</span><span id="line-336">         *</span>
<span class="source-line-no">337</span><span id="line-337">         * @param ca1</span>
<span class="source-line-no">338</span><span id="line-338">         *            An array of atoms in the first structure</span>
<span class="source-line-no">339</span><span id="line-339">         * @param ca2</span>
<span class="source-line-no">340</span><span id="line-340">         *            An array of atoms in the second structure</span>
<span class="source-line-no">341</span><span id="line-341">         * @param residues1</span>
<span class="source-line-no">342</span><span id="line-342">         *            An array of {@link ResidueNumber ResidueNumbers} in the first structure that are aligned. Only null ResidueNumbers are considered to be unaligned</span>
<span class="source-line-no">343</span><span id="line-343">         * @param residues2</span>
<span class="source-line-no">344</span><span id="line-344">         *            An array of {@link ResidueNumber ResidueNumbers} in the second structure that are aligned. Only null ResidueNumbers are considered to be unaligned</span>
<span class="source-line-no">345</span><span id="line-345">         * @throws StructureException</span>
<span class="source-line-no">346</span><span id="line-346">         */</span>
<span class="source-line-no">347</span><span id="line-347">        private static AFPChain buildAlignment(Atom[] ca1, Atom[] ca2, ResidueNumber[] residues1, ResidueNumber[] residues2)</span>
<span class="source-line-no">348</span><span id="line-348">                        throws StructureException {</span>
<span class="source-line-no">349</span><span id="line-349"></span>
<span class="source-line-no">350</span><span id="line-350">                // remove any gap</span>
<span class="source-line-no">351</span><span id="line-351">                // this includes the ones introduced by the nullifying above</span>
<span class="source-line-no">352</span><span id="line-352">                List&lt;ResidueNumber&gt; alignedResiduesList1 = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">353</span><span id="line-353">                List&lt;ResidueNumber&gt; alignedResiduesList2 = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">354</span><span id="line-354">                for (int i = 0; i &lt; residues1.length; i++) {</span>
<span class="source-line-no">355</span><span id="line-355">                        if (residues1[i] != null &amp;&amp; residues2[i] != null) {</span>
<span class="source-line-no">356</span><span id="line-356">                                alignedResiduesList1.add(residues1[i]);</span>
<span class="source-line-no">357</span><span id="line-357">                                alignedResiduesList2.add(residues2[i]);</span>
<span class="source-line-no">358</span><span id="line-358">                        }</span>
<span class="source-line-no">359</span><span id="line-359">                }</span>
<span class="source-line-no">360</span><span id="line-360"></span>
<span class="source-line-no">361</span><span id="line-361">                ResidueNumber[] alignedResidues1 = alignedResiduesList1.toArray(new ResidueNumber[alignedResiduesList1.size()]);</span>
<span class="source-line-no">362</span><span id="line-362">                ResidueNumber[] alignedResidues2 = alignedResiduesList2.toArray(new ResidueNumber[alignedResiduesList2.size()]);</span>
<span class="source-line-no">363</span><span id="line-363"></span>
<span class="source-line-no">364</span><span id="line-364">                AFPChain afpChain = AlignmentTools.createAFPChain(ca1, ca2, alignedResidues1, alignedResidues2);</span>
<span class="source-line-no">365</span><span id="line-365">                afpChain.setAlgorithmName("unknown");</span>
<span class="source-line-no">366</span><span id="line-366"></span>
<span class="source-line-no">367</span><span id="line-367">                AlignmentTools.updateSuperposition(afpChain, ca1, ca2);</span>
<span class="source-line-no">368</span><span id="line-368"></span>
<span class="source-line-no">369</span><span id="line-369">                afpChain.setBlockSize(new int[] {afpChain.getNrEQR()});</span>
<span class="source-line-no">370</span><span id="line-370">                afpChain.setBlockRmsd(new double[] {afpChain.getTotalRmsdOpt()});</span>
<span class="source-line-no">371</span><span id="line-371">                afpChain.setBlockGap(new int[] {afpChain.getGapLen()});</span>
<span class="source-line-no">372</span><span id="line-372"></span>
<span class="source-line-no">373</span><span id="line-373">                return afpChain;</span>
<span class="source-line-no">374</span><span id="line-374"></span>
<span class="source-line-no">375</span><span id="line-375">        }</span>
<span class="source-line-no">376</span><span id="line-376"></span>
<span class="source-line-no">377</span><span id="line-377">        /**</span>
<span class="source-line-no">378</span><span id="line-378">         * Takes a protein sequence string with capital and lowercase letters and sets its {@link ProteinSequence#getUserCollection() user collection} to record which letters are uppercase (aligned) and which are lowercase (unaligned).</span>
<span class="source-line-no">379</span><span id="line-379">         * @param sequence Make sure &lt;em&gt;not&lt;/em&gt; to use {@link ProteinSequence#getSequenceAsString()} for this, as it won't preserve upper- and lower-case</span>
<span class="source-line-no">380</span><span id="line-380">         */</span>
<span class="source-line-no">381</span><span id="line-381">        public static List&lt;Object&gt; getAlignedUserCollection(String sequence) {</span>
<span class="source-line-no">382</span><span id="line-382">                List&lt;Object&gt; aligned = new ArrayList&lt;&gt;(sequence.length());</span>
<span class="source-line-no">383</span><span id="line-383">                for (char c : sequence.toCharArray()) {</span>
<span class="source-line-no">384</span><span id="line-384">                        aligned.add(Character.isUpperCase(c));</span>
<span class="source-line-no">385</span><span id="line-385">                }</span>
<span class="source-line-no">386</span><span id="line-386">                return aligned;</span>
<span class="source-line-no">387</span><span id="line-387">        }</span>
<span class="source-line-no">388</span><span id="line-388"></span>
<span class="source-line-no">389</span><span id="line-389">        /**</span>
<span class="source-line-no">390</span><span id="line-390">         * Prints out the XML representation of an AFPChain from a file containing exactly two FASTA sequences.</span>
<span class="source-line-no">391</span><span id="line-391">         *</span>
<span class="source-line-no">392</span><span id="line-392">         * @param args</span>
<span class="source-line-no">393</span><span id="line-393">         *            A String array of fasta-file structure-1-name structure-2-name</span>
<span class="source-line-no">394</span><span id="line-394">         * @throws StructureException</span>
<span class="source-line-no">395</span><span id="line-395">         * @throws IOException</span>
<span class="source-line-no">396</span><span id="line-396">         */</span>
<span class="source-line-no">397</span><span id="line-397">        public static void main(String[] args) throws StructureException, IOException {</span>
<span class="source-line-no">398</span><span id="line-398">                if (args.length != 3) {</span>
<span class="source-line-no">399</span><span id="line-399">                        System.err.println("Usage: FastaAFPChainConverter fasta-file structure-1-name structure-2-name");</span>
<span class="source-line-no">400</span><span id="line-400">                        return;</span>
<span class="source-line-no">401</span><span id="line-401">                }</span>
<span class="source-line-no">402</span><span id="line-402">                File fasta = new File(args[0]);</span>
<span class="source-line-no">403</span><span id="line-403">                Structure structure1 = StructureTools.getStructure(args[1]);</span>
<span class="source-line-no">404</span><span id="line-404">                Structure structure2 = StructureTools.getStructure(args[2]);</span>
<span class="source-line-no">405</span><span id="line-405">                if (structure1 == null) throw new IllegalArgumentException("No structure for " + args[1] + " was found");</span>
<span class="source-line-no">406</span><span id="line-406">                if (structure2 == null) throw new IllegalArgumentException("No structure for " + args[2] + " was found");</span>
<span class="source-line-no">407</span><span id="line-407">                AFPChain afpChain = fastaFileToAfpChain(fasta, structure1, structure2);</span>
<span class="source-line-no">408</span><span id="line-408">                String xml = AFPChainXMLConverter.toXML(afpChain);</span>
<span class="source-line-no">409</span><span id="line-409">                System.out.println(xml);</span>
<span class="source-line-no">410</span><span id="line-410">        }</span>
<span class="source-line-no">411</span><span id="line-411"></span>
<span class="source-line-no">412</span><span id="line-412">}</span>




























































</pre>
</div>
</main>
</body>
</html>
