<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.structure.io, class: FastaStructureParser">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> */</span>
<span class="source-line-no">021</span><span id="line-21">package org.biojava.nbio.structure.io;</span>
<span class="source-line-no">022</span><span id="line-22"></span>
<span class="source-line-no">023</span><span id="line-23">import org.biojava.nbio.structure.ResidueNumber;</span>
<span class="source-line-no">024</span><span id="line-24">import org.biojava.nbio.structure.Structure;</span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.structure.StructureException;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.structure.align.util.AtomCache;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.sequence.io.FastaReader;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;</span>
<span class="source-line-no">032</span><span id="line-32"></span>
<span class="source-line-no">033</span><span id="line-33">import java.io.File;</span>
<span class="source-line-no">034</span><span id="line-34">import java.io.FileNotFoundException;</span>
<span class="source-line-no">035</span><span id="line-35">import java.io.IOException;</span>
<span class="source-line-no">036</span><span id="line-36">import java.io.InputStream;</span>
<span class="source-line-no">037</span><span id="line-37">import java.util.LinkedHashMap;</span>
<span class="source-line-no">038</span><span id="line-38">import java.util.Map;</span>
<span class="source-line-no">039</span><span id="line-39"></span>
<span class="source-line-no">040</span><span id="line-40"></span>
<span class="source-line-no">041</span><span id="line-41">/**</span>
<span class="source-line-no">042</span><span id="line-42"> * Reads a protein sequence from a fasta file and attempts to match it to a</span>
<span class="source-line-no">043</span><span id="line-43"> * 3D structure. Any gaps ('-') in the fasta file are preserved as null atoms in</span>
<span class="source-line-no">044</span><span id="line-44"> * the output, allowing structural alignments to be read from fasta files.</span>
<span class="source-line-no">045</span><span id="line-45"> *</span>
<span class="source-line-no">046</span><span id="line-46"> * &lt;p&gt;Structures are loaded from an AtomCache. For this to work, the accession</span>
<span class="source-line-no">047</span><span id="line-47"> * for each protein should be parsed from the fasta header line into a form</span>
<span class="source-line-no">048</span><span id="line-48"> * understood by {@link AtomCache#getStructure(String)}.</span>
<span class="source-line-no">049</span><span id="line-49"> *</span>
<span class="source-line-no">050</span><span id="line-50"> * &lt;p&gt;Lowercase letters are sometimes used to specify unaligned residues.</span>
<span class="source-line-no">051</span><span id="line-51"> * This information can be preserved by using a CasePreservingSequenceCreator,</span>
<span class="source-line-no">052</span><span id="line-52"> * which allows the case of residues to be accessed through the</span>
<span class="source-line-no">053</span><span id="line-53"> * {@link ProteinSequence#getUserCollection()} method.</span>
<span class="source-line-no">054</span><span id="line-54"> *</span>
<span class="source-line-no">055</span><span id="line-55"> * @author Spencer Bliven</span>
<span class="source-line-no">056</span><span id="line-56"> *</span>
<span class="source-line-no">057</span><span id="line-57"> */</span>
<span class="source-line-no">058</span><span id="line-58">public class FastaStructureParser {</span>
<span class="source-line-no">059</span><span id="line-59"></span>
<span class="source-line-no">060</span><span id="line-60">        // inputs</span>
<span class="source-line-no">061</span><span id="line-61">        private FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; reader;</span>
<span class="source-line-no">062</span><span id="line-62">        private AtomCache cache;</span>
<span class="source-line-no">063</span><span id="line-63"></span>
<span class="source-line-no">064</span><span id="line-64">        // cache processed data</span>
<span class="source-line-no">065</span><span id="line-65">        private String[] accessions;</span>
<span class="source-line-no">066</span><span id="line-66">        private ProteinSequence[] sequences;</span>
<span class="source-line-no">067</span><span id="line-67">        private Structure[] structures;</span>
<span class="source-line-no">068</span><span id="line-68">        private ResidueNumber[][] residues;</span>
<span class="source-line-no">069</span><span id="line-69"></span>
<span class="source-line-no">070</span><span id="line-70">        public FastaStructureParser(InputStream is,</span>
<span class="source-line-no">071</span><span id="line-71">                        SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser,</span>
<span class="source-line-no">072</span><span id="line-72">                        SequenceCreatorInterface&lt;AminoAcidCompound&gt; sequenceCreator,</span>
<span class="source-line-no">073</span><span id="line-73">                        AtomCache cache)</span>
<span class="source-line-no">074</span><span id="line-74">        {</span>
<span class="source-line-no">075</span><span id="line-75">                this(new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(</span>
<span class="source-line-no">076</span><span id="line-76">                                is, headerParser, sequenceCreator),cache);</span>
<span class="source-line-no">077</span><span id="line-77">        }</span>
<span class="source-line-no">078</span><span id="line-78"></span>
<span class="source-line-no">079</span><span id="line-79">        public FastaStructureParser(File file,</span>
<span class="source-line-no">080</span><span id="line-80">                        SequenceHeaderParserInterface&lt;ProteinSequence, AminoAcidCompound&gt; headerParser,</span>
<span class="source-line-no">081</span><span id="line-81">                        SequenceCreatorInterface&lt;AminoAcidCompound&gt; sequenceCreator,</span>
<span class="source-line-no">082</span><span id="line-82">                        AtomCache cache) throws FileNotFoundException</span>
<span class="source-line-no">083</span><span id="line-83">        {</span>
<span class="source-line-no">084</span><span id="line-84">                this(new FastaReader&lt;ProteinSequence, AminoAcidCompound&gt;(</span>
<span class="source-line-no">085</span><span id="line-85">                                file, headerParser, sequenceCreator), cache);</span>
<span class="source-line-no">086</span><span id="line-86">        }</span>
<span class="source-line-no">087</span><span id="line-87"></span>
<span class="source-line-no">088</span><span id="line-88">        public FastaStructureParser(FastaReader&lt;ProteinSequence, AminoAcidCompound&gt; reader,</span>
<span class="source-line-no">089</span><span id="line-89">                        AtomCache cache) {</span>
<span class="source-line-no">090</span><span id="line-90">                this.reader = reader;</span>
<span class="source-line-no">091</span><span id="line-91">                this.cache = cache;</span>
<span class="source-line-no">092</span><span id="line-92">                this.accessions = null;</span>
<span class="source-line-no">093</span><span id="line-93">                this.sequences = null;</span>
<span class="source-line-no">094</span><span id="line-94">                this.structures = null;</span>
<span class="source-line-no">095</span><span id="line-95">                this.residues = null;</span>
<span class="source-line-no">096</span><span id="line-96">        }</span>
<span class="source-line-no">097</span><span id="line-97"></span>
<span class="source-line-no">098</span><span id="line-98"></span>
<span class="source-line-no">099</span><span id="line-99">        /**</span>
<span class="source-line-no">100</span><span id="line-100">         * Parses the fasta file and loads it into memory.</span>
<span class="source-line-no">101</span><span id="line-101">         *</span>
<span class="source-line-no">102</span><span id="line-102">         * Information can be subsequently accessed through</span>
<span class="source-line-no">103</span><span id="line-103">         * {@link #getSequences()},</span>
<span class="source-line-no">104</span><span id="line-104">         * {@link #getStructures()},</span>
<span class="source-line-no">105</span><span id="line-105">         * {@link #getResidues()}, and</span>
<span class="source-line-no">106</span><span id="line-106">         * {@link #getAccessions()}.</span>
<span class="source-line-no">107</span><span id="line-107">         *</span>
<span class="source-line-no">108</span><span id="line-108">         * @throws IOException</span>
<span class="source-line-no">109</span><span id="line-109">         * @throws StructureException</span>
<span class="source-line-no">110</span><span id="line-110">         */</span>
<span class="source-line-no">111</span><span id="line-111">        public void process() throws IOException, StructureException {</span>
<span class="source-line-no">112</span><span id="line-112">                if(sequences == null) { // only process once, then return cached values</span>
<span class="source-line-no">113</span><span id="line-113">                        Map&lt;String, ProteinSequence&gt; sequenceMap = reader.process();</span>
<span class="source-line-no">114</span><span id="line-114"></span>
<span class="source-line-no">115</span><span id="line-115">                        sequences = sequenceMap.values().toArray(new ProteinSequence[0]);</span>
<span class="source-line-no">116</span><span id="line-116">                        accessions = new String[sequences.length];</span>
<span class="source-line-no">117</span><span id="line-117">                        structures = new Structure[sequences.length];</span>
<span class="source-line-no">118</span><span id="line-118">                        residues = new ResidueNumber[sequences.length][];</span>
<span class="source-line-no">119</span><span id="line-119"></span>
<span class="source-line-no">120</span><span id="line-120">                        // Match each sequence  to a series of PDB Residue numbers</span>
<span class="source-line-no">121</span><span id="line-121">                        for(int i=0;i&lt;sequences.length;i++) {</span>
<span class="source-line-no">122</span><span id="line-122">                                accessions[i] = sequences[i].getAccession().getID();</span>
<span class="source-line-no">123</span><span id="line-123"></span>
<span class="source-line-no">124</span><span id="line-124">                                //System.out.println("Fetching "+accession);</span>
<span class="source-line-no">125</span><span id="line-125">                                structures[i] = cache.getStructure(accessions[i]);</span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">                                residues[i] = StructureSequenceMatcher.matchSequenceToStructure(sequences[i], structures[i]);</span>
<span class="source-line-no">128</span><span id="line-128"></span>
<span class="source-line-no">129</span><span id="line-129">                                assert( residues[i].length == sequences[i].getLength());</span>
<span class="source-line-no">130</span><span id="line-130">                        }</span>
<span class="source-line-no">131</span><span id="line-131">                }</span>
<span class="source-line-no">132</span><span id="line-132">        }</span>
<span class="source-line-no">133</span><span id="line-133"></span>
<span class="source-line-no">134</span><span id="line-134"></span>
<span class="source-line-no">135</span><span id="line-135">        /**</span>
<span class="source-line-no">136</span><span id="line-136">         * Gets the protein sequences read from the Fasta file.</span>
<span class="source-line-no">137</span><span id="line-137">         * Returns null if {@link #process()} has not been called.</span>
<span class="source-line-no">138</span><span id="line-138">         * @return An array ProteinSequences from</span>
<span class="source-line-no">139</span><span id="line-139">         *  parsing the fasta file, or null if process() hasn't been called.</span>
<span class="source-line-no">140</span><span id="line-140">         */</span>
<span class="source-line-no">141</span><span id="line-141">        public ProteinSequence[] getSequences() {</span>
<span class="source-line-no">142</span><span id="line-142">                return sequences;</span>
<span class="source-line-no">143</span><span id="line-143">        }</span>
<span class="source-line-no">144</span><span id="line-144"></span>
<span class="source-line-no">145</span><span id="line-145">        /**</span>
<span class="source-line-no">146</span><span id="line-146">         * Gets the protein structures mapped from the Fasta file.</span>
<span class="source-line-no">147</span><span id="line-147">         * Returns null if {@link #process()} has not been called.</span>
<span class="source-line-no">148</span><span id="line-148">         * @return An array of Structures for each protein</span>
<span class="source-line-no">149</span><span id="line-149">         *  in the fasta file, or null if process() hasn't been called.</span>
<span class="source-line-no">150</span><span id="line-150">         */</span>
<span class="source-line-no">151</span><span id="line-151">        public Structure[] getStructures() {</span>
<span class="source-line-no">152</span><span id="line-152">                return structures;</span>
<span class="source-line-no">153</span><span id="line-153">        }</span>
<span class="source-line-no">154</span><span id="line-154"></span>
<span class="source-line-no">155</span><span id="line-155">        /**</span>
<span class="source-line-no">156</span><span id="line-156">         * For each residue in the fasta file, return the ResidueNumber in the</span>
<span class="source-line-no">157</span><span id="line-157">         * corresponding structure. If the residue cannot be found in the structure,</span>
<span class="source-line-no">158</span><span id="line-158">         * that entry will be null. This can happen if that residue was not included</span>
<span class="source-line-no">159</span><span id="line-159">         * in the PDB file (eg disordered residues), if the fasta sequence does not</span>
<span class="source-line-no">160</span><span id="line-160">         * match the PDB sequence, or if errors occur during the matching process.</span>
<span class="source-line-no">161</span><span id="line-161">         * @return A 2D array of ResidueNumbers, or null if process() hasn't been called.</span>
<span class="source-line-no">162</span><span id="line-162">         * @see StructureSequenceMatcher#matchSequenceToStructure(ProteinSequence, Structure)</span>
<span class="source-line-no">163</span><span id="line-163">         */</span>
<span class="source-line-no">164</span><span id="line-164">        public ResidueNumber[][] getResidues() {</span>
<span class="source-line-no">165</span><span id="line-165">                return residues;</span>
<span class="source-line-no">166</span><span id="line-166">        }</span>
<span class="source-line-no">167</span><span id="line-167"></span>
<span class="source-line-no">168</span><span id="line-168">        /**</span>
<span class="source-line-no">169</span><span id="line-169">         * Gets the protein accessions mapped from the Fasta file.</span>
<span class="source-line-no">170</span><span id="line-170">         * Returns null if {@link #process()} has not been called.</span>
<span class="source-line-no">171</span><span id="line-171">         * @return An array of Structures for each protein</span>
<span class="source-line-no">172</span><span id="line-172">         *  in the fasta file, or null if process() hasn't been called.</span>
<span class="source-line-no">173</span><span id="line-173">         */</span>
<span class="source-line-no">174</span><span id="line-174">        public String[] getAccessions() {</span>
<span class="source-line-no">175</span><span id="line-175">                return accessions;</span>
<span class="source-line-no">176</span><span id="line-176">        }</span>
<span class="source-line-no">177</span><span id="line-177">}</span>




























































</pre>
</div>
</main>
</body>
</html>
