<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.structure.io, class: SeqRes2AtomAligner">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/*</span>
<span class="source-line-no">002</span><span id="line-2"> *                  BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created on Aug 23, 2007</span>
<span class="source-line-no">021</span><span id="line-21"> *</span>
<span class="source-line-no">022</span><span id="line-22"> */</span>
<span class="source-line-no">023</span><span id="line-23"></span>
<span class="source-line-no">024</span><span id="line-24">package org.biojava.nbio.structure.io;</span>
<span class="source-line-no">025</span><span id="line-25"></span>
<span class="source-line-no">026</span><span id="line-26"></span>
<span class="source-line-no">027</span><span id="line-27">import java.util.ArrayList;</span>
<span class="source-line-no">028</span><span id="line-28">import java.util.HashMap;</span>
<span class="source-line-no">029</span><span id="line-29">import java.util.Iterator;</span>
<span class="source-line-no">030</span><span id="line-30">import java.util.List;</span>
<span class="source-line-no">031</span><span id="line-31">import java.util.Map;</span>
<span class="source-line-no">032</span><span id="line-32"></span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.alignment.Alignments;</span>
<span class="source-line-no">034</span><span id="line-34">import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;</span>
<span class="source-line-no">035</span><span id="line-35">import org.biojava.nbio.alignment.SimpleGapPenalty;</span>
<span class="source-line-no">036</span><span id="line-36">import org.biojava.nbio.alignment.template.GapPenalty;</span>
<span class="source-line-no">037</span><span id="line-37">import org.biojava.nbio.alignment.template.PairwiseSequenceAligner;</span>
<span class="source-line-no">038</span><span id="line-38">import org.biojava.nbio.core.alignment.matrices.SubstitutionMatrixHelper;</span>
<span class="source-line-no">039</span><span id="line-39">import org.biojava.nbio.core.alignment.template.SequencePair;</span>
<span class="source-line-no">040</span><span id="line-40">import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;</span>
<span class="source-line-no">041</span><span id="line-41">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">042</span><span id="line-42">import org.biojava.nbio.core.sequence.DNASequence;</span>
<span class="source-line-no">043</span><span id="line-43">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">044</span><span id="line-44">import org.biojava.nbio.core.sequence.RNASequence;</span>
<span class="source-line-no">045</span><span id="line-45">import org.biojava.nbio.core.sequence.compound.AmbiguityDNACompoundSet;</span>
<span class="source-line-no">046</span><span id="line-46">import org.biojava.nbio.core.sequence.compound.AmbiguityDNARNAHybridCompoundSet;</span>
<span class="source-line-no">047</span><span id="line-47">import org.biojava.nbio.core.sequence.compound.AmbiguityRNACompoundSet;</span>
<span class="source-line-no">048</span><span id="line-48">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">049</span><span id="line-49">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span>
<span class="source-line-no">050</span><span id="line-50">import org.biojava.nbio.core.sequence.compound.DNACompoundSet;</span>
<span class="source-line-no">051</span><span id="line-51">import org.biojava.nbio.core.sequence.compound.NucleotideCompound;</span>
<span class="source-line-no">052</span><span id="line-52">import org.biojava.nbio.core.sequence.template.Compound;</span>
<span class="source-line-no">053</span><span id="line-53">import org.biojava.nbio.core.sequence.template.Sequence;</span>
<span class="source-line-no">054</span><span id="line-54">import org.biojava.nbio.structure.AminoAcid;</span>
<span class="source-line-no">055</span><span id="line-55">import org.biojava.nbio.structure.Atom;</span>
<span class="source-line-no">056</span><span id="line-56">import org.biojava.nbio.structure.Chain;</span>
<span class="source-line-no">057</span><span id="line-57">import org.biojava.nbio.structure.Group;</span>
<span class="source-line-no">058</span><span id="line-58">import org.biojava.nbio.structure.GroupType;</span>
<span class="source-line-no">059</span><span id="line-59">import org.biojava.nbio.structure.HetatomImpl;</span>
<span class="source-line-no">060</span><span id="line-60">import org.biojava.nbio.structure.NucleotideImpl;</span>
<span class="source-line-no">061</span><span id="line-61">import org.biojava.nbio.structure.ResidueNumber;</span>
<span class="source-line-no">062</span><span id="line-62">import org.biojava.nbio.structure.Structure;</span>
<span class="source-line-no">063</span><span id="line-63">import org.biojava.nbio.structure.chem.ChemComp;</span>
<span class="source-line-no">064</span><span id="line-64">import org.biojava.nbio.structure.chem.PolymerType;</span>
<span class="source-line-no">065</span><span id="line-65">import org.biojava.nbio.structure.chem.ResidueType;</span>
<span class="source-line-no">066</span><span id="line-66">import org.slf4j.Logger;</span>
<span class="source-line-no">067</span><span id="line-67">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">068</span><span id="line-68"></span>
<span class="source-line-no">069</span><span id="line-69"></span>
<span class="source-line-no">070</span><span id="line-70">/**</span>
<span class="source-line-no">071</span><span id="line-71"> * Aligns the SEQRES residues to the ATOM residues.</span>
<span class="source-line-no">072</span><span id="line-72"> * The AminoAcids that can be matched between the two of them will be set in the SEQRES</span>
<span class="source-line-no">073</span><span id="line-73"> * chains</span>
<span class="source-line-no">074</span><span id="line-74"> *</span>
<span class="source-line-no">075</span><span id="line-75"> *</span>
<span class="source-line-no">076</span><span id="line-76"> * @author Andreas Prlic</span>
<span class="source-line-no">077</span><span id="line-77"> * @author Jose Duarte</span>
<span class="source-line-no">078</span><span id="line-78"> */</span>
<span class="source-line-no">079</span><span id="line-79">public class SeqRes2AtomAligner {</span>
<span class="source-line-no">080</span><span id="line-80"></span>
<span class="source-line-no">081</span><span id="line-81">        private static final Logger logger = LoggerFactory.getLogger(SeqRes2AtomAligner.class);</span>
<span class="source-line-no">082</span><span id="line-82"></span>
<span class="source-line-no">083</span><span id="line-83"></span>
<span class="source-line-no">084</span><span id="line-84"></span>
<span class="source-line-no">085</span><span id="line-85">        private String alignmentString;</span>
<span class="source-line-no">086</span><span id="line-86"></span>
<span class="source-line-no">087</span><span id="line-87">        public SeqRes2AtomAligner(){</span>
<span class="source-line-no">088</span><span id="line-88">                logger.debug("initialising SeqRes2AtomAligner");</span>
<span class="source-line-no">089</span><span id="line-89">                alignmentString = "";</span>
<span class="source-line-no">090</span><span id="line-90">        }</span>
<span class="source-line-no">091</span><span id="line-91"></span>
<span class="source-line-no">092</span><span id="line-92">        public String getAlignmentString() {</span>
<span class="source-line-no">093</span><span id="line-93">                return alignmentString;</span>
<span class="source-line-no">094</span><span id="line-94">        }</span>
<span class="source-line-no">095</span><span id="line-95"></span>
<span class="source-line-no">096</span><span id="line-96">        /**</span>
<span class="source-line-no">097</span><span id="line-97">         *</span>
<span class="source-line-no">098</span><span id="line-98">         * @param seqRes</span>
<span class="source-line-no">099</span><span id="line-99">         * @param atomList</span>
<span class="source-line-no">100</span><span id="line-100">         * @param useChainId if true chainId (Chain.getId) is used for matching,</span>
<span class="source-line-no">101</span><span id="line-101">         * if false chainName (Chain.getName) is used</span>
<span class="source-line-no">102</span><span id="line-102">         * @return</span>
<span class="source-line-no">103</span><span id="line-103">         */</span>
<span class="source-line-no">104</span><span id="line-104">        public static Chain getMatchingAtomRes(Chain seqRes, List&lt;Chain&gt; atomList, boolean useChainId)</span>
<span class="source-line-no">105</span><span id="line-105">        {</span>
<span class="source-line-no">106</span><span id="line-106">                Iterator&lt;Chain&gt; iter = atomList.iterator();</span>
<span class="source-line-no">107</span><span id="line-107">                while(iter.hasNext()){</span>
<span class="source-line-no">108</span><span id="line-108">                        Chain atomChain = iter.next();</span>
<span class="source-line-no">109</span><span id="line-109"></span>
<span class="source-line-no">110</span><span id="line-110">                        String atomChainId = null;</span>
<span class="source-line-no">111</span><span id="line-111">                        String seqResChainId = null;</span>
<span class="source-line-no">112</span><span id="line-112">                        if (useChainId) {</span>
<span class="source-line-no">113</span><span id="line-113">                                atomChainId = atomChain.getId();</span>
<span class="source-line-no">114</span><span id="line-114">                                seqResChainId = seqRes.getId();</span>
<span class="source-line-no">115</span><span id="line-115">                        } else {</span>
<span class="source-line-no">116</span><span id="line-116">                                atomChainId = atomChain.getName();</span>
<span class="source-line-no">117</span><span id="line-117">                                seqResChainId = seqRes.getName();</span>
<span class="source-line-no">118</span><span id="line-118"></span>
<span class="source-line-no">119</span><span id="line-119">                        }</span>
<span class="source-line-no">120</span><span id="line-120"></span>
<span class="source-line-no">121</span><span id="line-121">                        if ( atomChainId.equals(seqResChainId)){</span>
<span class="source-line-no">122</span><span id="line-122">                                return atomChain;</span>
<span class="source-line-no">123</span><span id="line-123">                        }</span>
<span class="source-line-no">124</span><span id="line-124"></span>
<span class="source-line-no">125</span><span id="line-125">                }</span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">                logger.info("Could not match SEQRES chainID asymId:" + seqRes.getId() + " authId:"+ seqRes.getName() +"  to ATOM chains!, size of atom chain: " + atomList.size());</span>
<span class="source-line-no">128</span><span id="line-128">                return null;</span>
<span class="source-line-no">129</span><span id="line-129">        }</span>
<span class="source-line-no">130</span><span id="line-130"></span>
<span class="source-line-no">131</span><span id="line-131"></span>
<span class="source-line-no">132</span><span id="line-132"></span>
<span class="source-line-no">133</span><span id="line-133"></span>
<span class="source-line-no">134</span><span id="line-134">        public void align(Structure s, List&lt;Chain&gt; seqResList){</span>
<span class="source-line-no">135</span><span id="line-135"></span>
<span class="source-line-no">136</span><span id="line-136">                List&lt;Chain&gt; atomList   = s.getModel(0);</span>
<span class="source-line-no">137</span><span id="line-137"></span>
<span class="source-line-no">138</span><span id="line-138"></span>
<span class="source-line-no">139</span><span id="line-139">                for (Chain seqRes: seqResList){</span>
<span class="source-line-no">140</span><span id="line-140"></span>
<span class="source-line-no">141</span><span id="line-141">                                Chain atomRes = getMatchingAtomRes(seqRes,atomList,false);</span>
<span class="source-line-no">142</span><span id="line-142">                                if ( atomRes == null)</span>
<span class="source-line-no">143</span><span id="line-143">                                        continue;</span>
<span class="source-line-no">144</span><span id="line-144"></span>
<span class="source-line-no">145</span><span id="line-145">                                mapSeqresRecords(atomRes,seqRes);</span>
<span class="source-line-no">146</span><span id="line-146"></span>
<span class="source-line-no">147</span><span id="line-147"></span>
<span class="source-line-no">148</span><span id="line-148">                }</span>
<span class="source-line-no">149</span><span id="line-149"></span>
<span class="source-line-no">150</span><span id="line-150">        }</span>
<span class="source-line-no">151</span><span id="line-151"></span>
<span class="source-line-no">152</span><span id="line-152">        /**</span>
<span class="source-line-no">153</span><span id="line-153">         * Map the seqRes groups to the atomRes chain.</span>
<span class="source-line-no">154</span><span id="line-154">         * Updates the atomRes chain object with the mapped data</span>
<span class="source-line-no">155</span><span id="line-155">         * The seqRes chain should not be needed after this and atomRes should be further used.</span>
<span class="source-line-no">156</span><span id="line-156">         *</span>
<span class="source-line-no">157</span><span id="line-157">         * @param atomRes the chain containing ATOM groups (in atomGroups slot). This chain</span>
<span class="source-line-no">158</span><span id="line-158">         * is modified to contain in its seqresGroups slot the mapped atom groups</span>
<span class="source-line-no">159</span><span id="line-159">         * @param seqRes the chain containing SEQRES groups (in atomGroups slot). This chain</span>
<span class="source-line-no">160</span><span id="line-160">         * is not modified</span>
<span class="source-line-no">161</span><span id="line-161">         */</span>
<span class="source-line-no">162</span><span id="line-162">        public void mapSeqresRecords(Chain atomRes, Chain seqRes) {</span>
<span class="source-line-no">163</span><span id="line-163">                List&lt;Group&gt; seqResGroups = seqRes.getAtomGroups();</span>
<span class="source-line-no">164</span><span id="line-164">                List&lt;Group&gt; atmResGroups = atomRes.getAtomGroups();</span>
<span class="source-line-no">165</span><span id="line-165"></span>
<span class="source-line-no">166</span><span id="line-166"></span>
<span class="source-line-no">167</span><span id="line-167"></span>
<span class="source-line-no">168</span><span id="line-168">                logger.debug("Comparing ATOM {} ({} groups) to SEQRES {} ({} groups) ",</span>
<span class="source-line-no">169</span><span id="line-169">                                atomRes.getId(), atmResGroups.size(), seqRes.getId(), seqResGroups.size());</span>
<span class="source-line-no">170</span><span id="line-170"></span>
<span class="source-line-no">171</span><span id="line-171"></span>
<span class="source-line-no">172</span><span id="line-172">                List&lt;Group&gt; matchedGroups = trySimpleMatch(seqResGroups, atmResGroups);</span>
<span class="source-line-no">173</span><span id="line-173"></span>
<span class="source-line-no">174</span><span id="line-174">                if ( matchedGroups != null) {</span>
<span class="source-line-no">175</span><span id="line-175">                        // update the new SEQRES list</span>
<span class="source-line-no">176</span><span id="line-176">                        atomRes.setSeqResGroups(matchedGroups);</span>
<span class="source-line-no">177</span><span id="line-177">                        return;</span>
<span class="source-line-no">178</span><span id="line-178">                }</span>
<span class="source-line-no">179</span><span id="line-179"></span>
<span class="source-line-no">180</span><span id="line-180">                logger.debug("Could not map SEQRES to ATOM records easily, need to align...");</span>
<span class="source-line-no">181</span><span id="line-181"></span>
<span class="source-line-no">182</span><span id="line-182">                int numAminosSeqres = seqRes.getAtomGroups(GroupType.AMINOACID).size();</span>
<span class="source-line-no">183</span><span id="line-183">                int numNucleotidesSeqres = seqRes.getAtomGroups(GroupType.NUCLEOTIDE).size();</span>
<span class="source-line-no">184</span><span id="line-184"></span>
<span class="source-line-no">185</span><span id="line-185">                if ( numAminosSeqres &lt; 1) {</span>
<span class="source-line-no">186</span><span id="line-186"></span>
<span class="source-line-no">187</span><span id="line-187">                        if ( numNucleotidesSeqres &gt; 1) {</span>
<span class="source-line-no">188</span><span id="line-188"></span>
<span class="source-line-no">189</span><span id="line-189">                                logger.debug("SEQRES chain {} is a nucleotide chain ({} nucleotides), aligning nucleotides...", seqRes.getId(), numNucleotidesSeqres);</span>
<span class="source-line-no">190</span><span id="line-190"></span>
<span class="source-line-no">191</span><span id="line-191">                                alignNucleotideChains(seqRes,atomRes);</span>
<span class="source-line-no">192</span><span id="line-192">                                return;</span>
<span class="source-line-no">193</span><span id="line-193">                        } else {</span>
<span class="source-line-no">194</span><span id="line-194"></span>
<span class="source-line-no">195</span><span id="line-195">                                logger.debug("SEQRES chain {} contains {} amino acids and {} nucleotides, ignoring...", seqRes.getId(),numAminosSeqres,numNucleotidesSeqres);</span>
<span class="source-line-no">196</span><span id="line-196"></span>
<span class="source-line-no">197</span><span id="line-197">                                return;</span>
<span class="source-line-no">198</span><span id="line-198">                        }</span>
<span class="source-line-no">199</span><span id="line-199">                }</span>
<span class="source-line-no">200</span><span id="line-200"></span>
<span class="source-line-no">201</span><span id="line-201">                if ( atomRes.getAtomGroups(GroupType.AMINOACID).size() &lt; 1) {</span>
<span class="source-line-no">202</span><span id="line-202">                        logger.debug("ATOM chain {} does not contain amino acids, ignoring...", atomRes.getId());</span>
<span class="source-line-no">203</span><span id="line-203">                        return;</span>
<span class="source-line-no">204</span><span id="line-204">                }</span>
<span class="source-line-no">205</span><span id="line-205"></span>
<span class="source-line-no">206</span><span id="line-206">                logger.debug("Proceeding to do protein alignment for chain {}", atomRes.getId() );</span>
<span class="source-line-no">207</span><span id="line-207"></span>
<span class="source-line-no">208</span><span id="line-208"></span>
<span class="source-line-no">209</span><span id="line-209">                boolean noMatchFound = alignProteinChains(seqResGroups,atomRes.getAtomGroups());</span>
<span class="source-line-no">210</span><span id="line-210">                if ( ! noMatchFound){</span>
<span class="source-line-no">211</span><span id="line-211">                        atomRes.setSeqResGroups(seqResGroups);</span>
<span class="source-line-no">212</span><span id="line-212">                }</span>
<span class="source-line-no">213</span><span id="line-213"></span>
<span class="source-line-no">214</span><span id="line-214">        }</span>
<span class="source-line-no">215</span><span id="line-215"></span>
<span class="source-line-no">216</span><span id="line-216">        private void alignNucleotideChains(Chain seqRes, Chain atomRes) {</span>
<span class="source-line-no">217</span><span id="line-217"></span>
<span class="source-line-no">218</span><span id="line-218">                if ( atomRes.getAtomGroups(GroupType.NUCLEOTIDE).size() &lt; 1) {</span>
<span class="source-line-no">219</span><span id="line-219">                        logger.debug("ATOM chain {} does not contain nucleotides, ignoring...", atomRes.getId());</span>
<span class="source-line-no">220</span><span id="line-220"></span>
<span class="source-line-no">221</span><span id="line-221">                        return;</span>
<span class="source-line-no">222</span><span id="line-222">                }</span>
<span class="source-line-no">223</span><span id="line-223">                logger.debug("Alignment for chain {}", atomRes.getId() );</span>
<span class="source-line-no">224</span><span id="line-224"></span>
<span class="source-line-no">225</span><span id="line-225">                List&lt;Group&gt; seqResGroups = seqRes.getAtomGroups();</span>
<span class="source-line-no">226</span><span id="line-226">                boolean noMatchFound = alignNucleotideGroups(seqResGroups,atomRes.getAtomGroups());</span>
<span class="source-line-no">227</span><span id="line-227">                if ( ! noMatchFound){</span>
<span class="source-line-no">228</span><span id="line-228">                        atomRes.setSeqResGroups(seqResGroups);</span>
<span class="source-line-no">229</span><span id="line-229">                }</span>
<span class="source-line-no">230</span><span id="line-230"></span>
<span class="source-line-no">231</span><span id="line-231">        }</span>
<span class="source-line-no">232</span><span id="line-232"></span>
<span class="source-line-no">233</span><span id="line-233">        /**</span>
<span class="source-line-no">234</span><span id="line-234">         * A simple matching approach that tries to do a 1:1 mapping between SEQRES and ATOM records</span>
<span class="source-line-no">235</span><span id="line-235">         *</span>
<span class="source-line-no">236</span><span id="line-236">         * @param seqResGroups list of seqREs groups</span>
<span class="source-line-no">237</span><span id="line-237">         * @param atmResGroups list of atmRes Groups</span>
<span class="source-line-no">238</span><span id="line-238">         * @return the matching or null if the matching didn't work</span>
<span class="source-line-no">239</span><span id="line-239">         */</span>
<span class="source-line-no">240</span><span id="line-240">        private List&lt;Group&gt; trySimpleMatch(List&lt;Group&gt; seqResGroups,List&lt;Group&gt; atmResGroups) {</span>
<span class="source-line-no">241</span><span id="line-241">                // by default first ATOM position is 1</span>
<span class="source-line-no">242</span><span id="line-242">                //</span>
<span class="source-line-no">243</span><span id="line-243"></span>
<span class="source-line-no">244</span><span id="line-244">                @SuppressWarnings("unchecked")</span>
<span class="source-line-no">245</span><span id="line-245">                List&lt;Group&gt; newSeqResGroups = (ArrayList&lt;Group&gt;)((ArrayList&lt;Group&gt;)seqResGroups).clone();</span>
<span class="source-line-no">246</span><span id="line-246"></span>
<span class="source-line-no">247</span><span id="line-247">                boolean startAt1 = true;</span>
<span class="source-line-no">248</span><span id="line-248"></span>
<span class="source-line-no">249</span><span id="line-249">                for ( int atomResPos = 0 ; atomResPos &lt; atmResGroups.size() ; atomResPos++){</span>
<span class="source-line-no">250</span><span id="line-250"></span>
<span class="source-line-no">251</span><span id="line-251">                        // let's try to match this case</span>
<span class="source-line-no">252</span><span id="line-252">                        Group atomResGroup = atmResGroups.get(atomResPos);</span>
<span class="source-line-no">253</span><span id="line-253"></span>
<span class="source-line-no">254</span><span id="line-254">                        // let's ignore waters</span>
<span class="source-line-no">255</span><span id="line-255">                        if ( atomResGroup.isWater()){</span>
<span class="source-line-no">256</span><span id="line-256">                                continue;</span>
<span class="source-line-no">257</span><span id="line-257">                        }</span>
<span class="source-line-no">258</span><span id="line-258"></span>
<span class="source-line-no">259</span><span id="line-259">                        ResidueNumber atomResNum = atomResGroup.getResidueNumber();</span>
<span class="source-line-no">260</span><span id="line-260"></span>
<span class="source-line-no">261</span><span id="line-261">                        int seqResPos = atomResNum.getSeqNum();</span>
<span class="source-line-no">262</span><span id="line-262"></span>
<span class="source-line-no">263</span><span id="line-263"></span>
<span class="source-line-no">264</span><span id="line-264"></span>
<span class="source-line-no">265</span><span id="line-265">                        if ( seqResPos &lt; 0) {</span>
<span class="source-line-no">266</span><span id="line-266">                                logger.debug("ATOM residue number &lt; 0 : {}", seqResPos);</span>
<span class="source-line-no">267</span><span id="line-267">                                return null;</span>
<span class="source-line-no">268</span><span id="line-268">                        }</span>
<span class="source-line-no">269</span><span id="line-269"></span>
<span class="source-line-no">270</span><span id="line-270">                        if ( seqResPos == 0){</span>
<span class="source-line-no">271</span><span id="line-271">                                // make sure the first SEQRES is matching.</span>
<span class="source-line-no">272</span><span id="line-272">                                Group seqResGroup = seqResGroups.get(0);</span>
<span class="source-line-no">273</span><span id="line-273">                                if (  seqResGroup.getPDBName().equals(atomResGroup.getPDBName())){</span>
<span class="source-line-no">274</span><span id="line-274">                                        // they match! seems in this case the numbering starts with 0...</span>
<span class="source-line-no">275</span><span id="line-275">                                        startAt1 = false;</span>
<span class="source-line-no">276</span><span id="line-276">                                } else {</span>
<span class="source-line-no">277</span><span id="line-277"></span>
<span class="source-line-no">278</span><span id="line-278">                                        logger.debug("SEQRES position 1 ({}) does not match ATOM PDB res num 0 ({})",</span>
<span class="source-line-no">279</span><span id="line-279">                                                        seqResGroup.getPDBName(), atomResGroup.getPDBName());</span>
<span class="source-line-no">280</span><span id="line-280"></span>
<span class="source-line-no">281</span><span id="line-281"></span>
<span class="source-line-no">282</span><span id="line-282">                                        return null;</span>
<span class="source-line-no">283</span><span id="line-283"></span>
<span class="source-line-no">284</span><span id="line-284">                                }</span>
<span class="source-line-no">285</span><span id="line-285">                        }</span>
<span class="source-line-no">286</span><span id="line-286"></span>
<span class="source-line-no">287</span><span id="line-287"></span>
<span class="source-line-no">288</span><span id="line-288">                        if ( startAt1 )</span>
<span class="source-line-no">289</span><span id="line-289">                                seqResPos--;</span>
<span class="source-line-no">290</span><span id="line-290"></span>
<span class="source-line-no">291</span><span id="line-291">                        // another check that would require the alignment approach</span>
<span class="source-line-no">292</span><span id="line-292">                        if ( startAt1 &amp;&amp; seqResPos &gt;=  seqResGroups.size()  )</span>
<span class="source-line-no">293</span><span id="line-293">                        {</span>
<span class="source-line-no">294</span><span id="line-294"></span>
<span class="source-line-no">295</span><span id="line-295">                                // could be a HETATOM...</span>
<span class="source-line-no">296</span><span id="line-296">                                if ( atomResGroup instanceof AminoAcid) {</span>
<span class="source-line-no">297</span><span id="line-297">                                        logger.debug(" ATOM residue nr: " + seqResPos + " &gt; seqres! " + seqResGroups.size() + " " + atomResGroup);</span>
<span class="source-line-no">298</span><span id="line-298">                                        return null;</span>
<span class="source-line-no">299</span><span id="line-299">                                } else if ( atomResGroup instanceof NucleotideImpl) {</span>
<span class="source-line-no">300</span><span id="line-300">                                        logger.debug(" NUCLEOTIDE residue nr: " + seqResPos + " &gt; seqres! " + seqResGroups.size() + " " + atomResGroup);</span>
<span class="source-line-no">301</span><span id="line-301">                                        return null;</span>
<span class="source-line-no">302</span><span id="line-302">                                } else {</span>
<span class="source-line-no">303</span><span id="line-303">                                        // we won't map HETATOM groups...</span>
<span class="source-line-no">304</span><span id="line-304">                                        continue;</span>
<span class="source-line-no">305</span><span id="line-305">                                }</span>
<span class="source-line-no">306</span><span id="line-306">                        }</span>
<span class="source-line-no">307</span><span id="line-307"></span>
<span class="source-line-no">308</span><span id="line-308"></span>
<span class="source-line-no">309</span><span id="line-309">                        //                      if ( seqResPos &lt; 0){</span>
<span class="source-line-no">310</span><span id="line-310">                        //</span>
<span class="source-line-no">311</span><span id="line-311">                        //                              System.err.println("What is going on??? " + atomRes.getChainID() + " " + atomResGroup);</span>
<span class="source-line-no">312</span><span id="line-312">                        //                      }</span>
<span class="source-line-no">313</span><span id="line-313"></span>
<span class="source-line-no">314</span><span id="line-314">                        if ( seqResPos &gt;= seqResGroups.size()){</span>
<span class="source-line-no">315</span><span id="line-315">                                logger.debug("seqres groups don't match atom indices {}", seqResPos);</span>
<span class="source-line-no">316</span><span id="line-316">                                if ( atomResGroup instanceof AminoAcid )</span>
<span class="source-line-no">317</span><span id="line-317">                                        return null;</span>
<span class="source-line-no">318</span><span id="line-318">                                else</span>
<span class="source-line-no">319</span><span id="line-319">                                        continue;</span>
<span class="source-line-no">320</span><span id="line-320">                        }</span>
<span class="source-line-no">321</span><span id="line-321"></span>
<span class="source-line-no">322</span><span id="line-322">                        Group seqResGroup = seqResGroups.get(seqResPos );</span>
<span class="source-line-no">323</span><span id="line-323"></span>
<span class="source-line-no">324</span><span id="line-324">                        if ( ! seqResGroup.getPDBName().trim().equals(atomResGroup.getPDBName().trim())){</span>
<span class="source-line-no">325</span><span id="line-325">                                // a mismatch! something is wrong in the mapping and we need to do an alignment</span>
<span class="source-line-no">326</span><span id="line-326">                                logger.debug("Mismatch of SEQRES pos " + seqResPos + " and ATOM record: " + atomResGroup + " | " + seqResGroup);</span>
<span class="source-line-no">327</span><span id="line-327">                                return null;</span>
<span class="source-line-no">328</span><span id="line-328">                        }</span>
<span class="source-line-no">329</span><span id="line-329"></span>
<span class="source-line-no">330</span><span id="line-330">                        // the two groups are identical and we can merge them</span>
<span class="source-line-no">331</span><span id="line-331">                        // replace the SEQRES group with the ATOM group...</span>
<span class="source-line-no">332</span><span id="line-332"></span>
<span class="source-line-no">333</span><span id="line-333">                        Group replacedGroup = newSeqResGroups.set(seqResPos, atomResGroup);</span>
<span class="source-line-no">334</span><span id="line-334">                        logger.debug("Merging index {}: replaced seqres group {} ({}) with atom group {} ({})",</span>
<span class="source-line-no">335</span><span id="line-335">                                        seqResPos,</span>
<span class="source-line-no">336</span><span id="line-336">                                        replacedGroup.getResidueNumber(), replacedGroup.getPDBName(),</span>
<span class="source-line-no">337</span><span id="line-337">                                        atomResGroup.getResidueNumber(), atomResGroup.getPDBName());</span>
<span class="source-line-no">338</span><span id="line-338"></span>
<span class="source-line-no">339</span><span id="line-339">                }</span>
<span class="source-line-no">340</span><span id="line-340"></span>
<span class="source-line-no">341</span><span id="line-341">                // all went ok. copy over the updated list to the original one.</span>
<span class="source-line-no">342</span><span id="line-342">                // note: if something went wrong, we did not modifiy the original list.</span>
<span class="source-line-no">343</span><span id="line-343">                //seqResGroups = newSeqResGroups;</span>
<span class="source-line-no">344</span><span id="line-344"></span>
<span class="source-line-no">345</span><span id="line-345"></span>
<span class="source-line-no">346</span><span id="line-346">                //                      int pos = -1;</span>
<span class="source-line-no">347</span><span id="line-347">                //                      for (Group g: seqResGroups){</span>
<span class="source-line-no">348</span><span id="line-348">                //                              pos++;</span>
<span class="source-line-no">349</span><span id="line-349">                //                              logger.debug(pos + " " + g);</span>
<span class="source-line-no">350</span><span id="line-350">                //                      }</span>
<span class="source-line-no">351</span><span id="line-351"></span>
<span class="source-line-no">352</span><span id="line-352">                //System.out.println("I:" + seqResGroups);</span>
<span class="source-line-no">353</span><span id="line-353">                // all atom records could get matched correctly!</span>
<span class="source-line-no">354</span><span id="line-354">                return newSeqResGroups;</span>
<span class="source-line-no">355</span><span id="line-355"></span>
<span class="source-line-no">356</span><span id="line-356">        }</span>
<span class="source-line-no">357</span><span id="line-357"></span>
<span class="source-line-no">358</span><span id="line-358">        /**</span>
<span class="source-line-no">359</span><span id="line-359">         * Returns the full sequence of the Atom records of a parent</span>
<span class="source-line-no">360</span><span id="line-360">         * with X instead of HETATMSs. The advantage of this is</span>
<span class="source-line-no">361</span><span id="line-361">         * that it allows us to also align HETATM groups back to the SEQRES.</span>
<span class="source-line-no">362</span><span id="line-362">         * @param groups the list of groups in a parent</span>
<span class="source-line-no">363</span><span id="line-363">         * @param positionIndex a Map to keep track of which group is at which sequence position</span>
<span class="source-line-no">364</span><span id="line-364">         * @param isNucleotideChain whether the atom groups are predominantly nucleotides (the groups represent a nucleotide chain), if true</span>
<span class="source-line-no">365</span><span id="line-365">         * non-standard nucleotides will be represented with ambiguous letter 'N' instead of 'X', if false all non-standard residues will be 'X'</span>
<span class="source-line-no">366</span><span id="line-366">         * @return string representations</span>
<span class="source-line-no">367</span><span id="line-367">         */</span>
<span class="source-line-no">368</span><span id="line-368">        public static String getFullAtomSequence(List&lt;Group&gt; groups, Map&lt;Integer, Integer&gt; positionIndex, boolean isNucleotideChain){</span>
<span class="source-line-no">369</span><span id="line-369"></span>
<span class="source-line-no">370</span><span id="line-370">                StringBuffer sequence = new StringBuffer() ;</span>
<span class="source-line-no">371</span><span id="line-371">                int seqIndex = 0; // track sequence.length()</span>
<span class="source-line-no">372</span><span id="line-372">                for ( int i=0 ; i&lt; groups.size(); i++){</span>
<span class="source-line-no">373</span><span id="line-373">                        Group g = groups.get(i);</span>
<span class="source-line-no">374</span><span id="line-374"></span>
<span class="source-line-no">375</span><span id="line-375">                        if ( g instanceof AminoAcid ){</span>
<span class="source-line-no">376</span><span id="line-376">                                AminoAcid a = (AminoAcid)g;</span>
<span class="source-line-no">377</span><span id="line-377">                                char oneLetter =a.getAminoType();</span>
<span class="source-line-no">378</span><span id="line-378">                                if ( oneLetter == '?')</span>
<span class="source-line-no">379</span><span id="line-379">                                        oneLetter = 'X';</span>
<span class="source-line-no">380</span><span id="line-380"></span>
<span class="source-line-no">381</span><span id="line-381">                                positionIndex.put(seqIndex,i);</span>
<span class="source-line-no">382</span><span id="line-382">                                sequence.append(oneLetter );</span>
<span class="source-line-no">383</span><span id="line-383">                                seqIndex++;</span>
<span class="source-line-no">384</span><span id="line-384">                        } else {</span>
<span class="source-line-no">385</span><span id="line-385"></span>
<span class="source-line-no">386</span><span id="line-386">                                // exclude solvents</span>
<span class="source-line-no">387</span><span id="line-387">                                if (  g.isWater())</span>
<span class="source-line-no">388</span><span id="line-388">                                        continue;</span>
<span class="source-line-no">389</span><span id="line-389"></span>
<span class="source-line-no">390</span><span id="line-390">                                // exclude metals</span>
<span class="source-line-no">391</span><span id="line-391">                                if ( g.size() == 1 ) {</span>
<span class="source-line-no">392</span><span id="line-392"></span>
<span class="source-line-no">393</span><span id="line-393">                                        Atom a = g.getAtom(0);</span>
<span class="source-line-no">394</span><span id="line-394">                                        if ( a == null)</span>
<span class="source-line-no">395</span><span id="line-395">                                                continue;</span>
<span class="source-line-no">396</span><span id="line-396">                                        if (a.getElement().isMetal())</span>
<span class="source-line-no">397</span><span id="line-397">                                                continue;</span>
<span class="source-line-no">398</span><span id="line-398"></span>
<span class="source-line-no">399</span><span id="line-399">                                }</span>
<span class="source-line-no">400</span><span id="line-400"></span>
<span class="source-line-no">401</span><span id="line-401">                                ChemComp cc = g.getChemComp();</span>
<span class="source-line-no">402</span><span id="line-402">                                if ( cc == null) {</span>
<span class="source-line-no">403</span><span id="line-403">                                        logger.debug("No chem comp available for group {}",g.toString());</span>
<span class="source-line-no">404</span><span id="line-404">                                        // not sure what to do in that case!</span>
<span class="source-line-no">405</span><span id="line-405">                                        continue;</span>
<span class="source-line-no">406</span><span id="line-406">                                }</span>
<span class="source-line-no">407</span><span id="line-407">                                if (</span>
<span class="source-line-no">408</span><span id="line-408">                                                ResidueType.lPeptideLinking.equals(cc.getResidueType()) ||</span>
<span class="source-line-no">409</span><span id="line-409">                                                PolymerType.PROTEIN_ONLY.contains(cc.getPolymerType())  ||</span>
<span class="source-line-no">410</span><span id="line-410">                                                PolymerType.POLYNUCLEOTIDE_ONLY.contains(cc.getPolymerType())</span>
<span class="source-line-no">411</span><span id="line-411">                                                ) {</span>
<span class="source-line-no">412</span><span id="line-412">                                        //System.out.println(cc.getOne_letter_code());</span>
<span class="source-line-no">413</span><span id="line-413">                                        String c = cc.getOneLetterCode();</span>
<span class="source-line-no">414</span><span id="line-414">                                        if ( "?".equals(c)) {</span>
<span class="source-line-no">415</span><span id="line-415">                                                if (isNucleotideChain &amp;&amp; PolymerType.POLYNUCLEOTIDE_ONLY.contains(cc.getPolymerType())) {</span>
<span class="source-line-no">416</span><span id="line-416">                                                        // the way to represent unknown nucleotides is with 'N', see https://en.wikipedia.org/wiki/Nucleic_acid_notation</span>
<span class="source-line-no">417</span><span id="line-417">                                                        c = "N";</span>
<span class="source-line-no">418</span><span id="line-418">                                                } else {</span>
<span class="source-line-no">419</span><span id="line-419">                                                        c = "X";</span>
<span class="source-line-no">420</span><span id="line-420">                                                }</span>
<span class="source-line-no">421</span><span id="line-421">                                        }</span>
<span class="source-line-no">422</span><span id="line-422"></span>
<span class="source-line-no">423</span><span id="line-423">                                        // For some unusual cases the het residue can map to 2 or more standard aas and thus give an</span>
<span class="source-line-no">424</span><span id="line-424">                                        // insertion of length &gt;1.</span>
<span class="source-line-no">425</span><span id="line-425">                                        //      e.g. 1: SUI maps to DG  (in 1oew,A)</span>
<span class="source-line-no">426</span><span id="line-426">                                        //              e.g. 2: NRQ maps to MYG (in 3cfh,A)</span>
<span class="source-line-no">427</span><span id="line-427">                                        if (c.length()&gt;1) {</span>
<span class="source-line-no">428</span><span id="line-428">                                                logger.info("Group '{}' maps to more than 1 standard aminoacid: {}.",</span>
<span class="source-line-no">429</span><span id="line-429">                                                                g.toString(), c);</span>
<span class="source-line-no">430</span><span id="line-430">                                        }</span>
<span class="source-line-no">431</span><span id="line-431">                                        // because of the mapping to more than 1 aminoacid, we have</span>
<span class="source-line-no">432</span><span id="line-432">                                        // to loop through it (99% of cases c will have length 1 anyway)</span>
<span class="source-line-no">433</span><span id="line-433">                                        for (int cIdx=0;cIdx&lt;c.length();cIdx++) {</span>
<span class="source-line-no">434</span><span id="line-434">                                                positionIndex.put(seqIndex,i);</span>
<span class="source-line-no">435</span><span id="line-435">                                                sequence.append(c.charAt(cIdx));</span>
<span class="source-line-no">436</span><span id="line-436">                                                seqIndex++;</span>
<span class="source-line-no">437</span><span id="line-437">                                        }</span>
<span class="source-line-no">438</span><span id="line-438">                                } else {</span>
<span class="source-line-no">439</span><span id="line-439">                                        logger.debug("Group {} is not lPeptideLinked, nor PROTEIN_ONLY, nor POLYNUCLEOTIDE_ONLY",g.toString());</span>
<span class="source-line-no">440</span><span id="line-440">                                        continue;</span>
<span class="source-line-no">441</span><span id="line-441">                                }</span>
<span class="source-line-no">442</span><span id="line-442"></span>
<span class="source-line-no">443</span><span id="line-443"></span>
<span class="source-line-no">444</span><span id="line-444">                                //sequence.append("X");</span>
<span class="source-line-no">445</span><span id="line-445">                        }</span>
<span class="source-line-no">446</span><span id="line-446"></span>
<span class="source-line-no">447</span><span id="line-447">                }</span>
<span class="source-line-no">448</span><span id="line-448"></span>
<span class="source-line-no">449</span><span id="line-449">                return sequence.toString();</span>
<span class="source-line-no">450</span><span id="line-450"></span>
<span class="source-line-no">451</span><span id="line-451">        }</span>
<span class="source-line-no">452</span><span id="line-452"></span>
<span class="source-line-no">453</span><span id="line-453"></span>
<span class="source-line-no">454</span><span id="line-454">        private boolean alignNucleotideGroups(List&lt;Group&gt; seqRes, List&lt;Group&gt; atomRes) {</span>
<span class="source-line-no">455</span><span id="line-455"></span>
<span class="source-line-no">456</span><span id="line-456">                Map&lt;Integer,Integer&gt; seqresIndexPosition = new HashMap&lt;&gt;();</span>
<span class="source-line-no">457</span><span id="line-457">                Map&lt;Integer,Integer&gt; atomIndexPosition   = new HashMap&lt;&gt;();</span>
<span class="source-line-no">458</span><span id="line-458"></span>
<span class="source-line-no">459</span><span id="line-459">                String seq1 = getFullAtomSequence(seqRes, seqresIndexPosition, true);</span>
<span class="source-line-no">460</span><span id="line-460">                //</span>
<span class="source-line-no">461</span><span id="line-461">                String seq2 = getFullAtomSequence(atomRes, atomIndexPosition, true);</span>
<span class="source-line-no">462</span><span id="line-462"></span>
<span class="source-line-no">463</span><span id="line-463">                if (seq1.isEmpty() || seq2.isEmpty()) {</span>
<span class="source-line-no">464</span><span id="line-464">                        logger.warn("Could not align nucleotide sequences, at least one of them is empty");</span>
<span class="source-line-no">465</span><span id="line-465">                        return true;</span>
<span class="source-line-no">466</span><span id="line-466">                }</span>
<span class="source-line-no">467</span><span id="line-467"></span>
<span class="source-line-no">468</span><span id="line-468">                logger.debug("align seq1 ("+ seq1.length()+") " + seq1);</span>
<span class="source-line-no">469</span><span id="line-469">                logger.debug("align seq2 ("+ seq2.length()+") " + seq2);</span>
<span class="source-line-no">470</span><span id="line-470"></span>
<span class="source-line-no">471</span><span id="line-471">                Sequence&lt;NucleotideCompound&gt; s1 = getNucleotideSequence(seq1);</span>
<span class="source-line-no">472</span><span id="line-472">                Sequence&lt;NucleotideCompound&gt; s2 = getNucleotideSequence(seq2);</span>
<span class="source-line-no">473</span><span id="line-473"></span>
<span class="source-line-no">474</span><span id="line-474">                if (s1==null || s2==null) return true;</span>
<span class="source-line-no">475</span><span id="line-475"></span>
<span class="source-line-no">476</span><span id="line-476">                if ( ! s1.getCompoundSet().equals(s2.getCompoundSet()) ) {</span>
<span class="source-line-no">477</span><span id="line-477">                        // e.g. trying to align a DNA and an RNA sequence...</span>
<span class="source-line-no">478</span><span id="line-478">                        // test PDB ID: 1A34</span>
<span class="source-line-no">479</span><span id="line-479">                        // try to make both RNA sequence...</span>
<span class="source-line-no">480</span><span id="line-480">                        if ( ! s1.getCompoundSet().equals(AmbiguityRNACompoundSet.getRNACompoundSet())) {</span>
<span class="source-line-no">481</span><span id="line-481">                                try {</span>
<span class="source-line-no">482</span><span id="line-482">                                        s1 = new RNASequence(seq1,AmbiguityRNACompoundSet.getRNACompoundSet());</span>
<span class="source-line-no">483</span><span id="line-483">                                } catch (CompoundNotFoundException ex) {</span>
<span class="source-line-no">484</span><span id="line-484">                                        logger.warn("Could not align DNA and RNA compound sets: " + seq1);</span>
<span class="source-line-no">485</span><span id="line-485">                                        return true;</span>
<span class="source-line-no">486</span><span id="line-486">                                }</span>
<span class="source-line-no">487</span><span id="line-487">                        }</span>
<span class="source-line-no">488</span><span id="line-488"></span>
<span class="source-line-no">489</span><span id="line-489">                        if( ! s2.getCompoundSet().equals(AmbiguityRNACompoundSet.getRNACompoundSet())) {</span>
<span class="source-line-no">490</span><span id="line-490">                                try {</span>
<span class="source-line-no">491</span><span id="line-491">                                        s2 = new RNASequence(seq2,AmbiguityRNACompoundSet.getRNACompoundSet());</span>
<span class="source-line-no">492</span><span id="line-492">                                } catch (CompoundNotFoundException ex) {</span>
<span class="source-line-no">493</span><span id="line-493">                                        logger.warn("Could not align DNA and RNA compound sets: " + seq2);</span>
<span class="source-line-no">494</span><span id="line-494">                                        return true;</span>
<span class="source-line-no">495</span><span id="line-495">                                }</span>
<span class="source-line-no">496</span><span id="line-496">                        }</span>
<span class="source-line-no">497</span><span id="line-497">                }</span>
<span class="source-line-no">498</span><span id="line-498"></span>
<span class="source-line-no">499</span><span id="line-499"></span>
<span class="source-line-no">500</span><span id="line-500">                SubstitutionMatrix&lt;NucleotideCompound&gt; matrix = SubstitutionMatrixHelper.getNuc4_4();</span>
<span class="source-line-no">501</span><span id="line-501"></span>
<span class="source-line-no">502</span><span id="line-502">                GapPenalty penalty = new SimpleGapPenalty(8,1);</span>
<span class="source-line-no">503</span><span id="line-503"></span>
<span class="source-line-no">504</span><span id="line-504">                PairwiseSequenceAligner&lt;Sequence&lt;NucleotideCompound&gt;, NucleotideCompound&gt; smithWaterman =</span>
<span class="source-line-no">505</span><span id="line-505">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.LOCAL, penalty, matrix);</span>
<span class="source-line-no">506</span><span id="line-506"></span>
<span class="source-line-no">507</span><span id="line-507"></span>
<span class="source-line-no">508</span><span id="line-508"></span>
<span class="source-line-no">509</span><span id="line-509">                SequencePair&lt;Sequence&lt;NucleotideCompound&gt;, NucleotideCompound&gt; pair = smithWaterman.getPair();</span>
<span class="source-line-no">510</span><span id="line-510"></span>
<span class="source-line-no">511</span><span id="line-511"></span>
<span class="source-line-no">512</span><span id="line-512"></span>
<span class="source-line-no">513</span><span id="line-513">                if ( pair == null) {</span>
<span class="source-line-no">514</span><span id="line-514">                        logger.warn("Could not align nucleotide sequences. ATOM and SEQRES groups will not be aligned.");</span>
<span class="source-line-no">515</span><span id="line-515">                        logger.warn("Sequences: ");</span>
<span class="source-line-no">516</span><span id="line-516">                        logger.warn(seq1);</span>
<span class="source-line-no">517</span><span id="line-517">                        logger.warn(seq2);</span>
<span class="source-line-no">518</span><span id="line-518">                        return true;</span>
<span class="source-line-no">519</span><span id="line-519"></span>
<span class="source-line-no">520</span><span id="line-520">                }</span>
<span class="source-line-no">521</span><span id="line-521"></span>
<span class="source-line-no">522</span><span id="line-522"></span>
<span class="source-line-no">523</span><span id="line-523"></span>
<span class="source-line-no">524</span><span id="line-524">                logger.debug("Alignment:\n{}", pair.toString(100));</span>
<span class="source-line-no">525</span><span id="line-525"></span>
<span class="source-line-no">526</span><span id="line-526"></span>
<span class="source-line-no">527</span><span id="line-527">                boolean noMatchFound = mapDNAChains(seqRes,atomRes,pair,seqresIndexPosition, atomIndexPosition );</span>
<span class="source-line-no">528</span><span id="line-528"></span>
<span class="source-line-no">529</span><span id="line-529">                return noMatchFound;</span>
<span class="source-line-no">530</span><span id="line-530"></span>
<span class="source-line-no">531</span><span id="line-531">        }</span>
<span class="source-line-no">532</span><span id="line-532"></span>
<span class="source-line-no">533</span><span id="line-533">        private Sequence&lt;NucleotideCompound&gt; getNucleotideSequence(String seq) {</span>
<span class="source-line-no">534</span><span id="line-534">                Sequence&lt;NucleotideCompound&gt; s = null;</span>
<span class="source-line-no">535</span><span id="line-535"></span>
<span class="source-line-no">536</span><span id="line-536">                // first we try DNA, then RNA, them hybrid</span>
<span class="source-line-no">537</span><span id="line-537"></span>
<span class="source-line-no">538</span><span id="line-538">                try {</span>
<span class="source-line-no">539</span><span id="line-539"></span>
<span class="source-line-no">540</span><span id="line-540">                        s = new DNASequence(seq, AmbiguityDNACompoundSet.getDNACompoundSet());</span>
<span class="source-line-no">541</span><span id="line-541">                } catch (CompoundNotFoundException e){</span>
<span class="source-line-no">542</span><span id="line-542"></span>
<span class="source-line-no">543</span><span id="line-543">                        try {</span>
<span class="source-line-no">544</span><span id="line-544">                                s= new RNASequence(seq, AmbiguityRNACompoundSet.getRNACompoundSet());</span>
<span class="source-line-no">545</span><span id="line-545">                        } catch (CompoundNotFoundException ex) {</span>
<span class="source-line-no">546</span><span id="line-546"></span>
<span class="source-line-no">547</span><span id="line-547">                                try {</span>
<span class="source-line-no">548</span><span id="line-548">                                        // it could still be a hybrid, e.g. 3hoz, chain T, what to do in that case?</span>
<span class="source-line-no">549</span><span id="line-549">                                        s = new DNASequence(seq, AmbiguityDNARNAHybridCompoundSet.getDNARNAHybridCompoundSet());</span>
<span class="source-line-no">550</span><span id="line-550">                                        logger.warn("Hybrid RNA/DNA sequence detected for sequence {}", seq);</span>
<span class="source-line-no">551</span><span id="line-551">                                } catch (CompoundNotFoundException exc) {</span>
<span class="source-line-no">552</span><span id="line-552">                                        // not DNA, nor RNA, nor hybrid</span>
<span class="source-line-no">553</span><span id="line-553">                                        logger.warn("Could not determine compound set (neither DNA, RNA nor hybrid) for nucleotide sequence " + seq);</span>
<span class="source-line-no">554</span><span id="line-554">                                        return null;</span>
<span class="source-line-no">555</span><span id="line-555">                                }</span>
<span class="source-line-no">556</span><span id="line-556"></span>
<span class="source-line-no">557</span><span id="line-557">                        }</span>
<span class="source-line-no">558</span><span id="line-558">                }</span>
<span class="source-line-no">559</span><span id="line-559">                return s;</span>
<span class="source-line-no">560</span><span id="line-560">        }</span>
<span class="source-line-no">561</span><span id="line-561"></span>
<span class="source-line-no">562</span><span id="line-562"></span>
<span class="source-line-no">563</span><span id="line-563">        /**</span>
<span class="source-line-no">564</span><span id="line-564">         * Aligns two chains of groups, where the first parent is representing the</span>
<span class="source-line-no">565</span><span id="line-565">         * list of amino acids as obtained from the SEQRES records, and the second parent</span>
<span class="source-line-no">566</span><span id="line-566">         * represents the groups obtained from the ATOM records (and containing the actual ATOM information).</span>
<span class="source-line-no">567</span><span id="line-567">         * This does the actual alignment and if a group can be mapped to a position in the SEQRES then the corresponding</span>
<span class="source-line-no">568</span><span id="line-568">         * position is replaced with the group that contains the atoms.</span>
<span class="source-line-no">569</span><span id="line-569">         *</span>
<span class="source-line-no">570</span><span id="line-570">         * @param seqRes</span>
<span class="source-line-no">571</span><span id="line-571">         * @param atomRes</span>
<span class="source-line-no">572</span><span id="line-572">         * @return true if no match has been found</span>
<span class="source-line-no">573</span><span id="line-573">         */</span>
<span class="source-line-no">574</span><span id="line-574">        private boolean alignProteinChains(List&lt;Group&gt; seqRes, List&lt;Group&gt; atomRes) {</span>
<span class="source-line-no">575</span><span id="line-575"></span>
<span class="source-line-no">576</span><span id="line-576">                Map&lt;Integer,Integer&gt; seqresIndexPosition = new HashMap&lt;&gt;();</span>
<span class="source-line-no">577</span><span id="line-577">                Map&lt;Integer,Integer&gt; atomIndexPosition   = new HashMap&lt;&gt;();</span>
<span class="source-line-no">578</span><span id="line-578"></span>
<span class="source-line-no">579</span><span id="line-579">                String seq1 = getFullAtomSequence(seqRes, seqresIndexPosition, false);</span>
<span class="source-line-no">580</span><span id="line-580">                //</span>
<span class="source-line-no">581</span><span id="line-581">                String seq2 = getFullAtomSequence(atomRes, atomIndexPosition, false);</span>
<span class="source-line-no">582</span><span id="line-582"></span>
<span class="source-line-no">583</span><span id="line-583"></span>
<span class="source-line-no">584</span><span id="line-584">                logger.debug("Protein seq1 to align (length "+ seq1.length()+"): " + seq1);</span>
<span class="source-line-no">585</span><span id="line-585">                logger.debug("Protein seq2 to align (length "+ seq2.length()+"): " + seq2);</span>
<span class="source-line-no">586</span><span id="line-586"></span>
<span class="source-line-no">587</span><span id="line-587">                ProteinSequence s1;</span>
<span class="source-line-no">588</span><span id="line-588">                ProteinSequence s2;</span>
<span class="source-line-no">589</span><span id="line-589">                try {</span>
<span class="source-line-no">590</span><span id="line-590">                        s1 = new ProteinSequence(seq1);</span>
<span class="source-line-no">591</span><span id="line-591">                        s2 = new ProteinSequence(seq2);</span>
<span class="source-line-no">592</span><span id="line-592">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">593</span><span id="line-593">                        logger.warn("Could not create protein sequences ({}) to align ATOM and SEQRES groups, they will remain unaligned.", e.getMessage());</span>
<span class="source-line-no">594</span><span id="line-594">                        return true;</span>
<span class="source-line-no">595</span><span id="line-595">                }</span>
<span class="source-line-no">596</span><span id="line-596"></span>
<span class="source-line-no">597</span><span id="line-597"></span>
<span class="source-line-no">598</span><span id="line-598">                SubstitutionMatrix&lt;AminoAcidCompound&gt; matrix = SubstitutionMatrixHelper.getBlosum65();</span>
<span class="source-line-no">599</span><span id="line-599"></span>
<span class="source-line-no">600</span><span id="line-600">                GapPenalty penalty = new SimpleGapPenalty(8, 1);</span>
<span class="source-line-no">601</span><span id="line-601"></span>
<span class="source-line-no">602</span><span id="line-602"></span>
<span class="source-line-no">603</span><span id="line-603">                PairwiseSequenceAligner&lt;ProteinSequence, AminoAcidCompound&gt; smithWaterman =</span>
<span class="source-line-no">604</span><span id="line-604">                                Alignments.getPairwiseAligner(s1, s2, PairwiseSequenceAlignerType.LOCAL, penalty, matrix);</span>
<span class="source-line-no">605</span><span id="line-605"></span>
<span class="source-line-no">606</span><span id="line-606">                SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; pair = smithWaterman.getPair();</span>
<span class="source-line-no">607</span><span id="line-607"></span>
<span class="source-line-no">608</span><span id="line-608"></span>
<span class="source-line-no">609</span><span id="line-609">                // sequences that are only X (e.g. 1jnv chain A) produced empty alignments, because nothing aligns to nothing and thus the local alignment is empty</span>
<span class="source-line-no">610</span><span id="line-610">                // to avoid those empty alignments we catch them here with pair.getLength()==0</span>
<span class="source-line-no">611</span><span id="line-611">                if ( pair == null || pair.getLength()==0) {</span>
<span class="source-line-no">612</span><span id="line-612">                        logger.warn("Could not align protein sequences. ATOM and SEQRES groups will not be aligned.");</span>
<span class="source-line-no">613</span><span id="line-613">                        logger.warn("Sequences: ");</span>
<span class="source-line-no">614</span><span id="line-614">                        logger.warn(seq1);</span>
<span class="source-line-no">615</span><span id="line-615">                        logger.warn(seq2);</span>
<span class="source-line-no">616</span><span id="line-616">                        return true;</span>
<span class="source-line-no">617</span><span id="line-617">                }</span>
<span class="source-line-no">618</span><span id="line-618"></span>
<span class="source-line-no">619</span><span id="line-619"></span>
<span class="source-line-no">620</span><span id="line-620">                logger.debug("Alignment:\n{}", pair.toString(100));</span>
<span class="source-line-no">621</span><span id="line-621"></span>
<span class="source-line-no">622</span><span id="line-622"></span>
<span class="source-line-no">623</span><span id="line-623">                boolean noMatchFound = mapChains(seqRes,atomRes,pair,seqresIndexPosition, atomIndexPosition );</span>
<span class="source-line-no">624</span><span id="line-624"></span>
<span class="source-line-no">625</span><span id="line-625">                return noMatchFound;</span>
<span class="source-line-no">626</span><span id="line-626"></span>
<span class="source-line-no">627</span><span id="line-627"></span>
<span class="source-line-no">628</span><span id="line-628">        }</span>
<span class="source-line-no">629</span><span id="line-629"></span>
<span class="source-line-no">630</span><span id="line-630"></span>
<span class="source-line-no">631</span><span id="line-631">        private boolean mapChains(List&lt;Group&gt; seqResGroups, List&lt;Group&gt; atomRes,</span>
<span class="source-line-no">632</span><span id="line-632">                        SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; pair,</span>
<span class="source-line-no">633</span><span id="line-633">                        Map&lt;Integer,Integer&gt; seqresIndexPosition,</span>
<span class="source-line-no">634</span><span id="line-634">                        Map&lt;Integer,Integer&gt; atomIndexPosition )   {</span>
<span class="source-line-no">635</span><span id="line-635"></span>
<span class="source-line-no">636</span><span id="line-636"></span>
<span class="source-line-no">637</span><span id="line-637"></span>
<span class="source-line-no">638</span><span id="line-638">                // at the present stage the future seqRes are still stored as Atom groups in the seqRes parent...</span>
<span class="source-line-no">639</span><span id="line-639"></span>
<span class="source-line-no">640</span><span id="line-640"></span>
<span class="source-line-no">641</span><span id="line-641">                int aligLength = pair.getLength();</span>
<span class="source-line-no">642</span><span id="line-642"></span>
<span class="source-line-no">643</span><span id="line-643">                // make sure we actually find an alignment</span>
<span class="source-line-no">644</span><span id="line-644">                boolean noMatchFound = true;</span>
<span class="source-line-no">645</span><span id="line-645"></span>
<span class="source-line-no">646</span><span id="line-646">                Compound gapSymbol =  AminoAcidCompoundSet.getAminoAcidCompoundSet().getCompoundForString("-");</span>
<span class="source-line-no">647</span><span id="line-647"></span>
<span class="source-line-no">648</span><span id="line-648">                mainLoop:</span>
<span class="source-line-no">649</span><span id="line-649">                        for (int i = 1; i &lt;= aligLength ; i++) {</span>
<span class="source-line-no">650</span><span id="line-650"></span>
<span class="source-line-no">651</span><span id="line-651">                                Compound s =  pair.getCompoundAt(1, i);</span>
<span class="source-line-no">652</span><span id="line-652">                                Compound a =  pair.getCompoundAt(2, i);</span>
<span class="source-line-no">653</span><span id="line-653"></span>
<span class="source-line-no">654</span><span id="line-654">                                // alignment is using internal index start at 1...</span>
<span class="source-line-no">655</span><span id="line-655">                                int posSeq  = pair.getIndexInQueryAt(i)  - 1;</span>
<span class="source-line-no">656</span><span id="line-656">                                int posAtom = pair.getIndexInTargetAt(i) - 1;</span>
<span class="source-line-no">657</span><span id="line-657"></span>
<span class="source-line-no">658</span><span id="line-658">                                if (  s.equals(gapSymbol) || a.equals(gapSymbol)){</span>
<span class="source-line-no">659</span><span id="line-659">                                        continue;</span>
<span class="source-line-no">660</span><span id="line-660">                                }</span>
<span class="source-line-no">661</span><span id="line-661"></span>
<span class="source-line-no">662</span><span id="line-662">                                if ( s.equals(a)){</span>
<span class="source-line-no">663</span><span id="line-663"></span>
<span class="source-line-no">664</span><span id="line-664">                                        // the atom record can be aligned to the SeqRes record!</span>
<span class="source-line-no">665</span><span id="line-665">                                        // replace the SeqRes group with the Atom group!</span>
<span class="source-line-no">666</span><span id="line-666"></span>
<span class="source-line-no">667</span><span id="line-667">                                        Group s1 = seqResGroups.get(seqresIndexPosition.get(posSeq));</span>
<span class="source-line-no">668</span><span id="line-668">                                        Group a1 = atomRes.get(atomIndexPosition.get(posAtom));</span>
<span class="source-line-no">669</span><span id="line-669"></span>
<span class="source-line-no">670</span><span id="line-670">                                        if ( s1 == null || a1 == null){</span>
<span class="source-line-no">671</span><span id="line-671">                                                /// can't map this position...</span>
<span class="source-line-no">672</span><span id="line-672">                                                logger.warn("can't map " + i + ":" + s + " " + posSeq +" " + s1 + " atom: " + posAtom + " " + a1 );</span>
<span class="source-line-no">673</span><span id="line-673">                                                continue mainLoop;</span>
<span class="source-line-no">674</span><span id="line-674">                                        }</span>
<span class="source-line-no">675</span><span id="line-675"></span>
<span class="source-line-no">676</span><span id="line-676">                                        // need to trim the names to allow matching e.g in</span>
<span class="source-line-no">677</span><span id="line-677">                                        // pdb1b2m</span>
<span class="source-line-no">678</span><span id="line-678">                                        String pdbNameS = s1.getPDBName();</span>
<span class="source-line-no">679</span><span id="line-679">                                        String pdbNameA = a1.getPDBName();</span>
<span class="source-line-no">680</span><span id="line-680"></span>
<span class="source-line-no">681</span><span id="line-681">                                        if ( pdbNameS == null || pdbNameA == null ){</span>
<span class="source-line-no">682</span><span id="line-682">                                                logger.warn("null value for group.getPDBName found at {} when trying to align {} and {} {}",posSeq, s1, a1, posAtom);</span>
<span class="source-line-no">683</span><span id="line-683">                                                logger.warn("ATOM and SEQRES sequences will not be aligned.");</span>
<span class="source-line-no">684</span><span id="line-684">                                                return true;</span>
<span class="source-line-no">685</span><span id="line-685">                                        }</span>
<span class="source-line-no">686</span><span id="line-686"></span>
<span class="source-line-no">687</span><span id="line-687">                                        if ( ! pdbNameA.trim().equals(pdbNameS.trim())) {</span>
<span class="source-line-no">688</span><span id="line-688"></span>
<span class="source-line-no">689</span><span id="line-689">                                                String msg = "'"+ s1 + "' (position " + posSeq + ") does not align with '" + a1+ "' (position " + posAtom + "), should be: " + s + " : " + a;</span>
<span class="source-line-no">690</span><span id="line-690"></span>
<span class="source-line-no">691</span><span id="line-691">                                                if ( s1.getType().equals(HetatomImpl.type) &amp;&amp; a1.getType().equals(HetatomImpl.type)){</span>
<span class="source-line-no">692</span><span id="line-692">                                                        logger.info(msg + ". They seem to be hetatoms, so ignoring mismatch.");</span>
<span class="source-line-no">693</span><span id="line-693">                                                }</span>
<span class="source-line-no">694</span><span id="line-694">                                                else {</span>
<span class="source-line-no">695</span><span id="line-695">                                                        logger.warn(msg + ". This could be a problem because they aren't both hetatoms");</span>
<span class="source-line-no">696</span><span id="line-696">                                                }</span>
<span class="source-line-no">697</span><span id="line-697"></span>
<span class="source-line-no">698</span><span id="line-698">                                        }</span>
<span class="source-line-no">699</span><span id="line-699"></span>
<span class="source-line-no">700</span><span id="line-700">                                        // do the actual replacing of the SEQRES group with the ATOM group</span>
<span class="source-line-no">701</span><span id="line-701">                                        //                                      if ( s1.getChain().getChainID().equals("A")) {</span>
<span class="source-line-no">702</span><span id="line-702">                                        //                                              System.out.println(" setting " + posSeq +" " + a1);</span>
<span class="source-line-no">703</span><span id="line-703">                                        //                                      }</span>
<span class="source-line-no">704</span><span id="line-704">                                        seqResGroups.set(seqresIndexPosition.get(posSeq),a1);</span>
<span class="source-line-no">705</span><span id="line-705">                                        noMatchFound = false;</span>
<span class="source-line-no">706</span><span id="line-706">                                }</span>
<span class="source-line-no">707</span><span id="line-707">                        }</span>
<span class="source-line-no">708</span><span id="line-708"></span>
<span class="source-line-no">709</span><span id="line-709"></span>
<span class="source-line-no">710</span><span id="line-710">                // now we merge the two chains into one</span>
<span class="source-line-no">711</span><span id="line-711">                // the Groups that can be aligned are now pointing to the</span>
<span class="source-line-no">712</span><span id="line-712">                // groups in the Atom records.</span>
<span class="source-line-no">713</span><span id="line-713">                if (  noMatchFound) {</span>
<span class="source-line-no">714</span><span id="line-714"></span>
<span class="source-line-no">715</span><span id="line-715">                        logger.debug("no alignment found!");</span>
<span class="source-line-no">716</span><span id="line-716">                }</span>
<span class="source-line-no">717</span><span id="line-717">                return noMatchFound;</span>
<span class="source-line-no">718</span><span id="line-718"></span>
<span class="source-line-no">719</span><span id="line-719">        }</span>
<span class="source-line-no">720</span><span id="line-720"></span>
<span class="source-line-no">721</span><span id="line-721">        private boolean mapDNAChains(List&lt;Group&gt; seqResGroups, List&lt;Group&gt; atomRes,</span>
<span class="source-line-no">722</span><span id="line-722">                        SequencePair&lt;Sequence&lt;NucleotideCompound&gt;, NucleotideCompound&gt; pair,</span>
<span class="source-line-no">723</span><span id="line-723">                        Map&lt;Integer,Integer&gt; seqresIndexPosition,</span>
<span class="source-line-no">724</span><span id="line-724">                        Map&lt;Integer,Integer&gt; atomIndexPosition)   {</span>
<span class="source-line-no">725</span><span id="line-725"></span>
<span class="source-line-no">726</span><span id="line-726"></span>
<span class="source-line-no">727</span><span id="line-727">                // at the present stage the future seqREs are still stored as Atom groups in the seqRes parent...</span>
<span class="source-line-no">728</span><span id="line-728"></span>
<span class="source-line-no">729</span><span id="line-729"></span>
<span class="source-line-no">730</span><span id="line-730">                int aligLength = pair.getLength();</span>
<span class="source-line-no">731</span><span id="line-731"></span>
<span class="source-line-no">732</span><span id="line-732">                //              System.out.println("sequence length seqres:");</span>
<span class="source-line-no">733</span><span id="line-733">                //              System.out.println(seqresIndexPosition.keySet().size());</span>
<span class="source-line-no">734</span><span id="line-734">                //              System.out.println("alignment length: " + aligLength);</span>
<span class="source-line-no">735</span><span id="line-735">                // System.out.println(gapSymbol.getName());</span>
<span class="source-line-no">736</span><span id="line-736"></span>
<span class="source-line-no">737</span><span id="line-737">                // make sure we actually find an alignment</span>
<span class="source-line-no">738</span><span id="line-738">                boolean noMatchFound = true;</span>
<span class="source-line-no">739</span><span id="line-739"></span>
<span class="source-line-no">740</span><span id="line-740">                Compound gapSymbol =  DNACompoundSet.getDNACompoundSet().getCompoundForString("-");</span>
<span class="source-line-no">741</span><span id="line-741"></span>
<span class="source-line-no">742</span><span id="line-742">                mainLoop:</span>
<span class="source-line-no">743</span><span id="line-743">                        for (int i = 1; i &lt;= aligLength ; i++) {</span>
<span class="source-line-no">744</span><span id="line-744"></span>
<span class="source-line-no">745</span><span id="line-745">                                Compound s =  pair.getCompoundAt(1, i);</span>
<span class="source-line-no">746</span><span id="line-746">                                Compound a =  pair.getCompoundAt(2,i);</span>
<span class="source-line-no">747</span><span id="line-747"></span>
<span class="source-line-no">748</span><span id="line-748">                                // alignment is using internal index start at 1...</span>
<span class="source-line-no">749</span><span id="line-749">                                int posSeq = pair.getIndexInQueryAt(i)  -1;</span>
<span class="source-line-no">750</span><span id="line-750">                                int posAtom = pair.getIndexInTargetAt(i) -1;</span>
<span class="source-line-no">751</span><span id="line-751"></span>
<span class="source-line-no">752</span><span id="line-752">                                if (  s.equals(gapSymbol) || a.equals(gapSymbol)){</span>
<span class="source-line-no">753</span><span id="line-753">                                        continue;</span>
<span class="source-line-no">754</span><span id="line-754">                                }</span>
<span class="source-line-no">755</span><span id="line-755"></span>
<span class="source-line-no">756</span><span id="line-756">                                if ( s.equals(a)){</span>
<span class="source-line-no">757</span><span id="line-757"></span>
<span class="source-line-no">758</span><span id="line-758">                                        // the atom record can be aligned to the SeqRes record!</span>
<span class="source-line-no">759</span><span id="line-759">                                        // replace the SeqRes group with the Atom group!</span>
<span class="source-line-no">760</span><span id="line-760"></span>
<span class="source-line-no">761</span><span id="line-761">                                        Group s1 = seqResGroups.get(seqresIndexPosition.get(posSeq));</span>
<span class="source-line-no">762</span><span id="line-762">                                        Group a1 = atomRes.get(atomIndexPosition.get(posAtom));</span>
<span class="source-line-no">763</span><span id="line-763"></span>
<span class="source-line-no">764</span><span id="line-764">                                        if ( s1 == null || a1 == null){</span>
<span class="source-line-no">765</span><span id="line-765">                                                /// can't map this position...</span>
<span class="source-line-no">766</span><span id="line-766">                                                logger.warn("can't map " + i + ":" + s + " " + posSeq +" " + s1 + " atom: " + posAtom + " " + a1 );</span>
<span class="source-line-no">767</span><span id="line-767">                                                continue mainLoop;</span>
<span class="source-line-no">768</span><span id="line-768">                                        }</span>
<span class="source-line-no">769</span><span id="line-769"></span>
<span class="source-line-no">770</span><span id="line-770">                                        // need to trim the names to allow matching e.g in</span>
<span class="source-line-no">771</span><span id="line-771">                                        // pdb1b2m</span>
<span class="source-line-no">772</span><span id="line-772">                                        String pdbNameS = s1.getPDBName();</span>
<span class="source-line-no">773</span><span id="line-773">                                        String pdbNameA = a1.getPDBName();</span>
<span class="source-line-no">774</span><span id="line-774">                                        if ( pdbNameS == null || pdbNameA == null ){</span>
<span class="source-line-no">775</span><span id="line-775">                                                logger.warn("null value found for group.getPDBName() at " + posSeq + " when trying to align " + s1 + " and " + a1 + " " + posAtom);</span>
<span class="source-line-no">776</span><span id="line-776">                                                logger.warn("ATOM and SEQRES sequences will not be aligned.");</span>
<span class="source-line-no">777</span><span id="line-777">                                        }</span>
<span class="source-line-no">778</span><span id="line-778">                                        if (! pdbNameA.equals(pdbNameS)){</span>
<span class="source-line-no">779</span><span id="line-779">                                                if ( ! pdbNameA.trim().equals(pdbNameS.trim())) {</span>
<span class="source-line-no">780</span><span id="line-780">                                                        logger.info(s1 + " " + posSeq + " does not align with " + a1+ " " + posAtom + " should be: " + s + " : " + a);</span>
<span class="source-line-no">781</span><span id="line-781">                                                        if ( s1.getType().equals(HetatomImpl.type) &amp;&amp; a1.getType().equals(HetatomImpl.type)){</span>
<span class="source-line-no">782</span><span id="line-782">                                                                logger.info("they seem to be hetatoms, so ignoring mismatch.");</span>
<span class="source-line-no">783</span><span id="line-783">                                                        }</span>
<span class="source-line-no">784</span><span id="line-784">                                                        else {</span>
<span class="source-line-no">785</span><span id="line-785">                                                                //  System.exit(0);// for debug only</span>
<span class="source-line-no">786</span><span id="line-786">                                                                //System.out.println(lst1.seqString());</span>
<span class="source-line-no">787</span><span id="line-787">                                                                //System.out.println(lst2.seqString());</span>
<span class="source-line-no">788</span><span id="line-788">                                                                logger.warn("Could not match residues " + s1 + " " + a1);</span>
<span class="source-line-no">789</span><span id="line-789">                                                        }</span>
<span class="source-line-no">790</span><span id="line-790"></span>
<span class="source-line-no">791</span><span id="line-791">                                                }</span>
<span class="source-line-no">792</span><span id="line-792">                                        }</span>
<span class="source-line-no">793</span><span id="line-793"></span>
<span class="source-line-no">794</span><span id="line-794">                                        // do the actual replacing of the SEQRES group with the ATOM group</span>
<span class="source-line-no">795</span><span id="line-795">                                        //                                      if ( s1.getChain().getChainID().equals("A")) {</span>
<span class="source-line-no">796</span><span id="line-796">                                        //                                              System.out.println(" setting " + posSeq +" " + a1);</span>
<span class="source-line-no">797</span><span id="line-797">                                        //                                      }</span>
<span class="source-line-no">798</span><span id="line-798">                                        seqResGroups.set(seqresIndexPosition.get(posSeq),a1);</span>
<span class="source-line-no">799</span><span id="line-799">                                        noMatchFound = false;</span>
<span class="source-line-no">800</span><span id="line-800">                                }</span>
<span class="source-line-no">801</span><span id="line-801">                        }</span>
<span class="source-line-no">802</span><span id="line-802"></span>
<span class="source-line-no">803</span><span id="line-803"></span>
<span class="source-line-no">804</span><span id="line-804">                // now we merge the two chains into one</span>
<span class="source-line-no">805</span><span id="line-805">                // the Groups that can be aligned are now pointing to the</span>
<span class="source-line-no">806</span><span id="line-806">                // groups in the Atom records.</span>
<span class="source-line-no">807</span><span id="line-807">                if (  noMatchFound) {</span>
<span class="source-line-no">808</span><span id="line-808"></span>
<span class="source-line-no">809</span><span id="line-809">                        logger.debug("no alignment found!");</span>
<span class="source-line-no">810</span><span id="line-810">                }</span>
<span class="source-line-no">811</span><span id="line-811">                return noMatchFound;</span>
<span class="source-line-no">812</span><span id="line-812"></span>
<span class="source-line-no">813</span><span id="line-813">        }</span>
<span class="source-line-no">814</span><span id="line-814"></span>
<span class="source-line-no">815</span><span id="line-815">        /**</span>
<span class="source-line-no">816</span><span id="line-816">         * Storing unaligned SEQRES groups in a Structure.</span>
<span class="source-line-no">817</span><span id="line-817">         * @param structure</span>
<span class="source-line-no">818</span><span id="line-818">         * @param seqResChains</span>
<span class="source-line-no">819</span><span id="line-819">         */</span>
<span class="source-line-no">820</span><span id="line-820">        public static void storeUnAlignedSeqRes(Structure structure, List&lt;Chain&gt; seqResChains, boolean headerOnly) {</span>
<span class="source-line-no">821</span><span id="line-821"></span>
<span class="source-line-no">822</span><span id="line-822">                if (headerOnly) {</span>
<span class="source-line-no">823</span><span id="line-823">                        List&lt;Chain&gt; atomChains = new ArrayList&lt;&gt;();</span>
<span class="source-line-no">824</span><span id="line-824">                        for (Chain seqRes: seqResChains) {</span>
<span class="source-line-no">825</span><span id="line-825">                                // In header-only mode skip ATOM records.</span>
<span class="source-line-no">826</span><span id="line-826">                                // Here we store chains with SEQRES instead of AtomGroups.</span>
<span class="source-line-no">827</span><span id="line-827">                                seqRes.setSeqResGroups(seqRes.getAtomGroups());</span>
<span class="source-line-no">828</span><span id="line-828">                                seqRes.setAtomGroups(new ArrayList&lt;&gt;()); // clear out the atom groups.</span>
<span class="source-line-no">829</span><span id="line-829">                                atomChains.add(seqRes);</span>
<span class="source-line-no">830</span><span id="line-830">                        }</span>
<span class="source-line-no">831</span><span id="line-831">                        structure.setChains(0, atomChains);</span>
<span class="source-line-no">832</span><span id="line-832"></span>
<span class="source-line-no">833</span><span id="line-833">                } else {</span>
<span class="source-line-no">834</span><span id="line-834"></span>
<span class="source-line-no">835</span><span id="line-835">                        for (int i = 0; i &lt; structure.nrModels(); i++) {</span>
<span class="source-line-no">836</span><span id="line-836">                                List&lt;Chain&gt; atomChains   = structure.getModel(i);</span>
<span class="source-line-no">837</span><span id="line-837">                                if (seqResChains.isEmpty()) {</span>
<span class="source-line-no">838</span><span id="line-838">                                        // in files without SEQRES, seqResChains object is empty: we replace by atomChains resulting below in a trivial alignment and a copy of atom groups to seqres groups</span>
<span class="source-line-no">839</span><span id="line-839">                                        seqResChains = atomChains;</span>
<span class="source-line-no">840</span><span id="line-840">                                }</span>
<span class="source-line-no">841</span><span id="line-841">                                for (Chain seqRes: seqResChains){</span>
<span class="source-line-no">842</span><span id="line-842">                                        // Otherwise, we find a chain with AtomGroups</span>
<span class="source-line-no">843</span><span id="line-843">                                        // and set this as SEQRES groups.</span>
<span class="source-line-no">844</span><span id="line-844">                                        // TODO no idea if new parameter useChainId should be false or true here, used true as a guess - JD 2016-05-09</span>
<span class="source-line-no">845</span><span id="line-845">                                        Chain atomRes = SeqRes2AtomAligner.getMatchingAtomRes(seqRes,atomChains,true);</span>
<span class="source-line-no">846</span><span id="line-846">                                        if ( atomRes != null)</span>
<span class="source-line-no">847</span><span id="line-847">                                                atomRes.setSeqResGroups(seqRes.getAtomGroups());</span>
<span class="source-line-no">848</span><span id="line-848">                                        else</span>
<span class="source-line-no">849</span><span id="line-849">                                                logger.warn("Could not find atom records for chain " + seqRes.getId());</span>
<span class="source-line-no">850</span><span id="line-850">                                }</span>
<span class="source-line-no">851</span><span id="line-851">                        }</span>
<span class="source-line-no">852</span><span id="line-852">                }</span>
<span class="source-line-no">853</span><span id="line-853">        }</span>
<span class="source-line-no">854</span><span id="line-854">}</span>




























































</pre>
</div>
</main>
</body>
</html>
