<!DOCTYPE HTML>
<html lang>
<head>
<!-- Generated by javadoc (21) -->
<title>Source code</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="source: package: org.biojava.nbio.structure.io, class: StructureSequenceMatcher">
<meta name="generator" content="javadoc/SourceToHTMLConverter">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body class="source-page">
<main role="main">
<div class="source-container">
<pre><span class="source-line-no">001</span><span id="line-1">/**</span>
<span class="source-line-no">002</span><span id="line-2"> *                    BioJava development code</span>
<span class="source-line-no">003</span><span id="line-3"> *</span>
<span class="source-line-no">004</span><span id="line-4"> * This code may be freely distributed and modified under the</span>
<span class="source-line-no">005</span><span id="line-5"> * terms of the GNU Lesser General Public Licence.  This should</span>
<span class="source-line-no">006</span><span id="line-6"> * be distributed with the code.  If you do not have a copy,</span>
<span class="source-line-no">007</span><span id="line-7"> * see:</span>
<span class="source-line-no">008</span><span id="line-8"> *</span>
<span class="source-line-no">009</span><span id="line-9"> *      http://www.gnu.org/copyleft/lesser.html</span>
<span class="source-line-no">010</span><span id="line-10"> *</span>
<span class="source-line-no">011</span><span id="line-11"> * Copyright for this code is held jointly by the individual</span>
<span class="source-line-no">012</span><span id="line-12"> * authors.  These should be listed in @author doc comments.</span>
<span class="source-line-no">013</span><span id="line-13"> *</span>
<span class="source-line-no">014</span><span id="line-14"> * For more information on the BioJava project and its aims,</span>
<span class="source-line-no">015</span><span id="line-15"> * or to join the biojava-l mailing list, visit the home page</span>
<span class="source-line-no">016</span><span id="line-16"> * at:</span>
<span class="source-line-no">017</span><span id="line-17"> *</span>
<span class="source-line-no">018</span><span id="line-18"> *      http://www.biojava.org/</span>
<span class="source-line-no">019</span><span id="line-19"> *</span>
<span class="source-line-no">020</span><span id="line-20"> * Created by Spencer Bliven</span>
<span class="source-line-no">021</span><span id="line-21"> *</span>
<span class="source-line-no">022</span><span id="line-22"> */</span>
<span class="source-line-no">023</span><span id="line-23">package org.biojava.nbio.structure.io;</span>
<span class="source-line-no">024</span><span id="line-24"></span>
<span class="source-line-no">025</span><span id="line-25">import org.biojava.nbio.alignment.Alignments;</span>
<span class="source-line-no">026</span><span id="line-26">import org.biojava.nbio.alignment.Alignments.PairwiseSequenceAlignerType;</span>
<span class="source-line-no">027</span><span id="line-27">import org.biojava.nbio.alignment.SimpleGapPenalty;</span>
<span class="source-line-no">028</span><span id="line-28">import org.biojava.nbio.core.alignment.matrices.SimpleSubstitutionMatrix;</span>
<span class="source-line-no">029</span><span id="line-29">import org.biojava.nbio.core.alignment.template.AlignedSequence;</span>
<span class="source-line-no">030</span><span id="line-30">import org.biojava.nbio.core.alignment.template.SequencePair;</span>
<span class="source-line-no">031</span><span id="line-31">import org.biojava.nbio.core.alignment.template.SubstitutionMatrix;</span>
<span class="source-line-no">032</span><span id="line-32">import org.biojava.nbio.structure.*;</span>
<span class="source-line-no">033</span><span id="line-33">import org.biojava.nbio.core.exceptions.CompoundNotFoundException;</span>
<span class="source-line-no">034</span><span id="line-34">import org.biojava.nbio.core.sequence.ProteinSequence;</span>
<span class="source-line-no">035</span><span id="line-35">import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;</span>
<span class="source-line-no">036</span><span id="line-36">import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;</span>
<span class="source-line-no">037</span><span id="line-37">import org.biojava.nbio.core.sequence.template.CompoundSet;</span>
<span class="source-line-no">038</span><span id="line-38">import org.slf4j.Logger;</span>
<span class="source-line-no">039</span><span id="line-39">import org.slf4j.LoggerFactory;</span>
<span class="source-line-no">040</span><span id="line-40"></span>
<span class="source-line-no">041</span><span id="line-41">import java.io.InputStreamReader;</span>
<span class="source-line-no">042</span><span id="line-42">import java.util.*;</span>
<span class="source-line-no">043</span><span id="line-43"></span>
<span class="source-line-no">044</span><span id="line-44"></span>
<span class="source-line-no">045</span><span id="line-45">/**</span>
<span class="source-line-no">046</span><span id="line-46"> * A utility class with methods for matching ProteinSequences with</span>
<span class="source-line-no">047</span><span id="line-47"> * Structures.</span>
<span class="source-line-no">048</span><span id="line-48"> * @author Spencer Bliven</span>
<span class="source-line-no">049</span><span id="line-49"> *</span>
<span class="source-line-no">050</span><span id="line-50"> */</span>
<span class="source-line-no">051</span><span id="line-51">public class StructureSequenceMatcher {</span>
<span class="source-line-no">052</span><span id="line-52"></span>
<span class="source-line-no">053</span><span id="line-53">        private static final Logger logger = LoggerFactory.getLogger(StructureSequenceMatcher.class);</span>
<span class="source-line-no">054</span><span id="line-54"></span>
<span class="source-line-no">055</span><span id="line-55">        /**</span>
<span class="source-line-no">056</span><span id="line-56">         * Get a substructure of {@code wholeStructure} containing only the {@link Group Groups} that are included in</span>
<span class="source-line-no">057</span><span id="line-57">         * {@code sequence}. The resulting structure will contain only {@code ATOM} residues; the SEQ-RES will be empty.</span>
<span class="source-line-no">058</span><span id="line-58">         * The {@link Chain Chains} of the Structure will be new instances (cloned), but the {@link Group Groups} will not.</span>
<span class="source-line-no">059</span><span id="line-59">         * @param sequence The input protein sequence</span>
<span class="source-line-no">060</span><span id="line-60">         * @param wholeStructure The structure from which to take a substructure</span>
<span class="source-line-no">061</span><span id="line-61">         * @return The resulting structure</span>
<span class="source-line-no">062</span><span id="line-62">         * @see #matchSequenceToStructure(ProteinSequence, Structure)</span>
<span class="source-line-no">063</span><span id="line-63">         */</span>
<span class="source-line-no">064</span><span id="line-64">        public static Structure getSubstructureMatchingProteinSequence(ProteinSequence sequence, Structure wholeStructure) {</span>
<span class="source-line-no">065</span><span id="line-65">                ResidueNumber[] rns = matchSequenceToStructure(sequence, wholeStructure);</span>
<span class="source-line-no">066</span><span id="line-66">                Structure structure = wholeStructure.clone();</span>
<span class="source-line-no">067</span><span id="line-67">                structure.setChains(new ArrayList&lt;&gt;());</span>
<span class="source-line-no">068</span><span id="line-68">//              structure.getHetGroups().clear();</span>
<span class="source-line-no">069</span><span id="line-69">                Chain currentChain = null;</span>
<span class="source-line-no">070</span><span id="line-70">                for (ResidueNumber rn : rns) {</span>
<span class="source-line-no">071</span><span id="line-71">                        if (rn == null) continue;</span>
<span class="source-line-no">072</span><span id="line-72">                        Group group; // note that we don't clone</span>
<span class="source-line-no">073</span><span id="line-73">                        try {</span>
<span class="source-line-no">074</span><span id="line-74">                                group = StructureTools.getGroupByPDBResidueNumber(wholeStructure, rn);</span>
<span class="source-line-no">075</span><span id="line-75">                        } catch (StructureException e) {</span>
<span class="source-line-no">076</span><span id="line-76">                                throw new IllegalArgumentException("Could not find residue " + rn + " in structure", e);</span>
<span class="source-line-no">077</span><span id="line-77">                        }</span>
<span class="source-line-no">078</span><span id="line-78">                        Chain chain = new ChainImpl();</span>
<span class="source-line-no">079</span><span id="line-79">                        chain.setName(group.getChain().getName());</span>
<span class="source-line-no">080</span><span id="line-80">                        chain.setId(group.getChain().getId());</span>
<span class="source-line-no">081</span><span id="line-81">                        if (currentChain == null || !currentChain.getId().equals(chain.getId())) {</span>
<span class="source-line-no">082</span><span id="line-82">                                structure.addChain(chain);</span>
<span class="source-line-no">083</span><span id="line-83">                                chain.setEntityInfo(group.getChain().getEntityInfo());</span>
<span class="source-line-no">084</span><span id="line-84">                                chain.setStructure(structure);</span>
<span class="source-line-no">085</span><span id="line-85">                                chain.setId(group.getChain().getId());</span>
<span class="source-line-no">086</span><span id="line-86">                                chain.setName(group.getChain().getName());</span>
<span class="source-line-no">087</span><span id="line-87">                                currentChain = chain;</span>
<span class="source-line-no">088</span><span id="line-88">                        }</span>
<span class="source-line-no">089</span><span id="line-89">                        currentChain.addGroup(group);</span>
<span class="source-line-no">090</span><span id="line-90">                }</span>
<span class="source-line-no">091</span><span id="line-91">                return structure;</span>
<span class="source-line-no">092</span><span id="line-92">        }</span>
<span class="source-line-no">093</span><span id="line-93"></span>
<span class="source-line-no">094</span><span id="line-94">        /**</span>
<span class="source-line-no">095</span><span id="line-95">         * Generates a ProteinSequence corresponding to the sequence of struct,</span>
<span class="source-line-no">096</span><span id="line-96">         * and maintains a mapping from the sequence back to the original groups.</span>
<span class="source-line-no">097</span><span id="line-97">         *</span>
<span class="source-line-no">098</span><span id="line-98">         * Chains are appended to one another. 'X' is used for heteroatoms.</span>
<span class="source-line-no">099</span><span id="line-99">         *</span>
<span class="source-line-no">100</span><span id="line-100">         * @param struct Input structure</span>
<span class="source-line-no">101</span><span id="line-101">         * @param groupIndexPosition An empty map, which will be populated with</span>
<span class="source-line-no">102</span><span id="line-102">         *  (residue index in returned ProteinSequence) -&gt; (Group within struct)</span>
<span class="source-line-no">103</span><span id="line-103">         * @return A ProteinSequence with the full sequence of struct. Chains are</span>
<span class="source-line-no">104</span><span id="line-104">         *  concatenated in the same order as the input structures</span>
<span class="source-line-no">105</span><span id="line-105">         *</span>
<span class="source-line-no">106</span><span id="line-106">         * @see SeqRes2AtomAligner#getFullAtomSequence(List, Map, boolean)</span>
<span class="source-line-no">107</span><span id="line-107">         *</span>
<span class="source-line-no">108</span><span id="line-108">         */</span>
<span class="source-line-no">109</span><span id="line-109">        public static ProteinSequence getProteinSequenceForStructure(Structure struct, Map&lt;Integer,Group&gt; groupIndexPosition ) {</span>
<span class="source-line-no">110</span><span id="line-110"></span>
<span class="source-line-no">111</span><span id="line-111">                if( groupIndexPosition != null) {</span>
<span class="source-line-no">112</span><span id="line-112">                        groupIndexPosition.clear();</span>
<span class="source-line-no">113</span><span id="line-113">                }</span>
<span class="source-line-no">114</span><span id="line-114"></span>
<span class="source-line-no">115</span><span id="line-115">                StringBuilder seqStr = new StringBuilder();</span>
<span class="source-line-no">116</span><span id="line-116"></span>
<span class="source-line-no">117</span><span id="line-117">                for(Chain chain : struct.getChains()) {</span>
<span class="source-line-no">118</span><span id="line-118">                        List&lt;Group&gt; groups = chain.getAtomGroups();</span>
<span class="source-line-no">119</span><span id="line-119">                        Map&lt;Integer,Integer&gt; chainIndexPosition = new HashMap&lt;&gt;();</span>
<span class="source-line-no">120</span><span id="line-120">                        int prevLen = seqStr.length();</span>
<span class="source-line-no">121</span><span id="line-121"></span>
<span class="source-line-no">122</span><span id="line-122">                        // get the sequence for this chain</span>
<span class="source-line-no">123</span><span id="line-123">                        String chainSeq = SeqRes2AtomAligner.getFullAtomSequence(groups, chainIndexPosition, false);</span>
<span class="source-line-no">124</span><span id="line-124">                        seqStr.append(chainSeq);</span>
<span class="source-line-no">125</span><span id="line-125"></span>
<span class="source-line-no">126</span><span id="line-126"></span>
<span class="source-line-no">127</span><span id="line-127">                        // fix up the position to include previous chains, and map the value back to a Group</span>
<span class="source-line-no">128</span><span id="line-128">                        for(Integer seqIndex : chainIndexPosition.keySet()) {</span>
<span class="source-line-no">129</span><span id="line-129">                                Integer groupIndex = chainIndexPosition.get(seqIndex);</span>
<span class="source-line-no">130</span><span id="line-130">                                groupIndexPosition.put(prevLen + seqIndex, groups.get(groupIndex));</span>
<span class="source-line-no">131</span><span id="line-131">                        }</span>
<span class="source-line-no">132</span><span id="line-132">                }</span>
<span class="source-line-no">133</span><span id="line-133"></span>
<span class="source-line-no">134</span><span id="line-134">                ProteinSequence s = null;</span>
<span class="source-line-no">135</span><span id="line-135">                try {</span>
<span class="source-line-no">136</span><span id="line-136">                        s = new ProteinSequence(seqStr.toString());</span>
<span class="source-line-no">137</span><span id="line-137">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">138</span><span id="line-138">                        // I believe this can't happen, please correct this if I'm wrong - JD 2014-10-24</span>
<span class="source-line-no">139</span><span id="line-139">                        // we can log an error if it does, it would mean there's a bad bug somewhere</span>
<span class="source-line-no">140</span><span id="line-140">                        logger.error("Could not create protein sequence, unknown compounds in string: {}", e.getMessage());</span>
<span class="source-line-no">141</span><span id="line-141">                }</span>
<span class="source-line-no">142</span><span id="line-142"></span>
<span class="source-line-no">143</span><span id="line-143">                return s;</span>
<span class="source-line-no">144</span><span id="line-144">        }</span>
<span class="source-line-no">145</span><span id="line-145"></span>
<span class="source-line-no">146</span><span id="line-146">        /**</span>
<span class="source-line-no">147</span><span id="line-147">         * Given a sequence and the corresponding Structure, get the ResidueNumber</span>
<span class="source-line-no">148</span><span id="line-148">         * for each residue in the sequence.</span>
<span class="source-line-no">149</span><span id="line-149">         *</span>
<span class="source-line-no">150</span><span id="line-150">         * &lt;p&gt;Smith-Waterman alignment is used to match the sequences. Residues</span>
<span class="source-line-no">151</span><span id="line-151">         * in the sequence but not the structure or mismatched between sequence</span>
<span class="source-line-no">152</span><span id="line-152">         * and structure will have a null atom, while</span>
<span class="source-line-no">153</span><span id="line-153">         * residues in the structure but not the sequence are ignored with a warning.</span>
<span class="source-line-no">154</span><span id="line-154">         * @param seq The protein sequence. Should match the sequence of struct very</span>
<span class="source-line-no">155</span><span id="line-155">         *      closely.</span>
<span class="source-line-no">156</span><span id="line-156">         * @param struct The corresponding protein structure</span>
<span class="source-line-no">157</span><span id="line-157">         * @return A list of ResidueNumbers of the same length as seq, containing</span>
<span class="source-line-no">158</span><span id="line-158">         *  either the corresponding residue or null.</span>
<span class="source-line-no">159</span><span id="line-159">         */</span>
<span class="source-line-no">160</span><span id="line-160">        public static ResidueNumber[] matchSequenceToStructure(ProteinSequence seq, Structure struct) {</span>
<span class="source-line-no">161</span><span id="line-161"></span>
<span class="source-line-no">162</span><span id="line-162">                //1. Create ProteinSequence for struct while remembering to which group each residue corresponds</span>
<span class="source-line-no">163</span><span id="line-163">                Map&lt;Integer,Group&gt; atomIndexPosition   = new HashMap&lt;&gt;();</span>
<span class="source-line-no">164</span><span id="line-164"></span>
<span class="source-line-no">165</span><span id="line-165">                ProteinSequence structSeq = getProteinSequenceForStructure(struct,atomIndexPosition);</span>
<span class="source-line-no">166</span><span id="line-166"></span>
<span class="source-line-no">167</span><span id="line-167">                // TODO This should really be semi-global alignment, though global for either sequence OR structure (we don't know which)</span>
<span class="source-line-no">168</span><span id="line-168">                //2. Run Smith-Waterman to get the alignment</span>
<span class="source-line-no">169</span><span id="line-169">                // Identity substitution matrix with +1 for match, -1 for mismatch</span>
<span class="source-line-no">170</span><span id="line-170">                // TODO</span>
<span class="source-line-no">171</span><span id="line-171">                SubstitutionMatrix&lt;AminoAcidCompound&gt; matrix =</span>
<span class="source-line-no">172</span><span id="line-172">                        new SimpleSubstitutionMatrix&lt;&gt;(</span>
<span class="source-line-no">173</span><span id="line-173">                                        AminoAcidCompoundSet.getAminoAcidCompoundSet(),</span>
<span class="source-line-no">174</span><span id="line-174">                                        (short)1, (short)-1 );</span>
<span class="source-line-no">175</span><span id="line-175">                matrix = new SimpleSubstitutionMatrix&lt;&gt;(</span>
<span class="source-line-no">176</span><span id="line-176">                                AminoAcidCompoundSet.getAminoAcidCompoundSet(),</span>
<span class="source-line-no">177</span><span id="line-177">                                new InputStreamReader(</span>
<span class="source-line-no">178</span><span id="line-178">                                                SimpleSubstitutionMatrix.class.getResourceAsStream("/matrices/blosum100.txt")),</span>
<span class="source-line-no">179</span><span id="line-179">                "blosum100");</span>
<span class="source-line-no">180</span><span id="line-180">                SequencePair&lt;ProteinSequence, AminoAcidCompound&gt; pair =</span>
<span class="source-line-no">181</span><span id="line-181">                        Alignments.getPairwiseAlignment(seq, structSeq,</span>
<span class="source-line-no">182</span><span id="line-182">                                        PairwiseSequenceAlignerType.GLOBAL, new SimpleGapPenalty(), matrix);</span>
<span class="source-line-no">183</span><span id="line-183"></span>
<span class="source-line-no">184</span><span id="line-184">                //System.out.print(pair.toString());</span>
<span class="source-line-no">185</span><span id="line-185"></span>
<span class="source-line-no">186</span><span id="line-186">                //3. Convert the alignment back to Atoms</span>
<span class="source-line-no">187</span><span id="line-187">                AlignedSequence&lt;ProteinSequence,AminoAcidCompound&gt; alignedSeq = pair.getQuery();</span>
<span class="source-line-no">188</span><span id="line-188">                AlignedSequence&lt;ProteinSequence,AminoAcidCompound&gt; alignedStruct = pair.getTarget();</span>
<span class="source-line-no">189</span><span id="line-189"></span>
<span class="source-line-no">190</span><span id="line-190"></span>
<span class="source-line-no">191</span><span id="line-191">                assert(alignedSeq.getLength() == alignedStruct.getLength());</span>
<span class="source-line-no">192</span><span id="line-192"></span>
<span class="source-line-no">193</span><span id="line-193">//              System.out.println(pair.toString(80));</span>
<span class="source-line-no">194</span><span id="line-194">//              System.out.format("%d/min{%d,%d}\n", pair.getNumIdenticals(),</span>
<span class="source-line-no">195</span><span id="line-195">//                              alignedSeq.getLength()-alignedSeq.getNumGaps(),</span>
<span class="source-line-no">196</span><span id="line-196">//                              alignedStruct.getLength()-alignedStruct.getNumGaps());</span>
<span class="source-line-no">197</span><span id="line-197"></span>
<span class="source-line-no">198</span><span id="line-198">                ResidueNumber[] ca = new ResidueNumber[seq.getLength()];</span>
<span class="source-line-no">199</span><span id="line-199"></span>
<span class="source-line-no">200</span><span id="line-200">                for( int pos = alignedSeq.getStart().getPosition(); pos &lt;= alignedSeq.getEnd().getPosition(); pos++ ) { // 1-indexed</span>
<span class="source-line-no">201</span><span id="line-201">                        //skip missing residues from sequence. These probably represent alignment errors</span>
<span class="source-line-no">202</span><span id="line-202">                        if(alignedSeq.isGap(pos)) {</span>
<span class="source-line-no">203</span><span id="line-203">                                int structIndex = alignedStruct.getSequenceIndexAt(pos)-1;</span>
<span class="source-line-no">204</span><span id="line-204">                                assert(structIndex &gt; 0);//should be defined since seq gap</span>
<span class="source-line-no">205</span><span id="line-205"></span>
<span class="source-line-no">206</span><span id="line-206">                                Group g = atomIndexPosition.get(structIndex);</span>
<span class="source-line-no">207</span><span id="line-207"></span>
<span class="source-line-no">208</span><span id="line-208">                                logger.warn("Chain {} residue {} in the Structure {} has no corresponding amino acid in the sequence.",</span>
<span class="source-line-no">209</span><span id="line-209">                                                g.getChainId(),</span>
<span class="source-line-no">210</span><span id="line-210">                                                g.getResidueNumber().toString(),</span>
<span class="source-line-no">211</span><span id="line-211">                                                g.getChain().getStructure().getPDBCode() );</span>
<span class="source-line-no">212</span><span id="line-212">                                continue;</span>
<span class="source-line-no">213</span><span id="line-213">                        }</span>
<span class="source-line-no">214</span><span id="line-214"></span>
<span class="source-line-no">215</span><span id="line-215">                        if(! alignedStruct.isGap(pos) ) {</span>
<span class="source-line-no">216</span><span id="line-216">                                int seqIndex = alignedSeq.getSequenceIndexAt(pos)-1;//1-indexed</span>
<span class="source-line-no">217</span><span id="line-217">                                int structIndex = alignedStruct.getSequenceIndexAt(pos)-1;//1-indexed</span>
<span class="source-line-no">218</span><span id="line-218">                                Group g = atomIndexPosition.get(structIndex);</span>
<span class="source-line-no">219</span><span id="line-219"></span>
<span class="source-line-no">220</span><span id="line-220">                                assert(0&lt;=seqIndex &amp;&amp; seqIndex &lt; ca.length);</span>
<span class="source-line-no">221</span><span id="line-221"></span>
<span class="source-line-no">222</span><span id="line-222">                                ca[seqIndex] = g.getResidueNumber(); //remains null for gaps</span>
<span class="source-line-no">223</span><span id="line-223">                        }</span>
<span class="source-line-no">224</span><span id="line-224">                }</span>
<span class="source-line-no">225</span><span id="line-225">                return ca;</span>
<span class="source-line-no">226</span><span id="line-226">        }</span>
<span class="source-line-no">227</span><span id="line-227"></span>
<span class="source-line-no">228</span><span id="line-228"></span>
<span class="source-line-no">229</span><span id="line-229">        /**</span>
<span class="source-line-no">230</span><span id="line-230">         * Removes all gaps ('-') from a protein sequence</span>
<span class="source-line-no">231</span><span id="line-231">         * @param gapped</span>
<span class="source-line-no">232</span><span id="line-232">         * @return</span>
<span class="source-line-no">233</span><span id="line-233">         */</span>
<span class="source-line-no">234</span><span id="line-234">        public static ProteinSequence removeGaps(ProteinSequence gapped) {</span>
<span class="source-line-no">235</span><span id="line-235">                final String[] gapStrings = {"-","."};</span>
<span class="source-line-no">236</span><span id="line-236"></span>
<span class="source-line-no">237</span><span id="line-237">                StringBuilder seq = new StringBuilder();</span>
<span class="source-line-no">238</span><span id="line-238"></span>
<span class="source-line-no">239</span><span id="line-239">                CompoundSet&lt;AminoAcidCompound&gt; aaSet = gapped.getCompoundSet();</span>
<span class="source-line-no">240</span><span id="line-240">                AminoAcidCompound[] gaps = new AminoAcidCompound[gapStrings.length];</span>
<span class="source-line-no">241</span><span id="line-241"></span>
<span class="source-line-no">242</span><span id="line-242">                for(int i=0;i&lt;gapStrings.length;i++) {</span>
<span class="source-line-no">243</span><span id="line-243">                        gaps[i] = aaSet.getCompoundForString(gapStrings[i]);</span>
<span class="source-line-no">244</span><span id="line-244">                }</span>
<span class="source-line-no">245</span><span id="line-245"></span>
<span class="source-line-no">246</span><span id="line-246">                for(int i=1; i&lt;=gapped.getLength();i++) { //1-indexed</span>
<span class="source-line-no">247</span><span id="line-247">                        AminoAcidCompound aa = gapped.getCompoundAt(i);</span>
<span class="source-line-no">248</span><span id="line-248">                        boolean isGap = false;</span>
<span class="source-line-no">249</span><span id="line-249">                        for(AminoAcidCompound gap : gaps) {</span>
<span class="source-line-no">250</span><span id="line-250">                                if( aa.equals(gap)) {</span>
<span class="source-line-no">251</span><span id="line-251">                                        isGap = true;</span>
<span class="source-line-no">252</span><span id="line-252">                                        break;</span>
<span class="source-line-no">253</span><span id="line-253">                                }</span>
<span class="source-line-no">254</span><span id="line-254">                        }</span>
<span class="source-line-no">255</span><span id="line-255">                        if(!isGap) {</span>
<span class="source-line-no">256</span><span id="line-256">                                seq.append(aa.getShortName());</span>
<span class="source-line-no">257</span><span id="line-257">                        }</span>
<span class="source-line-no">258</span><span id="line-258">                }</span>
<span class="source-line-no">259</span><span id="line-259"></span>
<span class="source-line-no">260</span><span id="line-260">                ProteinSequence ungapped = null;</span>
<span class="source-line-no">261</span><span id="line-261">                try {</span>
<span class="source-line-no">262</span><span id="line-262">                        ungapped = new ProteinSequence(seq.toString());</span>
<span class="source-line-no">263</span><span id="line-263">                } catch (CompoundNotFoundException e) {</span>
<span class="source-line-no">264</span><span id="line-264">                        // this can't happen, if it does there's a bug somewhere</span>
<span class="source-line-no">265</span><span id="line-265">                        logger.error("Could not create ungapped protein sequence, found unknown compounds: {}. This is most likely a bug.", e.getMessage());</span>
<span class="source-line-no">266</span><span id="line-266">                }</span>
<span class="source-line-no">267</span><span id="line-267"></span>
<span class="source-line-no">268</span><span id="line-268">                return ungapped;</span>
<span class="source-line-no">269</span><span id="line-269">        }</span>
<span class="source-line-no">270</span><span id="line-270"></span>
<span class="source-line-no">271</span><span id="line-271">        /**</span>
<span class="source-line-no">272</span><span id="line-272">         * Creates a new list consisting of all columns of gapped where no row</span>
<span class="source-line-no">273</span><span id="line-273">         * contained a null value.</span>
<span class="source-line-no">274</span><span id="line-274">         *</span>
<span class="source-line-no">275</span><span id="line-275">         * Here, "row" refers to the first index and "column" to the second, eg</span>
<span class="source-line-no">276</span><span id="line-276">         * gapped.get(row).get(column)</span>
<span class="source-line-no">277</span><span id="line-277">         * @param gapped A rectangular matrix containing null to mark gaps</span>
<span class="source-line-no">278</span><span id="line-278">         * @return A new List without columns containing nulls</span>
<span class="source-line-no">279</span><span id="line-279">         */</span>
<span class="source-line-no">280</span><span id="line-280">        public static &lt;T&gt; T[][] removeGaps(final T[][] gapped) {</span>
<span class="source-line-no">281</span><span id="line-281">                if(gapped == null ) return null;</span>
<span class="source-line-no">282</span><span id="line-282">                if(gapped.length &lt; 1) return Arrays.copyOf(gapped, gapped.length);</span>
<span class="source-line-no">283</span><span id="line-283"></span>
<span class="source-line-no">284</span><span id="line-284">                final int nProts = gapped.length;</span>
<span class="source-line-no">285</span><span id="line-285">                final int protLen = gapped[0].length; // length of gapped proteins</span>
<span class="source-line-no">286</span><span id="line-286"></span>
<span class="source-line-no">287</span><span id="line-287">                // Verify that input is rectangular</span>
<span class="source-line-no">288</span><span id="line-288">                for(int i=0;i&lt;nProts;i++) {</span>
<span class="source-line-no">289</span><span id="line-289">                        if(gapped[i].length != protLen) {</span>
<span class="source-line-no">290</span><span id="line-290">                                throw new IllegalArgumentException(String.format(</span>
<span class="source-line-no">291</span><span id="line-291">                                                "Expected a rectangular array, but row 0 has %d elements " +</span>
<span class="source-line-no">292</span><span id="line-292">                                                "while row %d has %d.", protLen,i,gapped[i].length));</span>
<span class="source-line-no">293</span><span id="line-293"></span>
<span class="source-line-no">294</span><span id="line-294">                        }</span>
<span class="source-line-no">295</span><span id="line-295">                }</span>
<span class="source-line-no">296</span><span id="line-296"></span>
<span class="source-line-no">297</span><span id="line-297">                // determine where gaps exist in any structures</span>
<span class="source-line-no">298</span><span id="line-298">                boolean[] isGap = new boolean[protLen];</span>
<span class="source-line-no">299</span><span id="line-299">                int gaps = 0;</span>
<span class="source-line-no">300</span><span id="line-300">                for(int j=0;j&lt;protLen;j++) {</span>
<span class="source-line-no">301</span><span id="line-301">                        for(int i=0;i&lt;nProts;i++) {</span>
<span class="source-line-no">302</span><span id="line-302">                                if(gapped[i][j] == null ) {</span>
<span class="source-line-no">303</span><span id="line-303">                                        isGap[j] = true;</span>
<span class="source-line-no">304</span><span id="line-304">                                        gaps++;</span>
<span class="source-line-no">305</span><span id="line-305">                                        break; //go to next position</span>
<span class="source-line-no">306</span><span id="line-306">                                }</span>
<span class="source-line-no">307</span><span id="line-307">                        }</span>
<span class="source-line-no">308</span><span id="line-308">                }</span>
<span class="source-line-no">309</span><span id="line-309"></span>
<span class="source-line-no">310</span><span id="line-310">                // Create ungapped array</span>
<span class="source-line-no">311</span><span id="line-311">                T[][] ungapped = Arrays.copyOf(gapped,nProts);</span>
<span class="source-line-no">312</span><span id="line-312">                final int ungappedLen = protLen-gaps;</span>
<span class="source-line-no">313</span><span id="line-313">                for(int i=0;i&lt;nProts;i++) {</span>
<span class="source-line-no">314</span><span id="line-314">                        ungapped[i] = Arrays.copyOf(gapped[i],ungappedLen);</span>
<span class="source-line-no">315</span><span id="line-315">                        int k = 0;</span>
<span class="source-line-no">316</span><span id="line-316">                        for(int j=0;j&lt;protLen;j++) {</span>
<span class="source-line-no">317</span><span id="line-317">                                if(!isGap[j]) { //skip gaps</span>
<span class="source-line-no">318</span><span id="line-318">                                        assert(gapped[i][j] != null);</span>
<span class="source-line-no">319</span><span id="line-319">                                        ungapped[i][k] = gapped[i][j];</span>
<span class="source-line-no">320</span><span id="line-320">                                        k++;</span>
<span class="source-line-no">321</span><span id="line-321">                                }</span>
<span class="source-line-no">322</span><span id="line-322">                        }</span>
<span class="source-line-no">323</span><span id="line-323">                        assert(k == ungappedLen);</span>
<span class="source-line-no">324</span><span id="line-324">                }</span>
<span class="source-line-no">325</span><span id="line-325"></span>
<span class="source-line-no">326</span><span id="line-326">                return ungapped;</span>
<span class="source-line-no">327</span><span id="line-327">        }</span>
<span class="source-line-no">328</span><span id="line-328">}</span>




























































</pre>
</div>
</main>
</body>
</html>
