<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:51 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Class org.biojava.nbio.structure.align.model.AFPChain (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class org.biojava.nbio.structure.align.model.AFPChain (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/structure/align/model/class-use/AFPChain.html" target="_top">Frames</a></li>
<li><a href="AFPChain.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class org.biojava.nbio.structure.align.model.AFPChain" class="title">Uses of Class<br>org.biojava.nbio.structure.align.model.AFPChain</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align">org.biojava.nbio.structure.align</a></td>
<td class="colLast">
<div class="block">Classes for the alignment of structures.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.ce">org.biojava.nbio.structure.align.ce</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the CE alignment algorithm, here called jCE.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.client">org.biojava.nbio.structure.align.client</a></td>
<td class="colLast">
<div class="block">This package deals with the server communication for auto-downloading pre-calculated alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.fatcat">org.biojava.nbio.structure.align.fatcat</a></td>
<td class="colLast">
<div class="block">Classes related to the implementation of the FATCAT alignment algorithm, here called jFATCAT.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.fatcat.calc">org.biojava.nbio.structure.align.fatcat.calc</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui">org.biojava.nbio.structure.align.gui</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.aligpanel">org.biojava.nbio.structure.align.gui.aligpanel</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.gui.jmol">org.biojava.nbio.structure.align.gui.jmol</a></td>
<td class="colLast">
<div class="block">Utility methods for better interaction with Jmol.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.model">org.biojava.nbio.structure.align.model</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.multiple">org.biojava.nbio.structure.align.multiple</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.seq">org.biojava.nbio.structure.align.seq</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.util">org.biojava.nbio.structure.align.util</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.align.xml">org.biojava.nbio.structure.align.xml</a></td>
<td class="colLast">
<div class="block">Serialization and deserialization for structure alignment results (as AFPChain object) to XML.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.io">org.biojava.nbio.structure.io</a></td>
<td class="colLast">
<div class="block">Input and Output of Structures</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.internal">org.biojava.nbio.structure.symmetry.internal</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.structure.symmetry.utils">org.biojava.nbio.structure.symmetry.utils</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="org.biojava.nbio.structure.align">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>abstract <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Run an alignment while specifying the atoms to be aligned.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">BioJavaStructureAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/BioJavaStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>abstract <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AbstractStructureAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/AbstractStructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StructureAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/StructureAlignment.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;params)</code>
<div class="block">run an alignment and also send a bean containing the parameters.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CallableStructureAlignment.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/CallableStructureAlignment.html#call()">call</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/package-summary.html">org.biojava.nbio.structure.align</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/AFPTwister.html#afp2Res(org.biojava.nbio.structure.align.model.AFPChain,%20int,%20int[],%20int)">afp2Res</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       int&nbsp;afpn,
       int[]&nbsp;afpPositions,
       int&nbsp;listStart)</code>
<div class="block">Set the list of equivalent residues in the two proteins given a list of AFPs

 WARNING: changes the values for FocusRes1, focusRes2 and FocusResn in afpChain!</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistOptimized(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistOptimized</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">superimposing according to the optimized alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">AFPTwister.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/AFPTwister.html#twistPDB(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">twistPDB</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">calculate the total rmsd of the blocks
 output a merged pdb file for both proteins
          protein 1, in chain A
         protein 2 is twisted according to the twists detected, in chain B</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.ce">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Aligns ca1 and ca2 using a heuristic to check for CPs.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeMain.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>
<div class="block">Align ca2 onto ca1.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20org.biojava.nbio.structure.align.model.AFPChain[])">alignOptimal</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
            <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignPermuted(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20int)">alignPermuted</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
             int&nbsp;cp)</code>
<div class="block">Aligns ca1 with ca2 permuted by <i>cp</i> residues.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">filterDuplicateAFPs</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code>
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECPParameters)">filterDuplicateAFPs</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
                   <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#invertAlignment(org.biojava.nbio.structure.align.model.AFPChain)">invertAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a)</code>
<div class="block">Swaps the order of structures in an AFPChain</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator)">postProcessAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.align.ce.CECPParameters)">postProcessAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/ce/package-summary.html">org.biojava.nbio.structure.align.ce</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">OptimalCECPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/OptimalCECPMain.html#alignOptimal(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object,%20org.biojava.nbio.structure.align.model.AFPChain[])">alignOptimal</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param,
            <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]&nbsp;alignments)</code>
<div class="block">Finds the optimal alignment between two proteins allowing for a circular
 permutation (CP).</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">convertAfpChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#convertAfpChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">convertAfpChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">copy data from this class into AFPChain container object.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">display</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#extractFragments(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractFragments</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">filterDuplicateAFPs</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated)</code>
<div class="block">Takes as input an AFPChain where ca2 has been artificially duplicated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#filterDuplicateAFPs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECPParameters)">filterDuplicateAFPs</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;ceCalc,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2duplicated,
                   <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AbstractUserArgumentProcessor.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/AbstractUserArgumentProcessor.html#getDbSearchResult(org.biojava.nbio.structure.align.model.AFPChain)">getDbSearchResult</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#invertAlignment(org.biojava.nbio.structure.align.model.AFPChain)">invertAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a)</code>
<div class="block">Swaps the order of structures in an AFPChain</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#nextStep(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">nextStep</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator)">postProcessAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeCPMain.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCPMain.html#postProcessAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.align.ce.CECPParameters)">postProcessAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2m,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECPParameters.html" title="class in org.biojava.nbio.structure.align.ce">CECPParameters</a>&nbsp;param)</code>
<div class="block">Circular permutation specific code to be run after the standard CE alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">prepareGroupsForDisplay</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">GuiWrapper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/GuiWrapper.html#showAlignmentImage(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">showAlignmentImage</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;jmol)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeCalculatorEnhanced.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CeCalculatorEnhanced.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">traceFragmentMatrix</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CECalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html#traceFragmentMatrix(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">traceFragmentMatrix</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.client">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAFPChainFromServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#getAFPChainFromServer(java.lang.String,%20java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">getAFPChainFromServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;method,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                     int&nbsp;timeout)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/client/package-summary.html">org.biojava.nbio.structure.align.client</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">JFatCatClient.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/client/JFatCatClient.html#sendAFPChainToServer(java.lang.String,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">sendAFPChainToServer</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;serverLocation,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.fatcat">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/package-summary.html">org.biojava.nbio.structure.align.fatcat</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20boolean)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
     boolean&nbsp;doRigid)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatRigid.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCatRigid.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatFlexible.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCatFlexible.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;param)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignFlexible(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignFlexible</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">alignRigid</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCat.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/FatCat.html#alignRigid(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters)">alignRigid</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
          <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.fatcat.calc">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FatCatAligner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatAligner.html#getAfpChain()">getAfpChain</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/package-summary.html">org.biojava.nbio.structure.align.fatcat.calc</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="strong">AFPChainer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html#afpPairConn(int,%20int,%20org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain)">afpPairConn</a></strong>(int&nbsp;afp1,
           int&nbsp;afp2,
           <a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">//Key function: calculate the connectivity of AFP pairs
        //no compatibility criteria is executed
        //note: afp1 is previous to afp2 in terms of the position
                 //this module must be optimized</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPOptimizer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html#blockInfo(org.biojava.nbio.structure.align.model.AFPChain)">blockInfo</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">get the afp list and residue list for each block</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected static double</code></td>
<td class="colLast"><span class="strong">AFPChainer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html#calAfpRmsd(int,%20int[],%20int,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">calAfpRmsd</a></strong>(int&nbsp;afpn,
          int[]&nbsp;afpPositions,
          int&nbsp;listStart,
          <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">//return the rmsd of the residues from the segments that form the given AFP list
        //this value can be a measurement (1) for the connectivity of the AFPs</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">SigEva.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/SigEva.html#calNS(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain)">calNS</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
     <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>double</code></td>
<td class="colLast"><span class="strong">SigEva.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/SigEva.html#calSigAll(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain)">calSigAll</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
         <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPChainer.html#doChainAfp(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">doChainAfp</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
          <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">// Key function: chain (assembly) the AFPs
        // a AFP (k) is defined as (i, j, k), with i and j are staring points
        // AFP extension (eg.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html#extractAFPChains(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">extractAFPChains</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
                <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPOptimizer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html#optimizeAln(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">optimizeAln</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">optimize the alignment by dynamic programming</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPPostProcessor.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPPostProcessor.html#postProcess(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">postProcess</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
           <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
           <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPCalculator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPCalculator.html#sortAfps(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">sortAfps</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPOptimizer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/AFPOptimizer.html#updateScore(org.biojava.nbio.structure.align.fatcat.calc.FatCatParameters,%20org.biojava.nbio.structure.align.model.AFPChain)">updateScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/fatcat/calc/FatCatParameters.html" title="class in org.biojava.nbio.structure.align.fatcat.calc">FatCatParameters</a>&nbsp;params,
           <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">to update the chaining score after block delete and merge processed
 the blockScore value is important for significance evaluation</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">createArtificalStructure</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Create a "fake" Structure objects that contains the two sets of atoms aligned on top of each other.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">display</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Display an AFPChain alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">StructureAlignmentJmol</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#display(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Group[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.List,%20java.util.List)">display</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
       <a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>&gt;&nbsp;hetatms2)</code>
<div class="block">Note: ca2, hetatoms2 and nucleotides2 should not be rotated.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing">JMenuBar</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#getAlignmentPanelMenu(javax.swing.JFrame,%20java.awt.event.ActionListener,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAlignmentPanelMenu</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing">JFrame</a>&nbsp;frame,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/awt/event/ActionListener.html?is-external=true" title="class or interface in java.awt.event">ActionListener</a>&nbsp;actionListener,
                     <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                     <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Create the menu for the Alignment Panel representation of
 Structural Alignments.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing">JMenuBar</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#getAlignmentTextMenu(javax.swing.JFrame,%20java.awt.event.ActionListener,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getAlignmentTextMenu</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing">JFrame</a>&nbsp;frame,
                    <a href="http://docs.oracle.com/javase/7/docs/api/java/awt/event/ActionListener.html?is-external=true" title="class or interface in java.awt.event">ActionListener</a>&nbsp;actionListener,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                    <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Create the menu for the Text representations of Structural Alignments.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getAtomForAligPos(org.biojava.nbio.structure.align.model.AFPChain,%20int,%20int,%20org.biojava.nbio.structure.Atom[],%20boolean)">getAtomForAligPos</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 int&nbsp;chainNr,
                 int&nbsp;aligPos,
                 <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca,
                 boolean&nbsp;getPrevious)</code>
<div class="block">return the atom at alignment position aligPos.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getBlockNrForAlignPos(org.biojava.nbio.structure.align.model.AFPChain,%20int)">getBlockNrForAlignPos</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                     int&nbsp;aligPos)</code>
<div class="block"><strong>Deprecated.</strong>&nbsp;
<div class="block"><i>use AFPAlignmentDisplay.getBlockNrForAlignPos instead...</i></div>
</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getEQRAlignmentPos(org.biojava.nbio.structure.align.model.AFPChain)">getEQRAlignmentPos</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&gt;</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#getPDBresnum(int,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getPDBresnum</a></strong>(int&nbsp;aligPos,
            <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>
<div class="block">Return a list of pdb Strings corresponding to the aligned positions of the molecule.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuItem.html?is-external=true" title="class or interface in javax.swing">JMenuItem</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#getSaveAlignmentMenuItem(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">getSaveAlignmentMenuItem</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JMenuBar.html?is-external=true" title="class or interface in javax.swing">JMenuBar</a></code></td>
<td class="colLast"><span class="strong">MenuCreator.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.html#initJmolMenu(javax.swing.JFrame,%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">initJmolMenu</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/javax/swing/JFrame.html?is-external=true" title="class or interface in javax.swing">JFrame</a>&nbsp;frame,
            <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;parent,
            <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Provide a JMenuBar that can be added to a JFrame containing
 a JmolPanel.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]</code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#prepareGroupsForDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">prepareGroupsForDisplay</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Rotate the Atoms/Groups so they are aligned for the 3D visualisation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">StructureAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/StructureAlignmentDisplay.html#shiftCA2(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.jama.Matrix,%20org.biojava.nbio.structure.Atom,%20org.biojava.nbio.structure.Group[])">shiftCA2</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;m,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>&nbsp;shift,
        <a href="../../../../../../../org/biojava/nbio/structure/Group.html" title="interface in org.biojava.nbio.structure">Group</a>[]&nbsp;twistedGroups)</code>
<div class="block">only shift CA positions.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#showAlignmentImage(org.biojava.nbio.structure.align.model.AFPChain,%20java.lang.String)">showAlignmentImage</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;result)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">DisplayAFP.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DisplayAFP.html#showAlignmentPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">showAlignmentPanel</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                  <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                  <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/package-summary.html">org.biojava.nbio.structure.align.gui</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/DotPlotPanel.html#DotPlotPanel(org.biojava.nbio.structure.align.model.AFPChain)">DotPlotPanel</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;alignment)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MenuCreator.DotPlotListener.html#MenuCreator.DotPlotListener(org.biojava.nbio.structure.align.model.AFPChain)">MenuCreator.DotPlotListener</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MySaveFileListener.html#MySaveFileListener(org.biojava.nbio.structure.align.model.AFPChain)">MySaveFileListener</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/MySaveFileListener.html#MySaveFileListener(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.align.multiple.MultipleAlignment)">MySaveFileListener</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                  <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a>&nbsp;msa)</code>
<div class="block">Constructor to avoid checking which of the two is null before
 instantiating this class.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.aligpanel">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">StatusDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html#getAfpChain()">getAfpChain</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AligPanel.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html#getAFPChain()">getAFPChain</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">StatusDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/StatusDisplay.html#setAfpChain(org.biojava.nbio.structure.align.model.AFPChain)">setAfpChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AFPChainCoordManager.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/AFPChainCoordManager.html#setAFPChain(org.biojava.nbio.structure.align.model.AFPChain)">setAFPChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">AligPanel.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/AligPanel.html#setAFPChain(org.biojava.nbio.structure.align.model.AFPChain)">setAFPChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/package-summary.html">org.biojava.nbio.structure.align.gui.aligpanel</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/aligpanel/MultipleAligPanel.html#MultipleAligPanel(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.align.gui.jmol.AbstractAlignmentJmol)">MultipleAligPanel</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                 <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/AbstractAlignmentJmol.html" title="class in org.biojava.nbio.structure.align.gui.jmol">AbstractAlignmentJmol</a>&nbsp;jmol)</code>
<div class="block">Constructor using an afpChain and the atom arrays for pairwise
 alignments.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.gui.jmol">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentJmol.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#getJmolScript4Block(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">getJmolScript4Block</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                   int&nbsp;blockNr)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">StructureAlignmentJmol.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#getJmolString(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getJmolString</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/package-summary.html">org.biojava.nbio.structure.align.gui.jmol</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/gui/jmol/StructureAlignmentJmol.html#StructureAlignmentJmol(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">StructureAlignmentJmol</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.model">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/model/package-summary.html">org.biojava.nbio.structure.align.model</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toAlignedPairs(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toAlignedPairs</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Prints the alignment in the simplest form: a list of aligned residues.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toCE(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toCE</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
    <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toDBSearchResult(org.biojava.nbio.structure.align.model.AFPChain)">toDBSearchResult</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toFatCat(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toFatCat</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toFatCatCore(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean,%20boolean,%20boolean,%20boolean)">toFatCatCore</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
            <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
            <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
            boolean&nbsp;printLegend,
            boolean&nbsp;longHeader,
            boolean&nbsp;showHTML,
            boolean&nbsp;showAlignmentBlock)</code>
<div class="block">Output in FatCatCore format</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toRotMat(org.biojava.nbio.structure.align.model.AFPChain)">toRotMat</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toScoresList(org.biojava.nbio.structure.align.model.AFPChain)">toScoresList</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toWebSiteDisplay</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AfpChainWriter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AfpChainWriter.html#toWebSiteDisplay(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">toWebSiteDisplay</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                boolean&nbsp;showAlignmentBlock)</code>
<div class="block">Prints the afpChain as a nicely formatted alignment, including alignment
 statistics, the aligned sequences themselves, and information about the
 superposition.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/model/package-summary.html">org.biojava.nbio.structure.align.model</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#AFPChain(org.biojava.nbio.structure.align.model.AFPChain)">AFPChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;o)</code>
<div class="block">Copy constructor</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.multiple">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/package-summary.html">org.biojava.nbio.structure.align.multiple</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignmentEnsembleImpl.html#MultipleAlignmentEnsembleImpl(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">MultipleAlignmentEnsembleImpl</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afp,
                             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                             boolean&nbsp;flexible)</code>
<div class="block">Constructor from an AFPChain instance.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.seq">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/seq/package-summary.html">org.biojava.nbio.structure.align.seq</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SmithWaterman3Daligner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/seq/SmithWaterman3Daligner.html#align(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.Object)">align</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a>&nbsp;parameters)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.util">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#createAFPChain(org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.ResidueNumber[],%20org.biojava.nbio.structure.ResidueNumber[])">createAFPChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="../../../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned1,
              <a href="../../../../../../../org/biojava/nbio/structure/ResidueNumber.html" title="class in org.biojava.nbio.structure">ResidueNumber</a>[]&nbsp;aligned2)</code>
<div class="block">Fundamentally, an alignment is just a list of aligned residues in each
 protein.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int[],%20int[][][])">replaceOptAln</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;blockNum,
             int[]&nbsp;optLens,
             int[][][]&nbsp;optAln)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.Map)">replaceOptAln</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;alignment)</code>
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(int[][][],%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">replaceOptAln</a></strong>(int[][][]&nbsp;newAlgn,
             <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">splitBlocksByTopology</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#alignmentAsMap(org.biojava.nbio.structure.align.model.AFPChain)">alignmentAsMap</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">Creates a Map specifying the alignment as a mapping between residue indices
 of protein 1 and residue indices of protein 2.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#alignmentToSIF(java.io.Writer,%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.lang.String,%20java.lang.String)">alignmentToSIF</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/Writer.html?is-external=true" title="class or interface in java.io">Writer</a>&nbsp;out,
              <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;backboneInteraction,
              <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;alignmentInteraction)</code>
<div class="block">Creates a simple interaction format (SIF) file for an alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#createArtificalStructure(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">createArtificalStructure</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAlign</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Extract the alignment output</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlign(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20boolean)">getAlign</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
        boolean&nbsp;showSeq)</code>
<div class="block">Sets the following properties:
 
 The alignment strings <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnseq1(char[])"><code>alnseq1</code></a>,
  <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnseq2(char[])"><code>alnseq2</code></a>,
  and <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnsymb(char[])"><code>alnsymb</code></a>
 <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnbeg1(int)"><code>alnbeg1</code></a> and 2
 <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setAlnLength(int)"><code>alnLength</code></a> and
  <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html#setGapLen(int)"><code>gapLen</code></a>
 </div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlignedAtoms1(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getAlignedAtoms1</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getAlignedAtoms2(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">getAlignedAtoms2</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">RotationAxis.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain)">getAngle</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">Calculate the rotation angle for a structure</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getBlockNrForAlignPos(org.biojava.nbio.structure.align.model.AFPChain,%20int)">getBlockNrForAlignPos</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                     int&nbsp;aligPos)</code>
<div class="block">get the block number for an aligned position</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&gt;&gt;</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#getOptAlnAsList(org.biojava.nbio.structure.align.model.AFPChain)">getOptAlnAsList</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">Retrieves the optimum alignment from an AFPChain and returns it as a
 java collection.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getRotMax(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getRotMax</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
         <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
         <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static int</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#getSymmetryOrder(org.biojava.nbio.structure.align.model.AFPChain,%20int,%20float)">getSymmetryOrder</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                int&nbsp;maxSymmetry,
                float&nbsp;minimumMetricChange)</code>
<div class="block">Guesses the order of symmetry in an alignment</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">AFPChainScorer.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPChainScorer.html#getTMScore(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTMScore</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;align,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
          <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a></code></td>
<td class="colLast"><span class="strong">AFPAlignmentDisplay.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AFPAlignmentDisplay.html#getTranslation(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getTranslation</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static boolean</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#isSequentialAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20boolean)">isSequentialAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                     boolean&nbsp;checkWithinBlocks)</code>
<div class="block">Checks that the alignment given by afpChain is sequential.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int,%20int[],%20int[][][])">replaceOptAln</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             int&nbsp;blockNum,
             int[]&nbsp;optLens,
             int[][][]&nbsp;optAln)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20java.util.Map)">replaceOptAln</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
             <a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&nbsp;alignment)</code>
<div class="block">Takes an AFPChain and replaces the optimal alignment based on an alignment map</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#replaceOptAln(int[][][],%20org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">replaceOptAln</a></strong>(int[][][]&nbsp;newAlgn,
             <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
             <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">It replaces an optimal alignment of an AFPChain and calculates all the new alignment scores and variables.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#splitBlocksByTopology(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">splitBlocksByTopology</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;a,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AlignmentTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/AlignmentTools.html#updateSuperposition(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">updateSuperposition</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">After the alignment changes (optAln, optLen, blockNum, at a minimum),
 many other properties which depend on the superposition will be invalid.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing constructors, and an explanation">
<caption><span>Constructors in <a href="../../../../../../../org/biojava/nbio/structure/align/util/package-summary.html">org.biojava.nbio.structure.align.util</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colLast"><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/util/RotationAxis.html#RotationAxis(org.biojava.nbio.structure.align.model.AFPChain)">RotationAxis</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>
<div class="block">Calculate the rotation axis for the first block of an AFPChain</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.align.xml">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainFlipper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainFlipper.html#flipChain(org.biojava.nbio.structure.align.model.AFPChain)">flipChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;o)</code>
<div class="block">Flip the position of name1 and name2 (as well as all underlying data) in an AFPChain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">fromXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#fromXML(java.lang.String,%20java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">fromXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name1,
       <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;name2,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">new utility method that checks that the order of the pair in the XML alignment is correct and flips the direction if needed</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>[]</code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#parseMultiXML(java.lang.String)">parseMultiXML</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;xml)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/align/xml/package-summary.html">org.biojava.nbio.structure.align.xml</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">AFPChainFlipper.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainFlipper.html#flipChain(org.biojava.nbio.structure.align.model.AFPChain)">flipChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;o)</code>
<div class="block">Flip the position of name1 and name2 (as well as all underlying data) in an AFPChain.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#printXMLEQRInferPositions(org.biojava.nbio.core.util.PrettyXMLWriter,%20org.biojava.nbio.structure.align.model.AFPChain,%20int,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">printXMLEQRInferPositions</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
                         <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                         int&nbsp;bk,
                         <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
                         <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#printXMLHeader(org.biojava.nbio.core.util.PrettyXMLWriter,%20org.biojava.nbio.structure.align.model.AFPChain)">printXMLHeader</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/util/PrettyXMLWriter.html" title="class in org.biojava.nbio.core.util">PrettyXMLWriter</a>&nbsp;xml,
              <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLParser.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLParser.html#rebuildAFPChain(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">rebuildAFPChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
               <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
               <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">replace the PDB res nums with atom positions:</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain)">toXML</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toXML</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Convert an afpChain to a simple XML representation</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static void</code></td>
<td class="colLast"><span class="strong">AFPChainXMLConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/align/xml/AFPChainXMLConverter.html#toXML(org.biojava.nbio.structure.align.model.AFPChain,%20java.io.StringWriter,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">toXML</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
     <a href="http://docs.oracle.com/javase/7/docs/api/java/io/StringWriter.html?is-external=true" title="class or interface in java.io">StringWriter</a>&nbsp;swriter,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
     <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Write the XML representation to a StringWriter</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.io">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/io/package-summary.html">org.biojava.nbio.structure.io</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.io.File,%20org.biojava.nbio.structure.Structure,%20int)">cpFastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
                 <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.structure.Structure,%20int)">cpFastaToAfpChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;first,
                 <a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;second,
                 <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>
<div class="block">Takes a structure and sequence corresponding to an alignment between a structure or sequence and itself (or even a structure with a sequence), where the result has a circular permutation site
 <code>cpSite</code> residues to the right.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#cpFastaToAfpChain(java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20int)">cpFastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;first,
                 <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;second,
                 <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure,
                 int&nbsp;cpSite)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaFileToAfpChain(java.io.File,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaFileToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/io/File.html?is-external=true" title="class or interface in java.io">File</a>&nbsp;fastaFile,
                   <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                   <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Reads the file <code>fastaFile</code>, expecting exactly two sequences which give a pairwise alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaStringToAfpChain(java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaStringToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence1,
                     <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence2,
                     <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
                     <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(java.util.Map,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/Map.html?is-external=true" title="class or interface in java.util">Map</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>,<a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&gt;&nbsp;sequences,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Uses two sequences each with a corresponding structure to create an AFPChain corresponding to the alignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.core.sequence.ProteinSequence,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence1,
               <a href="../../../../../../../org/biojava/nbio/core/sequence/ProteinSequence.html" title="class in org.biojava.nbio.core.sequence">ProteinSequence</a>&nbsp;sequence2,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Returns an AFPChain corresponding to the alignment between <code>structure1</code> and <code>structure2</code>, which is given by the gapped protein sequences <code>sequence1</code> and <code>sequence2</code>.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(org.biojava.nbio.core.alignment.template.SequencePair,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="../../../../../../../org/biojava/nbio/core/alignment/template/SequencePair.html" title="interface in org.biojava.nbio.core.alignment.template">SequencePair</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/template/Sequence.html" title="interface in org.biojava.nbio.core.sequence.template">Sequence</a>&lt;<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;,<a href="../../../../../../../org/biojava/nbio/core/sequence/compound/AminoAcidCompound.html" title="class in org.biojava.nbio.core.sequence.compound">AminoAcidCompound</a>&gt;&nbsp;alignment,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">Provided only for convenience.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">FastaAFPChainConverter.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/io/FastaAFPChainConverter.html#fastaToAfpChain(java.lang.String,%20java.lang.String,%20org.biojava.nbio.structure.Structure,%20org.biojava.nbio.structure.Structure)">fastaToAfpChain</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence1,
               <a href="http://docs.oracle.com/javase/7/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;sequence2,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure1,
               <a href="../../../../../../../org/biojava/nbio/structure/Structure.html" title="interface in org.biojava.nbio.structure">Structure</a>&nbsp;structure2)</code>
<div class="block">TODO Write comment</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.internal">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> that return <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">CeSymmResult.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html#getSelfAlignment()">getSelfAlignment</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">refineSymmetry</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;k)</code>
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/package-summary.html">org.biojava.nbio.structure.symmetry.internal</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">AngleOrderDetectorPlus.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/AngleOrderDetectorPlus.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">GraphComponentOrderDetector.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/GraphComponentOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">SequenceFunctionOrderDetector.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionOrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">OrderDetector.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/OrderDetector.html#calculateOrder(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">calculateOrder</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SymmetryRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>
<div class="block">Returns a refined symmetry alignment, where the repeat residues are
 aligned consistently in a MultipleAlignment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">GraphComponentRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/GraphComponentRefiner.html#refine(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int)">refine</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment,
      <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
      int&nbsp;order)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></code></td>
<td class="colLast"><span class="strong">SequenceFunctionRefiner.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/SequenceFunctionRefiner.html#refineSymmetry(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[],%20int)">refineSymmetry</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
              <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
              int&nbsp;k)</code>
<div class="block">Refines a CE-Symm alignment so that it is perfectly symmetric.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">CeSymmResult.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/internal/CeSymmResult.html#setSelfAlignment(org.biojava.nbio.structure.align.model.AFPChain)">setSelfAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.structure.symmetry.utils">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a> in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with parameters of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static boolean[][]</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#blankOutBreakFlag(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20boolean[][],%20int)">blankOutBreakFlag</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                 <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                 int&nbsp;rows,
                 int&nbsp;cols,
                 <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                 boolean[][]&nbsp;breakFlag,
                 int&nbsp;blankWindowSize)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#blankOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.jama.Matrix,%20int)">blankOutPreviousAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                         <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                         int&nbsp;rows,
                         int&nbsp;cols,
                         <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                         <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
                         int&nbsp;blankWindowSize)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static org.jgrapht.UndirectedGraph&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>,org.jgrapht.graph.DefaultEdge&gt;</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#buildSymmetryGraph(org.biojava.nbio.structure.align.model.AFPChain)">buildSymmetryGraph</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;selfAlignment)</code>
<div class="block">Converts a self alignment into a directed jGraphT of aligned residues,
 where each vertex is a residue and each edge means the equivalence
 between the two residues in the self-alignment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/align/multiple/MultipleAlignment.html" title="interface in org.biojava.nbio.structure.align.multiple">MultipleAlignment</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#fromAFP(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[])">fromAFP</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;symm,
       <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms)</code>
<div class="block">Converts a refined symmetry AFPChain alignment into the standard
 representation of symmetry in a MultipleAlignment, that contains the
 entire Atom array of the strcuture and the symmetric repeats are orgaized
 in different rows in a single Block.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>static double</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#getAngle(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20org.biojava.nbio.structure.Atom[])">getAngle</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca1,
        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2)</code>
<div class="block">Returns the <em>magnitude</em> of the angle between the first and second
 blocks of <code>afpChain</code>, measured in degrees.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>static <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a></code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#grayOutPreviousAlignment(org.biojava.nbio.structure.align.model.AFPChain,%20org.biojava.nbio.structure.Atom[],%20int,%20int,%20org.biojava.nbio.structure.align.ce.CECalculator,%20org.biojava.nbio.structure.jama.Matrix,%20int,%20double[],%20double)">grayOutPreviousAlignment</a></strong>(<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&nbsp;afpChain,
                        <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;ca2,
                        int&nbsp;rows,
                        int&nbsp;cols,
                        <a href="../../../../../../../org/biojava/nbio/structure/align/ce/CECalculator.html" title="class in org.biojava.nbio.structure.align.ce">CECalculator</a>&nbsp;calculator,
                        <a href="../../../../../../../org/biojava/nbio/structure/jama/Matrix.html" title="class in org.biojava.nbio.structure.jama">Matrix</a>&nbsp;max,
                        int&nbsp;blankWindowSize,
                        double[]&nbsp;gradientPolyCoeff,
                        double&nbsp;gradientExpCoeff)</code>&nbsp;</td>
</tr>
</tbody>
</table>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Method parameters in <a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/package-summary.html">org.biojava.nbio.structure.symmetry.utils</a> with type arguments of type <a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static <a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="http://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&gt;&gt;</code></td>
<td class="colLast"><span class="strong">SymmetryTools.</span><code><strong><a href="../../../../../../../org/biojava/nbio/structure/symmetry/utils/SymmetryTools.html#buildSymmetryGraph(java.util.List,%20org.biojava.nbio.structure.Atom[],%20boolean)">buildSymmetryGraph</a></strong>(<a href="http://docs.oracle.com/javase/7/docs/api/java/util/List.html?is-external=true" title="class or interface in java.util">List</a>&lt;<a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">AFPChain</a>&gt;&nbsp;afps,
                  <a href="../../../../../../../org/biojava/nbio/structure/Atom.html" title="interface in org.biojava.nbio.structure">Atom</a>[]&nbsp;atoms,
                  boolean&nbsp;undirected)</code>
<div class="block">Converts a set of AFP alignments into a Graph of aligned residues, where
 each vertex is a residue and each edge means the connection between the
 two residues in one of the alignments.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../../../org/biojava/nbio/structure/align/model/AFPChain.html" title="class in org.biojava.nbio.structure.align.model">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../../index.html?org/biojava/nbio/structure/align/model/class-use/AFPChain.html" target="_top">Frames</a></li>
<li><a href="AFPChain.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
