<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_79) on Thu Mar 10 15:47:51 PST 2016 -->
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<title>Uses of Package org.biojava.nbio.core.sequence.features (biojava 4.2.0 API)</title>
<meta name="date" content="2016-03-10">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Package org.biojava.nbio.core.sequence.features (biojava 4.2.0 API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/features/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h1 title="Uses of Package org.biojava.nbio.core.sequence.features" class="title">Uses of Package<br>org.biojava.nbio.core.sequence.features</h1>
</div>
<div class="contentContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.features">org.biojava.nbio.core.sequence.features</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.io">org.biojava.nbio.core.sequence.io</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.loader">org.biojava.nbio.core.sequence.loader</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#org.biojava.nbio.core.sequence.template">org.biojava.nbio.core.sequence.template</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#org.biojava.nbio.sequencing.io.fastq">org.biojava.nbio.sequencing.io.fastq</a></td>
<td class="colLast">
<div class="block">FASTQ and variants sequence format I/O.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.features">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.features">AbstractFeature</a>
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.features">DBReferenceInfo</a>
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.features">FeatureInterface</a>
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/Qualifier.html#org.biojava.nbio.core.sequence.features">Qualifier</a>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.io">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../../../../../../org/biojava/nbio/core/sequence/io/package-summary.html">org.biojava.nbio.core.sequence.io</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.io">AbstractFeature</a>
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.io">DBReferenceInfo</a>
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.io">FeatureInterface</a>
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.loader">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../../../../../../org/biojava/nbio/core/sequence/loader/package-summary.html">org.biojava.nbio.core.sequence.loader</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/AbstractFeature.html#org.biojava.nbio.core.sequence.loader">AbstractFeature</a>
<div class="block">A feature is currently any descriptive item that can be associated with a sequence position(s)
 A feature has a type and a source which is currently a string to allow flexibility for the user
 Ideally well defined features should have a class to describe attributes of that feature</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/DatabaseReferenceInterface.html#org.biojava.nbio.core.sequence.loader">DatabaseReferenceInterface</a>
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/DBReferenceInfo.html#org.biojava.nbio.core.sequence.loader">DBReferenceInfo</a>
<div class="block">If you have a uniprot ID then it is possible to get a collection
 of other id(s) that the protein is known by.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/FeatureRetriever.html#org.biojava.nbio.core.sequence.loader">FeatureRetriever</a>
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/FeaturesKeyWordInterface.html#org.biojava.nbio.core.sequence.loader">FeaturesKeyWordInterface</a>
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.core.sequence.template">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../../../../../../org/biojava/nbio/core/sequence/template/package-summary.html">org.biojava.nbio.core.sequence.template</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/DatabaseReferenceInterface.html#org.biojava.nbio.core.sequence.template">DatabaseReferenceInterface</a>
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list of cross reference id(s)</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/FeatureInterface.html#org.biojava.nbio.core.sequence.template">FeatureInterface</a>
<div class="block">Interface class to handle describing arbitrary features.</div>
</td>
</tr>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/FeatureRetriever.html#org.biojava.nbio.core.sequence.template">FeatureRetriever</a>
<div class="block">If a SequenceProxyReader implements this interface then that external source
 has a list features</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/FeaturesKeyWordInterface.html#org.biojava.nbio.core.sequence.template">FeaturesKeyWordInterface</a>
<div class="block">Models the keywords that are annotated for a protein sequence at Uniprot.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="org.biojava.nbio.sequencing.io.fastq">
<!--   -->
</a>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing classes, and an explanation">
<caption><span>Classes in <a href="../../../../../../org/biojava/nbio/core/sequence/features/package-summary.html">org.biojava.nbio.core.sequence.features</a> used by <a href="../../../../../../org/biojava/nbio/sequencing/io/fastq/package-summary.html">org.biojava.nbio.sequencing.io.fastq</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Class and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/QualityFeature.html#org.biojava.nbio.sequencing.io.fastq">QualityFeature</a>
<div class="block">DNA Sequences produced by modern sequencers usually have quality informaion
 attached to them.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><a href="../../../../../../org/biojava/nbio/core/sequence/features/class-use/QuantityFeature.html#org.biojava.nbio.sequencing.io.fastq">QuantityFeature</a>
<div class="block">It is common to have a numerical value or values associated with a feature.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li>Class</li>
<li class="navBarCell1Rev">Use</li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
<div class="aboutLanguage"><em><script src="http://www.google-analytics.com/urchin.js" 							type="text/javascript"></script><script 							type="text/javascript">_uacct = 							"UA-1326640-1";urchinTracker();</script></em></div>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?org/biojava/nbio/core/sequence/features/package-use.html" target="_top">Frames</a></li>
<li><a href="package-use.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2000&#x2013;2016 <a href="http://www.biojava.org">BioJava</a>. All rights reserved.</small></p>
</body>
</html>
